<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>RNA-seq gene expression and pathway analysis</title>
  <meta name="description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="RNA-seq gene expression and pathway analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="RNA-seq gene expression and pathway analysis" />
  
  <meta name="twitter:description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher, Helen Hipperson, Steve Paterson, Bert Overduin, Matthew Gemmell, Xuan Liu" />


<meta name="date" content="2024-06-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />

<link rel="next" href="02-Background.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>RNA-seq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="part"><span><b>Generating gene counts</b></span></li>
<li class="chapter" data-level="4" data-path="04-QC.html"><a href="04-QC.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC.html"><a href="04-QC.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-QC.html"><a href="04-QC.html#quality-assessment"><i class="fa fa-check"></i><b>4.2</b> Quality assessment</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-questions"><i class="fa fa-check"></i><b>4.2.1</b> Quality assessment: questions</a></li>
<li class="chapter" data-level="4.2.2" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-summary"><i class="fa fa-check"></i><b>4.2.2</b> Quality assessment: summary</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="04-QC.html"><a href="04-QC.html#quality-control"><i class="fa fa-check"></i><b>4.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="04-QC.html"><a href="04-QC.html#post-quality-control-check"><i class="fa fa-check"></i><b>4.3.1</b> Post Quality control check</a></li>
<li class="chapter" data-level="4.3.2" data-path="04-QC.html"><a href="04-QC.html#quality-control-questions"><i class="fa fa-check"></i><b>4.3.2</b> Quality control: questions</a></li>
<li class="chapter" data-level="4.3.3" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-summary-1"><i class="fa fa-check"></i><b>4.3.3</b> Quality assessment: summary</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Aligning.html"><a href="05-Aligning.html"><i class="fa fa-check"></i><b>5</b> Aligning Illumina transcriptome data to a reference genome</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Aligning.html"><a href="05-Aligning.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>5.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="5.2" data-path="05-Aligning.html"><a href="05-Aligning.html#read-mapping"><i class="fa fa-check"></i><b>5.2</b> Read mapping</a></li>
<li class="chapter" data-level="5.3" data-path="05-Aligning.html"><a href="05-Aligning.html#assess-mapping-quality"><i class="fa fa-check"></i><b>5.3</b> Assess mapping quality</a></li>
<li class="chapter" data-level="5.4" data-path="05-Aligning.html"><a href="05-Aligning.html#visualising-the-mapping"><i class="fa fa-check"></i><b>5.4</b> Visualising the mapping</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Counting_reads.html"><a href="06-Counting_reads.html"><i class="fa fa-check"></i><b>6</b> Counting Reads using HTSeq-Count</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Counting_reads.html"><a href="06-Counting_reads.html#exercise"><i class="fa fa-check"></i><b>6.1</b> Exercise</a></li>
</ul></li>
<li class="part"><span><b>Jupyter &amp; R</b></span></li>
<li class="chapter" data-level="7" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html"><i class="fa fa-check"></i><b>7</b> Jupyter</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>7.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="7.2" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#create-r-notebook"><i class="fa fa-check"></i><b>7.2</b> Create R notebook</a></li>
<li class="chapter" data-level="7.3" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#cells-and-code"><i class="fa fa-check"></i><b>7.3</b> Cells and code</a></li>
<li class="chapter" data-level="7.4" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#create-new-cells"><i class="fa fa-check"></i><b>7.4</b> Create new cells</a></li>
<li class="chapter" data-level="7.5" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#running-code"><i class="fa fa-check"></i><b>7.5</b> Running code</a></li>
<li class="chapter" data-level="7.6" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#saving-the-file"><i class="fa fa-check"></i><b>7.6</b> Saving the file</a></li>
<li class="chapter" data-level="7.7" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>7.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="7.8" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#close-the-notebook"><i class="fa fa-check"></i><b>7.8</b> Close the notebook</a></li>
<li class="chapter" data-level="7.9" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#video-tutorial"><i class="fa fa-check"></i><b>7.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html"><i class="fa fa-check"></i><b>8</b> Basic Introduction to R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#open-a-new-jupyter-notebook"><i class="fa fa-check"></i><b>8.1</b> Open a new Jupyter notebook</a></li>
<li class="chapter" data-level="8.2" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#your-working-directory"><i class="fa fa-check"></i><b>8.2</b> Your working directory</a></li>
<li class="chapter" data-level="8.3" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#packages"><i class="fa fa-check"></i><b>8.3</b> Packages</a></li>
<li class="chapter" data-level="8.4" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#functions"><i class="fa fa-check"></i><b>8.4</b> Functions</a></li>
<li class="chapter" data-level="8.5" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#assigning-values-to-variables"><i class="fa fa-check"></i><b>8.5</b> Assigning values to variables</a></li>
<li class="chapter" data-level="8.6" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#a-few-useful-commands-and-tips"><i class="fa fa-check"></i><b>8.6</b> A few useful commands and tips</a></li>
</ul></li>
<li class="part"><span><b>Differential Gene Expression</b></span></li>
<li class="chapter" data-level="9" data-path="09-Differential_Gene_Expression.html"><a href="09-Differential_Gene_Expression.html"><i class="fa fa-check"></i><b>9</b> Intro to DGE and DESeq2</a></li>
<li class="chapter" data-level="10" data-path="10-Fly_dataset.html"><a href="10-Fly_dataset.html"><i class="fa fa-check"></i><b>10</b> The fly dataset</a></li>
<li class="chapter" data-level="11" data-path="11-Load_data.html"><a href="11-Load_data.html"><i class="fa fa-check"></i><b>11</b> Load data</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Load_data.html"><a href="11-Load_data.html#dge-notebook"><i class="fa fa-check"></i><b>11.1</b> DGE notebook</a></li>
<li class="chapter" data-level="11.2" data-path="11-Load_data.html"><a href="11-Load_data.html#loading-in-the-data"><i class="fa fa-check"></i><b>11.2</b> Loading in the data</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="11-Load_data.html"><a href="11-Load_data.html#gene-count-data"><i class="fa fa-check"></i><b>11.2.1</b> Gene count data</a></li>
<li class="chapter" data-level="11.2.2" data-path="11-Load_data.html"><a href="11-Load_data.html#metadata"><i class="fa fa-check"></i><b>11.2.2</b> Metadata</a></li>
<li class="chapter" data-level="11.2.3" data-path="11-Load_data.html"><a href="11-Load_data.html#factorise-metadata"><i class="fa fa-check"></i><b>11.2.3</b> Factorise metadata</a></li>
<li class="chapter" data-level="11.2.4" data-path="11-Load_data.html"><a href="11-Load_data.html#compare-count-and-metadata-sample-orders"><i class="fa fa-check"></i><b>11.2.4</b> Compare count and metadata sample orders</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="11-Load_data.html"><a href="11-Load_data.html#deseqdataset"><i class="fa fa-check"></i><b>11.3</b> DESeqDataset</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Filtering_the_data.html"><a href="12-Filtering_the_data.html"><i class="fa fa-check"></i><b>12</b> Filter the data</a></li>
<li class="chapter" data-level="13" data-path="13-DGE_analysis.html"><a href="13-DGE_analysis.html"><i class="fa fa-check"></i><b>13</b> DGE analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-DGE_analysis.html"><a href="13-DGE_analysis.html#dge-run"><i class="fa fa-check"></i><b>13.1</b> DGE: Run</a></li>
<li class="chapter" data-level="13.2" data-path="13-DGE_analysis.html"><a href="13-DGE_analysis.html#dge-extract-results"><i class="fa fa-check"></i><b>13.2</b> DGE: Extract results</a></li>
<li class="chapter" data-level="13.3" data-path="13-DGE_analysis.html"><a href="13-DGE_analysis.html#dge-order-by-significance"><i class="fa fa-check"></i><b>13.3</b> DGE: Order by significance</a></li>
<li class="chapter" data-level="13.4" data-path="13-DGE_analysis.html"><a href="13-DGE_analysis.html#dge-plot-counts-of-significant-features"><i class="fa fa-check"></i><b>13.4</b> DGE: Plot counts of significant features</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="14-DGE_visualisation.html"><a href="14-DGE_visualisation.html"><i class="fa fa-check"></i><b>14</b> Visualising the results</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-DGE_visualisation.html"><a href="14-DGE_visualisation.html#ma-plot"><i class="fa fa-check"></i><b>14.1</b> MA plot</a></li>
<li class="chapter" data-level="14.2" data-path="14-DGE_visualisation.html"><a href="14-DGE_visualisation.html#volcano-plot"><i class="fa fa-check"></i><b>14.2</b> Volcano plot</a></li>
<li class="chapter" data-level="14.3" data-path="14-DGE_visualisation.html"><a href="14-DGE_visualisation.html#heatmap"><i class="fa fa-check"></i><b>14.3</b> Heatmap</a></li>
<li class="chapter" data-level="14.4" data-path="14-DGE_visualisation.html"><a href="14-DGE_visualisation.html#principle-component-analysis"><i class="fa fa-check"></i><b>14.4</b> Principle Component Analysis</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Export_results.html"><a href="15-Export_results.html"><i class="fa fa-check"></i><b>15</b> Export the results</a></li>
<li class="part"><span><b>Gene Ontology</b></span></li>
<li class="chapter" data-level="16" data-path="16-GO.html"><a href="16-GO.html"><i class="fa fa-check"></i><b>16</b> Gene Ontology enrichment analysis</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-GO.html"><a href="16-GO.html#go-setup"><i class="fa fa-check"></i><b>16.1</b> GO: Setup</a></li>
<li class="chapter" data-level="16.2" data-path="16-GO.html"><a href="16-GO.html#go-extract-top-differential-genes"><i class="fa fa-check"></i><b>16.2</b> GO: Extract top differential genes</a></li>
<li class="chapter" data-level="16.3" data-path="16-GO.html"><a href="16-GO.html#go-map-gos-to-genes"><i class="fa fa-check"></i><b>16.3</b> GO: Map GOs to genes</a></li>
<li class="chapter" data-level="16.4" data-path="16-GO.html"><a href="16-GO.html#go-create-topgodata-object"><i class="fa fa-check"></i><b>16.4</b> GO: Create topGOdata object</a></li>
<li class="chapter" data-level="16.5" data-path="16-GO.html"><a href="16-GO.html#go-enrichment-analysis"><i class="fa fa-check"></i><b>16.5</b> GO: enrichment analysis</a></li>
<li class="chapter" data-level="16.6" data-path="16-GO.html"><a href="16-GO.html#go-enrichment-network-plot"><i class="fa fa-check"></i><b>16.6</b> GO: Enrichment network plot</a></li>
</ul></li>
<li class="part"><span><b>Recap &amp; appendix</b></span></li>
<li class="chapter" data-level="17" data-path="17-Recap.html"><a href="17-Recap.html"><i class="fa fa-check"></i><b>17</b> Final recap</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="18-Appendix.html"><a href="18-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="18-Appendix.html"><a href="18-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
<li class="chapter" data-level="A.2" data-path="18-Appendix.html"><a href="18-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>A.2</b> Jupyter-notebook</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RNA-seq gene expression and pathway analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">RNA-seq gene expression and pathway analysis</h1>
<p class="author"><em>Katy Maher, Helen Hipperson, Steve Paterson, Bert Overduin, Matthew Gemmell, Xuan Liu</em></p>
<p class="date"><em>2024-06-10</em></p>
</div>
<center>
<img src="figures/NEOF.png" style="width:400px; border-radius:15px" />
</center>
<div id="introduction" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Introduction<a href="01-Intro.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/rna.png" style="width:300px" />
</center>
<p>RNA-Seq (AKA whole transcriptomic shotgun sequencing) aims to determine the presence and quantity of RNA in a biological sample at a given moment in time. This allows you to determine the level of expression of these genes and therefore compare these expressions across different sample groups, i.e. Differential Gene Expression (DGE).</p>
<p>Sessions will start with a brief presentation followed by self-paced computer practicals guided by an online interactive book. The book will contain theory and practice code. Multiple choice questions will guide the interpretation of results.</p>
<p>At the end of the course learners will be able to:</p>
<ul>
<li>Quality control Illumina RNA-seq data.</li>
<li>Map RNA-seq data to a prepared reference transcriptome.</li>
<li>Visualise the mapping with IGV.</li>
<li>Count the number of reads per gene and exon with htseq.</li>
<li>Carry out differential gene expression analysis (DGE) with DESeq2.</li>
<li>Visualise DGE results via MA plots, volcano plots, heatmaps, and PCA plots.</li>
<li>Carry out Gene Ontology enrichment analysis.</li>
</ul>
<p>There are <a href="https://neof-workshops.github.io/RNAseq_mzar4p/Course/">supplemental</a> materials including creating a htseq count table, further analysis with DESeq2, and using loops to process multiple samples. These can be run through after the course by the learner.</p>
<div id="table-of-contents" class="section level2 unnumbered hasAnchor">
<h2>Table of contents<a href="01-Intro.html#table-of-contents" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table class="table" style="margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:center;">
<a href="02-Background.html#background"><strong>Background</strong></a>
</td>
<td style="text-align:center;">
<a href="03-Cluster_Introduction.html#cluster"><strong>VNC &amp; data</strong></a>
</td>
</tr>
<tr>
<td style="text-align:center;">
<a href="02-Background.html#background"><img src="figures/Drosophila.png" style="height:150px; border-radius:15px" /></a>
</td>
<td style="text-align:center;">
<a href="03-Cluster_Introduction.html#cluster"><img src="figures/cluster.png" height="150" /></a>
</td>
</tr>
<tr>
<td style="text-align:center;">
<a href="04-QC.html#QC"><strong>Quality control</strong></a>
</td>
<td style="text-align:center;">
<a href="05-Aligning.html#map"><strong>Alignment to reference genome</strong></a>
</td>
</tr>
<tr>
<td style="text-align:center;">
<a href="04-QC.html#QC"><img src="figures/mangify_glass.png" height="150" /></a>
</td>
<td style="text-align:center;">
<a href="05-Aligning.html#map"><img src="figures/map.png" height="150" /></a>
</td>
</tr>
<tr>
<td style="text-align:center;">
<a href="06-Counting_reads.html#count"><strong>Counting reads</strong></a>
</td>
<td style="text-align:center;">
<a href="07-Jupyter_tutorial.html#Jupyter"><strong>Jupyter and R</strong></a>
</td>
</tr>
<tr>
<td style="text-align:center;">
<a href="06-Counting_reads.html#count"><img src="figures/count.png" style="height:150px; border-radius:15px" /></a>
</td>
<td style="text-align:center;">
<a href="07-Jupyter_tutorial.html#Jupyter"><img src="figures/jupyter_logo.png" style="height:150px; border-radius:15px; background:white" /></a>
</td>
</tr>
<tr>
<td style="text-align:center;">
<a href="09-Differential_Gene_Expression.html#DGE"><strong>Differential gene expression</strong></a>
</td>
<td style="text-align:center;">
<a href="16-GO.html#GO"><strong>Gene Ontology enrichment analysis</strong></a>
</td>
</tr>
<tr>
<td style="text-align:center;">
<a href="09-Differential_Gene_Expression.html#DGE"><img src="figures/DESeq2.png" height="150" /></a>
</td>
<td style="text-align:center;">
<a href="16-GO.html#GO"><img src="figures/go-logo.large.png" style="height:100px; border-radius:15px; background:white" /></a>
</td>
</tr>
</tbody>
</table>
<p><br />
<a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img alt="Creative Commons Licence" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.</p>

</div>
</div>



<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>

<a href="02-Background.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
