<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Aligning Illumina transcriptome data to a reference genome | RNA-seq gene expression and pathway analysis</title>
  <meta name="description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Aligning Illumina transcriptome data to a reference genome | RNA-seq gene expression and pathway analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Aligning Illumina transcriptome data to a reference genome | RNA-seq gene expression and pathway analysis" />
  
  <meta name="twitter:description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher, Helen Hipperson, Steve Paterson, Bert Overduin, Matthew Gemmell, Xuan Liu" />


<meta name="date" content="2023-09-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="04-QC.html"/>
<link rel="next" href="06-Counting_reads.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>RNA-seq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="part"><span><b>Generating gene counts</b></span></li>
<li class="chapter" data-level="4" data-path="04-QC.html"><a href="04-QC.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC.html"><a href="04-QC.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-QC.html"><a href="04-QC.html#quality-assessment"><i class="fa fa-check"></i><b>4.2</b> Quality assessment</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-questions"><i class="fa fa-check"></i><b>4.2.1</b> Quality assessment: questions</a></li>
<li class="chapter" data-level="4.2.2" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-summary"><i class="fa fa-check"></i><b>4.2.2</b> Quality assessment: summary</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="04-QC.html"><a href="04-QC.html#quality-control"><i class="fa fa-check"></i><b>4.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="04-QC.html"><a href="04-QC.html#post-quality-control-check"><i class="fa fa-check"></i><b>4.3.1</b> Post Quality control check</a></li>
<li class="chapter" data-level="4.3.2" data-path="04-QC.html"><a href="04-QC.html#quality-control-questions"><i class="fa fa-check"></i><b>4.3.2</b> Quality control: questions</a></li>
<li class="chapter" data-level="4.3.3" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-summary-1"><i class="fa fa-check"></i><b>4.3.3</b> Quality assessment: summary</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Aligning.html"><a href="05-Aligning.html"><i class="fa fa-check"></i><b>5</b> Aligning Illumina transcriptome data to a reference genome</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Aligning.html"><a href="05-Aligning.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>5.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="5.2" data-path="05-Aligning.html"><a href="05-Aligning.html#read-mapping"><i class="fa fa-check"></i><b>5.2</b> Read mapping</a></li>
<li class="chapter" data-level="5.3" data-path="05-Aligning.html"><a href="05-Aligning.html#assess-mapping-quality"><i class="fa fa-check"></i><b>5.3</b> Assess mapping quality</a></li>
<li class="chapter" data-level="5.4" data-path="05-Aligning.html"><a href="05-Aligning.html#visualising-the-mapping"><i class="fa fa-check"></i><b>5.4</b> Visualising the mapping</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Counting_reads.html"><a href="06-Counting_reads.html"><i class="fa fa-check"></i><b>6</b> Counting Reads using HTSeq-Count</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Counting_reads.html"><a href="06-Counting_reads.html#exercise"><i class="fa fa-check"></i><b>6.1</b> Exercise</a></li>
</ul></li>
<li class="part"><span><b>Gene expression analysis</b></span></li>
<li class="chapter" data-level="7" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html"><i class="fa fa-check"></i><b>7</b> Differential Gene Expression using DeSeq2</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#jupyter"><i class="fa fa-check"></i><b>7.1</b> Jupyter</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>7.1.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="7.1.2" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#create-r-notebook"><i class="fa fa-check"></i><b>7.1.2</b> Create R notebook</a></li>
<li class="chapter" data-level="7.1.3" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#cells-and-code"><i class="fa fa-check"></i><b>7.1.3</b> Cells and code</a></li>
<li class="chapter" data-level="7.1.4" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#create-new-cells"><i class="fa fa-check"></i><b>7.1.4</b> Create new cells</a></li>
<li class="chapter" data-level="7.1.5" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#running-code"><i class="fa fa-check"></i><b>7.1.5</b> Running code</a></li>
<li class="chapter" data-level="7.1.6" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#saving-the-file"><i class="fa fa-check"></i><b>7.1.6</b> Saving the file</a></li>
<li class="chapter" data-level="7.1.7" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>7.1.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="7.1.8" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#close-the-notebook"><i class="fa fa-check"></i><b>7.1.8</b> Close the notebook</a></li>
<li class="chapter" data-level="7.1.9" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#video-tutorial"><i class="fa fa-check"></i><b>7.1.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#r"><i class="fa fa-check"></i><b>7.2</b> R</a></li>
<li class="chapter" data-level="7.3" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#introduction-to-dge-and-deseq2"><i class="fa fa-check"></i><b>7.3</b> Introduction to DGE and DESeq2</a></li>
<li class="chapter" data-level="7.4" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#the-dataset"><i class="fa fa-check"></i><b>7.4</b> The dataset</a></li>
<li class="chapter" data-level="7.5" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#differential-expression-analysis"><i class="fa fa-check"></i><b>7.5</b> Differential expression analysis</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#loading-in-the-data"><i class="fa fa-check"></i><b>7.5.1</b> Loading in the data</a></li>
<li class="chapter" data-level="7.5.2" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#filtering-the-data"><i class="fa fa-check"></i><b>7.5.2</b> Filtering the data</a></li>
<li class="chapter" data-level="7.5.3" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#running-the-analysis-and-obtaining-the-dge-results"><i class="fa fa-check"></i><b>7.5.3</b> Running the analysis and obtaining the DGE results</a></li>
<li class="chapter" data-level="7.5.4" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#plot-the-expression-differences-for-the-most-significant-differentially-expressed-genes"><i class="fa fa-check"></i><b>7.5.4</b> Plot the expression differences for the most significant differentially expressed genes</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#visualising-the-results"><i class="fa fa-check"></i><b>7.6</b> Visualising the results</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#ma-plot"><i class="fa fa-check"></i><b>7.6.1</b> MA plot</a></li>
<li class="chapter" data-level="7.6.2" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#volcano-plot"><i class="fa fa-check"></i><b>7.6.2</b> Volcano plot</a></li>
<li class="chapter" data-level="7.6.3" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#heatmap"><i class="fa fa-check"></i><b>7.6.3</b> Heatmap</a></li>
<li class="chapter" data-level="7.6.4" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#principle-component-analysis"><i class="fa fa-check"></i><b>7.6.4</b> Principle Component Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#export-the-results"><i class="fa fa-check"></i><b>7.7</b> Export the results</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-GO.html"><a href="08-GO.html"><i class="fa fa-check"></i><b>8</b> Gene Ontology enrichment analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-GO.html"><a href="08-GO.html#final-recap"><i class="fa fa-check"></i><b>8.1</b> Final recap</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RNA-seq gene expression and pathway analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="map" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Aligning Illumina transcriptome data to a reference genome<a href="05-Aligning.html#map" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><a href="http://daehwankimlab.github.io/hisat2/manual/">HISAT2</a> is a fast and memory efficient alignment program for mapping next-generation sequencing reads (whole-genome, transcriptome, and exome sequencing data) against a reference genome. It is also a splice-aware aligner. Splice-aware aligners map reads over exon/intro junctions and are appropriate for aligning reads to a genome reference. Splice-unaware aligners are not aware of exon/intron junctions and are therefore only appropriate for mapping RNA-sequencing data to a transcriptome.</p>
<div id="reference-sequence-preparation" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Reference sequence preparation<a href="05-Aligning.html#reference-sequence-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/index.png" width="40%" style="display: block; margin: auto;" /></p>
<p><code>HISAT2</code> uses indexing of the reference genome to speed up the mapping. The index needs to be prepared before we map our reads. To index the genome run</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="05-Aligning.html#cb16-1" tabindex="-1"></a><span class="ex">hisat2-build</span> <span class="at">-f</span> slice.fa slice</span>
<span id="cb16-2"><a href="05-Aligning.html#cb16-2" tabindex="-1"></a><span class="co"># view the index files</span></span>
<span id="cb16-3"><a href="05-Aligning.html#cb16-3" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>This produces index files with the extension '.ht2' that will be automatically detected and used in the mapping step below.</p>
</div>
<div id="read-mapping" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Read mapping<a href="05-Aligning.html#read-mapping" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/map.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Now we are ready to map our reads. First make a directory for our output.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="05-Aligning.html#cb17-1" tabindex="-1"></a><span class="fu">mkdir</span> aligned</span></code></pre></div>
<p>To map our paired-end, trimmed reads type the following:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="05-Aligning.html#cb18-1" tabindex="-1"></a><span class="ex">hisat2</span> <span class="dt">\</span></span>
<span id="cb18-2"><a href="05-Aligning.html#cb18-2" tabindex="-1"></a>  <span class="at">-x</span> slice <span class="dt">\</span></span>
<span id="cb18-3"><a href="05-Aligning.html#cb18-3" tabindex="-1"></a>  <span class="at">-1</span> trimmed/17_slice_R1_out_paired.fastq <span class="dt">\</span></span>
<span id="cb18-4"><a href="05-Aligning.html#cb18-4" tabindex="-1"></a>  <span class="at">-2</span> trimmed/17_slice_R2_out_paired.fastq <span class="dt">\</span></span>
<span id="cb18-5"><a href="05-Aligning.html#cb18-5" tabindex="-1"></a>  <span class="at">-p</span> 8 <span class="dt">\</span></span>
<span id="cb18-6"><a href="05-Aligning.html#cb18-6" tabindex="-1"></a>  <span class="at">-q</span> <span class="dt">\</span></span>
<span id="cb18-7"><a href="05-Aligning.html#cb18-7" tabindex="-1"></a>  <span class="at">--met-file</span> aligned/17_slice.stats <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb18-8"><a href="05-Aligning.html#cb18-8" tabindex="-1"></a>  <span class="ex">samtools</span> sort <span class="at">-O</span> BAM <span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb18-9"><a href="05-Aligning.html#cb18-9" tabindex="-1"></a>  aligned/17_slice.sorted.bam</span></code></pre></div>
<p>The options used are:</p>
<p><code>-x</code> The index filename prefix</p>
<p><code>-1</code> Path to forward paired-end reads to use for aligning</p>
<p><code>-2</code> Path to reverse paired-end reads to use for aligning.</p>
<p><code>-p</code> Use this many threads to align reads. The default is 1</p>
<p><code>-q</code> Reads are FASTQ files. FASTQ files usually have extension .fq or .fastq. FASTQ is the default format</p>
<p><code>--met-file path</code> Write hisat2 metrics to file. Having alignment metrics can be useful for debugging certain problems, especially performance issues. See also: --met. Default: metrics disabled.</p>
<p>There are many different mapping parameters you can specify, see <a href="http://daehwankimlab.github.io/hisat2/manual/">here</a>. While it is often sufficient to run HISAT2 with default settings, we recommend reading the manual and considering your own dataset carefully when running any analysis.</p>
<p>HISAT2 outputs a SAM file <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">(Sequence Alignment Map format)</a>. However, here we pipe the output to <a href="http://www.htslib.org/"><code>samtools</code></a>, a program for writing, viewing and manipulating alignment files, to sort and generate a BAM format, a binary, compressed version of SAM format. This sorts the SAM file by position (this is the default) and outputs it as a BAM file to save space.</p>
<p>We can view this file using <code>samtools view</code>, and <code>head</code> to display only the first few lines:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="05-Aligning.html#cb19-1" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-h</span> aligned/17_slice.sorted.bam <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<p>The header section consists of lines starting with '@'. In this case we have a header line showing the SAM/BAM format version (<span class="citation">@HD</span> VN:), information on the reference sequence (<span class="citation">@SQ</span>) contig names (SN:) and lengths (LN:) and program information on how the SAM/BAM file was generated (<span class="citation">@PG</span>).</p>
<p>To look at the information in the alignment section we can leave out the <code>-h</code> option:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="05-Aligning.html#cb20-1" tabindex="-1"></a><span class="ex">samtools</span> view aligned/17_slice.sorted.bam <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<p>Here is an example of one line, representing information from one sequence read.</p>
<p><img src="figures/sam1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>Each column contains information on where and how well the read aligns to the reference. After column 11 there can be several optional tags, often specific to the aligner used and can be useful for downstream analysis.</p>
<p><img src="figures/sam2.png" width="60%" style="display: block; margin: auto auto auto 0;" /></p>
<p>The image above is from a different alignment file to the one you have generated here. On the first line of your alignment file what is the query sequence name? What is the reference contig name?</p>
</div>
<div id="assess-mapping-quality" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Assess mapping quality<a href="05-Aligning.html#assess-mapping-quality" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/mangify_glass.png" width="20%" style="display: block; margin: auto;" /></p>
<p>We can use the Samtools command <code>flagstat</code> to find information on how the reads mapped:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="05-Aligning.html#cb21-1" tabindex="-1"></a><span class="ex">samtools</span> flagstat aligned/17_slice.sorted.bam</span></code></pre></div>
How many reads have mapped?
<div id="radio_ZDOVXPXSIM" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ZDOVXPXSIM" value=""></input> <span>5,734</span></label><label><input type="radio" autocomplete="off" name="radio_ZDOVXPXSIM" value=""></input> <span>824,268</span></label><label><input type="radio" autocomplete="off" name="radio_ZDOVXPXSIM" value="answer"></input> <span>762,382</span></label><label><input type="radio" autocomplete="off" name="radio_ZDOVXPXSIM" value=""></input> <span>700,842</span></label>
</div>
<p>This information is a summary of the 'FLAG' values - the second field in the sam/bam file. These values can be used if you want to extract or exclude a specific set of reads for downstream analysis.
Here we will run a command that will keep only high quality, unique alignments and discard unmapped reads and unmapped mates.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="05-Aligning.html#cb22-1" tabindex="-1"></a><span class="fu">mkdir</span> aligned_clean</span>
<span id="cb22-2"><a href="05-Aligning.html#cb22-2" tabindex="-1"></a></span>
<span id="cb22-3"><a href="05-Aligning.html#cb22-3" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-b</span> <span class="at">-q</span> 40 <span class="at">-f</span> 2 <span class="dt">\</span></span>
<span id="cb22-4"><a href="05-Aligning.html#cb22-4" tabindex="-1"></a>-F 12 aligned/17_slice.sorted.bam <span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb22-5"><a href="05-Aligning.html#cb22-5" tabindex="-1"></a>aligned_clean/17_slice.sorted.clean.bam</span></code></pre></div>
<p>The options used are:</p>
<p><code>-b</code> output in bam format</p>
<p><code>-q</code> have a mapping quality greater or equal to the number specified</p>
<p><code>-f</code> to include the reads with the specified flag</p>
<p><code>-F</code> only include reads without this flag</p>
<p>The Broad Institute has a useful <a href="https://broadinstitute.github.io/picard/explain-flags.html">website to interpret the FLAG values</a>, likewise they are <a href="http://www.htslib.org/doc/samtools-flags.html">listed in the samtools documentation</a>.</p>
<p>Run flagstat on the bam file from which we have excluded unmapped reads.</p>
<ol style="list-style-type: decimal">
<li>How many reads are in the cleaned file in total?
<div id="radio_ISIPKYOJWQ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ISIPKYOJWQ" value="answer"></input> <span>690,248</span></label><label><input type="radio" autocomplete="off" name="radio_ISIPKYOJWQ" value=""></input> <span>824,268</span></label><label><input type="radio" autocomplete="off" name="radio_ISIPKYOJWQ" value=""></input> <span>700,842</span></label><label><input type="radio" autocomplete="off" name="radio_ISIPKYOJWQ" value=""></input> <span>345,124</span></label>
</div></li>
<li>What proportion of the reads in this file are now mapped?
<div id="radio_DUOEDZOVQQ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_DUOEDZOVQQ" value=""></input> <span>0%</span></label><label><input type="radio" autocomplete="off" name="radio_DUOEDZOVQQ" value=""></input> <span>84%</span></label><label><input type="radio" autocomplete="off" name="radio_DUOEDZOVQQ" value=""></input> <span>98%</span></label><label><input type="radio" autocomplete="off" name="radio_DUOEDZOVQQ" value="answer"></input> <span>100%</span></label>
</div></li>
</ol>
</div>
<div id="visualising-the-mapping" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Visualising the mapping<a href="05-Aligning.html#visualising-the-mapping" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/eye_purple.png" width="15%" style="display: block; margin: auto;" /></p>
<p>To get a better feel for the mapping we have just done, we will visualise our mappings in IGV, the <a href="https://www.broadinstitute.org/igv/">Integrative Genomics Viewer</a>.</p>
<p>First, we need to index the BAM file in order for the viewer to rapidly go to any bit of the genome:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="05-Aligning.html#cb23-1" tabindex="-1"></a><span class="ex">samtools</span> index aligned_clean/17_slice.sorted.clean.bam</span></code></pre></div>
<p>To launch IGV first <strong>open a new terminal window</strong>: right click on the main screen, choose <code>Terminal</code> and load the conda environment <code>. usernaseq</code></p>
<p>Then type:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="05-Aligning.html#cb24-1" tabindex="-1"></a><span class="ex">igv</span> <span class="kw">&amp;</span></span></code></pre></div>
<p>The &amp; (ampersand) opens IGV “in the background’, so you can still enter commands in the terminal window if you want.</p>
<p>Once IGV has opened (this can take a few minutes) import the genome, mapped reads and gene annotations:</p>
<p>To import the genome:</p>
<ol style="list-style-type: decimal">
<li>Select ‘Genomes’ -&gt; ‘Load Genome from File…’.</li>
<li>Select slice.fa (which should be in rnaseq/Practical_one).</li>
<li>Click [Open].</li>
</ol>
<p>To import the mapped reads:</p>
<ol start="4" style="list-style-type: decimal">
<li>Select ‘File’ -&gt; ‘Load from File…’.</li>
<li>Select 17_slice.sorted.clean.bam (which should be in rnaseq/Practical_one/aligned_clean).</li>
<li>Click [Open].</li>
</ol>
<p>To import the gene annotations:</p>
<ol start="7" style="list-style-type: decimal">
<li>Select ‘File’ -&gt; ‘Load from File…’.</li>
<li>Select slice.FlyBase.gff (which should be in rnaseq/Practical_one).</li>
<li>Click [Open].</li>
</ol>
<p>And show this track in ‘expanded’ mode:</p>
<ol start="10" style="list-style-type: decimal">
<li>Right-click on the slice.FlyBase.gff track label.</li>
<li>Select ‘Expanded’.</li>
</ol>
<p>Now type Chr2:1,014,346-1,015,966 into the white box with ‘Go’ next to it.</p>
<p>You should now be looking at gene FBgn0072602:
<img src="figures/igv.png" width="100%" style="display: block; margin: auto;" /></p>
<p>Mapped reads are shown as fat grey arrows. Some of these are split across an intron. The blue track at the bottom shows the FlyBase gene annotations.</p>
<p>You can move to different parts of the genome and you may see that some reads map well to the gene models and some don’t. The latter will be due to a combination of 5’ and 3’-UTRs (which are not part of the gene annotations), reads transcribed from non-coding DNA, transposons, or from regions where the gene prediction program used to generate the models has missed a gene. The RNA-seq data here can be used to refine gene models, but that’s beyond the scope of this practical.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="04-QC.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="06-Counting_reads.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
