<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Differential Gene Expression using DeSeq2 | RNA-seq gene expression and pathway analysis</title>
  <meta name="description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Differential Gene Expression using DeSeq2 | RNA-seq gene expression and pathway analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Differential Gene Expression using DeSeq2 | RNA-seq gene expression and pathway analysis" />
  
  <meta name="twitter:description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher, Helen Hipperson, Steve Paterson, Bert Overduin, Matthew Gemmell, Xuan Liu" />


<meta name="date" content="2024-04-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="07-Jupyter_and_R.html"/>
<link rel="next" href="09-GO.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>RNA-seq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="part"><span><b>Generating gene counts</b></span></li>
<li class="chapter" data-level="4" data-path="04-QC.html"><a href="04-QC.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC.html"><a href="04-QC.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-QC.html"><a href="04-QC.html#quality-assessment"><i class="fa fa-check"></i><b>4.2</b> Quality assessment</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-questions"><i class="fa fa-check"></i><b>4.2.1</b> Quality assessment: questions</a></li>
<li class="chapter" data-level="4.2.2" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-summary"><i class="fa fa-check"></i><b>4.2.2</b> Quality assessment: summary</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="04-QC.html"><a href="04-QC.html#quality-control"><i class="fa fa-check"></i><b>4.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="04-QC.html"><a href="04-QC.html#post-quality-control-check"><i class="fa fa-check"></i><b>4.3.1</b> Post Quality control check</a></li>
<li class="chapter" data-level="4.3.2" data-path="04-QC.html"><a href="04-QC.html#quality-control-questions"><i class="fa fa-check"></i><b>4.3.2</b> Quality control: questions</a></li>
<li class="chapter" data-level="4.3.3" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-summary-1"><i class="fa fa-check"></i><b>4.3.3</b> Quality assessment: summary</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Aligning.html"><a href="05-Aligning.html"><i class="fa fa-check"></i><b>5</b> Aligning Illumina transcriptome data to a reference genome</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Aligning.html"><a href="05-Aligning.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>5.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="5.2" data-path="05-Aligning.html"><a href="05-Aligning.html#read-mapping"><i class="fa fa-check"></i><b>5.2</b> Read mapping</a></li>
<li class="chapter" data-level="5.3" data-path="05-Aligning.html"><a href="05-Aligning.html#assess-mapping-quality"><i class="fa fa-check"></i><b>5.3</b> Assess mapping quality</a></li>
<li class="chapter" data-level="5.4" data-path="05-Aligning.html"><a href="05-Aligning.html#visualising-the-mapping"><i class="fa fa-check"></i><b>5.4</b> Visualising the mapping</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Counting_reads.html"><a href="06-Counting_reads.html"><i class="fa fa-check"></i><b>6</b> Counting Reads using HTSeq-Count</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Counting_reads.html"><a href="06-Counting_reads.html#exercise"><i class="fa fa-check"></i><b>6.1</b> Exercise</a></li>
</ul></li>
<li class="part"><span><b>Gene expression analysis</b></span></li>
<li class="chapter" data-level="7" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html"><i class="fa fa-check"></i><b>7</b> Jupyter and R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html#jupyter"><i class="fa fa-check"></i><b>7.1</b> Jupyter</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>7.1.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="7.1.2" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html#create-r-notebook"><i class="fa fa-check"></i><b>7.1.2</b> Create R notebook</a></li>
<li class="chapter" data-level="7.1.3" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html#cells-and-code"><i class="fa fa-check"></i><b>7.1.3</b> Cells and code</a></li>
<li class="chapter" data-level="7.1.4" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html#create-new-cells"><i class="fa fa-check"></i><b>7.1.4</b> Create new cells</a></li>
<li class="chapter" data-level="7.1.5" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html#running-code"><i class="fa fa-check"></i><b>7.1.5</b> Running code</a></li>
<li class="chapter" data-level="7.1.6" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html#saving-the-file"><i class="fa fa-check"></i><b>7.1.6</b> Saving the file</a></li>
<li class="chapter" data-level="7.1.7" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>7.1.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="7.1.8" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html#close-the-notebook"><i class="fa fa-check"></i><b>7.1.8</b> Close the notebook</a></li>
<li class="chapter" data-level="7.1.9" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html#video-tutorial"><i class="fa fa-check"></i><b>7.1.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="07-Jupyter_and_R.html"><a href="07-Jupyter_and_R.html#r"><i class="fa fa-check"></i><b>7.2</b> R</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html"><i class="fa fa-check"></i><b>8</b> Differential Gene Expression using DeSeq2</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#introduction-to-dge-and-deseq2"><i class="fa fa-check"></i><b>8.1</b> Introduction to DGE and DESeq2</a></li>
<li class="chapter" data-level="8.2" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#the-dataset"><i class="fa fa-check"></i><b>8.2</b> The dataset</a></li>
<li class="chapter" data-level="8.3" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#differential-expression-analysis"><i class="fa fa-check"></i><b>8.3</b> Differential expression analysis</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#loading-in-the-data"><i class="fa fa-check"></i><b>8.3.1</b> Loading in the data</a></li>
<li class="chapter" data-level="8.3.2" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#filtering-the-data"><i class="fa fa-check"></i><b>8.3.2</b> Filtering the data</a></li>
<li class="chapter" data-level="8.3.3" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#running-the-analysis-and-obtaining-the-dge-results"><i class="fa fa-check"></i><b>8.3.3</b> Running the analysis and obtaining the DGE results</a></li>
<li class="chapter" data-level="8.3.4" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#plot-the-expression-differences-for-the-most-significant-differentially-expressed-genes"><i class="fa fa-check"></i><b>8.3.4</b> Plot the expression differences for the most significant differentially expressed genes</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#visualising-the-results"><i class="fa fa-check"></i><b>8.4</b> Visualising the results</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#ma-plot"><i class="fa fa-check"></i><b>8.4.1</b> MA plot</a></li>
<li class="chapter" data-level="8.4.2" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#volcano-plot"><i class="fa fa-check"></i><b>8.4.2</b> Volcano plot</a></li>
<li class="chapter" data-level="8.4.3" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#heatmap"><i class="fa fa-check"></i><b>8.4.3</b> Heatmap</a></li>
<li class="chapter" data-level="8.4.4" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#principle-component-analysis"><i class="fa fa-check"></i><b>8.4.4</b> Principle Component Analysis</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="08-Differential_Gene_Expression.html"><a href="08-Differential_Gene_Expression.html#export-the-results"><i class="fa fa-check"></i><b>8.5</b> Export the results</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-GO.html"><a href="09-GO.html"><i class="fa fa-check"></i><b>9</b> Gene Ontology enrichment analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-GO.html"><a href="09-GO.html#final-recap"><i class="fa fa-check"></i><b>9.1</b> Final recap</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RNA-seq gene expression and pathway analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="DGE" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Differential Gene Expression using DeSeq2<a href="08-Differential_Gene_Expression.html#DGE" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="introduction-to-dge-and-deseq2" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Introduction to DGE and DESeq2<a href="08-Differential_Gene_Expression.html#introduction-to-dge-and-deseq2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Differential gene expression (DGE) is an analysis used to find genes that are differentially expressed in response to a treatment or variable and it estimates the size and direction of the effect. Most RNA-seq experiments involve comparing two (or more) treatments. The method we will be using uses generalised linear models to explicitly model the count data.</p>
<center>
<img src="figures/DESeq2.png" style="width:200px" />
</center>
<p>We will be using the R package <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8"><code>DESeq2</code></a> to run the differential expression analysis. <code>DESeq2</code> performs differential expression analysis using a table of count data for each gene per sample. It tests for differential expression based on a model using the negative
binomial distribution and uses a shrinkage estimation for dispersions and fold changes to improve stability and interpretability of estimates. For more information see the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">publication</a>, <a href="https://bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf">manual</a> and <a href="http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">vignette</a>.</p>
<p>To summarise the general workflow we will be following we will:</p>
<center>
<img src="figures/workflow_deseq2.png" style="width:600px; border-radius:15px" />
</center>
<ol style="list-style-type: decimal">
<li><p>Read the data into R. This comes in the form of 2 files, a count data file and a metadata file. We then merge these together into a single data structure.</p></li>
<li><p>Filter the data. Many genes have either no or virtually no counts recorded for them. These can cause problems for the statistical analysis downstream so are often removed. (Plus, if they’re not expressed in any treatment they won’t be differentially expressed).</p></li>
<li><p>Normalise the data. In addition to interesting biological reasons why counts of reads from genes will be different, there are plenty of dull technical reasons why they differ that we have to control for. Normalisation controls for the number of reads in total from the library, the fact that genes at a low expression level tend to vary more than highly expressed genes, and that different genes have different amounts of variation. DESeq2 uses a modelling approach to perform normalisation, internally correcting for library size.</p></li>
<li><p>Find differentially expressed genes. This is where we identify genes that differ systematically between treatments.</p></li>
</ol>
</div>
<div id="the-dataset" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> The dataset<a href="08-Differential_Gene_Expression.html#the-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/Dros_lines.png" style="width:200px; border-radius:15px" />
</center>
<p>To do this we will use a pre-prepared data set which contains multiple individuals belonging to different treatment groups needed for DGE analysis. The data comes from Mike Ritchie and Paris Veltsos, University of St Andrews. The full experiment involved 96 samples, but here we just use subsets of these. <em>Drosophila</em> were selected for many generations with either one or many tissue samples taken from males or females, head or thorax and from virgin or mated flies.
In this subset there are three regimes:</p>
<ul>
<li><p>B: Baseline population</p></li>
<li><p>M: Multiple mates. These were flies with elevated polyandrous activity.</p></li>
<li><p>E: Evolved. These flies were forced to have monogamous activity.</p></li>
</ul>
<div class="l-body-outset">
<table>
<thead>
<tr class="header">
<th>Factor</th>
<th>Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Regime (Base, Multiple mates, Evolved)</td>
<td>B, M, E</td>
</tr>
<tr class="even">
<td>Sex (Male or Female)</td>
<td>M, F</td>
</tr>
<tr class="odd">
<td>Status (control or virgin)</td>
<td>C, V</td>
</tr>
<tr class="even">
<td>Tissue (Head or body)</td>
<td>H, B</td>
</tr>
<tr class="odd">
<td>Line (replicate selection lines)</td>
<td>1 - 4</td>
</tr>
</tbody>
</table>
</div>
<p>The subset we will use contains 24 samples. All samples are head tissue (H), from males (M) and females (F), grown under three different regimes (B, M and E).</p>
<p><strong>Important note:</strong> In real experiments you should have biological replicates. The absolute minimum should be 3. It will be difficult to
distinguish any real patterns of expression between conditions from background biological variation
present within conditions without sufficient replicates. Even using 3 replicates, it is likely that you can expect false-positives within your dataset and it will be harder to detect genes with smaller effects.</p>
</div>
<div id="differential-expression-analysis" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Differential expression analysis<a href="08-Differential_Gene_Expression.html#differential-expression-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Create a new notebook called <strong>"Chp07-Differential_gene_expression"</strong>.
- Add a markdown cell with the first level header: # Differential gene expression
- Add the below to a code cell and run the cell to load the <code>DESeq2</code> library.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb36-1"><a href="08-Differential_Gene_Expression.html#cb36-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;DESeq2&quot;</span>)</span></code></pre></div>
<div id="loading-in-the-data" class="section level3 hasAnchor" number="8.3.1">
<h3><span class="header-section-number">8.3.1</span> Loading in the data<a href="08-Differential_Gene_Expression.html#loading-in-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/loading_bar.png" style="width:200px" />
</center>
<p>Create a new code cell. First we will read in a table which contains the count data per gene for all the individuals we will be examining.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb37-1"><a href="08-Differential_Gene_Expression.html#cb37-1" tabindex="-1"></a><span class="co"># The row names are your gene ids</span></span>
<span id="cb37-2"><a href="08-Differential_Gene_Expression.html#cb37-2" tabindex="-1"></a>counts<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;genecount.set1.tsv&quot;</span>, <span class="at">header=</span>T, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb37-3"><a href="08-Differential_Gene_Expression.html#cb37-3" tabindex="-1"></a><span class="co"># Print the start of the dataframe to look at its structure. </span></span>
<span id="cb37-4"><a href="08-Differential_Gene_Expression.html#cb37-4" tabindex="-1"></a><span class="fu">head</span>(counts)</span></code></pre></div>
<p>In a new cell we will read in the metadata for the samples. This table includes information about the regime, sex, status, tissue and line for each sample (for more information see above).</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb38-1"><a href="08-Differential_Gene_Expression.html#cb38-1" tabindex="-1"></a>coldata<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;design.set1.tsv&quot;</span>, <span class="at">header=</span>T, <span class="at">row.names =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>From now on you will get less instructions on your notebook structure. Please create your own coding and markdown cells where you think appropriate.</p>
<p>We are going to run a simple analysis to examine which genes are differentially expressed between males and females (i.e. sex-biased genes). Sex-biased gene expression is common in animals and sex-biased genes are thought to drive many behavioural, physiological and morphological differences between the sexes.</p>
<p>In our model we need to make sure that sex is treated as a factor. To do this we can run the following command.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb39-1"><a href="08-Differential_Gene_Expression.html#cb39-1" tabindex="-1"></a>coldata<span class="sc">$</span>sex <span class="ot">&lt;-</span> <span class="fu">factor</span>(coldata<span class="sc">$</span>sex)</span></code></pre></div>
<p>It is essential that the order of the sample names in your metadata and count tables are in the same order. We can check this by comparing whether all the row names in our metadata match the column names in our count data.</p>
<p>Run the following function. If <code>TRUE</code> is printed to the screen then all the sample names match and we can proceed.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb40-1"><a href="08-Differential_Gene_Expression.html#cb40-1" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(coldata) <span class="sc">==</span> <span class="fu">colnames</span>(counts))</span></code></pre></div>
<p>Now we set up our DESeqDataset, which will store our read counts and metadata. We must also specify an associated design formula, which expresses the variables which will be used in modeling. In this instance we are interested in how sex affects gene expression patterns so we specify this as a term of interest after a tilde (~) in our formula design.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb41-1"><a href="08-Differential_Gene_Expression.html#cb41-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb41-2"><a href="08-Differential_Gene_Expression.html#cb41-2" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb41-3"><a href="08-Differential_Gene_Expression.html#cb41-3" tabindex="-1"></a>                              <span class="at">design=</span> <span class="sc">~</span> sex)</span></code></pre></div>
</div>
<div id="filtering-the-data" class="section level3 hasAnchor" number="8.3.2">
<h3><span class="header-section-number">8.3.2</span> Filtering the data<a href="08-Differential_Gene_Expression.html#filtering-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/water_filter.png" style="width:200px" />
</center>
<p>We will now filter the data to remove genes that have extremely low expression in all samples. You do not have to do this step but there are a couple of reasons why this can be useful:</p>
<ol style="list-style-type: decimal">
<li>removing genes (rows) which have very few reads we can reduce the memory needed to process the dds data object</li>
<li>this will speed up the analysis.</li>
</ol>
<p>In this case we will remove low count genes with less than 10 reads in total across samples. (A stricter filtering is applied to increase power via an independent filtering on the mean of normalised counts when we run the results function further down).</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb42-1"><a href="08-Differential_Gene_Expression.html#cb42-1" tabindex="-1"></a><span class="co"># check the number of genes contained in the dataframe</span></span>
<span id="cb42-2"><a href="08-Differential_Gene_Expression.html#cb42-2" tabindex="-1"></a><span class="fu">nrow</span>(dds)</span>
<span id="cb42-3"><a href="08-Differential_Gene_Expression.html#cb42-3" tabindex="-1"></a><span class="co"># we start with 16,707 genes</span></span>
<span id="cb42-4"><a href="08-Differential_Gene_Expression.html#cb42-4" tabindex="-1"></a></span>
<span id="cb42-5"><a href="08-Differential_Gene_Expression.html#cb42-5" tabindex="-1"></a><span class="co"># remove genes with less than or equal to 10 reads across all samples</span></span>
<span id="cb42-6"><a href="08-Differential_Gene_Expression.html#cb42-6" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">&gt;=</span> <span class="dv">10</span></span>
<span id="cb42-7"><a href="08-Differential_Gene_Expression.html#cb42-7" tabindex="-1"></a>dds<span class="ot">&lt;-</span> dds[keep,]</span>
<span id="cb42-8"><a href="08-Differential_Gene_Expression.html#cb42-8" tabindex="-1"></a></span>
<span id="cb42-9"><a href="08-Differential_Gene_Expression.html#cb42-9" tabindex="-1"></a><span class="co"># re-count the number of genes we have retained after filtering</span></span>
<span id="cb42-10"><a href="08-Differential_Gene_Expression.html#cb42-10" tabindex="-1"></a><span class="fu">nrow</span>(dds)</span>
<span id="cb42-11"><a href="08-Differential_Gene_Expression.html#cb42-11" tabindex="-1"></a><span class="co"># we have 15,675 genes after filtering</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>How many genes do we have after filtering?
<div id="radio_UGPVSCJIRO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_UGPVSCJIRO" value=""></input> <span>67,569</span></label><label><input type="radio" autocomplete="off" name="radio_UGPVSCJIRO" value=""></input> <span>12,675</span></label><label><input type="radio" autocomplete="off" name="radio_UGPVSCJIRO" value="answer"></input> <span>15,675</span></label>
</div></li>
</ol>
<p>Note: The reference level of for factors used within the model are by default based on alphabetical order. It might not matter which factor is set as the reference in your analysis but if for example you had a control/wild type group it would make sense to set this as the reference to compare against. One way you can tell R how to set your reference is using the relevel function. For example if we had a term called condition and we wanted to make sure our wild type (wt) samples were set as the reference compared to the mutant (mt) we could do this using the following command:
<code>dds$condition &lt;- relevel(dds$condition, ref = "wt")</code>. This would mean that positive fold changes in gene expression in our results would mean that expression was higher in the mutant. You would need to do this before running the DESeq function below for the re-leveling to be reflected in your results.</p>
</div>
<div id="running-the-analysis-and-obtaining-the-dge-results" class="section level3 hasAnchor" number="8.3.3">
<h3><span class="header-section-number">8.3.3</span> Running the analysis and obtaining the DGE results<a href="08-Differential_Gene_Expression.html#running-the-analysis-and-obtaining-the-dge-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/play_blue.png" style="width:200px" />
</center>
<p>Now we will run the DESeq function on our filtered DESeqDataSet. This function performs a differential expression analysis based on the negative binomial distribution. It will:</p>
<ol style="list-style-type: decimal">
<li>estimate size factors</li>
<li>estimate dispertions</li>
<li>fit a negative binomial general linear model and wald statistics</li>
</ol>
<div class="sourceCode" id="cb43"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb43-1"><a href="08-Differential_Gene_Expression.html#cb43-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<p>For more details of each step you can check the help section of the DESeq2 help using:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb44-1"><a href="08-Differential_Gene_Expression.html#cb44-1" tabindex="-1"></a>?DESeq</span></code></pre></div>
<p>You can also read more about the estimation steps <a href="http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#theory">here</a> or in the DESeq2 <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">publication</a>.</p>
<p>We will now extract the results table for our DGE analysis using the <code>results</code> function.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb45-1"><a href="08-Differential_Gene_Expression.html#cb45-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span></code></pre></div>
<p>We can look at the start of our results object by running.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb46-1"><a href="08-Differential_Gene_Expression.html#cb46-1" tabindex="-1"></a><span class="fu">head</span>(res)</span></code></pre></div>
<p>You can see we have several columns in our results:</p>
<ul>
<li>baseMean: the average of the normalized count values, dividing by size factors, taken over all samples</li>
<li>log2FoldChange: the effect size estimate. It tells us how much the gene’s expression seems to have changed due to the "treatment". This value is reported on a logarithmic scale to base 2. (In this case a value of 1 means that the expression level for a gene in males is twice that in females).</li>
<li>lfcSE: The log2 fold change estimate standard error (the uncertainty associated with the estimate)</li>
<li>stat: Wald statistic</li>
<li>pvalue: Wald test p-value (The p-value will be reported as NA if all counts for a particular gene are zero, or if the gene was excluded from analysis because it contained an extreme count outlier).</li>
<li>padj: Benjamini-Hochberg adjusted p-value (This is a p-value adjusted to account for multiple testing. We need to apply a correction as we have run thousands of tests which means that the chance of getting a false significant result is much higher. For more info see this <a href="http://www.nature.com/nbt/journal/v27/n12/full/nbt1209-1135.html">link</a>.</li>
</ul>
<p>We can summarise the results by running the following command.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb47-1"><a href="08-Differential_Gene_Expression.html#cb47-1" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<p>This shows us that we have 646 genes with a positive log fold change (higher expression in males and lower expression in females)</p>
How many genes have a negative log fold change (higher expression in females and lower expression in males)?
<div id="radio_IEBDBZITGI" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_IEBDBZITGI" value=""></input> <span>646</span></label><label><input type="radio" autocomplete="off" name="radio_IEBDBZITGI" value="answer"></input> <span>690</span></label><label><input type="radio" autocomplete="off" name="radio_IEBDBZITGI" value=""></input> <span>3,661</span></label>
</div>
<p>We can order our results by the adjusted p-value to look at the most significant DEG.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb48-1"><a href="08-Differential_Gene_Expression.html#cb48-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb48-2"><a href="08-Differential_Gene_Expression.html#cb48-2" tabindex="-1"></a><span class="fu">head</span>(res)</span></code></pre></div>
Which gene has the largest log fold change difference of the top 6 genes shown?
<div id="radio_LOFJNUKHDR" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_LOFJNUKHDR" value=""></input> <span>FBgn0079420</span></label><label><input type="radio" autocomplete="off" name="radio_LOFJNUKHDR" value=""></input> <span>FBgn0245773</span></label><label><input type="radio" autocomplete="off" name="radio_LOFJNUKHDR" value="answer"></input> <span>FBgn0075575</span></label>
</div>
</div>
<div id="plot-the-expression-differences-for-the-most-significant-differentially-expressed-genes" class="section level3 hasAnchor" number="8.3.4">
<h3><span class="header-section-number">8.3.4</span> Plot the expression differences for the most significant differentially expressed genes<a href="08-Differential_Gene_Expression.html#plot-the-expression-differences-for-the-most-significant-differentially-expressed-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can visually plot these genes by typing</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb49-1"><a href="08-Differential_Gene_Expression.html#cb49-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb49-2"><a href="08-Differential_Gene_Expression.html#cb49-2" tabindex="-1"></a></span>
<span id="cb49-3"><a href="08-Differential_Gene_Expression.html#cb49-3" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">&quot;FBgn0245773&quot;</span>, <span class="at">intgroup=</span><span class="st">&quot;sex&quot;</span>)</span>
<span id="cb49-4"><a href="08-Differential_Gene_Expression.html#cb49-4" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">&quot;FBgn0075575&quot;</span>, <span class="at">intgroup=</span><span class="st">&quot;sex&quot;</span>)</span>
<span id="cb49-5"><a href="08-Differential_Gene_Expression.html#cb49-5" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">&quot;FBgn0071323&quot;</span>, <span class="at">intgroup=</span><span class="st">&quot;sex&quot;</span>)</span>
<span id="cb49-6"><a href="08-Differential_Gene_Expression.html#cb49-6" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">&quot;FBgn0072657&quot;</span>, <span class="at">intgroup=</span><span class="st">&quot;sex&quot;</span>)</span>
<span id="cb49-7"><a href="08-Differential_Gene_Expression.html#cb49-7" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">&quot;FBgn0249407&quot;</span>, <span class="at">intgroup=</span><span class="st">&quot;sex&quot;</span>)</span>
<span id="cb49-8"><a href="08-Differential_Gene_Expression.html#cb49-8" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">&quot;FBgn0079420&quot;</span>, <span class="at">intgroup=</span><span class="st">&quot;sex&quot;</span>)</span></code></pre></div>
<center>
<img src="figures/6_genes.png" style="width:700px; border-radius:15px" />
</center>
<p>You can see that of the 6 genes with the lowest adjusted p-values, 5 have negative log fold changes and have higher normalised counts in females compared to males, and 1 has a positive log fold change with higher normalised counts in males than females.</p>
</div>
</div>
<div id="visualising-the-results" class="section level2 hasAnchor" number="8.4">
<h2><span class="header-section-number">8.4</span> Visualising the results<a href="08-Differential_Gene_Expression.html#visualising-the-results" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/eye_brown.png" style="width:200px" />
</center>
<p>We will now go through several different methods of visualising the results from our DGE analysis.</p>
<div id="ma-plot" class="section level3 hasAnchor" number="8.4.1">
<h3><span class="header-section-number">8.4.1</span> MA plot<a href="08-Differential_Gene_Expression.html#ma-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We will first plot an MA plot. This gives us an overview of the experiment and how DESeq2 calls significant DEG. Each gene is represented by a dot. Genes with significant adjusted p-values (default &lt; 0.01) are coloured in blue. This shows that DESeq2 will only call significant DGE in genes with a large average normalised count as only these genes contain sufficient information to be able to do so.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb50-1"><a href="08-Differential_Gene_Expression.html#cb50-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb50-2"><a href="08-Differential_Gene_Expression.html#cb50-2" tabindex="-1"></a></span>
<span id="cb50-3"><a href="08-Differential_Gene_Expression.html#cb50-3" tabindex="-1"></a><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>))</span></code></pre></div>
<center>
<img src="figures/MA.png" style="width:700px; border-radius:15px" />
</center>
</div>
<div id="volcano-plot" class="section level3 hasAnchor" number="8.4.2">
<h3><span class="header-section-number">8.4.2</span> Volcano plot<a href="08-Differential_Gene_Expression.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>One nice way to visualise your results is to plot a volcano plot.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb51-1"><a href="08-Differential_Gene_Expression.html#cb51-1" tabindex="-1"></a><span class="co"># Make the basic volcano plot</span></span>
<span id="cb51-2"><a href="08-Differential_Gene_Expression.html#cb51-2" tabindex="-1"></a><span class="fu">with</span>(res, <span class="fu">plot</span>(log2FoldChange, </span>
<span id="cb51-3"><a href="08-Differential_Gene_Expression.html#cb51-3" tabindex="-1"></a>               <span class="sc">-</span><span class="fu">log10</span>(pvalue), </span>
<span id="cb51-4"><a href="08-Differential_Gene_Expression.html#cb51-4" tabindex="-1"></a>               <span class="at">pch=</span><span class="dv">20</span>, <span class="at">main=</span><span class="st">&quot;Volcano plot&quot;</span>, </span>
<span id="cb51-5"><a href="08-Differential_Gene_Expression.html#cb51-5" tabindex="-1"></a>               <span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>,<span class="dv">10</span>),  <span class="at">col=</span><span class="st">&quot;grey&quot;</span>))</span>
<span id="cb51-6"><a href="08-Differential_Gene_Expression.html#cb51-6" tabindex="-1"></a></span>
<span id="cb51-7"><a href="08-Differential_Gene_Expression.html#cb51-7" tabindex="-1"></a><span class="co"># Colour points light blue if padj&lt;0.01</span></span>
<span id="cb51-8"><a href="08-Differential_Gene_Expression.html#cb51-8" tabindex="-1"></a><span class="fu">with</span>(<span class="fu">subset</span>(res, padj<span class="sc">&lt;</span><span class="fl">0.01</span> ), </span>
<span id="cb51-9"><a href="08-Differential_Gene_Expression.html#cb51-9" tabindex="-1"></a>     <span class="fu">points</span>(log2FoldChange, </span>
<span id="cb51-10"><a href="08-Differential_Gene_Expression.html#cb51-10" tabindex="-1"></a>            <span class="sc">-</span><span class="fu">log10</span>(pvalue), </span>
<span id="cb51-11"><a href="08-Differential_Gene_Expression.html#cb51-11" tabindex="-1"></a>            <span class="at">pch=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;#a6cee3&quot;</span>))</span>
<span id="cb51-12"><a href="08-Differential_Gene_Expression.html#cb51-12" tabindex="-1"></a><span class="co"># Colour points dark blue if log2FC&gt;1 and padj&lt;0.01)</span></span>
<span id="cb51-13"><a href="08-Differential_Gene_Expression.html#cb51-13" tabindex="-1"></a><span class="fu">with</span>(<span class="fu">subset</span>(res, padj<span class="sc">&lt;</span><span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange)<span class="sc">&gt;</span><span class="dv">1</span>), </span>
<span id="cb51-14"><a href="08-Differential_Gene_Expression.html#cb51-14" tabindex="-1"></a>     <span class="fu">points</span>(log2FoldChange, </span>
<span id="cb51-15"><a href="08-Differential_Gene_Expression.html#cb51-15" tabindex="-1"></a>            <span class="sc">-</span><span class="fu">log10</span>(pvalue), </span>
<span id="cb51-16"><a href="08-Differential_Gene_Expression.html#cb51-16" tabindex="-1"></a>            <span class="at">pch=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;#1f78b4&quot;</span>))</span></code></pre></div>
<center>
<img src="figures/volcano.png" style="width:700px; border-radius:15px" />
</center>
<p>This allows us to visualise the number of differentially expressed genes which are significant (light blue). We can also visualise how many genes have an expression level which is at least twice as high in males than in females (log2FoldChange&gt;1) and those which are twice as high in females than in males (log2FoldChange&lt;1).</p>
<p>In our case, we have a fairly even number of genes both up and down regulated in males compared to females.</p>
</div>
<div id="heatmap" class="section level3 hasAnchor" number="8.4.3">
<h3><span class="header-section-number">8.4.3</span> Heatmap<a href="08-Differential_Gene_Expression.html#heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Usually, the variation between samples within a sample group (within-group variation) is smaller than that between samples from different sample groups (overall variation) because the former consists of technical and biological variation only, while the latter also contains variation due to the effects of factors. When the factor effect is the dominant contributor of variation, the sample groups can be clearly separated. However, if the factor effect is weak compared to the technical and biological variation within sample groups, it will be difficult to discriminate groups from the data.</p>
<p>PCA and Correlation Heatmap plots visualise the outcomes of the assessment of the variation within and between sample groups.</p>
<p>Before we plot these we will transform the data. For the differential expression analysis we use raw counts but for visualisation and clustering purposes it is common to work with transformed data. There are two main ways suggested for transforming data for visualisation and clustering in DESeq2: VST (variance stabilising transformations) and the rlog (regularised logarithm). For more information see <a href="http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#data-transformations-and-visualization">here</a>. Here we will use the rlog transformation before plotting the Heatmap and PCA plots.</p>
<p>First use the rlog transformation to normalise the data.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb52-1"><a href="08-Differential_Gene_Expression.html#cb52-1" tabindex="-1"></a>rld <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds, <span class="at">blind=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p>The argument blind, refers to whether the transformation should be blind to the sample information specified by the design formula. This is not appopriate to use as we are expecting many of the genes to have counts which are explained by the experimental design we specified (i.e. sex), so we have set this argument to false.</p>
<p>We will use the package <code>pheatmap</code> to plot a heatmap for our samples and see how they cluster.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb53-1"><a href="08-Differential_Gene_Expression.html#cb53-1" tabindex="-1"></a><span class="co"># load pheatmap &amp; ggplot2</span></span>
<span id="cb53-2"><a href="08-Differential_Gene_Expression.html#cb53-2" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb53-3"><a href="08-Differential_Gene_Expression.html#cb53-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb53-4"><a href="08-Differential_Gene_Expression.html#cb53-4" tabindex="-1"></a></span>
<span id="cb53-5"><a href="08-Differential_Gene_Expression.html#cb53-5" tabindex="-1"></a><span class="co"># extract assay of numeric values from the rld object needed by pheatmap</span></span>
<span id="cb53-6"><a href="08-Differential_Gene_Expression.html#cb53-6" tabindex="-1"></a>rld_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(rld)</span>
<span id="cb53-7"><a href="08-Differential_Gene_Expression.html#cb53-7" tabindex="-1"></a><span class="co"># compute the pairwise correlation values for samples and make a matrix of correlations</span></span>
<span id="cb53-8"><a href="08-Differential_Gene_Expression.html#cb53-8" tabindex="-1"></a>rld_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(rld_mat)</span>
<span id="cb53-9"><a href="08-Differential_Gene_Expression.html#cb53-9" tabindex="-1"></a><span class="co"># rename the coldata by the sex column in the metadata</span></span>
<span id="cb53-10"><a href="08-Differential_Gene_Expression.html#cb53-10" tabindex="-1"></a><span class="fu">rownames</span>(rld_cor)<span class="ot">&lt;-</span>coldata<span class="sc">$</span>sex</span>
<span id="cb53-11"><a href="08-Differential_Gene_Expression.html#cb53-11" tabindex="-1"></a><span class="fu">head</span>(rld_cor)</span></code></pre></div>
<div class="sourceCode" id="cb54"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb54-1"><a href="08-Differential_Gene_Expression.html#cb54-1" tabindex="-1"></a><span class="co"># plot heatmap</span></span>
<span id="cb54-2"><a href="08-Differential_Gene_Expression.html#cb54-2" tabindex="-1"></a><span class="fu">pheatmap</span>(rld_cor)</span></code></pre></div>
<center>
<img src="figures/heatmap.png" style="width:700px; border-radius:15px" />
</center>
<p>We can see that in general samples seem to cluster on the heatmap by sex.</p>
</div>
<div id="principle-component-analysis" class="section level3 hasAnchor" number="8.4.4">
<h3><span class="header-section-number">8.4.4</span> Principle Component Analysis<a href="08-Differential_Gene_Expression.html#principle-component-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We will now plot the PCA and colour the points based on sex. We specify that females will be coloured green and males orange.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb55-1"><a href="08-Differential_Gene_Expression.html#cb55-1" tabindex="-1"></a><span class="fu">plotPCA</span>(rld, <span class="at">intgroup=</span><span class="st">&quot;sex&quot;</span>)<span class="sc">+</span></span>
<span id="cb55-2"><a href="08-Differential_Gene_Expression.html#cb55-2" tabindex="-1"></a>            <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb55-3"><a href="08-Differential_Gene_Expression.html#cb55-3" tabindex="-1"></a>            <span class="fu">theme</span>(<span class="at">legend.title=</span><span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb55-4"><a href="08-Differential_Gene_Expression.html#cb55-4" tabindex="-1"></a>            <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#33a02c&quot;</span>, <span class="st">&quot;#ff7f00&quot;</span>))</span></code></pre></div>
<center>
<img src="figures/PCA_sex.png" style="width:700px; border-radius:15px" />
</center>
<p>The PCA shows us that 32% of the variance can be explained by PC1 and looking at how the points are clustered we can suggest that sex is driving the division based on the x axis.</p>
<p>24% assigned to PC2. We can recolour the plot based on the group variable from our metadata to see if this sheds some more light on what could be driving the differences between samples on the y axis.</p>
<p>Plot again and colour based on group.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb56-1"><a href="08-Differential_Gene_Expression.html#cb56-1" tabindex="-1"></a><span class="fu">plotPCA</span>(rld, <span class="at">intgroup=</span><span class="st">&quot;group&quot;</span>)<span class="sc">+</span></span>
<span id="cb56-2"><a href="08-Differential_Gene_Expression.html#cb56-2" tabindex="-1"></a>                <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb56-3"><a href="08-Differential_Gene_Expression.html#cb56-3" tabindex="-1"></a>                <span class="fu">theme</span>(<span class="at">legend.title=</span><span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb56-4"><a href="08-Differential_Gene_Expression.html#cb56-4" tabindex="-1"></a>                <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#a6cee3&quot;</span>, <span class="st">&quot;#1f78b4&quot;</span>, <span class="st">&quot;#fb9a99&quot;</span>, </span>
<span id="cb56-5"><a href="08-Differential_Gene_Expression.html#cb56-5" tabindex="-1"></a>                                              <span class="st">&quot;#e31a1c&quot;</span>, <span class="st">&quot;#b2df8a&quot;</span>, <span class="st">&quot;#33a02c&quot;</span>))</span></code></pre></div>
<center>
<img src="figures/PCA_group.png" style="width:700px; border-radius:15px" />
</center>
<p>All samples in this dataset are from control samples and head tissue. We specify different colours based on the regime (<strong>B</strong>ase, <strong>M</strong>ultiple mates, <strong>E</strong>volved). Darker points are samples from males and lighter colours are from females.</p>
<p>We can see that the diversity on the y axis appears to be driven by the regime, so it would be interesting to repeat our analysis adding regime to the formula. That regime seems to have a some affect on gene expression patterns makes sense as mating system variation has been shown to influence sex-biased gene expression in some animal species.</p>
<p>For more information on how you would run more complex models please see Chapter 4 of the supplemental bookdown.</p>
</div>
</div>
<div id="export-the-results" class="section level2 hasAnchor" number="8.5">
<h2><span class="header-section-number">8.5</span> Export the results<a href="08-Differential_Gene_Expression.html#export-the-results" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/export.png" style="width:200px" />
</center>
<p>Export the results as an R data file.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb57-1"><a href="08-Differential_Gene_Expression.html#cb57-1" tabindex="-1"></a><span class="fu">save</span>(res, <span class="at">file=</span> <span class="st">&quot;DESeq2_DGE_results_sex.RData&quot;</span>)</span></code></pre></div>
<p>To save the results to a <code>csv</code> file you can type the following.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb58-1"><a href="08-Differential_Gene_Expression.html#cb58-1" tabindex="-1"></a><span class="fu">write.csv</span>(res, <span class="st">&quot;DESeq2_DGE_results_sex.csv&quot;</span>)</span></code></pre></div>
<p>You can now close and halt your notebook.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="07-Jupyter_and_R.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="09-GO.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
