<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Gene Ontology enrichment analysis | RNA-seq gene expression and pathway analysis</title>
  <meta name="description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Gene Ontology enrichment analysis | RNA-seq gene expression and pathway analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Gene Ontology enrichment analysis | RNA-seq gene expression and pathway analysis" />
  
  <meta name="twitter:description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher, Helen Hipperson, Steve Paterson, Bert Overduin, Matthew Gemmell, Xuan Liu" />


<meta name="date" content="2023-05-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="07-Differential_Gene_Expression.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>RNA-seq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="part"><span><b>Generating gene counts</b></span></li>
<li class="chapter" data-level="4" data-path="04-QC.html"><a href="04-QC.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC.html"><a href="04-QC.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-QC.html"><a href="04-QC.html#quality-assessment"><i class="fa fa-check"></i><b>4.2</b> Quality assessment</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-questions"><i class="fa fa-check"></i><b>4.2.1</b> Quality assessment: questions</a></li>
<li class="chapter" data-level="4.2.2" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-summary"><i class="fa fa-check"></i><b>4.2.2</b> Quality assessment: summary</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="04-QC.html"><a href="04-QC.html#quality-control"><i class="fa fa-check"></i><b>4.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="04-QC.html"><a href="04-QC.html#post-quality-control-check"><i class="fa fa-check"></i><b>4.3.1</b> Post Quality control check</a></li>
<li class="chapter" data-level="4.3.2" data-path="04-QC.html"><a href="04-QC.html#quality-control-questions"><i class="fa fa-check"></i><b>4.3.2</b> Quality control: questions</a></li>
<li class="chapter" data-level="4.3.3" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-summary-1"><i class="fa fa-check"></i><b>4.3.3</b> Quality assessment: summary</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Aligning.html"><a href="05-Aligning.html"><i class="fa fa-check"></i><b>5</b> Aligning Illumina transcriptome data to a reference genome</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Aligning.html"><a href="05-Aligning.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>5.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="5.2" data-path="05-Aligning.html"><a href="05-Aligning.html#read-mapping"><i class="fa fa-check"></i><b>5.2</b> Read mapping</a></li>
<li class="chapter" data-level="5.3" data-path="05-Aligning.html"><a href="05-Aligning.html#assess-mapping-quality"><i class="fa fa-check"></i><b>5.3</b> Assess mapping quality</a></li>
<li class="chapter" data-level="5.4" data-path="05-Aligning.html"><a href="05-Aligning.html#visualising-the-mapping"><i class="fa fa-check"></i><b>5.4</b> Visualising the mapping</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Counting_reads.html"><a href="06-Counting_reads.html"><i class="fa fa-check"></i><b>6</b> Counting Reads using HTSeq-Count</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Counting_reads.html"><a href="06-Counting_reads.html#exercise"><i class="fa fa-check"></i><b>6.1</b> Exercise</a></li>
</ul></li>
<li class="part"><span><b>Gene expression analysis</b></span></li>
<li class="chapter" data-level="7" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html"><i class="fa fa-check"></i><b>7</b> Differential Gene Expression using DeSeq2</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#jupyter"><i class="fa fa-check"></i><b>7.1</b> Jupyter</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>7.1.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="7.1.2" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#create-r-notebook"><i class="fa fa-check"></i><b>7.1.2</b> Create R notebook</a></li>
<li class="chapter" data-level="7.1.3" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#cells-and-code"><i class="fa fa-check"></i><b>7.1.3</b> Cells and code</a></li>
<li class="chapter" data-level="7.1.4" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#create-new-cells"><i class="fa fa-check"></i><b>7.1.4</b> Create new cells</a></li>
<li class="chapter" data-level="7.1.5" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#running-code"><i class="fa fa-check"></i><b>7.1.5</b> Running code</a></li>
<li class="chapter" data-level="7.1.6" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#saving-the-file"><i class="fa fa-check"></i><b>7.1.6</b> Saving the file</a></li>
<li class="chapter" data-level="7.1.7" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>7.1.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="7.1.8" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#close-the-notebook"><i class="fa fa-check"></i><b>7.1.8</b> Close the notebook</a></li>
<li class="chapter" data-level="7.1.9" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#video-tutorial"><i class="fa fa-check"></i><b>7.1.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#r"><i class="fa fa-check"></i><b>7.2</b> R</a></li>
<li class="chapter" data-level="7.3" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#introduction-to-dge-and-deseq2"><i class="fa fa-check"></i><b>7.3</b> Introduction to DGE and DESeq2</a></li>
<li class="chapter" data-level="7.4" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#the-dataset"><i class="fa fa-check"></i><b>7.4</b> The dataset</a></li>
<li class="chapter" data-level="7.5" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#differential-expression-analysis"><i class="fa fa-check"></i><b>7.5</b> Differential expression analysis</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#loading-in-the-data"><i class="fa fa-check"></i><b>7.5.1</b> Loading in the data</a></li>
<li class="chapter" data-level="7.5.2" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#filtering-the-data"><i class="fa fa-check"></i><b>7.5.2</b> Filtering the data</a></li>
<li class="chapter" data-level="7.5.3" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#running-the-analysis-and-obtaining-the-dge-results"><i class="fa fa-check"></i><b>7.5.3</b> Running the analysis and obtaining the DGE results</a></li>
<li class="chapter" data-level="7.5.4" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#plot-the-expression-differences-for-the-most-significant-differentially-expressed-genes"><i class="fa fa-check"></i><b>7.5.4</b> Plot the expression differences for the most significant differentially expressed genes</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#visualising-the-results"><i class="fa fa-check"></i><b>7.6</b> Visualising the results</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#ma-plot"><i class="fa fa-check"></i><b>7.6.1</b> MA plot</a></li>
<li class="chapter" data-level="7.6.2" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#volcano-plot"><i class="fa fa-check"></i><b>7.6.2</b> Volcano plot</a></li>
<li class="chapter" data-level="7.6.3" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#heatmap"><i class="fa fa-check"></i><b>7.6.3</b> Heatmap</a></li>
<li class="chapter" data-level="7.6.4" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#principle-component-analysis"><i class="fa fa-check"></i><b>7.6.4</b> Principle Component Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="07-Differential_Gene_Expression.html"><a href="07-Differential_Gene_Expression.html#export-the-results"><i class="fa fa-check"></i><b>7.7</b> Export the results</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-GO.html"><a href="08-GO.html"><i class="fa fa-check"></i><b>8</b> Gene Ontology enrichment analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-GO.html"><a href="08-GO.html#final-recap"><i class="fa fa-check"></i><b>8.1</b> Final recap</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RNA-seq gene expression and pathway analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="GO" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Gene Ontology enrichment analysis<a href="08-GO.html#GO" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Gene Ontology (GO) is a method off classifying gene function in a structured way and is split into three subgroupings: "biological process (e.g., signal transduction), molecular function (e.g., ATPase activity) and cellular component (e.g., ribosome)". Genes are associated with particular GO terms via GO annotations and each gene can have multiple annotations associated with it. You can obtain GO annotations from the Gene Ontology website, from species-specific databases, or they can be annotated to a reference gennome using software such as blast2GO.</p>
<p>GO enrichment analysis finds which GO terms are over-represented (or under-represented) using annotations for a set of genes (in this case genes which are significantly differentially expressed between males and females).</p>
<p>We will perform GO enrichment analysis using the topGO package in R. We strongly recommend reading through the topGO <a href="https://bioconductor.org/packages/release/bioc/vignettes/topGO/inst/doc/topGO.pdf">manual</a> when you come to analyse your own data.</p>
<p>First create a new notebook called <strong>"Chp08-GO_enrichment"</strong>.</p>
<p>Load the topGO package and load an example gene dataset, this also loads a function to define a list of differentially expressed genes that we will use in our analysis. We will also re-load our results from the DGE analysis.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="08-GO.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(topGO)</span>
<span id="cb59-2"><a href="08-GO.html#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Rgraphviz)</span>
<span id="cb59-3"><a href="08-GO.html#cb59-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-4"><a href="08-GO.html#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList)</span>
<span id="cb59-5"><a href="08-GO.html#cb59-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-6"><a href="08-GO.html#cb59-6" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;DESeq2_DGE_results_sex.RData&quot;</span>)</span></code></pre></div>
<p>We will now modify the geneList object to contain the results we obtained from our DGE analysis containing the adjusted p-values and the names of the genes.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="08-GO.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># extract the adjusted p-values from  our results object</span></span>
<span id="cb60-2"><a href="08-GO.html#cb60-2" aria-hidden="true" tabindex="-1"></a>geneList <span class="ot">&lt;-</span> res<span class="sc">$</span>padj</span>
<span id="cb60-3"><a href="08-GO.html#cb60-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-4"><a href="08-GO.html#cb60-4" aria-hidden="true" tabindex="-1"></a><span class="co"># add the gene names from our results to our list of adjusted p-values</span></span>
<span id="cb60-5"><a href="08-GO.html#cb60-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneList) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res)</span>
<span id="cb60-6"><a href="08-GO.html#cb60-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-7"><a href="08-GO.html#cb60-7" aria-hidden="true" tabindex="-1"></a><span class="co"># remove genes with an adjusted p-value recorded as NA</span></span>
<span id="cb60-8"><a href="08-GO.html#cb60-8" aria-hidden="true" tabindex="-1"></a>geneList<span class="ot">&lt;-</span><span class="fu">na.omit</span>(geneList)</span></code></pre></div>
<p>We can use the topDiffGenes function to extract genes with a adjusted p-value of less than 0.01. This will be used as our list of genes of interest for our GO enrichment analysis.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="08-GO.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">topDiffGenes</span>(geneList))</span></code></pre></div>
<p>We have 666 genes which are significantly differentially expressed when using a cutoff of 0.01.</p>
<p>We will now read in a list of genes and their associated biological process GO annotations.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="08-GO.html#cb62-1" aria-hidden="true" tabindex="-1"></a>geneID2GO <span class="ot">&lt;-</span> <span class="fu">readMappings</span>(<span class="st">&quot;geneanno_goBio.tsv&quot;</span>)</span>
<span id="cb62-2"><a href="08-GO.html#cb62-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geneID2GO)</span></code></pre></div>
<p>You can see we now have a list of GO annotations associated with each gene ID.</p>
<p>We now have all the data we need to make a topGOdata object. This will contain all the information needed to perform the enrichment analysis including the gene lists we've made, the GO annotations and the GO hierarchical structure.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="08-GO.html#cb63-1" aria-hidden="true" tabindex="-1"></a>sampleGOdata <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">&quot;topGOdata&quot;</span>,</span>
<span id="cb63-2"><a href="08-GO.html#cb63-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">ontology =</span> <span class="st">&quot;BP&quot;</span>, </span>
<span id="cb63-3"><a href="08-GO.html#cb63-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">allGenes =</span> geneList,</span>
<span id="cb63-4"><a href="08-GO.html#cb63-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">geneSel =</span> topDiffGenes,</span>
<span id="cb63-5"><a href="08-GO.html#cb63-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">nodeSize =</span> <span class="dv">10</span>,</span>
<span id="cb63-6"><a href="08-GO.html#cb63-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">annot =</span> annFUN.gene2GO, </span>
<span id="cb63-7"><a href="08-GO.html#cb63-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">gene2GO =</span> geneID2GO)</span></code></pre></div>
<ul>
<li>ontology = specifies which ontology to use (biological processes, cellular components or molecular function)</li>
<li>allGenes = a list of all genes which we tested in our DGE analysis with their associated adjusted p-values</li>
<li>geneSel = select genes using the topDiffGenes function (genes with a adjusted p-value &lt; 0.01) to use as our set of genes of interest</li>
<li>nodeSize = remove any annotated GO terms which have less than X number of genes associated with them (in this case 10)</li>
<li>annot = annFUN.gene2GO function is used to extract the gene-to-GO mappings from the geneID2GO object</li>
<li>gene2GO = our list of genes and their associated GOs</li>
</ul>
<p>If you check the topGOdata object we can see a summary of the data we added.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="08-GO.html#cb64-1" aria-hidden="true" tabindex="-1"></a>sampleGOdata</span></code></pre></div>
<p>Now we have the topGOdata object ready we can start with the enrichment analysis. Here we run the analysis using the Fisherâ€™s exact test statistic using the runTest function and test whether there is an over-representation of GO terms within the group of differentially expressed genes. Each GO category is tested independently with the algorithm classic.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="08-GO.html#cb65-1" aria-hidden="true" tabindex="-1"></a>resultFisher <span class="ot">&lt;-</span> <span class="fu">runTest</span>(sampleGOdata, <span class="at">algorithm =</span> <span class="st">&quot;classic&quot;</span>, <span class="at">statistic =</span> <span class="st">&quot;fisher&quot;</span>)</span>
<span id="cb65-2"><a href="08-GO.html#cb65-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-3"><a href="08-GO.html#cb65-3" aria-hidden="true" tabindex="-1"></a>resultFisher</span></code></pre></div>
<p>The enrichment analysis has identified 63 significantly enriched GO categories (p &lt; 0.01).</p>
<p>We can then pull out the most significant GO terms from the results and the corresponding p-values using the function GenTable. In this case we will print the top 10 GO categories.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="08-GO.html#cb66-1" aria-hidden="true" tabindex="-1"></a>allRes <span class="ot">&lt;-</span> <span class="fu">GenTable</span>(sampleGOdata, <span class="at">classicFisher =</span> resultFisher,</span>
<span id="cb66-2"><a href="08-GO.html#cb66-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">orderBy =</span> <span class="st">&quot;classicFisher&quot;</span>, <span class="at">topNodes =</span> <span class="dv">10</span>)</span>
<span id="cb66-3"><a href="08-GO.html#cb66-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-4"><a href="08-GO.html#cb66-4" aria-hidden="true" tabindex="-1"></a>allRes</span></code></pre></div>
<p>Plot a figure of the most significant GO terms. We will plot the top 5 to make it easier to view the plot.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="08-GO.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">showSigOfNodes</span>(sampleGOdata, <span class="fu">score</span>(resultFisher), <span class="at">firstSigNodes =</span> <span class="dv">5</span>, <span class="at">useInfo =</span> <span class="st">&#39;all&#39;</span>)</span></code></pre></div>
<p>This plot is nice but it's quite hard to read. Instead we will write this plot to a png file to view it.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="08-GO.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename =</span> <span class="st">&quot;GO.png&quot;</span>, <span class="at">res =</span> <span class="dv">300</span>,</span>
<span id="cb68-2"><a href="08-GO.html#cb68-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">200</span>, <span class="at">width =</span> <span class="dv">200</span>)</span>
<span id="cb68-3"><a href="08-GO.html#cb68-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-4"><a href="08-GO.html#cb68-4" aria-hidden="true" tabindex="-1"></a><span class="fu">showSigOfNodes</span>(sampleGOdata, <span class="fu">score</span>(resultFisher), <span class="at">firstSigNodes =</span> <span class="dv">5</span>, <span class="at">useInfo =</span> <span class="st">&#39;all&#39;</span>)</span>
<span id="cb68-5"><a href="08-GO.html#cb68-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-6"><a href="08-GO.html#cb68-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p>You can open the png file you just made by clicking on the Jupyter 'home page' tab and then clicking on the GO.png file you just created. You will then be able to zoom in to see the text of the plot in more details.</p>
<p><img src="figures/GO.png" width="80%" style="display: block; margin: auto;" /></p>
<p>This plots the top 5 GO terms which have been identified using the Fishers test. The 5 most significant terms are plotted as rectangles. The colour of the rectangles represent significance (dark red most significant, yellow least significant). Within each node, useful information is printed:</p>
<ul>
<li>Line 1: GO identifier</li>
<li>Line 2: trimmed GO name</li>
<li>Line 3: p-value</li>
<li>Line 4: number of significant genes and the total number of genes annotated to the respective GO term</li>
</ul>
<p>Our top 5 enriched GO terms for sex-biased genes were found to be:</p>
<ul>
<li>ribosome biogenesis</li>
<li>ribonucleoprotein complex biogenesis</li>
<li>cellular amide metabolic process</li>
<li>peptide metabolic process</li>
<li>translation</li>
</ul>
<div id="final-recap" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Final recap<a href="08-GO.html#final-recap" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/recap.png" width="20%" style="display: block; margin: auto;" /></p>
<p>In this workshop we started with raw reads, which we cleaned (<code>Trimmomatic</code>), aligned to a reference genome (<code>HISAT2</code>) and then produced a set of gene counts per sample (<code>htseq-count</code>). We then performed a differential gene expression analysis (<code>DeSeq2</code>) and GO enrichment analysis (<code>topGO</code>) to find out which genes and pathways were differentially expressed and enriched in response to a variable of interest.</p>
<p>We performed this analysis using a small subset of samples and fitted a simple model to examine a single variable. In your own analysis it is extremely important to consider your experimental design, the number of replicates you need, the amount of sequencing per sample and the best model to fit to be able to answer your specific biological question.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="07-Differential_Gene_Expression.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
