<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 14 Visualising the results | RNA-seq gene expression and pathway analysis</title>
  <meta name="description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 14 Visualising the results | RNA-seq gene expression and pathway analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 14 Visualising the results | RNA-seq gene expression and pathway analysis" />
  
  <meta name="twitter:description" content="NEOF book for the RNA-seq gene expression and pathway analysis workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher, Helen Hipperson, Ewan Harney, Steve Paterson, Bert Overduin, Matthew Gemmell, Xuan Liu" />


<meta name="date" content="2025-09-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="13-DGE_analysis.html"/>
<link rel="next" href="15-Export_results.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/tippyjs-3.2.0/themes/google.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/light-border.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/light.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/translucent.css" rel="stylesheet" />
<script src="libs/tippyjs-3.2.0/tippy.all.min.js"></script>
<script src="libs/tippy-binding-0.1.0/tippy.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>RNA-seq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="part"><span><b>Generating gene counts</b></span></li>
<li class="chapter" data-level="4" data-path="04-QC.html"><a href="04-QC.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC.html"><a href="04-QC.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-QC.html"><a href="04-QC.html#quality-assessment"><i class="fa fa-check"></i><b>4.2</b> Quality assessment</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-questions"><i class="fa fa-check"></i><b>4.2.1</b> Quality assessment: questions</a></li>
<li class="chapter" data-level="4.2.2" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-summary"><i class="fa fa-check"></i><b>4.2.2</b> Quality assessment: summary</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="04-QC.html"><a href="04-QC.html#quality-control"><i class="fa fa-check"></i><b>4.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="04-QC.html"><a href="04-QC.html#post-quality-control-check"><i class="fa fa-check"></i><b>4.3.1</b> Post Quality control check</a></li>
<li class="chapter" data-level="4.3.2" data-path="04-QC.html"><a href="04-QC.html#quality-control-questions"><i class="fa fa-check"></i><b>4.3.2</b> Quality control: questions</a></li>
<li class="chapter" data-level="4.3.3" data-path="04-QC.html"><a href="04-QC.html#quality-assessment-summary-1"><i class="fa fa-check"></i><b>4.3.3</b> Quality assessment: summary</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Aligning.html"><a href="05-Aligning.html"><i class="fa fa-check"></i><b>5</b> Aligning Illumina transcriptome data to a reference genome</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Aligning.html"><a href="05-Aligning.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>5.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="5.2" data-path="05-Aligning.html"><a href="05-Aligning.html#read-mapping"><i class="fa fa-check"></i><b>5.2</b> Read mapping</a></li>
<li class="chapter" data-level="5.3" data-path="05-Aligning.html"><a href="05-Aligning.html#assess-mapping-quality"><i class="fa fa-check"></i><b>5.3</b> Assess mapping quality</a></li>
<li class="chapter" data-level="5.4" data-path="05-Aligning.html"><a href="05-Aligning.html#visualising-the-mapping"><i class="fa fa-check"></i><b>5.4</b> Visualising the mapping</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Counting_reads.html"><a href="06-Counting_reads.html"><i class="fa fa-check"></i><b>6</b> Counting Reads using HTSeq-Count</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Counting_reads.html"><a href="06-Counting_reads.html#exercise"><i class="fa fa-check"></i><b>6.1</b> Exercise</a></li>
</ul></li>
<li class="part"><span><b>Jupyter &amp; R</b></span></li>
<li class="chapter" data-level="7" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html"><i class="fa fa-check"></i><b>7</b> Jupyter</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>7.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="7.2" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#create-r-notebook"><i class="fa fa-check"></i><b>7.2</b> Create R notebook</a></li>
<li class="chapter" data-level="7.3" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#cells-and-code"><i class="fa fa-check"></i><b>7.3</b> Cells and code</a></li>
<li class="chapter" data-level="7.4" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#create-new-cells"><i class="fa fa-check"></i><b>7.4</b> Create new cells</a></li>
<li class="chapter" data-level="7.5" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#running-code"><i class="fa fa-check"></i><b>7.5</b> Running code</a></li>
<li class="chapter" data-level="7.6" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#saving-the-file"><i class="fa fa-check"></i><b>7.6</b> Saving the file</a></li>
<li class="chapter" data-level="7.7" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>7.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="7.8" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#close-the-notebook"><i class="fa fa-check"></i><b>7.8</b> Close the notebook</a></li>
<li class="chapter" data-level="7.9" data-path="07-Jupyter_tutorial.html"><a href="07-Jupyter_tutorial.html#video-tutorial"><i class="fa fa-check"></i><b>7.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html"><i class="fa fa-check"></i><b>8</b> Basic Introduction to R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#create-a-new-jupyter-notebook"><i class="fa fa-check"></i><b>8.1</b> Create a new Jupyter notebook</a></li>
<li class="chapter" data-level="8.2" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#your-working-directory"><i class="fa fa-check"></i><b>8.2</b> Your working directory</a></li>
<li class="chapter" data-level="8.3" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#packages"><i class="fa fa-check"></i><b>8.3</b> Packages</a></li>
<li class="chapter" data-level="8.4" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#functions"><i class="fa fa-check"></i><b>8.4</b> Functions</a></li>
<li class="chapter" data-level="8.5" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#assigning-values-to-variables"><i class="fa fa-check"></i><b>8.5</b> Assigning values to variables</a></li>
<li class="chapter" data-level="8.6" data-path="08-R_tutorials.html"><a href="08-R_tutorials.html#a-few-useful-commands-and-tips"><i class="fa fa-check"></i><b>8.6</b> A few useful commands and tips</a></li>
</ul></li>
<li class="part"><span><b>Differential Gene Expression</b></span></li>
<li class="chapter" data-level="9" data-path="09-Differential_Gene_Expression.html"><a href="09-Differential_Gene_Expression.html"><i class="fa fa-check"></i><b>9</b> Intro to DGE and DESeq2</a></li>
<li class="chapter" data-level="10" data-path="10-Fly_dataset.html"><a href="10-Fly_dataset.html"><i class="fa fa-check"></i><b>10</b> The fly dataset</a></li>
<li class="chapter" data-level="11" data-path="11-Load_data.html"><a href="11-Load_data.html"><i class="fa fa-check"></i><b>11</b> Load data</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Load_data.html"><a href="11-Load_data.html#dge-notebook"><i class="fa fa-check"></i><b>11.1</b> DGE notebook</a></li>
<li class="chapter" data-level="11.2" data-path="11-Load_data.html"><a href="11-Load_data.html#loading-in-the-data"><i class="fa fa-check"></i><b>11.2</b> Loading in the data</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="11-Load_data.html"><a href="11-Load_data.html#gene-count-data"><i class="fa fa-check"></i><b>11.2.1</b> Gene count data</a></li>
<li class="chapter" data-level="11.2.2" data-path="11-Load_data.html"><a href="11-Load_data.html#metadata"><i class="fa fa-check"></i><b>11.2.2</b> Metadata</a></li>
<li class="chapter" data-level="11.2.3" data-path="11-Load_data.html"><a href="11-Load_data.html#factorise-metadata"><i class="fa fa-check"></i><b>11.2.3</b> Factorise metadata</a></li>
<li class="chapter" data-level="11.2.4" data-path="11-Load_data.html"><a href="11-Load_data.html#compare-count-and-metadata-sample-orders"><i class="fa fa-check"></i><b>11.2.4</b> Compare count and metadata sample orders</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="11-Load_data.html"><a href="11-Load_data.html#deseqdataset"><i class="fa fa-check"></i><b>11.3</b> DESeqDataset</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Filtering_the_data.html"><a href="12-Filtering_the_data.html"><i class="fa fa-check"></i><b>12</b> Filter the data</a></li>
<li class="chapter" data-level="13" data-path="13-DGE_analysis.html"><a href="13-DGE_analysis.html"><i class="fa fa-check"></i><b>13</b> DGE analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-DGE_analysis.html"><a href="13-DGE_analysis.html#dge-run"><i class="fa fa-check"></i><b>13.1</b> DGE: Run</a></li>
<li class="chapter" data-level="13.2" data-path="13-DGE_analysis.html"><a href="13-DGE_analysis.html#dge-extract-results"><i class="fa fa-check"></i><b>13.2</b> DGE: Extract results</a></li>
<li class="chapter" data-level="13.3" data-path="13-DGE_analysis.html"><a href="13-DGE_analysis.html#dge-order-by-significance"><i class="fa fa-check"></i><b>13.3</b> DGE: Order by significance</a></li>
<li class="chapter" data-level="13.4" data-path="13-DGE_analysis.html"><a href="13-DGE_analysis.html#dge-plot-counts-of-significant-features"><i class="fa fa-check"></i><b>13.4</b> DGE: Plot counts of significant features</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="14-DGE_visualisation.html"><a href="14-DGE_visualisation.html"><i class="fa fa-check"></i><b>14</b> Visualising the results</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-DGE_visualisation.html"><a href="14-DGE_visualisation.html#ma-plot"><i class="fa fa-check"></i><b>14.1</b> MA plot</a></li>
<li class="chapter" data-level="14.2" data-path="14-DGE_visualisation.html"><a href="14-DGE_visualisation.html#volcano-plot"><i class="fa fa-check"></i><b>14.2</b> Volcano plot</a></li>
<li class="chapter" data-level="14.3" data-path="14-DGE_visualisation.html"><a href="14-DGE_visualisation.html#heatmap"><i class="fa fa-check"></i><b>14.3</b> Heatmap</a></li>
<li class="chapter" data-level="14.4" data-path="14-DGE_visualisation.html"><a href="14-DGE_visualisation.html#principle-component-analysis"><i class="fa fa-check"></i><b>14.4</b> Principle Component Analysis</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Export_results.html"><a href="15-Export_results.html"><i class="fa fa-check"></i><b>15</b> Export the results</a></li>
<li class="part"><span><b>Gene Ontology</b></span></li>
<li class="chapter" data-level="16" data-path="16-GO.html"><a href="16-GO.html"><i class="fa fa-check"></i><b>16</b> Gene Ontology enrichment analysis</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-GO.html"><a href="16-GO.html#go-setup"><i class="fa fa-check"></i><b>16.1</b> GO: Setup</a></li>
<li class="chapter" data-level="16.2" data-path="16-GO.html"><a href="16-GO.html#go-extract-top-differential-genes"><i class="fa fa-check"></i><b>16.2</b> GO: Extract top differential genes</a></li>
<li class="chapter" data-level="16.3" data-path="16-GO.html"><a href="16-GO.html#go-map-gos-to-genes"><i class="fa fa-check"></i><b>16.3</b> GO: Map GOs to genes</a></li>
<li class="chapter" data-level="16.4" data-path="16-GO.html"><a href="16-GO.html#go-create-topgodata-object"><i class="fa fa-check"></i><b>16.4</b> GO: Create topGOdata object</a></li>
<li class="chapter" data-level="16.5" data-path="16-GO.html"><a href="16-GO.html#go-enrichment-analysis"><i class="fa fa-check"></i><b>16.5</b> GO: enrichment analysis</a></li>
<li class="chapter" data-level="16.6" data-path="16-GO.html"><a href="16-GO.html#go-enrichment-network-plot"><i class="fa fa-check"></i><b>16.6</b> GO: Enrichment network plot</a></li>
</ul></li>
<li class="part"><span><b>Recap &amp; appendix</b></span></li>
<li class="chapter" data-level="17" data-path="17-Recap.html"><a href="17-Recap.html"><i class="fa fa-check"></i><b>17</b> Final recap</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="18-Appendix.html"><a href="18-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="18-Appendix.html"><a href="18-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
<li class="chapter" data-level="A.2" data-path="18-Appendix.html"><a href="18-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>A.2</b> Jupyter-notebook</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RNA-seq gene expression and pathway analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visualising-the-results" class="section level1 hasAnchor" number="14">
<h1><span class="header-section-number">Chapter 14</span> Visualising the results<a href="14-DGE_visualisation.html#visualising-the-results" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/eye_brown.png" style="width:300px" />
</center>
<p>We will now go through several different methods of visualising the results from our DGE analysis.</p>
<div id="ma-plot" class="section level2 hasAnchor" number="14.1">
<h2><span class="header-section-number">14.1</span> MA plot<a href="14-DGE_visualisation.html#ma-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/MA_horseshoe.png" style="width:200px" />
</center>
<p>We will first plot an MA plot. This gives us an overview of the experiment and how DESeq2 calls significant DEG. Each gene is represented by a dot. Genes with significant adjusted p-values (default &lt; 0.01) are coloured in blue. This shows that DESeq2 will only call significant DGE in genes with a large average normalised count as only these genes contain sufficient information to be able to do so.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb59-1"><a href="14-DGE_visualisation.html#cb59-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb59-2"><a href="14-DGE_visualisation.html#cb59-2" tabindex="-1"></a></span>
<span id="cb59-3"><a href="14-DGE_visualisation.html#cb59-3" tabindex="-1"></a><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>))</span></code></pre></div>
<center>
<img src="figures/MA.png" style="width:700px; border-radius:15px" />
</center>
</div>
<div id="volcano-plot" class="section level2 hasAnchor" number="14.2">
<h2><span class="header-section-number">14.2</span> Volcano plot<a href="14-DGE_visualisation.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/volcano_icon.png" style="width:200px" />
</center>
<p>One nice way to visualise your results is to plot a volcano plot.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb60-1"><a href="14-DGE_visualisation.html#cb60-1" tabindex="-1"></a><span class="co"># Make the basic volcano plot</span></span>
<span id="cb60-2"><a href="14-DGE_visualisation.html#cb60-2" tabindex="-1"></a><span class="fu">with</span>(res, <span class="fu">plot</span>(log2FoldChange, </span>
<span id="cb60-3"><a href="14-DGE_visualisation.html#cb60-3" tabindex="-1"></a>               <span class="sc">-</span><span class="fu">log10</span>(pvalue), </span>
<span id="cb60-4"><a href="14-DGE_visualisation.html#cb60-4" tabindex="-1"></a>               <span class="at">pch=</span><span class="dv">20</span>, <span class="at">main=</span><span class="st">&quot;Volcano plot&quot;</span>, </span>
<span id="cb60-5"><a href="14-DGE_visualisation.html#cb60-5" tabindex="-1"></a>               <span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>,<span class="dv">10</span>),  <span class="at">col=</span><span class="st">&quot;grey&quot;</span>))</span>
<span id="cb60-6"><a href="14-DGE_visualisation.html#cb60-6" tabindex="-1"></a></span>
<span id="cb60-7"><a href="14-DGE_visualisation.html#cb60-7" tabindex="-1"></a><span class="co"># Colour points light blue if padj&lt;0.01</span></span>
<span id="cb60-8"><a href="14-DGE_visualisation.html#cb60-8" tabindex="-1"></a><span class="fu">with</span>(<span class="fu">subset</span>(res, padj<span class="sc">&lt;</span><span class="fl">0.01</span> ), </span>
<span id="cb60-9"><a href="14-DGE_visualisation.html#cb60-9" tabindex="-1"></a>     <span class="fu">points</span>(log2FoldChange, </span>
<span id="cb60-10"><a href="14-DGE_visualisation.html#cb60-10" tabindex="-1"></a>            <span class="sc">-</span><span class="fu">log10</span>(pvalue), </span>
<span id="cb60-11"><a href="14-DGE_visualisation.html#cb60-11" tabindex="-1"></a>            <span class="at">pch=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;#a6cee3&quot;</span>))</span>
<span id="cb60-12"><a href="14-DGE_visualisation.html#cb60-12" tabindex="-1"></a><span class="co"># Colour points dark blue if log2FC&gt;1 and padj&lt;0.01</span></span>
<span id="cb60-13"><a href="14-DGE_visualisation.html#cb60-13" tabindex="-1"></a><span class="fu">with</span>(<span class="fu">subset</span>(res, padj<span class="sc">&lt;</span><span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange)<span class="sc">&gt;</span><span class="dv">1</span>), </span>
<span id="cb60-14"><a href="14-DGE_visualisation.html#cb60-14" tabindex="-1"></a>     <span class="fu">points</span>(log2FoldChange, </span>
<span id="cb60-15"><a href="14-DGE_visualisation.html#cb60-15" tabindex="-1"></a>            <span class="sc">-</span><span class="fu">log10</span>(pvalue), </span>
<span id="cb60-16"><a href="14-DGE_visualisation.html#cb60-16" tabindex="-1"></a>            <span class="at">pch=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;#1f78b4&quot;</span>))</span></code></pre></div>
<center>
<img src="figures/volcano.png" style="width:700px; border-radius:15px" />
</center>
<p>This allows us to visualise the number of differentially expressed genes which are significant (light blue). We can also visualise how many genes have an expression level which is at least twice as high in males than in females (log2FoldChange&gt;1) and those which are twice as high in females than in males (log2FoldChange&lt;1).</p>
<p>In our case, we have a fairly even number of genes both up and down regulated in males compared to females.</p>
</div>
<div id="heatmap" class="section level2 hasAnchor" number="14.3">
<h2><span class="header-section-number">14.3</span> Heatmap<a href="14-DGE_visualisation.html#heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/heatmap_icon.png" style="width:200px" />
</center>
<p>Usually, the variation between samples within a sample group (within-group variation) is smaller than that between samples from different sample groups (overall variation) because the former consists of technical and biological variation only, while the latter also contains variation due to the effects of factors. When the factor effect is the dominant contributor of variation, the sample groups can be clearly separated. However, if the factor effect is weak compared to the technical and biological variation within sample groups, it will be difficult to discriminate groups from the data.</p>
<p>PCA and Correlation Heatmap plots visualise the outcomes of the assessment of the variation within and between sample groups.</p>
<p>Before we plot these we will transform the data. For the differential expression analysis we use raw counts but for visualisation and clustering purposes it is common to work with transformed data. There are two main ways suggested for transforming data for visualisation and clustering in DESeq2: VST (variance stabilising transformations) and the rlog (regularised logarithm). For more information see <a href="http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#data-transformations-and-visualization">here</a>. Here we will use the rlog transformation before plotting the Heatmap and PCA plots.</p>
<p>First use the rlog transformation to normalise the data.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb61-1"><a href="14-DGE_visualisation.html#cb61-1" tabindex="-1"></a>rld <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds, <span class="at">blind=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p>The argument blind, refers to whether the transformation should be blind to the sample information specified by the design formula. This is not appopriate to use as we are expecting many of the genes to have counts which are explained by the experimental design we specified (i.e. sex), so we have set this argument to false.</p>
<p>We will use the package <code>pheatmap</code> to plot a heatmap for our samples and see how they cluster.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb62-1"><a href="14-DGE_visualisation.html#cb62-1" tabindex="-1"></a><span class="co"># load pheatmap &amp; ggplot2</span></span>
<span id="cb62-2"><a href="14-DGE_visualisation.html#cb62-2" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb62-3"><a href="14-DGE_visualisation.html#cb62-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb62-4"><a href="14-DGE_visualisation.html#cb62-4" tabindex="-1"></a></span>
<span id="cb62-5"><a href="14-DGE_visualisation.html#cb62-5" tabindex="-1"></a><span class="co"># extract assay of numeric values from the rld object needed by pheatmap</span></span>
<span id="cb62-6"><a href="14-DGE_visualisation.html#cb62-6" tabindex="-1"></a>rld_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(rld)</span>
<span id="cb62-7"><a href="14-DGE_visualisation.html#cb62-7" tabindex="-1"></a><span class="co"># compute the pairwise correlation values for samples and make a matrix of correlations</span></span>
<span id="cb62-8"><a href="14-DGE_visualisation.html#cb62-8" tabindex="-1"></a>rld_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(rld_mat)</span>
<span id="cb62-9"><a href="14-DGE_visualisation.html#cb62-9" tabindex="-1"></a><span class="co"># rename the coldata by the sex column in the metadata</span></span>
<span id="cb62-10"><a href="14-DGE_visualisation.html#cb62-10" tabindex="-1"></a><span class="fu">rownames</span>(rld_cor)<span class="ot">&lt;-</span>coldata<span class="sc">$</span>sex</span>
<span id="cb62-11"><a href="14-DGE_visualisation.html#cb62-11" tabindex="-1"></a><span class="fu">head</span>(rld_cor)</span></code></pre></div>
<div class="sourceCode" id="cb63"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb63-1"><a href="14-DGE_visualisation.html#cb63-1" tabindex="-1"></a><span class="co"># plot heatmap</span></span>
<span id="cb63-2"><a href="14-DGE_visualisation.html#cb63-2" tabindex="-1"></a><span class="fu">pheatmap</span>(rld_cor)</span></code></pre></div>
<center>
<img src="figures/heatmap.png" style="width:700px; border-radius:15px" />
</center>
<p>We can see that in general samples seem to cluster on the heatmap by sex.</p>
</div>
<div id="principle-component-analysis" class="section level2 hasAnchor" number="14.4">
<h2><span class="header-section-number">14.4</span> Principle Component Analysis<a href="14-DGE_visualisation.html#principle-component-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/Scatterplot_2.png" style="width:200px" />
</center>
<p>We will now plot the PCA and colour the points based on sex. We specify that females will be coloured green and males orange.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb64-1"><a href="14-DGE_visualisation.html#cb64-1" tabindex="-1"></a><span class="fu">plotPCA</span>(rld, <span class="at">intgroup=</span><span class="st">&quot;sex&quot;</span>)<span class="sc">+</span></span>
<span id="cb64-2"><a href="14-DGE_visualisation.html#cb64-2" tabindex="-1"></a>            <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb64-3"><a href="14-DGE_visualisation.html#cb64-3" tabindex="-1"></a>            <span class="fu">theme</span>(<span class="at">legend.title=</span><span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb64-4"><a href="14-DGE_visualisation.html#cb64-4" tabindex="-1"></a>            <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#33a02c&quot;</span>, <span class="st">&quot;#ff7f00&quot;</span>))</span></code></pre></div>
<center>
<img src="figures/PCA_sex.png" style="width:700px; border-radius:15px" />
</center>
<p>The PCA shows us that 32% of the variance can be explained by PC1 and looking at how the points are clustered we can suggest that sex is driving the division based on the x axis.</p>
<p>24% assigned to PC2. We can recolour the plot based on the group variable from our metadata to see if this sheds some more light on what could be driving the differences between samples on the y axis.</p>
<p>Plot again and colour based on group.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb65-1"><a href="14-DGE_visualisation.html#cb65-1" tabindex="-1"></a><span class="fu">plotPCA</span>(rld, <span class="at">intgroup=</span><span class="st">&quot;group&quot;</span>)<span class="sc">+</span></span>
<span id="cb65-2"><a href="14-DGE_visualisation.html#cb65-2" tabindex="-1"></a>                <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb65-3"><a href="14-DGE_visualisation.html#cb65-3" tabindex="-1"></a>                <span class="fu">theme</span>(<span class="at">legend.title=</span><span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb65-4"><a href="14-DGE_visualisation.html#cb65-4" tabindex="-1"></a>                <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#a6cee3&quot;</span>, <span class="st">&quot;#1f78b4&quot;</span>, <span class="st">&quot;#fb9a99&quot;</span>, </span>
<span id="cb65-5"><a href="14-DGE_visualisation.html#cb65-5" tabindex="-1"></a>                                              <span class="st">&quot;#e31a1c&quot;</span>, <span class="st">&quot;#b2df8a&quot;</span>, <span class="st">&quot;#33a02c&quot;</span>))</span></code></pre></div>
<center>
<img src="figures/PCA_group.png" style="width:700px; border-radius:15px" />
</center>
<p>All samples in this dataset are from control samples and head tissue. We specify different colours based on the regime (<strong>B</strong>ase, <strong>M</strong>ultiple mates, <strong>E</strong>volved). Darker points are samples from males and lighter colours are from females.</p>
<p>We can see that the diversity on the y axis appears to be driven by the regime, so it would be interesting to repeat our analysis adding regime to the formula. Regime having some affect on gene expression patterns makes sense as mating system variation has been shown to influence sex-biased gene expression in some animal species.</p>
<p>For more information on how you would run more complex models please see Chapter 4 of the supplemental bookdown.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="13-DGE_analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="15-Export_results.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

</body>

</html>
