<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Identifying and removing primers | Metabarcoding for diet analysis and environmental DNA</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Identifying and removing primers | Metabarcoding for diet analysis and environmental DNA" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Identifying and removing primers | Metabarcoding for diet analysis and environmental DNA" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher and Helen Hipperson" />


<meta name="date" content="2023-09-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="05-Loading_data.html"/>
<link rel="next" href="07-quality_checks.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Environmental metabarcoding</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Metabarcoding.html"><a href="01-Metabarcoding.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Example_dataset.html"><a href="02-Example_dataset.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html"><i class="fa fa-check"></i><b>3</b> Introduction to DADA2 and R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#dada2"><i class="fa fa-check"></i><b>3.1</b> DADA2</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#otus-vs-asvs"><i class="fa fa-check"></i><b>3.1.1</b> OTUs vs ASVs</a></li>
<li class="chapter" data-level="3.1.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#workflow"><i class="fa fa-check"></i><b>3.1.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#r"><i class="fa fa-check"></i><b>3.2</b> R</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#the-console-window"><i class="fa fa-check"></i><b>3.2.1</b> The console window</a></li>
<li class="chapter" data-level="3.2.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#your-working-directory"><i class="fa fa-check"></i><b>3.2.2</b> Your working directory</a></li>
<li class="chapter" data-level="3.2.3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#packages"><i class="fa fa-check"></i><b>3.2.3</b> Packages</a></li>
<li class="chapter" data-level="3.2.4" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#functions"><i class="fa fa-check"></i><b>3.2.4</b> Functions</a></li>
<li class="chapter" data-level="3.2.5" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#assigning-values-to-variables"><i class="fa fa-check"></i><b>3.2.5</b> Assigning values to variables</a></li>
<li class="chapter" data-level="3.2.6" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#a-few-useful-commands-and-tips"><i class="fa fa-check"></i><b>3.2.6</b> A few useful commands and tips</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>DADA2</b></span></li>
<li class="chapter" data-level="5" data-path="05-Loading_data.html"><a href="05-Loading_data.html"><i class="fa fa-check"></i><b>5</b> Loading data into R and DADA2</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#jupyter"><i class="fa fa-check"></i><b>5.1</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.1.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#create-r-notebook"><i class="fa fa-check"></i><b>5.1.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.1.3" data-path="05-Loading_data.html"><a href="05-Loading_data.html#cells-and-code"><i class="fa fa-check"></i><b>5.1.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.1.4" data-path="05-Loading_data.html"><a href="05-Loading_data.html#create-new-cells"><i class="fa fa-check"></i><b>5.1.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.1.5" data-path="05-Loading_data.html"><a href="05-Loading_data.html#running-code"><i class="fa fa-check"></i><b>5.1.5</b> Running code</a></li>
<li class="chapter" data-level="5.1.6" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-file"><i class="fa fa-check"></i><b>5.1.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.1.7" data-path="05-Loading_data.html"><a href="05-Loading_data.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.1.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.1.8" data-path="05-Loading_data.html"><a href="05-Loading_data.html#close-the-notebook"><i class="fa fa-check"></i><b>5.1.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.1.9" data-path="05-Loading_data.html"><a href="05-Loading_data.html#video-tutorial"><i class="fa fa-check"></i><b>5.1.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#r-1"><i class="fa fa-check"></i><b>5.2</b> R</a></li>
<li class="chapter" data-level="5.3" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-dada2"><i class="fa fa-check"></i><b>5.3</b> Loading DADA2</a></li>
<li class="chapter" data-level="5.4" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-our-data"><i class="fa fa-check"></i><b>5.4</b> Loading our data</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-paths-for-our-input-and-output-directories"><i class="fa fa-check"></i><b>5.4.1</b> Saving the paths for our input and output directories</a></li>
<li class="chapter" data-level="5.4.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#inputting-the-forward-and-reverse-reads"><i class="fa fa-check"></i><b>5.4.2</b> Inputting the forward and reverse reads</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-primer_removal.html"><a href="06-primer_removal.html"><i class="fa fa-check"></i><b>6</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-primer_removal.html"><a href="06-primer_removal.html#primer-orientation-checking"><i class="fa fa-check"></i><b>6.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="6.2" data-path="06-primer_removal.html"><a href="06-primer_removal.html#cutadapt"><i class="fa fa-check"></i><b>6.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-quality_checks.html"><a href="07-quality_checks.html"><i class="fa fa-check"></i><b>7</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="8" data-path="08-clean.html"><a href="08-clean.html"><i class="fa fa-check"></i><b>8</b> Cleaning your data</a></li>
<li class="chapter" data-level="9" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html"><i class="fa fa-check"></i><b>9</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#generate-an-error-model"><i class="fa fa-check"></i><b>9.1</b> Generate an error model</a></li>
<li class="chapter" data-level="9.2" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dereplication"><i class="fa fa-check"></i><b>9.2</b> Dereplication</a></li>
<li class="chapter" data-level="9.3" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#inferrence-of-asvs"><i class="fa fa-check"></i><b>9.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="9.4" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#merging-paired-end-reads"><i class="fa fa-check"></i><b>9.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="9.5" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#making-our-asv-matrix"><i class="fa fa-check"></i><b>9.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="9.6" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#chimera-detection-and-removal"><i class="fa fa-check"></i><b>9.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="9.7" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#sequence-tracking-sanity-check"><i class="fa fa-check"></i><b>9.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Assign_taxonomy.html"><a href="10-Assign_taxonomy.html"><i class="fa fa-check"></i><b>10</b> Assigning taxonomy</a></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="11" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html"><i class="fa fa-check"></i><b>11</b> Further analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#r-packages-for-further-analysis"><i class="fa fa-check"></i><b>11.1</b> R packages for further analysis</a></li>
<li class="chapter" data-level="11.2" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#rarefaction-curves"><i class="fa fa-check"></i><b>11.2</b> Rarefaction curves</a></li>
<li class="chapter" data-level="11.3" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#alpha-diversity"><i class="fa fa-check"></i><b>11.3</b> Alpha diversity</a></li>
<li class="chapter" data-level="11.4" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#beta-diversity"><i class="fa fa-check"></i><b>11.4</b> Beta diversity</a></li>
</ul></li>
<li class="part"><span><b>Second marker</b></span></li>
<li class="chapter" data-level="12" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html"><i class="fa fa-check"></i><b>12</b> Analysing the second primer pair</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html#the-primer-sequences"><i class="fa fa-check"></i><b>12.1</b> The primer sequences</a></li>
<li class="chapter" data-level="12.2" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html#questions"><i class="fa fa-check"></i><b>12.2</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html"><i class="fa fa-check"></i><b>13</b> Guide to analysing the second primer pair</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#loading-the-data"><i class="fa fa-check"></i><b>13.1</b> Loading the data</a></li>
<li class="chapter" data-level="13.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#identifying-and-removing-primers"><i class="fa fa-check"></i><b>13.2</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#primer-orientation-checking-1"><i class="fa fa-check"></i><b>13.2.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="13.2.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#cutadapt-1"><i class="fa fa-check"></i><b>13.2.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#checking-the-quality-of-your-data"><i class="fa fa-check"></i><b>13.3</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="13.4" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#cleaning-your-data"><i class="fa fa-check"></i><b>13.4</b> Cleaning your data</a></li>
<li class="chapter" data-level="13.5" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#identification-of-asvs"><i class="fa fa-check"></i><b>13.5</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#generate-an-error-model-1"><i class="fa fa-check"></i><b>13.5.1</b> Generate an error model</a></li>
<li class="chapter" data-level="13.5.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#dereplication-1"><i class="fa fa-check"></i><b>13.5.2</b> Dereplication</a></li>
<li class="chapter" data-level="13.5.3" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#inferrence-of-asvs-1"><i class="fa fa-check"></i><b>13.5.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="13.5.4" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#merging-paired-end-reads-1"><i class="fa fa-check"></i><b>13.5.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="13.5.5" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#making-our-asv-matrix-1"><i class="fa fa-check"></i><b>13.5.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="13.5.6" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#chimera-detection-and-removal-1"><i class="fa fa-check"></i><b>13.5.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="13.5.7" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#sequence-tracking-sanity-check-1"><i class="fa fa-check"></i><b>13.5.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#assigning-taxonomy"><i class="fa fa-check"></i><b>13.6</b> Assigning taxonomy</a></li>
<li class="chapter" data-level="13.7" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#further-analysis"><i class="fa fa-check"></i><b>13.7</b> Further analysis</a></li>
<li class="chapter" data-level="13.8" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#rarefaction-curves-1"><i class="fa fa-check"></i><b>13.8</b> Rarefaction curves</a></li>
<li class="chapter" data-level="13.9" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#alpha-diversity-1"><i class="fa fa-check"></i><b>13.9</b> Alpha diversity</a></li>
<li class="chapter" data-level="13.10" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#beta-diversity-1"><i class="fa fa-check"></i><b>13.10</b> Beta diversity</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Metabarcoding for diet analysis and environmental DNA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="removeprimer" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Identifying and removing primers<a href="06-primer_removal.html#removeprimer" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Now you will carry out primer removal. DADA2 requires the primers you used to be trimmed off the forward and reverse of your reads. We will use the software cutadapt for this from within R.</p>
<p>Read in your forward and reverse primer sequences for our MiFish-U primerset. The MiFish-U primers do not contain any degenerate bases but cutadapt is able to handle any degenerate bases you may have in primer sequences for your own projects.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="06-primer_removal.html#cb20-1" aria-hidden="true" tabindex="-1"></a>FWD <span class="ot">&lt;-</span> <span class="st">&quot;GTCGGTAAAACTCGTGCCAGC&quot;</span></span>
<span id="cb20-2"><a href="06-primer_removal.html#cb20-2" aria-hidden="true" tabindex="-1"></a>REV <span class="ot">&lt;-</span> <span class="st">&quot;CATAGTGGGGTATCTAATCCCAGTTTG&quot;</span></span></code></pre></div>
<p>We have written the primer sequences out below with space dividers to make it easier to type in. Remember to remove the spaces when you type this into R!</p>
<p><strong>Forward primer</strong></p>
<ul>
<li>GTC GGT AAA ACT CGT GCC AGC</li>
</ul>
<p><strong>Reverse primer</strong></p>
<ul>
<li>CAT AGT GGG GTA TCT AAT CCC AGT TTG</li>
</ul>
<div id="primer-orientation-checking" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Primer orientation checking<a href="06-primer_removal.html#primer-orientation-checking" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It might be useful to check the orientation of the primers on our sequences so we know the best way to remove them. We know how our libraries were created and sequenced so we are expecting that the forward primer should be located in a forward orientation in the forward read and the reverse primer should be located in the forward orientation in the reverse read.</p>
<p>For your own data you might be less certain as to which orientation your primers are in your reads.</p>
<p>We use the following code to make a list of all orientations of our primer sequences (forward, complement, reverse and reverse completment).</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="06-primer_removal.html#cb21-1" aria-hidden="true" tabindex="-1"></a>allOrients <span class="ot">&lt;-</span> <span class="cf">function</span>(primer) {</span>
<span id="cb21-2"><a href="06-primer_removal.html#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create all orientations of the input sequence</span></span>
<span id="cb21-3"><a href="06-primer_removal.html#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(Biostrings)</span>
<span id="cb21-4"><a href="06-primer_removal.html#cb21-4" aria-hidden="true" tabindex="-1"></a>  dna <span class="ot">&lt;-</span> <span class="fu">DNAString</span>(primer)  <span class="co"># The Biostrings works w/ DNAString </span></span>
<span id="cb21-5"><a href="06-primer_removal.html#cb21-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="co">#objects rather than character vectors</span></span>
<span id="cb21-6"><a href="06-primer_removal.html#cb21-6" aria-hidden="true" tabindex="-1"></a>  orients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">Forward =</span> dna, <span class="at">Complement =</span> <span class="fu">complement</span>(dna), <span class="at">Reverse =</span> <span class="fu">reverse</span>(dna), </span>
<span id="cb21-7"><a href="06-primer_removal.html#cb21-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">RevComp =</span> <span class="fu">reverseComplement</span>(dna))</span>
<span id="cb21-8"><a href="06-primer_removal.html#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sapply</span>(orients, toString))  <span class="co"># Convert back to character vector</span></span>
<span id="cb21-9"><a href="06-primer_removal.html#cb21-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-10"><a href="06-primer_removal.html#cb21-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-11"><a href="06-primer_removal.html#cb21-11" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(FWD)</span>
<span id="cb21-12"><a href="06-primer_removal.html#cb21-12" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="co"># print all orientations of the forward primer to the console</span></span>
<span id="cb21-13"><a href="06-primer_removal.html#cb21-13" aria-hidden="true" tabindex="-1"></a>REV.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(REV)</span>
<span id="cb21-14"><a href="06-primer_removal.html#cb21-14" aria-hidden="true" tabindex="-1"></a>REV.orients <span class="co"># print all orientations of the reverse primer to the console</span></span></code></pre></div>
<p>It is a good idea to pre-filter your data before searching for the primer sequences to remove Ns (ambiguously called bases) from the dataset and improve mapping of the primers. To do this we can use the <code>filterAndTrim</code> function. For more information on <code>filterAndTrim</code> and the parameters you can specify type into the console.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="06-primer_removal.html#cb22-1" aria-hidden="true" tabindex="-1"></a>?filterAndTrim</span></code></pre></div>
<p>(Close the pop-up window).</p>
<p>For now we are only filtering Ns but we will come back to this function later after removing our primers. Specify the path to a new output directory to contain the N filtered files. We will call this directory <code>filtN</code>, and a copy of each of the forward and reverse fastq files with any ‘N’ containing sequences removed will be written here.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="06-primer_removal.html#cb23-1" aria-hidden="true" tabindex="-1"></a>fnFs.filtN <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output.path, <span class="st">&quot;filtN&quot;</span>, <span class="fu">basename</span>(fnFs)) </span>
<span id="cb23-2"><a href="06-primer_removal.html#cb23-2" aria-hidden="true" tabindex="-1"></a>fnRs.filtN <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output.path, <span class="st">&quot;filtN&quot;</span>, <span class="fu">basename</span>(fnRs))</span></code></pre></div>
<p>Then run <code>filterAndTrim</code> with <code>maxN = 0</code> (sequences with more than 0 Ns will be discarded) and <code>multithread=TRUE</code> (all input files are filtered in parallel, which is faster).</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="06-primer_removal.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">filterAndTrim</span>(fnFs, fnFs.filtN, fnRs, fnRs.filtN, <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>For this function we have specified:</p>
<ul>
<li>the path to our raw forward read fatsqs <code>fnFs</code></li>
<li>the path to write our N-trimmed forward fastqs <code>fnFs.filtN</code></li>
<li>the path to our raw reverse read fatsqs <code>fnRs</code></li>
<li>the path to write our N-trimmed reverse fastqs <code>fnRs.filtN</code></li>
<li>and the function-specific options for <code>maxN</code> and <code>multithread</code></li>
</ul>
<p>This will take a few minutes to run. Wait for <code>filterAndTrim</code> to finish running before running the next step. This is indicated by the colour of the circle next to <code>R</code> on the right-hand side of the screen. Grey=running, clear=idle:</p>
<p><img src="figures/running_circle.png" width="100%" style="display: block; margin: auto;" /></p>
<p>The following function will use all possible primer combinations to count the number of times a primer is found in the forward and reverse read in each orientation.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="06-primer_removal.html#cb25-1" aria-hidden="true" tabindex="-1"></a>primerHits <span class="ot">&lt;-</span> <span class="cf">function</span>(primer, fn) {</span>
<span id="cb25-2"><a href="06-primer_removal.html#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Counts number of reads in which the primer is found</span></span>
<span id="cb25-3"><a href="06-primer_removal.html#cb25-3" aria-hidden="true" tabindex="-1"></a>  nhits <span class="ot">&lt;-</span> <span class="fu">vcountPattern</span>(primer, <span class="fu">sread</span>(<span class="fu">readFastq</span>(fn)), <span class="at">fixed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-4"><a href="06-primer_removal.html#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sum</span>(nhits <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb25-5"><a href="06-primer_removal.html#cb25-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb25-6"><a href="06-primer_removal.html#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="06-primer_removal.html#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.filtN[[<span class="dv">1</span>]]), </span>
<span id="cb25-8"><a href="06-primer_removal.html#cb25-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnRs.filtN[[<span class="dv">1</span>]]), </span>
<span id="cb25-9"><a href="06-primer_removal.html#cb25-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.filtN[[<span class="dv">1</span>]]), </span>
<span id="cb25-10"><a href="06-primer_removal.html#cb25-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnRs.filtN[[<span class="dv">1</span>]]))</span></code></pre></div>
<p>The <code>[[1]]</code> part of the rbind command runs the function on the first file in our list. To specify the second file you would change this to <code>[[2]]</code>. However, we do not need to run this command more than once (which is good as it can take a little while to run!) because we know that the libraries for each sample were made using the same protocol. Therefore, all the samples should look similar.</p>
<p>We can see that although most of our primers are found as expected in the forward orientation on our forward/reverse read in this sample, we have a few sequences that also contain the reverse complement. This is likely due to the fact that there is some variation in the length of this amplicon between species so in some cases the sequencing machine has cycled through the full region and into the the forward/reverse primer.</p>
</div>
<div id="cutadapt" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Cutadapt<a href="06-primer_removal.html#cutadapt" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now we know the orientation of our primers we can use the software cutadapt to remove the primers from our samples.</p>
<p>See <a href="https://cutadapt.readthedocs.io/en/stable/" class="uri">https://cutadapt.readthedocs.io/en/stable/</a> for more information on how to install and run cutadapt.</p>
<p>Specify the path to the cutadapt software so R knows where to find and run it.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="06-primer_removal.html#cb26-1" aria-hidden="true" tabindex="-1"></a>cutadapt <span class="ot">&lt;-</span> <span class="st">&quot;/pub39/tea/nsc006/anaconda3/envs/qiime2-2021.2/bin/cutadapt&quot;</span></span></code></pre></div>
<p>We can use the <code>system2</code> command to run a shell command from within R.</p>
<p>Specify the path to the output files where we want to put the cutadapt output files.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="06-primer_removal.html#cb27-1" aria-hidden="true" tabindex="-1"></a>path.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output.path, <span class="st">&quot;cutadapt&quot;</span>)</span>
<span id="cb27-2"><a href="06-primer_removal.html#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut)) <span class="fu">dir.create</span>(path.cut)</span>
<span id="cb27-3"><a href="06-primer_removal.html#cb27-3" aria-hidden="true" tabindex="-1"></a>fnFs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnFs))</span>
<span id="cb27-4"><a href="06-primer_removal.html#cb27-4" aria-hidden="true" tabindex="-1"></a>fnRs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnRs))</span></code></pre></div>
<p>Here we specify the options needed by cutadapt in order to trim the forward orientation of the forward and reverse primer off the forward and reverse read and the reverse complement off the end of the forward and reverse reads.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="06-primer_removal.html#cb28-1" aria-hidden="true" tabindex="-1"></a>FWD.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD)</span>
<span id="cb28-2"><a href="06-primer_removal.html#cb28-2" aria-hidden="true" tabindex="-1"></a>REV.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV)</span>
<span id="cb28-3"><a href="06-primer_removal.html#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Trim FWD and the reverse-complement of REV off of R1 (forward reads)</span></span>
<span id="cb28-4"><a href="06-primer_removal.html#cb28-4" aria-hidden="true" tabindex="-1"></a>R1.flags <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;-g&quot;</span>, FWD, <span class="st">&quot;-a&quot;</span>, REV.RC) </span>
<span id="cb28-5"><a href="06-primer_removal.html#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Trim REV and the reverse-complement of FWD off of R2 (reverse reads)</span></span>
<span id="cb28-6"><a href="06-primer_removal.html#cb28-6" aria-hidden="true" tabindex="-1"></a>R2.flags <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;-G&quot;</span>, REV, <span class="st">&quot;-A&quot;</span>, FWD.RC) </span></code></pre></div>
<p>What do the <code>-g</code> <code>-a</code> <code>-G</code> and <code>-A</code> parameters mean? (Hint: use the following command to check the helpfile for cutadapt)</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="06-primer_removal.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="st">&quot;--help&quot;</span>)</span></code></pre></div>
<p>This should open the cutadapt help information in the Linux terminal session rather than Jupyter-Notebooks directly.</p>
<p>In addition to trimming the primers from the reads we will also specify a couple of extra useful parameters.</p>
<p><code>--discard-untrimmed</code> this tells cutadapt to discard any read where the primers haven’t been trimmed off. This is especially important for our data as our files at the moment contain sequences amplified using both MiFish-U and 12S-V5 primer sets. We only want to keep sequences matching the MiFish-U primer set for this analysis.</p>
<p><code>--minimum-length 60</code> discard reads shorter than 60bp. This will remove unexpected short reads and help speed up further analysis.</p>
<p>Run cutadapt. (Note. The cutadapt step is time intensive so this might take a little while to run, probably about 15 minutes, and a lot of output will be written to the Linux terminal screen while it is running).</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="06-primer_removal.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {</span>
<span id="cb30-2"><a href="06-primer_removal.html#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(R1.flags, R2.flags, <span class="st">&quot;-n&quot;</span>, <span class="dv">2</span>, </span>
<span id="cb30-3"><a href="06-primer_removal.html#cb30-3" aria-hidden="true" tabindex="-1"></a>                             <span class="co"># -n 2 required to remove FWD and REV</span></span>
<span id="cb30-4"><a href="06-primer_removal.html#cb30-4" aria-hidden="true" tabindex="-1"></a>                             <span class="st">&quot;-o&quot;</span>, fnFs.cut[i], <span class="st">&quot;-p&quot;</span>, fnRs.cut[i], <span class="co"># output files</span></span>
<span id="cb30-5"><a href="06-primer_removal.html#cb30-5" aria-hidden="true" tabindex="-1"></a>                             fnFs.filtN[i], fnRs.filtN[i], <span class="co"># input files</span></span>
<span id="cb30-6"><a href="06-primer_removal.html#cb30-6" aria-hidden="true" tabindex="-1"></a>                             <span class="st">&quot;--discard-untrimmed&quot;</span>,</span>
<span id="cb30-7"><a href="06-primer_removal.html#cb30-7" aria-hidden="true" tabindex="-1"></a>                             <span class="st">&quot;--minimum-length 60&quot;</span>))</span>
<span id="cb30-8"><a href="06-primer_removal.html#cb30-8" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>We can now check whether all the primers have been removed using the primerHits function we specified earlier.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="06-primer_removal.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">1</span>]]), </span>
<span id="cb31-2"><a href="06-primer_removal.html#cb31-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">1</span>]]), </span>
<span id="cb31-3"><a href="06-primer_removal.html#cb31-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">1</span>]]), </span>
<span id="cb31-4"><a href="06-primer_removal.html#cb31-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">1</span>]]))</span></code></pre></div>
<p>We now have no primers remaining in our file.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="05-Loading_data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="07-quality_checks.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
