<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Identification of ASVs | Metabarcoding for diet analysis and environmental DNA</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Identification of ASVs | Metabarcoding for diet analysis and environmental DNA" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Identification of ASVs | Metabarcoding for diet analysis and environmental DNA" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher and Helen Hipperson" />


<meta name="date" content="2023-09-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="08-clean.html"/>
<link rel="next" href="10-Assign_taxonomy.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Environmental metabarcoding</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Metabarcoding.html"><a href="01-Metabarcoding.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Example_dataset.html"><a href="02-Example_dataset.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html"><i class="fa fa-check"></i><b>3</b> Introduction to DADA2 and R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#dada2"><i class="fa fa-check"></i><b>3.1</b> DADA2</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#otus-vs-asvs"><i class="fa fa-check"></i><b>3.1.1</b> OTUs vs ASVs</a></li>
<li class="chapter" data-level="3.1.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#workflow"><i class="fa fa-check"></i><b>3.1.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#r"><i class="fa fa-check"></i><b>3.2</b> R</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#the-console-window"><i class="fa fa-check"></i><b>3.2.1</b> The console window</a></li>
<li class="chapter" data-level="3.2.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#your-working-directory"><i class="fa fa-check"></i><b>3.2.2</b> Your working directory</a></li>
<li class="chapter" data-level="3.2.3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#packages"><i class="fa fa-check"></i><b>3.2.3</b> Packages</a></li>
<li class="chapter" data-level="3.2.4" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#functions"><i class="fa fa-check"></i><b>3.2.4</b> Functions</a></li>
<li class="chapter" data-level="3.2.5" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#assigning-values-to-variables"><i class="fa fa-check"></i><b>3.2.5</b> Assigning values to variables</a></li>
<li class="chapter" data-level="3.2.6" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#a-few-useful-commands-and-tips"><i class="fa fa-check"></i><b>3.2.6</b> A few useful commands and tips</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>DADA2</b></span></li>
<li class="chapter" data-level="5" data-path="05-Loading_data.html"><a href="05-Loading_data.html"><i class="fa fa-check"></i><b>5</b> Loading data into R and DADA2</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#jupyter"><i class="fa fa-check"></i><b>5.1</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.1.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#create-r-notebook"><i class="fa fa-check"></i><b>5.1.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.1.3" data-path="05-Loading_data.html"><a href="05-Loading_data.html#cells-and-code"><i class="fa fa-check"></i><b>5.1.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.1.4" data-path="05-Loading_data.html"><a href="05-Loading_data.html#create-new-cells"><i class="fa fa-check"></i><b>5.1.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.1.5" data-path="05-Loading_data.html"><a href="05-Loading_data.html#running-code"><i class="fa fa-check"></i><b>5.1.5</b> Running code</a></li>
<li class="chapter" data-level="5.1.6" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-file"><i class="fa fa-check"></i><b>5.1.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.1.7" data-path="05-Loading_data.html"><a href="05-Loading_data.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.1.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.1.8" data-path="05-Loading_data.html"><a href="05-Loading_data.html#close-the-notebook"><i class="fa fa-check"></i><b>5.1.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.1.9" data-path="05-Loading_data.html"><a href="05-Loading_data.html#video-tutorial"><i class="fa fa-check"></i><b>5.1.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#r-1"><i class="fa fa-check"></i><b>5.2</b> R</a></li>
<li class="chapter" data-level="5.3" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-dada2"><i class="fa fa-check"></i><b>5.3</b> Loading DADA2</a></li>
<li class="chapter" data-level="5.4" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-our-data"><i class="fa fa-check"></i><b>5.4</b> Loading our data</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-paths-for-our-input-and-output-directories"><i class="fa fa-check"></i><b>5.4.1</b> Saving the paths for our input and output directories</a></li>
<li class="chapter" data-level="5.4.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#inputting-the-forward-and-reverse-reads"><i class="fa fa-check"></i><b>5.4.2</b> Inputting the forward and reverse reads</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-primer_removal.html"><a href="06-primer_removal.html"><i class="fa fa-check"></i><b>6</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-primer_removal.html"><a href="06-primer_removal.html#primer-orientation-checking"><i class="fa fa-check"></i><b>6.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="6.2" data-path="06-primer_removal.html"><a href="06-primer_removal.html#cutadapt"><i class="fa fa-check"></i><b>6.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-quality_checks.html"><a href="07-quality_checks.html"><i class="fa fa-check"></i><b>7</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="8" data-path="08-clean.html"><a href="08-clean.html"><i class="fa fa-check"></i><b>8</b> Cleaning your data</a></li>
<li class="chapter" data-level="9" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html"><i class="fa fa-check"></i><b>9</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#generate-an-error-model"><i class="fa fa-check"></i><b>9.1</b> Generate an error model</a></li>
<li class="chapter" data-level="9.2" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dereplication"><i class="fa fa-check"></i><b>9.2</b> Dereplication</a></li>
<li class="chapter" data-level="9.3" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#inferrence-of-asvs"><i class="fa fa-check"></i><b>9.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="9.4" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#merging-paired-end-reads"><i class="fa fa-check"></i><b>9.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="9.5" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#making-our-asv-matrix"><i class="fa fa-check"></i><b>9.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="9.6" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#chimera-detection-and-removal"><i class="fa fa-check"></i><b>9.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="9.7" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#sequence-tracking-sanity-check"><i class="fa fa-check"></i><b>9.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Assign_taxonomy.html"><a href="10-Assign_taxonomy.html"><i class="fa fa-check"></i><b>10</b> Assigning taxonomy</a></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="11" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html"><i class="fa fa-check"></i><b>11</b> Further analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#r-packages-for-further-analysis"><i class="fa fa-check"></i><b>11.1</b> R packages for further analysis</a></li>
<li class="chapter" data-level="11.2" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#rarefaction-curves"><i class="fa fa-check"></i><b>11.2</b> Rarefaction curves</a></li>
<li class="chapter" data-level="11.3" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#alpha-diversity"><i class="fa fa-check"></i><b>11.3</b> Alpha diversity</a></li>
<li class="chapter" data-level="11.4" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#beta-diversity"><i class="fa fa-check"></i><b>11.4</b> Beta diversity</a></li>
</ul></li>
<li class="part"><span><b>Second marker</b></span></li>
<li class="chapter" data-level="12" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html"><i class="fa fa-check"></i><b>12</b> Analysing the second primer pair</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html#the-primer-sequences"><i class="fa fa-check"></i><b>12.1</b> The primer sequences</a></li>
<li class="chapter" data-level="12.2" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html#questions"><i class="fa fa-check"></i><b>12.2</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html"><i class="fa fa-check"></i><b>13</b> Guide to analysing the second primer pair</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#loading-the-data"><i class="fa fa-check"></i><b>13.1</b> Loading the data</a></li>
<li class="chapter" data-level="13.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#identifying-and-removing-primers"><i class="fa fa-check"></i><b>13.2</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#primer-orientation-checking-1"><i class="fa fa-check"></i><b>13.2.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="13.2.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#cutadapt-1"><i class="fa fa-check"></i><b>13.2.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#checking-the-quality-of-your-data"><i class="fa fa-check"></i><b>13.3</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="13.4" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#cleaning-your-data"><i class="fa fa-check"></i><b>13.4</b> Cleaning your data</a></li>
<li class="chapter" data-level="13.5" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#identification-of-asvs"><i class="fa fa-check"></i><b>13.5</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#generate-an-error-model-1"><i class="fa fa-check"></i><b>13.5.1</b> Generate an error model</a></li>
<li class="chapter" data-level="13.5.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#dereplication-1"><i class="fa fa-check"></i><b>13.5.2</b> Dereplication</a></li>
<li class="chapter" data-level="13.5.3" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#inferrence-of-asvs-1"><i class="fa fa-check"></i><b>13.5.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="13.5.4" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#merging-paired-end-reads-1"><i class="fa fa-check"></i><b>13.5.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="13.5.5" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#making-our-asv-matrix-1"><i class="fa fa-check"></i><b>13.5.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="13.5.6" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#chimera-detection-and-removal-1"><i class="fa fa-check"></i><b>13.5.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="13.5.7" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#sequence-tracking-sanity-check-1"><i class="fa fa-check"></i><b>13.5.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#assigning-taxonomy"><i class="fa fa-check"></i><b>13.6</b> Assigning taxonomy</a></li>
<li class="chapter" data-level="13.7" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#further-analysis"><i class="fa fa-check"></i><b>13.7</b> Further analysis</a></li>
<li class="chapter" data-level="13.8" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#rarefaction-curves-1"><i class="fa fa-check"></i><b>13.8</b> Rarefaction curves</a></li>
<li class="chapter" data-level="13.9" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#alpha-diversity-1"><i class="fa fa-check"></i><b>13.9</b> Alpha diversity</a></li>
<li class="chapter" data-level="13.10" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#beta-diversity-1"><i class="fa fa-check"></i><b>13.10</b> Beta diversity</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Metabarcoding for diet analysis and environmental DNA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="idasvs" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Identification of ASVs<a href="09-ASV_identification.html#idasvs" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="generate-an-error-model" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Generate an error model<a href="09-ASV_identification.html#generate-an-error-model" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First we need to model the error rates of our dataset using both the forward and reverse reads. Each dataset will have a specific error-signature with errors introduced by PCR amplification and sequencing.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="09-ASV_identification.html#cb37-1" aria-hidden="true" tabindex="-1"></a>errF <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtFs, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb37-2"><a href="09-ASV_identification.html#cb37-2" aria-hidden="true" tabindex="-1"></a>errR <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtRs, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>We can use the <code>plotErrors</code> function to check the estimated error rates.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="09-ASV_identification.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>You will see the
<code>Warning messages: 1: Transformation introduced infinite values in continuous y-axis  2: Transformation introduced infinite values in continuous y-axis</code></p>
<p>Do not worry about these warning messages. This is a message from the plotting function to let you know that there were some zero values in the data plotted (which turn into infinities on the log-scale). This is expected, it results from the fact that not every combination of error type (e.g. A-&gt;C) and quality score (e.g. 33) is observed in your data, which is normal.</p>
<p>Interpreting the plots.</p>
<p><img src="figures/errors.png" width="80%" style="display: block; margin: auto;" /></p>
<ul>
<li><p>The error rates for each possible transition (e.g. A→C, A→G) are shown</p></li>
<li><p>Red line - expected based on the quality score. (These are plotted when <code>nominalQ = TRUE</code> is included in the plot command)</p></li>
<li><p>Black line - estimate</p></li>
<li><p>Black dots - observed</p></li>
</ul>
<p>What we are expecting to see here is that the observed match up with estimates. Here we can see that the black dots track well with the black line. We can also see that the error rates drop with increasing quality score as we would expect. Sanity checks complete, we can proceed with the analysis.</p>
<ul>
<li>If you are worried about what your error plots look like when you fit your own dataset, one possible way to to improve the fit is to try increasing the number of bases the function is using (the default is 100 million).</li>
</ul>
</div>
<div id="dereplication" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Dereplication<a href="09-ASV_identification.html#dereplication" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The next step is to dereplicate identical reads. This is a common step in many workflows used for processing amplicons. This saves time and processing power as indentical reads are collapsed together. For example instead of processing 100 identical sequences, only one is processed but the original number (i.e. 100) is associated with it.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="09-ASV_identification.html#cb39-1" aria-hidden="true" tabindex="-1"></a>exists <span class="ot">&lt;-</span> <span class="fu">file.exists</span>(filtFs) </span>
<span id="cb39-2"><a href="09-ASV_identification.html#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="co"># check that all the samples are still present after filtering</span></span>
<span id="cb39-3"><a href="09-ASV_identification.html#cb39-3" aria-hidden="true" tabindex="-1"></a>derepFs <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtFs[exists], <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb39-4"><a href="09-ASV_identification.html#cb39-4" aria-hidden="true" tabindex="-1"></a>derepRs <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtRs[exists], <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb39-5"><a href="09-ASV_identification.html#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Name the derep-class objects by the sample names</span></span>
<span id="cb39-6"><a href="09-ASV_identification.html#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derepFs) <span class="ot">&lt;-</span> sample.names[exists]</span>
<span id="cb39-7"><a href="09-ASV_identification.html#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derepRs) <span class="ot">&lt;-</span> sample.names[exists]</span></code></pre></div>
</div>
<div id="inferrence-of-asvs" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Inferrence of ASVs<a href="09-ASV_identification.html#inferrence-of-asvs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now we are ready to infer the ASVs in our dataset. To do this DADA2 uses the error models created above to infer the true sample composition (follow <a href="https://www.nature.com/articles/nmeth.3869#methods">this link</a> for more details).</p>
<p>Here we will run the inference algorithm on single samples to save time but it is also possible to run samples together in pseudo-pools to increase the ability to identify ASVs of low abundance. Low abundance ASVs in a sample may be filtered out when run separately but if they are found in higher numbers in another sample then the chance of that ASV being real increases. Pseudo-pooling increases the chances of these “real” low abundance ASVs being retained within samples. Whether or not to use the pseudo-pooling option will depend on your dataset and experimental design (see <a href="https://benjjneb.github.io/dada2/pseudo.html#Pseudo-pooling" class="uri">https://benjjneb.github.io/dada2/pseudo.html#Pseudo-pooling</a> for more information).</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="09-ASV_identification.html#cb40-1" aria-hidden="true" tabindex="-1"></a>dadaFs <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepFs, <span class="at">err =</span> errF, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb40-2"><a href="09-ASV_identification.html#cb40-2" aria-hidden="true" tabindex="-1"></a>dadaRs <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepRs, <span class="at">err =</span> errR, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>It is important to note that you want to run the error and inference steps on datasets generated from a single Illumina run. Data generated from different runs can have different error structures.</p>
</div>
<div id="merging-paired-end-reads" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Merging paired end reads<a href="09-ASV_identification.html#merging-paired-end-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Up until now we have carried out all filtering, error correction and inference on the forward and reverse reads separately. It is now time to merge the two files. By default the minimum overlap allowed between the two samples is 12 bp.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="09-ASV_identification.html#cb41-1" aria-hidden="true" tabindex="-1"></a>mergers <span class="ot">&lt;-</span> <span class="fu">mergePairs</span>(dadaFs, derepFs, dadaRs, derepRs, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="making-our-asv-matrix" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Making our ASV matrix<a href="09-ASV_identification.html#making-our-asv-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now it is time to make the counts table. Each column represents a single ASV and each row is an individual sample.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="09-ASV_identification.html#cb42-1" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> <span class="fu">makeSequenceTable</span>(mergers)</span>
<span id="cb42-2"><a href="09-ASV_identification.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab) </span></code></pre></div>
<p>Q. How many ASVs are in our matrix?</p>
</div>
<div id="chimera-detection-and-removal" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> Chimera detection and removal<a href="09-ASV_identification.html#chimera-detection-and-removal" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The last step in generating our ASV matrix is to detect and remove any chimeric sequences.</p>
<p>Chimeric sequences are formed when two or more biological sequences joined together. This is fairly common in amplicon sequencing. DADA2 uses a method whereby it combines the left and right segments of abundant reads and compares these with lower abundant sequences. Any low abundant sequences that match are removed.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="09-ASV_identification.html#cb43-1" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">&lt;-</span> <span class="fu">removeBimeraDenovo</span>(seqtab, <span class="at">method=</span><span class="st">&quot;consensus&quot;</span>, </span>
<span id="cb43-2"><a href="09-ASV_identification.html#cb43-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb43-3"><a href="09-ASV_identification.html#cb43-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-4"><a href="09-ASV_identification.html#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab.nochim)</span></code></pre></div>
<p>Q. How many ASVs remain after filtering out chimeras?</p>
<p>Although this is a large proportion of sequence variants it should be a smaller proportion of the total sequences.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="09-ASV_identification.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(seqtab.nochim)<span class="sc">/</span><span class="fu">sum</span>(seqtab)</span></code></pre></div>
<p>In these data ~ 17% of the merged sequence reads were identified as chimeric.</p>
<p>Check the range of ASV lengths:</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="09-ASV_identification.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim)))</span></code></pre></div>
<p>Sequences that are much longer or shorter than the expected amplicon length could be the result of non-specific priming, and can be removed from your sequence table (e.g. <code>seqtab.len &lt;- seqtab[,nchar(colnames(seqtab)) %in% 150:180]</code>). This is analogous to “cutting a band” in-silico to get amplicons of the targeted length, e.g. the above command would remove amplicons between 150 bp and 180 bp. This is dependent on the amplicon and you should only remove these if you do not expect any length variation, or after further investigation of what these short or long sequences are. For this example we will proceed without any amplicon length filtering.</p>
</div>
<div id="sequence-tracking-sanity-check" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> Sequence tracking sanity check<a href="09-ASV_identification.html#sequence-tracking-sanity-check" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The last thing to do in this section is to track the number of sequences through the pipeline to check whether everything has run as expected and whether there are any steps where we loose a disproportionate number of sequences. If we end up with too few reads to run further analysis we can use this table to identify any step which might require further investigation and optimisation.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="09-ASV_identification.html#cb46-1" aria-hidden="true" tabindex="-1"></a>getN <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">sum</span>(<span class="fu">getUniques</span>(x))</span>
<span id="cb46-2"><a href="09-ASV_identification.html#cb46-2" aria-hidden="true" tabindex="-1"></a>track <span class="ot">&lt;-</span> <span class="fu">cbind</span>(out, <span class="fu">sapply</span>(dadaFs, getN), </span>
<span id="cb46-3"><a href="09-ASV_identification.html#cb46-3" aria-hidden="true" tabindex="-1"></a>               <span class="fu">sapply</span>(dadaRs, getN), </span>
<span id="cb46-4"><a href="09-ASV_identification.html#cb46-4" aria-hidden="true" tabindex="-1"></a>               <span class="fu">sapply</span>(mergers, getN), </span>
<span id="cb46-5"><a href="09-ASV_identification.html#cb46-5" aria-hidden="true" tabindex="-1"></a>               <span class="fu">rowSums</span>(seqtab.nochim))</span>
<span id="cb46-6"><a href="09-ASV_identification.html#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(track) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;input&quot;</span>, <span class="st">&quot;filtered&quot;</span>, </span>
<span id="cb46-7"><a href="09-ASV_identification.html#cb46-7" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;denoisedF&quot;</span>, <span class="st">&quot;denoisedR&quot;</span>, </span>
<span id="cb46-8"><a href="09-ASV_identification.html#cb46-8" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;merged&quot;</span>, <span class="st">&quot;nonchim&quot;</span>)</span>
<span id="cb46-9"><a href="09-ASV_identification.html#cb46-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(track) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb46-10"><a href="09-ASV_identification.html#cb46-10" aria-hidden="true" tabindex="-1"></a>track</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="08-clean.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="10-Assign_taxonomy.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
