<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Further analysis | Metabarcoding for diet analysis and environmental DNA</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Further analysis | Metabarcoding for diet analysis and environmental DNA" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Further analysis | Metabarcoding for diet analysis and environmental DNA" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher and Helen Hipperson" />


<meta name="date" content="2023-08-31" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="10-Assign_taxonomy.html"/>
<link rel="next" href="12-Analysing_another_marker.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Environmental metabarcoding</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Metabarcoding.html"><a href="01-Metabarcoding.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Example_dataset.html"><a href="02-Example_dataset.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html"><i class="fa fa-check"></i><b>3</b> Introduction to DADA2 and R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#dada2"><i class="fa fa-check"></i><b>3.1</b> DADA2</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#otus-vs-asvs"><i class="fa fa-check"></i><b>3.1.1</b> OTUs vs ASVs</a></li>
<li class="chapter" data-level="3.1.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#workflow"><i class="fa fa-check"></i><b>3.1.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#r"><i class="fa fa-check"></i><b>3.2</b> R</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#the-console-window"><i class="fa fa-check"></i><b>3.2.1</b> The console window</a></li>
<li class="chapter" data-level="3.2.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#your-working-directory"><i class="fa fa-check"></i><b>3.2.2</b> Your working directory</a></li>
<li class="chapter" data-level="3.2.3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#packages"><i class="fa fa-check"></i><b>3.2.3</b> Packages</a></li>
<li class="chapter" data-level="3.2.4" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#functions"><i class="fa fa-check"></i><b>3.2.4</b> Functions</a></li>
<li class="chapter" data-level="3.2.5" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#assigning-values-to-variables"><i class="fa fa-check"></i><b>3.2.5</b> Assigning values to variables</a></li>
<li class="chapter" data-level="3.2.6" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#a-few-useful-commands-and-tips"><i class="fa fa-check"></i><b>3.2.6</b> A few useful commands and tips</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>DADA2</b></span></li>
<li class="chapter" data-level="5" data-path="05-Loading_data.html"><a href="05-Loading_data.html"><i class="fa fa-check"></i><b>5</b> Loading data into R and DADA2</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#jupyter"><i class="fa fa-check"></i><b>5.1</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.1.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#create-r-notebook"><i class="fa fa-check"></i><b>5.1.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.1.3" data-path="05-Loading_data.html"><a href="05-Loading_data.html#cells-and-code"><i class="fa fa-check"></i><b>5.1.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.1.4" data-path="05-Loading_data.html"><a href="05-Loading_data.html#create-new-cells"><i class="fa fa-check"></i><b>5.1.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.1.5" data-path="05-Loading_data.html"><a href="05-Loading_data.html#running-code"><i class="fa fa-check"></i><b>5.1.5</b> Running code</a></li>
<li class="chapter" data-level="5.1.6" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-file"><i class="fa fa-check"></i><b>5.1.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.1.7" data-path="05-Loading_data.html"><a href="05-Loading_data.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.1.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.1.8" data-path="05-Loading_data.html"><a href="05-Loading_data.html#close-the-notebook"><i class="fa fa-check"></i><b>5.1.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.1.9" data-path="05-Loading_data.html"><a href="05-Loading_data.html#video-tutorial"><i class="fa fa-check"></i><b>5.1.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#r-1"><i class="fa fa-check"></i><b>5.2</b> R</a></li>
<li class="chapter" data-level="5.3" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-dada2"><i class="fa fa-check"></i><b>5.3</b> Loading DADA2</a></li>
<li class="chapter" data-level="5.4" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-our-data"><i class="fa fa-check"></i><b>5.4</b> Loading our data</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-paths-for-our-input-and-output-directories"><i class="fa fa-check"></i><b>5.4.1</b> Saving the paths for our input and output directories</a></li>
<li class="chapter" data-level="5.4.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#inputting-the-forward-and-reverse-reads"><i class="fa fa-check"></i><b>5.4.2</b> Inputting the forward and reverse reads</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-primer_removal.html"><a href="06-primer_removal.html"><i class="fa fa-check"></i><b>6</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-primer_removal.html"><a href="06-primer_removal.html#primer-orientation-checking"><i class="fa fa-check"></i><b>6.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="6.2" data-path="06-primer_removal.html"><a href="06-primer_removal.html#cutadapt"><i class="fa fa-check"></i><b>6.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-quality_checks.html"><a href="07-quality_checks.html"><i class="fa fa-check"></i><b>7</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="8" data-path="08-clean.html"><a href="08-clean.html"><i class="fa fa-check"></i><b>8</b> Cleaning your data</a></li>
<li class="chapter" data-level="9" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html"><i class="fa fa-check"></i><b>9</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#generate-an-error-model"><i class="fa fa-check"></i><b>9.1</b> Generate an error model</a></li>
<li class="chapter" data-level="9.2" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dereplication"><i class="fa fa-check"></i><b>9.2</b> Dereplication</a></li>
<li class="chapter" data-level="9.3" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#inferrence-of-asvs"><i class="fa fa-check"></i><b>9.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="9.4" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#merging-paired-end-reads"><i class="fa fa-check"></i><b>9.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="9.5" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#making-our-asv-matrix"><i class="fa fa-check"></i><b>9.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="9.6" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#chimera-detection-and-removal"><i class="fa fa-check"></i><b>9.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="9.7" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#sequence-tracking-sanity-check"><i class="fa fa-check"></i><b>9.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Assign_taxonomy.html"><a href="10-Assign_taxonomy.html"><i class="fa fa-check"></i><b>10</b> Assigning taxonomy</a></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="11" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html"><i class="fa fa-check"></i><b>11</b> Further analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#r-packages-for-further-analysis"><i class="fa fa-check"></i><b>11.1</b> R packages for further analysis</a></li>
<li class="chapter" data-level="11.2" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#rarefaction-curves"><i class="fa fa-check"></i><b>11.2</b> Rarefaction curves</a></li>
<li class="chapter" data-level="11.3" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#alpha-diversity"><i class="fa fa-check"></i><b>11.3</b> Alpha diversity</a></li>
<li class="chapter" data-level="11.4" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#beta-diversity"><i class="fa fa-check"></i><b>11.4</b> Beta diversity</a></li>
</ul></li>
<li class="part"><span><b>Second marker</b></span></li>
<li class="chapter" data-level="12" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html"><i class="fa fa-check"></i><b>12</b> Analysing the second primer pair</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html#the-primer-sequences"><i class="fa fa-check"></i><b>12.1</b> The primer sequences</a></li>
<li class="chapter" data-level="12.2" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html#questions"><i class="fa fa-check"></i><b>12.2</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html"><i class="fa fa-check"></i><b>13</b> Guide to analysing the second primer pair</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#loading-the-data"><i class="fa fa-check"></i><b>13.1</b> Loading the data</a></li>
<li class="chapter" data-level="13.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#identifying-and-removing-primers"><i class="fa fa-check"></i><b>13.2</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#primer-orientation-checking-1"><i class="fa fa-check"></i><b>13.2.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="13.2.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#cutadapt-1"><i class="fa fa-check"></i><b>13.2.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#checking-the-quality-of-your-data"><i class="fa fa-check"></i><b>13.3</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="13.4" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#cleaning-your-data"><i class="fa fa-check"></i><b>13.4</b> Cleaning your data</a></li>
<li class="chapter" data-level="13.5" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#identification-of-asvs"><i class="fa fa-check"></i><b>13.5</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#generate-an-error-model-1"><i class="fa fa-check"></i><b>13.5.1</b> Generate an error model</a></li>
<li class="chapter" data-level="13.5.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#dereplication-1"><i class="fa fa-check"></i><b>13.5.2</b> Dereplication</a></li>
<li class="chapter" data-level="13.5.3" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#inferrence-of-asvs-1"><i class="fa fa-check"></i><b>13.5.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="13.5.4" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#merging-paired-end-reads-1"><i class="fa fa-check"></i><b>13.5.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="13.5.5" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#making-our-asv-matrix-1"><i class="fa fa-check"></i><b>13.5.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="13.5.6" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#chimera-detection-and-removal-1"><i class="fa fa-check"></i><b>13.5.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="13.5.7" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#sequence-tracking-sanity-check-1"><i class="fa fa-check"></i><b>13.5.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#assigning-taxonomy"><i class="fa fa-check"></i><b>13.6</b> Assigning taxonomy</a></li>
<li class="chapter" data-level="13.7" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#further-analysis"><i class="fa fa-check"></i><b>13.7</b> Further analysis</a></li>
<li class="chapter" data-level="13.8" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#rarefaction-curves-1"><i class="fa fa-check"></i><b>13.8</b> Rarefaction curves</a></li>
<li class="chapter" data-level="13.9" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#alpha-diversity-1"><i class="fa fa-check"></i><b>13.9</b> Alpha diversity</a></li>
<li class="chapter" data-level="13.10" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#beta-diversity-1"><i class="fa fa-check"></i><b>13.10</b> Beta diversity</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Metabarcoding for diet analysis and environmental DNA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="furtheranalysis" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Further analysis<a href="11-Further_analysis.html#furtheranalysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this section we will briefly discuss a few basic types of multivariate data analysis and data visualisation which are often used in metabarcoding studies. It is important to remember that there are several different ways to plot and analyse your data and we have presented only a few examples here. It might be that for your own data set and questions it would be better to approach your analysis in a different way.</p>
<p>First we will load in a table which contains the meta data.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="11-Further_analysis.html#cb48-1" aria-hidden="true" tabindex="-1"></a>meta<span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&quot;/pub39/tea/nsc006/NEOF/metabarcoding_workshop/sample_info.csv&quot;</span>, </span>
<span id="cb48-2"><a href="11-Further_analysis.html#cb48-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">row.names =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>Before we start we will clean up the dataset. Our negative controls seem to be mostly clean of contaminants (with only 0 and 4 reads assigned to them). We will remove these for further analysis. Here we remove rows 25 and 31 which contain the filter (S62) and extraction (S72) control samples from our ASV table and metadata table.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="11-Further_analysis.html#cb49-1" aria-hidden="true" tabindex="-1"></a>seqtab.rmcontrol<span class="ot">&lt;-</span>seqtab.nochim[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">31</span>),]</span>
<span id="cb49-2"><a href="11-Further_analysis.html#cb49-2" aria-hidden="true" tabindex="-1"></a>meta.rmcontrol<span class="ot">&lt;-</span>meta[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">31</span>),]</span></code></pre></div>
<p>If you are concerned about contaminants in your own analysis then it could be useful to explore the R package <a href="https://benjjneb.github.io/decontam/vignettes/decontam_intro.html">decontam</a> which is designed to identify and filter contaminants.</p>
<p>We will also remove samples which appear to have failed/have very low sequence numbers in our dataset.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="11-Further_analysis.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Check number of sequences per sample</span></span>
<span id="cb50-2"><a href="11-Further_analysis.html#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rowSums</span>(seqtab.rmcontrol)</span>
<span id="cb50-3"><a href="11-Further_analysis.html#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="co"># S40, S41, S50 all have low sequence counts. So we will remove these rows.</span></span>
<span id="cb50-4"><a href="11-Further_analysis.html#cb50-4" aria-hidden="true" tabindex="-1"></a>seqtab.rmlow<span class="ot">&lt;-</span>seqtab.rmcontrol[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">13</span>,<span class="dv">14</span>,<span class="dv">23</span>),]</span>
<span id="cb50-5"><a href="11-Further_analysis.html#cb50-5" aria-hidden="true" tabindex="-1"></a>meta.rmlow<span class="ot">&lt;-</span>meta.rmcontrol[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">13</span>,<span class="dv">14</span>,<span class="dv">23</span>),]</span>
<span id="cb50-6"><a href="11-Further_analysis.html#cb50-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-7"><a href="11-Further_analysis.html#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the minimum sequence number in one sample.</span></span>
<span id="cb50-8"><a href="11-Further_analysis.html#cb50-8" aria-hidden="true" tabindex="-1"></a><span class="fu">min</span>(<span class="fu">rowSums</span>(seqtab.rmlow))</span>
<span id="cb50-9"><a href="11-Further_analysis.html#cb50-9" aria-hidden="true" tabindex="-1"></a><span class="co"># The lowest number of sequences in one sample is now 26667.</span></span></code></pre></div>
<div id="r-packages-for-further-analysis" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> R packages for further analysis<a href="11-Further_analysis.html#r-packages-for-further-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will use three more handy R packages in order to further explore our data, <a href="https://joey711.github.io/phyloseq/">phyloseq</a> and <a href="https://cran.r-project.org/web/packages/vegan/vegan.pdf">vegan</a> to explore the diverity in our data, plus the visualisation package ggplot2 for graphics. Load these libraries before proceeding with the steps below.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="11-Further_analysis.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb51-2"><a href="11-Further_analysis.html#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb51-3"><a href="11-Further_analysis.html#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
</div>
<div id="rarefaction-curves" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Rarefaction curves<a href="11-Further_analysis.html#rarefaction-curves" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The more deeply we sequence a sample the more species we discover, until this accumulation levels off as we have found all or as many ASVs/species as we are going to find. This becomes a problem as samples are often sequenced at different depths so will have reached different points in the curve. This is a common difficulty as pooling and sequencing equal amounts of each sample can be tricky.</p>
<p>One way to visualise this is to plot a rarefaction curve for each sample.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="11-Further_analysis.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rarecurve</span>(seqtab.rmlow, <span class="at">step=</span><span class="dv">100</span>, <span class="at">col=</span>meta.rmlow<span class="sc">$</span>COLOUR, <span class="at">lwd=</span><span class="dv">2</span>, <span class="at">ylab=</span><span class="st">&quot;ASVs&quot;</span>, <span class="at">label=</span>F)</span>
<span id="cb52-2"><a href="11-Further_analysis.html#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="co"># add a vertical line to represent the fewest sequences in any sample</span></span>
<span id="cb52-3"><a href="11-Further_analysis.html#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v=</span>(<span class="fu">min</span>(<span class="fu">rowSums</span>(seqtab.rmlow))))</span></code></pre></div>
<p><img src="figures/rarefaction_ed.png" width="80%" style="display: block; margin: auto;" /></p>
<p>You can see that each sample levels off at a different sequencing depth and each sample has been sequenced at different depths. This also gives us an indication of how many ASVs are in each sample, with the aquarium samples here (dark blue lines) amongst the least diverse, as we’d expect.</p>
<p>Rarefaction is one of the common approaches people use to correct for differences in library sequencing depth between samples and was originally proposed for traditional ecological studies (Sanders 1968). It involves randomly removing reads until you reach a number often equal to or less than the number of reads in the smallest sample. Random subsampling can result in a loss of data and generate artificial variation. Due to these concerns other methods of transformation have been suggested.</p>
<p>Other methods of normalisation include normalising counts as a proportion of the total library size (McMurdie and Holmes 2014), centered log-ratio transformations (Gloor et al. 2017), geometric mean pairwise ratios (Chen et al. 2018), and also variance stabilising transformations and relative log expressions (Badri et al. 2018).</p>
<p>Choosing the type of normalisation method to use for your own dataset is not trivial and the most appropriate method can vary between studies and datasets. Further reading is recommended. A couple of papers to get you started are listed below:</p>
<ul>
<li><a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003531">McMurdie and Holmes (2014)</a></li>
<li><a href="https://www.frontiersin.org/articles/10.3389/fmicb.2019.02407/full">Willis (2019)</a></li>
<li><a href="https://www.biorxiv.org/content/10.1101/2020.09.09.290049v1.full">Cameron <em>et al.</em> (preprint)</a></li>
</ul>
</div>
<div id="alpha-diversity" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> Alpha diversity<a href="11-Further_analysis.html#alpha-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Measures of alpha diversity are used to describe diversity within a sample.</p>
<p>We will use the R package <a href="https://joey711.github.io/phyloseq/">phyloseq</a> to plot alpha diversity. For this example we will proceed with unnormalised data. It is generally recommended not to normalise count data before calculating alpha diversity measures in the <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-FAQ.html#should-i-normalize-my-data-before-alpha-diversity-analysis">phyloseq FAQ</a>.</p>
<p>First make a phyloseq object. To do this we first read in our ASV, taxonomy and metadata tables before making the plyloseq object <code>phylo</code>.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="11-Further_analysis.html#cb53-1" aria-hidden="true" tabindex="-1"></a>seqtab.rmlow2<span class="ot">&lt;-</span><span class="fu">t</span>(<span class="fu">as.data.frame</span>(seqtab.rmlow))</span>
<span id="cb53-2"><a href="11-Further_analysis.html#cb53-2" aria-hidden="true" tabindex="-1"></a>phylo_asv <span class="ot">&lt;-</span> <span class="fu">otu_table</span>(seqtab.rmlow2, <span class="at">taxa_are_rows=</span><span class="cn">TRUE</span>)</span>
<span id="cb53-3"><a href="11-Further_analysis.html#cb53-3" aria-hidden="true" tabindex="-1"></a>phylo_tax <span class="ot">&lt;-</span> <span class="fu">tax_table</span>(taxa)</span>
<span id="cb53-4"><a href="11-Further_analysis.html#cb53-4" aria-hidden="true" tabindex="-1"></a>phylo_samples <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(meta.rmlow)</span>
<span id="cb53-5"><a href="11-Further_analysis.html#cb53-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-6"><a href="11-Further_analysis.html#cb53-6" aria-hidden="true" tabindex="-1"></a>phylo <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(phylo_asv, phylo_tax, phylo_samples)</span>
<span id="cb53-7"><a href="11-Further_analysis.html#cb53-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-8"><a href="11-Further_analysis.html#cb53-8" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_names</span>(phylo)</span>
<span id="cb53-9"><a href="11-Further_analysis.html#cb53-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rank_names</span>(phylo)</span>
<span id="cb53-10"><a href="11-Further_analysis.html#cb53-10" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_variables</span>(phylo) </span></code></pre></div>
<p>The two alpha diversity metrics we will plot are:</p>
<p><strong>1. Shannon’s diversity index</strong></p>
<p>A measure of diversity where a higher number means higher diversity. Shannon’s index accounts for the abundance and evenness of the features present. If richness and evenness increase the diversity score increases. Values can range from one (in case of a single dominant species) to the total number of all species (in case of all species having equal abundance).</p>
<p><strong>Equation</strong>:</p>
<p><span class="math display">\[
H =
-\sum_{i=1}^{S} p_i lnp_i
\]</span></p>
<ul>
<li>H = Shannon diversity index</li>
<li>pi = is the proportion of species i</li>
<li>S = Number of species</li>
</ul>
<p><strong>2. Simpson diversity index</strong>
A measure of diversity based on number of features present and the relative abundance of each feature. If richness and evenness increase the diversity score increases. The values range from 0 (no diversity) to 1 (infinite diversity).</p>
<p><strong>Equation</strong>:
<span class="math display">\[
D = 1 -
{\sum_{i=1}^{S} {p_i} ^{2}}
\]</span></p>
<ul>
<li>D = Simpson diversity index</li>
<li>pi = is the proportion of species i</li>
<li>S = Number of species</li>
</ul>
<p>Plot the two methods of calculating alpha diversity.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="11-Further_analysis.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_richness</span>(phylo, </span>
<span id="cb54-2"><a href="11-Further_analysis.html#cb54-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">measures=</span><span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>), </span>
<span id="cb54-3"><a href="11-Further_analysis.html#cb54-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">color =</span> <span class="st">&quot;SITE&quot;</span>)</span>
<span id="cb54-4"><a href="11-Further_analysis.html#cb54-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_richness</span>(phylo, <span class="at">x=</span><span class="st">&quot;SITE&quot;</span>, <span class="at">measures=</span><span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>), </span>
<span id="cb54-5"><a href="11-Further_analysis.html#cb54-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">color =</span> <span class="st">&quot;SITE&quot;</span>) <span class="sc">+</span> <span class="fu">geom_boxplot</span>()</span></code></pre></div>
<p><img src="figures/alpha1.png" width="80%" style="display: block; margin: auto;" /></p>
<p><img src="figures/alpha2.png" width="80%" style="display: block; margin: auto;" /></p>
<p>The first figure plots the diversity measure per sample and colours the output by site.
The second figure combines the replicates to plot as a boxplot.</p>
</div>
<div id="beta-diversity" class="section level2 hasAnchor" number="11.4">
<h2><span class="header-section-number">11.4</span> Beta diversity<a href="11-Further_analysis.html#beta-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Beta diversity compares the difference in diversity between two sites, or to put it another way it calculates the number of species that are not the same in the two sites.</p>
<p>We will normalise the data before running the beta diversity calculation. We will transform the data into proportions to be used for Bray-Curtis distances.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="11-Further_analysis.html#cb55-1" aria-hidden="true" tabindex="-1"></a>ps.prop <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(phylo, <span class="cf">function</span>(otu) otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span></code></pre></div>
<p>We then generate and plot the NMDS (Non-metric MultiDimenstional Scaling) using Bray-Curtis distances.</p>
<p><strong>Equation</strong>:
<span class="math display">\[
d_jk =
\sum\frac{|x_{ij}-x_{ik}|} {(x_{ij}+x_{ik})}
\]</span></p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="11-Further_analysis.html#cb56-1" aria-hidden="true" tabindex="-1"></a>ord.nmds.bray <span class="ot">&lt;-</span> <span class="fu">ordinate</span>(ps.prop, <span class="at">method=</span><span class="st">&quot;NMDS&quot;</span>, <span class="at">distance=</span><span class="st">&quot;bray&quot;</span>)</span>
<span id="cb56-2"><a href="11-Further_analysis.html#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ordination</span>(ps.prop, ord.nmds.bray, <span class="at">color=</span><span class="st">&quot;SITE&quot;</span>, <span class="at">title=</span><span class="st">&quot;Bray NMDS&quot;</span>)</span></code></pre></div>
<p><img src="figures/beta.png" width="80%" style="display: block; margin: auto;" /></p>
<p>In general the samples do seem to cluster roughly by site in the NMDS plot.</p>
<p>We will then calculate the Bray–Curtis distances using the <code>distance</code> function and perform a PERMANOVA (permutational multivariate analysis of variance) using the <code>adonis</code> function from Vegan to check whether the separation of samples by site is significantly different.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="11-Further_analysis.html#cb57-1" aria-hidden="true" tabindex="-1"></a>bray.dist<span class="ot">&lt;-</span><span class="fu">distance</span>(ps.prop, <span class="at">method=</span><span class="st">&quot;bray&quot;</span>)</span>
<span id="cb57-2"><a href="11-Further_analysis.html#cb57-2" aria-hidden="true" tabindex="-1"></a>sampledf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(phylo))</span>
<span id="cb57-3"><a href="11-Further_analysis.html#cb57-3" aria-hidden="true" tabindex="-1"></a><span class="fu">adonis</span>(bray.dist <span class="sc">~</span> SITE, <span class="at">data =</span> sampledf)</span></code></pre></div>
<p>The PERMANOVA results suggest that there is a statistical difference in communities between sites.</p>
<p>Lastly, let’s plot the proportion of ASV sequences within each sample that belong to different taxonomic families.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="11-Further_analysis.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_bar</span>(ps.prop, <span class="at">fill =</span> <span class="st">&quot;Family&quot;</span>)<span class="sc">+</span></span>
<span id="cb58-2"><a href="11-Further_analysis.html#cb58-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">color=</span>Family, <span class="at">fill=</span>Family), <span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">position=</span><span class="st">&quot;stack&quot;</span>)<span class="sc">+</span></span>
<span id="cb58-3"><a href="11-Further_analysis.html#cb58-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>SITE, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>)</span></code></pre></div>
<p><img src="figures/Family_abundance_MiFish.png" width="80%" style="display: block; margin: auto;" /></p>
<p>We can see that the five aquarium samples contain mostly Cichlidae sequences, apart from a very small number of unassigned (NA) and misassigned (there were only cichlid fish in the aquarium) sequences. Site 11 has more than 50% unassigned sequences in all three replicate samples. All of the samples containing &gt; 10% Procatopodidae sequences are grouped together in the NMDS plot (especially sites 19 &amp; 20, but also two samples from site 2, and the one sample from site 18), suggesting this may be in part driving the observed patterns of community similarity.</p>

</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="10-Assign_taxonomy.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="12-Analysing_another_marker.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
