<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 Analysing the second primer pair | Metabarcoding for diet analysis and environmental DNA</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 Analysing the second primer pair | Metabarcoding for diet analysis and environmental DNA" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 Analysing the second primer pair | Metabarcoding for diet analysis and environmental DNA" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher and Helen Hipperson" />


<meta name="date" content="2023-08-31" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="11-Further_analysis.html"/>
<link rel="next" href="13-Others_marker_steps.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Environmental metabarcoding</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Metabarcoding.html"><a href="01-Metabarcoding.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Example_dataset.html"><a href="02-Example_dataset.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html"><i class="fa fa-check"></i><b>3</b> Introduction to DADA2 and R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#dada2"><i class="fa fa-check"></i><b>3.1</b> DADA2</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#otus-vs-asvs"><i class="fa fa-check"></i><b>3.1.1</b> OTUs vs ASVs</a></li>
<li class="chapter" data-level="3.1.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#workflow"><i class="fa fa-check"></i><b>3.1.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#r"><i class="fa fa-check"></i><b>3.2</b> R</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#the-console-window"><i class="fa fa-check"></i><b>3.2.1</b> The console window</a></li>
<li class="chapter" data-level="3.2.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#your-working-directory"><i class="fa fa-check"></i><b>3.2.2</b> Your working directory</a></li>
<li class="chapter" data-level="3.2.3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#packages"><i class="fa fa-check"></i><b>3.2.3</b> Packages</a></li>
<li class="chapter" data-level="3.2.4" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#functions"><i class="fa fa-check"></i><b>3.2.4</b> Functions</a></li>
<li class="chapter" data-level="3.2.5" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#assigning-values-to-variables"><i class="fa fa-check"></i><b>3.2.5</b> Assigning values to variables</a></li>
<li class="chapter" data-level="3.2.6" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#a-few-useful-commands-and-tips"><i class="fa fa-check"></i><b>3.2.6</b> A few useful commands and tips</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>DADA2</b></span></li>
<li class="chapter" data-level="5" data-path="05-Loading_data.html"><a href="05-Loading_data.html"><i class="fa fa-check"></i><b>5</b> Loading data into R and DADA2</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#jupyter"><i class="fa fa-check"></i><b>5.1</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.1.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#create-r-notebook"><i class="fa fa-check"></i><b>5.1.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.1.3" data-path="05-Loading_data.html"><a href="05-Loading_data.html#cells-and-code"><i class="fa fa-check"></i><b>5.1.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.1.4" data-path="05-Loading_data.html"><a href="05-Loading_data.html#create-new-cells"><i class="fa fa-check"></i><b>5.1.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.1.5" data-path="05-Loading_data.html"><a href="05-Loading_data.html#running-code"><i class="fa fa-check"></i><b>5.1.5</b> Running code</a></li>
<li class="chapter" data-level="5.1.6" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-file"><i class="fa fa-check"></i><b>5.1.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.1.7" data-path="05-Loading_data.html"><a href="05-Loading_data.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.1.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.1.8" data-path="05-Loading_data.html"><a href="05-Loading_data.html#close-the-notebook"><i class="fa fa-check"></i><b>5.1.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.1.9" data-path="05-Loading_data.html"><a href="05-Loading_data.html#video-tutorial"><i class="fa fa-check"></i><b>5.1.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#r-1"><i class="fa fa-check"></i><b>5.2</b> R</a></li>
<li class="chapter" data-level="5.3" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-dada2"><i class="fa fa-check"></i><b>5.3</b> Loading DADA2</a></li>
<li class="chapter" data-level="5.4" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-our-data"><i class="fa fa-check"></i><b>5.4</b> Loading our data</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-paths-for-our-input-and-output-directories"><i class="fa fa-check"></i><b>5.4.1</b> Saving the paths for our input and output directories</a></li>
<li class="chapter" data-level="5.4.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#inputting-the-forward-and-reverse-reads"><i class="fa fa-check"></i><b>5.4.2</b> Inputting the forward and reverse reads</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-primer_removal.html"><a href="06-primer_removal.html"><i class="fa fa-check"></i><b>6</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-primer_removal.html"><a href="06-primer_removal.html#primer-orientation-checking"><i class="fa fa-check"></i><b>6.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="6.2" data-path="06-primer_removal.html"><a href="06-primer_removal.html#cutadapt"><i class="fa fa-check"></i><b>6.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-quality_checks.html"><a href="07-quality_checks.html"><i class="fa fa-check"></i><b>7</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="8" data-path="08-clean.html"><a href="08-clean.html"><i class="fa fa-check"></i><b>8</b> Cleaning your data</a></li>
<li class="chapter" data-level="9" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html"><i class="fa fa-check"></i><b>9</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#generate-an-error-model"><i class="fa fa-check"></i><b>9.1</b> Generate an error model</a></li>
<li class="chapter" data-level="9.2" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dereplication"><i class="fa fa-check"></i><b>9.2</b> Dereplication</a></li>
<li class="chapter" data-level="9.3" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#inferrence-of-asvs"><i class="fa fa-check"></i><b>9.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="9.4" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#merging-paired-end-reads"><i class="fa fa-check"></i><b>9.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="9.5" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#making-our-asv-matrix"><i class="fa fa-check"></i><b>9.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="9.6" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#chimera-detection-and-removal"><i class="fa fa-check"></i><b>9.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="9.7" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#sequence-tracking-sanity-check"><i class="fa fa-check"></i><b>9.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Assign_taxonomy.html"><a href="10-Assign_taxonomy.html"><i class="fa fa-check"></i><b>10</b> Assigning taxonomy</a></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="11" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html"><i class="fa fa-check"></i><b>11</b> Further analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#r-packages-for-further-analysis"><i class="fa fa-check"></i><b>11.1</b> R packages for further analysis</a></li>
<li class="chapter" data-level="11.2" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#rarefaction-curves"><i class="fa fa-check"></i><b>11.2</b> Rarefaction curves</a></li>
<li class="chapter" data-level="11.3" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#alpha-diversity"><i class="fa fa-check"></i><b>11.3</b> Alpha diversity</a></li>
<li class="chapter" data-level="11.4" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#beta-diversity"><i class="fa fa-check"></i><b>11.4</b> Beta diversity</a></li>
</ul></li>
<li class="part"><span><b>Second marker</b></span></li>
<li class="chapter" data-level="12" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html"><i class="fa fa-check"></i><b>12</b> Analysing the second primer pair</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html#the-primer-sequences"><i class="fa fa-check"></i><b>12.1</b> The primer sequences</a></li>
<li class="chapter" data-level="12.2" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html#questions"><i class="fa fa-check"></i><b>12.2</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html"><i class="fa fa-check"></i><b>13</b> Guide to analysing the second primer pair</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#loading-the-data"><i class="fa fa-check"></i><b>13.1</b> Loading the data</a></li>
<li class="chapter" data-level="13.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#identifying-and-removing-primers"><i class="fa fa-check"></i><b>13.2</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#primer-orientation-checking-1"><i class="fa fa-check"></i><b>13.2.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="13.2.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#cutadapt-1"><i class="fa fa-check"></i><b>13.2.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#checking-the-quality-of-your-data"><i class="fa fa-check"></i><b>13.3</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="13.4" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#cleaning-your-data"><i class="fa fa-check"></i><b>13.4</b> Cleaning your data</a></li>
<li class="chapter" data-level="13.5" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#identification-of-asvs"><i class="fa fa-check"></i><b>13.5</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#generate-an-error-model-1"><i class="fa fa-check"></i><b>13.5.1</b> Generate an error model</a></li>
<li class="chapter" data-level="13.5.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#dereplication-1"><i class="fa fa-check"></i><b>13.5.2</b> Dereplication</a></li>
<li class="chapter" data-level="13.5.3" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#inferrence-of-asvs-1"><i class="fa fa-check"></i><b>13.5.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="13.5.4" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#merging-paired-end-reads-1"><i class="fa fa-check"></i><b>13.5.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="13.5.5" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#making-our-asv-matrix-1"><i class="fa fa-check"></i><b>13.5.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="13.5.6" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#chimera-detection-and-removal-1"><i class="fa fa-check"></i><b>13.5.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="13.5.7" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#sequence-tracking-sanity-check-1"><i class="fa fa-check"></i><b>13.5.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#assigning-taxonomy"><i class="fa fa-check"></i><b>13.6</b> Assigning taxonomy</a></li>
<li class="chapter" data-level="13.7" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#further-analysis"><i class="fa fa-check"></i><b>13.7</b> Further analysis</a></li>
<li class="chapter" data-level="13.8" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#rarefaction-curves-1"><i class="fa fa-check"></i><b>13.8</b> Rarefaction curves</a></li>
<li class="chapter" data-level="13.9" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#alpha-diversity-1"><i class="fa fa-check"></i><b>13.9</b> Alpha diversity</a></li>
<li class="chapter" data-level="13.10" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#beta-diversity-1"><i class="fa fa-check"></i><b>13.10</b> Beta diversity</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Metabarcoding for diet analysis and environmental DNA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="secondmarker" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">Chapter 12</span> Analysing the second primer pair<a href="12-Analysing_another_marker.html#secondmarker" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The sequencing data files we have analysed for the marker MiFish-U also contain sequences for a second marker, 12S-V5.</p>
<p>Both of these primer pairs amplify up short regions of the 12S gene. As shown below, the 12S-V5 region neighbours that amplified by the MiFish-U primers, and has a shorter expected length.</p>
<p><img src="figures/12S.png" width="80%" style="display: block; margin: auto;" /></p>
<p>It can be common to use multiple genes or multiple markers of the same gene in metabarcoding studies. Depending on the taxa that you are targeting, multiple primer pairs can complement each other by increasing taxonomic coverage or resolution.</p>
<div id="the-primer-sequences" class="section level2 hasAnchor" number="12.1">
<h2><span class="header-section-number">12.1</span> The primer sequences<a href="12-Analysing_another_marker.html#the-primer-sequences" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Below are the forward and reverse primer sequences for 12S-V5.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="12-Analysing_another_marker.html#cb59-1" aria-hidden="true" tabindex="-1"></a>FWD2 <span class="ot">&lt;-</span> <span class="st">&quot;ACTGGGATTAGATACCCC&quot;</span></span>
<span id="cb59-2"><a href="12-Analysing_another_marker.html#cb59-2" aria-hidden="true" tabindex="-1"></a>REV2 <span class="ot">&lt;-</span> <span class="st">&quot;TAGAACAGGCTCCTCTAG&quot;</span></span></code></pre></div>
<p>Here are the primer sequences again, but formatted so they are a little easier to read. When you type them into R though you should do this with no spaces, as above.</p>
<p>12S-V5 FWD: ACT GGG ATT AGA TAC CCC</p>
<p>12S-V5 REV: TAG AAC AGG CTC CTC TAG</p>
<p>There is also a fasta reference database file for 12S-V5, which is here:
<code>/pub39/tea/nsc006/NEOF/metabarcoding_workshop/taxonomy/12S-V5_Reference_Database_taxonomy.fasta</code></p>
<p>Run through the pipeline for the 12S-V5 primer pair, starting from identifying and removing the primers and through to the further diversity analyses.</p>
<p>Start by making a new notebook called <strong>“DADA2_analysis_pipeline_12S-V5_primer_set”</strong>.</p>
<p>Try to work through this on your own using the commands run for the MiFish-U primers as a guide. You will be able to copy and paste commands from your MiFish-U notebook. However, if you do get stuck with any of the code then you can look at the next section for help.</p>
<p>The questions below will hopefully help guide your interpretation of the different processing steps and output. Answers/discussion can also be found in the next section under each of the relevant headings.</p>
</div>
<div id="questions" class="section level2 hasAnchor" number="12.2">
<h2><span class="header-section-number">12.2</span> Questions<a href="12-Analysing_another_marker.html#questions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Primer orientation checking</strong></p>
<ul>
<li>Do we see the same pattern of the orientation of the 12S-V5 primers in the forward and reverse reads as we did for the MiFish-U primers? If not, why do you think this is?</li>
</ul>
<p><strong>Checking the data quality</strong></p>
<ul>
<li>Is the sequence data of similar quality to the MiFish reads? Are there any differences you can notice between the quality plots for the MiFish-U and 12S-V5 sequences?</li>
</ul>
<p><strong>Cleaning the data</strong></p>
<ul>
<li>How many of the samples have fewer than 1000 clean reads remaining after quality trimming? Are these the same samples with low sequence numbers as those for the MiFish primers?</li>
</ul>
<p><strong>Making the ASV matrix and chimera removal</strong></p>
<ul>
<li>How many ASVs are identified for the 12S-V5 primer data, and how many non-chimeric ASVs? Does this number differ from the number for the MiFish data, and if so why do you think this could be?</li>
</ul>
<p><strong>Sequence tracking sanity check</strong></p>
<ul>
<li>How many samples have low sequence numbers, and does this include our two negative samples (S62 and S72)?</li>
</ul>
<p><strong>Assigning taxonomy</strong></p>
<ul>
<li>How many of the top six ASVs have species-level taxonomic assignments?</li>
</ul>
<p><strong>Alpha diversity</strong></p>
<ul>
<li>How do the measures of alpha diversity compare between MiFish-U and 12S-V5, in terms of level of diversity and variability within sampling site?</li>
<li>Does our filter negative control sample have low or high diversity compared to the actual samples?</li>
</ul>
<p><strong>Beta diversity</strong></p>
<ul>
<li>How does the proportion of taxonomically unassigned reads compare between the 12S-V5 and MiFish ASVs?</li>
<li>Does the negative filter control sample resemble any of the other samples?</li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="11-Further_analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="13-Others_marker_steps.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
