<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Let’s now analyse the second 12S primer pair | Metabarcoding for diet analysis and environmental DNA</title>
  <meta name="description" content="Chapter 13 Let’s now analyse the second 12S primer pair | Metabarcoding for diet analysis and environmental DNA" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Let’s now analyse the second 12S primer pair | Metabarcoding for diet analysis and environmental DNA" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Let’s now analyse the second 12S primer pair | Metabarcoding for diet analysis and environmental DNA" />
  
  
  

<meta name="author" content="Katy Maher and Helen Hipperson" />


<meta name="date" content="2021-05-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="11-Further_analysis.html"/>

<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a>Environmental metabarcoding</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Metabarcoding.html"><a href="01-Metabarcoding.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Example_dataset.html"><a href="02-Example_dataset.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html"><i class="fa fa-check"></i><b>3</b> Introduction to DADA2 and R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#dada2"><i class="fa fa-check"></i><b>3.1</b> DADA2</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#otus-vs-asvs"><i class="fa fa-check"></i><b>3.1.1</b> OTUs vs ASVs</a></li>
<li class="chapter" data-level="3.1.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#workflow"><i class="fa fa-check"></i><b>3.1.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#r"><i class="fa fa-check"></i><b>3.2</b> R</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#the-console-window"><i class="fa fa-check"></i><b>3.2.1</b> The console window</a></li>
<li class="chapter" data-level="3.2.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#your-working-directory"><i class="fa fa-check"></i><b>3.2.2</b> Your working directory</a></li>
<li class="chapter" data-level="3.2.3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#packages"><i class="fa fa-check"></i><b>3.2.3</b> Packages</a></li>
<li class="chapter" data-level="3.2.4" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#functions"><i class="fa fa-check"></i><b>3.2.4</b> Functions</a></li>
<li class="chapter" data-level="3.2.5" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#assigning-values-to-variables"><i class="fa fa-check"></i><b>3.2.5</b> Assigning values to variables</a></li>
<li class="chapter" data-level="3.2.6" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#a-few-useful-commands-and-tips"><i class="fa fa-check"></i><b>3.2.6</b> A few useful commands and tips</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Loading_data.html"><a href="05-Loading_data.html"><i class="fa fa-check"></i><b>5</b> Loading data into R and DADA2</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-r"><i class="fa fa-check"></i><b>5.1</b> Loading R</a></li>
<li class="chapter" data-level="5.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-dada2"><i class="fa fa-check"></i><b>5.2</b> Loading DADA2</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-our-data"><i class="fa fa-check"></i><b>6</b> Loading our data</a>
<ul>
<li class="chapter" data-level="6.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-paths-for-our-input-and-output-directories"><i class="fa fa-check"></i><b>6.1</b> Saving the paths for our input and output directories</a></li>
<li class="chapter" data-level="6.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#inputting-the-forward-and-reverse-reads"><i class="fa fa-check"></i><b>6.2</b> Inputting the forward and reverse reads</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="06-primer_removal.html"><a href="06-primer_removal.html"><i class="fa fa-check"></i><b>7</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="7.1" data-path="06-primer_removal.html"><a href="06-primer_removal.html#primer-orientation-checking"><i class="fa fa-check"></i><b>7.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="7.2" data-path="06-primer_removal.html"><a href="06-primer_removal.html#cutadapt"><i class="fa fa-check"></i><b>7.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="07-quality_checks.html"><a href="07-quality_checks.html"><i class="fa fa-check"></i><b>8</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="9" data-path="08-clean.html"><a href="08-clean.html"><i class="fa fa-check"></i><b>9</b> Cleaning your data</a></li>
<li class="chapter" data-level="10" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html"><i class="fa fa-check"></i><b>10</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="10.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#generate-an-error-model"><i class="fa fa-check"></i><b>10.1</b> Generate an error model</a></li>
<li class="chapter" data-level="10.2" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dereplication"><i class="fa fa-check"></i><b>10.2</b> Dereplication</a></li>
<li class="chapter" data-level="10.3" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#inferrence-of-asvs"><i class="fa fa-check"></i><b>10.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="10.4" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#merging-paired-end-reads"><i class="fa fa-check"></i><b>10.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="10.5" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#making-our-asv-matrix"><i class="fa fa-check"></i><b>10.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="10.6" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#chimera-detection-and-removal"><i class="fa fa-check"></i><b>10.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="10.7" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#sequence-tracking-sanity-check"><i class="fa fa-check"></i><b>10.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="10-Assign_taxonomy.html"><a href="10-Assign_taxonomy.html"><i class="fa fa-check"></i><b>11</b> Assigning taxonomy</a></li>
<li class="chapter" data-level="12" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html"><i class="fa fa-check"></i><b>12</b> Further analysis</a>
<ul>
<li class="chapter" data-level="12.1" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#r-packages-for-further-analysis"><i class="fa fa-check"></i><b>12.1</b> R packages for further analysis</a></li>
<li class="chapter" data-level="12.2" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#rarefaction-curves"><i class="fa fa-check"></i><b>12.2</b> Rarefaction curves</a></li>
<li class="chapter" data-level="12.3" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#alpha-diversity"><i class="fa fa-check"></i><b>12.3</b> Alpha diversity</a></li>
<li class="chapter" data-level="12.4" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#beta-diversity"><i class="fa fa-check"></i><b>12.4</b> Beta diversity</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html"><i class="fa fa-check"></i><b>13</b> Let’s now analyse the second 12S primer pair</a>
<ul>
<li class="chapter" data-level="13.1" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#identifying-and-removing-primers-1"><i class="fa fa-check"></i><b>13.1</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#primer-orientation-checking-1"><i class="fa fa-check"></i><b>13.1.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="13.1.2" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#cutadapt-1"><i class="fa fa-check"></i><b>13.1.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#checking-the-quality-of-your-data-1"><i class="fa fa-check"></i><b>13.2</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="13.3" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#cleaning-your-data-1"><i class="fa fa-check"></i><b>13.3</b> Cleaning your data</a></li>
<li class="chapter" data-level="13.4" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#identification-of-asvs-1"><i class="fa fa-check"></i><b>13.4</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#generate-an-error-model-1"><i class="fa fa-check"></i><b>13.4.1</b> Generate an error model</a></li>
<li class="chapter" data-level="13.4.2" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#dereplication-1"><i class="fa fa-check"></i><b>13.4.2</b> Dereplication</a></li>
<li class="chapter" data-level="13.4.3" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#inferrence-of-asvs-1"><i class="fa fa-check"></i><b>13.4.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="13.4.4" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#merging-paired-end-reads-1"><i class="fa fa-check"></i><b>13.4.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="13.4.5" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#making-our-asv-matrix-1"><i class="fa fa-check"></i><b>13.4.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="13.4.6" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#chimera-detection-and-removal-1"><i class="fa fa-check"></i><b>13.4.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="13.4.7" data-path="12-Others_12S_primer_pair_steps.html"><a href="12-Others_12S_primer_pair_steps.html#sequence-tracking-sanity-check-1"><i class="fa fa-check"></i><b>13.4.7</b> Sequence tracking sanity check</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Metabarcoding for diet analysis and environmental DNA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="lets-now-analyse-the-second-12s-primer-pair" class="section level1" number="13">
<h1><span class="header-section-number">Chapter 13</span> Let’s now analyse the second 12S primer pair</h1>
<p>The two primer pairs were multiplexed together for each sample, so our input path and lists of file names and locations (fnFs and fnRs) remain the same.</p>
<div id="identifying-and-removing-primers-1" class="section level2" number="13.1">
<h2><span class="header-section-number">13.1</span> Identifying and removing primers</h2>
<p>Now you will carry out primer removal. DADA2 requires the primers you used to be trimmed off the forward and reverse of your reads. We will use the software cutadapt for this from within R.</p>
<p>Read in your forward and reverse primer sequences for the 12S-V5 primer set. The 12S-V5 primers do not contain any degenerate bases but cutadapt is able to handle any degenerate bases you may have in your own primer sequences.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="12-Others_12S_primer_pair_steps.html#cb54-1" aria-hidden="true" tabindex="-1"></a>FWD2 <span class="ot">&lt;-</span> <span class="st">&quot;ACTGGGATTAGATACCCC&quot;</span></span>
<span id="cb54-2"><a href="12-Others_12S_primer_pair_steps.html#cb54-2" aria-hidden="true" tabindex="-1"></a>REV2 <span class="ot">&lt;-</span> <span class="st">&quot;TAGAACAGGCTCCTCTAG&quot;</span></span></code></pre></div>
<div id="primer-orientation-checking-1" class="section level3" number="13.1.1">
<h3><span class="header-section-number">13.1.1</span> Primer orientation checking</h3>
<p>It might be useful to check the orientation of the primers on our sequences so we know the best way to remove them. We know how our libraries were created and sequenced so we are expecting that the forward primer should be located in a forward orientation in the forward read and the reverse primer should be located in the forward orientation in the reverse read.</p>
<p>For your own data you might be less certain as to which orientation your primers are in your reads.</p>
<p>We use the <code>allOrients</code> function again to make a list of all orientations of our primer sequences (forward, complement, reverse and reverse complement).</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="12-Others_12S_primer_pair_steps.html#cb55-1" aria-hidden="true" tabindex="-1"></a>FWD2.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(FWD2)</span>
<span id="cb55-2"><a href="12-Others_12S_primer_pair_steps.html#cb55-2" aria-hidden="true" tabindex="-1"></a>FWD2.orients <span class="co"># print all orientations of the forward primer to the console</span></span>
<span id="cb55-3"><a href="12-Others_12S_primer_pair_steps.html#cb55-3" aria-hidden="true" tabindex="-1"></a>REV2.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(REV2)</span>
<span id="cb55-4"><a href="12-Others_12S_primer_pair_steps.html#cb55-4" aria-hidden="true" tabindex="-1"></a>REV2.orients <span class="co"># print all orientations of the reverse primer to the console</span></span></code></pre></div>
<p>We have already filtered any Ns from our raw data when we processed the MiFish primer sequences, so we can use those output files to find and trim the 12S-V5 primer sequences.</p>
<p>The <code>primerHits</code> function will use all possible 12S-V5 primer combinations to count the number of times a primer is found in the forward and reverse read in each orientation.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="12-Others_12S_primer_pair_steps.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD2.ForwardReads =</span> <span class="fu">sapply</span>(FWD2.orients, primerHits, <span class="at">fn =</span> fnFs.filtN[[<span class="dv">1</span>]]), </span>
<span id="cb56-2"><a href="12-Others_12S_primer_pair_steps.html#cb56-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD2.ReverseReads =</span> <span class="fu">sapply</span>(FWD2.orients, primerHits, <span class="at">fn =</span> fnRs.filtN[[<span class="dv">1</span>]]), </span>
<span id="cb56-3"><a href="12-Others_12S_primer_pair_steps.html#cb56-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV2.ForwardReads =</span> <span class="fu">sapply</span>(REV2.orients, primerHits, <span class="at">fn =</span> fnFs.filtN[[<span class="dv">1</span>]]), </span>
<span id="cb56-4"><a href="12-Others_12S_primer_pair_steps.html#cb56-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV2.ReverseReads =</span> <span class="fu">sapply</span>(REV2.orients, primerHits, <span class="at">fn =</span> fnRs.filtN[[<span class="dv">1</span>]]))</span></code></pre></div>
<p>The <code>[[1]]</code> part of the rbind command runs the function on the first file in our list, and it will take a little while to run. Because we know that the libraries for each sample were made using the same protocol we’ll just check the primer orientations in the first file again as all the samples should look similar.</p>
<p>##As with MiFish, we can see that although most of our primers are found as expected in the forward orientation on our forward/reverse read in this sample. ##</p>
<p>We also have a few sequences that also contain the reverse complement. This is likely due to the fact that there is some variation in the length of this amplicon between species so in some cases the sequencing machine has cycled through the full region and into the the forward/reverse primer.</p>
</div>
<div id="cutadapt-1" class="section level3" number="13.1.2">
<h3><span class="header-section-number">13.1.2</span> Cutadapt</h3>
<p>Now we know the orientation of our primers we can use the software cutadapt to remove the primers from our samples.</p>
<p>See <a href="https://cutadapt.readthedocs.io/en/stable/" class="uri">https://cutadapt.readthedocs.io/en/stable/</a> for more information on how to install and run cutadapt.</p>
<p>Specify the path to the cutadapt software so R knows where to find and run it.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="12-Others_12S_primer_pair_steps.html#cb57-1" aria-hidden="true" tabindex="-1"></a>cutadapt <span class="ot">&lt;-</span> <span class="st">&quot;/pub39/tea/nsc006/anaconda3/envs/qiime2-2021.2/bin/cutadapt&quot;</span></span></code></pre></div>
<p>We can use the <code>system2</code> command to run a shell command from within R.</p>
<p>Specify the path to a new output directory where we want to put the 12S-V5 cutadapt output files.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="12-Others_12S_primer_pair_steps.html#cb58-1" aria-hidden="true" tabindex="-1"></a>path.cut2 <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output.path, <span class="st">&quot;cutadapt2&quot;</span>)</span>
<span id="cb58-2"><a href="12-Others_12S_primer_pair_steps.html#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut2)) <span class="fu">dir.create</span>(path.cut2)</span>
<span id="cb58-3"><a href="12-Others_12S_primer_pair_steps.html#cb58-3" aria-hidden="true" tabindex="-1"></a>fnFs.cut2 <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut2, <span class="fu">basename</span>(fnFs))</span>
<span id="cb58-4"><a href="12-Others_12S_primer_pair_steps.html#cb58-4" aria-hidden="true" tabindex="-1"></a>fnRs.cut2 <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut2, <span class="fu">basename</span>(fnRs))</span></code></pre></div>
<p>Here we specify the options needed by cutadapt in order to trim the forward orientation of the forward and reverse primer off the forward and reverse read and the reverse complement off the end of the forward and reverse reads.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="12-Others_12S_primer_pair_steps.html#cb59-1" aria-hidden="true" tabindex="-1"></a>FWD2.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD2)</span>
<span id="cb59-2"><a href="12-Others_12S_primer_pair_steps.html#cb59-2" aria-hidden="true" tabindex="-1"></a>REV2.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV2)</span>
<span id="cb59-3"><a href="12-Others_12S_primer_pair_steps.html#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Trim FWD and the reverse-complement of REV off of R1 (forward reads)</span></span>
<span id="cb59-4"><a href="12-Others_12S_primer_pair_steps.html#cb59-4" aria-hidden="true" tabindex="-1"></a>R1.flags2 <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;-g&quot;</span>, FWD2, <span class="st">&quot;-a&quot;</span>, REV2.RC) </span>
<span id="cb59-5"><a href="12-Others_12S_primer_pair_steps.html#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Trim REV and the reverse-complement of FWD off of R2 (reverse reads)</span></span>
<span id="cb59-6"><a href="12-Others_12S_primer_pair_steps.html#cb59-6" aria-hidden="true" tabindex="-1"></a>R2.flags2 <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;-G&quot;</span>, REV2, <span class="st">&quot;-A&quot;</span>, FWD2.RC) </span></code></pre></div>
<p>In addition to trimming the primers from the reads we will also specify a couple of extra useful parameters.</p>
<p><code>--discard-untrimmed</code> this tells cutadapt to discard any read where the primers haven’t been trimmed off. This is especially important for our data as our files at the moment contain sequences amplified using both MiFish-U and 12S-V5 primer sets. We only want to keep sequences matching the 12S-V5 primer set for this analysis.</p>
<p><code>--minimum-length 60</code> discard reads shorter than 60bp. This will remove unexpected short reads and help speed up further analysis.</p>
<p>Run cutadapt. (Note. The cutadapt step is time intensive so this might take a little while to run, probably about 15 minutes).</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="12-Others_12S_primer_pair_steps.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {</span>
<span id="cb60-2"><a href="12-Others_12S_primer_pair_steps.html#cb60-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(R1.flags2, R2.flags2, <span class="st">&quot;-n&quot;</span>, <span class="dv">2</span>, <span class="co"># -n 2 required to remove FWD and </span></span>
<span id="cb60-3"><a href="12-Others_12S_primer_pair_steps.html#cb60-3" aria-hidden="true" tabindex="-1"></a>                             <span class="st">&quot;-o&quot;</span>, fnFs.cut2[i], <span class="st">&quot;-p&quot;</span>, fnRs.cut2[i], <span class="co"># output files</span></span>
<span id="cb60-4"><a href="12-Others_12S_primer_pair_steps.html#cb60-4" aria-hidden="true" tabindex="-1"></a>                             fnFs.filtN[i], fnRs.filtN[i], <span class="co"># input files</span></span>
<span id="cb60-5"><a href="12-Others_12S_primer_pair_steps.html#cb60-5" aria-hidden="true" tabindex="-1"></a>                             <span class="st">&quot;--discard-untrimmed&quot;</span>,</span>
<span id="cb60-6"><a href="12-Others_12S_primer_pair_steps.html#cb60-6" aria-hidden="true" tabindex="-1"></a>                             <span class="st">&quot;--minimum-length 60&quot;</span>))</span>
<span id="cb60-7"><a href="12-Others_12S_primer_pair_steps.html#cb60-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>We can now check whether all the primers have been removed using the primerHits function we specified earlier.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="12-Others_12S_primer_pair_steps.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD2.ForwardReads =</span> <span class="fu">sapply</span>(FWD2.orients, primerHits, <span class="at">fn =</span> fnFs.cut2[[<span class="dv">1</span>]]), </span>
<span id="cb61-2"><a href="12-Others_12S_primer_pair_steps.html#cb61-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD2.ReverseReads =</span> <span class="fu">sapply</span>(FWD2.orients, primerHits, <span class="at">fn =</span> fnRs.cut2[[<span class="dv">1</span>]]), </span>
<span id="cb61-3"><a href="12-Others_12S_primer_pair_steps.html#cb61-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV2.ForwardReads =</span> <span class="fu">sapply</span>(REV2.orients, primerHits, <span class="at">fn =</span> fnFs.cut2[[<span class="dv">1</span>]]), </span>
<span id="cb61-4"><a href="12-Others_12S_primer_pair_steps.html#cb61-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV2.ReverseReads =</span> <span class="fu">sapply</span>(REV2.orients, primerHits, <span class="at">fn =</span> fnRs.cut2[[<span class="dv">1</span>]]))</span></code></pre></div>
<p>We now have no primers remaining in our file.</p>
</div>
</div>
<div id="checking-the-quality-of-your-data-1" class="section level2" number="13.2">
<h2><span class="header-section-number">13.2</span> Checking the quality of your data</h2>
<p>Those of you who attended our quality Introduction to sequencing data and quality control course we used FastQC to check the quality of our data. DADA2 has its own quality control option, which plots a similar read length by quality figure.</p>
<p>To run first import the cutadapt files and extract the sample names and then run the <code>plotQualityProfile</code> function</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="12-Others_12S_primer_pair_steps.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify the paths and file names the forward and reverse primer cleaned files </span></span>
<span id="cb62-2"><a href="12-Others_12S_primer_pair_steps.html#cb62-2" aria-hidden="true" tabindex="-1"></a>cutFs2 <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path.cut2, <span class="at">pattern =</span> <span class="st">&quot;_L001_R1_001.fastq&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb62-3"><a href="12-Others_12S_primer_pair_steps.html#cb62-3" aria-hidden="true" tabindex="-1"></a>cutRs2 <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path.cut2, <span class="at">pattern =</span> <span class="st">&quot;_L001_R2_001.fastq&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb62-4"><a href="12-Others_12S_primer_pair_steps.html#cb62-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-5"><a href="12-Others_12S_primer_pair_steps.html#cb62-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract sample names Need to do this again or should be the same ???</span></span>
<span id="cb62-6"><a href="12-Others_12S_primer_pair_steps.html#cb62-6" aria-hidden="true" tabindex="-1"></a><span class="co">#get.sample.name &lt;- function(fname) strsplit(basename(fname), &quot;_&quot;)[[1]][2]</span></span>
<span id="cb62-7"><a href="12-Others_12S_primer_pair_steps.html#cb62-7" aria-hidden="true" tabindex="-1"></a><span class="co">#sample.names &lt;- unname(sapply(cutFs, get.sample.name))</span></span>
<span id="cb62-8"><a href="12-Others_12S_primer_pair_steps.html#cb62-8" aria-hidden="true" tabindex="-1"></a><span class="co">#head(sample.names)</span></span>
<span id="cb62-9"><a href="12-Others_12S_primer_pair_steps.html#cb62-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-10"><a href="12-Others_12S_primer_pair_steps.html#cb62-10" aria-hidden="true" tabindex="-1"></a><span class="co"># check the quality for the first file</span></span>
<span id="cb62-11"><a href="12-Others_12S_primer_pair_steps.html#cb62-11" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(cutFs2[<span class="dv">1</span><span class="sc">:</span><span class="dv">1</span>])</span></code></pre></div>
<p>To interpret this plot, the gray-scale heatmap shows the the frequency of each quality score along the forward read length.The green line is the median quality score and the orange lines are the quartiles. The red line at the bottom of the plot represents the proportion of reads of that particular length.</p>
<p>The quality is very good for our forward reads. You can also see that the majority of the forward reads are ~130bp long after having the primers removed with cutadapt.</p>
<p>Now check the quality of the reverse file in the same way</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="12-Others_12S_primer_pair_steps.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(cutRs2[<span class="dv">1</span><span class="sc">:</span><span class="dv">1</span>])</span></code></pre></div>
<p>The reverse reads also look like they are good quality.</p>
<p>To check the quality of the second and third fastq files we would type.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="12-Others_12S_primer_pair_steps.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(cutFs2[<span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>])</span>
<span id="cb64-2"><a href="12-Others_12S_primer_pair_steps.html#cb64-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-3"><a href="12-Others_12S_primer_pair_steps.html#cb64-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(cutRs2[<span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>])</span></code></pre></div>
</div>
<div id="cleaning-your-data-1" class="section level2" number="13.3">
<h2><span class="header-section-number">13.3</span> Cleaning your data</h2>
<p>We will now filter our data to remove any poor quality reads.</p>
<p>First set the path to a directory to store the filtered output files called <code>filtered</code>.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="12-Others_12S_primer_pair_steps.html#cb65-1" aria-hidden="true" tabindex="-1"></a>filtFs2 <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut2, <span class="st">&quot;../filtered&quot;</span>, <span class="fu">basename</span>(cutFs2))</span>
<span id="cb65-2"><a href="12-Others_12S_primer_pair_steps.html#cb65-2" aria-hidden="true" tabindex="-1"></a>filtRs2 <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut2, <span class="st">&quot;../filtered&quot;</span>, <span class="fu">basename</span>(cutRs2))</span></code></pre></div>
<p>Now run <code>filterAndTrim</code>. This time we use the standard filtering parameters:</p>
<ul>
<li><code>maxN=0</code> After truncation, sequences with more than 0 Ns will be discarded. (DADA2 requires sequences contain no Ns)</li>
<li><code>truncQ = 2</code> Truncate reads at the first instance of a quality score less than or equal to 2</li>
<li><code>rm.phix = TRUE</code> Discard reads that match against the phiX genome</li>
<li><code>maxEE=2</code> After truncation, reads with higher than 2 “expected errors” will be discarded</li>
<li><code>minLen = 60</code> Remove reads with length less than 60 (note these should have already been removed by cutadapt)</li>
<li><code>multithread = TRUE</code> input files are filtered in parallel</li>
</ul>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="12-Others_12S_primer_pair_steps.html#cb66-1" aria-hidden="true" tabindex="-1"></a>out2 <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(cutFs2, filtFs2, cutRs2, filtRs2, <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">maxEE =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>), </span>
<span id="cb66-2"><a href="12-Others_12S_primer_pair_steps.html#cb66-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">truncQ =</span> <span class="dv">2</span>, <span class="at">minLen =</span> <span class="dv">60</span>, <span class="at">rm.phix =</span> <span class="cn">TRUE</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb66-3"><a href="12-Others_12S_primer_pair_steps.html#cb66-3" aria-hidden="true" tabindex="-1"></a>out2</span></code></pre></div>
<p>##Some samples have lost all reads in this filtering step. These could be poor quality samples but we also have negatives controls in this dataset so we would expect these to contain zero or very few reads.##</p>
</div>
<div id="identification-of-asvs-1" class="section level2" number="13.4">
<h2><span class="header-section-number">13.4</span> Identification of ASVs</h2>
<div id="generate-an-error-model-1" class="section level3" number="13.4.1">
<h3><span class="header-section-number">13.4.1</span> Generate an error model</h3>
<p>First we need to model the error rates of our dataset using both the forward and reverse reads. Each dataset will have a specific error-signature with errors introduced by PCR amplification and sequencing.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="12-Others_12S_primer_pair_steps.html#cb67-1" aria-hidden="true" tabindex="-1"></a>errF2 <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtFs2, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb67-2"><a href="12-Others_12S_primer_pair_steps.html#cb67-2" aria-hidden="true" tabindex="-1"></a>errR2 <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtRs2, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>We can use the <code>plotErrors</code> function to check the estimated error rates.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="12-Others_12S_primer_pair_steps.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF2, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>You will see the
<code>Warning messages: 1: Transformation introduced infinite values in continuous y-axis  2: Transformation introduced infinite values in continuous y-axis</code></p>
<p>Do not worry about these warning messages. This is a message from the plotting function to let you know that there were some zero values in the data plotted (which turn into infinities on the log-scale). This is expected, it results from the fact that not every combination of error type (e.g. A-&gt;C) and quality score (e.g. 33) is observed in your data which is normal.</p>
<p>Interpreting the plots.</p>
<ul>
<li><p>The error rates for each possible transition (e.g. A→C, A→G) are shown</p></li>
<li><p>Red line - expected based on the quality score</p></li>
<li><p>Black line - estimate</p></li>
<li><p>Black dots - observed</p></li>
</ul>
<p>What we are expecting to see here is that the observed to match up with estimates. Here we can see that the black dots track well with the black line. We can also see that the error rates drop with the quality score as we would expect. Sanity checks complete, we can proceed with the analysis.</p>
<ul>
<li>If you are worried about what your error plots look like when you fit your own dataset, one possible way to to improve the fit is to try increasing the number of bases the function is using (the default is 100 million)</li>
</ul>
</div>
<div id="dereplication-1" class="section level3" number="13.4.2">
<h3><span class="header-section-number">13.4.2</span> Dereplication</h3>
<p>The next step is to dereplicate identical reads. This is a common step in many workflows used for processing amplicons. This saves time and processing power as indentical reads are collapsed together. For example instead of processing 100 identical sequences, only one is processed but the original number (e.g. 100) is associated with it.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="12-Others_12S_primer_pair_steps.html#cb69-1" aria-hidden="true" tabindex="-1"></a>exists2 <span class="ot">&lt;-</span> <span class="fu">file.exists</span>(filtFs2) <span class="co"># check that all the samples are still present after filtering</span></span>
<span id="cb69-2"><a href="12-Others_12S_primer_pair_steps.html#cb69-2" aria-hidden="true" tabindex="-1"></a>derepFs2 <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtFs2[exists2], <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb69-3"><a href="12-Others_12S_primer_pair_steps.html#cb69-3" aria-hidden="true" tabindex="-1"></a>derepRs2 <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtRs2[exists2], <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb69-4"><a href="12-Others_12S_primer_pair_steps.html#cb69-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Name the derep-class objects by the sample names</span></span>
<span id="cb69-5"><a href="12-Others_12S_primer_pair_steps.html#cb69-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derepFs2) <span class="ot">&lt;-</span> sample.names2[exists2]</span>
<span id="cb69-6"><a href="12-Others_12S_primer_pair_steps.html#cb69-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derepRs2) <span class="ot">&lt;-</span> sample.names2[exists2]</span></code></pre></div>
</div>
<div id="inferrence-of-asvs-1" class="section level3" number="13.4.3">
<h3><span class="header-section-number">13.4.3</span> Inferrence of ASVs</h3>
<p>Now we are ready to infer the ASVs in our dataset. To do this DADA2 used the error models above to infer the true sample composition (see <a href="https://www.nature.com/articles/nmeth.3869#methods" class="uri">https://www.nature.com/articles/nmeth.3869#methods</a> for more details).</p>
<p>Here we will run the inference algorithm on single samples to save time but it is also possible to run samples together in pseudo-pools to increase the ability to identify ASVs of low abundance. Low abundance ASVs in a sample may be filtered out when run separately but if they are found in higher numbers in another sample then the chance of that ASV being real increases. Pseudo-pooling increases the chances of these “real” low abundance ASVs being retained within samples. Whether or not to use the pseudo-pooling option will depend on your dataset and experimental design (see <a href="https://benjjneb.github.io/dada2/pseudo.html#Pseudo-pooling" class="uri">https://benjjneb.github.io/dada2/pseudo.html#Pseudo-pooling</a> for more information).</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="12-Others_12S_primer_pair_steps.html#cb70-1" aria-hidden="true" tabindex="-1"></a>dadaFs2 <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepFs2, <span class="at">err =</span> errF2, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb70-2"><a href="12-Others_12S_primer_pair_steps.html#cb70-2" aria-hidden="true" tabindex="-1"></a>dadaRs2 <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepRs2, <span class="at">err =</span> errR2, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>It is important to note that you want to run the error and inference steps on datasets generated from a single Illumina run. Data generated from different runs can have different error structures.</p>
</div>
<div id="merging-paired-end-reads-1" class="section level3" number="13.4.4">
<h3><span class="header-section-number">13.4.4</span> Merging paired end reads</h3>
<p>Up until now we have carried out all filtering, error correction and inference on the forward and reverse reads seperately. It is now time to merge the two files. By default the minimum overlap allowed between the two samples is 12 bp.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="12-Others_12S_primer_pair_steps.html#cb71-1" aria-hidden="true" tabindex="-1"></a>mergers2 <span class="ot">&lt;-</span> <span class="fu">mergePairs</span>(dadaFs2, derepFs2, dadaRs2, derepRs2, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="making-our-asv-matrix-1" class="section level3" number="13.4.5">
<h3><span class="header-section-number">13.4.5</span> Making our ASV matrix</h3>
<p>Now it is time to make the counts table. Each column represents a single ASV and each row is an individual sample.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="12-Others_12S_primer_pair_steps.html#cb72-1" aria-hidden="true" tabindex="-1"></a>seqtab2 <span class="ot">&lt;-</span> <span class="fu">makeSequenceTable</span>(mergers2)</span>
<span id="cb72-2"><a href="12-Others_12S_primer_pair_steps.html#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab2) </span></code></pre></div>
<p>There are 583 ASVs in our matrix.</p>
</div>
<div id="chimera-detection-and-removal-1" class="section level3" number="13.4.6">
<h3><span class="header-section-number">13.4.6</span> Chimera detection and removal</h3>
<p>The last step in generating our ASV matrix is to detect and remove any chimeric sequences.</p>
<p>Chimeric sequences are formed when two or more biological sequences joined together. This is fairly common in amplicon sequencing. DADA2 uses a method whereby it combines the left and right segments of abundant reads and compares these with lower abundant sequences. Any low abundant sequences that match are removed.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="12-Others_12S_primer_pair_steps.html#cb73-1" aria-hidden="true" tabindex="-1"></a>seqtab2.nochim <span class="ot">&lt;-</span> <span class="fu">removeBimeraDenovo</span>(seqtab2, <span class="at">method=</span><span class="st">&quot;consensus&quot;</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb73-2"><a href="12-Others_12S_primer_pair_steps.html#cb73-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-3"><a href="12-Others_12S_primer_pair_steps.html#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab2.nochim)</span></code></pre></div>
<p>We have 212 ASVs remaining after filtering out chimeras. Although this is a large proportion of sequence variants it should be a smaller proportion of the total sequences.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="12-Others_12S_primer_pair_steps.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(seqtab2.nochim)<span class="sc">/</span><span class="fu">sum</span>(seqtab2)</span></code></pre></div>
<p>In these data ~ 7.5% of the merged sequence reads were identified as chimeric.</p>
<p>CHeck the range of ASV lengths:</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="12-Others_12S_primer_pair_steps.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab2.nochim)))</span></code></pre></div>
</div>
<div id="sequence-tracking-sanity-check-1" class="section level3" number="13.4.7">
<h3><span class="header-section-number">13.4.7</span> Sequence tracking sanity check</h3>
<p>The last thing to do in this section is to track the number of sequences through the pipeline to check whether everything has run as expected and whether there are any steps where we loose a disproportionate number of sequences. If we end up with too few reads to run further analysis we can use this table to identify any step which might require further investigation and optimisation.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="12-Others_12S_primer_pair_steps.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co">#getN &lt;- function(x) sum(getUniques(x))</span></span>
<span id="cb76-2"><a href="12-Others_12S_primer_pair_steps.html#cb76-2" aria-hidden="true" tabindex="-1"></a>track2 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(out2, <span class="fu">sapply</span>(dadaFs2, getN), <span class="fu">sapply</span>(dadaRs2, getN), <span class="fu">sapply</span>(mergers2, getN), <span class="fu">rowSums</span>(seqtab.nochim2))</span>
<span id="cb76-3"><a href="12-Others_12S_primer_pair_steps.html#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(track2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;input&quot;</span>, <span class="st">&quot;filtered&quot;</span>, <span class="st">&quot;denoisedF&quot;</span>, <span class="st">&quot;denoisedR&quot;</span>, <span class="st">&quot;merged&quot;</span>, <span class="st">&quot;nonchim&quot;</span>)</span>
<span id="cb76-4"><a href="12-Others_12S_primer_pair_steps.html#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(track2) <span class="ot">&lt;-</span> sample.names2</span>
<span id="cb76-5"><a href="12-Others_12S_primer_pair_steps.html#cb76-5" aria-hidden="true" tabindex="-1"></a>track2</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="11-Further_analysis.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
