<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Optional: guide to analysing the second primer pair | Metabarcoding and data sharing for environmental DNA</title>
  <meta name="description" content="NEOF book for the Metabarcoding and data sharing for environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Optional: guide to analysing the second primer pair | Metabarcoding and data sharing for environmental DNA" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding and data sharing for environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Optional: guide to analysing the second primer pair | Metabarcoding and data sharing for environmental DNA" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding and data sharing for environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher and Helen Hipperson" />


<meta name="date" content="2024-03-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="12-Analysing_another_marker.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://www.ebi.ac.uk/training/events/metabarcoding-and-data-sharing-environmental-dna/" target="blank"><img src="figures/EBI_CEH_NEOF.png"></a></li>
<li><a>Environmental metabarcoding</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Metabarcoding.html"><a href="01-Metabarcoding.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Example_dataset.html"><a href="02-Example_dataset.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html"><i class="fa fa-check"></i><b>3</b> Introduction to DADA2 and R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#dada2"><i class="fa fa-check"></i><b>3.1</b> DADA2</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#otus-vs-asvs"><i class="fa fa-check"></i><b>3.1.1</b> OTUs vs ASVs</a></li>
<li class="chapter" data-level="3.1.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#workflow"><i class="fa fa-check"></i><b>3.1.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#r"><i class="fa fa-check"></i><b>3.2</b> R</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#the-console-window"><i class="fa fa-check"></i><b>3.2.1</b> The console window</a></li>
<li class="chapter" data-level="3.2.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#your-working-directory"><i class="fa fa-check"></i><b>3.2.2</b> Your working directory</a></li>
<li class="chapter" data-level="3.2.3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#packages"><i class="fa fa-check"></i><b>3.2.3</b> Packages</a></li>
<li class="chapter" data-level="3.2.4" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#functions"><i class="fa fa-check"></i><b>3.2.4</b> Functions</a></li>
<li class="chapter" data-level="3.2.5" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#assigning-values-to-variables"><i class="fa fa-check"></i><b>3.2.5</b> Assigning values to variables</a></li>
<li class="chapter" data-level="3.2.6" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#a-few-useful-commands-and-tips"><i class="fa fa-check"></i><b>3.2.6</b> A few useful commands and tips</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>DADA2</b></span></li>
<li class="chapter" data-level="5" data-path="05-Loading_data.html"><a href="05-Loading_data.html"><i class="fa fa-check"></i><b>5</b> Loading data into R and DADA2</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#jupyter"><i class="fa fa-check"></i><b>5.1</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.1.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#create-r-notebook"><i class="fa fa-check"></i><b>5.1.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.1.3" data-path="05-Loading_data.html"><a href="05-Loading_data.html#cells-and-code"><i class="fa fa-check"></i><b>5.1.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.1.4" data-path="05-Loading_data.html"><a href="05-Loading_data.html#create-new-cells"><i class="fa fa-check"></i><b>5.1.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.1.5" data-path="05-Loading_data.html"><a href="05-Loading_data.html#running-code"><i class="fa fa-check"></i><b>5.1.5</b> Running code</a></li>
<li class="chapter" data-level="5.1.6" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-file"><i class="fa fa-check"></i><b>5.1.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.1.7" data-path="05-Loading_data.html"><a href="05-Loading_data.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.1.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.1.8" data-path="05-Loading_data.html"><a href="05-Loading_data.html#close-the-notebook"><i class="fa fa-check"></i><b>5.1.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.1.9" data-path="05-Loading_data.html"><a href="05-Loading_data.html#video-tutorial"><i class="fa fa-check"></i><b>5.1.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#r-1"><i class="fa fa-check"></i><b>5.2</b> R</a></li>
<li class="chapter" data-level="5.3" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-dada2"><i class="fa fa-check"></i><b>5.3</b> Loading DADA2</a></li>
<li class="chapter" data-level="5.4" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-our-data"><i class="fa fa-check"></i><b>5.4</b> Loading our data</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-paths-for-our-input-and-output-directories"><i class="fa fa-check"></i><b>5.4.1</b> Saving the paths for our input and output directories</a></li>
<li class="chapter" data-level="5.4.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#inputting-the-forward-and-reverse-reads"><i class="fa fa-check"></i><b>5.4.2</b> Inputting the forward and reverse reads</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-primer_removal.html"><a href="06-primer_removal.html"><i class="fa fa-check"></i><b>6</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-primer_removal.html"><a href="06-primer_removal.html#primer-orientation-checking"><i class="fa fa-check"></i><b>6.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="6.2" data-path="06-primer_removal.html"><a href="06-primer_removal.html#cutadapt"><i class="fa fa-check"></i><b>6.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-quality_checks.html"><a href="07-quality_checks.html"><i class="fa fa-check"></i><b>7</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="8" data-path="08-clean.html"><a href="08-clean.html"><i class="fa fa-check"></i><b>8</b> Cleaning your data</a></li>
<li class="chapter" data-level="9" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html"><i class="fa fa-check"></i><b>9</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#generate-an-error-model"><i class="fa fa-check"></i><b>9.1</b> Generate an error model</a></li>
<li class="chapter" data-level="9.2" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dereplication"><i class="fa fa-check"></i><b>9.2</b> Dereplication</a></li>
<li class="chapter" data-level="9.3" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#inferrence-of-asvs"><i class="fa fa-check"></i><b>9.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="9.4" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#merging-paired-end-reads"><i class="fa fa-check"></i><b>9.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="9.5" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#making-our-asv-matrix"><i class="fa fa-check"></i><b>9.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="9.6" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#chimera-detection-and-removal"><i class="fa fa-check"></i><b>9.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="9.7" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#sequence-tracking-sanity-check"><i class="fa fa-check"></i><b>9.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Assign_taxonomy.html"><a href="10-Assign_taxonomy.html"><i class="fa fa-check"></i><b>10</b> Assigning taxonomy</a></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="11" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html"><i class="fa fa-check"></i><b>11</b> Further analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#r-packages-for-further-analysis"><i class="fa fa-check"></i><b>11.1</b> R packages for further analysis</a></li>
<li class="chapter" data-level="11.2" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#rarefaction-curves"><i class="fa fa-check"></i><b>11.2</b> Rarefaction curves</a></li>
<li class="chapter" data-level="11.3" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#alpha-diversity"><i class="fa fa-check"></i><b>11.3</b> Alpha diversity</a></li>
<li class="chapter" data-level="11.4" data-path="11-Further_analysis.html"><a href="11-Further_analysis.html#beta-diversity"><i class="fa fa-check"></i><b>11.4</b> Beta diversity</a></li>
</ul></li>
<li class="part"><span><b>Optional Extras</b></span></li>
<li class="chapter" data-level="12" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html"><i class="fa fa-check"></i><b>12</b> Optional: analysing the second primer pair</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html#the-primer-sequences"><i class="fa fa-check"></i><b>12.1</b> The primer sequences</a></li>
<li class="chapter" data-level="12.2" data-path="12-Analysing_another_marker.html"><a href="12-Analysing_another_marker.html#questions"><i class="fa fa-check"></i><b>12.2</b> Questions</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html"><i class="fa fa-check"></i><b>13</b> Optional: guide to analysing the second primer pair</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#loading-the-data"><i class="fa fa-check"></i><b>13.1</b> Loading the data</a></li>
<li class="chapter" data-level="13.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#identifying-and-removing-primers"><i class="fa fa-check"></i><b>13.2</b> Identifying and removing primers</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#primer-orientation-checking-1"><i class="fa fa-check"></i><b>13.2.1</b> Primer orientation checking</a></li>
<li class="chapter" data-level="13.2.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#cutadapt-1"><i class="fa fa-check"></i><b>13.2.2</b> Cutadapt</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#checking-the-quality-of-your-data"><i class="fa fa-check"></i><b>13.3</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="13.4" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#cleaning-your-data"><i class="fa fa-check"></i><b>13.4</b> Cleaning your data</a></li>
<li class="chapter" data-level="13.5" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#identification-of-asvs"><i class="fa fa-check"></i><b>13.5</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#generate-an-error-model-1"><i class="fa fa-check"></i><b>13.5.1</b> Generate an error model</a></li>
<li class="chapter" data-level="13.5.2" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#dereplication-1"><i class="fa fa-check"></i><b>13.5.2</b> Dereplication</a></li>
<li class="chapter" data-level="13.5.3" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#inferrence-of-asvs-1"><i class="fa fa-check"></i><b>13.5.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="13.5.4" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#merging-paired-end-reads-1"><i class="fa fa-check"></i><b>13.5.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="13.5.5" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#making-our-asv-matrix-1"><i class="fa fa-check"></i><b>13.5.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="13.5.6" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#chimera-detection-and-removal-1"><i class="fa fa-check"></i><b>13.5.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="13.5.7" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#sequence-tracking-sanity-check-1"><i class="fa fa-check"></i><b>13.5.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#assigning-taxonomy"><i class="fa fa-check"></i><b>13.6</b> Assigning taxonomy</a></li>
<li class="chapter" data-level="13.7" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#further-analysis"><i class="fa fa-check"></i><b>13.7</b> Further analysis</a></li>
<li class="chapter" data-level="13.8" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#rarefaction-curves-1"><i class="fa fa-check"></i><b>13.8</b> Rarefaction curves</a></li>
<li class="chapter" data-level="13.9" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#alpha-diversity-1"><i class="fa fa-check"></i><b>13.9</b> Alpha diversity</a></li>
<li class="chapter" data-level="13.10" data-path="13-Others_marker_steps.html"><a href="13-Others_marker_steps.html#beta-diversity-1"><i class="fa fa-check"></i><b>13.10</b> Beta diversity</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Metabarcoding and data sharing for environmental DNA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="secondmarkerguide" class="section level1 hasAnchor" number="13">
<h1><span class="header-section-number">Chapter 13</span> Optional: guide to analysing the second primer pair<a href="13-Others_marker_steps.html#secondmarkerguide" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="loading-the-data" class="section level2 hasAnchor" number="13.1">
<h2><span class="header-section-number">13.1</span> Loading the data<a href="13-Others_marker_steps.html#loading-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Load the DAD2 packages needed for analysis as before.</p>
<p>The two primer pairs were multiplexed together for each sample, so our input path and lists of file names and locations (fnFs and fnRs) remain the same.</p>
</div>
<div id="identifying-and-removing-primers" class="section level2 hasAnchor" number="13.2">
<h2><span class="header-section-number">13.2</span> Identifying and removing primers<a href="13-Others_marker_steps.html#identifying-and-removing-primers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now to carry out primer removal. DADA2 requires the primers you used to be trimmed off the forward and reverse of your reads. We will use the software cutadapt again for this, which will this time identify and remove the 12S-V5 primers and discard any other sequences (including those we have already analysed for MiFish-U).</p>
<p>Read in your forward and reverse primer sequences for the 12S-V5 primer set, but give them different variable names. The 12S-V5 primers do not contain any degenerate bases but cutadapt is able to handle any degenerate bases you may have in your own primer sequences.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="13-Others_marker_steps.html#cb60-1" tabindex="-1"></a>FWD2 <span class="ot">&lt;-</span> <span class="st">&quot;ACTGGGATTAGATACCCC&quot;</span></span>
<span id="cb60-2"><a href="13-Others_marker_steps.html#cb60-2" tabindex="-1"></a>REV2 <span class="ot">&lt;-</span> <span class="st">&quot;TAGAACAGGCTCCTCTAG&quot;</span></span></code></pre></div>
<p>Here are the primer sequences again, but formatted so they are a little easier to read. When you type them into R though you should do this with no spaces, as above.</p>
<p>12S-V5 FWD: ACT GGG ATT AGA TAC CCC</p>
<p>12S-V5 REV: TAG AAC AGG CTC CTC TAG</p>
<div id="primer-orientation-checking-1" class="section level3 hasAnchor" number="13.2.1">
<h3><span class="header-section-number">13.2.1</span> Primer orientation checking<a href="13-Others_marker_steps.html#primer-orientation-checking-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s check the orientation of these primers. We know how our libraries were created and sequenced so we are expecting that the forward primer should be located in a forward orientation in the forward read and the reverse primer should be located in the forward orientation in the reverse read.</p>
<p>For your own data you might be less certain as to which orientation your primers are in your reads.</p>
<p>We can use the <code>allOrients</code> function again to make a list of all orientations of our primer sequences (forward, complement, reverse and reverse complement, again make sure to give these R variables a unique name). Note that if you can copy this from your previous notebook.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="13-Others_marker_steps.html#cb61-1" tabindex="-1"></a>FWD2.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(FWD2)</span>
<span id="cb61-2"><a href="13-Others_marker_steps.html#cb61-2" tabindex="-1"></a>FWD2.orients <span class="co"># print all orientations of the forward primer to the console</span></span>
<span id="cb61-3"><a href="13-Others_marker_steps.html#cb61-3" tabindex="-1"></a>REV2.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(REV2)</span>
<span id="cb61-4"><a href="13-Others_marker_steps.html#cb61-4" tabindex="-1"></a>REV2.orients <span class="co"># print all orientations of the reverse primer to the console</span></span></code></pre></div>
<p>We have already filtered any Ns from our raw data when we processed the MiFish primer sequences, so we can use those output files to find and trim the 12S-V5 primer sequences.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="13-Others_marker_steps.html#cb62-1" tabindex="-1"></a>fnFs.filtN <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output.path, <span class="st">&quot;filtN&quot;</span>, <span class="fu">basename</span>(fnFs)) </span>
<span id="cb62-2"><a href="13-Others_marker_steps.html#cb62-2" tabindex="-1"></a>fnRs.filtN <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output.path, <span class="st">&quot;filtN&quot;</span>, <span class="fu">basename</span>(fnRs))</span></code></pre></div>
<p>The <code>primerHits</code> function will use all possible 12S-V5 primer combinations to count the number of times a primer is found in the forward and reverse read in each orientation.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="13-Others_marker_steps.html#cb63-1" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD2.ForwardReads =</span> <span class="fu">sapply</span>(FWD2.orients, primerHits, <span class="at">fn =</span> fnFs.filtN[[<span class="dv">1</span>]]), </span>
<span id="cb63-2"><a href="13-Others_marker_steps.html#cb63-2" tabindex="-1"></a>      <span class="at">FWD2.ReverseReads =</span> <span class="fu">sapply</span>(FWD2.orients, primerHits, <span class="at">fn =</span> fnRs.filtN[[<span class="dv">1</span>]]), </span>
<span id="cb63-3"><a href="13-Others_marker_steps.html#cb63-3" tabindex="-1"></a>      <span class="at">REV2.ForwardReads =</span> <span class="fu">sapply</span>(REV2.orients, primerHits, <span class="at">fn =</span> fnFs.filtN[[<span class="dv">1</span>]]), </span>
<span id="cb63-4"><a href="13-Others_marker_steps.html#cb63-4" tabindex="-1"></a>      <span class="at">REV2.ReverseReads =</span> <span class="fu">sapply</span>(REV2.orients, primerHits, <span class="at">fn =</span> fnRs.filtN[[<span class="dv">1</span>]]))</span></code></pre></div>
<p>The <code>[[1]]</code> part of the rbind command runs the function on the first file in our list, and it will take a little while to run.</p>
<p>As with MiFish, we have both primers found as expected in the forward orientation on our forward and reverse read in this sample. However, there are also the same number or more hits for the reverse complements of the primer sequences. This is actually a bit of an odd looking sample as there are very few reads (&lt; 2,000) matching the primer sequence in the forward orientation, so let’s check another sample to get a better idea of the 12S-V5 primers in these data.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="13-Others_marker_steps.html#cb64-1" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD2.ForwardReads =</span> <span class="fu">sapply</span>(FWD2.orients, primerHits, <span class="at">fn =</span> fnFs.filtN[[<span class="dv">5</span>]]), </span>
<span id="cb64-2"><a href="13-Others_marker_steps.html#cb64-2" tabindex="-1"></a>      <span class="at">FWD2.ReverseReads =</span> <span class="fu">sapply</span>(FWD2.orients, primerHits, <span class="at">fn =</span> fnRs.filtN[[<span class="dv">5</span>]]), </span>
<span id="cb64-3"><a href="13-Others_marker_steps.html#cb64-3" tabindex="-1"></a>      <span class="at">REV2.ForwardReads =</span> <span class="fu">sapply</span>(REV2.orients, primerHits, <span class="at">fn =</span> fnFs.filtN[[<span class="dv">5</span>]]), </span>
<span id="cb64-4"><a href="13-Others_marker_steps.html#cb64-4" tabindex="-1"></a>      <span class="at">REV2.ReverseReads =</span> <span class="fu">sapply</span>(REV2.orients, primerHits, <span class="at">fn =</span> fnRs.filtN[[<span class="dv">5</span>]]))</span></code></pre></div>
<p>Here we can see that (a) the reverse complement of the forward primer sequence is frequently seen in the reverse reads, and (b) the reverse complement of the reverse primer is frequently seen in the forward reads.</p>
<ol style="list-style-type: lower-alpha">
<li><p>The 12S-V5 forward primer is actually situated in the same location as the MiFish-U reverse primer (see the figure in section 13), so in this case we are seeing it in both the 12S-V5 and the MiFish-U sequences.</p></li>
<li><p>The expected length of the 12S-V5 amplicons is ~106 bp, and hence the Illumina reads have sequenced through the full amplicon and into the reverse primer.</p></li>
</ol>
<p>This explains the numbers, and neither of these are problematic for cutadapt to identify and trim these primers to leave us with the 12S-V5 sequences for further analysis.</p>
</div>
<div id="cutadapt-1" class="section level3 hasAnchor" number="13.2.2">
<h3><span class="header-section-number">13.2.2</span> Cutadapt<a href="13-Others_marker_steps.html#cutadapt-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Copy and paste the path to cutadapt into your new notebook.</p>
<p>Specify the path to a new output directory (‘cutadapt2’) where we will put the 12S-V5 cutadapt output files.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="13-Others_marker_steps.html#cb65-1" tabindex="-1"></a>path.cut2 <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output.path, <span class="st">&quot;cutadapt2&quot;</span>)</span>
<span id="cb65-2"><a href="13-Others_marker_steps.html#cb65-2" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut2)) <span class="fu">dir.create</span>(path.cut2)</span>
<span id="cb65-3"><a href="13-Others_marker_steps.html#cb65-3" tabindex="-1"></a>fnFs.cut2 <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut2, <span class="fu">basename</span>(fnFs))</span>
<span id="cb65-4"><a href="13-Others_marker_steps.html#cb65-4" tabindex="-1"></a>fnRs.cut2 <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut2, <span class="fu">basename</span>(fnRs))</span></code></pre></div>
<p>Here we specify the options needed by cutadapt in order to trim the forward orientation of the forward and reverse primer off the forward and reverse read and the reverse complement off the end of the forward and reverse reads.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="13-Others_marker_steps.html#cb66-1" tabindex="-1"></a>FWD2.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD2)</span>
<span id="cb66-2"><a href="13-Others_marker_steps.html#cb66-2" tabindex="-1"></a>REV2.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV2)</span>
<span id="cb66-3"><a href="13-Others_marker_steps.html#cb66-3" tabindex="-1"></a><span class="co"># Trim FWD and the reverse-complement of REV off of R1 (forward reads)</span></span>
<span id="cb66-4"><a href="13-Others_marker_steps.html#cb66-4" tabindex="-1"></a>R1.flags2 <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;-g&quot;</span>, FWD2, <span class="st">&quot;-a&quot;</span>, REV2.RC) </span>
<span id="cb66-5"><a href="13-Others_marker_steps.html#cb66-5" tabindex="-1"></a><span class="co"># Trim REV and the reverse-complement of FWD off of R2 (reverse reads)</span></span>
<span id="cb66-6"><a href="13-Others_marker_steps.html#cb66-6" tabindex="-1"></a>R2.flags2 <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;-G&quot;</span>, REV2, <span class="st">&quot;-A&quot;</span>, FWD2.RC) </span></code></pre></div>
<p>In addition to trimming the primers from the reads we will also specify a couple of extra useful parameters.</p>
<p><code>--discard-untrimmed</code> this tells cutadapt to discard any read where the primers haven’t been trimmed off. This is especially important for our data as our files at the moment contain sequences amplified using both MiFish-U and 12S-V5 primer sets. We only want to keep sequences matching the 12S-V5 primer set for this analysis.</p>
<p><code>--minimum-length 60</code> discard reads shorter than 60bp. This will remove unexpected short reads and help speed up further analysis.</p>
<p>Run cutadapt. (Note. The cutadapt step is time intensive so this might take a little while to run, probably about 15 minutes).</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="13-Others_marker_steps.html#cb67-1" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {</span>
<span id="cb67-2"><a href="13-Others_marker_steps.html#cb67-2" tabindex="-1"></a>  <span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(R1.flags2, R2.flags2, <span class="st">&quot;-n&quot;</span>, <span class="dv">2</span>, </span>
<span id="cb67-3"><a href="13-Others_marker_steps.html#cb67-3" tabindex="-1"></a>                             <span class="co"># -n 2 required to remove FWD and REV</span></span>
<span id="cb67-4"><a href="13-Others_marker_steps.html#cb67-4" tabindex="-1"></a>                             <span class="st">&quot;-o&quot;</span>, fnFs.cut2[i], <span class="st">&quot;-p&quot;</span>, fnRs.cut2[i], <span class="co"># output files</span></span>
<span id="cb67-5"><a href="13-Others_marker_steps.html#cb67-5" tabindex="-1"></a>                             fnFs.filtN[i], fnRs.filtN[i], <span class="co"># input files</span></span>
<span id="cb67-6"><a href="13-Others_marker_steps.html#cb67-6" tabindex="-1"></a>                             <span class="st">&quot;--discard-untrimmed&quot;</span>,</span>
<span id="cb67-7"><a href="13-Others_marker_steps.html#cb67-7" tabindex="-1"></a>                             <span class="st">&quot;--minimum-length 60&quot;</span>))</span>
<span id="cb67-8"><a href="13-Others_marker_steps.html#cb67-8" tabindex="-1"></a>}</span></code></pre></div>
<p>We can now check whether all the primers have been removed using the primerHits function we specified earlier.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="13-Others_marker_steps.html#cb68-1" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD2.ForwardReads =</span> <span class="fu">sapply</span>(FWD2.orients, primerHits, <span class="at">fn =</span> fnFs.cut2[[<span class="dv">1</span>]]), </span>
<span id="cb68-2"><a href="13-Others_marker_steps.html#cb68-2" tabindex="-1"></a>      <span class="at">FWD2.ReverseReads =</span> <span class="fu">sapply</span>(FWD2.orients, primerHits, <span class="at">fn =</span> fnRs.cut2[[<span class="dv">1</span>]]), </span>
<span id="cb68-3"><a href="13-Others_marker_steps.html#cb68-3" tabindex="-1"></a>      <span class="at">REV2.ForwardReads =</span> <span class="fu">sapply</span>(REV2.orients, primerHits, <span class="at">fn =</span> fnFs.cut2[[<span class="dv">1</span>]]), </span>
<span id="cb68-4"><a href="13-Others_marker_steps.html#cb68-4" tabindex="-1"></a>      <span class="at">REV2.ReverseReads =</span> <span class="fu">sapply</span>(REV2.orients, primerHits, <span class="at">fn =</span> fnRs.cut2[[<span class="dv">1</span>]]))</span></code></pre></div>
<p>We now have no primers remaining in our file.</p>
</div>
</div>
<div id="checking-the-quality-of-your-data" class="section level2 hasAnchor" number="13.3">
<h2><span class="header-section-number">13.3</span> Checking the quality of your data<a href="13-Others_marker_steps.html#checking-the-quality-of-your-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Those of you who attended our quality Introduction to sequencing data and quality control course we used FastQC to check the quality of our data. DADA2 has its own quality control option, which plots a similar read length by quality figure.</p>
<p>To run first import the cutadapt files and then run the <code>plotQualityProfile</code> function</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="13-Others_marker_steps.html#cb69-1" tabindex="-1"></a><span class="co"># Specify the paths and file names the forward and reverse primer cleaned files </span></span>
<span id="cb69-2"><a href="13-Others_marker_steps.html#cb69-2" tabindex="-1"></a>cutFs2 <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path.cut2, </span>
<span id="cb69-3"><a href="13-Others_marker_steps.html#cb69-3" tabindex="-1"></a>                          <span class="at">pattern =</span> <span class="st">&quot;_L001_R1_001.fastq&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb69-4"><a href="13-Others_marker_steps.html#cb69-4" tabindex="-1"></a>cutRs2 <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path.cut2, </span>
<span id="cb69-5"><a href="13-Others_marker_steps.html#cb69-5" tabindex="-1"></a>                          <span class="at">pattern =</span> <span class="st">&quot;_L001_R2_001.fastq&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb69-6"><a href="13-Others_marker_steps.html#cb69-6" tabindex="-1"></a></span>
<span id="cb69-7"><a href="13-Others_marker_steps.html#cb69-7" tabindex="-1"></a><span class="co"># Extract sample names</span></span>
<span id="cb69-8"><a href="13-Others_marker_steps.html#cb69-8" tabindex="-1"></a>get.sample.name <span class="ot">&lt;-</span> <span class="cf">function</span>(fname) <span class="fu">strsplit</span>(<span class="fu">basename</span>(fname), <span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]</span>
<span id="cb69-9"><a href="13-Others_marker_steps.html#cb69-9" tabindex="-1"></a>sample.names <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">sapply</span>(cutFs2, get.sample.name))</span>
<span id="cb69-10"><a href="13-Others_marker_steps.html#cb69-10" tabindex="-1"></a><span class="fu">head</span>(sample.names)</span>
<span id="cb69-11"><a href="13-Others_marker_steps.html#cb69-11" tabindex="-1"></a></span>
<span id="cb69-12"><a href="13-Others_marker_steps.html#cb69-12" tabindex="-1"></a><span class="co"># check the quality for the first file</span></span>
<span id="cb69-13"><a href="13-Others_marker_steps.html#cb69-13" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(cutFs2[<span class="dv">1</span><span class="sc">:</span><span class="dv">1</span>])</span></code></pre></div>
<p>To interpret this plot, the gray-scale heatmap shows the the frequency of each quality score along the forward read length.The green line is the median quality score and the orange lines are the quartiles. The red line at the bottom of the plot represents the proportion of reads of that particular length.</p>
<p>The overall quality is good for our forward reads, as for the MiFish sequences. The majority of the reads are just over 100bp long, as we expect with this shorter amplicon.</p>
<p><img src="figures/quality_12Sv5.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
<div id="cleaning-your-data" class="section level2 hasAnchor" number="13.4">
<h2><span class="header-section-number">13.4</span> Cleaning your data<a href="13-Others_marker_steps.html#cleaning-your-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will now filter our data to remove any poor quality reads.</p>
<p>First set the path to a directory to store the filtered output files called <code>filtered2</code>.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="13-Others_marker_steps.html#cb70-1" tabindex="-1"></a>filtFs2 <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut2, <span class="st">&quot;../filtered2&quot;</span>, <span class="fu">basename</span>(cutFs2))</span>
<span id="cb70-2"><a href="13-Others_marker_steps.html#cb70-2" tabindex="-1"></a>filtRs2 <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut2, <span class="st">&quot;../filtered2&quot;</span>, <span class="fu">basename</span>(cutRs2))</span></code></pre></div>
<p>Now run <code>filterAndTrim</code>, using the standard filtering parameters:</p>
<ul>
<li><code>maxN=0</code> After truncation, sequences with more than 0 Ns will be discarded. (DADA2 requires sequences contain no Ns)</li>
<li><code>truncQ = 2</code> Truncate reads at the first instance of a quality score less than or equal to 2</li>
<li><code>rm.phix = TRUE</code> Discard reads that match against the phiX genome</li>
<li><code>maxEE=c(2, 2)</code> After truncation, reads with higher than 2 “expected errors” will be discarded</li>
<li><code>minLen = 60</code> Remove reads with length less than 60 (note these should have already been removed by cutadapt)</li>
<li><code>multithread = TRUE</code> input files are filtered in parallel</li>
</ul>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="13-Others_marker_steps.html#cb71-1" tabindex="-1"></a>out2 <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(cutFs2, filtFs2, cutRs2, filtRs2, <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">maxEE =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>), </span>
<span id="cb71-2"><a href="13-Others_marker_steps.html#cb71-2" tabindex="-1"></a>                     <span class="at">truncQ =</span> <span class="dv">2</span>, <span class="at">minLen =</span> <span class="dv">60</span>, <span class="at">rm.phix =</span> <span class="cn">TRUE</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>, </span>
<span id="cb71-3"><a href="13-Others_marker_steps.html#cb71-3" tabindex="-1"></a>                     <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb71-4"><a href="13-Others_marker_steps.html#cb71-4" tabindex="-1"></a>out2</span></code></pre></div>
<p>Some samples have low read numbers. These could be poor quality samples or the negatives controls. Some of the samples with low read numbers are different to those with low reads for MiFish-U. This might be because (through sample composition or by chance) the PCR or sequencing has worked better for one primer combination than the other for the same sample.</p>
</div>
<div id="identification-of-asvs" class="section level2 hasAnchor" number="13.5">
<h2><span class="header-section-number">13.5</span> Identification of ASVs<a href="13-Others_marker_steps.html#identification-of-asvs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="generate-an-error-model-1" class="section level3 hasAnchor" number="13.5.1">
<h3><span class="header-section-number">13.5.1</span> Generate an error model<a href="13-Others_marker_steps.html#generate-an-error-model-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First we need to model the error rates of our dataset using both the forward and reverse reads. Each dataset will have a specific error-signature with errors introduced by PCR amplification and sequencing.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="13-Others_marker_steps.html#cb72-1" tabindex="-1"></a>errF2 <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtFs2, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb72-2"><a href="13-Others_marker_steps.html#cb72-2" tabindex="-1"></a>errR2 <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtRs2, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>We can use the <code>plotErrors</code> function to check the estimated error rates.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="13-Others_marker_steps.html#cb73-1" tabindex="-1"></a><span class="fu">plotErrors</span>(errF2, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>Interpreting the plots, as a reminder:</p>
<ul>
<li><p>The error rates for each possible transition (e.g. A→C, A→G) are shown</p></li>
<li><p>Red line - expected based on the quality score</p></li>
<li><p>Black line - estimate</p></li>
<li><p>Black dots - observed</p></li>
</ul>
<p>The error plots look very similar to those for the MiFish-U sequences - the observed dots track well with the expected line, and the error rates drop with increasing quality scores. Sanity checks complete, we can proceed with the analysis.</p>
</div>
<div id="dereplication-1" class="section level3 hasAnchor" number="13.5.2">
<h3><span class="header-section-number">13.5.2</span> Dereplication<a href="13-Others_marker_steps.html#dereplication-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The next step is to dereplicate identical reads, after first checking again that all the samples are present after filtering.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="13-Others_marker_steps.html#cb74-1" tabindex="-1"></a>exists2 <span class="ot">&lt;-</span> <span class="fu">file.exists</span>(filtFs2) <span class="co"># check that all the samples </span></span>
<span id="cb74-2"><a href="13-Others_marker_steps.html#cb74-2" tabindex="-1"></a>                                    <span class="co">#are still present after filtering</span></span>
<span id="cb74-3"><a href="13-Others_marker_steps.html#cb74-3" tabindex="-1"></a>derepFs2 <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtFs2[exists2], <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb74-4"><a href="13-Others_marker_steps.html#cb74-4" tabindex="-1"></a>derepRs2 <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtRs2[exists2], <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb74-5"><a href="13-Others_marker_steps.html#cb74-5" tabindex="-1"></a><span class="co"># Name the derep-class objects by the sample names</span></span>
<span id="cb74-6"><a href="13-Others_marker_steps.html#cb74-6" tabindex="-1"></a><span class="fu">names</span>(derepFs2) <span class="ot">&lt;-</span> sample.names[exists2]</span>
<span id="cb74-7"><a href="13-Others_marker_steps.html#cb74-7" tabindex="-1"></a><span class="fu">names</span>(derepRs2) <span class="ot">&lt;-</span> sample.names[exists2]</span></code></pre></div>
</div>
<div id="inferrence-of-asvs-1" class="section level3 hasAnchor" number="13.5.3">
<h3><span class="header-section-number">13.5.3</span> Inferrence of ASVs<a href="13-Others_marker_steps.html#inferrence-of-asvs-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now we are ready to infer the ASVs in our dataset.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="13-Others_marker_steps.html#cb75-1" tabindex="-1"></a>dadaFs2 <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepFs2, <span class="at">err =</span> errF2, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb75-2"><a href="13-Others_marker_steps.html#cb75-2" tabindex="-1"></a>dadaRs2 <span class="ot">&lt;-</span> <span class="fu">dada</span>(derepRs2, <span class="at">err =</span> errR2, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="merging-paired-end-reads-1" class="section level3 hasAnchor" number="13.5.4">
<h3><span class="header-section-number">13.5.4</span> Merging paired end reads<a href="13-Others_marker_steps.html#merging-paired-end-reads-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>It is now time to merge the two files. By default the minimum overlap allowed between the two samples is 12 bp, and no mismatches are permitted.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="13-Others_marker_steps.html#cb76-1" tabindex="-1"></a>mergers2 <span class="ot">&lt;-</span> <span class="fu">mergePairs</span>(dadaFs2, derepFs2, dadaRs2, derepRs2, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="making-our-asv-matrix-1" class="section level3 hasAnchor" number="13.5.5">
<h3><span class="header-section-number">13.5.5</span> Making our ASV matrix<a href="13-Others_marker_steps.html#making-our-asv-matrix-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now it is time to make the counts table. Each column represents a single ASV and each row is an individual sample.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="13-Others_marker_steps.html#cb77-1" tabindex="-1"></a>seqtab2 <span class="ot">&lt;-</span> <span class="fu">makeSequenceTable</span>(mergers2)</span>
<span id="cb77-2"><a href="13-Others_marker_steps.html#cb77-2" tabindex="-1"></a><span class="fu">dim</span>(seqtab2) </span></code></pre></div>
<p>There are 583 ASVs in our matrix.</p>
</div>
<div id="chimera-detection-and-removal-1" class="section level3 hasAnchor" number="13.5.6">
<h3><span class="header-section-number">13.5.6</span> Chimera detection and removal<a href="13-Others_marker_steps.html#chimera-detection-and-removal-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The last step in generating our ASV matrix is to detect and remove any chimeric sequences.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="13-Others_marker_steps.html#cb78-1" tabindex="-1"></a>seqtab2.nochim <span class="ot">&lt;-</span> <span class="fu">removeBimeraDenovo</span>(seqtab2, <span class="at">method=</span><span class="st">&quot;consensus&quot;</span>, </span>
<span id="cb78-2"><a href="13-Others_marker_steps.html#cb78-2" tabindex="-1"></a>                                     <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb78-3"><a href="13-Others_marker_steps.html#cb78-3" tabindex="-1"></a></span>
<span id="cb78-4"><a href="13-Others_marker_steps.html#cb78-4" tabindex="-1"></a><span class="fu">dim</span>(seqtab2.nochim)</span></code></pre></div>
<p>We have 212 ASVs remaining after filtering out chimeras. There are fewer ASVs for the 12S-V5 data than for the MiFish data. This could be as it is a shorter amplicon and therefore has less potential for variation, or potentially the primers could be less efficient at/have more mismatches when amplifying some taxa.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="13-Others_marker_steps.html#cb79-1" tabindex="-1"></a><span class="fu">sum</span>(seqtab2.nochim)<span class="sc">/</span><span class="fu">sum</span>(seqtab2)</span></code></pre></div>
<p>In these data ~ 7.5% of the merged sequence reads were identified as chimeric.</p>
<p>CHeck the range of ASV lengths:</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="13-Others_marker_steps.html#cb80-1" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab2.nochim)))</span></code></pre></div>
</div>
<div id="sequence-tracking-sanity-check-1" class="section level3 hasAnchor" number="13.5.7">
<h3><span class="header-section-number">13.5.7</span> Sequence tracking sanity check<a href="13-Others_marker_steps.html#sequence-tracking-sanity-check-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The last thing to do in this section is to track the number of sequences through the pipeline to check whether everything has run as expected and whether there are any steps where we loose a disproportionate number of sequences. If we end up with too few reads to run further analysis we can use this table to identify any step which might require further investigation and optimisation.</p>
<p>We will generate the number of sequences using the <code>getN</code> function. You can copy and paste this from the previous notebook.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="13-Others_marker_steps.html#cb81-1" tabindex="-1"></a>getN <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">sum</span>(<span class="fu">getUniques</span>(x)) </span>
<span id="cb81-2"><a href="13-Others_marker_steps.html#cb81-2" tabindex="-1"></a></span>
<span id="cb81-3"><a href="13-Others_marker_steps.html#cb81-3" tabindex="-1"></a>track2 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(out2, <span class="fu">sapply</span>(dadaFs2, getN), </span>
<span id="cb81-4"><a href="13-Others_marker_steps.html#cb81-4" tabindex="-1"></a>                <span class="fu">sapply</span>(dadaRs2, getN), </span>
<span id="cb81-5"><a href="13-Others_marker_steps.html#cb81-5" tabindex="-1"></a>                <span class="fu">sapply</span>(mergers2, getN), </span>
<span id="cb81-6"><a href="13-Others_marker_steps.html#cb81-6" tabindex="-1"></a>                <span class="fu">rowSums</span>(seqtab2.nochim))</span>
<span id="cb81-7"><a href="13-Others_marker_steps.html#cb81-7" tabindex="-1"></a><span class="fu">colnames</span>(track2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;input&quot;</span>, <span class="st">&quot;filtered&quot;</span>, </span>
<span id="cb81-8"><a href="13-Others_marker_steps.html#cb81-8" tabindex="-1"></a>                      <span class="st">&quot;denoisedF&quot;</span>, <span class="st">&quot;denoisedR&quot;</span>, </span>
<span id="cb81-9"><a href="13-Others_marker_steps.html#cb81-9" tabindex="-1"></a>                      <span class="st">&quot;merged&quot;</span>, <span class="st">&quot;nonchim&quot;</span>)</span>
<span id="cb81-10"><a href="13-Others_marker_steps.html#cb81-10" tabindex="-1"></a><span class="fu">rownames</span>(track2) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb81-11"><a href="13-Others_marker_steps.html#cb81-11" tabindex="-1"></a>track2</span></code></pre></div>
<p>There are five samples with fewer than 2000 reads. However, our eDNA filter negative control (S62) has &gt;100000 sequences.</p>
</div>
</div>
<div id="assigning-taxonomy" class="section level2 hasAnchor" number="13.6">
<h2><span class="header-section-number">13.6</span> Assigning taxonomy<a href="13-Others_marker_steps.html#assigning-taxonomy" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To assign taxonomy we will use a custom reference database containing fish sequences available for the 12S-V5 amplicon region for species found in Lake Tanganyika and its broader catchment area.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="13-Others_marker_steps.html#cb82-1" tabindex="-1"></a>taxa12S <span class="ot">&lt;-</span> <span class="fu">assignTaxonomy</span>(seqtab2.nochim, </span>
<span id="cb82-2"><a href="13-Others_marker_steps.html#cb82-2" tabindex="-1"></a>                          <span class="st">&quot;/pub39/tea/nsc006/NEOF/metabarcoding_workshop/taxonomy/12S-V5_Reference_Database.fasta&quot;</span>, </span>
<span id="cb82-3"><a href="13-Others_marker_steps.html#cb82-3" tabindex="-1"></a>                          <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">verbose =</span> T)</span>
<span id="cb82-4"><a href="13-Others_marker_steps.html#cb82-4" tabindex="-1"></a></span>
<span id="cb82-5"><a href="13-Others_marker_steps.html#cb82-5" tabindex="-1"></a>taxa12S.print <span class="ot">&lt;-</span> taxa12S</span>
<span id="cb82-6"><a href="13-Others_marker_steps.html#cb82-6" tabindex="-1"></a><span class="fu">rownames</span>(taxa12S.print) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb82-7"><a href="13-Others_marker_steps.html#cb82-7" tabindex="-1"></a><span class="fu">head</span>(taxa12S.print)</span></code></pre></div>
<p>For this marker all of the first six ASVs have been assigned to class Actinopteri, four to the family-level (all Cichlidae), and one to the species-level (<em>Pseudosimochromis babaulti</em>).</p>
</div>
<div id="further-analysis" class="section level2 hasAnchor" number="13.7">
<h2><span class="header-section-number">13.7</span> Further analysis<a href="13-Others_marker_steps.html#further-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As the 12S-V5 sequences are all from the same samples then we can use the same meta data table as for the MiFish analysis.</p>
<p>Before we start we will clean up the dataset. One of our negative controls seems to be mostly clean of contaminants (S72, with only 9 reads assigned). We will remove this from further analysis, but retain our filter control negative (S62) for further investigation.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="13-Others_marker_steps.html#cb83-1" tabindex="-1"></a>meta<span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&quot;/pub39/tea/nsc006/NEOF/metabarcoding_workshop/sample_info.csv&quot;</span>, </span>
<span id="cb83-2"><a href="13-Others_marker_steps.html#cb83-2" tabindex="-1"></a>               <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb83-3"><a href="13-Others_marker_steps.html#cb83-3" tabindex="-1"></a></span>
<span id="cb83-4"><a href="13-Others_marker_steps.html#cb83-4" tabindex="-1"></a>seqtab2.rmcontrol<span class="ot">&lt;-</span>seqtab2.nochim[<span class="sc">-</span><span class="dv">31</span>,]</span>
<span id="cb83-5"><a href="13-Others_marker_steps.html#cb83-5" tabindex="-1"></a>meta2.rmcontrol<span class="ot">&lt;-</span>meta[<span class="sc">-</span><span class="dv">31</span>,]</span></code></pre></div>
<p>We will also remove samples which appear to have failed/have very low sequence numbers in our dataset.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="13-Others_marker_steps.html#cb84-1" tabindex="-1"></a><span class="co">#Check number of sequences per sample</span></span>
<span id="cb84-2"><a href="13-Others_marker_steps.html#cb84-2" tabindex="-1"></a><span class="fu">rowSums</span>(seqtab2.rmcontrol)</span>
<span id="cb84-3"><a href="13-Others_marker_steps.html#cb84-3" tabindex="-1"></a><span class="co"># S16, S25, S40 and S43 all have low sequence counts. So we will remove these rows.</span></span>
<span id="cb84-4"><a href="13-Others_marker_steps.html#cb84-4" tabindex="-1"></a>seqtab2.rmlow<span class="ot">&lt;-</span>seqtab2.rmcontrol[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">13</span>,<span class="dv">16</span>),]</span>
<span id="cb84-5"><a href="13-Others_marker_steps.html#cb84-5" tabindex="-1"></a>meta2.rmlow<span class="ot">&lt;-</span>meta2.rmcontrol[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">13</span>,<span class="dv">16</span>),]</span>
<span id="cb84-6"><a href="13-Others_marker_steps.html#cb84-6" tabindex="-1"></a></span>
<span id="cb84-7"><a href="13-Others_marker_steps.html#cb84-7" tabindex="-1"></a><span class="co"># Print the minimum sequence number in one sample.</span></span>
<span id="cb84-8"><a href="13-Others_marker_steps.html#cb84-8" tabindex="-1"></a><span class="fu">min</span>(<span class="fu">rowSums</span>(seqtab2.rmlow))</span>
<span id="cb84-9"><a href="13-Others_marker_steps.html#cb84-9" tabindex="-1"></a><span class="co"># The lowest number of sequences in one sample is now 26841.</span></span></code></pre></div>
</div>
<div id="rarefaction-curves-1" class="section level2 hasAnchor" number="13.8">
<h2><span class="header-section-number">13.8</span> Rarefaction curves<a href="13-Others_marker_steps.html#rarefaction-curves-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The more deeply we sequence a sample the more species we discover, until this accumulation levels off as we have found all or as many ASVs/species as we are going to find. This becomes a problem as samples are often sequenced at different depths so will have reached different points in the curve. This is a common difficulty as pooling and sequencing equal amounts of each sample can be tricky.</p>
<p>One way to visualise this is to plot a rarefaction curve for each sample.</p>
<p>Note: remember to load the required packages first.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="13-Others_marker_steps.html#cb85-1" tabindex="-1"></a><span class="fu">rarecurve</span>(seqtab2.rmlow, <span class="at">step=</span><span class="dv">100</span>, <span class="at">col=</span>meta2.rmlow<span class="sc">$</span>COLOUR, <span class="at">lwd=</span><span class="dv">2</span>, <span class="at">ylab=</span><span class="st">&quot;ASVs&quot;</span>, <span class="at">label=</span>F)</span>
<span id="cb85-2"><a href="13-Others_marker_steps.html#cb85-2" tabindex="-1"></a><span class="co"># add a vertical line to represent the fewest sequences in any sample</span></span>
<span id="cb85-3"><a href="13-Others_marker_steps.html#cb85-3" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v=</span>(<span class="fu">min</span>(<span class="fu">rowSums</span>(seqtab2.rmlow))))</span></code></pre></div>
<p><img src="figures/rarefaction_12Sv5.png" width="80%" style="display: block; margin: auto;" /></p>
<p>You can see that each sample levels off at a different sequencing depth and each sample has been sequenced at different depths.</p>
<p>Overall the number of ASVs is lower, but as with MiFish the ASV numbers level off for almost all samples before the minimum sequencing depth is reached.</p>
</div>
<div id="alpha-diversity-1" class="section level2 hasAnchor" number="13.9">
<h2><span class="header-section-number">13.9</span> Alpha diversity<a href="13-Others_marker_steps.html#alpha-diversity-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Measures of alpha diversity are used to describe diversity within a sample.</p>
<p>First make a phyloseq object. To do this we first read in our ASV, taxonomy and metadata tables before making the plyloseq object <code>phylo2</code>.</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="13-Others_marker_steps.html#cb86-1" tabindex="-1"></a>seqtab2.rmlow2<span class="ot">&lt;-</span><span class="fu">t</span>(<span class="fu">as.data.frame</span>(seqtab2.rmlow))</span>
<span id="cb86-2"><a href="13-Others_marker_steps.html#cb86-2" tabindex="-1"></a>phylo_asv2 <span class="ot">&lt;-</span> <span class="fu">otu_table</span>(seqtab2.rmlow2, <span class="at">taxa_are_rows=</span><span class="cn">TRUE</span>)</span>
<span id="cb86-3"><a href="13-Others_marker_steps.html#cb86-3" tabindex="-1"></a>phylo_tax2 <span class="ot">&lt;-</span> <span class="fu">tax_table</span>(taxa12S)</span>
<span id="cb86-4"><a href="13-Others_marker_steps.html#cb86-4" tabindex="-1"></a>phylo_samples2 <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(meta2.rmlow)</span>
<span id="cb86-5"><a href="13-Others_marker_steps.html#cb86-5" tabindex="-1"></a></span>
<span id="cb86-6"><a href="13-Others_marker_steps.html#cb86-6" tabindex="-1"></a>phylo2 <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(phylo_asv2, phylo_tax2, phylo_samples2)</span>
<span id="cb86-7"><a href="13-Others_marker_steps.html#cb86-7" tabindex="-1"></a></span>
<span id="cb86-8"><a href="13-Others_marker_steps.html#cb86-8" tabindex="-1"></a><span class="fu">sample_names</span>(phylo2)</span>
<span id="cb86-9"><a href="13-Others_marker_steps.html#cb86-9" tabindex="-1"></a><span class="fu">rank_names</span>(phylo2)</span>
<span id="cb86-10"><a href="13-Others_marker_steps.html#cb86-10" tabindex="-1"></a><span class="fu">sample_variables</span>(phylo2) </span></code></pre></div>
<p>Plot the two alpha diversity metrics.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="13-Others_marker_steps.html#cb87-1" tabindex="-1"></a><span class="fu">plot_richness</span>(phylo2, <span class="at">measures=</span><span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>), </span>
<span id="cb87-2"><a href="13-Others_marker_steps.html#cb87-2" tabindex="-1"></a>              <span class="at">color =</span> <span class="st">&quot;SITE&quot;</span>)</span>
<span id="cb87-3"><a href="13-Others_marker_steps.html#cb87-3" tabindex="-1"></a><span class="fu">plot_richness</span>(phylo2, <span class="at">x=</span><span class="st">&quot;SITE&quot;</span>, <span class="at">measures=</span><span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>), </span>
<span id="cb87-4"><a href="13-Others_marker_steps.html#cb87-4" tabindex="-1"></a>              <span class="at">color =</span> <span class="st">&quot;SITE&quot;</span>) <span class="sc">+</span> <span class="fu">geom_boxplot</span>()</span></code></pre></div>
<p><img src="figures/alpha1_12Sv5.png" width="80%" style="display: block; margin: auto;" /></p>
<p><img src="figures/alpha2_12Sv5.png" width="80%" style="display: block; margin: auto;" /></p>
<p>The first figure plots the diversity measure per sample and colours the output by site. The second figure combines the replicates to plot as a boxplot. We can see that both the Shannon and Simpson diversity is lower for 12S-V5, and variability within sampling sites is generally higher. The filter negative control has very low diversity (only S41 is lower).</p>
</div>
<div id="beta-diversity-1" class="section level2 hasAnchor" number="13.10">
<h2><span class="header-section-number">13.10</span> Beta diversity<a href="13-Others_marker_steps.html#beta-diversity-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Beta diversity compares the difference in diversity between two sites, or to put it another way it calculates the number of species that are not the same in the two sites.</p>
<p>We will normalise the data before running the beta diversity calculation. We will transform the data into proportions to be used for Bray-Curtis distances.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="13-Others_marker_steps.html#cb88-1" tabindex="-1"></a>ps.prop2 <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(phylo2, <span class="cf">function</span>(otu) otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span></code></pre></div>
<p>We then generate and plot the NMDS (Non-metric MultiDimenstional Scaling) using Bray-Curtis distances.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="13-Others_marker_steps.html#cb89-1" tabindex="-1"></a>ord.nmds.bray2 <span class="ot">&lt;-</span> <span class="fu">ordinate</span>(ps.prop2, <span class="at">method=</span><span class="st">&quot;NMDS&quot;</span>, <span class="at">distance=</span><span class="st">&quot;bray&quot;</span>)</span>
<span id="cb89-2"><a href="13-Others_marker_steps.html#cb89-2" tabindex="-1"></a><span class="fu">plot_ordination</span>(ps.prop2, ord.nmds.bray2, </span>
<span id="cb89-3"><a href="13-Others_marker_steps.html#cb89-3" tabindex="-1"></a>                <span class="at">color=</span><span class="st">&quot;SITE&quot;</span>, <span class="at">title=</span><span class="st">&quot;Bray NMDS 12S-V5&quot;</span>)</span></code></pre></div>
<p><img src="figures/bray_12Sv5.png" width="80%" style="display: block; margin: auto;" /></p>
<p>This is a very different pattern to that for the MiFish data - all of the lake site samples are clustered together, as our the aquarium samples, and the filter control is very distant to both of these groupings.</p>
<p>Before deciding how to proceed let’s plot the proportion of ASV sequences within each sample that belong to different taxonomic families.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="13-Others_marker_steps.html#cb90-1" tabindex="-1"></a><span class="fu">plot_bar</span>(ps.prop2, <span class="at">fill =</span> <span class="st">&quot;Family&quot;</span>)<span class="sc">+</span></span>
<span id="cb90-2"><a href="13-Others_marker_steps.html#cb90-2" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">color=</span>Family, <span class="at">fill=</span>Family), </span>
<span id="cb90-3"><a href="13-Others_marker_steps.html#cb90-3" tabindex="-1"></a>           <span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">position=</span><span class="st">&quot;stack&quot;</span>)<span class="sc">+</span></span>
<span id="cb90-4"><a href="13-Others_marker_steps.html#cb90-4" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>SITE, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>)</span></code></pre></div>
<p><img src="figures/Family_abundance_12Sv5.png" width="80%" style="display: block; margin: auto;" /></p>
<p>There are a far higher proportion of ASVs unassigned with this primer combination (the grey NA portion of the bars). The filter control sample has almost entirely unassigned ASVs. Given the low diversity, high differentiation from other samples and lack of ASVs of the fish sequences of interest we can remove this negative control, and re-plot the NMDS. Let’s remove the aquarium samples as well, and re-plot the community similarity for the lake filter samples.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="13-Others_marker_steps.html#cb91-1" tabindex="-1"></a><span class="co">#To remove specific samples from a phyloseq object</span></span>
<span id="cb91-2"><a href="13-Others_marker_steps.html#cb91-2" tabindex="-1"></a>SamplesToRemove <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;S62&quot;</span>, <span class="st">&quot;S63&quot;</span>, <span class="st">&quot;S64&quot;</span>, <span class="st">&quot;S65&quot;</span>, <span class="st">&quot;S66&quot;</span>, <span class="st">&quot;S67&quot;</span>)</span>
<span id="cb91-3"><a href="13-Others_marker_steps.html#cb91-3" tabindex="-1"></a>phylo2.nofilt <span class="ot">&lt;-</span> <span class="fu">subset_samples</span>(phylo2, <span class="sc">!</span>(<span class="fu">sample_names</span>(phylo2) <span class="sc">%in%</span> SamplesToRemove))</span>
<span id="cb91-4"><a href="13-Others_marker_steps.html#cb91-4" tabindex="-1"></a></span>
<span id="cb91-5"><a href="13-Others_marker_steps.html#cb91-5" tabindex="-1"></a><span class="co">#Transform the data into proportions and re-plot NMDS</span></span>
<span id="cb91-6"><a href="13-Others_marker_steps.html#cb91-6" tabindex="-1"></a>ps.prop.nofilt <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(phylo2.nofilt, <span class="cf">function</span>(otu) otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb91-7"><a href="13-Others_marker_steps.html#cb91-7" tabindex="-1"></a>ord.nmds.bray.nofilt <span class="ot">&lt;-</span> <span class="fu">ordinate</span>(ps.prop.nofilt, <span class="at">method=</span><span class="st">&quot;NMDS&quot;</span>, <span class="at">distance=</span><span class="st">&quot;bray&quot;</span>)</span>
<span id="cb91-8"><a href="13-Others_marker_steps.html#cb91-8" tabindex="-1"></a><span class="fu">plot_ordination</span>(ps.prop.nofilt, ord.nmds.bray.nofilt, <span class="at">color=</span><span class="st">&quot;SITE&quot;</span>, </span>
<span id="cb91-9"><a href="13-Others_marker_steps.html#cb91-9" tabindex="-1"></a>                <span class="at">title=</span><span class="st">&quot;Bray NMDS 12S-V5 lake samples&quot;</span>)</span></code></pre></div>
<p><img src="figures/bray_12Sv5_lake.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Lastly, we can calculate the Bray–Curtis distances using the <code>distance</code> function and perform a PERMANOVA (permutational multivariate analysis of variance) using the <code>adonis</code> function from Vegan to check whether there is any significantly different separation of samples by site.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="13-Others_marker_steps.html#cb92-1" tabindex="-1"></a>bray.dist.nofilt<span class="ot">&lt;-</span><span class="fu">distance</span>(ps.prop.nofilt, <span class="at">method=</span><span class="st">&quot;bray&quot;</span>)</span>
<span id="cb92-2"><a href="13-Others_marker_steps.html#cb92-2" tabindex="-1"></a>sampledf.nofilt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(phylo2.nofilt))</span>
<span id="cb92-3"><a href="13-Others_marker_steps.html#cb92-3" tabindex="-1"></a><span class="fu">adonis2</span>(bray.dist.nofilt <span class="sc">~</span> SITE, <span class="at">data =</span> sampledf.nofilt)</span></code></pre></div>
<p>The PERMANOVA results suggest that there is also a statistical difference in communities between sites with the 12S-V5 data.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="12-Analysing_another_marker.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
