<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Genome pre-assembly | 05-Pre_assembly.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Genome pre-assembly | 05-Pre_assembly.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Genome pre-assembly | 05-Pre_assembly.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  


<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>Eukaryote genome assembly</a></li>

<li class="divider"></li>
<li class="part"><span><b>Assembly steps</b></span></li>
<li class="chapter" data-level="1" data-path=""><a href="#preassembly"><i class="fa fa-check"></i><b>1</b> Genome pre-assembly</a>
<ul>
<li class="chapter" data-level="1.1" data-path=""><a href="#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction</a></li>
<li class="chapter" data-level="1.2" data-path=""><a href="#obtaining-the-read-data"><i class="fa fa-check"></i><b>1.2</b> Obtaining the read data</a></li>
<li class="chapter" data-level="1.3" data-path=""><a href="#counting-reads-and-bases"><i class="fa fa-check"></i><b>1.3</b> Counting reads and bases</a></li>
<li class="chapter" data-level="1.4" data-path=""><a href="#counting-and-analysing-k-mers"><i class="fa fa-check"></i><b>1.4</b> Counting and analysing k-mers</a></li>
<li class="chapter" data-level="1.5" data-path=""><a href="#summary"><i class="fa fa-check"></i><b>1.5</b> Summary</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<!--bookdown:title:end-->
<!--bookdown:title:start-->



<div id="preassembly" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Genome pre-assembly<a href="#preassembly" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/magnify_glass_27AUG2020.png" style="width:200px" />
</center>
<div id="introduction" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Introduction<a href="#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Assembling a eukaryotic genome using short reads used to be a time consuming process. One had to do a lot of QC checks to see if the reads had any quality issues or remaining adapters. With PacBio Hifi reads, the quality and length and adapters are not usually an issue as the PacBio HiFi platform has already done the circular consensus step so what you get is high quality long read data.</p>
<p>However, it's good to have a sense of whether your read coverage is adequate, and whether your genome is haploid, diploid or polyploid. To do this we analyse the k-mers present in the HiFi reads. The broad steps are to first run a k-mer counter, and then run the resulting histograms through a tool like <code>GenomeScope</code>.</p>
</div>
<div id="obtaining-the-read-data" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Obtaining the read data<a href="#obtaining-the-read-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/Oscheius_dolichura.png" style="width:200px" />
</center>
<p>We are going to assemble the genome of a nematode <em>Oscheius onirici</em> which we expect to be 60-100 million bases or Megabases (Mb) in size because other <em>Oscheius</em> species are approximately that size.</p>
<p>First, we need to create a directory to put the data in and then change the directory to it.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> ~/eukaryote_assembly/1-preassembly</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="bu">cd</span> ~/eukaryote_assembly/1-preassembly</span></code></pre></div>
<p>Now we can generate a symbolic link (i.e. shortcut) to the raw sequence data file, which will appear in the current directory:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub14/tea/nsc206/NEOF/Euk_assembly/data/ERR7979900.fastq</span></code></pre></div>
<p>If you would like to know more about the <code>ln</code> command please check out this <a href="https://linuxize.com/post/how-to-create-symbolic-links-in-linux-using-the-ln-command/">guide</a>.</p>
<p>Now, check the symbolic link is in your current directory:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>The file name should appear in blue. If it appears in red with a black background this indicates that the file has not been found. If you see this, check the file path and name carefully for typos and try again.</p>
<p>The file has one run of PacBio HiFi data for <em>Oscheius onirici</em>.</p>
</div>
<div id="counting-reads-and-bases" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Counting reads and bases<a href="#counting-reads-and-bases" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/count.png" style="width:200px; background:white; border-radius:15px" />
</center>
<p>A simple QC step is to count how many reads and bases are in the raw data. To get an estimate of how much data is in this file we will use <a href="https://bioinf.shenwei.me/seqkit/"><code>seqkit</code></a>:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">seqkit</span> stats ERR7979900.fastq</span></code></pre></div>
<center>
<img src="figures/seqkit_stats.png" style="width:1000px" />
</center>
<p>The output shows that there are almost 1 million sequences (num_seqs = 976,950) totalling 10,800 Mb of raw sequence data (sum_len = 10,801,040,319). This means that for a genome of size 60-100 Mb, we have at least 10,800 / 100 = 108X depth. The "X" can be read as "times" - i.e. the number of times - we expect each base in the genome assembly to be covered by a raw sequence read. Some people also refer to sequencing depth as sequencing coverage. A good rule of thumb for PacBio HiFi sequencing is that we should have at least 30X sequencing depth for a good assembly, so a minimum of 108X depth is more than fine for our needs and we can proceed.</p>
</div>
<div id="counting-and-analysing-k-mers" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> Counting and analysing k-mers<a href="#counting-and-analysing-k-mers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/k-mer_pic.png" style="width:600px" />
</center>
<p>Another QC step that can be run very fast is k-mer analysis. K-mers are short, overlapping sub-sequences of length k from a longer sequence. For example, here are all the 3-mers (i.e. k=3) from the read AACGATT: AAC ACG CGA GAT ATT</p>
<p>There are many great reviews and tutorials on why k-mers work and how you can use them to analyse read sets and assemblies. But for now, we're going to use them in a simple way to estimate the size of the genome and its heterozygosity. We will use the software <a href="https://github.com/refresh-bio/KMC"><code>kmc</code></a> to do this:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">kmc</span> <span class="at">-cs2000</span> <span class="at">-k21</span> ERR7979900.fastq ERR7979900k21 .</span></code></pre></div>
<ul>
<li><code>-cs&lt;value&gt;</code> - maximal value of a counter (default: 255). When you have ~108X depth (from just counting reads and bases), it is a good idea to bump this up to 2,000 or even 10,000</li>
<li><code>-k&lt;len&gt;</code> - k-mer length (k from 1 to 256; default: 25). We picked k21 to match the default for GenomeScope which we will use in the next step. You can try different values like k25 or k31. Just remember to change the value in the GenomeScope input form as well. The end result should be roughly the same</li>
<li><code>ERR7979900k21</code> is the prefix of the output files</li>
<li><code>.</code> Use the current directory for storing temporary files</li>
</ul>
<p>This step should take about 2-3 minutes to count all the k-mers and store them in kmc's own binary file formats with the <code>.kmc_suf</code> and <code>.kmc_pre</code> extensions.</p>
<p>Now use the <code>kmc</code>'s <code>transform</code> tool to extract a k-mer frequency histogram:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">kmc_tools</span> transform ERR7979900k21 histogram ERR7979900k21.histo</span></code></pre></div>
<ul>
<li><code>transform</code> is the kmc_tools function</li>
<li><code>ERR7979900k21</code> is the prefix of the input files</li>
<li><code>histogram</code> is the command to kmc_tools to create that kind of output</li>
<li><code>ERR7979900k21.histo</code> is the output histogram file name</li>
</ul>
<p>The output file <code>ERR7979900k21.histo</code> is a simple text file with two columns: Column 1 has the number of copies of a k-mer of size k (21 in this case), and column 2 has the frequency, i.e. how many different k-mers have that copy number.</p>
<p>The next step is to view the histogram.
But first we will talk about <strong>background</strong> and <strong>foreground</strong> processes.</p>
<p>Normally when you run a command it will run the process in the <strong>foreground</strong>.
This means your command line will be locked whilst the process is running.
You will only be able to type a new command in the terminal once the <strong>foreground</strong> process is finished.</p>
<p>This is normally wanted but not always, for example when you run a process like <code>firefox</code>.
To run a process in the <strong>background</strong>, so you can type and run new commands, you can add the symbol <code>&amp;</code> to the end of a command.</p>
<center>
<img src="figures/firefox.png" style="width:100px" />
</center>
<p>Using <code>firefox</code> open the GenomeScope web page and upload the <code>.histo</code> file:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">firefox</span> http://genomescope.org/genomescope2.0/ <span class="kw">&amp;</span></span></code></pre></div>
<p>To upload this file click in the box where it says "Click or drop .histo file here to upload”. It may take a little while for the “File Upload” pop-up to appear. When it does make sure you are in your “Home” directory and double-click on your "1-preassembly" directory (in the “eukaryote_assembly” directory) and choose the <code>ERR7979900k21.histo</code> file. You can leave the rest of the fields as they are and then click <strong>Submit</strong>, then <strong>Continue</strong>. After a minute or two of processing the plots will appear in the browser.</p>
<p>The first plot in the output should look like this:</p>
<center>
<img src="figures/genomescope_plot1_bad.png" style="width:900px; border-radius:15px" />
</center>
<p>According to this plot, the length of the genome estimated from k-mers is <strong>len:258,425bp</strong> - which is way too small for a nematode genome that we expect to be around 60-100 Mb. Something has clearly gone wrong!</p>
<p>The problem is that, because there is so much data in our raw sequence file, the k-mer coverage on the x-axis goes all the way up to 3000, and GenomeScope automatically decided that the little peak near 2000X coverage (on the x axis) was the right peak for the genome assembly. The area under that peak was only ~258 kb, so GenomeScope reported that, and it counted the real peak of the data around 100X on the x axis as low-frequency errors, shown in the orange curve.</p>
<p>This may happen when you are analysing your own data as well. If this happens, we can force GenomeScope to disregard the high coverage k-mers (which are likely to be mitochondria or ribosomal RNA repeats) by setting <strong>Max k-mer coverage: 1000</strong> in this case.</p>
<p>Open a NEW browser tab, type <a href="http://genomescope.org/genomescope2.0/" class="uri">http://genomescope.org/genomescope2.0/</a> into the address bar and upload the <code>ERR7979900k21.histo</code> file again. This time, set the <strong>Max k-mer coverage</strong> to 1000 and then click Submit and Continue.</p>
<p>After a minute or two you should see a set of plots, and the first plot should look like this:</p>
<center>
<img src="figures/genomescope_plot2_good.png" style="width:900px; border-radius:15px" />
</center>
<p>This time the genome length estimate of <strong>len:95,828,752bp</strong> seems more in line with what we expect. GenomeScope also reports <strong>aa:99.8%</strong> i.e 99.8% of this dataset is homozygous and the plot itself is a single peak, which means that the two copies of each diploid chromosome are almost identical. If GenomeScope had reported a homozygosity % like <strong>aa:97%</strong> and shown two peaks, that would have meant a higher level of heterozygosity. Low heterozygosity in diploid or polyploid samples can aid the assembly process and so inbred lines or strains are often used when this is practical to do so. For most taxa, however, this is not possible and a higher level of heterozygosity would be expected.</p>
<p>For more details, see <a href="https://www.nature.com/articles/s41467-020-14998-3">GenomeScope 2.0 and Smudgeplot for reference-free profiling of polyploid genomes</a></p>
</div>
<div id="summary" class="section level2 hasAnchor" number="1.5">
<h2><span class="header-section-number">1.5</span> Summary<a href="#summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/sum_black.png" style="width:200px" />
</center>
<p>Counting reads and bases and k-mers in a set of read files can give you a good sense of whether your genome assembly is going to work, and whether you need to worry about low coverage, or highly erroneous k-mers, etc.</p>
<p>The length of the genome assembly should be roughly what you expect the genome size to be. It may not be the exact size of your species of interest if the sample also includes DNA from other co-bionts such as contaminants or parasites or endosymbionts.</p>
</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>


    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": false,
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
