<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Genome assembly | Eukaryote genome assembly</title>
  <meta name="description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Genome assembly | Eukaryote genome assembly" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Genome assembly | Eukaryote genome assembly" />
  
  <meta name="twitter:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sujai Kumar &amp; Helen Hipperson" />


<meta name="date" content="2023-04-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="05-Pre_assembly.html"/>
<link rel="next" href="07-Genome_assessment.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Eukaryote genome assembly</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Eukaryote_genome_assembly.html"><a href="01-Eukaryote_genome_assembly.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="chapter" data-level="5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html"><i class="fa fa-check"></i><b>5</b> Genome pre-assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#introduction-1"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#obtaining-the-read-data"><i class="fa fa-check"></i><b>5.2</b> Obtaining the read data</a></li>
<li class="chapter" data-level="5.3" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-reads-and-bases"><i class="fa fa-check"></i><b>5.3</b> Counting reads and bases</a></li>
<li class="chapter" data-level="5.4" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-and-analysing-k-mers"><i class="fa fa-check"></i><b>5.4</b> Counting and analysing k-mers</a></li>
<li class="chapter" data-level="5.5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#summary"><i class="fa fa-check"></i><b>5.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Assembly.html"><a href="06-Assembly.html"><i class="fa fa-check"></i><b>6</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Assembly.html"><a href="06-Assembly.html#overview-1"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="06-Assembly.html"><a href="06-Assembly.html#downsample-raw-data"><i class="fa fa-check"></i><b>6.2</b> Downsample raw data</a></li>
<li class="chapter" data-level="6.3" data-path="06-Assembly.html"><a href="06-Assembly.html#run-the-hifiasm-assembler"><i class="fa fa-check"></i><b>6.3</b> Run the Hifiasm assembler</a></li>
<li class="chapter" data-level="6.4" data-path="06-Assembly.html"><a href="06-Assembly.html#summary-1"><i class="fa fa-check"></i><b>6.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html"><i class="fa fa-check"></i><b>7</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#obtaining-the-full-assembly"><i class="fa fa-check"></i><b>7.1</b> Obtaining the full assembly</a></li>
<li class="chapter" data-level="7.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#basic-assembly-statistics"><i class="fa fa-check"></i><b>7.2</b> Basic assembly statistics</a></li>
<li class="chapter" data-level="7.3" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#check-for-contamination"><i class="fa fa-check"></i><b>7.3</b> Check for contamination</a></li>
<li class="chapter" data-level="7.4" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco"><i class="fa fa-check"></i><b>7.4</b> BUSCO</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-lineages"><i class="fa fa-check"></i><b>7.4.1</b> BUSCO lineages</a></li>
<li class="chapter" data-level="7.4.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#run-busco"><i class="fa fa-check"></i><b>7.4.2</b> Run BUSCO</a></li>
<li class="chapter" data-level="7.4.3" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-output"><i class="fa fa-check"></i><b>7.4.3</b> BUSCO output</a></li>
<li class="chapter" data-level="7.4.4" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-questions"><i class="fa fa-check"></i><b>7.4.4</b> BUSCO questions</a></li>
<li class="chapter" data-level="7.4.5" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-summary"><i class="fa fa-check"></i><b>7.4.5</b> BUSCO summary</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#purging-duplications-using-purge_dups"><i class="fa fa-check"></i><b>7.5</b> Purging duplications using purge_dups</a></li>
<li class="chapter" data-level="7.6" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#assessment-recap"><i class="fa fa-check"></i><b>7.6</b> Assessment recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html"><i class="fa fa-check"></i><b>8</b> Mitochondrial assembly</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#introduction-2"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#set-up"><i class="fa fa-check"></i><b>8.2</b> Set up</a></li>
<li class="chapter" data-level="8.3" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#run-mitohifi"><i class="fa fa-check"></i><b>8.3</b> Run MitoHiFi</a></li>
<li class="chapter" data-level="8.4" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#run-mitohifi-on-the-organelle-reference"><i class="fa fa-check"></i><b>8.4</b> Run MitoHifi on the organelle reference</a></li>
<li class="chapter" data-level="8.5" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#summary-2"><i class="fa fa-check"></i><b>8.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html"><i class="fa fa-check"></i><b>9</b> Scaffolding using HiC</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#introduction-3"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#set-up-1"><i class="fa fa-check"></i><b>9.2</b> Set up</a></li>
<li class="chapter" data-level="9.3" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#map-the-hi-c-reads-to-the-genome-assembly"><i class="fa fa-check"></i><b>9.3</b> Map the Hi-C reads to the genome assembly</a></li>
<li class="chapter" data-level="9.4" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#visualise-the-scaffolding-data-in-juicer"><i class="fa fa-check"></i><b>9.4</b> Visualise the scaffolding data in Juicer</a></li>
<li class="chapter" data-level="9.5" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#summary-3"><i class="fa fa-check"></i><b>9.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html"><i class="fa fa-check"></i><b>10</b> Optional Exercises</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#introduction-4"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#preassembly-1"><i class="fa fa-check"></i><b>10.2</b> Preassembly</a></li>
<li class="chapter" data-level="10.3" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#assembly-1"><i class="fa fa-check"></i><b>10.3</b> Assembly</a></li>
<li class="chapter" data-level="10.4" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#assessment"><i class="fa fa-check"></i><b>10.4</b> Assessment</a></li>
<li class="chapter" data-level="10.5" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#mitochondrial-assembly"><i class="fa fa-check"></i><b>10.5</b> Mitochondrial Assembly</a></li>
<li class="chapter" data-level="10.6" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#scaffolding-1"><i class="fa fa-check"></i><b>10.6</b> Scaffolding</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Conclusion.html"><a href="11-Conclusion.html"><i class="fa fa-check"></i><b>11</b> Conclusion</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Eukaryote genome assembly</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="assembly" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Genome assembly<a href="06-Assembly.html#assembly" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/string.png" width="20%" style="display: block; margin: auto;" /></p>
<div id="overview-1" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Overview<a href="06-Assembly.html#overview-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this section we're going to assemble a eukaryotic genome from long and highly accurate (&gt;Q30) PacBio HiFi reads.</p>
<p>The following workflow is courtesy <a href="https://www.sanger.ac.uk/person/stevens-lewis/">Lewis Stevens</a> who uses it to assemble nematode genomes (which range in size from 50-700 Mbp). Broadly, the same steps are also used by the <a href="https://www.darwintreeoflife.org">Darwin Tree of Life Project</a>, which is sequencing 70,000 eukaryotic species, ranging in size from 15 Mbp fungal genomes, to 100 Gbp plants genomes.</p>
</div>
<div id="downsample-raw-data" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Downsample raw data<a href="06-Assembly.html#downsample-raw-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, we need to create a directory to put the data in and then change directory to it.</p>
<pre><code>mkdir -p ~/eukaryote_assembly/2-assembly
cd ~/eukaryote_assembly/2-assembly</code></pre>
<p>Assembling the whole raw read file takes about 3 hours on our computers. So we are going to downsample the raw reads file so that it runs in a few minutes for this tutorial. If you were doing a real genome assembly project, you would give the whole file as an input.</p>
<p>The original file had 10,800 Mb, whereas the expected genome size is &lt;100 Mb, or ~108X sequencing depth. For this exercise we will sample only 6% of the input data using a popular tool called <code>seqtk</code>. 6% of the data is roughly equal to 6X sequencing depth/coverage. A good pacbio hifi assembly usually needs at least 30X depth.</p>
<p>Even if you are doing a real genome assembly project and not a tutorial/workshop like this one, it can be helpful to work with downsampled data that runs fast to check that the software tools are all installed correctly.</p>
<p><strong>Note</strong>:</p>
<ul>
<li>The below command can be run over one line excluding the <code>\</code></li>
<li>Alternatively, you can type <code>\</code> during a command and then press the enter key. The next line on the command line will start with <code>&gt;</code>. This will allow you to continue typing the command on the line. This can be used to type one command over multiple lines. We often present the command lines like this in the workbook to help make them easier to read.</li>
</ul>
<p>To downsample the data to only 6% of the original size:</p>
<pre><code>seqtk sample ~/eukaryote_assembly/1-preassembly/ERR7979900.fastq 0.06 | \
gzip -c &gt; ERR7979900.6pct.fastq.gz</code></pre>
<p>The <code>seqtk sample</code> command takes two arguments <code>input_file</code> and <code>percentage_to_sample</code> so we gave it the original raw read file from the previous directory, and 0.06 or 6%. The <code>| gzip -c &gt;</code> part pipes the output of the first command through the gzip programme and stores the result in <code>ERR7979900.6pct.fastq.gz</code></p>
</div>
<div id="run-the-hifiasm-assembler" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Run the Hifiasm assembler<a href="06-Assembly.html#run-the-hifiasm-assembler" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We are now ready to use <a href="https://hifiasm.readthedocs.io/en/latest/index.html">hifiasm</a> to assemble the 6% downsampled data. hifiasm is a very fast genome assembler for PacBio HiFi data:</p>
<pre><code>hifiasm -l 0 -t 16 -o ERR7979900_hifiasm6l0 ERR7979900.6pct.fastq.gz</code></pre>
<p>This command should take ~8 minutes, during which time information about the progress of the assembly is printed to the terminal. Further information and discussion on this can be found <a href="https://hifiasm.readthedocs.io/en/latest/interpreting-output.html#hifiasm-log-interpretation">here</a>.</p>
<p>Explanation of the parameters:</p>
<ul>
<li><code>-l</code> (lower case letter l) Level of purge duplication. 0 to disable, 1 to only purge contained haplotigs, 2 to purge all types of haplotigs, 3 to purge all types of haplotigs in the most aggressive way. Default 3. We changed it to 0 as the GenomeScope plot only had one clear peak indicating a highly homozygous assembly.</li>
<li><code>-o</code> (lower case letter o) Output prefix. We used the input read file name followed by *_hifiasm6* to indicate that it is a hifiasm assembly on only 6% of the reads, followed by <em>l0</em> to indicate the level of duplicate purging applied</li>
<li><code>-t</code> Number of CPU threads</li>
</ul>
<p>Hifiasm returns several <code>.gfa</code> files which have the graph structure of the assembled fasta contigs. We need the file with the suffix <code>.bp.p_ctg.gfa</code> as that has the primary assembled contigs. To convert it to fasta, we use an awk command that prints out the sequence headers and the sequence in fasta format:</p>
<pre><code>awk &#39;/^S/{print &quot;&gt;&quot;$2;print $3}&#39; ERR7979900_hifiasm6l0.bp.p_ctg.gfa &gt; ERR7979900_hifiasm6l0.bp.p_ctg.fa</code></pre>
<p>If you run the UNIX command <code>ls -lh</code> to see the size of the files, you should see it is about 80 Megabytes in size:</p>
<pre><code>ls -lh ERR7979900_hifiasm6l0.bp.p_ctg.fa</code></pre>
</div>
<div id="summary-1" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Summary<a href="06-Assembly.html#summary-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this exercise, you have successfully downsampled the raw read sequence data and run the hifiasm assembler. In a real genome assembly project, you would not downsample the data.</p>
<p>However, sometimes you may have more than 500X or 1000X sequencing depth if you are assembling a species with a small genome size. In those cases, you may want to downsample the raw data to about 50X using the method shown here, because otherwise the assembly will take very long, and might give you too many alternate contigs.</p>
<p>In the next section you will look at tools and approaches to assess what you have asssembled.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="05-Pre_assembly.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="07-Genome_assessment.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
