<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Genome assembly assessment | Eukaryote genome assembly</title>
  <meta name="description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Genome assembly assessment | Eukaryote genome assembly" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Genome assembly assessment | Eukaryote genome assembly" />
  
  <meta name="twitter:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sujai Kumar &amp; Helen Hipperson" />


<meta name="date" content="2023-04-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="06-Assembly.html"/>
<link rel="next" href="08-Mitochondrial_assembly.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Eukaryote genome assembly</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Eukaryote_genome_assembly.html"><a href="01-Eukaryote_genome_assembly.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="chapter" data-level="5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html"><i class="fa fa-check"></i><b>5</b> Genome pre-assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#introduction-1"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#obtaining-the-read-data"><i class="fa fa-check"></i><b>5.2</b> Obtaining the read data</a></li>
<li class="chapter" data-level="5.3" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-reads-and-bases"><i class="fa fa-check"></i><b>5.3</b> Counting reads and bases</a></li>
<li class="chapter" data-level="5.4" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-and-analysing-k-mers"><i class="fa fa-check"></i><b>5.4</b> Counting and analysing k-mers</a></li>
<li class="chapter" data-level="5.5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#summary"><i class="fa fa-check"></i><b>5.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Assembly.html"><a href="06-Assembly.html"><i class="fa fa-check"></i><b>6</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Assembly.html"><a href="06-Assembly.html#overview-1"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="06-Assembly.html"><a href="06-Assembly.html#downsample-raw-data"><i class="fa fa-check"></i><b>6.2</b> Downsample raw data</a></li>
<li class="chapter" data-level="6.3" data-path="06-Assembly.html"><a href="06-Assembly.html#run-the-hifiasm-assembler"><i class="fa fa-check"></i><b>6.3</b> Run the Hifiasm assembler</a></li>
<li class="chapter" data-level="6.4" data-path="06-Assembly.html"><a href="06-Assembly.html#summary-1"><i class="fa fa-check"></i><b>6.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html"><i class="fa fa-check"></i><b>7</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#obtaining-the-full-assembly"><i class="fa fa-check"></i><b>7.1</b> Obtaining the full assembly</a></li>
<li class="chapter" data-level="7.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#basic-assembly-statistics"><i class="fa fa-check"></i><b>7.2</b> Basic assembly statistics</a></li>
<li class="chapter" data-level="7.3" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#check-for-contamination"><i class="fa fa-check"></i><b>7.3</b> Check for contamination</a></li>
<li class="chapter" data-level="7.4" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco"><i class="fa fa-check"></i><b>7.4</b> BUSCO</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-lineages"><i class="fa fa-check"></i><b>7.4.1</b> BUSCO lineages</a></li>
<li class="chapter" data-level="7.4.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#run-busco"><i class="fa fa-check"></i><b>7.4.2</b> Run BUSCO</a></li>
<li class="chapter" data-level="7.4.3" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-output"><i class="fa fa-check"></i><b>7.4.3</b> BUSCO output</a></li>
<li class="chapter" data-level="7.4.4" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-questions"><i class="fa fa-check"></i><b>7.4.4</b> BUSCO questions</a></li>
<li class="chapter" data-level="7.4.5" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-summary"><i class="fa fa-check"></i><b>7.4.5</b> BUSCO summary</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#purging-duplications-using-purge_dups"><i class="fa fa-check"></i><b>7.5</b> Purging duplications using purge_dups</a></li>
<li class="chapter" data-level="7.6" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#assessment-recap"><i class="fa fa-check"></i><b>7.6</b> Assessment recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html"><i class="fa fa-check"></i><b>8</b> Mitochondrial assembly</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#introduction-2"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#set-up"><i class="fa fa-check"></i><b>8.2</b> Set up</a></li>
<li class="chapter" data-level="8.3" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#run-mitohifi"><i class="fa fa-check"></i><b>8.3</b> Run MitoHiFi</a></li>
<li class="chapter" data-level="8.4" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#run-mitohifi-on-the-organelle-reference"><i class="fa fa-check"></i><b>8.4</b> Run MitoHifi on the organelle reference</a></li>
<li class="chapter" data-level="8.5" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#summary-2"><i class="fa fa-check"></i><b>8.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html"><i class="fa fa-check"></i><b>9</b> Scaffolding using HiC</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#introduction-3"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#set-up-1"><i class="fa fa-check"></i><b>9.2</b> Set up</a></li>
<li class="chapter" data-level="9.3" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#map-the-hi-c-reads-to-the-genome-assembly"><i class="fa fa-check"></i><b>9.3</b> Map the Hi-C reads to the genome assembly</a></li>
<li class="chapter" data-level="9.4" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#visualise-the-scaffolding-data-in-juicer"><i class="fa fa-check"></i><b>9.4</b> Visualise the scaffolding data in Juicer</a></li>
<li class="chapter" data-level="9.5" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#summary-3"><i class="fa fa-check"></i><b>9.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html"><i class="fa fa-check"></i><b>10</b> Optional Exercises</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#introduction-4"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#preassembly-1"><i class="fa fa-check"></i><b>10.2</b> Preassembly</a></li>
<li class="chapter" data-level="10.3" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#assembly-1"><i class="fa fa-check"></i><b>10.3</b> Assembly</a></li>
<li class="chapter" data-level="10.4" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#assessment"><i class="fa fa-check"></i><b>10.4</b> Assessment</a></li>
<li class="chapter" data-level="10.5" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#mitochondrial-assembly"><i class="fa fa-check"></i><b>10.5</b> Mitochondrial Assembly</a></li>
<li class="chapter" data-level="10.6" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#scaffolding-1"><i class="fa fa-check"></i><b>10.6</b> Scaffolding</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Conclusion.html"><a href="11-Conclusion.html"><i class="fa fa-check"></i><b>11</b> Conclusion</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Eukaryote genome assembly</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genomeassemblyassessment" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Genome assembly assessment<a href="07-Genome_assessment.html#genomeassemblyassessment" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/inspect.png" width="20%" style="display: block; margin: auto;" /></p>
<p>After carrying out a genome assembly, you should always do some basic checks on it to make sure it has the expected size, and that it matches the species that you sampled.</p>
<p>In this section we will use a few different tools to check the quality of the assembly:</p>
<ul>
<li>seqkit - which you used to check the raw read files as well</li>
<li>tiara - a very fast bacterial and prokaryote DNA classifier</li>
<li>busco - for finding universal single copy orthologues for that clade</li>
</ul>
<p>We will also learn about purge_dups, a tool for removing duplicate contigs so that you are left with only a haploid genome assembly.</p>
<p>In the previous section you saw how to run the hifiasm assembler on a smaller downsampled subset of reads. In this section we will be using the full genome assembly (which took about 3 hours to complete).</p>
<div id="obtaining-the-full-assembly" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Obtaining the full assembly<a href="07-Genome_assessment.html#obtaining-the-full-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We need to create a directory to put the assembly in and then change directory to it.</p>
<pre><code>mkdir -p ~/eukaryote_assembly/3-assessment
cd ~/eukaryote_assembly/3-assessment</code></pre>
<p>Create a symbolic link (i.e. shortcut) to the fasta file with the full assembly</p>
<pre><code>ln -s /pub14/tea/nsc202/ERR7979900_hifiasm.bp.p_ctg.fa</code></pre>
</div>
<div id="basic-assembly-statistics" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Basic assembly statistics<a href="07-Genome_assessment.html#basic-assembly-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Run seqkit stats to find some basic statistics about the genome assembly:</p>
<pre><code>seqkit stats -a ERR7979900_hifiasm.bp.p_ctg.fa</code></pre>
<ul>
<li><code>-a</code> specifies all stats, including n50</li>
</ul>
<p>You should see an output like this:</p>
<pre><code>file                            format  type  num_seqs      sum_len  min_len    avg_len     max_len        Q1      Q2        Q3  sum_gap        N50  Q20(%)  Q30(%)  GC(%)
ERR7979900_hifiasm.bp.p_ctg.fa  FASTA   DNA        548  122,586,855    6,222  223,698.7  13,381,880  21,636.5  27,578  40,509.5        0  5,147,866       0       0   51.9
</code></pre>
<p>Statistics such as num_seqs, sum_len, min_len, avg_len, max_len etc are self explanatory, but you can get a full reference <a href="https://bioinf.shenwei.me/seqkit/usage/#stats">here</a>. The Q20, Q30 values are only for FastQ files which have quality scores as well. Hifiasm creates a contig assembly, i.e. it creates contiguous sequences with no gaps, therefore sum_gap = 0.</p>
<p>You may not be aware of the N50 metric, which is calculated using the following method:</p>
<ol style="list-style-type: decimal">
<li>Order the contigs from largest to smallest.</li>
<li>Find the point that is 50% the length of the assembly starting from the start of the largest contig going toward the smallest contig.</li>
<li>Determine which contig this point belongs to.</li>
<li>The N50 is equal to the length of that contig.</li>
</ol>
<p>In this assembly, 122 Mb of total sequence is assembled in only 548 sequences. The contig N50 is &gt; 5Mb which is good as anything above 1Mb is generally considered to be high quality.</p>
<p>Some things you will want to look out for in future projects:</p>
<ul>
<li>High number of contigs, with small N50: This may indicate the genome assembly is very uncontiguous and so the assembler had issues with assembly.
<ul>
<li>It is possible the genome is a particularly difficult one to resolve.</li>
<li>The input data could be poor. Examples include:
<ul>
<li>Low quality.</li>
<li>Low coverage.</li>
<li>Average coverage could be good but it is possible that there are areas of the genome that have low coverage, so these areas might not be assembled well.</li>
</ul></li>
</ul></li>
<li>Length is not correct: Could be caused by several reasons:
<ul>
<li>Parts of the genome are duplicated in the assembly causing the assembly to be too long. This can happen when a consensus could not be reached effectively for part of the genome.</li>
<li>A short assembly can indicate that parts of the genome are missing.</li>
<li>A drastically increased length can indicate the presence of contamination from another organism. The GC(%) can give a clue if this may be the case if it is different from what you might expect for your study species.</li>
</ul></li>
</ul>
<p>Further assessment can elucidate if there are any issues and what they may be.</p>
</div>
<div id="check-for-contamination" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Check for contamination<a href="07-Genome_assessment.html#check-for-contamination" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>When assembling the genome of a eukaryote, it is very common to accidentally also sequence bacteria or other microorganisms that live alongside, on the surface of, or inside the target organism. There might also be endosymbionts (bacteria that live inside the cell of the eukaryote, eg Wolbachia), symbionts (eg Cyanobacteria when sequencing lichen), parasites (eg apicomplexan parasites inside some insects), or even the host organism if the target organism is a parasite (eg a host plant if you are sequencing a parasitoid gall wasp). In general, a non-target organism in your sample can be called a co-biont, i.e. an organism present alongside your organism of interest in a given sample.</p>
<p>A full discussion of methods for separating these cobionts is beyond the scope of this course. However, when doing a eukaryote genome assembly, it is always a good idea to do a quick check for bacteria and other prokaryotes using Tiara. <a href="https://academic.oup.com/bioinformatics/article/38/2/344/6375939">Tiara</a> is a very fast deep-learning based tool for identifying archaea, bacteria, prokarya, eukarya, and organelle (mitochondria and plastids) sequences.</p>
<p>First we need to open a new terminal window (<code>right click -&gt; click terminal</code>), load the tiara <code>environment</code> and then navigate to your <code>3-assessment</code> directory.</p>
<pre><code>. usetiara

cd ~/eukaryote_assembly/3-assessment</code></pre>
<p>In this example, we have used a bash environment variable called <code>ASSEMBLY</code>, so that if you want to run it again, you just have to change the value of this variable in the first line.</p>
<pre><code>ASSEMBLY=ERR7979900_hifiasm.bp.p_ctg.fa
tiara -i $ASSEMBLY -o $ASSEMBLY.tiara -t 16 --pr --tf all</code></pre>
<ul>
<li><code>-i</code> is the input sequence fasta file</li>
<li><code>-o</code> is the name of the main output file</li>
<li><code>-t</code> is the number of CPU threads to use</li>
<li><code>--pr</code> outputs probabilities for each class (bacteria, eukarya, etc) for each sequence</li>
<li><code>--tf all</code> splits the input fasta files into separate fasta files for each high probability assignment.</li>
</ul>
<p>Take a look inside the main output file <code>ERR7979900_hifiasm.bp.p_ctg.fa.tiara</code>. The first few lines should look like this:</p>
<pre><code>sequence_id   class_fst_stage   class_snd_stage   org       bac       arc       euk       unk1        pla     unk2  mit
ptg000001l    eukarya           n/a               0.001638  0.001968    0.000518    0.995876    0.000000    n/a     n/a   n/a
ptg000002l    eukarya           n/a               0.000000  0.000059    0.000574    0.999367    0.000000    n/a     n/a   n/a
ptg000003l    eukarya           n/a               0.000011  0.001375    0.001749    0.996865    0.000000    n/a     n/a   n/a
ptg000004l    eukarya           n/a               0.000000  0.001148    0.000726    0.998127    0.000000    n/a     n/a   n/a
ptg000005c    bacteria          n/a               0.000000  0.999976    0.000024    0.000000    0.000000    n/a     n/a   n/a
ptg000006l    bacteria          n/a               0.000000  0.999999    0.000001    0.000000    0.000000    n/a     n/a   n/a
ptg000007l    eukarya           n/a               0.000019  0.001668    0.004887    0.993426    0.000000    n/a     n/a   n/a
ptg000008l    eukarya           n/a               0.000001  0.000232    0.000022    0.999745    0.000000    n/a     n/a   n/a
ptg000009c    bacteria          n/a               0.000000  0.999979    0.000001    0.000020    0.000000    n/a     n/a   n/a
</code></pre>
<p>Columns 2 and 3 show you that Tiara runs in two stages:</p>
<ul>
<li>the first stage identifies organelles, bacteria, archaea, eukarya, and unknown (unk1)</li>
<li>the second stage further identifies organelle sequences as plastid, mitochondria, and unknown (unk2)</li>
</ul>
<p>The final set of probabilities for each sequence is in the remaining columns. If the probability for a particular class (eg eukarya) is above a certain threshold, it will put the fasta sequence for that class in a separate <code>CLASS_$ASSEMBLY</code> file, (eg <code>eukarya_ERR7979900_hifiasm.bp.p_ctg.fa</code>). You can see the default thresholds by just typing the <code>tiara</code> command on its own.</p>
<p>Let's see how many sequences got assigned to each category. We can use the <code>seqkit stats</code> that we used earlier and give all the files created. First go back to your terminal window with the <code>assembly</code> environment loaded (hint: the command prompt will start with <code>(assembly)</code>). First you'll need to specify the <code>$ASSEMBLY</code> variable again for this terminal, tnen run <code>seqkit</code>:</p>
<pre><code>ASSEMBLY=ERR7979900_hifiasm.bp.p_ctg.fa
seqkit stats *$ASSEMBLY</code></pre>
<ul>
<li><code>*$ASSEMBLY</code> will list all files that begin with the wildcard <code>*</code> (0 or more characters) and end with <code>$ASSEMBLY</code>, in this case all files that end with ERR7979900_hifiasm.bp.p_ctg.fa:</li>
</ul>
<p>The output should look like this:</p>
<pre><code>file                                          format  type  num_seqs      sum_len  min_len      avg_len     max_len
ERR7979900_hifiasm.bp.p_ctg.fa                FASTA   DNA        548  122,586,855    6,222    223,698.7  13,381,880
bacteria_ERR7979900_hifiasm.bp.p_ctg.fa       FASTA   DNA        493   56,346,101    6,222    114,292.3   5,147,866
eukarya_ERR7979900_hifiasm.bp.p_ctg.fa        FASTA   DNA         38   65,597,190   10,925  1,726,241.8  13,381,880
plastid_ERR7979900_hifiasm.bp.p_ctg.fa        FASTA   DNA          1       16,568   16,568       16,568      16,568
unknown_ERR7979900_hifiasm.bp.p_ctg.fa        FASTA   DNA         16      627,026   23,742     39,189.1      77,760</code></pre>
<p>Based on this table, most of the assembly by number (493 out of 548 sequences), and almost half by span (56 Mb out of 122 Mb) is actually bacterial in origin.</p>
<ol style="list-style-type: decimal">
<li>The default threshold for assigning a sequence to one of the above classes is 0.65. If you change that to 0.9 (Hint: use the <code>-p</code> option in tiara), what is the span of eukaryotic sequences?
<div id="radio_YDORHOVXDJ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_YDORHOVXDJ" value=""></input> <span>55 Mb</span></label><label><input type="radio" autocomplete="off" name="radio_YDORHOVXDJ" value="answer"></input> <span>65 Mb</span></label><label><input type="radio" autocomplete="off" name="radio_YDORHOVXDJ" value=""></input> <span>75 Mb</span></label><label><input type="radio" autocomplete="off" name="radio_YDORHOVXDJ" value=""></input> <span>85 Mb</span></label>
</div></li>
</ol>
<p>In this case, Tiara has done a very good job of separating the eukaryotic nematode sequences from the rest, because we know that the <em>Oscheius onirici</em> genome is 65 Mb in size.</p>
<p>However, the 16 kb sequence labelled plastid is odd as only plants should have plastids. We can check this sequence by Blasting it. Using firefox, load the BLAST webpage and upload the plastid fasta file.</p>
<pre><code>firefox https://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn &amp;</code></pre>
<p>Click on <code>Nucleotide BLAST</code> &gt; <code>Browse</code> next to where it says 'upload file' and choose the plastid assembly file.</p>
<p>Under 'Organism' enter 'Oscheius (taxid:42476)' and run a megablast search.</p>
<p><img src="figures/blast_search.png" width="90%" style="display: block; margin: auto;" /></p>
<p>After a few minutes the results will appear, and you can see that this is in fact the mitochondrial genome.</p>
<p><img src="figures/blast_result.png" width="90%" style="display: block; margin: auto;" /></p>
<p>A few things to remember:</p>
<ul>
<li>Tiara sometimes labels mitochondrial sequences as plastids</li>
<li>Tiara is not perfect. It is a lightweight and fast way to get a sense of what is going on in your assembly. A very good tool that requires more resources but is also fast is the <a href="https://github.com/ncbi/fcs">NCBI Foreign Contamination Screen</a></li>
</ul>
</div>
<div id="busco" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> BUSCO<a href="07-Genome_assessment.html#busco" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/universe.png" width="15%" style="display: block; margin: auto;" /></p>
<p><code>BUSCO</code> stands for Benchmarking Universal Single-Copy Orthologs. The tool is used to estimate the completeness of genome assemblies by determining how many BUSCOs are present.</p>
<p>BUSCOs are genes that are:</p>
<ul>
<li><strong>Universal</strong>: The genes are present in all organisms of a specific lineage.</li>
<li><strong>Single-Copy</strong>: The genes are only found as one copy.</li>
</ul>
<div id="busco-lineages" class="section level3 hasAnchor" number="7.4.1">
<h3><span class="header-section-number">7.4.1</span> BUSCO lineages<a href="07-Genome_assessment.html#busco-lineages" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/phylo_tree.png" width="15%" style="display: block; margin: auto;" /></p>
<p><code>BUSCO</code> has many different lineage datasets. These are datasets of the different groups of BUSCOs found in various different lineages. Carry out the below command to see all these different datasets.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="07-Genome_assessment.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="ex">busco</span> <span class="at">--list-datasets</span> <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p>You will notice that the lineage datasets are at various taxonomic levels going from high-level kingdoms (bacteria, archaea, &amp; eukaryota) down to lower-level taxon lineages (primates, insecta, saccharomycetes etc). It is always good to be as specific as you can to your organism's lineage. A more specific lineage will have more BUSCOs than a more generic one.</p>
<p><strong>Note</strong>: All datasets end with <code>_odb10</code> which stands for <a href="https://www.orthodb.org/">OrthoDB version 10</a>.</p>
<p>Looking through the choices (and using the <a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI taxonomy website</a>) what lineage dataset would you use for the following genera?</p>
<ol style="list-style-type: decimal">
<li><strong>Heliconius</strong> (Taxonomy ID: 33416)
<div id="radio_ECDNBZZCVC" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ECDNBZZCVC" value="answer"></input> <span>lepidoptera_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_ECDNBZZCVC" value=""></input> <span>diptera_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_ECDNBZZCVC" value=""></input> <span>arthropoda_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_ECDNBZZCVC" value=""></input> <span>eukaryota_odb10</span></label>
</div></li>
<li><strong>Saccharomyces</strong> (Taxonomy ID: 4930)
<div id="radio_ARXSSVALKF" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ARXSSVALKF" value=""></input> <span>eukaryota_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_ARXSSVALKF" value="answer"></input> <span>saccharomycetes_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_ARXSSVALKF" value=""></input> <span>bacteria_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_ARXSSVALKF" value=""></input> <span>fungi_odb10</span></label>
</div></li>
<li><strong>Oscheius</strong> (Taxonomy ID: 42476)
<div id="radio_MBCJYKNKEM" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_MBCJYKNKEM" value=""></input> <span>insecta_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_MBCJYKNKEM" value=""></input> <span>metazoa_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_MBCJYKNKEM" value="answer"></input> <span>nematoda_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_MBCJYKNKEM" value=""></input> <span>eukaryota_odb10</span></label>
</div></li>
<li><strong>Neoparamoeba</strong> (Taxonomy ID: 1641714)
<div id="radio_TKJWEIPQUI" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_TKJWEIPQUI" value=""></input> <span>alveolata_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_TKJWEIPQUI" value=""></input> <span>coccidia_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_TKJWEIPQUI" value=""></input> <span>plasmodium_odb10</span></label><label><input type="radio" autocomplete="off" name="radio_TKJWEIPQUI" value="answer"></input> <span>eukaryota_odb10</span></label>
</div></li>
</ol>
<div class="webex-solution">
<button>
Click to reveal links to NCBI taxonomies
</button>
<ol style="list-style-type: decimal">
<li><a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=33416"><strong>Heliconius</strong> (Taxonomy ID: 33416)</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=4930"><strong>Saccharomyces</strong> (Taxonomy ID: 4930)</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=42476"><strong>Oscheius</strong> (Taxonomy ID: 42476)</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=1641714"><strong>Neoparamoeba</strong> (Taxonomy ID: 1641714)</a></li>
</ol>
</div>
</div>
<div id="run-busco" class="section level3 hasAnchor" number="7.4.2">
<h3><span class="header-section-number">7.4.2</span> Run BUSCO<a href="07-Genome_assessment.html#run-busco" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/enter_key.png" width="15%" style="display: block; margin: auto;" /></p>
<p>For our <em>Oscheius onerici</em> example, we will choose the <code>nematoda_odb10</code> as our lineage dataset. With that we can run <code>BUSCO</code> on our genome assembly.</p>
<p><strong>Note</strong>: The parameters <code>-i</code>, <code>-l</code>, <code>-m</code>, &amp; <code>-o</code> are all letters.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="07-Genome_assessment.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="ex">busco</span> <span class="dt">\</span></span>
<span id="cb27-2"><a href="07-Genome_assessment.html#cb27-2" aria-hidden="true" tabindex="-1"></a>-i <span class="va">$ASSEMBLY</span> <span class="dt">\</span></span>
<span id="cb27-3"><a href="07-Genome_assessment.html#cb27-3" aria-hidden="true" tabindex="-1"></a>-l nematoda_odb10 <span class="dt">\</span></span>
<span id="cb27-4"><a href="07-Genome_assessment.html#cb27-4" aria-hidden="true" tabindex="-1"></a>-m geno <span class="dt">\</span></span>
<span id="cb27-5"><a href="07-Genome_assessment.html#cb27-5" aria-hidden="true" tabindex="-1"></a>-o <span class="va">$ASSEMBLY</span><span class="dt">\_</span>nematoda_odb10 <span class="dt">\</span></span>
<span id="cb27-6"><a href="07-Genome_assessment.html#cb27-6" aria-hidden="true" tabindex="-1"></a>--cpu 16</span></code></pre></div>
<p><strong>Parameter choice</strong></p>
<ul>
<li><code>-i</code> : The input genome assembly in fasta format.</li>
<li><code>-l</code> : The lineage dataset to be used. This has to end with the <code>_odb10</code> part. This will download the dataset (if it does not exist) into a directory called <code>busco_downloads</code> in the current directory. The directory will be created if it doesn't exist.</li>
<li><code>-m</code> : The assessment mode.
<ul>
<li><code>geno</code> = genome assemblies (DNA)</li>
<li><code>tran</code> = transcriptome assemblies (DNA)</li>
<li><code>prot</code> = annotated gene sets (protein)</li>
</ul></li>
<li><code>-o</code> : The run name. Output folders and files will be labelled with this name.</li>
<li><code>--cpu</code> : The number of cpu threads to use</li>
</ul>
<p>This command should take ~10 minutes to run.</p>
</div>
<div id="busco-output" class="section level3 hasAnchor" number="7.4.3">
<h3><span class="header-section-number">7.4.3</span> BUSCO output<a href="07-Genome_assessment.html#busco-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/output.png" width="15%" style="display: block; margin: auto;" /></p>
<p>There are many output files in the directory <code>ERR7979900_hifiasm.bp.p_ctg.fa_nematoda_odb10</code>. A full list of the output can be found in the following link: <a href="https://busco.ezlab.org/busco_userguide.html#outputs" class="uri">https://busco.ezlab.org/busco_userguide.html#outputs</a></p>
<p>I ignore the majority of the output files most of the time, instead only looking at the short summary file.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="07-Genome_assessment.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> ERR7979900_hifiasm.bp.p_ctg.fa_nematoda_odb10/short_summary.specific.nematoda_odb10.ERR7979900_hifiasm.bp.p_ctg.fa_nematoda_odb10.txt</span></code></pre></div>
<p>This provides us with 6 values. These values are presented in terms of percentage (along the top line of the Results) and total numbers (the next 6 lines). The values represent:</p>
<ul>
<li><strong>Complete BUSCOs</strong> : The number of complete BUSCOs discovered. The closer this value is to 100% the better. This number is the sum of the next two values.</li>
<li><strong>Complete and single-copy BUSCOs</strong>: The number of BUSCOs found as complete and single-copy. The closer this number is to 100% the better. A good assembly will generally be &gt;90%.</li>
<li><strong>Complete and duplicated BUSCOs</strong>: Complete BUSCOs that have been found two or more times within the assembly. This indicates there are duplicated areas of the genome. Preferably we would like this number to be low.</li>
<li><strong>Fragmented BUSCOs</strong>: This could indicate that there are genes that are only partially present within the genome. This could be due to misassmebly.</li>
<li><strong>Missing BUSCOs</strong>: These are BUSCOs that could not be found in the genome assembly. This could indicate that parts of the genome are missing in the assembly.</li>
<li><strong>Total BUSCO groups searched</strong>: This represents the total number of BUSCOs within the lineage dataset used for analysis.</li>
</ul>
<p>For more info on interpreting the results please see the following link: <a href="https://busco.ezlab.org/busco_userguide.html#interpreting-the-results" class="uri">https://busco.ezlab.org/busco_userguide.html#interpreting-the-results</a></p>
</div>
<div id="busco-questions" class="section level3 hasAnchor" number="7.4.4">
<h3><span class="header-section-number">7.4.4</span> BUSCO questions<a href="07-Genome_assessment.html#busco-questions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/question_bubble_blue.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Answer the below MCQs based on the results in your short summary file.</p>
<ol style="list-style-type: decimal">
<li>Ignoring "total BUSCO groups searched", which value is the highest?
<div id="radio_BZVTOHYBGP" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BZVTOHYBGP" value="answer"></input> <span>Complete BUSCOs (C)</span></label><label><input type="radio" autocomplete="off" name="radio_BZVTOHYBGP" value=""></input> <span>Fragmented BUSCOs (F)</span></label><label><input type="radio" autocomplete="off" name="radio_BZVTOHYBGP" value=""></input> <span>Missing BUSCOs (M)</span></label>
</div></li>
<li>What is the sum of fragmented and missing BUSCOs?
<div id="radio_VUOKMMENOJ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_VUOKMMENOJ" value=""></input> <span>1</span></label><label><input type="radio" autocomplete="off" name="radio_VUOKMMENOJ" value=""></input> <span>151</span></label><label><input type="radio" autocomplete="off" name="radio_VUOKMMENOJ" value="answer"></input> <span>616</span></label>
</div></li>
<li>Which value would you want to be highest for a complete non duplicated assembly?
<div id="radio_ZAZTPHNIFZ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ZAZTPHNIFZ" value="answer"></input> <span>Complete and single-copy BUSCOs (S)</span></label><label><input type="radio" autocomplete="off" name="radio_ZAZTPHNIFZ" value=""></input> <span>Complete and duplicated BUSCOs (D)</span></label><label><input type="radio" autocomplete="off" name="radio_ZAZTPHNIFZ" value=""></input> <span>Fragmented BUSCOs (F)</span></label><label><input type="radio" autocomplete="off" name="radio_ZAZTPHNIFZ" value=""></input> <span>Missing BUSCOs (M)</span></label>
</div></li>
</ol>
</div>
<div id="busco-summary" class="section level3 hasAnchor" number="7.4.5">
<h3><span class="header-section-number">7.4.5</span> BUSCO summary<a href="07-Genome_assessment.html#busco-summary" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/sum_blue.png" width="15%" style="display: block; margin: auto;" /></p>
<p>The <code>BUSCO</code> stats are OK for this assembly. There are more fragmented and missing BUSCOs than we would like. We'll try to improve this with scaffolding. Even though the BUSCO gene sets are meant to be universal (i.e. applicable to all species within a busco lineage), we have found that some clades are never more than 80-85% complete even when the full chromosomal level genome assembly is available.</p>
</div>
</div>
<div id="purging-duplications-using-purge_dups" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Purging duplications using purge_dups<a href="07-Genome_assessment.html#purging-duplications-using-purge_dups" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If you are assembling a diploid genome, then the small allelic variations between the two copies of a genome can cause several alternate duplicated contigs to be assembled. Hifiasm tries to account for this by generating alternate haplotig assembly files. However, sometimes hifiasm isn't able to clearly distinguish two contigs as being alternate haplotigs of the same genomic region.</p>
<p>This is where <em>purge_dups</em> comes in. It is a piece of software that maps the original reads back to the assembly and tries to figure out which bits are alternate haplotigs and which bits are genuine haploid genome representations.</p>
<p>The genomescope plot for our workshop example species <em>Oscheius onirici</em> didn't indicate much heterozygosity, so it is very likely that purge_dups will not find many extra contigs to remove.</p>
<p>For this part of the workshop, we will use only the eukarya part of the genome assembly as identified by Tiara.</p>
<p>First make a sub-folder for purge_dups and change into it:</p>
<pre><code>mkdir -p ~/eukaryote_assembly/3-assessment/purge_dups
cd ~/eukaryote_assembly/3-assessment/purge_dups</code></pre>
<p>Now make a symlink to the eukarya_ part of the assembly, and to the reads from the assembly directory:</p>
<pre><code>ln -s ~/eukaryote_assembly/3-assessment/eukarya_ERR7979900_hifiasm.bp.p_ctg.fa
ln -s ~/eukaryote_assembly/1-preassembly/ERR7979900.fastq</code></pre>
<p>From now on, we're going to use bash variables extensively, so that you just have to change the names of the files in the variables at the start, and then you can reuse the commands that follow on your own data in future:</p>
<pre><code>READS=ERR7979900.fastq
CONTIGS=eukarya_ERR7979900_hifiasm.bp.p_ctg.fa
CPU=16</code></pre>
<ol style="list-style-type: decimal">
<li>Map the reads to the assembly (this step will take the longest, ~2 minutes)</li>
</ol>
<pre><code>minimap2 -t $CPU -x map-hifi $CONTIGS $READS | \
gzip -c - &gt;${CONTIGS}.paf.gz</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Calculate read depth and cutoffs:</li>
</ol>
<pre><code>pbcstat ${CONTIGS}.paf.gz

calcuts PB.stat &gt; ${CONTIGS}_cutoffs 2&gt; \
${CONTIGS}_calcults.log</code></pre>
<p>The first step above gets read depth statistics. The second step uses that information to decide cutoffs, e.g., if the majority of the contigs are around 20-25 read depth, then a contig with read depth between 10-12 might be a duplicated contig representing the alternate heterozygous part of the assembly.</p>
<ol start="3" style="list-style-type: decimal">
<li>Split assembly and do self alignment:</li>
</ol>
<pre><code>split_fa $CONTIGS &gt;${CONTIGS}.split

minimap2 -t $CPU -xasm5 \
-DP ${CONTIGS}.split \
${CONTIGS}.split | \
gzip -c - &gt;${CONTIGS}.split.self.paf.gz</code></pre>
<p>This step is necessary to see which contigs are almost identical to other contigs.</p>
<ol start="4" style="list-style-type: decimal">
<li>Purge haplotigs:</li>
</ol>
<pre><code>purge_dups -2 -T ${PREFIX}_cutoffs \
-c PB.base.cov \
${CONTIGS}.split.self.paf.gz &gt;${CONTIGS}_dups.bed \
2&gt;${CONTIGS}_purge_dups.log

get_seqs -e ${CONTIGS}_dups.bed $CONTIGS

mv purged.fa ${CONTIGS}_purged.fa

mv hap.fa ${CONTIGS}_hap.fa</code></pre>
<p>A haplotig is a contig that comes from the same haplotype. This is the final step where purge_dups uses three pieces of information: the cutoffs file, the coverage (or read depth file), and the information about which contigs are almost identical to others. If a contig is lower than the cutoff, and is almost identical to another contig, then it is marked as a haplotig. The output of this step is <code>hap.fa</code> - the file with alternate haplotigs, and <code>purged.fa</code> - the file with haplotigs removed (i.e. purged).</p>
<p>Let's take a look at the stats of the two files created:</p>
<pre><code>seqkit stats ${CONTIGS} \
${CONTIGS}_purged.fa \
${CONTIGS}_hap.fa</code></pre>
<p>The output should look something like this:</p>
<pre><code>file                                              format  type  num_seqs     sum_len  min_len      avg_len     max_len
eukarya_ERR7979900_hifiasm.bp.p_ctg.fa            FASTA   DNA         18  64,881,769   10,925  3,604,542.7  13,381,880
eukarya_ERR7979900_hifiasm.bp.p_ctg.fa_purged.fa  FASTA   DNA         12  64,697,286   16,318  5,391,440.5  13,381,880
eukarya_ERR7979900_hifiasm.bp.p_ctg.fa_hap.fa     FASTA   DNA          8     184,483   10,925     23,060.4      45,255</code></pre>
<p>As we suspected, only a tiny span (184,483 bp) was removed from the assembly, indicating this was a mostly homozygous assembly with almost no alternate haplotigs.</p>
</div>
<div id="assessment-recap" class="section level2 hasAnchor" number="7.6">
<h2><span class="header-section-number">7.6</span> Assessment recap<a href="07-Genome_assessment.html#assessment-recap" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/recap.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We used the following tools to assess our assembly:</p>
<ol style="list-style-type: decimal">
<li><code>seqkit stats</code> to get assembly metrics</li>
<li><code>tiara</code> to see how much bacterial/archaeal sequence was in the assembly</li>
<li><code>busco</code> to see how complete the eukaryote part of the assembly was</li>
<li><code>purge_dups</code> to get rid of any duplicated sequences as a result of heterozygosity</li>
</ol>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="06-Assembly.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="08-Mitochondrial_assembly.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
