<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Check for contamination | My Book</title>
  <meta name="description" content="This book ..." />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Check for contamination | My Book" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This book ..." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Check for contamination | My Book" />
  
  <meta name="twitter:description" content="This book ..." />
  

<meta name="author" content="Me" />


<meta name="date" content="2024-06-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="07-Genome_assessment.html"/>
<link rel="next" href="09-BUSCO.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>Eukaryote genome assembly</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Webexercises</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#example-questions"><i class="fa fa-check"></i><b>0.1</b> Example Questions</a>
<ul>
<li class="chapter" data-level="0.1.1" data-path="index.html"><a href="index.html#fill-in-the-blanks-fitb"><i class="fa fa-check"></i><b>0.1.1</b> Fill-In-The-Blanks (<code>fitb()</code>)</a></li>
<li class="chapter" data-level="0.1.2" data-path="index.html"><a href="index.html#multiple-choice-mcq"><i class="fa fa-check"></i><b>0.1.2</b> Multiple Choice (<code>mcq()</code>)</a></li>
<li class="chapter" data-level="0.1.3" data-path="index.html"><a href="index.html#true-or-false-torf"><i class="fa fa-check"></i><b>0.1.3</b> True or False (<code>torf()</code>)</a></li>
<li class="chapter" data-level="0.1.4" data-path="index.html"><a href="index.html#longer-mcqs-longmcq"><i class="fa fa-check"></i><b>0.1.4</b> Longer MCQs (<code>longmcq()</code>)</a></li>
</ul></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#checked-sections"><i class="fa fa-check"></i><b>0.2</b> Checked sections</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#hidden-solutions-and-hints"><i class="fa fa-check"></i><b>0.3</b> Hidden solutions and hints</a></li>
</ul></li>
<li class="part"><span><b>Intro</b></span></li>
<li class="chapter" data-level="1" data-path="01-Eukaryote_genome_assembly.html"><a href="01-Eukaryote_genome_assembly.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Eukaryote_genome_assembly.html"><a href="01-Eukaryote_genome_assembly.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="part"><span><b>Assembly steps</b></span></li>
<li class="chapter" data-level="5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html"><i class="fa fa-check"></i><b>5</b> Genome pre-assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#introduction-1"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#obtaining-the-read-data"><i class="fa fa-check"></i><b>5.2</b> Obtaining the read data</a></li>
<li class="chapter" data-level="5.3" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-reads-and-bases"><i class="fa fa-check"></i><b>5.3</b> Counting reads and bases</a></li>
<li class="chapter" data-level="5.4" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-and-analysing-k-mers"><i class="fa fa-check"></i><b>5.4</b> Counting and analysing k-mers</a></li>
<li class="chapter" data-level="5.5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#summary"><i class="fa fa-check"></i><b>5.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Assembly.html"><a href="06-Assembly.html"><i class="fa fa-check"></i><b>6</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Assembly.html"><a href="06-Assembly.html#overview-1"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="06-Assembly.html"><a href="06-Assembly.html#downsample-raw-data"><i class="fa fa-check"></i><b>6.2</b> Downsample raw data</a></li>
<li class="chapter" data-level="6.3" data-path="06-Assembly.html"><a href="06-Assembly.html#run-the-hifiasm-assembler"><i class="fa fa-check"></i><b>6.3</b> Run the Hifiasm assembler</a></li>
<li class="chapter" data-level="6.4" data-path="06-Assembly.html"><a href="06-Assembly.html#summary-1"><i class="fa fa-check"></i><b>6.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Assembly assessment</b></span></li>
<li class="chapter" data-level="7" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html"><i class="fa fa-check"></i><b>7</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#obtaining-the-full-assembly"><i class="fa fa-check"></i><b>7.1</b> Obtaining the full assembly</a></li>
<li class="chapter" data-level="7.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#basic-assembly-statistics"><i class="fa fa-check"></i><b>7.2</b> Basic assembly statistics</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Contamination_check.html"><a href="08-Contamination_check.html"><i class="fa fa-check"></i><b>8</b> Check for contamination</a></li>
<li class="chapter" data-level="9" data-path="09-BUSCO.html"><a href="09-BUSCO.html"><i class="fa fa-check"></i><b>9</b> BUSCO</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-lineages"><i class="fa fa-check"></i><b>9.0.1</b> BUSCO lineages</a></li>
<li class="chapter" data-level="9.0.2" data-path="09-BUSCO.html"><a href="09-BUSCO.html#run-busco"><i class="fa fa-check"></i><b>9.0.2</b> Run BUSCO</a></li>
<li class="chapter" data-level="9.0.3" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-output"><i class="fa fa-check"></i><b>9.0.3</b> BUSCO output</a></li>
<li class="chapter" data-level="9.0.4" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-questions"><i class="fa fa-check"></i><b>9.0.4</b> BUSCO questions</a></li>
<li class="chapter" data-level="9.0.5" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-summary"><i class="fa fa-check"></i><b>9.0.5</b> BUSCO summary</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Purge_duplicates.html"><a href="10-Purge_duplicates.html"><i class="fa fa-check"></i><b>10</b> Purging duplications</a></li>
<li class="chapter" data-level="11" data-path="11-Assessment_recap.html"><a href="11-Assessment_recap.html"><i class="fa fa-check"></i><b>11</b> Assessment recap</a></li>
<li class="part"><span><b>Post-assembly</b></span></li>
<li class="chapter" data-level="12" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html"><i class="fa fa-check"></i><b>12</b> Mitochondrial assembly</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#introduction-2"><i class="fa fa-check"></i><b>12.1</b> Introduction</a></li>
<li class="chapter" data-level="12.2" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#set-up"><i class="fa fa-check"></i><b>12.2</b> Set up</a></li>
<li class="chapter" data-level="12.3" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#run-mitohifi"><i class="fa fa-check"></i><b>12.3</b> Run MitoHiFi</a></li>
<li class="chapter" data-level="12.4" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#run-mitohifi-on-the-organelle-reference"><i class="fa fa-check"></i><b>12.4</b> Run MitoHifi on the organelle reference</a></li>
<li class="chapter" data-level="12.5" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#summary-2"><i class="fa fa-check"></i><b>12.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html"><i class="fa fa-check"></i><b>13</b> Scaffolding using HiC</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#introduction-3"><i class="fa fa-check"></i><b>13.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#set-up-1"><i class="fa fa-check"></i><b>13.2</b> Set up</a></li>
<li class="chapter" data-level="13.3" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#map-the-hi-c-reads-to-the-genome-assembly"><i class="fa fa-check"></i><b>13.3</b> Map the Hi-C reads to the genome assembly</a></li>
<li class="chapter" data-level="13.4" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#visualise-the-scaffolding-data-in-juicer"><i class="fa fa-check"></i><b>13.4</b> Visualise the scaffolding data in Juicer</a></li>
<li class="chapter" data-level="13.5" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#summary-3"><i class="fa fa-check"></i><b>13.5</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Outro</b></span></li>
<li class="chapter" data-level="14" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html"><i class="fa fa-check"></i><b>14</b> Optional Exercises</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#introduction-4"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#preassembly-1"><i class="fa fa-check"></i><b>14.2</b> Preassembly</a></li>
<li class="chapter" data-level="14.3" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#assembly-1"><i class="fa fa-check"></i><b>14.3</b> Assembly</a></li>
<li class="chapter" data-level="14.4" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#assessment"><i class="fa fa-check"></i><b>14.4</b> Assessment</a></li>
<li class="chapter" data-level="14.5" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#mitochondrial-assembly"><i class="fa fa-check"></i><b>14.5</b> Mitochondrial Assembly</a></li>
<li class="chapter" data-level="14.6" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#scaffolding-1"><i class="fa fa-check"></i><b>14.6</b> Scaffolding</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Conclusion.html"><a href="15-Conclusion.html"><i class="fa fa-check"></i><b>15</b> Conclusion</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">My Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="check-for-contamination" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Check for contamination<a href="08-Contamination_check.html#check-for-contamination" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/tiara.png" style="width:200px" />
</center>
<p>When assembling the genome of a eukaryote, it is very common to accidentally also sequence bacteria or other microorganisms that live alongside, on the surface of, or inside the target organism.</p>
<p>Other contaminants may come from:</p>
<ul>
<li><strong>Endosymbionts:</strong> Bacteria that live inside the cell of the eukaryote, e.g., <em>Wolbachia</em></li>
<li><strong>Symbionts:</strong> E.g., Cyanobacteria when sequencing lichen</li>
<li><strong>Parasites:</strong> E.g., apicomplexan parasites inside some insects</li>
<li><strong>The host organism:</strong> E.g. if the target organism is a parasite such as a host plant if you are sequencing a parasitoid gall wasp</li>
</ul>
<p>In general, a non-target organism in your sample can be called a co-biont, i.e., an organism present alongside your organism of interest in a given sample.</p>
<p>A full discussion of methods for separating these cobionts is beyond the scope of this course. However, when doing a eukaryote genome assembly, it is always a good idea to do a quick check for bacteria and other prokaryotes using <code>Tiara</code>. <a href="https://academic.oup.com/bioinformatics/article/38/2/344/6375939"><code>Tiara</code></a> is a very fast deep-learning based tool for identifying archaea, bacteria, prokarya, eukarya and organelle (mitochondria and plastids) sequences.</p>
<p>First we need to open a new terminal window (<code>right click -&gt; click terminal</code>), load the <code>tiara environment</code> and then navigate to your <code>3-assessment</code> directory.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="08-Contamination_check.html#cb1-1" tabindex="-1"></a><span class="bu">.</span> usetiara</span>
<span id="cb1-2"><a href="08-Contamination_check.html#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="08-Contamination_check.html#cb1-3" tabindex="-1"></a><span class="bu">cd</span> ~/eukaryote_assembly/3-assessment</span></code></pre></div>
<p>In this example, we will use a bash environment variable called <code>ASSEMBLY</code>. This means if you want to run it again with another sample you can change the value of this variable in the first line and then rerun the subsequent command.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="08-Contamination_check.html#cb2-1" tabindex="-1"></a><span class="va">ASSEMBLY</span><span class="op">=</span>ERR7979900_hifiasm.bp.p_ctg.fa</span>
<span id="cb2-2"><a href="08-Contamination_check.html#cb2-2" tabindex="-1"></a><span class="ex">tiara</span> <span class="at">-i</span> <span class="va">$ASSEMBLY</span> <span class="at">-o</span> <span class="va">$ASSEMBLY</span>.tiara <span class="at">-t</span> 16 <span class="at">--pr</span> <span class="at">--tf</span> all</span></code></pre></div>
<ul>
<li><code>-i</code> is the input sequence fasta file</li>
<li><code>-o</code> is the name of the main output file</li>
<li><code>-t</code> is the number of CPU threads to use</li>
<li><code>--pr</code> outputs probabilities for each class (bacteria, eukarya, etc.) for each sequence</li>
<li><code>--tf all</code> splits the input fasta files into separate fasta files for each high probability assignment.</li>
</ul>
<p>Take a look inside the main output file <code>ERR7979900_hifiasm.bp.p_ctg.fa.tiara</code>. The first few lines should look like this:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="08-Contamination_check.html#cb3-1" tabindex="-1"></a><span class="ex">sequence_id</span>   class_fst_stage   class_snd_stage   org       bac       arc       euk       unk1        pla     unk2  mit</span>
<span id="cb3-2"><a href="08-Contamination_check.html#cb3-2" tabindex="-1"></a><span class="ex">ptg000001l</span>    eukarya           n/a               0.001638  0.001968    0.000518    0.995876    0.000000    n/a     n/a   n/a</span>
<span id="cb3-3"><a href="08-Contamination_check.html#cb3-3" tabindex="-1"></a><span class="ex">ptg000002l</span>    eukarya           n/a               0.000000  0.000059    0.000574    0.999367    0.000000    n/a     n/a   n/a</span>
<span id="cb3-4"><a href="08-Contamination_check.html#cb3-4" tabindex="-1"></a><span class="ex">ptg000003l</span>    eukarya           n/a               0.000011  0.001375    0.001749    0.996865    0.000000    n/a     n/a   n/a</span>
<span id="cb3-5"><a href="08-Contamination_check.html#cb3-5" tabindex="-1"></a><span class="ex">ptg000004l</span>    eukarya           n/a               0.000000  0.001148    0.000726    0.998127    0.000000    n/a     n/a   n/a</span>
<span id="cb3-6"><a href="08-Contamination_check.html#cb3-6" tabindex="-1"></a><span class="ex">ptg000005c</span>    bacteria          n/a               0.000000  0.999976    0.000024    0.000000    0.000000    n/a     n/a   n/a</span>
<span id="cb3-7"><a href="08-Contamination_check.html#cb3-7" tabindex="-1"></a><span class="ex">ptg000006l</span>    bacteria          n/a               0.000000  0.999999    0.000001    0.000000    0.000000    n/a     n/a   n/a</span>
<span id="cb3-8"><a href="08-Contamination_check.html#cb3-8" tabindex="-1"></a><span class="ex">ptg000007l</span>    eukarya           n/a               0.000019  0.001668    0.004887    0.993426    0.000000    n/a     n/a   n/a</span>
<span id="cb3-9"><a href="08-Contamination_check.html#cb3-9" tabindex="-1"></a><span class="ex">ptg000008l</span>    eukarya           n/a               0.000001  0.000232    0.000022    0.999745    0.000000    n/a     n/a   n/a</span>
<span id="cb3-10"><a href="08-Contamination_check.html#cb3-10" tabindex="-1"></a><span class="ex">ptg000009c</span>    bacteria          n/a               0.000000  0.999979    0.000001    0.000020    0.000000    n/a     n/a   n/a</span></code></pre></div>
<p>Columns 2 and 3 show you that Tiara runs in two stages:</p>
<ul>
<li>the first stage identifies organelles, bacteria, archaea, eukarya, and unknown (unk1)</li>
<li>the second stage further identifies organelle sequences as plastid, mitochondria, and unknown (unk2)</li>
</ul>
<p>The final set of probabilities for each sequence is in the remaining columns. If the probability for a particular class (e.g., eukarya) is above a certain threshold, it will put the fasta sequence for that class in a separate <code>CLASS_$ASSEMBLY</code> file, (eg <code>eukarya_ERR7979900_hifiasm.bp.p_ctg.fa</code>). You can see the default thresholds by just typing the <code>tiara</code> command on its own.</p>
<p>Let's see how many sequences got assigned to each category. We can use the <code>seqkit stats</code> that we used earlier and give all the files created. First go back to your terminal window with the <code>assembly</code> environment loaded (hint: the command prompt will start with <code>(assembly)</code>). First you'll need to specify the <code>$ASSEMBLY</code> variable again for this terminal, tnen run <code>seqkit</code>:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="08-Contamination_check.html#cb4-1" tabindex="-1"></a><span class="va">ASSEMBLY</span><span class="op">=</span>ERR7979900_hifiasm.bp.p_ctg.fa</span>
<span id="cb4-2"><a href="08-Contamination_check.html#cb4-2" tabindex="-1"></a><span class="ex">seqkit</span> stats <span class="pp">*</span><span class="va">$ASSEMBLY</span></span></code></pre></div>
<ul>
<li><code>*$ASSEMBLY</code> will list all files that begin with the wildcard <code>*</code> (0 or more characters) and end with <code>$ASSEMBLY</code>, in this case all files that end with ERR7979900_hifiasm.bp.p_ctg.fa:</li>
</ul>
<p>The output should look like this:</p>
<center>
<img src="figures/tiara_stats.png" style="width:1500px" />
</center>
<p>Based on this table, most of the assembly by number (490 out of 548 sequences), and almost half by span (56 Mb out of 122 Mb) is actually bacterial in origin.</p>
<ol style="list-style-type: decimal">
<li>The default threshold for assigning a sequence to one of the above classes is 0.65. If you change that to 0.9 (Hint: use the <code>-p</code> option in <code>tiara</code>), what is the span of eukaryotic sequences?
<div id="radio_NPIPOYNEFC" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_NPIPOYNEFC" value=""></input> <span>55 Mb</span></label><label><input type="radio" autocomplete="off" name="radio_NPIPOYNEFC" value="answer"></input> <span>65 Mb</span></label><label><input type="radio" autocomplete="off" name="radio_NPIPOYNEFC" value=""></input> <span>75 Mb</span></label><label><input type="radio" autocomplete="off" name="radio_NPIPOYNEFC" value=""></input> <span>85 Mb</span></label>
</div></li>
</ol>
<p>In this case, <code>Tiara</code> has done a very good job of separating the eukaryotic nematode sequences from the rest, because we know that the <em>Oscheius onirici</em> genome is 65 Mb in size.</p>
<p>However, the 16 kb sequence labelled plastid is odd as only plants should have plastids. We can check this sequence by Blasting it. Using <code>firefox</code>, load the BLAST webpage and upload the plastid fasta file.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="08-Contamination_check.html#cb5-1" tabindex="-1"></a><span class="ex">firefox</span> https://blast.ncbi.nlm.nih.gov/Blast.cgi<span class="pp">?</span>PROGRAM=blastn <span class="kw">&amp;</span></span></code></pre></div>
<p>Click on <code>Nucleotide BLAST</code> &gt; <code>Browse</code> next to where it says 'upload file'. It may take a little while for the “File Upload” pop-up to appear. When it does, choose the plastid assembly file.</p>
<p>Under 'Organism' enter 'Oscheius (taxid:42476)' and run a megablast search.</p>
<center>
<img src="figures/blast_search.png" style="width:900px; border-radius:5px" />
</center>
<p>After a few minutes the results will appear, and you can see that this is in fact the mitochondrial genome.</p>
<center>
<img src="figures/blast_result.png" style="width:900px; border-radius:5px" />
</center>
<p>A few things to remember:</p>
<ul>
<li><code>Tiara</code> sometimes labels mitochondrial sequences as plastids</li>
<li><code>Tiara</code> is not perfect. It is a lightweight and fast way to get a sense of what is going on in your assembly. A very good tool that requires more resources but is also fast is the <a href="https://github.com/ncbi/fcs">NCBI Foreign Contamination Screen</a></li>
</ul>

</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="07-Genome_assessment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="09-BUSCO.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
