<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Scaffolding using HiC | Eukaryote genome assembly</title>
  <meta name="description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Scaffolding using HiC | Eukaryote genome assembly" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Scaffolding using HiC | Eukaryote genome assembly" />
  
  <meta name="twitter:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sujai Kumar &amp; Helen Hipperson" />


<meta name="date" content="2023-04-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="08-Mitochondrial_assembly.html"/>
<link rel="next" href="10-Optional_exercises.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Eukaryote genome assembly</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Eukaryote_genome_assembly.html"><a href="01-Eukaryote_genome_assembly.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="chapter" data-level="5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html"><i class="fa fa-check"></i><b>5</b> Genome pre-assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#introduction-1"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#obtaining-the-read-data"><i class="fa fa-check"></i><b>5.2</b> Obtaining the read data</a></li>
<li class="chapter" data-level="5.3" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-reads-and-bases"><i class="fa fa-check"></i><b>5.3</b> Counting reads and bases</a></li>
<li class="chapter" data-level="5.4" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-and-analysing-k-mers"><i class="fa fa-check"></i><b>5.4</b> Counting and analysing k-mers</a></li>
<li class="chapter" data-level="5.5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#summary"><i class="fa fa-check"></i><b>5.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Assembly.html"><a href="06-Assembly.html"><i class="fa fa-check"></i><b>6</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Assembly.html"><a href="06-Assembly.html#overview-1"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="06-Assembly.html"><a href="06-Assembly.html#downsample-raw-data"><i class="fa fa-check"></i><b>6.2</b> Downsample raw data</a></li>
<li class="chapter" data-level="6.3" data-path="06-Assembly.html"><a href="06-Assembly.html#run-the-hifiasm-assembler"><i class="fa fa-check"></i><b>6.3</b> Run the Hifiasm assembler</a></li>
<li class="chapter" data-level="6.4" data-path="06-Assembly.html"><a href="06-Assembly.html#summary-1"><i class="fa fa-check"></i><b>6.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html"><i class="fa fa-check"></i><b>7</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#obtaining-the-full-assembly"><i class="fa fa-check"></i><b>7.1</b> Obtaining the full assembly</a></li>
<li class="chapter" data-level="7.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#basic-assembly-statistics"><i class="fa fa-check"></i><b>7.2</b> Basic assembly statistics</a></li>
<li class="chapter" data-level="7.3" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#check-for-contamination"><i class="fa fa-check"></i><b>7.3</b> Check for contamination</a></li>
<li class="chapter" data-level="7.4" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco"><i class="fa fa-check"></i><b>7.4</b> BUSCO</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-lineages"><i class="fa fa-check"></i><b>7.4.1</b> BUSCO lineages</a></li>
<li class="chapter" data-level="7.4.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#run-busco"><i class="fa fa-check"></i><b>7.4.2</b> Run BUSCO</a></li>
<li class="chapter" data-level="7.4.3" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-output"><i class="fa fa-check"></i><b>7.4.3</b> BUSCO output</a></li>
<li class="chapter" data-level="7.4.4" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-questions"><i class="fa fa-check"></i><b>7.4.4</b> BUSCO questions</a></li>
<li class="chapter" data-level="7.4.5" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-summary"><i class="fa fa-check"></i><b>7.4.5</b> BUSCO summary</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#purging-duplications-using-purge_dups"><i class="fa fa-check"></i><b>7.5</b> Purging duplications using purge_dups</a></li>
<li class="chapter" data-level="7.6" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#assessment-recap"><i class="fa fa-check"></i><b>7.6</b> Assessment recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html"><i class="fa fa-check"></i><b>8</b> Mitochondrial assembly</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#introduction-2"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#set-up"><i class="fa fa-check"></i><b>8.2</b> Set up</a></li>
<li class="chapter" data-level="8.3" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#run-mitohifi"><i class="fa fa-check"></i><b>8.3</b> Run MitoHiFi</a></li>
<li class="chapter" data-level="8.4" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#run-mitohifi-on-the-organelle-reference"><i class="fa fa-check"></i><b>8.4</b> Run MitoHifi on the organelle reference</a></li>
<li class="chapter" data-level="8.5" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#summary-2"><i class="fa fa-check"></i><b>8.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html"><i class="fa fa-check"></i><b>9</b> Scaffolding using HiC</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#introduction-3"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#set-up-1"><i class="fa fa-check"></i><b>9.2</b> Set up</a></li>
<li class="chapter" data-level="9.3" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#map-the-hi-c-reads-to-the-genome-assembly"><i class="fa fa-check"></i><b>9.3</b> Map the Hi-C reads to the genome assembly</a></li>
<li class="chapter" data-level="9.4" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#visualise-the-scaffolding-data-in-juicer"><i class="fa fa-check"></i><b>9.4</b> Visualise the scaffolding data in Juicer</a></li>
<li class="chapter" data-level="9.5" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#summary-3"><i class="fa fa-check"></i><b>9.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html"><i class="fa fa-check"></i><b>10</b> Optional Exercises</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#introduction-4"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#preassembly-1"><i class="fa fa-check"></i><b>10.2</b> Preassembly</a></li>
<li class="chapter" data-level="10.3" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#assembly-1"><i class="fa fa-check"></i><b>10.3</b> Assembly</a></li>
<li class="chapter" data-level="10.4" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#assessment"><i class="fa fa-check"></i><b>10.4</b> Assessment</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Conclusion.html"><a href="11-Conclusion.html"><i class="fa fa-check"></i><b>11</b> Conclusion</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Conclusion.html"><a href="11-Conclusion.html#report"><i class="fa fa-check"></i><b>11.1</b> Report</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Conclusion.html"><a href="11-Conclusion.html#report-conda-environment-directory"><i class="fa fa-check"></i><b>11.1.1</b> Report: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="11.1.2" data-path="11-Conclusion.html"><a href="11-Conclusion.html#report-quast"><i class="fa fa-check"></i><b>11.1.2</b> Report: QUAST</a></li>
<li class="chapter" data-level="11.1.3" data-path="11-Conclusion.html"><a href="11-Conclusion.html#report-busco"><i class="fa fa-check"></i><b>11.1.3</b> Report: BUSCO</a></li>
<li class="chapter" data-level="11.1.4" data-path="11-Conclusion.html"><a href="11-Conclusion.html#report-multiqc"><i class="fa fa-check"></i><b>11.1.4</b> Report: MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Conclusion.html"><a href="11-Conclusion.html#final-recap"><i class="fa fa-check"></i><b>11.2</b> Final recap</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="12-Appendix.html"><a href="12-Appendix.html"><i class="fa fa-check"></i><b>A</b> Next steps</a></li>
<li class="chapter" data-level="B" data-path="12-Appendix.html"><a href="12-Appendix.html#manuals"><i class="fa fa-check"></i><b>B</b> Manuals</a></li>
<li class="chapter" data-level="C" data-path="12-Appendix.html"><a href="12-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>C</b> Obtaining Read Data</a></li>
<li class="chapter" data-level="D" data-path="12-Appendix.html"><a href="12-Appendix.html#scaffolding-1"><i class="fa fa-check"></i><b>D</b> Scaffolding</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Eukaryote genome assembly</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="scaffolding" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Scaffolding using HiC<a href="09-Scaffolding.html#scaffolding" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/scaffold.png" width="18%" style="display: block; margin: auto;" /></p>
<div id="introduction-3" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Introduction<a href="09-Scaffolding.html#introduction-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the section on genome assessment, we separated the high-probability bacteria contigs from the eukarya contigs, and then ran purge_dups on the eukarya contigs. We were left with a very high contiguity nematode genome assembly.</p>
<p>In this exercise we will use our best assembly from the previous step and paired Illumina Hi-C reads to scaffold the already very good assembly into chromosome sized scaffolds.</p>
</div>
<div id="set-up-1" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Set up<a href="09-Scaffolding.html#set-up-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Open a new terminal and load the Hi-C environment:</p>
<pre><code>. usehic</code></pre>
<p>Create a new directory, change into it, and link to the files we need:</p>
<pre><code>mkdir -p ~/eukaryote_assembly/5-scaffolding
cd ~/eukaryote_assembly/5-scaffolding</code></pre>
<p>Link to the purged eukaryote-only contigs:</p>
<pre><code>ln -s ~/eukaryote_assembly/3-assessment/eukarya_ERR7979900_hifiasm.bp.p_ctg.fa_purged.fa</code></pre>
<p>Link to the paired Hi-C reads. The original files were downloaded from <a href="https://www.ebi.ac.uk/ena/browser/view/ERR5967937" class="uri">https://www.ebi.ac.uk/ena/browser/view/ERR5967937</a> but were very large and would have taken too long for this exercise, so we sampled the first 10 million reads:</p>
<pre><code>ln -s /pub14/tea/nsc206/NEOF/Euk_assembly/data/test10m1.fq.gz
ln -s /pub14/tea/nsc206/NEOF/Euk_assembly/data/test10m2.fq.gz</code></pre>
</div>
<div id="map-the-hi-c-reads-to-the-genome-assembly" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Map the Hi-C reads to the genome assembly<a href="09-Scaffolding.html#map-the-hi-c-reads-to-the-genome-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As in the previous steps, we're going to use BASH variables so that in future you can just copy the commands:</p>
<pre><code>REF=eukarya_ERR7979900_hifiasm.bp.p_ctg.fa_purged.fa
FASTQ1=test10m1.fq.gz
FASTQ2=test10m2.fq.gz
PREFIX=purged10m
CPU=16
MEM=100G</code></pre>
<p>Index the reference and map the reads. The <code>bwa index</code> command will take a minute or so to run. Make sure to wait for it to finish before entering the second and third commands:</p>
<pre><code>bwa index $REF

samtools faidx $REF

bwa mem -t $CPU $REF $FASTQ1 $FASTQ2 | \
samtools view -b - &gt; $PREFIX.bam</code></pre>
<p>This will take a few minutes to run. When it's finished run the actual scaffolding step using <a href="https://github.com/c-zhou/yahss">YAHS</a> (Yet Another Hi-C Scaffolder)</p>
<pre><code>yahs $REF $PREFIX.bam -o $PREFIX</code></pre>
</div>
<div id="visualise-the-scaffolding-data-in-juicer" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Visualise the scaffolding data in Juicer<a href="09-Scaffolding.html#visualise-the-scaffolding-data-in-juicer" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Juicer and Juicebox are tools that allow you to visualise scaffolds and manipulate them.</p>
<p>These yahs scripts create a file ready for juicer:</p>
<pre><code>juicer pre ${PREFIX}.bin ${PREFIX}_scaffolds_final.agp $REF.fai | \
sort -k2,2d -k6,6d --parallel=$CPU -S$MEM | \
awk &#39;NF&#39; &gt; \
${PREFIX}_alignments_sorted.txt

awk &#39;/^&gt;/{if (l!=&quot;&quot;) print l; printf &quot;%s\t&quot;, $1; l=0; next}{l+=length($0)}END{print l}&#39; \
${PREFIX}_scaffolds_final.fa | \
sed &quot;s/&gt;//&quot; &gt; \
${PREFIX}_scaffolds_final.fa_chromosome_sizes.txt</code></pre>
<p>Now run juicer_tools to create a file that can be seen in Juicebox:</p>
<pre><code>java -jar -Xmx$MEM /pub14/tea/nsc206/jars/juicebox/juicer_tools_2.15.08.jar \
pre --threads $CPU \
${PREFIX}_alignments_sorted.txt \
${PREFIX}.hic \
${PREFIX}_scaffolds_final.fa_chromosome_sizes.txt</code></pre>
<p>This step takes ~10 minutes to run.
When it's finished launch juicebox:</p>
<pre><code>java -jar /pub14/tea/nsc206/jars/juicebox/juicebox_2.15.08.jar</code></pre>
<p>Go to the top menu File -&gt; Open, click on 'Local', navigate to the your <code>~/eukaryote_assembly/5-scaffolding</code> directory and load purged10m.hic</p>
<p>You should see something that looks like this:
<img src="figures/juicer.jpg" alt="Screenshot of juicer with the scaffolded assembly loaded, showing 6 chromosomes and several unplaced contigs" />
This figure clearly shows that there are 6 major scaffolds (and a few tiny ones at the end), and we know that <em>Oscheius onirici</em> has 6 chromosomes.</p>
</div>
<div id="summary-3" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Summary<a href="09-Scaffolding.html#summary-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Not all examples will work as well as this. But with high quality PacBio Hifi data, and high quality Illumina Hi-C data - and tools like Hifiasm and YAHS, we can often get a reference quality chromosomal level assembly.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="08-Mitochondrial_assembly.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="10-Optional_exercises.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
