<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Optional Exercises | Eukaryote genome assembly</title>
  <meta name="description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Optional Exercises | Eukaryote genome assembly" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Optional Exercises | Eukaryote genome assembly" />
  
  <meta name="twitter:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sujai Kumar &amp; Helen Hipperson" />


<meta name="date" content="2023-04-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="09-Scaffolding.html"/>
<link rel="next" href="11-Conclusion.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Eukaryote genome assembly</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Eukaryote_genome_assembly.html"><a href="01-Eukaryote_genome_assembly.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="chapter" data-level="5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html"><i class="fa fa-check"></i><b>5</b> Genome pre-assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#introduction-1"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#obtaining-the-read-data"><i class="fa fa-check"></i><b>5.2</b> Obtaining the read data</a></li>
<li class="chapter" data-level="5.3" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-reads-and-bases"><i class="fa fa-check"></i><b>5.3</b> Counting reads and bases</a></li>
<li class="chapter" data-level="5.4" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-and-analysing-k-mers"><i class="fa fa-check"></i><b>5.4</b> Counting and analysing k-mers</a></li>
<li class="chapter" data-level="5.5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#summary"><i class="fa fa-check"></i><b>5.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Assembly.html"><a href="06-Assembly.html"><i class="fa fa-check"></i><b>6</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Assembly.html"><a href="06-Assembly.html#overview-1"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="06-Assembly.html"><a href="06-Assembly.html#downsample-raw-data"><i class="fa fa-check"></i><b>6.2</b> Downsample raw data</a></li>
<li class="chapter" data-level="6.3" data-path="06-Assembly.html"><a href="06-Assembly.html#run-the-hifiasm-assembler"><i class="fa fa-check"></i><b>6.3</b> Run the Hifiasm assembler</a></li>
<li class="chapter" data-level="6.4" data-path="06-Assembly.html"><a href="06-Assembly.html#summary-1"><i class="fa fa-check"></i><b>6.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html"><i class="fa fa-check"></i><b>7</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#obtaining-the-full-assembly"><i class="fa fa-check"></i><b>7.1</b> Obtaining the full assembly</a></li>
<li class="chapter" data-level="7.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#basic-assembly-statistics"><i class="fa fa-check"></i><b>7.2</b> Basic assembly statistics</a></li>
<li class="chapter" data-level="7.3" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#check-for-contamination"><i class="fa fa-check"></i><b>7.3</b> Check for contamination</a></li>
<li class="chapter" data-level="7.4" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco"><i class="fa fa-check"></i><b>7.4</b> BUSCO</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-lineages"><i class="fa fa-check"></i><b>7.4.1</b> BUSCO lineages</a></li>
<li class="chapter" data-level="7.4.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#run-busco"><i class="fa fa-check"></i><b>7.4.2</b> Run BUSCO</a></li>
<li class="chapter" data-level="7.4.3" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-output"><i class="fa fa-check"></i><b>7.4.3</b> BUSCO output</a></li>
<li class="chapter" data-level="7.4.4" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-questions"><i class="fa fa-check"></i><b>7.4.4</b> BUSCO questions</a></li>
<li class="chapter" data-level="7.4.5" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-summary"><i class="fa fa-check"></i><b>7.4.5</b> BUSCO summary</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#purging-duplications-using-purge_dups"><i class="fa fa-check"></i><b>7.5</b> Purging duplications using purge_dups</a></li>
<li class="chapter" data-level="7.6" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#assessment-recap"><i class="fa fa-check"></i><b>7.6</b> Assessment recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html"><i class="fa fa-check"></i><b>8</b> Mitochondrial assembly</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#introduction-2"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#set-up"><i class="fa fa-check"></i><b>8.2</b> Set up</a></li>
<li class="chapter" data-level="8.3" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#run-mitohifi"><i class="fa fa-check"></i><b>8.3</b> Run MitoHiFi</a></li>
<li class="chapter" data-level="8.4" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#run-mitohifi-on-the-organelle-reference"><i class="fa fa-check"></i><b>8.4</b> Run MitoHifi on the organelle reference</a></li>
<li class="chapter" data-level="8.5" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#summary-2"><i class="fa fa-check"></i><b>8.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html"><i class="fa fa-check"></i><b>9</b> Scaffolding using HiC</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#introduction-3"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#set-up-1"><i class="fa fa-check"></i><b>9.2</b> Set up</a></li>
<li class="chapter" data-level="9.3" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#map-the-hi-c-reads-to-the-genome-assembly"><i class="fa fa-check"></i><b>9.3</b> Map the Hi-C reads to the genome assembly</a></li>
<li class="chapter" data-level="9.4" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#visualise-the-scaffolding-data-in-juicer"><i class="fa fa-check"></i><b>9.4</b> Visualise the scaffolding data in Juicer</a></li>
<li class="chapter" data-level="9.5" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#summary-3"><i class="fa fa-check"></i><b>9.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html"><i class="fa fa-check"></i><b>10</b> Optional Exercises</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#introduction-4"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#preassembly-1"><i class="fa fa-check"></i><b>10.2</b> Preassembly</a></li>
<li class="chapter" data-level="10.3" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#assembly-1"><i class="fa fa-check"></i><b>10.3</b> Assembly</a></li>
<li class="chapter" data-level="10.4" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#assessment"><i class="fa fa-check"></i><b>10.4</b> Assessment</a></li>
<li class="chapter" data-level="10.5" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#mitochondrial-assembly"><i class="fa fa-check"></i><b>10.5</b> Mitochondrial Assembly</a></li>
<li class="chapter" data-level="10.6" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#scaffolding-1"><i class="fa fa-check"></i><b>10.6</b> Scaffolding</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Conclusion.html"><a href="11-Conclusion.html"><i class="fa fa-check"></i><b>11</b> Conclusion</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Eukaryote genome assembly</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="optional" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Optional Exercises<a href="10-Optional_exercises.html#optional" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/exercise.png" width="18%" style="display: block; margin: auto;" /></p>
<div id="introduction-4" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Introduction<a href="10-Optional_exercises.html#introduction-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Here is another PacBio Hifi + Illumina Hi-C dataset for you to practice on. This is a fungus, a type of budding yeast called <em>Metschnikowia zobellii</em></p>
<p>Follow all the steps from pre-assembly through to scaffolding and see if the same steps work here or if you need to do anything different.</p>
<p>Hint: We recommend subsampling the Pacbio Hifi read set to 10 percent of the raw data, and we have already subsampled the Illumina data to 1M read pairs so that this exercise will run in a reasonable amount of time in this course. In reality you would run the whole dataset.</p>
</div>
<div id="preassembly-1" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Preassembly<a href="10-Optional_exercises.html#preassembly-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Make sure you are in the <code>. useassembly</code> terminal window.</p>
<pre><code>READACC=ERR9588941</code></pre>
<pre><code>mkdir -p ~/$READACC/1-preassembly
cd ~/$READACC/1-preassembly

# change these to actual location:
ln -s /pub14/tea/nsc206/NEOF/Euk_assembly/data/ERR9588941.fastq.gz
ln -s /pub14/tea/nsc206/NEOF/Euk_assembly/data/ERR9503460.1m.1.fq.gz
ln -s /pub14/tea/nsc206/NEOF/Euk_assembly/data/ERR9503460.1m.2.fq.gz

seqkit stats $READACC.fastq.gz

kmc -cs2000 -k21 ${READACC}.fastq.gz ${READACC}k21 .
kmc_tools transform ${READACC}k21 histogram ${READACC}k21.histo

# upload the .histo file in http://qb.cshl.edu/genomescope/genomescope2.0/</code></pre>
<p><img src="figures/question_bubble.png" width="18%" style="display: block; margin: auto;" /></p>
How many reads are in the ERR9588941.fastq.gz file?
<div id="radio_IJWULQACAZ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_IJWULQACAZ" value=""></input> <span><strong>61</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IJWULQACAZ" value="answer"></input> <span><strong>121,002</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IJWULQACAZ" value=""></input> <span><strong>1,411,251,333</strong></span></label>
</div>
What is the average length of the reads are in the ERR9588941.fastq.gz file?
<div id="radio_TOWZKNFBYD" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_TOWZKNFBYD" value=""></input> <span><strong>61</strong></span></label><label><input type="radio" autocomplete="off" name="radio_TOWZKNFBYD" value="answer"></input> <span><strong>11,663</strong></span></label><label><input type="radio" autocomplete="off" name="radio_TOWZKNFBYD" value=""></input> <span><strong>29,884</strong></span></label>
</div>
What is the estimated genome size in GenomeScope?
<div id="radio_EVJNHKCJRW" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_EVJNHKCJRW" value=""></input> <span><strong>21 Mb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_EVJNHKCJRW" value="answer"></input> <span><strong>14.35 Mb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_EVJNHKCJRW" value=""></input> <span><strong>13.18 Mb</strong></span></label>
</div>
<p>How does this compare to the known size of the assembly for <em>Metschnikowia zobellii</em>? (You can look this up on NCBI!)</p>
</div>
<div id="assembly-1" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Assembly<a href="10-Optional_exercises.html#assembly-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre><code>mkdir -p ~/$READACC/2-assembly
cd ~/$READACC/2-assembly

seqtk sample ../1-preassembly/${READACC}.fastq.gz 0.10 | gzip -c &gt; ${READACC}.10p.fastq.gz

hifiasm -l 0 -t 16 -o ${READACC}_hifiasm10pl0 ${READACC}.10p.fastq.gz
# should take ~2 min
# you can also try -l 3 and see the difference, remember to change the -o as well
# take care with the file names. Here we have added the suffix `hifiasm` for the assembler, `10p` to indicate that we have used 10% of the data and `l0` to indicate we have set the `l` (lower case letter l) paramter to `0`.

awk &#39;/^S/{print &quot;&gt;&quot;$2;print $3}&#39; ${READACC}_hifiasm10pl0.bp.p_ctg.gfa &gt; ${READACC}_hifiasm10pl0.bp.p_ctg.fa</code></pre>
</div>
<div id="assessment" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Assessment<a href="10-Optional_exercises.html#assessment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre><code>mkdir -p ~/$READACC/3-assessment
cd ~/$READACC/3-assessment

ln -s ../2-assembly/${READACC}_hifiasm10pl0.bp.p_ctg.fa

seqkit stats -a *.fa</code></pre>
<p>Move into your <code>. usetiara</code> terminal window.</p>
<pre><code>READACC=ERR9588941

cd ~/$READACC/3-assessment

ASSEMBLY=${READACC}_hifiasml0pl0.bp.p_ctg.fa

tiara -i $ASSEMBLY -o $ASSEMBLY.tiara -t 16 --pr --tf all</code></pre>
<p>Move back into your <code>. useassembly</code> window.</p>
<pre><code>seqkit stats -a *$ASSEMBLY

# note large number of contigs

busco \
-i ${ASSEMBLY} \
-l saccharomycetes_odb10 \
-m geno \
-o ${ASSEMBLY}_saccharomycetes_odb10 \
--cpu 16</code></pre>
<p><img src="figures/question_bubble.png" width="18%" style="display: block; margin: auto;" /></p>
How many contigs are in the ERR9588941_hifiasml0pl0.bp.p_ctg.fa assembly?
<div id="radio_QGRGDWRWXK" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_QGRGDWRWXK" value=""></input> <span><strong>1</strong></span></label><label><input type="radio" autocomplete="off" name="radio_QGRGDWRWXK" value=""></input> <span><strong>236</strong></span></label><label><input type="radio" autocomplete="off" name="radio_QGRGDWRWXK" value="answer"></input> <span><strong>238</strong></span></label>
</div>
What is the N50 of the ERR9588941_hifiasml0pl0.bp.p_ctg.fa assembly?
<div id="radio_AHSAMWZBKR" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_AHSAMWZBKR" value="answer"></input> <span><strong>149 Kb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_AHSAMWZBKR" value=""></input> <span><strong>104 Kb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_AHSAMWZBKR" value=""></input> <span><strong>14 Kb</strong></span></label>
</div>
What is the percentage of complete and single-copy BUSCOs found?
<div id="radio_CQJINOUTNI" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_CQJINOUTNI" value=""></input> <span><strong>96.6%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CQJINOUTNI" value="answer"></input> <span><strong>32.8%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CQJINOUTNI" value=""></input> <span><strong>63.8%</strong></span></label>
</div>
<pre><code># purge_dups
# no bacteria to remove, so keep them all

ln -s ../1-preassembly/${READACC}.fastq.gz

READS=${READACC}.fastq.gz
CONTIGS=$ASSEMBLY
CPU=16

minimap2 -t $CPU -x map-hifi $CONTIGS $READS | gzip -c - &gt;${CONTIGS}.paf.gz

pbcstat ${CONTIGS}.paf.gz
calcuts PB.stat &gt; ${CONTIGS}_cutoffs 2&gt; ${CONTIGS}_calcults.log

split_fa $CONTIGS &gt;${CONTIGS}.split
minimap2 -t $CPU -xasm5 -DP ${CONTIGS}.split ${CONTIGS}.split | gzip -c - &gt;${CONTIGS}.split.self.paf.gz

purge_dups -2 -T ${PREFIX}_cutoffs -c PB.base.cov ${CONTIGS}.split.self.paf.gz &gt;${CONTIGS}_dups.bed 2&gt;${CONTIGS}_purge_dups.log
get_seqs -e ${CONTIGS}_dups.bed $CONTIGS
mv purged.fa ${CONTIGS}_purged.fa
mv hap.fa ${CONTIGS}_hap.fa

seqkit stats ${CONTIGS} ${CONTIGS}_purged.fa ${CONTIGS}_hap.fa

# this time a lot more sequence got purged, as this was a more heterozygous assembly, as we saw in genomescope</code></pre>
<p><img src="figures/question_bubble.png" width="18%" style="display: block; margin: auto;" /></p>
How many contigs are in the ERR9588941_hifiasml0pl0.bp.p_ctg.fa_purged.fa assembly?
<div id="radio_IQLKUJNAPO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_IQLKUJNAPO" value="answer"></input> <span><strong>110</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IQLKUJNAPO" value=""></input> <span><strong>174</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IQLKUJNAPO" value=""></input> <span><strong>238</strong></span></label>
</div>
What is the size of the ERR9588941_hifiasml0pl0.bp.p_ctg.fa_purged.fa assembly?
<div id="radio_TMSLZONSSJ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_TMSLZONSSJ" value=""></input> <span><strong>23.9 Mb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_TMSLZONSSJ" value="answer"></input> <span><strong>13.6 Mb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_TMSLZONSSJ" value=""></input> <span><strong>10.3 Mb</strong></span></label>
</div>
</div>
<div id="mitochondrial-assembly" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> Mitochondrial Assembly<a href="10-Optional_exercises.html#mitochondrial-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Use the identified mitochondrial contig from the tiara assessment.</p>
<p>First move into your <code>. usemitohifi</code> terminal window.</p>
<pre><code>READACC=ERR9588941

mkdir -p ~/$READACC/4-mito
cd ~/$READACC/4-mito

cp ~/$READACC/3-assessment/mitochondrion_ERR9588941_hifiasm10pl0.bp.p_ctg.fa .

# Find closely related mitogenome

python /pub14/tea/nsc206/git_installs/MitoHiFi/findMitoReference.py \
--species &quot;Metschnikowia zobellii&quot; \
--outfolder ./ \
--min_length 16000

# Run MitoHiFi

python /pub14/tea/nsc206/git_installs/MitoHiFi/mitohifi.py \
-c mitochondrion_ERR9588941_hifiasm10pl0.bp.p_ctg.fa \
-f NC_057617.1.fasta -g NC_057617.1.gb \
-t 1 -o 3

# Note we have used a different mitochondrial genetic code - check this on the NCBI taxonomy page for the study species.

feh final_mitogenome.annotation.png</code></pre>
<p><img src="figures/question_bubble.png" width="18%" style="display: block; margin: auto;" /></p>
What is the size of the final mitogenome?
<div id="radio_RYPJWUXSBX" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_RYPJWUXSBX" value=""></input> <span><strong>16 Kb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_RYPJWUXSBX" value="answer"></input> <span><strong>51 Kb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_RYPJWUXSBX" value=""></input> <span><strong>97 Kb</strong></span></label>
</div>
How many genes are there?
<div id="radio_LQDNPBLSMO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_LQDNPBLSMO" value="answer"></input> <span><strong>39</strong></span></label><label><input type="radio" autocomplete="off" name="radio_LQDNPBLSMO" value=""></input> <span><strong>40</strong></span></label><label><input type="radio" autocomplete="off" name="radio_LQDNPBLSMO" value=""></input> <span><strong>41</strong></span></label>
</div>
<p>Yeast can have mitochondrial genomes up to 100 Kb.</p>
</div>
<div id="scaffolding-1" class="section level2 hasAnchor" number="10.6">
<h2><span class="header-section-number">10.6</span> Scaffolding<a href="10-Optional_exercises.html#scaffolding-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Move into your <code>. usehic</code> terminal window.</p>
<pre><code>READACC=ERR9588941

mkdir -p ~/$READACC/5-scaffolding
cd ~/$READACC/5-scaffolding

ln -s ../3-assessment/${READACC}_hifiasm10pl0.bp.p_ctg.fa_purged.fa
ln -s /pub14/tea/nsc206/NEOF/Euk_assembly/data/ERR9503460.1m.1.fq.gz
ln -s /pub14/tea/nsc206/NEOF/Euk_assembly/data/ERR9503460.1m.2.fq.gz

REF=${READACC}_hifiasm10pl0.bp.p_ctg.fa_purged.fa
FASTQ1=ERR9503460.1m.1.fq.gz
FASTQ2=ERR9503460.1m.2.fq.gz
PREFIX=purged1m
CPU=16
MEM=100G

bwa index $REF
samtools faidx $REF

bwa mem -t $CPU $REF $FASTQ1 $FASTQ2 | samtools view -b - &gt; $PREFIX.bam

yahs $REF $PREFIX.bam -o $PREFIX

juicer pre ${PREFIX}.bin ${PREFIX}_scaffolds_final.agp $REF.fai \
| sort -k2,2d -k6,6d --parallel=$CPU -S$MEM \
| awk &#39;NF&#39; \
&gt; ${PREFIX}_alignments_sorted.txt

awk &#39;/^&gt;/{if (l!=&quot;&quot;) print l; printf &quot;%s\t&quot;, $1; l=0; next}{l+=length($0)}END{print l}&#39; ${PREFIX}_scaffolds_final.fa \
| sed &quot;s/&gt;//&quot; \
&gt; ${PREFIX}_scaffolds_final.fa_chromosome_sizes.txt

java -jar -Xmx$MEM /pub14/tea/nsc206/jars/juicebox/juicer_tools_2.15.08.jar pre --threads $CPU ${PREFIX}_alignments_sorted.txt ${PREFIX}.hic ${PREFIX}_scaffolds_final.fa_chromosome_sizes.txt

java -jar /pub14/tea/nsc206/jars/juicebox/juicebox_2.15.08.jar</code></pre>
<p>You should see something that looks like this:
<img src="figures/funguspurged1m.jpg" alt="Screenshot of juicer with the scaffolded assembly loaded, showing 6 chromosomes and several unplaced contigs" /></p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="09-Scaffolding.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="11-Conclusion.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
