<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Purging duplications | Eukaryote genome assembly</title>
  <meta name="description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Purging duplications | Eukaryote genome assembly" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Purging duplications | Eukaryote genome assembly" />
  
  <meta name="twitter:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sujai Kumar &amp; Helen Hipperson" />


<meta name="date" content="2024-06-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="09-BUSCO.html"/>
<link rel="next" href="11-Assessment_recap.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>Eukaryote genome assembly</a></li>

<li class="divider"></li>
<li class="part"><span><b>Intro</b></span></li>
<li class="chapter" data-level="1" data-path="01-Eukaryote_genome_assembly.html"><a href="01-Eukaryote_genome_assembly.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Eukaryote_genome_assembly.html"><a href="01-Eukaryote_genome_assembly.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="part"><span><b>Assembly steps</b></span></li>
<li class="chapter" data-level="5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html"><i class="fa fa-check"></i><b>5</b> Genome pre-assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#introduction-1"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#obtaining-the-read-data"><i class="fa fa-check"></i><b>5.2</b> Obtaining the read data</a></li>
<li class="chapter" data-level="5.3" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-reads-and-bases"><i class="fa fa-check"></i><b>5.3</b> Counting reads and bases</a></li>
<li class="chapter" data-level="5.4" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-and-analysing-k-mers"><i class="fa fa-check"></i><b>5.4</b> Counting and analysing k-mers</a></li>
<li class="chapter" data-level="5.5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#summary"><i class="fa fa-check"></i><b>5.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Assembly.html"><a href="06-Assembly.html"><i class="fa fa-check"></i><b>6</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Assembly.html"><a href="06-Assembly.html#overview-1"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="06-Assembly.html"><a href="06-Assembly.html#downsample-raw-data"><i class="fa fa-check"></i><b>6.2</b> Downsample raw data</a></li>
<li class="chapter" data-level="6.3" data-path="06-Assembly.html"><a href="06-Assembly.html#run-the-hifiasm-assembler"><i class="fa fa-check"></i><b>6.3</b> Run the Hifiasm assembler</a></li>
<li class="chapter" data-level="6.4" data-path="06-Assembly.html"><a href="06-Assembly.html#summary-1"><i class="fa fa-check"></i><b>6.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Assembly assessment</b></span></li>
<li class="chapter" data-level="7" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html"><i class="fa fa-check"></i><b>7</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#obtaining-the-full-assembly"><i class="fa fa-check"></i><b>7.1</b> Obtaining the full assembly</a></li>
<li class="chapter" data-level="7.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#basic-assembly-statistics"><i class="fa fa-check"></i><b>7.2</b> Basic assembly statistics</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Contamination_check.html"><a href="08-Contamination_check.html"><i class="fa fa-check"></i><b>8</b> Check for contamination</a></li>
<li class="chapter" data-level="9" data-path="09-BUSCO.html"><a href="09-BUSCO.html"><i class="fa fa-check"></i><b>9</b> BUSCO</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-lineages"><i class="fa fa-check"></i><b>9.0.1</b> BUSCO lineages</a></li>
<li class="chapter" data-level="9.0.2" data-path="09-BUSCO.html"><a href="09-BUSCO.html#run-busco"><i class="fa fa-check"></i><b>9.0.2</b> Run BUSCO</a></li>
<li class="chapter" data-level="9.0.3" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-output"><i class="fa fa-check"></i><b>9.0.3</b> BUSCO output</a></li>
<li class="chapter" data-level="9.0.4" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-questions"><i class="fa fa-check"></i><b>9.0.4</b> BUSCO questions</a></li>
<li class="chapter" data-level="9.0.5" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-summary"><i class="fa fa-check"></i><b>9.0.5</b> BUSCO summary</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Purge_duplicates.html"><a href="10-Purge_duplicates.html"><i class="fa fa-check"></i><b>10</b> Purging duplications</a></li>
<li class="chapter" data-level="11" data-path="11-Assessment_recap.html"><a href="11-Assessment_recap.html"><i class="fa fa-check"></i><b>11</b> Assessment recap</a></li>
<li class="part"><span><b>Post-assembly</b></span></li>
<li class="chapter" data-level="12" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html"><i class="fa fa-check"></i><b>12</b> Mitochondrial assembly</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#introduction-2"><i class="fa fa-check"></i><b>12.1</b> Introduction</a></li>
<li class="chapter" data-level="12.2" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#set-up"><i class="fa fa-check"></i><b>12.2</b> Set up</a></li>
<li class="chapter" data-level="12.3" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#run-mitohifi"><i class="fa fa-check"></i><b>12.3</b> Run MitoHiFi</a></li>
<li class="chapter" data-level="12.4" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#run-mitohifi-on-the-organelle-reference"><i class="fa fa-check"></i><b>12.4</b> Run MitoHifi on the organelle reference</a></li>
<li class="chapter" data-level="12.5" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#summary-2"><i class="fa fa-check"></i><b>12.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html"><i class="fa fa-check"></i><b>13</b> Scaffolding using HiC</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#introduction-3"><i class="fa fa-check"></i><b>13.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#set-up-1"><i class="fa fa-check"></i><b>13.2</b> Set up</a></li>
<li class="chapter" data-level="13.3" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#map-the-hi-c-reads-to-the-genome-assembly"><i class="fa fa-check"></i><b>13.3</b> Map the Hi-C reads to the genome assembly</a></li>
<li class="chapter" data-level="13.4" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#visualise-the-scaffolding-data-in-juicer"><i class="fa fa-check"></i><b>13.4</b> Visualise the scaffolding data in Juicer</a></li>
<li class="chapter" data-level="13.5" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#summary-3"><i class="fa fa-check"></i><b>13.5</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Outro</b></span></li>
<li class="chapter" data-level="14" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html"><i class="fa fa-check"></i><b>14</b> Optional Exercises</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#introduction-4"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#preassembly-1"><i class="fa fa-check"></i><b>14.2</b> Preassembly</a></li>
<li class="chapter" data-level="14.3" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#assembly-1"><i class="fa fa-check"></i><b>14.3</b> Assembly</a></li>
<li class="chapter" data-level="14.4" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#assessment"><i class="fa fa-check"></i><b>14.4</b> Assessment</a></li>
<li class="chapter" data-level="14.5" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#mitochondrial-assembly"><i class="fa fa-check"></i><b>14.5</b> Mitochondrial Assembly</a></li>
<li class="chapter" data-level="14.6" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#scaffolding-1"><i class="fa fa-check"></i><b>14.6</b> Scaffolding</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Conclusion.html"><a href="15-Conclusion.html"><i class="fa fa-check"></i><b>15</b> Conclusion</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Eukaryote genome assembly</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="purging-duplications" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Purging duplications<a href="10-Purge_duplicates.html#purging-duplications" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/collapse.png" style="width:200px" />
</center>
<p>If you are assembling a diploid genome, then the small allelic variations between the two copies of a genome can cause several alternate duplicated contigs to be assembled. <code>Hifiasm</code> tries to account for this by generating alternate haplotig assembly files. However, sometimes <code>hifiasm</code> isn't able to clearly distinguish two contigs as being alternate haplotigs of the same genomic region.</p>
<p>This is where <code>purge_dups</code> comes in. It is a piece of software that maps the original reads back to the assembly and tries to figure out which bits are alternate haplotigs and which bits are genuine haploid genome representations.</p>
<p>The GenomeScope plot for our workshop example species <em>Oscheius onirici</em> didn't indicate much heterozygosity, so it is very likely that <code>purge_dups</code> will not find many extra contigs to remove.</p>
<p>For this part of the workshop, we will use only the eukarya part of the genome assembly as identified by <code>Tiara</code>.</p>
<p>First make a sub-folder for <code>purge_dups</code> and change into it:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb28-1"><a href="10-Purge_duplicates.html#cb28-1" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> ~/eukaryote_assembly/3-assessment/purge_dups</span>
<span id="cb28-2"><a href="10-Purge_duplicates.html#cb28-2" tabindex="-1"></a><span class="bu">cd</span> ~/eukaryote_assembly/3-assessment/purge_dups</span></code></pre></div>
<p>Now make a symlink to the eukarya_ part of the assembly, and to the reads from the assembly directory:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb29-1"><a href="10-Purge_duplicates.html#cb29-1" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/eukaryote_assembly/3-assessment/eukarya_ERR7979900_hifiasm.bp.p_ctg.fa</span>
<span id="cb29-2"><a href="10-Purge_duplicates.html#cb29-2" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/eukaryote_assembly/1-preassembly/ERR7979900.fastq</span></code></pre></div>
<p>From now on, we're going to use bash variables extensively, so that you just have to change the names of the files in the variables at the start, and then you can reuse the commands that follow on your own data in future:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb30-1"><a href="10-Purge_duplicates.html#cb30-1" tabindex="-1"></a><span class="va">READS</span><span class="op">=</span>ERR7979900.fastq</span>
<span id="cb30-2"><a href="10-Purge_duplicates.html#cb30-2" tabindex="-1"></a><span class="va">CONTIGS</span><span class="op">=</span>eukarya_ERR7979900_hifiasm.bp.p_ctg.fa</span>
<span id="cb30-3"><a href="10-Purge_duplicates.html#cb30-3" tabindex="-1"></a><span class="va">CPU</span><span class="op">=</span>16</span></code></pre></div>
<div id="mapping" class="section level4 unnumbered hasAnchor">
<h4>Mapping<a href="10-Purge_duplicates.html#mapping" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Map the reads to the assembly (this step will take the longest, ~2 minutes)</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb31-1"><a href="10-Purge_duplicates.html#cb31-1" tabindex="-1"></a><span class="ex">minimap2</span> <span class="at">-t</span> <span class="va">$CPU</span> <span class="at">-x</span> map-hifi <span class="va">$CONTIGS</span> <span class="va">$READS</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb31-2"><a href="10-Purge_duplicates.html#cb31-2" tabindex="-1"></a><span class="fu">gzip</span> <span class="at">-c</span> <span class="at">-</span> <span class="op">&gt;</span> <span class="va">${CONTIGS}</span>.paf.gz</span></code></pre></div>
</div>
<div id="calculate-read-depth-and-cutoffs" class="section level4 unnumbered hasAnchor">
<h4>Calculate read depth and cutoffs<a href="10-Purge_duplicates.html#calculate-read-depth-and-cutoffs" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The first step below (<code>pbcstat</code>) calculates the read depth statistics.</p>
<p>The second step uses the read depth information to decide cutoffs.
If the majority of the contigs are around 20-25 read depth, then a contig with read depth between 10-12 might be a duplicated contig representing the alternate heterozygous part of the assembly.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb32-1"><a href="10-Purge_duplicates.html#cb32-1" tabindex="-1"></a><span class="ex">pbcstat</span> <span class="va">${CONTIGS}</span>.paf.gz</span>
<span id="cb32-2"><a href="10-Purge_duplicates.html#cb32-2" tabindex="-1"></a></span>
<span id="cb32-3"><a href="10-Purge_duplicates.html#cb32-3" tabindex="-1"></a><span class="ex">calcuts</span> PB.stat <span class="op">&gt;</span> <span class="va">${CONTIGS}</span>_cutoffs <span class="dv">2</span><span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb32-4"><a href="10-Purge_duplicates.html#cb32-4" tabindex="-1"></a><span class="va">${CONTIGS}</span>_calcults.log</span></code></pre></div>
</div>
<div id="split-assembly-self-alignment" class="section level4 unnumbered hasAnchor">
<h4>Split assembly &amp; self alignment<a href="10-Purge_duplicates.html#split-assembly-self-alignment" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This step is necessary to see which contigs are almost identical to other contigs.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb33-1"><a href="10-Purge_duplicates.html#cb33-1" tabindex="-1"></a><span class="ex">split_fa</span> <span class="va">$CONTIGS</span> <span class="op">&gt;</span> <span class="va">${CONTIGS}</span>.split</span>
<span id="cb33-2"><a href="10-Purge_duplicates.html#cb33-2" tabindex="-1"></a></span>
<span id="cb33-3"><a href="10-Purge_duplicates.html#cb33-3" tabindex="-1"></a><span class="ex">minimap2</span> <span class="at">-t</span> <span class="va">$CPU</span> <span class="at">-xasm5</span> <span class="dt">\</span></span>
<span id="cb33-4"><a href="10-Purge_duplicates.html#cb33-4" tabindex="-1"></a>-DP <span class="va">${CONTIGS}</span>.split <span class="dt">\</span></span>
<span id="cb33-5"><a href="10-Purge_duplicates.html#cb33-5" tabindex="-1"></a><span class="va">${CONTIGS}</span>.split <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb33-6"><a href="10-Purge_duplicates.html#cb33-6" tabindex="-1"></a><span class="fu">gzip</span> <span class="at">-c</span> <span class="at">-</span> <span class="op">&gt;</span> <span class="va">${CONTIGS}</span>.split.self.paf.gz</span></code></pre></div>
</div>
<div id="purge-haplotigs" class="section level4 unnumbered hasAnchor">
<h4>Purge haplotigs<a href="10-Purge_duplicates.html#purge-haplotigs" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>A haplotig is a contig that comes from the same haplotype.
This is the final step where purge_dups uses three pieces of information: the cutoffs file, the coverage (or read depth file), and the information about which contigs are almost identical to others.
If a contig is lower than the cutoff, and is almost identical to another contig, then it is marked as a haplotig.
The output of this step is <code>hap.fa</code> - the file with alternate haplotigs, and <code>purged.fa</code> - the file with haplotigs removed (i.e., purged).</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb34-1"><a href="10-Purge_duplicates.html#cb34-1" tabindex="-1"></a><span class="ex">purge_dups</span> <span class="at">-2</span> <span class="at">-T</span> <span class="va">${CONTIGS}</span>_cutoffs <span class="dt">\</span></span>
<span id="cb34-2"><a href="10-Purge_duplicates.html#cb34-2" tabindex="-1"></a>-c PB.base.cov <span class="dt">\</span></span>
<span id="cb34-3"><a href="10-Purge_duplicates.html#cb34-3" tabindex="-1"></a><span class="va">${CONTIGS}</span>.split.self.paf.gz <span class="op">&gt;</span><span class="va">${CONTIGS}</span>_dups.bed <span class="dt">\</span></span>
<span id="cb34-4"><a href="10-Purge_duplicates.html#cb34-4" tabindex="-1"></a>2<span class="op">&gt;</span> <span class="va">${CONTIGS}</span>_purge_dups.log</span>
<span id="cb34-5"><a href="10-Purge_duplicates.html#cb34-5" tabindex="-1"></a></span>
<span id="cb34-6"><a href="10-Purge_duplicates.html#cb34-6" tabindex="-1"></a><span class="ex">get_seqs</span> <span class="at">-e</span> <span class="va">${CONTIGS}</span>_dups.bed <span class="va">$CONTIGS</span></span>
<span id="cb34-7"><a href="10-Purge_duplicates.html#cb34-7" tabindex="-1"></a></span>
<span id="cb34-8"><a href="10-Purge_duplicates.html#cb34-8" tabindex="-1"></a><span class="fu">mv</span> purged.fa <span class="va">${CONTIGS}</span>_purged.fa</span>
<span id="cb34-9"><a href="10-Purge_duplicates.html#cb34-9" tabindex="-1"></a></span>
<span id="cb34-10"><a href="10-Purge_duplicates.html#cb34-10" tabindex="-1"></a><span class="fu">mv</span> hap.fa <span class="va">${CONTIGS}</span>_hap.fa</span></code></pre></div>
</div>
<div id="output" class="section level4 unnumbered hasAnchor">
<h4>Output<a href="10-Purge_duplicates.html#output" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Let's take a look at the stats of the two files created:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb35-1"><a href="10-Purge_duplicates.html#cb35-1" tabindex="-1"></a><span class="ex">seqkit</span> stats <span class="va">${CONTIGS}</span> <span class="dt">\</span></span>
<span id="cb35-2"><a href="10-Purge_duplicates.html#cb35-2" tabindex="-1"></a><span class="va">${CONTIGS}</span>_purged.fa <span class="dt">\</span></span>
<span id="cb35-3"><a href="10-Purge_duplicates.html#cb35-3" tabindex="-1"></a><span class="va">${CONTIGS}</span>_hap.fa</span></code></pre></div>
<p>The output should look something like this:</p>
<center>
<img src="figures/pd_stats.png" style="width:1800px" />
</center>
<p>As we suspected, only a small span (226,413 bp) was removed from the assembly, indicating this was a mostly homozygous assembly with almost no alternate haplotigs.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="09-BUSCO.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="11-Assessment_recap.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
