<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Conclusion | Eukaryote genome assembly</title>
  <meta name="description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Conclusion | Eukaryote genome assembly" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Conclusion | Eukaryote genome assembly" />
  
  <meta name="twitter:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sujai Kumar &amp; Helen Hipperson" />


<meta name="date" content="2023-04-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="10-Optional_exercises.html"/>
<link rel="next" href="12-Appendix.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Eukaryote genome assembly</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Eukaryote_genome_assembly.html"><a href="01-Eukaryote_genome_assembly.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="chapter" data-level="5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html"><i class="fa fa-check"></i><b>5</b> Genome pre-assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#introduction-1"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#obtaining-the-read-data"><i class="fa fa-check"></i><b>5.2</b> Obtaining the read data</a></li>
<li class="chapter" data-level="5.3" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-reads-and-bases"><i class="fa fa-check"></i><b>5.3</b> Counting reads and bases</a></li>
<li class="chapter" data-level="5.4" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-and-analysing-k-mers"><i class="fa fa-check"></i><b>5.4</b> Counting and analysing k-mers</a></li>
<li class="chapter" data-level="5.5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#summary"><i class="fa fa-check"></i><b>5.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Assembly.html"><a href="06-Assembly.html"><i class="fa fa-check"></i><b>6</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Assembly.html"><a href="06-Assembly.html#overview-1"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="06-Assembly.html"><a href="06-Assembly.html#downsample-raw-data"><i class="fa fa-check"></i><b>6.2</b> Downsample raw data</a></li>
<li class="chapter" data-level="6.3" data-path="06-Assembly.html"><a href="06-Assembly.html#run-the-hifiasm-assembler"><i class="fa fa-check"></i><b>6.3</b> Run the Hifiasm assembler</a></li>
<li class="chapter" data-level="6.4" data-path="06-Assembly.html"><a href="06-Assembly.html#summary-1"><i class="fa fa-check"></i><b>6.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html"><i class="fa fa-check"></i><b>7</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#obtaining-the-full-assembly"><i class="fa fa-check"></i><b>7.1</b> Obtaining the full assembly</a></li>
<li class="chapter" data-level="7.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#basic-assembly-statistics"><i class="fa fa-check"></i><b>7.2</b> Basic assembly statistics</a></li>
<li class="chapter" data-level="7.3" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#check-for-contamination"><i class="fa fa-check"></i><b>7.3</b> Check for contamination</a></li>
<li class="chapter" data-level="7.4" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco"><i class="fa fa-check"></i><b>7.4</b> BUSCO</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-lineages"><i class="fa fa-check"></i><b>7.4.1</b> BUSCO lineages</a></li>
<li class="chapter" data-level="7.4.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#run-busco"><i class="fa fa-check"></i><b>7.4.2</b> Run BUSCO</a></li>
<li class="chapter" data-level="7.4.3" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-output"><i class="fa fa-check"></i><b>7.4.3</b> BUSCO output</a></li>
<li class="chapter" data-level="7.4.4" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-questions"><i class="fa fa-check"></i><b>7.4.4</b> BUSCO questions</a></li>
<li class="chapter" data-level="7.4.5" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#busco-summary"><i class="fa fa-check"></i><b>7.4.5</b> BUSCO summary</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#purging-duplications-using-purge_dups"><i class="fa fa-check"></i><b>7.5</b> Purging duplications using purge_dups</a></li>
<li class="chapter" data-level="7.6" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#assessment-recap"><i class="fa fa-check"></i><b>7.6</b> Assessment recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html"><i class="fa fa-check"></i><b>8</b> Mitochondrial assembly</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#introduction-2"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#set-up"><i class="fa fa-check"></i><b>8.2</b> Set up</a></li>
<li class="chapter" data-level="8.3" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#run-mitohifi"><i class="fa fa-check"></i><b>8.3</b> Run MitoHiFi</a></li>
<li class="chapter" data-level="8.4" data-path="08-Mitochondrial_assembly.html"><a href="08-Mitochondrial_assembly.html#run-mitohifi-on-the-organelle-reference"><i class="fa fa-check"></i><b>8.4</b> Run MitoHifi on the organelle reference</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html"><i class="fa fa-check"></i><b>9</b> Scaffolding using HiC</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#introduction-3"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#set-up-1"><i class="fa fa-check"></i><b>9.2</b> Set up</a></li>
<li class="chapter" data-level="9.3" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#map-the-hi-c-reads-to-the-genome-assembly"><i class="fa fa-check"></i><b>9.3</b> Map the Hi-C reads to the genome assembly</a></li>
<li class="chapter" data-level="9.4" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#visualise-the-scaffolding-data-in-juicer"><i class="fa fa-check"></i><b>9.4</b> Visualise the scaffolding data in Juicer</a></li>
<li class="chapter" data-level="9.5" data-path="09-Scaffolding.html"><a href="09-Scaffolding.html#summary-2"><i class="fa fa-check"></i><b>9.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html"><i class="fa fa-check"></i><b>10</b> Optional Exercises</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Optional_exercises.html"><a href="10-Optional_exercises.html#introduction-4"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Conclusion.html"><a href="11-Conclusion.html"><i class="fa fa-check"></i><b>11</b> Conclusion</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Conclusion.html"><a href="11-Conclusion.html#report"><i class="fa fa-check"></i><b>11.1</b> Report</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Conclusion.html"><a href="11-Conclusion.html#report-conda-environment-directory"><i class="fa fa-check"></i><b>11.1.1</b> Report: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="11.1.2" data-path="11-Conclusion.html"><a href="11-Conclusion.html#report-quast"><i class="fa fa-check"></i><b>11.1.2</b> Report: QUAST</a></li>
<li class="chapter" data-level="11.1.3" data-path="11-Conclusion.html"><a href="11-Conclusion.html#report-busco"><i class="fa fa-check"></i><b>11.1.3</b> Report: BUSCO</a></li>
<li class="chapter" data-level="11.1.4" data-path="11-Conclusion.html"><a href="11-Conclusion.html#report-multiqc"><i class="fa fa-check"></i><b>11.1.4</b> Report: MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Conclusion.html"><a href="11-Conclusion.html#final-recap"><i class="fa fa-check"></i><b>11.2</b> Final recap</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="12-Appendix.html"><a href="12-Appendix.html"><i class="fa fa-check"></i><b>A</b> Next steps</a></li>
<li class="chapter" data-level="B" data-path="12-Appendix.html"><a href="12-Appendix.html#manuals"><i class="fa fa-check"></i><b>B</b> Manuals</a></li>
<li class="chapter" data-level="C" data-path="12-Appendix.html"><a href="12-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>C</b> Obtaining Read Data</a></li>
<li class="chapter" data-level="D" data-path="12-Appendix.html"><a href="12-Appendix.html#scaffolding-1"><i class="fa fa-check"></i><b>D</b> Scaffolding</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Eukaryote genome assembly</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="conclusion" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Conclusion<a href="11-Conclusion.html#conclusion" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/final_report.png" width="20%" style="display: block; margin: auto;" /></p>
<div id="report" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Report<a href="11-Conclusion.html#report" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Once you have your final assmeblies it is always a good idea to make a final report. This will involve running <code>QUAST</code> and <code>BUSCO</code> on the final assemblies. Additionally, we will use <code>MultiQC</code> to make a html report including the metrics/stats from <code>QUAST</code>, <code>BUSCO</code>, and <code>Prokka</code>.</p>
<div id="report-conda-environment-directory" class="section level3 hasAnchor" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> Report: Conda environment &amp; directory<a href="11-Conclusion.html#report-conda-environment-directory" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will use the <code>bacterial_genome_assessment</code> conda environment for our report generation. Either:</p>
<ul>
<li>Use your currently open <code>bacterial_genome_assessment</code> terminal.</li>
<li>If you need a new <code>bacterial_genome_assessment</code> terminal see the <a href="#assessmentconda">assembly assessment conda activation instructions</a></li>
</ul>
<p>Ensure you are in the <code>final_assembly</code> directory</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb63-1"><a href="11-Conclusion.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/bacterial_assembly/standard_workflow/final_assembly</span></code></pre></div>
</div>
<div id="report-quast" class="section level3 hasAnchor" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> Report: QUAST<a href="11-Conclusion.html#report-quast" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/measuring_tape.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will run <code>QUAST</code> but this time we will include a reference. Make a softlink for the genome reference. This reference genome is the genome used to create the PacBio simulated reads we have been using for assembly.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb64-1"><a href="11-Conclusion.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub39/tea/nsc006/NEOF/bact_assembly/ref/ ~/bacterial_assembly/</span></code></pre></div>
<p>Now we will run <code>QUAST</code> with the added option of <code>-r</code> specifying the reference fasta to be used.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb65-1"><a href="11-Conclusion.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Make QUAST output directory</span></span>
<span id="cb65-2"><a href="11-Conclusion.html#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> quast</span>
<span id="cb65-3"><a href="11-Conclusion.html#cb65-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Run QUAST</span></span>
<span id="cb65-4"><a href="11-Conclusion.html#cb65-4" aria-hidden="true" tabindex="-1"></a><span class="ex">quast</span> <span class="at">-r</span> ../../ref/ecoli_ref.fasta <span class="at">-o</span> quast assembly/ecoli.fasta</span></code></pre></div>
<p>This <code>-r</code> option will add comparison metrics of the provided assembly/assemblies to the reference assembly. This is similar to <code>dnadiff</code> but more basic. We will look at the `QUAST metrics further down.</p>
</div>
<div id="report-busco" class="section level3 hasAnchor" number="11.1.3">
<h3><span class="header-section-number">11.1.3</span> Report: BUSCO<a href="11-Conclusion.html#report-busco" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/universe.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Now to run <code>BUSCO</code> for the assembly.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb66-1"><a href="11-Conclusion.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Make BUSCO directory</span></span>
<span id="cb66-2"><a href="11-Conclusion.html#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> busco</span>
<span id="cb66-3"><a href="11-Conclusion.html#cb66-3" aria-hidden="true" tabindex="-1"></a><span class="co">#run BUSCO</span></span>
<span id="cb66-4"><a href="11-Conclusion.html#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="ex">busco</span> <span class="at">-i</span> assembly/ecoli.fasta <span class="at">-l</span> enterobacterales_odb10 <span class="at">-m</span> geno <span class="at">-o</span> ecoli <span class="at">--out_path</span> ./busco/</span></code></pre></div>
</div>
<div id="report-multiqc" class="section level3 hasAnchor" number="11.1.4">
<h3><span class="header-section-number">11.1.4</span> Report: MultiQC<a href="11-Conclusion.html#report-multiqc" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/multiqc.png" width="25%" style="display: block; margin: auto;" /></p>
<p><code>MultiQC</code> will create a html report from the output from various tools. A list of compatible tools and output files can be found at: <a href="https://multiqc.info/docs/#multiqc-modules" class="uri">https://multiqc.info/docs/#multiqc-modules</a></p>
<p>When running <code>MultiQC</code> the user points it to a directory or multiple directories which contain the output you want included in the report. <code>MultiQC</code> will then search through the files looking for all the compatible files and ignoring the incompatible files.</p>
<p>However, I find <code>MultiQC</code> is very slow to search through files, especially for tools that create a lot of output files. I therefore like to create an <code>input</code> directory where I copy all the input for <code>MultiQC</code> so it will run much quicker. Let's do that now.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb67-1"><a href="11-Conclusion.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Make MultiQC directory</span></span>
<span id="cb67-2"><a href="11-Conclusion.html#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> multiqc</span>
<span id="cb67-3"><a href="11-Conclusion.html#cb67-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Move into MultiQC directory</span></span>
<span id="cb67-4"><a href="11-Conclusion.html#cb67-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> multiqc</span>
<span id="cb67-5"><a href="11-Conclusion.html#cb67-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Make a input directory</span></span>
<span id="cb67-6"><a href="11-Conclusion.html#cb67-6" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> input</span>
<span id="cb67-7"><a href="11-Conclusion.html#cb67-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Copy the relevant QUAST, BUSCO, and Prokka files to be used as MultiQC input</span></span>
<span id="cb67-8"><a href="11-Conclusion.html#cb67-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> ../quast/report.tsv input/</span>
<span id="cb67-9"><a href="11-Conclusion.html#cb67-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> ../busco/ecoli/short_summary.specific.enterobacterales_odb10.ecoli.txt input/</span>
<span id="cb67-10"><a href="11-Conclusion.html#cb67-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> ../prokka/ecoli/ecoli.txt input/</span></code></pre></div>
<p>Now we can run <code>MultiQC</code>.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb68-1"><a href="11-Conclusion.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> <span class="at">--cl-config</span> <span class="st">&quot;prokka_fn_snames: True&quot;</span> <span class="at">-o</span> . input/</span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li><code>--cl-config</code> : This parameter allows you to config how MultiQC will analyse/present data.
<ul>
<li><code>"prokka_fn_snames: True"</code>: <code>MultiQC</code> will use the file names as the sample names. Other wise the sample name is generated using the first line of the <code>Prokka</code> <code>.txt</code> file.</li>
<li>More info on configuring <code>MultiQC</code>: <a href="https://multiqc.info/docs/#configuring-multiqc" class="uri">https://multiqc.info/docs/#configuring-multiqc</a></li>
</ul></li>
<li><code>-o</code> : Output directory.</li>
<li>At the end of the command all the input directories (directories with the <code>MultiQC</code> inputs) are supplied. In this case it is just <code>input/</code> as it contains all our input.</li>
</ul>
<p>Inspect the html report.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb69-1"><a href="11-Conclusion.html#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="ex">firefox</span> multiqc_report.html</span></code></pre></div>
<p>In the <code>QUAST</code> <code>Assembly Statistics</code> table you will see normal contiguity metrics plus new metrics based on comparing the assembly to the reference assembly:</p>
<ul>
<li><strong>Misassemblies</strong>: Number of misassmebly positions in contigs.</li>
<li><strong>Mismatches/100kbp</strong>: Average number of mismatches per 100000 aligned bases.</li>
<li><strong>Indels/100kbp</strong>: Average number of indels per 100000 aligned bases.</li>
<li><strong>Genome Fraction</strong>: Percentage of bases in the reference that aligned to the genome assembly.</li>
</ul>
<p>More info on metrics: <a href="http://quast.sourceforge.net/docs/manual.html#sec3.1" class="uri">http://quast.sourceforge.net/docs/manual.html#sec3.1</a></p>
<p><strong>Questions</strong></p>
<ul>
<li>What percentage of reference bases aligned to the genome assembly?</li>
<li>When compared to the reference is there a higher number of Mismatches or Indels? Do you think Illumina sequencing would cause the same issue?</li>
<li>How have the BUSCO metrics changed from the BUSCO of the original assembly?</li>
</ul>
</div>
</div>
<div id="final-recap" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Final recap<a href="11-Conclusion.html#final-recap" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/recap.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We have produced a circularised, and polished genome assembly. Additionally we have annotated this assembly and assessed it for contiguity and completeness.</p>
<p>The assembly has very good contiguity but not the highest completeness estimation which appears to be due to missassemblies from mismatches but primarily indels (as evidenced by comparing the assembly to the reference).</p>
<p>Below are a few suggestions for how this could be improved in a real project:</p>
<ul>
<li>Better QC of the initial data.
<ul>
<li>Filter out small reads (&lt;500-1kbp).</li>
<li>Correct reads prior to the first assembly.</li>
</ul></li>
<li>Try out other long read assemblers.</li>
<li>Further polishing
<ul>
<li>Polishing after circularisation.</li>
<li>Try out a different polisher.</li>
</ul></li>
<li>Include Illumina reads
<ul>
<li>Hybrid assembly with PacBio and Illumina reads</li>
<li>Polish PacBio reads with Illumina reads</li>
<li>Polish the assembly with the Illumina reads rather than the PacBio reads. PacBio &amp; ONT reads are less accurate and more susceptible to indel errors than Illumina reads.</li>
</ul></li>
</ul>
<p>As you can see there are a lot of ways to try to improve an assembly. That is why it is always important to carry out regular assembly assessment to determine if you are taking the best route. Additionally, it is also good to know if you have reached the optimum assembly from your data, if so there is not much point spending a lot more bioinformatics time to improve the assembly. Some times you need better/more data.</p>

</div>
</div>



<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="10-Optional_exercises.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="12-Appendix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
