<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 14 Optional Exercises | Eukaryote genome assembly</title>
  <meta name="description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 14 Optional Exercises | Eukaryote genome assembly" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 14 Optional Exercises | Eukaryote genome assembly" />
  
  <meta name="twitter:description" content="NEOF book for the Eukaryote genome assembly workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sujai Kumar &amp; Helen Hipperson" />


<meta name="date" content="2024-06-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="13-Scaffolding.html"/>
<link rel="next" href="15-Conclusion.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>Eukaryote genome assembly</a></li>

<li class="divider"></li>
<li class="part"><span><b>Intro</b></span></li>
<li class="chapter" data-level="1" data-path="01-Eukaryote_genome_assembly.html"><a href="01-Eukaryote_genome_assembly.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Eukaryote_genome_assembly.html"><a href="01-Eukaryote_genome_assembly.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="part"><span><b>Assembly steps</b></span></li>
<li class="chapter" data-level="5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html"><i class="fa fa-check"></i><b>5</b> Genome pre-assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#introduction-1"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#obtaining-the-read-data"><i class="fa fa-check"></i><b>5.2</b> Obtaining the read data</a></li>
<li class="chapter" data-level="5.3" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-reads-and-bases"><i class="fa fa-check"></i><b>5.3</b> Counting reads and bases</a></li>
<li class="chapter" data-level="5.4" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#counting-and-analysing-k-mers"><i class="fa fa-check"></i><b>5.4</b> Counting and analysing k-mers</a></li>
<li class="chapter" data-level="5.5" data-path="05-Pre_assembly.html"><a href="05-Pre_assembly.html#summary"><i class="fa fa-check"></i><b>5.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Assembly.html"><a href="06-Assembly.html"><i class="fa fa-check"></i><b>6</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Assembly.html"><a href="06-Assembly.html#overview-1"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="06-Assembly.html"><a href="06-Assembly.html#downsample-raw-data"><i class="fa fa-check"></i><b>6.2</b> Downsample raw data</a></li>
<li class="chapter" data-level="6.3" data-path="06-Assembly.html"><a href="06-Assembly.html#run-the-hifiasm-assembler"><i class="fa fa-check"></i><b>6.3</b> Run the Hifiasm assembler</a></li>
<li class="chapter" data-level="6.4" data-path="06-Assembly.html"><a href="06-Assembly.html#summary-1"><i class="fa fa-check"></i><b>6.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Assembly assessment</b></span></li>
<li class="chapter" data-level="7" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html"><i class="fa fa-check"></i><b>7</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#obtaining-the-full-assembly"><i class="fa fa-check"></i><b>7.1</b> Obtaining the full assembly</a></li>
<li class="chapter" data-level="7.2" data-path="07-Genome_assessment.html"><a href="07-Genome_assessment.html#basic-assembly-statistics"><i class="fa fa-check"></i><b>7.2</b> Basic assembly statistics</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Contamination_check.html"><a href="08-Contamination_check.html"><i class="fa fa-check"></i><b>8</b> Check for contamination</a></li>
<li class="chapter" data-level="9" data-path="09-BUSCO.html"><a href="09-BUSCO.html"><i class="fa fa-check"></i><b>9</b> BUSCO</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-lineages"><i class="fa fa-check"></i><b>9.0.1</b> BUSCO lineages</a></li>
<li class="chapter" data-level="9.0.2" data-path="09-BUSCO.html"><a href="09-BUSCO.html#run-busco"><i class="fa fa-check"></i><b>9.0.2</b> Run BUSCO</a></li>
<li class="chapter" data-level="9.0.3" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-output"><i class="fa fa-check"></i><b>9.0.3</b> BUSCO output</a></li>
<li class="chapter" data-level="9.0.4" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-questions"><i class="fa fa-check"></i><b>9.0.4</b> BUSCO questions</a></li>
<li class="chapter" data-level="9.0.5" data-path="09-BUSCO.html"><a href="09-BUSCO.html#busco-summary"><i class="fa fa-check"></i><b>9.0.5</b> BUSCO summary</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Purge_duplicates.html"><a href="10-Purge_duplicates.html"><i class="fa fa-check"></i><b>10</b> Purging duplications</a></li>
<li class="chapter" data-level="11" data-path="11-Assessment_recap.html"><a href="11-Assessment_recap.html"><i class="fa fa-check"></i><b>11</b> Assessment recap</a></li>
<li class="part"><span><b>Post-assembly</b></span></li>
<li class="chapter" data-level="12" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html"><i class="fa fa-check"></i><b>12</b> Mitochondrial assembly</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#introduction-2"><i class="fa fa-check"></i><b>12.1</b> Introduction</a></li>
<li class="chapter" data-level="12.2" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#set-up"><i class="fa fa-check"></i><b>12.2</b> Set up</a></li>
<li class="chapter" data-level="12.3" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#run-mitohifi"><i class="fa fa-check"></i><b>12.3</b> Run MitoHiFi</a></li>
<li class="chapter" data-level="12.4" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#run-mitohifi-on-the-organelle-reference"><i class="fa fa-check"></i><b>12.4</b> Run MitoHifi on the organelle reference</a></li>
<li class="chapter" data-level="12.5" data-path="12-Mitochondrial_assembly.html"><a href="12-Mitochondrial_assembly.html#summary-2"><i class="fa fa-check"></i><b>12.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html"><i class="fa fa-check"></i><b>13</b> Scaffolding using HiC</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#introduction-3"><i class="fa fa-check"></i><b>13.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#set-up-1"><i class="fa fa-check"></i><b>13.2</b> Set up</a></li>
<li class="chapter" data-level="13.3" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#map-the-hi-c-reads-to-the-genome-assembly"><i class="fa fa-check"></i><b>13.3</b> Map the Hi-C reads to the genome assembly</a></li>
<li class="chapter" data-level="13.4" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#visualise-the-scaffolding-data-in-juicer"><i class="fa fa-check"></i><b>13.4</b> Visualise the scaffolding data in Juicer</a></li>
<li class="chapter" data-level="13.5" data-path="13-Scaffolding.html"><a href="13-Scaffolding.html#summary-3"><i class="fa fa-check"></i><b>13.5</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Outro</b></span></li>
<li class="chapter" data-level="14" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html"><i class="fa fa-check"></i><b>14</b> Optional Exercises</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#introduction-4"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#preassembly-1"><i class="fa fa-check"></i><b>14.2</b> Preassembly</a></li>
<li class="chapter" data-level="14.3" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#assembly-1"><i class="fa fa-check"></i><b>14.3</b> Assembly</a></li>
<li class="chapter" data-level="14.4" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#assessment"><i class="fa fa-check"></i><b>14.4</b> Assessment</a></li>
<li class="chapter" data-level="14.5" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#mitochondrial-assembly"><i class="fa fa-check"></i><b>14.5</b> Mitochondrial Assembly</a></li>
<li class="chapter" data-level="14.6" data-path="14-Optional_exercises.html"><a href="14-Optional_exercises.html#scaffolding-1"><i class="fa fa-check"></i><b>14.6</b> Scaffolding</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Conclusion.html"><a href="15-Conclusion.html"><i class="fa fa-check"></i><b>15</b> Conclusion</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Eukaryote genome assembly</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="optional" class="section level1 hasAnchor" number="14">
<h1><span class="header-section-number">Chapter 14</span> Optional Exercises<a href="14-Optional_exercises.html#optional" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/exercise.png" style="width:200px" />
</center>
<div id="introduction-4" class="section level2 hasAnchor" number="14.1">
<h2><span class="header-section-number">14.1</span> Introduction<a href="14-Optional_exercises.html#introduction-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Here is another PacBio Hifi + Illumina Hi-C dataset for you to practice on. This is a fungus, a type of budding yeast called <em>Metschnikowia zobellii</em></p>
<p>Follow all the steps from pre-assembly through to scaffolding and see if the same steps work here or if you need to do anything different.</p>
<p>Hint: We recommend subsampling the Pacbio Hifi read set to 10 percent of the raw data, and we have already subsampled the Illumina data to 1M read pairs so that this exercise will run in a reasonable amount of time in this course. In reality you would run the whole dataset.</p>
</div>
<div id="preassembly-1" class="section level2 hasAnchor" number="14.2">
<h2><span class="header-section-number">14.2</span> Preassembly<a href="14-Optional_exercises.html#preassembly-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/magnify_glass_27AUG2020.png" style="width:200px" />
</center>
<p>Make sure you are in the <code>. useassembly</code> terminal window.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb59-1"><a href="14-Optional_exercises.html#cb59-1" tabindex="-1"></a><span class="va">READACC</span><span class="op">=</span>ERR9588941</span></code></pre></div>
<div class="sourceCode" id="cb60"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb60-1"><a href="14-Optional_exercises.html#cb60-1" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> ~/<span class="va">$READACC</span>/1-preassembly</span>
<span id="cb60-2"><a href="14-Optional_exercises.html#cb60-2" tabindex="-1"></a><span class="bu">cd</span> ~/<span class="va">$READACC</span>/1-preassembly</span>
<span id="cb60-3"><a href="14-Optional_exercises.html#cb60-3" tabindex="-1"></a></span>
<span id="cb60-4"><a href="14-Optional_exercises.html#cb60-4" tabindex="-1"></a><span class="co"># link to the raw data files:</span></span>
<span id="cb60-5"><a href="14-Optional_exercises.html#cb60-5" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub14/tea/nsc206/NEOF/Euk_assembly/data/ERR9588941.fastq.gz</span>
<span id="cb60-6"><a href="14-Optional_exercises.html#cb60-6" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub14/tea/nsc206/NEOF/Euk_assembly/data/ERR9503460.1m.1.fq.gz</span>
<span id="cb60-7"><a href="14-Optional_exercises.html#cb60-7" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub14/tea/nsc206/NEOF/Euk_assembly/data/ERR9503460.1m.2.fq.gz</span>
<span id="cb60-8"><a href="14-Optional_exercises.html#cb60-8" tabindex="-1"></a></span>
<span id="cb60-9"><a href="14-Optional_exercises.html#cb60-9" tabindex="-1"></a><span class="ex">seqkit</span> stats <span class="va">$READACC</span>.fastq.gz</span>
<span id="cb60-10"><a href="14-Optional_exercises.html#cb60-10" tabindex="-1"></a></span>
<span id="cb60-11"><a href="14-Optional_exercises.html#cb60-11" tabindex="-1"></a><span class="ex">kmc</span> <span class="at">-cs2000</span> <span class="at">-k21</span> <span class="va">${READACC}</span>.fastq.gz <span class="va">${READACC}</span>k21 .</span>
<span id="cb60-12"><a href="14-Optional_exercises.html#cb60-12" tabindex="-1"></a><span class="ex">kmc_tools</span> transform <span class="va">${READACC}</span>k21 histogram <span class="va">${READACC}</span>k21.histo</span>
<span id="cb60-13"><a href="14-Optional_exercises.html#cb60-13" tabindex="-1"></a></span>
<span id="cb60-14"><a href="14-Optional_exercises.html#cb60-14" tabindex="-1"></a><span class="co"># upload the .histo file in http://genomescope.org/genomescope2.0/</span></span></code></pre></div>
<center>
<img src="figures/question_bubble.png" style="width:200px" />
</center>
How many reads are in the ERR9588941.fastq.gz file?
<div id="radio_WGPTJYHBRK" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_WGPTJYHBRK" value=""></input> <span><strong>61</strong></span></label><label><input type="radio" autocomplete="off" name="radio_WGPTJYHBRK" value="answer"></input> <span><strong>121,002</strong></span></label><label><input type="radio" autocomplete="off" name="radio_WGPTJYHBRK" value=""></input> <span><strong>1,411,251,333</strong></span></label>
</div>
What is the average length of the reads are in the ERR9588941.fastq.gz file?
<div id="radio_YTSUTHMPXR" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_YTSUTHMPXR" value=""></input> <span><strong>61</strong></span></label><label><input type="radio" autocomplete="off" name="radio_YTSUTHMPXR" value="answer"></input> <span><strong>11,663</strong></span></label><label><input type="radio" autocomplete="off" name="radio_YTSUTHMPXR" value=""></input> <span><strong>29,884</strong></span></label>
</div>
What is the estimated genome size in GenomeScope?
<div id="radio_LQRTKFVQLI" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_LQRTKFVQLI" value=""></input> <span><strong>21 Mb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_LQRTKFVQLI" value="answer"></input> <span><strong>14.35 Mb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_LQRTKFVQLI" value="answer"></input> <span><strong>13.18 Mb</strong></span></label>
</div>
<p>If you set <strong>Max k-mer coverage: 1000</strong> you will get a slightly different estimate than if you leave this as the default option in GenomeScope. With this data set we don't see the error with the very high coverage we had with the nematode in chapter 5, and setting the coverage limit doesn't affect the genome size estimation (and it is only an estimation) too much.</p>
<p>How does this compare to the known size of the assembly for <em>Metschnikowia zobellii</em>? (You can look this up on NCBI!)</p>
</div>
<div id="assembly-1" class="section level2 hasAnchor" number="14.3">
<h2><span class="header-section-number">14.3</span> Assembly<a href="14-Optional_exercises.html#assembly-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/string.png" style="width:200px" />
</center>
<div class="sourceCode" id="cb61"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb61-1"><a href="14-Optional_exercises.html#cb61-1" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> ~/<span class="va">$READACC</span>/2-assembly</span>
<span id="cb61-2"><a href="14-Optional_exercises.html#cb61-2" tabindex="-1"></a><span class="bu">cd</span> ~/<span class="va">$READACC</span>/2-assembly</span>
<span id="cb61-3"><a href="14-Optional_exercises.html#cb61-3" tabindex="-1"></a></span>
<span id="cb61-4"><a href="14-Optional_exercises.html#cb61-4" tabindex="-1"></a><span class="ex">seqtk</span> sample ../1-preassembly/<span class="va">${READACC}</span>.fastq.gz 0.10 <span class="kw">|</span> <span class="fu">gzip</span> <span class="at">-c</span> <span class="op">&gt;</span> <span class="va">${READACC}</span>.10p.fastq.gz</span>
<span id="cb61-5"><a href="14-Optional_exercises.html#cb61-5" tabindex="-1"></a></span>
<span id="cb61-6"><a href="14-Optional_exercises.html#cb61-6" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="at">-l</span> 0 <span class="at">-t</span> 16 <span class="at">-o</span> <span class="va">${READACC}</span>_hifiasm10pl0 <span class="va">${READACC}</span>.10p.fastq.gz</span>
<span id="cb61-7"><a href="14-Optional_exercises.html#cb61-7" tabindex="-1"></a><span class="co"># should take ~2 min</span></span>
<span id="cb61-8"><a href="14-Optional_exercises.html#cb61-8" tabindex="-1"></a><span class="co"># you can also try -l 3 and see the difference, remember to change the -o as well</span></span>
<span id="cb61-9"><a href="14-Optional_exercises.html#cb61-9" tabindex="-1"></a><span class="co"># take care with the file names. </span></span>
<span id="cb61-10"><a href="14-Optional_exercises.html#cb61-10" tabindex="-1"></a><span class="co"># Here we have added the suffix `hifiasm` for the assembler, </span></span>
<span id="cb61-11"><a href="14-Optional_exercises.html#cb61-11" tabindex="-1"></a><span class="co"># `10p` to indicate that we have used 10% of the data and </span></span>
<span id="cb61-12"><a href="14-Optional_exercises.html#cb61-12" tabindex="-1"></a><span class="co"># `l0` to indicate we have set the `l` (lower case letter l) paramter to `0`.</span></span>
<span id="cb61-13"><a href="14-Optional_exercises.html#cb61-13" tabindex="-1"></a></span>
<span id="cb61-14"><a href="14-Optional_exercises.html#cb61-14" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;/^S/{print &quot;&gt;&quot;$2;print $3}&#39;</span> <span class="va">${READACC}</span>_hifiasm10pl0.bp.p_ctg.gfa <span class="op">&gt;</span> <span class="va">${READACC}</span>_hifiasm10pl0.bp.p_ctg.fa</span></code></pre></div>
</div>
<div id="assessment" class="section level2 hasAnchor" number="14.4">
<h2><span class="header-section-number">14.4</span> Assessment<a href="14-Optional_exercises.html#assessment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/inspect.png" style="width:200px" />
</center>
<div class="sourceCode" id="cb62"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb62-1"><a href="14-Optional_exercises.html#cb62-1" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> ~/<span class="va">$READACC</span>/3-assessment</span>
<span id="cb62-2"><a href="14-Optional_exercises.html#cb62-2" tabindex="-1"></a><span class="bu">cd</span> ~/<span class="va">$READACC</span>/3-assessment</span>
<span id="cb62-3"><a href="14-Optional_exercises.html#cb62-3" tabindex="-1"></a></span>
<span id="cb62-4"><a href="14-Optional_exercises.html#cb62-4" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../2-assembly/<span class="va">${READACC}</span>_hifiasm10pl0.bp.p_ctg.fa</span>
<span id="cb62-5"><a href="14-Optional_exercises.html#cb62-5" tabindex="-1"></a></span>
<span id="cb62-6"><a href="14-Optional_exercises.html#cb62-6" tabindex="-1"></a><span class="ex">seqkit</span> stats <span class="at">-a</span> <span class="pp">*</span>.fa</span></code></pre></div>
<p>Move into your <code>. usetiara</code> terminal window.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb63-1"><a href="14-Optional_exercises.html#cb63-1" tabindex="-1"></a><span class="va">READACC</span><span class="op">=</span>ERR9588941</span>
<span id="cb63-2"><a href="14-Optional_exercises.html#cb63-2" tabindex="-1"></a></span>
<span id="cb63-3"><a href="14-Optional_exercises.html#cb63-3" tabindex="-1"></a><span class="bu">cd</span> ~/<span class="va">$READACC</span>/3-assessment</span>
<span id="cb63-4"><a href="14-Optional_exercises.html#cb63-4" tabindex="-1"></a></span>
<span id="cb63-5"><a href="14-Optional_exercises.html#cb63-5" tabindex="-1"></a><span class="va">ASSEMBLY</span><span class="op">=</span><span class="va">${READACC}</span>_hifiasm10pl0.bp.p_ctg.fa</span>
<span id="cb63-6"><a href="14-Optional_exercises.html#cb63-6" tabindex="-1"></a></span>
<span id="cb63-7"><a href="14-Optional_exercises.html#cb63-7" tabindex="-1"></a><span class="ex">tiara</span> <span class="at">-i</span> <span class="va">$ASSEMBLY</span> <span class="at">-o</span> <span class="va">$ASSEMBLY</span>.tiara <span class="at">-t</span> 16 <span class="at">--pr</span> <span class="at">--tf</span> all</span></code></pre></div>
<p>Move back into your <code>. useassembly</code> window.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb64-1"><a href="14-Optional_exercises.html#cb64-1" tabindex="-1"></a><span class="va">ASSEMBLY</span><span class="op">=</span><span class="va">${READACC}</span>_hifiasm10pl0.bp.p_ctg.fa</span>
<span id="cb64-2"><a href="14-Optional_exercises.html#cb64-2" tabindex="-1"></a></span>
<span id="cb64-3"><a href="14-Optional_exercises.html#cb64-3" tabindex="-1"></a><span class="ex">seqkit</span> stats <span class="at">-a</span> <span class="pp">*</span><span class="va">$ASSEMBLY</span></span>
<span id="cb64-4"><a href="14-Optional_exercises.html#cb64-4" tabindex="-1"></a></span>
<span id="cb64-5"><a href="14-Optional_exercises.html#cb64-5" tabindex="-1"></a><span class="co"># note large number of contigs</span></span>
<span id="cb64-6"><a href="14-Optional_exercises.html#cb64-6" tabindex="-1"></a></span>
<span id="cb64-7"><a href="14-Optional_exercises.html#cb64-7" tabindex="-1"></a><span class="ex">busco</span> <span class="dt">\</span></span>
<span id="cb64-8"><a href="14-Optional_exercises.html#cb64-8" tabindex="-1"></a>-i <span class="va">${ASSEMBLY}</span> <span class="dt">\</span></span>
<span id="cb64-9"><a href="14-Optional_exercises.html#cb64-9" tabindex="-1"></a>-l saccharomycetes_odb10 <span class="dt">\</span></span>
<span id="cb64-10"><a href="14-Optional_exercises.html#cb64-10" tabindex="-1"></a>-m geno <span class="dt">\</span></span>
<span id="cb64-11"><a href="14-Optional_exercises.html#cb64-11" tabindex="-1"></a>-o <span class="va">${ASSEMBLY}</span>_saccharomycetes_odb10 <span class="dt">\</span></span>
<span id="cb64-12"><a href="14-Optional_exercises.html#cb64-12" tabindex="-1"></a>--cpu 16</span></code></pre></div>
<center>
<img src="figures/question_bubble.png" style="width:200px" />
</center>
How many contigs are in the ERR9588941_hifiasml0pl0.bp.p_ctg.fa assembly?
<div id="radio_PELLFFMEIC" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_PELLFFMEIC" value=""></input> <span><strong>1</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PELLFFMEIC" value=""></input> <span><strong>236</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PELLFFMEIC" value="answer"></input> <span><strong>238</strong></span></label>
</div>
What is the N50 of the ERR9588941_hifiasml0pl0.bp.p_ctg.fa assembly?
<div id="radio_OKAPJEBEAO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_OKAPJEBEAO" value="answer"></input> <span><strong>149 Kb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OKAPJEBEAO" value=""></input> <span><strong>104 Kb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OKAPJEBEAO" value=""></input> <span><strong>14 Kb</strong></span></label>
</div>
What is the percentage of complete and single-copy BUSCOs found?
<div id="radio_ZQCENXIASF" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ZQCENXIASF" value=""></input> <span><strong>96.6%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ZQCENXIASF" value="answer"></input> <span><strong>32.8%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ZQCENXIASF" value=""></input> <span><strong>63.8%</strong></span></label>
</div>
<div class="sourceCode" id="cb65"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb65-1"><a href="14-Optional_exercises.html#cb65-1" tabindex="-1"></a><span class="co"># purge_dups</span></span>
<span id="cb65-2"><a href="14-Optional_exercises.html#cb65-2" tabindex="-1"></a><span class="co"># no bacteria to remove, so keep them all</span></span>
<span id="cb65-3"><a href="14-Optional_exercises.html#cb65-3" tabindex="-1"></a></span>
<span id="cb65-4"><a href="14-Optional_exercises.html#cb65-4" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../1-preassembly/<span class="va">${READACC}</span>.fastq.gz</span>
<span id="cb65-5"><a href="14-Optional_exercises.html#cb65-5" tabindex="-1"></a></span>
<span id="cb65-6"><a href="14-Optional_exercises.html#cb65-6" tabindex="-1"></a><span class="va">READS</span><span class="op">=</span><span class="va">${READACC}</span>.fastq.gz</span>
<span id="cb65-7"><a href="14-Optional_exercises.html#cb65-7" tabindex="-1"></a><span class="va">CONTIGS</span><span class="op">=</span><span class="va">$ASSEMBLY</span></span>
<span id="cb65-8"><a href="14-Optional_exercises.html#cb65-8" tabindex="-1"></a><span class="va">CPU</span><span class="op">=</span>16</span>
<span id="cb65-9"><a href="14-Optional_exercises.html#cb65-9" tabindex="-1"></a></span>
<span id="cb65-10"><a href="14-Optional_exercises.html#cb65-10" tabindex="-1"></a><span class="ex">minimap2</span> <span class="at">-t</span> <span class="va">$CPU</span> <span class="at">-x</span> map-hifi <span class="va">$CONTIGS</span> <span class="va">$READS</span> <span class="kw">|</span> <span class="fu">gzip</span> <span class="at">-c</span> <span class="at">-</span> <span class="op">&gt;</span><span class="va">${CONTIGS}</span>.paf.gz</span>
<span id="cb65-11"><a href="14-Optional_exercises.html#cb65-11" tabindex="-1"></a></span>
<span id="cb65-12"><a href="14-Optional_exercises.html#cb65-12" tabindex="-1"></a><span class="ex">pbcstat</span> <span class="va">${CONTIGS}</span>.paf.gz</span>
<span id="cb65-13"><a href="14-Optional_exercises.html#cb65-13" tabindex="-1"></a><span class="ex">calcuts</span> PB.stat <span class="op">&gt;</span> <span class="va">${CONTIGS}</span>_cutoffs <span class="dv">2</span><span class="op">&gt;</span> <span class="va">${CONTIGS}</span>_calcults.log</span>
<span id="cb65-14"><a href="14-Optional_exercises.html#cb65-14" tabindex="-1"></a></span>
<span id="cb65-15"><a href="14-Optional_exercises.html#cb65-15" tabindex="-1"></a><span class="ex">split_fa</span> <span class="va">$CONTIGS</span> <span class="op">&gt;</span><span class="va">${CONTIGS}</span>.split</span>
<span id="cb65-16"><a href="14-Optional_exercises.html#cb65-16" tabindex="-1"></a><span class="ex">minimap2</span> <span class="at">-t</span> <span class="va">$CPU</span> <span class="at">-xasm5</span> <span class="at">-DP</span> <span class="va">${CONTIGS}</span>.split <span class="va">${CONTIGS}</span>.split <span class="kw">|</span> <span class="fu">gzip</span> <span class="at">-c</span> <span class="at">-</span> <span class="op">&gt;</span><span class="va">${CONTIGS}</span>.split.self.paf.gz</span>
<span id="cb65-17"><a href="14-Optional_exercises.html#cb65-17" tabindex="-1"></a></span>
<span id="cb65-18"><a href="14-Optional_exercises.html#cb65-18" tabindex="-1"></a><span class="ex">purge_dups</span> <span class="at">-2</span> <span class="at">-T</span> <span class="va">${CONTIGS}</span>_cutoffs <span class="at">-c</span> PB.base.cov <span class="va">${CONTIGS}</span>.split.self.paf.gz <span class="dt">\</span></span>
<span id="cb65-19"><a href="14-Optional_exercises.html#cb65-19" tabindex="-1"></a><span class="op">&gt;</span><span class="va">${CONTIGS}</span>_dups.bed <span class="dv">2</span><span class="op">&gt;</span><span class="va">${CONTIGS}</span>_purge_dups.log</span>
<span id="cb65-20"><a href="14-Optional_exercises.html#cb65-20" tabindex="-1"></a><span class="ex">get_seqs</span> <span class="at">-e</span> <span class="va">${CONTIGS}</span>_dups.bed <span class="va">$CONTIGS</span></span>
<span id="cb65-21"><a href="14-Optional_exercises.html#cb65-21" tabindex="-1"></a><span class="fu">mv</span> purged.fa <span class="va">${CONTIGS}</span>_purged.fa</span>
<span id="cb65-22"><a href="14-Optional_exercises.html#cb65-22" tabindex="-1"></a><span class="fu">mv</span> hap.fa <span class="va">${CONTIGS}</span>_hap.fa</span>
<span id="cb65-23"><a href="14-Optional_exercises.html#cb65-23" tabindex="-1"></a></span>
<span id="cb65-24"><a href="14-Optional_exercises.html#cb65-24" tabindex="-1"></a><span class="ex">seqkit</span> stats <span class="va">${CONTIGS}</span> <span class="va">${CONTIGS}</span>_purged.fa <span class="va">${CONTIGS}</span>_hap.fa</span>
<span id="cb65-25"><a href="14-Optional_exercises.html#cb65-25" tabindex="-1"></a></span>
<span id="cb65-26"><a href="14-Optional_exercises.html#cb65-26" tabindex="-1"></a><span class="co"># this time a lot more sequence got purged, as this was a more heterozygous assembly, as we saw in genomescope</span></span></code></pre></div>
<center>
<img src="figures/question_bubble.png" style="width:200px" />
</center>
How many contigs are in the ERR9588941_hifiasml0pl0.bp.p_ctg.fa_purged.fa assembly?
<div id="radio_XHXNIINPPG" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_XHXNIINPPG" value="answer"></input> <span><strong>103</strong></span></label><label><input type="radio" autocomplete="off" name="radio_XHXNIINPPG" value=""></input> <span><strong>175</strong></span></label><label><input type="radio" autocomplete="off" name="radio_XHXNIINPPG" value=""></input> <span><strong>238</strong></span></label>
</div>
What is the size of the ERR9588941_hifiasml0pl0.bp.p_ctg.fa_purged.fa assembly?
<div id="radio_VNQCVRNUZO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_VNQCVRNUZO" value=""></input> <span><strong>23.9 Mb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VNQCVRNUZO" value="answer"></input> <span><strong>13.4 Mb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VNQCVRNUZO" value=""></input> <span><strong>10.3 Mb</strong></span></label>
</div>
</div>
<div id="mitochondrial-assembly" class="section level2 hasAnchor" number="14.5">
<h2><span class="header-section-number">14.5</span> Mitochondrial Assembly<a href="14-Optional_exercises.html#mitochondrial-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/circlator.png" style="width:200px" />
</center>
<p>Use the identified mitochondrial contig from the <code>tiara</code> assessment.</p>
<p>First move into your <code>. usemitohifi</code> terminal window.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb66-1"><a href="14-Optional_exercises.html#cb66-1" tabindex="-1"></a><span class="va">READACC</span><span class="op">=</span>ERR9588941</span>
<span id="cb66-2"><a href="14-Optional_exercises.html#cb66-2" tabindex="-1"></a></span>
<span id="cb66-3"><a href="14-Optional_exercises.html#cb66-3" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> ~/<span class="va">$READACC</span>/4-mito</span>
<span id="cb66-4"><a href="14-Optional_exercises.html#cb66-4" tabindex="-1"></a><span class="bu">cd</span> ~/<span class="va">$READACC</span>/4-mito</span>
<span id="cb66-5"><a href="14-Optional_exercises.html#cb66-5" tabindex="-1"></a></span>
<span id="cb66-6"><a href="14-Optional_exercises.html#cb66-6" tabindex="-1"></a><span class="fu">cp</span> ~/<span class="va">$READACC</span>/3-assessment/mitochondrion_ERR9588941_hifiasm10pl0.bp.p_ctg.fa .</span>
<span id="cb66-7"><a href="14-Optional_exercises.html#cb66-7" tabindex="-1"></a></span>
<span id="cb66-8"><a href="14-Optional_exercises.html#cb66-8" tabindex="-1"></a><span class="co"># Find closely related mitogenome</span></span>
<span id="cb66-9"><a href="14-Optional_exercises.html#cb66-9" tabindex="-1"></a></span>
<span id="cb66-10"><a href="14-Optional_exercises.html#cb66-10" tabindex="-1"></a><span class="ex">python</span> /pub14/tea/nsc206/git_installs/MitoHiFi/findMitoReference.py <span class="dt">\</span></span>
<span id="cb66-11"><a href="14-Optional_exercises.html#cb66-11" tabindex="-1"></a>--species <span class="st">&quot;Metschnikowia zobellii&quot;</span> <span class="dt">\</span></span>
<span id="cb66-12"><a href="14-Optional_exercises.html#cb66-12" tabindex="-1"></a>--outfolder ./ <span class="dt">\</span></span>
<span id="cb66-13"><a href="14-Optional_exercises.html#cb66-13" tabindex="-1"></a>--min_length 16000</span>
<span id="cb66-14"><a href="14-Optional_exercises.html#cb66-14" tabindex="-1"></a></span>
<span id="cb66-15"><a href="14-Optional_exercises.html#cb66-15" tabindex="-1"></a><span class="co"># Run MitoHiFi</span></span>
<span id="cb66-16"><a href="14-Optional_exercises.html#cb66-16" tabindex="-1"></a></span>
<span id="cb66-17"><a href="14-Optional_exercises.html#cb66-17" tabindex="-1"></a><span class="ex">python</span> /pub14/tea/nsc206/git_installs/MitoHiFi/mitohifi.py <span class="dt">\</span></span>
<span id="cb66-18"><a href="14-Optional_exercises.html#cb66-18" tabindex="-1"></a>-c mitochondrion_ERR9588941_hifiasm10pl0.bp.p_ctg.fa <span class="dt">\</span></span>
<span id="cb66-19"><a href="14-Optional_exercises.html#cb66-19" tabindex="-1"></a>-f NC_057617.1.fasta <span class="at">-g</span> NC_057617.1.gb <span class="dt">\</span></span>
<span id="cb66-20"><a href="14-Optional_exercises.html#cb66-20" tabindex="-1"></a>-t 1 <span class="at">-o</span> 3</span>
<span id="cb66-21"><a href="14-Optional_exercises.html#cb66-21" tabindex="-1"></a></span>
<span id="cb66-22"><a href="14-Optional_exercises.html#cb66-22" tabindex="-1"></a><span class="co"># Note we have used a different mitochondrial genetic code - </span></span>
<span id="cb66-23"><a href="14-Optional_exercises.html#cb66-23" tabindex="-1"></a><span class="co"># check this on the NCBI taxonomy page for the study species.</span></span>
<span id="cb66-24"><a href="14-Optional_exercises.html#cb66-24" tabindex="-1"></a></span>
<span id="cb66-25"><a href="14-Optional_exercises.html#cb66-25" tabindex="-1"></a><span class="ex">feh</span> final_mitogenome.annotation.png</span></code></pre></div>
<center>
<img src="figures/question_bubble.png" style="width:200px" />
</center>
What is the size of the final mitogenome?
<div id="radio_GYKNNZWBFW" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_GYKNNZWBFW" value=""></input> <span><strong>16 Kb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_GYKNNZWBFW" value="answer"></input> <span><strong>51 Kb</strong></span></label><label><input type="radio" autocomplete="off" name="radio_GYKNNZWBFW" value=""></input> <span><strong>97 Kb</strong></span></label>
</div>
How many genes are there?
<div id="radio_SRTOIXJAWT" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_SRTOIXJAWT" value="answer"></input> <span><strong>39</strong></span></label><label><input type="radio" autocomplete="off" name="radio_SRTOIXJAWT" value=""></input> <span><strong>40</strong></span></label><label><input type="radio" autocomplete="off" name="radio_SRTOIXJAWT" value=""></input> <span><strong>41</strong></span></label>
</div>
<p>Yeast can have mitochondrial genomes up to 100 Kb.</p>
</div>
<div id="scaffolding-1" class="section level2 hasAnchor" number="14.6">
<h2><span class="header-section-number">14.6</span> Scaffolding<a href="14-Optional_exercises.html#scaffolding-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/scaffold.png" style="width:200px" />
</center>
<p>Move into your <code>. usehic</code> terminal window.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb67-1"><a href="14-Optional_exercises.html#cb67-1" tabindex="-1"></a><span class="va">READACC</span><span class="op">=</span>ERR9588941</span>
<span id="cb67-2"><a href="14-Optional_exercises.html#cb67-2" tabindex="-1"></a></span>
<span id="cb67-3"><a href="14-Optional_exercises.html#cb67-3" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> ~/<span class="va">$READACC</span>/5-scaffolding</span>
<span id="cb67-4"><a href="14-Optional_exercises.html#cb67-4" tabindex="-1"></a><span class="bu">cd</span> ~/<span class="va">$READACC</span>/5-scaffolding</span>
<span id="cb67-5"><a href="14-Optional_exercises.html#cb67-5" tabindex="-1"></a></span>
<span id="cb67-6"><a href="14-Optional_exercises.html#cb67-6" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../3-assessment/<span class="va">${READACC}</span>_hifiasm10pl0.bp.p_ctg.fa_purged.fa</span>
<span id="cb67-7"><a href="14-Optional_exercises.html#cb67-7" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub14/tea/nsc206/NEOF/Euk_assembly/data/ERR9503460.1m.1.fq.gz</span>
<span id="cb67-8"><a href="14-Optional_exercises.html#cb67-8" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub14/tea/nsc206/NEOF/Euk_assembly/data/ERR9503460.1m.2.fq.gz</span>
<span id="cb67-9"><a href="14-Optional_exercises.html#cb67-9" tabindex="-1"></a></span>
<span id="cb67-10"><a href="14-Optional_exercises.html#cb67-10" tabindex="-1"></a><span class="va">REF</span><span class="op">=</span><span class="va">${READACC}</span>_hifiasm10pl0.bp.p_ctg.fa_purged.fa</span>
<span id="cb67-11"><a href="14-Optional_exercises.html#cb67-11" tabindex="-1"></a><span class="va">FASTQ1</span><span class="op">=</span>ERR9503460.1m.1.fq.gz</span>
<span id="cb67-12"><a href="14-Optional_exercises.html#cb67-12" tabindex="-1"></a><span class="va">FASTQ2</span><span class="op">=</span>ERR9503460.1m.2.fq.gz</span>
<span id="cb67-13"><a href="14-Optional_exercises.html#cb67-13" tabindex="-1"></a><span class="va">PREFIX</span><span class="op">=</span>purged1m</span>
<span id="cb67-14"><a href="14-Optional_exercises.html#cb67-14" tabindex="-1"></a><span class="va">CPU</span><span class="op">=</span>16</span>
<span id="cb67-15"><a href="14-Optional_exercises.html#cb67-15" tabindex="-1"></a><span class="va">MEM</span><span class="op">=</span>100G</span>
<span id="cb67-16"><a href="14-Optional_exercises.html#cb67-16" tabindex="-1"></a></span>
<span id="cb67-17"><a href="14-Optional_exercises.html#cb67-17" tabindex="-1"></a><span class="ex">bwa</span> index <span class="va">$REF</span></span>
<span id="cb67-18"><a href="14-Optional_exercises.html#cb67-18" tabindex="-1"></a><span class="ex">samtools</span> faidx <span class="va">$REF</span></span>
<span id="cb67-19"><a href="14-Optional_exercises.html#cb67-19" tabindex="-1"></a></span>
<span id="cb67-20"><a href="14-Optional_exercises.html#cb67-20" tabindex="-1"></a><span class="ex">bwa</span> mem <span class="at">-t</span> <span class="va">$CPU</span> <span class="va">$REF</span> <span class="va">$FASTQ1</span> <span class="va">$FASTQ2</span> <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">-b</span> <span class="at">-</span> <span class="op">&gt;</span> <span class="va">$PREFIX</span>.bam</span>
<span id="cb67-21"><a href="14-Optional_exercises.html#cb67-21" tabindex="-1"></a></span>
<span id="cb67-22"><a href="14-Optional_exercises.html#cb67-22" tabindex="-1"></a><span class="ex">yahs</span> <span class="va">$REF</span> <span class="va">$PREFIX</span>.bam <span class="at">-o</span> <span class="va">$PREFIX</span></span>
<span id="cb67-23"><a href="14-Optional_exercises.html#cb67-23" tabindex="-1"></a></span>
<span id="cb67-24"><a href="14-Optional_exercises.html#cb67-24" tabindex="-1"></a><span class="ex">juicer</span> pre <span class="va">${PREFIX}</span>.bin <span class="va">${PREFIX}</span>_scaffolds_final.agp <span class="va">$REF</span>.fai <span class="dt">\</span></span>
<span id="cb67-25"><a href="14-Optional_exercises.html#cb67-25" tabindex="-1"></a><span class="kw">|</span> <span class="fu">sort</span> <span class="at">-k2,2d</span> <span class="at">-k6,6d</span> <span class="at">--parallel</span><span class="op">=</span><span class="va">$CPU</span> <span class="at">-S</span><span class="va">$MEM</span> <span class="dt">\</span></span>
<span id="cb67-26"><a href="14-Optional_exercises.html#cb67-26" tabindex="-1"></a><span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;NF&#39;</span> <span class="dt">\</span></span>
<span id="cb67-27"><a href="14-Optional_exercises.html#cb67-27" tabindex="-1"></a><span class="op">&gt;</span> <span class="va">${PREFIX}</span>_alignments_sorted.txt</span>
<span id="cb67-28"><a href="14-Optional_exercises.html#cb67-28" tabindex="-1"></a></span>
<span id="cb67-29"><a href="14-Optional_exercises.html#cb67-29" tabindex="-1"></a><span class="ex">samtools</span> faidx <span class="va">${PREFIX}</span>_scaffolds_final.fa</span>
<span id="cb67-30"><a href="14-Optional_exercises.html#cb67-30" tabindex="-1"></a></span>
<span id="cb67-31"><a href="14-Optional_exercises.html#cb67-31" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-f1,2</span> <span class="va">${PREFIX}</span>_scaffolds_final.fa.fai <span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb67-32"><a href="14-Optional_exercises.html#cb67-32" tabindex="-1"></a><span class="va">${PREFIX}</span>_scaffolds_final.fa_chromosome_sizes.txt</span>
<span id="cb67-33"><a href="14-Optional_exercises.html#cb67-33" tabindex="-1"></a></span>
<span id="cb67-34"><a href="14-Optional_exercises.html#cb67-34" tabindex="-1"></a><span class="ex">java</span> <span class="at">-jar</span> <span class="at">-Xmx</span><span class="va">$MEM</span> /pub14/tea/nsc206/jars/juicebox/juicer_tools_2.15.08.jar pre <span class="at">--threads</span> <span class="va">$CPU</span> <span class="dt">\</span></span>
<span id="cb67-35"><a href="14-Optional_exercises.html#cb67-35" tabindex="-1"></a><span class="va">${PREFIX}</span>_alignments_sorted.txt <span class="va">${PREFIX}</span>.hic <span class="va">${PREFIX}</span>_scaffolds_final.fa_chromosome_sizes.txt</span>
<span id="cb67-36"><a href="14-Optional_exercises.html#cb67-36" tabindex="-1"></a></span>
<span id="cb67-37"><a href="14-Optional_exercises.html#cb67-37" tabindex="-1"></a><span class="ex">java</span> <span class="at">-jar</span> /pub14/tea/nsc206/jars/juicebox/juicebox_2.15.08.jar</span></code></pre></div>
You should see something that looks like this:
<center>
<img src="figures/juicer2.png" style="width:900px; border-radius:5px" />
</center>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="13-Scaffolding.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="15-Conclusion.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
