<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Importing | R community analysis supplemental</title>
  <meta name="description" content="NEOF book for the R community analysis supplemental" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Importing | R community analysis supplemental" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the R community analysis supplemental" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Importing | R community analysis supplemental" />
  
  <meta name="twitter:description" content="NEOF book for the R community analysis supplemental" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2024-09-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="01-R_community_supplemental.html"/>
<link rel="next" href="03-Plotly.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img 
src="figures/neof_rounded_corners_300ppi.png" 
width=250px></a></li>
<li><a>R community analysis</a></li>

<li class="divider"></li>
<li class="part"><span><b>Intro</b></span></li>
<li class="chapter" data-level="1" data-path="01-R_community_supplemental.html"><a href="01-R_community_supplemental.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-R_community_supplemental.html"><a href="01-R_community_supplemental.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Importing.html"><a href="02-Importing.html"><i class="fa fa-check"></i><b>2</b> Importing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Importing.html"><a href="02-Importing.html#bracken"><i class="fa fa-check"></i><b>2.1</b> Bracken</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="02-Importing.html"><a href="02-Importing.html#bracken-setup"><i class="fa fa-check"></i><b>2.1.1</b> Bracken: setup</a></li>
<li class="chapter" data-level="2.1.2" data-path="02-Importing.html"><a href="02-Importing.html#bracken-convert-to-biom"><i class="fa fa-check"></i><b>2.1.2</b> Bracken: Convert to biom</a></li>
<li class="chapter" data-level="2.1.3" data-path="02-Importing.html"><a href="02-Importing.html#jupyter-notebook"><i class="fa fa-check"></i><b>2.1.3</b> Jupyter notebook</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="02-Importing.html"><a href="02-Importing.html#plain-text-files"><i class="fa fa-check"></i><b>2.2</b> Plain text files</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Plotly.html"><a href="03-Plotly.html"><i class="fa fa-check"></i><b>3</b> R plotly</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Plotly.html"><a href="03-Plotly.html#jupyter-notebook-1"><i class="fa fa-check"></i><b>3.1</b> Jupyter notebook</a></li>
<li class="chapter" data-level="3.2" data-path="03-Plotly.html"><a href="03-Plotly.html#penguin-data"><i class="fa fa-check"></i><b>3.2</b> Penguin data</a></li>
<li class="chapter" data-level="3.3" data-path="03-Plotly.html"><a href="03-Plotly.html#png-with-ggplot2"><i class="fa fa-check"></i><b>3.3</b> PNG with ggplot2</a></li>
<li class="chapter" data-level="3.4" data-path="03-Plotly.html"><a href="03-Plotly.html#html-with-plotly"><i class="fa fa-check"></i><b>3.4</b> HTML with plotly</a></li>
<li class="chapter" data-level="3.5" data-path="03-Plotly.html"><a href="03-Plotly.html#recap"><i class="fa fa-check"></i><b>3.5</b> Recap</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R community analysis supplemental</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="importchap" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Importing<a href="02-Importing.html#importchap" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/data.png" style="width:200px" />
</center>
<p>Apart from <code>QIIME2</code> artifact files, other file types from other programs can be imported into a <code>phyloseq</code> object.</p>
<div id="bracken" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Bracken<a href="02-Importing.html#bracken" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/bracken.png" style="width:200px" />
</center>
<p>A common way to carry out shotgun metagenomics taxonomic abundance profiling is with <code>Kraken2</code> and <code>Bracken</code>.
This section will show you how to import the <code>bracken</code> output.</p>
<p>Carry out the below steps in bash in your home directory (<code>cd ~</code>).</p>
<div id="bracken-setup" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Bracken: setup<a href="02-Importing.html#bracken-setup" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/setup_3.png" style="width:200px" />
</center>
<p>First we need the <code>Bracken</code> output in the <code>Kraken2</code> report output file format.
We can copy over the output made during the NEOF Shotgun metagenomics workshop.</p>
<p><strong>Note</strong>: If you are using <code>Bracken</code> ensure you are using the latest version as older versions do not provide this output.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb1-1"><a href="02-Importing.html#cb1-1" tabindex="-1"></a><span class="fu">mkdir</span> bracken_r_comm</span>
<span id="cb1-2"><a href="02-Importing.html#cb1-2" tabindex="-1"></a><span class="bu">cd</span> bracken_r_comm</span>
<span id="cb1-3"><a href="02-Importing.html#cb1-3" tabindex="-1"></a><span class="fu">cp</span> /pub14/tea/nsc206/NEOF/R_community/bracken_reports/<span class="pp">*</span>breport2 .</span></code></pre></div>
<p>We can also copy over the metadata file which we will use later.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb2-1"><a href="02-Importing.html#cb2-1" tabindex="-1"></a><span class="fu">cp</span> /pub14/tea/nsc206/NEOF/R_community/bracken_reports/KW_metadata.tsv .</span></code></pre></div>
</div>
<div id="bracken-convert-to-biom" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Bracken: Convert to biom<a href="02-Importing.html#bracken-convert-to-biom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/biom_tie.png" style="width:100px" />
</center>
<p>Unfortunately we can't use the <code>bracken</code> report files.
Instead we need to convert them to a single collated biom file with <code>kraken-biom</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb3-1"><a href="02-Importing.html#cb3-1" tabindex="-1"></a><span class="ex">kraken-biom</span> <span class="pp">*</span>breport2 <span class="at">-o</span> KW.biom </span></code></pre></div>
<p><a href="https://biom-format.org/"><strong>Biom</strong></a> files are not human readable.
Therefore to check the file is correct we can create a tsv (tab separated value) file to inspect.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb4-1"><a href="02-Importing.html#cb4-1" tabindex="-1"></a><span class="co">#Create tsv file</span></span>
<span id="cb4-2"><a href="02-Importing.html#cb4-2" tabindex="-1"></a><span class="ex">biom</span> convert <span class="at">-i</span> KW.biom <span class="at">-o</span> KW.tsv <span class="at">--header-key</span> taxonomy <span class="at">--to-tsv</span></span>
<span id="cb4-3"><a href="02-Importing.html#cb4-3" tabindex="-1"></a><span class="co">#inspect tsv file</span></span>
<span id="cb4-4"><a href="02-Importing.html#cb4-4" tabindex="-1"></a><span class="fu">less</span> KW.tsv</span></code></pre></div>
<p>If your file has a header starting with "#OTU ID" and the sample names, followed by rows of OTUs with abundance numbers it should be good.
You'll also have the taxonomy lineage as the last column because we added in <code>--header-key taxonomy</code>.</p>
</div>
<div id="jupyter-notebook" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> Jupyter notebook<a href="02-Importing.html#jupyter-notebook" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/jupyter_notebook.png" style="width:200px" />
</center>
<p>Now we are ready to carry out some R analysis.</p>
<p>Open up <code>jupyter-notebook</code> and create a new <code>R</code> notebook called "Bracken_phyloseq_creation".</p>
<p>At the top of the notebook create a cell to load the libraries we'll need.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb5-1"><a href="02-Importing.html#cb5-1" tabindex="-1"></a><span class="co">#Libraries</span></span>
<span id="cb5-2"><a href="02-Importing.html#cb5-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb5-3"><a href="02-Importing.html#cb5-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;microbiome&quot;</span>)</span></code></pre></div>
<div id="import-biom-into-phyloseq" class="section level4 unnumbered hasAnchor">
<h4>Import biom into phyloseq<a href="02-Importing.html#import-biom-into-phyloseq" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<center>
<img src="figures/phyloseq_logo.png" style="width:100px" />
</center>
<p>The first step is to import the biom file.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb6-1"><a href="02-Importing.html#cb6-1" tabindex="-1"></a><span class="co">#Read in bracken biom file</span></span>
<span id="cb6-2"><a href="02-Importing.html#cb6-2" tabindex="-1"></a>pseq <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">import_biom</span>(<span class="st">&quot;./KW.biom&quot;</span>)</span></code></pre></div>
</div>
<div id="import-sample-data" class="section level4 unnumbered hasAnchor">
<h4>Import sample data<a href="02-Importing.html#import-sample-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<center>
<img src="figures/samples.png" style="width:100px" />
</center>
<p>Next we need to import the sample data.
We have a tab delimited file we will read into R as a <strong>data frame</strong>.
Then we can assign the sample data of <code>pseq</code> to the <strong>data frame</strong>.
As long as the row names contain the same sample names, like in the biom file, it should work correctly.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb7-1"><a href="02-Importing.html#cb7-1" tabindex="-1"></a><span class="co">#Read in metadata</span></span>
<span id="cb7-2"><a href="02-Importing.html#cb7-2" tabindex="-1"></a>metadf <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;KW_metadata.tsv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">check.names =</span> <span class="cn">FALSE</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb7-3"><a href="02-Importing.html#cb7-3" tabindex="-1"></a><span class="co">#Check metadf</span></span>
<span id="cb7-4"><a href="02-Importing.html#cb7-4" tabindex="-1"></a><span class="fu">head</span>(metadf)</span>
<span id="cb7-5"><a href="02-Importing.html#cb7-5" tabindex="-1"></a><span class="co">#Assign sample data</span></span>
<span id="cb7-6"><a href="02-Importing.html#cb7-6" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq) <span class="ot">&lt;-</span> metadf</span></code></pre></div>
<p>We can then summarise our <code>phyloseq</code> object to make sure it seems ok.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb8-1"><a href="02-Importing.html#cb8-1" tabindex="-1"></a><span class="co">#Summarise</span></span>
<span id="cb8-2"><a href="02-Importing.html#cb8-2" tabindex="-1"></a>microbiome<span class="sc">::</span><span class="fu">summarize_phyloseq</span>(pseq)</span></code></pre></div>
</div>
<div id="taxonomy" class="section level4 unnumbered hasAnchor">
<h4>Taxonomy<a href="02-Importing.html#taxonomy" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<center>
<img src="figures/taxa.png" style="width:100px" />
</center>
<p>Let's check out the taxonomy</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb9-1"><a href="02-Importing.html#cb9-1" tabindex="-1"></a><span class="co">#Check taxonomy</span></span>
<span id="cb9-2"><a href="02-Importing.html#cb9-2" tabindex="-1"></a><span class="fu">head</span>(phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(pseq))</span></code></pre></div>
<p>It looks good but we'll carry out 2 tasks.</p>
<p>First we'll change the taxonomy headers from Rank1... to Kingdom....
We know we can do this due to the prefix of the taxonomies ("k__", "p__"...)</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb10-1"><a href="02-Importing.html#cb10-1" tabindex="-1"></a><span class="co">#Change tax column names</span></span>
<span id="cb10-2"><a href="02-Importing.html#cb10-2" tabindex="-1"></a><span class="fu">colnames</span>(phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(pseq)) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb10-3"><a href="02-Importing.html#cb10-3" tabindex="-1"></a><span class="fu">head</span>(phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(pseq))</span></code></pre></div>
<p>This next step is optional.
We are going to aggregate the <code>phyloseq</code> object from taxids to genus.
Shotgun metagenomics is not always the best tool at differentiating between species so genus classifications will be more reliable.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb11-1"><a href="02-Importing.html#cb11-1" tabindex="-1"></a><span class="co">#Aggregate to genus</span></span>
<span id="cb11-2"><a href="02-Importing.html#cb11-2" tabindex="-1"></a>genus_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_taxa</span>(pseq, <span class="st">&quot;Genus&quot;</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-3"><a href="02-Importing.html#cb11-3" tabindex="-1"></a><span class="co">#Summarise</span></span>
<span id="cb11-4"><a href="02-Importing.html#cb11-4" tabindex="-1"></a>microbiome<span class="sc">::</span><span class="fu">summarize_phyloseq</span>(genus_pseq)</span>
<span id="cb11-5"><a href="02-Importing.html#cb11-5" tabindex="-1"></a><span class="co">#Check number of genera</span></span>
<span id="cb11-6"><a href="02-Importing.html#cb11-6" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Genera: &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(genus_pseq)))</span></code></pre></div>
<p>In your own analysis you may even want to use family or order to aggregate to, or to keep using genus.
Know your data, look at the literature, and make an informed decision.
Of course it can be useful to look at multiple levels of taxonomy.</p>
</div>
</div>
</div>
<div id="plain-text-files" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Plain text files<a href="02-Importing.html#plain-text-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/plain_text.png" style="width:100px; background: white; border-radius:5px" />
</center>
<p>If you have plain text files you can read them into R and then phyloseq-ize them.
The phyloseq website has a nice tutorial on how to carry this out: <a href="https://joey711.github.io/phyloseq/import-data" class="uri">https://joey711.github.io/phyloseq/import-data</a>.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="01-R_community_supplemental.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="03-Plotly.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
