<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Basic Introduction to R | RNA-seq gene expression and pathway analysis - supplemental material</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Basic Introduction to R | RNA-seq gene expression and pathway analysis - supplemental material" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Basic Introduction to R | RNA-seq gene expression and pathway analysis - supplemental material" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher, Helen Hipperson, Steve Paterson, Bert Overduin, Matthew Gemmell, Xuan Liu" />


<meta name="date" content="2022-11-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="01-Supplemental.html"/>
<link rel="next" href="03-Count_table.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>RNA-seq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Supplemental.html"><a href="01-Supplemental.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-R_intro.html"><a href="02-R_intro.html"><i class="fa fa-check"></i><b>2</b> Basic Introduction to R</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-R_intro.html"><a href="02-R_intro.html#starting-and-quitting-an-r-session-in-linux"><i class="fa fa-check"></i><b>2.1</b> Starting and quitting an R session in Linux</a></li>
<li class="chapter" data-level="2.2" data-path="02-R_intro.html"><a href="02-R_intro.html#the-console-window"><i class="fa fa-check"></i><b>2.2</b> The console window</a></li>
<li class="chapter" data-level="2.3" data-path="02-R_intro.html"><a href="02-R_intro.html#your-working-directory"><i class="fa fa-check"></i><b>2.3</b> Your working directory</a></li>
<li class="chapter" data-level="2.4" data-path="02-R_intro.html"><a href="02-R_intro.html#packages"><i class="fa fa-check"></i><b>2.4</b> Packages</a></li>
<li class="chapter" data-level="2.5" data-path="02-R_intro.html"><a href="02-R_intro.html#functions"><i class="fa fa-check"></i><b>2.5</b> Functions</a></li>
<li class="chapter" data-level="2.6" data-path="02-R_intro.html"><a href="02-R_intro.html#assigning-values-to-variables"><i class="fa fa-check"></i><b>2.6</b> Assigning values to variables</a></li>
<li class="chapter" data-level="2.7" data-path="02-R_intro.html"><a href="02-R_intro.html#a-few-useful-commands-and-tips"><i class="fa fa-check"></i><b>2.7</b> A few useful commands and tips</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Count_table.html"><a href="03-Count_table.html"><i class="fa fa-check"></i><b>3</b> Making a count table from HTSeq output files</a></li>
<li class="chapter" data-level="4" data-path="04-Further_DESeq2.html"><a href="04-Further_DESeq2.html"><i class="fa fa-check"></i><b>4</b> Further analysis with DESeq2</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Further_DESeq2.html"><a href="04-Further_DESeq2.html#control-for-sex-of-fly"><i class="fa fa-check"></i><b>4.1</b> Control for sex of fly</a></li>
<li class="chapter" data-level="4.2" data-path="04-Further_DESeq2.html"><a href="04-Further_DESeq2.html#fit-with-multiple-groups"><i class="fa fa-check"></i><b>4.2</b> Fit with multiple groups</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RNA-seq gene expression and pathway analysis - supplemental material</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="R" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Basic Introduction to R<a href="02-R_intro.html#R" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/R.png" width="20%" style="display: block; margin: auto;" /></p>
<p>R is an open-source free language and environment which was designed for statistical computing and graphics. It is possible to install R on many UNIX platforms, Windows and MacOS. In addition to its base environment, that contains capabilities for data manipulation, calculation and graphical display, there are many packages (including DADA2) that can be installed and contain additional functions, data and code for analysing, manipulating and plotting different types of data.</p>
<p>You do not need to understand all the code you will be running today but a basic foundation in R can be useful when you come to analysing your own data and running further analysis. Here are some useful links to get you started if you want to learn more about R after this course.</p>
<p><a href="https://swirlstats.com/">Tutorial</a></p>
<p><a href="https://www.r-project.org/">R Website</a></p>
<p><a href="https://cran.r-project.org/">Cran Website</a></p>
<p><a href="https://www.rstudio.com/">Rstudio Website</a></p>
<p>If you are not familiar with R we have describe some of the common terminology and put a few hints to help you understand how to run a few basic R commands below.</p>
<div id="starting-and-quitting-an-r-session-in-linux" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Starting and quitting an R session in Linux<a href="02-R_intro.html#starting-and-quitting-an-r-session-in-linux" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To start an R session type</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="02-R_intro.html#cb2-1" aria-hidden="true" tabindex="-1"></a>R</span></code></pre></div>
<p>To quit an R session type</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="02-R_intro.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">q</span>()</span></code></pre></div>
</div>
<div id="the-console-window" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> The console window<a href="02-R_intro.html#the-console-window" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The console window is the place where the user can type R commands to submit and then view the results, just like the terminal window in linux.</p>
<p>R shows it is ready to accept commands by showing a <code>&gt;</code> prompt. You can then submit a command using the enter button. If you haven’t completed a command (for example you forget to close a set of parentheses) the <code>+</code> symbol will show on a new line until you finish the command.</p>
<p><img src="figures/R_hang.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="your-working-directory" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Your working directory<a href="02-R_intro.html#your-working-directory" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Your working directory is the folder you are currently working on on your computer. If you set this when you read in or write an output to a file R will automatically look for and save files in this directory.</p>
<p>You can set your working directory using:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="02-R_intro.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/path/to/working_directory/&quot;</span>)</span></code></pre></div>
</div>
<div id="packages" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Packages<a href="02-R_intro.html#packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Packages contain additional functions, data and code for analysing, manipulating and plotting different types of data. Many common packages will be installed as default when you install R. Other more specialised packages, such as the DADA2 package, must be installed by the user.</p>
<p>Packages found on The Comprehensive R Archive Network (CRAN) which is R’s central software repository can be installed easily using the following command.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="02-R_intro.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;package_name&quot;</span>)</span></code></pre></div>
<p>Every time you reload R you will need to load the packages you need if they are not one of the ones installed by default. To do this type:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="02-R_intro.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;package_name&quot;</span>)</span></code></pre></div>
</div>
<div id="functions" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Functions<a href="02-R_intro.html#functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Functions are a set of commands that are organised to perform a specific task. There are a lot of functions that come pre-installed in packages in R and others can be installed as parts of packages. It is also possible for users to create their own functions.</p>
<p>For example a specific function can be called like this, we will use the sum function:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="02-R_intro.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="dv">1</span>,<span class="dv">2</span>) <span class="co"># which sums 1+2 and prints 3 to the console window.</span></span></code></pre></div>
<p>And to write your own function you would type something like this:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="02-R_intro.html#cb8-1" aria-hidden="true" tabindex="-1"></a>function_name <span class="ot">&lt;-</span> <span class="cf">function</span>(arg_1, arg_2, ...) {</span>
<span id="cb8-2"><a href="02-R_intro.html#cb8-2" aria-hidden="true" tabindex="-1"></a>   Function body </span>
<span id="cb8-3"><a href="02-R_intro.html#cb8-3" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="assigning-values-to-variables" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Assigning values to variables<a href="02-R_intro.html#assigning-values-to-variables" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A variable provides a means of accessing the data stored in Rs memory. Data can be assigned to a variable using either <code>&lt;-</code> or <code>=</code>. For example:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="02-R_intro.html#cb9-1" aria-hidden="true" tabindex="-1"></a>x<span class="ot">&lt;-</span><span class="fu">sum</span>(<span class="dv">1</span>,<span class="dv">2</span>)</span></code></pre></div>
<p>If you run this command and then type <code>x</code> into the R console <code>3</code> will be printed to the screen. You can then use this variable in another function and assign to another variable if desired:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="02-R_intro.html#cb10-1" aria-hidden="true" tabindex="-1"></a>y<span class="ot">&lt;-</span><span class="fu">sum</span>(x,x)</span>
<span id="cb10-2"><a href="02-R_intro.html#cb10-2" aria-hidden="true" tabindex="-1"></a>y <span class="co"># y would be assigned the value 6 and be printed to the console </span></span></code></pre></div>
<p>When you have a large script containing many commands it is beneficial to give variables logical names which can help you remember what they contain and help you call them quickly when needed.</p>
</div>
<div id="a-few-useful-commands-and-tips" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> A few useful commands and tips<a href="02-R_intro.html#a-few-useful-commands-and-tips" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><code>rm(list=ls())</code> can be used to clear your environment when you start a new analysis. This will delete all the variables you have stored in Rs memory so be careful when you use this.</li>
<li>The <code>read.table()</code> can be used to read in a file. R likes to work with simple formats such as <code>.csv</code> and <code>.txt</code> files.</li>
<li><code>write.table()</code> can be used to write a data frame or a matrix to a file.</li>
<li>Just like linux, remember you can press the up arrow on your keyboard to list previous commands run in the current R session.</li>
<li>To get the more information and help on the function you want to use you can call using <code>?_function.name_</code> e.g.:</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="02-R_intro.html#cb11-1" aria-hidden="true" tabindex="-1"></a>?sum</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="01-Supplemental.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="03-Count_table.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
