<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Further analysis with DESeq2 | RNA-seq gene expression and pathway analysis - supplemental material</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Further analysis with DESeq2 | RNA-seq gene expression and pathway analysis - supplemental material" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Further analysis with DESeq2 | RNA-seq gene expression and pathway analysis - supplemental material" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher, Helen Hipperson, Steve Paterson, Bert Overduin, Matthew Gemmell, Xuan Liu" />


<meta name="date" content="2023-06-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="03-Count_table.html"/>
<link rel="next" href="05-Multiple_samples_loops.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>RNA-seq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Supplemental.html"><a href="01-Supplemental.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-R_intro.html"><a href="02-R_intro.html"><i class="fa fa-check"></i><b>2</b> Basic Introduction to R</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-R_intro.html"><a href="02-R_intro.html#open-a-new-jupyter-notebook"><i class="fa fa-check"></i><b>2.1</b> Open a new Jupyter notebook</a></li>
<li class="chapter" data-level="2.2" data-path="02-R_intro.html"><a href="02-R_intro.html#your-working-directory"><i class="fa fa-check"></i><b>2.2</b> Your working directory</a></li>
<li class="chapter" data-level="2.3" data-path="02-R_intro.html"><a href="02-R_intro.html#packages"><i class="fa fa-check"></i><b>2.3</b> Packages</a></li>
<li class="chapter" data-level="2.4" data-path="02-R_intro.html"><a href="02-R_intro.html#functions"><i class="fa fa-check"></i><b>2.4</b> Functions</a></li>
<li class="chapter" data-level="2.5" data-path="02-R_intro.html"><a href="02-R_intro.html#assigning-values-to-variables"><i class="fa fa-check"></i><b>2.5</b> Assigning values to variables</a></li>
<li class="chapter" data-level="2.6" data-path="02-R_intro.html"><a href="02-R_intro.html#a-few-useful-commands-and-tips"><i class="fa fa-check"></i><b>2.6</b> A few useful commands and tips</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Count_table.html"><a href="03-Count_table.html"><i class="fa fa-check"></i><b>3</b> Making a count table from HTSeq output files</a></li>
<li class="chapter" data-level="4" data-path="04-Further_DESeq2.html"><a href="04-Further_DESeq2.html"><i class="fa fa-check"></i><b>4</b> Further analysis with DESeq2</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Further_DESeq2.html"><a href="04-Further_DESeq2.html#control-for-sex-of-fly"><i class="fa fa-check"></i><b>4.1</b> Control for sex of fly</a></li>
<li class="chapter" data-level="4.2" data-path="04-Further_DESeq2.html"><a href="04-Further_DESeq2.html#fit-with-multiple-groups"><i class="fa fa-check"></i><b>4.2</b> Fit with multiple groups</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Multiple_samples_loops.html"><a href="05-Multiple_samples_loops.html"><i class="fa fa-check"></i><b>5</b> Using loops to process multiple samples</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Multiple_samples_loops.html"><a href="05-Multiple_samples_loops.html#simple-loop-example"><i class="fa fa-check"></i><b>5.1</b> Simple loop example</a></li>
<li class="chapter" data-level="5.2" data-path="05-Multiple_samples_loops.html"><a href="05-Multiple_samples_loops.html#trimmomatic-loop-example"><i class="fa fa-check"></i><b>5.2</b> Trimmomatic loop example</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RNA-seq gene expression and pathway analysis - supplemental material</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="DESeq2continued" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Further analysis with DESeq2<a href="04-Further_DESeq2.html#DESeq2continued" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="control-for-sex-of-fly" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Control for sex of fly<a href="04-Further_DESeq2.html#control-for-sex-of-fly" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/Dros_no_sex.png" width="40%" style="display: block; margin: auto;" /></p>
<p>Here we will briefly cover a couple of slightly more complex models and how you would code them and extract the results. In our first model we looked at a single term (sex).</p>
<p>For the second model, we may be primarily interested in the selection regime of the lines of flies. But, given that there are big differences due to sex of fly, we might want to control for this. We can fit a model of the form expression = sex + regime.</p>
<p>Make sure you are in the ‘Practical_two’ directory, have the ‘usernaseq’ environment activated, open a new Jupyter notebook called <strong>“Suppl03-Further_DESeq2”</strong> and read in the count data and the meta data for the experiment into a new code cell.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="04-Further_DESeq2.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;DESeq2&quot;</span>)</span>
<span id="cb18-2"><a href="04-Further_DESeq2.html#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="04-Further_DESeq2.html#cb18-3" aria-hidden="true" tabindex="-1"></a>counts<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;genecount.set1.tsv&quot;</span>, <span class="at">header=</span>T, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb18-4"><a href="04-Further_DESeq2.html#cb18-4" aria-hidden="true" tabindex="-1"></a>coldata<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;design.set1.tsv&quot;</span>, <span class="at">header=</span>T, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb18-5"><a href="04-Further_DESeq2.html#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="04-Further_DESeq2.html#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure the metadata columns sex and regime are read as factors</span></span>
<span id="cb18-7"><a href="04-Further_DESeq2.html#cb18-7" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>sex <span class="ot">&lt;-</span> <span class="fu">factor</span>(coldata<span class="sc">$</span>sex)</span>
<span id="cb18-8"><a href="04-Further_DESeq2.html#cb18-8" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>regime <span class="ot">&lt;-</span> <span class="fu">factor</span>(coldata<span class="sc">$</span>regime)</span></code></pre></div>
<p>We can now create the DESeq data set for the additive model. As the variable of interest in this case is the “regime”, we put it at the end of the formula and control for the factor “sex”.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="04-Further_DESeq2.html#cb19-1" aria-hidden="true" tabindex="-1"></a>dds2 <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb19-2"><a href="04-Further_DESeq2.html#cb19-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb19-3"><a href="04-Further_DESeq2.html#cb19-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design=</span> <span class="sc">~</span> sex <span class="sc">+</span> regime)</span></code></pre></div>
<p>As before remove lowly expressed genes and run the differential expression analysis.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="04-Further_DESeq2.html#cb20-1" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds2)) <span class="sc">&gt;=</span> <span class="dv">10</span></span>
<span id="cb20-2"><a href="04-Further_DESeq2.html#cb20-2" aria-hidden="true" tabindex="-1"></a>dds2<span class="ot">&lt;-</span> dds2[keep,]</span>
<span id="cb20-3"><a href="04-Further_DESeq2.html#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="04-Further_DESeq2.html#cb20-4" aria-hidden="true" tabindex="-1"></a>dds2 <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds2)</span></code></pre></div>
<p>We are now ready to extract our results from the analysis we have run. First we will compare the evolved (monogamous) regime against the baseline population (the order that we specify the contrast here is important, with B being the reference).</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="04-Further_DESeq2.html#cb21-1" aria-hidden="true" tabindex="-1"></a>resEB<span class="ot">&lt;-</span><span class="fu">results</span>(dds2, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">&quot;regime&quot;</span>, <span class="st">&quot;E&quot;</span>, <span class="st">&quot;B&quot;</span>))</span>
<span id="cb21-2"><a href="04-Further_DESeq2.html#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(resEB)</span>
<span id="cb21-3"><a href="04-Further_DESeq2.html#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(resEB)</span></code></pre></div>
<p>Positive fold changes in gene expression in our results mean that expression was higher for these genes in the evolved line compared to the baseline.</p>
<p>We can do the same again but compare the multiply mated regime against the baseline:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="04-Further_DESeq2.html#cb22-1" aria-hidden="true" tabindex="-1"></a>resMB<span class="ot">&lt;-</span><span class="fu">results</span>(dds2, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">&quot;regime&quot;</span>, <span class="st">&quot;M&quot;</span>,<span class="st">&quot;B&quot;</span>))</span>
<span id="cb22-2"><a href="04-Further_DESeq2.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(resMB)</span>
<span id="cb22-3"><a href="04-Further_DESeq2.html#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(resMB)</span></code></pre></div>
</div>
<div id="fit-with-multiple-groups" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Fit with multiple groups<a href="04-Further_DESeq2.html#fit-with-multiple-groups" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/Dros_3_regimes.png" width="40%" style="display: block; margin: auto;" /></p>
<p>The last model we will show is how to fit a model with multiple groups. In the fly example here, it might be the case that a gene is differentially expressed between regimes M and E, but only in one sex of fly. This is an experiment on sexual selection, so it’s reasonable that selection might act on different genes in the males and the females.</p>
<p>Interaction terms can be added to the design formula, but it can be much simpler to combine the factors of interest into a single factor with all combinations of the original factors. We can then change the design formula to include just this factor. We have done this for the “group” factor in our metadata. For example, BFCH is a female fly from regime B and we can compare the other all other groups (e.g. male-B, female-E, male-E, female-M &amp; male-M) against each other.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="04-Further_DESeq2.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure the metadata column group is set up as a factor</span></span>
<span id="cb23-2"><a href="04-Further_DESeq2.html#cb23-2" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(coldata<span class="sc">$</span>group)</span>
<span id="cb23-3"><a href="04-Further_DESeq2.html#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="04-Further_DESeq2.html#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the new model</span></span>
<span id="cb23-5"><a href="04-Further_DESeq2.html#cb23-5" aria-hidden="true" tabindex="-1"></a>dds3 <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb23-6"><a href="04-Further_DESeq2.html#cb23-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb23-7"><a href="04-Further_DESeq2.html#cb23-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design=</span> <span class="sc">~</span> group)</span></code></pre></div>
<p>Filter and run the analysis as before.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="04-Further_DESeq2.html#cb24-1" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds3)) <span class="sc">&gt;=</span> <span class="dv">10</span></span>
<span id="cb24-2"><a href="04-Further_DESeq2.html#cb24-2" aria-hidden="true" tabindex="-1"></a>dds3<span class="ot">&lt;-</span> dds3[keep,]</span>
<span id="cb24-3"><a href="04-Further_DESeq2.html#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="04-Further_DESeq2.html#cb24-4" aria-hidden="true" tabindex="-1"></a>dds3 <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds3)</span></code></pre></div>
<p>We can now extract the results of interest as we did in the previous example. Let’s look at the regime M vs E contrast, first in females:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="04-Further_DESeq2.html#cb25-1" aria-hidden="true" tabindex="-1"></a>resMFvsEF<span class="ot">&lt;-</span><span class="fu">results</span>(dds3, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">&quot;group&quot;</span>, <span class="st">&quot;MFCH&quot;</span>, <span class="st">&quot;EFCH&quot;</span>))</span>
<span id="cb25-2"><a href="04-Further_DESeq2.html#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(resMFvsEF)</span>
<span id="cb25-3"><a href="04-Further_DESeq2.html#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(resMFvsEF)</span></code></pre></div>
<p>In this case positive expression changes mean that gene expression was higher in multiply mated females compared to evolved (monogamous) females.</p>
<p>And then in males:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="04-Further_DESeq2.html#cb26-1" aria-hidden="true" tabindex="-1"></a>resMMvsEM<span class="ot">&lt;-</span><span class="fu">results</span>(dds3, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">&quot;group&quot;</span>, <span class="st">&quot;MMCH&quot;</span>, <span class="st">&quot;EMCH&quot;</span>))</span>
<span id="cb26-2"><a href="04-Further_DESeq2.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(resMMvsEM)</span>
<span id="cb26-3"><a href="04-Further_DESeq2.html#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(resMMvsEM)</span></code></pre></div>
<p>For more information on model design and set up see the DESeq2 <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#interactions">vignette</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="03-Count_table.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="05-Multiple_samples_loops.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
