<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Gene Co-expression Network Analysis with WGCNA | RNA-seq gene expression and pathway analysis - supplemental material</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Gene Co-expression Network Analysis with WGCNA | RNA-seq gene expression and pathway analysis - supplemental material" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Gene Co-expression Network Analysis with WGCNA | RNA-seq gene expression and pathway analysis - supplemental material" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher, Helen Hipperson, Ewan Harney, Steve Paterson, Bert Overduin, Matthew Gemmell, Xuan Liu" />


<meta name="date" content="2024-06-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="04-Multiple_samples_loops.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>RNA-seq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Supplemental.html"><a href="01-Supplemental.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Supplemental.html"><a href="01-Supplemental.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Count_table.html"><a href="02-Count_table.html"><i class="fa fa-check"></i><b>2</b> Making a count table from HTSeq output files</a></li>
<li class="chapter" data-level="3" data-path="03-Further_DESeq2.html"><a href="03-Further_DESeq2.html"><i class="fa fa-check"></i><b>3</b> Further analysis with DESeq2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Further_DESeq2.html"><a href="03-Further_DESeq2.html#control-for-sex-of-fly"><i class="fa fa-check"></i><b>3.1</b> Control for sex of fly</a></li>
<li class="chapter" data-level="3.2" data-path="03-Further_DESeq2.html"><a href="03-Further_DESeq2.html#fit-with-multiple-groups"><i class="fa fa-check"></i><b>3.2</b> Fit with multiple groups</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Multiple_samples_loops.html"><a href="04-Multiple_samples_loops.html"><i class="fa fa-check"></i><b>4</b> Using loops to process multiple samples</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Multiple_samples_loops.html"><a href="04-Multiple_samples_loops.html#simple-loop-example"><i class="fa fa-check"></i><b>4.1</b> Simple loop example</a></li>
<li class="chapter" data-level="4.2" data-path="04-Multiple_samples_loops.html"><a href="04-Multiple_samples_loops.html#trimmomatic-loop-example"><i class="fa fa-check"></i><b>4.2</b> Trimmomatic loop example</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-WGCNA.html"><a href="05-WGCNA.html"><i class="fa fa-check"></i><b>5</b> Gene Co-expression Network Analysis with WGCNA</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-WGCNA.html"><a href="05-WGCNA.html#introduction-to-wgcna"><i class="fa fa-check"></i><b>5.1</b> Introduction to WGCNA</a></li>
<li class="chapter" data-level="5.2" data-path="05-WGCNA.html"><a href="05-WGCNA.html#set-up"><i class="fa fa-check"></i><b>5.2</b> Set up</a></li>
<li class="chapter" data-level="5.3" data-path="05-WGCNA.html"><a href="05-WGCNA.html#checking-hierarchical-clustering"><i class="fa fa-check"></i><b>5.3</b> Checking hierarchical clustering</a></li>
<li class="chapter" data-level="5.4" data-path="05-WGCNA.html"><a href="05-WGCNA.html#network-construction"><i class="fa fa-check"></i><b>5.4</b> Network construction</a></li>
<li class="chapter" data-level="5.5" data-path="05-WGCNA.html"><a href="05-WGCNA.html#module-detection"><i class="fa fa-check"></i><b>5.5</b> Module detection</a></li>
<li class="chapter" data-level="5.6" data-path="05-WGCNA.html"><a href="05-WGCNA.html#module-treatment-correlation"><i class="fa fa-check"></i><b>5.6</b> Module-treatment correlation</a></li>
<li class="chapter" data-level="5.7" data-path="05-WGCNA.html"><a href="05-WGCNA.html#network-plots-and-hub-genes"><i class="fa fa-check"></i><b>5.7</b> Network plots and hub genes</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RNA-seq gene expression and pathway analysis - supplemental material</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="WGCNA" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Gene Co-expression Network Analysis with WGCNA<a href="05-WGCNA.html#WGCNA" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/correlation.png" style="width:200px" />
</center>
<div id="introduction-to-wgcna" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Introduction to WGCNA<a href="05-WGCNA.html#introduction-to-wgcna" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Weighted Gene Co-Expression Network Analysis (<a href="https://doi.org/10.2202/1544-6115.1128">WGCNA</a>) is a statistical method for identifying groups (modules) of genes with correlated expression (co-expression) in microarray and RNA-seq data. Modular gene expression is summarised using the cocept of module eigengenes, and these values can be correlated with phenotypic traits or experimental treatments. Looking within modules can also reveal 'hub' genes that have a high number of intra-modular connections.</p>
<p>The analysis in this chapter takes transformed count data from a DESeq2 dds object and uses the <a href="https://doi.org/10.1186/1471-2105-9-559">WGCNA</a> R package to analyse similarity of gene expression among genes, group them into modules, and correlate them with phenotypic traits or experimental treatments. Detection of co-expression modules is aided by the <a href="http://www.jstatsoft.org/v46/i11/">flashClust</a> package and the interactions of genes within modules is visualised by plotting network graphs with the <a href="https://r.igraph.org/">igraph</a> package.</p>
<p>This tutorial is adapted from the original tutorial created by WGCNA authors Peter Langfelder and Steve Horvath. Although no longer hosted on the UCLA website, the tutorial is still available from Peter Langfelder's <a href="https://www.dropbox.com/scl/fo/4vqfiysan6rlurfo2pbnk/h?rlkey=thqg8wlpdn4spu3ihjuc1kmlu&amp;e=3&amp;dl=0">private repository</a>. There are also several newer tutorials on the web <a href="https://rpubs.com/natmurad/WGCNA">here</a>, <a href="https://github.com/Lindseynicer/WGCNA_tutorial">here</a>, and <a href="https://bioinformaticsworkbook.org/tutorials/wgcna.html#gsc.tab=0">here</a>, and in the last few years some new packages have appeared, including <a href="https://doi.org/10.1002/tpg2.20323">SimpleTidy_GeneCoEx</a> and <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04179-4">GWENA</a> which perform similar analyses. This tutorial provides an introduction to WGCNA; additonal functions and capabilities are explored more fully in some of these other resources.</p>
</div>
<div id="set-up" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Set up<a href="05-WGCNA.html#set-up" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/setup_4.png" style="width:200px" />
</center>
<p>Make sure you are in the ‘rnaseq/tmp’ directory, have the ‘usernaseq’ environment activated, open a new Jupyter notebook called “Suppl04-WGCNA”. In the first code chunk of the script we need to load a few packages to run these analyses:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb22-1"><a href="05-WGCNA.html#cb22-1" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span>
<span id="cb22-2"><a href="05-WGCNA.html#cb22-2" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb22-3"><a href="05-WGCNA.html#cb22-3" tabindex="-1"></a><span class="fu">library</span>(flashClust)</span>
<span id="cb22-4"><a href="05-WGCNA.html#cb22-4" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb22-5"><a href="05-WGCNA.html#cb22-5" tabindex="-1"></a><span class="fu">library</span>(repr)</span></code></pre></div>
<p>The first step is to import the data. We will use the same data that we used in the DESeq2 analysis. The files are preloaded in this directory. This time we are going to specify the variable of interest as treatment group.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb23-1"><a href="05-WGCNA.html#cb23-1" tabindex="-1"></a><span class="co"># read in counts and metadata</span></span>
<span id="cb23-2"><a href="05-WGCNA.html#cb23-2" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;genecount.set1.tsv&quot;</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb23-3"><a href="05-WGCNA.html#cb23-3" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;design.set1.tsv&quot;</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb23-4"><a href="05-WGCNA.html#cb23-4" tabindex="-1"></a></span>
<span id="cb23-5"><a href="05-WGCNA.html#cb23-5" tabindex="-1"></a><span class="co"># factorise metadata</span></span>
<span id="cb23-6"><a href="05-WGCNA.html#cb23-6" tabindex="-1"></a>coldata<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(coldata<span class="sc">$</span>group)</span></code></pre></div>
<p>As before we are going to create a dds object and run the DESeq function, but with group in the design.
This time we will filter out genes with less than 100 reads across all samples. This is partly in the interests of time - the analysis will run quicker with a smaller dataset - but as with DESeq, it is often useful to remove genes that are relatively uncommon.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb24-1"><a href="05-WGCNA.html#cb24-1" tabindex="-1"></a><span class="co"># create the deseq object</span></span>
<span id="cb24-2"><a href="05-WGCNA.html#cb24-2" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb24-3"><a href="05-WGCNA.html#cb24-3" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb24-4"><a href="05-WGCNA.html#cb24-4" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> group)</span>
<span id="cb24-5"><a href="05-WGCNA.html#cb24-5" tabindex="-1"></a></span>
<span id="cb24-6"><a href="05-WGCNA.html#cb24-6" tabindex="-1"></a><span class="co"># remove genes with less than or equal to 100 reads across all samples</span></span>
<span id="cb24-7"><a href="05-WGCNA.html#cb24-7" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">&gt;=</span> <span class="dv">100</span></span>
<span id="cb24-8"><a href="05-WGCNA.html#cb24-8" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep,]</span>
<span id="cb24-9"><a href="05-WGCNA.html#cb24-9" tabindex="-1"></a></span>
<span id="cb24-10"><a href="05-WGCNA.html#cb24-10" tabindex="-1"></a><span class="co"># Run DESeq</span></span>
<span id="cb24-11"><a href="05-WGCNA.html#cb24-11" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<p>Now that we have run DESeq, we will carry out a transformation of the count data, extract the counts and reformat them for input into WGCNA. The variance stabilisation transformation, or VST, is useful to apply for RNA-seq datasets. Many statistical techniques assume variance is not dependent on the mean, but in RNA-seq, variance does often increase with the mean. VST helps to reduce this dependence.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb25-1"><a href="05-WGCNA.html#cb25-1" tabindex="-1"></a><span class="co"># transform data for use in WGCNA</span></span>
<span id="cb25-2"><a href="05-WGCNA.html#cb25-2" tabindex="-1"></a>t_dds <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-3"><a href="05-WGCNA.html#cb25-3" tabindex="-1"></a></span>
<span id="cb25-4"><a href="05-WGCNA.html#cb25-4" tabindex="-1"></a><span class="co"># export normalised counts</span></span>
<span id="cb25-5"><a href="05-WGCNA.html#cb25-5" tabindex="-1"></a>a_dds <span class="ot">&lt;-</span> <span class="fu">assay</span>(t_dds)</span>
<span id="cb25-6"><a href="05-WGCNA.html#cb25-6" tabindex="-1"></a></span>
<span id="cb25-7"><a href="05-WGCNA.html#cb25-7" tabindex="-1"></a><span class="co"># transpose data frame (swap rows for columns)</span></span>
<span id="cb25-8"><a href="05-WGCNA.html#cb25-8" tabindex="-1"></a>ncount <span class="ot">&lt;-</span> <span class="fu">t</span>(a_dds)</span></code></pre></div>
</div>
<div id="checking-hierarchical-clustering" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Checking hierarchical clustering<a href="05-WGCNA.html#checking-hierarchical-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/fruit_fly_orange.png" style="width:200px" />
</center>
<p>Before proceeding with the network construction, we are going to check how well expression in our samples relates to our treatment groups, and make sure that there are no outliers with vastly different expression.</p>
<p>To begin with we need to reformat our metadata slightly. Our treatment data is categorical, but WGCNA generally prefers the metadata in a binary format. Fortunately, this is easily achieved with a helper function within WGCNA.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb26-1"><a href="05-WGCNA.html#cb26-1" tabindex="-1"></a><span class="co"># create binary variables for group</span></span>
<span id="cb26-2"><a href="05-WGCNA.html#cb26-2" tabindex="-1"></a>bgroup <span class="ot">&lt;-</span> <span class="fu">binarizeCategoricalVariable</span>(coldata<span class="sc">$</span>group,</span>
<span id="cb26-3"><a href="05-WGCNA.html#cb26-3" tabindex="-1"></a>                                  <span class="at">includePairwise =</span> <span class="cn">FALSE</span>,</span>
<span id="cb26-4"><a href="05-WGCNA.html#cb26-4" tabindex="-1"></a>                                  <span class="at">includeLevelVsAll =</span> <span class="cn">TRUE</span>)</span>
<span id="cb26-5"><a href="05-WGCNA.html#cb26-5" tabindex="-1"></a></span>
<span id="cb26-6"><a href="05-WGCNA.html#cb26-6" tabindex="-1"></a><span class="co"># give the new metadata the same names as the previous metadata</span></span>
<span id="cb26-7"><a href="05-WGCNA.html#cb26-7" tabindex="-1"></a><span class="fu">rownames</span>(bgroup) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(coldata)</span>
<span id="cb26-8"><a href="05-WGCNA.html#cb26-8" tabindex="-1"></a></span>
<span id="cb26-9"><a href="05-WGCNA.html#cb26-9" tabindex="-1"></a><span class="co"># makes sure the metadata and counts data are in the same order</span></span>
<span id="cb26-10"><a href="05-WGCNA.html#cb26-10" tabindex="-1"></a>meta <span class="ot">&lt;-</span> bgroup[<span class="fu">rownames</span>(ncount),]</span></code></pre></div>
<p>Now we will cluster our samples based on their expression using the flashClust function. We can then plot that with our treatment information underneath. This will reveal whether we have any outlying samples, and how well the replicates group together.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb27-1"><a href="05-WGCNA.html#cb27-1" tabindex="-1"></a><span class="co"># Cluster samples based on the gene count data</span></span>
<span id="cb27-2"><a href="05-WGCNA.html#cb27-2" tabindex="-1"></a>sampleTree2 <span class="ot">&lt;-</span> <span class="fu">flashClust</span>(<span class="fu">dist</span>(ncount), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb27-3"><a href="05-WGCNA.html#cb27-3" tabindex="-1"></a></span>
<span id="cb27-4"><a href="05-WGCNA.html#cb27-4" tabindex="-1"></a><span class="co"># Convert traits to a color representation: white means 0, red means 1</span></span>
<span id="cb27-5"><a href="05-WGCNA.html#cb27-5" tabindex="-1"></a>traitColors <span class="ot">&lt;-</span> <span class="fu">numbers2colors</span>(meta)</span>
<span id="cb27-6"><a href="05-WGCNA.html#cb27-6" tabindex="-1"></a></span>
<span id="cb27-7"><a href="05-WGCNA.html#cb27-7" tabindex="-1"></a><span class="co"># Plot the sample dendrogram and the colors underneath.</span></span>
<span id="cb27-8"><a href="05-WGCNA.html#cb27-8" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(sampleTree2, traitColors,</span>
<span id="cb27-9"><a href="05-WGCNA.html#cb27-9" tabindex="-1"></a>                    <span class="at">groupLabels =</span> <span class="fu">colnames</span>(meta))</span></code></pre></div>
<center>
<img src="figures/Dendrogram_counts_phenotype.png" style="width:600px; border-radius:15px" />
</center>
<p>As we can see, there are no outliers in the data set which is good! The replicates cluster together for some of our treatment groups, although for others the clustering is less clear.</p>
Which treatment group shows the strongest clustering of replicate samples (ignoring difference between the sexes)?
<div id="radio_UPBMNEMAGN" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_UPBMNEMAGN" value="answer"></input> <span>B - Baseline</span></label><label><input type="radio" autocomplete="off" name="radio_UPBMNEMAGN" value=""></input> <span>E - Evolved monogamy</span></label><label><input type="radio" autocomplete="off" name="radio_UPBMNEMAGN" value=""></input> <span>M - Multiple mates</span></label>
</div>
<p>However, the fact that the replicates do not cluster together may not be that unusual in a comparison where the treatments are relatively slight (we would expect to see stronger clustering when comparing different tissues or development stages for example).</p>
</div>
<div id="network-construction" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Network construction<a href="05-WGCNA.html#network-construction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/topology.png" style="width:200px" />
</center>
<p>The "Weighted" in Weighted Gene Co-expression Analysis refers to the fact that the connections between the nodes (genes) in our network are not simply present or absent (if they were, we would be carrying out Unweighted Gene Co-expression Analysis). Instead there are many interactions that have varying strengths, or weights. Determining the weights is the clever bit of WGCNA.</p>
<p>Without going into too much detail, the network analysis will assign a connection weight to each gene pair in our data set that is based on the similarity of their expression. Raising the co-expression similarity between gene pairs to the power of an exponent (the "soft threshold") enables the topology of our network (the arrangement of nodes and connections) to approximate that of a scale-free network, i.e. one where most nodes have few connections but some nodes have many connections and act as hubs which is what we would expect in a gene expression network.</p>
<p>In order to decide what power we want to use as our soft threshold, we can analyse the network topology at various exponents and see how well it fits a scale-free topology model. The creators of WGCNA suggest to ideally choose a soft threshold at the lowest power for which scale-free topology fit reaches 0.9, although using a slightly lower fit (0.8) is also valid if the values do not reach 0.9.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb28-1"><a href="05-WGCNA.html#cb28-1" tabindex="-1"></a><span class="co"># Call the network topology analysis function</span></span>
<span id="cb28-2"><a href="05-WGCNA.html#cb28-2" tabindex="-1"></a>sft <span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(ncount, <span class="at">powerVector =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">16</span>), <span class="at">verbose =</span> <span class="dv">5</span>)</span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb29-1"><a href="05-WGCNA.html#cb29-1" tabindex="-1"></a><span class="co"># Change some graphical parameters</span></span>
<span id="cb29-2"><a href="05-WGCNA.html#cb29-2" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb29-3"><a href="05-WGCNA.html#cb29-3" tabindex="-1"></a></span>
<span id="cb29-4"><a href="05-WGCNA.html#cb29-4" tabindex="-1"></a><span class="co"># create the plot of scale-free topology fit index as a function of the soft-thresholding power</span></span>
<span id="cb29-5"><a href="05-WGCNA.html#cb29-5" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], </span>
<span id="cb29-6"><a href="05-WGCNA.html#cb29-6" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>])<span class="sc">*</span>sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb29-7"><a href="05-WGCNA.html#cb29-7" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,</span>
<span id="cb29-8"><a href="05-WGCNA.html#cb29-8" tabindex="-1"></a>     <span class="at">ylab=</span><span class="st">&quot;SFT Model Fit, R2&quot;</span>,</span>
<span id="cb29-9"><a href="05-WGCNA.html#cb29-9" tabindex="-1"></a>     <span class="at">type=</span><span class="st">&quot;n&quot;</span>) <span class="sc">+</span></span>
<span id="cb29-10"><a href="05-WGCNA.html#cb29-10" tabindex="-1"></a>     <span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>])<span class="sc">*</span>sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb29-11"><a href="05-WGCNA.html#cb29-11" tabindex="-1"></a>     <span class="at">labels=</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">16</span>),<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb29-12"><a href="05-WGCNA.html#cb29-12" tabindex="-1"></a></span>
<span id="cb29-13"><a href="05-WGCNA.html#cb29-13" tabindex="-1"></a><span class="co"># plot line corresponding to R^2 cut-off of h</span></span>
<span id="cb29-14"><a href="05-WGCNA.html#cb29-14" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fl">0.9</span>,<span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span></code></pre></div>
<center>
<img src="figures/SFT_vs_softThresh.png" style="width:500px; border-radius:15px" />
</center>
<p>As we can see, the topology of our network approaches a scale-free topology fit of 0.9 at around 6, so we will use this as our soft power.</p>
<p>The next steps will:
1) Apply the chosen soft thresholding power to calculate the adjacency of gene pairs, i.e. how close every gene is to every other gene,
2) Calculate the Topological Overlap Matrix (TOM). Nodes which overlap many other nodes have high connectivity or centrality and are assigned high TOM scores,
3) Calculate the dissimilarity in TOM, which will allow us to identify discrete modules of distinct gene expression.</p>
<p>Running the following code chunk may take a minute or two. While it is running, the book icon in the firefox tab for the jupyter notebook will be replaced by a sand timer. When it has finished running you should see the following message:</p>
<p><left>
<img src="figures/TOM_output_message.png" style="width:300px; border-radius:15px" />
</left></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb30-1"><a href="05-WGCNA.html#cb30-1" tabindex="-1"></a><span class="co"># select our soft power value </span></span>
<span id="cb30-2"><a href="05-WGCNA.html#cb30-2" tabindex="-1"></a>softPower <span class="ot">&lt;-</span> <span class="dv">6</span></span>
<span id="cb30-3"><a href="05-WGCNA.html#cb30-3" tabindex="-1"></a></span>
<span id="cb30-4"><a href="05-WGCNA.html#cb30-4" tabindex="-1"></a><span class="co"># calculate network adjacency</span></span>
<span id="cb30-5"><a href="05-WGCNA.html#cb30-5" tabindex="-1"></a>adjacency <span class="ot">&lt;-</span> <span class="fu">adjacency</span>(ncount, <span class="at">power =</span> softPower)</span>
<span id="cb30-6"><a href="05-WGCNA.html#cb30-6" tabindex="-1"></a></span>
<span id="cb30-7"><a href="05-WGCNA.html#cb30-7" tabindex="-1"></a><span class="co"># turn adjacency into topological overlap matrix (TOM)</span></span>
<span id="cb30-8"><a href="05-WGCNA.html#cb30-8" tabindex="-1"></a>TOM <span class="ot">&lt;-</span> <span class="fu">TOMsimilarity</span>(adjacency)</span>
<span id="cb30-9"><a href="05-WGCNA.html#cb30-9" tabindex="-1"></a></span>
<span id="cb30-10"><a href="05-WGCNA.html#cb30-10" tabindex="-1"></a><span class="co"># we use the dissimilarity (1-overlap) for module detection</span></span>
<span id="cb30-11"><a href="05-WGCNA.html#cb30-11" tabindex="-1"></a>dissTOM <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">-</span>TOM</span>
<span id="cb30-12"><a href="05-WGCNA.html#cb30-12" tabindex="-1"></a></span>
<span id="cb30-13"><a href="05-WGCNA.html#cb30-13" tabindex="-1"></a><span class="co"># remove the adjacency object, which we no longer need</span></span>
<span id="cb30-14"><a href="05-WGCNA.html#cb30-14" tabindex="-1"></a><span class="fu">rm</span>(adjacency)</span></code></pre></div>
</div>
<div id="module-detection" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> Module detection<a href="05-WGCNA.html#module-detection" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/dendrogram.png" style="width:200px" />
</center>
<p>Now that we have our weighted gene co-expression network for all genes, we can look at connectivity by splitting groups of highly connecting genes into co-expression modules. This couples a hierarchical clustering approach with the TOM dissimilarity we calculated in the last step. We also have to specify a minimum module size. The creators of WGCNA recommend a minimum module size of 30. Here we are going to use 50, but with your own data you may want to experiment with different minimum sizes (&gt;30).</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb31-1"><a href="05-WGCNA.html#cb31-1" tabindex="-1"></a><span class="co"># cluster genes based on TOM dissimilarity</span></span>
<span id="cb31-2"><a href="05-WGCNA.html#cb31-2" tabindex="-1"></a>geneTree <span class="ot">&lt;-</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(dissTOM), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb31-3"><a href="05-WGCNA.html#cb31-3" tabindex="-1"></a></span>
<span id="cb31-4"><a href="05-WGCNA.html#cb31-4" tabindex="-1"></a><span class="co"># set a minimum module size</span></span>
<span id="cb31-5"><a href="05-WGCNA.html#cb31-5" tabindex="-1"></a>minModuleSize <span class="ot">&lt;-</span> <span class="dv">50</span></span>
<span id="cb31-6"><a href="05-WGCNA.html#cb31-6" tabindex="-1"></a></span>
<span id="cb31-7"><a href="05-WGCNA.html#cb31-7" tabindex="-1"></a><span class="co"># module identification</span></span>
<span id="cb31-8"><a href="05-WGCNA.html#cb31-8" tabindex="-1"></a>dynamicMods <span class="ot">&lt;-</span> <span class="fu">cutreeDynamic</span>(<span class="at">dendro =</span> geneTree, <span class="at">distM =</span> dissTOM,</span>
<span id="cb31-9"><a href="05-WGCNA.html#cb31-9" tabindex="-1"></a>                            <span class="at">deepSplit =</span> <span class="dv">2</span>, <span class="at">pamRespectsDendro =</span> <span class="cn">FALSE</span>,</span>
<span id="cb31-10"><a href="05-WGCNA.html#cb31-10" tabindex="-1"></a>                            <span class="at">minClusterSize =</span> minModuleSize)</span>
<span id="cb31-11"><a href="05-WGCNA.html#cb31-11" tabindex="-1"></a></span>
<span id="cb31-12"><a href="05-WGCNA.html#cb31-12" tabindex="-1"></a><span class="co"># show modules</span></span>
<span id="cb31-13"><a href="05-WGCNA.html#cb31-13" tabindex="-1"></a><span class="fu">table</span>(dynamicMods)</span></code></pre></div>
Excluding the '0', how many modules have been detected by the hierarchical clustering?
<div id="radio_DFVKMPLPRF" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_DFVKMPLPRF" value=""></input> <span>15</span></label><label><input type="radio" autocomplete="off" name="radio_DFVKMPLPRF" value="answer"></input> <span>24</span></label><label><input type="radio" autocomplete="off" name="radio_DFVKMPLPRF" value=""></input> <span>31</span></label>
</div>
<p>You should be able to see that the modules contain between 74 and 1697 genes. The number of modules depends on the dataset - because the differences between treatments in this experiment are fairly subtle, it is likely to have a relatively high number of modules, while an experiment with greater differences between treatments may have fewer but more distinct modules. We can assess module similarity by comparing the gene expression in each of the modules using module eignegenes (the first principal component of the expression matrix).</p>
<p>But first, it's time to get colourful. Rather than using numbers to label modules, WGCNA uses colours. This is not only for aesthetic reasons, but makes it easier to keep a track of how different modules relate to one another and to treatments or phenotypes.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb32-1"><a href="05-WGCNA.html#cb32-1" tabindex="-1"></a><span class="co"># Give the modules colour name labels</span></span>
<span id="cb32-2"><a href="05-WGCNA.html#cb32-2" tabindex="-1"></a>dynamicCols <span class="ot">&lt;-</span> <span class="fu">labels2colors</span>(dynamicMods)</span>
<span id="cb32-3"><a href="05-WGCNA.html#cb32-3" tabindex="-1"></a></span>
<span id="cb32-4"><a href="05-WGCNA.html#cb32-4" tabindex="-1"></a><span class="co"># show colourfully labelled modules</span></span>
<span id="cb32-5"><a href="05-WGCNA.html#cb32-5" tabindex="-1"></a><span class="fu">table</span>(dynamicCols)</span></code></pre></div>
<p>Note that the 0 module has been relabelled as "grey". This module contains all the genes that do not show sufficient similarity in gene expression to any other genes (and thus should not be considered in further analysis). Now that we have colourful module labels, we can create our module eigengenes and look at the similarity of expression between different modules.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb33-1"><a href="05-WGCNA.html#cb33-1" tabindex="-1"></a><span class="co"># Calculate eigengenes</span></span>
<span id="cb33-2"><a href="05-WGCNA.html#cb33-2" tabindex="-1"></a>MEList <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(ncount, <span class="at">colors =</span> dynamicCols)</span>
<span id="cb33-3"><a href="05-WGCNA.html#cb33-3" tabindex="-1"></a>MEs <span class="ot">&lt;-</span> MEList<span class="sc">$</span>eigengenes</span>
<span id="cb33-4"><a href="05-WGCNA.html#cb33-4" tabindex="-1"></a></span>
<span id="cb33-5"><a href="05-WGCNA.html#cb33-5" tabindex="-1"></a><span class="co"># Calculate dissimilarity of module eigengenes</span></span>
<span id="cb33-6"><a href="05-WGCNA.html#cb33-6" tabindex="-1"></a>MEDiss <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(MEs, <span class="at">use =</span> <span class="st">&quot;complete&quot;</span>)</span>
<span id="cb33-7"><a href="05-WGCNA.html#cb33-7" tabindex="-1"></a></span>
<span id="cb33-8"><a href="05-WGCNA.html#cb33-8" tabindex="-1"></a><span class="co"># Cluster modules by eigengene similarity</span></span>
<span id="cb33-9"><a href="05-WGCNA.html#cb33-9" tabindex="-1"></a>METree <span class="ot">&lt;-</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(MEDiss), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb33-10"><a href="05-WGCNA.html#cb33-10" tabindex="-1"></a></span>
<span id="cb33-11"><a href="05-WGCNA.html#cb33-11" tabindex="-1"></a><span class="co"># plot tree of modules</span></span>
<span id="cb33-12"><a href="05-WGCNA.html#cb33-12" tabindex="-1"></a><span class="fu">plot</span>(METree, <span class="at">main =</span> <span class="st">&quot;Clustering of module eigengenes&quot;</span>,</span>
<span id="cb33-13"><a href="05-WGCNA.html#cb33-13" tabindex="-1"></a>     <span class="at">sub=</span> <span class="st">&quot;&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb33-14"><a href="05-WGCNA.html#cb33-14" tabindex="-1"></a></span>
<span id="cb33-15"><a href="05-WGCNA.html#cb33-15" tabindex="-1"></a><span class="co"># Set a merge threshold</span></span>
<span id="cb33-16"><a href="05-WGCNA.html#cb33-16" tabindex="-1"></a>MEDissThres <span class="ot">=</span> <span class="fl">0.25</span></span>
<span id="cb33-17"><a href="05-WGCNA.html#cb33-17" tabindex="-1"></a></span>
<span id="cb33-18"><a href="05-WGCNA.html#cb33-18" tabindex="-1"></a><span class="co"># Plot the cut line into the dendrogram</span></span>
<span id="cb33-19"><a href="05-WGCNA.html#cb33-19" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span>MEDissThres)</span></code></pre></div>
<center>
<img src="figures/module_eigengenes.png" style="width:600px; border-radius:15px" />
</center>
<p>We can see that there are quite a few modules with fairly similar module eigengenes, such as the darkgrey and lightgreen, and the darkgreen and tan. We can merge these modules to reduce the overall number of modules in our network.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb34-1"><a href="05-WGCNA.html#cb34-1" tabindex="-1"></a><span class="co"># Call an automatic merging function</span></span>
<span id="cb34-2"><a href="05-WGCNA.html#cb34-2" tabindex="-1"></a>merge <span class="ot">&lt;-</span> <span class="fu">mergeCloseModules</span>(ncount, dynamicCols, <span class="at">cutHeight =</span> MEDissThres, <span class="at">verbose =</span> <span class="dv">3</span>)</span>
<span id="cb34-3"><a href="05-WGCNA.html#cb34-3" tabindex="-1"></a></span>
<span id="cb34-4"><a href="05-WGCNA.html#cb34-4" tabindex="-1"></a><span class="co"># new label for merged module colors</span></span>
<span id="cb34-5"><a href="05-WGCNA.html#cb34-5" tabindex="-1"></a>mergedCols <span class="ot">&lt;-</span> merge<span class="sc">$</span>colors</span>
<span id="cb34-6"><a href="05-WGCNA.html#cb34-6" tabindex="-1"></a></span>
<span id="cb34-7"><a href="05-WGCNA.html#cb34-7" tabindex="-1"></a><span class="co"># new label for eigengenes of newly merged modules:</span></span>
<span id="cb34-8"><a href="05-WGCNA.html#cb34-8" tabindex="-1"></a>mergedMEs <span class="ot">&lt;-</span> merge<span class="sc">$</span>newMEs</span>
<span id="cb34-9"><a href="05-WGCNA.html#cb34-9" tabindex="-1"></a></span>
<span id="cb34-10"><a href="05-WGCNA.html#cb34-10" tabindex="-1"></a><span class="co"># show colourfully labelled merged modules</span></span>
<span id="cb34-11"><a href="05-WGCNA.html#cb34-11" tabindex="-1"></a><span class="fu">table</span>(mergedCols)</span></code></pre></div>
<p>Finally we can look at our gene tree based on TOM dissimilarity together with the original modules of co-expression and the merged modules of co-expression.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb35-1"><a href="05-WGCNA.html#cb35-1" tabindex="-1"></a><span class="co"># Plot a dendrogram of gene TOM with modules before and after merging</span></span>
<span id="cb35-2"><a href="05-WGCNA.html#cb35-2" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(geneTree, <span class="fu">cbind</span>(dynamicCols, mergedCols),</span>
<span id="cb35-3"><a href="05-WGCNA.html#cb35-3" tabindex="-1"></a>                    <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>, <span class="at">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb35-4"><a href="05-WGCNA.html#cb35-4" tabindex="-1"></a>                    <span class="at">addGuide =</span> <span class="cn">TRUE</span>, <span class="at">guideHang =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<center>
<img src="figures/gene_module_dendrogram.png" style="width:600px; border-radius:15px" />
</center>
<p>Compared to some data sets, that you may see, this does not cluster that clearly into discrete modules of gene co-expression, with many of the modules rather mixed together. Nonetheless, we can still see some well-defined blocks of colour at the bottom which highlight clear modules, such as the salmon coloured module on the left, the yellow and greenyellow coloured modules in the middle, and the blue coloured module on the right. In the next step we will see how they relate to our treatment groups.</p>
</div>
<div id="module-treatment-correlation" class="section level2 hasAnchor" number="5.6">
<h2><span class="header-section-number">5.6</span> Module-treatment correlation<a href="05-WGCNA.html#module-treatment-correlation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/heatmap_3.png" style="width:200px" />
</center>
<p>Now that we have our co-expression modules, we want to see how these relate to our treatments. To do this we are going to take the module eigengenes - the first principal component of an expression matrix for that module - and correlate these with our binary treatment variables for each of the groups. The first step requires us to calculate the correlations and their P-values and put them in a matrix.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb36-1"><a href="05-WGCNA.html#cb36-1" tabindex="-1"></a><span class="co"># Calculate correlations between module eigengenes and treatments</span></span>
<span id="cb36-2"><a href="05-WGCNA.html#cb36-2" tabindex="-1"></a>moduleTraitCor <span class="ot">&lt;-</span> <span class="fu">cor</span>(mergedMEs, meta, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>)</span>
<span id="cb36-3"><a href="05-WGCNA.html#cb36-3" tabindex="-1"></a></span>
<span id="cb36-4"><a href="05-WGCNA.html#cb36-4" tabindex="-1"></a><span class="co"># How many genes are in our dataset?</span></span>
<span id="cb36-5"><a href="05-WGCNA.html#cb36-5" tabindex="-1"></a>nSamples <span class="ot">&lt;-</span> <span class="fu">nrow</span>(ncount)</span>
<span id="cb36-6"><a href="05-WGCNA.html#cb36-6" tabindex="-1"></a></span>
<span id="cb36-7"><a href="05-WGCNA.html#cb36-7" tabindex="-1"></a><span class="co"># Caluclate the P values for these correlations</span></span>
<span id="cb36-8"><a href="05-WGCNA.html#cb36-8" tabindex="-1"></a>moduleTraitPvalue <span class="ot">=</span> <span class="fu">corPvalueStudent</span>(moduleTraitCor, nSamples)</span>
<span id="cb36-9"><a href="05-WGCNA.html#cb36-9" tabindex="-1"></a></span>
<span id="cb36-10"><a href="05-WGCNA.html#cb36-10" tabindex="-1"></a><span class="co"># combine correlations and P values into a matrix</span></span>
<span id="cb36-11"><a href="05-WGCNA.html#cb36-11" tabindex="-1"></a>textMatrix <span class="ot">&lt;-</span>  <span class="fu">paste</span>(<span class="fu">signif</span>(moduleTraitCor, <span class="dv">2</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">(&quot;</span>,</span>
<span id="cb36-12"><a href="05-WGCNA.html#cb36-12" tabindex="-1"></a>                    <span class="fu">signif</span>(moduleTraitPvalue, <span class="dv">1</span>), <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb36-13"><a href="05-WGCNA.html#cb36-13" tabindex="-1"></a></span>
<span id="cb36-14"><a href="05-WGCNA.html#cb36-14" tabindex="-1"></a><span class="co"># update dimensions of testMatrix</span></span>
<span id="cb36-15"><a href="05-WGCNA.html#cb36-15" tabindex="-1"></a><span class="fu">dim</span>(textMatrix) <span class="ot">&lt;-</span> <span class="fu">dim</span>(moduleTraitCor)</span></code></pre></div>
<p>Now that we have the matrix we can plot our heatmap. We are going to alter some graphical parameters first so that the plot fits nicely into the window.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb37-1"><a href="05-WGCNA.html#cb37-1" tabindex="-1"></a><span class="co"># change some graphical parameters</span></span>
<span id="cb37-2"><a href="05-WGCNA.html#cb37-2" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">9</span>, <span class="dv">4</span>, <span class="dv">2</span>))</span>
<span id="cb37-3"><a href="05-WGCNA.html#cb37-3" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb37-4"><a href="05-WGCNA.html#cb37-4" tabindex="-1"></a><span class="fu">options</span>(<span class="at">repr.plot.width=</span><span class="dv">7</span>, <span class="at">repr.plot.height =</span> <span class="dv">9</span>)</span>
<span id="cb37-5"><a href="05-WGCNA.html#cb37-5" tabindex="-1"></a></span>
<span id="cb37-6"><a href="05-WGCNA.html#cb37-6" tabindex="-1"></a><span class="co"># plot the heatmap</span></span>
<span id="cb37-7"><a href="05-WGCNA.html#cb37-7" tabindex="-1"></a><span class="fu">labeledHeatmap</span>(<span class="at">Matrix =</span> moduleTraitCor,</span>
<span id="cb37-8"><a href="05-WGCNA.html#cb37-8" tabindex="-1"></a>               <span class="at">xLabels =</span> <span class="fu">colnames</span>(meta),</span>
<span id="cb37-9"><a href="05-WGCNA.html#cb37-9" tabindex="-1"></a>               <span class="at">yLabels =</span> <span class="fu">names</span>(mergedMEs),</span>
<span id="cb37-10"><a href="05-WGCNA.html#cb37-10" tabindex="-1"></a>               <span class="at">ySymbols =</span> <span class="fu">names</span>(mergedMEs),</span>
<span id="cb37-11"><a href="05-WGCNA.html#cb37-11" tabindex="-1"></a>               <span class="at">colors =</span> <span class="fu">blueWhiteRed</span>(<span class="dv">50</span>),</span>
<span id="cb37-12"><a href="05-WGCNA.html#cb37-12" tabindex="-1"></a>               <span class="at">textMatrix =</span> textMatrix,</span>
<span id="cb37-13"><a href="05-WGCNA.html#cb37-13" tabindex="-1"></a>               <span class="at">setStdMargins =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<center>
<img src="figures/Eigengene_pheno_heatmap.png" style="width:600px; border-radius:15px" />
</center>
<p>The heatmap shows that some of the module eigengenes have quite high correlations with our treatment groups. The best correlation is the blue module, which strongly positively correlates with the baseline male group (r = 0.75, p = 0.00002). However, we would probably be more interested in knowing the modules that correlate with the experimental treatment groups.</p>
What module eigengene shows the strongest correlation in any of the M (multiple matings) or E (Evolved monogamy) treatment groups?
<div id="radio_HZTEIMHVMO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_HZTEIMHVMO" value=""></input> <span>salmon</span></label><label><input type="radio" autocomplete="off" name="radio_HZTEIMHVMO" value="answer"></input> <span>greenyellow</span></label><label><input type="radio" autocomplete="off" name="radio_HZTEIMHVMO" value=""></input> <span>lightcyan</span></label>
</div>
<p>One module that shows an interesting pattern of co-expression is the lightyellow module, which shows a negative correlation with the evolved monogamy male group (r = -0.51, p = 0.01), but also a positive correlation with the the multiple mated male group (r = 0.41, p = 0.04). We are going to use this module as an example for the final stage of this network analysis - plotting a network graph.</p>
</div>
<div id="network-plots-and-hub-genes" class="section level2 hasAnchor" number="5.7">
<h2><span class="header-section-number">5.7</span> Network plots and hub genes<a href="05-WGCNA.html#network-plots-and-hub-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/spiderweb.png" style="width:200px" />
</center>
<p>WGCNA uses the concept of module membership (MM) to identify hub genes. Module membership is effectively the correlation between a given gene's gene expression profile and the module eigengene (a value between -1 and 1). High MM scores are close to -1 or 1, while low MM scores are close to 0. For each gene, the MM (and it's p value) are calculated for all the modules, but it will have the highest MM for the module to which it was assigned during the clustering analysis.</p>
<p>Higher module membership is associated with intramodular connectivity, and thus can be used as a metric for identifying hub genes within a module. We are going to create a data frame of all MM scores and their p values for all genes and all modules, which we can query in the following steps.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb38-1"><a href="05-WGCNA.html#cb38-1" tabindex="-1"></a><span class="co"># names (colors) of the modules</span></span>
<span id="cb38-2"><a href="05-WGCNA.html#cb38-2" tabindex="-1"></a>modNames <span class="ot">&lt;-</span> <span class="fu">substring</span>(<span class="fu">names</span>(MEs), <span class="dv">3</span>)</span>
<span id="cb38-3"><a href="05-WGCNA.html#cb38-3" tabindex="-1"></a></span>
<span id="cb38-4"><a href="05-WGCNA.html#cb38-4" tabindex="-1"></a><span class="co"># Caluclate gene module membership and p value</span></span>
<span id="cb38-5"><a href="05-WGCNA.html#cb38-5" tabindex="-1"></a>geneModMem <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(ncount, MEs, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb38-6"><a href="05-WGCNA.html#cb38-6" tabindex="-1"></a>MMPvalue <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneModMem), nSamples))</span>
<span id="cb38-7"><a href="05-WGCNA.html#cb38-7" tabindex="-1"></a></span>
<span id="cb38-8"><a href="05-WGCNA.html#cb38-8" tabindex="-1"></a><span class="co"># Rename columns of these data frames</span></span>
<span id="cb38-9"><a href="05-WGCNA.html#cb38-9" tabindex="-1"></a><span class="fu">names</span>(geneModMem) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;MM&quot;</span>, modNames, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb38-10"><a href="05-WGCNA.html#cb38-10" tabindex="-1"></a><span class="fu">names</span>(MMPvalue) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;p.MM&quot;</span>, modNames, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb38-11"><a href="05-WGCNA.html#cb38-11" tabindex="-1"></a></span>
<span id="cb38-12"><a href="05-WGCNA.html#cb38-12" tabindex="-1"></a><span class="co"># merge data frames</span></span>
<span id="cb38-13"><a href="05-WGCNA.html#cb38-13" tabindex="-1"></a>MMdata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">merge</span>(<span class="at">x =</span> geneModMem, <span class="at">y =</span> MMPvalue, <span class="at">by =</span> <span class="st">&quot;row.names&quot;</span>), <span class="at">row.names =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>Now we are going to specify the module that we are interested in, select the top n genes for that module according to their module membership scores, and then extract the TOM dissimilarity information for these genes, which can be used as an input for our network plot.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb39-1"><a href="05-WGCNA.html#cb39-1" tabindex="-1"></a><span class="co"># Select a module of interest</span></span>
<span id="cb39-2"><a href="05-WGCNA.html#cb39-2" tabindex="-1"></a>modules <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;lightyellow&quot;</span>)</span>
<span id="cb39-3"><a href="05-WGCNA.html#cb39-3" tabindex="-1"></a></span>
<span id="cb39-4"><a href="05-WGCNA.html#cb39-4" tabindex="-1"></a><span class="co"># Obtain the genes that are in that module</span></span>
<span id="cb39-5"><a href="05-WGCNA.html#cb39-5" tabindex="-1"></a>inModuleCol <span class="ot">&lt;-</span> <span class="fu">is.finite</span>(<span class="fu">match</span>(mergedCols, modules))</span>
<span id="cb39-6"><a href="05-WGCNA.html#cb39-6" tabindex="-1"></a>modGenes <span class="ot">&lt;-</span> <span class="fu">colnames</span>(ncount)[inModuleCol]</span>
<span id="cb39-7"><a href="05-WGCNA.html#cb39-7" tabindex="-1"></a></span>
<span id="cb39-8"><a href="05-WGCNA.html#cb39-8" tabindex="-1"></a><span class="co"># how many genes are in the module?</span></span>
<span id="cb39-9"><a href="05-WGCNA.html#cb39-9" tabindex="-1"></a><span class="fu">length</span>(modGenes)</span>
<span id="cb39-10"><a href="05-WGCNA.html#cb39-10" tabindex="-1"></a></span>
<span id="cb39-11"><a href="05-WGCNA.html#cb39-11" tabindex="-1"></a><span class="co"># select module membership values for the module of interest</span></span>
<span id="cb39-12"><a href="05-WGCNA.html#cb39-12" tabindex="-1"></a>modMMdata <span class="ot">&lt;-</span> MMdata[<span class="fu">rownames</span>(MMdata) <span class="sc">%in%</span> modGenes,</span>
<span id="cb39-13"><a href="05-WGCNA.html#cb39-13" tabindex="-1"></a>                    <span class="fu">c</span>(<span class="fu">paste</span>(<span class="st">&quot;MM&quot;</span>, modules, <span class="at">sep=</span><span class="st">&quot;&quot;</span>),<span class="fu">paste</span>(<span class="st">&quot;p.MM&quot;</span>, modules, <span class="at">sep=</span><span class="st">&quot;&quot;</span>))]</span>
<span id="cb39-14"><a href="05-WGCNA.html#cb39-14" tabindex="-1"></a></span>
<span id="cb39-15"><a href="05-WGCNA.html#cb39-15" tabindex="-1"></a><span class="co"># How many genes do we want to include? 20 may be a a good number</span></span>
<span id="cb39-16"><a href="05-WGCNA.html#cb39-16" tabindex="-1"></a>nnode <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb39-17"><a href="05-WGCNA.html#cb39-17" tabindex="-1"></a></span>
<span id="cb39-18"><a href="05-WGCNA.html#cb39-18" tabindex="-1"></a><span class="co"># For our chosen module, select the top genes ordered by module membership. </span></span>
<span id="cb39-19"><a href="05-WGCNA.html#cb39-19" tabindex="-1"></a><span class="co"># Notice that we use &#39;abs&#39; to choose genes with high or low values </span></span>
<span id="cb39-20"><a href="05-WGCNA.html#cb39-20" tabindex="-1"></a><span class="co">#(module membership is scored between -1 and 1).</span></span>
<span id="cb39-21"><a href="05-WGCNA.html#cb39-21" tabindex="-1"></a>topnode <span class="ot">&lt;-</span> modMMdata[<span class="fu">order</span>(</span>
<span id="cb39-22"><a href="05-WGCNA.html#cb39-22" tabindex="-1"></a>  <span class="sc">-</span><span class="fu">abs</span>(modMMdata[,<span class="fu">paste</span>(<span class="st">&quot;MM&quot;</span>, modules, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)])</span>
<span id="cb39-23"><a href="05-WGCNA.html#cb39-23" tabindex="-1"></a>  ),][<span class="dv">1</span><span class="sc">:</span>nnode,]</span>
<span id="cb39-24"><a href="05-WGCNA.html#cb39-24" tabindex="-1"></a></span>
<span id="cb39-25"><a href="05-WGCNA.html#cb39-25" tabindex="-1"></a><span class="co"># Select the TOM values for these genes</span></span>
<span id="cb39-26"><a href="05-WGCNA.html#cb39-26" tabindex="-1"></a>topnodeGenes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(topnode)</span>
<span id="cb39-27"><a href="05-WGCNA.html#cb39-27" tabindex="-1"></a>inModuleGene <span class="ot">&lt;-</span> <span class="fu">colnames</span>(ncount) <span class="sc">%in%</span> topnodeGenes</span>
<span id="cb39-28"><a href="05-WGCNA.html#cb39-28" tabindex="-1"></a>modTOM <span class="ot">&lt;-</span> TOM[inModuleGene, inModuleGene]</span>
<span id="cb39-29"><a href="05-WGCNA.html#cb39-29" tabindex="-1"></a><span class="fu">dimnames</span>(modTOM) <span class="ot">&lt;-</span> <span class="fu">list</span>(topnodeGenes, topnodeGenes)</span></code></pre></div>
How many genes does the lightyellow module contain?
<div id="radio_KQJFHFVQJA" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_KQJFHFVQJA" value="answer"></input> <span>681</span></label><label><input type="radio" autocomplete="off" name="radio_KQJFHFVQJA" value=""></input> <span>912</span></label><label><input type="radio" autocomplete="off" name="radio_KQJFHFVQJA" value=""></input> <span>1337</span></label>
</div>
<p>Now we are going to visualise the connections between the nodes using the igraph package in R. In igraph nodes are referred to as vertices, and connections are referred to as edges. These terms are commonly used in network analysis and are synonymous. Although we have limited the network to only 20 genes, it is likely to retain a high number of edges (connections). To make it a little easier to interpret we can exclude some of the connections with low weights, as well as excluding nodes that are floating (those which have no connections after filtering by weight).</p>
<p>Deciding the threshold for filtering is generally a case of trial and error, and will depend on how many genes you have selected to display and the topology of the module (e.g. does it have many hub genes or relatively few). In our case we are going to use a filter set to 0.1.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb40-1"><a href="05-WGCNA.html#cb40-1" tabindex="-1"></a><span class="co"># build the graph object</span></span>
<span id="cb40-2"><a href="05-WGCNA.html#cb40-2" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">graph_from_adjacency_matrix</span>(modTOM, </span>
<span id="cb40-3"><a href="05-WGCNA.html#cb40-3" tabindex="-1"></a>                                   <span class="at">mode =</span> <span class="st">&quot;upper&quot;</span>, </span>
<span id="cb40-4"><a href="05-WGCNA.html#cb40-4" tabindex="-1"></a>                                   <span class="at">weighted =</span> <span class="cn">TRUE</span>, </span>
<span id="cb40-5"><a href="05-WGCNA.html#cb40-5" tabindex="-1"></a>                                   <span class="at">diag =</span> <span class="cn">FALSE</span>)</span>
<span id="cb40-6"><a href="05-WGCNA.html#cb40-6" tabindex="-1"></a></span>
<span id="cb40-7"><a href="05-WGCNA.html#cb40-7" tabindex="-1"></a><span class="co"># delete edges with a low weight ( &lt; 0.1)</span></span>
<span id="cb40-8"><a href="05-WGCNA.html#cb40-8" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">delete_edges</span>(net, <span class="fu">which</span>(<span class="fu">E</span>(net)<span class="sc">$</span>weight <span class="sc">&lt;</span> <span class="fl">0.1</span>))</span>
<span id="cb40-9"><a href="05-WGCNA.html#cb40-9" tabindex="-1"></a></span>
<span id="cb40-10"><a href="05-WGCNA.html#cb40-10" tabindex="-1"></a><span class="co"># delete floating nodes</span></span>
<span id="cb40-11"><a href="05-WGCNA.html#cb40-11" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">delete_vertices</span>(net, <span class="fu">which</span>(<span class="fu">degree</span>(net)<span class="sc">==</span><span class="dv">0</span>))</span>
<span id="cb40-12"><a href="05-WGCNA.html#cb40-12" tabindex="-1"></a></span>
<span id="cb40-13"><a href="05-WGCNA.html#cb40-13" tabindex="-1"></a><span class="co"># adjust graphics parameters</span></span>
<span id="cb40-14"><a href="05-WGCNA.html#cb40-14" tabindex="-1"></a><span class="fu">options</span>(<span class="at">repr.plot.width=</span><span class="dv">6</span>, <span class="at">repr.plot.height =</span> <span class="dv">6</span>)</span>
<span id="cb40-15"><a href="05-WGCNA.html#cb40-15" tabindex="-1"></a></span>
<span id="cb40-16"><a href="05-WGCNA.html#cb40-16" tabindex="-1"></a><span class="co"># plot the module network</span></span>
<span id="cb40-17"><a href="05-WGCNA.html#cb40-17" tabindex="-1"></a><span class="fu">plot</span>(net,</span>
<span id="cb40-18"><a href="05-WGCNA.html#cb40-18" tabindex="-1"></a>     <span class="at">vertex.size =</span> <span class="dv">8</span>, </span>
<span id="cb40-19"><a href="05-WGCNA.html#cb40-19" tabindex="-1"></a>     <span class="at">vertex.label.cex =</span> <span class="fl">0.8</span>,</span>
<span id="cb40-20"><a href="05-WGCNA.html#cb40-20" tabindex="-1"></a>     <span class="at">edge.color =</span> <span class="st">&quot;grey85&quot;</span>)</span></code></pre></div>
<center>
<img src="figures/LY_network_v4.png" style="width:600px; border-radius:15px" />
</center>
<p>This gives us a network plot of 20 fairly well connected genes. If some of the gene names overlap, try rerunning the the code. You can also view a larger version of the image by right clicking on the figure and selection "Open image in new tab". You can also dissociate the network by increasing the number for <code>delete_edges</code> (for example from 0.1 to 0.12).</p>
<p>Now we have our network plot, the final steps of the analysis are to inspect the module membership information and identify which are the best candidates for hub genes, and to load some functional annotation information about these genes.</p>
<p>The functional information here was obtained by submitting a list of gene symbols for <em>Drosophila pseudoobscura</em> to the <a href="https://www.uniprot.org/id-mapping">Uniprot Knowledge Base</a>.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb41-1"><a href="05-WGCNA.html#cb41-1" tabindex="-1"></a><span class="co"># read in uniprot annotations</span></span>
<span id="cb41-2"><a href="05-WGCNA.html#cb41-2" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;updated_annot.tsv&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">na.strings =</span> <span class="st">&quot;NA&quot;</span>, <span class="at">quote =</span> <span class="st">&quot;&quot;</span>, <span class="at">header =</span> T)</span>
<span id="cb41-3"><a href="05-WGCNA.html#cb41-3" tabindex="-1"></a></span>
<span id="cb41-4"><a href="05-WGCNA.html#cb41-4" tabindex="-1"></a><span class="co"># create a data frame with annotation and module membership information</span></span>
<span id="cb41-5"><a href="05-WGCNA.html#cb41-5" tabindex="-1"></a>hubs<span class="ot">&lt;-</span><span class="fu">merge</span>(topnode, annot, <span class="at">by.x =</span> <span class="dv">0</span>, <span class="at">by.y =</span> <span class="st">&quot;FBgn&quot;</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-6"><a href="05-WGCNA.html#cb41-6" tabindex="-1"></a></span>
<span id="cb41-7"><a href="05-WGCNA.html#cb41-7" tabindex="-1"></a><span class="co"># display the top 10 genes by module membership</span></span>
<span id="cb41-8"><a href="05-WGCNA.html#cb41-8" tabindex="-1"></a><span class="fu">head</span>(hubs[<span class="fu">order</span>(<span class="sc">-</span>hubs[,<span class="fu">paste</span>(<span class="st">&quot;MM&quot;</span>, modules, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)]),], <span class="at">n =</span> <span class="dv">10</span>)</span></code></pre></div>
What is the protein name of the gene with the highest module membership score?
<div id="radio_VKIYBANGVL" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_VKIYBANGVL" value=""></input> <span>NPC intracellular cholesterol transporter 1</span></label><label><input type="radio" autocomplete="off" name="radio_VKIYBANGVL" value="answer"></input> <span>Nicalin</span></label><label><input type="radio" autocomplete="off" name="radio_VKIYBANGVL" value=""></input> <span>Adapter molecule Crk</span></label>
</div>
<p>You will notice that the gene with the highest module membership score does not appear to sit at the centre of our network. Because we have limited ourselves to plotting only the top 20 genes, we are excluding a lot of information about genes in this module with more moderate MM scores. It is likely that the gene with the highest MM score is somewhat connected to a larger number of genes overall within the module, even if some of the other genes in the top 20 seem to be better connected among themselves. However, all these genes have very high scores, and could represent interesting candidates for further investigation.</p>
<p>We can also plot the functional information directly onto the network graph with a small modification to our code.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb42-1"><a href="05-WGCNA.html#cb42-1" tabindex="-1"></a><span class="co"># Take the protein names that match the gene names in this module</span></span>
<span id="cb42-2"><a href="05-WGCNA.html#cb42-2" tabindex="-1"></a>modProts <span class="ot">&lt;-</span> annot<span class="sc">$</span>Protein_names[<span class="fu">match</span>(modGenes, annot<span class="sc">$</span>FBgn)]</span>
<span id="cb42-3"><a href="05-WGCNA.html#cb42-3" tabindex="-1"></a></span>
<span id="cb42-4"><a href="05-WGCNA.html#cb42-4" tabindex="-1"></a><span class="co"># Update the modTOM object</span></span>
<span id="cb42-5"><a href="05-WGCNA.html#cb42-5" tabindex="-1"></a><span class="fu">dimnames</span>(modTOM) <span class="ot">&lt;-</span> <span class="fu">list</span>(modProts, modProts)</span></code></pre></div>
<p>Copy the 5 lines of code from <code>#build the graph object</code> up to and including <code>#plot the module network</code> and paste it into a new cell. When you run this code you'll see that it now displays the proteins names directly on the network graph. Unfortunately the names of our proteins are rather long, which makes the figure a little messy. However, if your annotations are shorter, then this can be a convenient way of displaying the functions of the hub genes in your network figures. The full list of genes within the module could also be exported for <a href="https://neof-workshops.github.io/RNAseq_mzar4p/Course/16-GO.html">Gene Ontology enrichment analysis</a>.</p>
<p>When it comes to plotting network figures, igraph provides a nice first approach to visualise the results. Within R, the <a href="https://briatte.github.io/ggnet/">ggnet2</a> package provides options to fine tune their appearance. Output files from WGCNA can also be exported with the <code>exportNetworkToCytoscape()</code> function and visualised with <a href="https://cytoscape.org/">Cytoscape</a> software, an open-source desktop app that provides a greater level of control in plotting networks.</p>
<p>Once you have done this you can close and halt your workbook.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="04-Multiple_samples_loops.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
