<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Psychrobacter arcticus genome QC | NERC ATSC Genomics</title>
  <meta name="description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  <meta name="generator" content="bookdown 0.31 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Psychrobacter arcticus genome QC | NERC ATSC Genomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Psychrobacter arcticus genome QC | NERC ATSC Genomics" />
  
  <meta name="twitter:description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-02-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="02-NEOF.html"/>
<link rel="next" href="04-b_cerues_qc.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>NERC ATSC MAR 2023</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="01-Intro.html"><a href="01-Intro.html#example-mcqs"><i class="fa fa-check"></i><b>1.1</b> Example MCQs</a></li>
</ul></li>
<li class="part"><span><b>NEOF</b></span></li>
<li class="chapter" data-level="2" data-path="02-NEOF.html"><a href="02-NEOF.html"><i class="fa fa-check"></i><b>2</b> NEOF</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-NEOF.html"><a href="02-NEOF.html#about-neof"><i class="fa fa-check"></i><b>2.1</b> About NEOF</a></li>
<li class="chapter" data-level="2.2" data-path="02-NEOF.html"><a href="02-NEOF.html#capabilities"><i class="fa fa-check"></i><b>2.2</b> Capabilities</a></li>
<li class="chapter" data-level="2.3" data-path="02-NEOF.html"><a href="02-NEOF.html#visitor-facility-and-training"><i class="fa fa-check"></i><b>2.3</b> Visitor facility and training</a></li>
<li class="chapter" data-level="2.4" data-path="02-NEOF.html"><a href="02-NEOF.html#mailing-list-contact"><i class="fa fa-check"></i><b>2.4</b> Mailing list &amp; contact</a></li>
</ul></li>
<li class="part"><span><b>Illumina</b></span></li>
<li class="chapter" data-level="3" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html"><i class="fa fa-check"></i><b>3</b> Psychrobacter arcticus genome QC</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-fastqc"><i class="fa fa-check"></i><b>3.1</b> P. arcticus: fastQC</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-fastqc-mcqs"><i class="fa fa-check"></i><b>3.1.1</b> P. arcticus: fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-summary"><i class="fa fa-check"></i><b>3.2</b> P. arcticus: summary</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html"><i class="fa fa-check"></i><b>4</b> Bacillus cereus genome QC</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-raw-fastqc"><i class="fa fa-check"></i><b>4.1</b> B. cereus: raw fastQC</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-raw-fastqc-mcqs"><i class="fa fa-check"></i><b>4.1.1</b> B. cereus: raw fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-quality-control"><i class="fa fa-check"></i><b>4.2</b> B. cereus: Quality control</a></li>
<li class="chapter" data-level="4.3" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-trimmed-fastqc"><i class="fa fa-check"></i><b>4.3</b> B. cereus: trimmed fastQC</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-trimmed-fastqc-mcqs"><i class="fa fa-check"></i><b>4.3.1</b> B. cereus: trimmed fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-summary"><i class="fa fa-check"></i><b>4.4</b> B. cereus: summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html"><i class="fa fa-check"></i><b>5</b> Whole shotgun metagenomics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-raw-fastqc"><i class="fa fa-check"></i><b>5.1</b> Metagenome: raw fastqc</a></li>
<li class="chapter" data-level="5.2" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-trimmed-fastqc"><i class="fa fa-check"></i><b>5.2</b> Metagenome: trimmed fastqc</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-trimmed-fastqc-mcqs"><i class="fa fa-check"></i><b>5.2.1</b> Metagenome: trimmed fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-summary"><i class="fa fa-check"></i><b>5.3</b> Metagenome: summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-16S_qc.html"><a href="06-16S_qc.html"><i class="fa fa-check"></i><b>6</b> 16S rRNA Metagenetics QC</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-16S_qc.html"><a href="06-16S_qc.html#s-fastqc"><i class="fa fa-check"></i><b>6.1</b> 16S: fastQC</a></li>
<li class="chapter" data-level="6.2" data-path="06-16S_qc.html"><a href="06-16S_qc.html#s-summary"><i class="fa fa-check"></i><b>6.2</b> 16S: summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-multiQC.html"><a href="07-multiQC.html"><i class="fa fa-check"></i><b>7</b> MultiQC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-multiQC.html"><a href="07-multiQC.html#multiqc-of-the-raw-data"><i class="fa fa-check"></i><b>7.1</b> MultiQC of the raw data</a></li>
<li class="chapter" data-level="7.2" data-path="07-multiQC.html"><a href="07-multiQC.html#fastqc-of-the-trimmed-data"><i class="fa fa-check"></i><b>7.2</b> Fastqc of the trimmed data</a></li>
<li class="chapter" data-level="7.3" data-path="07-multiQC.html"><a href="07-multiQC.html#read-data"><i class="fa fa-check"></i><b>7.3</b> Read data</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NERC ATSC Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="psychrobacter-arcticus-genome-qc" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Psychrobacter arcticus genome QC<a href="03-p_arcticus_qc.html#psychrobacter-arcticus-genome-qc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/iceberg.png" width="20%" style="display: block; margin: auto;" /></p>
<p>In this section you will view and interpret the output from fastQC from different datasets.</p>
<p>The first dataset will comprise of good quality reads for the genome of <em>Psychrobacter arcticus</em>.</p>
<p><em>P. arcticus</em> is a bacteria that was first isolated from Siberian permafrost.
I am fond of <em>Psychrobacter</em> as my BSc project involved studying it.</p>
<div id="p.-arcticus-fastqc" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> P. arcticus: fastQC<a href="03-p_arcticus_qc.html#p.-arcticus-fastqc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/magnifying_glass_blue.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Below are the links to the <code>fastQC</code> reports for the forward (R1) and reverse (R2) reads of one <em>P. arcticus</em> genome sequencing sample.</p>
<ul>
<li><a href="https://neof-workshops.github.io/NERC_ATSC_MAR_2023_ztkban/files/Illumina_QC/p_arcticus/raw_fastqc_output/p_arcticus_R1_fastqc.html"><strong>p_arcticus_R1_fastqc.html</strong></a></li>
<li><a href="https://neof-workshops.github.io/NERC_ATSC_MAR_2023_ztkban/files/Illumina_QC/p_arcticus/raw_fastqc_output/p_arcticus_R2_fastqc.html"><strong>p_arcticus_R2_fastqc.html</strong></a></li>
</ul>
<div id="p.-arcticus-fastqc-mcqs" class="section level3 hasAnchor" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> P. arcticus: fastQC MCQs<a href="03-p_arcticus_qc.html#p.-arcticus-fastqc-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/question_bubble_blue.png" width="20%" style="display: block; margin: auto;" /></p>
<p>With the <code>fastQC</code> report, attempt the following MCQs, you can reattempt as many times as needed.
Please feel free to collaborate/help each other and ask the instructor questions.</p>
<ol style="list-style-type: decimal">
<li>How many total reads/sequences are there in the R1 file?
<div id="radio_PXSRABPXWB" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_PXSRABPXWB" value=""></input> <span><strong>42</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PXSRABPXWB" value="answer"></input> <span><strong>175,697</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PXSRABPXWB" value=""></input> <span><strong>150</strong></span></label>
</div></li>
<li>What is the length of the reads?
<div id="radio_GZKCAUGYDU" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_GZKCAUGYDU" value=""></input> <span><strong>42</strong></span></label><label><input type="radio" autocomplete="off" name="radio_GZKCAUGYDU" value=""></input> <span><strong>175,697</strong></span></label><label><input type="radio" autocomplete="off" name="radio_GZKCAUGYDU" value="answer"></input> <span><strong>150</strong></span></label>
</div></li>
<li>What is the average GC (%) content of the reads?
<div id="radio_BVAXZUDFTO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BVAXZUDFTO" value="answer"></input> <span><strong>42</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BVAXZUDFTO" value=""></input> <span><strong>175,697</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BVAXZUDFTO" value=""></input> <span><strong>150</strong></span></label>
</div></li>
<li>In R1 at what quality range is base position 1?
<div id="radio_AFNJZLCAVB" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_AFNJZLCAVB" value=""></input> <span><strong>0-40</strong></span></label><label><input type="radio" autocomplete="off" name="radio_AFNJZLCAVB" value="answer"></input> <span><strong>20-40</strong></span></label><label><input type="radio" autocomplete="off" name="radio_AFNJZLCAVB" value=""></input> <span><strong>28-40</strong></span></label>
</div></li>
<li>In R2 at what quality range are base positions 15-19?
<div id="radio_LKKBTFKDOD" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_LKKBTFKDOD" value=""></input> <span><strong>0-40</strong></span></label><label><input type="radio" autocomplete="off" name="radio_LKKBTFKDOD" value=""></input> <span><strong>20-40</strong></span></label><label><input type="radio" autocomplete="off" name="radio_LKKBTFKDOD" value="answer"></input> <span><strong>28-40</strong></span></label>
</div></li>
<li>In R1 at what quality range are base positions 45-49?
<div id="radio_BVNXZBXTZD" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BVNXZBXTZD" value=""></input> <span><strong>0-40</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BVNXZBXTZD" value=""></input> <span><strong>20-40</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BVNXZBXTZD" value="answer"></input> <span><strong>28-40</strong></span></label>
</div></li>
<li>In R2 at what quality range is base position 150?
<div id="radio_HBJYEZMVSS" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_HBJYEZMVSS" value=""></input> <span><strong>0-40</strong></span></label><label><input type="radio" autocomplete="off" name="radio_HBJYEZMVSS" value="answer"></input> <span><strong>20-40</strong></span></label><label><input type="radio" autocomplete="off" name="radio_HBJYEZMVSS" value=""></input> <span><strong>28-40</strong></span></label>
</div></li>
</ol>
</div>
</div>
<div id="p.-arcticus-summary" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> P. arcticus: summary<a href="03-p_arcticus_qc.html#p.-arcticus-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_blue.png" width="20%" style="display: block; margin: auto;" /></p>
<p>In this case the reads have been simulated from a genome assembly.
This is the reason it looks so nice.
Real data from a sequencing machine will unfortunately not look as good and will require quality trimming and filtering.</p>
<p>Generally even if data does look very nice we would carry out quality control to get rid of any poor data that is masked by the very good data.
However, for the purposes of time we will not look at this for this dataset.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="02-NEOF.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="04-b_cerues_qc.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
