<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Bacillus cereus genome QC | NERC ATSC Genomics</title>
  <meta name="description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  <meta name="generator" content="bookdown 0.31 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Bacillus cereus genome QC | NERC ATSC Genomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Bacillus cereus genome QC | NERC ATSC Genomics" />
  
  <meta name="twitter:description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-02-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="03-p_arcticus_qc.html"/>
<link rel="next" href="05-shotgun_metagenomic_qc.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>NERC ATSC MAR 2023</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="01-Intro.html"><a href="01-Intro.html#example-mcqs"><i class="fa fa-check"></i><b>1.1</b> Example MCQs</a></li>
</ul></li>
<li class="part"><span><b>NEOF</b></span></li>
<li class="chapter" data-level="2" data-path="02-NEOF.html"><a href="02-NEOF.html"><i class="fa fa-check"></i><b>2</b> NEOF</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-NEOF.html"><a href="02-NEOF.html#about-neof"><i class="fa fa-check"></i><b>2.1</b> About NEOF</a></li>
<li class="chapter" data-level="2.2" data-path="02-NEOF.html"><a href="02-NEOF.html#capabilities"><i class="fa fa-check"></i><b>2.2</b> Capabilities</a></li>
<li class="chapter" data-level="2.3" data-path="02-NEOF.html"><a href="02-NEOF.html#visitor-facility-and-training"><i class="fa fa-check"></i><b>2.3</b> Visitor facility and training</a></li>
<li class="chapter" data-level="2.4" data-path="02-NEOF.html"><a href="02-NEOF.html#mailing-list-contact"><i class="fa fa-check"></i><b>2.4</b> Mailing list &amp; contact</a></li>
</ul></li>
<li class="part"><span><b>Illumina</b></span></li>
<li class="chapter" data-level="3" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html"><i class="fa fa-check"></i><b>3</b> Psychrobacter arcticus genome QC</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-fastqc"><i class="fa fa-check"></i><b>3.1</b> P. arcticus: fastQC</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-fastqc-mcqs"><i class="fa fa-check"></i><b>3.1.1</b> P. arcticus: fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-summary"><i class="fa fa-check"></i><b>3.2</b> P. arcticus: summary</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html"><i class="fa fa-check"></i><b>4</b> Bacillus cereus genome QC</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-raw-fastqc"><i class="fa fa-check"></i><b>4.1</b> B. cereus: raw fastQC</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-raw-fastqc-mcqs"><i class="fa fa-check"></i><b>4.1.1</b> B. cereus: raw fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-quality-control"><i class="fa fa-check"></i><b>4.2</b> B. cereus: Quality control</a></li>
<li class="chapter" data-level="4.3" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-trimmed-fastqc"><i class="fa fa-check"></i><b>4.3</b> B. cereus: trimmed fastQC</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-trimmed-fastqc-mcqs"><i class="fa fa-check"></i><b>4.3.1</b> B. cereus: trimmed fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-summary"><i class="fa fa-check"></i><b>4.4</b> B. cereus: summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html"><i class="fa fa-check"></i><b>5</b> Whole shotgun metagenomics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-raw-fastqc"><i class="fa fa-check"></i><b>5.1</b> Metagenome: raw fastqc</a></li>
<li class="chapter" data-level="5.2" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-trimmed-fastqc"><i class="fa fa-check"></i><b>5.2</b> Metagenome: trimmed fastqc</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-trimmed-fastqc-mcqs"><i class="fa fa-check"></i><b>5.2.1</b> Metagenome: trimmed fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-summary"><i class="fa fa-check"></i><b>5.3</b> Metagenome: summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-16S_qc.html"><a href="06-16S_qc.html"><i class="fa fa-check"></i><b>6</b> 16S rRNA Metagenetics QC</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-16S_qc.html"><a href="06-16S_qc.html#s-fastqc"><i class="fa fa-check"></i><b>6.1</b> 16S: fastQC</a></li>
<li class="chapter" data-level="6.2" data-path="06-16S_qc.html"><a href="06-16S_qc.html#s-summary"><i class="fa fa-check"></i><b>6.2</b> 16S: summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-multiQC.html"><a href="07-multiQC.html"><i class="fa fa-check"></i><b>7</b> MultiQC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-multiQC.html"><a href="07-multiQC.html#multiqc-of-the-raw-data"><i class="fa fa-check"></i><b>7.1</b> MultiQC of the raw data</a></li>
<li class="chapter" data-level="7.2" data-path="07-multiQC.html"><a href="07-multiQC.html#fastqc-of-the-trimmed-data"><i class="fa fa-check"></i><b>7.2</b> Fastqc of the trimmed data</a></li>
<li class="chapter" data-level="7.3" data-path="07-multiQC.html"><a href="07-multiQC.html#read-data"><i class="fa fa-check"></i><b>7.3</b> Read data</a></li>
</ul></li>
<li class="part"><span><b>Shotgun metagenomics</b></span></li>
<li class="chapter" data-level="8" data-path="08-Shotgun_metagenomics.html"><a href="08-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>8</b> Shotgun metagenomics</a></li>
<li class="chapter" data-level="9" data-path="09-Krona.html"><a href="09-Krona.html"><i class="fa fa-check"></i><b>9</b> Krona</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="09-Krona.html"><a href="09-Krona.html#krona-mcqs"><i class="fa fa-check"></i><b>9.0.1</b> Krona: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Lefse.html"><a href="10-Lefse.html"><i class="fa fa-check"></i><b>10</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Lefse.html"><a href="10-Lefse.html#lefse-mcqs"><i class="fa fa-check"></i><b>10.1</b> LEfSe: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Humann.html"><a href="11-Humann.html"><i class="fa fa-check"></i><b>11</b> HUMAnN Functional profiling</a></li>
<li class="part"><span><b>Metabarcoding</b></span></li>
<li class="chapter" data-level="12" data-path="12-Metabarcoding.html"><a href="12-Metabarcoding.html"><i class="fa fa-check"></i><b>12</b> Background</a></li>
<li class="chapter" data-level="13" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html"><i class="fa fa-check"></i><b>13</b> Biodiversity</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#alpha-diversity"><i class="fa fa-check"></i><b>13.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#alpha-diversity-box-plots"><i class="fa fa-check"></i><b>13.1.1</b> Alpha diversity box plots</a></li>
<li class="chapter" data-level="13.1.2" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#alpha-diversity-mcqs"><i class="fa fa-check"></i><b>13.1.2</b> Alpha diversity MCQs</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#beta-diversity"><i class="fa fa-check"></i><b>13.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#beta-diveristy-nmds"><i class="fa fa-check"></i><b>13.2.1</b> Beta diveristy NMDS</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#taxonomic-composition"><i class="fa fa-check"></i><b>13.3</b> Taxonomic composition</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NERC ATSC Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bacillus-cereus-genome-qc" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Bacillus cereus genome QC<a href="04-b_cerues_qc.html#bacillus-cereus-genome-qc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/bacillus_cereus.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Now we will look at some read data.
We will be looking at <em>Bacillus cereus</em> MiSeq data that can be found in the following link: <a href="https://ccb.jhu.edu/gage_b/datasets/index.html" class="uri">https://ccb.jhu.edu/gage_b/datasets/index.html</a> (no need to download).</p>
<p>For the purposes of the tutorial I subsampled the reads down to a more manageable 50,000 pairs, normally read data will not come in such round numbers.</p>
<div id="b.-cereus-raw-fastqc" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> B. cereus: raw fastQC<a href="04-b_cerues_qc.html#b.-cereus-raw-fastqc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/magnifying_glass_red.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Below are the links to the <code>fastQC</code> reports for the forward (R1) and reverse (R2) reads of the <em>B. cereus</em> sequencing data.
This is the raw data, i.e. the data that comes straight from the sequencing machine.
It is always good to check the raw data before carrying out quality trimming and filtering.
We will look at the quality controlled data after this.</p>
<ul>
<li><a href="https://neof-workshops.github.io/NERC_ATSC_MAR_2023_ztkban/files/Illumina_QC/b_cereus/raw_fastqc_output/B_cereus_R1_fastqc.html"><strong>B_cereus_R1_fastqc.html</strong></a></li>
<li><a href="https://neof-workshops.github.io/NERC_ATSC_MAR_2023_ztkban/files/Illumina_QC/b_cereus/raw_fastqc_output/B_cereus_R2_fastqc.html"><strong>B_cereus_R2_fastqc.html</strong></a></li>
</ul>
<div id="b.-cereus-raw-fastqc-mcqs" class="section level3 hasAnchor" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> B. cereus: raw fastQC MCQs<a href="04-b_cerues_qc.html#b.-cereus-raw-fastqc-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/question_bubble_red.png" width="20%" style="display: block; margin: auto;" /></p>
<p>With the <code>fastQC</code> report, attempt the following MCQs, you can reattempt as many times as needed.
Please feel free to collaborate/help each other and ask the instructor questions.</p>
<ol style="list-style-type: decimal">
<li>How many total reads/sequences are there in the R2 file?
<div id="radio_KUJLDIWBRX" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_KUJLDIWBRX" value=""></input> <span><strong>36</strong></span></label><label><input type="radio" autocomplete="off" name="radio_KUJLDIWBRX" value=""></input> <span><strong>35-251</strong></span></label><label><input type="radio" autocomplete="off" name="radio_KUJLDIWBRX" value="answer"></input> <span><strong>50,000</strong></span></label>
</div></li>
<li>What is the length range of the data?
<div id="radio_JCVFDFSXSY" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_JCVFDFSXSY" value=""></input> <span><strong>36</strong></span></label><label><input type="radio" autocomplete="off" name="radio_JCVFDFSXSY" value="answer"></input> <span><strong>35-251</strong></span></label><label><input type="radio" autocomplete="off" name="radio_JCVFDFSXSY" value=""></input> <span><strong>50,000</strong></span></label>
</div></li>
<li>What is the average GC (%) content of the data?
<div id="radio_DEYYKADDEL" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_DEYYKADDEL" value="answer"></input> <span><strong>36</strong></span></label><label><input type="radio" autocomplete="off" name="radio_DEYYKADDEL" value=""></input> <span><strong>35-251</strong></span></label><label><input type="radio" autocomplete="off" name="radio_DEYYKADDEL" value=""></input> <span><strong>50,000</strong></span></label>
</div></li>
<li>At what base position/s does the IQR (Interquartile Range/yellow box) drop below a quality of Q28 for the R1 data?
<div id="radio_DAQJDQABPY" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_DAQJDQABPY" value=""></input> <span><strong>15-19</strong></span></label><label><input type="radio" autocomplete="off" name="radio_DAQJDQABPY" value=""></input> <span><strong>235-239</strong></span></label><label><input type="radio" autocomplete="off" name="radio_DAQJDQABPY" value="answer"></input> <span><strong>251</strong></span></label>
</div></li>
<li>At what base position/s does the IQR and whiskers (lines) first completely go above a quality of Q28 for the R2 data?
<div id="radio_ISLPSITPIR" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ISLPSITPIR" value="answer"></input> <span><strong>15-19</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ISLPSITPIR" value=""></input> <span><strong>235-239</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ISLPSITPIR" value=""></input> <span><strong>251</strong></span></label>
</div></li>
<li>At what base position/s does the IQR (Interquartile Range/yellow box) drop below a quality of Q20 for the R2 data?
<div id="radio_CPDZFVMBWU" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_CPDZFVMBWU" value=""></input> <span><strong>15-19</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CPDZFVMBWU" value="answer"></input> <span><strong>235-239</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CPDZFVMBWU" value=""></input> <span><strong>251</strong></span></label>
</div></li>
<li>What is the most common "mean sequence quality" for the reads in R2 (see the peak of the "Per sequence quality scores" graph)?
<div id="radio_KKMYCROLKF" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_KKMYCROLKF" value="answer"></input> <span><strong>37</strong></span></label><label><input type="radio" autocomplete="off" name="radio_KKMYCROLKF" value=""></input> <span><strong>129</strong></span></label><label><input type="radio" autocomplete="off" name="radio_KKMYCROLKF" value=""></input> <span><strong>250-252</strong></span></label>
</div></li>
<li>What is the total count of overrepresented sequences in R1?
<div id="radio_XGJUELGGGU" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_XGJUELGGGU" value=""></input> <span><strong>37</strong></span></label><label><input type="radio" autocomplete="off" name="radio_XGJUELGGGU" value="answer"></input> <span><strong>129</strong></span></label><label><input type="radio" autocomplete="off" name="radio_XGJUELGGGU" value=""></input> <span><strong>250-252</strong></span></label>
</div></li>
<li>What is the most common base length range for R2 (see the peak of the "Sequence Length Distribution" graph)?
<div id="radio_PDZRTWENHF" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_PDZRTWENHF" value=""></input> <span><strong>37</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PDZRTWENHF" value=""></input> <span><strong>129</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PDZRTWENHF" value="answer"></input> <span><strong>250-252</strong></span></label>
</div></li>
</ol>
<p>This data is pretty typical of what would normally be seen.
We are therefore going to quality trim and filter the data.
Unfortunately data straight from the sequencer always needs quality control.</p>
</div>
</div>
<div id="b.-cereus-quality-control" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> B. cereus: Quality control<a href="04-b_cerues_qc.html#b.-cereus-quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/quality_trimming_and_filtering.png" width="30%" style="display: block; margin: auto;" /></p>
<p>Quality control generally comes in two forms:</p>
<ol style="list-style-type: decimal">
<li><strong>Trimming</strong>: This is directly cutting off bits of sequence. This is typical in the form of trimming off low quality bases from the end of reads and trimming off adapters at the start of reads.</li>
<li><strong>Filtering</strong>: This occurs when entire reads are removed. A typical occurrence of this is when a read is too small as we do not want reads below a certain length.</li>
</ol>
<p>The data was quality controlled with <a href="http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf"><code>trimmomatic</code></a>.
A sliding window size of 4 with a quality threshold of Q20 was used.
Additionally, a minimum length of 50 was used for filtering.</p>
</div>
<div id="b.-cereus-trimmed-fastqc" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> B. cereus: trimmed fastQC<a href="04-b_cerues_qc.html#b.-cereus-trimmed-fastqc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/magnifying_glass_purple.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Below are the links to the <code>fastQC</code> reports for the trimmed forward (R1) and reverse (R2) reads of the <em>B. cereus</em> sequencing data.</p>
<ul>
<li><a href="https://neof-workshops.github.io/NERC_ATSC_MAR_2023_ztkban/files/Illumina_QC/b_cereus/trimmomatic_fastqc_output/B_cereus_trimmed_R1_fastqc.html"><strong>B_cereus_trimmed_R1_fastqc.html</strong></a></li>
<li><a href="https://neof-workshops.github.io/NERC_ATSC_MAR_2023_ztkban/files/Illumina_QC/b_cereus/trimmomatic_fastqc_output/B_cereus_trimmed_R2_fastqc.html"><strong>B_cereus_trimmed_R2_fastqc.html</strong></a></li>
</ul>
<div id="b.-cereus-trimmed-fastqc-mcqs" class="section level3 hasAnchor" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> B. cereus: trimmed fastQC MCQs<a href="04-b_cerues_qc.html#b.-cereus-trimmed-fastqc-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/question_bubble_purple.png" width="20%" style="display: block; margin: auto;" /></p>
<p>To see how well the reads have improved answer the below questions, based on the trimmed <code>fastQC</code> reports, and compare to them to the answers of the raw read <code>fastQC</code> questions.</p>
<ol style="list-style-type: decimal">
<li>How many total reads/sequences are there in the R2 file?
<div id="radio_OCDAGHNIPC" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_OCDAGHNIPC" value=""></input> <span><strong>35</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OCDAGHNIPC" value=""></input> <span><strong>50-251</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OCDAGHNIPC" value="answer"></input> <span><strong>38,309</strong></span></label>
</div></li>
<li>What is the length range of the data?
<div id="radio_ISAGRZRHQZ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ISAGRZRHQZ" value=""></input> <span><strong>35</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ISAGRZRHQZ" value="answer"></input> <span><strong>50-251</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ISAGRZRHQZ" value=""></input> <span><strong>38,309</strong></span></label>
</div></li>
<li>What is the average GC (%) content of the data?
<div id="radio_KNDBJWBPYP" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_KNDBJWBPYP" value="answer"></input> <span><strong>35</strong></span></label><label><input type="radio" autocomplete="off" name="radio_KNDBJWBPYP" value=""></input> <span><strong>50-251</strong></span></label><label><input type="radio" autocomplete="off" name="radio_KNDBJWBPYP" value=""></input> <span><strong>38,309</strong></span></label>
</div></li>
<li>In R1, do the IQRs and/or whiskers drop below Q28 in the "Per base sequence quality" graph?
<div id="radio_PGAZPTAXNK" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_PGAZPTAXNK" value=""></input> <span><strong>Yes</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PGAZPTAXNK" value="answer"></input> <span><strong>No</strong></span></label>
</div></li>
<li>In R2, do the IQRs and/or whiskers drop below Q28 in the "Per base sequence quality" graph?
<div id="radio_OXOHDYVCON" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_OXOHDYVCON" value="answer"></input> <span><strong>Yes</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OXOHDYVCON" value=""></input> <span><strong>No</strong></span></label>
</div></li>
<li>In R2, do the IQRs drop below Q28 in the "Per base sequence quality" graph?
<div id="radio_OABGZDGYTK" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_OABGZDGYTK" value=""></input> <span><strong>Yes</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OABGZDGYTK" value="answer"></input> <span><strong>No</strong></span></label>
</div></li>
<li>What is the most common "mean sequence quality" for the reads in R2 (see the peak of the "Per sequence quality scores" graph)?
<div id="radio_MBUEEIWQZQ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_MBUEEIWQZQ" value=""></input> <span><strong>0</strong></span></label><label><input type="radio" autocomplete="off" name="radio_MBUEEIWQZQ" value="answer"></input> <span><strong>37</strong></span></label><label><input type="radio" autocomplete="off" name="radio_MBUEEIWQZQ" value=""></input> <span><strong>250-251</strong></span></label>
</div></li>
<li>What is the total count of overrepresented sequences in R1?
<div id="radio_BOBPDNPRBE" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BOBPDNPRBE" value="answer"></input> <span><strong>0</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BOBPDNPRBE" value=""></input> <span><strong>37</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BOBPDNPRBE" value=""></input> <span><strong>250-251</strong></span></label>
</div></li>
<li>What is the most common base length range for R2 (see the peak of the "Sequence Length Distribution" graph)?
<div id="radio_OHEDJYTDWH" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_OHEDJYTDWH" value=""></input> <span><strong>0</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OHEDJYTDWH" value=""></input> <span><strong>37</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OHEDJYTDWH" value="answer"></input> <span><strong>250-251</strong></span></label>
</div></li>
</ol>
</div>
</div>
<div id="b.-cereus-summary" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> B. cereus: summary<a href="04-b_cerues_qc.html#b.-cereus-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_purple.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Some things to note:</p>
<ul>
<li>The amount of reads has significantly decreased due to quality control. This is expected, however this will need to be taken into account. If you do not have enough reads for the downstream analysis you will need to be less stringent on the trimming or filtering. What is the minimum number of reads required for downstream analysis? Unfortunately this depends on the specific analysis at hand. For genome assemblies generally the rule of thumb is you want at least 20 times as much bases in the reads as would be in the final assembly (based on the genome size of the organisms).</li>
<li>R2 quality is still worse than R1 quality. Again this is normal.</li>
<li>The quality controlled reads of the reverse reads still drop below Q28 towards the end of the reads. This is pretty typical. You could be more stringent but as previously mentioned you may be removing too many reads.</li>
<li>The overrepresented sequences, which were Illumina adapters needed for sequencing, were removed by <code>trimmomatic</code>. This is good as we don't need/want them any more.</li>
</ul>
<p><strong>Note</strong>: If you are getting data from a sequencing centre they will have most likely removed the adapters already.
Been sure to check they are not appearing in the Overrepresented sequences or the Adapter Content section of the <code>fastqc</code> report after trimming before continuing.</p>
<p>If adapters are present you can use <code>Cutadapt</code> to remove them: <a href="https://cutadapt.readthedocs.io/en/stable/guide.html" class="uri">https://cutadapt.readthedocs.io/en/stable/guide.html</a></p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="03-p_arcticus_qc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="05-shotgun_metagenomic_qc.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
