<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Krona | NERC ATSC Genomics</title>
  <meta name="description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Krona | NERC ATSC Genomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Krona | NERC ATSC Genomics" />
  
  <meta name="twitter:description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-10-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="08-Shotgun_metagenomics.html"/>
<link rel="next" href="10-Lefse.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>NERC ATSC MAR 2023</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="01-Intro.html"><a href="01-Intro.html#example-mcqs"><i class="fa fa-check"></i><b>1.1</b> Example MCQs</a></li>
</ul></li>
<li class="part"><span><b>NEOF</b></span></li>
<li class="chapter" data-level="2" data-path="02-NEOF.html"><a href="02-NEOF.html"><i class="fa fa-check"></i><b>2</b> NEOF</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-NEOF.html"><a href="02-NEOF.html#about-neof"><i class="fa fa-check"></i><b>2.1</b> About NEOF</a></li>
<li class="chapter" data-level="2.2" data-path="02-NEOF.html"><a href="02-NEOF.html#capabilities"><i class="fa fa-check"></i><b>2.2</b> Capabilities</a></li>
<li class="chapter" data-level="2.3" data-path="02-NEOF.html"><a href="02-NEOF.html#visitor-facility-and-training"><i class="fa fa-check"></i><b>2.3</b> Visitor facility and training</a></li>
<li class="chapter" data-level="2.4" data-path="02-NEOF.html"><a href="02-NEOF.html#mailing-list-contact"><i class="fa fa-check"></i><b>2.4</b> Mailing list &amp; contact</a></li>
</ul></li>
<li class="part"><span><b>Illumina</b></span></li>
<li class="chapter" data-level="3" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html"><i class="fa fa-check"></i><b>3</b> Psychrobacter arcticus genome QC</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-fastqc"><i class="fa fa-check"></i><b>3.1</b> P. arcticus: fastQC</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-fastqc-mcqs"><i class="fa fa-check"></i><b>3.1.1</b> P. arcticus: fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-summary"><i class="fa fa-check"></i><b>3.2</b> P. arcticus: summary</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html"><i class="fa fa-check"></i><b>4</b> Bacillus cereus genome QC</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-raw-fastqc"><i class="fa fa-check"></i><b>4.1</b> B. cereus: raw fastQC</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-raw-fastqc-mcqs"><i class="fa fa-check"></i><b>4.1.1</b> B. cereus: raw fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-quality-control"><i class="fa fa-check"></i><b>4.2</b> B. cereus: Quality control</a></li>
<li class="chapter" data-level="4.3" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-trimmed-fastqc"><i class="fa fa-check"></i><b>4.3</b> B. cereus: trimmed fastQC</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-trimmed-fastqc-mcqs"><i class="fa fa-check"></i><b>4.3.1</b> B. cereus: trimmed fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-summary"><i class="fa fa-check"></i><b>4.4</b> B. cereus: summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html"><i class="fa fa-check"></i><b>5</b> Whole shotgun metagenomics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-raw-fastqc"><i class="fa fa-check"></i><b>5.1</b> Metagenome: raw fastqc</a></li>
<li class="chapter" data-level="5.2" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-trimmed-fastqc"><i class="fa fa-check"></i><b>5.2</b> Metagenome: trimmed fastqc</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-trimmed-fastqc-mcqs"><i class="fa fa-check"></i><b>5.2.1</b> Metagenome: trimmed fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-summary"><i class="fa fa-check"></i><b>5.3</b> Metagenome: summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-16S_qc.html"><a href="06-16S_qc.html"><i class="fa fa-check"></i><b>6</b> 16S rRNA Metagenetics QC</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-16S_qc.html"><a href="06-16S_qc.html#s-fastqc"><i class="fa fa-check"></i><b>6.1</b> 16S: fastQC</a></li>
<li class="chapter" data-level="6.2" data-path="06-16S_qc.html"><a href="06-16S_qc.html#s-summary"><i class="fa fa-check"></i><b>6.2</b> 16S: summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-multiQC.html"><a href="07-multiQC.html"><i class="fa fa-check"></i><b>7</b> MultiQC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-multiQC.html"><a href="07-multiQC.html#multiqc-of-the-raw-data"><i class="fa fa-check"></i><b>7.1</b> MultiQC of the raw data</a></li>
<li class="chapter" data-level="7.2" data-path="07-multiQC.html"><a href="07-multiQC.html#fastqc-of-the-trimmed-data"><i class="fa fa-check"></i><b>7.2</b> Fastqc of the trimmed data</a></li>
<li class="chapter" data-level="7.3" data-path="07-multiQC.html"><a href="07-multiQC.html#read-data"><i class="fa fa-check"></i><b>7.3</b> Read data</a></li>
</ul></li>
<li class="part"><span><b>Shotgun metagenomics</b></span></li>
<li class="chapter" data-level="8" data-path="08-Shotgun_metagenomics.html"><a href="08-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>8</b> Shotgun metagenomics</a></li>
<li class="chapter" data-level="9" data-path="09-Krona.html"><a href="09-Krona.html"><i class="fa fa-check"></i><b>9</b> Krona</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="09-Krona.html"><a href="09-Krona.html#krona-mcqs"><i class="fa fa-check"></i><b>9.0.1</b> Krona: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Lefse.html"><a href="10-Lefse.html"><i class="fa fa-check"></i><b>10</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Lefse.html"><a href="10-Lefse.html#lefse-mcqs"><i class="fa fa-check"></i><b>10.1</b> LEfSe: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Humann.html"><a href="11-Humann.html"><i class="fa fa-check"></i><b>11</b> HUMAnN Functional profiling</a></li>
<li class="part"><span><b>Metabarcoding</b></span></li>
<li class="chapter" data-level="12" data-path="12-Metabarcoding.html"><a href="12-Metabarcoding.html"><i class="fa fa-check"></i><b>12</b> Background</a></li>
<li class="chapter" data-level="13" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html"><i class="fa fa-check"></i><b>13</b> Biodiversity</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#alpha-diversity"><i class="fa fa-check"></i><b>13.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#alpha-diversity-box-plots"><i class="fa fa-check"></i><b>13.1.1</b> Alpha diversity box plots</a></li>
<li class="chapter" data-level="13.1.2" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#alpha-diversity-mcqs"><i class="fa fa-check"></i><b>13.1.2</b> Alpha diversity MCQs</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#beta-diversity"><i class="fa fa-check"></i><b>13.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#beta-diveristy-nmds"><i class="fa fa-check"></i><b>13.2.1</b> Beta diveristy NMDS</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#taxonomic-composition"><i class="fa fa-check"></i><b>13.3</b> Taxonomic composition</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NERC ATSC Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="krona" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Krona<a href="09-Krona.html#krona" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/krona.png" width="15%" style="display: block; margin: auto;" /></p>
<p><code>Krona</code> is an interactive metagenome species abundance visualisation tool. We will use it to view output from Kraken2.</p>
<p>We can view our interactive chart in a web browser. This shows the percentage of reads that were classified to various taxonomies at different levels.</p>
<p><strong>Note</strong>: This only contains 3 samples; 2 western samples (W1 &amp; W2) and one Korean sample (K1).</p>
<p><a href="https://neof-workshops.github.io/NERC_ATSC_MAR_2023_ztkban/files/Shotgun_metagenomics/kraken2.krona.html"><strong>Krona plot</strong></a></p>
<p>This is an interactive multi-tiered pie chart with many options. Some of the interactivity is described below:</p>
<ul>
<li>To choose a sample to view, click on the sample name in the top left.</li>
<li>To zoom into a lower taxonomy, double click on the taxonomy's name on the pie chart.
<ul>
<li>E.g. double click the word "Bacteria" on the pie chart to zoom into Bacteria and therefore ignore Eukaryota, and "[other Root]".</li>
</ul></li>
<li>To zoom out to a higher taxonomy, click on the taxonomy's name in the middle of the pie chart.
<ul>
<li>E.g. click the word "root" in the middle of the pie chart to zoom back out from any lower level of taxonomy.</li>
</ul></li>
<li>To view percentage stats of a taxonomy, click on the taxonomy name on the pie chart.
<ul>
<li>This will display the percentage this taxonomy covers of all the different taxonomies above it that it belongs to. This is displayed on the right side of the screen.</li>
<li>E.g. Click on Proteobacteria to see the percentage of the Root it accounts for and to see the percentage of Bacteria it accounts for in the sample.</li>
</ul></li>
</ul>
<div id="krona-mcqs" class="section level3 hasAnchor" number="9.0.1">
<h3><span class="header-section-number">9.0.1</span> Krona: MCQs<a href="09-Krona.html#krona-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/question_bubble_red.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Viewing the Krona, attempt the below MCQs. Please collaborate and ask questions to the instructor.</p>
<ol style="list-style-type: decimal">
<li>What percentage of the <strong>root</strong> was classified to <strong>Bacteria</strong> for <strong>K1</strong>?
<div id="radio_DTLUAPHSHA" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_DTLUAPHSHA" value=""></input> <span><strong>43%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_DTLUAPHSHA" value="answer"></input> <span><strong>35%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_DTLUAPHSHA" value=""></input> <span><strong>14%</strong></span></label>
</div></li>
<li>What percentage of <strong>bacteria</strong> was classified to <strong>Proteobacteria</strong> for <strong>K2</strong>?
<div id="radio_VWYDPGTXUK" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_VWYDPGTXUK" value="answer"></input> <span><strong>43%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VWYDPGTXUK" value=""></input> <span><strong>35%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VWYDPGTXUK" value=""></input> <span><strong>14%</strong></span></label>
</div></li>
<li>What percentage of the <strong>root</strong> was classified to <strong>Azorhizobium</strong> for <strong>W1</strong>?
<div id="radio_VAYPIQFZUB" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_VAYPIQFZUB" value=""></input> <span><strong>43%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VAYPIQFZUB" value=""></input> <span><strong>35%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VAYPIQFZUB" value="answer"></input> <span><strong>14%</strong></span></label>
</div></li>
<li>What percentage of the <strong>Terrabacteria groups</strong> was classified to <strong>Cellulomonas gilvus</strong> for <strong>K2</strong>?
<div id="radio_FNROJKUECE" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_FNROJKUECE" value="answer"></input> <span><strong>39%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_FNROJKUECE" value=""></input> <span><strong>27%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_FNROJKUECE" value=""></input> <span><strong>6%</strong></span></label>
</div></li>
<li>What percentage of <strong>Alphaproteobacteria</strong> was classified to <strong>Phenylobacterium immobile</strong> for <strong>K1</strong>?
<div id="radio_NGEQFUHSUR" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_NGEQFUHSUR" value=""></input> <span><strong>39%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_NGEQFUHSUR" value=""></input> <span><strong>27%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_NGEQFUHSUR" value="answer"></input> <span><strong>6%</strong></span></label>
</div></li>
<li>What percentage of <strong>Viridiplantae</strong> (in Eukaryota) was classified to <strong>Zygnemataceae</strong> for <strong>W1</strong>?
<div id="radio_YALSALQRFG" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_YALSALQRFG" value=""></input> <span><strong>39%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_YALSALQRFG" value="answer"></input> <span><strong>27%</strong></span></label><label><input type="radio" autocomplete="off" name="radio_YALSALQRFG" value=""></input> <span><strong>6%</strong></span></label>
</div></li>
</ol>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="08-Shotgun_metagenomics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="10-Lefse.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
