<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Biodiversity | NERC ATSC Genomics</title>
  <meta name="description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Biodiversity | NERC ATSC Genomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Biodiversity | NERC ATSC Genomics" />
  
  <meta name="twitter:description" content="NERC book for the Genomics section of the March 2023 ATSC" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-10-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="12-Metabarcoding.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>NERC ATSC MAR 2023</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="01-Intro.html"><a href="01-Intro.html#example-mcqs"><i class="fa fa-check"></i><b>1.1</b> Example MCQs</a></li>
</ul></li>
<li class="part"><span><b>NEOF</b></span></li>
<li class="chapter" data-level="2" data-path="02-NEOF.html"><a href="02-NEOF.html"><i class="fa fa-check"></i><b>2</b> NEOF</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-NEOF.html"><a href="02-NEOF.html#about-neof"><i class="fa fa-check"></i><b>2.1</b> About NEOF</a></li>
<li class="chapter" data-level="2.2" data-path="02-NEOF.html"><a href="02-NEOF.html#capabilities"><i class="fa fa-check"></i><b>2.2</b> Capabilities</a></li>
<li class="chapter" data-level="2.3" data-path="02-NEOF.html"><a href="02-NEOF.html#visitor-facility-and-training"><i class="fa fa-check"></i><b>2.3</b> Visitor facility and training</a></li>
<li class="chapter" data-level="2.4" data-path="02-NEOF.html"><a href="02-NEOF.html#mailing-list-contact"><i class="fa fa-check"></i><b>2.4</b> Mailing list &amp; contact</a></li>
</ul></li>
<li class="part"><span><b>Illumina</b></span></li>
<li class="chapter" data-level="3" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html"><i class="fa fa-check"></i><b>3</b> Psychrobacter arcticus genome QC</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-fastqc"><i class="fa fa-check"></i><b>3.1</b> P. arcticus: fastQC</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-fastqc-mcqs"><i class="fa fa-check"></i><b>3.1.1</b> P. arcticus: fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-p_arcticus_qc.html"><a href="03-p_arcticus_qc.html#p.-arcticus-summary"><i class="fa fa-check"></i><b>3.2</b> P. arcticus: summary</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html"><i class="fa fa-check"></i><b>4</b> Bacillus cereus genome QC</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-raw-fastqc"><i class="fa fa-check"></i><b>4.1</b> B. cereus: raw fastQC</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-raw-fastqc-mcqs"><i class="fa fa-check"></i><b>4.1.1</b> B. cereus: raw fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-quality-control"><i class="fa fa-check"></i><b>4.2</b> B. cereus: Quality control</a></li>
<li class="chapter" data-level="4.3" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-trimmed-fastqc"><i class="fa fa-check"></i><b>4.3</b> B. cereus: trimmed fastQC</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-trimmed-fastqc-mcqs"><i class="fa fa-check"></i><b>4.3.1</b> B. cereus: trimmed fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="04-b_cerues_qc.html"><a href="04-b_cerues_qc.html#b.-cereus-summary"><i class="fa fa-check"></i><b>4.4</b> B. cereus: summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html"><i class="fa fa-check"></i><b>5</b> Whole shotgun metagenomics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-raw-fastqc"><i class="fa fa-check"></i><b>5.1</b> Metagenome: raw fastqc</a></li>
<li class="chapter" data-level="5.2" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-trimmed-fastqc"><i class="fa fa-check"></i><b>5.2</b> Metagenome: trimmed fastqc</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-trimmed-fastqc-mcqs"><i class="fa fa-check"></i><b>5.2.1</b> Metagenome: trimmed fastQC MCQs</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="05-shotgun_metagenomic_qc.html"><a href="05-shotgun_metagenomic_qc.html#metagenome-summary"><i class="fa fa-check"></i><b>5.3</b> Metagenome: summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-16S_qc.html"><a href="06-16S_qc.html"><i class="fa fa-check"></i><b>6</b> 16S rRNA Metagenetics QC</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-16S_qc.html"><a href="06-16S_qc.html#s-fastqc"><i class="fa fa-check"></i><b>6.1</b> 16S: fastQC</a></li>
<li class="chapter" data-level="6.2" data-path="06-16S_qc.html"><a href="06-16S_qc.html#s-summary"><i class="fa fa-check"></i><b>6.2</b> 16S: summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-multiQC.html"><a href="07-multiQC.html"><i class="fa fa-check"></i><b>7</b> MultiQC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-multiQC.html"><a href="07-multiQC.html#multiqc-of-the-raw-data"><i class="fa fa-check"></i><b>7.1</b> MultiQC of the raw data</a></li>
<li class="chapter" data-level="7.2" data-path="07-multiQC.html"><a href="07-multiQC.html#fastqc-of-the-trimmed-data"><i class="fa fa-check"></i><b>7.2</b> Fastqc of the trimmed data</a></li>
<li class="chapter" data-level="7.3" data-path="07-multiQC.html"><a href="07-multiQC.html#read-data"><i class="fa fa-check"></i><b>7.3</b> Read data</a></li>
</ul></li>
<li class="part"><span><b>Shotgun metagenomics</b></span></li>
<li class="chapter" data-level="8" data-path="08-Shotgun_metagenomics.html"><a href="08-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>8</b> Shotgun metagenomics</a></li>
<li class="chapter" data-level="9" data-path="09-Krona.html"><a href="09-Krona.html"><i class="fa fa-check"></i><b>9</b> Krona</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="09-Krona.html"><a href="09-Krona.html#krona-mcqs"><i class="fa fa-check"></i><b>9.0.1</b> Krona: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Lefse.html"><a href="10-Lefse.html"><i class="fa fa-check"></i><b>10</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Lefse.html"><a href="10-Lefse.html#lefse-mcqs"><i class="fa fa-check"></i><b>10.1</b> LEfSe: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Humann.html"><a href="11-Humann.html"><i class="fa fa-check"></i><b>11</b> HUMAnN Functional profiling</a></li>
<li class="part"><span><b>Metabarcoding</b></span></li>
<li class="chapter" data-level="12" data-path="12-Metabarcoding.html"><a href="12-Metabarcoding.html"><i class="fa fa-check"></i><b>12</b> Background</a></li>
<li class="chapter" data-level="13" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html"><i class="fa fa-check"></i><b>13</b> Biodiversity</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#alpha-diversity"><i class="fa fa-check"></i><b>13.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#alpha-diversity-box-plots"><i class="fa fa-check"></i><b>13.1.1</b> Alpha diversity box plots</a></li>
<li class="chapter" data-level="13.1.2" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#alpha-diversity-mcqs"><i class="fa fa-check"></i><b>13.1.2</b> Alpha diversity MCQs</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#beta-diversity"><i class="fa fa-check"></i><b>13.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#beta-diveristy-nmds"><i class="fa fa-check"></i><b>13.2.1</b> Beta diveristy NMDS</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Biodiversity.html"><a href="13-Biodiversity.html#taxonomic-composition"><i class="fa fa-check"></i><b>13.3</b> Taxonomic composition</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NERC ATSC Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="biodiversity" class="section level1 hasAnchor" number="13">
<h1><span class="header-section-number">Chapter 13</span> Biodiversity<a href="13-Biodiversity.html#biodiversity" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this section you will look at some biodiversity plots of the Lake Tanganyika dataset. These encompas only a few methods to look at biodiversity data.</p>
<div id="alpha-diversity" class="section level2 hasAnchor" number="13.1">
<h2><span class="header-section-number">13.1</span> Alpha diversity<a href="13-Biodiversity.html#alpha-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/alpha_div.png" width="15%" style="display: block; margin: auto;" /></p>
<p>The alpha diversity plots will use the following metrics.</p>
<p><strong>1. Shannon's diversity index</strong></p>
<p>A measure of diversity where a higher number means higher diversity. Shannon's index accounts for the abundance and evenness of the features present. If richness and evenness increase the diversity score increases. Values can range from one (in case of a single dominant species) to the total number of all species (in case of all species having equal abundance).</p>
<p><strong>Equation</strong>:</p>
<p><span class="math display">\[
H =
-\sum_{i=1}^{S} p_i lnp_i
\]</span></p>
<ul>
<li>H = Shannon diversity index</li>
<li>pi = is the proportion of species i</li>
<li>S = Number of species</li>
</ul>
<p><strong>2. Simpson diversity index</strong>
A measure of diversity based on number of features present and the relative abundance of each feature. If richness and evenness increase the diversity score increases. The values range from 0 (no diversity) to 1 (infinite diversity).</p>
<p><strong>Equation</strong>:
<span class="math display">\[
D = 1 -
{\sum_{i=1}^{S} {p_i} ^{2}}
\]</span></p>
<ul>
<li>D = Simpson diversity index</li>
<li>pi = is the proportion of species i</li>
<li>S = Number of species</li>
</ul>
<div id="alpha-diversity-box-plots" class="section level3 hasAnchor" number="13.1.1">
<h3><span class="header-section-number">13.1.1</span> Alpha diversity box plots<a href="13-Biodiversity.html#alpha-diversity-box-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Below are alpha diversity box plots grouped by the sites.</p>
<p><img src="figures/alpha2.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="alpha-diversity-mcqs" class="section level3 hasAnchor" number="13.1.2">
<h3><span class="header-section-number">13.1.2</span> Alpha diversity MCQs<a href="13-Biodiversity.html#alpha-diversity-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Attempt the following MCQs</p>
Which site has the the lowest diversity?
<div id="radio_YISXAMONDV" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_YISXAMONDV" value="answer"></input> <span>Aquarium</span></label><label><input type="radio" autocomplete="off" name="radio_YISXAMONDV" value=""></input> <span>Site 2</span></label><label><input type="radio" autocomplete="off" name="radio_YISXAMONDV" value=""></input> <span>Site 10</span></label>
</div>
Which site, of the options provided, has the highest diversity?
<div id="radio_UUNSLDAKMF" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_UUNSLDAKMF" value=""></input> <span>Aquarium</span></label><label><input type="radio" autocomplete="off" name="radio_UUNSLDAKMF" value=""></input> <span>Site 2</span></label><label><input type="radio" autocomplete="off" name="radio_UUNSLDAKMF" value="answer"></input> <span>Site 10</span></label>
</div>
Which site has the greatest range of Shannon values (i.e. largest difference between minimum and maximum values.)?
<div id="radio_SFESOFBVJT" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_SFESOFBVJT" value=""></input> <span>Aquarium</span></label><label><input type="radio" autocomplete="off" name="radio_SFESOFBVJT" value="answer"></input> <span>Site 2</span></label><label><input type="radio" autocomplete="off" name="radio_SFESOFBVJT" value=""></input> <span>Site 10</span></label>
</div>
How many sites have a median Shannon value greater than 2.0?
<div id="radio_HTOONOAHQQ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_HTOONOAHQQ" value=""></input> <span>2</span></label><label><input type="radio" autocomplete="off" name="radio_HTOONOAHQQ" value=""></input> <span>3</span></label><label><input type="radio" autocomplete="off" name="radio_HTOONOAHQQ" value="answer"></input> <span>4</span></label>
</div>
How many sites have a median Simpson value less than 0.6?
<div id="radio_ZPJFMQIYLE" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ZPJFMQIYLE" value="answer"></input> <span>2</span></label><label><input type="radio" autocomplete="off" name="radio_ZPJFMQIYLE" value=""></input> <span>3</span></label><label><input type="radio" autocomplete="off" name="radio_ZPJFMQIYLE" value=""></input> <span>4</span></label>
</div>
</div>
</div>
<div id="beta-diversity" class="section level2 hasAnchor" number="13.2">
<h2><span class="header-section-number">13.2</span> Beta diversity<a href="13-Biodiversity.html#beta-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/beta_div.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Beta diversity compares the difference in diversity between two sites, or to put it another way it calculates the number of species that are not the same in the two sites.</p>
<div id="beta-diveristy-nmds" class="section level3 hasAnchor" number="13.2.1">
<h3><span class="header-section-number">13.2.1</span> Beta diveristy NMDS<a href="13-Biodiversity.html#beta-diveristy-nmds" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Below is the NMDS (Non-metric MultiDimenstional Scaling) plot using Bray-Curtis distances. The closer the points are the more similar they are in terms of taxonomic composition.</p>
<div class="webex-solution">
<button>
Bray-Curtis info
</button>
<p>The Bray-Curtis metric is a dissimilarity measure that can quantify the level of difference between two samples. Two identical samples would have a Bray-Curtis measure of 0 (i.e. they have 0 dissimilarity).</p>
<p>There are 2 versions but the one used by QIIME2 looks at the number of features shared by the 2 samples.</p>
<p><strong>Equation</strong>:</p>
<p><span class="math display">\[
CBC = 1 –
(\frac{2c}
{a + b})
\]</span></p>
<ul>
<li>c = # features present in both samples</li>
<li>a = # features present in sample a</li>
<li>b = # features present in sample b</li>
</ul>
</div>
<p><img src="figures/beta.png" width="80%" style="display: block; margin: auto;" /></p>
<p>In general the samples do seem to cluster roughly by site in the NMDS plot.</p>
</div>
</div>
<div id="taxonomic-composition" class="section level2 hasAnchor" number="13.3">
<h2><span class="header-section-number">13.3</span> Taxonomic composition<a href="13-Biodiversity.html#taxonomic-composition" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The below plot displays the relative abundances of the different families detected in the samples.</p>
<p><img src="figures/Family_abundance_MiFish.png" width="80%" style="display: block; margin: auto;" /></p>
<p>We can see that the five aquarium samples contain mostly Cichlidae sequences, apart from a very small number of unassigned (NA) and misassigned (there were only cichlid fish in the aquarium) sequences. Site 11 has more than 50% unassigned sequences in all three replicate samples. All of the samples containing &gt; 10% Procatopodidae sequences are grouped together in the NMDS plot (especially sites 19 &amp; 20, but also two samples from site 2, and the one sample from site 18), suggesting this may be in part driving the observed patterns of community similarity.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="12-Metabarcoding.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
