<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Other types of data | Python for Bioinformatics - Supplementary Materials</title>
  <meta name="description" content="CGR book for the Python for Bioinformatics workshop - Supplementary materials" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Other types of data | Python for Bioinformatics - Supplementary Materials" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/CGR_large.png" />
  <meta property="og:description" content="CGR book for the Python for Bioinformatics workshop - Supplementary materials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Other types of data | Python for Bioinformatics - Supplementary Materials" />
  
  <meta name="twitter:description" content="CGR book for the Python for Bioinformatics workshop - Supplementary materials" />
  <meta name="twitter:image" content="/figures/CGR_large.png" />

<meta name="author" content="Graeme Fox &amp; Matthew R. Gemmell" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/CGR_favicon.png" type="image/x-icon" />
<link rel="prev" href="01-Intro.html"/>
<link rel="next" href="03_plotting_data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://www.liverpool.ac.uk/genomic-research/" target="blank">
<img src="figures/CGR_large.png" width=250px></a></li>
<li><a>Python for Bioinformatics Supplemental materials</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02_other_data.html"><a href="02_other_data.html"><i class="fa fa-check"></i><b>2</b> Other types of data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02_other_data.html"><a href="02_other_data.html#background-info-on-the-example-data"><i class="fa fa-check"></i><b>2.1</b> Background info on the example data</a></li>
<li class="chapter" data-level="2.2" data-path="02_other_data.html"><a href="02_other_data.html#read-a-.csv-file-with-pandas"><i class="fa fa-check"></i><b>2.2</b> Read a .csv file with pandas</a>
<ul>
<li class="chapter" data-level="" data-path="02_other_data.html"><a href="02_other_data.html#import-module"><i class="fa fa-check"></i>Import module</a></li>
<li class="chapter" data-level="" data-path="02_other_data.html"><a href="02_other_data.html#read-data"><i class="fa fa-check"></i>Read data</a></li>
<li class="chapter" data-level="" data-path="02_other_data.html"><a href="02_other_data.html#data-types"><i class="fa fa-check"></i>Data types</a></li>
<li class="chapter" data-level="" data-path="02_other_data.html"><a href="02_other_data.html#columns-and-index"><i class="fa fa-check"></i>Columns and index</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="02_other_data.html"><a href="02_other_data.html#make-more-complex-selections"><i class="fa fa-check"></i><b>2.3</b> Make more complex selections</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="02_other_data.html"><a href="02_other_data.html#select-columns-by-name"><i class="fa fa-check"></i><b>2.3.1</b> Select columns by name</a></li>
<li class="chapter" data-level="2.3.2" data-path="02_other_data.html"><a href="02_other_data.html#select-records-by-their-properties"><i class="fa fa-check"></i><b>2.3.2</b> Select records by their properties</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="02_other_data.html"><a href="02_other_data.html#applying-functions-and-methods-to-pandas-objects"><i class="fa fa-check"></i><b>2.4</b> Applying functions and methods to pandas objects</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03_plotting_data.html"><a href="03_plotting_data.html"><i class="fa fa-check"></i><b>3</b> Plotting using Python and Matplotlib</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03_plotting_data.html"><a href="03_plotting_data.html#import-matplotlib"><i class="fa fa-check"></i><b>3.1</b> Import Matplotlib</a></li>
<li class="chapter" data-level="3.2" data-path="03_plotting_data.html"><a href="03_plotting_data.html#bar-chart"><i class="fa fa-check"></i><b>3.2</b> Bar Chart</a></li>
<li class="chapter" data-level="3.3" data-path="03_plotting_data.html"><a href="03_plotting_data.html#scatter-plot"><i class="fa fa-check"></i><b>3.3</b> Scatter plot</a></li>
<li class="chapter" data-level="3.4" data-path="03_plotting_data.html"><a href="03_plotting_data.html#recap"><i class="fa fa-check"></i><b>3.4</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04_further_reading.html"><a href="04_further_reading.html"><i class="fa fa-check"></i><b>4</b> Further Reading and Resources</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04_further_reading.html"><a href="04_further_reading.html#websites-of-the-various-packages-used-in-the-workshop"><i class="fa fa-check"></i><b>4.1</b> Websites of the various packages used in the workshop</a></li>
<li class="chapter" data-level="4.2" data-path="04_further_reading.html"><a href="04_further_reading.html#ebooks"><i class="fa fa-check"></i><b>4.2</b> eBooks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Python for Bioinformatics - Supplementary Materials</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pandas" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Other types of data<a href="02_other_data.html#pandas" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/pandas_icon.jpg" style="width:200px; border-radius:5px" />
</center>
<p>Often scientists store data in delimited text files. These are typically
separated by commas or tabs. We can <strong>read</strong> these into Python, <strong>loop</strong>
through them and manipulate their values. We will do this using
similar/identical approaches that we used with our sequence data
previously.</p>
<p>We are going to load a new example dataset; a table of comma delimited
data. We will then use the Python module
<a href="https://pandas.pydata.org/">pandas</a> to filter and manipulate the data.</p>
<p>Create a new notebook called <strong>"10-Pandas"</strong> for this chapter.</p>
<div id="background-info-on-the-example-data" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Background info on the example data<a href="02_other_data.html#background-info-on-the-example-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/microbiome_icon.webp" style="width:200px; border-radius:5px" />
</center>
<p>The example data we are working with is a subset of the
<a href="https://microbiome.github.io/tutorials/Atlas.html">Atlas1006</a>
dataset, which itself is example data provided with the R package
<a href="https://microbiome.github.io/tutorials/">microbiome</a>.</p>
<p>It contains information on 1006 westerners with no reported health
complications. The westerners had their gut microbiome profiled and several
other characteristics recorded.</p>
<p>The data we will be looking at contains the following variables:</p>
<ul>
<li>SampleID</li>
<li>Age</li>
<li>Gender</li>
<li>Nationality</li>
<li>Diversity of gut microbiome</li>
<li>A BMI category</li>
</ul>
</div>
<div id="read-a-.csv-file-with-pandas" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Read a .csv file with pandas<a href="02_other_data.html#read-a-.csv-file-with-pandas" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/pandas_icon.jpg" style="width:200px; border-radius: 5px" />
</center>
<p>The Python <strong>library</strong> <a href="https://pandas.pydata.org/">pandas</a> is
specifically designed for reading delimited text files. It pre-processes
text data files and makes them easier to process and use in analysis.</p>
<p><a href="https://pandas.pydata.org/">Pandas</a> allows us to access certain
elements within each row. For example, the gender of each patient.</p>
<p>With <a href="https://pandas.pydata.org/">pandas</a> let us <strong>read</strong> in the file
<strong>"patients_gut_data.csv"</strong>.</p>
<div id="import-module" class="section level3 unnumbered hasAnchor">
<h3>Import module<a href="02_other_data.html#import-module" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First we need to load the <a href="https://pandas.pydata.org/">pandas</a>
<strong>module</strong>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb1-1"><a href="02_other_data.html#cb1-1" tabindex="-1"></a><span class="co">#load pandas</span></span>
<span id="cb1-2"><a href="02_other_data.html#cb1-2" tabindex="-1"></a><span class="im">import</span> pandas</span></code></pre></div>
</div>
<div id="read-data" class="section level3 unnumbered hasAnchor">
<h3>Read data<a href="02_other_data.html#read-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next we will <strong>read</strong> in the data with the
<a href="https://pandas.pydata.org/">pandas</a> <strong>function</strong> <code>read_csv()</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb2-1"><a href="02_other_data.html#cb2-1" tabindex="-1"></a><span class="co">#Read in csv file</span></span>
<span id="cb2-2"><a href="02_other_data.html#cb2-2" tabindex="-1"></a>patients <span class="op">=</span> pandas.read_csv(<span class="st">&quot;patients_gut_data.csv&quot;</span>)</span></code></pre></div>
<p>We have read in a <strong><a href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html">DataFrame</a></strong>.
Look at the top of this <strong>DataFrame</strong> with the <code>.head()</code> <strong>method</strong>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb3-1"><a href="02_other_data.html#cb3-1" tabindex="-1"></a>patients.head()</span></code></pre></div>
<p>One very useful feature of <a href="https://pandas.pydata.org/">pandas</a> is that
the <strong>object</strong> created when <strong>reading</strong> in the file will not dissappear
when it is used. This means you can <strong>read</strong> in the file once and do
many tasks with the resulting <strong>object</strong>.</p>
</div>
<div id="data-types" class="section level3 unnumbered hasAnchor">
<h3>Data types<a href="02_other_data.html#data-types" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, we can take a look at how <a href="https://pandas.pydata.org/">pandas</a> has stored the data by looking at the <strong>".dtypes"</strong> (data types).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb4-1"><a href="02_other_data.html#cb4-1" tabindex="-1"></a><span class="bu">print</span>(patients.dtypes)</span></code></pre></div>
<p>Pandas has automatically used the column headers as the variable names,
and assigned each variable a type based on the data:</p>
<ul>
<li><strong>int64</strong>: An integer.</li>
<li><strong>float64</strong>: A floating point number.</li>
<li><strong>object</strong>: A catch all term for string types</li>
</ul>
</div>
<div id="columns-and-index" class="section level3 unnumbered hasAnchor">
<h3>Columns and index<a href="02_other_data.html#columns-and-index" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can access a single column of data by it's variable name assigned by
<a href="https://pandas.pydata.org/">pandas</a>. For example, to access the
'diversity' column:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb5-1"><a href="02_other_data.html#cb5-1" tabindex="-1"></a><span class="bu">print</span>(patients.diversity)</span></code></pre></div>
<p>To access individual records, we need to first tell
<a href="https://pandas.pydata.org/">pandas</a> which column contains the index, or
sample names.</p>
<p>We can set the <strong>index</strong> column when <strong>reading</strong> in the file with the
option <code>index_col</code>. In the case of the <strong>"patients_gut_data.csv"</strong> data
file, the identifiers are in the first column, which Python counts as
column zero.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb6-1"><a href="02_other_data.html#cb6-1" tabindex="-1"></a><span class="co">#Read in file and use 1st column as index</span></span>
<span id="cb6-2"><a href="02_other_data.html#cb6-2" tabindex="-1"></a>patients <span class="op">=</span> pandas.read_csv(<span class="st">&quot;patients_gut_data.csv&quot;</span>, index_col<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb6-3"><a href="02_other_data.html#cb6-3" tabindex="-1"></a><span class="co">#Print nationality information for Sample-21</span></span>
<span id="cb6-4"><a href="02_other_data.html#cb6-4" tabindex="-1"></a><span class="bu">print</span>(patients.nationality[<span class="st">&quot;Sample-21&quot;</span>])</span></code></pre></div>
</div>
</div>
<div id="make-more-complex-selections" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Make more complex selections<a href="02_other_data.html#make-more-complex-selections" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/filter_icon.png" style="width:200px; border-radius:5px" />
</center>
<p>Pandas is very powerful for selecting columns and records from the data
by any conditions we want.</p>
<div id="select-columns-by-name" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Select columns by name<a href="02_other_data.html#select-columns-by-name" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let's select the data relating to the age, gut microbiome diversity, and
bmi_group and save it as a separate variable.</p>
<p><b><em>NOTE</em> the double square brackets containing the variable names: [[]]</b>. This is used because:</p>
<ul>
<li>We are choosing columns from the <strong>patients data frame</strong> by indexing, carried out
by the outer square brackets (<b>[</b>[]<b>]</b>).</li>
<li>We're using a <strong>list</strong> of the column names to select columns to
index. This is carried out by the inner square brackets ([<b>[]</b>]).</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb7-1"><a href="02_other_data.html#cb7-1" tabindex="-1"></a><span class="co">#Subset patients info</span></span>
<span id="cb7-2"><a href="02_other_data.html#cb7-2" tabindex="-1"></a>age_div_bmi <span class="op">=</span> patients[[<span class="st">&quot;age&quot;</span>, <span class="st">&quot;diversity&quot;</span>, <span class="st">&quot;bmi_group&quot;</span>]]</span>
<span id="cb7-3"><a href="02_other_data.html#cb7-3" tabindex="-1"></a><span class="co">#Print</span></span>
<span id="cb7-4"><a href="02_other_data.html#cb7-4" tabindex="-1"></a><span class="bu">print</span>(age_div_bmi)</span>
<span id="cb7-5"><a href="02_other_data.html#cb7-5" tabindex="-1"></a><span class="bu">print</span>(<span class="bu">type</span>(age_div_bmi))</span></code></pre></div>
<p>The below code carries out the same task but demonstrates the usage of the inner square brackets.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb8-1"><a href="02_other_data.html#cb8-1" tabindex="-1"></a><span class="co">#List of column names to extract</span></span>
<span id="cb8-2"><a href="02_other_data.html#cb8-2" tabindex="-1"></a>column_names <span class="op">=</span> [<span class="st">&quot;age&quot;</span>, <span class="st">&quot;diversity&quot;</span>, <span class="st">&quot;bmi_group&quot;</span>]</span>
<span id="cb8-3"><a href="02_other_data.html#cb8-3" tabindex="-1"></a><span class="co">#Subset patients info</span></span>
<span id="cb8-4"><a href="02_other_data.html#cb8-4" tabindex="-1"></a>age_div_bmi <span class="op">=</span> patients[column_names]</span>
<span id="cb8-5"><a href="02_other_data.html#cb8-5" tabindex="-1"></a><span class="co">#Print</span></span>
<span id="cb8-6"><a href="02_other_data.html#cb8-6" tabindex="-1"></a><span class="bu">print</span>(age_div_bmi)</span>
<span id="cb8-7"><a href="02_other_data.html#cb8-7" tabindex="-1"></a><span class="bu">print</span>(<span class="bu">type</span>(age_div_bmi))</span></code></pre></div>
<p>We can see our new dataframe contains just the columns we selected.
By checking it with the <code>type()</code> function, we know this new dataframe is in the <a href="https://pandas.pydata.org/">pandas</a> format.</p>
</div>
<div id="select-records-by-their-properties" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Select records by their properties<a href="02_other_data.html#select-records-by-their-properties" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/venn_icon.jpg" style="width:200px; border-radius:5px" />
</center>
<p>Let's choose records which fit certain criteria.</p>
<p>First we'll choose records from the patients data in which the patient
is aged 25 years or more <strong>AND</strong> are overweight. The two statements are
linked together by the <strong>AND</strong> operator (<code>&amp;</code>).</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb9-1"><a href="02_other_data.html#cb9-1" tabindex="-1"></a><span class="co">#Select 25 or older and overweight</span></span>
<span id="cb9-2"><a href="02_other_data.html#cb9-2" tabindex="-1"></a>overweight_over25 <span class="op">=</span> patients[(patients[<span class="st">&quot;age&quot;</span>] <span class="op">&gt;=</span> <span class="dv">25</span>) <span class="op">&amp;</span> (patients[<span class="st">&quot;bmi_group&quot;</span>] <span class="op">==</span> <span class="st">&quot;overweight&quot;</span>)]</span>
<span id="cb9-3"><a href="02_other_data.html#cb9-3" tabindex="-1"></a><span class="co">#print</span></span>
<span id="cb9-4"><a href="02_other_data.html#cb9-4" tabindex="-1"></a><span class="bu">print</span>(overweight_over25)</span></code></pre></div>
<p>We can also do something similar but using the <em>OR</em> operator (<code>|</code>).</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb10-1"><a href="02_other_data.html#cb10-1" tabindex="-1"></a><span class="co">#Select entries who are overweight or severeobese</span></span>
<span id="cb10-2"><a href="02_other_data.html#cb10-2" tabindex="-1"></a>obese_or_severeobese <span class="op">=</span> patients[(patients[<span class="st">&quot;bmi_group&quot;</span>] <span class="op">==</span> <span class="st">&quot;obese&quot;</span>) <span class="op">|</span> (patients[<span class="st">&quot;bmi_group&quot;</span>] <span class="op">==</span> <span class="st">&quot;severeobese&quot;</span>)]</span>
<span id="cb10-3"><a href="02_other_data.html#cb10-3" tabindex="-1"></a><span class="co">#print</span></span>
<span id="cb10-4"><a href="02_other_data.html#cb10-4" tabindex="-1"></a><span class="bu">print</span>(obese_or_severeobese)</span></code></pre></div>
</div>
</div>
<div id="applying-functions-and-methods-to-pandas-objects" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Applying functions and methods to pandas objects<a href="02_other_data.html#applying-functions-and-methods-to-pandas-objects" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/function.jpg" style="width:200px; background: white; border-radius:5px; border: white solid 5px" />
</center>
<p><a href="https://pandas.pydata.org/">Pandas</a> provides useful methods for working
with data tables, as we have seen. All of the previous approaches that
we used in the main book still apply to data which has been
processed using <a href="https://pandas.pydata.org/">pandas</a>.</p>
<p>For example, we can use pandas to select the age column, and convert the
number of years to months:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb11-1"><a href="02_other_data.html#cb11-1" tabindex="-1"></a><span class="bu">print</span>(patients.age<span class="op">*</span><span class="dv">12</span>)</span></code></pre></div>
<p>We can also pass pandas data to functions, as before.</p>
<p>Or, we could define a new function and use that with pandas data.</p>
<p>The below code calculates the average diversity.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode python Rchunk"><code class="sourceCode python"><span id="cb12-1"><a href="02_other_data.html#cb12-1" tabindex="-1"></a><span class="co">#Set variable for total diversity</span></span>
<span id="cb12-2"><a href="02_other_data.html#cb12-2" tabindex="-1"></a>div_total <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb12-3"><a href="02_other_data.html#cb12-3" tabindex="-1"></a><span class="co">#Loop through diversity values in patients</span></span>
<span id="cb12-4"><a href="02_other_data.html#cb12-4" tabindex="-1"></a><span class="cf">for</span> div <span class="kw">in</span> patients.diversity:</span>
<span id="cb12-5"><a href="02_other_data.html#cb12-5" tabindex="-1"></a>  <span class="co">#Add value to div_total</span></span>
<span id="cb12-6"><a href="02_other_data.html#cb12-6" tabindex="-1"></a>  div_total <span class="op">=</span> div_total <span class="op">+</span> div</span>
<span id="cb12-7"><a href="02_other_data.html#cb12-7" tabindex="-1"></a><span class="co">#Print average diversity</span></span>
<span id="cb12-8"><a href="02_other_data.html#cb12-8" tabindex="-1"></a><span class="bu">print</span>(div_total <span class="op">/</span> <span class="bu">len</span>(patients.diversity))</span></code></pre></div>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="01-Intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="03_plotting_data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

</body>

</html>
