{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "08397e6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Cell 1\n",
    "#Load package/library\n",
    "library(\"qiime2R\")\n",
    "\n",
    "#Import data\n",
    "#Paths set to \"data/XXX\" as there is a subdirectory with the data in the same directory as this notebook\n",
    "pseq <- qiime2R::qza_to_phyloseq(\n",
    "    features = \"data/table-dada2.qza\",\n",
    "    tree = \"data/rooted-tree.qza\",\n",
    "    taxonomy = \"data/taxonomy.sklearn.qza\",\n",
    "    metadata = \"data/media_metadata.txt\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "43dcf290",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "\n",
      "Loading required package: ggplot2\n",
      "\n",
      "\n",
      "microbiome R package (microbiome.github.com)\n",
      "    \n",
      "\n",
      "\n",
      " Copyright (C) 2011-2022 Leo Lahti, \n",
      "    Sudarshan Shetty et al. <microbiome.github.io>\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘microbiome’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:ggplot2’:\n",
      "\n",
      "    alpha\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    transform\n",
      "\n",
      "\n",
      "Compositional = NO2\n",
      "\n",
      "1] Min. number of reads = 104332] Max. number of reads = 170053] Total number of reads = 4948314] Average number of reads = 13745.30555555565] Median number of reads = 14050.57] Sparsity = 0.9389890674680956] Any OTU sum to 1 or less? NO8] Number of singletons = 09] Percent of OTUs that are singletons \n",
      "        (i.e. exactly one read detected across all samples)010] Number of sample variables are: 5sample.namesitemediareplicatesite.media2\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<ol>\n",
       "\t<li>'1] Min. number of reads = 10433'</li>\n",
       "\t<li>'2] Max. number of reads = 17005'</li>\n",
       "\t<li>'3] Total number of reads = 494831'</li>\n",
       "\t<li>'4] Average number of reads = 13745.3055555556'</li>\n",
       "\t<li>'5] Median number of reads = 14050.5'</li>\n",
       "\t<li>'7] Sparsity = 0.938989067468095'</li>\n",
       "\t<li>'6] Any OTU sum to 1 or less? NO'</li>\n",
       "\t<li>'8] Number of singletons = 0'</li>\n",
       "\t<li><span style=white-space:pre-wrap>'9] Percent of OTUs that are singletons \\n        (i.e. exactly one read detected across all samples)0'</span></li>\n",
       "\t<li>'10] Number of sample variables are: 5'</li>\n",
       "\t<li><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'sample.name'</li><li>'site'</li><li>'media'</li><li>'replicate'</li><li>'site.media'</li></ol>\n",
       "</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate}\n",
       "\\item '1{]} Min. number of reads = 10433'\n",
       "\\item '2{]} Max. number of reads = 17005'\n",
       "\\item '3{]} Total number of reads = 494831'\n",
       "\\item '4{]} Average number of reads = 13745.3055555556'\n",
       "\\item '5{]} Median number of reads = 14050.5'\n",
       "\\item '7{]} Sparsity = 0.938989067468095'\n",
       "\\item '6{]} Any OTU sum to 1 or less? NO'\n",
       "\\item '8{]} Number of singletons = 0'\n",
       "\\item '9{]} Percent of OTUs that are singletons \\textbackslash{}n        (i.e. exactly one read detected across all samples)0'\n",
       "\\item '10{]} Number of sample variables are: 5'\n",
       "\\item \\begin{enumerate*}\n",
       "\\item 'sample.name'\n",
       "\\item 'site'\n",
       "\\item 'media'\n",
       "\\item 'replicate'\n",
       "\\item 'site.media'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\end{enumerate}\n"
      ],
      "text/markdown": [
       "1. '1] Min. number of reads = 10433'\n",
       "2. '2] Max. number of reads = 17005'\n",
       "3. '3] Total number of reads = 494831'\n",
       "4. '4] Average number of reads = 13745.3055555556'\n",
       "5. '5] Median number of reads = 14050.5'\n",
       "6. '7] Sparsity = 0.938989067468095'\n",
       "7. '6] Any OTU sum to 1 or less? NO'\n",
       "8. '8] Number of singletons = 0'\n",
       "9. <span style=white-space:pre-wrap>'9] Percent of OTUs that are singletons \\n        (i.e. exactly one read detected across all samples)0'</span>\n",
       "10. '10] Number of sample variables are: 5'\n",
       "11. 1. 'sample.name'\n",
       "2. 'site'\n",
       "3. 'media'\n",
       "4. 'replicate'\n",
       "5. 'site.media'\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[[1]]\n",
       "[1] \"1] Min. number of reads = 10433\"\n",
       "\n",
       "[[2]]\n",
       "[1] \"2] Max. number of reads = 17005\"\n",
       "\n",
       "[[3]]\n",
       "[1] \"3] Total number of reads = 494831\"\n",
       "\n",
       "[[4]]\n",
       "[1] \"4] Average number of reads = 13745.3055555556\"\n",
       "\n",
       "[[5]]\n",
       "[1] \"5] Median number of reads = 14050.5\"\n",
       "\n",
       "[[6]]\n",
       "[1] \"7] Sparsity = 0.938989067468095\"\n",
       "\n",
       "[[7]]\n",
       "[1] \"6] Any OTU sum to 1 or less? NO\"\n",
       "\n",
       "[[8]]\n",
       "[1] \"8] Number of singletons = 0\"\n",
       "\n",
       "[[9]]\n",
       "[1] \"9] Percent of OTUs that are singletons \\n        (i.e. exactly one read detected across all samples)0\"\n",
       "\n",
       "[[10]]\n",
       "[1] \"10] Number of sample variables are: 5\"\n",
       "\n",
       "[[11]]\n",
       "[1] \"sample.name\" \"site\"        \"media\"       \"replicate\"   \"site.media\" \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Cell 2\n",
    "#Load microbiome library\n",
    "library(\"microbiome\")\n",
    "#Summary of phyloseq object\n",
    "microbiome::summarize_phyloseq(pseq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "11c9e3ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Cell 3\n",
    "#Save phyloseq as file\n",
    "save(pseq, file = \"phyloseq.RData\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
