<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 QC Tutorials | Population Genomics</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 QC Tutorials | Population Genomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 QC Tutorials | Population Genomics" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Helen Hipperson, Katy Maher, Victor Soria-Carrasco, Graeme Fox" />


<meta name="date" content="2023-06-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="03-Cluster_Introduction.html"/>
<link rel="next" href="05-Mapping.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Population Genomics</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-whole-genome-resequencing-of-killer-whales"><i class="fa fa-check"></i><b>2.1</b> Example data: Whole-genome resequencing of killer whales</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="part"><span><b>Generating SNP datasets</b></span></li>
<li class="chapter" data-level="4" data-path="04-QC.html"><a href="04-QC.html"><i class="fa fa-check"></i><b>4</b> QC Tutorials</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC.html"><a href="04-QC.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-QC.html"><a href="04-QC.html#killer-whale-genomic-dataset"><i class="fa fa-check"></i><b>4.2</b> Killer whale genomic dataset</a></li>
<li class="chapter" data-level="4.3" data-path="04-QC.html"><a href="04-QC.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-QC.html"><a href="04-QC.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality control</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="04-QC.html"><a href="04-QC.html#post-quality-control-check"><i class="fa fa-check"></i><b>4.4.1</b> Post Quality control check</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Mapping.html"><a href="05-Mapping.html"><i class="fa fa-check"></i><b>5</b> Mapping sequences to the reference genome using BWA</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Mapping.html"><a href="05-Mapping.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>5.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="5.2" data-path="05-Mapping.html"><a href="05-Mapping.html#read-mapping"><i class="fa fa-check"></i><b>5.2</b> Read mapping</a></li>
<li class="chapter" data-level="5.3" data-path="05-Mapping.html"><a href="05-Mapping.html#assessing-mapping-results"><i class="fa fa-check"></i><b>5.3</b> Assessing mapping results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-SNPs.html"><a href="06-SNPs.html"><i class="fa fa-check"></i><b>6</b> SNP and genotype calling using SAMtools and BCFtools</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-SNPs.html"><a href="06-SNPs.html#file-preparation"><i class="fa fa-check"></i><b>6.1</b> File preparation</a></li>
<li class="chapter" data-level="6.2" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-calling"><i class="fa fa-check"></i><b>6.2</b> Variant calling</a></li>
<li class="chapter" data-level="6.3" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-filtering"><i class="fa fa-check"></i><b>6.3</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-SNPs.html"><a href="06-SNPs.html#extracting-information"><i class="fa fa-check"></i><b>6.3.1</b> Extracting information</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-SNPs.html"><a href="06-SNPs.html#subsetting-files"><i class="fa fa-check"></i><b>6.3.2</b> Subsetting files</a></li>
<li class="chapter" data-level="6.3.3" data-path="06-SNPs.html"><a href="06-SNPs.html#filtering"><i class="fa fa-check"></i><b>6.3.3</b> Filtering</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="7" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html"><i class="fa fa-check"></i><b>7</b> Population structure with NGSadmix and PCA</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#ngsadmix"><i class="fa fa-check"></i><b>7.1</b> NGSadmix</a></li>
<li class="chapter" data-level="7.2" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#pca"><i class="fa fa-check"></i><b>7.2</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html"><i class="fa fa-check"></i><b>8</b> Calculating population summary statistics using PopGenome</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html#getting-started"><i class="fa fa-check"></i><b>8.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html"><i class="fa fa-check"></i><b>9</b> Delimitation of contiguous regions of differentiation using Hidden Markov Models (HMM)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#initial-set-up"><i class="fa fa-check"></i><b>9.1</b> Initial set up</a></li>
<li class="chapter" data-level="9.2" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#data-formatting"><i class="fa fa-check"></i><b>9.2</b> Data formatting</a></li>
<li class="chapter" data-level="9.3" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#allele-frequency-estimation"><i class="fa fa-check"></i><b>9.3</b> Allele frequency estimation</a></li>
<li class="chapter" data-level="9.4" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#fst-estimation"><i class="fa fa-check"></i><b>9.4</b> F<sub>ST</sub> estimation</a></li>
<li class="chapter" data-level="9.5" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#delimitation-of-contiguous-regions-of-differentiation-using-a-hmm-model"><i class="fa fa-check"></i><b>9.5</b> Delimitation of contiguous regions of differentiation using a HMM model</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-GWAS.html"><a href="10-GWAS.html"><i class="fa fa-check"></i><b>10</b> Genetic architecture of traits using multi-locus Genome Wide Association (GWA) mapping with GEMMA</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-GWAS.html"><a href="10-GWAS.html#initial-set-up-1"><i class="fa fa-check"></i><b>10.1</b> Initial set up</a></li>
<li class="chapter" data-level="10.2" data-path="10-GWAS.html"><a href="10-GWAS.html#data-formatting-1"><i class="fa fa-check"></i><b>10.2</b> Data formatting</a></li>
<li class="chapter" data-level="10.3" data-path="10-GWAS.html"><a href="10-GWAS.html#running-gemma-to-fit-a-bayesian-sparse-linear-mixed-model-bslmm"><i class="fa fa-check"></i><b>10.3</b> Running GEMMA to fit a Bayesian sparse linear mixed model (BSLMM)</a></li>
<li class="chapter" data-level="10.4" data-path="10-GWAS.html"><a href="10-GWAS.html#analysing-gemma-bslmm-output"><i class="fa fa-check"></i><b>10.4</b> Analysing GEMMA BSLMM output</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="11-Appendix.html"><a href="11-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="11-Appendix.html"><a href="11-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
<li class="chapter" data-level="A.2" data-path="11-Appendix.html"><a href="11-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>A.2</b> Jupyter-notebook</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Population Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sequencedata" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> QC Tutorials<a href="04-QC.html#sequencedata" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/mangify_glass.png" width="20%" style="display: block; margin: auto;" />
This tutorial will give hands on experience with quality control of genomic Illumina data. We will first look at the quality of the data</p>
<div id="workshop-data" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Workshop data<a href="04-QC.html#workshop-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before we can carry on with the workshop ensure you are in your working directory.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="04-QC.html#cb4-1" tabindex="-1"></a><span class="bu">cd</span> ~/popgenomics</span></code></pre></div>
<p>Before running any examples let us change directory to the raw fastq data directory.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="04-QC.html#cb5-1" tabindex="-1"></a><span class="bu">cd</span> data/fastq/</span></code></pre></div>
</div>
<div id="killer-whale-genomic-dataset" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Killer whale genomic dataset<a href="04-QC.html#killer-whale-genomic-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/orca.png" width="20%" style="display: block; margin: auto;" /></p>
<p>This dataset contains an example fastq file from one individual (sample name SRR8861574) of a re-sequencing project on killer whales.</p>
<p>The directory contains the forward (1) and reverse (2) read fastq files, and has been downsized to allow processing in a short amount of time for the workshop.</p>
<p>List the contents of the current directory to confirm the two files are there.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="04-QC.html#cb6-1" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>Have a look at the structure of the input fastq file:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="04-QC.html#cb7-1" tabindex="-1"></a><span class="fu">zcat</span> SRR8861574_1_small.fastq.gz <span class="kw">|</span> <span class="fu">head</span> </span></code></pre></div>
<p>These fastq files are compressed (with the .gz file extension). <code>zcat</code> is a command for viewing the contents of compressed files, the pipe <code>|</code> symbol then passes this to <code>head</code> to display only the first few lines of the file.</p>
<p>Fastq files contain a header line, the nucleotide sequence, and its corresponding quality scores.</p>
</div>
<div id="quality-assessment" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Quality assessment<a href="04-QC.html#quality-assessment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We’ll run the raw sequence data through FastQC to summarise the data quality.</p>
<p>First make a directory for the output.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="04-QC.html#cb8-1" tabindex="-1"></a><span class="fu">mkdir</span> raw_fastqc_output</span></code></pre></div>
<p><img src="figures/magnify_glass_27AUG2020.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Now we can run <strong>fastQC</strong>.</p>
<p><strong>Note</strong>:</p>
<ul>
<li>The below command can be run over one line excluding the <code>\</code></li>
<li>Alternatively, you can type <code>\</code> during a command and then press the enter key. The next line on the command line will start with <code>&gt;</code>. This will allow you to continue typing the command on the line. This can be used to type one command over multiple lines.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="04-QC.html#cb9-1" tabindex="-1"></a><span class="ex">fastqc</span> <span class="at">-o</span> raw_fastqc_output <span class="dt">\</span></span>
<span id="cb9-2"><a href="04-QC.html#cb9-2" tabindex="-1"></a>SRR8861574_1_small.fastq.gz SRR8861574_2_small.fastq.gz </span></code></pre></div>
<p>Normally when you run a command it will run the process in the foreground. This means your command line will be locked whilst the process is running. You will only be able to type a new command in the terminal once the foreground process is finished.</p>
<p>This is normally wanted but not always, for example when you run a process like firefox. To run a process in the background, so you can type and run new commands, you can add the symbol <code>&amp;</code> to the end of a command. We will use this for running firefox.</p>
<p>Please see the following link for more info on the foreground and background: <a href="https://www.lifewire.com/multitasking-background-foreground-process-2180219" class="uri">https://www.lifewire.com/multitasking-background-foreground-process-2180219</a></p>
<p>Using firefox have a look at the output html reports and answer the following questions.</p>
<p>To look at R1 fastQC output</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="04-QC.html#cb10-1" tabindex="-1"></a><span class="ex">firefox</span> raw_fastqc_output/SRR8861574_1_small_fastqc.html <span class="kw">&amp;</span></span></code></pre></div>
<p>You may see a warning message on your terminal, but you can press enter to continue to type in the terminal.</p>
<p>To look at R2 fastQC output</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="04-QC.html#cb11-1" tabindex="-1"></a><span class="ex">firefox</span> raw_fastqc_output/SRR8861574_2_small_fastqc.html <span class="kw">&amp;</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>How many total reads are there in both the files?</li>
<li>What is the length of the reads.</li>
<li>In the read 2 file, at what base position does the quality of all the reads go below 28 (I.e. the position where not all of the box plot is in the green)?</li>
<li>Are there any sequences with adapter content or overrepresented sequences?</li>
</ol>
<p>In this case the reads seem to be very high quality. This is likely as these reads are relatively short for an Illumina sequencing run. Longer sequences can be of poorer quality and may contain adapter sequence, especially towards the ends of the reads.</p>
<p>Often data from a sequencing machine will unfortunately not look so nice and will require quality trimming and filtering.</p>
<p>For most datasets:</p>
<ul>
<li>The quality decreases towards the end of the reads</li>
<li>The R2 reads have poorer quality than the R1 reads</li>
<li>The read sizes have a range compared to all being one size, however most of the reads are towards the long end of the range.</li>
</ul>
<p>Generally even if data does look very nice we would carry out quality control to get rid of any poor data that is masked by the very good data and to remove any adapter sequences.</p>
</div>
<div id="quality-control" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Quality control<a href="04-QC.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/quality_trimming_and_filtering.png" width="30%" style="display: block; margin: auto;" /></p>
<p>Quality control generally comes in two forms:</p>
<ol style="list-style-type: decimal">
<li><strong>Trimming</strong>: This is directly cutting off bits of sequence. This is typical in the form of trimming off low quality bases from the end of reads and trimming off adapters at the start of reads.</li>
<li><strong>Filtering</strong>: This occurs when entire reads are removed. A typical occurrence of this is when a read is too small as we do not want reads below a certain length.</li>
</ol>
<p>To carry this out we are going to use Trimmomatic.
<a href="http://www.usadellab.org/cms/index.php?page=trimmomatic" class="uri">http://www.usadellab.org/cms/index.php?page=trimmomatic</a></p>
<p>To run Trimmomatic with the reads we will use the below command.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="04-QC.html#cb12-1" tabindex="-1"></a><span class="ex">trimmomatic</span> <span class="dt">\</span></span>
<span id="cb12-2"><a href="04-QC.html#cb12-2" tabindex="-1"></a>  PE <span class="at">-phred33</span> <span class="dt">\</span></span>
<span id="cb12-3"><a href="04-QC.html#cb12-3" tabindex="-1"></a>  SRR8861574_1_small.fastq.gz SRR8861574_2_small.fastq.gz <span class="dt">\</span></span>
<span id="cb12-4"><a href="04-QC.html#cb12-4" tabindex="-1"></a>  SRR8861574_1_out_paired.fastq SRR8861574_1_out_unpaired.fastq <span class="dt">\</span></span>
<span id="cb12-5"><a href="04-QC.html#cb12-5" tabindex="-1"></a>  SRR8861574_2_out_paired.fastq SRR8861574_2_out_unpaired.fastq<span class="dt">\</span></span>
<span id="cb12-6"><a href="04-QC.html#cb12-6" tabindex="-1"></a>  ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 <span class="dt">\</span></span>
<span id="cb12-7"><a href="04-QC.html#cb12-7" tabindex="-1"></a>  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:30 MINLEN:50</span></code></pre></div>
<p>The parameter meanings are:</p>
<ul>
<li><code>PE</code>: Input data consists of paired-end reads.</li>
<li><code>-phred33</code>: Type of quality Phred encoding. In this case it is phred33 as our fastq files use the sanger encoding like most Illumina data.</li>
<li>Next the raw input forward (1) and then reverse (2) read files for quality control are specified.</li>
<li>On the next line the output files for read 1 are specified, first for paired reads and second for unpaired reads. This comes about when one read from a pair is filtered out but the other one is not. We can normally ignore this file after the trimming. However, trimmomatic must have this value to run.</li>
<li>Next the output files for read 2 are specified, first paired then unpaired.</li>
<li><code>ILLUMINACLIP</code>: These settings are used to find and remove Illumina adapters. First a fasta file of known adapter sequences is given, followed by the number of mismatches allowed between the adapter and read sequence and then thresholds for how accurate the alignment is between the adapter and read sequence.</li>
<li><code>LEADING</code>: The minimum quality value required to keep a base at the start of the read.</li>
<li><code>TRAILING</code>: The minimum quality value required to keep a base at the end of the read.</li>
<li><code>SLIDINGWINDOW</code>: This specifies to scan the read quality over a 4bp window, cutting when the average quality drops below 30.</li>
<li><code>MINLEN</code>: This specifies the minimum length of a read to keep, any ones shorter than 50bp are discarded.</li>
</ul>
<p>Once the command is run it will give you some useful output to the screen.
Based on it, approximately what percentage of paired reads were kept?</p>
<div id="post-quality-control-check" class="section level3 hasAnchor" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> Post Quality control check<a href="04-QC.html#post-quality-control-check" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/magnify_glass_good.png" width="20%" style="display: block; margin: auto;" />
To see how successful the quality control has been we need to run fastQC on the trimmomatic output.</p>
<p>First create a new directory for the fastQC output</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="04-QC.html#cb13-1" tabindex="-1"></a><span class="fu">mkdir</span> trim_fastqc_output</span></code></pre></div>
<p>Now run fastQC on the two paired read files</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="04-QC.html#cb14-1" tabindex="-1"></a><span class="ex">fastqc</span> <span class="at">-o</span> trim_fastqc_output <span class="dt">\</span></span>
<span id="cb14-2"><a href="04-QC.html#cb14-2" tabindex="-1"></a>SRR8861574_1_out_paired.fastq SRR8861574_2_out_paired.fastq</span></code></pre></div>
<p>We can open the fastQC output with the following commands</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="04-QC.html#cb15-1" tabindex="-1"></a><span class="ex">firefox</span> <span class="dt">\</span></span>
<span id="cb15-2"><a href="04-QC.html#cb15-2" tabindex="-1"></a>trim_fastqc_output/SRR8861574_1_out_paired_fastqc.html <span class="kw">&amp;</span></span>
<span id="cb15-3"><a href="04-QC.html#cb15-3" tabindex="-1"></a><span class="ex">firefox</span> <span class="dt">\</span></span>
<span id="cb15-4"><a href="04-QC.html#cb15-4" tabindex="-1"></a>trim_fastqc_output/SRR8861574_2_out_paired_fastqc.html <span class="kw">&amp;</span></span></code></pre></div>
<p>To see how well the reads have improved let’s answer the below questions and compare to them to the answers of the raw read fastQC questions.</p>
<ol style="list-style-type: decimal">
<li>How many total reads are there in both the files?</li>
<li>What is the length of the reads?</li>
<li>In the read 2 file, at what base position range does the quality of all the reads go below 28 (I.e. the position where not all of the box plot is in the green)?</li>
<li>What is the average GC content of the reads?</li>
</ol>
<p>Some things to note:</p>
<ul>
<li>The amount of reads has significantly decreased due to quality control. This is expected, however this will need to be taken into account. If you do not have enough reads for the downstream analysis you will need to be less stringent on the trimming or filtering. What is the minimum number of reads required for downstream analysis? Unfortunately this depends on the specific analysis at hand. For genome assemblies generally the rule of thumb is you want at least 20 times as many bases in the reads as would be in the final assembly (based on the genome size of the organisms). For calling SNPs you need less coverage.</li>
<li>R2 quality is still slightly worse than R1 quality towards the end of the reads. Again this is normal. You could be more stringent but as previously mentioned you may be removing too many reads.</li>
</ul>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="03-Cluster_Introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="05-Mapping.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
