<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 SNP and genotype calling using SAMtools and BCFtools | Population Genomics</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 SNP and genotype calling using SAMtools and BCFtools | Population Genomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 SNP and genotype calling using SAMtools and BCFtools | Population Genomics" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Helen Hipperson, Katy Maher, Victor Soria-Carrasco, Graeme Fox, Gavin Gouws" />


<meta name="date" content="2024-05-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="05-Mapping.html"/>
<link rel="next" href="07-NGSAdmix.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li 
  class="toc-logo">
  <a 
    href="https://neof.org.uk/" 
    target="blank"><img src="figures/neof_bordered_small.png">
  </a>
</li>
<li><a>Population Genomics</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-whole-genome-resequencing-of-killer-whales"><i class="fa fa-check"></i><b>2.1</b> Example data: Whole-genome resequencing of killer whales</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="part"><span><b>Generating SNP datasets</b></span></li>
<li class="chapter" data-level="4" data-path="04-QC.html"><a href="04-QC.html"><i class="fa fa-check"></i><b>4</b> QC Tutorials</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC.html"><a href="04-QC.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-QC.html"><a href="04-QC.html#killer-whale-genomic-dataset"><i class="fa fa-check"></i><b>4.2</b> Killer whale genomic dataset</a></li>
<li class="chapter" data-level="4.3" data-path="04-QC.html"><a href="04-QC.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-QC.html"><a href="04-QC.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality control</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="04-QC.html"><a href="04-QC.html#post-quality-control-check"><i class="fa fa-check"></i><b>4.4.1</b> Post Quality control check</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Mapping.html"><a href="05-Mapping.html"><i class="fa fa-check"></i><b>5</b> Mapping sequences to the reference genome using BWA</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Mapping.html"><a href="05-Mapping.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>5.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="5.2" data-path="05-Mapping.html"><a href="05-Mapping.html#read-mapping"><i class="fa fa-check"></i><b>5.2</b> Read mapping</a></li>
<li class="chapter" data-level="5.3" data-path="05-Mapping.html"><a href="05-Mapping.html#assessing-mapping-results"><i class="fa fa-check"></i><b>5.3</b> Assessing mapping results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-SNPs.html"><a href="06-SNPs.html"><i class="fa fa-check"></i><b>6</b> SNP and genotype calling using SAMtools and BCFtools</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-SNPs.html"><a href="06-SNPs.html#file-preparation"><i class="fa fa-check"></i><b>6.1</b> File preparation</a></li>
<li class="chapter" data-level="6.2" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-calling"><i class="fa fa-check"></i><b>6.2</b> Variant calling</a></li>
<li class="chapter" data-level="6.3" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-filtering"><i class="fa fa-check"></i><b>6.3</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-SNPs.html"><a href="06-SNPs.html#extracting-information"><i class="fa fa-check"></i><b>6.3.1</b> Extracting information</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-SNPs.html"><a href="06-SNPs.html#subsetting-files"><i class="fa fa-check"></i><b>6.3.2</b> Subsetting files</a></li>
<li class="chapter" data-level="6.3.3" data-path="06-SNPs.html"><a href="06-SNPs.html#filtering"><i class="fa fa-check"></i><b>6.3.3</b> Filtering</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="7" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html"><i class="fa fa-check"></i><b>7</b> Population structure with NGSadmix and PCA</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#ngsadmix"><i class="fa fa-check"></i><b>7.1</b> NGSadmix</a></li>
<li class="chapter" data-level="7.2" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#jupyter"><i class="fa fa-check"></i><b>7.2</b> Jupyter</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>7.2.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="7.2.2" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#create-r-notebook"><i class="fa fa-check"></i><b>7.2.2</b> Create R notebook</a></li>
<li class="chapter" data-level="7.2.3" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#cells-and-code"><i class="fa fa-check"></i><b>7.2.3</b> Cells and code</a></li>
<li class="chapter" data-level="7.2.4" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#create-new-cells"><i class="fa fa-check"></i><b>7.2.4</b> Create new cells</a></li>
<li class="chapter" data-level="7.2.5" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#running-code"><i class="fa fa-check"></i><b>7.2.5</b> Running code</a></li>
<li class="chapter" data-level="7.2.6" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#saving-the-file"><i class="fa fa-check"></i><b>7.2.6</b> Saving the file</a></li>
<li class="chapter" data-level="7.2.7" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>7.2.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="7.2.8" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#close-the-notebook"><i class="fa fa-check"></i><b>7.2.8</b> Close the notebook</a></li>
<li class="chapter" data-level="7.2.9" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#video-tutorial"><i class="fa fa-check"></i><b>7.2.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#admixture-plots"><i class="fa fa-check"></i><b>7.3</b> Admixture plots</a></li>
<li class="chapter" data-level="7.4" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#pca"><i class="fa fa-check"></i><b>7.4</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html"><i class="fa fa-check"></i><b>8</b> Calculating population summary statistics using VCFtools</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html#getting-started"><i class="fa fa-check"></i><b>8.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html"><i class="fa fa-check"></i><b>9</b> Delimitation of contiguous regions of differentiation using Hidden Markov Models (HMM)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#initial-set-up"><i class="fa fa-check"></i><b>9.1</b> Initial set up</a></li>
<li class="chapter" data-level="9.2" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#data-formatting"><i class="fa fa-check"></i><b>9.2</b> Data formatting</a></li>
<li class="chapter" data-level="9.3" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#allele-frequency-estimation"><i class="fa fa-check"></i><b>9.3</b> Allele frequency estimation</a></li>
<li class="chapter" data-level="9.4" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#fst-estimation"><i class="fa fa-check"></i><b>9.4</b> F<sub>ST</sub> estimation</a></li>
<li class="chapter" data-level="9.5" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#delimitation-of-contiguous-regions-of-differentiation-using-a-hmm-model"><i class="fa fa-check"></i><b>9.5</b> Delimitation of contiguous regions of differentiation using a HMM model</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-GWAS.html"><a href="10-GWAS.html"><i class="fa fa-check"></i><b>10</b> Genetic architecture of traits using multi-locus Genome Wide Association (GWA) mapping with GEMMA</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-GWAS.html"><a href="10-GWAS.html#initial-set-up-1"><i class="fa fa-check"></i><b>10.1</b> Initial set up</a></li>
<li class="chapter" data-level="10.2" data-path="10-GWAS.html"><a href="10-GWAS.html#data-formatting-1"><i class="fa fa-check"></i><b>10.2</b> Data formatting</a></li>
<li class="chapter" data-level="10.3" data-path="10-GWAS.html"><a href="10-GWAS.html#running-gemma-to-fit-a-bayesian-sparse-linear-mixed-model-bslmm"><i class="fa fa-check"></i><b>10.3</b> Running GEMMA to fit a Bayesian sparse linear mixed model (BSLMM)</a></li>
<li class="chapter" data-level="10.4" data-path="10-GWAS.html"><a href="10-GWAS.html#analysing-gemma-bslmm-output"><i class="fa fa-check"></i><b>10.4</b> Analysing GEMMA BSLMM output</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="11-Appendix.html"><a href="11-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="11-Appendix.html"><a href="11-Appendix.html#mamba-installation-and-environment"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="11-Appendix.html"><a href="11-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>B</b> Jupyter-notebook</a></li>
<li class="chapter" data-level="C" data-path="11-Appendix.html"><a href="11-Appendix.html#loops"><i class="fa fa-check"></i><b>C</b> Using loops to process multiple samples</a>
<ul>
<li class="chapter" data-level="C.1" data-path="11-Appendix.html"><a href="11-Appendix.html#simple-loop-example"><i class="fa fa-check"></i><b>C.1</b> Simple loop example</a></li>
<li class="chapter" data-level="C.2" data-path="11-Appendix.html"><a href="11-Appendix.html#trimmomatic-loop-example"><i class="fa fa-check"></i><b>C.2</b> Trimmomatic loop example</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="11-Appendix.html"><a href="11-Appendix.html#an-overview-of-bayesian-statistics"><i class="fa fa-check"></i><b>D</b> An overview of Bayesian statistics</a></li>
<li class="chapter" data-level="E" data-path="11-Appendix.html"><a href="11-Appendix.html#additional-approaches-for-detecting-selection-and-for-gwas"><i class="fa fa-check"></i><b>E</b> Additional approaches for detecting selection and for GWAS</a>
<ul>
<li class="chapter" data-level="E.1" data-path="11-Appendix.html"><a href="11-Appendix.html#rehh"><i class="fa fa-check"></i><b>E.1</b> rehh</a></li>
<li class="chapter" data-level="E.2" data-path="11-Appendix.html"><a href="11-Appendix.html#gcta"><i class="fa fa-check"></i><b>E.2</b> GCTA</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Population Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="SNPcalling" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> SNP and genotype calling using SAMtools and BCFtools<a href="06-SNPs.html#SNPcalling" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/snp.png" style="width:300px" />
</center>
<p>BCFtools is a very popular program to call SNPs and genotypes (and also to manipulate and filter vcf/bcf files, as we will see afterwards). SNP calling is a relatively intensive process, so to speed things up we will be restricting variant calling to one scaffold from the killer whale genome (NW_004438415.1). We will call SNPs for three samples (SRR8861574, SRR8861575 and SRR8861576). The alignments have been provided for you in '.bam' format - this is a binary, compressed version of a sam file.</p>
<p>First navigate to the 'aligned' directory:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb22-1"><a href="06-SNPs.html#cb22-1" tabindex="-1"></a><span class="bu">cd</span> ~/popgenomics/data/aligned/</span></code></pre></div>
<p>Check that you have bam files for the three samples and their corresponding index files using <code>ls</code>.</p>
<div id="file-preparation" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> File preparation<a href="06-SNPs.html#file-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/index%20-%20Copy.png" style="width:150px" />
</center>
<p>Before we start SNP calling, we will copy over the genome assembly file and index it using <code>samtools</code>:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb23-1"><a href="06-SNPs.html#cb23-1" tabindex="-1"></a><span class="fu">cp</span> /pub14/tea/nsc206/NEOF/popgenomics/genome/GCF_000331955.2_Oorc_1.1_genomic.fna .</span>
<span id="cb23-2"><a href="06-SNPs.html#cb23-2" tabindex="-1"></a></span>
<span id="cb23-3"><a href="06-SNPs.html#cb23-3" tabindex="-1"></a><span class="ex">samtools</span> faidx GCF_000331955.2_Oorc_1.1_genomic.fna</span></code></pre></div>
<p>This will produce a <code>.fai</code> index file.</p>
<p>Also, we need to make a list of the BAM files before we start:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb24-1"><a href="06-SNPs.html#cb24-1" tabindex="-1"></a><span class="fu">ls</span> <span class="pp">*</span>.bam <span class="op">&gt;</span> bamFiles.txt</span></code></pre></div>
<p>We will use this list file to call SNPs from all three samples rather than having to specify each sample name in the command.</p>
</div>
<div id="variant-calling" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Variant calling<a href="06-SNPs.html#variant-calling" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/calling.png" style="width:150px" />
</center>
<p>We will use BCFtools <code>mpileup</code> to produce genotype likelihoods for the three samples/BAM files, and pipe this output to BCFtools <code>call</code> to generate a file of genetic variants.</p>
<p>Here is the BCFtools command:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb25-1"><a href="06-SNPs.html#cb25-1" tabindex="-1"></a><span class="ex">bcftools</span> mpileup <span class="at">-Ou</span> <span class="dt">\</span></span>
<span id="cb25-2"><a href="06-SNPs.html#cb25-2" tabindex="-1"></a>--max-depth 10000 <span class="at">-q</span> 20 <span class="at">-Q</span> 20 <span class="dt">\</span></span>
<span id="cb25-3"><a href="06-SNPs.html#cb25-3" tabindex="-1"></a>-P ILLUMINA <span class="at">-a</span> FORMAT/DP,FORMAT/AD <span class="dt">\</span></span>
<span id="cb25-4"><a href="06-SNPs.html#cb25-4" tabindex="-1"></a>-f GCF_000331955.2_Oorc_1.1_genomic.fna <span class="dt">\</span></span>
<span id="cb25-5"><a href="06-SNPs.html#cb25-5" tabindex="-1"></a>-r NW_004438415.1 <span class="dt">\</span></span>
<span id="cb25-6"><a href="06-SNPs.html#cb25-6" tabindex="-1"></a>-b bamFiles.txt <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb25-7"><a href="06-SNPs.html#cb25-7" tabindex="-1"></a><span class="ex">bcftools</span> call <span class="at">-m</span> <span class="at">-v</span> <span class="at">-P</span> 1e-6 <span class="at">-f</span> GQ <span class="dt">\</span></span>
<span id="cb25-8"><a href="06-SNPs.html#cb25-8" tabindex="-1"></a>-O b <span class="at">-o</span> orca_3samp_1scaf.bcf</span></code></pre></div>
<p>The options we are using are:</p>
<p><code>mpileup</code></p>
<ul>
<li><p><code>-Ou</code>: ouput an uncompressed bam file. This is the option to use when piping the output to another command for optimum performance/speed.</p></li>
<li><p><code>--max-depth 10000</code>: the maximum number of sequences considered per position</p></li>
<li><p><code>-q 20</code>: filter out alignments with mapping quality &lt;20</p></li>
<li><p><code>-Q 20</code>: filter out bases with QS &lt; 20</p></li>
<li><p><code>-P ILLUMINA</code>: use Illumina platform for indels</p></li>
<li><p><code>-a FORMAT/DP,FORMAT/AD</code>: output depth and allelic depth</p></li>
<li><p><code>-f</code>: specify the genome reference file, which must be faidx-indexed</p></li>
<li><p><code>-r</code>: restrict mpileup output to this specific region</p></li>
<li><p><code>-b</code>: list of input bam alignment files</p>
<p><code>call</code></p></li>
<li><p><code>-m</code>: use the multiallelic caller</p></li>
<li><p><code>-v</code>: output variants only</p></li>
<li><p><code>-P 1e-6</code>: prior for expected substitution rate</p></li>
<li><p><code>-f GQ</code>: output genotype quality</p></li>
<li><p><code>-O b</code>: output in BCF format</p></li>
</ul>
<p>This command generates a '.bcf' file, which is a binary version of a '.vcf'.</p>
<p>One of the most popular text-file formats for storing genetic variation information is the Variant Call Format (VCF). There are different versions of the format, but the core elements are the same. You can find a full description of the latest iteration <a href="https://samtools.github.io/hts-specs/VCFv4.3.pdf">here</a>. Due to the large amount of data usually involved, files tend to be stored compressed as a BCF file. BCF is designed to be much more efficient to process and store large amounts of data. The practical downside is that the contents can only be accessed with <code>bcftools view</code>.</p>
<p>The VCF format is composed of meta-information lines (prefixed wih ##), a header line (prefixed with #), and data lines each containing information about a position in the genome and genotype information on samples for each position (text fields separated by tabs). There are 8 fixed field lines, usually followed by an additional FORMAT field and an arbitrary number of sample fields when genotypes are included.</p>
<p>Description of VCF fields</p>
<p><left>
<img src="figures/vcf.png" style="width:700px; border-radius:15px; border:5px solid #333333; background:null" />
</left></p>
<p>Some notes about genotype fields:</p>
<p>GT: Allele values are 0 for the reference and 1+ for alternate alleles. Alleles are separated with / if unphased and with | if phased. There are as many alleles separated by / or | as the ploidy. A . is used for missing data, i.e. when a genotype could not be called. More information on phasing is provided in the Appendix.</p>
<p>GL or PL: Genotype likelihoods as floating-point log10-scaled or phred-scaled format. A diploid biallelic site has three values: homozygote for reference, heterozygote, and homozygote for alternate. Triallelic sites have 6 values: homozygote for reference, heterozygote of reference and alternate allele 1, homozygote for alternate allele 1, heterozygote of reference and alternate allele 2, heterozygote of alternate alleles 1 and 2, and homozygote for alternate allele 2.</p>
<p>A graphical explanation of the different parts and fields can be found <a href="http://vcftools.sourceforge.net/VCF-poster.pdf">here</a>.</p>
<p>Let's have a look at our BCF file:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb26-1"><a href="06-SNPs.html#cb26-1" tabindex="-1"></a><span class="ex">bcftools</span> view orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p>The file starts with a few lines specifying the VCF and bcftools versions, the command used with <code>bcftools mpileup</code>, followed by a long list of the &gt;1,000 scaffolds that comprise the genome assembly (starting with ##contig), the descriptions of the fields for ALT, INFO and FORMAT, and any additional commands that were executed to produce this file (<code>bcftools call</code> in this case).</p>
<p>To focus only on the header or the SNPs, the flags <code>-h</code> and <code>-H</code> can be used:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb27-1"><a href="06-SNPs.html#cb27-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-h</span> orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span>
<span id="cb27-2"><a href="06-SNPs.html#cb27-2" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>Are all of the variants SNPs or are there also INDELs in the file?
<div id="radio_JMHZBVWMYN" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_JMHZBVWMYN" value=""></input> <span>Only SNPs</span></label><label><input type="radio" autocomplete="off" name="radio_JMHZBVWMYN" value=""></input> <span>A good mix of SNPs and indels</span></label><label><input type="radio" autocomplete="off" name="radio_JMHZBVWMYN" value="answer"></input> <span>A few indels</span></label>
</div></li>
<li>Are the variant quality scores (QUAL field) all high (&gt; 20)?
<div id="radio_BPWPTOHFNP" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BPWPTOHFNP" value=""></input> <span>Yes</span></label><label><input type="radio" autocomplete="off" name="radio_BPWPTOHFNP" value="answer"></input> <span>No</span></label>
</div></li>
</ol>
<p>Information in the BCF file can be used to filter to retain, for example, only biallelic SNPs, or to filter out variants with a low quality score or read depth.</p>
<p>A list of the samples contained in the file can be obtained using <code>bcftools query</code>, and can be counted with <code>wc</code>:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb28-1"><a href="06-SNPs.html#cb28-1" tabindex="-1"></a><span class="ex">bcftools</span> query <span class="at">-l</span> orca_3samp_1scaf.bcf</span>
<span id="cb28-2"><a href="06-SNPs.html#cb28-2" tabindex="-1"></a><span class="ex">bcftools</span> query <span class="at">-l</span> orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre></div>
<p>Lastly, let's index the bcf file in order to be able to process it below:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb29-1"><a href="06-SNPs.html#cb29-1" tabindex="-1"></a><span class="ex">bcftools</span> index orca_3samp_1scaf.bcf</span></code></pre></div>
</div>
<div id="variant-filtering" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Variant filtering<a href="06-SNPs.html#variant-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/filter.png" style="width:150px" />
</center>
<p>The number of variants can be counted using bcftools <code>view -H</code> and <code>wc</code>, but it includes all kinds of variants. To check particular ones, we need to use the flags <code>-v/-V</code> (or <code>--types/--exclude-types</code>) to include or exclude certain variant:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb30-1"><a href="06-SNPs.html#cb30-1" tabindex="-1"></a><span class="co"># all variants</span></span>
<span id="cb30-2"><a href="06-SNPs.html#cb30-2" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb30-3"><a href="06-SNPs.html#cb30-3" tabindex="-1"></a><span class="co"># only SNPs</span></span>
<span id="cb30-4"><a href="06-SNPs.html#cb30-4" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> <span class="at">-v</span> snps orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb30-5"><a href="06-SNPs.html#cb30-5" tabindex="-1"></a><span class="co"># only indels</span></span>
<span id="cb30-6"><a href="06-SNPs.html#cb30-6" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> <span class="at">-v</span> indels orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre></div>
<p>Biallelic SNPs (with just 1 alternate allele) can be extracted with the <code>-m</code> and <code>-M</code> flags:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb31-1"><a href="06-SNPs.html#cb31-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> <span class="at">-v</span> snps <span class="at">-m</span> 2 <span class="at">-M</span> 2 orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p><code>-m 2</code> is the bcftools command that means “-m, –min-alleles INT: print sites with at least INT alleles listed in REF and ALT columns”. So here we are asking for all lines where there at least 2 REF and ALT alleles in total.</p>
<p><code>-M 2</code> is the bcftools command that means “-M, –max-alleles INT: print sites with at most INT alleles listed in REF and ALT columns”. So here we are asking for all lines where there at most 2 REF and ALT alleles in total.</p>
<div id="extracting-information" class="section level3 hasAnchor" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Extracting information<a href="06-SNPs.html#extracting-information" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/exctractor.png" style="width:150px" />
</center>
<p>Partial information can be extracted using <code>bcftools query</code>. In the example above, we saw how to get the list of samples using the <code>-l</code> option, but it can also be used to extract any fields using the <code>-f</code> option. For example, you can simply extract the list of positions with:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb32-1"><a href="06-SNPs.html#cb32-1" tabindex="-1"></a><span class="ex">bcftools</span> query <span class="at">-f</span> <span class="st">&#39;%POS\n&#39;</span> orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p>Or combine multiple fields in the output, for example, scaffold/chromosome, position, reference and alternate alleles:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb33-1"><a href="06-SNPs.html#cb33-1" tabindex="-1"></a><span class="ex">bcftools</span> query <span class="at">-f</span> <span class="st">&#39;%CHROM\t%POS\t%REF\t%ALT\n&#39;</span> orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p>You can also extract genotypes (notice the position of the tab to avoid adding an extra tab at the end of the lines):</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb34-1"><a href="06-SNPs.html#cb34-1" tabindex="-1"></a><span class="ex">bcftools</span> query <span class="at">-f</span> <span class="st">&#39;%CHROM\t%POS\t%REF\t%ALT[\t%GT]\n&#39;</span> <span class="dt">\</span></span>
<span id="cb34-2"><a href="06-SNPs.html#cb34-2" tabindex="-1"></a>orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p>Or genotype likelihoods:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb35-1"><a href="06-SNPs.html#cb35-1" tabindex="-1"></a><span class="ex">bcftools</span> query <span class="at">-f</span> <span class="st">&#39;%CHROM\t%POS\t%REF\t%ALT[\t%PL]\n&#39;</span> <span class="dt">\</span></span>
<span id="cb35-2"><a href="06-SNPs.html#cb35-2" tabindex="-1"></a>orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p>You can even combine this with awk, Perl or other tools to do simple calculations. For example, you can calculate allele frequencies from the AC and AN counts:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb36-1"><a href="06-SNPs.html#cb36-1" tabindex="-1"></a><span class="ex">bcftools</span> query <span class="at">-f</span> <span class="st">&#39;%CHROM\t%POS\t%REF\t%ALT\t%AC\t%AN\n&#39;</span> <span class="dt">\</span></span>
<span id="cb36-2"><a href="06-SNPs.html#cb36-2" tabindex="-1"></a>orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span>
<span id="cb36-3"><a href="06-SNPs.html#cb36-3" tabindex="-1"></a></span>
<span id="cb36-4"><a href="06-SNPs.html#cb36-4" tabindex="-1"></a><span class="ex">bcftools</span> query <span class="at">-f</span> <span class="st">&#39;%CHROM\t%POS\t%REF\t%ALT\t%AC\t%AN\n&#39;</span> <span class="dt">\</span></span>
<span id="cb36-5"><a href="06-SNPs.html#cb36-5" tabindex="-1"></a>orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;{print $1,$2,$3,$4,$5,$6,$5/$6}&#39;</span> <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p>Or the mean genotype quality:</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb37-1"><a href="06-SNPs.html#cb37-1" tabindex="-1"></a><span class="co"># For all genotypes (uncalled ones = 0)</span></span>
<span id="cb37-2"><a href="06-SNPs.html#cb37-2" tabindex="-1"></a><span class="ex">bcftools</span> query <span class="at">-f</span> <span class="st">&#39;%CHROM\t%POS\t%REF\t%ALT[\t%GQ]\n&#39;</span> <span class="dt">\</span></span>
<span id="cb37-3"><a href="06-SNPs.html#cb37-3" tabindex="-1"></a>orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb37-4"><a href="06-SNPs.html#cb37-4" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{SUM=0; N=0; for(i=5; i&lt;=NF; i++){N+=1; SUM+=$i}; \</span></span>
<span id="cb37-5"><a href="06-SNPs.html#cb37-5" tabindex="-1"></a><span class="st">AVG=SUM/N; print $1,$2,$3,$4,AVG}&#39;</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb37-6"><a href="06-SNPs.html#cb37-6" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span></span>
<span id="cb37-7"><a href="06-SNPs.html#cb37-7" tabindex="-1"></a></span>
<span id="cb37-8"><a href="06-SNPs.html#cb37-8" tabindex="-1"></a><span class="co"># Only for called genotypes:</span></span>
<span id="cb37-9"><a href="06-SNPs.html#cb37-9" tabindex="-1"></a><span class="ex">bcftools</span> query <span class="at">-f</span> <span class="st">&#39;%CHROM\t%POS\t%REF\t%ALT[\t%GQ]\n&#39;</span> <span class="dt">\</span></span>
<span id="cb37-10"><a href="06-SNPs.html#cb37-10" tabindex="-1"></a>orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb37-11"><a href="06-SNPs.html#cb37-11" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{SUM=0; N=0; for(i=5; i&lt;=NF; i++){SUM+=$i; if ($i&gt;0) N+=1}; \</span></span>
<span id="cb37-12"><a href="06-SNPs.html#cb37-12" tabindex="-1"></a><span class="st">AVG=SUM/N; print $1,$2,$3,$4,AVG}&#39;</span> <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
</div>
<div id="subsetting-files" class="section level3 hasAnchor" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Subsetting files<a href="06-SNPs.html#subsetting-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/subsample.png" style="width:150px; border-radius:15px; border:5px solid #333333; background:white" />
</center>
<p>We can subset SNPs within particular regions. For example, we can extract all the SNPs within a region on a particular chromosome/scaffold:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb38-1"><a href="06-SNPs.html#cb38-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> orca_3samp_1scaf.bcf NW_004438415.1:50000-100000 <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb38-2"><a href="06-SNPs.html#cb38-2" tabindex="-1"></a></span>
<span id="cb38-3"><a href="06-SNPs.html#cb38-3" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> orca_3samp_1scaf.bcf NW_004438415.1:50000-100000 <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p>Subsetting a number of samples is also possible, for example to get only the first 2 samples:</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb39-1"><a href="06-SNPs.html#cb39-1" tabindex="-1"></a><span class="va">SAMPLES</span><span class="op">=</span><span class="va">$(</span><span class="ex">bcftools</span> query <span class="at">-l</span> orca_3samp_1scaf.bcf <span class="kw">|</span> <span class="fu">head</span> <span class="at">-n</span> 2 <span class="kw">|</span> <span class="fu">tr</span> <span class="st">&#39;\n/&#39;</span> <span class="st">&#39;,&#39;</span><span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/,$//&#39;</span><span class="va">)</span></span>
<span id="cb39-2"><a href="06-SNPs.html#cb39-2" tabindex="-1"></a></span>
<span id="cb39-3"><a href="06-SNPs.html#cb39-3" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-s</span> <span class="va">$SAMPLES</span> orca_3samp_1scaf.bcf <span class="at">-O</span> b <span class="at">-o</span> orca_3samp_1scaf_first2.bcf</span>
<span id="cb39-4"><a href="06-SNPs.html#cb39-4" tabindex="-1"></a></span>
<span id="cb39-5"><a href="06-SNPs.html#cb39-5" tabindex="-1"></a><span class="ex">bcftools</span> index orca_3samp_1scaf_first2.bcf</span></code></pre></div>
</div>
<div id="filtering" class="section level3 hasAnchor" number="6.3.3">
<h3><span class="header-section-number">6.3.3</span> Filtering<a href="06-SNPs.html#filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Usually, we want to apply some filters depending on the sort of downstream analyses that we intend to do. Filtering usually consists of establishing some sort of threshold for the SNP quality in the QUAL field or some of the variables encoded in the INFO field (e.g. depth, mapping quality, etc.), but more sophisticated filtering is possible based on genotype-specific (i.e. per-sample) variables (e.g. discarding genotypes below a certain quality or with too few reads). Although it is possible to do soft-filtering of SNPs and simply tag them as PASSED or FAILED in the VCF/BCF file, it is more useful in practice to do hard-filtering and simply remove all those SNPs (and genotypes) that did not meet our thresholds. We are going to focus on the SNPs only, so let’s extract only the SNPs from all the variants:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb40-1"><a href="06-SNPs.html#cb40-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-v</span> snps orca_3samp_1scaf.bcf <span class="at">-O</span> b <span class="op">&gt;</span> orca_3samp_1scaf_snps.bcf</span>
<span id="cb40-2"><a href="06-SNPs.html#cb40-2" tabindex="-1"></a></span>
<span id="cb40-3"><a href="06-SNPs.html#cb40-3" tabindex="-1"></a><span class="ex">bcftools</span> index orca_3samp_1scaf_snps.bcf</span>
<span id="cb40-4"><a href="06-SNPs.html#cb40-4" tabindex="-1"></a></span>
<span id="cb40-5"><a href="06-SNPs.html#cb40-5" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> orca_3samp_1scaf_snps.bcf <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre></div>
<p>bcftools allows applying filters on many of its commands, but usually they are used with <code>bcftools view</code> or with <code>bcftools filter</code>. Filtering can be done using information encoded in the QUAL or INFO fields, also allowing expression with multiple conditions and basic arithmetics <a href="https://samtools.github.io/bcftools/bcftools.html#expressions">more details here</a>.
Here are some examples:</p>
<p><strong>SNP quality</strong></p>
<p>An obvious filter is to exclude (<code>-e</code>) SNPs below a quality threshold:</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb41-1"><a href="06-SNPs.html#cb41-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-e</span> <span class="st">&#39;QUAL&lt;20&#39;</span> <span class="at">-O</span> b orca_3samp_1scaf_snps.bcf <span class="op">&gt;</span> orca_3samp_1scaf_snps.QS20.bcf</span></code></pre></div>
<p>Filters can also be specified as include (<code>-i</code>), the equivalent of the one above is:</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb42-1"><a href="06-SNPs.html#cb42-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-i</span> <span class="st">&#39;QUAL&gt;=20&#39;</span> <span class="at">-O</span> b orca_3samp_1scaf_snps.bcf <span class="op">&gt;</span> orca_3samp_1scaf_snps.QS20i.bcf</span></code></pre></div>
<p>You can see they have the same number of SNPs:</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb43-1"><a href="06-SNPs.html#cb43-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> orca_3samp_1scaf_snps.QS20.bcf <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb43-2"><a href="06-SNPs.html#cb43-2" tabindex="-1"></a></span>
<span id="cb43-3"><a href="06-SNPs.html#cb43-3" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> orca_3samp_1scaf_snps.QS20i.bcf <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre></div>
<p><strong>SNP depth</strong></p>
<p>An example to exclude SNPs with depth &lt;10:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb44-1"><a href="06-SNPs.html#cb44-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-e</span> <span class="st">&#39;INFO/DP&lt;10&#39;</span> <span class="at">-O</span> b orca_3samp_1scaf_snps.bcf <span class="op">&gt;</span> orca_3samp_1scaf_snps.DP10.bcf</span>
<span id="cb44-2"><a href="06-SNPs.html#cb44-2" tabindex="-1"></a><span class="co"># You can see this is quite a stringent filter for this subset of data:</span></span>
<span id="cb44-3"><a href="06-SNPs.html#cb44-3" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> orca_3samp_1scaf_snps.DP10.bcf <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre></div>
How many of the 267 SNPs have a sequencing depth of 10 or more reads?
<div id="radio_SYYXYOMUZP" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_SYYXYOMUZP" value="answer"></input> <span>2</span></label><label><input type="radio" autocomplete="off" name="radio_SYYXYOMUZP" value=""></input> <span>3</span></label><label><input type="radio" autocomplete="off" name="radio_SYYXYOMUZP" value=""></input> <span>8</span></label><label><input type="radio" autocomplete="off" name="radio_SYYXYOMUZP" value=""></input> <span>29</span></label>
</div>
<p>An important note here is to be aware that if the file is further processed so that only part of the individuals are used, the fields in INFO may not be updated and it would be then unreliable to filter out using any information from that field.</p>
<p><strong>Mapping quality</strong></p>
<p>Another typical filter is to remove SNPs with low mapping quality (RMS MQ: root mean square of the mapping quality of reads across all samples). For examples, all SNPs with RMS MQ &lt; 40 can be discarded with the following command:</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb45-1"><a href="06-SNPs.html#cb45-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-e</span> <span class="st">&#39;MQ&lt;40&#39;</span> <span class="at">-O</span> b orca_3samp_1scaf_snps.bcf <span class="op">&gt;</span> orca_3samp_1scaf_snps.MQ40.bcf</span></code></pre></div>
<p>How many SNPs has this removed?</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb46-1"><a href="06-SNPs.html#cb46-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> orca_3samp_1scaf_snps.bcf <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb46-2"><a href="06-SNPs.html#cb46-2" tabindex="-1"></a></span>
<span id="cb46-3"><a href="06-SNPs.html#cb46-3" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> orca_3samp_1scaf_snps.MQ40.bcf <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre></div>
<p>Eleven of the SNPs had MQ&gt;40, and most have MQ=60:</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb47-1"><a href="06-SNPs.html#cb47-1" tabindex="-1"></a><span class="ex">bcftools</span> query <span class="at">-f</span> <span class="st">&#39;%MQ\n&#39;</span> orca_3samp_1scaf_snps.bcf <span class="kw">|</span> <span class="fu">sort</span> <span class="kw">|</span> <span class="fu">uniq</span> <span class="at">-c</span></span></code></pre></div>
<p><strong>Allele frequency</strong></p>
<p>For population genetic analyses, it is frequently advised to remove variants below a certain allele frequency, as these ones are difficult to tell apart from sequencing errors. For example, to exclude all SNPs with a minor allele frequency (MAF) below 5% we would run:</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb48-1"><a href="06-SNPs.html#cb48-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-e</span> <span class="st">&#39;MAF&lt;0.05&#39;</span> <span class="at">-O</span> b orca_3samp_1scaf_snps.bcf <span class="op">&gt;</span> orca_3samp_1scaf_snps.MAF005.bcf</span></code></pre></div>
How many SNPs have a minor allele frequency less than 5% (i.e. 0.05)?
<div id="radio_XYXKQFANQU" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_XYXKQFANQU" value=""></input> <span>8</span></label><label><input type="radio" autocomplete="off" name="radio_XYXKQFANQU" value=""></input> <span>29</span></label><label><input type="radio" autocomplete="off" name="radio_XYXKQFANQU" value=""></input> <span>257</span></label><label><input type="radio" autocomplete="off" name="radio_XYXKQFANQU" value="answer"></input> <span>259</span></label>
</div>
<p><strong>Sample coverage</strong></p>
<p>Another typical situation is to want to exclude all SNPs for which only a small fracion of all the individuals have sequence data. That requires using the total number of alleles in the called genotypes (AN). In the case of a diploid species, this can be used to filter SNPs by number of individuals genotyped. For example, we can remove all SNPs genotyped for less than 3 individuals:</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb49-1"><a href="06-SNPs.html#cb49-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-e</span> <span class="st">&#39;AN/2&lt;3&#39;</span> <span class="at">-O</span> b orca_3samp_1scaf_snps.bcf <span class="op">&gt;</span> orca_3samp_1scaf_snps.SAMP3.bcf</span></code></pre></div>
How many SNPs have been called and genotyped all three of the samples/individuals?
<div id="radio_GKTDCOPKRW" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_GKTDCOPKRW" value=""></input> <span>3</span></label><label><input type="radio" autocomplete="off" name="radio_GKTDCOPKRW" value=""></input> <span>8</span></label><label><input type="radio" autocomplete="off" name="radio_GKTDCOPKRW" value="answer"></input> <span>10</span></label><label><input type="radio" autocomplete="off" name="radio_GKTDCOPKRW" value=""></input> <span>29</span></label>
</div>
<p><strong>Genotype-based filtering</strong></p>
<p>Filtering using the genotype fields can allow for some more precise filtering. For example, in cases of high depth heterogeneity among samples, it may be better to filter out by median genotype depth than by total depth across all samples. An example removing all SNPs where the mean genotype depth is below 3:</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb50-1"><a href="06-SNPs.html#cb50-1" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-e</span> <span class="st">&#39;AVG(FMT/DP)&lt;3&#39;</span> <span class="at">-O</span> b orca_3samp_1scaf_snps.bcf <span class="op">&gt;</span> orca_3samp_1scaf_snps.MEANGTDP3.bcf</span></code></pre></div>
How many SNPs have a mean genotyping depth of 3 or more?
<div id="radio_QBXNUAJFJI" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_QBXNUAJFJI" value="answer"></input> <span>3</span></label><label><input type="radio" autocomplete="off" name="radio_QBXNUAJFJI" value=""></input> <span>8</span></label><label><input type="radio" autocomplete="off" name="radio_QBXNUAJFJI" value=""></input> <span>10</span></label><label><input type="radio" autocomplete="off" name="radio_QBXNUAJFJI" value=""></input> <span>29</span></label>
</div>
<p>Sometimes it is reasonable to ignore genotype calls based on few reads. The following command remove all genotype calls (i.e. genotypes are substituted by ./.) informed by less than 5 reads:</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb51-1"><a href="06-SNPs.html#cb51-1" tabindex="-1"></a><span class="ex">bcftools</span> filter <span class="at">-S</span> . <span class="at">-e</span> <span class="st">&#39;FMT/DP&lt;5&#39;</span> <span class="at">-O</span> b orca_3samp_1scaf_snps.bcf <span class="op">&gt;</span> orca_3samp_1scaf_snps.NOGTDP5.bcf</span></code></pre></div>
<p><strong>Combining multiple filters</strong></p>
<p>Multiple filters can be combined in a single command using or piping several ones. For example, we can combine a few of the filters we have used above:</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb52-1"><a href="06-SNPs.html#cb52-1" tabindex="-1"></a><span class="ex">bcftools</span> filter <span class="at">-S</span> . <span class="at">-e</span> <span class="st">&#39;FMT/DP&lt;3&#39;</span> orca_3samp_1scaf_snps.bcf <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb52-2"><a href="06-SNPs.html#cb52-2" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-e</span> <span class="st">&#39;AVG(FMT/DP)&lt;3 || QUAL&lt;20 || AN/2&lt;2&#39;</span> <span class="at">-O</span> b <span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb52-3"><a href="06-SNPs.html#cb52-3" tabindex="-1"></a>orca_3samp_1scaf_snps.NOGTDP3.MEANGTDP3.Q20.SAMP2.bcf</span>
<span id="cb52-4"><a href="06-SNPs.html#cb52-4" tabindex="-1"></a><span class="co"># This results in quite a dramatic reduction in the number of SNPs:</span></span>
<span id="cb52-5"><a href="06-SNPs.html#cb52-5" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> orca_3samp_1scaf_snps.NOGTDP3.MEANGTDP3.Q20.SAMP2.bcf <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre></div>
<p>Final note: It is important to be careful with the order of the filters, as different combinations can result in different end results, especially if we subsample the individuals at some point. Also, some positions may end up being invariant or only variant with respect to the reference (i.e. private), or not having any individuals genotyped. We have also worked on a small bcf file with these examples - just one scaffold for three individuals. With larger amounts of data you will likely want to change some of the threshold values from those used here.</p>

</div>
</div>
</div>



<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="05-Mapping.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="07-NGSAdmix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
