<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Calculating population summary statistics using PopGenome | Population Genomics</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Calculating population summary statistics using PopGenome | Population Genomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Calculating population summary statistics using PopGenome | Population Genomics" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Helen Hipperson, Katy Maher, Victor Soria-Carrasco, Graeme Fox" />


<<<<<<< Updated upstream
<meta name="date" content="2023-06-19" />
=======
<meta name="date" content="2023-07-14" />
>>>>>>> Stashed changes

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="07-NGSAdmix.html"/>
<link rel="next" href="09-Detecting_selection.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Population Genomics</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-whole-genome-resequencing-of-killer-whales"><i class="fa fa-check"></i><b>2.1</b> Example data: Whole-genome resequencing of killer whales</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="part"><span><b>Generating SNP datasets</b></span></li>
<li class="chapter" data-level="4" data-path="04-QC.html"><a href="04-QC.html"><i class="fa fa-check"></i><b>4</b> QC Tutorials</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC.html"><a href="04-QC.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-QC.html"><a href="04-QC.html#killer-whale-genomic-dataset"><i class="fa fa-check"></i><b>4.2</b> Killer whale genomic dataset</a></li>
<li class="chapter" data-level="4.3" data-path="04-QC.html"><a href="04-QC.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-QC.html"><a href="04-QC.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality control</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="04-QC.html"><a href="04-QC.html#post-quality-control-check"><i class="fa fa-check"></i><b>4.4.1</b> Post Quality control check</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Mapping.html"><a href="05-Mapping.html"><i class="fa fa-check"></i><b>5</b> Mapping sequences to the reference genome using BWA</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Mapping.html"><a href="05-Mapping.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>5.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="5.2" data-path="05-Mapping.html"><a href="05-Mapping.html#read-mapping"><i class="fa fa-check"></i><b>5.2</b> Read mapping</a></li>
<li class="chapter" data-level="5.3" data-path="05-Mapping.html"><a href="05-Mapping.html#assessing-mapping-results"><i class="fa fa-check"></i><b>5.3</b> Assessing mapping results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-SNPs.html"><a href="06-SNPs.html"><i class="fa fa-check"></i><b>6</b> SNP and genotype calling using SAMtools and BCFtools</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-SNPs.html"><a href="06-SNPs.html#file-preparation"><i class="fa fa-check"></i><b>6.1</b> File preparation</a></li>
<li class="chapter" data-level="6.2" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-calling"><i class="fa fa-check"></i><b>6.2</b> Variant calling</a></li>
<li class="chapter" data-level="6.3" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-filtering"><i class="fa fa-check"></i><b>6.3</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-SNPs.html"><a href="06-SNPs.html#extracting-information"><i class="fa fa-check"></i><b>6.3.1</b> Extracting information</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-SNPs.html"><a href="06-SNPs.html#subsetting-files"><i class="fa fa-check"></i><b>6.3.2</b> Subsetting files</a></li>
<li class="chapter" data-level="6.3.3" data-path="06-SNPs.html"><a href="06-SNPs.html#filtering"><i class="fa fa-check"></i><b>6.3.3</b> Filtering</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="7" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html"><i class="fa fa-check"></i><b>7</b> Population structure with NGSadmix and PCA</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#ngsadmix"><i class="fa fa-check"></i><b>7.1</b> NGSadmix</a></li>
<li class="chapter" data-level="7.2" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#jupyter"><i class="fa fa-check"></i><b>7.2</b> Jupyter</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>7.2.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="7.2.2" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#create-r-notebook"><i class="fa fa-check"></i><b>7.2.2</b> Create R notebook</a></li>
<li class="chapter" data-level="7.2.3" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#cells-and-code"><i class="fa fa-check"></i><b>7.2.3</b> Cells and code</a></li>
<li class="chapter" data-level="7.2.4" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#create-new-cells"><i class="fa fa-check"></i><b>7.2.4</b> Create new cells</a></li>
<li class="chapter" data-level="7.2.5" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#running-code"><i class="fa fa-check"></i><b>7.2.5</b> Running code</a></li>
<li class="chapter" data-level="7.2.6" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#saving-the-file"><i class="fa fa-check"></i><b>7.2.6</b> Saving the file</a></li>
<li class="chapter" data-level="7.2.7" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>7.2.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="7.2.8" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#close-the-notebook"><i class="fa fa-check"></i><b>7.2.8</b> Close the notebook</a></li>
<li class="chapter" data-level="7.2.9" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#video-tutorial"><i class="fa fa-check"></i><b>7.2.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#admixture-plots"><i class="fa fa-check"></i><b>7.3</b> Admixture plots</a></li>
<li class="chapter" data-level="7.4" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#pca"><i class="fa fa-check"></i><b>7.4</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html"><i class="fa fa-check"></i><b>8</b> Calculating population summary statistics using PopGenome</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html#getting-started"><i class="fa fa-check"></i><b>8.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html"><i class="fa fa-check"></i><b>9</b> Delimitation of contiguous regions of differentiation using Hidden Markov Models (HMM)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#initial-set-up"><i class="fa fa-check"></i><b>9.1</b> Initial set up</a></li>
<li class="chapter" data-level="9.2" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#data-formatting"><i class="fa fa-check"></i><b>9.2</b> Data formatting</a></li>
<li class="chapter" data-level="9.3" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#allele-frequency-estimation"><i class="fa fa-check"></i><b>9.3</b> Allele frequency estimation</a></li>
<li class="chapter" data-level="9.4" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#fst-estimation"><i class="fa fa-check"></i><b>9.4</b> F<sub>ST</sub> estimation</a></li>
<li class="chapter" data-level="9.5" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#delimitation-of-contiguous-regions-of-differentiation-using-a-hmm-model"><i class="fa fa-check"></i><b>9.5</b> Delimitation of contiguous regions of differentiation using a HMM model</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-GWAS.html"><a href="10-GWAS.html"><i class="fa fa-check"></i><b>10</b> Genetic architecture of traits using multi-locus Genome Wide Association (GWA) mapping with GEMMA</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-GWAS.html"><a href="10-GWAS.html#initial-set-up-1"><i class="fa fa-check"></i><b>10.1</b> Initial set up</a></li>
<li class="chapter" data-level="10.2" data-path="10-GWAS.html"><a href="10-GWAS.html#data-formatting-1"><i class="fa fa-check"></i><b>10.2</b> Data formatting</a></li>
<li class="chapter" data-level="10.3" data-path="10-GWAS.html"><a href="10-GWAS.html#running-gemma-to-fit-a-bayesian-sparse-linear-mixed-model-bslmm"><i class="fa fa-check"></i><b>10.3</b> Running GEMMA to fit a Bayesian sparse linear mixed model (BSLMM)</a></li>
<li class="chapter" data-level="10.4" data-path="10-GWAS.html"><a href="10-GWAS.html#analysing-gemma-bslmm-output"><i class="fa fa-check"></i><b>10.4</b> Analysing GEMMA BSLMM output</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="11-Appendix.html"><a href="11-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="11-Appendix.html"><a href="11-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
<<<<<<< Updated upstream
<li class="chapter" data-level="A.2" data-path="11-Appendix.html"><a href="11-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>A.2</b> Jupyter-notebook</a></li>
=======
</ul></li>
<li class="chapter" data-level="B" data-path="11-Appendix.html"><a href="11-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>B</b> Jupyter-notebook</a></li>
<li class="chapter" data-level="C" data-path="11-Appendix.html"><a href="11-Appendix.html#loops"><i class="fa fa-check"></i><b>C</b> Using loops to process multiple samples</a>
<ul>
<li class="chapter" data-level="C.1" data-path="11-Appendix.html"><a href="11-Appendix.html#simple-loop-example"><i class="fa fa-check"></i><b>C.1</b> Simple loop example</a></li>
<li class="chapter" data-level="C.2" data-path="11-Appendix.html"><a href="11-Appendix.html#trimmomatic-loop-example"><i class="fa fa-check"></i><b>C.2</b> Trimmomatic loop example</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="11-Appendix.html"><a href="11-Appendix.html#an-overview-of-bayesian-statistics"><i class="fa fa-check"></i><b>D</b> An overview of Bayesian statistics</a></li>
<li class="chapter" data-level="E" data-path="11-Appendix.html"><a href="11-Appendix.html#additional-approaches-for-detecting-selection-and-for-gwas"><i class="fa fa-check"></i><b>E</b> Additional approaches for detecting selection and for GWAS</a>
<ul>
<li class="chapter" data-level="E.1" data-path="11-Appendix.html"><a href="11-Appendix.html#rehh"><i class="fa fa-check"></i><b>E.1</b> rehh</a></li>
<li class="chapter" data-level="E.2" data-path="11-Appendix.html"><a href="11-Appendix.html#gcta"><i class="fa fa-check"></i><b>E.2</b> GCTA</a></li>
>>>>>>> Stashed changes
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Population Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="popstats" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Calculating population summary statistics using PopGenome<a href="08-Summary_stats.html#popstats" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Generating summary statistics of your dataset can be useful to help you get a feel for your data and complement the visual structure plots and PCAs created in the previous chapter.</p>
<p>We will now generate a summary of nucleotide diversity and <em>F<sub>ST</sub></em> using one scaffold (to save time) from the vcf file we have generated using the package <code>PopGenome</code> in R.</p>
<p><img src="figures/popgenome.png" width="20%" style="display: block; margin: auto;" /></p>
<p><code>PopGenome</code> is a powerful and flexible R package that can be used to calculate a diverse range of population genetics statistics. We will use only a couple of very basic features and analyses here, but we strongly recommend reading the <a href="https://academic.oup.com/mbe/article/31/7/1929/2925788">paper</a> and <a href="https://cran.r-project.org/web/packages/PopGenome/vignettes/An_introduction_to_the_PopGenome_package.pdf">manual</a> for greater understanding of its capabilities. Advantages of PopGenome include its abilty to:</p>
<ul>
<li>Read data in a variety of input formats</li>
<li>Implement a comprehensive range of population genetics/genomics analyses and statistics</li>
<li>Read associated annotation files and allow to systematically select regions of interest</li>
<li>Analyse individual loci, multiple loci, and sliding windows</li>
<li>Open source and be easily extendable by the scientific community to incorporate new types of analyses</li>
<li>Integrated with powerful numerical and graphical capabilities</li>
<li>Platform independent</li>
</ul>
<p>We will now generate a couple of commonly used population genomic statistics, <em>F<sub>ST</sub></em> and nucleutide diversity (<em>π</em>).</p>
<p><strong><em>F<sub>ST</sub></em></strong></p>
<p>F statistics are one of the of the most common methods for quantifying genetic differentiation between subpopulations. <em>F<sub>ST</sub></em> (fixation index) estimates genetic differentiation between subpopulations. It reflects the degree of inbreeding within a subpopulation relative to the total population, as the probability that two alleles drawn at random from a subpopulation are identical by descent. <em>F<sub>ST</sub></em> is bound between 0 and 1. If <em>F<sub>ST</sub></em> = 0, there is no genetic differentiation between populations. If <em>F<sub>ST</sub></em> = 1, the two populations are very strongly separated.</p>
<p><strong>Equation</strong>:
<span class="math display">\[
F _{ST} =
\frac{(H_T - H_S)} {H_S}
\]</span></p>
<ul>
<li><em>H<sub>T</sub></em> expected heterozygosity of the total population</li>
<li><em>H<sub>S</sub></em> expected heterozygosity if the subpopulation were in Hardy–Weinberg Equilibrium</li>
</ul>
<p><strong><em>π</em></strong></p>
<p>Nucleotide diversity (<em>π</em>) quantifies the mean divergence between sequences and reflects the extent to which sequences differ from each other. <em>π</em> is the average pairwise difference between all possible pairs of individuals in your sample.</p>
<p><strong>Equation</strong>:
<span class="math display">\[
\pi =
\sum f_i f_jp_{ij}
\]</span></p>
<ul>
<li><em>f<sub>i</sub></em> and <em>f<sub>i</sub></em> frequencies of the _i_th and _j_th haplotypes in the population</li>
<li><em>p<sub>ij</sub></em> sequence divergence between haplotypes</li>
</ul>
<div id="getting-started" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Getting started<a href="08-Summary_stats.html#getting-started" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/genome_pop.png" width="20%" style="display: block; margin: auto;" /></p>
<<<<<<< Updated upstream
<p>Now launch R by typing <code>R</code>into the terminal.</p>
<p>Then load the PopGenome package:</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="08-Summary_stats.html#cb69-1" tabindex="-1"></a><span class="fu">library</span>(PopGenome)</span></code></pre></div>
<p>We will now load in our SNP data, PopGenome can handle vcf files.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="08-Summary_stats.html#cb70-1" tabindex="-1"></a><span class="co"># Load the data</span></span>
<span id="cb70-2"><a href="08-Summary_stats.html#cb70-2" tabindex="-1"></a>GENOME.class <span class="ot">&lt;-</span> <span class="fu">readData</span>(<span class="st">&quot;/pub14/tea/nsc206/NEOF/popgenomics/data/popgenome/&quot;</span>, </span>
<span id="cb70-3"><a href="08-Summary_stats.html#cb70-3" tabindex="-1"></a>                         <span class="at">format=</span><span class="st">&quot;VCF&quot;</span>)</span></code></pre></div>
<p>This makes a complex object which we can use to extract useful data.</p>
<p>First we can get a summary of the vcf we have imported.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="08-Summary_stats.html#cb71-1" tabindex="-1"></a><span class="co"># This is complex object, with several slots</span></span>
<span id="cb71-2"><a href="08-Summary_stats.html#cb71-2" tabindex="-1"></a><span class="fu">get.sum.data</span>(GENOME.class)</span></code></pre></div>
=======
<p>In jupyter notebook, create a new notebook called <strong>"Chp08-PopGenome"</strong>.</p>
<p>Create a new markdown cell titled # Summary Statistics using PopGenome</p>
<p>Load the PopGenome package in a new code cell.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="08-Summary_stats.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PopGenome)</span></code></pre></div>
<p>We will now load in our SNP data, PopGenome can handle vcf files.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="08-Summary_stats.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the data</span></span>
<span id="cb71-2"><a href="08-Summary_stats.html#cb71-2" aria-hidden="true" tabindex="-1"></a>GENOME.class <span class="ot">&lt;-</span> <span class="fu">readData</span>(<span class="st">&quot;/pub14/tea/nsc206/NEOF/popgenomics/data/popgenome/&quot;</span>, </span>
<span id="cb71-3"><a href="08-Summary_stats.html#cb71-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">format=</span><span class="st">&quot;VCF&quot;</span>)</span></code></pre></div>
<p>This makes a complex object which we can use to extract useful data.</p>
<p>First we can get a summary of the vcf we have imported.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="08-Summary_stats.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This is complex object, with several slots</span></span>
<span id="cb72-2"><a href="08-Summary_stats.html#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="fu">get.sum.data</span>(GENOME.class)</span></code></pre></div>
>>>>>>> Stashed changes
<ul>
<li>n.sites - this is the position of the last SNP in the vcf file - in this case 17,452,779</li>
<li>n.biallelic.sites - Number of biallelic sites (SNPs) - there are 7,511 biallelic SNPs in this scaffold file - note SNPs are only counted if they have a 'PASS' description in the filter column (you can get the full number of SNPs in the vcf by typing <code>GENOME.class@n.sites2</code>)</li>
<li>n.gaps - Number of gaps observed in the data - 0</li>
<li>n.unknowns - Number of unknown positions - 778 this corresponds to the number of SNPs without the 'PASS' description in the filter column</li>
<li>n.valid.sites - Number of sites without gaps (number reported as NA for vcf files)</li>
<li>n.polyallelic.sites - Number of sites with more than two variants - 0</li>
<li>trans.transv.ratio - Transition (changes from A ↔︎ G and C ↔︎ T) - transversion (changes
from A ↔︎ C, A ↔︎T, G ↔︎ C or G ↔︎ T) ratio</li>
</ul>
<<<<<<< Updated upstream
<p>The object <code>GENOME.class</code> contains all the information of the different data that you can import and assign to the vcf and the statistics that can be run. To get a summary type:</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="08-Summary_stats.html#cb72-1" tabindex="-1"></a><span class="fu">show.slots</span>(GENOME.class)</span></code></pre></div>
<p>These slots can be accessed by using the “@” sign. For example the number of sites:</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="08-Summary_stats.html#cb73-1" tabindex="-1"></a>GENOME.class<span class="sc">@</span>n.sites</span></code></pre></div>
<p>For more information on the slots available using PopGenome please refer to the <a href="https://cran.r-project.org/web/packages/PopGenome/vignettes/An_introduction_to_the_PopGenome_package.pdf">manual</a></p>
<p>We can now assign population information to the object. This is needed to calculate <em>F<sub>ST</sub></em> and <em>π</em>. For now we will specify just two of the populations.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="08-Summary_stats.html#cb74-1" tabindex="-1"></a><span class="co"># Make a list of individuals for each population</span></span>
<span id="cb74-2"><a href="08-Summary_stats.html#cb74-2" tabindex="-1"></a><span class="co"># Note: the sample names in the vcf contain the path and the .bam file extension</span></span>
<span id="cb74-3"><a href="08-Summary_stats.html#cb74-3" tabindex="-1"></a><span class="co"># Here we are just making a list of sample names as they appear in the vcf -</span></span>
<span id="cb74-4"><a href="08-Summary_stats.html#cb74-4" tabindex="-1"></a><span class="co"># we are not importing any additional files</span></span>
<span id="cb74-5"><a href="08-Summary_stats.html#cb74-5" tabindex="-1"></a></span>
<span id="cb74-6"><a href="08-Summary_stats.html#cb74-6" tabindex="-1"></a>pop1<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;./aligned/SRR8861576_paired.bam&quot;</span>,<span class="st">&quot;./aligned/SRR8861581_paired.bam&quot;</span>,</span>
<span id="cb74-7"><a href="08-Summary_stats.html#cb74-7" tabindex="-1"></a>        <span class="st">&quot;./aligned/SRR8861586_paired.bam&quot;</span>,<span class="st">&quot;/aligned/SRR8861592_paired.bam&quot;</span>)</span>
<span id="cb74-8"><a href="08-Summary_stats.html#cb74-8" tabindex="-1"></a>pop2<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;./aligned/SRR8861578_paired.bam&quot;</span>,<span class="st">&quot;./aligned/SRR8861583_paired.bam&quot;</span>,</span>
<span id="cb74-9"><a href="08-Summary_stats.html#cb74-9" tabindex="-1"></a>        <span class="st">&quot;./aligned/SRR8861589_paired.bam&quot;</span>,<span class="st">&quot;./aligned/SRR8861591_paired.bam&quot;</span>)</span>
<span id="cb74-10"><a href="08-Summary_stats.html#cb74-10" tabindex="-1"></a></span>
<span id="cb74-11"><a href="08-Summary_stats.html#cb74-11" tabindex="-1"></a><span class="co"># Set populations</span></span>
<span id="cb74-12"><a href="08-Summary_stats.html#cb74-12" tabindex="-1"></a>GENOME.class  <span class="ot">&lt;-</span> <span class="fu">set.populations</span>(GENOME.class, <span class="fu">list</span>(pop1,pop2))</span>
<span id="cb74-13"><a href="08-Summary_stats.html#cb74-13" tabindex="-1"></a></span>
<span id="cb74-14"><a href="08-Summary_stats.html#cb74-14" tabindex="-1"></a><span class="co"># check if setting the populations worked</span></span>
<span id="cb74-15"><a href="08-Summary_stats.html#cb74-15" tabindex="-1"></a>GENOME.class<span class="sc">@</span>populations <span class="co"># You can see that the sample names have been assigned to </span></span>
<span id="cb74-16"><a href="08-Summary_stats.html#cb74-16" tabindex="-1"></a>                            <span class="co">#two different within  @populations</span></span></code></pre></div>
<p>Since the calculation of certain population genetic parameters can be computationally intensive, they have to be executed separately beforehand. For this modules have to be run. To calculate <em>F<sub>ST</sub></em> and <em>π</em> we can use the F_st.stats module.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="08-Summary_stats.html#cb75-1" tabindex="-1"></a><span class="co"># Perform the calculations and add the results to the appropriate slots</span></span>
<span id="cb75-2"><a href="08-Summary_stats.html#cb75-2" tabindex="-1"></a>GENOME.class <span class="ot">&lt;-</span> <span class="fu">F_ST.stats</span>(GENOME.class) </span>
<span id="cb75-3"><a href="08-Summary_stats.html#cb75-3" tabindex="-1"></a></span>
<span id="cb75-4"><a href="08-Summary_stats.html#cb75-4" tabindex="-1"></a><span class="co"># Print FST</span></span>
<span id="cb75-5"><a href="08-Summary_stats.html#cb75-5" tabindex="-1"></a>GENOME.class<span class="sc">@</span>nucleotide.F_ST <span class="co"># Fst is very low between these two populations</span></span>
<span id="cb75-6"><a href="08-Summary_stats.html#cb75-6" tabindex="-1"></a><span class="co"># Print nucleotide diversity along the scaffold and average nucleotide diversity per site</span></span>
<span id="cb75-7"><a href="08-Summary_stats.html#cb75-7" tabindex="-1"></a>GENOME.class<span class="sc">@</span>Pi</span>
<span id="cb75-8"><a href="08-Summary_stats.html#cb75-8" tabindex="-1"></a>GENOME.class<span class="sc">@</span>Pi<span class="sc">/</span>GENOME.class<span class="sc">@</span>n.sites2</span></code></pre></div>
=======
<p>The object <code>GENOME.class</code> contains all the information of the different data that you can import and assign to the vcf and the statistics that can be run. To get a summary, type:</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="08-Summary_stats.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show.slots</span>(GENOME.class)</span></code></pre></div>
<p>These slots can be accessed by using the "@" sign. For example the number of sites:</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="08-Summary_stats.html#cb74-1" aria-hidden="true" tabindex="-1"></a>GENOME.class<span class="sc">@</span>n.sites</span></code></pre></div>
<p>For more information on the slots available using PopGenome please refer to the <a href="https://cran.r-project.org/web/packages/PopGenome/vignettes/An_introduction_to_the_PopGenome_package.pdf">manual</a></p>
<p>We can now assign population information to the object. This is needed to calculate <em>F<sub>ST</sub></em> and <em>π</em>. For now we will specify just two of the populations.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="08-Summary_stats.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a list of individuals for each population</span></span>
<span id="cb75-2"><a href="08-Summary_stats.html#cb75-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: the sample names in the vcf contain the path and the .bam file extension</span></span>
<span id="cb75-3"><a href="08-Summary_stats.html#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Here we are just making a list of sample names as they appear in the vcf -</span></span>
<span id="cb75-4"><a href="08-Summary_stats.html#cb75-4" aria-hidden="true" tabindex="-1"></a><span class="co"># we are not importing any additional files</span></span>
<span id="cb75-5"><a href="08-Summary_stats.html#cb75-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-6"><a href="08-Summary_stats.html#cb75-6" aria-hidden="true" tabindex="-1"></a>pop1<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;./aligned/SRR8861576_paired.bam&quot;</span>,<span class="st">&quot;./aligned/SRR8861581_paired.bam&quot;</span>,</span>
<span id="cb75-7"><a href="08-Summary_stats.html#cb75-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;./aligned/SRR8861586_paired.bam&quot;</span>,<span class="st">&quot;/aligned/SRR8861592_paired.bam&quot;</span>)</span>
<span id="cb75-8"><a href="08-Summary_stats.html#cb75-8" aria-hidden="true" tabindex="-1"></a>pop2<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;./aligned/SRR8861578_paired.bam&quot;</span>,<span class="st">&quot;./aligned/SRR8861583_paired.bam&quot;</span>,</span>
<span id="cb75-9"><a href="08-Summary_stats.html#cb75-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;./aligned/SRR8861589_paired.bam&quot;</span>,<span class="st">&quot;./aligned/SRR8861591_paired.bam&quot;</span>)</span>
<span id="cb75-10"><a href="08-Summary_stats.html#cb75-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-11"><a href="08-Summary_stats.html#cb75-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Set populations</span></span>
<span id="cb75-12"><a href="08-Summary_stats.html#cb75-12" aria-hidden="true" tabindex="-1"></a>GENOME.class  <span class="ot">&lt;-</span> <span class="fu">set.populations</span>(GENOME.class, <span class="fu">list</span>(pop1,pop2))</span>
<span id="cb75-13"><a href="08-Summary_stats.html#cb75-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-14"><a href="08-Summary_stats.html#cb75-14" aria-hidden="true" tabindex="-1"></a><span class="co"># check if setting the populations worked</span></span>
<span id="cb75-15"><a href="08-Summary_stats.html#cb75-15" aria-hidden="true" tabindex="-1"></a>GENOME.class<span class="sc">@</span>populations <span class="co"># You can see that the sample names have been assigned to </span></span>
<span id="cb75-16"><a href="08-Summary_stats.html#cb75-16" aria-hidden="true" tabindex="-1"></a>                            <span class="co">#two different within  @populations</span></span></code></pre></div>
<p>Since the calculation of certain population genetic parameters can be computationally intensive, they have to be executed separately beforehand. For this, modules have to be run. To calculate <em>F<sub>ST</sub></em> and <em>π</em> we can use the F_st.stats module.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="08-Summary_stats.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform the calculations and add the results to the appropriate slots</span></span>
<span id="cb76-2"><a href="08-Summary_stats.html#cb76-2" aria-hidden="true" tabindex="-1"></a>GENOME.class <span class="ot">&lt;-</span> <span class="fu">F_ST.stats</span>(GENOME.class) </span>
<span id="cb76-3"><a href="08-Summary_stats.html#cb76-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-4"><a href="08-Summary_stats.html#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Print FST</span></span>
<span id="cb76-5"><a href="08-Summary_stats.html#cb76-5" aria-hidden="true" tabindex="-1"></a>GENOME.class<span class="sc">@</span>nucleotide.F_ST <span class="co"># Fst is very low between these two populations</span></span>
<span id="cb76-6"><a href="08-Summary_stats.html#cb76-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-7"><a href="08-Summary_stats.html#cb76-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Print nucleotide diversity along the scaffold and average nucleotide diversity per site</span></span>
<span id="cb76-8"><a href="08-Summary_stats.html#cb76-8" aria-hidden="true" tabindex="-1"></a>GENOME.class<span class="sc">@</span>Pi</span>
<span id="cb76-9"><a href="08-Summary_stats.html#cb76-9" aria-hidden="true" tabindex="-1"></a>GENOME.class<span class="sc">@</span>Pi<span class="sc">/</span>GENOME.class<span class="sc">@</span>n.sites2</span></code></pre></div>
>>>>>>> Stashed changes
<p>This can give you a quick feel for your dataset but in reality diversity will vary throughout the genome. It is often more appropriate and interesting to examine how different regions of the genome have diverged between populations. We will explore one method for estimating differentiation between a pair of populations in the next chapter.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="07-NGSAdmix.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="09-Detecting_selection.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
