<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Calculating population summary statistics using VCFtools | Population Genomics</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Calculating population summary statistics using VCFtools | Population Genomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Calculating population summary statistics using VCFtools | Population Genomics" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Helen Hipperson, Katy Maher, Victor Soria-Carrasco, Graeme Fox, Gavin Gouws" />


<meta name="date" content="2025-09-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="07-NGSAdmix.html"/>
<link rel="next" href="09-Detecting_selection.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/tippyjs-3.2.0/themes/google.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/light-border.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/light.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/translucent.css" rel="stylesheet" />
<script src="libs/tippyjs-3.2.0/tippy.all.min.js"></script>
<script src="libs/tippy-binding-0.1.0/tippy.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_rounded_corners.png" width="250"></a></li>
<li><a>Population Genomics</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-whole-genome-resequencing-of-killer-whales"><i class="fa fa-check"></i><b>2.1</b> Example data: Whole-genome resequencing of killer whales</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="part"><span><b>Generating SNP datasets</b></span></li>
<li class="chapter" data-level="4" data-path="04-QC.html"><a href="04-QC.html"><i class="fa fa-check"></i><b>4</b> QC Tutorials</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC.html"><a href="04-QC.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-QC.html"><a href="04-QC.html#killer-whale-genomic-dataset"><i class="fa fa-check"></i><b>4.2</b> Killer whale genomic dataset</a></li>
<li class="chapter" data-level="4.3" data-path="04-QC.html"><a href="04-QC.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-QC.html"><a href="04-QC.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality control</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="04-QC.html"><a href="04-QC.html#post-quality-control-check"><i class="fa fa-check"></i><b>4.4.1</b> Post Quality control check</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Mapping.html"><a href="05-Mapping.html"><i class="fa fa-check"></i><b>5</b> Mapping sequences to the reference genome using BWA</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Mapping.html"><a href="05-Mapping.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>5.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="5.2" data-path="05-Mapping.html"><a href="05-Mapping.html#read-mapping"><i class="fa fa-check"></i><b>5.2</b> Read mapping</a></li>
<li class="chapter" data-level="5.3" data-path="05-Mapping.html"><a href="05-Mapping.html#assessing-mapping-results"><i class="fa fa-check"></i><b>5.3</b> Assessing mapping results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-SNPs.html"><a href="06-SNPs.html"><i class="fa fa-check"></i><b>6</b> SNP and genotype calling using SAMtools and BCFtools</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-SNPs.html"><a href="06-SNPs.html#file-preparation"><i class="fa fa-check"></i><b>6.1</b> File preparation</a></li>
<li class="chapter" data-level="6.2" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-calling"><i class="fa fa-check"></i><b>6.2</b> Variant calling</a></li>
<li class="chapter" data-level="6.3" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-filtering"><i class="fa fa-check"></i><b>6.3</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-SNPs.html"><a href="06-SNPs.html#extracting-information"><i class="fa fa-check"></i><b>6.3.1</b> Extracting information</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-SNPs.html"><a href="06-SNPs.html#subsetting-files"><i class="fa fa-check"></i><b>6.3.2</b> Subsetting files</a></li>
<li class="chapter" data-level="6.3.3" data-path="06-SNPs.html"><a href="06-SNPs.html#filtering"><i class="fa fa-check"></i><b>6.3.3</b> Filtering</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="7" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html"><i class="fa fa-check"></i><b>7</b> Population structure with NGSadmix and PCA</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#ngsadmix"><i class="fa fa-check"></i><b>7.1</b> NGSadmix</a></li>
<li class="chapter" data-level="7.2" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#jupyter"><i class="fa fa-check"></i><b>7.2</b> Jupyter</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>7.2.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="7.2.2" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#create-r-notebook"><i class="fa fa-check"></i><b>7.2.2</b> Create R notebook</a></li>
<li class="chapter" data-level="7.2.3" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#cells-and-code"><i class="fa fa-check"></i><b>7.2.3</b> Cells and code</a></li>
<li class="chapter" data-level="7.2.4" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#create-new-cells"><i class="fa fa-check"></i><b>7.2.4</b> Create new cells</a></li>
<li class="chapter" data-level="7.2.5" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#running-code"><i class="fa fa-check"></i><b>7.2.5</b> Running code</a></li>
<li class="chapter" data-level="7.2.6" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#saving-the-file"><i class="fa fa-check"></i><b>7.2.6</b> Saving the file</a></li>
<li class="chapter" data-level="7.2.7" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>7.2.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="7.2.8" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#close-the-notebook"><i class="fa fa-check"></i><b>7.2.8</b> Close the notebook</a></li>
<li class="chapter" data-level="7.2.9" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#video-tutorial"><i class="fa fa-check"></i><b>7.2.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#admixture-plots"><i class="fa fa-check"></i><b>7.3</b> Admixture plots</a></li>
<li class="chapter" data-level="7.4" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#pca"><i class="fa fa-check"></i><b>7.4</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html"><i class="fa fa-check"></i><b>8</b> Calculating population summary statistics using VCFtools</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html#getting-started"><i class="fa fa-check"></i><b>8.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html"><i class="fa fa-check"></i><b>9</b> Delimitation of contiguous regions of differentiation using Hidden Markov Models (HMM)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#initial-set-up"><i class="fa fa-check"></i><b>9.1</b> Initial set up</a></li>
<li class="chapter" data-level="9.2" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#data-formatting"><i class="fa fa-check"></i><b>9.2</b> Data formatting</a></li>
<li class="chapter" data-level="9.3" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#allele-frequency-estimation"><i class="fa fa-check"></i><b>9.3</b> Allele frequency estimation</a></li>
<li class="chapter" data-level="9.4" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#fst-estimation"><i class="fa fa-check"></i><b>9.4</b> F<sub>ST</sub> estimation</a></li>
<li class="chapter" data-level="9.5" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#delimitation-of-contiguous-regions-of-differentiation-using-a-hmm-model"><i class="fa fa-check"></i><b>9.5</b> Delimitation of contiguous regions of differentiation using a HMM model</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-GWAS.html"><a href="10-GWAS.html"><i class="fa fa-check"></i><b>10</b> Genetic architecture of traits using multi-locus Genome Wide Association (GWA) mapping with GEMMA</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-GWAS.html"><a href="10-GWAS.html#initial-set-up-1"><i class="fa fa-check"></i><b>10.1</b> Initial set up</a></li>
<li class="chapter" data-level="10.2" data-path="10-GWAS.html"><a href="10-GWAS.html#data-formatting-1"><i class="fa fa-check"></i><b>10.2</b> Data formatting</a></li>
<li class="chapter" data-level="10.3" data-path="10-GWAS.html"><a href="10-GWAS.html#running-gemma-to-fit-a-bayesian-sparse-linear-mixed-model-bslmm"><i class="fa fa-check"></i><b>10.3</b> Running GEMMA to fit a Bayesian sparse linear mixed model (BSLMM)</a></li>
<li class="chapter" data-level="10.4" data-path="10-GWAS.html"><a href="10-GWAS.html#analysing-gemma-bslmm-output"><i class="fa fa-check"></i><b>10.4</b> Analysing GEMMA BSLMM output</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="11-Appendix.html"><a href="11-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="11-Appendix.html"><a href="11-Appendix.html#mamba-installation-and-environment"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="11-Appendix.html"><a href="11-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>B</b> Jupyter-notebook</a></li>
<li class="chapter" data-level="C" data-path="11-Appendix.html"><a href="11-Appendix.html#loops"><i class="fa fa-check"></i><b>C</b> Using loops to process multiple samples</a>
<ul>
<li class="chapter" data-level="C.1" data-path="11-Appendix.html"><a href="11-Appendix.html#simple-loop-example"><i class="fa fa-check"></i><b>C.1</b> Simple loop example</a></li>
<li class="chapter" data-level="C.2" data-path="11-Appendix.html"><a href="11-Appendix.html#trimmomatic-loop-example"><i class="fa fa-check"></i><b>C.2</b> Trimmomatic loop example</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="11-Appendix.html"><a href="11-Appendix.html#an-overview-of-bayesian-statistics"><i class="fa fa-check"></i><b>D</b> An overview of Bayesian statistics</a></li>
<li class="chapter" data-level="E" data-path="11-Appendix.html"><a href="11-Appendix.html#additional-approaches-for-detecting-selection-and-for-gwas"><i class="fa fa-check"></i><b>E</b> Additional approaches for detecting selection and for GWAS</a>
<ul>
<li class="chapter" data-level="E.1" data-path="11-Appendix.html"><a href="11-Appendix.html#rehh"><i class="fa fa-check"></i><b>E.1</b> rehh</a></li>
<li class="chapter" data-level="E.2" data-path="11-Appendix.html"><a href="11-Appendix.html#gcta"><i class="fa fa-check"></i><b>E.2</b> GCTA</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Population Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="popstats" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Calculating population summary statistics using VCFtools<a href="08-Summary_stats.html#popstats" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Generating summary statistics of your dataset can be useful to help you get a feel for your data and complement the visual structure plots and PCAs created in the previous chapter. Depending on your research questions, differences in diversity among populations or the differentiation among them may be important and you may need to discuss these quantitatively.</p>
<center>
<img src="figures/statistics.png" style="width:150px" />
</center>
<p>We'll be using <a href="https://vcftools.github.io"><code>VCFtools</code></a> to generate a summaries of nucleotide diversity, <em>F<sub>ST</sub></em> and Tajima's <em>D</em> using the vcf (variant called format) file we were working with previously. To save time, we will only use variants from one scaffold. Like <code>BCFtools</code>, which we used to manipulate the binary version of the vcf in Chapter 6, <code>VCFtools</code> is widely used to manipulate, filter and query vcf files, and also offers a set of rudimentary analyses. There are many other tools and applications available to generate diversity and differentiation summary statistics, such as <a href="https://www.popgen.dk/angsd/index.php/ANGSD">ANGSD</a>, <a href="http://www.ub.edu/dnasp/">DnaSP</a>, <a href="http://cmpg.unibe.ch/software/arlequin35/">Arlequin</a>, <a href="https://www.megasoftware.net">MEGA</a>, <a href="https://github.com/ksamuk/pixy_analysis">Pixy</a> and <a href="https://biology-assets.anu.edu.au/GenAlEx/Welcome.html">GenAlEx</a>, <strong>but</strong> (1) many will require you to convert your data to another format, (2) some may struggle with large popgenomic data sets, and (3) several are desktop applications, which may not easily work on an HPC, and your local machine may struggle with the computational demands of large data sets. For simplicity, we'll use <code>VCFtools</code> and work with the vcf file directly.</p>
<p>We will now generate a couple of commonly-used population genomic statistics, nucleotide diversity (<em>π</em>), <em>F<sub>ST</sub></em> and Tajima's <em>D</em>.</p>
<p><strong><em>π</em></strong></p>
<p>Nucleotide diversity (<em>π</em>) quantifies the mean divergence between sequences and reflects the extent to which sequences differ from each other. <em>π</em> is the average pairwise difference between all possible pairs of individuals in your sample.</p>
<p><strong>Equation</strong>:
<span class="math display">\[
\pi =
\sum f_i f_jp_{ij}
\]</span></p>
<ul>
<li><em>f<sub>i</sub></em> and <em>f<sub>i</sub></em> frequencies of the _i_th and _j_th haplotypes in the population</li>
<li><em>p<sub>ij</sub></em> sequence divergence between haplotypes</li>
</ul>
<p><strong><em>F<sub>ST</sub></em></strong></p>
<p><em>F</em>-statistics are one of the of the most common methods for quantifying genetic differentiation between subpopulations. <em>F<sub>ST</sub></em> (fixation index) estimates genetic differentiation between subpopulations. It reflects the degree of inbreeding within a subpopulation relative to the total population, as the probability that two alleles drawn at random from a subpopulation are identical by descent. <em>F<sub>ST</sub></em> is bound between 0 and 1. If <em>F<sub>ST</sub></em> = 0, there is no genetic differentiation between populations. If <em>F<sub>ST</sub></em> = 1, the two populations are very strongly separated.</p>
<p><strong>Equation</strong>:
<span class="math display">\[
F _{ST} =
\frac{(H_T - H_S)} {H_S}
\]</span></p>
<ul>
<li><em>H<sub>T</sub></em> expected heterozygosity of the total population</li>
<li><em>H<sub>S</sub></em> expected heterozygosity if the subpopulation were in Hardy–Weinberg Equilibrium</li>
</ul>
<p><strong>Tajima's <em>D</em></strong></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1203831/pdf/ge1233585.pdf">Tajima's (1989)</a> <em>D</em>-statistic is used to determine whether sequences (and populations) are evolving randomly under neutral expectations or whether they are under directional or balancing selection. The statistic itself is the relationship between two genetic measures of genetic diversity from pairwise comparisons of sequences: the number of pairwise differences between sequences and the number of possibly segregating or mutating sites. These should be scaled and should be the same under neutral scenarios. When there are deviations in the relationship between these components in a population, or variation among estimates of <em>D</em> for different regions in the genome, this can be taken as evidence of selection in the population or genomic region. Alternatively, if we have reason to believe that our sequence data are neutral, Tajima's <em>D</em> is often used to evaluate demographic changes, such as population bottlenecks or demographic expansion.</p>
<div id="getting-started" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Getting started<a href="08-Summary_stats.html#getting-started" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/genome_pop.png" style="width:200px" />
</center>
<p>Save, close and halt your Jupyter notebook session from the last chapter. When you are back in the terminal window session, you should still be in the 'structure' directory. Navigate to the 'sum_stats' directory.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb69-1"><a href="08-Summary_stats.html#cb69-1" tabindex="-1"></a>cd ..<span class="sc">/</span>sum_stats</span></code></pre></div>
<p>List the files in the directory to confirm that the data are there.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb70-1"><a href="08-Summary_stats.html#cb70-1" tabindex="-1"></a>ls</span></code></pre></div>
<p>You should see a single vcf file: NW_004438430.1.vcf. This contains the 8,289 SNPs called from the scaffold NW_004438430, which is itself 17,494,547 bases long.</p>
<p>We'll first calculate nucleotide diversity (<em>π</em>) across all 16 samples included in the vcf file.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb71-1"><a href="08-Summary_stats.html#cb71-1" tabindex="-1"></a>vcftools <span class="sc">--</span>vcf NW_004438430.<span class="fl">1.</span>vcf <span class="sc">--</span>window<span class="sc">-</span>pi <span class="dv">17494547</span> <span class="sc">--</span>out all_samples</span></code></pre></div>
<p>In this code,</p>
<ul>
<li><code>--vcf</code> specifies our input data file and indicates that it is in vcf format.</li>
<li><code>--window-pi</code>: With this command, we are specifying that we want to calculate an average nucleotide diversity over a given window or bin. An alternative command (<code>sites-pi</code>) will calculate nucleotide diversity for each variant site, which, given that our data only includes variant sites (mostly biallelic SNPs), is not particularly useful or informative. Using a window or bin that can move along the scaffold or genome and calculate an average nucleotide diversity for that window is a handy way to examine how diversity is distributed or varies along the genome. In this case, we are specifying a window size of 17,494,547 in order to get an average nucleotide diversity for the full scaffold.</li>
<li><code>--out</code> specifies the prefix to be used for our output files.</li>
</ul>
<p>Once run, examine the output. You may see some warning messages which relate to the formatting of the header of the file. These can be ignored, the analysis should run successfully.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb72-1"><a href="08-Summary_stats.html#cb72-1" tabindex="-1"></a>less all_samples.windowed.pi</span></code></pre></div>
<ol style="list-style-type: decimal">
<li>What is the nucleotide diversity across all 16 samples?
<div id="radio_WQVCVFDJSG" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_WQVCVFDJSG" value="answer"></input> <span>0.000190465</span></label><label><input type="radio" autocomplete="off" name="radio_WQVCVFDJSG" value=""></input> <span>0.000202824</span></label><label><input type="radio" autocomplete="off" name="radio_WQVCVFDJSG" value=""></input> <span>0.000204345</span></label>
</div></li>
</ol>
<p>Having calculated nucleotide diversity for the full set of 16 samples, we will now calculate the same for two arbitrarily-defined populations of four samples each. For each of these we'll need to create a file to indicate which individuals belong to that population. In your own study, you will need to create a file for each population you wish to examine.</p>
<p>We'll use the text editor <code>nano</code> to compile these.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb73-1"><a href="08-Summary_stats.html#cb73-1" tabindex="-1"></a>nano</span></code></pre></div>
<p>This should open an "empty" text editor within your terminal. Enter the following text:</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb74-1"><a href="08-Summary_stats.html#cb74-1" tabindex="-1"></a>.<span class="sc">/</span>aligned<span class="sc">/</span>SRR8861576_paired.bam</span>
<span id="cb74-2"><a href="08-Summary_stats.html#cb74-2" tabindex="-1"></a>.<span class="sc">/</span>aligned<span class="sc">/</span>SRR8861581_paired.bam</span>
<span id="cb74-3"><a href="08-Summary_stats.html#cb74-3" tabindex="-1"></a>.<span class="sc">/</span>aligned<span class="sc">/</span>SRR8861586_paired.bam</span>
<span id="cb74-4"><a href="08-Summary_stats.html#cb74-4" tabindex="-1"></a>.<span class="sc">/</span>aligned<span class="sc">/</span>SRR8861592_paired.bam</span></code></pre></div>
<p>Note that, although these look like paths to the .bam files, we not actually creating any paths to data files. This is how the samples are named in the vcf file, resulting from the way that we called the variants with <code>bwa mpileup</code> before.</p>
<p>We can save the newly-created file, by
(1) <code>CTRL + X</code> (to exit <code>nano</code>),
(2) <code>Y</code> (confirming that we wish to save our changes), and
(3) Giving the file the name 'population1'</p>
<p>We can create the file for the second population by editing the one we've just
created.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb75-1"><a href="08-Summary_stats.html#cb75-1" tabindex="-1"></a>nano population1</span></code></pre></div>
<p>Edit the existing file to include the following samples:</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb76-1"><a href="08-Summary_stats.html#cb76-1" tabindex="-1"></a>.<span class="sc">/</span>aligned<span class="sc">/</span>SRR8861578_paired.bam</span>
<span id="cb76-2"><a href="08-Summary_stats.html#cb76-2" tabindex="-1"></a>.<span class="sc">/</span>aligned<span class="sc">/</span>SRR8861583_paired.bam</span>
<span id="cb76-3"><a href="08-Summary_stats.html#cb76-3" tabindex="-1"></a>.<span class="sc">/</span>aligned<span class="sc">/</span>SRR8861589_paired.bam</span>
<span id="cb76-4"><a href="08-Summary_stats.html#cb76-4" tabindex="-1"></a>.<span class="sc">/</span>aligned<span class="sc">/</span>SRR8861591_paired.bam</span></code></pre></div>
<p>Rename this file as 'population2' when saving it. At the final step, you will be asked if you wish to save it under a different name. Press <code>Y</code>.</p>
<p>We can now calculate nucleotide diversity for our first population.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb77-1"><a href="08-Summary_stats.html#cb77-1" tabindex="-1"></a>vcftools <span class="sc">--</span>vcf NW_004438430.<span class="fl">1.</span>vcf <span class="sc">--</span>keep population1 <span class="sc">--</span>window<span class="sc">-</span>pi <span class="dv">17494547</span> \</span>
<span id="cb77-2"><a href="08-Summary_stats.html#cb77-2" tabindex="-1"></a><span class="sc">--</span>out population1</span></code></pre></div>
<p>The only additional component in this code is <code>--keep</code> which specifies that we only want to include those samples listed in the 'population1' file.</p>
<p>View the 'population1.windowed.pi' fill to see the nucleotide diversity for this population. Note that only 7,628 SNPs are variable in this population (opposed to the 8,289 in the full sample set):</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb78-1"><a href="08-Summary_stats.html#cb78-1" tabindex="-1"></a>less population1.windowed.pi</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>What is the nucleotide diversity for population 1?
<div id="radio_XCPTHLUTWD" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_XCPTHLUTWD" value=""></input> <span>0.000190465</span></label><label><input type="radio" autocomplete="off" name="radio_XCPTHLUTWD" value=""></input> <span>0.000202824</span></label><label><input type="radio" autocomplete="off" name="radio_XCPTHLUTWD" value="answer"></input> <span>0.000204345</span></label>
</div></li>
</ol>
<p>We can now do the same for the second population. <strong>Tip:</strong> Use your Linux history and overwrite the '1' in the file names.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb79-1"><a href="08-Summary_stats.html#cb79-1" tabindex="-1"></a>vcftools <span class="sc">--</span>vcf NW_004438430.<span class="fl">1.</span>vcf <span class="sc">--</span>keep population2 <span class="sc">--</span>window<span class="sc">-</span>pi <span class="dv">17494547</span> \</span>
<span id="cb79-2"><a href="08-Summary_stats.html#cb79-2" tabindex="-1"></a><span class="sc">--</span>out population2</span></code></pre></div>
<div class="sourceCode" id="cb80"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb80-1"><a href="08-Summary_stats.html#cb80-1" tabindex="-1"></a>less population2.windowed.pi</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>How many variable SNPs are in population 2?
<div id="radio_DSCQOWAVCX" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_DSCQOWAVCX" value="answer"></input> <span>7,587</span></label><label><input type="radio" autocomplete="off" name="radio_DSCQOWAVCX" value=""></input> <span>7,628</span></label><label><input type="radio" autocomplete="off" name="radio_DSCQOWAVCX" value=""></input> <span>8,289</span></label>
</div></li>
</ol>
<p>Unfortunately, nucleotide diversities need to be calculated for each population individually. If you have a large number of populations, you could use a loop to cycle through each of your populations and calculate the diversity. The appendix of this workbook introduces the use of loops.</p>
<p>We will now proceed to examine differentiation among our two hypothetical populations, by calculating the <em>F<sub>ST</sub></em>-value for their comparison. In this case, it is <a href="https://www.researchgate.net/publication/200102299_Weir_BS_Cockerham_CC_Estimating_F-Statistics_for_the_Analysis_of_Population-Structure_Evolution_38_1358-1370">Weir &amp; Cockerham's (1984)</a> <em>F<sub>ST</sub></em>-estimate. By default, <em>F<sub>ST</sub></em>-values are calculated on a per site basis, and there is no need to define a window or bin, as before, to get the estimate for the full scaffold. Of course, it is possible to define bins if you wanted to get <em>F<sub>ST</sub></em>-values for different sections of your scaffold or genome (as one would do to look for islands of divergence or outlier regions).</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb81-1"><a href="08-Summary_stats.html#cb81-1" tabindex="-1"></a>vcftools <span class="sc">--</span>vcf NW_004438430.<span class="fl">1.</span>vcf <span class="sc">--</span>weir<span class="sc">-</span>fst<span class="sc">-</span>pop population1 \</span>
<span id="cb81-2"><a href="08-Summary_stats.html#cb81-2" tabindex="-1"></a><span class="sc">--</span>weir<span class="sc">-</span>fst<span class="sc">-</span>pop population2 <span class="sc">--</span>out pop1_vs_pop2</span></code></pre></div>
<p>In this command,</p>
<ul>
<li><code>--vcf</code> and <code>--out</code>, respectively, specify the input data file and the prefix for the output files, as above.</li>
<li><code>--weir-fst-pop</code> is used to specify, one at a time, the populations files that we wish to compare, with the argument being the file we created to indicate which individuals belong to the population. This command can be repeated as many times as you have populations to add to the matrix. If you were interested in performing the calculations over bins/windows or a given number of steps, we would specify these with <code>--fst-window-size</code> or <code>--fst-window-step</code>.</li>
</ul>
<p>Once the command is run, the <em>F<sub>ST</sub></em>-values (the mean value and a weighted value) are presented in the terminal. We can see that there is zero to negligible differentiation among our two hypothetical populations. If we are interested in the <em>F<sub>ST</sub></em>-estimates of differentiation among the two populations for each SNP, we can find these by viewing the '.weir.fst' output file.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb82-1"><a href="08-Summary_stats.html#cb82-1" tabindex="-1"></a>less pop1_vs_pop2.weir.fst</span></code></pre></div>
<p>Lastly, we will calculate and compare Tajima's <em>D</em> for various bins along the scaffold. We'll include all 16 individuals. Of course, you could also do this for individual populations (with the <code>--keep</code> command), or specify the entire length of the scaffold as your bin or window if you want an overall estimate to compare among populations. With the scaffold being 17,494,547 bases, we'll specify a window size of 1.8 million bases, which would give us ten bins (nine complete and one partial bin) for this scaffold. We specify this with the <code>--TajimaD</code> command in the code we will run below.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb83-1"><a href="08-Summary_stats.html#cb83-1" tabindex="-1"></a>vcftools <span class="sc">--</span>vcf NW_004438430.<span class="fl">1.</span>vcf <span class="sc">--</span>TajimaD <span class="dv">1800000</span> <span class="sc">--</span>out all_tajima</span></code></pre></div>
<p>We can now examine the results which are in a file with the '.Tajima.D' extension.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb84-1"><a href="08-Summary_stats.html#cb84-1" tabindex="-1"></a>less all_tajima.Tajima.D</span></code></pre></div>
<p>From this, we can see that the diversity is not evenly distributed along the scaffold. We can also see that Tajima's <em>D</em>-estimates are not similar across the bins.</p>
<ol start="4" style="list-style-type: decimal">
<li>Which bin (identified by the starting position) has the most SNPs?
<div id="radio_DRPQUJXQOO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_DRPQUJXQOO" value=""></input> <span>1,800,000</span></label><label><input type="radio" autocomplete="off" name="radio_DRPQUJXQOO" value=""></input> <span>12,600,000</span></label><label><input type="radio" autocomplete="off" name="radio_DRPQUJXQOO" value="answer"></input> <span>16,200,000</span></label>
</div></li>
<li>Which bin has the highest estimate of D?
<div id="radio_ZZZBRYAMIQ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ZZZBRYAMIQ" value="answer"></input> <span>0</span></label><label><input type="radio" autocomplete="off" name="radio_ZZZBRYAMIQ" value=""></input> <span>5,400,000</span></label><label><input type="radio" autocomplete="off" name="radio_ZZZBRYAMIQ" value=""></input> <span>16,200,000</span></label>
</div></li>
<li>What is the lowest value of Tajima's <em>D</em> observed?
<div id="radio_LXNNOSIGJW" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_LXNNOSIGJW" value=""></input> <span>0.05721</span></label><label><input type="radio" autocomplete="off" name="radio_LXNNOSIGJW" value="answer"></input> <span>0.755336</span></label><label><input type="radio" autocomplete="off" name="radio_LXNNOSIGJW" value=""></input> <span>1.69975</span></label>
</div></li>
</ol>
<p>In interpreting these values, we generally consider values greater than 2 or less than -2 to indicate that the region may be under selection, whereas values between -2 and 2 indicate that the region (or population) conforms to neutral expectations. Along this scaffold, it does seem that there are regions that are under greater selective pressure than others, and this can probably prompt some further genome scanning or outlier analyses to identify islands of divergence or conservation.</p>
<p>Summary statistics can give you an overall feel for your data set, but the reality is that diversity varies throughout the genome. Rather than consider divergence or diversity across all ones data, it is often more appropriate and interesting to examine how different regions of the genome have diverged between populations. We will explore one method for estimating differentiation between a pair of populations in the next chapter.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="07-NGSAdmix.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="09-Detecting_selection.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
