<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Genetic architecture of traits using multi-locus Genome Wide Association (GWA) mapping with GEMMA | Population Genomics</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Genetic architecture of traits using multi-locus Genome Wide Association (GWA) mapping with GEMMA | Population Genomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Genetic architecture of traits using multi-locus Genome Wide Association (GWA) mapping with GEMMA | Population Genomics" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Helen Hipperson, Katy Maher, Victor Soria-Carrasco, Graeme Fox, Gavin Gouws" />


<meta name="date" content="2025-03-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="09-Detecting_selection.html"/>
<link rel="next" href="11-Appendix.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_rounded_corners.png" width="250"></a></li>
<li><a>Population Genomics</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-whole-genome-resequencing-of-killer-whales"><i class="fa fa-check"></i><b>2.1</b> Example data: Whole-genome resequencing of killer whales</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="part"><span><b>Generating SNP datasets</b></span></li>
<li class="chapter" data-level="4" data-path="04-QC.html"><a href="04-QC.html"><i class="fa fa-check"></i><b>4</b> QC Tutorials</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC.html"><a href="04-QC.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-QC.html"><a href="04-QC.html#killer-whale-genomic-dataset"><i class="fa fa-check"></i><b>4.2</b> Killer whale genomic dataset</a></li>
<li class="chapter" data-level="4.3" data-path="04-QC.html"><a href="04-QC.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-QC.html"><a href="04-QC.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality control</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="04-QC.html"><a href="04-QC.html#post-quality-control-check"><i class="fa fa-check"></i><b>4.4.1</b> Post Quality control check</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Mapping.html"><a href="05-Mapping.html"><i class="fa fa-check"></i><b>5</b> Mapping sequences to the reference genome using BWA</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Mapping.html"><a href="05-Mapping.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>5.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="5.2" data-path="05-Mapping.html"><a href="05-Mapping.html#read-mapping"><i class="fa fa-check"></i><b>5.2</b> Read mapping</a></li>
<li class="chapter" data-level="5.3" data-path="05-Mapping.html"><a href="05-Mapping.html#assessing-mapping-results"><i class="fa fa-check"></i><b>5.3</b> Assessing mapping results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-SNPs.html"><a href="06-SNPs.html"><i class="fa fa-check"></i><b>6</b> SNP and genotype calling using SAMtools and BCFtools</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-SNPs.html"><a href="06-SNPs.html#file-preparation"><i class="fa fa-check"></i><b>6.1</b> File preparation</a></li>
<li class="chapter" data-level="6.2" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-calling"><i class="fa fa-check"></i><b>6.2</b> Variant calling</a></li>
<li class="chapter" data-level="6.3" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-filtering"><i class="fa fa-check"></i><b>6.3</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-SNPs.html"><a href="06-SNPs.html#extracting-information"><i class="fa fa-check"></i><b>6.3.1</b> Extracting information</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-SNPs.html"><a href="06-SNPs.html#subsetting-files"><i class="fa fa-check"></i><b>6.3.2</b> Subsetting files</a></li>
<li class="chapter" data-level="6.3.3" data-path="06-SNPs.html"><a href="06-SNPs.html#filtering"><i class="fa fa-check"></i><b>6.3.3</b> Filtering</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="7" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html"><i class="fa fa-check"></i><b>7</b> Population structure with NGSadmix and PCA</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#ngsadmix"><i class="fa fa-check"></i><b>7.1</b> NGSadmix</a></li>
<li class="chapter" data-level="7.2" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#jupyter"><i class="fa fa-check"></i><b>7.2</b> Jupyter</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>7.2.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="7.2.2" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#create-r-notebook"><i class="fa fa-check"></i><b>7.2.2</b> Create R notebook</a></li>
<li class="chapter" data-level="7.2.3" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#cells-and-code"><i class="fa fa-check"></i><b>7.2.3</b> Cells and code</a></li>
<li class="chapter" data-level="7.2.4" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#create-new-cells"><i class="fa fa-check"></i><b>7.2.4</b> Create new cells</a></li>
<li class="chapter" data-level="7.2.5" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#running-code"><i class="fa fa-check"></i><b>7.2.5</b> Running code</a></li>
<li class="chapter" data-level="7.2.6" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#saving-the-file"><i class="fa fa-check"></i><b>7.2.6</b> Saving the file</a></li>
<li class="chapter" data-level="7.2.7" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>7.2.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="7.2.8" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#close-the-notebook"><i class="fa fa-check"></i><b>7.2.8</b> Close the notebook</a></li>
<li class="chapter" data-level="7.2.9" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#video-tutorial"><i class="fa fa-check"></i><b>7.2.9</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#admixture-plots"><i class="fa fa-check"></i><b>7.3</b> Admixture plots</a></li>
<li class="chapter" data-level="7.4" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#pca"><i class="fa fa-check"></i><b>7.4</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html"><i class="fa fa-check"></i><b>8</b> Calculating population summary statistics using VCFtools</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html#getting-started"><i class="fa fa-check"></i><b>8.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html"><i class="fa fa-check"></i><b>9</b> Delimitation of contiguous regions of differentiation using Hidden Markov Models (HMM)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#initial-set-up"><i class="fa fa-check"></i><b>9.1</b> Initial set up</a></li>
<li class="chapter" data-level="9.2" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#data-formatting"><i class="fa fa-check"></i><b>9.2</b> Data formatting</a></li>
<li class="chapter" data-level="9.3" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#allele-frequency-estimation"><i class="fa fa-check"></i><b>9.3</b> Allele frequency estimation</a></li>
<li class="chapter" data-level="9.4" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#fst-estimation"><i class="fa fa-check"></i><b>9.4</b> F<sub>ST</sub> estimation</a></li>
<li class="chapter" data-level="9.5" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#delimitation-of-contiguous-regions-of-differentiation-using-a-hmm-model"><i class="fa fa-check"></i><b>9.5</b> Delimitation of contiguous regions of differentiation using a HMM model</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-GWAS.html"><a href="10-GWAS.html"><i class="fa fa-check"></i><b>10</b> Genetic architecture of traits using multi-locus Genome Wide Association (GWA) mapping with GEMMA</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-GWAS.html"><a href="10-GWAS.html#initial-set-up-1"><i class="fa fa-check"></i><b>10.1</b> Initial set up</a></li>
<li class="chapter" data-level="10.2" data-path="10-GWAS.html"><a href="10-GWAS.html#data-formatting-1"><i class="fa fa-check"></i><b>10.2</b> Data formatting</a></li>
<li class="chapter" data-level="10.3" data-path="10-GWAS.html"><a href="10-GWAS.html#running-gemma-to-fit-a-bayesian-sparse-linear-mixed-model-bslmm"><i class="fa fa-check"></i><b>10.3</b> Running GEMMA to fit a Bayesian sparse linear mixed model (BSLMM)</a></li>
<li class="chapter" data-level="10.4" data-path="10-GWAS.html"><a href="10-GWAS.html#analysing-gemma-bslmm-output"><i class="fa fa-check"></i><b>10.4</b> Analysing GEMMA BSLMM output</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="11-Appendix.html"><a href="11-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="11-Appendix.html"><a href="11-Appendix.html#mamba-installation-and-environment"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="11-Appendix.html"><a href="11-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>B</b> Jupyter-notebook</a></li>
<li class="chapter" data-level="C" data-path="11-Appendix.html"><a href="11-Appendix.html#loops"><i class="fa fa-check"></i><b>C</b> Using loops to process multiple samples</a>
<ul>
<li class="chapter" data-level="C.1" data-path="11-Appendix.html"><a href="11-Appendix.html#simple-loop-example"><i class="fa fa-check"></i><b>C.1</b> Simple loop example</a></li>
<li class="chapter" data-level="C.2" data-path="11-Appendix.html"><a href="11-Appendix.html#trimmomatic-loop-example"><i class="fa fa-check"></i><b>C.2</b> Trimmomatic loop example</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="11-Appendix.html"><a href="11-Appendix.html#an-overview-of-bayesian-statistics"><i class="fa fa-check"></i><b>D</b> An overview of Bayesian statistics</a></li>
<li class="chapter" data-level="E" data-path="11-Appendix.html"><a href="11-Appendix.html#additional-approaches-for-detecting-selection-and-for-gwas"><i class="fa fa-check"></i><b>E</b> Additional approaches for detecting selection and for GWAS</a>
<ul>
<li class="chapter" data-level="E.1" data-path="11-Appendix.html"><a href="11-Appendix.html#rehh"><i class="fa fa-check"></i><b>E.1</b> rehh</a></li>
<li class="chapter" data-level="E.2" data-path="11-Appendix.html"><a href="11-Appendix.html#gcta"><i class="fa fa-check"></i><b>E.2</b> GCTA</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Population Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Genetic architecture of traits using multi-locus Genome Wide Association (GWA) mapping with GEMMA<a href="10-GWAS.html#gwas" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The aim of this chapter is to use a multi-SNP method to study the genetic architecture of colour pattern phenotype (i.e. white stripe) in <em>Timema cristinae</em> stick insects. In particular, we will use the Bayesian Sparse Linear Mixed Model (BSLMM) model implemented in the program <code>GEMMA</code>, a piece of software that also allows taking into account any potential population structure. This model is a hybrid approach that allows inferring whether the genetic basis of a trait is highly polygenic or rather determined by a few loci (i.e. oligogenic), as well as identifying the loci that show the strongest association with the trait. We will use the dataset of <a href="https://www.cell.com/current-biology/fulltext/S0960-9822(15)00661-2">Comeault <em>et al. 2015</em></a>, which consists of 602 individuals sampled from a phenotypically variable population called FHA. We will infer the proportion of phenotypic variance explained by the genotypes, what fraction of such variance can be explained by a reduced number of major effect loci, and will identify what loci show the most significant association with this trait.</p>
<center>
<img src="figures/timema.png" style="width:800px; border-radius:15px; border:5px solid #333333; background:null" />
</center>
<p>Resources</p>
<ul>
<li><p><a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003264">Zhou <em>et al.</em> 2013</a> - Paper where the Bayesian Sparse Linear Mixed Model approach implemented in GEMMA is described.</p></li>
<li><p><a href="http://www.xzlab.org/software.html">Zhou Lab website</a> - You can obtain GEMMA and a few other interesting programs for GWAS here.</p></li>
<li><p><a href="http://www.xzlab.org/software/GEMMAmanual.pdf">GEMMA manual</a></p></li>
</ul>
<div id="initial-set-up-1" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Initial set up<a href="10-GWAS.html#initial-set-up-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First you will need move into the folder for this practical.</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb123-1"><a href="10-GWAS.html#cb123-1" tabindex="-1"></a><span class="bu">cd</span> ~/popgenomics/data/gwas_gemma</span></code></pre></div>
</div>
<div id="data-formatting-1" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Data formatting<a href="10-GWAS.html#data-formatting-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb124"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb124-1"><a href="10-GWAS.html#cb124-1" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span> data</span></code></pre></div>
<center>
<img src="figures/gwas_list_files.png" style="width:800px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<p>There is a vcf file containing single nucleotide polymorphisms (SNPs) from RAD data of 602 individuals from a single polymorphic population of <em>Timema cristinae</em> (population code FHA). You can have a look at the file content with the following commands:</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb125-1"><a href="10-GWAS.html#cb125-1" tabindex="-1"></a><span class="fu">zless</span> <span class="at">-S</span> data/fha.vcf.gz</span>
<span id="cb125-2"><a href="10-GWAS.html#cb125-2" tabindex="-1"></a><span class="co"># or with bcftools</span></span>
<span id="cb125-3"><a href="10-GWAS.html#cb125-3" tabindex="-1"></a><span class="ex">bcftools</span> view data/fha.vcf.gz <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span>
<span id="cb125-4"><a href="10-GWAS.html#cb125-4" tabindex="-1"></a><span class="co"># excluding long header</span></span>
<span id="cb125-5"><a href="10-GWAS.html#cb125-5" tabindex="-1"></a><span class="ex">bcftools</span> view <span class="at">-H</span> data/fha.vcf.gz <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p>There are two files containing the phenotypes of the same individuals in the same order as in the vcf file (NA when the phenotype is missing). These phenotypes encode the dorsal white stripe either as a continuous trait (standardised white stripe area; <code>fha.pheno</code>) or as a discrete binary trait (presence/absence of the stripe; <code>fha.pheno2</code>). We will be using the first one for this practical (but it would be a good exercise to repeat the analyses using the other one with the probit model using - <code>bslmm 3</code>).</p>
<p>You can have a look at the files content:</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb126-1"><a href="10-GWAS.html#cb126-1" tabindex="-1"></a><span class="fu">head</span> data/fha.pheno</span></code></pre></div>
<center>
<img src="figures/gwas_pheno.png" style="width:800px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<div class="sourceCode" id="cb127"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb127-1"><a href="10-GWAS.html#cb127-1" tabindex="-1"></a><span class="fu">head</span> data/fha.pheno2</span></code></pre></div>
<center>
<img src="figures/gwas_pheno2.png" style="width:800px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<p><code>GEMMA</code>, the program we will be using to carry out multi-variant GWA, needs two input files: one with the genotypes and another one with the phenotypes. Most GWA programs rely on called genotypes, but <code>GEMMA</code> can work with genotype probabilities, which allows incorporating genotype uncertainty in the analyses. <code>GEMMA</code> accepts a number of formats for genotypes, we are going to use the mean genotype format (based on the BIMBAM format), where genotypes are encoded as mean genotypes. A mean genotype is a value between 0 and 2 that can be interpreted as the minor allele dosage: 0 is homozygous for the major allele, 1 is a heterozygote, and 2 is a homozygote for the minor allele. Thus, intermediate values reflect the uncertainty in genotypes.</p>
<p>We are going to use the custom Perl script <code>bcf2bbgeno.pl</code> to calculate the mean genotypes. First, the script removes all the SNPs that have more than two alleles. This is common practice, because most of the models used for inference are based on biallelic variants. Then, the script calculates empirical posterior genotype probabilities from the genotype likelihoods in the vcf file under the assumption that the population is in Hardy-Weinberg equilibrium (HWE). Specifically, the script uses inferred allele frequencies to set HWE priors:</p>
<p><strong>Equation</strong>:
<span class="math display">\[
p(AA) =
p ^{2}; p(aa) =
(1-p)^{2}; p(Aa) =
2p(1-p)
\]</span>
<em>p</em> being the allele frequency of major/reference allele <em>A</em>. Genotype likelihoods are multiplied by these priors to obtain genotype posterior probabilities that are then encoded as mean genotypes and saved to a .bbgeno file.</p>
<p>You can get some info about how to run the Perl script:</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb128-1"><a href="10-GWAS.html#cb128-1" tabindex="-1"></a><span class="co"># show help</span></span>
<span id="cb128-2"><a href="10-GWAS.html#cb128-2" tabindex="-1"></a><span class="fu">perl</span> scripts/bcf2bbgeno.pl <span class="at">-h</span></span></code></pre></div>
<p>To calculate the genotype posterior probabilites and save them in mean genotype format, we would need to run a command like this one (followed by compression to save some space, given that <code>GEMMA</code> can handle gzipped files):</p>
<blockquote>
<p>perl scripts/bcf2bbgeno.pl -i data/fha.vcf.gz -o fha.bbgeno -p H-W -s -r</p>
</blockquote>
<blockquote>
<p>gzip fha.bbgeno</p>
</blockquote>
<p>However, this process may take quite a while. Therefore it would be better to use a batch job (see <code>bcf2bbgeno.sh</code>). Even this will take some time, so lets skip this step altogether for the purpose of this tutorial and use the .bbgeno.gz and .bbgeno.ids.txt files from the results folder. First cancel the command if you ran it (<code>Control</code>+<code>C</code>) and then copy the files needed from the results folder.</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb129-1"><a href="10-GWAS.html#cb129-1" tabindex="-1"></a><span class="fu">cp</span> results/<span class="pp">*</span>.bbgeno.<span class="pp">*</span> .</span></code></pre></div>
<p>You should now have two new files: fha.bbgeno.gz, which contains the mean genotypes, and fha.bbgeno.ids.txt, which contains the IDs of the samples (i.e. individuals) in the same order as shown in the genotypes file. You can have a look at them:</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb130-1"><a href="10-GWAS.html#cb130-1" tabindex="-1"></a><span class="fu">zless</span> <span class="at">-S</span> fha.bbgeno.gz</span></code></pre></div>
<center>
<img src="figures/gwas_genotypes.png" style="width:1000px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<p>The first column is the SNP ID, second one is major/reference allele and the third one is the minor/alternate allele. Having major and minor or reference and alternate alleles depends on how you called variants. This does not matter for downstream analyses with <code>GEMMA</code>, but should be kept in mind when interpreting the results. The rest of the columns are the mean genotypes for all the individuals (one for each).</p>
<p>The other file is simply to keep track of the order of the individuals in the genotype file:</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb131-1"><a href="10-GWAS.html#cb131-1" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> fha.bbgeno.ids.txt</span></code></pre></div>
<center>
<img src="figures/gwas_ids.png" style="width:200px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
</div>
<div id="running-gemma-to-fit-a-bayesian-sparse-linear-mixed-model-bslmm" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Running GEMMA to fit a Bayesian sparse linear mixed model (BSLMM)<a href="10-GWAS.html#running-gemma-to-fit-a-bayesian-sparse-linear-mixed-model-bslmm" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>GEMMA</code> (Genome-wide Efficient Mixed Model Association) is a complex piece of software with many options. It is highly recommended that you read the <a href="http://www.xzlab.org/software/GEMMAmanual.pdf">manual</a> carefully. Let’s have a look at the options, with emphasis on the specific ones we will be using.</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb132-1"><a href="10-GWAS.html#cb132-1" tabindex="-1"></a><span class="co"># general help</span></span>
<span id="cb132-2"><a href="10-GWAS.html#cb132-2" tabindex="-1"></a><span class="ex">gemma</span> <span class="at">-h</span></span>
<span id="cb132-3"><a href="10-GWAS.html#cb132-3" tabindex="-1"></a></span>
<span id="cb132-4"><a href="10-GWAS.html#cb132-4" tabindex="-1"></a><span class="co"># quick guide</span></span>
<span id="cb132-5"><a href="10-GWAS.html#cb132-5" tabindex="-1"></a><span class="ex">gemma</span> <span class="at">-h</span> 1</span>
<span id="cb132-6"><a href="10-GWAS.html#cb132-6" tabindex="-1"></a></span>
<span id="cb132-7"><a href="10-GWAS.html#cb132-7" tabindex="-1"></a><span class="co">#We will be using the Bayesian sparse linear mixed model (BSLMM):</span></span>
<span id="cb132-8"><a href="10-GWAS.html#cb132-8" tabindex="-1"></a><span class="ex">gemma</span> <span class="at">-h</span> 10</span>
<span id="cb132-9"><a href="10-GWAS.html#cb132-9" tabindex="-1"></a></span>
<span id="cb132-10"><a href="10-GWAS.html#cb132-10" tabindex="-1"></a><span class="co">#but first we will have to calculate the relatedness matrix to take into account population structure:</span></span>
<span id="cb132-11"><a href="10-GWAS.html#cb132-11" tabindex="-1"></a><span class="ex">gemma</span> <span class="at">-h</span> 4</span>
<span id="cb132-12"><a href="10-GWAS.html#cb132-12" tabindex="-1"></a></span>
<span id="cb132-13"><a href="10-GWAS.html#cb132-13" tabindex="-1"></a><span class="co">#and it would also be interesting do some filtering such as excluding rare variants, which are difficult to tell apart from sequencing errors:</span></span>
<span id="cb132-14"><a href="10-GWAS.html#cb132-14" tabindex="-1"></a><span class="ex">gemma</span> <span class="at">-h</span> 3</span></code></pre></div>
<p><code>GEMMA</code> uses a relatedness matrix to account for any population structure in the data that may affect results. The easiest way to obtain such matrix is using <code>GEMMA</code> itself. We are going to use the submission script <code>gemma_relmatrix.sh</code>. Let’s have a look at the script:</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb133-1"><a href="10-GWAS.html#cb133-1" tabindex="-1"></a><span class="fu">less</span> scripts/gemma_relmatrix.sh</span></code></pre></div>
<center>
<img src="figures/gemma_relmatrix.png" style="width:1000px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<p>Now let’s run the script (it should take just a few minutes):</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb134-1"><a href="10-GWAS.html#cb134-1" tabindex="-1"></a><span class="fu">chmod</span> +x scripts/gemma_relmatrix.sh</span>
<span id="cb134-2"><a href="10-GWAS.html#cb134-2" tabindex="-1"></a><span class="ex">scripts/gemma_relmatrix.sh</span></span></code></pre></div>
<p>When finished, it should have two files in the newly created output directory: relmatrix.log.txt, a log file, and relmatrix.cXX.txt, the actual relatedness matrix. If you have a look at them, they should look similar to these:</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb135-1"><a href="10-GWAS.html#cb135-1" tabindex="-1"></a><span class="fu">less</span> output/relmatrix.log.txt</span></code></pre></div>
<center>
<img src="figures/gwas_rel_log.png" style="width:1000px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<div class="sourceCode" id="cb136"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb136-1"><a href="10-GWAS.html#cb136-1" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> output/relmatrix.cXX.txt</span></code></pre></div>
<center>
<img src="figures/relmatrix.png" style="width:1000px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<p>Now we are going to run <code>GEMMA</code> to fit a BSLMM model. Since this will take a long time using the teaching cluster, we are not going to submit it now. The script has been designed to run as batch job <code>gemma_bslmm.sh</code>. <code>GEMMA</code> is run using the relatedness matrix we estimated before, using the BSLMM model with the continuous phenotypes and excluding variants with a minor allele frequency of less than 1%. We will run a MCMC chain for 1,000,000 steps, after an initial burn-in of 250,000, saving every 100th step, so that we will end up with 10,000 samples to approximate the posterior distribution of the parameters. Let’s have a look :</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb137-1"><a href="10-GWAS.html#cb137-1" tabindex="-1"></a><span class="fu">less</span> scripts/gemma_bslmm.sh</span></code></pre></div>
<center>
<img src="figures/gemma_script.png" style="width:1000px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<p>Do not run this now. However, if you were to submit the script you would do so using the following:</p>
<blockquote>
<p>scripts/gemma_bslmm.sh</p>
</blockquote>
<p>Copy the following files into the output folder:</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb138-1"><a href="10-GWAS.html#cb138-1" tabindex="-1"></a><span class="fu">cp</span>  results/output/bslmm<span class="pp">*</span> output/</span>
<span id="cb138-2"><a href="10-GWAS.html#cb138-2" tabindex="-1"></a></span>
<span id="cb138-3"><a href="10-GWAS.html#cb138-3" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span> output</span></code></pre></div>
<center>
<img src="figures/list_files.png" style="width:700px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<p>The new files generated are:</p>
<ul>
<li><p>bslmm.bv.txt -&gt; posterior samples of breeding values (~estimated random effects)</p></li>
<li><p>bslmm.gamma.txt -&gt; posterior samples for the gamma (i.e. sparse effects) listing the SNPs included in the model in each iteration, that is the loci with detectable (large) effects</p></li>
<li><p>bslmm.hyp.txt -&gt; posterior samples of hyper-parameters, including the proportion of the variance explained (pve), the proportion of the genetic variance (i.e. pve) explained by the sparse effect loci (pge), the proportion of SNPs with non-zero effects (pi) and the estimated number of loci with large effects (n_gamma)</p></li>
<li><p>bslmm.log.txt -&gt; log file</p></li>
<li><p>bslmm.param.txt -&gt; posterior samples of parameters: random effects (alpha), fixed effects (beta), and sparse effects (gamma)</p></li>
</ul>
<p>Let’s have a look at the log. Notice the number of individuals with both genotypic and phenotypic data is 546 out of 602 and the number of SNPs was reduced from 518,232 to 346,660 after excluding SNPs missing genotypes for over 5% individuals (default threshold) and rare variants (minor allele frequency of less than 1%):</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb139-1"><a href="10-GWAS.html#cb139-1" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> output/bslmm.log.txt</span></code></pre></div>
<center>
<img src="figures/bslmm_log.png" style="width:1000px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<p>the hyperparameters:</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb140-1"><a href="10-GWAS.html#cb140-1" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> output/bslmm.hyp.txt</span></code></pre></div>
<center>
<img src="figures/hyp.png" style="width:1000px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<ul>
<li>h - approximation to PVE: proportion of phenotypic variance explained by loci</li>
<li>pve - proportion of phenotypic variance explained by the genotypes</li>
<li>rho - approximation to PGE: proportion of genetic variance explained by sparse effect terms (~major effect loci) (rho=0 =&gt; pure LMM, highly polygenic; rho=1 =&gt; pure BVSR, few loci)</li>
<li>pge - proportion of genetic variance explained by major effect loci</li>
<li>pi - proportion of variants with non-zero effects</li>
<li>n_gamma - number of variants with major effect</li>
</ul>
<p>and the parameters (notice we didn’t specify chromosomes (chr) or positions (pos) and therefore there are only SNP IDs (rs column):</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb141-1"><a href="10-GWAS.html#cb141-1" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> output/bslmm.param.txt</span></code></pre></div>
<center>
<img src="figures/param.png" style="width:1000px; border-radius:5px; border:5px solid #333333; background:null" />
</center>
<p>For more information please see the <code>GEMMA</code> <a href="http://www.xzlab.org/software/GEMMAmanual.pdf">manual</a></p>
</div>
<div id="analysing-gemma-bslmm-output" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Analysing GEMMA BSLMM output<a href="10-GWAS.html#analysing-gemma-bslmm-output" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will now make some plots to help us to visualise the results from the hyperparamter and paramter files.</p>
<p>Create a new Jupyter notebook called <strong>"Chp10-GWAS"</strong>.</p>
<p>First, we are going to play with the hyperparameter estimates, which will inform us of the genetic architecture of the trait.</p>
<p>First we change the working directory:</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb142-1"><a href="10-GWAS.html#cb142-1" tabindex="-1"></a><span class="ex">setwd</span><span class="er">(</span><span class="st">&quot;~/popgenomics/data/gwas_gemma/output&quot;</span><span class="kw">)</span></span></code></pre></div>
<p>And then load the hyperparameter file:</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb143-1"><a href="10-GWAS.html#cb143-1" tabindex="-1"></a><span class="co"># Load hyperparameter file</span></span>
<span id="cb143-2"><a href="10-GWAS.html#cb143-2" tabindex="-1"></a><span class="co"># ==============================================================================</span></span>
<span id="cb143-3"><a href="10-GWAS.html#cb143-3" tabindex="-1"></a><span class="ex">hyp.params</span><span class="op">&lt;</span>-read.table<span class="er">(</span><span class="st">&quot;bslmm.hyp.txt&quot;</span><span class="ex">,header=T</span><span class="kw">)</span></span>
<span id="cb143-4"><a href="10-GWAS.html#cb143-4" tabindex="-1"></a><span class="co"># ==============================================================================</span></span></code></pre></div>
<p>Let’s get some statistics (mean, median, and 95% ETPI - <a href="https://cran.r-project.org/web/packages/bayestestR/vignettes/credible_interval.html">equal-tail posterior probability intervals</a>):</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb144-1"><a href="10-GWAS.html#cb144-1" tabindex="-1"></a><span class="co"># Get mean, median, and 95% ETPI of hyperparameters</span></span>
<span id="cb144-2"><a href="10-GWAS.html#cb144-2" tabindex="-1"></a><span class="co"># ==============================================================================</span></span>
<span id="cb144-3"><a href="10-GWAS.html#cb144-3" tabindex="-1"></a><span class="co"># pve -&gt; proportion of phenotypic variance explained by the genotypes</span></span>
<span id="cb144-4"><a href="10-GWAS.html#cb144-4" tabindex="-1"></a><span class="ex">pve</span><span class="op">&lt;</span>-c<span class="er">(</span><span class="st">&quot;PVE&quot;</span><span class="ex">,</span> mean<span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pve</span><span class="kw">)</span><span class="ex">,quantile</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pve</span><span class="ex">,</span> probs=c<span class="er">(</span><span class="ex">0.5,0.025,0.975</span><span class="kw">)))</span></span>
<span id="cb144-5"><a href="10-GWAS.html#cb144-5" tabindex="-1"></a></span>
<span id="cb144-6"><a href="10-GWAS.html#cb144-6" tabindex="-1"></a><span class="co"># pge -&gt; proportion of genetic variance explained by major effect loci</span></span>
<span id="cb144-7"><a href="10-GWAS.html#cb144-7" tabindex="-1"></a><span class="ex">pge</span><span class="op">&lt;</span>-c<span class="er">(</span><span class="st">&quot;PGE&quot;</span><span class="ex">,mean</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pge</span><span class="kw">)</span><span class="ex">,quantile</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pge</span><span class="ex">,</span> probs=c<span class="er">(</span><span class="ex">0.5,0.025,0.975</span><span class="kw">)))</span></span>
<span id="cb144-8"><a href="10-GWAS.html#cb144-8" tabindex="-1"></a></span>
<span id="cb144-9"><a href="10-GWAS.html#cb144-9" tabindex="-1"></a><span class="co"># pi -&gt; proportion of variants with non-zero effects</span></span>
<span id="cb144-10"><a href="10-GWAS.html#cb144-10" tabindex="-1"></a><span class="ex">pi</span><span class="op">&lt;</span>-c<span class="er">(</span><span class="st">&quot;pi&quot;</span><span class="ex">,mean</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pi</span><span class="kw">)</span><span class="ex">,quantile</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pi</span><span class="ex">,</span> probs=c<span class="er">(</span><span class="ex">0.5,0.025,0.975</span><span class="kw">)))</span></span>
<span id="cb144-11"><a href="10-GWAS.html#cb144-11" tabindex="-1"></a></span>
<span id="cb144-12"><a href="10-GWAS.html#cb144-12" tabindex="-1"></a><span class="co"># n.gamma -&gt; number of variants with major effect</span></span>
<span id="cb144-13"><a href="10-GWAS.html#cb144-13" tabindex="-1"></a><span class="ex">n.gamma</span><span class="op">&lt;</span>-c<span class="er">(</span><span class="st">&quot;n.gamma&quot;</span><span class="ex">,mean</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$n_gamma</span><span class="kw">)</span><span class="ex">,quantile</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$n_gamma</span><span class="ex">,</span> probs=c<span class="er">(</span><span class="ex">0.5,0.025,0.975</span><span class="kw">)))</span></span>
<span id="cb144-14"><a href="10-GWAS.html#cb144-14" tabindex="-1"></a><span class="co"># ==============================================================================</span></span>
<span id="cb144-15"><a href="10-GWAS.html#cb144-15" tabindex="-1"></a></span>
<span id="cb144-16"><a href="10-GWAS.html#cb144-16" tabindex="-1"></a><span class="co"># get table of hyperparameters</span></span>
<span id="cb144-17"><a href="10-GWAS.html#cb144-17" tabindex="-1"></a><span class="co"># ==============================================================================</span></span>
<span id="cb144-18"><a href="10-GWAS.html#cb144-18" tabindex="-1"></a><span class="ex">hyp.params.table</span><span class="op">&lt;</span>-as.data.frame<span class="er">(</span><span class="ex">rbind</span><span class="er">(</span><span class="ex">pve,pge,pi,n.gamma</span><span class="kw">)</span><span class="ex">,row.names=F</span><span class="kw">)</span></span>
<span id="cb144-19"><a href="10-GWAS.html#cb144-19" tabindex="-1"></a><span class="ex">colnames</span><span class="er">(</span><span class="ex">hyp.params.table</span><span class="kw">)</span><span class="op">&lt;</span>-c<span class="kw">(</span><span class="st">&quot;hyperparam&quot;</span><span class="ex">,</span> <span class="st">&quot;mean&quot;</span>,<span class="st">&quot;median&quot;</span>,<span class="st">&quot;2.5%&quot;</span>, <span class="st">&quot;97.5%&quot;</span><span class="kw">)</span></span>
<span id="cb144-20"><a href="10-GWAS.html#cb144-20" tabindex="-1"></a><span class="co"># show table</span></span>
<span id="cb144-21"><a href="10-GWAS.html#cb144-21" tabindex="-1"></a><span class="ex">hyp.params.table</span></span>
<span id="cb144-22"><a href="10-GWAS.html#cb144-22" tabindex="-1"></a><span class="co"># write table to file</span></span>
<span id="cb144-23"><a href="10-GWAS.html#cb144-23" tabindex="-1"></a><span class="ex">write.table</span><span class="er">(</span><span class="ex">hyp.params.table,</span> file=<span class="st">&quot;hyperparameters.dsv&quot;</span>, sep=<span class="st">&quot;\t&quot;</span>, quote=F<span class="kw">)</span></span>
<span id="cb144-24"><a href="10-GWAS.html#cb144-24" tabindex="-1"></a><span class="co"># ==============================================================================</span></span></code></pre></div>
<p>Values should be similar to this:</p>
<div class="l-body-outset">
<table>
<colgroup>
<col width="30%" />
<col width="16%" />
<col width="21%" />
<col width="16%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th>hyperparam</th>
<th>mean</th>
<th>median</th>
<th>2.5%</th>
<th>97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PVE</td>
<td>0.50758534164</td>
<td>0.49616535</td>
<td>0.3290844075</td>
<td>0.753359795</td>
</tr>
<tr class="even">
<td>PGE</td>
<td>0.70427977221</td>
<td>0.70412075</td>
<td>0.4118065175</td>
<td>0.9789640325</td>
</tr>
<tr class="odd">
<td>pi</td>
<td>1.29564889986e-05</td>
<td>9.376132e-06</td>
<td>3.25441665e-06</td>
<td>4.3316011e-05</td>
</tr>
<tr class="even">
<td>n.gamma</td>
<td>4.574</td>
<td>4</td>
<td>1</td>
<td>14</td>
</tr>
</tbody>
</table>
</div>
<p>Now let’s plot the MCMC traces and the posterior distributions of the hyperparameters and save them into a pdf file.</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb145-1"><a href="10-GWAS.html#cb145-1" tabindex="-1"></a><span class="co"># plot traces and distributions of hyperparameters</span></span>
<span id="cb145-2"><a href="10-GWAS.html#cb145-2" tabindex="-1"></a><span class="co"># ==============================================================================</span></span>
<span id="cb145-3"><a href="10-GWAS.html#cb145-3" tabindex="-1"></a><span class="ex">layout</span><span class="er">(</span><span class="ex">matrix</span><span class="er">(</span><span class="ex">c</span><span class="er">(</span><span class="ex">1,1,2,3,4,4,5,6</span><span class="kw">)</span><span class="ex">,</span> 4, 2, byrow = TRUE<span class="kw">))</span></span>
<span id="cb145-4"><a href="10-GWAS.html#cb145-4" tabindex="-1"></a></span>
<span id="cb145-5"><a href="10-GWAS.html#cb145-5" tabindex="-1"></a><span class="co"># PVE</span></span>
<span id="cb145-6"><a href="10-GWAS.html#cb145-6" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb145-7"><a href="10-GWAS.html#cb145-7" tabindex="-1"></a><span class="ex">plot</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pve</span><span class="ex">,</span> type=<span class="st">&quot;l&quot;</span>, ylab=<span class="st">&quot;PVE&quot;</span>, main=<span class="st">&quot;PVE - trace&quot;</span><span class="kw">)</span></span>
<span id="cb145-8"><a href="10-GWAS.html#cb145-8" tabindex="-1"></a><span class="ex">hist</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pve</span><span class="ex">,</span> main=<span class="st">&quot;PVE - posterior distribution&quot;</span>, xlab=<span class="st">&quot;PVE&quot;</span><span class="kw">)</span></span>
<span id="cb145-9"><a href="10-GWAS.html#cb145-9" tabindex="-1"></a><span class="ex">plot</span><span class="er">(</span><span class="ex">density</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pve</span><span class="kw">)</span><span class="ex">,</span> main=<span class="st">&quot;PVE - posterior distribution&quot;</span>, xlab=<span class="st">&quot;PVE&quot;</span><span class="kw">)</span></span>
<span id="cb145-10"><a href="10-GWAS.html#cb145-10" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb145-11"><a href="10-GWAS.html#cb145-11" tabindex="-1"></a></span>
<span id="cb145-12"><a href="10-GWAS.html#cb145-12" tabindex="-1"></a><span class="co"># PGE</span></span>
<span id="cb145-13"><a href="10-GWAS.html#cb145-13" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb145-14"><a href="10-GWAS.html#cb145-14" tabindex="-1"></a><span class="ex">plot</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pge</span><span class="ex">,</span> type=<span class="st">&quot;l&quot;</span>, ylab=<span class="st">&quot;PGE&quot;</span>, main=<span class="st">&quot;PGE - trace&quot;</span><span class="kw">)</span></span>
<span id="cb145-15"><a href="10-GWAS.html#cb145-15" tabindex="-1"></a><span class="ex">hist</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pge</span><span class="ex">,</span> main=<span class="st">&quot;PGE - posterior distribution&quot;</span>, xlab=<span class="st">&quot;PGE&quot;</span><span class="kw">)</span></span>
<span id="cb145-16"><a href="10-GWAS.html#cb145-16" tabindex="-1"></a><span class="ex">plot</span><span class="er">(</span><span class="ex">density</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pge</span><span class="kw">)</span><span class="ex">,</span> main=<span class="st">&quot;PGE - posterior distribution&quot;</span>, xlab=<span class="st">&quot;PGE&quot;</span><span class="kw">)</span></span>
<span id="cb145-17"><a href="10-GWAS.html#cb145-17" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb145-18"><a href="10-GWAS.html#cb145-18" tabindex="-1"></a></span>
<span id="cb145-19"><a href="10-GWAS.html#cb145-19" tabindex="-1"></a><span class="co"># pi</span></span>
<span id="cb145-20"><a href="10-GWAS.html#cb145-20" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb145-21"><a href="10-GWAS.html#cb145-21" tabindex="-1"></a><span class="ex">plot</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pi</span><span class="ex">,</span> type=<span class="st">&quot;l&quot;</span>, ylab=<span class="st">&quot;pi&quot;</span>, main=<span class="st">&quot;pi&quot;</span><span class="kw">)</span></span>
<span id="cb145-22"><a href="10-GWAS.html#cb145-22" tabindex="-1"></a><span class="ex">hist</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pi</span><span class="ex">,</span> main=<span class="st">&quot;pi&quot;</span>, xlab=<span class="st">&quot;pi&quot;</span><span class="kw">)</span></span>
<span id="cb145-23"><a href="10-GWAS.html#cb145-23" tabindex="-1"></a><span class="ex">plot</span><span class="er">(</span><span class="ex">density</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pi</span><span class="kw">)</span><span class="ex">,</span> main=<span class="st">&quot;pi&quot;</span>, xlab=<span class="st">&quot;pi&quot;</span><span class="kw">)</span></span>
<span id="cb145-24"><a href="10-GWAS.html#cb145-24" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb145-25"><a href="10-GWAS.html#cb145-25" tabindex="-1"></a></span>
<span id="cb145-26"><a href="10-GWAS.html#cb145-26" tabindex="-1"></a><span class="co"># No gamma</span></span>
<span id="cb145-27"><a href="10-GWAS.html#cb145-27" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb145-28"><a href="10-GWAS.html#cb145-28" tabindex="-1"></a><span class="ex">plot</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$n_gamma</span><span class="ex">,</span> type=<span class="st">&quot;l&quot;</span>, ylab=<span class="st">&quot;n_gamma&quot;</span>, main=<span class="st">&quot;n_gamma - trace&quot;</span><span class="kw">)</span></span>
<span id="cb145-29"><a href="10-GWAS.html#cb145-29" tabindex="-1"></a><span class="ex">hist</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$n_gamma</span><span class="ex">,</span> main=<span class="st">&quot;n_gamma - posterior distribution&quot;</span>, xlab=<span class="st">&quot;n_gamma&quot;</span><span class="kw">)</span></span>
<span id="cb145-30"><a href="10-GWAS.html#cb145-30" tabindex="-1"></a><span class="ex">plot</span><span class="er">(</span><span class="ex">density</span><span class="er">(</span><span class="ex">hyp.params</span><span class="va">$pi</span><span class="kw">)</span><span class="ex">,</span> main=<span class="st">&quot;n_gamma - posterior distribution&quot;</span>, xlab=<span class="st">&quot;n_gamma&quot;</span><span class="kw">)</span></span>
<span id="cb145-31"><a href="10-GWAS.html#cb145-31" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb145-32"><a href="10-GWAS.html#cb145-32" tabindex="-1"></a><span class="co"># ==============================================================================</span></span></code></pre></div>
<p>Ideally, the MCMC traces should look like a caterpillar and distributions should be generally unimodal, indicating that mixing was good. Also it is highly recommended to run <code>GEMMA</code> multiple times (at least 3) and compare both the traces and the posterior distributions of the hyperparameters obtained from different runs and check that all are converging into the same values. However, we will not do this in this practical for time reasons. For instance, PVE should look like this:</p>
<center>
<img src="figures/pve.png" style="width:800px; border-radius:15px; border:5px solid #333333; background:null" />
</center>
<p>We can now look at the parameter estimates to see if there are any SNPs that are strongly and consistently associated with the trait.</p>
<p>We will load a library to help speed up the loading of large tables and we will then load the parameter estimates:</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb146-1"><a href="10-GWAS.html#cb146-1" tabindex="-1"></a><span class="co"># library to speed up loading of big tables</span></span>
<span id="cb146-2"><a href="10-GWAS.html#cb146-2" tabindex="-1"></a><span class="ex">library</span><span class="er">(</span><span class="ex">data.table</span><span class="kw">)</span></span>
<span id="cb146-3"><a href="10-GWAS.html#cb146-3" tabindex="-1"></a></span>
<span id="cb146-4"><a href="10-GWAS.html#cb146-4" tabindex="-1"></a><span class="co"># Load parameters output</span></span>
<span id="cb146-5"><a href="10-GWAS.html#cb146-5" tabindex="-1"></a><span class="co"># ==============================================================================</span></span>
<span id="cb146-6"><a href="10-GWAS.html#cb146-6" tabindex="-1"></a><span class="ex">params</span><span class="op">&lt;</span>-fread<span class="er">(</span><span class="st">&quot;bslmm.param.txt&quot;</span><span class="ex">,header=T,sep=</span><span class="st">&quot;\t&quot;</span><span class="ex">,</span> data.table=F<span class="kw">)</span></span>
<span id="cb146-7"><a href="10-GWAS.html#cb146-7" tabindex="-1"></a><span class="co"># ==============================================================================</span></span></code></pre></div>
<p>Now we will get the SNPs that have a sparse effect (i.e. detectable large effect) on the phenotype. In the table, beta is actually β|γ == 1 for a given SNP, β being the small effect all variants have and γ the additional large effect only some variants have. Thus, the sparse effect size for each SNP is calculated multiplying β by γ.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb147-1"><a href="10-GWAS.html#cb147-1" tabindex="-1"></a><span class="co"># Get variants with sparse effect size on phenotypes </span></span>
<span id="cb147-2"><a href="10-GWAS.html#cb147-2" tabindex="-1"></a><span class="co"># ==============================================================================</span></span>
<span id="cb147-3"><a href="10-GWAS.html#cb147-3" tabindex="-1"></a><span class="co"># add sparse effect size (= beta * gamma) to data frame</span></span>
<span id="cb147-4"><a href="10-GWAS.html#cb147-4" tabindex="-1"></a><span class="va">params</span><span class="op">[</span><span class="st">&quot;eff&quot;</span><span class="op">]&lt;</span>-abs<span class="kw">(</span><span class="ex">params</span><span class="va">$beta</span><span class="ex">*params</span><span class="va">$gamma</span><span class="kw">)</span></span></code></pre></div>
<p>Then we will extract the SNPs that have a detectable large effect size, sort them by effect size, and save the top SNPs using several thresholds (1%, 0.1%, 0.01%):</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb148-1"><a href="10-GWAS.html#cb148-1" tabindex="-1"></a><span class="co"># get variants with effect size &gt; 0</span></span>
<span id="cb148-2"><a href="10-GWAS.html#cb148-2" tabindex="-1"></a><span class="ex">params.effects</span><span class="op">&lt;</span>-params[params<span class="va">$eff</span><span class="op">&gt;</span>0,]</span>
<span id="cb148-3"><a href="10-GWAS.html#cb148-3" tabindex="-1"></a></span>
<span id="cb148-4"><a href="10-GWAS.html#cb148-4" tabindex="-1"></a><span class="co"># show number of variants with measurable effect</span></span>
<span id="cb148-5"><a href="10-GWAS.html#cb148-5" tabindex="-1"></a><span class="ex">nrow</span><span class="er">(</span><span class="ex">params.effects</span><span class="kw">)</span></span>
<span id="cb148-6"><a href="10-GWAS.html#cb148-6" tabindex="-1"></a></span>
<span id="cb148-7"><a href="10-GWAS.html#cb148-7" tabindex="-1"></a><span class="co"># sort by descending effect size</span></span>
<span id="cb148-8"><a href="10-GWAS.html#cb148-8" tabindex="-1"></a><span class="ex">params.effects.sort</span><span class="op">&lt;</span>-params.effects[order<span class="er">(</span><span class="ex">-params.effects</span><span class="va">$eff</span><span class="kw">)</span><span class="ex">,]</span></span>
<span id="cb148-9"><a href="10-GWAS.html#cb148-9" tabindex="-1"></a></span>
<span id="cb148-10"><a href="10-GWAS.html#cb148-10" tabindex="-1"></a><span class="co"># show top 10 variants with highest effect</span></span>
<span id="cb148-11"><a href="10-GWAS.html#cb148-11" tabindex="-1"></a><span class="fu">head</span><span class="er">(</span><span class="ex">params.effects.sort,</span> 10<span class="kw">)</span> </span>
<span id="cb148-12"><a href="10-GWAS.html#cb148-12" tabindex="-1"></a></span>
<span id="cb148-13"><a href="10-GWAS.html#cb148-13" tabindex="-1"></a><span class="co"># variants with the highest sparse effects</span></span>
<span id="cb148-14"><a href="10-GWAS.html#cb148-14" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb148-15"><a href="10-GWAS.html#cb148-15" tabindex="-1"></a><span class="co"># top 1% variants (above 99% quantile)</span></span>
<span id="cb148-16"><a href="10-GWAS.html#cb148-16" tabindex="-1"></a><span class="ex">top1</span><span class="op">&lt;</span>-params.effects.sort[params.effects.sort<span class="va">$eff</span><span class="op">&gt;</span>quantile<span class="er">(</span><span class="ex">params.effects.sort</span><span class="va">$eff</span><span class="ex">,0.99</span><span class="kw">)</span><span class="ex">,]</span></span>
<span id="cb148-17"><a href="10-GWAS.html#cb148-17" tabindex="-1"></a><span class="co"># top 0.1% variants (above 99.9% quantile)</span></span>
<span id="cb148-18"><a href="10-GWAS.html#cb148-18" tabindex="-1"></a><span class="ex">top01</span><span class="op">&lt;</span>-params.effects.sort[params.effects.sort<span class="va">$eff</span><span class="op">&gt;</span>quantile<span class="er">(</span><span class="ex">params.effects.sort</span><span class="va">$eff</span><span class="ex">,0.999</span><span class="kw">)</span><span class="ex">,]</span></span>
<span id="cb148-19"><a href="10-GWAS.html#cb148-19" tabindex="-1"></a><span class="co"># top 0.01% variants (above 99.99% quantile)</span></span>
<span id="cb148-20"><a href="10-GWAS.html#cb148-20" tabindex="-1"></a><span class="ex">top001</span><span class="op">&lt;</span>-params.effects.sort[params.effects.sort<span class="va">$eff</span><span class="op">&gt;</span>quantile<span class="er">(</span><span class="ex">params.effects.sort</span><span class="va">$eff</span><span class="ex">,0.9999</span><span class="kw">)</span><span class="ex">,]</span></span>
<span id="cb148-21"><a href="10-GWAS.html#cb148-21" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb148-22"><a href="10-GWAS.html#cb148-22" tabindex="-1"></a></span>
<span id="cb148-23"><a href="10-GWAS.html#cb148-23" tabindex="-1"></a><span class="co"># write tables</span></span>
<span id="cb148-24"><a href="10-GWAS.html#cb148-24" tabindex="-1"></a><span class="ex">write.table</span><span class="er">(</span><span class="ex">top1,</span> file=<span class="st">&quot;top1eff.dsv&quot;</span>, quote=F, row.names=F, sep=<span class="st">&quot;\t&quot;</span><span class="kw">)</span></span>
<span id="cb148-25"><a href="10-GWAS.html#cb148-25" tabindex="-1"></a><span class="ex">write.table</span><span class="er">(</span><span class="ex">top01,</span> file=<span class="st">&quot;top0.1eff.dsv&quot;</span>, quote=F, row.names=F, sep=<span class="st">&quot;\t&quot;</span><span class="kw">)</span></span>
<span id="cb148-26"><a href="10-GWAS.html#cb148-26" tabindex="-1"></a><span class="ex">write.table</span><span class="er">(</span><span class="ex">top001,</span> file=<span class="st">&quot;top0.01eff.dsv&quot;</span>, quote=F, row.names=F, sep=<span class="st">&quot;\t&quot;</span><span class="kw">)</span></span></code></pre></div>
<p>There should be around 15595 SNPs with detectable sparse effects and the top SNPs should look like this (notice we didn’t specify the chromosomes or the positions and therefore there are only SNP IDs, i.e. rs):</p>
<div class="l-body-outset">
<table>
<colgroup>
<col width="2%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="19%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th></th>
<th>chr</th>
<th>rs</th>
<th>ps</th>
<th>n_miss</th>
<th>alpha</th>
<th>beta</th>
<th>gamma</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>-9</td>
<td>lg8_ord45_scaf1036-131573</td>
<td>-9</td>
<td>0</td>
<td>-1.229901e-05</td>
<td>-1.2065980</td>
<td>0.5364</td>
<td>4929 0.64721917</td>
</tr>
<tr class="even">
<td>-9</td>
<td>lg8_ord55_scaf1512-149001</td>
<td>-9</td>
<td>0</td>
<td>-6.640361e-05</td>
<td>-0.7905428</td>
<td>0.5635</td>
<td>4943 0.48103043</td>
</tr>
<tr class="odd">
<td>-9</td>
<td>lg8_ord66_scaf318-448145</td>
<td>-9</td>
<td>0</td>
<td>-1.133074e-04</td>
<td>-0.6136137</td>
<td>0.0566</td>
<td>341474</td>
</tr>
<tr class="even">
<td>-9</td>
<td>lgNA_ordNA_scaf784-237602</td>
<td>-9</td>
<td>0</td>
<td>1.004314e-04</td>
<td>0.7119540</td>
<td>0.0378</td>
<td>330495</td>
</tr>
<tr class="odd">
<td>-9</td>
<td>lg3_ord81_scaf488-29866</td>
<td>-9</td>
<td>0</td>
<td>7.241678e-05</td>
<td>0.9273204</td>
<td>0.0232</td>
<td>315197</td>
</tr>
<tr class="even">
<td>-9</td>
<td>lgNA_ordNA_scaf154-274617</td>
<td>-9</td>
<td>0</td>
<td>5.164412e-05</td>
<td>1.1116970</td>
<td>0.0157</td>
<td>5105</td>
</tr>
<tr class="odd">
<td>-9</td>
<td>lg6_ord32_scaf531-110990</td>
<td>-9</td>
<td>0</td>
<td>6.656697e-05</td>
<td>0.9619961</td>
<td>0.0169</td>
<td>183101</td>
</tr>
<tr class="even">
<td>-9</td>
<td>lg6_ord32_scaf531-110990</td>
<td>-9</td>
<td>0</td>
<td>6.656697e-05</td>
<td>0.9619961</td>
<td>0.0169</td>
<td>138641</td>
</tr>
<tr class="odd">
<td>-9</td>
<td>lg6_ord32_scaf531-110989</td>
<td>-9</td>
<td>0</td>
<td>6.606357e-05</td>
<td>0.9763403</td>
<td>0.0144</td>
<td>138636</td>
</tr>
<tr class="even">
<td>-9</td>
<td>lg3_ord28_scaf970-18573</td>
<td>-9</td>
<td>0</td>
<td>5.868428e-05</td>
<td>1.1262150</td>
<td>0.0109</td>
<td>138636</td>
</tr>
</tbody>
</table>
</div>
<p>We are going to estimate the Posterior Inclusion Probability (PIP), that is the frequency a SNP is estimated to have a detectable large effect in the MCMC (i.e. γ). This can be used as a measure of the strength of the association of a SNP with a phenotype. As with the effect sizes we will sort them by PIP, and save the top SNPs using several thresholds (0.01, 0.1, 0.25, 0.5):</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb149-1"><a href="10-GWAS.html#cb149-1" tabindex="-1"></a><span class="co"># ==============================================================================</span></span>
<span id="cb149-2"><a href="10-GWAS.html#cb149-2" tabindex="-1"></a><span class="co"># Get variants with high Posterior Inclusion Probability (PIP) == gamma</span></span>
<span id="cb149-3"><a href="10-GWAS.html#cb149-3" tabindex="-1"></a><span class="co"># ==============================================================================</span></span>
<span id="cb149-4"><a href="10-GWAS.html#cb149-4" tabindex="-1"></a><span class="co"># PIP is the frequency a variant is estimated to have a sparse effect in the MCMC</span></span>
<span id="cb149-5"><a href="10-GWAS.html#cb149-5" tabindex="-1"></a></span>
<span id="cb149-6"><a href="10-GWAS.html#cb149-6" tabindex="-1"></a><span class="co"># sort variants by descending PIP</span></span>
<span id="cb149-7"><a href="10-GWAS.html#cb149-7" tabindex="-1"></a><span class="ex">params.pipsort</span><span class="op">&lt;</span>-params[order<span class="er">(</span><span class="ex">-params</span><span class="va">$gamma</span><span class="kw">)</span><span class="ex">,]</span></span>
<span id="cb149-8"><a href="10-GWAS.html#cb149-8" tabindex="-1"></a></span>
<span id="cb149-9"><a href="10-GWAS.html#cb149-9" tabindex="-1"></a><span class="co"># Show top 10 variants with highest PIP</span></span>
<span id="cb149-10"><a href="10-GWAS.html#cb149-10" tabindex="-1"></a><span class="fu">head</span><span class="er">(</span><span class="ex">params.pipsort,10</span><span class="kw">)</span></span>
<span id="cb149-11"><a href="10-GWAS.html#cb149-11" tabindex="-1"></a></span>
<span id="cb149-12"><a href="10-GWAS.html#cb149-12" tabindex="-1"></a><span class="co"># sets of variants above a certain threshold</span></span>
<span id="cb149-13"><a href="10-GWAS.html#cb149-13" tabindex="-1"></a><span class="co"># variants with effect in 1% MCMC samples or more</span></span>
<span id="cb149-14"><a href="10-GWAS.html#cb149-14" tabindex="-1"></a><span class="ex">pip01</span><span class="op">&lt;</span>-params.pipsort[params.pipsort<span class="va">$gamma</span><span class="op">&gt;</span>=0.01,]</span>
<span id="cb149-15"><a href="10-GWAS.html#cb149-15" tabindex="-1"></a><span class="co"># variants with effect in 10% MCMC samples or more</span></span>
<span id="cb149-16"><a href="10-GWAS.html#cb149-16" tabindex="-1"></a><span class="ex">pip10</span><span class="op">&lt;</span>-params.pipsort[params.pipsort<span class="va">$gamma</span><span class="op">&gt;</span>=0.10,]</span>
<span id="cb149-17"><a href="10-GWAS.html#cb149-17" tabindex="-1"></a><span class="co"># variants with effect in 25% MCMC samples or more</span></span>
<span id="cb149-18"><a href="10-GWAS.html#cb149-18" tabindex="-1"></a><span class="ex">pip25</span><span class="op">&lt;</span>-params.pipsort[params.pipsort<span class="va">$gamma</span><span class="op">&gt;</span>=0.25,]</span>
<span id="cb149-19"><a href="10-GWAS.html#cb149-19" tabindex="-1"></a><span class="co"># variants with effect in 50% MCMC samples or more</span></span>
<span id="cb149-20"><a href="10-GWAS.html#cb149-20" tabindex="-1"></a><span class="ex">pip50</span><span class="op">&lt;</span>-params.pipsort[params.pipsort<span class="va">$gamma</span><span class="op">&gt;</span>=0.50,]</span>
<span id="cb149-21"><a href="10-GWAS.html#cb149-21" tabindex="-1"></a></span>
<span id="cb149-22"><a href="10-GWAS.html#cb149-22" tabindex="-1"></a><span class="co"># write tables</span></span>
<span id="cb149-23"><a href="10-GWAS.html#cb149-23" tabindex="-1"></a><span class="ex">write.table</span><span class="er">(</span><span class="ex">pip01,</span> file=<span class="st">&quot;pip01.dsv&quot;</span>, quote=F, row.names=F, sep=<span class="st">&quot;\t&quot;</span><span class="kw">)</span></span>
<span id="cb149-24"><a href="10-GWAS.html#cb149-24" tabindex="-1"></a><span class="ex">write.table</span><span class="er">(</span><span class="ex">pip10,</span> file=<span class="st">&quot;pip10.dsv&quot;</span>, quote=F, row.names=F, sep=<span class="st">&quot;\t&quot;</span><span class="kw">)</span></span>
<span id="cb149-25"><a href="10-GWAS.html#cb149-25" tabindex="-1"></a><span class="ex">write.table</span><span class="er">(</span><span class="ex">pip25,</span> file=<span class="st">&quot;pip25.dsv&quot;</span>, quote=F, row.names=F, sep=<span class="st">&quot;\t&quot;</span><span class="kw">)</span></span>
<span id="cb149-26"><a href="10-GWAS.html#cb149-26" tabindex="-1"></a><span class="ex">write.table</span><span class="er">(</span><span class="ex">pip50,</span> file=<span class="st">&quot;pip50.dsv&quot;</span>, quote=F, row.names=F, sep=<span class="st">&quot;\t&quot;</span><span class="kw">)</span></span></code></pre></div>
<p>You can see the SNPs are very similar but not exactly the same ones than when sorted by effect size:</p>
<div class="l-body-outset">
<table>
<colgroup>
<col width="2%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="19%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th></th>
<th>chr</th>
<th>rs</th>
<th>ps</th>
<th>n_miss</th>
<th>alpha</th>
<th>beta</th>
<th>gamma</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>-9</td>
<td>lg8_ord55_scaf1512-149001</td>
<td>-9</td>
<td>0</td>
<td>-6.640361e-05</td>
<td>-0.7905428</td>
<td>0.5635</td>
<td>0.445470868</td>
</tr>
<tr class="even">
<td>-9</td>
<td>lg8_ord45_scaf1036-131573</td>
<td>-9</td>
<td>0</td>
<td>-1.229901e-05</td>
<td>-1.2065980</td>
<td>0.5364</td>
<td>0.647219167</td>
</tr>
<tr class="odd">
<td>-9</td>
<td>lg8_ord45_scaf1036-131605</td>
<td>-9</td>
<td>0</td>
<td>-1.096165e-05</td>
<td>-1.0375980</td>
<td>0.4636</td>
<td>0.481030433</td>
</tr>
<tr class="even">
<td>-9</td>
<td>lg8_ord66_scaf318-448145</td>
<td>-9</td>
<td>0</td>
<td>-1.133074e-04</td>
<td>-0.6136137</td>
<td>0.0566</td>
<td>0.034730535</td>
</tr>
<tr class="odd">
<td>-9</td>
<td>lgNA_ordNA_scaf784-237602</td>
<td>-9</td>
<td>0</td>
<td>1.004314e-04</td>
<td>0.7119540</td>
<td>0.0378</td>
<td>0.026911861</td>
</tr>
<tr class="even">
<td>-9</td>
<td>lg1_ord96_scaf190-725578</td>
<td>-9</td>
<td>0</td>
<td>-2.183769e-04</td>
<td>-0.3418733</td>
<td>0.0326</td>
<td>0.011145070</td>
</tr>
<tr class="odd">
<td>-9</td>
<td>lg3_ord81_scaf488-29866</td>
<td>-9</td>
<td>0</td>
<td>7.241678e-05</td>
<td>0.9273204</td>
<td>0.0232</td>
<td>0.021513833</td>
</tr>
<tr class="even">
<td>-9</td>
<td>lg10_ord64_scaf380-30883</td>
<td>-9</td>
<td>0</td>
<td>-2.363161e-04</td>
<td>-0.3245005</td>
<td>0.0180</td>
<td>0.005841009</td>
</tr>
<tr class="odd">
<td>-9</td>
<td>lg8_ord60_scaf2482-78371</td>
<td>-9</td>
<td>0</td>
<td>1.514078e-04</td>
<td>0.4569336</td>
<td>0.0174</td>
<td>0.007950645</td>
</tr>
<tr class="even">
<td>-9</td>
<td>lg6_ord32_scaf531-110990</td>
<td>-9</td>
<td>0</td>
<td>6.656697e-05</td>
<td>0.9619961</td>
<td>0.0169</td>
<td>0.016257734</td>
</tr>
</tbody>
</table>
</div>
<p>Lastly we can generate a Manhattan plot of the PIPs to visualize the distribution of associated SNPs across the genome. We will highlight the top candidates in the genome and make the size of the dots reflect the effect size.</p>
<div class="sourceCode" id="cb150"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb150-1"><a href="10-GWAS.html#cb150-1" tabindex="-1"></a><span class="co"># ==============================================================================</span></span>
<span id="cb150-2"><a href="10-GWAS.html#cb150-2" tabindex="-1"></a><span class="co"># plot variants PIPs across linkage groups/chromosomes</span></span>
<span id="cb150-3"><a href="10-GWAS.html#cb150-3" tabindex="-1"></a><span class="co"># ==============================================================================</span></span>
<span id="cb150-4"><a href="10-GWAS.html#cb150-4" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb150-5"><a href="10-GWAS.html#cb150-5" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-6"><a href="10-GWAS.html#cb150-6" tabindex="-1"></a><span class="co"># add linkage group column (chr)</span></span>
<span id="cb150-7"><a href="10-GWAS.html#cb150-7" tabindex="-1"></a><span class="ex">chr</span><span class="op">&lt;</span>-gsub<span class="er">(</span><span class="st">&quot;lg|_.+&quot;</span><span class="ex">,</span><span class="st">&quot;&quot;</span><span class="ex">,params</span><span class="va">$rs</span><span class="kw">)</span></span>
<span id="cb150-8"><a href="10-GWAS.html#cb150-8" tabindex="-1"></a><span class="va">params</span><span class="op">[</span><span class="st">&quot;chr&quot;</span><span class="op">]&lt;</span>-chr</span>
<span id="cb150-9"><a href="10-GWAS.html#cb150-9" tabindex="-1"></a></span>
<span id="cb150-10"><a href="10-GWAS.html#cb150-10" tabindex="-1"></a><span class="co"># sort by linkage group and position</span></span>
<span id="cb150-11"><a href="10-GWAS.html#cb150-11" tabindex="-1"></a><span class="ex">params.sort</span><span class="op">&lt;</span>-params[order<span class="er">(</span><span class="ex">as.numeric</span><span class="er">(</span><span class="ex">params</span><span class="va">$chr</span><span class="kw">)</span><span class="ex">,</span> params<span class="va">$rs</span><span class="kw">)</span><span class="ex">,]</span></span>
<span id="cb150-12"><a href="10-GWAS.html#cb150-12" tabindex="-1"></a></span>
<span id="cb150-13"><a href="10-GWAS.html#cb150-13" tabindex="-1"></a><span class="co"># get list of linkage groups/chromosomes</span></span>
<span id="cb150-14"><a href="10-GWAS.html#cb150-14" tabindex="-1"></a><span class="ex">chrs</span><span class="op">&lt;</span>-sort<span class="er">(</span><span class="ex">as.numeric</span><span class="er">(</span><span class="ex">unique</span><span class="er">(</span><span class="ex">chr</span><span class="kw">)))</span></span>
<span id="cb150-15"><a href="10-GWAS.html#cb150-15" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-16"><a href="10-GWAS.html#cb150-16" tabindex="-1"></a></span>
<span id="cb150-17"><a href="10-GWAS.html#cb150-17" tabindex="-1"></a><span class="co"># Plot figure</span></span>
<span id="cb150-18"><a href="10-GWAS.html#cb150-18" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-19"><a href="10-GWAS.html#cb150-19" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-20"><a href="10-GWAS.html#cb150-20" tabindex="-1"></a><span class="co"># set up empty plot</span></span>
<span id="cb150-21"><a href="10-GWAS.html#cb150-21" tabindex="-1"></a><span class="ex">plot</span><span class="er">(</span><span class="ex">-1,-1,xlim=c</span><span class="er">(</span><span class="ex">0,nrow</span><span class="er">(</span><span class="ex">params.sort</span><span class="kw">))</span><span class="ex">,ylim=c</span><span class="er">(</span><span class="ex">0,1</span><span class="kw">)</span><span class="ex">,ylab=</span><span class="st">&quot;PIP&quot;</span><span class="ex">,xlab=</span><span class="st">&quot;linkage group&quot;</span><span class="ex">,</span> xaxt=<span class="st">&quot;n&quot;</span><span class="kw">)</span></span>
<span id="cb150-22"><a href="10-GWAS.html#cb150-22" tabindex="-1"></a></span>
<span id="cb150-23"><a href="10-GWAS.html#cb150-23" tabindex="-1"></a></span>
<span id="cb150-24"><a href="10-GWAS.html#cb150-24" tabindex="-1"></a><span class="co"># plot grey bands for chromosome/linkage groups</span></span>
<span id="cb150-25"><a href="10-GWAS.html#cb150-25" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-26"><a href="10-GWAS.html#cb150-26" tabindex="-1"></a><span class="ex">start</span><span class="op">&lt;</span>-1</span>
<span id="cb150-27"><a href="10-GWAS.html#cb150-27" tabindex="-1"></a><span class="ex">lab.pos</span><span class="op">&lt;</span>-vector<span class="er">(</span><span class="kw">)</span></span>
<span id="cb150-28"><a href="10-GWAS.html#cb150-28" tabindex="-1"></a><span class="cf">for</span> <span class="kw">(</span><span class="ex">ch</span> in chrs<span class="kw">){</span></span>
<span id="cb150-29"><a href="10-GWAS.html#cb150-29" tabindex="-1"></a>  <span class="fu">size</span><span class="op">&lt;</span>-nrow<span class="er">(</span><span class="ex">params.sort[params.sort</span><span class="va">$chr</span><span class="ex">==ch,]</span><span class="kw">)</span></span>
<span id="cb150-30"><a href="10-GWAS.html#cb150-30" tabindex="-1"></a>  <span class="fu">cat</span> <span class="er">(</span><span class="st">&quot;CH: &quot;</span><span class="ex">,</span> ch, <span class="st">&quot;\n&quot;</span><span class="kw">)</span></span>
<span id="cb150-31"><a href="10-GWAS.html#cb150-31" tabindex="-1"></a>  <span class="ex">colour</span><span class="op">&lt;</span>-<span class="st">&quot;light grey&quot;</span></span>
<span id="cb150-32"><a href="10-GWAS.html#cb150-32" tabindex="-1"></a>  <span class="cf">if</span> <span class="kw">(</span><span class="ex">ch%%2</span> <span class="op">&gt;</span> 0<span class="kw">){</span></span>
<span id="cb150-33"><a href="10-GWAS.html#cb150-33" tabindex="-1"></a>    <span class="ex">polygon</span><span class="er">(</span><span class="ex">c</span><span class="er">(</span><span class="ex">start,start,start+size,start+size,start</span><span class="kw">)</span><span class="ex">,</span> c<span class="er">(</span><span class="ex">0,1,1,0,0</span><span class="kw">)</span><span class="ex">,</span> col=colour, border=colour<span class="kw">)</span></span>
<span id="cb150-34"><a href="10-GWAS.html#cb150-34" tabindex="-1"></a>  <span class="kw">}</span></span>
<span id="cb150-35"><a href="10-GWAS.html#cb150-35" tabindex="-1"></a>  <span class="fu">cat</span><span class="er">(</span><span class="st">&quot;CHR: &quot;</span><span class="ex">,</span> ch, <span class="st">&quot; variants: &quot;</span>, size, <span class="st">&quot;(total: &quot;</span>, <span class="er">(</span><span class="ex">start+size</span><span class="kw">)</span><span class="ex">,</span> <span class="st">&quot;)\n&quot;</span><span class="kw">)</span></span>
<span id="cb150-36"><a href="10-GWAS.html#cb150-36" tabindex="-1"></a>  <span class="ex">txtpos</span><span class="op">&lt;</span>-start+size/2</span>
<span id="cb150-37"><a href="10-GWAS.html#cb150-37" tabindex="-1"></a>  <span class="ex">lab.pos</span><span class="op">&lt;</span>-c<span class="er">(</span><span class="ex">lab.pos,</span> txtpos<span class="kw">)</span></span>
<span id="cb150-38"><a href="10-GWAS.html#cb150-38" tabindex="-1"></a></span>
<span id="cb150-39"><a href="10-GWAS.html#cb150-39" tabindex="-1"></a>  <span class="ex">start</span><span class="op">&lt;</span>-start+size</span>
<span id="cb150-40"><a href="10-GWAS.html#cb150-40" tabindex="-1"></a><span class="kw">}</span></span>
<span id="cb150-41"><a href="10-GWAS.html#cb150-41" tabindex="-1"></a><span class="co"># Add variants outside linkage groups</span></span>
<span id="cb150-42"><a href="10-GWAS.html#cb150-42" tabindex="-1"></a><span class="ex">chrs</span><span class="op">&lt;</span>-c<span class="er">(</span><span class="ex">chrs,</span><span class="st">&quot;NA&quot;</span><span class="kw">)</span></span>
<span id="cb150-43"><a href="10-GWAS.html#cb150-43" tabindex="-1"></a><span class="fu">size</span><span class="op">&lt;</span>-nrow<span class="er">(</span><span class="ex">params.sort[params.sort</span><span class="va">$chr</span><span class="ex">==</span><span class="st">&quot;NA&quot;</span><span class="ex">,]</span><span class="kw">)</span></span>
<span id="cb150-44"><a href="10-GWAS.html#cb150-44" tabindex="-1"></a><span class="ex">lab.pos</span><span class="op">&lt;</span>-c<span class="er">(</span><span class="ex">lab.pos,</span> start+size/2<span class="kw">)</span></span>
<span id="cb150-45"><a href="10-GWAS.html#cb150-45" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-46"><a href="10-GWAS.html#cb150-46" tabindex="-1"></a></span>
<span id="cb150-47"><a href="10-GWAS.html#cb150-47" tabindex="-1"></a><span class="co"># Add x axis labels</span></span>
<span id="cb150-48"><a href="10-GWAS.html#cb150-48" tabindex="-1"></a><span class="ex">axis</span><span class="er">(</span><span class="va">side</span><span class="op">=</span>1,at=lab.pos,labels=chrs,tick=F<span class="kw">)</span></span>
<span id="cb150-49"><a href="10-GWAS.html#cb150-49" tabindex="-1"></a></span>
<span id="cb150-50"><a href="10-GWAS.html#cb150-50" tabindex="-1"></a><span class="co"># plot PIP for all variants</span></span>
<span id="cb150-51"><a href="10-GWAS.html#cb150-51" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-52"><a href="10-GWAS.html#cb150-52" tabindex="-1"></a><span class="co"># rank of variants across linkage groups</span></span>
<span id="cb150-53"><a href="10-GWAS.html#cb150-53" tabindex="-1"></a><span class="ex">x</span><span class="op">&lt;</span>-seq<span class="er">(</span><span class="ex">1,length</span><span class="er">(</span><span class="ex">params.sort</span><span class="va">$gamma</span><span class="kw">)</span><span class="ex">,1</span><span class="kw">)</span></span>
<span id="cb150-54"><a href="10-GWAS.html#cb150-54" tabindex="-1"></a><span class="co"># PIP </span></span>
<span id="cb150-55"><a href="10-GWAS.html#cb150-55" tabindex="-1"></a><span class="ex">y</span><span class="op">&lt;</span>-params.sort<span class="va">$gamma</span></span>
<span id="cb150-56"><a href="10-GWAS.html#cb150-56" tabindex="-1"></a><span class="co"># sparse effect size, used for dot size</span></span>
<span id="cb150-57"><a href="10-GWAS.html#cb150-57" tabindex="-1"></a><span class="ex">z</span><span class="op">&lt;</span>-params.sort<span class="va">$eff</span></span>
<span id="cb150-58"><a href="10-GWAS.html#cb150-58" tabindex="-1"></a><span class="co"># log-transform to enhance visibility</span></span>
<span id="cb150-59"><a href="10-GWAS.html#cb150-59" tabindex="-1"></a><span class="va">z</span><span class="op">[</span>z==0<span class="op">]&lt;</span>-0.00000000001</span>
<span id="cb150-60"><a href="10-GWAS.html#cb150-60" tabindex="-1"></a><span class="ex">z</span><span class="op">&lt;</span>-1/abs<span class="er">(</span><span class="ex">log</span><span class="er">(</span><span class="ex">z</span><span class="kw">))</span></span>
<span id="cb150-61"><a href="10-GWAS.html#cb150-61" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb150-62"><a href="10-GWAS.html#cb150-62" tabindex="-1"></a><span class="ex">symbols</span><span class="er">(</span><span class="ex">x,y,circles=z,</span> bg=<span class="st">&quot;black&quot;</span>,inches=1/5, fg=NULL,add=T<span class="kw">)</span></span>
<span id="cb150-63"><a href="10-GWAS.html#cb150-63" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-64"><a href="10-GWAS.html#cb150-64" tabindex="-1"></a></span>
<span id="cb150-65"><a href="10-GWAS.html#cb150-65" tabindex="-1"></a><span class="co"># highlight high PIP variants (PIP&gt;=0.25)</span></span>
<span id="cb150-66"><a href="10-GWAS.html#cb150-66" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-67"><a href="10-GWAS.html#cb150-67" tabindex="-1"></a><span class="co"># plot threshold line</span></span>
<span id="cb150-68"><a href="10-GWAS.html#cb150-68" tabindex="-1"></a><span class="ex">abline</span><span class="er">(</span><span class="va">h</span><span class="op">=</span>0.25,lty=3,col=<span class="st">&quot;dark grey&quot;</span><span class="kw">)</span></span>
<span id="cb150-69"><a href="10-GWAS.html#cb150-69" tabindex="-1"></a><span class="co"># rank of high PIP variants across linkage groups</span></span>
<span id="cb150-70"><a href="10-GWAS.html#cb150-70" tabindex="-1"></a><span class="ex">x</span><span class="op">&lt;</span>-match<span class="er">(</span><span class="ex">params.sort</span><span class="va">$gamma</span><span class="ex">[params.sort</span><span class="va">$gamma</span><span class="op">&gt;</span>=0.25],params.sort<span class="va">$gamma</span><span class="kw">)</span></span>
<span id="cb150-71"><a href="10-GWAS.html#cb150-71" tabindex="-1"></a><span class="co"># PIP</span></span>
<span id="cb150-72"><a href="10-GWAS.html#cb150-72" tabindex="-1"></a><span class="ex">y</span><span class="op">&lt;</span>-params.sort<span class="va">$gamma</span>[params.sort<span class="va">$gamma</span><span class="op">&gt;</span>=0.25]</span>
<span id="cb150-73"><a href="10-GWAS.html#cb150-73" tabindex="-1"></a><span class="co"># sparse effect size, used for dot size</span></span>
<span id="cb150-74"><a href="10-GWAS.html#cb150-74" tabindex="-1"></a><span class="ex">z</span><span class="op">&lt;</span>-params.sort<span class="va">$eff</span>[params.sort<span class="va">$gamma</span><span class="op">&gt;</span>=0.25]</span>
<span id="cb150-75"><a href="10-GWAS.html#cb150-75" tabindex="-1"></a><span class="ex">z</span><span class="op">&lt;</span>-1/abs<span class="er">(</span><span class="ex">log</span><span class="er">(</span><span class="ex">z</span><span class="kw">))</span></span>
<span id="cb150-76"><a href="10-GWAS.html#cb150-76" tabindex="-1"></a></span>
<span id="cb150-77"><a href="10-GWAS.html#cb150-77" tabindex="-1"></a><span class="ex">symbols</span><span class="er">(</span><span class="ex">x,y,circles=z,</span> bg=<span class="st">&quot;red&quot;</span>,inches=1/5,fg=NULL,add=T<span class="kw">)</span></span>
<span id="cb150-78"><a href="10-GWAS.html#cb150-78" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-79"><a href="10-GWAS.html#cb150-79" tabindex="-1"></a></span>
<span id="cb150-80"><a href="10-GWAS.html#cb150-80" tabindex="-1"></a><span class="co"># add label high PIP variants</span></span>
<span id="cb150-81"><a href="10-GWAS.html#cb150-81" tabindex="-1"></a><span class="ex">text</span><span class="er">(</span><span class="ex">x,y,labels=params.sort</span><span class="va">$rs</span><span class="ex">[params.sort</span><span class="va">$gamma</span><span class="op">&gt;</span>=0.25], adj=c<span class="er">(</span><span class="ex">0,0</span><span class="kw">)</span><span class="ex">,</span> cex=0.8<span class="kw">)</span></span>
<span id="cb150-82"><a href="10-GWAS.html#cb150-82" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-83"><a href="10-GWAS.html#cb150-83" tabindex="-1"></a><span class="co"># ------------------------------------------------------------------------------</span></span>
<span id="cb150-84"><a href="10-GWAS.html#cb150-84" tabindex="-1"></a></span>
<span id="cb150-85"><a href="10-GWAS.html#cb150-85" tabindex="-1"></a><span class="co"># ==============================================================================</span></span></code></pre></div>
<p>The PIP plot shows there are three large effect size SNPs strongly associated with colour pattern in two non-contiguous scaffolds that belong to linkage group 8:</p>
<center>
<img src="figures/pip_plot.png" style="width:800px; border-radius:15px; border:5px solid #333333; background:null" />
</center>
<p>The code to produce these plots is contained in two <code>R</code> scripts. If you were running these on your own data the scripts could be run in batch mode by executing: <code>Rscript gemma_hyperparam.R</code>. We will provide the scripts in the materials we send around after the course.</p>

</div>
</div>



<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="09-Detecting_selection.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="11-Appendix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
