<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Import | R community analysis</title>
  <meta name="description" content="CGR book for the R community analysis workflow" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Import | R community analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/CGR_strips.png" />
  <meta property="og:description" content="CGR book for the R community analysis workflow" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Import | R community analysis" />
  
  <meta name="twitter:description" content="CGR book for the R community analysis workflow" />
  <meta name="twitter:image" content="/figures/CGR_strips.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-07-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/CGR_favicon.png" type="image/x-icon" />
<link rel="prev" href="06-Preprocess_part.html"/>
<link rel="next" href="08-Summarise_phyloseq.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://www.liverpool.ac.uk/genomic-research/" target="blank"><img src="figures/CGR_strips.png"></a></li>
<li><a>R community analysis</a></li>

<li class="divider"></li>
<li class="part"><span><b>Intro</b></span></li>
<li class="chapter" data-level="1" data-path="01-R_community_main_workflow.html"><a href="01-R_community_main_workflow.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html"><i class="fa fa-check"></i><b>2</b> Dataset &amp; workflow</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#dataset"><i class="fa fa-check"></i><b>2.1</b> Dataset</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sites"><i class="fa fa-check"></i><b>2.1.1</b> Sites</a></li>
<li class="chapter" data-level="2.1.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#culture-media"><i class="fa fa-check"></i><b>2.1.2</b> Culture media</a></li>
<li class="chapter" data-level="2.1.3" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sum_and_qs"><i class="fa fa-check"></i><b>2.1.3</b> Summary &amp; questions</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#workflow"><i class="fa fa-check"></i><b>2.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-R_packages.html"><a href="03-R_packages.html"><i class="fa fa-check"></i><b>3</b> R Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_packages.html"><a href="03-R_packages.html#r-packageslibraries"><i class="fa fa-check"></i><b>3.1</b> R packages/libraries</a></li>
<li class="chapter" data-level="3.2" data-path="03-R_packages.html"><a href="03-R_packages.html#the-grammar-of-graphics"><i class="fa fa-check"></i><b>3.2</b> The grammar of graphics</a></li>
<li class="chapter" data-level="3.3" data-path="03-R_packages.html"><a href="03-R_packages.html#phyloseq"><i class="fa fa-check"></i><b>3.3</b> phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Setup.html"><a href="04-Setup.html"><i class="fa fa-check"></i><b>4</b> Set-up</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Setup.html"><a href="04-Setup.html#cluster"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Setup.html"><a href="04-Setup.html#mamba"><i class="fa fa-check"></i><b>4.2</b> Mamba</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Jupyter.html"><a href="05-Jupyter.html"><i class="fa fa-check"></i><b>5</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Jupyter.html"><a href="05-Jupyter.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.2" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-r-notebook"><i class="fa fa-check"></i><b>5.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.3" data-path="05-Jupyter.html"><a href="05-Jupyter.html#cells-and-code"><i class="fa fa-check"></i><b>5.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.4" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-new-cells"><i class="fa fa-check"></i><b>5.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.5" data-path="05-Jupyter.html"><a href="05-Jupyter.html#running-code"><i class="fa fa-check"></i><b>5.5</b> Running code</a></li>
<li class="chapter" data-level="5.6" data-path="05-Jupyter.html"><a href="05-Jupyter.html#saving-the-file"><i class="fa fa-check"></i><b>5.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.7" data-path="05-Jupyter.html"><a href="05-Jupyter.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.8" data-path="05-Jupyter.html"><a href="05-Jupyter.html#close-the-notebook"><i class="fa fa-check"></i><b>5.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.9" data-path="05-Jupyter.html"><a href="05-Jupyter.html#video-tutorial"><i class="fa fa-check"></i><b>5.9</b> Video tutorial</a></li>
</ul></li>
<li class="part"><span><b>Data preparation</b></span></li>
<li class="chapter" data-level="6" data-path="06-Preprocess_part.html"><a href="06-Preprocess_part.html"><i class="fa fa-check"></i><b>6</b> Data prep intro</a></li>
<li class="chapter" data-level="7" data-path="07-Import.html"><a href="07-Import.html"><i class="fa fa-check"></i><b>7</b> Import</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Import.html"><a href="07-Import.html#import-notebook"><i class="fa fa-check"></i><b>7.1</b> Import: notebook</a></li>
<li class="chapter" data-level="7.2" data-path="07-Import.html"><a href="07-Import.html#qiime2r"><i class="fa fa-check"></i><b>7.2</b> qiime2R</a></li>
<li class="chapter" data-level="7.3" data-path="07-Import.html"><a href="07-Import.html#import-summarise-phyloseq"><i class="fa fa-check"></i><b>7.3</b> Import: Summarise phyloseq</a></li>
<li class="chapter" data-level="7.4" data-path="07-Import.html"><a href="07-Import.html#save-the-phyloseq-object"><i class="fa fa-check"></i><b>7.4</b> Save the phyloseq object</a></li>
<li class="chapter" data-level="7.5" data-path="07-Import.html"><a href="07-Import.html#import-recap"><i class="fa fa-check"></i><b>7.5</b> Import: recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html"><i class="fa fa-check"></i><b>8</b> Summarise phyloseq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-setup"><i class="fa fa-check"></i><b>8.1</b> Summarise: setup</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-header"><i class="fa fa-check"></i><b>8.1.1</b> Summarise header</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-phyloseq"><i class="fa fa-check"></i><b>8.1.2</b> Summarise phyloseq</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#reads-per-sample"><i class="fa fa-check"></i><b>8.1.3</b> Reads per sample</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#display-png"><i class="fa fa-check"></i><b>8.1.4</b> Display png</a></li>
<li class="chapter" data-level="8.1.5" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#asvs-per-sample"><i class="fa fa-check"></i><b>8.1.5</b> ASVs per sample</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-recap"><i class="fa fa-check"></i><b>8.2</b> Summarise: recap</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html"><i class="fa fa-check"></i><b>9</b> Minimum read depth</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#considerations"><i class="fa fa-check"></i><b>9.1</b> Considerations</a></li>
<li class="chapter" data-level="9.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minimum-read-depth-section-title"><i class="fa fa-check"></i><b>9.2</b> Minimum read depth section title</a></li>
<li class="chapter" data-level="9.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#read-depth-vector-and-histogram"><i class="fa fa-check"></i><b>9.3</b> Read depth vector and histogram</a></li>
<li class="chapter" data-level="9.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#depth-boxplots"><i class="fa fa-check"></i><b>9.4</b> Sample depth boxplot</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#creating-a-data-frame-for-boxplots"><i class="fa fa-check"></i><b>9.4.1</b> Creating a data frame for boxplots</a></li>
<li class="chapter" data-level="9.4.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#ggplot2-boxplot"><i class="fa fa-check"></i><b>9.4.2</b> ggplot2 boxplot</a></li>
<li class="chapter" data-level="9.4.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#save-ggplot-as-png"><i class="fa fa-check"></i><b>9.4.3</b> Save ggplot as png</a></li>
<li class="chapter" data-level="9.4.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#media-by-depth-boxplot"><i class="fa fa-check"></i><b>9.4.4</b> Media by depth boxplot</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#chap9rarefaction"><i class="fa fa-check"></i><b>9.5</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#asv-abundance-data-frame"><i class="fa fa-check"></i><b>9.5.1</b> ASV abundance data frame</a></li>
<li class="chapter" data-level="9.5.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#vegans-rarecurve"><i class="fa fa-check"></i><b>9.5.2</b> vegan's rarecurve</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#filtering-by-minimum-read-depth"><i class="fa fa-check"></i><b>9.6</b> Filtering by minimum read depth</a></li>
<li class="chapter" data-level="9.7" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minmum-read-depth-summary"><i class="fa fa-check"></i><b>9.7</b> Minmum read depth: Summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy</b></span></li>
<li class="chapter" data-level="10" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html"><i class="fa fa-check"></i><b>10</b> Taxa relative abundance</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-setup"><i class="fa fa-check"></i><b>10.1</b> Taxa relative abundance: setup</a></li>
<li class="chapter" data-level="10.2" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#relative-abundance-transformation"><i class="fa fa-check"></i><b>10.2</b> Relative abundance transformation</a></li>
<li class="chapter" data-level="10.3" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#aggregate-taxa"><i class="fa fa-check"></i><b>10.3</b> Aggregate taxa</a></li>
<li class="chapter" data-level="10.4" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-summary"><i class="fa fa-check"></i><b>10.4</b> Taxa Relative abundance: summary</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html"><i class="fa fa-check"></i><b>11</b> Taxa plots</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#simple-bar-chart"><i class="fa fa-check"></i><b>11.1</b> Simple bar chart</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#editing-the-plot"><i class="fa fa-check"></i><b>11.1.1</b> Editing the plot</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-heatmap"><i class="fa fa-check"></i><b>11.2</b> Taxa heatmap</a></li>
<li class="chapter" data-level="11.3" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#aggregateraretaxa"><i class="fa fa-check"></i><b>11.3</b> Aggregate rare taxa</a></li>
<li class="chapter" data-level="11.4" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-media"><i class="fa fa-check"></i><b>11.4</b> Plot by media</a></li>
<li class="chapter" data-level="11.5" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-site-and-media"><i class="fa fa-check"></i><b>11.5</b> Plot by site and media</a></li>
<li class="chapter" data-level="11.6" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-plots-summary"><i class="fa fa-check"></i><b>11.6</b> Taxa plots: summary</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html"><i class="fa fa-check"></i><b>12</b> Family &amp; Genus plots</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-taxa-plots"><i class="fa fa-check"></i><b>12.1</b> Family taxa plots</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#aggregate-families"><i class="fa fa-check"></i><b>12.1.1</b> Aggregate families</a></li>
<li class="chapter" data-level="12.1.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-heatmap"><i class="fa fa-check"></i><b>12.1.2</b> Family heatmap</a></li>
<li class="chapter" data-level="12.1.3" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-rare-aggregation"><i class="fa fa-check"></i><b>12.1.3</b> Family rare aggregation</a></li>
<li class="chapter" data-level="12.1.4" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-bar-chart"><i class="fa fa-check"></i><b>12.1.4</b> Family bar chart</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots"><i class="fa fa-check"></i><b>12.2</b> Genus taxa plots</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots-solutions"><i class="fa fa-check"></i><b>12.2.1</b> Genus taxa plots solutions</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Diversity analysis</b></span></li>
<li class="chapter" data-level="13" data-path="13-Diversity_analysis.html"><a href="13-Diversity_analysis.html"><i class="fa fa-check"></i><b>13</b> Diversity analysis intro</a></li>
<li class="chapter" data-level="14" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html"><i class="fa fa-check"></i><b>14</b> Rarefaction</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-setup"><i class="fa fa-check"></i><b>14.1</b> Rarefaction: setup</a></li>
<li class="chapter" data-level="14.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve"><i class="fa fa-check"></i><b>14.2</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#bird-analogy"><i class="fa fa-check"></i><b>14.2.1</b> Bird analogy</a></li>
<li class="chapter" data-level="14.2.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-simple-plot"><i class="fa fa-check"></i><b>14.2.2</b> Rarefaction curve: simple plot</a></li>
<li class="chapter" data-level="14.2.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-better-plot"><i class="fa fa-check"></i><b>14.2.3</b> Rarefaction curve: better plot</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-slope"><i class="fa fa-check"></i><b>14.3</b> Rarefaction slope</a></li>
<li class="chapter" data-level="14.4" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefy-data"><i class="fa fa-check"></i><b>14.4</b> Rarefy data</a></li>
<li class="chapter" data-level="14.5" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-summary"><i class="fa fa-check"></i><b>14.5</b> Rarefaction: summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Alpha.html"><a href="15-Alpha.html"><i class="fa fa-check"></i><b>15</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-setup"><i class="fa fa-check"></i><b>15.1</b> Alpha: setup</a></li>
<li class="chapter" data-level="15.2" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-simple-plot"><i class="fa fa-check"></i><b>15.2</b> Alpha: simple plot</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="15-Alpha.html"><a href="15-Alpha.html#metric-choice"><i class="fa fa-check"></i><b>15.2.1</b> Metric choice</a></li>
<li class="chapter" data-level="15.2.2" data-path="15-Alpha.html"><a href="15-Alpha.html#sample-grouping-box-plot"><i class="fa fa-check"></i><b>15.2.2</b> Sample grouping &amp; box plot</a></li>
<li class="chapter" data-level="15.2.3" data-path="15-Alpha.html"><a href="15-Alpha.html#png-height-and-width-choice"><i class="fa fa-check"></i><b>15.2.3</b> PNG height and width choice</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats"><i class="fa fa-check"></i><b>15.3</b> Alpha: stats</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats-task"><i class="fa fa-check"></i><b>15.3.1</b> Alpha stats task</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-diversity-violin-plot"><i class="fa fa-check"></i><b>15.4</b> Alpha diversity: violin plot</a></li>
<li class="chapter" data-level="15.5" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-site-plots-and-statistics"><i class="fa fa-check"></i><b>15.5</b> Alpha: Site plots and statistics</a></li>
<li class="chapter" data-level="15.6" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-summary"><i class="fa fa-check"></i><b>15.6</b> Alpha: summary</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Beta.html"><a href="16-Beta.html"><i class="fa fa-check"></i><b>16</b> Beta</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Beta.html"><a href="16-Beta.html#beta-setup"><i class="fa fa-check"></i><b>16.1</b> Beta: setup</a></li>
<li class="chapter" data-level="16.2" data-path="16-Beta.html"><a href="16-Beta.html#beta-ordination"><i class="fa fa-check"></i><b>16.2</b> Beta: ordination</a></li>
<li class="chapter" data-level="16.3" data-path="16-Beta.html"><a href="16-Beta.html#beta-plot-ordination"><i class="fa fa-check"></i><b>16.3</b> Beta: plot ordination</a></li>
<li class="chapter" data-level="16.4" data-path="16-Beta.html"><a href="16-Beta.html#beta-statistics"><i class="fa fa-check"></i><b>16.4</b> Beta: statistics</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="16-Beta.html"><a href="16-Beta.html#permanova"><i class="fa fa-check"></i><b>16.4.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="16.4.2" data-path="16-Beta.html"><a href="16-Beta.html#pairwise-permanova"><i class="fa fa-check"></i><b>16.4.2</b> Pairwise PERMANOVA</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="16-Beta.html"><a href="16-Beta.html#beta-facet-zoom"><i class="fa fa-check"></i><b>16.5</b> Beta: Facet zoom</a></li>
<li class="chapter" data-level="16.6" data-path="16-Beta.html"><a href="16-Beta.html#beta-subset-samples"><i class="fa fa-check"></i><b>16.6</b> Beta: subset samples</a></li>
<li class="chapter" data-level="16.7" data-path="16-Beta.html"><a href="16-Beta.html#permanova-tasks"><i class="fa fa-check"></i><b>16.7</b> PERMANOVA tasks</a></li>
<li class="chapter" data-level="16.8" data-path="16-Beta.html"><a href="16-Beta.html#beta-summary"><i class="fa fa-check"></i><b>16.8</b> Beta: Summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html"><i class="fa fa-check"></i><b>17</b> Differential abundance (DA) analysis</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-setup"><i class="fa fa-check"></i><b>17.1</b> DA: setup</a></li>
<li class="chapter" data-level="17.2" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-preprocess"><i class="fa fa-check"></i><b>17.2</b> DA: Preprocess</a></li>
<li class="chapter" data-level="17.3" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-analysis"><i class="fa fa-check"></i><b>17.3</b> DA: ANCOM-BC analysis</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#ancombc2-options"><i class="fa fa-check"></i><b>17.3.1</b> ANCOMBC2 options</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-results"><i class="fa fa-check"></i><b>17.4</b> DA: ANCOM-BC results</a></li>
<li class="chapter" data-level="17.5" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-create-signifcant-lfc-long-data-frame"><i class="fa fa-check"></i><b>17.5</b> DA: Create signifcant LFC long data frame</a></li>
<li class="chapter" data-level="17.6" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-biomarker-lfc-heatmap"><i class="fa fa-check"></i><b>17.6</b> DA: Biomarker LFC heatmap</a></li>
<li class="chapter" data-level="17.7" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-matching-bar-plot"><i class="fa fa-check"></i><b>17.7</b> DA: matching bar plot</a></li>
<li class="chapter" data-level="17.8" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-summary"><i class="fa fa-check"></i><b>17.8</b> DA: summary</a></li>
</ul></li>
<li class="part"><span><b>Outro</b></span></li>
<li class="chapter" data-level="18" data-path="18-Summary.html"><a href="18-Summary.html"><i class="fa fa-check"></i><b>18</b> Summary</a>
<ul>
<li class="chapter" data-level="18.1" data-path="18-Summary.html"><a href="18-Summary.html#dataset-1"><i class="fa fa-check"></i><b>18.1</b> Dataset</a></li>
<li class="chapter" data-level="18.2" data-path="18-Summary.html"><a href="18-Summary.html#data-preparation"><i class="fa fa-check"></i><b>18.2</b> Data preparation</a></li>
<li class="chapter" data-level="18.3" data-path="18-Summary.html"><a href="18-Summary.html#taxonomy"><i class="fa fa-check"></i><b>18.3</b> Taxonomy</a></li>
<li class="chapter" data-level="18.4" data-path="18-Summary.html"><a href="18-Summary.html#diversity-analysis"><i class="fa fa-check"></i><b>18.4</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="18-Summary.html"><a href="18-Summary.html#alpha-beta"><i class="fa fa-check"></i><b>18.4.1</b> Alpha &amp; beta</a></li>
<li class="chapter" data-level="18.4.2" data-path="18-Summary.html"><a href="18-Summary.html#differential-abundance-anlaysis"><i class="fa fa-check"></i><b>18.4.2</b> Differential abundance anlaysis</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="18-Summary.html"><a href="18-Summary.html#findings"><i class="fa fa-check"></i><b>18.5</b> Findings</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="18-Summary.html"><a href="18-Summary.html#anthropisation-gradient"><i class="fa fa-check"></i><b>18.5.1</b> Anthropisation gradient</a></li>
<li class="chapter" data-level="18.5.2" data-path="18-Summary.html"><a href="18-Summary.html#replicate-differences"><i class="fa fa-check"></i><b>18.5.2</b> Replicate differences</a></li>
<li class="chapter" data-level="18.5.3" data-path="18-Summary.html"><a href="18-Summary.html#site-or-media"><i class="fa fa-check"></i><b>18.5.3</b> Site or media</a></li>
<li class="chapter" data-level="18.5.4" data-path="18-Summary.html"><a href="18-Summary.html#media-versus-env-samples"><i class="fa fa-check"></i><b>18.5.4</b> Media versus ENV samples</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="18-Summary.html"><a href="18-Summary.html#conclusion"><i class="fa fa-check"></i><b>18.6</b> Conclusion</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="19-Appendix.html"><a href="19-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="19-Appendix.html"><a href="19-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
<li class="chapter" data-level="A.2" data-path="19-Appendix.html"><a href="19-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>A.2</b> Jupyter-notebook</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="19-Appendix.html"><a href="19-Appendix.html#r-packages-1"><i class="fa fa-check"></i><b>B</b> R packages</a>
<ul>
<li class="chapter" data-level="B.1" data-path="19-Appendix.html"><a href="19-Appendix.html#ggplot2_appendix"><i class="fa fa-check"></i><b>B.1</b> ggplot2</a></li>
<li class="chapter" data-level="B.2" data-path="19-Appendix.html"><a href="19-Appendix.html#phyloseq_appendix"><i class="fa fa-check"></i><b>B.2</b> Phyloseq</a></li>
<li class="chapter" data-level="B.3" data-path="19-Appendix.html"><a href="19-Appendix.html#importing-into-phyloseq"><i class="fa fa-check"></i><b>B.3</b> Importing into phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-metrics"><i class="fa fa-check"></i><b>C</b> Diversity metrics</a>
<ul>
<li class="chapter" data-level="C.1" data-path="19-Appendix.html"><a href="19-Appendix.html#alpha_appendix"><i class="fa fa-check"></i><b>C.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="19-Appendix.html"><a href="19-Appendix.html#obvs"><i class="fa fa-check"></i><b>C.1.1</b> Observed features</a></li>
<li class="chapter" data-level="C.1.2" data-path="19-Appendix.html"><a href="19-Appendix.html#chao"><i class="fa fa-check"></i><b>C.1.2</b> Chao</a></li>
<li class="chapter" data-level="C.1.3" data-path="19-Appendix.html"><a href="19-Appendix.html#evenness"><i class="fa fa-check"></i><b>C.1.3</b> Evenness</a></li>
<li class="chapter" data-level="C.1.4" data-path="19-Appendix.html"><a href="19-Appendix.html#faith"><i class="fa fa-check"></i><b>C.1.4</b> Faith's PD (phylogenetic diversity)</a></li>
<li class="chapter" data-level="C.1.5" data-path="19-Appendix.html"><a href="19-Appendix.html#simpson"><i class="fa fa-check"></i><b>C.1.5</b> Simpson</a></li>
<li class="chapter" data-level="C.1.6" data-path="19-Appendix.html"><a href="19-Appendix.html#simpsone"><i class="fa fa-check"></i><b>C.1.6</b> Simpson evenness measure (simpson_e)</a></li>
<li class="chapter" data-level="C.1.7" data-path="19-Appendix.html"><a href="19-Appendix.html#shannon"><i class="fa fa-check"></i><b>C.1.7</b> Shannon</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="19-Appendix.html"><a href="19-Appendix.html#beta_appendix"><i class="fa fa-check"></i><b>C.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="19-Appendix.html"><a href="19-Appendix.html#unifrac_explanation"><i class="fa fa-check"></i><b>C.2.1</b> Weighted and unweighted UniFrac distances</a></li>
<li class="chapter" data-level="C.2.2" data-path="19-Appendix.html"><a href="19-Appendix.html#bray_explanation"><i class="fa fa-check"></i><b>C.2.2</b> Bray-Curtis</a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-resources"><i class="fa fa-check"></i><b>C.3</b> Diversity Resources</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R community analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="import_chap" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Import<a href="07-Import.html#import_chap" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/import.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Before carrying out any analysis we first need to import our QIIME2 artifacts into a <code>phyloseq</code> object. Thankfully there is an R package called <code>qiime2R</code>.</p>
<div id="import-notebook" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Import: notebook<a href="07-Import.html#import-notebook" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/r_script.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Prior to any coding, we will create a new directory, our analysis directory.
afterwards, reate a new notebook called <strong>"01-Import.ipynb"</strong> in it.
We will be creating many notebooks, so we'll number them so we can easily see the order of scripts.</p>
<p>First create a new directory.</p>
<ul>
<li>In the notebook file explorer, click the <strong>"New"</strong> button.</li>
<li>Select <strong>"Folder"</strong>.</li>
</ul>
<p>You will have an <strong>"Untitled Folder"</strong>. To rename it:</p>
<ul>
<li>Click on the box left of the name.</li>
<li>Press the <strong>"Rename"</strong> button that appeared.</li>
<li>Change the name to <strong>"R_community_workshop"</strong>.</li>
<li>Click <strong>"Rename"</strong>.</li>
</ul>
<p>Click on your <strong>"R_community_workshop"</strong> folder to move into it.</p>
<p>Next step is to create a new R notebook, rename it to <strong>"01-Import"</strong>, and save it.</p>
</div>
<div id="qiime2r" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> qiime2R<a href="07-Import.html#qiime2r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/qiime2r.png" width="40%" style="display: block; margin: auto;" /></p>
<p><a href="https://github.com/jbisanz/qiime2R"><code>qiime2R</code></a> is an R package for importing QIIME2 artifacts into a R <code>phyloseq</code> object. The package contains many different commands. Its function <code>read_qza()</code> can read a single artifact at a time.</p>
<p>The best way to import all your QIIME2 artifacts is with the <code>qiime2R::qza_to_phyloseq()</code> function. In your <strong>"01-Import.ipynb"</strong> script, add the following and run the commands.</p>
<p><strong>Tip</strong>: You can tab complete and/or copy and paste file paths within the webVNC.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="07-Import.html#cb12-1" tabindex="-1"></a><span class="co">#Cell 1</span></span>
<span id="cb12-2"><a href="07-Import.html#cb12-2" tabindex="-1"></a><span class="co">#Load the package/library</span></span>
<span id="cb12-3"><a href="07-Import.html#cb12-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;qiime2R&quot;</span>)</span>
<span id="cb12-4"><a href="07-Import.html#cb12-4" tabindex="-1"></a></span>
<span id="cb12-5"><a href="07-Import.html#cb12-5" tabindex="-1"></a><span class="co">#Import data</span></span>
<span id="cb12-6"><a href="07-Import.html#cb12-6" tabindex="-1"></a>pseq <span class="ot">&lt;-</span> qiime2R<span class="sc">::</span><span class="fu">qza_to_phyloseq</span>(</span>
<span id="cb12-7"><a href="07-Import.html#cb12-7" tabindex="-1"></a>  <span class="at">features =</span> <span class="st">&quot;/pub14/tea/nsc206/NEOF/R_community/data/table-dada2.qza&quot;</span>,</span>
<span id="cb12-8"><a href="07-Import.html#cb12-8" tabindex="-1"></a>  <span class="at">tree =</span> <span class="st">&quot;/pub14/tea/nsc206/NEOF/R_community/data/rooted-tree.qza&quot;</span>,</span>
<span id="cb12-9"><a href="07-Import.html#cb12-9" tabindex="-1"></a>  <span class="at">taxonomy =</span> <span class="st">&quot;/pub14/tea/nsc206/NEOF/R_community/data/taxonomy.sklearn.qza&quot;</span>,</span>
<span id="cb12-10"><a href="07-Import.html#cb12-10" tabindex="-1"></a>  <span class="at">metadata =</span> <span class="st">&quot;/pub14/tea/nsc206/NEOF/R_community/data/media_metadata.txt&quot;</span></span>
<span id="cb12-11"><a href="07-Import.html#cb12-11" tabindex="-1"></a>)</span></code></pre></div>
<p>This command creates a phyloseq object named <code>pseq</code>. It contains:</p>
<ul>
<li>The ASV abundance table (<code>features = "table-dada2.qza"</code>).</li>
<li>The rooted phylogenetic tree (<code>tree = "rooted-tree.qza"</code>).</li>
<li>The taxonomic classifications of the ASVs (<code>taxonomy = "taxonomy.sklearn.qza"</code>).</li>
<li>The sample metadata (<code>metadata = "media_metadata.txt"</code>)</li>
</ul>
</div>
<div id="import-summarise-phyloseq" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Import: Summarise phyloseq<a href="07-Import.html#import-summarise-phyloseq" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/list_blue.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Now that we have imported the data we can extract some summary information from it.</p>
<p>First we will use the <code>microbiome</code> package with its <code>summarize_phyloseq()</code> function.</p>
<p>Create a new <strong>cell</strong> and write + run the below in it.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="07-Import.html#cb13-1" tabindex="-1"></a><span class="co">#Cell 2</span></span>
<span id="cb13-2"><a href="07-Import.html#cb13-2" tabindex="-1"></a><span class="co">#Load microbiome library</span></span>
<span id="cb13-3"><a href="07-Import.html#cb13-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;microbiome&quot;</span>)</span>
<span id="cb13-4"><a href="07-Import.html#cb13-4" tabindex="-1"></a><span class="co">#Summary of phyloseq object</span></span>
<span id="cb13-5"><a href="07-Import.html#cb13-5" tabindex="-1"></a>microbiome<span class="sc">::</span><span class="fu">summarize_phyloseq</span>(pseq)</span></code></pre></div>
<p>This gives us a plethora of information:</p>
<ul>
<li>The top line tells us if the data is compositional (relative abundance).</li>
<li>We get the following list of values in a paragraph and via a list.
<ul>
<li><strong>Min. number of reads:</strong> Number of reads in the sample with the lowest number of reads.</li>
<li><strong>Max. number of reads:</strong> Number of reads in the sample with the largest number of reads.</li>
<li><strong>Total number of reads:</strong> Sum of all reads across all samples.</li>
<li><strong>Average number of reads:</strong> Sum of all reads / number of samples.</li>
<li><strong>Median number of reads:</strong> Midpoint read abundance across samples.</li>
<li><strong>Sparsity:</strong> See expandable box further down.</li>
<li><strong>Any OTU sum to 1 or less?:</strong> States if there are any ASVs with a summed abundance of 1 or less across all the samples.</li>
<li><strong>Number of singletons:</strong> Number of ASVs with a sum of 1 across all samples.</li>
<li><strong>Percent of OTUs that are singletons:</strong> Percentage of ASVs that only contain one read across all the samples.</li>
<li><strong>Number of sample variables are:</strong> Number of sample variables/groupings in our metadata.</li>
<li>The last line shows the names of the sample variables/groupings in our metadata.</li>
</ul></li>
</ul>
<div class="webex-solution">
<button>
Sparsity
</button>
<p>Sparsity is a measure of the number of 0s in a table. It can be represented by the following equation:</p>
<p><span class="math display">\[
sparsity = Z/C
\]</span></p>
<p>Where:</p>
<ul>
<li>Z = The number of cells that equal zero.</li>
<li>C = The total number of cells.</li>
</ul>
<p>Let's look at an example of an abundance table with a small amount of ASVs and Samples.</p>
<table>
<thead>
<tr class="header">
<th></th>
<th>Sample1</th>
<th>Sample2</th>
<th>Sample3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ASV1</td>
<td>0</td>
<td>10</td>
<td>24</td>
</tr>
<tr class="even">
<td>ASV2</td>
<td>1</td>
<td>0</td>
<td>37</td>
</tr>
<tr class="odd">
<td>ASV3</td>
<td>6</td>
<td>25</td>
<td>0</td>
</tr>
<tr class="even">
<td>ASV4</td>
<td>51</td>
<td>2</td>
<td>0</td>
</tr>
</tbody>
</table>
<ul>
<li>This abundance table has 12 cells (3 samples * 4 ASVs).</li>
<li>Of these cells, 4 have an abundance of zero.</li>
<li>4/12 = 0.3333, therefore its sparsity is 0.3333.</li>
</ul>
<p>Sparsity can be any value from 0-1. The higher the value the more sparse it is, with a value of 1 meaning all the cells have an abundance of zero. The lower the value the less sparse it is, with a value of 0 meaning all the cells have an abundance of 1 or more.</p>
<p>16S data is known to be sparse so high sparsity is not unexpected. Keep in mind that lower levels of taxa (ASVs, Species, &amp; Genera) will generally have more sparse tables that higher levels of taxa (Kingdom, Phylum, Class).</p>
</div>
<p>If you would like to see how the function calculates its values you can view the <a href="https://rdrr.io/github/microbiome/microbiome/src/R/summarize_phyloseq.R">source code online</a>.</p>
</div>
<div id="save-the-phyloseq-object" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Save the phyloseq object<a href="07-Import.html#save-the-phyloseq-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/r_save.png" width="15%" style="display: block; margin: auto;" /></p>
<p>When using multiple notebooks/scripts for analysis it is useful to save the R objects that will be used in different notebooks/scripts. This can be carried out with the function <code>save()</code>.</p>
<p>Write and run the following code in a third <strong>cell</strong>.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="07-Import.html#cb14-1" tabindex="-1"></a><span class="co">#Cell 3</span></span>
<span id="cb14-2"><a href="07-Import.html#cb14-2" tabindex="-1"></a><span class="co">#Save phyloseq as file</span></span>
<span id="cb14-3"><a href="07-Import.html#cb14-3" tabindex="-1"></a><span class="fu">save</span>(pseq, <span class="at">file =</span> <span class="st">&quot;phyloseq.RData&quot;</span>)</span></code></pre></div>
<p>This saves our object <code>pseq</code> into the file <code>phyloseq.RData</code>. The suffix <code>.RData</code> is the normal convention.</p>
<p>We have saved our final object of the notebook. Close and halt it.</p>
</div>
<div id="import-recap" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Import: recap<a href="07-Import.html#import-recap" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/recap.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We have imported our QIIME2 artifacts into one <code>phyloseq</code> object so we can analyse the data in R. This object has been saved into a ".RData" file which we will load in the next chapter.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="06-Preprocess_part.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="08-Summarise_phyloseq.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
