<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Minimum read depth | R community analysis</title>
  <meta name="description" content="CGR book for the R community analysis workflow" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Minimum read depth | R community analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/CGR_strips.png" />
  <meta property="og:description" content="CGR book for the R community analysis workflow" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Minimum read depth | R community analysis" />
  
  <meta name="twitter:description" content="CGR book for the R community analysis workflow" />
  <meta name="twitter:image" content="/figures/CGR_strips.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-07-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/CGR_favicon.png" type="image/x-icon" />
<link rel="prev" href="08-Summarise_phyloseq.html"/>
<link rel="next" href="10-Relative_abundance.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://www.liverpool.ac.uk/genomic-research/" target="blank"><img src="figures/CGR_strips.png"></a></li>
<li><a>R community analysis</a></li>

<li class="divider"></li>
<li class="part"><span><b>Intro</b></span></li>
<li class="chapter" data-level="1" data-path="01-R_community_main_workflow.html"><a href="01-R_community_main_workflow.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html"><i class="fa fa-check"></i><b>2</b> Dataset &amp; workflow</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#dataset"><i class="fa fa-check"></i><b>2.1</b> Dataset</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sites"><i class="fa fa-check"></i><b>2.1.1</b> Sites</a></li>
<li class="chapter" data-level="2.1.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#culture-media"><i class="fa fa-check"></i><b>2.1.2</b> Culture media</a></li>
<li class="chapter" data-level="2.1.3" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sum_and_qs"><i class="fa fa-check"></i><b>2.1.3</b> Summary &amp; questions</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#workflow"><i class="fa fa-check"></i><b>2.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-R_packages.html"><a href="03-R_packages.html"><i class="fa fa-check"></i><b>3</b> R Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_packages.html"><a href="03-R_packages.html#r-packageslibraries"><i class="fa fa-check"></i><b>3.1</b> R packages/libraries</a></li>
<li class="chapter" data-level="3.2" data-path="03-R_packages.html"><a href="03-R_packages.html#the-grammar-of-graphics"><i class="fa fa-check"></i><b>3.2</b> The grammar of graphics</a></li>
<li class="chapter" data-level="3.3" data-path="03-R_packages.html"><a href="03-R_packages.html#phyloseq"><i class="fa fa-check"></i><b>3.3</b> phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Setup.html"><a href="04-Setup.html"><i class="fa fa-check"></i><b>4</b> Set-up</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Setup.html"><a href="04-Setup.html#cluster"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Setup.html"><a href="04-Setup.html#mamba"><i class="fa fa-check"></i><b>4.2</b> Mamba</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Jupyter.html"><a href="05-Jupyter.html"><i class="fa fa-check"></i><b>5</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Jupyter.html"><a href="05-Jupyter.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.2" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-r-notebook"><i class="fa fa-check"></i><b>5.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.3" data-path="05-Jupyter.html"><a href="05-Jupyter.html#cells-and-code"><i class="fa fa-check"></i><b>5.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.4" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-new-cells"><i class="fa fa-check"></i><b>5.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.5" data-path="05-Jupyter.html"><a href="05-Jupyter.html#running-code"><i class="fa fa-check"></i><b>5.5</b> Running code</a></li>
<li class="chapter" data-level="5.6" data-path="05-Jupyter.html"><a href="05-Jupyter.html#saving-the-file"><i class="fa fa-check"></i><b>5.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.7" data-path="05-Jupyter.html"><a href="05-Jupyter.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.8" data-path="05-Jupyter.html"><a href="05-Jupyter.html#close-the-notebook"><i class="fa fa-check"></i><b>5.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.9" data-path="05-Jupyter.html"><a href="05-Jupyter.html#video-tutorial"><i class="fa fa-check"></i><b>5.9</b> Video tutorial</a></li>
</ul></li>
<li class="part"><span><b>Data preparation</b></span></li>
<li class="chapter" data-level="6" data-path="06-Preprocess_part.html"><a href="06-Preprocess_part.html"><i class="fa fa-check"></i><b>6</b> Data prep intro</a></li>
<li class="chapter" data-level="7" data-path="07-Import.html"><a href="07-Import.html"><i class="fa fa-check"></i><b>7</b> Import</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Import.html"><a href="07-Import.html#import-notebook"><i class="fa fa-check"></i><b>7.1</b> Import: notebook</a></li>
<li class="chapter" data-level="7.2" data-path="07-Import.html"><a href="07-Import.html#qiime2r"><i class="fa fa-check"></i><b>7.2</b> qiime2R</a></li>
<li class="chapter" data-level="7.3" data-path="07-Import.html"><a href="07-Import.html#import-summarise-phyloseq"><i class="fa fa-check"></i><b>7.3</b> Import: Summarise phyloseq</a></li>
<li class="chapter" data-level="7.4" data-path="07-Import.html"><a href="07-Import.html#save-the-phyloseq-object"><i class="fa fa-check"></i><b>7.4</b> Save the phyloseq object</a></li>
<li class="chapter" data-level="7.5" data-path="07-Import.html"><a href="07-Import.html#import-recap"><i class="fa fa-check"></i><b>7.5</b> Import: recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html"><i class="fa fa-check"></i><b>8</b> Summarise phyloseq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-setup"><i class="fa fa-check"></i><b>8.1</b> Summarise: setup</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-header"><i class="fa fa-check"></i><b>8.1.1</b> Summarise header</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-phyloseq"><i class="fa fa-check"></i><b>8.1.2</b> Summarise phyloseq</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#reads-per-sample"><i class="fa fa-check"></i><b>8.1.3</b> Reads per sample</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#display-png"><i class="fa fa-check"></i><b>8.1.4</b> Display png</a></li>
<li class="chapter" data-level="8.1.5" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#asvs-per-sample"><i class="fa fa-check"></i><b>8.1.5</b> ASVs per sample</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-recap"><i class="fa fa-check"></i><b>8.2</b> Summarise: recap</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html"><i class="fa fa-check"></i><b>9</b> Minimum read depth</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#considerations"><i class="fa fa-check"></i><b>9.1</b> Considerations</a></li>
<li class="chapter" data-level="9.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minimum-read-depth-section-title"><i class="fa fa-check"></i><b>9.2</b> Minimum read depth section title</a></li>
<li class="chapter" data-level="9.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#read-depth-vector-and-histogram"><i class="fa fa-check"></i><b>9.3</b> Read depth vector and histogram</a></li>
<li class="chapter" data-level="9.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#depth-boxplots"><i class="fa fa-check"></i><b>9.4</b> Sample depth boxplot</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#creating-a-data-frame-for-boxplots"><i class="fa fa-check"></i><b>9.4.1</b> Creating a data frame for boxplots</a></li>
<li class="chapter" data-level="9.4.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#ggplot2-boxplot"><i class="fa fa-check"></i><b>9.4.2</b> ggplot2 boxplot</a></li>
<li class="chapter" data-level="9.4.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#save-ggplot-as-png"><i class="fa fa-check"></i><b>9.4.3</b> Save ggplot as png</a></li>
<li class="chapter" data-level="9.4.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#media-by-depth-boxplot"><i class="fa fa-check"></i><b>9.4.4</b> Media by depth boxplot</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#chap9rarefaction"><i class="fa fa-check"></i><b>9.5</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#asv-abundance-data-frame"><i class="fa fa-check"></i><b>9.5.1</b> ASV abundance data frame</a></li>
<li class="chapter" data-level="9.5.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#vegans-rarecurve"><i class="fa fa-check"></i><b>9.5.2</b> vegan's rarecurve</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#filtering-by-minimum-read-depth"><i class="fa fa-check"></i><b>9.6</b> Filtering by minimum read depth</a></li>
<li class="chapter" data-level="9.7" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minmum-read-depth-summary"><i class="fa fa-check"></i><b>9.7</b> Minmum read depth: Summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy</b></span></li>
<li class="chapter" data-level="10" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html"><i class="fa fa-check"></i><b>10</b> Taxa relative abundance</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-setup"><i class="fa fa-check"></i><b>10.1</b> Taxa relative abundance: setup</a></li>
<li class="chapter" data-level="10.2" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#relative-abundance-transformation"><i class="fa fa-check"></i><b>10.2</b> Relative abundance transformation</a></li>
<li class="chapter" data-level="10.3" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#aggregate-taxa"><i class="fa fa-check"></i><b>10.3</b> Aggregate taxa</a></li>
<li class="chapter" data-level="10.4" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-summary"><i class="fa fa-check"></i><b>10.4</b> Taxa Relative abundance: summary</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html"><i class="fa fa-check"></i><b>11</b> Taxa plots</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#simple-bar-chart"><i class="fa fa-check"></i><b>11.1</b> Simple bar chart</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#editing-the-plot"><i class="fa fa-check"></i><b>11.1.1</b> Editing the plot</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-heatmap"><i class="fa fa-check"></i><b>11.2</b> Taxa heatmap</a></li>
<li class="chapter" data-level="11.3" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#aggregateraretaxa"><i class="fa fa-check"></i><b>11.3</b> Aggregate rare taxa</a></li>
<li class="chapter" data-level="11.4" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-media"><i class="fa fa-check"></i><b>11.4</b> Plot by media</a></li>
<li class="chapter" data-level="11.5" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-site-and-media"><i class="fa fa-check"></i><b>11.5</b> Plot by site and media</a></li>
<li class="chapter" data-level="11.6" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-plots-summary"><i class="fa fa-check"></i><b>11.6</b> Taxa plots: summary</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html"><i class="fa fa-check"></i><b>12</b> Family &amp; Genus plots</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-taxa-plots"><i class="fa fa-check"></i><b>12.1</b> Family taxa plots</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#aggregate-families"><i class="fa fa-check"></i><b>12.1.1</b> Aggregate families</a></li>
<li class="chapter" data-level="12.1.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-heatmap"><i class="fa fa-check"></i><b>12.1.2</b> Family heatmap</a></li>
<li class="chapter" data-level="12.1.3" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-rare-aggregation"><i class="fa fa-check"></i><b>12.1.3</b> Family rare aggregation</a></li>
<li class="chapter" data-level="12.1.4" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-bar-chart"><i class="fa fa-check"></i><b>12.1.4</b> Family bar chart</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots"><i class="fa fa-check"></i><b>12.2</b> Genus taxa plots</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots-solutions"><i class="fa fa-check"></i><b>12.2.1</b> Genus taxa plots solutions</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Diversity analysis</b></span></li>
<li class="chapter" data-level="13" data-path="13-Diversity_analysis.html"><a href="13-Diversity_analysis.html"><i class="fa fa-check"></i><b>13</b> Diversity analysis intro</a></li>
<li class="chapter" data-level="14" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html"><i class="fa fa-check"></i><b>14</b> Rarefaction</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-setup"><i class="fa fa-check"></i><b>14.1</b> Rarefaction: setup</a></li>
<li class="chapter" data-level="14.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve"><i class="fa fa-check"></i><b>14.2</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#bird-analogy"><i class="fa fa-check"></i><b>14.2.1</b> Bird analogy</a></li>
<li class="chapter" data-level="14.2.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-simple-plot"><i class="fa fa-check"></i><b>14.2.2</b> Rarefaction curve: simple plot</a></li>
<li class="chapter" data-level="14.2.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-better-plot"><i class="fa fa-check"></i><b>14.2.3</b> Rarefaction curve: better plot</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-slope"><i class="fa fa-check"></i><b>14.3</b> Rarefaction slope</a></li>
<li class="chapter" data-level="14.4" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefy-data"><i class="fa fa-check"></i><b>14.4</b> Rarefy data</a></li>
<li class="chapter" data-level="14.5" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-summary"><i class="fa fa-check"></i><b>14.5</b> Rarefaction: summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Alpha.html"><a href="15-Alpha.html"><i class="fa fa-check"></i><b>15</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-setup"><i class="fa fa-check"></i><b>15.1</b> Alpha: setup</a></li>
<li class="chapter" data-level="15.2" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-simple-plot"><i class="fa fa-check"></i><b>15.2</b> Alpha: simple plot</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="15-Alpha.html"><a href="15-Alpha.html#metric-choice"><i class="fa fa-check"></i><b>15.2.1</b> Metric choice</a></li>
<li class="chapter" data-level="15.2.2" data-path="15-Alpha.html"><a href="15-Alpha.html#sample-grouping-box-plot"><i class="fa fa-check"></i><b>15.2.2</b> Sample grouping &amp; box plot</a></li>
<li class="chapter" data-level="15.2.3" data-path="15-Alpha.html"><a href="15-Alpha.html#png-height-and-width-choice"><i class="fa fa-check"></i><b>15.2.3</b> PNG height and width choice</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats"><i class="fa fa-check"></i><b>15.3</b> Alpha: stats</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats-task"><i class="fa fa-check"></i><b>15.3.1</b> Alpha stats task</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-diversity-violin-plot"><i class="fa fa-check"></i><b>15.4</b> Alpha diversity: violin plot</a></li>
<li class="chapter" data-level="15.5" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-site-plots-and-statistics"><i class="fa fa-check"></i><b>15.5</b> Alpha: Site plots and statistics</a></li>
<li class="chapter" data-level="15.6" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-summary"><i class="fa fa-check"></i><b>15.6</b> Alpha: summary</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Beta.html"><a href="16-Beta.html"><i class="fa fa-check"></i><b>16</b> Beta</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Beta.html"><a href="16-Beta.html#beta-setup"><i class="fa fa-check"></i><b>16.1</b> Beta: setup</a></li>
<li class="chapter" data-level="16.2" data-path="16-Beta.html"><a href="16-Beta.html#beta-ordination"><i class="fa fa-check"></i><b>16.2</b> Beta: ordination</a></li>
<li class="chapter" data-level="16.3" data-path="16-Beta.html"><a href="16-Beta.html#beta-plot-ordination"><i class="fa fa-check"></i><b>16.3</b> Beta: plot ordination</a></li>
<li class="chapter" data-level="16.4" data-path="16-Beta.html"><a href="16-Beta.html#beta-statistics"><i class="fa fa-check"></i><b>16.4</b> Beta: statistics</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="16-Beta.html"><a href="16-Beta.html#permanova"><i class="fa fa-check"></i><b>16.4.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="16.4.2" data-path="16-Beta.html"><a href="16-Beta.html#pairwise-permanova"><i class="fa fa-check"></i><b>16.4.2</b> Pairwise PERMANOVA</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="16-Beta.html"><a href="16-Beta.html#beta-facet-zoom"><i class="fa fa-check"></i><b>16.5</b> Beta: Facet zoom</a></li>
<li class="chapter" data-level="16.6" data-path="16-Beta.html"><a href="16-Beta.html#beta-subset-samples"><i class="fa fa-check"></i><b>16.6</b> Beta: subset samples</a></li>
<li class="chapter" data-level="16.7" data-path="16-Beta.html"><a href="16-Beta.html#permanova-tasks"><i class="fa fa-check"></i><b>16.7</b> PERMANOVA tasks</a></li>
<li class="chapter" data-level="16.8" data-path="16-Beta.html"><a href="16-Beta.html#beta-summary"><i class="fa fa-check"></i><b>16.8</b> Beta: Summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html"><i class="fa fa-check"></i><b>17</b> Differential abundance (DA) analysis</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-setup"><i class="fa fa-check"></i><b>17.1</b> DA: setup</a></li>
<li class="chapter" data-level="17.2" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-preprocess"><i class="fa fa-check"></i><b>17.2</b> DA: Preprocess</a></li>
<li class="chapter" data-level="17.3" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-analysis"><i class="fa fa-check"></i><b>17.3</b> DA: ANCOM-BC analysis</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#ancombc2-options"><i class="fa fa-check"></i><b>17.3.1</b> ANCOMBC2 options</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-results"><i class="fa fa-check"></i><b>17.4</b> DA: ANCOM-BC results</a></li>
<li class="chapter" data-level="17.5" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-create-signifcant-lfc-long-data-frame"><i class="fa fa-check"></i><b>17.5</b> DA: Create signifcant LFC long data frame</a></li>
<li class="chapter" data-level="17.6" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-biomarker-lfc-heatmap"><i class="fa fa-check"></i><b>17.6</b> DA: Biomarker LFC heatmap</a></li>
<li class="chapter" data-level="17.7" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-matching-bar-plot"><i class="fa fa-check"></i><b>17.7</b> DA: matching bar plot</a></li>
<li class="chapter" data-level="17.8" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-summary"><i class="fa fa-check"></i><b>17.8</b> DA: summary</a></li>
</ul></li>
<li class="part"><span><b>Outro</b></span></li>
<li class="chapter" data-level="18" data-path="18-Summary.html"><a href="18-Summary.html"><i class="fa fa-check"></i><b>18</b> Summary</a>
<ul>
<li class="chapter" data-level="18.1" data-path="18-Summary.html"><a href="18-Summary.html#dataset-1"><i class="fa fa-check"></i><b>18.1</b> Dataset</a></li>
<li class="chapter" data-level="18.2" data-path="18-Summary.html"><a href="18-Summary.html#data-preparation"><i class="fa fa-check"></i><b>18.2</b> Data preparation</a></li>
<li class="chapter" data-level="18.3" data-path="18-Summary.html"><a href="18-Summary.html#taxonomy"><i class="fa fa-check"></i><b>18.3</b> Taxonomy</a></li>
<li class="chapter" data-level="18.4" data-path="18-Summary.html"><a href="18-Summary.html#diversity-analysis"><i class="fa fa-check"></i><b>18.4</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="18-Summary.html"><a href="18-Summary.html#alpha-beta"><i class="fa fa-check"></i><b>18.4.1</b> Alpha &amp; beta</a></li>
<li class="chapter" data-level="18.4.2" data-path="18-Summary.html"><a href="18-Summary.html#differential-abundance-anlaysis"><i class="fa fa-check"></i><b>18.4.2</b> Differential abundance anlaysis</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="18-Summary.html"><a href="18-Summary.html#findings"><i class="fa fa-check"></i><b>18.5</b> Findings</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="18-Summary.html"><a href="18-Summary.html#anthropisation-gradient"><i class="fa fa-check"></i><b>18.5.1</b> Anthropisation gradient</a></li>
<li class="chapter" data-level="18.5.2" data-path="18-Summary.html"><a href="18-Summary.html#replicate-differences"><i class="fa fa-check"></i><b>18.5.2</b> Replicate differences</a></li>
<li class="chapter" data-level="18.5.3" data-path="18-Summary.html"><a href="18-Summary.html#site-or-media"><i class="fa fa-check"></i><b>18.5.3</b> Site or media</a></li>
<li class="chapter" data-level="18.5.4" data-path="18-Summary.html"><a href="18-Summary.html#media-versus-env-samples"><i class="fa fa-check"></i><b>18.5.4</b> Media versus ENV samples</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="18-Summary.html"><a href="18-Summary.html#conclusion"><i class="fa fa-check"></i><b>18.6</b> Conclusion</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="19-Appendix.html"><a href="19-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="19-Appendix.html"><a href="19-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
<li class="chapter" data-level="A.2" data-path="19-Appendix.html"><a href="19-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>A.2</b> Jupyter-notebook</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="19-Appendix.html"><a href="19-Appendix.html#r-packages-1"><i class="fa fa-check"></i><b>B</b> R packages</a>
<ul>
<li class="chapter" data-level="B.1" data-path="19-Appendix.html"><a href="19-Appendix.html#ggplot2_appendix"><i class="fa fa-check"></i><b>B.1</b> ggplot2</a></li>
<li class="chapter" data-level="B.2" data-path="19-Appendix.html"><a href="19-Appendix.html#phyloseq_appendix"><i class="fa fa-check"></i><b>B.2</b> Phyloseq</a></li>
<li class="chapter" data-level="B.3" data-path="19-Appendix.html"><a href="19-Appendix.html#importing-into-phyloseq"><i class="fa fa-check"></i><b>B.3</b> Importing into phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-metrics"><i class="fa fa-check"></i><b>C</b> Diversity metrics</a>
<ul>
<li class="chapter" data-level="C.1" data-path="19-Appendix.html"><a href="19-Appendix.html#alpha_appendix"><i class="fa fa-check"></i><b>C.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="19-Appendix.html"><a href="19-Appendix.html#obvs"><i class="fa fa-check"></i><b>C.1.1</b> Observed features</a></li>
<li class="chapter" data-level="C.1.2" data-path="19-Appendix.html"><a href="19-Appendix.html#chao"><i class="fa fa-check"></i><b>C.1.2</b> Chao</a></li>
<li class="chapter" data-level="C.1.3" data-path="19-Appendix.html"><a href="19-Appendix.html#evenness"><i class="fa fa-check"></i><b>C.1.3</b> Evenness</a></li>
<li class="chapter" data-level="C.1.4" data-path="19-Appendix.html"><a href="19-Appendix.html#faith"><i class="fa fa-check"></i><b>C.1.4</b> Faith's PD (phylogenetic diversity)</a></li>
<li class="chapter" data-level="C.1.5" data-path="19-Appendix.html"><a href="19-Appendix.html#simpson"><i class="fa fa-check"></i><b>C.1.5</b> Simpson</a></li>
<li class="chapter" data-level="C.1.6" data-path="19-Appendix.html"><a href="19-Appendix.html#simpsone"><i class="fa fa-check"></i><b>C.1.6</b> Simpson evenness measure (simpson_e)</a></li>
<li class="chapter" data-level="C.1.7" data-path="19-Appendix.html"><a href="19-Appendix.html#shannon"><i class="fa fa-check"></i><b>C.1.7</b> Shannon</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="19-Appendix.html"><a href="19-Appendix.html#beta_appendix"><i class="fa fa-check"></i><b>C.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="19-Appendix.html"><a href="19-Appendix.html#unifrac_explanation"><i class="fa fa-check"></i><b>C.2.1</b> Weighted and unweighted UniFrac distances</a></li>
<li class="chapter" data-level="C.2.2" data-path="19-Appendix.html"><a href="19-Appendix.html#bray_explanation"><i class="fa fa-check"></i><b>C.2.2</b> Bray-Curtis</a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-resources"><i class="fa fa-check"></i><b>C.3</b> Diversity Resources</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R community analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mindepthchap" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Minimum read depth<a href="09-Minimum_read_depth.html#mindepthchap" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/depth_diver.png" width="20%" style="display: block; margin: auto;" /></p>
<p>It is good to remove samples with a very low read depth (number of sequencing reads).
But determining an appropriate minimum read depth is not trivial.
This value will vary from study to study.
Normally, for 16S data, a depth of at least 20K per sample is suggested.
This is the general consensus for <strong>human microbiome</strong> data.</p>
<p>In this chapter we will:</p>
<ul>
<li>Cover a brief intro to considerations of what is an acceptable minimum depth for your dataset.</li>
<li>Reinvestigate our sample depths with a previously created histogram.</li>
<li>View the depth ranges of different sample groups (site &amp; media) with box plots.</li>
<li>Create a rarefaction curve to assess if the depth of our samples have captured a good amount of biodiversity.</li>
<li>Demonstrate how to filter samples by depth.</li>
</ul>
<div id="considerations" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Considerations<a href="09-Minimum_read_depth.html#considerations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/consider.png" width="15%" style="display: block; margin: auto;" /></p>
<p>There are 3 main considerations to take into account when choosing an appropriate depth for your dataset.</p>
<ul>
<li><strong>The biodiversity of your samples.</strong>
<ul>
<li>If your sample is very biodiverse, such as the human gut microbiome, you will need a good depth (&gt;20K per sample).</li>
<li>If your sample is less biodiverse, such as many geological environments or skin, then you will not need as much read depth.</li>
<li>Rarefaction curves are a good method to determine if your samples have enough depth. We will look at this in this chapter.</li>
</ul></li>
<li><strong>The biomass of your samples.</strong>
<ul>
<li>Some environments are hard to extract DNA from.</li>
<li>If this is the case for you, then people will hopefully accept that this is an unfortunate reality of life and you will use what you can.</li>
<li>Be careful of your conclusions. If you think your dataset doesn't have as much data as it could, do not make very definitive detailed claims.</li>
</ul></li>
<li><strong>Read depth of sample groups.</strong>
<ul>
<li>It may be possible that a few samples have a much lower depth than the rest. It may be a good idea to remove them.</li>
<li>However, these may all come from the same sample group and so you have lost all information of one group.</li>
<li>For instance, you may be comparing different geological surfaces and your rock samples have much lower read depths than the various soil samples.</li>
<li>For comparisons including the lower depth sample group (e.g. rock samples) you will need to retain the lower depth samples.</li>
</ul></li>
</ul>
<p>That is a brief overview of that topic. If you are interested in more I suggest you look at papers where they have studied an environment similar to yours.</p>
</div>
<div id="minimum-read-depth-section-title" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Minimum read depth section title<a href="09-Minimum_read_depth.html#minimum-read-depth-section-title" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/header_l2.png" width="10%" style="display: block; margin: auto;" /></p>
<p>In the next section of our <code>jupyter-notebook</code> we will investigate what the minimum read depth should be and remove samples with a lower depth.</p>
<p>Create a new <strong>"Markdown"</strong> <strong>cell</strong> and add the following 2nd level header.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="09-Minimum_read_depth.html#cb27-1" tabindex="-1"></a><span class="do">## Minimum read depth</span></span></code></pre></div>
</div>
<div id="read-depth-vector-and-histogram" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Read depth vector and histogram<a href="09-Minimum_read_depth.html#read-depth-vector-and-histogram" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/histogram.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We have already created a vector and histogram of the read depths across our samples. Scroll up your notebook to view these and answer the following MCQ:</p>
<ul>
<li>What is the approximate read depth range of our dataset?
<div id="radio_KMUKHUXIXZ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_KMUKHUXIXZ" value=""></input> <span><strong>1-8</strong></span></label><label><input type="radio" autocomplete="off" name="radio_KMUKHUXIXZ" value="answer"></input> <span><strong>10,000-18,000</strong></span></label><label><input type="radio" autocomplete="off" name="radio_KMUKHUXIXZ" value=""></input> <span><strong>20,000+</strong></span></label>
</div></li>
</ul>
<p>This is a good first step but what if we want to know how the read depths vary between sample groups?</p>
</div>
<div id="depth-boxplots" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Sample depth boxplot<a href="09-Minimum_read_depth.html#depth-boxplots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/boxplot.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We are going to use <code>ggplot2</code> to create a couple boxplots to show the sequencing depth ranges of the different sample groups (Site &amp; Media).</p>
<div id="creating-a-data-frame-for-boxplots" class="section level3 hasAnchor" number="9.4.1">
<h3><span class="header-section-number">9.4.1</span> Creating a data frame for boxplots<a href="09-Minimum_read_depth.html#creating-a-data-frame-for-boxplots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/metadata.png" width="10%" style="display: block; margin: auto;" /></p>
<p>First, we need to create an object containing our sample names, site &amp; media information, and the depth.
We will use this object to produce our boxplots.
Thankfully the metadata in our <code>phyloseq</code> object contains all this information except the depth.</p>
<p>In a new <strong>cell</strong> write and run the below code. This will extract the sample data (metadata) to a new object and display the top 6 rows of this new object.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="09-Minimum_read_depth.html#cb28-1" tabindex="-1"></a><span class="co">#Extract sample data as a separate R object</span></span>
<span id="cb28-2"><a href="09-Minimum_read_depth.html#cb28-2" tabindex="-1"></a>abundance_metadf <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq)</span>
<span id="cb28-3"><a href="09-Minimum_read_depth.html#cb28-3" tabindex="-1"></a><span class="co">#View top 6 rows of metadata data frame</span></span>
<span id="cb28-4"><a href="09-Minimum_read_depth.html#cb28-4" tabindex="-1"></a><span class="fu">head</span>(abundance_metadf)</span></code></pre></div>
<p>The 2 new functions above are:</p>
<ul>
<li><code>phyloseq::sample_data()</code>: Extracts the sample data (metadata) data frame from a <code>phyloseq</code> object.</li>
<li><code>head()</code>: Returns the first 6 parts of an R object by default.
<ul>
<li>It can be used for a vector, matrix, table, data frame, or function.</li>
<li>In the case of a data frame it returns the first 6 rows.</li>
</ul></li>
</ul>
<p>We need to add the depth information to our new data frame.
We have extracted this previously into an object called <code>sample_depths</code>.
However, before adding it we want to check if it has the same order of samples as the rows in <code>abundance_metadf</code>.</p>
<p>Write and run the below script in the same <strong>cell</strong>. The code uses <code>head()</code> to view the first 6 elements of <code>sample_depths</code> and the row names of <code>abundance_metadf</code>. Then the function <code>identical()</code> is used to see if they are identical (<code>TRUE</code>) or not (<code>FALSE</code>).</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="09-Minimum_read_depth.html#cb29-1" tabindex="-1"></a><span class="co">#Check if our vector of sample_depths has the same order as our metadata rows</span></span>
<span id="cb29-2"><a href="09-Minimum_read_depth.html#cb29-2" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">names</span>(sample_depths))</span>
<span id="cb29-3"><a href="09-Minimum_read_depth.html#cb29-3" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">row.names</span>(abundance_metadf))</span>
<span id="cb29-4"><a href="09-Minimum_read_depth.html#cb29-4" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">names</span>(sample_depths),<span class="fu">row.names</span>(abundance_metadf))</span></code></pre></div>
<p>The order of samples is identical so we can add the depth information to <code>abundance_metadf</code>. Carry this out in the same cell with the code below.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="09-Minimum_read_depth.html#cb30-1" tabindex="-1"></a><span class="co">#Add sample depths to metadata data frame</span></span>
<span id="cb30-2"><a href="09-Minimum_read_depth.html#cb30-2" tabindex="-1"></a>abundance_metadf[,<span class="st">&quot;depth&quot;</span>] <span class="ot">&lt;-</span> sample_depths</span>
<span id="cb30-3"><a href="09-Minimum_read_depth.html#cb30-3" tabindex="-1"></a><span class="co">#View top 6 rows of edited metadata dataframe</span></span>
<span id="cb30-4"><a href="09-Minimum_read_depth.html#cb30-4" tabindex="-1"></a><span class="fu">head</span>(abundance_metadf)</span></code></pre></div>
<p>Great! We will use this data frame to create 2 boxplots.</p>
</div>
<div id="ggplot2-boxplot" class="section level3 hasAnchor" number="9.4.2">
<h3><span class="header-section-number">9.4.2</span> ggplot2 boxplot<a href="09-Minimum_read_depth.html#ggplot2-boxplot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/boxplots_groups.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We are going to create 2 boxplots with <code>ggplot2</code>.
We won't go into too much detail on how the code works here, instead learning more later in this book.</p>
<p>The code below creates a <code>ggplot2</code> boxplot. We carry this out with 2 functions:</p>
<ul>
<li><code>ggplot()</code>: This creates a <code>ggplot2</code> object, storing the information and aesthetics.
<ul>
<li>The first option is the data we want to use for plotting (<code>abundance_metadf</code>).</li>
<li>The second option is the aesthetics (<code>aes()</code>) to plot. In this case we want the depth column to be plotted on the y-axis (<code>y=depth</code>) and the site column to be plotted on the x-axis (<code>x=site</code>).</li>
</ul></li>
<li><code>+</code>: We need to have a <code>+</code> at the end of the <code>ggplot()</code> function to add the next component of the plot.</li>
<li><code>geom_boxplot()</code>: This adds a <strong>layer</strong> to our <code>ggplot2</code> object. In this case it converts the <code>ggplot</code> object, which is just information, into a boxplot.</li>
</ul>
<p>Write and run the following code in a new cell:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="09-Minimum_read_depth.html#cb31-1" tabindex="-1"></a><span class="co">#Create ggplot2 boxplot of depth by site</span></span>
<span id="cb31-2"><a href="09-Minimum_read_depth.html#cb31-2" tabindex="-1"></a>boxplot <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(abundance_metadf, <span class="fu">aes</span>(<span class="at">y=</span>depth, <span class="at">x=</span>site)) <span class="sc">+</span></span>
<span id="cb31-3"><a href="09-Minimum_read_depth.html#cb31-3" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>()</span></code></pre></div>
<p>We have a saved <code>ggplot2</code> object but now we need to save it as a image file and display it.</p>
</div>
<div id="save-ggplot-as-png" class="section level3 hasAnchor" number="9.4.3">
<h3><span class="header-section-number">9.4.3</span> Save ggplot as png<a href="09-Minimum_read_depth.html#save-ggplot-as-png" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/png_file.png" width="10%" style="display: block; margin: auto;" /></p>
<p>You can save a <code>ggplot</code> object as an image file with <code>ggsave()</code>.
This allows you to select the size and resolution (<code>dpi</code>) of the image.
This can then be displayed with <code>IRdisplay::display_png()</code>.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="09-Minimum_read_depth.html#cb32-1" tabindex="-1"></a><span class="co">#Save ggplot object as png file</span></span>
<span id="cb32-2"><a href="09-Minimum_read_depth.html#cb32-2" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./boxplot_x_site_y_depth.png&quot;</span>, <span class="at">plot =</span> boxplot,</span>
<span id="cb32-3"><a href="09-Minimum_read_depth.html#cb32-3" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">100</span>, <span class="at">width =</span> <span class="dv">150</span>)</span>
<span id="cb32-4"><a href="09-Minimum_read_depth.html#cb32-4" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb32-5"><a href="09-Minimum_read_depth.html#cb32-5" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./boxplot_x_site_y_depth.png&quot;</span>)</span></code></pre></div>
<ul>
<li>Which site has the highest median depth (middle line of boxplot)?
<div id="radio_BSIIDNHDVD" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BSIIDNHDVD" value="answer"></input> <span><strong>LD (Lower Durance)</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BSIIDNHDVD" value=""></input> <span><strong>MD (Middle Durance)</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BSIIDNHDVD" value=""></input> <span><strong>UD (Upper Durance</strong></span></label>
</div></li>
</ul>
</div>
<div id="media-by-depth-boxplot" class="section level3 hasAnchor" number="9.4.4">
<h3><span class="header-section-number">9.4.4</span> Media by depth boxplot<a href="09-Minimum_read_depth.html#media-by-depth-boxplot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/petri.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will use the same code to plot the depth by media.
You can copy and paste the code changing all the instances of <strong>site</strong> to <strong>media</strong>.
Carry this out in the same cell and run the code.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="09-Minimum_read_depth.html#cb33-1" tabindex="-1"></a><span class="co">#Create ggplot2 boxplot of depth by media</span></span>
<span id="cb33-2"><a href="09-Minimum_read_depth.html#cb33-2" tabindex="-1"></a>boxplot <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(abundance_metadf, <span class="fu">aes</span>(<span class="at">y=</span>depth, <span class="at">x=</span>media)) <span class="sc">+</span></span>
<span id="cb33-3"><a href="09-Minimum_read_depth.html#cb33-3" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>()</span>
<span id="cb33-4"><a href="09-Minimum_read_depth.html#cb33-4" tabindex="-1"></a><span class="co">#Save ggplot object as png file</span></span>
<span id="cb33-5"><a href="09-Minimum_read_depth.html#cb33-5" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./boxplot_x_media_y_depth.png&quot;</span>, <span class="at">plot =</span> boxplot,</span>
<span id="cb33-6"><a href="09-Minimum_read_depth.html#cb33-6" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">100</span>, <span class="at">width =</span> <span class="dv">150</span>)</span>
<span id="cb33-7"><a href="09-Minimum_read_depth.html#cb33-7" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb33-8"><a href="09-Minimum_read_depth.html#cb33-8" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./boxplot_x_media_y_depth.png&quot;</span>)</span></code></pre></div>
<ul>
<li>Which media has the lowest median depth (middle line of boxplot)?
<div id="radio_IVJXDSHLSJ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_IVJXDSHLSJ" value="answer"></input> <span><strong>CVP</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IVJXDSHLSJ" value=""></input> <span><strong>ENV</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IVJXDSHLSJ" value=""></input> <span><strong>TSA</strong></span></label>
</div></li>
</ul>
<p>From the boxplots we can see there is no drastic difference between the depths of the different media groups. We will therefore continue and make some rarefaction curves to further assess the depth of our samples.</p>
<p>For more resources on <code>ggplot2</code> please see the <a href="19-Appendix.html#ggplot2_appendix">appendix</a> of this book.</p>
</div>
</div>
<div id="chap9rarefaction" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Rarefaction curve<a href="09-Minimum_read_depth.html#chap9rarefaction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/rarefaction.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Our read depths appear a bit low, each sample has &lt;20K reads.
However, this might be fine for our dataset since we are using surface water samples rather than human gut microbiome samples.
Let's see how our samples look with a rarefaction curve.</p>
<p><strong>Note</strong>: This is a quick example and we will go into more detail in the <a href="14-Rarefaction.html#rarefaction_chap">rarefaction chapter</a>.</p>
<p>Surprisingly, there is not a good method to produce a rarefaction curve with the <code>phyloseq</code> or <code>microbiome</code> packages.
We will therefore use the <a href="https://vegandevs.github.io/vegan/"><code>vegan</code></a> package.
<code>vegan</code> is an R package for community ecologists.
It has a variety of functions but it uses normal R data frames rather than <code>phyloseq</code> objects.
We will therefore only use it for rarefaction purposes.</p>
<div id="asv-abundance-data-frame" class="section level3 hasAnchor" number="9.5.1">
<h3><span class="header-section-number">9.5.1</span> ASV abundance data frame<a href="09-Minimum_read_depth.html#asv-abundance-data-frame" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/nice_table.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Before creating our rarefaction curve we will extract the ASV abundance table from the <code>phyloseq</code> object with <code>phyloseq</code>'s function <code>otu_table()</code>.
We need to transpose (<code>t()</code>) the table so it is in the correct orientation for the rarefaction function.
Additionally, we will ensure it is a data frame with the function <code>as.data.frame()</code>.</p>
<p>Carry this out in a new <strong>cell</strong>.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="09-Minimum_read_depth.html#cb34-1" tabindex="-1"></a><span class="co">#Rarefaction curve</span></span>
<span id="cb34-2"><a href="09-Minimum_read_depth.html#cb34-2" tabindex="-1"></a><span class="co">#Extract ASV table as transposed data frame</span></span>
<span id="cb34-3"><a href="09-Minimum_read_depth.html#cb34-3" tabindex="-1"></a>asv_abund_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(pseq)))</span></code></pre></div>
</div>
<div id="vegans-rarecurve" class="section level3 hasAnchor" number="9.5.2">
<h3><span class="header-section-number">9.5.2</span> vegan's rarecurve<a href="09-Minimum_read_depth.html#vegans-rarecurve" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/vegan.png" width="10%" style="display: block; margin: auto;" /></p>
<p>With this data frame we can create a rarefaction curve with <code>vegan</code>'s <code>rarecurve()</code> function.</p>
<p>Add the following code to the same cell and run it.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="09-Minimum_read_depth.html#cb35-1" tabindex="-1"></a><span class="co">#Save as png</span></span>
<span id="cb35-2"><a href="09-Minimum_read_depth.html#cb35-2" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename =</span> <span class="st">&quot;rarefaction.png&quot;</span>, <span class="at">res =</span> <span class="dv">300</span>,</span>
<span id="cb35-3"><a href="09-Minimum_read_depth.html#cb35-3" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">200</span>, <span class="at">width =</span> <span class="dv">200</span>)</span>
<span id="cb35-4"><a href="09-Minimum_read_depth.html#cb35-4" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb35-5"><a href="09-Minimum_read_depth.html#cb35-5" tabindex="-1"></a>vegan<span class="sc">::</span><span class="fu">rarecurve</span>(</span>
<span id="cb35-6"><a href="09-Minimum_read_depth.html#cb35-6" tabindex="-1"></a>  <span class="at">x =</span> asv_abund_df, <span class="at">step =</span> <span class="dv">50</span>,</span>
<span id="cb35-7"><a href="09-Minimum_read_depth.html#cb35-7" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">&quot;Read depth&quot;</span>,</span>
<span id="cb35-8"><a href="09-Minimum_read_depth.html#cb35-8" tabindex="-1"></a>  <span class="at">ylab =</span> <span class="st">&quot;ASVs&quot;</span></span>
<span id="cb35-9"><a href="09-Minimum_read_depth.html#cb35-9" tabindex="-1"></a>)</span>
<span id="cb35-10"><a href="09-Minimum_read_depth.html#cb35-10" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb35-11"><a href="09-Minimum_read_depth.html#cb35-11" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb35-12"><a href="09-Minimum_read_depth.html#cb35-12" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./rarefaction.png&quot;</span>)</span></code></pre></div>
<p><strong>Note</strong>: You will get a warning saying <code>"most observed count data have counts 1, but smallest count is 2"</code>. This can be ignored in this case.</p>
<p>In essence, we are hoping that the majority of samples have plateaued.
If the curves have flattened in relation to the y axis this indicates that most of the ASVs present in the sample have been captured.</p>
<p>Our samples have plateau'd or have gentle slopes towards the end.
With this we can be happy to continue and not remove samples by a minimum read depth.</p>
<p>If we saw some samples with steep curves and low depths we could carry out some more analysis with rarefaction.
However, more rarefaction functions, analysis, and theories will be covered in the <a href="14-Rarefaction.html#rarefaction_chap">rarefaction chapter</a>.
You could use some of these at this point to help you determine your minimum read depth for your own datasets.</p>
</div>
</div>
<div id="filtering-by-minimum-read-depth" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> Filtering by minimum read depth<a href="09-Minimum_read_depth.html#filtering-by-minimum-read-depth" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/water_filter.png" width="10%" style="display: block; margin: auto;" /></p>
<p>What if you want to filter samples by a minimum read depth?</p>
<p>You can use the <code>subset_samples()</code> function from <code>phyloseq()</code>.
We will use our previously created vector containing read depths (<code>sample_depths</code>) to remove sample with less than 11k reads.
We have chosen this depth as an example to remove some samples.</p>
<p>Write and run the below code in a new cell.
It will create a new subsetted <code>phyloseq</code> object.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="09-Minimum_read_depth.html#cb36-1" tabindex="-1"></a><span class="co">#Subset and keep samples with more than 11k reads</span></span>
<span id="cb36-2"><a href="09-Minimum_read_depth.html#cb36-2" tabindex="-1"></a>pseq_min11K <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(pseq, sample_depths <span class="sc">&gt;</span> <span class="dv">11000</span>)</span></code></pre></div>
<p>After removing samples it is also useful to remove ASVs with no abundance values.
This can occur when ASVs are only present in the samples which have been removed.</p>
<p>To remove these ASVs we can use two <code>phyloseq</code> functions:</p>
<ul>
<li><strong><code>taxa_sums()</code></strong>: Returns a vector showing sum of all taxa in the abundance table.
<ul>
<li>In our <code>phyloseq</code> object the ASVs are the taxa.</li>
<li>ASVs have long human unfriendly names that are unique to every single ASV possible.</li>
</ul></li>
<li><strong><code>prune_taxa()</code></strong>: This retains taxa/ASVs based on a provided vector.
<ul>
<li>We are creating a logical vector (<code>TRUE</code>/<code>FALSE</code>) where ASVs with 0 abundance are <code>FALSE</code> and ASVs with abundance &gt; 0 are <code>TRUE</code>.</li>
</ul></li>
</ul>
<p>We'll first write and run some commands with <code>taxa_sums()</code> to get some practice with it. Carry this out in the same <strong>cell</strong> as the <code>subset_samples()</code> command.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="09-Minimum_read_depth.html#cb37-1" tabindex="-1"></a><span class="co">#Abundance sums of the 1st six ASVs</span></span>
<span id="cb37-2"><a href="09-Minimum_read_depth.html#cb37-2" tabindex="-1"></a><span class="fu">head</span>(phyloseq<span class="sc">::</span><span class="fu">taxa_sums</span>(pseq_min11K))</span>
<span id="cb37-3"><a href="09-Minimum_read_depth.html#cb37-3" tabindex="-1"></a><span class="co">#View number of ASVs in our data</span></span>
<span id="cb37-4"><a href="09-Minimum_read_depth.html#cb37-4" tabindex="-1"></a><span class="fu">length</span>(phyloseq<span class="sc">::</span><span class="fu">taxa_sums</span>(pseq_min11K))</span></code></pre></div>
<p>In the same cell add the following.
This will filter out ASVs with no abundance.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="09-Minimum_read_depth.html#cb38-1" tabindex="-1"></a><span class="co">#Remove ASVs with no abundance</span></span>
<span id="cb38-2"><a href="09-Minimum_read_depth.html#cb38-2" tabindex="-1"></a>pseq_min11K <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">prune_taxa</span>(</span>
<span id="cb38-3"><a href="09-Minimum_read_depth.html#cb38-3" tabindex="-1"></a>  phyloseq<span class="sc">::</span><span class="fu">taxa_sums</span>(pseq_min11K) <span class="sc">&gt;</span> <span class="dv">0</span>, pseq_min11K</span>
<span id="cb38-4"><a href="09-Minimum_read_depth.html#cb38-4" tabindex="-1"></a>)</span></code></pre></div>
<p>Finally, summarise the contents of the <code>phyloseq</code> object.
Add the following in the same cell and then run the code in the cell.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="09-Minimum_read_depth.html#cb39-1" tabindex="-1"></a><span class="co">#Summarise subsetted phyloseq</span></span>
<span id="cb39-2"><a href="09-Minimum_read_depth.html#cb39-2" tabindex="-1"></a>microbiome<span class="sc">::</span><span class="fu">summarize_phyloseq</span>(pseq_min11K)</span>
<span id="cb39-3"><a href="09-Minimum_read_depth.html#cb39-3" tabindex="-1"></a>microbiome<span class="sc">::</span><span class="fu">readcount</span>(pseq_min11K)</span>
<span id="cb39-4"><a href="09-Minimum_read_depth.html#cb39-4" tabindex="-1"></a><span class="fu">length</span>(phyloseq<span class="sc">::</span><span class="fu">taxa_sums</span>(pseq_min11K))</span>
<span id="cb39-5"><a href="09-Minimum_read_depth.html#cb39-5" tabindex="-1"></a>pseq_min11K</span></code></pre></div>
<p>We can see that the data lost 26 ASVs (2551 - 2525).</p>
<p>Try to write your own R code in a new cell to answer the following questions:</p>
<ul>
<li>What is the difference of the minimum number of reads between <code>pseq_min11K</code> and <code>pseq</code>?
<div id="radio_CHDBOPWREY" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_CHDBOPWREY" value=""></input> <span><strong>4</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CHDBOPWREY" value="answer"></input> <span><strong>613</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CHDBOPWREY" value=""></input> <span><strong>42507</strong></span></label>
</div></li>
<li>How many samples were removed due to the read depth filtering?
<div id="radio_CYXAXAPVVG" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_CYXAXAPVVG" value="answer"></input> <span><strong>4</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CYXAXAPVVG" value=""></input> <span><strong>613</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CYXAXAPVVG" value=""></input> <span><strong>42507</strong></span></label>
</div></li>
<li>How many reads were removed by the minimum read filtering?
<div id="radio_QRZHMIRKIP" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_QRZHMIRKIP" value=""></input> <span><strong>4</strong></span></label><label><input type="radio" autocomplete="off" name="radio_QRZHMIRKIP" value=""></input> <span><strong>613</strong></span></label><label><input type="radio" autocomplete="off" name="radio_QRZHMIRKIP" value="answer"></input> <span><strong>42507</strong></span></label>
</div></li>
</ul>
<div class="webex-solution">
<button>
Tip
</button>
<p>Combining <code>microbiome::readcount()</code> with <code>min()</code>, <code>length()</code>, and <code>sum()</code> might help.</p>
</div>
<div class="webex-solution">
<button>
Code solutions + bonus
</button>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="09-Minimum_read_depth.html#cb40-1" tabindex="-1"></a><span class="co">#Difference of minimum read numbers</span></span>
<span id="cb40-2"><a href="09-Minimum_read_depth.html#cb40-2" tabindex="-1"></a><span class="fu">min</span>(microbiome<span class="sc">::</span><span class="fu">readcount</span>(pseq_min11K)) <span class="sc">-</span> <span class="fu">min</span>(microbiome<span class="sc">::</span><span class="fu">readcount</span>(pseq))</span>
<span id="cb40-3"><a href="09-Minimum_read_depth.html#cb40-3" tabindex="-1"></a><span class="co">#Number of samples lost</span></span>
<span id="cb40-4"><a href="09-Minimum_read_depth.html#cb40-4" tabindex="-1"></a><span class="fu">length</span>(microbiome<span class="sc">::</span><span class="fu">readcount</span>(pseq)) <span class="sc">-</span> <span class="fu">length</span>(microbiome<span class="sc">::</span><span class="fu">readcount</span>(pseq_min11K))</span>
<span id="cb40-5"><a href="09-Minimum_read_depth.html#cb40-5" tabindex="-1"></a><span class="co">#Number of reads removed</span></span>
<span id="cb40-6"><a href="09-Minimum_read_depth.html#cb40-6" tabindex="-1"></a><span class="fu">sum</span>(microbiome<span class="sc">::</span><span class="fu">readcount</span>(pseq)) <span class="sc">-</span> <span class="fu">sum</span>(microbiome<span class="sc">::</span><span class="fu">readcount</span>(pseq_min11K))</span>
<span id="cb40-7"><a href="09-Minimum_read_depth.html#cb40-7" tabindex="-1"></a><span class="co">#Bonus</span></span>
<span id="cb40-8"><a href="09-Minimum_read_depth.html#cb40-8" tabindex="-1"></a><span class="co">#List the removed samples</span></span>
<span id="cb40-9"><a href="09-Minimum_read_depth.html#cb40-9" tabindex="-1"></a><span class="fu">setdiff</span>(phyloseq<span class="sc">::</span><span class="fu">sample_names</span>(pseq), phyloseq<span class="sc">::</span><span class="fu">sample_names</span>(pseq_min11K))</span></code></pre></div>
</div>
<p>We won't actually use this subsetted file as we want to keep all the samples in this case.
Since we are not removing any samples we don't need to remove any ASVs as they should all have a total abundance &gt; 0.
You can therefore remove the new <code>phyloseq</code> object in a new <strong>cell</strong>.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="09-Minimum_read_depth.html#cb41-1" tabindex="-1"></a><span class="co">#Remove subsetted phyloseq</span></span>
<span id="cb41-2"><a href="09-Minimum_read_depth.html#cb41-2" tabindex="-1"></a><span class="fu">rm</span>(pseq_min11K)</span></code></pre></div>
<p>Once you are finished with this notebook you can save it then close and halt it.</p>
</div>
<div id="minmum-read-depth-summary" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> Minmum read depth: Summary<a href="09-Minimum_read_depth.html#minmum-read-depth-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_blue.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We have assessed the read depth in this chapter and decided to not remove any samples.
This assessment included:</p>
<ul>
<li>Viewing a histogram of sample read depths.</li>
<li>Creating boxplots to compare the sample read depths across sample groups (site and media).</li>
<li>Producing a rarefaction depth to determine if any samples did not represent a good amount of the ASVs present in the environment.</li>
</ul>
<p>Additionally, we created a new <code>phyloseq</code> object where the samples were filtered by depth.
Ultimately we did not keep the filtered <code>phyloseq</code> object but it is good to know how to do this for your future research.</p>

</div>
</div>



<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="08-Summarise_phyloseq.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="10-Relative_abundance.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
