<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 Family &amp; Genus plots | R community analysis</title>
  <meta name="description" content="CGR book for the R community analysis workflow" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 Family &amp; Genus plots | R community analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/CGR_strips.png" />
  <meta property="og:description" content="CGR book for the R community analysis workflow" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 Family &amp; Genus plots | R community analysis" />
  
  <meta name="twitter:description" content="CGR book for the R community analysis workflow" />
  <meta name="twitter:image" content="/figures/CGR_strips.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-07-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/CGR_favicon.png" type="image/x-icon" />
<link rel="prev" href="11-Taxa_plots.html"/>
<link rel="next" href="13-Diversity_analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://www.liverpool.ac.uk/genomic-research/" target="blank"><img src="figures/CGR_strips.png"></a></li>
<li><a>R community analysis</a></li>

<li class="divider"></li>
<li class="part"><span><b>Intro</b></span></li>
<li class="chapter" data-level="1" data-path="01-R_community_main_workflow.html"><a href="01-R_community_main_workflow.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html"><i class="fa fa-check"></i><b>2</b> Dataset &amp; workflow</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#dataset"><i class="fa fa-check"></i><b>2.1</b> Dataset</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sites"><i class="fa fa-check"></i><b>2.1.1</b> Sites</a></li>
<li class="chapter" data-level="2.1.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#culture-media"><i class="fa fa-check"></i><b>2.1.2</b> Culture media</a></li>
<li class="chapter" data-level="2.1.3" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sum_and_qs"><i class="fa fa-check"></i><b>2.1.3</b> Summary &amp; questions</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#workflow"><i class="fa fa-check"></i><b>2.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-R_packages.html"><a href="03-R_packages.html"><i class="fa fa-check"></i><b>3</b> R Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_packages.html"><a href="03-R_packages.html#r-packageslibraries"><i class="fa fa-check"></i><b>3.1</b> R packages/libraries</a></li>
<li class="chapter" data-level="3.2" data-path="03-R_packages.html"><a href="03-R_packages.html#the-grammar-of-graphics"><i class="fa fa-check"></i><b>3.2</b> The grammar of graphics</a></li>
<li class="chapter" data-level="3.3" data-path="03-R_packages.html"><a href="03-R_packages.html#phyloseq"><i class="fa fa-check"></i><b>3.3</b> phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Setup.html"><a href="04-Setup.html"><i class="fa fa-check"></i><b>4</b> Set-up</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Setup.html"><a href="04-Setup.html#cluster"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Setup.html"><a href="04-Setup.html#mamba"><i class="fa fa-check"></i><b>4.2</b> Mamba</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Jupyter.html"><a href="05-Jupyter.html"><i class="fa fa-check"></i><b>5</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Jupyter.html"><a href="05-Jupyter.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.2" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-r-notebook"><i class="fa fa-check"></i><b>5.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.3" data-path="05-Jupyter.html"><a href="05-Jupyter.html#cells-and-code"><i class="fa fa-check"></i><b>5.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.4" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-new-cells"><i class="fa fa-check"></i><b>5.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.5" data-path="05-Jupyter.html"><a href="05-Jupyter.html#running-code"><i class="fa fa-check"></i><b>5.5</b> Running code</a></li>
<li class="chapter" data-level="5.6" data-path="05-Jupyter.html"><a href="05-Jupyter.html#saving-the-file"><i class="fa fa-check"></i><b>5.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.7" data-path="05-Jupyter.html"><a href="05-Jupyter.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.8" data-path="05-Jupyter.html"><a href="05-Jupyter.html#close-the-notebook"><i class="fa fa-check"></i><b>5.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.9" data-path="05-Jupyter.html"><a href="05-Jupyter.html#video-tutorial"><i class="fa fa-check"></i><b>5.9</b> Video tutorial</a></li>
</ul></li>
<li class="part"><span><b>Data preparation</b></span></li>
<li class="chapter" data-level="6" data-path="06-Preprocess_part.html"><a href="06-Preprocess_part.html"><i class="fa fa-check"></i><b>6</b> Data prep intro</a></li>
<li class="chapter" data-level="7" data-path="07-Import.html"><a href="07-Import.html"><i class="fa fa-check"></i><b>7</b> Import</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Import.html"><a href="07-Import.html#import-notebook"><i class="fa fa-check"></i><b>7.1</b> Import: notebook</a></li>
<li class="chapter" data-level="7.2" data-path="07-Import.html"><a href="07-Import.html#qiime2r"><i class="fa fa-check"></i><b>7.2</b> qiime2R</a></li>
<li class="chapter" data-level="7.3" data-path="07-Import.html"><a href="07-Import.html#import-summarise-phyloseq"><i class="fa fa-check"></i><b>7.3</b> Import: Summarise phyloseq</a></li>
<li class="chapter" data-level="7.4" data-path="07-Import.html"><a href="07-Import.html#save-the-phyloseq-object"><i class="fa fa-check"></i><b>7.4</b> Save the phyloseq object</a></li>
<li class="chapter" data-level="7.5" data-path="07-Import.html"><a href="07-Import.html#import-recap"><i class="fa fa-check"></i><b>7.5</b> Import: recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html"><i class="fa fa-check"></i><b>8</b> Summarise phyloseq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-setup"><i class="fa fa-check"></i><b>8.1</b> Summarise: setup</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-header"><i class="fa fa-check"></i><b>8.1.1</b> Summarise header</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-phyloseq"><i class="fa fa-check"></i><b>8.1.2</b> Summarise phyloseq</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#reads-per-sample"><i class="fa fa-check"></i><b>8.1.3</b> Reads per sample</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#display-png"><i class="fa fa-check"></i><b>8.1.4</b> Display png</a></li>
<li class="chapter" data-level="8.1.5" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#asvs-per-sample"><i class="fa fa-check"></i><b>8.1.5</b> ASVs per sample</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-recap"><i class="fa fa-check"></i><b>8.2</b> Summarise: recap</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html"><i class="fa fa-check"></i><b>9</b> Minimum read depth</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#considerations"><i class="fa fa-check"></i><b>9.1</b> Considerations</a></li>
<li class="chapter" data-level="9.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minimum-read-depth-section-title"><i class="fa fa-check"></i><b>9.2</b> Minimum read depth section title</a></li>
<li class="chapter" data-level="9.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#read-depth-vector-and-histogram"><i class="fa fa-check"></i><b>9.3</b> Read depth vector and histogram</a></li>
<li class="chapter" data-level="9.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#depth-boxplots"><i class="fa fa-check"></i><b>9.4</b> Sample depth boxplot</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#creating-a-data-frame-for-boxplots"><i class="fa fa-check"></i><b>9.4.1</b> Creating a data frame for boxplots</a></li>
<li class="chapter" data-level="9.4.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#ggplot2-boxplot"><i class="fa fa-check"></i><b>9.4.2</b> ggplot2 boxplot</a></li>
<li class="chapter" data-level="9.4.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#save-ggplot-as-png"><i class="fa fa-check"></i><b>9.4.3</b> Save ggplot as png</a></li>
<li class="chapter" data-level="9.4.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#media-by-depth-boxplot"><i class="fa fa-check"></i><b>9.4.4</b> Media by depth boxplot</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#chap9rarefaction"><i class="fa fa-check"></i><b>9.5</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#asv-abundance-data-frame"><i class="fa fa-check"></i><b>9.5.1</b> ASV abundance data frame</a></li>
<li class="chapter" data-level="9.5.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#vegans-rarecurve"><i class="fa fa-check"></i><b>9.5.2</b> vegan's rarecurve</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#filtering-by-minimum-read-depth"><i class="fa fa-check"></i><b>9.6</b> Filtering by minimum read depth</a></li>
<li class="chapter" data-level="9.7" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minmum-read-depth-summary"><i class="fa fa-check"></i><b>9.7</b> Minmum read depth: Summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy</b></span></li>
<li class="chapter" data-level="10" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html"><i class="fa fa-check"></i><b>10</b> Taxa relative abundance</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-setup"><i class="fa fa-check"></i><b>10.1</b> Taxa relative abundance: setup</a></li>
<li class="chapter" data-level="10.2" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#relative-abundance-transformation"><i class="fa fa-check"></i><b>10.2</b> Relative abundance transformation</a></li>
<li class="chapter" data-level="10.3" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#aggregate-taxa"><i class="fa fa-check"></i><b>10.3</b> Aggregate taxa</a></li>
<li class="chapter" data-level="10.4" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-summary"><i class="fa fa-check"></i><b>10.4</b> Taxa Relative abundance: summary</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html"><i class="fa fa-check"></i><b>11</b> Taxa plots</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#simple-bar-chart"><i class="fa fa-check"></i><b>11.1</b> Simple bar chart</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#editing-the-plot"><i class="fa fa-check"></i><b>11.1.1</b> Editing the plot</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-heatmap"><i class="fa fa-check"></i><b>11.2</b> Taxa heatmap</a></li>
<li class="chapter" data-level="11.3" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#aggregateraretaxa"><i class="fa fa-check"></i><b>11.3</b> Aggregate rare taxa</a></li>
<li class="chapter" data-level="11.4" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-media"><i class="fa fa-check"></i><b>11.4</b> Plot by media</a></li>
<li class="chapter" data-level="11.5" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-site-and-media"><i class="fa fa-check"></i><b>11.5</b> Plot by site and media</a></li>
<li class="chapter" data-level="11.6" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-plots-summary"><i class="fa fa-check"></i><b>11.6</b> Taxa plots: summary</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html"><i class="fa fa-check"></i><b>12</b> Family &amp; Genus plots</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-taxa-plots"><i class="fa fa-check"></i><b>12.1</b> Family taxa plots</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#aggregate-families"><i class="fa fa-check"></i><b>12.1.1</b> Aggregate families</a></li>
<li class="chapter" data-level="12.1.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-heatmap"><i class="fa fa-check"></i><b>12.1.2</b> Family heatmap</a></li>
<li class="chapter" data-level="12.1.3" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-rare-aggregation"><i class="fa fa-check"></i><b>12.1.3</b> Family rare aggregation</a></li>
<li class="chapter" data-level="12.1.4" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-bar-chart"><i class="fa fa-check"></i><b>12.1.4</b> Family bar chart</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots"><i class="fa fa-check"></i><b>12.2</b> Genus taxa plots</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots-solutions"><i class="fa fa-check"></i><b>12.2.1</b> Genus taxa plots solutions</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Diversity analysis</b></span></li>
<li class="chapter" data-level="13" data-path="13-Diversity_analysis.html"><a href="13-Diversity_analysis.html"><i class="fa fa-check"></i><b>13</b> Diversity analysis intro</a></li>
<li class="chapter" data-level="14" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html"><i class="fa fa-check"></i><b>14</b> Rarefaction</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-setup"><i class="fa fa-check"></i><b>14.1</b> Rarefaction: setup</a></li>
<li class="chapter" data-level="14.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve"><i class="fa fa-check"></i><b>14.2</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#bird-analogy"><i class="fa fa-check"></i><b>14.2.1</b> Bird analogy</a></li>
<li class="chapter" data-level="14.2.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-simple-plot"><i class="fa fa-check"></i><b>14.2.2</b> Rarefaction curve: simple plot</a></li>
<li class="chapter" data-level="14.2.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-better-plot"><i class="fa fa-check"></i><b>14.2.3</b> Rarefaction curve: better plot</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-slope"><i class="fa fa-check"></i><b>14.3</b> Rarefaction slope</a></li>
<li class="chapter" data-level="14.4" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefy-data"><i class="fa fa-check"></i><b>14.4</b> Rarefy data</a></li>
<li class="chapter" data-level="14.5" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-summary"><i class="fa fa-check"></i><b>14.5</b> Rarefaction: summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Alpha.html"><a href="15-Alpha.html"><i class="fa fa-check"></i><b>15</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-setup"><i class="fa fa-check"></i><b>15.1</b> Alpha: setup</a></li>
<li class="chapter" data-level="15.2" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-simple-plot"><i class="fa fa-check"></i><b>15.2</b> Alpha: simple plot</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="15-Alpha.html"><a href="15-Alpha.html#metric-choice"><i class="fa fa-check"></i><b>15.2.1</b> Metric choice</a></li>
<li class="chapter" data-level="15.2.2" data-path="15-Alpha.html"><a href="15-Alpha.html#sample-grouping-box-plot"><i class="fa fa-check"></i><b>15.2.2</b> Sample grouping &amp; box plot</a></li>
<li class="chapter" data-level="15.2.3" data-path="15-Alpha.html"><a href="15-Alpha.html#png-height-and-width-choice"><i class="fa fa-check"></i><b>15.2.3</b> PNG height and width choice</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats"><i class="fa fa-check"></i><b>15.3</b> Alpha: stats</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats-task"><i class="fa fa-check"></i><b>15.3.1</b> Alpha stats task</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-diversity-violin-plot"><i class="fa fa-check"></i><b>15.4</b> Alpha diversity: violin plot</a></li>
<li class="chapter" data-level="15.5" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-site-plots-and-statistics"><i class="fa fa-check"></i><b>15.5</b> Alpha: Site plots and statistics</a></li>
<li class="chapter" data-level="15.6" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-summary"><i class="fa fa-check"></i><b>15.6</b> Alpha: summary</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Beta.html"><a href="16-Beta.html"><i class="fa fa-check"></i><b>16</b> Beta</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Beta.html"><a href="16-Beta.html#beta-setup"><i class="fa fa-check"></i><b>16.1</b> Beta: setup</a></li>
<li class="chapter" data-level="16.2" data-path="16-Beta.html"><a href="16-Beta.html#beta-ordination"><i class="fa fa-check"></i><b>16.2</b> Beta: ordination</a></li>
<li class="chapter" data-level="16.3" data-path="16-Beta.html"><a href="16-Beta.html#beta-plot-ordination"><i class="fa fa-check"></i><b>16.3</b> Beta: plot ordination</a></li>
<li class="chapter" data-level="16.4" data-path="16-Beta.html"><a href="16-Beta.html#beta-statistics"><i class="fa fa-check"></i><b>16.4</b> Beta: statistics</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="16-Beta.html"><a href="16-Beta.html#permanova"><i class="fa fa-check"></i><b>16.4.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="16.4.2" data-path="16-Beta.html"><a href="16-Beta.html#pairwise-permanova"><i class="fa fa-check"></i><b>16.4.2</b> Pairwise PERMANOVA</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="16-Beta.html"><a href="16-Beta.html#beta-facet-zoom"><i class="fa fa-check"></i><b>16.5</b> Beta: Facet zoom</a></li>
<li class="chapter" data-level="16.6" data-path="16-Beta.html"><a href="16-Beta.html#beta-subset-samples"><i class="fa fa-check"></i><b>16.6</b> Beta: subset samples</a></li>
<li class="chapter" data-level="16.7" data-path="16-Beta.html"><a href="16-Beta.html#permanova-tasks"><i class="fa fa-check"></i><b>16.7</b> PERMANOVA tasks</a></li>
<li class="chapter" data-level="16.8" data-path="16-Beta.html"><a href="16-Beta.html#beta-summary"><i class="fa fa-check"></i><b>16.8</b> Beta: Summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html"><i class="fa fa-check"></i><b>17</b> Differential abundance (DA) analysis</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-setup"><i class="fa fa-check"></i><b>17.1</b> DA: setup</a></li>
<li class="chapter" data-level="17.2" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-preprocess"><i class="fa fa-check"></i><b>17.2</b> DA: Preprocess</a></li>
<li class="chapter" data-level="17.3" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-analysis"><i class="fa fa-check"></i><b>17.3</b> DA: ANCOM-BC analysis</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#ancombc2-options"><i class="fa fa-check"></i><b>17.3.1</b> ANCOMBC2 options</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-results"><i class="fa fa-check"></i><b>17.4</b> DA: ANCOM-BC results</a></li>
<li class="chapter" data-level="17.5" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-create-signifcant-lfc-long-data-frame"><i class="fa fa-check"></i><b>17.5</b> DA: Create signifcant LFC long data frame</a></li>
<li class="chapter" data-level="17.6" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-biomarker-lfc-heatmap"><i class="fa fa-check"></i><b>17.6</b> DA: Biomarker LFC heatmap</a></li>
<li class="chapter" data-level="17.7" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-matching-bar-plot"><i class="fa fa-check"></i><b>17.7</b> DA: matching bar plot</a></li>
<li class="chapter" data-level="17.8" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-summary"><i class="fa fa-check"></i><b>17.8</b> DA: summary</a></li>
</ul></li>
<li class="part"><span><b>Outro</b></span></li>
<li class="chapter" data-level="18" data-path="18-Summary.html"><a href="18-Summary.html"><i class="fa fa-check"></i><b>18</b> Summary</a>
<ul>
<li class="chapter" data-level="18.1" data-path="18-Summary.html"><a href="18-Summary.html#dataset-1"><i class="fa fa-check"></i><b>18.1</b> Dataset</a></li>
<li class="chapter" data-level="18.2" data-path="18-Summary.html"><a href="18-Summary.html#data-preparation"><i class="fa fa-check"></i><b>18.2</b> Data preparation</a></li>
<li class="chapter" data-level="18.3" data-path="18-Summary.html"><a href="18-Summary.html#taxonomy"><i class="fa fa-check"></i><b>18.3</b> Taxonomy</a></li>
<li class="chapter" data-level="18.4" data-path="18-Summary.html"><a href="18-Summary.html#diversity-analysis"><i class="fa fa-check"></i><b>18.4</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="18-Summary.html"><a href="18-Summary.html#alpha-beta"><i class="fa fa-check"></i><b>18.4.1</b> Alpha &amp; beta</a></li>
<li class="chapter" data-level="18.4.2" data-path="18-Summary.html"><a href="18-Summary.html#differential-abundance-anlaysis"><i class="fa fa-check"></i><b>18.4.2</b> Differential abundance anlaysis</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="18-Summary.html"><a href="18-Summary.html#findings"><i class="fa fa-check"></i><b>18.5</b> Findings</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="18-Summary.html"><a href="18-Summary.html#anthropisation-gradient"><i class="fa fa-check"></i><b>18.5.1</b> Anthropisation gradient</a></li>
<li class="chapter" data-level="18.5.2" data-path="18-Summary.html"><a href="18-Summary.html#replicate-differences"><i class="fa fa-check"></i><b>18.5.2</b> Replicate differences</a></li>
<li class="chapter" data-level="18.5.3" data-path="18-Summary.html"><a href="18-Summary.html#site-or-media"><i class="fa fa-check"></i><b>18.5.3</b> Site or media</a></li>
<li class="chapter" data-level="18.5.4" data-path="18-Summary.html"><a href="18-Summary.html#media-versus-env-samples"><i class="fa fa-check"></i><b>18.5.4</b> Media versus ENV samples</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="18-Summary.html"><a href="18-Summary.html#conclusion"><i class="fa fa-check"></i><b>18.6</b> Conclusion</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="19-Appendix.html"><a href="19-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="19-Appendix.html"><a href="19-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
<li class="chapter" data-level="A.2" data-path="19-Appendix.html"><a href="19-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>A.2</b> Jupyter-notebook</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="19-Appendix.html"><a href="19-Appendix.html#r-packages-1"><i class="fa fa-check"></i><b>B</b> R packages</a>
<ul>
<li class="chapter" data-level="B.1" data-path="19-Appendix.html"><a href="19-Appendix.html#ggplot2_appendix"><i class="fa fa-check"></i><b>B.1</b> ggplot2</a></li>
<li class="chapter" data-level="B.2" data-path="19-Appendix.html"><a href="19-Appendix.html#phyloseq_appendix"><i class="fa fa-check"></i><b>B.2</b> Phyloseq</a></li>
<li class="chapter" data-level="B.3" data-path="19-Appendix.html"><a href="19-Appendix.html#importing-into-phyloseq"><i class="fa fa-check"></i><b>B.3</b> Importing into phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-metrics"><i class="fa fa-check"></i><b>C</b> Diversity metrics</a>
<ul>
<li class="chapter" data-level="C.1" data-path="19-Appendix.html"><a href="19-Appendix.html#alpha_appendix"><i class="fa fa-check"></i><b>C.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="19-Appendix.html"><a href="19-Appendix.html#obvs"><i class="fa fa-check"></i><b>C.1.1</b> Observed features</a></li>
<li class="chapter" data-level="C.1.2" data-path="19-Appendix.html"><a href="19-Appendix.html#chao"><i class="fa fa-check"></i><b>C.1.2</b> Chao</a></li>
<li class="chapter" data-level="C.1.3" data-path="19-Appendix.html"><a href="19-Appendix.html#evenness"><i class="fa fa-check"></i><b>C.1.3</b> Evenness</a></li>
<li class="chapter" data-level="C.1.4" data-path="19-Appendix.html"><a href="19-Appendix.html#faith"><i class="fa fa-check"></i><b>C.1.4</b> Faith's PD (phylogenetic diversity)</a></li>
<li class="chapter" data-level="C.1.5" data-path="19-Appendix.html"><a href="19-Appendix.html#simpson"><i class="fa fa-check"></i><b>C.1.5</b> Simpson</a></li>
<li class="chapter" data-level="C.1.6" data-path="19-Appendix.html"><a href="19-Appendix.html#simpsone"><i class="fa fa-check"></i><b>C.1.6</b> Simpson evenness measure (simpson_e)</a></li>
<li class="chapter" data-level="C.1.7" data-path="19-Appendix.html"><a href="19-Appendix.html#shannon"><i class="fa fa-check"></i><b>C.1.7</b> Shannon</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="19-Appendix.html"><a href="19-Appendix.html#beta_appendix"><i class="fa fa-check"></i><b>C.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="19-Appendix.html"><a href="19-Appendix.html#unifrac_explanation"><i class="fa fa-check"></i><b>C.2.1</b> Weighted and unweighted UniFrac distances</a></li>
<li class="chapter" data-level="C.2.2" data-path="19-Appendix.html"><a href="19-Appendix.html#bray_explanation"><i class="fa fa-check"></i><b>C.2.2</b> Bray-Curtis</a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-resources"><i class="fa fa-check"></i><b>C.3</b> Diversity Resources</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R community analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="family_genus_chap" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">Chapter 12</span> Family &amp; Genus plots<a href="12-Family_and_Genus_taxa.html#family_genus_chap" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/plots.png" width="20%" style="display: block; margin: auto;" /></p>
<p>In this chapter you will:</p>
<ul>
<li>Create Family plots by following example code.</li>
<li>Be tasked to create Genus plots with little instruction.</li>
</ul>
<p>I highly encourage you to copy previous code and edit it.
It is a major time saver.
However, it can be easy to overlook some edits.
It is a good idea to double check your edited code to ensure you have changed the relevant options, and object &amp; file names.</p>
<div id="family-taxa-plots" class="section level2 hasAnchor" number="12.1">
<h2><span class="header-section-number">12.1</span> Family taxa plots<a href="12-Family_and_Genus_taxa.html#family-taxa-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/family.png" width="15%" style="display: block; margin: auto;" /></p>
<p>This part will show you a nice workflow for creating some family based taxa plots.
Run this at the bottom of your "3-taxonomy_barchart.ipynb" notebook.</p>
<div id="aggregate-families" class="section level3 hasAnchor" number="12.1.1">
<h3><span class="header-section-number">12.1.1</span> Aggregate families<a href="12-Family_and_Genus_taxa.html#aggregate-families" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/aggregate_taxa.png" width="15%" style="display: block; margin: auto;" /></p>
<p>First we will create a <code>phyloseq</code> object by aggregating the families from the relative abundance <code>phyloseq</code> object.
Let us also then check the family <code>phyloseq</code> object.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="12-Family_and_Genus_taxa.html#cb54-1" tabindex="-1"></a><span class="co">#Family phyloseq</span></span>
<span id="cb54-2"><a href="12-Family_and_Genus_taxa.html#cb54-2" tabindex="-1"></a>family_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_taxa</span>(pseq_relabund, <span class="st">&quot;Family&quot;</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb54-3"><a href="12-Family_and_Genus_taxa.html#cb54-3" tabindex="-1"></a><span class="co">#Head of phylum relative abundance table</span></span>
<span id="cb54-4"><a href="12-Family_and_Genus_taxa.html#cb54-4" tabindex="-1"></a><span class="fu">head</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(family_pseq))</span>
<span id="cb54-5"><a href="12-Family_and_Genus_taxa.html#cb54-5" tabindex="-1"></a><span class="co">#Number of families</span></span>
<span id="cb54-6"><a href="12-Family_and_Genus_taxa.html#cb54-6" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Number of families: &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(family_pseq)))</span>
<span id="cb54-7"><a href="12-Family_and_Genus_taxa.html#cb54-7" tabindex="-1"></a><span class="co">#Summarise</span></span>
<span id="cb54-8"><a href="12-Family_and_Genus_taxa.html#cb54-8" tabindex="-1"></a>microbiome<span class="sc">::</span><span class="fu">summarize_phyloseq</span>(family_pseq)</span>
<span id="cb54-9"><a href="12-Family_and_Genus_taxa.html#cb54-9" tabindex="-1"></a>microbiome<span class="sc">::</span><span class="fu">readcount</span>(family_pseq)</span></code></pre></div>
<p>We have 112 families. Quite a bit but hopefully they will all fit into a heatmap.</p>
</div>
<div id="family-heatmap" class="section level3 hasAnchor" number="12.1.2">
<h3><span class="header-section-number">12.1.2</span> Family heatmap<a href="12-Family_and_Genus_taxa.html#family-heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/heat_topology.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We will create a heatmap as before with one added <code>ggplot2</code> component.
As there are a lot of Families (112) and fewer samples (36) it will be easier to view if we have the families on the y-axis.
By default <code>microbiome::plot_composition()</code> will plot the taxa/families on the x-axis.
Thankfully we can use the <code>ggplot2()</code> function/component <code>coord_flip()</code> to flip/swap the x and y axes.</p>
<p><strong>Note</strong>: The <code>xlab()</code> and <code>ylab()</code> are specified the before this flipping.
E.g. <code>xlab()</code> is set to "Family" which will appear on the y axis because of the flip.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="12-Family_and_Genus_taxa.html#cb55-1" tabindex="-1"></a><span class="co">#Family heatmap</span></span>
<span id="cb55-2"><a href="12-Family_and_Genus_taxa.html#cb55-2" tabindex="-1"></a>family_heatmap <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">plot_composition</span>(family_pseq, <span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>) <span class="sc">+</span></span>
<span id="cb55-3"><a href="12-Family_and_Genus_taxa.html#cb55-3" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Family&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Sample&quot;</span>) <span class="sc">+</span></span>
<span id="cb55-4"><a href="12-Family_and_Genus_taxa.html#cb55-4" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Family relative abundance heatmap&quot;</span>) <span class="sc">+</span></span>
<span id="cb55-5"><a href="12-Family_and_Genus_taxa.html#cb55-5" tabindex="-1"></a>  <span class="co">#Flip the x and y axes</span></span>
<span id="cb55-6"><a href="12-Family_and_Genus_taxa.html#cb55-6" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb55-7"><a href="12-Family_and_Genus_taxa.html#cb55-7" tabindex="-1"></a><span class="co">#Save ggplot object as png file</span></span>
<span id="cb55-8"><a href="12-Family_and_Genus_taxa.html#cb55-8" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./family_relabund_heatmap.png&quot;</span>, <span class="at">plot =</span> family_heatmap,</span>
<span id="cb55-9"><a href="12-Family_and_Genus_taxa.html#cb55-9" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">200</span>, <span class="at">width =</span> <span class="dv">300</span>)</span>
<span id="cb55-10"><a href="12-Family_and_Genus_taxa.html#cb55-10" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb55-11"><a href="12-Family_and_Genus_taxa.html#cb55-11" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./family_relabund_heatmap.png&quot;</span>)</span></code></pre></div>
<p>The plot is quite large.
You can right click on the image and click "Open Image in New Tab" to view the whole image in a new tab.</p>
<p>With this heatmap we can see that the KBC and TSA samples have a high relative abundance of <em>Pseudomonadaceae</em> whilst the TSA samples have a high relative abundance of <em>Aeromonadaceae</em>.
Additionally, we can see that a lot of families have relatively low relative abundances.</p>
<p>Once you are finished looking at the plot, close the tab containing the plot.
This will help to keep the used resources low on the cluster.
Next we'll carry out our rare aggregation and produce some bar charts.</p>
</div>
<div id="family-rare-aggregation" class="section level3 hasAnchor" number="12.1.3">
<h3><span class="header-section-number">12.1.3</span> Family rare aggregation<a href="12-Family_and_Genus_taxa.html#family-rare-aggregation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/elephant_origami.png" width="15%" style="display: block; margin: auto;" /></p>
<p>To aggregate our rare families we need to choose a suitable <a href="11-Taxa_plots.html#aggregateraretaxa">detection and prevalence threshold</a>.
Generally, this is a process of trial and error trying to get a total of 12 taxa.
Twelve is a good number as it is the total number of colours in the Paired palette from Colour brewer.
You may need more in your own analysis so it is not a hard limit.</p>
<p>Therefore, lets try out various thresholds and see how many families we get.
To get a good idea where to start we can get a summary of the mean relative abundance values of the families.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="12-Family_and_Genus_taxa.html#cb56-1" tabindex="-1"></a><span class="co">#Summary of row means (families) from the family otu table</span></span>
<span id="cb56-2"><a href="12-Family_and_Genus_taxa.html#cb56-2" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">rowMeans</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(family_pseq)))</span></code></pre></div>
<p>We have a lot of families so we will want to remove a good amount of them.
For our first attempt at rare aggregation we will use a value near the 3rd Quartile (0.0021601).
This will hopefully remove at least 3/4 of the families.</p>
<p>We'll set up the code a bit differently so we can easily copy and paste it to try out various other values of detection and threshold.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="12-Family_and_Genus_taxa.html#cb57-1" tabindex="-1"></a><span class="co">#Tests</span></span>
<span id="cb57-2"><a href="12-Family_and_Genus_taxa.html#cb57-2" tabindex="-1"></a><span class="co">#1</span></span>
<span id="cb57-3"><a href="12-Family_and_Genus_taxa.html#cb57-3" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fl">0.002</span></span>
<span id="cb57-4"><a href="12-Family_and_Genus_taxa.html#cb57-4" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">5</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb57-5"><a href="12-Family_and_Genus_taxa.html#cb57-5" tabindex="-1"></a>test_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb57-6"><a href="12-Family_and_Genus_taxa.html#cb57-6" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Family&quot;</span>,</span>
<span id="cb57-7"><a href="12-Family_and_Genus_taxa.html#cb57-7" tabindex="-1"></a>  <span class="at">detection =</span> d, <span class="at">prevalence =</span> p</span>
<span id="cb57-8"><a href="12-Family_and_Genus_taxa.html#cb57-8" tabindex="-1"></a>)</span>
<span id="cb57-9"><a href="12-Family_and_Genus_taxa.html#cb57-9" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Detection = &quot;</span>, d, <span class="st">&quot;, Prevalence = &quot;</span>, p,</span>
<span id="cb57-10"><a href="12-Family_and_Genus_taxa.html#cb57-10" tabindex="-1"></a>       <span class="st">&quot;, Number of families = &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(test_pseq)))</span></code></pre></div>
<p>That gives us 56 families which is too high for a nice visualisation. We'll therefore try a bunch of trial and error to get 12 families after aggregation.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="12-Family_and_Genus_taxa.html#cb58-1" tabindex="-1"></a><span class="co">#2 Increase prevalence threshold</span></span>
<span id="cb58-2"><a href="12-Family_and_Genus_taxa.html#cb58-2" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fl">0.002</span></span>
<span id="cb58-3"><a href="12-Family_and_Genus_taxa.html#cb58-3" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb58-4"><a href="12-Family_and_Genus_taxa.html#cb58-4" tabindex="-1"></a>test_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb58-5"><a href="12-Family_and_Genus_taxa.html#cb58-5" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Family&quot;</span>,</span>
<span id="cb58-6"><a href="12-Family_and_Genus_taxa.html#cb58-6" tabindex="-1"></a>  <span class="at">detection =</span> d, <span class="at">prevalence =</span> p</span>
<span id="cb58-7"><a href="12-Family_and_Genus_taxa.html#cb58-7" tabindex="-1"></a>)</span>
<span id="cb58-8"><a href="12-Family_and_Genus_taxa.html#cb58-8" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Detection = &quot;</span>, d, <span class="st">&quot;, Prevalence = &quot;</span>, p,</span>
<span id="cb58-9"><a href="12-Family_and_Genus_taxa.html#cb58-9" tabindex="-1"></a>       <span class="st">&quot;, Number of families = &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(test_pseq)))</span>
<span id="cb58-10"><a href="12-Family_and_Genus_taxa.html#cb58-10" tabindex="-1"></a><span class="co">#3 Increase detection threshold</span></span>
<span id="cb58-11"><a href="12-Family_and_Genus_taxa.html#cb58-11" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fl">0.005</span></span>
<span id="cb58-12"><a href="12-Family_and_Genus_taxa.html#cb58-12" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb58-13"><a href="12-Family_and_Genus_taxa.html#cb58-13" tabindex="-1"></a>test_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb58-14"><a href="12-Family_and_Genus_taxa.html#cb58-14" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Family&quot;</span>,</span>
<span id="cb58-15"><a href="12-Family_and_Genus_taxa.html#cb58-15" tabindex="-1"></a>  <span class="at">detection =</span> d, <span class="at">prevalence =</span> p</span>
<span id="cb58-16"><a href="12-Family_and_Genus_taxa.html#cb58-16" tabindex="-1"></a>)</span>
<span id="cb58-17"><a href="12-Family_and_Genus_taxa.html#cb58-17" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Detection = &quot;</span>, d, <span class="st">&quot;, Prevalence = &quot;</span>, p,</span>
<span id="cb58-18"><a href="12-Family_and_Genus_taxa.html#cb58-18" tabindex="-1"></a>       <span class="st">&quot;, Number of families = &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(test_pseq)))</span>
<span id="cb58-19"><a href="12-Family_and_Genus_taxa.html#cb58-19" tabindex="-1"></a><span class="co">#4 Increase detection threshold</span></span>
<span id="cb58-20"><a href="12-Family_and_Genus_taxa.html#cb58-20" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fl">0.02</span></span>
<span id="cb58-21"><a href="12-Family_and_Genus_taxa.html#cb58-21" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb58-22"><a href="12-Family_and_Genus_taxa.html#cb58-22" tabindex="-1"></a>test_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb58-23"><a href="12-Family_and_Genus_taxa.html#cb58-23" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Family&quot;</span>,</span>
<span id="cb58-24"><a href="12-Family_and_Genus_taxa.html#cb58-24" tabindex="-1"></a>  <span class="at">detection =</span> d, <span class="at">prevalence =</span> p</span>
<span id="cb58-25"><a href="12-Family_and_Genus_taxa.html#cb58-25" tabindex="-1"></a>)</span>
<span id="cb58-26"><a href="12-Family_and_Genus_taxa.html#cb58-26" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Detection = &quot;</span>, d, <span class="st">&quot;, Prevalence = &quot;</span>, p,</span>
<span id="cb58-27"><a href="12-Family_and_Genus_taxa.html#cb58-27" tabindex="-1"></a>       <span class="st">&quot;, Number of families = &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(test_pseq)))</span>
<span id="cb58-28"><a href="12-Family_and_Genus_taxa.html#cb58-28" tabindex="-1"></a><span class="co">#5 Increase prevalence threshold</span></span>
<span id="cb58-29"><a href="12-Family_and_Genus_taxa.html#cb58-29" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fl">0.02</span></span>
<span id="cb58-30"><a href="12-Family_and_Genus_taxa.html#cb58-30" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">20</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb58-31"><a href="12-Family_and_Genus_taxa.html#cb58-31" tabindex="-1"></a>test_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb58-32"><a href="12-Family_and_Genus_taxa.html#cb58-32" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Family&quot;</span>,</span>
<span id="cb58-33"><a href="12-Family_and_Genus_taxa.html#cb58-33" tabindex="-1"></a>  <span class="at">detection =</span> d, <span class="at">prevalence =</span> p</span>
<span id="cb58-34"><a href="12-Family_and_Genus_taxa.html#cb58-34" tabindex="-1"></a>)</span>
<span id="cb58-35"><a href="12-Family_and_Genus_taxa.html#cb58-35" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Detection = &quot;</span>, d, <span class="st">&quot;, Prevalence = &quot;</span>, p,</span>
<span id="cb58-36"><a href="12-Family_and_Genus_taxa.html#cb58-36" tabindex="-1"></a>       <span class="st">&quot;, Number of families = &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(test_pseq)))</span>
<span id="cb58-37"><a href="12-Family_and_Genus_taxa.html#cb58-37" tabindex="-1"></a><span class="co">#Remove unwanted test_pseq</span></span>
<span id="cb58-38"><a href="12-Family_and_Genus_taxa.html#cb58-38" tabindex="-1"></a><span class="fu">rm</span>(test_pseq)</span></code></pre></div>
<p>Super!
A Detection threshold of 0.02 and a prevalence threshold of 20/100 gives us 12 families.
We'll now create a new <code>phyloseq</code> object using this info in a new cell.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="12-Family_and_Genus_taxa.html#cb59-1" tabindex="-1"></a><span class="co">#Aggregate families phyla</span></span>
<span id="cb59-2"><a href="12-Family_and_Genus_taxa.html#cb59-2" tabindex="-1"></a>family_rareaggregate_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb59-3"><a href="12-Family_and_Genus_taxa.html#cb59-3" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Family&quot;</span>,</span>
<span id="cb59-4"><a href="12-Family_and_Genus_taxa.html#cb59-4" tabindex="-1"></a>  <span class="at">detection =</span> <span class="fl">0.02</span>, <span class="at">prevalence =</span> <span class="dv">20</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb59-5"><a href="12-Family_and_Genus_taxa.html#cb59-5" tabindex="-1"></a>)</span></code></pre></div>
<p>Choosing a good detection and prevalence threshold is quite difficult.
It takes a lot of trial and practice and I find the thresholds to not be very intuitive to understand.
One piece of advise is to only change one threshold at a time during your trial and errors.</p>
</div>
<div id="family-bar-chart" class="section level3 hasAnchor" number="12.1.4">
<h3><span class="header-section-number">12.1.4</span> Family bar chart<a href="12-Family_and_Genus_taxa.html#family-bar-chart" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/chocolate_bar.png" width="15%" style="display: block; margin: auto;" /></p>
<p>With our family rare aggregated <code>phyloseq</code> object we'll create a bar chart.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="12-Family_and_Genus_taxa.html#cb60-1" tabindex="-1"></a><span class="co">#Plot</span></span>
<span id="cb60-2"><a href="12-Family_and_Genus_taxa.html#cb60-2" tabindex="-1"></a>family_bar <span class="ot">&lt;-</span> </span>
<span id="cb60-3"><a href="12-Family_and_Genus_taxa.html#cb60-3" tabindex="-1"></a>  microbiome<span class="sc">::</span><span class="fu">plot_composition</span>(family_rareaggregate_pseq,</span>
<span id="cb60-4"><a href="12-Family_and_Genus_taxa.html#cb60-4" tabindex="-1"></a>                               <span class="co">#otu.sort to sort family by abundance</span></span>
<span id="cb60-5"><a href="12-Family_and_Genus_taxa.html#cb60-5" tabindex="-1"></a>                               <span class="at">otu.sort =</span> <span class="st">&quot;abundance&quot;</span>, <span class="at">group_by =</span> <span class="st">&quot;media&quot;</span>) <span class="sc">+</span></span>
<span id="cb60-6"><a href="12-Family_and_Genus_taxa.html#cb60-6" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Sample&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Relative abundance&quot;</span>) <span class="sc">+</span></span>
<span id="cb60-7"><a href="12-Family_and_Genus_taxa.html#cb60-7" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Family relative abundance bar chart&quot;</span>) <span class="sc">+</span></span>
<span id="cb60-8"><a href="12-Family_and_Genus_taxa.html#cb60-8" tabindex="-1"></a>  <span class="co">#Change colours of families to that of the &quot;Paired&quot; palette from colour brewer</span></span>
<span id="cb60-9"><a href="12-Family_and_Genus_taxa.html#cb60-9" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="st">&quot;Family&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;Paired&quot;</span>)</span>
<span id="cb60-10"><a href="12-Family_and_Genus_taxa.html#cb60-10" tabindex="-1"></a><span class="co">#Save ggplot object as png</span></span>
<span id="cb60-11"><a href="12-Family_and_Genus_taxa.html#cb60-11" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./family_relabund_rareaggregate.png&quot;</span>, <span class="at">plot =</span> family_bar,</span>
<span id="cb60-12"><a href="12-Family_and_Genus_taxa.html#cb60-12" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">200</span>, <span class="at">width =</span> <span class="dv">300</span>)</span>
<span id="cb60-13"><a href="12-Family_and_Genus_taxa.html#cb60-13" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb60-14"><a href="12-Family_and_Genus_taxa.html#cb60-14" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./family_relabund_rareaggregate.png&quot;</span>)</span></code></pre></div>
<p>Brilliant, we are now finished with the Family taxa plots. Onto something harder.</p>
</div>
</div>
<div id="genus-taxa-plots" class="section level2 hasAnchor" number="12.2">
<h2><span class="header-section-number">12.2</span> Genus taxa plots<a href="12-Family_and_Genus_taxa.html#genus-taxa-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/ginger.png" width="15%" style="display: block; margin: auto;" /></p>
<p>In this section you are running your own analysis for the Genus information in the "3-taxonomy_barchart.ipynb" notebook.
Carry out the following tasks with the relative abundance <code>phyloseq</code> object (<code>pseq_relabund</code>):</p>
<ul>
<li>Create an aggregated genus <code>phyloseq</code> object and summarise it (no rare aggregation).</li>
<li>Create a heatmap of the genera relative abundances.</li>
<li>Carry out genus rare aggregation so the resulting <code>phyloseq</code> object only contains 12 genera.</li>
<li>Use the rare aggregated <code>phyloseq</code> object to create a bar chart grouped by <strong>media</strong>.</li>
</ul>
<p>In essence you are going through the workflow for the family data but for genus instead.
Again, I encourage you to copy, paste, and edit previous code.
You will most likely need to change names of objects and also change various options.</p>
<div id="genus-taxa-plots-solutions" class="section level3 hasAnchor" number="12.2.1">
<h3><span class="header-section-number">12.2.1</span> Genus taxa plots solutions<a href="12-Family_and_Genus_taxa.html#genus-taxa-plots-solutions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/ginger_solutions.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Please give the task a good try before looking at the solutions in the expandable boxes.
However, if you are really stuck or want to compare your code to mine please have a look.
Please, remember that there are many ways to do the same thing.
If your code is different but it works and you understand it then that is fine.</p>
<div class="webex-solution">
<button>
Aggregate genus and summarise
</button>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="12-Family_and_Genus_taxa.html#cb61-1" tabindex="-1"></a><span class="co">#Genus phyloseq</span></span>
<span id="cb61-2"><a href="12-Family_and_Genus_taxa.html#cb61-2" tabindex="-1"></a>genus_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_taxa</span>(pseq_relabund, <span class="st">&quot;Genus&quot;</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb61-3"><a href="12-Family_and_Genus_taxa.html#cb61-3" tabindex="-1"></a><span class="co">#Head of genus relative abundance table</span></span>
<span id="cb61-4"><a href="12-Family_and_Genus_taxa.html#cb61-4" tabindex="-1"></a><span class="fu">head</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(genus_pseq))</span>
<span id="cb61-5"><a href="12-Family_and_Genus_taxa.html#cb61-5" tabindex="-1"></a><span class="co">#Number of genera</span></span>
<span id="cb61-6"><a href="12-Family_and_Genus_taxa.html#cb61-6" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Number of families: &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(genus_pseq)))</span>
<span id="cb61-7"><a href="12-Family_and_Genus_taxa.html#cb61-7" tabindex="-1"></a><span class="co">#Summarise</span></span>
<span id="cb61-8"><a href="12-Family_and_Genus_taxa.html#cb61-8" tabindex="-1"></a>microbiome<span class="sc">::</span><span class="fu">summarize_phyloseq</span>(genus_pseq)</span>
<span id="cb61-9"><a href="12-Family_and_Genus_taxa.html#cb61-9" tabindex="-1"></a>microbiome<span class="sc">::</span><span class="fu">readcount</span>(genus_pseq)</span></code></pre></div>
</div>
<div class="webex-solution">
<button>
Heatmap
</button>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="12-Family_and_Genus_taxa.html#cb62-1" tabindex="-1"></a><span class="co">#Genus heatmap</span></span>
<span id="cb62-2"><a href="12-Family_and_Genus_taxa.html#cb62-2" tabindex="-1"></a>genus_heatmap <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">plot_composition</span>(genus_pseq, <span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>) <span class="sc">+</span></span>
<span id="cb62-3"><a href="12-Family_and_Genus_taxa.html#cb62-3" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Genus&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Sample&quot;</span>) <span class="sc">+</span></span>
<span id="cb62-4"><a href="12-Family_and_Genus_taxa.html#cb62-4" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Genus relative abundance heatmap&quot;</span>) <span class="sc">+</span></span>
<span id="cb62-5"><a href="12-Family_and_Genus_taxa.html#cb62-5" tabindex="-1"></a>  <span class="co">#Flip the x and y axes</span></span>
<span id="cb62-6"><a href="12-Family_and_Genus_taxa.html#cb62-6" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb62-7"><a href="12-Family_and_Genus_taxa.html#cb62-7" tabindex="-1"></a><span class="co">#Save ggplot object as png file</span></span>
<span id="cb62-8"><a href="12-Family_and_Genus_taxa.html#cb62-8" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./genus_relabund_heatmap.png&quot;</span>, <span class="at">plot =</span> genus_heatmap,</span>
<span id="cb62-9"><a href="12-Family_and_Genus_taxa.html#cb62-9" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">450</span>, <span class="at">width =</span> <span class="dv">300</span>)</span>
<span id="cb62-10"><a href="12-Family_and_Genus_taxa.html#cb62-10" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb62-11"><a href="12-Family_and_Genus_taxa.html#cb62-11" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./genus_relabund_heatmap.png&quot;</span>)</span></code></pre></div>
</div>
<div class="webex-solution">
<button>
Aggregate rare genus
</button>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="12-Family_and_Genus_taxa.html#cb63-1" tabindex="-1"></a><span class="co">#Summary of row means (families) from the family otu table</span></span>
<span id="cb63-2"><a href="12-Family_and_Genus_taxa.html#cb63-2" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">rowMeans</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(genus_pseq)))</span></code></pre></div>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="12-Family_and_Genus_taxa.html#cb64-1" tabindex="-1"></a><span class="co">#Tests</span></span>
<span id="cb64-2"><a href="12-Family_and_Genus_taxa.html#cb64-2" tabindex="-1"></a><span class="co">#1</span></span>
<span id="cb64-3"><a href="12-Family_and_Genus_taxa.html#cb64-3" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fl">0.001</span></span>
<span id="cb64-4"><a href="12-Family_and_Genus_taxa.html#cb64-4" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">5</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb64-5"><a href="12-Family_and_Genus_taxa.html#cb64-5" tabindex="-1"></a>test_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb64-6"><a href="12-Family_and_Genus_taxa.html#cb64-6" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb64-7"><a href="12-Family_and_Genus_taxa.html#cb64-7" tabindex="-1"></a>  <span class="at">detection =</span> d, <span class="at">prevalence =</span> p</span>
<span id="cb64-8"><a href="12-Family_and_Genus_taxa.html#cb64-8" tabindex="-1"></a>)</span>
<span id="cb64-9"><a href="12-Family_and_Genus_taxa.html#cb64-9" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Detection = &quot;</span>, d, <span class="st">&quot;, Prevalence = &quot;</span>, p,</span>
<span id="cb64-10"><a href="12-Family_and_Genus_taxa.html#cb64-10" tabindex="-1"></a>       <span class="st">&quot;, Number of genera = &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(test_pseq)))</span>
<span id="cb64-11"><a href="12-Family_and_Genus_taxa.html#cb64-11" tabindex="-1"></a><span class="co">#2</span></span>
<span id="cb64-12"><a href="12-Family_and_Genus_taxa.html#cb64-12" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fl">0.01</span></span>
<span id="cb64-13"><a href="12-Family_and_Genus_taxa.html#cb64-13" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">5</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb64-14"><a href="12-Family_and_Genus_taxa.html#cb64-14" tabindex="-1"></a>test_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb64-15"><a href="12-Family_and_Genus_taxa.html#cb64-15" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb64-16"><a href="12-Family_and_Genus_taxa.html#cb64-16" tabindex="-1"></a>  <span class="at">detection =</span> d, <span class="at">prevalence =</span> p</span>
<span id="cb64-17"><a href="12-Family_and_Genus_taxa.html#cb64-17" tabindex="-1"></a>)</span>
<span id="cb64-18"><a href="12-Family_and_Genus_taxa.html#cb64-18" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Detection = &quot;</span>, d, <span class="st">&quot;, Prevalence = &quot;</span>, p,</span>
<span id="cb64-19"><a href="12-Family_and_Genus_taxa.html#cb64-19" tabindex="-1"></a>       <span class="st">&quot;, Number of genera = &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(test_pseq)))</span>
<span id="cb64-20"><a href="12-Family_and_Genus_taxa.html#cb64-20" tabindex="-1"></a><span class="co">#3</span></span>
<span id="cb64-21"><a href="12-Family_and_Genus_taxa.html#cb64-21" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fl">0.01</span></span>
<span id="cb64-22"><a href="12-Family_and_Genus_taxa.html#cb64-22" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">20</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb64-23"><a href="12-Family_and_Genus_taxa.html#cb64-23" tabindex="-1"></a>test_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb64-24"><a href="12-Family_and_Genus_taxa.html#cb64-24" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb64-25"><a href="12-Family_and_Genus_taxa.html#cb64-25" tabindex="-1"></a>  <span class="at">detection =</span> d, <span class="at">prevalence =</span> p</span>
<span id="cb64-26"><a href="12-Family_and_Genus_taxa.html#cb64-26" tabindex="-1"></a>)</span>
<span id="cb64-27"><a href="12-Family_and_Genus_taxa.html#cb64-27" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Detection = &quot;</span>, d, <span class="st">&quot;, Prevalence = &quot;</span>, p,</span>
<span id="cb64-28"><a href="12-Family_and_Genus_taxa.html#cb64-28" tabindex="-1"></a>       <span class="st">&quot;, Number of genera = &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(test_pseq)))</span>
<span id="cb64-29"><a href="12-Family_and_Genus_taxa.html#cb64-29" tabindex="-1"></a><span class="co">#4</span></span>
<span id="cb64-30"><a href="12-Family_and_Genus_taxa.html#cb64-30" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fl">0.005</span></span>
<span id="cb64-31"><a href="12-Family_and_Genus_taxa.html#cb64-31" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">20</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb64-32"><a href="12-Family_and_Genus_taxa.html#cb64-32" tabindex="-1"></a>test_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb64-33"><a href="12-Family_and_Genus_taxa.html#cb64-33" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb64-34"><a href="12-Family_and_Genus_taxa.html#cb64-34" tabindex="-1"></a>  <span class="at">detection =</span> d, <span class="at">prevalence =</span> p</span>
<span id="cb64-35"><a href="12-Family_and_Genus_taxa.html#cb64-35" tabindex="-1"></a>)</span>
<span id="cb64-36"><a href="12-Family_and_Genus_taxa.html#cb64-36" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Detection = &quot;</span>, d, <span class="st">&quot;, Prevalence = &quot;</span>, p,</span>
<span id="cb64-37"><a href="12-Family_and_Genus_taxa.html#cb64-37" tabindex="-1"></a>       <span class="st">&quot;, Number of genera = &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(test_pseq)))</span>
<span id="cb64-38"><a href="12-Family_and_Genus_taxa.html#cb64-38" tabindex="-1"></a><span class="co">#5</span></span>
<span id="cb64-39"><a href="12-Family_and_Genus_taxa.html#cb64-39" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fl">0.006</span></span>
<span id="cb64-40"><a href="12-Family_and_Genus_taxa.html#cb64-40" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">20</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb64-41"><a href="12-Family_and_Genus_taxa.html#cb64-41" tabindex="-1"></a>test_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb64-42"><a href="12-Family_and_Genus_taxa.html#cb64-42" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb64-43"><a href="12-Family_and_Genus_taxa.html#cb64-43" tabindex="-1"></a>  <span class="at">detection =</span> d, <span class="at">prevalence =</span> p</span>
<span id="cb64-44"><a href="12-Family_and_Genus_taxa.html#cb64-44" tabindex="-1"></a>)</span>
<span id="cb64-45"><a href="12-Family_and_Genus_taxa.html#cb64-45" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Detection = &quot;</span>, d, <span class="st">&quot;, Prevalence = &quot;</span>, p,</span>
<span id="cb64-46"><a href="12-Family_and_Genus_taxa.html#cb64-46" tabindex="-1"></a>       <span class="st">&quot;, Number of genera = &quot;</span>, <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(test_pseq)))</span>
<span id="cb64-47"><a href="12-Family_and_Genus_taxa.html#cb64-47" tabindex="-1"></a></span>
<span id="cb64-48"><a href="12-Family_and_Genus_taxa.html#cb64-48" tabindex="-1"></a><span class="co">#Remove unwanted test_pseq</span></span>
<span id="cb64-49"><a href="12-Family_and_Genus_taxa.html#cb64-49" tabindex="-1"></a><span class="fu">rm</span>(test_pseq)</span></code></pre></div>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="12-Family_and_Genus_taxa.html#cb65-1" tabindex="-1"></a><span class="co">#Aggregate genus phyla</span></span>
<span id="cb65-2"><a href="12-Family_and_Genus_taxa.html#cb65-2" tabindex="-1"></a>genus_rareaggregate_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_rare</span>(</span>
<span id="cb65-3"><a href="12-Family_and_Genus_taxa.html#cb65-3" tabindex="-1"></a>  pseq_relabund, <span class="at">level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb65-4"><a href="12-Family_and_Genus_taxa.html#cb65-4" tabindex="-1"></a>  <span class="at">detection =</span> <span class="fl">0.006</span>, <span class="at">prevalence =</span> <span class="dv">20</span><span class="sc">/</span><span class="dv">100</span></span>
<span id="cb65-5"><a href="12-Family_and_Genus_taxa.html#cb65-5" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div class="webex-solution">
<button>
Bar chart
</button>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="12-Family_and_Genus_taxa.html#cb66-1" tabindex="-1"></a><span class="co">#Plot</span></span>
<span id="cb66-2"><a href="12-Family_and_Genus_taxa.html#cb66-2" tabindex="-1"></a>genus_bar <span class="ot">&lt;-</span> </span>
<span id="cb66-3"><a href="12-Family_and_Genus_taxa.html#cb66-3" tabindex="-1"></a>  microbiome<span class="sc">::</span><span class="fu">plot_composition</span>(genus_rareaggregate_pseq,</span>
<span id="cb66-4"><a href="12-Family_and_Genus_taxa.html#cb66-4" tabindex="-1"></a>                               <span class="co">#otu.sort to sort genera by abundance</span></span>
<span id="cb66-5"><a href="12-Family_and_Genus_taxa.html#cb66-5" tabindex="-1"></a>                               <span class="at">otu.sort =</span> <span class="st">&quot;abundance&quot;</span>, <span class="at">group_by =</span> <span class="st">&quot;media&quot;</span>) <span class="sc">+</span></span>
<span id="cb66-6"><a href="12-Family_and_Genus_taxa.html#cb66-6" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Sample&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Relative abundance&quot;</span>) <span class="sc">+</span></span>
<span id="cb66-7"><a href="12-Family_and_Genus_taxa.html#cb66-7" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Genus relative abundance bar chart&quot;</span>) <span class="sc">+</span></span>
<span id="cb66-8"><a href="12-Family_and_Genus_taxa.html#cb66-8" tabindex="-1"></a>  <span class="co">#Change colours of genera to that of the &quot;Paired&quot; palette from colour brewer</span></span>
<span id="cb66-9"><a href="12-Family_and_Genus_taxa.html#cb66-9" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="st">&quot;Genus&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;Paired&quot;</span>)</span>
<span id="cb66-10"><a href="12-Family_and_Genus_taxa.html#cb66-10" tabindex="-1"></a><span class="co">#Save ggplot object as png</span></span>
<span id="cb66-11"><a href="12-Family_and_Genus_taxa.html#cb66-11" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./genus_relabund_rareaggregate.png&quot;</span>, <span class="at">plot =</span> genus_bar,</span>
<span id="cb66-12"><a href="12-Family_and_Genus_taxa.html#cb66-12" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">200</span>, <span class="at">width =</span> <span class="dv">300</span>)</span>
<span id="cb66-13"><a href="12-Family_and_Genus_taxa.html#cb66-13" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb66-14"><a href="12-Family_and_Genus_taxa.html#cb66-14" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./genus_relabund_rareaggregate.png&quot;</span>)</span></code></pre></div>
</div>
<p>Once you are happy you can <strong>Close and Halt</strong> the notebook.</p>

</div>
</div>
</div>



<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="11-Taxa_plots.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="13-Diversity_analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
