<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 14 Rarefaction | R community analysis</title>
  <meta name="description" content="CGR book for the R community analysis workflow" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 14 Rarefaction | R community analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/CGR_strips.png" />
  <meta property="og:description" content="CGR book for the R community analysis workflow" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 14 Rarefaction | R community analysis" />
  
  <meta name="twitter:description" content="CGR book for the R community analysis workflow" />
  <meta name="twitter:image" content="/figures/CGR_strips.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-07-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/CGR_favicon.png" type="image/x-icon" />
<link rel="prev" href="13-Diversity_analysis.html"/>
<link rel="next" href="15-Alpha.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://www.liverpool.ac.uk/genomic-research/" target="blank"><img src="figures/CGR_strips.png"></a></li>
<li><a>R community analysis</a></li>

<li class="divider"></li>
<li class="part"><span><b>Intro</b></span></li>
<li class="chapter" data-level="1" data-path="01-R_community_main_workflow.html"><a href="01-R_community_main_workflow.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html"><i class="fa fa-check"></i><b>2</b> Dataset &amp; workflow</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#dataset"><i class="fa fa-check"></i><b>2.1</b> Dataset</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sites"><i class="fa fa-check"></i><b>2.1.1</b> Sites</a></li>
<li class="chapter" data-level="2.1.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#culture-media"><i class="fa fa-check"></i><b>2.1.2</b> Culture media</a></li>
<li class="chapter" data-level="2.1.3" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sum_and_qs"><i class="fa fa-check"></i><b>2.1.3</b> Summary &amp; questions</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#workflow"><i class="fa fa-check"></i><b>2.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-R_packages.html"><a href="03-R_packages.html"><i class="fa fa-check"></i><b>3</b> R Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_packages.html"><a href="03-R_packages.html#r-packageslibraries"><i class="fa fa-check"></i><b>3.1</b> R packages/libraries</a></li>
<li class="chapter" data-level="3.2" data-path="03-R_packages.html"><a href="03-R_packages.html#the-grammar-of-graphics"><i class="fa fa-check"></i><b>3.2</b> The grammar of graphics</a></li>
<li class="chapter" data-level="3.3" data-path="03-R_packages.html"><a href="03-R_packages.html#phyloseq"><i class="fa fa-check"></i><b>3.3</b> phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Setup.html"><a href="04-Setup.html"><i class="fa fa-check"></i><b>4</b> Set-up</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Setup.html"><a href="04-Setup.html#cluster"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Setup.html"><a href="04-Setup.html#mamba"><i class="fa fa-check"></i><b>4.2</b> Mamba</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Jupyter.html"><a href="05-Jupyter.html"><i class="fa fa-check"></i><b>5</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Jupyter.html"><a href="05-Jupyter.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.2" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-r-notebook"><i class="fa fa-check"></i><b>5.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.3" data-path="05-Jupyter.html"><a href="05-Jupyter.html#cells-and-code"><i class="fa fa-check"></i><b>5.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.4" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-new-cells"><i class="fa fa-check"></i><b>5.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.5" data-path="05-Jupyter.html"><a href="05-Jupyter.html#running-code"><i class="fa fa-check"></i><b>5.5</b> Running code</a></li>
<li class="chapter" data-level="5.6" data-path="05-Jupyter.html"><a href="05-Jupyter.html#saving-the-file"><i class="fa fa-check"></i><b>5.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.7" data-path="05-Jupyter.html"><a href="05-Jupyter.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.8" data-path="05-Jupyter.html"><a href="05-Jupyter.html#close-the-notebook"><i class="fa fa-check"></i><b>5.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.9" data-path="05-Jupyter.html"><a href="05-Jupyter.html#video-tutorial"><i class="fa fa-check"></i><b>5.9</b> Video tutorial</a></li>
</ul></li>
<li class="part"><span><b>Data preparation</b></span></li>
<li class="chapter" data-level="6" data-path="06-Preprocess_part.html"><a href="06-Preprocess_part.html"><i class="fa fa-check"></i><b>6</b> Data prep intro</a></li>
<li class="chapter" data-level="7" data-path="07-Import.html"><a href="07-Import.html"><i class="fa fa-check"></i><b>7</b> Import</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Import.html"><a href="07-Import.html#import-notebook"><i class="fa fa-check"></i><b>7.1</b> Import: notebook</a></li>
<li class="chapter" data-level="7.2" data-path="07-Import.html"><a href="07-Import.html#qiime2r"><i class="fa fa-check"></i><b>7.2</b> qiime2R</a></li>
<li class="chapter" data-level="7.3" data-path="07-Import.html"><a href="07-Import.html#import-summarise-phyloseq"><i class="fa fa-check"></i><b>7.3</b> Import: Summarise phyloseq</a></li>
<li class="chapter" data-level="7.4" data-path="07-Import.html"><a href="07-Import.html#save-the-phyloseq-object"><i class="fa fa-check"></i><b>7.4</b> Save the phyloseq object</a></li>
<li class="chapter" data-level="7.5" data-path="07-Import.html"><a href="07-Import.html#import-recap"><i class="fa fa-check"></i><b>7.5</b> Import: recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html"><i class="fa fa-check"></i><b>8</b> Summarise phyloseq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-setup"><i class="fa fa-check"></i><b>8.1</b> Summarise: setup</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-header"><i class="fa fa-check"></i><b>8.1.1</b> Summarise header</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-phyloseq"><i class="fa fa-check"></i><b>8.1.2</b> Summarise phyloseq</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#reads-per-sample"><i class="fa fa-check"></i><b>8.1.3</b> Reads per sample</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#display-png"><i class="fa fa-check"></i><b>8.1.4</b> Display png</a></li>
<li class="chapter" data-level="8.1.5" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#asvs-per-sample"><i class="fa fa-check"></i><b>8.1.5</b> ASVs per sample</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-recap"><i class="fa fa-check"></i><b>8.2</b> Summarise: recap</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html"><i class="fa fa-check"></i><b>9</b> Minimum read depth</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#considerations"><i class="fa fa-check"></i><b>9.1</b> Considerations</a></li>
<li class="chapter" data-level="9.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minimum-read-depth-section-title"><i class="fa fa-check"></i><b>9.2</b> Minimum read depth section title</a></li>
<li class="chapter" data-level="9.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#read-depth-vector-and-histogram"><i class="fa fa-check"></i><b>9.3</b> Read depth vector and histogram</a></li>
<li class="chapter" data-level="9.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#depth-boxplots"><i class="fa fa-check"></i><b>9.4</b> Sample depth boxplot</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#creating-a-data-frame-for-boxplots"><i class="fa fa-check"></i><b>9.4.1</b> Creating a data frame for boxplots</a></li>
<li class="chapter" data-level="9.4.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#ggplot2-boxplot"><i class="fa fa-check"></i><b>9.4.2</b> ggplot2 boxplot</a></li>
<li class="chapter" data-level="9.4.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#save-ggplot-as-png"><i class="fa fa-check"></i><b>9.4.3</b> Save ggplot as png</a></li>
<li class="chapter" data-level="9.4.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#media-by-depth-boxplot"><i class="fa fa-check"></i><b>9.4.4</b> Media by depth boxplot</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#chap9rarefaction"><i class="fa fa-check"></i><b>9.5</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#asv-abundance-data-frame"><i class="fa fa-check"></i><b>9.5.1</b> ASV abundance data frame</a></li>
<li class="chapter" data-level="9.5.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#vegans-rarecurve"><i class="fa fa-check"></i><b>9.5.2</b> vegan's rarecurve</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#filtering-by-minimum-read-depth"><i class="fa fa-check"></i><b>9.6</b> Filtering by minimum read depth</a></li>
<li class="chapter" data-level="9.7" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minmum-read-depth-summary"><i class="fa fa-check"></i><b>9.7</b> Minmum read depth: Summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy</b></span></li>
<li class="chapter" data-level="10" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html"><i class="fa fa-check"></i><b>10</b> Taxa relative abundance</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-setup"><i class="fa fa-check"></i><b>10.1</b> Taxa relative abundance: setup</a></li>
<li class="chapter" data-level="10.2" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#relative-abundance-transformation"><i class="fa fa-check"></i><b>10.2</b> Relative abundance transformation</a></li>
<li class="chapter" data-level="10.3" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#aggregate-taxa"><i class="fa fa-check"></i><b>10.3</b> Aggregate taxa</a></li>
<li class="chapter" data-level="10.4" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-summary"><i class="fa fa-check"></i><b>10.4</b> Taxa Relative abundance: summary</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html"><i class="fa fa-check"></i><b>11</b> Taxa plots</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#simple-bar-chart"><i class="fa fa-check"></i><b>11.1</b> Simple bar chart</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#editing-the-plot"><i class="fa fa-check"></i><b>11.1.1</b> Editing the plot</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-heatmap"><i class="fa fa-check"></i><b>11.2</b> Taxa heatmap</a></li>
<li class="chapter" data-level="11.3" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#aggregateraretaxa"><i class="fa fa-check"></i><b>11.3</b> Aggregate rare taxa</a></li>
<li class="chapter" data-level="11.4" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-media"><i class="fa fa-check"></i><b>11.4</b> Plot by media</a></li>
<li class="chapter" data-level="11.5" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-site-and-media"><i class="fa fa-check"></i><b>11.5</b> Plot by site and media</a></li>
<li class="chapter" data-level="11.6" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-plots-summary"><i class="fa fa-check"></i><b>11.6</b> Taxa plots: summary</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html"><i class="fa fa-check"></i><b>12</b> Family &amp; Genus plots</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-taxa-plots"><i class="fa fa-check"></i><b>12.1</b> Family taxa plots</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#aggregate-families"><i class="fa fa-check"></i><b>12.1.1</b> Aggregate families</a></li>
<li class="chapter" data-level="12.1.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-heatmap"><i class="fa fa-check"></i><b>12.1.2</b> Family heatmap</a></li>
<li class="chapter" data-level="12.1.3" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-rare-aggregation"><i class="fa fa-check"></i><b>12.1.3</b> Family rare aggregation</a></li>
<li class="chapter" data-level="12.1.4" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-bar-chart"><i class="fa fa-check"></i><b>12.1.4</b> Family bar chart</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots"><i class="fa fa-check"></i><b>12.2</b> Genus taxa plots</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots-solutions"><i class="fa fa-check"></i><b>12.2.1</b> Genus taxa plots solutions</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Diversity analysis</b></span></li>
<li class="chapter" data-level="13" data-path="13-Diversity_analysis.html"><a href="13-Diversity_analysis.html"><i class="fa fa-check"></i><b>13</b> Diversity analysis intro</a></li>
<li class="chapter" data-level="14" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html"><i class="fa fa-check"></i><b>14</b> Rarefaction</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-setup"><i class="fa fa-check"></i><b>14.1</b> Rarefaction: setup</a></li>
<li class="chapter" data-level="14.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve"><i class="fa fa-check"></i><b>14.2</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#bird-analogy"><i class="fa fa-check"></i><b>14.2.1</b> Bird analogy</a></li>
<li class="chapter" data-level="14.2.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-simple-plot"><i class="fa fa-check"></i><b>14.2.2</b> Rarefaction curve: simple plot</a></li>
<li class="chapter" data-level="14.2.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-better-plot"><i class="fa fa-check"></i><b>14.2.3</b> Rarefaction curve: better plot</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-slope"><i class="fa fa-check"></i><b>14.3</b> Rarefaction slope</a></li>
<li class="chapter" data-level="14.4" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefy-data"><i class="fa fa-check"></i><b>14.4</b> Rarefy data</a></li>
<li class="chapter" data-level="14.5" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-summary"><i class="fa fa-check"></i><b>14.5</b> Rarefaction: summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Alpha.html"><a href="15-Alpha.html"><i class="fa fa-check"></i><b>15</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-setup"><i class="fa fa-check"></i><b>15.1</b> Alpha: setup</a></li>
<li class="chapter" data-level="15.2" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-simple-plot"><i class="fa fa-check"></i><b>15.2</b> Alpha: simple plot</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="15-Alpha.html"><a href="15-Alpha.html#metric-choice"><i class="fa fa-check"></i><b>15.2.1</b> Metric choice</a></li>
<li class="chapter" data-level="15.2.2" data-path="15-Alpha.html"><a href="15-Alpha.html#sample-grouping-box-plot"><i class="fa fa-check"></i><b>15.2.2</b> Sample grouping &amp; box plot</a></li>
<li class="chapter" data-level="15.2.3" data-path="15-Alpha.html"><a href="15-Alpha.html#png-height-and-width-choice"><i class="fa fa-check"></i><b>15.2.3</b> PNG height and width choice</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats"><i class="fa fa-check"></i><b>15.3</b> Alpha: stats</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats-task"><i class="fa fa-check"></i><b>15.3.1</b> Alpha stats task</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-diversity-violin-plot"><i class="fa fa-check"></i><b>15.4</b> Alpha diversity: violin plot</a></li>
<li class="chapter" data-level="15.5" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-site-plots-and-statistics"><i class="fa fa-check"></i><b>15.5</b> Alpha: Site plots and statistics</a></li>
<li class="chapter" data-level="15.6" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-summary"><i class="fa fa-check"></i><b>15.6</b> Alpha: summary</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Beta.html"><a href="16-Beta.html"><i class="fa fa-check"></i><b>16</b> Beta</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Beta.html"><a href="16-Beta.html#beta-setup"><i class="fa fa-check"></i><b>16.1</b> Beta: setup</a></li>
<li class="chapter" data-level="16.2" data-path="16-Beta.html"><a href="16-Beta.html#beta-ordination"><i class="fa fa-check"></i><b>16.2</b> Beta: ordination</a></li>
<li class="chapter" data-level="16.3" data-path="16-Beta.html"><a href="16-Beta.html#beta-plot-ordination"><i class="fa fa-check"></i><b>16.3</b> Beta: plot ordination</a></li>
<li class="chapter" data-level="16.4" data-path="16-Beta.html"><a href="16-Beta.html#beta-statistics"><i class="fa fa-check"></i><b>16.4</b> Beta: statistics</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="16-Beta.html"><a href="16-Beta.html#permanova"><i class="fa fa-check"></i><b>16.4.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="16.4.2" data-path="16-Beta.html"><a href="16-Beta.html#pairwise-permanova"><i class="fa fa-check"></i><b>16.4.2</b> Pairwise PERMANOVA</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="16-Beta.html"><a href="16-Beta.html#beta-facet-zoom"><i class="fa fa-check"></i><b>16.5</b> Beta: Facet zoom</a></li>
<li class="chapter" data-level="16.6" data-path="16-Beta.html"><a href="16-Beta.html#beta-subset-samples"><i class="fa fa-check"></i><b>16.6</b> Beta: subset samples</a></li>
<li class="chapter" data-level="16.7" data-path="16-Beta.html"><a href="16-Beta.html#permanova-tasks"><i class="fa fa-check"></i><b>16.7</b> PERMANOVA tasks</a></li>
<li class="chapter" data-level="16.8" data-path="16-Beta.html"><a href="16-Beta.html#beta-summary"><i class="fa fa-check"></i><b>16.8</b> Beta: Summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html"><i class="fa fa-check"></i><b>17</b> Differential abundance (DA) analysis</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-setup"><i class="fa fa-check"></i><b>17.1</b> DA: setup</a></li>
<li class="chapter" data-level="17.2" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-preprocess"><i class="fa fa-check"></i><b>17.2</b> DA: Preprocess</a></li>
<li class="chapter" data-level="17.3" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-analysis"><i class="fa fa-check"></i><b>17.3</b> DA: ANCOM-BC analysis</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#ancombc2-options"><i class="fa fa-check"></i><b>17.3.1</b> ANCOMBC2 options</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-results"><i class="fa fa-check"></i><b>17.4</b> DA: ANCOM-BC results</a></li>
<li class="chapter" data-level="17.5" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-create-signifcant-lfc-long-data-frame"><i class="fa fa-check"></i><b>17.5</b> DA: Create signifcant LFC long data frame</a></li>
<li class="chapter" data-level="17.6" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-biomarker-lfc-heatmap"><i class="fa fa-check"></i><b>17.6</b> DA: Biomarker LFC heatmap</a></li>
<li class="chapter" data-level="17.7" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-matching-bar-plot"><i class="fa fa-check"></i><b>17.7</b> DA: matching bar plot</a></li>
<li class="chapter" data-level="17.8" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-summary"><i class="fa fa-check"></i><b>17.8</b> DA: summary</a></li>
</ul></li>
<li class="part"><span><b>Outro</b></span></li>
<li class="chapter" data-level="18" data-path="18-Summary.html"><a href="18-Summary.html"><i class="fa fa-check"></i><b>18</b> Summary</a>
<ul>
<li class="chapter" data-level="18.1" data-path="18-Summary.html"><a href="18-Summary.html#dataset-1"><i class="fa fa-check"></i><b>18.1</b> Dataset</a></li>
<li class="chapter" data-level="18.2" data-path="18-Summary.html"><a href="18-Summary.html#data-preparation"><i class="fa fa-check"></i><b>18.2</b> Data preparation</a></li>
<li class="chapter" data-level="18.3" data-path="18-Summary.html"><a href="18-Summary.html#taxonomy"><i class="fa fa-check"></i><b>18.3</b> Taxonomy</a></li>
<li class="chapter" data-level="18.4" data-path="18-Summary.html"><a href="18-Summary.html#diversity-analysis"><i class="fa fa-check"></i><b>18.4</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="18-Summary.html"><a href="18-Summary.html#alpha-beta"><i class="fa fa-check"></i><b>18.4.1</b> Alpha &amp; beta</a></li>
<li class="chapter" data-level="18.4.2" data-path="18-Summary.html"><a href="18-Summary.html#differential-abundance-anlaysis"><i class="fa fa-check"></i><b>18.4.2</b> Differential abundance anlaysis</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="18-Summary.html"><a href="18-Summary.html#findings"><i class="fa fa-check"></i><b>18.5</b> Findings</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="18-Summary.html"><a href="18-Summary.html#anthropisation-gradient"><i class="fa fa-check"></i><b>18.5.1</b> Anthropisation gradient</a></li>
<li class="chapter" data-level="18.5.2" data-path="18-Summary.html"><a href="18-Summary.html#replicate-differences"><i class="fa fa-check"></i><b>18.5.2</b> Replicate differences</a></li>
<li class="chapter" data-level="18.5.3" data-path="18-Summary.html"><a href="18-Summary.html#site-or-media"><i class="fa fa-check"></i><b>18.5.3</b> Site or media</a></li>
<li class="chapter" data-level="18.5.4" data-path="18-Summary.html"><a href="18-Summary.html#media-versus-env-samples"><i class="fa fa-check"></i><b>18.5.4</b> Media versus ENV samples</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="18-Summary.html"><a href="18-Summary.html#conclusion"><i class="fa fa-check"></i><b>18.6</b> Conclusion</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="19-Appendix.html"><a href="19-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="19-Appendix.html"><a href="19-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
<li class="chapter" data-level="A.2" data-path="19-Appendix.html"><a href="19-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>A.2</b> Jupyter-notebook</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="19-Appendix.html"><a href="19-Appendix.html#r-packages-1"><i class="fa fa-check"></i><b>B</b> R packages</a>
<ul>
<li class="chapter" data-level="B.1" data-path="19-Appendix.html"><a href="19-Appendix.html#ggplot2_appendix"><i class="fa fa-check"></i><b>B.1</b> ggplot2</a></li>
<li class="chapter" data-level="B.2" data-path="19-Appendix.html"><a href="19-Appendix.html#phyloseq_appendix"><i class="fa fa-check"></i><b>B.2</b> Phyloseq</a></li>
<li class="chapter" data-level="B.3" data-path="19-Appendix.html"><a href="19-Appendix.html#importing-into-phyloseq"><i class="fa fa-check"></i><b>B.3</b> Importing into phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-metrics"><i class="fa fa-check"></i><b>C</b> Diversity metrics</a>
<ul>
<li class="chapter" data-level="C.1" data-path="19-Appendix.html"><a href="19-Appendix.html#alpha_appendix"><i class="fa fa-check"></i><b>C.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="19-Appendix.html"><a href="19-Appendix.html#obvs"><i class="fa fa-check"></i><b>C.1.1</b> Observed features</a></li>
<li class="chapter" data-level="C.1.2" data-path="19-Appendix.html"><a href="19-Appendix.html#chao"><i class="fa fa-check"></i><b>C.1.2</b> Chao</a></li>
<li class="chapter" data-level="C.1.3" data-path="19-Appendix.html"><a href="19-Appendix.html#evenness"><i class="fa fa-check"></i><b>C.1.3</b> Evenness</a></li>
<li class="chapter" data-level="C.1.4" data-path="19-Appendix.html"><a href="19-Appendix.html#faith"><i class="fa fa-check"></i><b>C.1.4</b> Faith's PD (phylogenetic diversity)</a></li>
<li class="chapter" data-level="C.1.5" data-path="19-Appendix.html"><a href="19-Appendix.html#simpson"><i class="fa fa-check"></i><b>C.1.5</b> Simpson</a></li>
<li class="chapter" data-level="C.1.6" data-path="19-Appendix.html"><a href="19-Appendix.html#simpsone"><i class="fa fa-check"></i><b>C.1.6</b> Simpson evenness measure (simpson_e)</a></li>
<li class="chapter" data-level="C.1.7" data-path="19-Appendix.html"><a href="19-Appendix.html#shannon"><i class="fa fa-check"></i><b>C.1.7</b> Shannon</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="19-Appendix.html"><a href="19-Appendix.html#beta_appendix"><i class="fa fa-check"></i><b>C.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="19-Appendix.html"><a href="19-Appendix.html#unifrac_explanation"><i class="fa fa-check"></i><b>C.2.1</b> Weighted and unweighted UniFrac distances</a></li>
<li class="chapter" data-level="C.2.2" data-path="19-Appendix.html"><a href="19-Appendix.html#bray_explanation"><i class="fa fa-check"></i><b>C.2.2</b> Bray-Curtis</a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-resources"><i class="fa fa-check"></i><b>C.3</b> Diversity Resources</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R community analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rarefaction_chap" class="section level1 hasAnchor" number="14">
<h1><span class="header-section-number">Chapter 14</span> Rarefaction<a href="14-Rarefaction.html#rarefaction_chap" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/whale.png" width="20%" style="display: block; margin: auto;" /></p>
<p>We are going to create a rarefied abundance table in this chapter.
This can be a controversial topic.
Please make your own decision if you want to do this or not in your own analysis.</p>
<div id="rarefaction-setup" class="section level2 hasAnchor" number="14.1">
<h2><span class="header-section-number">14.1</span> Rarefaction: setup<a href="14-Rarefaction.html#rarefaction-setup" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/library_1.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We will use a new notebook called <strong>"4-rarefaction.ipynb"</strong>.
Add the following to the top of this new notebook to load the required libraries/packages and data.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="14-Rarefaction.html#cb67-1" tabindex="-1"></a><span class="co">#Libraries</span></span>
<span id="cb67-2"><a href="14-Rarefaction.html#cb67-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb67-3"><a href="14-Rarefaction.html#cb67-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;microbiome&quot;</span>)</span>
<span id="cb67-4"><a href="14-Rarefaction.html#cb67-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;vegan&quot;</span>)</span>
<span id="cb67-5"><a href="14-Rarefaction.html#cb67-5" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;IRdisplay&quot;</span>)</span>
<span id="cb67-6"><a href="14-Rarefaction.html#cb67-6" tabindex="-1"></a><span class="co">#Load phyloseq object</span></span>
<span id="cb67-7"><a href="14-Rarefaction.html#cb67-7" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;phyloseq.RData&quot;</span>)</span>
<span id="cb67-8"><a href="14-Rarefaction.html#cb67-8" tabindex="-1"></a><span class="co">#Load ASV count vector</span></span>
<span id="cb67-9"><a href="14-Rarefaction.html#cb67-9" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;num_asvs_vec.RData&quot;</span>)</span></code></pre></div>
<p>Ensure you add markdown and code cells to this notebook to give yourself
a good structure.</p>
</div>
<div id="rarefaction-curve" class="section level2 hasAnchor" number="14.2">
<h2><span class="header-section-number">14.2</span> Rarefaction curve<a href="14-Rarefaction.html#rarefaction-curve" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/rarefaction.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Prior to rarefying our data we need to determine the depth we want to
use. We can carry this out with a rarefaction curve.</p>
<p>A rarefaction curve is produced by randomly sampling the sequences in a
sample (without replacement). The rarefaction curve extracts
the number of unique ASVs in the first N sequences of each sample.
N is equal to the step size which
we will set as 50 (<code>step = 50</code>).
This is followed by the number of unique ASVs found in 2N, then 3N etc.</p>
<p>The plot will show the number of total unique ASVs it has found
against the depth it has currently sampled. Therefore, after 10 steps it
will be at a depth of 500 on the x-axis (50 * 10) and it will show how
many unique ASVs it has currently discovered from all 10 samplings.</p>
<div id="bird-analogy" class="section level3 hasAnchor" number="14.2.1">
<h3><span class="header-section-number">14.2.1</span> Bird analogy<a href="14-Rarefaction.html#bird-analogy" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/ibyster.png" width="15%" style="display: block; margin: auto;" /></p>
<p>You are cataloguing the bird diversity (ASV diversity) in three forests/woods
in Hampshire (3 samples of our abundance table). You record your
findings over a week for each forest. Your record on the cumulative
amount of unique species you find over the 3 separate weeks are in the
table below:</p>
<table>
<caption>Cumulative bird species found in Hampshire forests/woods over a week</caption>
<colgroup>
<col width="20%" />
<col width="24%" />
<col width="27%" />
<col width="27%" />
</colgroup>
<thead>
<tr class="header">
<th>Day</th>
<th>New Forest (93900 acres)</th>
<th>Telegraph woods (50 acres)</th>
<th>Old Lords wood (4.67 acres)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>12</td>
<td>7</td>
<td>8</td>
</tr>
<tr class="even">
<td>2</td>
<td>21</td>
<td>12</td>
<td>10</td>
</tr>
<tr class="odd">
<td>3</td>
<td>32</td>
<td>16</td>
<td>11</td>
</tr>
<tr class="even">
<td>4</td>
<td>40</td>
<td>21</td>
<td>13</td>
</tr>
<tr class="odd">
<td>5</td>
<td>49</td>
<td>23</td>
<td>13</td>
</tr>
<tr class="even">
<td>6</td>
<td>60</td>
<td>24</td>
<td>13</td>
</tr>
<tr class="odd">
<td>7</td>
<td>65</td>
<td>25</td>
<td>13</td>
</tr>
</tbody>
</table>
<p>This can then be plotted in a rarefaction plot like below.</p>
<p><img src="CGR-R-community-analysis_files/figure-html/unnamed-chunk-171-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>In this case we can be fairly confident that seven days is enough to find all the birds we can in the Old Lords wood (green).
I.e. the sampling depth is enough.
Just like bird watching, it is very hard to find all the species present in a sample with metabarcoding or shotgun metagenomics.
This can be caused by our barcodes not picking up all species, our sampling wasn't perfect, some species are very rare, or errors that DNA sequencing introduces.</p>
<p>A week seems like a good amount for sampling the Telegraph woods (Blue) but it is possible that some more species have not been recorded.</p>
<p>Strikingly, a week does not appear to be enough time for the New Forest (red). The amount of new species decreased to 5 on day 7 but the curve has not yet plateaued.</p>
<p>This hopefully helps you understand why rarefaction curves are a useful measure of how well we have captured the diversity of samples at different depths.</p>
</div>
<div id="rarefaction-curve-simple-plot" class="section level3 hasAnchor" number="14.2.2">
<h3><span class="header-section-number">14.2.2</span> Rarefaction curve: simple plot<a href="14-Rarefaction.html#rarefaction-curve-simple-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/crayon.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We'll use <code>vegan</code> to create our rarefaction curve like we did in
<a href="09-Minimum_read_depth.html#chap9rarefaction">Chapter 9</a>.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="14-Rarefaction.html#cb68-1" tabindex="-1"></a><span class="co">#Extract ASV table as data frame</span></span>
<span id="cb68-2"><a href="14-Rarefaction.html#cb68-2" tabindex="-1"></a>asv_abund_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(pseq)))</span>
<span id="cb68-3"><a href="14-Rarefaction.html#cb68-3" tabindex="-1"></a><span class="co">#Rarefaction curve</span></span>
<span id="cb68-4"><a href="14-Rarefaction.html#cb68-4" tabindex="-1"></a><span class="co">#Save as png file</span></span>
<span id="cb68-5"><a href="14-Rarefaction.html#cb68-5" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename =</span> <span class="st">&quot;./rarefaction_plot.png&quot;</span>, <span class="at">res =</span> <span class="dv">300</span>,</span>
<span id="cb68-6"><a href="14-Rarefaction.html#cb68-6" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">200</span>, <span class="at">width =</span> <span class="dv">300</span>)</span>
<span id="cb68-7"><a href="14-Rarefaction.html#cb68-7" tabindex="-1"></a><span class="co">#Produce plot</span></span>
<span id="cb68-8"><a href="14-Rarefaction.html#cb68-8" tabindex="-1"></a>vegan<span class="sc">::</span><span class="fu">rarecurve</span>(</span>
<span id="cb68-9"><a href="14-Rarefaction.html#cb68-9" tabindex="-1"></a>  <span class="at">x =</span> asv_abund_df, <span class="at">step =</span> <span class="dv">50</span>,</span>
<span id="cb68-10"><a href="14-Rarefaction.html#cb68-10" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">&quot;Read depth&quot;</span>,</span>
<span id="cb68-11"><a href="14-Rarefaction.html#cb68-11" tabindex="-1"></a>  <span class="at">ylab =</span> <span class="st">&quot;ASVs&quot;</span></span>
<span id="cb68-12"><a href="14-Rarefaction.html#cb68-12" tabindex="-1"></a>)</span>
<span id="cb68-13"><a href="14-Rarefaction.html#cb68-13" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb68-14"><a href="14-Rarefaction.html#cb68-14" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb68-15"><a href="14-Rarefaction.html#cb68-15" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./rarefaction_plot.png&quot;</span>)</span></code></pre></div>
</div>
<div id="rarefaction-curve-better-plot" class="section level3 hasAnchor" number="14.2.3">
<h3><span class="header-section-number">14.2.3</span> Rarefaction curve: better plot<a href="14-Rarefaction.html#rarefaction-curve-better-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/palette_brush.png" width="15%" style="display: block; margin: auto;" /></p>
<p>It is a useful plot but we can make it better!
We will do that with the following addition:</p>
<ul>
<li>Adding extra options to <code>vegan::rarecurve()</code>:
<ul>
<li><code>lwd</code>: Sets the line width of the plot.</li>
<li><code>label</code>: Turn the sample labels on (<code>T</code>) or off (<code>F</code>). We will turn them off as it is hard to read them all.</li>
<li><code>sample</code>: Draws a vertical line at the specified depth.
Additionally, draws a horizontal line for each sample showing
how many ASVs were discovered at the sampling depth.</li>
</ul></li>
</ul>
<p>We will initially use our minimum sample depth as our rarefaction/sampling depth.
This is a good idea if it seems like a good amount as it will allow us to keep all our samples after rarefaction.
Any samples with a lower depth than are final rarefaction size will be removed.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="14-Rarefaction.html#cb69-1" tabindex="-1"></a><span class="co">#Improved plot saved as file</span></span>
<span id="cb69-2"><a href="14-Rarefaction.html#cb69-2" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename =</span> <span class="st">&quot;./rarefaction_plot.png&quot;</span>, <span class="at">res =</span> <span class="dv">300</span>,</span>
<span id="cb69-3"><a href="14-Rarefaction.html#cb69-3" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">200</span>, <span class="at">width =</span> <span class="dv">300</span>)</span>
<span id="cb69-4"><a href="14-Rarefaction.html#cb69-4" tabindex="-1"></a><span class="co">#Plot</span></span>
<span id="cb69-5"><a href="14-Rarefaction.html#cb69-5" tabindex="-1"></a>vegan<span class="sc">::</span><span class="fu">rarecurve</span>(</span>
<span id="cb69-6"><a href="14-Rarefaction.html#cb69-6" tabindex="-1"></a>  <span class="at">x =</span> asv_abund_df, <span class="at">step =</span> <span class="dv">50</span>,</span>
<span id="cb69-7"><a href="14-Rarefaction.html#cb69-7" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">&quot;Read depth&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;ASVs&quot;</span>, <span class="at">lwd=</span><span class="dv">1</span>, <span class="at">label =</span> F,</span>
<span id="cb69-8"><a href="14-Rarefaction.html#cb69-8" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">min</span>(microbiome<span class="sc">::</span><span class="fu">readcount</span>(pseq))</span>
<span id="cb69-9"><a href="14-Rarefaction.html#cb69-9" tabindex="-1"></a>)</span>
<span id="cb69-10"><a href="14-Rarefaction.html#cb69-10" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb69-11"><a href="14-Rarefaction.html#cb69-11" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb69-12"><a href="14-Rarefaction.html#cb69-12" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./rarefaction_plot.png&quot;</span>)</span></code></pre></div>
<p>From this plot we can see that most of the samples have plateu'd quite nicely at the minimum depth.
This indicates that this is a good sampling depth, i.e. depth to be chosen for rarefying the data.
The grey horizontal lines help show how many more ASVs are found in a sample after the chosen sampling depth.
This can be seen in the sample with the highest number of ASVs where 5-20 more ASVs are discovered with its final depth of ~16,000.</p>
</div>
</div>
<div id="rarefaction-slope" class="section level2 hasAnchor" number="14.3">
<h2><span class="header-section-number">14.3</span> Rarefaction slope<a href="14-Rarefaction.html#rarefaction-slope" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/hill_landscape.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We can calculate the slope of each sample at our specified sampling depth.
A slope of 0 shows a flat horizontal line (what we want).
A slope of 1 shows a flat vertical line (not what we want).</p>
<p>To get the slope of each sample we will use <code>vegan::rareslope()</code>.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="14-Rarefaction.html#cb70-1" tabindex="-1"></a><span class="co">#Rarefaction slopes</span></span>
<span id="cb70-2"><a href="14-Rarefaction.html#cb70-2" tabindex="-1"></a>rarefaction_slopes <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">rareslope</span>(</span>
<span id="cb70-3"><a href="14-Rarefaction.html#cb70-3" tabindex="-1"></a>  <span class="at">x =</span> asv_abund_df, <span class="at">sample =</span> <span class="fu">min</span>(microbiome<span class="sc">::</span><span class="fu">readcount</span>(pseq))</span>
<span id="cb70-4"><a href="14-Rarefaction.html#cb70-4" tabindex="-1"></a>)</span>
<span id="cb70-5"><a href="14-Rarefaction.html#cb70-5" tabindex="-1"></a><span class="co">#View slopes from lowest to highest value</span></span>
<span id="cb70-6"><a href="14-Rarefaction.html#cb70-6" tabindex="-1"></a><span class="fu">sort</span>(rarefaction_slopes)</span>
<span id="cb70-7"><a href="14-Rarefaction.html#cb70-7" tabindex="-1"></a><span class="co">#Summary of slopes</span></span>
<span id="cb70-8"><a href="14-Rarefaction.html#cb70-8" tabindex="-1"></a><span class="fu">summary</span>(rarefaction_slopes)</span>
<span id="cb70-9"><a href="14-Rarefaction.html#cb70-9" tabindex="-1"></a><span class="co">#Histogram of slopes</span></span>
<span id="cb70-10"><a href="14-Rarefaction.html#cb70-10" tabindex="-1"></a><span class="co">#Save as png</span></span>
<span id="cb70-11"><a href="14-Rarefaction.html#cb70-11" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename =</span> <span class="st">&quot;./rarefaction_slopes_histogram.png&quot;</span>, <span class="at">res =</span> <span class="dv">300</span>,</span>
<span id="cb70-12"><a href="14-Rarefaction.html#cb70-12" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">100</span>, <span class="at">width =</span> <span class="dv">200</span>)</span>
<span id="cb70-13"><a href="14-Rarefaction.html#cb70-13" tabindex="-1"></a><span class="co">#Plot</span></span>
<span id="cb70-14"><a href="14-Rarefaction.html#cb70-14" tabindex="-1"></a><span class="fu">hist</span>(rarefaction_slopes)</span>
<span id="cb70-15"><a href="14-Rarefaction.html#cb70-15" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb70-16"><a href="14-Rarefaction.html#cb70-16" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb70-17"><a href="14-Rarefaction.html#cb70-17" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./rarefaction_slopes_histogram.png&quot;</span>)</span></code></pre></div>
<p>With the calculated slopes attempt the following questions:</p>
<ol style="list-style-type: decimal">
<li>Which sample has the largest slope value?
<div id="radio_HXVAYKLJFJ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_HXVAYKLJFJ" value=""></input> <span><strong>UD_ENV_rep2</strong></span></label><label><input type="radio" autocomplete="off" name="radio_HXVAYKLJFJ" value="answer"></input> <span><strong>MD_ENV_rep3</strong></span></label><label><input type="radio" autocomplete="off" name="radio_HXVAYKLJFJ" value=""></input> <span><strong>LD_CVP_rep1</strong></span></label>
</div></li>
<li>Which sample has the lowest slope value?
<div id="radio_ELPZIZHLPJ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ELPZIZHLPJ" value=""></input> <span><strong>UD_ENV_rep2</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ELPZIZHLPJ" value=""></input> <span><strong>MD_ENV_rep3</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ELPZIZHLPJ" value="answer"></input> <span><strong>LD_CVP_rep1</strong></span></label>
</div></li>
<li>Which samples have larger slope values; the environmental samples or media samples?
<div id="radio_LFDKKLJQEG" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_LFDKKLJQEG" value="answer"></input> <span><strong>Environmental</strong></span></label><label><input type="radio" autocomplete="off" name="radio_LFDKKLJQEG" value=""></input> <span><strong>Media</strong></span></label>
</div></li>
</ol>
<p>Overall the slopes values are low (&lt;0.01).
It is noticeable that the environmental samples have higher slopes than the media samples.
Biologically, this makes sense as there is no bias being introduced by media.</p>
</div>
<div id="rarefy-data" class="section level2 hasAnchor" number="14.4">
<h2><span class="header-section-number">14.4</span> Rarefy data<a href="14-Rarefaction.html#rarefy-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/pangolin.png" width="20%" style="display: block; margin: auto;" /></p>
<p>The minimum sample depth appears to be a good choice for rarefying.
It keeps all the samples and at this depth the samples have a very good coverage of the ASVs present in the data.
This is represented by the good plateauing in the rarefaction plot and the low valued slopes (&lt;0.01).
In your own analyses you may need to balance the loss of samples with the loss of depth.</p>
<p>With that decision we will rarefy our data to the minimum depth.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="14-Rarefaction.html#cb71-1" tabindex="-1"></a><span class="co">#Rarefy to minimum depth</span></span>
<span id="cb71-2"><a href="14-Rarefaction.html#cb71-2" tabindex="-1"></a>pseq_rarefy <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">rarefy_even_depth</span>(</span>
<span id="cb71-3"><a href="14-Rarefaction.html#cb71-3" tabindex="-1"></a>  pseq, <span class="at">sample.size =</span> <span class="fu">min</span>(microbiome<span class="sc">::</span><span class="fu">readcount</span>(pseq)),</span>
<span id="cb71-4"><a href="14-Rarefaction.html#cb71-4" tabindex="-1"></a>  <span class="at">rngseed =</span> <span class="dv">1000</span></span>
<span id="cb71-5"><a href="14-Rarefaction.html#cb71-5" tabindex="-1"></a>  )</span></code></pre></div>
<p>The options we provided to the function <code>rarefy_even_depth</code> were:</p>
<ul>
<li><strong><code>pseq</code></strong>: The <code>phyloseq</code> object to rarefy.</li>
<li><strong><code>sample.size =</code></strong>: The sampling depth to rarefy to.</li>
<li><strong><code>rngseed</code></strong>: This is the seed used for random subsampling.
<ul>
<li>If you rarefy the data again with the same seed it will extract the same data.</li>
<li>This is useful so you and others can replicate your work.</li>
<li>Additionally, it means you will get the same output as me (the writer) so you can accurately compare your findings to mine.</li>
</ul></li>
</ul>
<p>As always, it is useful to check our data.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="14-Rarefaction.html#cb72-1" tabindex="-1"></a><span class="co">#Summarise and check sample counts which should each amount to 10433 (min depth)</span></span>
<span id="cb72-2"><a href="14-Rarefaction.html#cb72-2" tabindex="-1"></a>microbiome<span class="sc">::</span><span class="fu">summarize_phyloseq</span>(pseq_rarefy)</span>
<span id="cb72-3"><a href="14-Rarefaction.html#cb72-3" tabindex="-1"></a>microbiome<span class="sc">::</span><span class="fu">readcount</span>(pseq_rarefy)</span>
<span id="cb72-4"><a href="14-Rarefaction.html#cb72-4" tabindex="-1"></a><span class="co">#ASV counts</span></span>
<span id="cb72-5"><a href="14-Rarefaction.html#cb72-5" tabindex="-1"></a><span class="co">#Add relative abundance ASV count</span></span>
<span id="cb72-6"><a href="14-Rarefaction.html#cb72-6" tabindex="-1"></a>num_asvs_vec[<span class="st">&quot;rarefied&quot;</span>] <span class="ot">&lt;-</span> <span class="fu">nrow</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(pseq_rarefy))</span>
<span id="cb72-7"><a href="14-Rarefaction.html#cb72-7" tabindex="-1"></a>num_asvs_vec</span></code></pre></div>
<p>With this we can see that all our samples have the correct depth (10,433) and very few ASVs have been lost (2551-2498 = 53).
I would definetly be happy with this outcome.</p>
<p>Once you have viewed the outputs save your <code>phyloseq</code> object and ASV count vector.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="14-Rarefaction.html#cb73-1" tabindex="-1"></a><span class="co">#Phyloseq save</span></span>
<span id="cb73-2"><a href="14-Rarefaction.html#cb73-2" tabindex="-1"></a><span class="fu">save</span>(pseq_rarefy, <span class="at">file =</span><span class="st">&quot;phyloseq_rarefied.RData&quot;</span>)</span>
<span id="cb73-3"><a href="14-Rarefaction.html#cb73-3" tabindex="-1"></a><span class="co">#ASV count save</span></span>
<span id="cb73-4"><a href="14-Rarefaction.html#cb73-4" tabindex="-1"></a><span class="fu">save</span>(num_asvs_vec, <span class="at">file=</span><span class="st">&quot;num_asvs_vec.v2.RData&quot;</span>)</span></code></pre></div>
<p>Now you can close and halt the notebook.</p>
</div>
<div id="rarefaction-summary" class="section level2 hasAnchor" number="14.5">
<h2><span class="header-section-number">14.5</span> Rarefaction: summary<a href="14-Rarefaction.html#rarefaction-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_orange.png" width="15%" style="display: block; margin: auto;" /></p>
<p>In this chapter we:</p>
<ul>
<li>Determined the sampling depth to rarefy to with rarefaction curves and slopes.</li>
<li>Rarefied our samples to the minimum read depth. This retained all our samples whilst using a good depth.</li>
</ul>
<p>We now have an abundance table, relative abundance table, and rarefied abundance table. We will use these to carry out some analyses.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="13-Diversity_analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="15-Alpha.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
