<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 15 Alpha diversity | R community analysis</title>
  <meta name="description" content="CGR book for the R community analysis workflow" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 15 Alpha diversity | R community analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/CGR_strips.png" />
  <meta property="og:description" content="CGR book for the R community analysis workflow" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 15 Alpha diversity | R community analysis" />
  
  <meta name="twitter:description" content="CGR book for the R community analysis workflow" />
  <meta name="twitter:image" content="/figures/CGR_strips.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-07-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/CGR_favicon.png" type="image/x-icon" />
<link rel="prev" href="14-Rarefaction.html"/>
<link rel="next" href="16-Beta.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://www.liverpool.ac.uk/genomic-research/" target="blank"><img src="figures/CGR_strips.png"></a></li>
<li><a>R community analysis</a></li>

<li class="divider"></li>
<li class="part"><span><b>Intro</b></span></li>
<li class="chapter" data-level="1" data-path="01-R_community_main_workflow.html"><a href="01-R_community_main_workflow.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html"><i class="fa fa-check"></i><b>2</b> Dataset &amp; workflow</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#dataset"><i class="fa fa-check"></i><b>2.1</b> Dataset</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sites"><i class="fa fa-check"></i><b>2.1.1</b> Sites</a></li>
<li class="chapter" data-level="2.1.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#culture-media"><i class="fa fa-check"></i><b>2.1.2</b> Culture media</a></li>
<li class="chapter" data-level="2.1.3" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sum_and_qs"><i class="fa fa-check"></i><b>2.1.3</b> Summary &amp; questions</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#workflow"><i class="fa fa-check"></i><b>2.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-R_packages.html"><a href="03-R_packages.html"><i class="fa fa-check"></i><b>3</b> R Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_packages.html"><a href="03-R_packages.html#r-packageslibraries"><i class="fa fa-check"></i><b>3.1</b> R packages/libraries</a></li>
<li class="chapter" data-level="3.2" data-path="03-R_packages.html"><a href="03-R_packages.html#the-grammar-of-graphics"><i class="fa fa-check"></i><b>3.2</b> The grammar of graphics</a></li>
<li class="chapter" data-level="3.3" data-path="03-R_packages.html"><a href="03-R_packages.html#phyloseq"><i class="fa fa-check"></i><b>3.3</b> phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Setup.html"><a href="04-Setup.html"><i class="fa fa-check"></i><b>4</b> Set-up</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Setup.html"><a href="04-Setup.html#cluster"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Setup.html"><a href="04-Setup.html#mamba"><i class="fa fa-check"></i><b>4.2</b> Mamba</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Jupyter.html"><a href="05-Jupyter.html"><i class="fa fa-check"></i><b>5</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Jupyter.html"><a href="05-Jupyter.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.2" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-r-notebook"><i class="fa fa-check"></i><b>5.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.3" data-path="05-Jupyter.html"><a href="05-Jupyter.html#cells-and-code"><i class="fa fa-check"></i><b>5.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.4" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-new-cells"><i class="fa fa-check"></i><b>5.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.5" data-path="05-Jupyter.html"><a href="05-Jupyter.html#running-code"><i class="fa fa-check"></i><b>5.5</b> Running code</a></li>
<li class="chapter" data-level="5.6" data-path="05-Jupyter.html"><a href="05-Jupyter.html#saving-the-file"><i class="fa fa-check"></i><b>5.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.7" data-path="05-Jupyter.html"><a href="05-Jupyter.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.8" data-path="05-Jupyter.html"><a href="05-Jupyter.html#close-the-notebook"><i class="fa fa-check"></i><b>5.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.9" data-path="05-Jupyter.html"><a href="05-Jupyter.html#video-tutorial"><i class="fa fa-check"></i><b>5.9</b> Video tutorial</a></li>
</ul></li>
<li class="part"><span><b>Data preparation</b></span></li>
<li class="chapter" data-level="6" data-path="06-Preprocess_part.html"><a href="06-Preprocess_part.html"><i class="fa fa-check"></i><b>6</b> Data prep intro</a></li>
<li class="chapter" data-level="7" data-path="07-Import.html"><a href="07-Import.html"><i class="fa fa-check"></i><b>7</b> Import</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Import.html"><a href="07-Import.html#import-notebook"><i class="fa fa-check"></i><b>7.1</b> Import: notebook</a></li>
<li class="chapter" data-level="7.2" data-path="07-Import.html"><a href="07-Import.html#qiime2r"><i class="fa fa-check"></i><b>7.2</b> qiime2R</a></li>
<li class="chapter" data-level="7.3" data-path="07-Import.html"><a href="07-Import.html#import-summarise-phyloseq"><i class="fa fa-check"></i><b>7.3</b> Import: Summarise phyloseq</a></li>
<li class="chapter" data-level="7.4" data-path="07-Import.html"><a href="07-Import.html#save-the-phyloseq-object"><i class="fa fa-check"></i><b>7.4</b> Save the phyloseq object</a></li>
<li class="chapter" data-level="7.5" data-path="07-Import.html"><a href="07-Import.html#import-recap"><i class="fa fa-check"></i><b>7.5</b> Import: recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html"><i class="fa fa-check"></i><b>8</b> Summarise phyloseq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-setup"><i class="fa fa-check"></i><b>8.1</b> Summarise: setup</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-header"><i class="fa fa-check"></i><b>8.1.1</b> Summarise header</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-phyloseq"><i class="fa fa-check"></i><b>8.1.2</b> Summarise phyloseq</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#reads-per-sample"><i class="fa fa-check"></i><b>8.1.3</b> Reads per sample</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#display-png"><i class="fa fa-check"></i><b>8.1.4</b> Display png</a></li>
<li class="chapter" data-level="8.1.5" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#asvs-per-sample"><i class="fa fa-check"></i><b>8.1.5</b> ASVs per sample</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-recap"><i class="fa fa-check"></i><b>8.2</b> Summarise: recap</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html"><i class="fa fa-check"></i><b>9</b> Minimum read depth</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#considerations"><i class="fa fa-check"></i><b>9.1</b> Considerations</a></li>
<li class="chapter" data-level="9.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minimum-read-depth-section-title"><i class="fa fa-check"></i><b>9.2</b> Minimum read depth section title</a></li>
<li class="chapter" data-level="9.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#read-depth-vector-and-histogram"><i class="fa fa-check"></i><b>9.3</b> Read depth vector and histogram</a></li>
<li class="chapter" data-level="9.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#depth-boxplots"><i class="fa fa-check"></i><b>9.4</b> Sample depth boxplot</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#creating-a-data-frame-for-boxplots"><i class="fa fa-check"></i><b>9.4.1</b> Creating a data frame for boxplots</a></li>
<li class="chapter" data-level="9.4.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#ggplot2-boxplot"><i class="fa fa-check"></i><b>9.4.2</b> ggplot2 boxplot</a></li>
<li class="chapter" data-level="9.4.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#save-ggplot-as-png"><i class="fa fa-check"></i><b>9.4.3</b> Save ggplot as png</a></li>
<li class="chapter" data-level="9.4.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#media-by-depth-boxplot"><i class="fa fa-check"></i><b>9.4.4</b> Media by depth boxplot</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#chap9rarefaction"><i class="fa fa-check"></i><b>9.5</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#asv-abundance-data-frame"><i class="fa fa-check"></i><b>9.5.1</b> ASV abundance data frame</a></li>
<li class="chapter" data-level="9.5.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#vegans-rarecurve"><i class="fa fa-check"></i><b>9.5.2</b> vegan's rarecurve</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#filtering-by-minimum-read-depth"><i class="fa fa-check"></i><b>9.6</b> Filtering by minimum read depth</a></li>
<li class="chapter" data-level="9.7" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minmum-read-depth-summary"><i class="fa fa-check"></i><b>9.7</b> Minmum read depth: Summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy</b></span></li>
<li class="chapter" data-level="10" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html"><i class="fa fa-check"></i><b>10</b> Taxa relative abundance</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-setup"><i class="fa fa-check"></i><b>10.1</b> Taxa relative abundance: setup</a></li>
<li class="chapter" data-level="10.2" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#relative-abundance-transformation"><i class="fa fa-check"></i><b>10.2</b> Relative abundance transformation</a></li>
<li class="chapter" data-level="10.3" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#aggregate-taxa"><i class="fa fa-check"></i><b>10.3</b> Aggregate taxa</a></li>
<li class="chapter" data-level="10.4" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-summary"><i class="fa fa-check"></i><b>10.4</b> Taxa Relative abundance: summary</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html"><i class="fa fa-check"></i><b>11</b> Taxa plots</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#simple-bar-chart"><i class="fa fa-check"></i><b>11.1</b> Simple bar chart</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#editing-the-plot"><i class="fa fa-check"></i><b>11.1.1</b> Editing the plot</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-heatmap"><i class="fa fa-check"></i><b>11.2</b> Taxa heatmap</a></li>
<li class="chapter" data-level="11.3" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#aggregateraretaxa"><i class="fa fa-check"></i><b>11.3</b> Aggregate rare taxa</a></li>
<li class="chapter" data-level="11.4" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-media"><i class="fa fa-check"></i><b>11.4</b> Plot by media</a></li>
<li class="chapter" data-level="11.5" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-site-and-media"><i class="fa fa-check"></i><b>11.5</b> Plot by site and media</a></li>
<li class="chapter" data-level="11.6" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-plots-summary"><i class="fa fa-check"></i><b>11.6</b> Taxa plots: summary</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html"><i class="fa fa-check"></i><b>12</b> Family &amp; Genus plots</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-taxa-plots"><i class="fa fa-check"></i><b>12.1</b> Family taxa plots</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#aggregate-families"><i class="fa fa-check"></i><b>12.1.1</b> Aggregate families</a></li>
<li class="chapter" data-level="12.1.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-heatmap"><i class="fa fa-check"></i><b>12.1.2</b> Family heatmap</a></li>
<li class="chapter" data-level="12.1.3" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-rare-aggregation"><i class="fa fa-check"></i><b>12.1.3</b> Family rare aggregation</a></li>
<li class="chapter" data-level="12.1.4" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-bar-chart"><i class="fa fa-check"></i><b>12.1.4</b> Family bar chart</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots"><i class="fa fa-check"></i><b>12.2</b> Genus taxa plots</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots-solutions"><i class="fa fa-check"></i><b>12.2.1</b> Genus taxa plots solutions</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Diversity analysis</b></span></li>
<li class="chapter" data-level="13" data-path="13-Diversity_analysis.html"><a href="13-Diversity_analysis.html"><i class="fa fa-check"></i><b>13</b> Diversity analysis intro</a></li>
<li class="chapter" data-level="14" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html"><i class="fa fa-check"></i><b>14</b> Rarefaction</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-setup"><i class="fa fa-check"></i><b>14.1</b> Rarefaction: setup</a></li>
<li class="chapter" data-level="14.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve"><i class="fa fa-check"></i><b>14.2</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#bird-analogy"><i class="fa fa-check"></i><b>14.2.1</b> Bird analogy</a></li>
<li class="chapter" data-level="14.2.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-simple-plot"><i class="fa fa-check"></i><b>14.2.2</b> Rarefaction curve: simple plot</a></li>
<li class="chapter" data-level="14.2.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-better-plot"><i class="fa fa-check"></i><b>14.2.3</b> Rarefaction curve: better plot</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-slope"><i class="fa fa-check"></i><b>14.3</b> Rarefaction slope</a></li>
<li class="chapter" data-level="14.4" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefy-data"><i class="fa fa-check"></i><b>14.4</b> Rarefy data</a></li>
<li class="chapter" data-level="14.5" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-summary"><i class="fa fa-check"></i><b>14.5</b> Rarefaction: summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Alpha.html"><a href="15-Alpha.html"><i class="fa fa-check"></i><b>15</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-setup"><i class="fa fa-check"></i><b>15.1</b> Alpha: setup</a></li>
<li class="chapter" data-level="15.2" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-simple-plot"><i class="fa fa-check"></i><b>15.2</b> Alpha: simple plot</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="15-Alpha.html"><a href="15-Alpha.html#metric-choice"><i class="fa fa-check"></i><b>15.2.1</b> Metric choice</a></li>
<li class="chapter" data-level="15.2.2" data-path="15-Alpha.html"><a href="15-Alpha.html#sample-grouping-box-plot"><i class="fa fa-check"></i><b>15.2.2</b> Sample grouping &amp; box plot</a></li>
<li class="chapter" data-level="15.2.3" data-path="15-Alpha.html"><a href="15-Alpha.html#png-height-and-width-choice"><i class="fa fa-check"></i><b>15.2.3</b> PNG height and width choice</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats"><i class="fa fa-check"></i><b>15.3</b> Alpha: stats</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats-task"><i class="fa fa-check"></i><b>15.3.1</b> Alpha stats task</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-diversity-violin-plot"><i class="fa fa-check"></i><b>15.4</b> Alpha diversity: violin plot</a></li>
<li class="chapter" data-level="15.5" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-site-plots-and-statistics"><i class="fa fa-check"></i><b>15.5</b> Alpha: Site plots and statistics</a></li>
<li class="chapter" data-level="15.6" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-summary"><i class="fa fa-check"></i><b>15.6</b> Alpha: summary</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Beta.html"><a href="16-Beta.html"><i class="fa fa-check"></i><b>16</b> Beta</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Beta.html"><a href="16-Beta.html#beta-setup"><i class="fa fa-check"></i><b>16.1</b> Beta: setup</a></li>
<li class="chapter" data-level="16.2" data-path="16-Beta.html"><a href="16-Beta.html#beta-ordination"><i class="fa fa-check"></i><b>16.2</b> Beta: ordination</a></li>
<li class="chapter" data-level="16.3" data-path="16-Beta.html"><a href="16-Beta.html#beta-plot-ordination"><i class="fa fa-check"></i><b>16.3</b> Beta: plot ordination</a></li>
<li class="chapter" data-level="16.4" data-path="16-Beta.html"><a href="16-Beta.html#beta-statistics"><i class="fa fa-check"></i><b>16.4</b> Beta: statistics</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="16-Beta.html"><a href="16-Beta.html#permanova"><i class="fa fa-check"></i><b>16.4.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="16.4.2" data-path="16-Beta.html"><a href="16-Beta.html#pairwise-permanova"><i class="fa fa-check"></i><b>16.4.2</b> Pairwise PERMANOVA</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="16-Beta.html"><a href="16-Beta.html#beta-facet-zoom"><i class="fa fa-check"></i><b>16.5</b> Beta: Facet zoom</a></li>
<li class="chapter" data-level="16.6" data-path="16-Beta.html"><a href="16-Beta.html#beta-subset-samples"><i class="fa fa-check"></i><b>16.6</b> Beta: subset samples</a></li>
<li class="chapter" data-level="16.7" data-path="16-Beta.html"><a href="16-Beta.html#permanova-tasks"><i class="fa fa-check"></i><b>16.7</b> PERMANOVA tasks</a></li>
<li class="chapter" data-level="16.8" data-path="16-Beta.html"><a href="16-Beta.html#beta-summary"><i class="fa fa-check"></i><b>16.8</b> Beta: Summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html"><i class="fa fa-check"></i><b>17</b> Differential abundance (DA) analysis</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-setup"><i class="fa fa-check"></i><b>17.1</b> DA: setup</a></li>
<li class="chapter" data-level="17.2" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-preprocess"><i class="fa fa-check"></i><b>17.2</b> DA: Preprocess</a></li>
<li class="chapter" data-level="17.3" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-analysis"><i class="fa fa-check"></i><b>17.3</b> DA: ANCOM-BC analysis</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#ancombc2-options"><i class="fa fa-check"></i><b>17.3.1</b> ANCOMBC2 options</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-results"><i class="fa fa-check"></i><b>17.4</b> DA: ANCOM-BC results</a></li>
<li class="chapter" data-level="17.5" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-create-signifcant-lfc-long-data-frame"><i class="fa fa-check"></i><b>17.5</b> DA: Create signifcant LFC long data frame</a></li>
<li class="chapter" data-level="17.6" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-biomarker-lfc-heatmap"><i class="fa fa-check"></i><b>17.6</b> DA: Biomarker LFC heatmap</a></li>
<li class="chapter" data-level="17.7" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-matching-bar-plot"><i class="fa fa-check"></i><b>17.7</b> DA: matching bar plot</a></li>
<li class="chapter" data-level="17.8" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-summary"><i class="fa fa-check"></i><b>17.8</b> DA: summary</a></li>
</ul></li>
<li class="part"><span><b>Outro</b></span></li>
<li class="chapter" data-level="18" data-path="18-Summary.html"><a href="18-Summary.html"><i class="fa fa-check"></i><b>18</b> Summary</a>
<ul>
<li class="chapter" data-level="18.1" data-path="18-Summary.html"><a href="18-Summary.html#dataset-1"><i class="fa fa-check"></i><b>18.1</b> Dataset</a></li>
<li class="chapter" data-level="18.2" data-path="18-Summary.html"><a href="18-Summary.html#data-preparation"><i class="fa fa-check"></i><b>18.2</b> Data preparation</a></li>
<li class="chapter" data-level="18.3" data-path="18-Summary.html"><a href="18-Summary.html#taxonomy"><i class="fa fa-check"></i><b>18.3</b> Taxonomy</a></li>
<li class="chapter" data-level="18.4" data-path="18-Summary.html"><a href="18-Summary.html#diversity-analysis"><i class="fa fa-check"></i><b>18.4</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="18-Summary.html"><a href="18-Summary.html#alpha-beta"><i class="fa fa-check"></i><b>18.4.1</b> Alpha &amp; beta</a></li>
<li class="chapter" data-level="18.4.2" data-path="18-Summary.html"><a href="18-Summary.html#differential-abundance-anlaysis"><i class="fa fa-check"></i><b>18.4.2</b> Differential abundance anlaysis</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="18-Summary.html"><a href="18-Summary.html#findings"><i class="fa fa-check"></i><b>18.5</b> Findings</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="18-Summary.html"><a href="18-Summary.html#anthropisation-gradient"><i class="fa fa-check"></i><b>18.5.1</b> Anthropisation gradient</a></li>
<li class="chapter" data-level="18.5.2" data-path="18-Summary.html"><a href="18-Summary.html#replicate-differences"><i class="fa fa-check"></i><b>18.5.2</b> Replicate differences</a></li>
<li class="chapter" data-level="18.5.3" data-path="18-Summary.html"><a href="18-Summary.html#site-or-media"><i class="fa fa-check"></i><b>18.5.3</b> Site or media</a></li>
<li class="chapter" data-level="18.5.4" data-path="18-Summary.html"><a href="18-Summary.html#media-versus-env-samples"><i class="fa fa-check"></i><b>18.5.4</b> Media versus ENV samples</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="18-Summary.html"><a href="18-Summary.html#conclusion"><i class="fa fa-check"></i><b>18.6</b> Conclusion</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="19-Appendix.html"><a href="19-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="19-Appendix.html"><a href="19-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
<li class="chapter" data-level="A.2" data-path="19-Appendix.html"><a href="19-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>A.2</b> Jupyter-notebook</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="19-Appendix.html"><a href="19-Appendix.html#r-packages-1"><i class="fa fa-check"></i><b>B</b> R packages</a>
<ul>
<li class="chapter" data-level="B.1" data-path="19-Appendix.html"><a href="19-Appendix.html#ggplot2_appendix"><i class="fa fa-check"></i><b>B.1</b> ggplot2</a></li>
<li class="chapter" data-level="B.2" data-path="19-Appendix.html"><a href="19-Appendix.html#phyloseq_appendix"><i class="fa fa-check"></i><b>B.2</b> Phyloseq</a></li>
<li class="chapter" data-level="B.3" data-path="19-Appendix.html"><a href="19-Appendix.html#importing-into-phyloseq"><i class="fa fa-check"></i><b>B.3</b> Importing into phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-metrics"><i class="fa fa-check"></i><b>C</b> Diversity metrics</a>
<ul>
<li class="chapter" data-level="C.1" data-path="19-Appendix.html"><a href="19-Appendix.html#alpha_appendix"><i class="fa fa-check"></i><b>C.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="19-Appendix.html"><a href="19-Appendix.html#obvs"><i class="fa fa-check"></i><b>C.1.1</b> Observed features</a></li>
<li class="chapter" data-level="C.1.2" data-path="19-Appendix.html"><a href="19-Appendix.html#chao"><i class="fa fa-check"></i><b>C.1.2</b> Chao</a></li>
<li class="chapter" data-level="C.1.3" data-path="19-Appendix.html"><a href="19-Appendix.html#evenness"><i class="fa fa-check"></i><b>C.1.3</b> Evenness</a></li>
<li class="chapter" data-level="C.1.4" data-path="19-Appendix.html"><a href="19-Appendix.html#faith"><i class="fa fa-check"></i><b>C.1.4</b> Faith's PD (phylogenetic diversity)</a></li>
<li class="chapter" data-level="C.1.5" data-path="19-Appendix.html"><a href="19-Appendix.html#simpson"><i class="fa fa-check"></i><b>C.1.5</b> Simpson</a></li>
<li class="chapter" data-level="C.1.6" data-path="19-Appendix.html"><a href="19-Appendix.html#simpsone"><i class="fa fa-check"></i><b>C.1.6</b> Simpson evenness measure (simpson_e)</a></li>
<li class="chapter" data-level="C.1.7" data-path="19-Appendix.html"><a href="19-Appendix.html#shannon"><i class="fa fa-check"></i><b>C.1.7</b> Shannon</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="19-Appendix.html"><a href="19-Appendix.html#beta_appendix"><i class="fa fa-check"></i><b>C.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="19-Appendix.html"><a href="19-Appendix.html#unifrac_explanation"><i class="fa fa-check"></i><b>C.2.1</b> Weighted and unweighted UniFrac distances</a></li>
<li class="chapter" data-level="C.2.2" data-path="19-Appendix.html"><a href="19-Appendix.html#bray_explanation"><i class="fa fa-check"></i><b>C.2.2</b> Bray-Curtis</a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-resources"><i class="fa fa-check"></i><b>C.3</b> Diversity Resources</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R community analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="alpha_chap" class="section level1 hasAnchor" number="15">
<h1><span class="header-section-number">Chapter 15</span> Alpha diversity<a href="15-Alpha.html#alpha_chap" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/alpha.png" width="20%" style="display: block; margin: auto;" /></p>
<p>In this chapter we are going to produce alpha diversity plots and statistics with our rarefied data.
Alpha diversity is carried out by producing a metric for each sample.
These metrics are primarily interested in the evenness or diversity of a sample.</p>
<p>Generally, these metrics are carried out on the ASV data to see the most fine grain differences between samples.
However, you can also use different taxa levels which we will show later on.
This can be useful if there is too much difference in the ASVs from sample to sample or you can only accurately classify sequences to a high level taxonomy (such as with <code>KRAKEN2</code> output).</p>
<p>Once we have a value for each sample they can be used to compare samples and sample groups.
Box plots are used for visualisation.
Statistics are used to determine if there is a significant difference between groups.</p>
<p>We'll start out by producing a basic box plot and statistics.
Next we'll improve the plot with some extra R packages.
With these tools we will compare the metrics across the various metadata fields we have.</p>
<div id="alpha-setup" class="section level2 hasAnchor" number="15.1">
<h2><span class="header-section-number">15.1</span> Alpha: setup<a href="15-Alpha.html#alpha-setup" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/gear_setup.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We will use a new notebook called <strong>"5-alpha_diversity.ipynb"</strong>.
Add the following to the top of this new notebook to load the required libraries/packages and data.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="15-Alpha.html#cb74-1" tabindex="-1"></a><span class="co">#Libraries</span></span>
<span id="cb74-2"><a href="15-Alpha.html#cb74-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb74-3"><a href="15-Alpha.html#cb74-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;microbiome&quot;</span>)</span>
<span id="cb74-4"><a href="15-Alpha.html#cb74-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;IRdisplay&quot;</span>)</span>
<span id="cb74-5"><a href="15-Alpha.html#cb74-5" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggforce&quot;</span>)</span>
<span id="cb74-6"><a href="15-Alpha.html#cb74-6" tabindex="-1"></a><span class="co">#Load phyloseq object</span></span>
<span id="cb74-7"><a href="15-Alpha.html#cb74-7" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;phyloseq_rarefied.RData&quot;</span>)</span></code></pre></div>
<p>Reminder to use markdown and code cells in the notebook.</p>
</div>
<div id="alpha-simple-plot" class="section level2 hasAnchor" number="15.2">
<h2><span class="header-section-number">15.2</span> Alpha: simple plot<a href="15-Alpha.html#alpha-simple-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/simple_boxplot.png" width="15%" style="display: block; margin: auto;" /></p>
<p>The first step is to calculate and plot the alpha diversity metrics based on the ASV data.
Thankfully, the command <code>phyloseq::plot_richness()</code> will do both of these.
Try it out.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="15-Alpha.html#cb75-1" tabindex="-1"></a><span class="co">#Simple alpha plot</span></span>
<span id="cb75-2"><a href="15-Alpha.html#cb75-2" tabindex="-1"></a>alpha_plot <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_richness</span>(<span class="at">physeq =</span> pseq_rarefy)</span>
<span id="cb75-3"><a href="15-Alpha.html#cb75-3" tabindex="-1"></a><span class="co">#Save ggplot object as png file</span></span>
<span id="cb75-4"><a href="15-Alpha.html#cb75-4" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./alpha_div.png&quot;</span>, <span class="at">plot =</span> alpha_plot,</span>
<span id="cb75-5"><a href="15-Alpha.html#cb75-5" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">200</span>, <span class="at">width =</span> <span class="dv">300</span>)</span>
<span id="cb75-6"><a href="15-Alpha.html#cb75-6" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb75-7"><a href="15-Alpha.html#cb75-7" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./alpha_div.png&quot;</span>)</span></code></pre></div>
<p>The resulting plot is pretty good but is difficult to read.
We can do 3 major things to make the contents easier to read.</p>
<div id="metric-choice" class="section level3 hasAnchor" number="15.2.1">
<h3><span class="header-section-number">15.2.1</span> Metric choice<a href="15-Alpha.html#metric-choice" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/metric_square_1.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We can choose certain alpha diversity metrics to plot.
Currently it is showing 7 metrics (Observed, Chao1 etc.).
I like to use:</p>
<ul>
<li><strong>Observed</strong>: This is the number of observed features (ASVs, Genera, etc) in each sample.</li>
<li><strong>Chao1</strong>: This is an estimate number of the total number of features in each sample.</li>
<li><strong>Shannon</strong>: A measure of diversity where a higher number means higher diversity. Shannon's index accounts for both abundance and evenness of the features present.</li>
</ul>
<p>We'll choose these metrics with the <code>measures =</code> option.
Either make a new code block with new code or you can edit and run the previous code you wrote.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="15-Alpha.html#cb76-1" tabindex="-1"></a><span class="co">#Simple alpha plot</span></span>
<span id="cb76-2"><a href="15-Alpha.html#cb76-2" tabindex="-1"></a>alpha_plot <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_richness</span>(<span class="at">physeq =</span> pseq_rarefy, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Observed&quot;</span>,<span class="st">&quot;Chao1&quot;</span>,<span class="st">&quot;Shannon&quot;</span>))</span>
<span id="cb76-3"><a href="15-Alpha.html#cb76-3" tabindex="-1"></a><span class="co">#Save ggplot object as png file</span></span>
<span id="cb76-4"><a href="15-Alpha.html#cb76-4" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./alpha_div.png&quot;</span>, <span class="at">plot =</span> alpha_plot,</span>
<span id="cb76-5"><a href="15-Alpha.html#cb76-5" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">200</span>, <span class="at">width =</span> <span class="dv">300</span>)</span>
<span id="cb76-6"><a href="15-Alpha.html#cb76-6" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb76-7"><a href="15-Alpha.html#cb76-7" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./alpha_div.png&quot;</span>)</span></code></pre></div>
<p>To see which measures are available you can see the help page of <code>phyloseq::plot_richness()</code>.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="15-Alpha.html#cb77-1" tabindex="-1"></a>?phyloseq<span class="sc">::</span>plot_richness</span></code></pre></div>
<p>For a list and full description of some of the metrics please see the <a href="19-Appendix.html#alpha_appendix">APPENDIX</a>.</p>
</div>
<div id="sample-grouping-box-plot" class="section level3 hasAnchor" number="15.2.2">
<h3><span class="header-section-number">15.2.2</span> Sample grouping &amp; box plot<a href="15-Alpha.html#sample-grouping-box-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/boxplots_groups.png" width="15%" style="display: block; margin: auto;" /></p>
<p>The last plot looks a bit better.
To actually make them box plots we'll change the x axis so they are based on media rather than individual samples.</p>
<p>The <code>phyloseq::plot_richness()</code> function produces a <code>ggplot2</code> object.
To convert our dot plot into a box plot we can add the layer <code>geom_boxplt()</code> like with our <a href="09-Minimum_read_depth.html#depth-boxplots">depth boxplots</a>.
This will combine samples from one media into one IQR box.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="15-Alpha.html#cb78-1" tabindex="-1"></a><span class="co">#Box plot</span></span>
<span id="cb78-2"><a href="15-Alpha.html#cb78-2" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">plot_richness</span>(<span class="at">physeq =</span> pseq_rarefy, </span>
<span id="cb78-3"><a href="15-Alpha.html#cb78-3" tabindex="-1"></a>                        <span class="at">x =</span> <span class="st">&quot;media&quot;</span>,</span>
<span id="cb78-4"><a href="15-Alpha.html#cb78-4" tabindex="-1"></a>                        <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Observed&quot;</span>,<span class="st">&quot;Chao1&quot;</span>,<span class="st">&quot;Shannon&quot;</span>)) <span class="sc">+</span></span>
<span id="cb78-5"><a href="15-Alpha.html#cb78-5" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>()</span>
<span id="cb78-6"><a href="15-Alpha.html#cb78-6" tabindex="-1"></a><span class="co">#Save ggplot object as png file</span></span>
<span id="cb78-7"><a href="15-Alpha.html#cb78-7" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./alpha_div.png&quot;</span>, <span class="at">plot =</span> alpha_plot,</span>
<span id="cb78-8"><a href="15-Alpha.html#cb78-8" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">200</span>, <span class="at">width =</span> <span class="dv">300</span>)</span>
<span id="cb78-9"><a href="15-Alpha.html#cb78-9" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb78-10"><a href="15-Alpha.html#cb78-10" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./alpha_div.png&quot;</span>)</span></code></pre></div>
</div>
<div id="png-height-and-width-choice" class="section level3 hasAnchor" number="15.2.3">
<h3><span class="header-section-number">15.2.3</span> PNG height and width choice<a href="15-Alpha.html#png-height-and-width-choice" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/display_png.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Choosing a good height and width size can make plots easier to read.
We'll try 150 (height) and 250 (width).</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="15-Alpha.html#cb79-1" tabindex="-1"></a><span class="co"># Produce ggplot object of boxplot</span></span>
<span id="cb79-2"><a href="15-Alpha.html#cb79-2" tabindex="-1"></a>alpha_boxplot <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_richness</span>(<span class="at">physeq =</span> pseq_rarefy, </span>
<span id="cb79-3"><a href="15-Alpha.html#cb79-3" tabindex="-1"></a>                        <span class="at">x =</span> <span class="st">&quot;media&quot;</span>,</span>
<span id="cb79-4"><a href="15-Alpha.html#cb79-4" tabindex="-1"></a>                        <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Observed&quot;</span>,<span class="st">&quot;Chao1&quot;</span>,<span class="st">&quot;Shannon&quot;</span>)) <span class="sc">+</span></span>
<span id="cb79-5"><a href="15-Alpha.html#cb79-5" tabindex="-1"></a>                          ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>()</span>
<span id="cb79-6"><a href="15-Alpha.html#cb79-6" tabindex="-1"></a><span class="co">#Save ggplot2 object with ggsave</span></span>
<span id="cb79-7"><a href="15-Alpha.html#cb79-7" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./Alpha_diversity_media_boxplot.png&quot;</span>, <span class="at">plot =</span> alpha_boxplot,</span>
<span id="cb79-8"><a href="15-Alpha.html#cb79-8" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">150</span>, <span class="at">width =</span> <span class="dv">250</span>)</span>
<span id="cb79-9"><a href="15-Alpha.html#cb79-9" tabindex="-1"></a><span class="co">#Display plot</span></span>
<span id="cb79-10"><a href="15-Alpha.html#cb79-10" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./Alpha_diversity_media_boxplot.png&quot;</span>)</span></code></pre></div>
<p>How do I choose a good height and width?
This comes down to trial and error and experience.
You want something with a good aspect ratio for the plot and you want all the text to be clear.
Additionally, the size requirements of figures for journals will play a big consideration as well.
One of the nice things with creating jupyter-notebooks or R markdowns with all your code is that you can go back, edit, and rerun the code relatively easily and quickly.</p>
</div>
</div>
<div id="alpha-stats" class="section level2 hasAnchor" number="15.3">
<h2><span class="header-section-number">15.3</span> Alpha: stats<a href="15-Alpha.html#alpha-stats" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/pear.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Along with the plot it is also good to carry out statistical tests.
This will allow us to statistically determine if the alpha diversity values between groups are significantly different.</p>
<p>Before carrying out statistics we will need a data frame with the alpha diveristy values for each sample.
This can be carried out with the function<code>phyloseq::estimate_richness()</code>.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="15-Alpha.html#cb80-1" tabindex="-1"></a><span class="co">#Produce data frame of all alpha diversity values</span></span>
<span id="cb80-2"><a href="15-Alpha.html#cb80-2" tabindex="-1"></a>alpha_df <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">estimate_richness</span>(<span class="at">physeq =</span> pseq_rarefy)</span>
<span id="cb80-3"><a href="15-Alpha.html#cb80-3" tabindex="-1"></a><span class="fu">head</span>(alpha_df)</span></code></pre></div>
<p>We have all the alpha metrics in the new data frame.
Additionally, we also have the standard error for some of the metrics (e.g. <code>se.chao1</code>).</p>
<p>With this information we can carry out a pairwise comparison using Wilcoxon rank sum test (the same way as QIIME2).
Base R comes with the function <code>pairwsie.wilcox.test()</code> to carry this out.
We'll do this using media as our groups from the <code>sample_data()</code> and the Observed metric from the alpha_df.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="15-Alpha.html#cb81-1" tabindex="-1"></a><span class="co">#Paired wilcoxon test</span></span>
<span id="cb81-2"><a href="15-Alpha.html#cb81-2" tabindex="-1"></a><span class="co">#Observed</span></span>
<span id="cb81-3"><a href="15-Alpha.html#cb81-3" tabindex="-1"></a><span class="fu">pairwise.wilcox.test</span>(alpha_df<span class="sc">$</span>Observed, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq_rarefy)<span class="sc">$</span>media)</span></code></pre></div>
<p>You can ignore the warning messages in this case.</p>
<p>Observe the results.</p>
Are all the pairwise comparisons significant different (&lt;0.05)?
<div id="radio_YTNWHPRBCP" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_YTNWHPRBCP" value="answer"></input> <span><strong>Yes</strong></span></label><label><input type="radio" autocomplete="off" name="radio_YTNWHPRBCP" value=""></input> <span><strong>No</strong></span></label>
</div>
<p>The test has carried out P value adjustments using the Holm-Bonferroni method (<code>P value adjustment method: holm</code>) which is good.</p>
<p>In this case the p-values are all identical as the values do not vary greatly but there are clear differences.
The only group with a large variance is the ENV group with values going from &lt; 150 to &gt; 450.</p>
<p>For more info on Wilcoxon test <a href="https://en.wikipedia.org/wiki/Wilcoxon_signed-rank_test">Wikipedia</a> is a good start.</p>
<div id="alpha-stats-task" class="section level3 hasAnchor" number="15.3.1">
<h3><span class="header-section-number">15.3.1</span> Alpha stats task<a href="15-Alpha.html#alpha-stats-task" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/exercises.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Carry out the same statistical analysis (media as the grouping) with the <code>Chao1</code> and <code>Shannon</code> metrics.</p>
<div class="webex-solution">
<button>
Alpha stats solution
</button>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="15-Alpha.html#cb82-1" tabindex="-1"></a><span class="co">#Chao1</span></span>
<span id="cb82-2"><a href="15-Alpha.html#cb82-2" tabindex="-1"></a><span class="fu">pairwise.wilcox.test</span>(alpha_df<span class="sc">$</span>Chao1, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq_rarefy)<span class="sc">$</span>media)</span>
<span id="cb82-3"><a href="15-Alpha.html#cb82-3" tabindex="-1"></a><span class="co">#Shannon</span></span>
<span id="cb82-4"><a href="15-Alpha.html#cb82-4" tabindex="-1"></a><span class="fu">pairwise.wilcox.test</span>(alpha_df<span class="sc">$</span>Shannon, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq_rarefy)<span class="sc">$</span>media)</span></code></pre></div>
</div>
<p>With these we can see that the differences between the Chao1 values of the different media are significantly different.</p>
<p>However, the p-values when comparing the Shannon values are not all significant.
The only significant differences are between ENV &amp; KBC, and between KBC &amp; TSA.
Going back to the boxplot, does this look like the case?
I would say yes as the IQRs for CVP, KBC, and TSA all overlap.
There is little to no overlap of the IQRs of KBC and TSA against ENV.</p>
<p>We can therefore make these claims about the alpha diversity values of the groups.</p>
<ul>
<li>ENV has the highest number of observed ASVs and the highest diversity.</li>
<li>ENV appears to have a wider range of diversities across its samples.</li>
<li>The medias have much lower numbers of ASVs and diveristy compared to ENV.</li>
<li>Within each media grouping the samples have very similar numbers of ASVs present.</li>
</ul>
</div>
</div>
<div id="alpha-diversity-violin-plot" class="section level2 hasAnchor" number="15.4">
<h2><span class="header-section-number">15.4</span> Alpha diversity: violin plot<a href="15-Alpha.html#alpha-diversity-violin-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/violin.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Boxplots are quite nice but violin plots can be even nicer.
They show the distribution of the data in the IQR better and in fact you can easily display each value.</p>
<p>We can change our boxplot to a violin plot by changing <code>ggplot2:geom_boxplot()</code> to <code>ggplot2::geom_violin()</code>.</p>
<p>Copy and paste the previous boxplot code to a new cell at the bottom of your notebook.
Edit and run the code so it looks like the below.</p>
<p><strong>Tip:</strong> Remember to change the name of your png file in the <code>ggsave()</code> and <code>IRdisplay::display_png()</code> functions.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="15-Alpha.html#cb83-1" tabindex="-1"></a><span class="co"># Produce ggplot object of violin plot</span></span>
<span id="cb83-2"><a href="15-Alpha.html#cb83-2" tabindex="-1"></a>alpha_violinplot <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_richness</span>(<span class="at">physeq =</span> pseq_rarefy, </span>
<span id="cb83-3"><a href="15-Alpha.html#cb83-3" tabindex="-1"></a>                        <span class="at">x =</span> <span class="st">&quot;media&quot;</span>,</span>
<span id="cb83-4"><a href="15-Alpha.html#cb83-4" tabindex="-1"></a>                        <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Observed&quot;</span>,<span class="st">&quot;Chao1&quot;</span>,<span class="st">&quot;Shannon&quot;</span>)) <span class="sc">+</span></span>
<span id="cb83-5"><a href="15-Alpha.html#cb83-5" tabindex="-1"></a>                          ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>()</span>
<span id="cb83-6"><a href="15-Alpha.html#cb83-6" tabindex="-1"></a><span class="co">#Save ggplot2 object with ggsave</span></span>
<span id="cb83-7"><a href="15-Alpha.html#cb83-7" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./Alpha_diversity_media_violinplot.png&quot;</span>, <span class="at">plot =</span> alpha_violinplot,</span>
<span id="cb83-8"><a href="15-Alpha.html#cb83-8" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">150</span>, <span class="at">width =</span> <span class="dv">250</span>)</span>
<span id="cb83-9"><a href="15-Alpha.html#cb83-9" tabindex="-1"></a><span class="co">#Display plot</span></span>
<span id="cb83-10"><a href="15-Alpha.html#cb83-10" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./Alpha_diversity_media_violinplot.png&quot;</span>)</span></code></pre></div>
<p>We can also add semi transparent dots representing the values for each sample.
To do this, add the <code>ggplot2</code> layer <code>ggforce::geom_sina()</code>.
This has the option <code>alpha=0.5</code> where alpha resents the transparency of the dots and 0.5 represents 50% transparency.
Alpha is a common option used for many plots.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="15-Alpha.html#cb84-1" tabindex="-1"></a><span class="co"># Produce ggplot object of violin plot</span></span>
<span id="cb84-2"><a href="15-Alpha.html#cb84-2" tabindex="-1"></a>alpha_violinplot <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_richness</span>(<span class="at">physeq =</span> pseq_rarefy, </span>
<span id="cb84-3"><a href="15-Alpha.html#cb84-3" tabindex="-1"></a>                        <span class="at">x =</span> <span class="st">&quot;media&quot;</span>,</span>
<span id="cb84-4"><a href="15-Alpha.html#cb84-4" tabindex="-1"></a>                        <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Observed&quot;</span>,<span class="st">&quot;Chao1&quot;</span>,<span class="st">&quot;Shannon&quot;</span>)) <span class="sc">+</span></span>
<span id="cb84-5"><a href="15-Alpha.html#cb84-5" tabindex="-1"></a>                          ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>() <span class="sc">+</span></span>
<span id="cb84-6"><a href="15-Alpha.html#cb84-6" tabindex="-1"></a>                          ggforce<span class="sc">::</span><span class="fu">geom_sina</span>(<span class="at">alpha=</span><span class="fl">0.5</span>)</span>
<span id="cb84-7"><a href="15-Alpha.html#cb84-7" tabindex="-1"></a><span class="co">#Save ggplot2 object with ggsave</span></span>
<span id="cb84-8"><a href="15-Alpha.html#cb84-8" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./Alpha_diversity_media_violinplot.png&quot;</span>, <span class="at">plot =</span> alpha_violinplot,</span>
<span id="cb84-9"><a href="15-Alpha.html#cb84-9" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">150</span>, <span class="at">width =</span> <span class="dv">250</span>)</span>
<span id="cb84-10"><a href="15-Alpha.html#cb84-10" tabindex="-1"></a><span class="co">#Display plot</span></span>
<span id="cb84-11"><a href="15-Alpha.html#cb84-11" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./Alpha_diversity_media_violinplot.png&quot;</span>)</span></code></pre></div>
</div>
<div id="alpha-site-plots-and-statistics" class="section level2 hasAnchor" number="15.5">
<h2><span class="header-section-number">15.5</span> Alpha: Site plots and statistics<a href="15-Alpha.html#alpha-site-plots-and-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/river.png" width="15%" style="display: block; margin: auto;" /></p>
<p>With all the knowledge, skills, and code from this chapter carry out the following tasks:</p>
<ul>
<li>Produce a violin plot of Observed, Chao1, Shannon, and Inverse Simpson with the Site groups.</li>
<li>Carry out paired Wilcoxon test for Observed, Chao1, Shannon, and Inverse Simpson with the Site groups.</li>
</ul>
<div class="webex-solution">
<button>
Alpha sites solution
</button>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="15-Alpha.html#cb85-1" tabindex="-1"></a><span class="co"># Produce ggplot object of violin plot</span></span>
<span id="cb85-2"><a href="15-Alpha.html#cb85-2" tabindex="-1"></a>alpha_boxplot <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_richness</span>(<span class="at">physeq =</span> pseq_rarefy, </span>
<span id="cb85-3"><a href="15-Alpha.html#cb85-3" tabindex="-1"></a>                        <span class="at">x =</span> <span class="st">&quot;site&quot;</span>,</span>
<span id="cb85-4"><a href="15-Alpha.html#cb85-4" tabindex="-1"></a>                        <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Observed&quot;</span>,<span class="st">&quot;Chao1&quot;</span>,<span class="st">&quot;Shannon&quot;</span>)) <span class="sc">+</span></span>
<span id="cb85-5"><a href="15-Alpha.html#cb85-5" tabindex="-1"></a>                          ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb85-6"><a href="15-Alpha.html#cb85-6" tabindex="-1"></a>                          ggforce<span class="sc">::</span><span class="fu">geom_sina</span>(<span class="at">alpha=</span><span class="fl">0.5</span>)</span>
<span id="cb85-7"><a href="15-Alpha.html#cb85-7" tabindex="-1"></a><span class="co">#Save ggplot2 object with ggsave</span></span>
<span id="cb85-8"><a href="15-Alpha.html#cb85-8" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./Alpha_diversity_site_violinplot.png&quot;</span>, <span class="at">plot =</span> alpha_boxplot,</span>
<span id="cb85-9"><a href="15-Alpha.html#cb85-9" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">150</span>, <span class="at">width =</span> <span class="dv">250</span>)</span>
<span id="cb85-10"><a href="15-Alpha.html#cb85-10" tabindex="-1"></a><span class="co">#Display plot</span></span>
<span id="cb85-11"><a href="15-Alpha.html#cb85-11" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./Alpha_diversity_site_violinplot.png&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="15-Alpha.html#cb86-1" tabindex="-1"></a><span class="co">#Paired wilcoxon test</span></span>
<span id="cb86-2"><a href="15-Alpha.html#cb86-2" tabindex="-1"></a><span class="co">#Observed</span></span>
<span id="cb86-3"><a href="15-Alpha.html#cb86-3" tabindex="-1"></a><span class="fu">pairwise.wilcox.test</span>(alpha_df<span class="sc">$</span>Observed, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq_rarefy)<span class="sc">$</span>site)</span>
<span id="cb86-4"><a href="15-Alpha.html#cb86-4" tabindex="-1"></a><span class="co">#Chao1</span></span>
<span id="cb86-5"><a href="15-Alpha.html#cb86-5" tabindex="-1"></a><span class="fu">pairwise.wilcox.test</span>(alpha_df<span class="sc">$</span>Chao1, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq_rarefy)<span class="sc">$</span>site)</span>
<span id="cb86-6"><a href="15-Alpha.html#cb86-6" tabindex="-1"></a><span class="co">#Shannon</span></span>
<span id="cb86-7"><a href="15-Alpha.html#cb86-7" tabindex="-1"></a><span class="fu">pairwise.wilcox.test</span>(alpha_df<span class="sc">$</span>Shannon, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq_rarefy)<span class="sc">$</span>site)</span></code></pre></div>
</div>
<p>The plots and stats do not show anything particularly interesting in terms of site unfortunately.
There is too much difference caused by the media.
To view this we can add the option <code>aes(colour=media)</code> to <code>ggforce_sina()</code>.
This will colour the points by media.</p>
<p>If we only do this it will give us a legend with the title "colour".
We can fix this by adding the layer `ggplot2::labs(colour = "Media").</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="15-Alpha.html#cb87-1" tabindex="-1"></a><span class="co"># Produce ggplot object of violin plot</span></span>
<span id="cb87-2"><a href="15-Alpha.html#cb87-2" tabindex="-1"></a>alpha_violinplot <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_richness</span>(<span class="at">physeq =</span> pseq_rarefy, </span>
<span id="cb87-3"><a href="15-Alpha.html#cb87-3" tabindex="-1"></a>                        <span class="at">x =</span> <span class="st">&quot;site&quot;</span>,</span>
<span id="cb87-4"><a href="15-Alpha.html#cb87-4" tabindex="-1"></a>                        <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Observed&quot;</span>,<span class="st">&quot;Chao1&quot;</span>,<span class="st">&quot;Shannon&quot;</span>)) <span class="sc">+</span></span>
<span id="cb87-5"><a href="15-Alpha.html#cb87-5" tabindex="-1"></a>                          ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>() <span class="sc">+</span></span>
<span id="cb87-6"><a href="15-Alpha.html#cb87-6" tabindex="-1"></a>                          ggforce<span class="sc">::</span><span class="fu">geom_sina</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">colour=</span>media)) <span class="sc">+</span></span>
<span id="cb87-7"><a href="15-Alpha.html#cb87-7" tabindex="-1"></a>                          ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">colour =</span> <span class="st">&quot;Media&quot;</span>)</span>
<span id="cb87-8"><a href="15-Alpha.html#cb87-8" tabindex="-1"></a><span class="co">#Save ggplot2 object with ggsave</span></span>
<span id="cb87-9"><a href="15-Alpha.html#cb87-9" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./Alpha_diversity_site_violinplot.png&quot;</span>, <span class="at">plot =</span> alpha_violinplot,</span>
<span id="cb87-10"><a href="15-Alpha.html#cb87-10" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">150</span>, <span class="at">width =</span> <span class="dv">250</span>)</span>
<span id="cb87-11"><a href="15-Alpha.html#cb87-11" tabindex="-1"></a><span class="co">#Display plot</span></span>
<span id="cb87-12"><a href="15-Alpha.html#cb87-12" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./Alpha_diversity_site_violinplot.png&quot;</span>)</span></code></pre></div>
<p>When you are happy you can close and halt the notebook.</p>
</div>
<div id="alpha-summary" class="section level2 hasAnchor" number="15.6">
<h2><span class="header-section-number">15.6</span> Alpha: summary<a href="15-Alpha.html#alpha-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_blue.png" width="15%" style="display: block; margin: auto;" /></p>
<p>In this chapter we have created alpha diversity plots and run pairwise comparisons.
The biggest difference between samples is caused by the media choice which overwrites differences we can see between sites.
The environmental samples have much higher diversity samples than the media samples but this is expected.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="14-Rarefaction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="16-Beta.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
