<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 16 Beta | R community analysis</title>
  <meta name="description" content="CGR book for the R community analysis workflow" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 16 Beta | R community analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/CGR_strips.png" />
  <meta property="og:description" content="CGR book for the R community analysis workflow" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 16 Beta | R community analysis" />
  
  <meta name="twitter:description" content="CGR book for the R community analysis workflow" />
  <meta name="twitter:image" content="/figures/CGR_strips.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-07-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/CGR_favicon.png" type="image/x-icon" />
<link rel="prev" href="15-Alpha.html"/>
<link rel="next" href="17-Differential_abundance_analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://www.liverpool.ac.uk/genomic-research/" target="blank"><img src="figures/CGR_strips.png"></a></li>
<li><a>R community analysis</a></li>

<li class="divider"></li>
<li class="part"><span><b>Intro</b></span></li>
<li class="chapter" data-level="1" data-path="01-R_community_main_workflow.html"><a href="01-R_community_main_workflow.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html"><i class="fa fa-check"></i><b>2</b> Dataset &amp; workflow</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#dataset"><i class="fa fa-check"></i><b>2.1</b> Dataset</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sites"><i class="fa fa-check"></i><b>2.1.1</b> Sites</a></li>
<li class="chapter" data-level="2.1.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#culture-media"><i class="fa fa-check"></i><b>2.1.2</b> Culture media</a></li>
<li class="chapter" data-level="2.1.3" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sum_and_qs"><i class="fa fa-check"></i><b>2.1.3</b> Summary &amp; questions</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#workflow"><i class="fa fa-check"></i><b>2.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-R_packages.html"><a href="03-R_packages.html"><i class="fa fa-check"></i><b>3</b> R Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_packages.html"><a href="03-R_packages.html#r-packageslibraries"><i class="fa fa-check"></i><b>3.1</b> R packages/libraries</a></li>
<li class="chapter" data-level="3.2" data-path="03-R_packages.html"><a href="03-R_packages.html#the-grammar-of-graphics"><i class="fa fa-check"></i><b>3.2</b> The grammar of graphics</a></li>
<li class="chapter" data-level="3.3" data-path="03-R_packages.html"><a href="03-R_packages.html#phyloseq"><i class="fa fa-check"></i><b>3.3</b> phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Setup.html"><a href="04-Setup.html"><i class="fa fa-check"></i><b>4</b> Set-up</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Setup.html"><a href="04-Setup.html#cluster"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Setup.html"><a href="04-Setup.html#mamba"><i class="fa fa-check"></i><b>4.2</b> Mamba</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Jupyter.html"><a href="05-Jupyter.html"><i class="fa fa-check"></i><b>5</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Jupyter.html"><a href="05-Jupyter.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.2" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-r-notebook"><i class="fa fa-check"></i><b>5.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.3" data-path="05-Jupyter.html"><a href="05-Jupyter.html#cells-and-code"><i class="fa fa-check"></i><b>5.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.4" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-new-cells"><i class="fa fa-check"></i><b>5.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.5" data-path="05-Jupyter.html"><a href="05-Jupyter.html#running-code"><i class="fa fa-check"></i><b>5.5</b> Running code</a></li>
<li class="chapter" data-level="5.6" data-path="05-Jupyter.html"><a href="05-Jupyter.html#saving-the-file"><i class="fa fa-check"></i><b>5.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.7" data-path="05-Jupyter.html"><a href="05-Jupyter.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.8" data-path="05-Jupyter.html"><a href="05-Jupyter.html#close-the-notebook"><i class="fa fa-check"></i><b>5.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.9" data-path="05-Jupyter.html"><a href="05-Jupyter.html#video-tutorial"><i class="fa fa-check"></i><b>5.9</b> Video tutorial</a></li>
</ul></li>
<li class="part"><span><b>Data preparation</b></span></li>
<li class="chapter" data-level="6" data-path="06-Preprocess_part.html"><a href="06-Preprocess_part.html"><i class="fa fa-check"></i><b>6</b> Data prep intro</a></li>
<li class="chapter" data-level="7" data-path="07-Import.html"><a href="07-Import.html"><i class="fa fa-check"></i><b>7</b> Import</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Import.html"><a href="07-Import.html#import-notebook"><i class="fa fa-check"></i><b>7.1</b> Import: notebook</a></li>
<li class="chapter" data-level="7.2" data-path="07-Import.html"><a href="07-Import.html#qiime2r"><i class="fa fa-check"></i><b>7.2</b> qiime2R</a></li>
<li class="chapter" data-level="7.3" data-path="07-Import.html"><a href="07-Import.html#import-summarise-phyloseq"><i class="fa fa-check"></i><b>7.3</b> Import: Summarise phyloseq</a></li>
<li class="chapter" data-level="7.4" data-path="07-Import.html"><a href="07-Import.html#save-the-phyloseq-object"><i class="fa fa-check"></i><b>7.4</b> Save the phyloseq object</a></li>
<li class="chapter" data-level="7.5" data-path="07-Import.html"><a href="07-Import.html#import-recap"><i class="fa fa-check"></i><b>7.5</b> Import: recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html"><i class="fa fa-check"></i><b>8</b> Summarise phyloseq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-setup"><i class="fa fa-check"></i><b>8.1</b> Summarise: setup</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-header"><i class="fa fa-check"></i><b>8.1.1</b> Summarise header</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-phyloseq"><i class="fa fa-check"></i><b>8.1.2</b> Summarise phyloseq</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#reads-per-sample"><i class="fa fa-check"></i><b>8.1.3</b> Reads per sample</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#display-png"><i class="fa fa-check"></i><b>8.1.4</b> Display png</a></li>
<li class="chapter" data-level="8.1.5" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#asvs-per-sample"><i class="fa fa-check"></i><b>8.1.5</b> ASVs per sample</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-recap"><i class="fa fa-check"></i><b>8.2</b> Summarise: recap</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html"><i class="fa fa-check"></i><b>9</b> Minimum read depth</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#considerations"><i class="fa fa-check"></i><b>9.1</b> Considerations</a></li>
<li class="chapter" data-level="9.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minimum-read-depth-section-title"><i class="fa fa-check"></i><b>9.2</b> Minimum read depth section title</a></li>
<li class="chapter" data-level="9.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#read-depth-vector-and-histogram"><i class="fa fa-check"></i><b>9.3</b> Read depth vector and histogram</a></li>
<li class="chapter" data-level="9.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#depth-boxplots"><i class="fa fa-check"></i><b>9.4</b> Sample depth boxplot</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#creating-a-data-frame-for-boxplots"><i class="fa fa-check"></i><b>9.4.1</b> Creating a data frame for boxplots</a></li>
<li class="chapter" data-level="9.4.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#ggplot2-boxplot"><i class="fa fa-check"></i><b>9.4.2</b> ggplot2 boxplot</a></li>
<li class="chapter" data-level="9.4.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#save-ggplot-as-png"><i class="fa fa-check"></i><b>9.4.3</b> Save ggplot as png</a></li>
<li class="chapter" data-level="9.4.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#media-by-depth-boxplot"><i class="fa fa-check"></i><b>9.4.4</b> Media by depth boxplot</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#chap9rarefaction"><i class="fa fa-check"></i><b>9.5</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#asv-abundance-data-frame"><i class="fa fa-check"></i><b>9.5.1</b> ASV abundance data frame</a></li>
<li class="chapter" data-level="9.5.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#vegans-rarecurve"><i class="fa fa-check"></i><b>9.5.2</b> vegan's rarecurve</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#filtering-by-minimum-read-depth"><i class="fa fa-check"></i><b>9.6</b> Filtering by minimum read depth</a></li>
<li class="chapter" data-level="9.7" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minmum-read-depth-summary"><i class="fa fa-check"></i><b>9.7</b> Minmum read depth: Summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy</b></span></li>
<li class="chapter" data-level="10" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html"><i class="fa fa-check"></i><b>10</b> Taxa relative abundance</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-setup"><i class="fa fa-check"></i><b>10.1</b> Taxa relative abundance: setup</a></li>
<li class="chapter" data-level="10.2" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#relative-abundance-transformation"><i class="fa fa-check"></i><b>10.2</b> Relative abundance transformation</a></li>
<li class="chapter" data-level="10.3" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#aggregate-taxa"><i class="fa fa-check"></i><b>10.3</b> Aggregate taxa</a></li>
<li class="chapter" data-level="10.4" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-summary"><i class="fa fa-check"></i><b>10.4</b> Taxa Relative abundance: summary</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html"><i class="fa fa-check"></i><b>11</b> Taxa plots</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#simple-bar-chart"><i class="fa fa-check"></i><b>11.1</b> Simple bar chart</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#editing-the-plot"><i class="fa fa-check"></i><b>11.1.1</b> Editing the plot</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-heatmap"><i class="fa fa-check"></i><b>11.2</b> Taxa heatmap</a></li>
<li class="chapter" data-level="11.3" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#aggregateraretaxa"><i class="fa fa-check"></i><b>11.3</b> Aggregate rare taxa</a></li>
<li class="chapter" data-level="11.4" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-media"><i class="fa fa-check"></i><b>11.4</b> Plot by media</a></li>
<li class="chapter" data-level="11.5" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-site-and-media"><i class="fa fa-check"></i><b>11.5</b> Plot by site and media</a></li>
<li class="chapter" data-level="11.6" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-plots-summary"><i class="fa fa-check"></i><b>11.6</b> Taxa plots: summary</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html"><i class="fa fa-check"></i><b>12</b> Family &amp; Genus plots</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-taxa-plots"><i class="fa fa-check"></i><b>12.1</b> Family taxa plots</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#aggregate-families"><i class="fa fa-check"></i><b>12.1.1</b> Aggregate families</a></li>
<li class="chapter" data-level="12.1.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-heatmap"><i class="fa fa-check"></i><b>12.1.2</b> Family heatmap</a></li>
<li class="chapter" data-level="12.1.3" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-rare-aggregation"><i class="fa fa-check"></i><b>12.1.3</b> Family rare aggregation</a></li>
<li class="chapter" data-level="12.1.4" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-bar-chart"><i class="fa fa-check"></i><b>12.1.4</b> Family bar chart</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots"><i class="fa fa-check"></i><b>12.2</b> Genus taxa plots</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots-solutions"><i class="fa fa-check"></i><b>12.2.1</b> Genus taxa plots solutions</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Diversity analysis</b></span></li>
<li class="chapter" data-level="13" data-path="13-Diversity_analysis.html"><a href="13-Diversity_analysis.html"><i class="fa fa-check"></i><b>13</b> Diversity analysis intro</a></li>
<li class="chapter" data-level="14" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html"><i class="fa fa-check"></i><b>14</b> Rarefaction</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-setup"><i class="fa fa-check"></i><b>14.1</b> Rarefaction: setup</a></li>
<li class="chapter" data-level="14.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve"><i class="fa fa-check"></i><b>14.2</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#bird-analogy"><i class="fa fa-check"></i><b>14.2.1</b> Bird analogy</a></li>
<li class="chapter" data-level="14.2.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-simple-plot"><i class="fa fa-check"></i><b>14.2.2</b> Rarefaction curve: simple plot</a></li>
<li class="chapter" data-level="14.2.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-better-plot"><i class="fa fa-check"></i><b>14.2.3</b> Rarefaction curve: better plot</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-slope"><i class="fa fa-check"></i><b>14.3</b> Rarefaction slope</a></li>
<li class="chapter" data-level="14.4" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefy-data"><i class="fa fa-check"></i><b>14.4</b> Rarefy data</a></li>
<li class="chapter" data-level="14.5" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-summary"><i class="fa fa-check"></i><b>14.5</b> Rarefaction: summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Alpha.html"><a href="15-Alpha.html"><i class="fa fa-check"></i><b>15</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-setup"><i class="fa fa-check"></i><b>15.1</b> Alpha: setup</a></li>
<li class="chapter" data-level="15.2" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-simple-plot"><i class="fa fa-check"></i><b>15.2</b> Alpha: simple plot</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="15-Alpha.html"><a href="15-Alpha.html#metric-choice"><i class="fa fa-check"></i><b>15.2.1</b> Metric choice</a></li>
<li class="chapter" data-level="15.2.2" data-path="15-Alpha.html"><a href="15-Alpha.html#sample-grouping-box-plot"><i class="fa fa-check"></i><b>15.2.2</b> Sample grouping &amp; box plot</a></li>
<li class="chapter" data-level="15.2.3" data-path="15-Alpha.html"><a href="15-Alpha.html#png-height-and-width-choice"><i class="fa fa-check"></i><b>15.2.3</b> PNG height and width choice</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats"><i class="fa fa-check"></i><b>15.3</b> Alpha: stats</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats-task"><i class="fa fa-check"></i><b>15.3.1</b> Alpha stats task</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-diversity-violin-plot"><i class="fa fa-check"></i><b>15.4</b> Alpha diversity: violin plot</a></li>
<li class="chapter" data-level="15.5" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-site-plots-and-statistics"><i class="fa fa-check"></i><b>15.5</b> Alpha: Site plots and statistics</a></li>
<li class="chapter" data-level="15.6" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-summary"><i class="fa fa-check"></i><b>15.6</b> Alpha: summary</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Beta.html"><a href="16-Beta.html"><i class="fa fa-check"></i><b>16</b> Beta</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Beta.html"><a href="16-Beta.html#beta-setup"><i class="fa fa-check"></i><b>16.1</b> Beta: setup</a></li>
<li class="chapter" data-level="16.2" data-path="16-Beta.html"><a href="16-Beta.html#beta-ordination"><i class="fa fa-check"></i><b>16.2</b> Beta: ordination</a></li>
<li class="chapter" data-level="16.3" data-path="16-Beta.html"><a href="16-Beta.html#beta-plot-ordination"><i class="fa fa-check"></i><b>16.3</b> Beta: plot ordination</a></li>
<li class="chapter" data-level="16.4" data-path="16-Beta.html"><a href="16-Beta.html#beta-statistics"><i class="fa fa-check"></i><b>16.4</b> Beta: statistics</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="16-Beta.html"><a href="16-Beta.html#permanova"><i class="fa fa-check"></i><b>16.4.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="16.4.2" data-path="16-Beta.html"><a href="16-Beta.html#pairwise-permanova"><i class="fa fa-check"></i><b>16.4.2</b> Pairwise PERMANOVA</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="16-Beta.html"><a href="16-Beta.html#beta-facet-zoom"><i class="fa fa-check"></i><b>16.5</b> Beta: Facet zoom</a></li>
<li class="chapter" data-level="16.6" data-path="16-Beta.html"><a href="16-Beta.html#beta-subset-samples"><i class="fa fa-check"></i><b>16.6</b> Beta: subset samples</a></li>
<li class="chapter" data-level="16.7" data-path="16-Beta.html"><a href="16-Beta.html#permanova-tasks"><i class="fa fa-check"></i><b>16.7</b> PERMANOVA tasks</a></li>
<li class="chapter" data-level="16.8" data-path="16-Beta.html"><a href="16-Beta.html#beta-summary"><i class="fa fa-check"></i><b>16.8</b> Beta: Summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html"><i class="fa fa-check"></i><b>17</b> Differential abundance (DA) analysis</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-setup"><i class="fa fa-check"></i><b>17.1</b> DA: setup</a></li>
<li class="chapter" data-level="17.2" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-preprocess"><i class="fa fa-check"></i><b>17.2</b> DA: Preprocess</a></li>
<li class="chapter" data-level="17.3" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-analysis"><i class="fa fa-check"></i><b>17.3</b> DA: ANCOM-BC analysis</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#ancombc2-options"><i class="fa fa-check"></i><b>17.3.1</b> ANCOMBC2 options</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-results"><i class="fa fa-check"></i><b>17.4</b> DA: ANCOM-BC results</a></li>
<li class="chapter" data-level="17.5" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-create-signifcant-lfc-long-data-frame"><i class="fa fa-check"></i><b>17.5</b> DA: Create signifcant LFC long data frame</a></li>
<li class="chapter" data-level="17.6" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-biomarker-lfc-heatmap"><i class="fa fa-check"></i><b>17.6</b> DA: Biomarker LFC heatmap</a></li>
<li class="chapter" data-level="17.7" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-matching-bar-plot"><i class="fa fa-check"></i><b>17.7</b> DA: matching bar plot</a></li>
<li class="chapter" data-level="17.8" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-summary"><i class="fa fa-check"></i><b>17.8</b> DA: summary</a></li>
</ul></li>
<li class="part"><span><b>Outro</b></span></li>
<li class="chapter" data-level="18" data-path="18-Summary.html"><a href="18-Summary.html"><i class="fa fa-check"></i><b>18</b> Summary</a>
<ul>
<li class="chapter" data-level="18.1" data-path="18-Summary.html"><a href="18-Summary.html#dataset-1"><i class="fa fa-check"></i><b>18.1</b> Dataset</a></li>
<li class="chapter" data-level="18.2" data-path="18-Summary.html"><a href="18-Summary.html#data-preparation"><i class="fa fa-check"></i><b>18.2</b> Data preparation</a></li>
<li class="chapter" data-level="18.3" data-path="18-Summary.html"><a href="18-Summary.html#taxonomy"><i class="fa fa-check"></i><b>18.3</b> Taxonomy</a></li>
<li class="chapter" data-level="18.4" data-path="18-Summary.html"><a href="18-Summary.html#diversity-analysis"><i class="fa fa-check"></i><b>18.4</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="18-Summary.html"><a href="18-Summary.html#alpha-beta"><i class="fa fa-check"></i><b>18.4.1</b> Alpha &amp; beta</a></li>
<li class="chapter" data-level="18.4.2" data-path="18-Summary.html"><a href="18-Summary.html#differential-abundance-anlaysis"><i class="fa fa-check"></i><b>18.4.2</b> Differential abundance anlaysis</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="18-Summary.html"><a href="18-Summary.html#findings"><i class="fa fa-check"></i><b>18.5</b> Findings</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="18-Summary.html"><a href="18-Summary.html#anthropisation-gradient"><i class="fa fa-check"></i><b>18.5.1</b> Anthropisation gradient</a></li>
<li class="chapter" data-level="18.5.2" data-path="18-Summary.html"><a href="18-Summary.html#replicate-differences"><i class="fa fa-check"></i><b>18.5.2</b> Replicate differences</a></li>
<li class="chapter" data-level="18.5.3" data-path="18-Summary.html"><a href="18-Summary.html#site-or-media"><i class="fa fa-check"></i><b>18.5.3</b> Site or media</a></li>
<li class="chapter" data-level="18.5.4" data-path="18-Summary.html"><a href="18-Summary.html#media-versus-env-samples"><i class="fa fa-check"></i><b>18.5.4</b> Media versus ENV samples</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="18-Summary.html"><a href="18-Summary.html#conclusion"><i class="fa fa-check"></i><b>18.6</b> Conclusion</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="19-Appendix.html"><a href="19-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="19-Appendix.html"><a href="19-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
<li class="chapter" data-level="A.2" data-path="19-Appendix.html"><a href="19-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>A.2</b> Jupyter-notebook</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="19-Appendix.html"><a href="19-Appendix.html#r-packages-1"><i class="fa fa-check"></i><b>B</b> R packages</a>
<ul>
<li class="chapter" data-level="B.1" data-path="19-Appendix.html"><a href="19-Appendix.html#ggplot2_appendix"><i class="fa fa-check"></i><b>B.1</b> ggplot2</a></li>
<li class="chapter" data-level="B.2" data-path="19-Appendix.html"><a href="19-Appendix.html#phyloseq_appendix"><i class="fa fa-check"></i><b>B.2</b> Phyloseq</a></li>
<li class="chapter" data-level="B.3" data-path="19-Appendix.html"><a href="19-Appendix.html#importing-into-phyloseq"><i class="fa fa-check"></i><b>B.3</b> Importing into phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-metrics"><i class="fa fa-check"></i><b>C</b> Diversity metrics</a>
<ul>
<li class="chapter" data-level="C.1" data-path="19-Appendix.html"><a href="19-Appendix.html#alpha_appendix"><i class="fa fa-check"></i><b>C.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="19-Appendix.html"><a href="19-Appendix.html#obvs"><i class="fa fa-check"></i><b>C.1.1</b> Observed features</a></li>
<li class="chapter" data-level="C.1.2" data-path="19-Appendix.html"><a href="19-Appendix.html#chao"><i class="fa fa-check"></i><b>C.1.2</b> Chao</a></li>
<li class="chapter" data-level="C.1.3" data-path="19-Appendix.html"><a href="19-Appendix.html#evenness"><i class="fa fa-check"></i><b>C.1.3</b> Evenness</a></li>
<li class="chapter" data-level="C.1.4" data-path="19-Appendix.html"><a href="19-Appendix.html#faith"><i class="fa fa-check"></i><b>C.1.4</b> Faith's PD (phylogenetic diversity)</a></li>
<li class="chapter" data-level="C.1.5" data-path="19-Appendix.html"><a href="19-Appendix.html#simpson"><i class="fa fa-check"></i><b>C.1.5</b> Simpson</a></li>
<li class="chapter" data-level="C.1.6" data-path="19-Appendix.html"><a href="19-Appendix.html#simpsone"><i class="fa fa-check"></i><b>C.1.6</b> Simpson evenness measure (simpson_e)</a></li>
<li class="chapter" data-level="C.1.7" data-path="19-Appendix.html"><a href="19-Appendix.html#shannon"><i class="fa fa-check"></i><b>C.1.7</b> Shannon</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="19-Appendix.html"><a href="19-Appendix.html#beta_appendix"><i class="fa fa-check"></i><b>C.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="19-Appendix.html"><a href="19-Appendix.html#unifrac_explanation"><i class="fa fa-check"></i><b>C.2.1</b> Weighted and unweighted UniFrac distances</a></li>
<li class="chapter" data-level="C.2.2" data-path="19-Appendix.html"><a href="19-Appendix.html#bray_explanation"><i class="fa fa-check"></i><b>C.2.2</b> Bray-Curtis</a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-resources"><i class="fa fa-check"></i><b>C.3</b> Diversity Resources</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R community analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="beta_chap" class="section level1 hasAnchor" number="16">
<h1><span class="header-section-number">Chapter 16</span> Beta<a href="16-Beta.html#beta_chap" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/beta.png" width="20%" style="display: block; margin: auto;" /></p>
<p>In this chapter we are going to carry out beta diversity analysis.
This will include the production of ordination plots and statistical analysis.</p>
<p>When carrying out beta diversity all samples are compared in a pairwise manner.
Using various metrics a table of pairwise distances is created.
These distances are dissimilarity measures with smaller values indicating higher similarity and larger values representing higher dissimilarity.</p>
<p>With these dissimilarity measures we can produce scatter plots based visualisations (e.g. PCoA, &amp; NMDS plots).
Samples with low dissimilarity (i.e. similar samples) will be close to each other (cluster).
Samples with high dissimilarity will be apart from each other.
In a perfect example samples from the same group will form clusters that are distinct from the clusters of other groups.</p>
<p>In this chapter we will:</p>
<ul>
<li>Calculate beta diversity ordination metrics and plot them as a scatter plot.</li>
<li>Determine if there is a statistical difference between beta diversity metrics of sample groups.</li>
</ul>
<div id="beta-setup" class="section level2 hasAnchor" number="16.1">
<h2><span class="header-section-number">16.1</span> Beta: setup<a href="16-Beta.html#beta-setup" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/books.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We will use a new notebook called <strong>"6-beta_diversity.ipynb"</strong>.
Add the following to the top of this new notebook to load the required libraries/packages and data.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="16-Beta.html#cb88-1" tabindex="-1"></a><span class="co">#Libraries</span></span>
<span id="cb88-2"><a href="16-Beta.html#cb88-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb88-3"><a href="16-Beta.html#cb88-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;microbiome&quot;</span>)</span>
<span id="cb88-4"><a href="16-Beta.html#cb88-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;IRdisplay&quot;</span>)</span>
<span id="cb88-5"><a href="16-Beta.html#cb88-5" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggforce&quot;</span>)</span>
<span id="cb88-6"><a href="16-Beta.html#cb88-6" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;vegan&quot;</span>)</span>
<span id="cb88-7"><a href="16-Beta.html#cb88-7" tabindex="-1"></a><span class="co">#Load phyloseq object</span></span>
<span id="cb88-8"><a href="16-Beta.html#cb88-8" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;phyloseq_rarefied.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="beta-ordination" class="section level2 hasAnchor" number="16.2">
<h2><span class="header-section-number">16.2</span> Beta: ordination<a href="16-Beta.html#beta-ordination" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/ordinate.png" width="15%" style="display: block; margin: auto;" /></p>
<p>In this section we will ordinate our data with the <code>phyloseq::ordinate()</code> function.
We'll use the <a href="https://en.wikipedia.org/wiki/Multidimensional_scaling">MDS method</a> (AKA PCoA).
This will be done with the Weighted Unifrac metric as the beta diversity dissimilarity scores.
This uses the count data along with the phylogenetic distances to calculate pairwise dissimilarity.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="16-Beta.html#cb89-1" tabindex="-1"></a><span class="co">#Ordinate data</span></span>
<span id="cb89-2"><a href="16-Beta.html#cb89-2" tabindex="-1"></a>ord.mds.wunifrac <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">ordinate</span>(pseq_rarefy, <span class="at">method =</span> <span class="st">&quot;MDS&quot;</span>, <span class="at">distance =</span> <span class="st">&quot;wunifrac&quot;</span>)</span></code></pre></div>
<p>To see the different methods and distances that can be used please try <code>?phyloseq:ordinate</code> and <code>phyloseq::distanceMethodList</code> respectively.
For more information about the different distances please see the <a href="19-Appendix.html#beta_appendix">APPENDIX</a>.</p>
</div>
<div id="beta-plot-ordination" class="section level2 hasAnchor" number="16.3">
<h2><span class="header-section-number">16.3</span> Beta: plot ordination<a href="16-Beta.html#beta-plot-ordination" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/Scatterplot_2.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Now that we have our ordination we can plot it.
The script below uses <code>phyloseq::plot_ordination()</code> to plot our ordination.
It needs the <code>phyloseq</code> object used to create the ordination (for the metadata) and the ordination data itself.
We will add the options <code>color =</code> and <code>shape =</code> to colour the points by site and shape the points by media.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="16-Beta.html#cb90-1" tabindex="-1"></a><span class="co">#Plot ordination</span></span>
<span id="cb90-2"><a href="16-Beta.html#cb90-2" tabindex="-1"></a>nmds.wunifrac <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_ordination</span>(pseq_rarefy, ord.mds.wunifrac,</span>
<span id="cb90-3"><a href="16-Beta.html#cb90-3" tabindex="-1"></a>                                   <span class="at">color =</span> <span class="st">&quot;site&quot;</span>, <span class="at">shape =</span> <span class="st">&quot;media&quot;</span>)</span>
<span id="cb90-4"><a href="16-Beta.html#cb90-4" tabindex="-1"></a><span class="co">#Save ggplot2 object with ggsave</span></span>
<span id="cb90-5"><a href="16-Beta.html#cb90-5" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./Beta_diversity_mds_wunifrac_media_site.png&quot;</span>, <span class="at">plot =</span> nmds.wunifrac,</span>
<span id="cb90-6"><a href="16-Beta.html#cb90-6" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">125</span>, <span class="at">width =</span> <span class="dv">150</span>)</span>
<span id="cb90-7"><a href="16-Beta.html#cb90-7" tabindex="-1"></a><span class="co">#Display plot</span></span>
<span id="cb90-8"><a href="16-Beta.html#cb90-8" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./Beta_diversity_mds_wunifrac_media_site.png&quot;</span>)</span></code></pre></div>
<p>In this plot we can see that there are 3 main clusters:</p>
<ol style="list-style-type: decimal">
<li>TSA</li>
<li>ENV</li>
<li>CVP &amp; KBC</li>
</ol>
<p>A very important feature of ordination plots is the percentage values on the axes.
These are the <strong>percentage variation explained</strong> values.
These values show how much of the variation of the data is explained by each axis.
We need these as we are converting high dimensional data (many pairwise comparisons) into a 2 dimensional plot.</p>
<p>In our plot axis 1 has a very high percentage variation explained value of 72.5%.
This indicates that most of the variation in our data is explained by the differences between the media samples (CVP, KBC, &amp; TSA) and the ENV samples.
This is because the ENV samples are far removed (large distances) from the media samples in terms of axis 1.</p>
<p>On the other hand, axis.2 (15.8%) seems to be primarily showing the difference between the TSA samples and the other samples, and within the TSA samples.</p>
<p>The higher the total variation explained value is, the better our plot represents the beta diversity in our data.
In this case the total is &gt;88% which is very good.</p>
<ul>
<li><strong>&lt;30%</strong> is poor</li>
<li><strong>&gt;30%</strong> is not very good</li>
<li><strong>&gt;50%</strong> is good</li>
<li><strong>&gt;70%</strong> is very good</li>
<li><strong>&gt;90%</strong> is exceptional</li>
</ul>
<p>Additionally, we can see that there seems to be some separation of the sites within the different media groups. This is clear in TSA and ENV but not in CVP and KBC since the cluster is very tight (short distances).</p>
</div>
<div id="beta-statistics" class="section level2 hasAnchor" number="16.4">
<h2><span class="header-section-number">16.4</span> Beta: statistics<a href="16-Beta.html#beta-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/cluster_colours.png" width="15%" style="display: block; margin: auto;" /></p>
<p>For our beta diversity we will carry out PERMANOVA analysis for the media groupings.
This will be carried out for all the samples and in a pairwise manner.</p>
<p>PERMANOVA is a good test in this case to determine if there is a significance difference between the distances of groups.
This allows us to determine if the clustering of groups is significant.</p>
<p>For more info on PERMANOVA, <a href="https://en.wikipedia.org/wiki/Permutational_analysis_of_variance#:~:text=Permutational%20multivariate%20analysis%20of%20variance,are%20equivalent%20for%20all%20groups.">Wikipedia</a> is a good start.</p>
<div id="permanova" class="section level3 hasAnchor" number="16.4.1">
<h3><span class="header-section-number">16.4.1</span> PERMANOVA<a href="16-Beta.html#permanova" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/permanova.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We will carry out a PERMANOVA test on all the samples based on media.
This will help us determine if there are any differences between groupings but not which groupings.
Prior to PERMANOVA we need to create a matrix containing all the sample paired distance values.
We will use the function <code>phyloseq::distance()</code>.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="16-Beta.html#cb91-1" tabindex="-1"></a><span class="co">#PERMANOVA</span></span>
<span id="cb91-2"><a href="16-Beta.html#cb91-2" tabindex="-1"></a><span class="co">#Produce distance matrix</span></span>
<span id="cb91-3"><a href="16-Beta.html#cb91-3" tabindex="-1"></a>wunifrac_dist_mat <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(pseq_rarefy, <span class="at">method=</span><span class="st">&quot;wunifrac&quot;</span>)</span></code></pre></div>
<p>With the distance matrix we can carry out a PERMANOVA test with <a href="https://search.r-project.org/CRAN/refmans/vegan/html/adonis.html"><code>vegan::adonis2()</code></a>.
This requires the metadata in the form of a data frame.
In this case we can assess if there are any differences by media and by site (<code>media+site</code>).</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="16-Beta.html#cb92-1" tabindex="-1"></a><span class="co">#Extract metadata data frame from phyloseq</span></span>
<span id="cb92-2"><a href="16-Beta.html#cb92-2" tabindex="-1"></a>metadf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq_rarefy))</span>
<span id="cb92-3"><a href="16-Beta.html#cb92-3" tabindex="-1"></a><span class="co">#PERMANOVA/ADONIS of media and site</span></span>
<span id="cb92-4"><a href="16-Beta.html#cb92-4" tabindex="-1"></a>wunifrac_adonis <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(wunifrac_dist_mat <span class="sc">~</span> media<span class="sc">+</span>site, <span class="at">data =</span> metadf, <span class="at">by =</span> <span class="st">&quot;margin&quot;</span>)</span>
<span id="cb92-5"><a href="16-Beta.html#cb92-5" tabindex="-1"></a>wunifrac_adonis</span></code></pre></div>
<p>There is a lot of information in the resulting table.
We are primarily interested in the P-value column (<code>Pr(&gt;F)</code>).
We can see that significant differences are found within the groups for media (P = 0.001) and site (P = 0.018).</p>
</div>
<div id="pairwise-permanova" class="section level3 hasAnchor" number="16.4.2">
<h3><span class="header-section-number">16.4.2</span> Pairwise PERMANOVA<a href="16-Beta.html#pairwise-permanova" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/pear.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will now see which media groups are significantly different to each other.
We will do this with pairwise PERMANOVA tests.
Unfortunately the <code>vegan</code> library does not have a function to do this so we will need to do some scripting.</p>
<p>The first step is to get every combination of groupings with the <code>combn()</code> and <code>unique()</code> functions.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="16-Beta.html#cb93-1" tabindex="-1"></a><span class="co">#Pairwise comparisons</span></span>
<span id="cb93-2"><a href="16-Beta.html#cb93-2" tabindex="-1"></a><span class="co">#Get combinations of unique media values</span></span>
<span id="cb93-3"><a href="16-Beta.html#cb93-3" tabindex="-1"></a>cbn <span class="ot">&lt;-</span> <span class="fu">combn</span>(<span class="at">x =</span> <span class="fu">unique</span>(metadf<span class="sc">$</span>media), <span class="at">m =</span> <span class="dv">2</span>)</span>
<span id="cb93-4"><a href="16-Beta.html#cb93-4" tabindex="-1"></a>cbn</span>
<span id="cb93-5"><a href="16-Beta.html#cb93-5" tabindex="-1"></a><span class="co">#First combination</span></span>
<span id="cb93-6"><a href="16-Beta.html#cb93-6" tabindex="-1"></a>cbn[,<span class="dv">1</span>]</span></code></pre></div>
<div id="pairwise-permanova-on-one-combination" class="section level4 hasAnchor" number="16.4.2.1">
<h4><span class="header-section-number">16.4.2.1</span> Pairwise PERMANOVA on one combination<a href="16-Beta.html#pairwise-permanova-on-one-combination" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>For demonstration purposes we will first carry out the pairwise PERMANOVA on the 1st grouping combination.</p>
<p>With the first media combination (CVP &amp; ENV) we can subset our <code>phyloseq</code> object so it only contains those samples.
Additionally, we can extract the metadata from this subsetted <code>phyloseq</code> object.</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="16-Beta.html#cb94-1" tabindex="-1"></a><span class="co">#Subset phyloseq so it only contains CVP &amp; ENV samples (1st combination)</span></span>
<span id="cb94-2"><a href="16-Beta.html#cb94-2" tabindex="-1"></a>pseq_rarefy_subset <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(pseq_rarefy, media <span class="sc">%in%</span> cbn[,<span class="dv">1</span>])</span>
<span id="cb94-3"><a href="16-Beta.html#cb94-3" tabindex="-1"></a><span class="co">#Metadata data frame</span></span>
<span id="cb94-4"><a href="16-Beta.html#cb94-4" tabindex="-1"></a>metadf_subset <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq_rarefy_subset))</span>
<span id="cb94-5"><a href="16-Beta.html#cb94-5" tabindex="-1"></a>metadf_subset</span></code></pre></div>
<p>With this info we can calculate the distances and then carry out the PERMANOVA.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="16-Beta.html#cb95-1" tabindex="-1"></a><span class="co">#Weighted unifrac distance calculation</span></span>
<span id="cb95-2"><a href="16-Beta.html#cb95-2" tabindex="-1"></a>wunifrac_dist_mat <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(pseq_rarefy_subset, <span class="at">method=</span><span class="st">&quot;wunifrac&quot;</span>)</span>
<span id="cb95-3"><a href="16-Beta.html#cb95-3" tabindex="-1"></a><span class="co">#PERMANOVA/ADONIS of media</span></span>
<span id="cb95-4"><a href="16-Beta.html#cb95-4" tabindex="-1"></a>wunifrac_pairwise_adonis <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(wunifrac_dist_mat <span class="sc">~</span> media, <span class="at">data =</span> metadf_subset, <span class="at">by =</span> <span class="st">&quot;margin&quot;</span>)</span>
<span id="cb95-5"><a href="16-Beta.html#cb95-5" tabindex="-1"></a>wunifrac_pairwise_adonis</span></code></pre></div>
<p>We can see that there is a significant difference between the CVP and ENV samples (P = 0.01).</p>
</div>
<div id="pairwise-permanova-on-all-combinations" class="section level4 hasAnchor" number="16.4.2.2">
<h4><span class="header-section-number">16.4.2.2</span> Pairwise PERMANOVA on all combinations<a href="16-Beta.html#pairwise-permanova-on-all-combinations" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>It would be quite inefficient to type the above script for each combination.
We will therefore create a script with a loop to automatically run the code for each combination.
Additionally, we will create an empty data frame before the loop.
This will be used to store the comparison info and the P-values.</p>
<p>Create the below script in a new code block, reading the annotation to help understand how it works.</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="16-Beta.html#cb96-1" tabindex="-1"></a><span class="co">#Pairwise comparison loop</span></span>
<span id="cb96-2"><a href="16-Beta.html#cb96-2" tabindex="-1"></a><span class="co">#Get combinations of unique media values</span></span>
<span id="cb96-3"><a href="16-Beta.html#cb96-3" tabindex="-1"></a>cbn <span class="ot">&lt;-</span> <span class="fu">combn</span>(<span class="at">x =</span> <span class="fu">unique</span>(metadf<span class="sc">$</span>media), <span class="at">m =</span> <span class="dv">2</span>)</span>
<span id="cb96-4"><a href="16-Beta.html#cb96-4" tabindex="-1"></a><span class="co">#Create empty final data frame with 4 columns</span></span>
<span id="cb96-5"><a href="16-Beta.html#cb96-5" tabindex="-1"></a><span class="co"># and a number of rows equal to the the number of combinations</span></span>
<span id="cb96-6"><a href="16-Beta.html#cb96-6" tabindex="-1"></a>pairwise_permanova_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="at">data =</span> <span class="cn">NA</span>, <span class="at">nrow =</span> <span class="fu">ncol</span>(cbn), <span class="at">ncol =</span> <span class="dv">4</span>))</span>
<span id="cb96-7"><a href="16-Beta.html#cb96-7" tabindex="-1"></a><span class="co">#Add column names</span></span>
<span id="cb96-8"><a href="16-Beta.html#cb96-8" tabindex="-1"></a><span class="fu">colnames</span>(pairwise_permanova_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;1&quot;</span>,<span class="st">&quot;2&quot;</span>,<span class="st">&quot;p&quot;</span>,<span class="st">&quot;p.adj&quot;</span>)</span>
<span id="cb96-9"><a href="16-Beta.html#cb96-9" tabindex="-1"></a></span>
<span id="cb96-10"><a href="16-Beta.html#cb96-10" tabindex="-1"></a><span class="co">#Loop through the combinations</span></span>
<span id="cb96-11"><a href="16-Beta.html#cb96-11" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(cbn)){</span>
<span id="cb96-12"><a href="16-Beta.html#cb96-12" tabindex="-1"></a>  <span class="co">#Subset to media phyloseq</span></span>
<span id="cb96-13"><a href="16-Beta.html#cb96-13" tabindex="-1"></a>  <span class="co">#Ensure to change metadata group if using different grouping (media)</span></span>
<span id="cb96-14"><a href="16-Beta.html#cb96-14" tabindex="-1"></a>  pseq_rarefy_subset <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(pseq_rarefy, media <span class="sc">%in%</span> cbn[,i])</span>
<span id="cb96-15"><a href="16-Beta.html#cb96-15" tabindex="-1"></a>  <span class="co">#Metadata data frame</span></span>
<span id="cb96-16"><a href="16-Beta.html#cb96-16" tabindex="-1"></a>  metadf_subset <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq_rarefy_subset))</span>
<span id="cb96-17"><a href="16-Beta.html#cb96-17" tabindex="-1"></a>  <span class="co">#Weighted unifrac distance calculation</span></span>
<span id="cb96-18"><a href="16-Beta.html#cb96-18" tabindex="-1"></a>  wunifrac_dist_mat <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(pseq_rarefy_subset, <span class="at">method=</span><span class="st">&quot;wunifrac&quot;</span>)</span>
<span id="cb96-19"><a href="16-Beta.html#cb96-19" tabindex="-1"></a>  <span class="co">#PERMANOVA/ADONIS of media</span></span>
<span id="cb96-20"><a href="16-Beta.html#cb96-20" tabindex="-1"></a>  <span class="co">#Ensure to change metadata group if using different grouping (media)</span></span>
<span id="cb96-21"><a href="16-Beta.html#cb96-21" tabindex="-1"></a>  wunifrac_pairwise_adonis <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(wunifrac_dist_mat <span class="sc">~</span> media, <span class="at">data =</span> metadf_subset, <span class="at">by =</span> <span class="st">&quot;margin&quot;</span>)</span>
<span id="cb96-22"><a href="16-Beta.html#cb96-22" tabindex="-1"></a>  <span class="co">#Add the group names and p-value to the main data frame</span></span>
<span id="cb96-23"><a href="16-Beta.html#cb96-23" tabindex="-1"></a>  pairwise_permanova_df[i,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">&lt;-</span> cbn[,i]</span>
<span id="cb96-24"><a href="16-Beta.html#cb96-24" tabindex="-1"></a>  pairwise_permanova_df[i,<span class="dv">3</span>] <span class="ot">&lt;-</span> wunifrac_pairwise_adonis[<span class="dv">1</span>,<span class="st">&quot;Pr(&gt;F)&quot;</span>]</span>
<span id="cb96-25"><a href="16-Beta.html#cb96-25" tabindex="-1"></a>}</span>
<span id="cb96-26"><a href="16-Beta.html#cb96-26" tabindex="-1"></a><span class="co">#View data frame</span></span>
<span id="cb96-27"><a href="16-Beta.html#cb96-27" tabindex="-1"></a>pairwise_permanova_df</span></code></pre></div>
<p>All our media groupings are significantly different to each other (0.001).
You may have noticed that the <code>p.adj</code> column is empty.
We will add adjusted P-values to these columns.
It is always important to use adjusted P-values when carrying out multiple significant tests such as with pairwise comparisons.
We'll use the function <code>p.adjust()</code> to add Benjamin-Hochberg (<code>method = "BH"</code>) adjusted values values.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="16-Beta.html#cb97-1" tabindex="-1"></a><span class="co">#Add adjusted P-values</span></span>
<span id="cb97-2"><a href="16-Beta.html#cb97-2" tabindex="-1"></a>pairwise_permanova_df<span class="sc">$</span>p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(pairwise_permanova_df<span class="sc">$</span>p, <span class="at">method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb97-3"><a href="16-Beta.html#cb97-3" tabindex="-1"></a><span class="co">#View data frame</span></span>
<span id="cb97-4"><a href="16-Beta.html#cb97-4" tabindex="-1"></a>pairwise_permanova_df</span>
<span id="cb97-5"><a href="16-Beta.html#cb97-5" tabindex="-1"></a><span class="co">#Write to a file</span></span>
<span id="cb97-6"><a href="16-Beta.html#cb97-6" tabindex="-1"></a><span class="fu">write.table</span>(<span class="at">x =</span> pairwise_permanova_df, <span class="at">file =</span> <span class="st">&quot;pairwise_permanova_media_wunifrac.tsv&quot;</span>,</span>
<span id="cb97-7"><a href="16-Beta.html#cb97-7" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Brilliant, the adjusted P-values are still significant, in fact they did not even change.
This shows us that the different media groupings are significantly different.
This makes sense from our taxonomy, alpha diversity, and beta diversity analyses.
However, it is hard to see the separation of the CVP and KBC samples in the MDS plot.</p>
</div>
</div>
</div>
<div id="beta-facet-zoom" class="section level2 hasAnchor" number="16.5">
<h2><span class="header-section-number">16.5</span> Beta: Facet zoom<a href="16-Beta.html#beta-facet-zoom" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/telescope.png" width="15%" style="display: block; margin: auto;" /></p>
<p>The <a href="https://ggforce.data-imaginist.com/index.html"><code>ggforce</code> library</a> is a very powerful library to edit <code>ggplot2</code> objects.
One of its functions, <code>ggforce::facet_zoom()</code>, allows us to zoom into a specific part of a plot whilst also showing the main plot.
We will use this to zoom into the CVP and KBC samples within the weighted unifrac MDS plot.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="16-Beta.html#cb98-1" tabindex="-1"></a><span class="co">#Ordinate data</span></span>
<span id="cb98-2"><a href="16-Beta.html#cb98-2" tabindex="-1"></a>ord.mds.wunifrac <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">ordinate</span>(pseq_rarefy, <span class="at">method =</span> <span class="st">&quot;MDS&quot;</span>, <span class="at">distance =</span> <span class="st">&quot;wunifrac&quot;</span>)</span>
<span id="cb98-3"><a href="16-Beta.html#cb98-3" tabindex="-1"></a><span class="co">#Plot ordination</span></span>
<span id="cb98-4"><a href="16-Beta.html#cb98-4" tabindex="-1"></a>nmds.wunifrac <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_ordination</span>(pseq_rarefy, ord.mds.wunifrac,</span>
<span id="cb98-5"><a href="16-Beta.html#cb98-5" tabindex="-1"></a>                                   <span class="at">color =</span> <span class="st">&quot;site&quot;</span>, <span class="at">shape =</span> <span class="st">&quot;media&quot;</span>) <span class="sc">+</span></span>
<span id="cb98-6"><a href="16-Beta.html#cb98-6" tabindex="-1"></a>                  <span class="co">#Zoom into CVP and KBC samples specified by the metadata category</span></span>
<span id="cb98-7"><a href="16-Beta.html#cb98-7" tabindex="-1"></a>                  <span class="co">#Zoom.size = 1 so the zoom section is the same size as the main plot</span></span>
<span id="cb98-8"><a href="16-Beta.html#cb98-8" tabindex="-1"></a>                  <span class="co">#Zoom.size = 2 would mean the zoom plot is 2 times the size of the original</span></span>
<span id="cb98-9"><a href="16-Beta.html#cb98-9" tabindex="-1"></a>                  ggforce<span class="sc">::</span><span class="fu">facet_zoom</span>( <span class="at">xy =</span> media <span class="sc">==</span> <span class="fu">c</span>(<span class="st">&quot;CVP&quot;</span>,<span class="st">&quot;KBC&quot;</span>), <span class="at">zoom.size =</span> <span class="dv">1</span>)</span>
<span id="cb98-10"><a href="16-Beta.html#cb98-10" tabindex="-1"></a><span class="co">#Save ggplot2 object with ggsave</span></span>
<span id="cb98-11"><a href="16-Beta.html#cb98-11" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./Beta_diversity_mds_wunifrac_media_site_zoom.png&quot;</span>, <span class="at">plot =</span> nmds.wunifrac,</span>
<span id="cb98-12"><a href="16-Beta.html#cb98-12" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">100</span>, <span class="at">width =</span> <span class="dv">200</span>)</span>
<span id="cb98-13"><a href="16-Beta.html#cb98-13" tabindex="-1"></a><span class="co">#Display plot</span></span>
<span id="cb98-14"><a href="16-Beta.html#cb98-14" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./Beta_diversity_mds_wunifrac_media_site_zoom.png&quot;</span>)</span></code></pre></div>
<p>With the zoomed section we can more clearly see there is a difference between the CVP and KBC samples.
We can also see that the samples seem to slightly cluster by the site, especially for the KBC samples.</p>
<p>There are many other things you can do with <code>ggplot2</code> objects, too many to teach in this workshop.
How you want your plots will be highly dependant on your own data.
There are various resources in the <a href="19-Appendix.html#ggplot2_appendix">APPEDNDIX</a> to look at.</p>
</div>
<div id="beta-subset-samples" class="section level2 hasAnchor" number="16.6">
<h2><span class="header-section-number">16.6</span> Beta: subset samples<a href="16-Beta.html#beta-subset-samples" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/R_subset.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Another plot we can quickly make is to subset the <code>phyloseq</code> object to remove the ENV samples.
The ENV samples are causing the biggest difference between the samples.
To see the differences between the 3 media it will be good to remove it and produce an ordination plot.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="16-Beta.html#cb99-1" tabindex="-1"></a><span class="co">#Subset phyloseq to remove ENV samples</span></span>
<span id="cb99-2"><a href="16-Beta.html#cb99-2" tabindex="-1"></a>pseq_rarefy_no_env <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(pseq_rarefy, media <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;CVP&quot;</span>,<span class="st">&quot;KBC&quot;</span>,<span class="st">&quot;TSA&quot;</span>))</span>
<span id="cb99-3"><a href="16-Beta.html#cb99-3" tabindex="-1"></a><span class="co">#Ordinate data</span></span>
<span id="cb99-4"><a href="16-Beta.html#cb99-4" tabindex="-1"></a>ord.mds.wunifrac <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">ordinate</span>(pseq_rarefy_no_env, <span class="at">method =</span> <span class="st">&quot;MDS&quot;</span>, <span class="at">distance =</span> <span class="st">&quot;wunifrac&quot;</span>)</span>
<span id="cb99-5"><a href="16-Beta.html#cb99-5" tabindex="-1"></a><span class="co">#Plot ordination</span></span>
<span id="cb99-6"><a href="16-Beta.html#cb99-6" tabindex="-1"></a>nmds.wunifrac <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">plot_ordination</span>(pseq_rarefy_no_env, ord.mds.wunifrac,</span>
<span id="cb99-7"><a href="16-Beta.html#cb99-7" tabindex="-1"></a>                                   <span class="at">color =</span> <span class="st">&quot;site&quot;</span>, <span class="at">shape =</span> <span class="st">&quot;media&quot;</span>) <span class="sc">+</span></span>
<span id="cb99-8"><a href="16-Beta.html#cb99-8" tabindex="-1"></a>                  <span class="co">#Choose a range of the x axis to zoom into</span></span>
<span id="cb99-9"><a href="16-Beta.html#cb99-9" tabindex="-1"></a>                  <span class="co">#This was chosen after viewing the plot without the zoom</span></span>
<span id="cb99-10"><a href="16-Beta.html#cb99-10" tabindex="-1"></a>                  ggforce<span class="sc">::</span><span class="fu">facet_zoom</span>( <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.055</span>,<span class="sc">-</span><span class="fl">0.03</span>), <span class="at">zoom.size =</span> <span class="dv">1</span>)</span>
<span id="cb99-11"><a href="16-Beta.html#cb99-11" tabindex="-1"></a><span class="co">#Save ggplot2 object with ggsave</span></span>
<span id="cb99-12"><a href="16-Beta.html#cb99-12" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./Beta_diversity_mds_wunifrac_media_site_no_env.png&quot;</span>, <span class="at">plot =</span> nmds.wunifrac,</span>
<span id="cb99-13"><a href="16-Beta.html#cb99-13" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">150</span>, <span class="at">width =</span> <span class="dv">150</span>)</span>
<span id="cb99-14"><a href="16-Beta.html#cb99-14" tabindex="-1"></a><span class="co">#Display plot</span></span>
<span id="cb99-15"><a href="16-Beta.html#cb99-15" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./Beta_diversity_mds_wunifrac_media_site_no_env.png&quot;</span>)</span></code></pre></div>
<p>This has changed our plot in a few ways.
The biggest difference is that our 1st axis explains 94.7% of variation, a very large amount.
This axis separates the media groups with TSA being the most different.</p>
<p>We can clearly see separation of the sites within the media groups.
There are a couple outliers within the CVP samples.</p>
<p>Thankfully our pairwise PERMANOVA results are still valid as they use the <strong>distances</strong> which did not change compared to the <strong>ordination values</strong> which did change.</p>
</div>
<div id="permanova-tasks" class="section level2 hasAnchor" number="16.7">
<h2><span class="header-section-number">16.7</span> PERMANOVA tasks<a href="16-Beta.html#permanova-tasks" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/kettle_bell.png" width="15%" style="display: block; margin: auto;" /></p>
<p>For this task carry out pairwise PERMANOVA for the site metadata category.
I highly encourage you to copy, paste, and edit previous code you have created.</p>
<div class="webex-solution">
<button>
Site PERMANOVA solution
</button>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="16-Beta.html#cb100-1" tabindex="-1"></a><span class="co">#Pairwise comparison loop</span></span>
<span id="cb100-2"><a href="16-Beta.html#cb100-2" tabindex="-1"></a><span class="co">#Get combinations of unique site values</span></span>
<span id="cb100-3"><a href="16-Beta.html#cb100-3" tabindex="-1"></a>cbn <span class="ot">&lt;-</span> <span class="fu">combn</span>(<span class="at">x =</span> <span class="fu">unique</span>(metadf<span class="sc">$</span>site), <span class="at">m =</span> <span class="dv">2</span>)</span>
<span id="cb100-4"><a href="16-Beta.html#cb100-4" tabindex="-1"></a><span class="co">#Create empty final data frame with 4 columns</span></span>
<span id="cb100-5"><a href="16-Beta.html#cb100-5" tabindex="-1"></a><span class="co"># and a number of rows equal to the the number of combinations</span></span>
<span id="cb100-6"><a href="16-Beta.html#cb100-6" tabindex="-1"></a>pairwise_permanova_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="at">data =</span> <span class="cn">NA</span>, <span class="at">nrow =</span> <span class="fu">ncol</span>(cbn), <span class="at">ncol =</span> <span class="dv">4</span>))</span>
<span id="cb100-7"><a href="16-Beta.html#cb100-7" tabindex="-1"></a><span class="co">#Add column names</span></span>
<span id="cb100-8"><a href="16-Beta.html#cb100-8" tabindex="-1"></a><span class="fu">colnames</span>(pairwise_permanova_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;1&quot;</span>,<span class="st">&quot;2&quot;</span>,<span class="st">&quot;p&quot;</span>,<span class="st">&quot;p.adj&quot;</span>)</span>
<span id="cb100-9"><a href="16-Beta.html#cb100-9" tabindex="-1"></a></span>
<span id="cb100-10"><a href="16-Beta.html#cb100-10" tabindex="-1"></a><span class="co">#Loop trhough the combinations</span></span>
<span id="cb100-11"><a href="16-Beta.html#cb100-11" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(cbn)){</span>
<span id="cb100-12"><a href="16-Beta.html#cb100-12" tabindex="-1"></a>  <span class="co">#Subset to site phyloseq</span></span>
<span id="cb100-13"><a href="16-Beta.html#cb100-13" tabindex="-1"></a>  <span class="co">#Ensure to change metadata group if using different grouping (site)</span></span>
<span id="cb100-14"><a href="16-Beta.html#cb100-14" tabindex="-1"></a>  pseq_rarefy_subset <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(pseq_rarefy, site <span class="sc">%in%</span> cbn[,i])</span>
<span id="cb100-15"><a href="16-Beta.html#cb100-15" tabindex="-1"></a>  <span class="co">#Metadata data frame</span></span>
<span id="cb100-16"><a href="16-Beta.html#cb100-16" tabindex="-1"></a>  metadf_subset <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq_rarefy_subset))</span>
<span id="cb100-17"><a href="16-Beta.html#cb100-17" tabindex="-1"></a>  <span class="co">#Weighted unifrac distance calculation</span></span>
<span id="cb100-18"><a href="16-Beta.html#cb100-18" tabindex="-1"></a>  wunifrac_dist_mat <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(pseq_rarefy_subset, <span class="at">method=</span><span class="st">&quot;wunifrac&quot;</span>)</span>
<span id="cb100-19"><a href="16-Beta.html#cb100-19" tabindex="-1"></a>  <span class="co">#PERMANOVA/ADONIS of site</span></span>
<span id="cb100-20"><a href="16-Beta.html#cb100-20" tabindex="-1"></a>  <span class="co">#Ensure to change metadata group if using different grouping (site)</span></span>
<span id="cb100-21"><a href="16-Beta.html#cb100-21" tabindex="-1"></a>  wunifrac_pairwise_adonis <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(wunifrac_dist_mat <span class="sc">~</span> site, <span class="at">data =</span> metadf_subset, <span class="at">by =</span> <span class="st">&quot;margin&quot;</span>)</span>
<span id="cb100-22"><a href="16-Beta.html#cb100-22" tabindex="-1"></a>  <span class="co">#Add the gorup names and p-value to the main data frame</span></span>
<span id="cb100-23"><a href="16-Beta.html#cb100-23" tabindex="-1"></a>  pairwise_permanova_df[i,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">&lt;-</span> cbn[,i]</span>
<span id="cb100-24"><a href="16-Beta.html#cb100-24" tabindex="-1"></a>  pairwise_permanova_df[i,<span class="dv">3</span>] <span class="ot">&lt;-</span> wunifrac_pairwise_adonis[<span class="dv">1</span>,<span class="st">&quot;Pr(&gt;F)&quot;</span>]</span>
<span id="cb100-25"><a href="16-Beta.html#cb100-25" tabindex="-1"></a>}</span>
<span id="cb100-26"><a href="16-Beta.html#cb100-26" tabindex="-1"></a><span class="co">#Add adjusted P-values</span></span>
<span id="cb100-27"><a href="16-Beta.html#cb100-27" tabindex="-1"></a>pairwise_permanova_df<span class="sc">$</span>p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(pairwise_permanova_df<span class="sc">$</span>p, <span class="at">method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb100-28"><a href="16-Beta.html#cb100-28" tabindex="-1"></a><span class="co">#Write to a file</span></span>
<span id="cb100-29"><a href="16-Beta.html#cb100-29" tabindex="-1"></a><span class="fu">write.table</span>(<span class="at">x =</span> pairwise_permanova_df, <span class="at">file =</span> <span class="st">&quot;pairwise_permanova_site_wunifrac.tsv&quot;</span>,</span>
<span id="cb100-30"><a href="16-Beta.html#cb100-30" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb100-31"><a href="16-Beta.html#cb100-31" tabindex="-1"></a><span class="co">#View data frame</span></span>
<span id="cb100-32"><a href="16-Beta.html#cb100-32" tabindex="-1"></a>pairwise_permanova_df</span></code></pre></div>
</div>
<p>Unfortunately, we cannot detect any significant differences between the sites.
Can we detect differences between the different combinations of sites and media?
Use the <code>site.media</code> metadata category to carry out pairwise PERMANOVA tests.</p>
<div class="webex-solution">
<button>
Site and media PERMANOVA solution
</button>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="16-Beta.html#cb101-1" tabindex="-1"></a><span class="co">#Pairwise comparison loop</span></span>
<span id="cb101-2"><a href="16-Beta.html#cb101-2" tabindex="-1"></a><span class="co">#Get combinations of unique site &amp; media values</span></span>
<span id="cb101-3"><a href="16-Beta.html#cb101-3" tabindex="-1"></a>cbn <span class="ot">&lt;-</span> <span class="fu">combn</span>(<span class="at">x =</span> <span class="fu">unique</span>(metadf<span class="sc">$</span>site.media), <span class="at">m =</span> <span class="dv">2</span>)</span>
<span id="cb101-4"><a href="16-Beta.html#cb101-4" tabindex="-1"></a><span class="co">#Create empty final data frame with 4 columns</span></span>
<span id="cb101-5"><a href="16-Beta.html#cb101-5" tabindex="-1"></a><span class="co"># and a number of rows equal to the the number of combinations</span></span>
<span id="cb101-6"><a href="16-Beta.html#cb101-6" tabindex="-1"></a>pairwise_permanova_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="at">data =</span> <span class="cn">NA</span>, <span class="at">nrow =</span> <span class="fu">ncol</span>(cbn), <span class="at">ncol =</span> <span class="dv">4</span>))</span>
<span id="cb101-7"><a href="16-Beta.html#cb101-7" tabindex="-1"></a><span class="co">#Add column names</span></span>
<span id="cb101-8"><a href="16-Beta.html#cb101-8" tabindex="-1"></a><span class="fu">colnames</span>(pairwise_permanova_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;1&quot;</span>,<span class="st">&quot;2&quot;</span>,<span class="st">&quot;p&quot;</span>,<span class="st">&quot;p.adj&quot;</span>)</span>
<span id="cb101-9"><a href="16-Beta.html#cb101-9" tabindex="-1"></a></span>
<span id="cb101-10"><a href="16-Beta.html#cb101-10" tabindex="-1"></a><span class="co">#Loop trhough the combinations</span></span>
<span id="cb101-11"><a href="16-Beta.html#cb101-11" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(cbn)){</span>
<span id="cb101-12"><a href="16-Beta.html#cb101-12" tabindex="-1"></a>  <span class="co">#Subset to site &amp; media phyloseq</span></span>
<span id="cb101-13"><a href="16-Beta.html#cb101-13" tabindex="-1"></a>  <span class="co">#Ensure to change metadata group if using different grouping (site &amp; media)</span></span>
<span id="cb101-14"><a href="16-Beta.html#cb101-14" tabindex="-1"></a>  pseq_rarefy_subset <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(pseq_rarefy, site.media <span class="sc">%in%</span> cbn[,i])</span>
<span id="cb101-15"><a href="16-Beta.html#cb101-15" tabindex="-1"></a>  <span class="co">#Metadata data frame</span></span>
<span id="cb101-16"><a href="16-Beta.html#cb101-16" tabindex="-1"></a>  metadf_subset <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq_rarefy_subset))</span>
<span id="cb101-17"><a href="16-Beta.html#cb101-17" tabindex="-1"></a>  <span class="co">#Weighted unifrac distance calculation</span></span>
<span id="cb101-18"><a href="16-Beta.html#cb101-18" tabindex="-1"></a>  wunifrac_dist_mat <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(pseq_rarefy_subset, <span class="at">method=</span><span class="st">&quot;wunifrac&quot;</span>)</span>
<span id="cb101-19"><a href="16-Beta.html#cb101-19" tabindex="-1"></a>  <span class="co">#PERMANOVA/ADONIS of site &amp; media</span></span>
<span id="cb101-20"><a href="16-Beta.html#cb101-20" tabindex="-1"></a>  <span class="co">#Ensure to change metadata group if using different grouping (site &amp; media)</span></span>
<span id="cb101-21"><a href="16-Beta.html#cb101-21" tabindex="-1"></a>  wunifrac_pairwise_adonis <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(wunifrac_dist_mat <span class="sc">~</span> site.media, <span class="at">data =</span> metadf_subset, <span class="at">by =</span> <span class="st">&quot;margin&quot;</span>)</span>
<span id="cb101-22"><a href="16-Beta.html#cb101-22" tabindex="-1"></a>  <span class="co">#Add the gorup names and p-value to the main data frame</span></span>
<span id="cb101-23"><a href="16-Beta.html#cb101-23" tabindex="-1"></a>  pairwise_permanova_df[i,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">&lt;-</span> cbn[,i]</span>
<span id="cb101-24"><a href="16-Beta.html#cb101-24" tabindex="-1"></a>  pairwise_permanova_df[i,<span class="dv">3</span>] <span class="ot">&lt;-</span> wunifrac_pairwise_adonis[<span class="dv">1</span>,<span class="st">&quot;Pr(&gt;F)&quot;</span>]</span>
<span id="cb101-25"><a href="16-Beta.html#cb101-25" tabindex="-1"></a>}</span>
<span id="cb101-26"><a href="16-Beta.html#cb101-26" tabindex="-1"></a><span class="co">#Add adjusted P-values</span></span>
<span id="cb101-27"><a href="16-Beta.html#cb101-27" tabindex="-1"></a>pairwise_permanova_df<span class="sc">$</span>p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(pairwise_permanova_df<span class="sc">$</span>p, <span class="at">method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb101-28"><a href="16-Beta.html#cb101-28" tabindex="-1"></a><span class="co">#Write to a file</span></span>
<span id="cb101-29"><a href="16-Beta.html#cb101-29" tabindex="-1"></a><span class="fu">write.table</span>(<span class="at">x =</span> pairwise_permanova_df, <span class="at">file =</span> <span class="st">&quot;pairwise_permanova_site_media_wunifrac.tsv&quot;</span>,</span>
<span id="cb101-30"><a href="16-Beta.html#cb101-30" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb101-31"><a href="16-Beta.html#cb101-31" tabindex="-1"></a><span class="co">#View data frame</span></span>
<span id="cb101-32"><a href="16-Beta.html#cb101-32" tabindex="-1"></a>pairwise_permanova_df</span></code></pre></div>
</div>
<p>We are getting very high adjusted P-values for these comparisons.
Most likely there is not enough power for this analysis as there are only 3 replicates for each grouping.</p>
</div>
<div id="beta-summary" class="section level2 hasAnchor" number="16.8">
<h2><span class="header-section-number">16.8</span> Beta: Summary<a href="16-Beta.html#beta-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_orange.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We have produced Beta diversity plots and the statistics for weighted unifrac distances.
There are many other things you could do such as:</p>
<ul>
<li>Carry out beta diversity with taxa aggregated <code>phyloseq</code> objects.</li>
<li>Use different distance measures (<code>phyloseq::distanceMethodList</code>).
<ul>
<li>We have a phylogenetic tree in our <code>phyloseq</code> object which allows us to use <code>UniFrac</code> measures ("unifrac" and "wunifrac").</li>
<li>If you do not have this you will have to use non-phylogenetic aware <code>betadiver</code> distances such as "jaccard" or "bray".</li>
</ul></li>
<li>Use different different ordination methods (<code>?phyloseq:ordinate</code>).
<ul>
<li>Different methods will produce different <strong>percentage of variation explained</strong> values. If your plot has low values try another method to see if you can get higher values.</li>
<li>Many people will try out different methods and use the one with the highest <strong>percentage of variation explained</strong> values. This is perfectly acceptable and encouraged by many.</li>
</ul></li>
</ul>
<p>With our analyses we found out:</p>
<ul>
<li>The samples significantly cluster by media with ENV being the most different followed by TSA.</li>
<li>It appears that the samples cluster by site within each media group, however this is not backed up by statistics.</li>
<li>There appears to be different levels of variation within the media groups (tightness/looseness of clusters).
<ul>
<li>The CVP samples seem to have the lowest levels of variation (tight clusters), whilst the ENV samples seem to have the highest level of variation.</li>
<li>This could be explored further by <a href="https://microbiome.github.io/tutorials/PERMANOVA.html">"Checking the homogeneity condition"</a>.</li>
</ul></li>
</ul>
<p>We know there is a difference between the media groupings with the different medias causing different selections of certain taxa compared to the ENV samples.
Let's try to figure out which taxa are affected in which ways.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="15-Alpha.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="17-Differential_abundance_analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
