<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 17 Differential abundance (DA) analysis | R community analysis</title>
  <meta name="description" content="CGR book for the R community analysis workflow" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 17 Differential abundance (DA) analysis | R community analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/CGR_strips.png" />
  <meta property="og:description" content="CGR book for the R community analysis workflow" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 17 Differential abundance (DA) analysis | R community analysis" />
  
  <meta name="twitter:description" content="CGR book for the R community analysis workflow" />
  <meta name="twitter:image" content="/figures/CGR_strips.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-07-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/CGR_favicon.png" type="image/x-icon" />
<link rel="prev" href="16-Beta.html"/>
<link rel="next" href="18-Summary.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://www.liverpool.ac.uk/genomic-research/" target="blank"><img src="figures/CGR_strips.png"></a></li>
<li><a>R community analysis</a></li>

<li class="divider"></li>
<li class="part"><span><b>Intro</b></span></li>
<li class="chapter" data-level="1" data-path="01-R_community_main_workflow.html"><a href="01-R_community_main_workflow.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html"><i class="fa fa-check"></i><b>2</b> Dataset &amp; workflow</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#dataset"><i class="fa fa-check"></i><b>2.1</b> Dataset</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sites"><i class="fa fa-check"></i><b>2.1.1</b> Sites</a></li>
<li class="chapter" data-level="2.1.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#culture-media"><i class="fa fa-check"></i><b>2.1.2</b> Culture media</a></li>
<li class="chapter" data-level="2.1.3" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#sum_and_qs"><i class="fa fa-check"></i><b>2.1.3</b> Summary &amp; questions</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="02-Dataset_and_workflow.html"><a href="02-Dataset_and_workflow.html#workflow"><i class="fa fa-check"></i><b>2.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-R_packages.html"><a href="03-R_packages.html"><i class="fa fa-check"></i><b>3</b> R Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_packages.html"><a href="03-R_packages.html#r-packageslibraries"><i class="fa fa-check"></i><b>3.1</b> R packages/libraries</a></li>
<li class="chapter" data-level="3.2" data-path="03-R_packages.html"><a href="03-R_packages.html#the-grammar-of-graphics"><i class="fa fa-check"></i><b>3.2</b> The grammar of graphics</a></li>
<li class="chapter" data-level="3.3" data-path="03-R_packages.html"><a href="03-R_packages.html#phyloseq"><i class="fa fa-check"></i><b>3.3</b> phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Setup.html"><a href="04-Setup.html"><i class="fa fa-check"></i><b>4</b> Set-up</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Setup.html"><a href="04-Setup.html#cluster"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Setup.html"><a href="04-Setup.html#mamba"><i class="fa fa-check"></i><b>4.2</b> Mamba</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Jupyter.html"><a href="05-Jupyter.html"><i class="fa fa-check"></i><b>5</b> Jupyter</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Jupyter.html"><a href="05-Jupyter.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>5.1</b> Open Jupyter-notebook</a></li>
<li class="chapter" data-level="5.2" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-r-notebook"><i class="fa fa-check"></i><b>5.2</b> Create R notebook</a></li>
<li class="chapter" data-level="5.3" data-path="05-Jupyter.html"><a href="05-Jupyter.html#cells-and-code"><i class="fa fa-check"></i><b>5.3</b> Cells and code</a></li>
<li class="chapter" data-level="5.4" data-path="05-Jupyter.html"><a href="05-Jupyter.html#create-new-cells"><i class="fa fa-check"></i><b>5.4</b> Create new cells</a></li>
<li class="chapter" data-level="5.5" data-path="05-Jupyter.html"><a href="05-Jupyter.html#running-code"><i class="fa fa-check"></i><b>5.5</b> Running code</a></li>
<li class="chapter" data-level="5.6" data-path="05-Jupyter.html"><a href="05-Jupyter.html#saving-the-file"><i class="fa fa-check"></i><b>5.6</b> Saving the file</a></li>
<li class="chapter" data-level="5.7" data-path="05-Jupyter.html"><a href="05-Jupyter.html#title-cells-with-markdown"><i class="fa fa-check"></i><b>5.7</b> Title cells with markdown</a></li>
<li class="chapter" data-level="5.8" data-path="05-Jupyter.html"><a href="05-Jupyter.html#close-the-notebook"><i class="fa fa-check"></i><b>5.8</b> Close the notebook</a></li>
<li class="chapter" data-level="5.9" data-path="05-Jupyter.html"><a href="05-Jupyter.html#video-tutorial"><i class="fa fa-check"></i><b>5.9</b> Video tutorial</a></li>
</ul></li>
<li class="part"><span><b>Data preparation</b></span></li>
<li class="chapter" data-level="6" data-path="06-Preprocess_part.html"><a href="06-Preprocess_part.html"><i class="fa fa-check"></i><b>6</b> Data prep intro</a></li>
<li class="chapter" data-level="7" data-path="07-Import.html"><a href="07-Import.html"><i class="fa fa-check"></i><b>7</b> Import</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Import.html"><a href="07-Import.html#import-notebook"><i class="fa fa-check"></i><b>7.1</b> Import: notebook</a></li>
<li class="chapter" data-level="7.2" data-path="07-Import.html"><a href="07-Import.html#qiime2r"><i class="fa fa-check"></i><b>7.2</b> qiime2R</a></li>
<li class="chapter" data-level="7.3" data-path="07-Import.html"><a href="07-Import.html#import-summarise-phyloseq"><i class="fa fa-check"></i><b>7.3</b> Import: Summarise phyloseq</a></li>
<li class="chapter" data-level="7.4" data-path="07-Import.html"><a href="07-Import.html#save-the-phyloseq-object"><i class="fa fa-check"></i><b>7.4</b> Save the phyloseq object</a></li>
<li class="chapter" data-level="7.5" data-path="07-Import.html"><a href="07-Import.html#import-recap"><i class="fa fa-check"></i><b>7.5</b> Import: recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html"><i class="fa fa-check"></i><b>8</b> Summarise phyloseq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-setup"><i class="fa fa-check"></i><b>8.1</b> Summarise: setup</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-header"><i class="fa fa-check"></i><b>8.1.1</b> Summarise header</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-phyloseq"><i class="fa fa-check"></i><b>8.1.2</b> Summarise phyloseq</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#reads-per-sample"><i class="fa fa-check"></i><b>8.1.3</b> Reads per sample</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#display-png"><i class="fa fa-check"></i><b>8.1.4</b> Display png</a></li>
<li class="chapter" data-level="8.1.5" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#asvs-per-sample"><i class="fa fa-check"></i><b>8.1.5</b> ASVs per sample</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Summarise_phyloseq.html"><a href="08-Summarise_phyloseq.html#summarise-recap"><i class="fa fa-check"></i><b>8.2</b> Summarise: recap</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html"><i class="fa fa-check"></i><b>9</b> Minimum read depth</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#considerations"><i class="fa fa-check"></i><b>9.1</b> Considerations</a></li>
<li class="chapter" data-level="9.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minimum-read-depth-section-title"><i class="fa fa-check"></i><b>9.2</b> Minimum read depth section title</a></li>
<li class="chapter" data-level="9.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#read-depth-vector-and-histogram"><i class="fa fa-check"></i><b>9.3</b> Read depth vector and histogram</a></li>
<li class="chapter" data-level="9.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#depth-boxplots"><i class="fa fa-check"></i><b>9.4</b> Sample depth boxplot</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#creating-a-data-frame-for-boxplots"><i class="fa fa-check"></i><b>9.4.1</b> Creating a data frame for boxplots</a></li>
<li class="chapter" data-level="9.4.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#ggplot2-boxplot"><i class="fa fa-check"></i><b>9.4.2</b> ggplot2 boxplot</a></li>
<li class="chapter" data-level="9.4.3" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#save-ggplot-as-png"><i class="fa fa-check"></i><b>9.4.3</b> Save ggplot as png</a></li>
<li class="chapter" data-level="9.4.4" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#media-by-depth-boxplot"><i class="fa fa-check"></i><b>9.4.4</b> Media by depth boxplot</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#chap9rarefaction"><i class="fa fa-check"></i><b>9.5</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#asv-abundance-data-frame"><i class="fa fa-check"></i><b>9.5.1</b> ASV abundance data frame</a></li>
<li class="chapter" data-level="9.5.2" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#vegans-rarecurve"><i class="fa fa-check"></i><b>9.5.2</b> vegan's rarecurve</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#filtering-by-minimum-read-depth"><i class="fa fa-check"></i><b>9.6</b> Filtering by minimum read depth</a></li>
<li class="chapter" data-level="9.7" data-path="09-Minimum_read_depth.html"><a href="09-Minimum_read_depth.html#minmum-read-depth-summary"><i class="fa fa-check"></i><b>9.7</b> Minmum read depth: Summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy</b></span></li>
<li class="chapter" data-level="10" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html"><i class="fa fa-check"></i><b>10</b> Taxa relative abundance</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-setup"><i class="fa fa-check"></i><b>10.1</b> Taxa relative abundance: setup</a></li>
<li class="chapter" data-level="10.2" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#relative-abundance-transformation"><i class="fa fa-check"></i><b>10.2</b> Relative abundance transformation</a></li>
<li class="chapter" data-level="10.3" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#aggregate-taxa"><i class="fa fa-check"></i><b>10.3</b> Aggregate taxa</a></li>
<li class="chapter" data-level="10.4" data-path="10-Relative_abundance.html"><a href="10-Relative_abundance.html#taxa-relative-abundance-summary"><i class="fa fa-check"></i><b>10.4</b> Taxa Relative abundance: summary</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html"><i class="fa fa-check"></i><b>11</b> Taxa plots</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#simple-bar-chart"><i class="fa fa-check"></i><b>11.1</b> Simple bar chart</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#editing-the-plot"><i class="fa fa-check"></i><b>11.1.1</b> Editing the plot</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-heatmap"><i class="fa fa-check"></i><b>11.2</b> Taxa heatmap</a></li>
<li class="chapter" data-level="11.3" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#aggregateraretaxa"><i class="fa fa-check"></i><b>11.3</b> Aggregate rare taxa</a></li>
<li class="chapter" data-level="11.4" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-media"><i class="fa fa-check"></i><b>11.4</b> Plot by media</a></li>
<li class="chapter" data-level="11.5" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#plot-by-site-and-media"><i class="fa fa-check"></i><b>11.5</b> Plot by site and media</a></li>
<li class="chapter" data-level="11.6" data-path="11-Taxa_plots.html"><a href="11-Taxa_plots.html#taxa-plots-summary"><i class="fa fa-check"></i><b>11.6</b> Taxa plots: summary</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html"><i class="fa fa-check"></i><b>12</b> Family &amp; Genus plots</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-taxa-plots"><i class="fa fa-check"></i><b>12.1</b> Family taxa plots</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#aggregate-families"><i class="fa fa-check"></i><b>12.1.1</b> Aggregate families</a></li>
<li class="chapter" data-level="12.1.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-heatmap"><i class="fa fa-check"></i><b>12.1.2</b> Family heatmap</a></li>
<li class="chapter" data-level="12.1.3" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-rare-aggregation"><i class="fa fa-check"></i><b>12.1.3</b> Family rare aggregation</a></li>
<li class="chapter" data-level="12.1.4" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#family-bar-chart"><i class="fa fa-check"></i><b>12.1.4</b> Family bar chart</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots"><i class="fa fa-check"></i><b>12.2</b> Genus taxa plots</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="12-Family_and_Genus_taxa.html"><a href="12-Family_and_Genus_taxa.html#genus-taxa-plots-solutions"><i class="fa fa-check"></i><b>12.2.1</b> Genus taxa plots solutions</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Diversity analysis</b></span></li>
<li class="chapter" data-level="13" data-path="13-Diversity_analysis.html"><a href="13-Diversity_analysis.html"><i class="fa fa-check"></i><b>13</b> Diversity analysis intro</a></li>
<li class="chapter" data-level="14" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html"><i class="fa fa-check"></i><b>14</b> Rarefaction</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-setup"><i class="fa fa-check"></i><b>14.1</b> Rarefaction: setup</a></li>
<li class="chapter" data-level="14.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve"><i class="fa fa-check"></i><b>14.2</b> Rarefaction curve</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#bird-analogy"><i class="fa fa-check"></i><b>14.2.1</b> Bird analogy</a></li>
<li class="chapter" data-level="14.2.2" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-simple-plot"><i class="fa fa-check"></i><b>14.2.2</b> Rarefaction curve: simple plot</a></li>
<li class="chapter" data-level="14.2.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-curve-better-plot"><i class="fa fa-check"></i><b>14.2.3</b> Rarefaction curve: better plot</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-slope"><i class="fa fa-check"></i><b>14.3</b> Rarefaction slope</a></li>
<li class="chapter" data-level="14.4" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefy-data"><i class="fa fa-check"></i><b>14.4</b> Rarefy data</a></li>
<li class="chapter" data-level="14.5" data-path="14-Rarefaction.html"><a href="14-Rarefaction.html#rarefaction-summary"><i class="fa fa-check"></i><b>14.5</b> Rarefaction: summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Alpha.html"><a href="15-Alpha.html"><i class="fa fa-check"></i><b>15</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-setup"><i class="fa fa-check"></i><b>15.1</b> Alpha: setup</a></li>
<li class="chapter" data-level="15.2" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-simple-plot"><i class="fa fa-check"></i><b>15.2</b> Alpha: simple plot</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="15-Alpha.html"><a href="15-Alpha.html#metric-choice"><i class="fa fa-check"></i><b>15.2.1</b> Metric choice</a></li>
<li class="chapter" data-level="15.2.2" data-path="15-Alpha.html"><a href="15-Alpha.html#sample-grouping-box-plot"><i class="fa fa-check"></i><b>15.2.2</b> Sample grouping &amp; box plot</a></li>
<li class="chapter" data-level="15.2.3" data-path="15-Alpha.html"><a href="15-Alpha.html#png-height-and-width-choice"><i class="fa fa-check"></i><b>15.2.3</b> PNG height and width choice</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats"><i class="fa fa-check"></i><b>15.3</b> Alpha: stats</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-stats-task"><i class="fa fa-check"></i><b>15.3.1</b> Alpha stats task</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-diversity-violin-plot"><i class="fa fa-check"></i><b>15.4</b> Alpha diversity: violin plot</a></li>
<li class="chapter" data-level="15.5" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-site-plots-and-statistics"><i class="fa fa-check"></i><b>15.5</b> Alpha: Site plots and statistics</a></li>
<li class="chapter" data-level="15.6" data-path="15-Alpha.html"><a href="15-Alpha.html#alpha-summary"><i class="fa fa-check"></i><b>15.6</b> Alpha: summary</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Beta.html"><a href="16-Beta.html"><i class="fa fa-check"></i><b>16</b> Beta</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Beta.html"><a href="16-Beta.html#beta-setup"><i class="fa fa-check"></i><b>16.1</b> Beta: setup</a></li>
<li class="chapter" data-level="16.2" data-path="16-Beta.html"><a href="16-Beta.html#beta-ordination"><i class="fa fa-check"></i><b>16.2</b> Beta: ordination</a></li>
<li class="chapter" data-level="16.3" data-path="16-Beta.html"><a href="16-Beta.html#beta-plot-ordination"><i class="fa fa-check"></i><b>16.3</b> Beta: plot ordination</a></li>
<li class="chapter" data-level="16.4" data-path="16-Beta.html"><a href="16-Beta.html#beta-statistics"><i class="fa fa-check"></i><b>16.4</b> Beta: statistics</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="16-Beta.html"><a href="16-Beta.html#permanova"><i class="fa fa-check"></i><b>16.4.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="16.4.2" data-path="16-Beta.html"><a href="16-Beta.html#pairwise-permanova"><i class="fa fa-check"></i><b>16.4.2</b> Pairwise PERMANOVA</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="16-Beta.html"><a href="16-Beta.html#beta-facet-zoom"><i class="fa fa-check"></i><b>16.5</b> Beta: Facet zoom</a></li>
<li class="chapter" data-level="16.6" data-path="16-Beta.html"><a href="16-Beta.html#beta-subset-samples"><i class="fa fa-check"></i><b>16.6</b> Beta: subset samples</a></li>
<li class="chapter" data-level="16.7" data-path="16-Beta.html"><a href="16-Beta.html#permanova-tasks"><i class="fa fa-check"></i><b>16.7</b> PERMANOVA tasks</a></li>
<li class="chapter" data-level="16.8" data-path="16-Beta.html"><a href="16-Beta.html#beta-summary"><i class="fa fa-check"></i><b>16.8</b> Beta: Summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html"><i class="fa fa-check"></i><b>17</b> Differential abundance (DA) analysis</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-setup"><i class="fa fa-check"></i><b>17.1</b> DA: setup</a></li>
<li class="chapter" data-level="17.2" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-preprocess"><i class="fa fa-check"></i><b>17.2</b> DA: Preprocess</a></li>
<li class="chapter" data-level="17.3" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-analysis"><i class="fa fa-check"></i><b>17.3</b> DA: ANCOM-BC analysis</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#ancombc2-options"><i class="fa fa-check"></i><b>17.3.1</b> ANCOMBC2 options</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-ancom-bc-results"><i class="fa fa-check"></i><b>17.4</b> DA: ANCOM-BC results</a></li>
<li class="chapter" data-level="17.5" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-create-signifcant-lfc-long-data-frame"><i class="fa fa-check"></i><b>17.5</b> DA: Create signifcant LFC long data frame</a></li>
<li class="chapter" data-level="17.6" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-biomarker-lfc-heatmap"><i class="fa fa-check"></i><b>17.6</b> DA: Biomarker LFC heatmap</a></li>
<li class="chapter" data-level="17.7" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-matching-bar-plot"><i class="fa fa-check"></i><b>17.7</b> DA: matching bar plot</a></li>
<li class="chapter" data-level="17.8" data-path="17-Differential_abundance_analysis.html"><a href="17-Differential_abundance_analysis.html#da-summary"><i class="fa fa-check"></i><b>17.8</b> DA: summary</a></li>
</ul></li>
<li class="part"><span><b>Outro</b></span></li>
<li class="chapter" data-level="18" data-path="18-Summary.html"><a href="18-Summary.html"><i class="fa fa-check"></i><b>18</b> Summary</a>
<ul>
<li class="chapter" data-level="18.1" data-path="18-Summary.html"><a href="18-Summary.html#dataset-1"><i class="fa fa-check"></i><b>18.1</b> Dataset</a></li>
<li class="chapter" data-level="18.2" data-path="18-Summary.html"><a href="18-Summary.html#data-preparation"><i class="fa fa-check"></i><b>18.2</b> Data preparation</a></li>
<li class="chapter" data-level="18.3" data-path="18-Summary.html"><a href="18-Summary.html#taxonomy"><i class="fa fa-check"></i><b>18.3</b> Taxonomy</a></li>
<li class="chapter" data-level="18.4" data-path="18-Summary.html"><a href="18-Summary.html#diversity-analysis"><i class="fa fa-check"></i><b>18.4</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="18-Summary.html"><a href="18-Summary.html#alpha-beta"><i class="fa fa-check"></i><b>18.4.1</b> Alpha &amp; beta</a></li>
<li class="chapter" data-level="18.4.2" data-path="18-Summary.html"><a href="18-Summary.html#differential-abundance-anlaysis"><i class="fa fa-check"></i><b>18.4.2</b> Differential abundance anlaysis</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="18-Summary.html"><a href="18-Summary.html#findings"><i class="fa fa-check"></i><b>18.5</b> Findings</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="18-Summary.html"><a href="18-Summary.html#anthropisation-gradient"><i class="fa fa-check"></i><b>18.5.1</b> Anthropisation gradient</a></li>
<li class="chapter" data-level="18.5.2" data-path="18-Summary.html"><a href="18-Summary.html#replicate-differences"><i class="fa fa-check"></i><b>18.5.2</b> Replicate differences</a></li>
<li class="chapter" data-level="18.5.3" data-path="18-Summary.html"><a href="18-Summary.html#site-or-media"><i class="fa fa-check"></i><b>18.5.3</b> Site or media</a></li>
<li class="chapter" data-level="18.5.4" data-path="18-Summary.html"><a href="18-Summary.html#media-versus-env-samples"><i class="fa fa-check"></i><b>18.5.4</b> Media versus ENV samples</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="18-Summary.html"><a href="18-Summary.html#conclusion"><i class="fa fa-check"></i><b>18.6</b> Conclusion</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="19-Appendix.html"><a href="19-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="19-Appendix.html"><a href="19-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
<li class="chapter" data-level="A.2" data-path="19-Appendix.html"><a href="19-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>A.2</b> Jupyter-notebook</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="19-Appendix.html"><a href="19-Appendix.html#r-packages-1"><i class="fa fa-check"></i><b>B</b> R packages</a>
<ul>
<li class="chapter" data-level="B.1" data-path="19-Appendix.html"><a href="19-Appendix.html#ggplot2_appendix"><i class="fa fa-check"></i><b>B.1</b> ggplot2</a></li>
<li class="chapter" data-level="B.2" data-path="19-Appendix.html"><a href="19-Appendix.html#phyloseq_appendix"><i class="fa fa-check"></i><b>B.2</b> Phyloseq</a></li>
<li class="chapter" data-level="B.3" data-path="19-Appendix.html"><a href="19-Appendix.html#importing-into-phyloseq"><i class="fa fa-check"></i><b>B.3</b> Importing into phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-metrics"><i class="fa fa-check"></i><b>C</b> Diversity metrics</a>
<ul>
<li class="chapter" data-level="C.1" data-path="19-Appendix.html"><a href="19-Appendix.html#alpha_appendix"><i class="fa fa-check"></i><b>C.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="19-Appendix.html"><a href="19-Appendix.html#obvs"><i class="fa fa-check"></i><b>C.1.1</b> Observed features</a></li>
<li class="chapter" data-level="C.1.2" data-path="19-Appendix.html"><a href="19-Appendix.html#chao"><i class="fa fa-check"></i><b>C.1.2</b> Chao</a></li>
<li class="chapter" data-level="C.1.3" data-path="19-Appendix.html"><a href="19-Appendix.html#evenness"><i class="fa fa-check"></i><b>C.1.3</b> Evenness</a></li>
<li class="chapter" data-level="C.1.4" data-path="19-Appendix.html"><a href="19-Appendix.html#faith"><i class="fa fa-check"></i><b>C.1.4</b> Faith's PD (phylogenetic diversity)</a></li>
<li class="chapter" data-level="C.1.5" data-path="19-Appendix.html"><a href="19-Appendix.html#simpson"><i class="fa fa-check"></i><b>C.1.5</b> Simpson</a></li>
<li class="chapter" data-level="C.1.6" data-path="19-Appendix.html"><a href="19-Appendix.html#simpsone"><i class="fa fa-check"></i><b>C.1.6</b> Simpson evenness measure (simpson_e)</a></li>
<li class="chapter" data-level="C.1.7" data-path="19-Appendix.html"><a href="19-Appendix.html#shannon"><i class="fa fa-check"></i><b>C.1.7</b> Shannon</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="19-Appendix.html"><a href="19-Appendix.html#beta_appendix"><i class="fa fa-check"></i><b>C.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="19-Appendix.html"><a href="19-Appendix.html#unifrac_explanation"><i class="fa fa-check"></i><b>C.2.1</b> Weighted and unweighted UniFrac distances</a></li>
<li class="chapter" data-level="C.2.2" data-path="19-Appendix.html"><a href="19-Appendix.html#bray_explanation"><i class="fa fa-check"></i><b>C.2.2</b> Bray-Curtis</a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="19-Appendix.html"><a href="19-Appendix.html#diversity-resources"><i class="fa fa-check"></i><b>C.3</b> Diversity Resources</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R community analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="DA_chap" class="section level1 hasAnchor" number="17">
<h1><span class="header-section-number">Chapter 17</span> Differential abundance (DA) analysis<a href="17-Differential_abundance_analysis.html#DA_chap" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/difference.png" width="20%" style="display: block; margin: auto;" /></p>
<p>We are going to carry out differential abundance analysis with <a href="https://www.nature.com/articles/s41467-020-17041-7">ANCOM-BC (Analysis of compositions of microbiomes with bias correction)</a>.
This is an improvement of the original ANCOM method but with added bias correction.
This bias correction aims to account for the bias caused by the difference in sampling/depth across the samples.</p>
<p>Differential abundance analysis aims to detect features (ASVs, taxa, etc.) that have different abundances between groups.
This can be used to detect organisms in high abundance in diseased states versus healthy states.
This helps determine what are the disease causing organisms.
However, it may also detect organisms that are able to survive in the environment rather than those that cause the difference in environment.</p>
<p>This will be shown in our data.
Organisms in higher abundance in the media samples compared to the environmental samples are most likely caused by selection pressures the media introduces.
To keep our data manageable in a workshop we will look at the phyla abundances.
If a phyla is detected to have higher abundance in a media compared to the environmental sample, we can say the phyla is a biomarker of the media compared to the environmental sample.
Additionally, we will be able to detect phyla that have been depleted in the media samples versus the environmental sample.</p>
<div id="da-setup" class="section level2 hasAnchor" number="17.1">
<h2><span class="header-section-number">17.1</span> DA: setup<a href="17-Differential_abundance_analysis.html#da-setup" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/packages_1.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We will use a new notebook called <strong>"7-differential_abundance.ipynb"</strong>.
Add the following to the top of this new notebook to load the required libraries/packages and data.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="17-Differential_abundance_analysis.html#cb105-1" tabindex="-1"></a><span class="co">#Packages</span></span>
<span id="cb105-2"><a href="17-Differential_abundance_analysis.html#cb105-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb105-3"><a href="17-Differential_abundance_analysis.html#cb105-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;microbiome&quot;</span>)</span>
<span id="cb105-4"><a href="17-Differential_abundance_analysis.html#cb105-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;IRdisplay&quot;</span>)</span>
<span id="cb105-5"><a href="17-Differential_abundance_analysis.html#cb105-5" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ANCOMBC&quot;</span>)</span>
<span id="cb105-6"><a href="17-Differential_abundance_analysis.html#cb105-6" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;DT&quot;</span>)</span>
<span id="cb105-7"><a href="17-Differential_abundance_analysis.html#cb105-7" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;dplyr&quot;</span>)</span>
<span id="cb105-8"><a href="17-Differential_abundance_analysis.html#cb105-8" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;tidyr&quot;</span>)</span>
<span id="cb105-9"><a href="17-Differential_abundance_analysis.html#cb105-9" tabindex="-1"></a><span class="co">#Load phyloseq object</span></span>
<span id="cb105-10"><a href="17-Differential_abundance_analysis.html#cb105-10" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;phyloseq.RData&quot;</span>)</span></code></pre></div>
<p>We will be using the package <a href="https://github.com/FrederickHuangLin/ANCOM-BC-Code-Archive"><code>ANCOMBC</code></a> to carry out our DA analysis.
Instead of using our rarefied data we will use our original abundance data.
This is because ANCOM-BC uses absolute abundances and corrects for differences between samples.
This is a statistically heavy method and I recommend you read the <a href="https://www.nature.com/articles/s41467-020-17041-7">ANCOM-BC paper</a> if you are interested in how it works.
As bioinformaticians we are more interested in the output rather than the method.</p>
</div>
<div id="da-preprocess" class="section level2 hasAnchor" number="17.2">
<h2><span class="header-section-number">17.2</span> DA: Preprocess<a href="17-Differential_abundance_analysis.html#da-preprocess" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/water_surface.png" width="15%" style="display: block; margin: auto;" /></p>
<p>When carrying out ANCOM analysis you need to specify the reference group for the groupings you use.
The reference group will be the 1st level of the factor.
Therefore we will check our level orders for our media and site metadata categories.</p>
<div class="webex-solution">
<button>
What are factors?
</button>
<p>For an explanation on the factors class please see the following section in the R primer omics book:
<a href="https://neof-workshops.github.io/R_j4c0xh/14-Plots_scatterplot_and_box_plots.html#factors">Factors</a></p>
</div>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="17-Differential_abundance_analysis.html#cb106-1" tabindex="-1"></a><span class="co">#Check media and site levels</span></span>
<span id="cb106-2"><a href="17-Differential_abundance_analysis.html#cb106-2" tabindex="-1"></a><span class="fu">levels</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq)<span class="sc">$</span>media)</span>
<span id="cb106-3"><a href="17-Differential_abundance_analysis.html#cb106-3" tabindex="-1"></a><span class="fu">levels</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq)<span class="sc">$</span>site)</span></code></pre></div>
<p>The levels are not in the order we would like.
We'll therefore change the order with the <code>factor()</code> function and <code>levels =</code> option.</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="17-Differential_abundance_analysis.html#cb107-1" tabindex="-1"></a><span class="co">#Ensure ENV is our reference level for media</span></span>
<span id="cb107-2"><a href="17-Differential_abundance_analysis.html#cb107-2" tabindex="-1"></a><span class="fu">sample_data</span>(pseq)<span class="sc">$</span>media <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">sample_data</span>(pseq)<span class="sc">$</span>media,</span>
<span id="cb107-3"><a href="17-Differential_abundance_analysis.html#cb107-3" tabindex="-1"></a>                                  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;ENV&quot;</span>, <span class="st">&quot;CVP&quot;</span>, <span class="st">&quot;KBC&quot;</span>, <span class="st">&quot;TSA&quot;</span>))</span>
<span id="cb107-4"><a href="17-Differential_abundance_analysis.html#cb107-4" tabindex="-1"></a><span class="fu">levels</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq)<span class="sc">$</span>media)</span>
<span id="cb107-5"><a href="17-Differential_abundance_analysis.html#cb107-5" tabindex="-1"></a><span class="co">#Ensure UD is our reference level for site</span></span>
<span id="cb107-6"><a href="17-Differential_abundance_analysis.html#cb107-6" tabindex="-1"></a><span class="fu">sample_data</span>(pseq)<span class="sc">$</span>site <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">sample_data</span>(pseq)<span class="sc">$</span>site,</span>
<span id="cb107-7"><a href="17-Differential_abundance_analysis.html#cb107-7" tabindex="-1"></a>                                  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;UD&quot;</span>, <span class="st">&quot;MD&quot;</span>, <span class="st">&quot;LD&quot;</span>))</span>
<span id="cb107-8"><a href="17-Differential_abundance_analysis.html#cb107-8" tabindex="-1"></a><span class="fu">levels</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(pseq)<span class="sc">$</span>site)</span></code></pre></div>
</div>
<div id="da-ancom-bc-analysis" class="section level2 hasAnchor" number="17.3">
<h2><span class="header-section-number">17.3</span> DA: ANCOM-BC analysis<a href="17-Differential_abundance_analysis.html#da-ancom-bc-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/play_blue.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Now that we have our data the way we would like it we can carry out the ANCOM analysis.
We'll use the function <code>ANCOMBC::ancombc2()</code> for our ANCOM-BC analysis.</p>
<p><strong>Note:</strong> This command will take a few minutes to run.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="17-Differential_abundance_analysis.html#cb108-1" tabindex="-1"></a><span class="co">#Run ANCOMBC</span></span>
<span id="cb108-2"><a href="17-Differential_abundance_analysis.html#cb108-2" tabindex="-1"></a>ancom_class_out <span class="ot">&lt;-</span> ANCOMBC<span class="sc">::</span><span class="fu">ancombc2</span>(<span class="at">data =</span> pseq, <span class="at">fix_formula =</span> <span class="st">&quot;media + site&quot;</span>,</span>
<span id="cb108-3"><a href="17-Differential_abundance_analysis.html#cb108-3" tabindex="-1"></a>                                     <span class="at">p_adj_method =</span> <span class="st">&quot;holm&quot;</span>,</span>
<span id="cb108-4"><a href="17-Differential_abundance_analysis.html#cb108-4" tabindex="-1"></a>                                     <span class="at">tax_level =</span> <span class="st">&quot;Class&quot;</span>)</span></code></pre></div>
<p>The resulting object is very large so it can be a good idea to save it.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="17-Differential_abundance_analysis.html#cb109-1" tabindex="-1"></a><span class="co">#Save results</span></span>
<span id="cb109-2"><a href="17-Differential_abundance_analysis.html#cb109-2" tabindex="-1"></a><span class="fu">save</span>(ancom_class_out, <span class="at">file =</span> <span class="st">&quot;ANCOMBC2_class_analysis.RData&quot;</span>)</span></code></pre></div>
<div id="ancombc2-options" class="section level3 hasAnchor" number="17.3.1">
<h3><span class="header-section-number">17.3.1</span> ANCOMBC2 options<a href="17-Differential_abundance_analysis.html#ancombc2-options" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/parameter_blue.png" width="10%" style="display: block; margin: auto;" /></p>
<p>The options we use in the function are:</p>
<ul>
<li><code>data =</code>: The data to be analysed. This can be provided in various types of objects including the <code>phyloseq</code> object.</li>
<li><code>fix_formula =</code>: The groupings within your data. It is importnat to add all groupings that may have an influence on the microbial abundances.</li>
<li><code>p_adj_method =</code>: The p adjustment value to use.</li>
<li><code>tax_level =</code>: Te taxonomic level of interest.</li>
</ul>
<p>There are many more options that have been left as the default.
Please see the package <a href="https://bioc.ism.ac.jp/packages/devel/bioc/manuals/ANCOMBC/man/ANCOMBC.pdf">PDF</a> for more options and explanations.</p>
</div>
</div>
<div id="da-ancom-bc-results" class="section level2 hasAnchor" number="17.4">
<h2><span class="header-section-number">17.4</span> DA: ANCOM-BC results<a href="17-Differential_abundance_analysis.html#da-ancom-bc-results" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/output_file_blue.png" width="10%" style="display: block; margin: auto;" /></p>
<p>There is a lot of information in <code>ancom_class_out</code>.
Therefore, lets extract the results we want to look at and then remove the large object.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="17-Differential_abundance_analysis.html#cb110-1" tabindex="-1"></a><span class="co">#Output of analysis compared to reference groups</span></span>
<span id="cb110-2"><a href="17-Differential_abundance_analysis.html#cb110-2" tabindex="-1"></a>res_ancom_class <span class="ot">&lt;-</span> ancom_class_out<span class="sc">$</span>res</span>
<span id="cb110-3"><a href="17-Differential_abundance_analysis.html#cb110-3" tabindex="-1"></a><span class="co">#Remove large object</span></span>
<span id="cb110-4"><a href="17-Differential_abundance_analysis.html#cb110-4" tabindex="-1"></a><span class="fu">rm</span>(ancom_class_out)</span>
<span id="cb110-5"><a href="17-Differential_abundance_analysis.html#cb110-5" tabindex="-1"></a><span class="co">#Check top rows of results</span></span>
<span id="cb110-6"><a href="17-Differential_abundance_analysis.html#cb110-6" tabindex="-1"></a><span class="fu">head</span>(res_ancom_class)</span></code></pre></div>
<p>These reuslts only contain taxon where a significant result was found in at least one comparison.
This is based on the adjusted p-values (&lt;0.05).
There are a lot of columns (37) in our result data frame.
Apart from taxon they are split by the comparison group.
These contain the following information:</p>
<ul>
<li>taxon: The taxa groups found to be a biomarker in at least one comparison.</li>
<li>lfc_*: This contains the log (natural log) fold changes with respect to the reference group.
<ul>
<li>For example The values for lfc_mediaCVP would equal LFC(CVP - ENV).</li>
<li>A positive value means the abundances are higher in the comparison group (CVP) and lower in the ref (ENV).</li>
<li>A negative value means the abundances are lower in the comparison group (CVP) and higher in the ref (ENV).</li>
<li>The intercept value is the grand mean and is likely not of interest.</li>
</ul></li>
<li>se_*: The standard error for the LFC values.</li>
<li>W_*: The W statistic. The higher this value the more significant the feature is differentially abundant between groups.</li>
<li>p_*: The P-value.</li>
<li>q_*: The Q-value. This is the adjusted P-value.</li>
<li>diff*: This indicates whether there is a significant difference in the abundances of the comparison and reference group. In other words, is the Q-value &lt; 0.05.</li>
</ul>
<p>It is quite a large table so let's save it as a tsv (tab separated value) file.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="17-Differential_abundance_analysis.html#cb111-1" tabindex="-1"></a><span class="co">#Save as file</span></span>
<span id="cb111-2"><a href="17-Differential_abundance_analysis.html#cb111-2" tabindex="-1"></a><span class="fu">write.table</span>(<span class="at">x =</span> res_ancom_class, <span class="at">file =</span> <span class="st">&quot;ANCOMBC2_class_results.tsv&quot;</span>,</span>
<span id="cb111-3"><a href="17-Differential_abundance_analysis.html#cb111-3" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>This is good to open in excel but we don't have something like that on our cluster.
Therefore, we will use the package <code>DT</code> to create an interactive table with the function <code>datatable()</code>.
We can then save this object as an HTML with <code>htmlwidgets::saveWidget()</code>.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="17-Differential_abundance_analysis.html#cb112-1" tabindex="-1"></a><span class="co">#Create data table</span></span>
<span id="cb112-2"><a href="17-Differential_abundance_analysis.html#cb112-2" tabindex="-1"></a>m <span class="ot">&lt;-</span> DT<span class="sc">::</span><span class="fu">datatable</span>(res_ancom_class)</span>
<span id="cb112-3"><a href="17-Differential_abundance_analysis.html#cb112-3" tabindex="-1"></a><span class="co">#Save data table as html</span></span>
<span id="cb112-4"><a href="17-Differential_abundance_analysis.html#cb112-4" tabindex="-1"></a>htmlwidgets<span class="sc">::</span><span class="fu">saveWidget</span>(m, <span class="st">&quot;ANCOM_class_results.html&quot;</span>, <span class="at">selfcontained =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Go to the jupyter-notebook file explorer tab and click on the html file to open it in a new tab.
This gives a relatively nice way to view all the results.</p>
<p>One very clear pattern is that there are no biomarkers found for MD and LD.
I believe this is because we don't have enough replicates for the sites within the media samples and the differences between the medias are so large.
However, there is a good amount of biomarkers for the three different medias compared to ENV.</p>
</div>
<div id="da-create-signifcant-lfc-long-data-frame" class="section level2 hasAnchor" number="17.5">
<h2><span class="header-section-number">17.5</span> DA: Create signifcant LFC long data frame<a href="17-Differential_abundance_analysis.html#da-create-signifcant-lfc-long-data-frame" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/portrait.png" width="10%" style="display: block; margin: auto;" /></p>
<p>The results table is good but let's make a heatmap showing the LFCs (Log fold changes) of the significant biomarkers. First, we need to do some preprecessing.</p>
<p>Remove columns unrelated to the media comparisons.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="17-Differential_abundance_analysis.html#cb113-1" tabindex="-1"></a><span class="co">#Extract the taxon column and the media columns from the results</span></span>
<span id="cb113-2"><a href="17-Differential_abundance_analysis.html#cb113-2" tabindex="-1"></a><span class="co">#get positions of columns which contain &quot;media&quot;</span></span>
<span id="cb113-3"><a href="17-Differential_abundance_analysis.html#cb113-3" tabindex="-1"></a>media_colnames_pos <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;media&quot;</span>, <span class="fu">colnames</span>(res_ancom_class))</span>
<span id="cb113-4"><a href="17-Differential_abundance_analysis.html#cb113-4" tabindex="-1"></a><span class="co">#Taxon column and media column names</span></span>
<span id="cb113-5"><a href="17-Differential_abundance_analysis.html#cb113-5" tabindex="-1"></a>media_colnames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(res_ancom_class)[<span class="fu">c</span>(<span class="dv">1</span>,media_colnames_pos)]</span>
<span id="cb113-6"><a href="17-Differential_abundance_analysis.html#cb113-6" tabindex="-1"></a>media_colnames</span>
<span id="cb113-7"><a href="17-Differential_abundance_analysis.html#cb113-7" tabindex="-1"></a><span class="co">#Create new df with only taxon and media columns</span></span>
<span id="cb113-8"><a href="17-Differential_abundance_analysis.html#cb113-8" tabindex="-1"></a>res_ancom_class_media <span class="ot">&lt;-</span> res_ancom_class[,media_colnames]</span>
<span id="cb113-9"><a href="17-Differential_abundance_analysis.html#cb113-9" tabindex="-1"></a><span class="fu">head</span>(res_ancom_class_media)</span></code></pre></div>
<p>Because we had comparisons with more that 1 grouping (media &amp; site) we might have rows with no significnat results.
We will therefore remove these rows.</p>
<p><strong>Note:</strong> The next 2 code sections use the packages <code>dplyr</code> &amp; <code>tidyr</code> along with piping <code>%&gt;%</code>.
This is used as it would be very difficult to do these steps in base R and they are the code provided in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/ANCOMBC/inst/doc/ANCOMBC2.html"><code>ANCOMBC2</code> tutorial</a>.
If you would like to learn more about these packages please see the <a href="https://r4ds.had.co.nz/">R for Data Science book</a>.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="17-Differential_abundance_analysis.html#cb114-1" tabindex="-1"></a><span class="co">#Remove any taxon where there are no TRUE values</span></span>
<span id="cb114-2"><a href="17-Differential_abundance_analysis.html#cb114-2" tabindex="-1"></a><span class="co">#I.e. no significant difference found for the media samples</span></span>
<span id="cb114-3"><a href="17-Differential_abundance_analysis.html#cb114-3" tabindex="-1"></a><span class="co">#Number of biomarkers</span></span>
<span id="cb114-4"><a href="17-Differential_abundance_analysis.html#cb114-4" tabindex="-1"></a><span class="fu">nrow</span>(res_ancom_class_media)</span>
<span id="cb114-5"><a href="17-Differential_abundance_analysis.html#cb114-5" tabindex="-1"></a><span class="co">#Extract rows that contain at least one TRUE value</span></span>
<span id="cb114-6"><a href="17-Differential_abundance_analysis.html#cb114-6" tabindex="-1"></a>res_ancom_class_media_sig <span class="ot">&lt;-</span> res_ancom_class_media <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter_all</span>(</span>
<span id="cb114-7"><a href="17-Differential_abundance_analysis.html#cb114-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">any_vars</span>(. <span class="sc">%in%</span> <span class="cn">TRUE</span>)</span>
<span id="cb114-8"><a href="17-Differential_abundance_analysis.html#cb114-8" tabindex="-1"></a>)</span>
<span id="cb114-9"><a href="17-Differential_abundance_analysis.html#cb114-9" tabindex="-1"></a><span class="co">#Number of rows and the top of the new df</span></span>
<span id="cb114-10"><a href="17-Differential_abundance_analysis.html#cb114-10" tabindex="-1"></a><span class="fu">nrow</span>(res_ancom_class_media_sig)</span>
<span id="cb114-11"><a href="17-Differential_abundance_analysis.html#cb114-11" tabindex="-1"></a><span class="fu">head</span>(res_ancom_class_media_sig)</span></code></pre></div>
<p>Next we need to do a variety of preprocessing steps.
This will create an edited long data frame.
Long data frames are needed for <code>ggplot2()</code> when not using a <code>phyloseq</code> object.
Instead of having the LFC (log fold change) values over many columns there will be only one column for the values.
The long data frame will contain the following columns:</p>
<ul>
<li><strong>taxon</strong>: The biomarker name.</li>
<li><strong>group</strong>: The comparison group.</li>
<li><strong>value</strong>: The LFC value.</li>
</ul>
<p>Please see the code and its annotation below.
If you do not fully understand the code that is fine.
As long as you can edit it to work with your own future code.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="17-Differential_abundance_analysis.html#cb115-1" tabindex="-1"></a><span class="co">#Create edited long df to plot significant log fold changes on heatmap</span></span>
<span id="cb115-2"><a href="17-Differential_abundance_analysis.html#cb115-2" tabindex="-1"></a>res_ancom_class_media_sig_lfc_fig <span class="ot">&lt;-</span> res_ancom_class_media_sig <span class="sc">%&gt;%</span></span>
<span id="cb115-3"><a href="17-Differential_abundance_analysis.html#cb115-3" tabindex="-1"></a>  <span class="co">#Change log change values to 0 if diff is false</span></span>
<span id="cb115-4"><a href="17-Differential_abundance_analysis.html#cb115-4" tabindex="-1"></a>  <span class="co">#test: Is diff TRUE?</span></span>
<span id="cb115-5"><a href="17-Differential_abundance_analysis.html#cb115-5" tabindex="-1"></a>  <span class="co">#yes: Set LFC to original value (e.g. lfc_mediaCVP)</span></span>
<span id="cb115-6"><a href="17-Differential_abundance_analysis.html#cb115-6" tabindex="-1"></a>  <span class="co">#no: Set LFC to 0</span></span>
<span id="cb115-7"><a href="17-Differential_abundance_analysis.html#cb115-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">lfc_mediaCVP =</span> <span class="fu">ifelse</span>(<span class="at">test =</span> diff_mediaCVP <span class="sc">==</span> <span class="cn">TRUE</span>,</span>
<span id="cb115-8"><a href="17-Differential_abundance_analysis.html#cb115-8" tabindex="-1"></a>                                      <span class="at">yes =</span> lfc_mediaCVP, <span class="at">no =</span> <span class="dv">0</span>),</span>
<span id="cb115-9"><a href="17-Differential_abundance_analysis.html#cb115-9" tabindex="-1"></a>                <span class="at">lfc_mediaKBC =</span> <span class="fu">ifelse</span>(<span class="at">test =</span> diff_mediaKBC <span class="sc">==</span> <span class="cn">TRUE</span>,</span>
<span id="cb115-10"><a href="17-Differential_abundance_analysis.html#cb115-10" tabindex="-1"></a>                                      <span class="at">yes =</span> lfc_mediaKBC, <span class="at">no =</span> <span class="dv">0</span>),</span>
<span id="cb115-11"><a href="17-Differential_abundance_analysis.html#cb115-11" tabindex="-1"></a>                <span class="at">lfc_mediaTSA =</span> <span class="fu">ifelse</span>(<span class="at">test =</span> diff_mediaTSA <span class="sc">==</span> <span class="cn">TRUE</span>,</span>
<span id="cb115-12"><a href="17-Differential_abundance_analysis.html#cb115-12" tabindex="-1"></a>                                      <span class="at">yes =</span> lfc_mediaTSA, <span class="at">no =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb115-13"><a href="17-Differential_abundance_analysis.html#cb115-13" tabindex="-1"></a>  <span class="co">#Only retain taxon and log fold changes columns</span></span>
<span id="cb115-14"><a href="17-Differential_abundance_analysis.html#cb115-14" tabindex="-1"></a>  <span class="co">#Additionally, change column names and round LFC values to 2 decimal places</span></span>
<span id="cb115-15"><a href="17-Differential_abundance_analysis.html#cb115-15" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">transmute</span>(taxon,</span>
<span id="cb115-16"><a href="17-Differential_abundance_analysis.html#cb115-16" tabindex="-1"></a>                   <span class="at">CVP =</span> <span class="fu">round</span>(lfc_mediaCVP,<span class="dv">2</span>),</span>
<span id="cb115-17"><a href="17-Differential_abundance_analysis.html#cb115-17" tabindex="-1"></a>                   <span class="at">KBC =</span> <span class="fu">round</span>(lfc_mediaKBC,<span class="dv">2</span>),</span>
<span id="cb115-18"><a href="17-Differential_abundance_analysis.html#cb115-18" tabindex="-1"></a>                   <span class="at">TSA =</span> <span class="fu">round</span>(lfc_mediaTSA,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb115-19"><a href="17-Differential_abundance_analysis.html#cb115-19" tabindex="-1"></a>  <span class="co">#Convert to long data frame</span></span>
<span id="cb115-20"><a href="17-Differential_abundance_analysis.html#cb115-20" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> CVP<span class="sc">:</span>TSA,</span>
<span id="cb115-21"><a href="17-Differential_abundance_analysis.html#cb115-21" tabindex="-1"></a>                      <span class="at">names_to =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb115-22"><a href="17-Differential_abundance_analysis.html#cb115-22" tabindex="-1"></a>                      <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb115-23"><a href="17-Differential_abundance_analysis.html#cb115-23" tabindex="-1"></a>  <span class="co">#Arrange order of rows based on taxons</span></span>
<span id="cb115-24"><a href="17-Differential_abundance_analysis.html#cb115-24" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(taxon)</span>
<span id="cb115-25"><a href="17-Differential_abundance_analysis.html#cb115-25" tabindex="-1"></a><span class="co">#Check new long df</span></span>
<span id="cb115-26"><a href="17-Differential_abundance_analysis.html#cb115-26" tabindex="-1"></a><span class="fu">head</span>(res_ancom_class_media_sig_lfc_fig)</span></code></pre></div>
</div>
<div id="da-biomarker-lfc-heatmap" class="section level2 hasAnchor" number="17.6">
<h2><span class="header-section-number">17.6</span> DA: Biomarker LFC heatmap<a href="17-Differential_abundance_analysis.html#da-biomarker-lfc-heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/tiles.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We now have a long data frame with the significant LFC values.
We can use this to create a heatmap.</p>
<p>There are some specific functions and packages for R that can produce heatmaps.
These can include dendrograms on the y and x axes to show clustering.
We are not interested in having all the bells and whistles so we will use the <code>ggplot2</code> layer <code>geom_tile()</code>.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="17-Differential_abundance_analysis.html#cb116-1" tabindex="-1"></a><span class="co">#LFC heatmap</span></span>
<span id="cb116-2"><a href="17-Differential_abundance_analysis.html#cb116-2" tabindex="-1"></a>heatmap_media <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(res_ancom_class_media_sig_lfc_fig,</span>
<span id="cb116-3"><a href="17-Differential_abundance_analysis.html#cb116-3" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> taxon, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb116-4"><a href="17-Differential_abundance_analysis.html#cb116-4" tabindex="-1"></a>                  <span class="co">#Produce ggplot as tile/heatmap style plot</span></span>
<span id="cb116-5"><a href="17-Differential_abundance_analysis.html#cb116-5" tabindex="-1"></a>                  ggplot2<span class="sc">::</span><span class="fu">geom_tile</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb116-6"><a href="17-Differential_abundance_analysis.html#cb116-6" tabindex="-1"></a>                  <span class="co">#Add the LFC values as text in the cells</span></span>
<span id="cb116-7"><a href="17-Differential_abundance_analysis.html#cb116-7" tabindex="-1"></a>                  ggplot2<span class="sc">::</span><span class="fu">geom_text</span>(<span class="fu">aes</span>(group, taxon, <span class="at">label =</span> value),</span>
<span id="cb116-8"><a href="17-Differential_abundance_analysis.html#cb116-8" tabindex="-1"></a>                      <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb116-9"><a href="17-Differential_abundance_analysis.html#cb116-9" tabindex="-1"></a>                  <span class="co">#Remove the x and y labels (NULL) and add a title</span></span>
<span id="cb116-10"><a href="17-Differential_abundance_analysis.html#cb116-10" tabindex="-1"></a>                  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="cn">NULL</span>,</span>
<span id="cb116-11"><a href="17-Differential_abundance_analysis.html#cb116-11" tabindex="-1"></a>                    <span class="at">title =</span> <span class="st">&quot;Log fold changes compared to ENV samples&quot;</span>)</span>
<span id="cb116-12"><a href="17-Differential_abundance_analysis.html#cb116-12" tabindex="-1"></a><span class="co">#Save ggplot2 object with ggsave</span></span>
<span id="cb116-13"><a href="17-Differential_abundance_analysis.html#cb116-13" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./ANCOMBC_class_media_ref_ENV.png&quot;</span>, <span class="at">plot =</span> heatmap_media,</span>
<span id="cb116-14"><a href="17-Differential_abundance_analysis.html#cb116-14" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">125</span>, <span class="at">width =</span> <span class="dv">150</span>)</span>
<span id="cb116-15"><a href="17-Differential_abundance_analysis.html#cb116-15" tabindex="-1"></a><span class="co">#Display plot</span></span>
<span id="cb116-16"><a href="17-Differential_abundance_analysis.html#cb116-16" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./ANCOMBC_class_media_ref_ENV.png&quot;</span>)</span></code></pre></div>
<p>It is a good start but the colour gradient is not very useful.
We are going to use the layer <code>ggplot2::scale_fill_gradient2()</code> to customise this.
First, we need to decide on the minimum and maximum values.
Let's generate these with some code based on the minimum and maximum LFC values.
We will use 0 as the midpoint so we can easily differentiate between higher and lower abundance biomarkers.</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="17-Differential_abundance_analysis.html#cb117-1" tabindex="-1"></a><span class="co">#Get min and max log fold changes</span></span>
<span id="cb117-2"><a href="17-Differential_abundance_analysis.html#cb117-2" tabindex="-1"></a>lo <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">min</span>(res_ancom_class_media_sig_lfc_fig<span class="sc">$</span>value))</span>
<span id="cb117-3"><a href="17-Differential_abundance_analysis.html#cb117-3" tabindex="-1"></a>up <span class="ot">&lt;-</span> <span class="fu">ceiling</span>(<span class="fu">max</span>(res_ancom_class_media_sig_lfc_fig<span class="sc">$</span>value))</span>
<span id="cb117-4"><a href="17-Differential_abundance_analysis.html#cb117-4" tabindex="-1"></a>mid <span class="ot">&lt;-</span> <span class="dv">0</span></span></code></pre></div>
<p>With these values we can now make our upgraded heatmap.</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="17-Differential_abundance_analysis.html#cb118-1" tabindex="-1"></a><span class="co">#LFC heatmap</span></span>
<span id="cb118-2"><a href="17-Differential_abundance_analysis.html#cb118-2" tabindex="-1"></a>heatmap_media <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(res_ancom_class_media_sig_lfc_fig,</span>
<span id="cb118-3"><a href="17-Differential_abundance_analysis.html#cb118-3" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> taxon, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb118-4"><a href="17-Differential_abundance_analysis.html#cb118-4" tabindex="-1"></a>                  <span class="co">#Produce ggplot as tile/heatmap style plot</span></span>
<span id="cb118-5"><a href="17-Differential_abundance_analysis.html#cb118-5" tabindex="-1"></a>                  ggplot2<span class="sc">::</span><span class="fu">geom_tile</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb118-6"><a href="17-Differential_abundance_analysis.html#cb118-6" tabindex="-1"></a>                  <span class="co">#Customise colour gradient</span></span>
<span id="cb118-7"><a href="17-Differential_abundance_analysis.html#cb118-7" tabindex="-1"></a>                  ggplot2<span class="sc">::</span><span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">high =</span> <span class="st">&quot;red&quot;</span>, <span class="at">mid =</span> <span class="st">&quot;white&quot;</span>,</span>
<span id="cb118-8"><a href="17-Differential_abundance_analysis.html#cb118-8" tabindex="-1"></a>                                                <span class="at">na.value =</span> <span class="st">&quot;white&quot;</span>,</span>
<span id="cb118-9"><a href="17-Differential_abundance_analysis.html#cb118-9" tabindex="-1"></a>                                                <span class="at">midpoint =</span> mid, <span class="at">limit =</span> <span class="fu">c</span>(lo,up)) <span class="sc">+</span></span>
<span id="cb118-10"><a href="17-Differential_abundance_analysis.html#cb118-10" tabindex="-1"></a>                  <span class="co">#Add the LFC values as text in the cells</span></span>
<span id="cb118-11"><a href="17-Differential_abundance_analysis.html#cb118-11" tabindex="-1"></a>                  ggplot2<span class="sc">::</span><span class="fu">geom_text</span>(<span class="fu">aes</span>(group, taxon, <span class="at">label =</span> value),</span>
<span id="cb118-12"><a href="17-Differential_abundance_analysis.html#cb118-12" tabindex="-1"></a>                      <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb118-13"><a href="17-Differential_abundance_analysis.html#cb118-13" tabindex="-1"></a>                  <span class="co">#Remove the x and y labels (NULL) and add a title</span></span>
<span id="cb118-14"><a href="17-Differential_abundance_analysis.html#cb118-14" tabindex="-1"></a>                  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="cn">NULL</span>,</span>
<span id="cb118-15"><a href="17-Differential_abundance_analysis.html#cb118-15" tabindex="-1"></a>                    <span class="at">title =</span> <span class="st">&quot;Log fold changes compared to ENV samples&quot;</span>)</span>
<span id="cb118-16"><a href="17-Differential_abundance_analysis.html#cb118-16" tabindex="-1"></a><span class="co">#Save ggplot2 object with ggsave</span></span>
<span id="cb118-17"><a href="17-Differential_abundance_analysis.html#cb118-17" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./ANCOMBC_class_media_ref_ENV.png&quot;</span>, <span class="at">plot =</span> heatmap_media,</span>
<span id="cb118-18"><a href="17-Differential_abundance_analysis.html#cb118-18" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">125</span>, <span class="at">width =</span> <span class="dv">150</span>)</span>
<span id="cb118-19"><a href="17-Differential_abundance_analysis.html#cb118-19" tabindex="-1"></a><span class="co">#Display plot</span></span>
<span id="cb118-20"><a href="17-Differential_abundance_analysis.html#cb118-20" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./ANCOMBC_class_media_ref_ENV.png&quot;</span>)</span></code></pre></div>
<p>Great, now we can see the biomarker with higher abundances in the ENV sample (blue) and higher abundances in the comparison media (red).
Any 0 values are non-significant results we turned to 0 when we created the long data frame.</p>
<ul>
<li>Are there more biomarkers in higher abundance in ENV rather than the comparison media (i.e blue/negative results)?
<div id="radio_TVKGJFXUTY" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_TVKGJFXUTY" value="answer"></input> <span><strong>Yes</strong></span></label><label><input type="radio" autocomplete="off" name="radio_TVKGJFXUTY" value=""></input> <span><strong>No</strong></span></label>
</div></li>
<li>Which biomarker has the highest positive values?
<div id="radio_XKBMRJDQVJ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_XKBMRJDQVJ" value=""></input> <span><strong>Phylum:Verrucomicrobia</strong></span></label><label><input type="radio" autocomplete="off" name="radio_XKBMRJDQVJ" value="answer"></input> <span><strong>Class:Bacilli</strong></span></label><label><input type="radio" autocomplete="off" name="radio_XKBMRJDQVJ" value=""></input> <span><strong>Class:Alphaproteobacteria</strong></span></label>
</div></li>
<li>Which biomarker has the lowest value?
<div id="radio_SLFZJJZWBL" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_SLFZJJZWBL" value=""></input> <span><strong>Phylum:Verrucomicrobia</strong></span></label><label><input type="radio" autocomplete="off" name="radio_SLFZJJZWBL" value=""></input> <span><strong>Class:Bacilli</strong></span></label><label><input type="radio" autocomplete="off" name="radio_SLFZJJZWBL" value="answer"></input> <span><strong>Class:Alphaproteobacteria</strong></span></label>
</div></li>
</ul>
<p>From these results we can see that there are a lot of biomarkers with low abundances in the comparison media samples.
There are only 2 biomarkers with higher abundance in the media samples.</p>
<p>The LFC values between the different media are quite similar.
There are some biomarkers that are only siginifcnat in 1 or 2 of the media though.
It seems that CVP and KBC are more simialr to each other than to TSA.
This matches the clustering we found in the beta diversity.</p>
<p>This seems to indicate that the media samples are not a good representation of the actual environmental community.
They appear to select for a few bacterial classes and select against a lot of bacterial classes.</p>
</div>
<div id="da-matching-bar-plot" class="section level2 hasAnchor" number="17.7">
<h2><span class="header-section-number">17.7</span> DA: matching bar plot<a href="17-Differential_abundance_analysis.html#da-matching-bar-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/bar_chart.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Using code from the <a href="11-Taxa_plots.html#chaptaxaplots">taxa plots chapter</a> we'll make a quick class based taxa plot.
We will use the relative abundances for this.</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="17-Differential_abundance_analysis.html#cb119-1" tabindex="-1"></a><span class="co">#Quick class based relative abundance heatmap</span></span>
<span id="cb119-2"><a href="17-Differential_abundance_analysis.html#cb119-2" tabindex="-1"></a><span class="co">#Convert abundance table to class relative abundance (compositional) table</span></span>
<span id="cb119-3"><a href="17-Differential_abundance_analysis.html#cb119-3" tabindex="-1"></a>pseq_relabund <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">transform</span>(pseq, <span class="st">&quot;compositional&quot;</span>)</span>
<span id="cb119-4"><a href="17-Differential_abundance_analysis.html#cb119-4" tabindex="-1"></a><span class="co">#Class phyloseq</span></span>
<span id="cb119-5"><a href="17-Differential_abundance_analysis.html#cb119-5" tabindex="-1"></a>class_relabund_pseq <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">aggregate_taxa</span>(pseq_relabund, <span class="st">&quot;Class&quot;</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb119-6"><a href="17-Differential_abundance_analysis.html#cb119-6" tabindex="-1"></a><span class="co">#Produce heatmap ggplot</span></span>
<span id="cb119-7"><a href="17-Differential_abundance_analysis.html#cb119-7" tabindex="-1"></a>class_heatmap <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">plot_composition</span>(class_relabund_pseq, <span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>) <span class="sc">+</span></span>
<span id="cb119-8"><a href="17-Differential_abundance_analysis.html#cb119-8" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Class&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Sample&quot;</span>) <span class="sc">+</span></span>
<span id="cb119-9"><a href="17-Differential_abundance_analysis.html#cb119-9" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Class relative abundance heatmap&quot;</span>)</span>
<span id="cb119-10"><a href="17-Differential_abundance_analysis.html#cb119-10" tabindex="-1"></a><span class="co">#Save ggplot object as png file</span></span>
<span id="cb119-11"><a href="17-Differential_abundance_analysis.html#cb119-11" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">&quot;./class_relabund_heatmap.png&quot;</span>, <span class="at">plot =</span> class_heatmap,</span>
<span id="cb119-12"><a href="17-Differential_abundance_analysis.html#cb119-12" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="at">height =</span> <span class="dv">200</span>, <span class="at">width =</span> <span class="dv">200</span>)</span>
<span id="cb119-13"><a href="17-Differential_abundance_analysis.html#cb119-13" tabindex="-1"></a><span class="co">#Display the plot in jupyter notebook</span></span>
<span id="cb119-14"><a href="17-Differential_abundance_analysis.html#cb119-14" tabindex="-1"></a>IRdisplay<span class="sc">::</span><span class="fu">display_png</span>(<span class="at">file=</span><span class="st">&quot;./class_relabund_heatmap.png&quot;</span>)</span></code></pre></div>
<p>With this plot we can clearly see that there are many classess with abundances in the ENV samples that have very little in the media samples.
Most the the relative abundances of the media samples consist of Gammaproteobacteria.
The TSA and KBC samples also have a high abundance of Bacilli.
Overall, this is a nice visual confirmation of our ANCOM results.</p>
</div>
<div id="da-summary" class="section level2 hasAnchor" number="17.8">
<h2><span class="header-section-number">17.8</span> DA: summary<a href="17-Differential_abundance_analysis.html#da-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_blue.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We have used ANCOMBC to detect biomarkers in the media samples compared to the ENV samples.
This showed most biomarkers were caused by depletion of many classes and selection of a few classes by the media.</p>
<p>You could carry out more analysis in your future research in various ways:</p>
<ul>
<li>Carry out biomarker detection at different taxonomy levels such as genus.
<ul>
<li>We used class in this tutorial so we would have enough groups for a comparison but not too many to increase speed of analysis and so there wasn't too many results to look at.</li>
</ul></li>
<li>There are many other types of tests and considerations to keep in mind. Check out the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/ANCOMBC/inst/doc/ANCOMBC.html">ANCOMBC2 tutorial</a>, it is very in depth.
<ul>
<li>There is also a handy FAQ on the <a href="https://github.com/FrederickHuangLin/ANCOMBC">ANCOMBC github page</a>.</li>
</ul></li>
</ul>
<p>That is the last of the analysis. In the final chapter we will review the steps we carried out and the answers to the various questions we have about the dataset.</p>

</div>
</div>



<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="16-Beta.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="18-Summary.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
