<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A Useful tools and considerations | Illumina Quality Control</title>
  <meta name="description" content="NEOF book for the Illumina quality control workshop" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="A Useful tools and considerations | Illumina Quality Control" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Illumina quality control workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Useful tools and considerations | Illumina Quality Control" />
  
  <meta name="twitter:description" content="NEOF book for the Illumina quality control workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew Gemmell" />


<meta name="date" content="2022-03-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="04-QC_exercises.html"/>

<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Illumina QC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Illumina_Quality_Control.html"><a href="01-Illumina_Quality_Control.html"><i class="fa fa-check"></i><b>1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-QC_tutorials.html"><a href="03-QC_tutorials.html"><i class="fa fa-check"></i><b>3</b> QC tutorial<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-QC_tutorials.html"><a href="03-QC_tutorials.html#data"><i class="fa fa-check"></i><b>3.1</b> Workshop data<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="03-QC_tutorials.html"><a href="03-QC_tutorials.html#goodtutorial"><i class="fa fa-check"></i><b>3.2</b> Good quality Genomic dataset<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="03-QC_tutorials.html"><a href="03-QC_tutorials.html#poortutorial"><i class="fa fa-check"></i><b>3.3</b> Poorer quality Genomic dataset<span></span></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="03-QC_tutorials.html"><a href="03-QC_tutorials.html#quality-checking"><i class="fa fa-check"></i><b>3.3.1</b> Quality checking<span></span></a></li>
<li class="chapter" data-level="3.3.2" data-path="03-QC_tutorials.html"><a href="03-QC_tutorials.html#quality-control"><i class="fa fa-check"></i><b>3.3.2</b> Quality control<span></span></a></li>
<li class="chapter" data-level="3.3.3" data-path="03-QC_tutorials.html"><a href="03-QC_tutorials.html#post-quality-control-check"><i class="fa fa-check"></i><b>3.3.3</b> Post Quality control check<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-QC_exercises.html"><a href="04-QC_exercises.html"><i class="fa fa-check"></i><b>4</b> QC exercises<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC_exercises.html"><a href="04-QC_exercises.html#shotgun"><i class="fa fa-check"></i><b>4.1</b> Whole shotgun metagenomics<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="04-QC_exercises.html"><a href="04-QC_exercises.html#metagenetics"><i class="fa fa-check"></i><b>4.2</b> 16S rRNA Metagenetics<span></span></a></li>
</ul></li>
<li class="appendix"><span><b>Appendix<span></span></b></span></li>
<li class="chapter" data-level="A" data-path="05-Useful_Tools_and_Considerations.html"><a href="05-Useful_Tools_and_Considerations.html"><i class="fa fa-check"></i><b>A</b> Useful tools and considerations<span></span></a>
<ul>
<li class="chapter" data-level="A.1" data-path="05-Useful_Tools_and_Considerations.html"><a href="05-Useful_Tools_and_Considerations.html#quality-trimming-and-filtering-tools"><i class="fa fa-check"></i><b>A.1</b> Quality trimming and filtering tools<span></span></a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="05-Useful_Tools_and_Considerations.html"><a href="05-Useful_Tools_and_Considerations.html#trim-galore"><i class="fa fa-check"></i><b>A.1.1</b> Trim Galore<span></span></a></li>
<li class="chapter" data-level="A.1.2" data-path="05-Useful_Tools_and_Considerations.html"><a href="05-Useful_Tools_and_Considerations.html#trimmomatic"><i class="fa fa-check"></i><b>A.1.2</b> Trimmomatic<span></span></a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="05-Useful_Tools_and_Considerations.html"><a href="05-Useful_Tools_and_Considerations.html#multiqc"><i class="fa fa-check"></i><b>A.2</b> MultiQC<span></span></a></li>
<li class="chapter" data-level="A.3" data-path="05-Useful_Tools_and_Considerations.html"><a href="05-Useful_Tools_and_Considerations.html#fasta-and-fastq-processing-toolkits"><i class="fa fa-check"></i><b>A.3</b> Fasta and fastq processing toolkits<span></span></a>
<ul>
<li class="chapter" data-level="A.3.1" data-path="05-Useful_Tools_and_Considerations.html"><a href="05-Useful_Tools_and_Considerations.html#fastp"><i class="fa fa-check"></i><b>A.3.1</b> Fastp<span></span></a></li>
<li class="chapter" data-level="A.3.2" data-path="05-Useful_Tools_and_Considerations.html"><a href="05-Useful_Tools_and_Considerations.html#seqtk"><i class="fa fa-check"></i><b>A.3.2</b> Seqtk<span></span></a></li>
<li class="chapter" data-level="A.3.3" data-path="05-Useful_Tools_and_Considerations.html"><a href="05-Useful_Tools_and_Considerations.html#fastx-toolkit"><i class="fa fa-check"></i><b>A.3.3</b> FASTX-Toolkit<span></span></a></li>
<li class="chapter" data-level="A.3.4" data-path="05-Useful_Tools_and_Considerations.html"><a href="05-Useful_Tools_and_Considerations.html#bbtools"><i class="fa fa-check"></i><b>A.3.4</b> BBTools<span></span></a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="05-Useful_Tools_and_Considerations.html"><a href="05-Useful_Tools_and_Considerations.html#conda"><i class="fa fa-check"></i><b>A.4</b> Conda<span></span></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Illumina Quality Control</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="useful-tools-and-considerations" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">A</span> Useful tools and considerations<a href="05-Useful_Tools_and_Considerations.html#useful-tools-and-considerations" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/toolbox.png" width="20%" style="display: block; margin: auto;" /></p>
<p>There are various other useful tools for sequencing QC of Illumina data. Below are some suggestions but there are many more.</p>
<div id="quality-trimming-and-filtering-tools" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">A.1</span> Quality trimming and filtering tools<a href="05-Useful_Tools_and_Considerations.html#quality-trimming-and-filtering-tools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/quality_trimming_and_filtering.png" width="20%" style="display: block; margin: auto;" /></p>
<div id="trim-galore" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">A.1.1</span> Trim Galore<a href="05-Useful_Tools_and_Considerations.html#trim-galore" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>From the people who brought you FastQC, Trim Galore. This is a wrapper around cutadapt and FastQC. It has a lot of options and can output fastqc results of the quality controlled reads at the same time.</p>
<p><strong>Github page:</strong>
<a href="https://github.com/FelixKrueger/TrimGalore" class="uri">https://github.com/FelixKrueger/TrimGalore</a></p>
<p><strong>Manual:</strong>
<a href="https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md" class="uri">https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md</a></p>
</div>
<div id="trimmomatic" class="section level3 hasAnchor" number="5.1.2">
<h3><span class="header-section-number">A.1.2</span> Trimmomatic<a href="05-Useful_Tools_and_Considerations.html#trimmomatic" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A flexible read trimming tool for Illumina data.</p>
<p><strong>Main page:</strong>
<a href="http://www.usadellab.org/cms/?page=trimmomatic" class="uri">http://www.usadellab.org/cms/?page=trimmomatic</a></p>
<p><strong>Manual:</strong>
<a href="http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf" class="uri">http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf</a></p>
</div>
</div>
<div id="multiqc" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">A.2</span> MultiQC<a href="05-Useful_Tools_and_Considerations.html#multiqc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/MultiQC_logo.png" width="40%" style="display: block; margin: auto;" />
MultiQC is a brilliant tool that can be used to collate multiple fastQC reports into one interactive html file. This means if you have 100 sets of paired reads you can look at 2 htmls rather than 200 FastQC reports. I say 2 htmls because it is best to put the forward reads into 1 MultiQC report and the reverse reads into another.</p>
<p><strong>Main page:</strong> <a href="https://multiqc.info/" class="uri">https://multiqc.info/</a></p>
<p><strong>Documentation:</strong> <a href="https://multiqc.info/docs/" class="uri">https://multiqc.info/docs/</a></p>
<p>MultiQC can be to collate the output from many other bioinformatic tools.</p>
<p><strong>Supported tools:</strong> <a href="https://multiqc.info/#supported-tools" class="uri">https://multiqc.info/#supported-tools</a></p>
</div>
<div id="fasta-and-fastq-processing-toolkits" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">A.3</span> Fasta and fastq processing toolkits<a href="05-Useful_Tools_and_Considerations.html#fasta-and-fastq-processing-toolkits" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/process.png" width="20%" style="display: block; margin: auto;" />
Sometimes a fasta/fastq file will need to be modified to be properly used with a file or to remove a specific feature. Thankfully if you need to carry out a specific action someone else has probably needed to before you. Below are some tools that can be used to modify fasta/fastq files.</p>
<div id="fastp" class="section level3 hasAnchor" number="5.3.1">
<h3><span class="header-section-number">A.3.1</span> Fastp<a href="05-Useful_Tools_and_Considerations.html#fastp" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This is a tool designed to be a "all-in-one" preprocessor for fastq files. This includes quality control. I find it very useful as the tool will only do what you tell it to do. This means you can tell it to only carry out length filtering and it will only do length filtering (unlike Trim Galore which will insist on quality trimming).</p>
<p>You could therefore use the below command to trim all reads to a length of 120 and remove any read pairs where at least one read has a length shorter than 120. You’ll then only have reads that are a length of 120. This is required for a few tools.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="05-Useful_Tools_and_Considerations.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="ex">fastp</span> <span class="at">--in1</span> R1.fastq <span class="at">--in2</span> R2.fastq <span class="at">--out1</span> R1.120.fastq <span class="at">--out2</span> R2.120.fastq <span class="at">--max_len1</span> 120 <span class="at">--max_len2</span> 120 <span class="at">--length_required</span> 120</span></code></pre></div>
<p><strong>Main page:</strong> <a href="https://github.com/OpenGene/fastp" class="uri">https://github.com/OpenGene/fastp</a></p>
</div>
<div id="seqtk" class="section level3 hasAnchor" number="5.3.2">
<h3><span class="header-section-number">A.3.2</span> Seqtk<a href="05-Useful_Tools_and_Considerations.html#seqtk" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A tool to process fasta or fastq files. It can carry out many tasks but I primarily use it to:</p>
<ul>
<li>convert fastq files to fasta</li>
</ul>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="05-Useful_Tools_and_Considerations.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">seqtk</span> seq <span class="at">-a</span> in.fastq <span class="op">&gt;</span> out.fasta</span></code></pre></div>
<ul>
<li>subsample paired reads from fastq files</li>
</ul>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="05-Useful_Tools_and_Considerations.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">seqtk</span> sample <span class="at">-s100</span> read1.fastq 10000 <span class="op">&gt;</span> subsample1.fastq</span>
<span id="cb20-2"><a href="05-Useful_Tools_and_Considerations.html#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqtk</span> sample <span class="at">-s100</span> read2.fastq 10000 <span class="op">&gt;</span> subsample2.fastq</span></code></pre></div>
<p>For more information on how these commands work and for further commands please see the link below.</p>
<p><strong>Github:</strong> <a href="https://github.com/lh3/seqtk" class="uri">https://github.com/lh3/seqtk</a></p>
</div>
<div id="fastx-toolkit" class="section level3 hasAnchor" number="5.3.3">
<h3><span class="header-section-number">A.3.3</span> FASTX-Toolkit<a href="05-Useful_Tools_and_Considerations.html#fastx-toolkit" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A tool to process fasta or fastq files. It is quite useful but unfortunately does not take paired read information into context. This means if a read is removed from the reverse read it will not necessarily be removed from the forward file.</p>
<p>My favourite tool from this is 'fasta_formatter'. It can be used to set the maximum number of bases that will be on one line in a fasta file. I primarily use this so that each seq in a fasta file is on one line.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="05-Useful_Tools_and_Considerations.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">fasta_formatter</span> <span class="at">-i</span> in.fasta <span class="at">-o</span> out.fasta <span class="at">-w</span> 0</span></code></pre></div>
<p><strong>Main page:</strong> <a href="http://hannonlab.cshl.edu/fastx_toolkit/" class="uri">http://hannonlab.cshl.edu/fastx_toolkit/</a></p>
<p><strong>Commands:</strong> <a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html" class="uri">http://hannonlab.cshl.edu/fastx_toolkit/commandline.html</a></p>
</div>
<div id="bbtools" class="section level3 hasAnchor" number="5.3.4">
<h3><span class="header-section-number">A.3.4</span> BBTools<a href="05-Useful_Tools_and_Considerations.html#bbtools" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>BBTools is a collection of various tools. There are tools for deduplication, mapping, merging, and more. There is a lot so if you are interested it would be best to look at the links below.</p>
<p><strong>Main page</strong>: <a href="https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/" class="uri">https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/</a></p>
<p><strong>General usage guide:</strong> <a href="https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/usage-guide/" class="uri">https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/usage-guide/</a></p>
<p><strong>Data Preprocessing:</strong> <a href="https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/data-preprocessing/" class="uri">https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/data-preprocessing/</a></p>
</div>
</div>
<div id="conda" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">A.4</span> Conda<a href="05-Useful_Tools_and_Considerations.html#conda" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/conda.png" width="20%" style="display: block; margin: auto;" />
As you can tell there are many bioinformatic tools. It would take a lot of work to install these all if it was easy. However some tools can be difficult to install. Thankfully conda exists.</p>
<p>Conda is a very useful and highly used package and environmental management system. In other words you can use it to install many tools very easily. Additionally it is very easy to install conda.</p>
<p><strong>Getting started with conda:</strong> <a href="https://docs.conda.io/projects/conda/en/latest/user-guide/getting-started.html" class="uri">https://docs.conda.io/projects/conda/en/latest/user-guide/getting-started.html</a></p>
<p><strong>Main page:</strong> <a href="https://docs.conda.io/en/latest/" class="uri">https://docs.conda.io/en/latest/</a></p>
<p><strong>Installation instructions:</strong> <a href="https://docs.conda.io/projects/conda/en/latest/user-guide/install/" class="uri">https://docs.conda.io/projects/conda/en/latest/user-guide/install/</a></p>
<p><strong>Cheatsheet:</strong> <a href="https://docs.conda.io/projects/conda/en/latest/user-guide/cheatsheet.html" class="uri">https://docs.conda.io/projects/conda/en/latest/user-guide/cheatsheet.html</a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="04-QC_exercises.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
