<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 QC exercises | Illumina Quality Control</title>
  <meta name="description" content="NEOF book for the Illumina quality control workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 QC exercises | Illumina Quality Control" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Illumina quality control workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 QC exercises | Illumina Quality Control" />
  
  <meta name="twitter:description" content="NEOF book for the Illumina quality control workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-10-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="05-Poorer_quality_data.html"/>
<link rel="next" href="07-MultiQC.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Illumina QC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Illumina_Quality_Control.html"><a href="01-Illumina_Quality_Control.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Illumina_Quality_Control.html"><a href="01-Illumina_Quality_Control.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Setup.html"><a href="03-Setup.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Setup.html"><a href="03-Setup.html#data"><i class="fa fa-check"></i><b>3.1</b> Workshop data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Setup.html"><a href="03-Setup.html#mamba"><i class="fa fa-check"></i><b>3.2</b> Mamba</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Good_quality_data.html"><a href="04-Good_quality_data.html"><i class="fa fa-check"></i><b>4</b> Good quality genomic dataset</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Good_quality_data.html"><a href="04-Good_quality_data.html#run-fastqc"><i class="fa fa-check"></i><b>4.1</b> Run fastqc</a></li>
<li class="chapter" data-level="4.2" data-path="04-Good_quality_data.html"><a href="04-Good_quality_data.html#background-and-foreground-processes"><i class="fa fa-check"></i><b>4.2</b> Background and Foreground processes</a></li>
<li class="chapter" data-level="4.3" data-path="04-Good_quality_data.html"><a href="04-Good_quality_data.html#view-fastqc-output"><i class="fa fa-check"></i><b>4.3</b> View fastqc output</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Poorer_quality_data.html"><a href="05-Poorer_quality_data.html"><i class="fa fa-check"></i><b>5</b> Poorer quality genomic dataset</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Poorer_quality_data.html"><a href="05-Poorer_quality_data.html#quality-checking"><i class="fa fa-check"></i><b>5.1</b> Quality checking</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="05-Poorer_quality_data.html"><a href="05-Poorer_quality_data.html#quality-checking-mcqs"><i class="fa fa-check"></i><b>5.1.1</b> Quality checking MCQs</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="05-Poorer_quality_data.html"><a href="05-Poorer_quality_data.html#quality-control"><i class="fa fa-check"></i><b>5.2</b> Quality control</a></li>
<li class="chapter" data-level="5.3" data-path="05-Poorer_quality_data.html"><a href="05-Poorer_quality_data.html#post-quality-control-check"><i class="fa fa-check"></i><b>5.3</b> Post Quality control check</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="05-Poorer_quality_data.html"><a href="05-Poorer_quality_data.html#quality-checking-mcqs-1"><i class="fa fa-check"></i><b>5.3.1</b> Quality checking MCQs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-QC_exercises.html"><a href="06-QC_exercises.html"><i class="fa fa-check"></i><b>6</b> QC exercises</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-QC_exercises.html"><a href="06-QC_exercises.html#shotgun"><i class="fa fa-check"></i><b>6.1</b> Whole shotgun metagenomics</a></li>
<li class="chapter" data-level="6.2" data-path="06-QC_exercises.html"><a href="06-QC_exercises.html#metagenetics"><i class="fa fa-check"></i><b>6.2</b> 16S rRNA Metagenetics</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-MultiQC.html"><a href="07-MultiQC.html"><i class="fa fa-check"></i><b>7</b> MultiQC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-MultiQC.html"><a href="07-MultiQC.html#the-data"><i class="fa fa-check"></i><b>7.1</b> The data</a></li>
<li class="chapter" data-level="7.2" data-path="07-MultiQC.html"><a href="07-MultiQC.html#fastqc-of-the-raw-data"><i class="fa fa-check"></i><b>7.2</b> Fastqc of the raw data</a></li>
<li class="chapter" data-level="7.3" data-path="07-MultiQC.html"><a href="07-MultiQC.html#fastqc-of-the-trimmed-data"><i class="fa fa-check"></i><b>7.3</b> Fastqc of the trimmed data</a></li>
<li class="chapter" data-level="7.4" data-path="07-MultiQC.html"><a href="07-MultiQC.html#read-data"><i class="fa fa-check"></i><b>7.4</b> Read data</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="08-Useful_Tools_and_Considerations.html"><a href="08-Useful_Tools_and_Considerations.html"><i class="fa fa-check"></i><b>A</b> Useful tools and considerations</a>
<ul>
<li class="chapter" data-level="A.1" data-path="08-Useful_Tools_and_Considerations.html"><a href="08-Useful_Tools_and_Considerations.html#quality-trimming-and-filtering-tools"><i class="fa fa-check"></i><b>A.1</b> Quality trimming and filtering tools</a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="08-Useful_Tools_and_Considerations.html"><a href="08-Useful_Tools_and_Considerations.html#trim-galore"><i class="fa fa-check"></i><b>A.1.1</b> Trim Galore</a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="08-Useful_Tools_and_Considerations.html"><a href="08-Useful_Tools_and_Considerations.html#fasta-and-fastq-processing-toolkits"><i class="fa fa-check"></i><b>A.2</b> Fasta and fastq processing toolkits</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="08-Useful_Tools_and_Considerations.html"><a href="08-Useful_Tools_and_Considerations.html#fastp"><i class="fa fa-check"></i><b>A.2.1</b> Fastp</a></li>
<li class="chapter" data-level="A.2.2" data-path="08-Useful_Tools_and_Considerations.html"><a href="08-Useful_Tools_and_Considerations.html#seqtk"><i class="fa fa-check"></i><b>A.2.2</b> Seqtk</a></li>
<li class="chapter" data-level="A.2.3" data-path="08-Useful_Tools_and_Considerations.html"><a href="08-Useful_Tools_and_Considerations.html#fastx-toolkit"><i class="fa fa-check"></i><b>A.2.3</b> FASTX-Toolkit</a></li>
<li class="chapter" data-level="A.2.4" data-path="08-Useful_Tools_and_Considerations.html"><a href="08-Useful_Tools_and_Considerations.html#bbtools"><i class="fa fa-check"></i><b>A.2.4</b> BBTools</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="08-Useful_Tools_and_Considerations.html"><a href="08-Useful_Tools_and_Considerations.html#mamba-1"><i class="fa fa-check"></i><b>B</b> Mamba</a>
<ul>
<li class="chapter" data-level="B.1" data-path="08-Useful_Tools_and_Considerations.html"><a href="08-Useful_Tools_and_Considerations.html#mamba_install"><i class="fa fa-check"></i><b>B.1</b> Mamba installation and environment</a></li>
<li class="chapter" data-level="B.2" data-path="08-Useful_Tools_and_Considerations.html"><a href="08-Useful_Tools_and_Considerations.html#illumina-qc-environment"><i class="fa fa-check"></i><b>B.2</b> Illumina QC environment</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Illumina Quality Control</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="qcexercise" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> QC exercises<a href="06-QC_exercises.html#qcexercise" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/exercise.png" width="20%" style="display: block; margin: auto;" />
Now you will carry out Quality checking and control of a whole genome shotgun study and a 16S rRNA study. You will run through the Quality checking and Quality control using the examples to help you. There will be questions to answer after which there will be info on the data and what the QC has shown.</p>
<div id="shotgun" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Whole shotgun metagenomics<a href="06-QC_exercises.html#shotgun" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/kraken.png" width="20%" style="display: block; margin: auto;" />
The data from this exercise comes from the following link:</p>
<p><a href="https://www.ebi.ac.uk/ena/data/search?query=shotgun" class="uri">https://www.ebi.ac.uk/ena/data/search?query=shotgun</a></p>
<p>This contains the genomic content of many organisms.</p>
<p>The read data is within your <code>~/QC_workshop/Illumina_QC/exercise_1</code> directory (ensure to change directory to this). The reads have been subsampled so you will be working on 100,000 paired reads.</p>
<p>Carry out <code>fastqc</code> on the raw reads and ponder the below questions.</p>
<ol style="list-style-type: decimal">
<li>What may be causing the weird GC content at the start of the reads?</li>
<li>The "Per sequence GC content" has a weird peak. Why might this be?</li>
</ol>
<p>Carry out <code>trimmomatic</code> quality control, deciding on the sliding window and the length threshold. After <code>trimmomatic</code> carry out <code>fastqc</code> on the trimmed and filtered reads to determine if it was successful. Afterwards answer the following questions. If your answers don't match the correct answers below try to rerun trimmomatic with different parameter options.</p>
<ol style="list-style-type: decimal">
<li>Is the number of quality controlled reads above 60,000?
<div id="radio_ZEOFCUPGPP" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ZEOFCUPGPP" value="answer"></input> <span><strong>Yes</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ZEOFCUPGPP" value=""></input> <span><strong>No</strong></span></label>
</div></li>
<li>Does the quality for either R1 or R2 drop below Q24 (Including the whiskers)?
<div id="radio_SEBRRQPDKE" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_SEBRRQPDKE" value="answer"></input> <span><strong>Yes</strong></span></label><label><input type="radio" autocomplete="off" name="radio_SEBRRQPDKE" value=""></input> <span><strong>No</strong></span></label>
</div></li>
<li>Is the weird GC content still at the start of the read?
<div id="radio_ITSERSCIUP" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ITSERSCIUP" value="answer"></input> <span><strong>Yes</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ITSERSCIUP" value=""></input> <span><strong>No</strong></span></label>
</div></li>
<li>Does the "Per sequence GC content" still have a weird peak?
<div id="radio_SAMDAPIFGN" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_SAMDAPIFGN" value="answer"></input> <span><strong>Yes</strong></span></label><label><input type="radio" autocomplete="off" name="radio_SAMDAPIFGN" value=""></input> <span><strong>No</strong></span></label>
</div></li>
</ol>
<p><strong>Explanations</strong></p>
<p>In this case it helps to know the adapter sequence, used for the Illumina sequencing, was Nextera. Nextera is decent but I and others online have experienced it has a slight bias. It prefers to ligate to certain sequence patterns, this bias is then shown as the GC content bias at the start of reads. Fortunately the rest of the GC content is fine.</p>
<p>In previous tests I have trimmed off the first 20bps to remove this bias and not removed the first 20bp. I found there was little difference in the downstream results for the two methods.</p>
<p>As for the weird "Per sequence GC content" peak, We must remember that the metagenome sample contains many different organisms with different GC contents. Therefore a flat GC peak would be expected.</p>
</div>
<div id="metagenetics" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> 16S rRNA Metagenetics<a href="06-QC_exercises.html#metagenetics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/chime.png" width="20%" style="display: block; margin: auto;" />
Before starting this exercise I must note this is not the normal method to quality control 16S rRNA data. Both QIIME2 and MOTHUR have their own methods but it can be useful to check the quality of 16S rRNA data.</p>
<p>With that in mind we will only use <code>fastqc</code> to check the quality and features of a 16S rRNA dataset.</p>
<p>The dataset can be found in the following link:</p>
<p><a href="https://www.ebi.ac.uk/ena/data/view/SRR2081030" class="uri">https://www.ebi.ac.uk/ena/data/view/SRR2081030</a></p>
<p>The dataset in the cluster is the full dataset. It can be found in your <code>~/QC_workshop/Illumina_Q/exercise_2</code> directory.</p>
<p>Run <code>fastqc</code> on the data and view the reports. Looking at the report you may be worried by its appearance, however this is normal for 16S rRNA data. Some explanations for the features are:</p>
<ul>
<li>The base quality of R2 is very low towards the end. For most 16S rRNA studies the paired reads overlap and so poor quality bases at the end of R1 are corrected by the overlapping R2 reads and vice versa.</li>
<li>The "Per base sequence content" is heavily biased. This again is normal as we are targeting a specific region within the genome, a 16S rRNA region. This means a lot of the sequences will share very similar content. The fact we are targeting a specific region also makes the "Per sequence GC content", "Sequence duplication levels", and "overrepresented sequences" look worrying and warnings are given. These warnings are designed to help with whole genome data and so they are not useful in this circumstance.</li>
</ul>
<p>These features show that it is important to understand what your data contains and what analyses you will be carrying out with the data.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="05-Poorer_quality_data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="07-MultiQC.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
