<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 ONT background | Long read Quality Control</title>
  <meta name="description" content="NEOF book for the Long read quality control workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 ONT background | Long read Quality Control" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Long read quality control workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 ONT background | Long read Quality Control" />
  
  <meta name="twitter:description" content="NEOF book for the Long read quality control workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell &amp; Helen Hipperson" />


<meta name="date" content="2024-09-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="02-Intro.html"/>
<link rel="next" href="04-ONT_setup.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_rounded_corners.png" width="250"></a></li>
<li><a>Long read QC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Long_read_QC.html"><a href="01-Long_read_QC.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="" data-path="01-Long_read_QC.html"><a href="01-Long_read_QC.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Intro.html"><a href="02-Intro.html"><i class="fa fa-check"></i><b>2</b> Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Intro.html"><a href="02-Intro.html#the-data"><i class="fa fa-check"></i><b>2.1</b> The data</a></li>
</ul></li>
<li class="part"><span><b>ONT</b></span></li>
<li class="chapter" data-level="3" data-path="03-ONT_background.html"><a href="03-ONT_background.html"><i class="fa fa-check"></i><b>3</b> ONT background</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-ONT_background.html"><a href="03-ONT_background.html#ont_file_formats"><i class="fa fa-check"></i><b>3.1</b> File formats</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-ONT_background.html"><a href="03-ONT_background.html#pod5"><i class="fa fa-check"></i><b>3.1.1</b> POD5</a></li>
<li class="chapter" data-level="3.1.2" data-path="03-ONT_background.html"><a href="03-ONT_background.html#fast5"><i class="fa fa-check"></i><b>3.1.2</b> Fast5</a></li>
<li class="chapter" data-level="3.1.3" data-path="03-ONT_background.html"><a href="03-ONT_background.html#summary-file"><i class="fa fa-check"></i><b>3.1.3</b> Summary file</a></li>
<li class="chapter" data-level="3.1.4" data-path="03-ONT_background.html"><a href="03-ONT_background.html#bam-file"><i class="fa fa-check"></i><b>3.1.4</b> BAM file</a></li>
<li class="chapter" data-level="3.1.5" data-path="03-ONT_background.html"><a href="03-ONT_background.html#fastq-file"><i class="fa fa-check"></i><b>3.1.5</b> fastq file</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-ONT_background.html"><a href="03-ONT_background.html#ont_basecalling"><i class="fa fa-check"></i><b>3.2</b> Basecalling</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-ONT_setup.html"><a href="04-ONT_setup.html"><i class="fa fa-check"></i><b>4</b> ONT Setup</a></li>
<li class="chapter" data-level="5" data-path="05-NanoStat.html"><a href="05-NanoStat.html"><i class="fa fa-check"></i><b>5</b> NanoStat</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat-run"><i class="fa fa-check"></i><b>5.1</b> NanoStat: run</a></li>
<li class="chapter" data-level="5.2" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat_output"><i class="fa fa-check"></i><b>5.2</b> NanoStat: output</a></li>
<li class="chapter" data-level="5.3" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat-mcqs"><i class="fa fa-check"></i><b>5.3</b> NanoStat: MCQs</a></li>
<li class="chapter" data-level="5.4" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat-summary"><i class="fa fa-check"></i><b>5.4</b> NanoStat: summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Porechop.html"><a href="06-Porechop.html"><i class="fa fa-check"></i><b>6</b> Porechop</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Porechop.html"><a href="06-Porechop.html#porechop-run"><i class="fa fa-check"></i><b>6.1</b> Porechop: run</a></li>
<li class="chapter" data-level="6.2" data-path="06-Porechop.html"><a href="06-Porechop.html#porechop-mcqs"><i class="fa fa-check"></i><b>6.2</b> Porechop: MCQs</a></li>
<li class="chapter" data-level="6.3" data-path="06-Porechop.html"><a href="06-Porechop.html#porechop-summary"><i class="fa fa-check"></i><b>6.3</b> Porechop: summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html"><i class="fa fa-check"></i><b>7</b> NanoPlot</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-run"><i class="fa fa-check"></i><b>7.1</b> NanoPlot: run</a></li>
<li class="chapter" data-level="7.2" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-output"><i class="fa fa-check"></i><b>7.2</b> NanoPlot: output</a></li>
<li class="chapter" data-level="7.3" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-mcqs"><i class="fa fa-check"></i><b>7.3</b> NanoPlot: MCQs</a></li>
<li class="chapter" data-level="7.4" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-summary"><i class="fa fa-check"></i><b>7.4</b> NanoPlot: summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-NanoFilt.html"><a href="08-NanoFilt.html"><i class="fa fa-check"></i><b>8</b> NanoFilt</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-NanoFilt.html"><a href="08-NanoFilt.html#nanofilt-run"><i class="fa fa-check"></i><b>8.1</b> NanoFilt: run</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html"><i class="fa fa-check"></i><b>9</b> Final check</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#final-nanoplot"><i class="fa fa-check"></i><b>9.1</b> Final NanoPlot</a></li>
<li class="chapter" data-level="9.2" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#ont-qc-considerations"><i class="fa fa-check"></i><b>9.2</b> ONT QC considerations</a></li>
<li class="chapter" data-level="9.3" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#nanofilt-reattempts"><i class="fa fa-check"></i><b>9.3</b> NanoFilt Reattempts</a></li>
<li class="chapter" data-level="9.4" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#ont-recap"><i class="fa fa-check"></i><b>9.4</b> ONT Recap</a></li>
</ul></li>
<li class="part"><span><b>PacBio</b></span></li>
<li class="chapter" data-level="10" data-path="10-PB_background.html"><a href="10-PB_background.html"><i class="fa fa-check"></i><b>10</b> PacBio Background</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-PB_background.html"><a href="10-PB_background.html#file-formats"><i class="fa fa-check"></i><b>10.1</b> File formats</a></li>
<li class="chapter" data-level="10.2" data-path="10-PB_background.html"><a href="10-PB_background.html#clr-ccs"><i class="fa fa-check"></i><b>10.2</b> CLR &amp; CCS</a></li>
<li class="chapter" data-level="10.3" data-path="10-PB_background.html"><a href="10-PB_background.html#tools"><i class="fa fa-check"></i><b>10.3</b> Tools</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="10-PB_background.html"><a href="10-PB_background.html#sequeltools"><i class="fa fa-check"></i><b>10.3.1</b> SequelTools</a></li>
<li class="chapter" data-level="10.3.2" data-path="10-PB_background.html"><a href="10-PB_background.html#smrttools"><i class="fa fa-check"></i><b>10.3.2</b> smrttools</a></li>
<li class="chapter" data-level="10.3.3" data-path="10-PB_background.html"><a href="10-PB_background.html#rabbitqcplus"><i class="fa fa-check"></i><b>10.3.3</b> RabbitQCPlus</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-PB_CLR_setup.html"><a href="11-PB_CLR_setup.html"><i class="fa fa-check"></i><b>11</b> CLR setup</a></li>
<li class="chapter" data-level="12" data-path="12-PB_CLR_QC.html"><a href="12-PB_CLR_QC.html"><i class="fa fa-check"></i><b>12</b> CLR QC</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-PB_CLR_QC.html"><a href="12-PB_CLR_QC.html#clr-qc-run"><i class="fa fa-check"></i><b>12.1</b> CLR QC: run</a></li>
<li class="chapter" data-level="12.2" data-path="12-PB_CLR_QC.html"><a href="12-PB_CLR_QC.html#clr-qc-output"><i class="fa fa-check"></i><b>12.2</b> CLR QC: output</a></li>
<li class="chapter" data-level="12.3" data-path="12-PB_CLR_QC.html"><a href="12-PB_CLR_QC.html#clr-qc-mcqs"><i class="fa fa-check"></i><b>12.3</b> CLR QC: MCQs</a></li>
<li class="chapter" data-level="12.4" data-path="12-PB_CLR_QC.html"><a href="12-PB_CLR_QC.html#clr-qc-summary"><i class="fa fa-check"></i><b>12.4</b> CLR QC: summary</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-PB_CLR_filtering.html"><a href="13-PB_CLR_filtering.html"><i class="fa fa-check"></i><b>13</b> CLR filtering</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-PB_CLR_filtering.html"><a href="13-PB_CLR_filtering.html#clr-filtering-run"><i class="fa fa-check"></i><b>13.1</b> CLR filtering: run</a></li>
<li class="chapter" data-level="13.2" data-path="13-PB_CLR_filtering.html"><a href="13-PB_CLR_filtering.html#quality-check-filtered-data"><i class="fa fa-check"></i><b>13.2</b> Quality check filtered data</a></li>
<li class="chapter" data-level="13.3" data-path="13-PB_CLR_filtering.html"><a href="13-PB_CLR_filtering.html#bam-to-fastq"><i class="fa fa-check"></i><b>13.3</b> BAM to fastq</a></li>
<li class="chapter" data-level="13.4" data-path="13-PB_CLR_filtering.html"><a href="13-PB_CLR_filtering.html#clr-filtering-summary"><i class="fa fa-check"></i><b>13.4</b> CLR filtering: summary</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="14-CCS.html"><a href="14-CCS.html"><i class="fa fa-check"></i><b>14</b> Circular Consensus Sequences</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-CCS.html"><a href="14-CCS.html#qc-the-subreads"><i class="fa fa-check"></i><b>14.1</b> QC the subreads</a></li>
<li class="chapter" data-level="14.2" data-path="14-CCS.html"><a href="14-CCS.html#ccs-generation"><i class="fa fa-check"></i><b>14.2</b> CCS generation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="14-CCS.html"><a href="14-CCS.html#ccs-mcqs"><i class="fa fa-check"></i><b>14.2.1</b> CCS MCQs</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="14-CCS.html"><a href="14-CCS.html#rabbitqcplus-1"><i class="fa fa-check"></i><b>14.3</b> RabbitQCPlus</a></li>
<li class="chapter" data-level="14.4" data-path="14-CCS.html"><a href="14-CCS.html#final-summary"><i class="fa fa-check"></i><b>14.4</b> Final summary</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="15-Appendix.html"><a href="15-Appendix.html"><i class="fa fa-check"></i><b>A</b> Further resources</a>
<ul>
<li class="chapter" data-level="A.1" data-path="15-Appendix.html"><a href="15-Appendix.html#ont-1"><i class="fa fa-check"></i><b>A.1</b> ONT</a></li>
<li class="chapter" data-level="A.2" data-path="15-Appendix.html"><a href="15-Appendix.html#pacbio"><i class="fa fa-check"></i><b>A.2</b> PacBio</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="15-Appendix.html"><a href="15-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>B</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="B.1" data-path="15-Appendix.html"><a href="15-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>B.1</b> Mamba installation</a></li>
<li class="chapter" data-level="B.2" data-path="15-Appendix.html"><a href="15-Appendix.html#ont-qc-environment"><i class="fa fa-check"></i><b>B.2</b> ONT QC environment</a></li>
<li class="chapter" data-level="B.3" data-path="15-Appendix.html"><a href="15-Appendix.html#pb-qc-environment"><i class="fa fa-check"></i><b>B.3</b> PB QC environment</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Long read Quality Control</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ont" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> ONT background<a href="03-ONT_background.html#ont" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/ONT_logo.png" style="width:600px; border-radius:15px; border:5px solid white; background:null" />
</center>
<p>In this section we are going to carry out a quick QC of ONT data using the tool suite NanoPack and the tool Porechop. For this we will use the fastq files.</p>
<p>Before carrying out analysis we will cover some background of ONT data. This will include <a href="03-ONT_background.html#ont_file_formats">file formats</a>, &amp; <a href="03-ONT_background.html#ont_basecalling">basecalling</a>.</p>
<div id="ont_file_formats" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> File formats<a href="03-ONT_background.html#ont_file_formats" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/files.png" style="width:200px" />
</center>
<p>Although we will only be using fastq files for this tutorial it is important to know the other file formats you may encounter when working with ONT data.</p>
<div id="pod5" class="section level3 hasAnchor" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> POD5<a href="03-ONT_background.html#pod5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/arrow-logo_chevrons_black-txt_white-bg.png" style="width:150px; border-radius:15px" />
</center>
<p>The POD5 format is the ONT created format to store nanopore DNA data in an easily accessible manner. The files end with the extension <code>.pod5</code> and is normally created by the ont tool <code>MinKNOW</code>. It is a collection of <a href="https://arrow.apache.org/docs/java/table.html">Apache Arrow tables</a> (it is very unlikely you need to understand Apache Arrow tables).</p>
<p>The POD5 file format contains the following information:</p>
<ul>
<li><strong>Reads Table:</strong> Describes the metadata for each read.</li>
<li><strong>Signal Table:</strong> Contains the signal data from the sequencer.</li>
<li><strong>Run Info Table:</strong> Contains information on the <code>MinKNOW</code> runs that created the reads that are in the POD5 file.</li>
</ul>
<p>The POD5 format is quite new and was released as an early preview in May 2022. Prior to POD5 they used Fast5 files. They changed to POD5 with the following design goals in mind:</p>
<ul>
<li>Good write performance for MinKNOW</li>
<li>Recoverable if the writing process crashes</li>
<li>Good read performance for downstream tools, including basecall model generation</li>
<li>Efficient use of space</li>
<li>Straightforward to implement and maintain</li>
<li>Extensibility</li>
</ul>
<p>POD5 links:</p>
<ul>
<li><a href="https://github.com/nanoporetech/pod5-file-format">Main POD5 github page</a></li>
<li><a href="https://github.com/nanoporetech/pod5-file-format/blob/master/docs/DESIGN.md">POD5 File FOrmat Design Details</a></li>
<li><a href="https://github.com/nanoporetech/pod5-file-format/blob/master/docs/SPECIFICATION.md">POD5 Format Specification</a></li>
<li><a href="https://github.com/nanoporetech/pod5-file-format/blob/master/python/pod5/README.md#tools">POD5 Tools commands</a></li>
</ul>
</div>
<div id="fast5" class="section level3 hasAnchor" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> Fast5<a href="03-ONT_background.html#fast5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/fast5.png" style="width:150px" />
</center>
<p>The Fast5 format was the previous format used to store raw data from ONT machine. It is now a legacy format. However, if you use older data for your research you may encounter it.</p>
<p>The format contains the signal data from the pore which can be processed into more useful files.</p>
<p>Fast5 files can contain:</p>
<ul>
<li>Raw signal data</li>
<li>Run metadata</li>
<li>fastq-basecalls</li>
<li>Other additional analyses</li>
</ul>
<p>The Fast5 file format is an implementation of the HDF5 file format specifically for ONT sequencing data. For more information on Fast5 and a tool to interface with these types of file please see:
<a href="https://github.com/nanoporetech/ont_fast5_api" class="uri">https://github.com/nanoporetech/ont_fast5_api</a></p>
<p>You can convert a Fast5 file to a Pod5 file with the pod5 tools command <a href="https://github.com/nanoporetech/pod5-file-format/blob/master/python/pod5/README.md#pod5-convert-fast5"><code>pod5 convert fast5</code></a></p>
</div>
<div id="summary-file" class="section level3 hasAnchor" number="3.1.3">
<h3><span class="header-section-number">3.1.3</span> Summary file<a href="03-ONT_background.html#summary-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The MinION and GridION output a single sequencing summary file in addition to the POD5/Fast5 files. This file contains metdata which describes each sequenced read.</p>
<p>We will not use Fast5 or summary files for this tutorial as they are not needed most of the time. It is likely these files will not be initially provided to you by a sequencing service as they are very large. However, if you do require them you can always ask but be careful with how long the sequencing centre will retain your data.</p>
</div>
<div id="bam-file" class="section level3 hasAnchor" number="3.1.4">
<h3><span class="header-section-number">3.1.4</span> BAM file<a href="03-ONT_background.html#bam-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/bam.png" style="width:100px" />
</center>
<p><strong>BAM</strong> files (<code>.bam</code>) are the binary version of <strong>SAM</strong> files (<code>.sam</code>).</p>
<ul>
<li><strong>SAM</strong> stands for "Sequence Alignment/Map" (the B in <strong>BAM</strong> is Binary).</li>
<li><strong>SAM</strong> files are tab delimited and contain alignment information.</li>
<li><strong>BAM</strong> files are not human readable whilst <strong>SAM</strong> files are.</li>
<li><strong>SAM</strong> files are larger than <strong>BAM</strong>.</li>
</ul>
<p>Generally programs that can work on <strong>SAM</strong> files can also work on <strong>BAM</strong> files. Due to the size difference it is preferable to store data in <strong>BAM</strong> format over <strong>SAM</strong>. Even though a BAM file is smaller than its matching <strong>SAM</strong> file, <strong>BAM</strong> files are still very large and can be &gt;100GB.</p>
<p>It can be useful to contain unaligned reads from sequencing machines (e.g. PacBio and ONT) in <strong>BAM</strong> files as they can contain more metadata in the header and per-record auxiliary tags compared to <strong>fastq</strong> files.</p>
<p>If you are working with <strong>SAM</strong>/<strong>BAM</strong> files the following link will prove useful:
<a href="https://www.htslib.org/" class="uri">https://www.htslib.org/</a></p>
<p>For more information on the <strong>SAM</strong> and <strong>BAM</strong> format please see:
<a href="https://samtools.github.io/hts-specs/SAMv1.pdf" class="uri">https://samtools.github.io/hts-specs/SAMv1.pdf</a></p>
</div>
<div id="fastq-file" class="section level3 hasAnchor" number="3.1.5">
<h3><span class="header-section-number">3.1.5</span> fastq file<a href="03-ONT_background.html#fastq-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/fastq.png" style="width:150px" />
</center>
<p>The <strong>fastq</strong> file format is very consistent and so there is no real difference between <strong>fastq</strong> files for Illumina, PacBio, and ONT data.</p>
<p>All <strong>fastq</strong> files will contain a number of lines divisible by 4. This is because each entry/sequence will take up four lines consisting of the following information:</p>
<ol style="list-style-type: decimal">
<li>Header for <strong>fastq</strong> entry known as the <strong>fastq header</strong>.
<ul>
<li>This always begins with a <strong>'@'</strong></li>
<li>This is where you might see the most difference between different data. Different machines and different public databases will use different formats for <strong>fastq</strong> headers.</li>
</ul></li>
<li>Sequence content</li>
<li>Quality header
<ul>
<li>Always begins with a '+'. Sometimes also contains the same information as <strong>fastq</strong> header.<br />
</li>
</ul></li>
<li>Quality
<ul>
<li>Each base in the 2nd line will have a corresponding quality value in this line.</li>
<li>Note this uses Phred encoding, of which there are different formats. Most, if not all, new files will use Phred+33 but be careful if you are using older data as it may use a different one. See the "Encoding" section in the following link for more info: <a href="https://en.wikipedia.org/wiki/FASTQ_format" class="uri">https://en.wikipedia.org/wiki/FASTQ_format</a>.</li>
<li>NOTE: <strong>'@'</strong> can be used as a quality value.</li>
</ul></li>
</ol>
<p>An example of the information of one sequence/entry is:</p>
<p><strong>@Sequence 1</strong><br />
<strong>CTGTTAAATACCGACTTGCGTCAGGTGCGTGAACAACTGGGCCGCTTT</strong><br />
<strong>+</strong><br />
<strong>=&lt;&lt;&lt;=&gt;@@@ACDCBCDAC@BAA@BA@BBCBBDA@BB@&gt;CD@A@B?B@@</strong></p>
</div>
</div>
<div id="ont_basecalling" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Basecalling<a href="03-ONT_background.html#ont_basecalling" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/guppy.png" style="width:200px" />
</center>
<p>In short, basecalling is the converting of ONT signals to bases and quality. This step converts the <strong>POD5/Fast5</strong> files to <strong>BAM</strong> and/or <strong>fastq</strong> files.</p>
<p>There are many tools to carry out basecalling with ONT data. ONT sequencing machines carry this out with in built programs. However, if you are interested in the tools used for this the primary ones are MinKNOW, Guppy, Dorado, and Albacore. Unfortunately it is quite hard to find information about these tools unless you own an ONT machine.</p>
<p>We will not go into any more detail about basecalling as this is normally carried out by the machine/sequencing centre. Instead please find some useful links below if you would like to know more.</p>
<ul>
<li>MinKNOW
<ul>
<li><a href="https://community.nanoporetech.com/docs/prepare/library_prep_protocols/experiment-companion-minknow/v/mke_1013_v1_revde_11apr2016">Overview on working with MinKNOW</a></li>
<li><a href="https://community.nanoporetech.com/technical_documents/data-analysis/v/datd_5000_v1_revt_22aug2016/basecalling-algorithms">Basecalling algorithms</a></li>
</ul></li>
<li>Guppy
<ul>
<li><a href="https://denbi-nanopore-training-course.readthedocs.io/en/latest/basecalling/index.html">Basecalling with Guppy tutorials</a></li>
<li><a href="https://community.nanoporetech.com/docs/prepare/library_prep_protocols/Guppy-protocol/v/gpb_2003_v1_revax_14dec2018/input-and-output-files">Input and output files</a></li>
</ul></li>
<li>Dorado
<ul>
<li><a href="https://github.com/nanoporetech/dorado">Github page</a></li>
</ul></li>
<li>Literature
<ul>
<li><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-023-02903-2">Comprehensive benchmark and architectural analysis of deep learning models for nanopore sequencing basecalling</a></li>
</ul></li>
</ul>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="02-Intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="04-ONT_setup.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
