<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 NanoPlot | Long read Quality Control</title>
  <meta name="description" content="NEOF book for the Long read quality control workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 NanoPlot | Long read Quality Control" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Long read quality control workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 NanoPlot | Long read Quality Control" />
  
  <meta name="twitter:description" content="NEOF book for the Long read quality control workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell &amp; Helen Hipperson" />


<meta name="date" content="2023-11-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="06-Porechop.html"/>
<link rel="next" href="08-NanoFilt.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Long read QC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Long_read_QC.html"><a href="01-Long_read_QC.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="" data-path="01-Long_read_QC.html"><a href="01-Long_read_QC.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Intro.html"><a href="02-Intro.html"><i class="fa fa-check"></i><b>2</b> Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Intro.html"><a href="02-Intro.html#the-data"><i class="fa fa-check"></i><b>2.1</b> The data</a></li>
</ul></li>
<li class="part"><span><b>ONT</b></span></li>
<li class="chapter" data-level="3" data-path="03-ONT_background.html"><a href="03-ONT_background.html"><i class="fa fa-check"></i><b>3</b> ONT background</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-ONT_background.html"><a href="03-ONT_background.html#ont_file_formats"><i class="fa fa-check"></i><b>3.1</b> File formats</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-ONT_background.html"><a href="03-ONT_background.html#fast5"><i class="fa fa-check"></i><b>3.1.1</b> Fast5</a></li>
<li class="chapter" data-level="3.1.2" data-path="03-ONT_background.html"><a href="03-ONT_background.html#summary-file"><i class="fa fa-check"></i><b>3.1.2</b> Summary file</a></li>
<li class="chapter" data-level="3.1.3" data-path="03-ONT_background.html"><a href="03-ONT_background.html#bam-file"><i class="fa fa-check"></i><b>3.1.3</b> BAM file</a></li>
<li class="chapter" data-level="3.1.4" data-path="03-ONT_background.html"><a href="03-ONT_background.html#fastq-file"><i class="fa fa-check"></i><b>3.1.4</b> fastq file</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-ONT_background.html"><a href="03-ONT_background.html#ont_basecalling"><i class="fa fa-check"></i><b>3.2</b> Basecalling</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-ONT_setup.html"><a href="04-ONT_setup.html"><i class="fa fa-check"></i><b>4</b> ONT Setup</a></li>
<li class="chapter" data-level="5" data-path="05-NanoStat.html"><a href="05-NanoStat.html"><i class="fa fa-check"></i><b>5</b> NanoStat</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat-run"><i class="fa fa-check"></i><b>5.1</b> NanoStat: run</a></li>
<li class="chapter" data-level="5.2" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat_output"><i class="fa fa-check"></i><b>5.2</b> NanoStat: output</a></li>
<li class="chapter" data-level="5.3" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat-mcqs"><i class="fa fa-check"></i><b>5.3</b> NanoStat: MCQs</a></li>
<li class="chapter" data-level="5.4" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat-summary"><i class="fa fa-check"></i><b>5.4</b> NanoStat: summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Porechop.html"><a href="06-Porechop.html"><i class="fa fa-check"></i><b>6</b> Porechop</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Porechop.html"><a href="06-Porechop.html#porechop-run"><i class="fa fa-check"></i><b>6.1</b> Porechop: run</a></li>
<li class="chapter" data-level="6.2" data-path="06-Porechop.html"><a href="06-Porechop.html#porechop-mcqs"><i class="fa fa-check"></i><b>6.2</b> Porechop: MCQs</a></li>
<li class="chapter" data-level="6.3" data-path="06-Porechop.html"><a href="06-Porechop.html#porechop-summary"><i class="fa fa-check"></i><b>6.3</b> Porechop: summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html"><i class="fa fa-check"></i><b>7</b> NanoPlot</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-run"><i class="fa fa-check"></i><b>7.1</b> NanoPlot: run</a></li>
<li class="chapter" data-level="7.2" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-output"><i class="fa fa-check"></i><b>7.2</b> NanoPlot: output</a></li>
<li class="chapter" data-level="7.3" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-mcqs"><i class="fa fa-check"></i><b>7.3</b> NanoPlot: MCQs</a></li>
<li class="chapter" data-level="7.4" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-summary"><i class="fa fa-check"></i><b>7.4</b> NanoPlot: summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-NanoFilt.html"><a href="08-NanoFilt.html"><i class="fa fa-check"></i><b>8</b> NanoFilt</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-NanoFilt.html"><a href="08-NanoFilt.html#nanofilt-run"><i class="fa fa-check"></i><b>8.1</b> NanoFilt: run</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html"><i class="fa fa-check"></i><b>9</b> Final check</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#final-nanoplot"><i class="fa fa-check"></i><b>9.1</b> Final NanoPlot</a></li>
<li class="chapter" data-level="9.2" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#ont-qc-considerations"><i class="fa fa-check"></i><b>9.2</b> ONT QC considerations</a></li>
<li class="chapter" data-level="9.3" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#nanofilt-reattempts"><i class="fa fa-check"></i><b>9.3</b> NanoFilt Reattempts</a></li>
<li class="chapter" data-level="9.4" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#ont-recap"><i class="fa fa-check"></i><b>9.4</b> ONT Recap</a></li>
</ul></li>
<li class="part"><span><b>PacBio</b></span></li>
<li class="chapter" data-level="10" data-path="10-PB_background.html"><a href="10-PB_background.html"><i class="fa fa-check"></i><b>10</b> PacBio Background</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-PB_background.html"><a href="10-PB_background.html#file-formats"><i class="fa fa-check"></i><b>10.1</b> File formats</a></li>
<li class="chapter" data-level="10.2" data-path="10-PB_background.html"><a href="10-PB_background.html#clr-ccs"><i class="fa fa-check"></i><b>10.2</b> CLR &amp; CCS</a></li>
<li class="chapter" data-level="10.3" data-path="10-PB_background.html"><a href="10-PB_background.html#tools"><i class="fa fa-check"></i><b>10.3</b> Tools</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="10-PB_background.html"><a href="10-PB_background.html#sequeltools"><i class="fa fa-check"></i><b>10.3.1</b> SequelTools</a></li>
<li class="chapter" data-level="10.3.2" data-path="10-PB_background.html"><a href="10-PB_background.html#smrttools"><i class="fa fa-check"></i><b>10.3.2</b> smrttools</a></li>
<li class="chapter" data-level="10.3.3" data-path="10-PB_background.html"><a href="10-PB_background.html#rabbitqcplus"><i class="fa fa-check"></i><b>10.3.3</b> RabbitQCPlus</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-PB_CLR_setup.html"><a href="11-PB_CLR_setup.html"><i class="fa fa-check"></i><b>11</b> CLR setup</a></li>
<li class="chapter" data-level="12" data-path="12-PB_CLR_QC.html"><a href="12-PB_CLR_QC.html"><i class="fa fa-check"></i><b>12</b> CLR QC</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-PB_CLR_QC.html"><a href="12-PB_CLR_QC.html#clr-qc-run"><i class="fa fa-check"></i><b>12.1</b> CLR QC: run</a></li>
<li class="chapter" data-level="12.2" data-path="12-PB_CLR_QC.html"><a href="12-PB_CLR_QC.html#clr-qc-output"><i class="fa fa-check"></i><b>12.2</b> CLR QC: output</a></li>
<li class="chapter" data-level="12.3" data-path="12-PB_CLR_QC.html"><a href="12-PB_CLR_QC.html#clr-qc-mcqs"><i class="fa fa-check"></i><b>12.3</b> CLR QC: MCQs</a></li>
<li class="chapter" data-level="12.4" data-path="12-PB_CLR_QC.html"><a href="12-PB_CLR_QC.html#clr-qc-summary"><i class="fa fa-check"></i><b>12.4</b> CLR QC: summary</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-PB_CLR_filtering.html"><a href="13-PB_CLR_filtering.html"><i class="fa fa-check"></i><b>13</b> CLR filtering</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-PB_CLR_filtering.html"><a href="13-PB_CLR_filtering.html#clr-filtering-run"><i class="fa fa-check"></i><b>13.1</b> CLR filtering: run</a></li>
<li class="chapter" data-level="13.2" data-path="13-PB_CLR_filtering.html"><a href="13-PB_CLR_filtering.html#quality-check-filtered-data"><i class="fa fa-check"></i><b>13.2</b> Quality check filtered data</a></li>
<li class="chapter" data-level="13.3" data-path="13-PB_CLR_filtering.html"><a href="13-PB_CLR_filtering.html#bam-to-fastq"><i class="fa fa-check"></i><b>13.3</b> BAM to fastq</a></li>
<li class="chapter" data-level="13.4" data-path="13-PB_CLR_filtering.html"><a href="13-PB_CLR_filtering.html#clr-filtering-summary"><i class="fa fa-check"></i><b>13.4</b> CLR filtering: summary</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="14-CCS.html"><a href="14-CCS.html"><i class="fa fa-check"></i><b>14</b> Circular Consensus Sequences</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-CCS.html"><a href="14-CCS.html#qc-the-subreads"><i class="fa fa-check"></i><b>14.1</b> QC the subreads</a></li>
<li class="chapter" data-level="14.2" data-path="14-CCS.html"><a href="14-CCS.html#ccs-generation"><i class="fa fa-check"></i><b>14.2</b> CCS generation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="14-CCS.html"><a href="14-CCS.html#ccs-mcqs"><i class="fa fa-check"></i><b>14.2.1</b> CCS MCQs</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="14-CCS.html"><a href="14-CCS.html#rabbitqcplus-1"><i class="fa fa-check"></i><b>14.3</b> RabbitQCPlus</a></li>
<li class="chapter" data-level="14.4" data-path="14-CCS.html"><a href="14-CCS.html#final-summary"><i class="fa fa-check"></i><b>14.4</b> Final summary</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="15-Appendix.html"><a href="15-Appendix.html"><i class="fa fa-check"></i><b>A</b> Further resources</a>
<ul>
<li class="chapter" data-level="A.1" data-path="15-Appendix.html"><a href="15-Appendix.html#ont-1"><i class="fa fa-check"></i><b>A.1</b> ONT</a></li>
<li class="chapter" data-level="A.2" data-path="15-Appendix.html"><a href="15-Appendix.html#pacbio"><i class="fa fa-check"></i><b>A.2</b> PacBio</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="15-Appendix.html"><a href="15-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>B</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="B.1" data-path="15-Appendix.html"><a href="15-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>B.1</b> Mamba installation</a></li>
<li class="chapter" data-level="B.2" data-path="15-Appendix.html"><a href="15-Appendix.html#ont-qc-environment"><i class="fa fa-check"></i><b>B.2</b> ONT QC environment</a></li>
<li class="chapter" data-level="B.3" data-path="15-Appendix.html"><a href="15-Appendix.html#pb-qc-environment"><i class="fa fa-check"></i><b>B.3</b> PB QC environment</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Long read Quality Control</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="nanoplotchap" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> NanoPlot<a href="07-NanoPlot.html#nanoplotchap" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/nanoplot_hexes.png" width="15%" style="display: block; margin: auto;" /></p>
<p><code>NanoPlot</code> can be thought of as the <code>fastqc</code> for ONT data. It produces a lot of useful visualisations to investigate the quality of ONT sequencing data. It can be used for <strong>fastq</strong>, <strong>fasta</strong>, <strong>BAM</strong>, and sequencing summary files. The link for its github page is: <a href="https://github.com/wdecoster/NanoPlot" class="uri">https://github.com/wdecoster/NanoPlot</a></p>
<p>ONT data has much lower quality scores than Illumina with Q10 being good. If you have enough coverage and length the low quality can be corrected by downstream processes not covered in this tutorial.</p>
<div id="nanoplot-run" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> NanoPlot: run<a href="07-NanoPlot.html#nanoplot-run" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/enter_key_green.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Prior to running <code>NanoPlot</code> we will make a directory for the <code>NanoPlot</code> output. As <code>NanoPlot</code> creates a lot of files, we'll make a subdirectory for the <code>NanoPlot</code> output for the porechopped data.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb10-1"><a href="07-NanoPlot.html#cb10-1" tabindex="-1"></a><span class="fu">mkdir</span> nanoplot</span>
<span id="cb10-2"><a href="07-NanoPlot.html#cb10-2" tabindex="-1"></a><span class="fu">mkdir</span> nanoplot/porechop</span></code></pre></div>
<p>Now to run <code>NanoPlot</code>. The options we will use are:</p>
<ul>
<li><code>-t</code>: Number of threads to be used.</li>
<li><code>--fastq</code>: Specifies the input path which is a fastq file.</li>
<li><code>-o</code>: Directory where the output will be created.</li>
<li><code>-p</code>: Prefix of output files. It is useful to have "_" at the end of the prefix.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb11-1"><a href="07-NanoPlot.html#cb11-1" tabindex="-1"></a><span class="ex">NanoPlot</span> <span class="at">-t</span> 4 <span class="dt">\</span></span>
<span id="cb11-2"><a href="07-NanoPlot.html#cb11-2" tabindex="-1"></a>--fastq porechop/Acinetobacter.porechop.fastq <span class="dt">\</span></span>
<span id="cb11-3"><a href="07-NanoPlot.html#cb11-3" tabindex="-1"></a>-o nanoplot/porechop <span class="at">-p</span> Acinetobacter_ </span></code></pre></div>
</div>
<div id="nanoplot-output" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> NanoPlot: output<a href="07-NanoPlot.html#nanoplot-output" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/output_file_green.png" width="15%" style="display: block; margin: auto;" /></p>
<p>List the files in the output directory.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb12-1"><a href="07-NanoPlot.html#cb12-1" tabindex="-1"></a><span class="fu">ls</span> nanoplot/porechop/</span></code></pre></div>
<p>There are quite a few files. These should all start with "Acinetobacter_" thanks to the <code>-p</code> option. To quickly check all the results we can open the report html file with firefox.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb13-1"><a href="07-NanoPlot.html#cb13-1" tabindex="-1"></a><span class="ex">firefox</span> nanoplot/porechop/Acinetobacter_NanoPlot-report.html</span></code></pre></div>
<p>The first section contains <a href="05-NanoStat.html#nanostat_output"><code>NanoStat</code> output</a>. Quickly look over this and see how it compares to the <code>NanoStat</code> output of the pre-porechopped reads.</p>
<div class="webex-solution">
<button>
NanoStat of pre-porechopped reads
</button>
<table>
<thead>
<tr>
<th style="text-align:left;">
General summary
</th>
<th style="text-align:left;">
values
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Mean read length
</td>
<td style="text-align:left;">
7,070.8
</td>
</tr>
<tr>
<td style="text-align:left;">
Mean read quality
</td>
<td style="text-align:left;">
9.5
</td>
</tr>
<tr>
<td style="text-align:left;">
Median read length
</td>
<td style="text-align:left;">
6,700.5
</td>
</tr>
<tr>
<td style="text-align:left;">
Median read quality
</td>
<td style="text-align:left;">
10.4
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of reads
</td>
<td style="text-align:left;">
2,000.0
</td>
</tr>
<tr>
<td style="text-align:left;">
Read length N50
</td>
<td style="text-align:left;">
8,143.0
</td>
</tr>
<tr>
<td style="text-align:left;">
STDEV read length
</td>
<td style="text-align:left;">
3,336.0
</td>
</tr>
<tr>
<td style="text-align:left;">
Total bases
</td>
<td style="text-align:left;">
14,141,646.0
</td>
</tr>
<tr>
<td style="text-align:left;">
Number, percentage and megabases of reads above quality cutoffs
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
&gt;Q5
</td>
<td style="text-align:left;">
2000 (100.0%) 14.1Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
&gt;Q7
</td>
<td style="text-align:left;">
1998 (99.9%) 14.1Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
&gt;Q10
</td>
<td style="text-align:left;">
1231 (61.6%) 8.8Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
&gt;Q12
</td>
<td style="text-align:left;">
29 (1.5%) 0.2Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
&gt;Q15
</td>
<td style="text-align:left;">
0 (0.0%) 0.0Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
Top 5 highest mean basecall quality scores and their read lengths
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
12.3 (7039)
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
12.3 (5770)
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
12.3 (8052)
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
12.2 (8120)
</td>
</tr>
<tr>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
12.2 (3041)
</td>
</tr>
<tr>
<td style="text-align:left;">
Top 5 longest reads and their mean basecall quality score
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
34661 (9.2)
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
31549 (8.5)
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
25072 (10.0)
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
24871 (10.3)
</td>
</tr>
<tr>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
23185 (10.7)
</td>
</tr>
</tbody>
</table>
</div>
<p>After the <strong>Summary Statistics</strong> section there is a <strong>Plots</strong> section.
These are all plots created by <a href="https://plotly.com/">plotly</a> meaning they are interactive.
With these plots you can:</p>
<ul>
<li>Make boxes to zoom into a specific area.</li>
<li>Click and drag left or right only to zoom into a specific part of the x axis.</li>
<li>Click and drag up or down only to zoom into a specific part of the y axis.</li>
<li>Click the home icon (<svg viewBox="0 0 512 512" style="height:1em;position:relative;display:inline-block;top:.1em;" xmlns="http://www.w3.org/2000/svg"> <path d="M261.56,101.28a8,8,0,0,0-11.06,0L66.4,277.15a8,8,0,0,0-2.47,5.79L63.9,448a32,32,0,0,0,32,32H192a16,16,0,0,0,16-16V328a8,8,0,0,1,8-8h80a8,8,0,0,1,8,8l0,136a16,16,0,0,0,16,16h96.06a32,32,0,0,0,32-32l0-165.06a8,8,0,0,0-2.47-5.79Z"></path> <path d="M490.91,244.15l-74.8-71.56,0-108.59a16,16,0,0,0-16-16h-48a16,16,0,0,0-16,16l0,32L278.19,40.62C272.77,35.14,264.71,32,256,32h0c-8.68,0-16.72,3.14-22.14,8.63L21.16,244.13c-6.22,6-7,15.87-1.34,22.37A16,16,0,0,0,43,267.56L250.5,69.28a8,8,0,0,1,11.06,0L469.08,267.56a16,16,0,0,0,22.59-.44C497.81,260.76,497.3,250.26,490.91,244.15Z"></path></svg>) on the top right to reset the axes.</li>
<li>Hover over a points/bars to see the specific values of it.</li>
</ul>
<p>The plots are:</p>
<ul>
<li><strong>Weighted Histogram of read lengths</strong>
<ul>
<li>Histogram of "Number of <em>bases</em>" (y) against "Read length" (x).</li>
</ul></li>
<li><strong>Weighted Histogram of read lengths after log transformation</strong>
<ul>
<li>Histogram of "Number of <em>bases</em>" (y) against log transformed "Read length" (x).</li>
</ul></li>
<li><strong>Non weighted histogram of read lengths</strong>
<ul>
<li>Histogram of "Number of <em>reads</em>" (y) against "Read length" (x).</li>
</ul></li>
<li><strong>Non weighted histogram of read lengths after log transformations</strong>
<ul>
<li>Histogram of "Number of <em>reads</em>" (y) against log transformed "Read length" (x).</li>
</ul></li>
<li><strong>Yield by length</strong>
<ul>
<li>Plot showing the "Cumulative yield for minimal length" (y) by "Read length" (x).</li>
<li>The cumulative yield is measured in Gigabases (billion bases).</li>
<li>This plot is useful to know how many bases you would retain if you filtered reads based on read length.</li>
</ul></li>
<li><strong>Read lengths vs Average read quality plot using dots</strong>
<ul>
<li>Each dot represents a single read.</li>
<li>At the top of the plot is a histogram of Number of reads against read lengths.</li>
<li>At the right of the plot is a sideways histogram of number of reads against average read quality.</li>
</ul></li>
<li><strong>Read lengths vs Average read quality kde plot</strong>
<ul>
<li>Each dot represents a single read with a topology map showing the density of plots. THe darker the area the more dense.</li>
<li>This is the most informative plot.</li>
<li>At the top of the plot is a histogram of Number of reads against read lengths.</li>
<li>At the right of the plot is a sideways histogram of number of reads against average read quality.</li>
</ul></li>
</ul>
</div>
<div id="nanoplot-mcqs" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> NanoPlot: MCQs<a href="07-NanoPlot.html#nanoplot-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/question_bubble_green.png" width="15%" style="display: block; margin: auto;" />
Please attempt the below MCQs with the information in the report html.</p>
<ol style="list-style-type: decimal">
<li>What is the mean read quality for all the data?
<div id="radio_OPHJYAKCXZ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_OPHJYAKCXZ" value="answer"></input> <span><strong>9.5</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OPHJYAKCXZ" value=""></input> <span><strong>6,962.1</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OPHJYAKCXZ" value=""></input> <span><strong>8.8Mb</strong></span></label>
</div></li>
<li>What is the mean read length of the data?
<div id="radio_TQQEFMRDVJ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_TQQEFMRDVJ" value=""></input> <span><strong>9.5</strong></span></label><label><input type="radio" autocomplete="off" name="radio_TQQEFMRDVJ" value="answer"></input> <span><strong>6,962.1</strong></span></label><label><input type="radio" autocomplete="off" name="radio_TQQEFMRDVJ" value=""></input> <span><strong>8.8Mb</strong></span></label>
</div></li>
<li>How many Megabases are within reads with a quality cutoff of &gt;Q10?
<div id="radio_GGWUQRQZBK" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_GGWUQRQZBK" value=""></input> <span><strong>9.5</strong></span></label><label><input type="radio" autocomplete="off" name="radio_GGWUQRQZBK" value=""></input> <span><strong>6,962.1</strong></span></label><label><input type="radio" autocomplete="off" name="radio_GGWUQRQZBK" value="answer"></input> <span><strong>8.8Mb</strong></span></label>
</div></li>
<li>Approximately how many bases would you retain if you filtered out/removed reads with a length &lt;10,000bp (Yield by length)?
<div id="radio_BOARFXXFRO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BOARFXXFRO" value=""></input> <span><strong>10</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BOARFXXFRO" value=""></input> <span><strong>146</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BOARFXXFRO" value="answer"></input> <span><strong>4Mb (0.004Gb)</strong></span></label>
</div></li>
<li>Look at the "Read lengths vs Average read quality plot". In what average quality range are the majority of reads?
<div id="radio_QUEMMVPXLO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_QUEMMVPXLO" value=""></input> <span><strong>6-8</strong></span></label><label><input type="radio" autocomplete="off" name="radio_QUEMMVPXLO" value=""></input> <span><strong>8-10</strong></span></label><label><input type="radio" autocomplete="off" name="radio_QUEMMVPXLO" value="answer"></input> <span><strong>10-12</strong></span></label>
</div></li>
</ol>
</div>
<div id="nanoplot-summary" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> NanoPlot: summary<a href="07-NanoPlot.html#nanoplot-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_green.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We have use <code>NanoPlot</code> to produce read quality and length stats in table and plot form. Our data still looks good after porechopping but we still need to carry out some filtering.</p>
<p>You can use <code>NanoPlot</code> with different input files. This will give you different plots. For more details please see the "Plots Generated" section on: <a href="https://github.com/wdecoster/NanoPlot" class="uri">https://github.com/wdecoster/NanoPlot</a>.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="06-Porechop.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="08-NanoFilt.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
