<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Continuous Long Reads | Long read Quality Control</title>
  <meta name="description" content="NEOF book for the Long read quality control workshop" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Continuous Long Reads | Long read Quality Control" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Long read quality control workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Continuous Long Reads | Long read Quality Control" />
  
  <meta name="twitter:description" content="NEOF book for the Long read quality control workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell &amp; Helen Hipperson" />


<meta name="date" content="2022-08-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="10-PB_background.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.0.0/tabwid.css" rel="stylesheet" />
<link href="libs/tabwid-1.0.0/scrool.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Long read QC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Long_read_QC.html"><a href="01-Long_read_QC.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="02-Intro.html"><a href="02-Intro.html"><i class="fa fa-check"></i><b>2</b> Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Intro.html"><a href="02-Intro.html#the-data"><i class="fa fa-check"></i><b>2.1</b> The data</a></li>
</ul></li>
<li class="part"><span><b>ONT</b></span></li>
<li class="chapter" data-level="3" data-path="03-ONT_background.html"><a href="03-ONT_background.html"><i class="fa fa-check"></i><b>3</b> ONT background</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-ONT_background.html"><a href="03-ONT_background.html#ont_file_formats"><i class="fa fa-check"></i><b>3.1</b> File formats</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-ONT_background.html"><a href="03-ONT_background.html#fast5"><i class="fa fa-check"></i><b>3.1.1</b> Fast5</a></li>
<li class="chapter" data-level="3.1.2" data-path="03-ONT_background.html"><a href="03-ONT_background.html#summary-file"><i class="fa fa-check"></i><b>3.1.2</b> Summary file</a></li>
<li class="chapter" data-level="3.1.3" data-path="03-ONT_background.html"><a href="03-ONT_background.html#bam-file"><i class="fa fa-check"></i><b>3.1.3</b> BAM file</a></li>
<li class="chapter" data-level="3.1.4" data-path="03-ONT_background.html"><a href="03-ONT_background.html#fastq-file"><i class="fa fa-check"></i><b>3.1.4</b> fastq file</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-ONT_background.html"><a href="03-ONT_background.html#ont_basecalling"><i class="fa fa-check"></i><b>3.2</b> Basecalling</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-ONT_setup.html"><a href="04-ONT_setup.html"><i class="fa fa-check"></i><b>4</b> ONT Setup</a></li>
<li class="chapter" data-level="5" data-path="05-NanoStat.html"><a href="05-NanoStat.html"><i class="fa fa-check"></i><b>5</b> NanoStat</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat-run"><i class="fa fa-check"></i><b>5.1</b> NanoStat: run</a></li>
<li class="chapter" data-level="5.2" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat_output"><i class="fa fa-check"></i><b>5.2</b> NanoStat: output</a></li>
<li class="chapter" data-level="5.3" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat-mcqs"><i class="fa fa-check"></i><b>5.3</b> NanoStat: MCQs</a></li>
<li class="chapter" data-level="5.4" data-path="05-NanoStat.html"><a href="05-NanoStat.html#nanostat-summary"><i class="fa fa-check"></i><b>5.4</b> NanoStat: summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Porechop.html"><a href="06-Porechop.html"><i class="fa fa-check"></i><b>6</b> Porechop</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Porechop.html"><a href="06-Porechop.html#porechop-run"><i class="fa fa-check"></i><b>6.1</b> Porechop: run</a></li>
<li class="chapter" data-level="6.2" data-path="06-Porechop.html"><a href="06-Porechop.html#porechop-mcqs"><i class="fa fa-check"></i><b>6.2</b> Porechop: MCQs</a></li>
<li class="chapter" data-level="6.3" data-path="06-Porechop.html"><a href="06-Porechop.html#porechop-summary"><i class="fa fa-check"></i><b>6.3</b> Porechop: summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html"><i class="fa fa-check"></i><b>7</b> NanoPlot</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-run"><i class="fa fa-check"></i><b>7.1</b> NanoPlot: run</a></li>
<li class="chapter" data-level="7.2" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-output"><i class="fa fa-check"></i><b>7.2</b> NanoPlot: output</a></li>
<li class="chapter" data-level="7.3" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-mcqs"><i class="fa fa-check"></i><b>7.3</b> NanoPlot: MCQs</a></li>
<li class="chapter" data-level="7.4" data-path="07-NanoPlot.html"><a href="07-NanoPlot.html#nanoplot-summary"><i class="fa fa-check"></i><b>7.4</b> NanoPlot: summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-NanoFilt.html"><a href="08-NanoFilt.html"><i class="fa fa-check"></i><b>8</b> NanoFilt</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-NanoFilt.html"><a href="08-NanoFilt.html#nanofilt-run"><i class="fa fa-check"></i><b>8.1</b> NanoFilt: run</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html"><i class="fa fa-check"></i><b>9</b> Final check</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#final-nanoplot"><i class="fa fa-check"></i><b>9.1</b> Final NanoPlot</a></li>
<li class="chapter" data-level="9.2" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#ont-qc-considerations"><i class="fa fa-check"></i><b>9.2</b> ONT QC considerations</a></li>
<li class="chapter" data-level="9.3" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#nanofilt-reattempts"><i class="fa fa-check"></i><b>9.3</b> NanoFilt Reattempts</a></li>
<li class="chapter" data-level="9.4" data-path="09-ONT_final_check.html"><a href="09-ONT_final_check.html#ont-recap"><i class="fa fa-check"></i><b>9.4</b> ONT Recap</a></li>
</ul></li>
<li class="part"><span><b>PacBio</b></span></li>
<li class="chapter" data-level="10" data-path="10-PB_background.html"><a href="10-PB_background.html"><i class="fa fa-check"></i><b>10</b> PacBio Background</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-PB_background.html"><a href="10-PB_background.html#file-formats"><i class="fa fa-check"></i><b>10.1</b> File formats</a></li>
<li class="chapter" data-level="10.2" data-path="10-PB_background.html"><a href="10-PB_background.html#clr-ccs"><i class="fa fa-check"></i><b>10.2</b> CLR &amp; CCS</a></li>
<li class="chapter" data-level="10.3" data-path="10-PB_background.html"><a href="10-PB_background.html#tools"><i class="fa fa-check"></i><b>10.3</b> Tools</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="10-PB_background.html"><a href="10-PB_background.html#sequeltools"><i class="fa fa-check"></i><b>10.3.1</b> SequelTools</a></li>
<li class="chapter" data-level="10.3.2" data-path="10-PB_background.html"><a href="10-PB_background.html#smrttools"><i class="fa fa-check"></i><b>10.3.2</b> smrttools</a></li>
<li class="chapter" data-level="10.3.3" data-path="10-PB_background.html"><a href="10-PB_background.html#rabbitqc"><i class="fa fa-check"></i><b>10.3.3</b> RabbitQC</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-PB_CLR.html"><a href="11-PB_CLR.html"><i class="fa fa-check"></i><b>11</b> Continuous Long Reads</a>
<ul>
<li class="chapter" data-level="11.0.1" data-path="11-PB_CLR.html"><a href="11-PB_CLR.html#quality-check"><i class="fa fa-check"></i><b>11.0.1</b> Quality check</a></li>
<li class="chapter" data-level="11.0.2" data-path="11-PB_CLR.html"><a href="11-PB_CLR.html#filtering"><i class="fa fa-check"></i><b>11.0.2</b> Filtering</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Long read Quality Control</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="continuous-long-reads" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Continuous Long Reads<a href="11-PB_CLR.html#continuous-long-reads" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/clr.png" width="30%" style="display: block; margin: auto;" /></p>
<p>Before carrying out any specific commands we will first move into the relevant directory.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="11-PB_CLR.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/QC_workshop/PB_QC/data/CLR</span></code></pre></div>
<p>Use <code>ls</code> to list the contents of this directory. You will see that there are BAM subreads and scraps files for three samples, plus their BAM index files. These data were generated from high molecular weight DNA for <em>de novo</em> genome assembly.</p>
<p>We will use the SequelTools program to both assess and filter our PacBio CLR data. First we need to make a file of filenames (fofn) for the subreads files. You can do this with the following commands:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="11-PB_CLR.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span> <span class="at">-name</span> <span class="st">&quot;*subreads.bam&quot;</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="op">&gt;</span> CLR_subreads.txt</span></code></pre></div>
<p>This command line will find all of the files in our current working directory whose names end in 'subreads.bam', sorts them alphanumerically and prints these names into a new text file.</p>
<div id="quality-check" class="section level3 hasAnchor" number="11.0.1">
<h3><span class="header-section-number">11.0.1</span> Quality check<a href="11-PB_CLR.html#quality-check" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/qc_inspection.png" width="15%" style="display: block; margin: auto;" /></p>
<p>The SequelTools program has three different tools, specified with the <code>-t</code> argument, which are:</p>
<ul>
<li><code>Q</code> for quality control</li>
<li><code>S</code> for subsampling the data</li>
<li><code>F</code> for filtering the data</li>
</ul>
<p>We will first use <code>Q</code> to assess our data. Other options used are:</p>
<ul>
<li><code>-u</code> to specify the file containing the list of subread .bam files</li>
<li><code>-o</code> to specify the name of an output folder for the plots</li>
<li><code>-p</code> to specify which plots to produce: <code>b</code> for a few basic plots, <code>i</code> includes a few more detailed plots and <code>a</code> generates all available plots.</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="11-PB_CLR.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">SequelTools.sh</span> <span class="at">-t</span> Q <span class="at">-p</span> a <span class="at">-u</span> CLR_subreads.txt <span class="at">-o</span> CLR_QC</span></code></pre></div>
<p>When this has finished running the CLR_QC folder will contain a 'summaryTable.txt' file with values on the number and lengths of sequence reads for all three samples, plus a series of plots saved as pdf files. We can use firefox to view the pdf plots.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="11-PB_CLR.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="ex">firefox</span> CLR_QC/totalBasesBarplot.pdf</span></code></pre></div>
<p>This shows the total amount of sequence data for each of the three samples, both for all of the subreads present and 'longestSubs' (the longest subread within each CLR). We can see that Sample2 has the largest amount of data, and for all three samples most of the data is contained within the longest subreads. This is expected with CLR data - long fragments of DNA are extracted for sequenicng and we often achieve just a single pass of this insert during the PacBio sequenicng run.</p>
<p>There are also plots for the N50 and L50 of each sample:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="11-PB_CLR.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">firefox</span> CLR_QC/n50s.pdf CLR_QC/l50s.pdf</span></code></pre></div>
<ul>
<li>N50 = the median sequence length (in bp) of the data; 50% of the sequence data generated is in subreads equal to or greater than this length.</li>
<li>L50 = the minimum number of subreads whose length makes up the N50 value.</li>
</ul>
<p>We can see that the N50 is larger for Sample3 compared to Sample1 and Sample2, suggesting the subreads are longer for Sample3. Conversely, the L50 is higher for Sample2, suggesting the subreads are shorter for this sample as more subreads are required to make up the N50 value.</p>
<p>We can see information on the subread lengths in more detail as boxplots and histograms in the following plots:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="11-PB_CLR.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="ex">firefox</span> CLR_QC/subreadSizesBoxplots.pdf CLR_QC/<span class="pp">*</span>Hists.pdf</span></code></pre></div>
<p>The boxplots show that Sample3 does indeed have the longest subread lengths, and highest N50 as indicated by the blue diamond. Sample2 has a larger range of subread lengths than Sample1, but has a slightly lower median length and N50 value.</p>
<p>The histograms show the distribution of subread lengths in more detail. Sample3 shows a big spike of very short subreads - why do you think this is? (We'll come back to this after the filtering step!)</p>
</div>
<div id="filtering" class="section level3 hasAnchor" number="11.0.2">
<h3><span class="header-section-number">11.0.2</span> Filtering<a href="11-PB_CLR.html#filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/quality_filtering.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Our quality check of the PacBio data doesn't include any assessment of sequence quality. Unaligned PacBio data doesn't have a quality score in the same way as the Illumina data we looked at on Tuesday. Quality scores for PacBio are generated when the reads are aligned - either to a reference sequence or subreads aligned to each other to generate CCS reads. However, it is often a good idea to use just the longest PacBio reads for a <em>de novo</em> assembly.</p>
<p>To filter the data by minimum CLR length we will use SequelTools option <code>-t F</code>.</p>
<p>Other options used are:</p>
<ul>
<li><code>-u</code> to specify the file containing the list of subread .bam files</li>
<li><code>-c</code> to specify the file containing the list of scraps .bam files</li>
<li><code>-C</code> to specify to filter by minimum CLR length</li>
<li><code>-Z</code> to specify the minimum length to keep a CLR</li>
<li><code>-f</code> to specify the format of the output files: <code>s</code>=sam, <code>b</code>=bam, <code>2</code>=both</li>
<li><code>-o</code> to specify the name of an output folder for the filtered data files</li>
</ul>
<p>First we need to make a fofn for the scraps.bam files, as these are needed when using the filtering tool:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="11-PB_CLR.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span> <span class="at">-name</span> <span class="st">&quot;*scraps.bam&quot;</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="op">&gt;</span> CLR_scraps.txt</span></code></pre></div>
<p>Then we can run the filtering command. Here we will keep only CLRs of at least 10,000 bp:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="11-PB_CLR.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="ex">SequelTools.sh</span> <span class="at">-t</span> F <span class="at">-u</span> CLR_subreads.txt <span class="at">-c</span> CLR_scraps.txt <span class="dt">\</span></span>
<span id="cb26-2"><a href="11-PB_CLR.html#cb26-2" aria-hidden="true" tabindex="-1"></a>-C <span class="at">-Z</span> 10000 <span class="at">-f</span> b <span class="at">-o</span> filtered</span></code></pre></div>
<p>When this has finished running the 'filtered' folder will contain filtered BAM files for all three samples. Move into this folder using <code>cd</code>:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="11-PB_CLR.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> filtered</span></code></pre></div>
<p>From here let's run the quality control tool on the filtered files and compare the plots to those from the raw files. First we need a new fofn for the filtered subread files:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="11-PB_CLR.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span> <span class="at">-name</span> <span class="st">&quot;*subreads.bam&quot;</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="op">&gt;</span> filt_subreads.txt</span></code></pre></div>
<p>Then run the quality control tool again:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="11-PB_CLR.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="ex">SequelTools.sh</span> <span class="at">-t</span> Q <span class="at">-p</span> a <span class="at">-u</span> filt_subreads.txt <span class="at">-o</span> filt_QC</span></code></pre></div>
<p>When this has finished running the 'filt_QC' folder will contain a text file of summary values and a series of plots saved as pdf files. We can view these as before, for example:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="11-PB_CLR.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="ex">firefox</span> filt_QC/subreadSizesBoxplots.pdf</span></code></pre></div>
<p>And compare this to the boxplots from the unfiltered data:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb31-1"><a href="11-PB_CLR.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="ex">firefox</span> ~/QC_workshop/PB_QC/data/CLR/CLR_QC/subreadSizesBoxplots.pdf</span></code></pre></div>
<p>The difference looks quite subtle on the plots as the size range of reads is large, but the subread N50 value (indicated by the blue diamond) has increased by ~2,000 bp in the filtered data for Sample2 and Sample3.</p>
<p>There are also still subreads present in the data that are &lt; 10,000 bp. This is because we have filtered out CLRs that are &lt; 10,000 bp, but a CLR &gt; 10,000bp can still be made up of subreads smaller than this.</p>
<p>This is definitely the case for Sample1 and Sample2, where the DNA fragment length of the library was shorter than the CLR lengths, hence there are subreads &lt; 10,000 bp within longer CLRs.</p>
<p>Sample3 shows a more obvious difference before and after filtering:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb32-1"><a href="11-PB_CLR.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="ex">firefox</span> ~/QC_workshop/PB_QC/data/CLR/CLR_QC/Sample3.readLenHists.pdf <span class="dt">\</span></span>
<span id="cb32-2"><a href="11-PB_CLR.html#cb32-2" aria-hidden="true" tabindex="-1"></a>filt_QC/Sample3_flt.readLenHists.pdf</span></code></pre></div>
<p>For this sample the DNA fragment length of the library was much greater than 10,000 bp, and so the majority of CLRs consist of just one pass of this long insert. The large number of short subreads remaining in the filtered file (blue bars) represent subreads only partially sequenced on the 'return journey' of the circular library DNA molecule. These are not present when we look only at the longest subread present in each CLR (green bars).</p>
<p>SequelTools doesn't have an option for filtering subreads by length. To do this we can convert the filtered BAM file to a fastq or fasta file and use another downstream tool. The fastq file will not contain any useful information on sequence quality (every base is assigned a quality character of '!' which is equal to zero), but this file format is useful if only fastq format is accepted as input for a downstream tool.</p>
<p>To do this use the commands <code>pbindex</code> and <code>bam2fastq</code> from the smrttools package:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb33-1"><a href="11-PB_CLR.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pbindex</span> Sample1_flt.subreads.bam</span></code></pre></div>
<p>Note that you may see a long warning message ending with "No such file or directory". However, if you run <code>ls</code> you should see that the output file has been generated.</p>
<p>This command generates a .pbi index file that is needed before we can convert the BAM file to fastq format. The <code>bam2fastq</code> command requires an output file prefix name, specified by <code>-o</code>, and the name of the BAM file we want to convert:</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb34-1"><a href="11-PB_CLR.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bam2fastq</span> <span class="at">-o</span> Sample1_filtered Sample1_flt.subreads.bam</span></code></pre></div>
<p>You should now have a compressed fastq file for the CLR-filtered subreads from Sample1.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="10-PB_background.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
