<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Kraken2 | Shotgun Metagenomics</title>
  <meta name="description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Kraken2 | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Kraken2 | Shotgun Metagenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2025-04-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="08-Taxonomic_profiling.html"/>
<link rel="next" href="10-Krona.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Start_conda.html"><a href="04-Start_conda.html"><i class="fa fa-check"></i><b>4</b> Startup &amp; mamba</a></li>
<li class="part"><span><b>Data &amp; QC</b></span></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-summary"><i class="fa fa-check"></i><b>7.5</b> Host removal summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy of reads</b></span></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a></li>
<li class="chapter" data-level="9" data-path="09-Kraken2.html"><a href="09-Kraken2.html"><i class="fa fa-check"></i><b>9</b> Kraken2</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-run"><i class="fa fa-check"></i><b>9.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="9.2" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-output"><i class="fa fa-check"></i><b>9.2</b> Kraken2: output</a></li>
<li class="chapter" data-level="9.3" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken-2-mcqs"><i class="fa fa-check"></i><b>9.3</b> Kraken 2: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Krona.html"><a href="10-Krona.html"><i class="fa fa-check"></i><b>10</b> Krona plot</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Krona.html"><a href="10-Krona.html#krona-run"><i class="fa fa-check"></i><b>10.1</b> Krona: run</a></li>
<li class="chapter" data-level="10.2" data-path="10-Krona.html"><a href="10-Krona.html#krona-visualise"><i class="fa fa-check"></i><b>10.2</b> Krona: visualise</a></li>
<li class="chapter" data-level="10.3" data-path="10-Krona.html"><a href="10-Krona.html#krona-mcqs"><i class="fa fa-check"></i><b>10.3</b> Krona: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Bracken.html"><a href="11-Bracken.html"><i class="fa fa-check"></i><b>11</b> Bracken</a>
<ul>
<li class="chapter" data-level="11.0.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-run"><i class="fa fa-check"></i><b>11.0.1</b> Bracken: run</a></li>
<li class="chapter" data-level="11.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-output"><i class="fa fa-check"></i><b>11.1</b> Bracken: output</a></li>
<li class="chapter" data-level="11.2" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-mcqs"><i class="fa fa-check"></i><b>11.2</b> Bracken: MCQs</a></li>
<li class="chapter" data-level="11.3" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-merging-output"><i class="fa fa-check"></i><b>11.3</b> Bracken: merging output</a></li>
<li class="chapter" data-level="11.4" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-extracting-output"><i class="fa fa-check"></i><b>11.4</b> Bracken: extracting output</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Lefse.html"><a href="12-Lefse.html"><i class="fa fa-check"></i><b>12</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2_lefse_metadata"><i class="fa fa-check"></i><b>12.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="12.2" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-format"><i class="fa fa-check"></i><b>12.2</b> LEfSe: format</a></li>
<li class="chapter" data-level="12.3" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-run"><i class="fa fa-check"></i><b>12.3</b> LEfSe: run</a></li>
<li class="chapter" data-level="12.4" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-visualisation"><i class="fa fa-check"></i><b>12.4</b> LEfSe: visualisation</a></li>
<li class="chapter" data-level="12.5" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-mcqs"><i class="fa fa-check"></i><b>12.5</b> LEfSe: MCQs</a></li>
<li class="chapter" data-level="12.6" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>12.6</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="part"><span><b>Functional profiling of reads</b></span></li>
<li class="chapter" data-level="13" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html"><i class="fa fa-check"></i><b>13</b> HUMAnN Functional profiling</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann"><i class="fa fa-check"></i><b>13.1</b> HUMAnN</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-mamba-directories-and-files"><i class="fa fa-check"></i><b>13.1.1</b> HUMAnN: mamba, directories, and files</a></li>
<li class="chapter" data-level="13.1.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-run"><i class="fa fa-check"></i><b>13.1.2</b> HUMAnN: run</a></li>
<li class="chapter" data-level="13.1.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-output"><i class="fa fa-check"></i><b>13.1.3</b> HUMAnN: output</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humannmultisampleprocessing"><i class="fa fa-check"></i><b>13.2</b> Multi sample processing</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-combining-data"><i class="fa fa-check"></i><b>13.2.1</b> HUMAnN: combining data</a></li>
<li class="chapter" data-level="13.2.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-split-stratified-table"><i class="fa fa-check"></i><b>13.2.2</b> HUMAnN: split stratified table</a></li>
<li class="chapter" data-level="13.2.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humannrenormalise"><i class="fa fa-check"></i><b>13.2.3</b> HUMAnN: renormalising data</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#heatmap"><i class="fa fa-check"></i><b>13.3</b> Heatmap</a></li>
<li class="chapter" data-level="13.4" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann_lefse"><i class="fa fa-check"></i><b>13.4</b> LEfSe</a></li>
<li class="chapter" data-level="13.5" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#optional-task"><i class="fa fa-check"></i><b>13.5</b> Optional task</a></li>
</ul></li>
<li class="part"><span><b>Assembly Approach</b></span></li>
<li class="chapter" data-level="14" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html"><i class="fa fa-check"></i><b>14</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html#metagenome-assembly-mamba"><i class="fa fa-check"></i><b>14.1</b> Metagenome assembly: Mamba</a></li>
<li class="chapter" data-level="14.2" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>14.2</b> A primer on short read assembly</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html"><i class="fa fa-check"></i><b>15</b> Stitching read pairs</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-run"><i class="fa fa-check"></i><b>15.1</b> FLASH: run</a></li>
<li class="chapter" data-level="15.2" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-output"><i class="fa fa-check"></i><b>15.2</b> FLASH: Output</a></li>
<li class="chapter" data-level="15.3" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-mcqs"><i class="fa fa-check"></i><b>15.3</b> FLASH: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16_Megahit.html"><a href="16_Megahit.html"><i class="fa fa-check"></i><b>16</b> Megahit Assembly</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16_Megahit.html"><a href="16_Megahit.html#megahit"><i class="fa fa-check"></i><b>16.1</b> MEGAHIT</a></li>
<li class="chapter" data-level="16.2" data-path="16_Megahit.html"><a href="16_Megahit.html#quast"><i class="fa fa-check"></i><b>16.2</b> QUAST</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-run"><i class="fa fa-check"></i><b>16.2.1</b> QUAST: run</a></li>
<li class="chapter" data-level="16.2.2" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-visualise"><i class="fa fa-check"></i><b>16.2.2</b> QUAST: visualise</a></li>
<li class="chapter" data-level="16.2.3" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-mcqs"><i class="fa fa-check"></i><b>16.2.3</b> QUAST: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="16_Megahit.html"><a href="16_Megahit.html#metagenome-assembly-summary"><i class="fa fa-check"></i><b>16.3</b> Metagenome assembly summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html"><i class="fa fa-check"></i><b>17</b> Genome binning</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>17.1</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-depth-calculation"><i class="fa fa-check"></i><b>17.1.1</b> MetaBAT2: depth calculation</a></li>
<li class="chapter" data-level="17.1.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-run"><i class="fa fa-check"></i><b>17.1.2</b> MetaBAT2: run</a></li>
<li class="chapter" data-level="17.1.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-output"><i class="fa fa-check"></i><b>17.1.3</b> MetaBAT2: output</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm"><i class="fa fa-check"></i><b>17.2</b> CheckM</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-mamba"><i class="fa fa-check"></i><b>17.2.1</b> CheckM: Mamba</a></li>
<li class="chapter" data-level="17.2.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-run"><i class="fa fa-check"></i><b>17.2.2</b> CheckM: run</a></li>
<li class="chapter" data-level="17.2.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-output"><i class="fa fa-check"></i><b>17.2.3</b> CheckM: output</a></li>
<li class="chapter" data-level="17.2.4" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-quality-score"><i class="fa fa-check"></i><b>17.2.4</b> CheckM: Quality score</a></li>
<li class="chapter" data-level="17.2.5" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-mcqs"><i class="fa fa-check"></i><b>17.2.5</b> CheckM: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#binning-summary"><i class="fa fa-check"></i><b>17.3</b> Binning summary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html"><i class="fa fa-check"></i><b>18</b> Assembly functional annotation</a>
<ul>
<li class="chapter" data-level="18.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#taxonomic-annotation"><i class="fa fa-check"></i><b>18.1</b> Taxonomic annotation</a></li>
<li class="chapter" data-level="18.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#bakta"><i class="fa fa-check"></i><b>18.2</b> Bakta</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#bakta-run"><i class="fa fa-check"></i><b>18.2.1</b> Bakta: run</a></li>
<li class="chapter" data-level="18.2.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#bakta-premade-results"><i class="fa fa-check"></i><b>18.2.2</b> Bakta: premade results</a></li>
<li class="chapter" data-level="18.2.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#bakta-output"><i class="fa fa-check"></i><b>18.2.3</b> Bakta: output</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath"><i class="fa fa-check"></i><b>18.3</b> MinPath</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath-ec-extraction"><i class="fa fa-check"></i><b>18.3.1</b> MinPath: EC extraction</a></li>
<li class="chapter" data-level="18.3.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpathrun"><i class="fa fa-check"></i><b>18.3.2</b> MinPath: run</a></li>
<li class="chapter" data-level="18.3.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath-output"><i class="fa fa-check"></i><b>18.3.3</b> MinPath: output</a></li>
<li class="chapter" data-level="18.3.4" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#keggs"><i class="fa fa-check"></i><b>18.3.4</b> KEGGs</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="19-Appendix.html"><a href="19-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="19-Appendix.html"><a href="19-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="19-Appendix.html"><a href="19-Appendix.html#next-steps"><i class="fa fa-check"></i><b>B</b> Next steps</a></li>
<li class="chapter" data-level="C" data-path="19-Appendix.html"><a href="19-Appendix.html#manuals"><i class="fa fa-check"></i><b>C</b> Manuals</a></li>
<li class="chapter" data-level="D" data-path="19-Appendix.html"><a href="19-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>D</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="kraken2" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Kraken2<a href="09-Kraken2.html#kraken2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/kraken.png" style="width:150px" />
</center>
<p>Prior to running <code>Kraken2</code> we need to set a variable so <code>Kraken2</code> knows where to look for the databases it will use.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb23-1"><a href="09-Kraken2.html#cb23-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">KRAKEN2_DB_PATH</span><span class="op">=</span>/pub14/tea/nsc206/NEOF/Shotgun_metagenomics/kraken2_db</span></code></pre></div>
<p><strong>Note</strong>: You can look at the contents of the above directory to see it currently contains the MiniKraken database. This database contains only a subset of the bacteria, archaea, and viral <code>Kraken2</code> libraries. This is used in this practical due to restrictions on time and computational resources. For your own analyses we would recommend the full <code>Kraken2</code> database which uses all the bacteria, achaeal and viral complete genomes that are in Refseq at the time of building. See the following links for info on installing the databases.</p>
<ul>
<li>Standard <code>Kraken2</code> databases: <a href="https://github.com/DerrickWood/kraken2/wiki/Manual#standard-kraken-2-database" class="uri">https://github.com/DerrickWood/kraken2/wiki/Manual#standard-kraken-2-database</a></li>
<li>Custom <code>Kraken2</code> databases: <a href="https://github.com/DerrickWood/kraken2/wiki/Manual#custom-databases" class="uri">https://github.com/DerrickWood/kraken2/wiki/Manual#custom-databases</a></li>
</ul>
<div id="kraken2-run" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Kraken2: run<a href="09-Kraken2.html#kraken2-run" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/play_blue.png" style="width:100px" />
</center>
<p>Now, run <code>Kraken2</code> on sample K1 by running the following command.</p>
<p><strong>Note</strong>: We are not using the host removed data. This is to save time. In your own analysis ensure you are using host removed data.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb24-1"><a href="09-Kraken2.html#cb24-1" tabindex="-1"></a><span class="ex">kraken2</span> <span class="at">--paired</span> <span class="at">--db</span> minikraken2_v1_8GB <span class="dt">\</span></span>
<span id="cb24-2"><a href="09-Kraken2.html#cb24-2" tabindex="-1"></a>--output K1.kraken <span class="at">--report</span> K1.kreport2 <span class="dt">\</span></span>
<span id="cb24-3"><a href="09-Kraken2.html#cb24-3" tabindex="-1"></a>~/2-Trimmed/K1_R1.fq.gz ~/2-Trimmed/K1_R2.fq.gz</span></code></pre></div>
<div id="parameters-3" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="09-Kraken2.html#parameters-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<center>
<img src="figures/parameter_blue.png" style="width:100px" />
</center>
<ul>
<li><strong><code>--paired</code></strong>: Indicates that we are providing paired reads to <code>Kraken2</code>. Internally, <code>Kraken2</code> will concatenate the R1 and R2 reads into one sequence with an N between them.</li>
<li><strong><code>--db</code></strong>: Specify the <code>Kraken2</code> database to be used for taxonomic classification.
<ul>
<li>Previous to the command we set the <code>KRAKEN2_DB_PATH</code> so in this case the command will look for the directory called <code>minikraken2_v1_8GB</code> within <code>KRAKEN2_DB_PATH</code>.</li>
<li>Alternatively the full path of the required database could be provided (<code>/pub14/tea/nsc206/NEOF/Shotgun_metagenomics/kraken2_db/minikraken2_v1_8GB</code>).</li>
</ul></li>
<li><strong><code>--output</code></strong>: The output file. More info below.</li>
<li><strong><code>--report</code></strong>: The output report file. More info below.</li>
<li><strong><code>~/2-Trimmed/K1_R1.fq.gz ~/2-Trimmed/K1_R2.fq.gz</code></strong>: The trimmed read pairs for K1, which we will use as input.</li>
</ul>
</div>
</div>
<div id="kraken2-output" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Kraken2: output<a href="09-Kraken2.html#kraken2-output" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/output_file_blue.png" style="width:100px" />
</center>
<p>There are two major output formats from Kraken2.</p>
<div id="output-file" class="section level4 unnumbered hasAnchor">
<h4>Output file<a href="09-Kraken2.html#output-file" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <code>--output</code> parameter creates a <code>.kraken</code> file. Each sequence (or sequence pair, in the case of paired reads) classified by <code>Kraken2</code> results in a single line of output. <code>Kraken2</code>'s output lines contain five tab-delimited fields; from left to right, they are</p>
<ol style="list-style-type: decimal">
<li><strong>"C"/"U":</strong> a one letter code indicating that the sequence was either classified or unclassified.</li>
<li><strong>Sequence ID:</strong> Obtained from the FASTA/FASTQ header.</li>
<li><strong>Taxonomy ID:</strong> Assigned by <code>Kraken2</code>. This is 0 if the sequence is unclassified.</li>
<li><strong>Length of the sequence in bp:</strong> In the case of paired read data, this will be a string containing the lengths of the two sequences in bp, separated by a pipe character, e.g. "98|94".</li>
<li><strong>LCA mapping:</strong> A space-delimited list indicating the LCA (also known as <a href="https://en.wikipedia.org/wiki/Most_recent_common_ancestor">MRCA</a>) mapping of each k-mer in the sequence(s). For example, "562:13 561:4 A:31 0:1 562:3" would indicate that:
<ul>
<li>562:13 - The first 13 k-mers mapped to taxonomy ID #562</li>
<li>561:4 - The next 4 k-mers mapped to taxonomy ID #561</li>
<li>A:31 - The next 31 k-mers contained an ambiguous nucleotide</li>
<li>0:1 - The next k-mer was not in the database</li>
<li>562:3 - The last 3 k-mers mapped to taxonomy ID #562</li>
<li><strong>Note</strong>: that paired read data will contain a "|:|" token in this list to indicate the end of one read and the beginning of another.</li>
</ul></li>
</ol>
</div>
<div id="report-file" class="section level4 unnumbered hasAnchor">
<h4>Report file<a href="09-Kraken2.html#report-file" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <code>--report</code> parameter creates a <code>.kreport2</code> file. This is the report output format. This is required for bracken. It is tab-delimited with one line per taxon. The fields of the output, from left-to-right, are as follows:</p>
<ol style="list-style-type: decimal">
<li>Percentage of paired reads covered by the clade rooted at this taxon.</li>
<li>Number of paired reads covered by the clade rooted at this taxon.</li>
<li>Number of paired reads assigned directly to this taxon.</li>
<li>A rank code, indicating (U)nclassified, (R)oot, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. Taxa that are not at any of these 10 ranks have a rank code that is formed by using the rank code of the closest ancestor rank with a number indicating the distance from that rank. E.g., "G2" is a rank code indicating a taxon is between genus and species and the grandparent taxon is at the genus rank.</li>
<li>NCBI taxonomic ID number</li>
<li>Indented scientific name</li>
</ol>
</div>
<div id="screen-output" class="section level4 unnumbered hasAnchor">
<h4>Screen output<a href="09-Kraken2.html#screen-output" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The output to screen will show how many sequences are classified. This will be lower than normal as we are using a mini <code>Kraken2</code> database.</p>
</div>
<div id="confidence-threshold" class="section level4 unnumbered hasAnchor">
<h4>Confidence threshold<a href="09-Kraken2.html#confidence-threshold" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In a real analysis you may use the option <code>--confidence</code> which represents the <strong>"Confidence score threshold"</strong>. The default is <code>0.0</code>, which is the lowest, with the maximum value being <code>1</code>. A good place to start may be <code>0.1</code>. Too many classifications are removed if you attempt it with this dataset, due to the mini <code>Kraken2</code> database used. More info on the confidence scoring can be found at: <a href="https://github.com/DerrickWood/kraken2/wiki/Manual#confidence-scoring" class="uri">https://github.com/DerrickWood/kraken2/wiki/Manual#confidence-scoring</a></p>
</div>
<div id="task-1" class="section level4 unnumbered hasAnchor">
<h4>Task<a href="09-Kraken2.html#task-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Once the Kraken2 command has finished running, run it on the other two samples. Attempt the commands without looking at the help box.</p>
<p><strong>Hint</strong>: You will need to change all instances of K1 to K2 or W1 in the above command</p>
<div class="webex-solution">
<button>
K2 &amp; W1 Kraken 2 commands
</button>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb25-1"><a href="09-Kraken2.html#cb25-1" tabindex="-1"></a><span class="co">#K2</span></span>
<span id="cb25-2"><a href="09-Kraken2.html#cb25-2" tabindex="-1"></a><span class="ex">kraken2</span> <span class="at">--paired</span> <span class="at">--db</span> minikraken2_v1_8GB <span class="dt">\</span></span>
<span id="cb25-3"><a href="09-Kraken2.html#cb25-3" tabindex="-1"></a>--output K2.kraken <span class="at">--report</span> K2.kreport2 <span class="dt">\</span></span>
<span id="cb25-4"><a href="09-Kraken2.html#cb25-4" tabindex="-1"></a>~/2-Trimmed/K2_R1.fq.gz ~/2-Trimmed/K2_R2.fq.gz</span>
<span id="cb25-5"><a href="09-Kraken2.html#cb25-5" tabindex="-1"></a><span class="co">#W1</span></span>
<span id="cb25-6"><a href="09-Kraken2.html#cb25-6" tabindex="-1"></a><span class="ex">kraken2</span> <span class="at">--paired</span> <span class="at">--db</span> minikraken2_v1_8GB <span class="dt">\</span></span>
<span id="cb25-7"><a href="09-Kraken2.html#cb25-7" tabindex="-1"></a>--output W1.kraken <span class="at">--report</span> W1.kreport2 <span class="dt">\</span></span>
<span id="cb25-8"><a href="09-Kraken2.html#cb25-8" tabindex="-1"></a>~/2-Trimmed/W1_R1.fq.gz ~/2-Trimmed/W1_R2.fq.gz</span></code></pre></div>
</div>
</div>
</div>
<div id="kraken-2-mcqs" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Kraken 2: MCQs<a href="09-Kraken2.html#kraken-2-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/question_bubble_blue.png" style="width:100px" />
</center>
<p>Viewing the Kraken2 output files with your favourite text viewer (<code>less</code>, <code>nano</code>, <code>vim</code>, etc.), attempt the below MCQs.</p>
<ol style="list-style-type: decimal">
<li>How many paired reads were unclassified for K1?
<div id="radio_BNPMCFDFQF" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BNPMCFDFQF" value=""></input> <span><strong>2</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BNPMCFDFQF" value=""></input> <span><strong>486,723</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BNPMCFDFQF" value="answer"></input> <span><strong>555,162</strong></span></label>
</div></li>
<li>How many paired reads were classified for K2 (i.e. number of reads classified at root level and below)?
<div id="radio_BLVTSNTQAP" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BLVTSNTQAP" value=""></input> <span><strong>2</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BLVTSNTQAP" value="answer"></input> <span><strong>486,723</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BLVTSNTQAP" value=""></input> <span><strong>555,162</strong></span></label>
</div></li>
<li>How many paired reads were assigned directly to root level for W1?
<div id="radio_XLQAYDNWDG" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_XLQAYDNWDG" value="answer"></input> <span><strong>2</strong></span></label><label><input type="radio" autocomplete="off" name="radio_XLQAYDNWDG" value=""></input> <span><strong>486,723</strong></span></label><label><input type="radio" autocomplete="off" name="radio_XLQAYDNWDG" value=""></input> <span><strong>555,162</strong></span></label>
</div></li>
<li>What percentage of W1's paired reads were assigned to the clade of Bacteroidetes (Phylum)?
<div id="radio_RJUWOCPSEL" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_RJUWOCPSEL" value=""></input> <span><strong>0.12</strong></span></label><label><input type="radio" autocomplete="off" name="radio_RJUWOCPSEL" value=""></input> <span><strong>0.59</strong></span></label><label><input type="radio" autocomplete="off" name="radio_RJUWOCPSEL" value="answer"></input> <span><strong>14.89</strong></span></label>
</div></li>
<li>What percentage of K2's paired reads were assigned to the clade of Rikenellaceae (Family)?
<div id="radio_CUTTONCSKH" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_CUTTONCSKH" value=""></input> <span><strong>0.12</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CUTTONCSKH" value="answer"></input> <span><strong>0.59</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CUTTONCSKH" value=""></input> <span><strong>14.89</strong></span></label>
</div></li>
<li>What percentage of K1's paired reads were assigned to the clade of Bacteroides helcogenes (Species)?
<div id="radio_GDTOIYLVBG" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_GDTOIYLVBG" value="answer"></input> <span><strong>0.12</strong></span></label><label><input type="radio" autocomplete="off" name="radio_GDTOIYLVBG" value=""></input> <span><strong>0.59</strong></span></label><label><input type="radio" autocomplete="off" name="radio_GDTOIYLVBG" value=""></input> <span><strong>14.89</strong></span></label>
</div></li>
</ol>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="08-Taxonomic_profiling.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="10-Krona.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
