<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 LEfSe biomarker detection | Shotgun Metagenomics</title>
  <meta name="description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 LEfSe biomarker detection | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 LEfSe biomarker detection | Shotgun Metagenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2025-04-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="11-Bracken.html"/>
<link rel="next" href="13-Functional_profiling.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Start_conda.html"><a href="04-Start_conda.html"><i class="fa fa-check"></i><b>4</b> Startup &amp; mamba</a></li>
<li class="part"><span><b>Data &amp; QC</b></span></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-summary"><i class="fa fa-check"></i><b>7.5</b> Host removal summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy of reads</b></span></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a></li>
<li class="chapter" data-level="9" data-path="09-Kraken2.html"><a href="09-Kraken2.html"><i class="fa fa-check"></i><b>9</b> Kraken2</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-run"><i class="fa fa-check"></i><b>9.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="9.2" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-output"><i class="fa fa-check"></i><b>9.2</b> Kraken2: output</a></li>
<li class="chapter" data-level="9.3" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken-2-mcqs"><i class="fa fa-check"></i><b>9.3</b> Kraken 2: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Krona.html"><a href="10-Krona.html"><i class="fa fa-check"></i><b>10</b> Krona plot</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Krona.html"><a href="10-Krona.html#krona-run"><i class="fa fa-check"></i><b>10.1</b> Krona: run</a></li>
<li class="chapter" data-level="10.2" data-path="10-Krona.html"><a href="10-Krona.html#krona-visualise"><i class="fa fa-check"></i><b>10.2</b> Krona: visualise</a></li>
<li class="chapter" data-level="10.3" data-path="10-Krona.html"><a href="10-Krona.html#krona-mcqs"><i class="fa fa-check"></i><b>10.3</b> Krona: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Bracken.html"><a href="11-Bracken.html"><i class="fa fa-check"></i><b>11</b> Bracken</a>
<ul>
<li class="chapter" data-level="11.0.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-run"><i class="fa fa-check"></i><b>11.0.1</b> Bracken: run</a></li>
<li class="chapter" data-level="11.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-output"><i class="fa fa-check"></i><b>11.1</b> Bracken: output</a></li>
<li class="chapter" data-level="11.2" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-mcqs"><i class="fa fa-check"></i><b>11.2</b> Bracken: MCQs</a></li>
<li class="chapter" data-level="11.3" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-merging-output"><i class="fa fa-check"></i><b>11.3</b> Bracken: merging output</a></li>
<li class="chapter" data-level="11.4" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-extracting-output"><i class="fa fa-check"></i><b>11.4</b> Bracken: extracting output</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Lefse.html"><a href="12-Lefse.html"><i class="fa fa-check"></i><b>12</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2_lefse_metadata"><i class="fa fa-check"></i><b>12.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="12.2" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-format"><i class="fa fa-check"></i><b>12.2</b> LEfSe: format</a></li>
<li class="chapter" data-level="12.3" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-run"><i class="fa fa-check"></i><b>12.3</b> LEfSe: run</a></li>
<li class="chapter" data-level="12.4" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-visualisation"><i class="fa fa-check"></i><b>12.4</b> LEfSe: visualisation</a></li>
<li class="chapter" data-level="12.5" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-mcqs"><i class="fa fa-check"></i><b>12.5</b> LEfSe: MCQs</a></li>
<li class="chapter" data-level="12.6" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>12.6</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="part"><span><b>Functional profiling of reads</b></span></li>
<li class="chapter" data-level="13" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html"><i class="fa fa-check"></i><b>13</b> HUMAnN Functional profiling</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann"><i class="fa fa-check"></i><b>13.1</b> HUMAnN</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-mamba-directories-and-files"><i class="fa fa-check"></i><b>13.1.1</b> HUMAnN: mamba, directories, and files</a></li>
<li class="chapter" data-level="13.1.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-run"><i class="fa fa-check"></i><b>13.1.2</b> HUMAnN: run</a></li>
<li class="chapter" data-level="13.1.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-output"><i class="fa fa-check"></i><b>13.1.3</b> HUMAnN: output</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humannmultisampleprocessing"><i class="fa fa-check"></i><b>13.2</b> Multi sample processing</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-combining-data"><i class="fa fa-check"></i><b>13.2.1</b> HUMAnN: combining data</a></li>
<li class="chapter" data-level="13.2.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-split-stratified-table"><i class="fa fa-check"></i><b>13.2.2</b> HUMAnN: split stratified table</a></li>
<li class="chapter" data-level="13.2.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humannrenormalise"><i class="fa fa-check"></i><b>13.2.3</b> HUMAnN: renormalising data</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#heatmap"><i class="fa fa-check"></i><b>13.3</b> Heatmap</a></li>
<li class="chapter" data-level="13.4" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann_lefse"><i class="fa fa-check"></i><b>13.4</b> LEfSe</a></li>
<li class="chapter" data-level="13.5" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#optional-task"><i class="fa fa-check"></i><b>13.5</b> Optional task</a></li>
</ul></li>
<li class="part"><span><b>Assembly Approach</b></span></li>
<li class="chapter" data-level="14" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html"><i class="fa fa-check"></i><b>14</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html#metagenome-assembly-mamba"><i class="fa fa-check"></i><b>14.1</b> Metagenome assembly: Mamba</a></li>
<li class="chapter" data-level="14.2" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>14.2</b> A primer on short read assembly</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html"><i class="fa fa-check"></i><b>15</b> Stitching read pairs</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-run"><i class="fa fa-check"></i><b>15.1</b> FLASH: run</a></li>
<li class="chapter" data-level="15.2" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-output"><i class="fa fa-check"></i><b>15.2</b> FLASH: Output</a></li>
<li class="chapter" data-level="15.3" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-mcqs"><i class="fa fa-check"></i><b>15.3</b> FLASH: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16_Megahit.html"><a href="16_Megahit.html"><i class="fa fa-check"></i><b>16</b> Megahit Assembly</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16_Megahit.html"><a href="16_Megahit.html#megahit"><i class="fa fa-check"></i><b>16.1</b> MEGAHIT</a></li>
<li class="chapter" data-level="16.2" data-path="16_Megahit.html"><a href="16_Megahit.html#quast"><i class="fa fa-check"></i><b>16.2</b> QUAST</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-run"><i class="fa fa-check"></i><b>16.2.1</b> QUAST: run</a></li>
<li class="chapter" data-level="16.2.2" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-visualise"><i class="fa fa-check"></i><b>16.2.2</b> QUAST: visualise</a></li>
<li class="chapter" data-level="16.2.3" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-mcqs"><i class="fa fa-check"></i><b>16.2.3</b> QUAST: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="16_Megahit.html"><a href="16_Megahit.html#metagenome-assembly-summary"><i class="fa fa-check"></i><b>16.3</b> Metagenome assembly summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html"><i class="fa fa-check"></i><b>17</b> Genome binning</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>17.1</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-depth-calculation"><i class="fa fa-check"></i><b>17.1.1</b> MetaBAT2: depth calculation</a></li>
<li class="chapter" data-level="17.1.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-run"><i class="fa fa-check"></i><b>17.1.2</b> MetaBAT2: run</a></li>
<li class="chapter" data-level="17.1.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-output"><i class="fa fa-check"></i><b>17.1.3</b> MetaBAT2: output</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm"><i class="fa fa-check"></i><b>17.2</b> CheckM</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-mamba"><i class="fa fa-check"></i><b>17.2.1</b> CheckM: Mamba</a></li>
<li class="chapter" data-level="17.2.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-run"><i class="fa fa-check"></i><b>17.2.2</b> CheckM: run</a></li>
<li class="chapter" data-level="17.2.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-output"><i class="fa fa-check"></i><b>17.2.3</b> CheckM: output</a></li>
<li class="chapter" data-level="17.2.4" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-quality-score"><i class="fa fa-check"></i><b>17.2.4</b> CheckM: Quality score</a></li>
<li class="chapter" data-level="17.2.5" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-mcqs"><i class="fa fa-check"></i><b>17.2.5</b> CheckM: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#binning-summary"><i class="fa fa-check"></i><b>17.3</b> Binning summary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html"><i class="fa fa-check"></i><b>18</b> Assembly functional annotation</a>
<ul>
<li class="chapter" data-level="18.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#taxonomic-annotation"><i class="fa fa-check"></i><b>18.1</b> Taxonomic annotation</a></li>
<li class="chapter" data-level="18.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#bakta"><i class="fa fa-check"></i><b>18.2</b> Bakta</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#bakta-run"><i class="fa fa-check"></i><b>18.2.1</b> Bakta: run</a></li>
<li class="chapter" data-level="18.2.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#bakta-premade-results"><i class="fa fa-check"></i><b>18.2.2</b> Bakta: premade results</a></li>
<li class="chapter" data-level="18.2.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#bakta-output"><i class="fa fa-check"></i><b>18.2.3</b> Bakta: output</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath"><i class="fa fa-check"></i><b>18.3</b> MinPath</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath-ec-extraction"><i class="fa fa-check"></i><b>18.3.1</b> MinPath: EC extraction</a></li>
<li class="chapter" data-level="18.3.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpathrun"><i class="fa fa-check"></i><b>18.3.2</b> MinPath: run</a></li>
<li class="chapter" data-level="18.3.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath-output"><i class="fa fa-check"></i><b>18.3.3</b> MinPath: output</a></li>
<li class="chapter" data-level="18.3.4" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#keggs"><i class="fa fa-check"></i><b>18.3.4</b> KEGGs</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="19-Appendix.html"><a href="19-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="19-Appendix.html"><a href="19-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="19-Appendix.html"><a href="19-Appendix.html#next-steps"><i class="fa fa-check"></i><b>B</b> Next steps</a></li>
<li class="chapter" data-level="C" data-path="19-Appendix.html"><a href="19-Appendix.html#manuals"><i class="fa fa-check"></i><b>C</b> Manuals</a></li>
<li class="chapter" data-level="D" data-path="19-Appendix.html"><a href="19-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>D</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="lefse-biomarker-detection" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">Chapter 12</span> LEfSe biomarker detection<a href="12-Lefse.html#lefse-biomarker-detection" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/lefse.png" style="width:150px" />
</center>
<p>We will use <code>LEfSe</code> (Linear discriminant analysis Effect Size) to determine which taxa can most likely explain the differences between the Western and Korean diet. <code>LEfSe</code> couples standard tests for statistical significance with additional tests encoding biological consistency and effect relevance. It can be used with other features such as organisms, clades, operational taxonomic units, genes, or functions.</p>
<p>In essence it allows for the detection of biomarkers when comparing sample groups. In the <code>LEfSe</code> terminology the sample groups are called the class.</p>
<div id="kraken2_lefse_metadata" class="section level2 hasAnchor" number="12.1">
<h2><span class="header-section-number">12.1</span> LEfSe: add metadata<a href="12-Lefse.html#kraken2_lefse_metadata" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/header.png" style="width:100px; background:white;border-radius:5px" />
</center>
<p>We need to add metadata to our <code>Bracken</code> file to be ready for <code>LEfSe.</code> First we will copy the file so we have a backup in case we do anything wrong.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb34-1"><a href="12-Lefse.html#cb34-1" tabindex="-1"></a><span class="fu">cp</span> all_num.bracken all_num.lefse.bracken</span></code></pre></div>
<p>Using your favourite text editor (e.g. <code>nano</code>, <code>vim</code>, etc.) add the following line to the top of your <code>all_num.lefse.bracken</code> file. The words are separated by tabs. If you are not sure how to carry out this task please ask a demonstrator.</p>
<p><strong>diet K K K K K K K K K K K K W W W W W W W W W W W W</strong></p>
<p><strong>Note</strong>: The above is <strong>diet</strong> followed by 12 <strong>K</strong> and 12 <strong>W</strong>.</p>
<p>The singular line should match the order of your samples within the file. This is the metadata line that <code>LEfSe</code> will use to determine which samples belong to each sample group, and therefore which to compare. In this case it is Korean diet samples versus Western diet samples.</p>
<div class="webex-solution">
<button>
Issues with creating file?
</button>
<p>If you are having issues with creating and editing the file <code>all_num.lefse.bracken</code> you can copy a pre-made version.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb35-1"><a href="12-Lefse.html#cb35-1" tabindex="-1"></a><span class="fu">cp</span> /pub14/tea/nsc206/NEOF/Shotgun_metagenomics/lefse/all_num.lefse.bracken .</span></code></pre></div>
</div>
</div>
<div id="lefse-format" class="section level2 hasAnchor" number="12.2">
<h2><span class="header-section-number">12.2</span> LEfSe: format<a href="12-Lefse.html#lefse-format" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/lefse_format.png" style="width:100px;background:white;border-radius:5px" />
</center>
<p>We need to further format and preprocess our file with a <code>LEfSe</code> script.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb36-1"><a href="12-Lefse.html#cb36-1" tabindex="-1"></a><span class="ex">lefse_format_input.py</span> all_num.lefse.bracken all_num.lefse <span class="at">-c</span> 1 <span class="at">-u</span> 2 <span class="at">-o</span> 1000000</span></code></pre></div>
<div id="parameters-5" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="12-Lefse.html#parameters-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<center>
<img src="figures/parameter_purple.png" style="width:100px" />
</center>
<ul>
<li><code>all_num.lefse.bracken</code> : Input <code>Bracken</code> file.</li>
<li><code>all_num.lefse</code> : Output file formatted for the run_lefse command, which we will soon run.</li>
<li><code>-c 1</code> : Specifies the row with the class info. This is used to determine which samples will be compared against which samples. In this case it is the first row with the Ks and Ws.</li>
<li><code>-u 2</code> : Specifies the row with the sample names. This is the second row in this case.</li>
<li><code>-o 1000000</code> : An integer can be indicated to determine to what size (count sum value) each sample should be normalised to. <code>LEfSe</code> developers recommend 1000000 (1 million) when very low values are present. We generally always use 1 million for consistency.</li>
</ul>
</div>
</div>
<div id="lefse-run" class="section level2 hasAnchor" number="12.3">
<h2><span class="header-section-number">12.3</span> LEfSe: run<a href="12-Lefse.html#lefse-run" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/play_purple.png" style="width:100px" />
</center>
<p>Now to run <code>LEfSe.</code> All we need to do is run the command with the formatted input and provide an output file name.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb37-1"><a href="12-Lefse.html#cb37-1" tabindex="-1"></a><span class="ex">lefse_run.py</span> all_num.lefse all_num.lefse.out</span></code></pre></div>
<div id="output" class="section level4 unnumbered hasAnchor">
<h4>Output<a href="12-Lefse.html#output" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The output file is a tab-delimited file which contains a row for each species. Biomarkers will have the five columns below whilst non-biomarkers will have the first two followed by a "-" .</p>
<ul>
<li><strong>Biomarker name</strong></li>
<li><strong>Log of highest class average</strong>: I.e. get the class with the greater amounts of the biomarker, average the counts and then get the log of this value.</li>
<li><strong>Class with the greater amounts of biomarker</strong></li>
<li><strong>LDA effect size</strong>: A statistical figure for <code>LEfSe</code>.</li>
<li><strong>p-value</strong>: Biomarkers must have a p-value of &lt;0.05 to be considered significant.</li>
</ul>
<p>The <strong>LDA effect</strong> size indicates how much of an effect each biomarker has. The default is to only count a species with an LDA effect size of greater than 2 or less than -2 as a biomarker. The further the LDA effect size is from 0 the greater the effect the species causes. Generally, it can be thought of as the order of magnitude difference in the abundance of the biomarker between the sample groups.</p>
</div>
</div>
<div id="lefse-visualisation" class="section level2 hasAnchor" number="12.4">
<h2><span class="header-section-number">12.4</span> LEfSe: visualisation<a href="12-Lefse.html#lefse-visualisation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/bar_chart_horizontal.png" style="width:100px; background:white;border-radius:5px" />
</center>
<p>Next we can visualise the output.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb38-1"><a href="12-Lefse.html#cb38-1" tabindex="-1"></a><span class="ex">lefse_plot_res.py</span> <span class="at">--dpi</span> 200 <span class="at">--format</span> png all_num.lefse.out biomarkers.png</span></code></pre></div>
<ul>
<li><code>--dpi 200</code> : Dots per inch. This refers to the resolution of the output image. Normally publications want 300 dpi. We’ve chosen 200 as it is good quality and we will not be publishing these results.</li>
<li><code>--format png</code> : Format of output file. png is a commonly used file format for images.</li>
<li><code>all_num.lefse.out</code> : <code>LEfSe</code> output to visualise.</li>
<li><code>biomarkers.png</code> : Plot showing the LDA scores of the species detected as biomarkers. Colouring shows which class (K or W) the species is found in higher abundance.</li>
</ul>
<p>Look at the figure with <code>firefox</code>:</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb39-1"><a href="12-Lefse.html#cb39-1" tabindex="-1"></a><span class="ex">firefox</span> biomarkers.png</span></code></pre></div>
</div>
<div id="lefse-mcqs" class="section level2 hasAnchor" number="12.5">
<h2><span class="header-section-number">12.5</span> LEfSe: MCQs<a href="12-Lefse.html#lefse-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/question_bubble_purple.png" style="width:100px" />
</center>
<p>Interrogating the biomarkers.png plot and the all_num.lefse.out file, attempt the below MCQs.</p>
<p><strong>Note</strong>: In this instance green bars represent biomarkers in higher abundance in the W samples whilst the red bars represent biomarkers in higher abundance in the K samples.</p>
<ol style="list-style-type: decimal">
<li>Which species biomarker causes the biggest effect in the <strong>W</strong> class?
<div id="radio_NNGLPXJCKB" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_NNGLPXJCKB" value=""></input> <span><strong>Ruminococcuschampanellensis</strong></span></label><label><input type="radio" autocomplete="off" name="radio_NNGLPXJCKB" value="answer"></input> <span><strong>Alistipesshahii</strong></span></label><label><input type="radio" autocomplete="off" name="radio_NNGLPXJCKB" value=""></input> <span><strong>Methanosphaerastadtmanae</strong></span></label>
</div></li>
<li>Which species biomarker causes the biggest effect in the <strong>K</strong> class?
<div id="radio_KEMVWHPKXQ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_KEMVWHPKXQ" value=""></input> <span><strong>Ruminococcuschampanellensis</strong></span></label><label><input type="radio" autocomplete="off" name="radio_KEMVWHPKXQ" value=""></input> <span><strong>Alistipesshahii</strong></span></label><label><input type="radio" autocomplete="off" name="radio_KEMVWHPKXQ" value="answer"></input> <span><strong>Methanosphaerastadtmanae</strong></span></label>
</div></li>
<li>Which species biomarker (i.e. present in the plot) causes the lowest effect in the <strong>W</strong> class?
<div id="radio_VLQHDGBSRF" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_VLQHDGBSRF" value="answer"></input> <span><strong>Ruminococcuschampanellensis</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VLQHDGBSRF" value=""></input> <span><strong>Alistipesshahii</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VLQHDGBSRF" value=""></input> <span><strong>Methanosphaerastadtmanae</strong></span></label>
</div></li>
<li>Which class has more biomarkers associated with it?
<div id="radio_AFYYFRKBLC" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_AFYYFRKBLC" value=""></input> <span><strong>Korean</strong></span></label><label><input type="radio" autocomplete="off" name="radio_AFYYFRKBLC" value="answer"></input> <span><strong>Western</strong></span></label>
</div></li>
<li>What is the LEfSe p-value for <strong>Campylobactercoli</strong>?
<div id="radio_WVMXDFZZXO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_WVMXDFZZXO" value="answer"></input> <span><strong>0.0022136919545913936</strong></span></label><label><input type="radio" autocomplete="off" name="radio_WVMXDFZZXO" value=""></input> <span><strong>3.0271864317141715</strong></span></label><label><input type="radio" autocomplete="off" name="radio_WVMXDFZZXO" value=""></input> <span><strong>3.3224493820428482</strong></span></label>
</div></li>
<li>What is the Log of highest class average for <strong>Streptococcussuis</strong>?
<div id="radio_IPFRLNRZMD" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_IPFRLNRZMD" value=""></input> <span><strong>0.0022136919545913936</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IPFRLNRZMD" value="answer"></input> <span><strong>3.0271864317141715</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IPFRLNRZMD" value=""></input> <span><strong>3.3224493820428482</strong></span></label>
</div></li>
<li>What is the LDA effect size for <strong>Bifidobacteriumlongum</strong>?
<div id="radio_HKVNWIMFDA" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_HKVNWIMFDA" value=""></input> <span><strong>0.0022136919545913936</strong></span></label><label><input type="radio" autocomplete="off" name="radio_HKVNWIMFDA" value=""></input> <span><strong>3.0271864317141715</strong></span></label><label><input type="radio" autocomplete="off" name="radio_HKVNWIMFDA" value="answer"></input> <span><strong>3.3224493820428482</strong></span></label>
</div></li>
</ol>
</div>
<div id="kraken2-and-bracken-databases" class="section level2 hasAnchor" number="12.6">
<h2><span class="header-section-number">12.6</span> Kraken2 and Bracken databases<a href="12-Lefse.html#kraken2-and-bracken-databases" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/database.png" style="width:150px" />
</center>
<p>In your own future analysis you will need to create your own <code>Kraken2</code> and <code>Bracken</code> databases. Please see the following links on information for this:</p>
<ul>
<li><code>Kraken2</code>
<ul>
<li>Standard <code>Kraken2</code> databases: <a href="https://github.com/DerrickWood/kraken2/wiki/Manual#standard-kraken-2-database" class="uri">https://github.com/DerrickWood/kraken2/wiki/Manual#standard-kraken-2-database</a></li>
<li>Custom <code>Kraken2</code> databases: <a href="https://github.com/DerrickWood/kraken2/wiki/Manual#custom-databases" class="uri">https://github.com/DerrickWood/kraken2/wiki/Manual#custom-databases</a></li>
</ul></li>
<li><code>Bracken</code>
<ul>
<li><a href="https://ccb.jhu.edu/software/bracken/index.shtml?t=manual#step1" class="uri">https://ccb.jhu.edu/software/bracken/index.shtml?t=manual#step1</a></li>
<li>This requires a <code>Kraken2</code> database to be built first.</li>
</ul></li>
</ul>

</div>
</div>



<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="11-Bracken.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="13-Functional_profiling.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
