<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 15 Multi sample processing | Shotgun Metagenomics</title>
  <meta name="description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 15 Multi sample processing | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 15 Multi sample processing | Shotgun Metagenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2025-10-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="14-HUMAnN.html"/>
<link rel="next" href="16-Heatmap.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/tippyjs-3.2.0/themes/google.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/light-border.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/light.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/translucent.css" rel="stylesheet" />
<script src="libs/tippyjs-3.2.0/tippy.all.min.js"></script>
<script src="libs/tippy-binding-0.1.0/tippy.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Start_conda.html"><a href="04-Start_conda.html"><i class="fa fa-check"></i><b>4</b> Startup &amp; mamba</a></li>
<li class="part"><span><b>Data &amp; QC</b></span></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-summary"><i class="fa fa-check"></i><b>7.5</b> Host removal summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy of reads</b></span></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a></li>
<li class="chapter" data-level="9" data-path="09-Kraken2.html"><a href="09-Kraken2.html"><i class="fa fa-check"></i><b>9</b> Kraken2</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-run"><i class="fa fa-check"></i><b>9.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="9.2" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-output"><i class="fa fa-check"></i><b>9.2</b> Kraken2: output</a></li>
<li class="chapter" data-level="9.3" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken-2-mcqs"><i class="fa fa-check"></i><b>9.3</b> Kraken 2: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Krona.html"><a href="10-Krona.html"><i class="fa fa-check"></i><b>10</b> Krona plot</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Krona.html"><a href="10-Krona.html#krona-run"><i class="fa fa-check"></i><b>10.1</b> Krona: run</a></li>
<li class="chapter" data-level="10.2" data-path="10-Krona.html"><a href="10-Krona.html#krona-visualise"><i class="fa fa-check"></i><b>10.2</b> Krona: visualise</a></li>
<li class="chapter" data-level="10.3" data-path="10-Krona.html"><a href="10-Krona.html#krona-mcqs"><i class="fa fa-check"></i><b>10.3</b> Krona: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Bracken.html"><a href="11-Bracken.html"><i class="fa fa-check"></i><b>11</b> Bracken</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-run"><i class="fa fa-check"></i><b>11.1</b> Bracken: run</a></li>
<li class="chapter" data-level="11.2" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-output"><i class="fa fa-check"></i><b>11.2</b> Bracken: output</a></li>
<li class="chapter" data-level="11.3" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-mcqs"><i class="fa fa-check"></i><b>11.3</b> Bracken: MCQs</a></li>
<li class="chapter" data-level="11.4" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-merging-output"><i class="fa fa-check"></i><b>11.4</b> Bracken: merging output</a></li>
<li class="chapter" data-level="11.5" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-extracting-output"><i class="fa fa-check"></i><b>11.5</b> Bracken: extracting output</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Lefse.html"><a href="12-Lefse.html"><i class="fa fa-check"></i><b>12</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2_lefse_metadata"><i class="fa fa-check"></i><b>12.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="12.2" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-format"><i class="fa fa-check"></i><b>12.2</b> LEfSe: format</a></li>
<li class="chapter" data-level="12.3" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-run"><i class="fa fa-check"></i><b>12.3</b> LEfSe: run</a></li>
<li class="chapter" data-level="12.4" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-visualisation"><i class="fa fa-check"></i><b>12.4</b> LEfSe: visualisation</a></li>
<li class="chapter" data-level="12.5" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-mcqs"><i class="fa fa-check"></i><b>12.5</b> LEfSe: MCQs</a></li>
<li class="chapter" data-level="12.6" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>12.6</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="part"><span><b>Functional profiling of reads</b></span></li>
<li class="chapter" data-level="13" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html"><i class="fa fa-check"></i><b>13</b> Functional profiling</a></li>
<li class="chapter" data-level="14" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html"><i class="fa fa-check"></i><b>14</b> HUMAnN</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#humann-mamba-directories-and-files"><i class="fa fa-check"></i><b>14.1</b> HUMAnN: mamba, directories, and files</a></li>
<li class="chapter" data-level="14.2" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#humann-run"><i class="fa fa-check"></i><b>14.2</b> HUMAnN: run</a></li>
<li class="chapter" data-level="14.3" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#humann-output"><i class="fa fa-check"></i><b>14.3</b> HUMAnN: output</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html"><i class="fa fa-check"></i><b>15</b> Multi sample processing</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#combining-data"><i class="fa fa-check"></i><b>15.1</b> Combining data</a></li>
<li class="chapter" data-level="15.2" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#split-stratified-table"><i class="fa fa-check"></i><b>15.2</b> Split stratified table</a></li>
<li class="chapter" data-level="15.3" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#humannrenormalise"><i class="fa fa-check"></i><b>15.3</b> Renormalising data</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Heatmap.html"><a href="16-Heatmap.html"><i class="fa fa-check"></i><b>16</b> Heatmap</a></li>
<li class="chapter" data-level="17" data-path="17-LEfSe_humann.html"><a href="17-LEfSe_humann.html"><i class="fa fa-check"></i><b>17</b> LEfSe</a></li>
<li class="chapter" data-level="18" data-path="18-HUMAnN_task.html"><a href="18-HUMAnN_task.html"><i class="fa fa-check"></i><b>18</b> Optional task</a></li>
<li class="part"><span><b>Assembly Approach</b></span></li>
<li class="chapter" data-level="19" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html"><i class="fa fa-check"></i><b>19</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="19.1" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html#metagenome-assembly-mamba"><i class="fa fa-check"></i><b>19.1</b> Metagenome assembly: Mamba</a></li>
<li class="chapter" data-level="19.2" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>19.2</b> A primer on short read assembly</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html"><i class="fa fa-check"></i><b>20</b> Stitching read pairs</a>
<ul>
<li class="chapter" data-level="20.1" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html#flash-run"><i class="fa fa-check"></i><b>20.1</b> FLASH: run</a></li>
<li class="chapter" data-level="20.2" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html#flash-output"><i class="fa fa-check"></i><b>20.2</b> FLASH: Output</a></li>
<li class="chapter" data-level="20.3" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html#flash-mcqs"><i class="fa fa-check"></i><b>20.3</b> FLASH: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="21_Megahit.html"><a href="21_Megahit.html"><i class="fa fa-check"></i><b>21</b> Megahit Assembly</a>
<ul>
<li class="chapter" data-level="21.1" data-path="21_Megahit.html"><a href="21_Megahit.html#megahit"><i class="fa fa-check"></i><b>21.1</b> MEGAHIT</a></li>
<li class="chapter" data-level="21.2" data-path="21_Megahit.html"><a href="21_Megahit.html#quast"><i class="fa fa-check"></i><b>21.2</b> QUAST</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="21_Megahit.html"><a href="21_Megahit.html#quast-run"><i class="fa fa-check"></i><b>21.2.1</b> QUAST: run</a></li>
<li class="chapter" data-level="21.2.2" data-path="21_Megahit.html"><a href="21_Megahit.html#quast-visualise"><i class="fa fa-check"></i><b>21.2.2</b> QUAST: visualise</a></li>
<li class="chapter" data-level="21.2.3" data-path="21_Megahit.html"><a href="21_Megahit.html#quast-mcqs"><i class="fa fa-check"></i><b>21.2.3</b> QUAST: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="21_Megahit.html"><a href="21_Megahit.html#metagenome-assembly-summary"><i class="fa fa-check"></i><b>21.3</b> Metagenome assembly summary</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html"><i class="fa fa-check"></i><b>22</b> Genome binning</a>
<ul>
<li class="chapter" data-level="22.1" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>22.1</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="22.1.1" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2-depth-calculation"><i class="fa fa-check"></i><b>22.1.1</b> MetaBAT2: depth calculation</a></li>
<li class="chapter" data-level="22.1.2" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2-run"><i class="fa fa-check"></i><b>22.1.2</b> MetaBAT2: run</a></li>
<li class="chapter" data-level="22.1.3" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2-output"><i class="fa fa-check"></i><b>22.1.3</b> MetaBAT2: output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html"><i class="fa fa-check"></i><b>23</b> Bin quality</a>
<ul>
<li class="chapter" data-level="23.1" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#workflow-concept"><i class="fa fa-check"></i><b>23.1</b> Workflow concept</a>
<ul>
<li class="chapter" data-level="23.1.1" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#input-pre-processing"><i class="fa fa-check"></i><b>23.1.1</b> Input Pre-processing</a></li>
<li class="chapter" data-level="23.1.2" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#stage-i"><i class="fa fa-check"></i><b>23.1.2</b> Stage I</a></li>
<li class="chapter" data-level="23.1.3" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#stage-ii"><i class="fa fa-check"></i><b>23.1.3</b> Stage II</a></li>
</ul></li>
<li class="chapter" data-level="23.2" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#mamba"><i class="fa fa-check"></i><b>23.2</b> Mamba</a></li>
<li class="chapter" data-level="23.3" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#cocopye-run"><i class="fa fa-check"></i><b>23.3</b> CoCoPyE: run</a>
<ul>
<li class="chapter" data-level="" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#parameters-10"><i class="fa fa-check"></i>Parameters</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#cocopye-output"><i class="fa fa-check"></i><b>23.4</b> CoCoPyE: output</a></li>
<li class="chapter" data-level="23.5" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#cocopye-quality-score"><i class="fa fa-check"></i><b>23.5</b> CoCoPyE: Quality score</a>
<ul>
<li class="chapter" data-level="" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#quality-file"><i class="fa fa-check"></i>Quality file</a></li>
<li class="chapter" data-level="" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#calculate-quality-with-awk"><i class="fa fa-check"></i>Calculate quality with awk</a></li>
<li class="chapter" data-level="" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#add-quality-to-the-checkm-results-file"><i class="fa fa-check"></i>Add quality to the checkm results file</a></li>
</ul></li>
<li class="chapter" data-level="23.6" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#checkm-mcqs"><i class="fa fa-check"></i><b>23.6</b> CheckM: MCQs</a></li>
<li class="chapter" data-level="23.7" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#bin-quality-summary"><i class="fa fa-check"></i><b>23.7</b> Bin quality summary</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html"><i class="fa fa-check"></i><b>24</b> Assembly functional annotation</a>
<ul>
<li class="chapter" data-level="24.1" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#taxonomic-annotation"><i class="fa fa-check"></i><b>24.1</b> Taxonomic annotation</a></li>
<li class="chapter" data-level="24.2" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#bakta"><i class="fa fa-check"></i><b>24.2</b> Bakta</a>
<ul>
<li class="chapter" data-level="24.2.1" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#bakta-run"><i class="fa fa-check"></i><b>24.2.1</b> Bakta: run</a></li>
<li class="chapter" data-level="24.2.2" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#bakta-premade-results"><i class="fa fa-check"></i><b>24.2.2</b> Bakta: premade results</a></li>
<li class="chapter" data-level="24.2.3" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#bakta-output"><i class="fa fa-check"></i><b>24.2.3</b> Bakta: output</a></li>
</ul></li>
<li class="chapter" data-level="24.3" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#minpath"><i class="fa fa-check"></i><b>24.3</b> MinPath</a>
<ul>
<li class="chapter" data-level="24.3.1" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#minpath-ec-extraction"><i class="fa fa-check"></i><b>24.3.1</b> MinPath: EC extraction</a></li>
<li class="chapter" data-level="24.3.2" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#minpathrun"><i class="fa fa-check"></i><b>24.3.2</b> MinPath: run</a></li>
<li class="chapter" data-level="24.3.3" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#minpath-output"><i class="fa fa-check"></i><b>24.3.3</b> MinPath: output</a></li>
<li class="chapter" data-level="24.3.4" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#keggs"><i class="fa fa-check"></i><b>24.3.4</b> KEGGs</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="25-Appendix.html"><a href="25-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="25-Appendix.html"><a href="25-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="25-Appendix.html"><a href="25-Appendix.html#next-steps"><i class="fa fa-check"></i><b>B</b> Next steps</a></li>
<li class="chapter" data-level="C" data-path="25-Appendix.html"><a href="25-Appendix.html#manuals"><i class="fa fa-check"></i><b>C</b> Manuals</a></li>
<li class="chapter" data-level="D" data-path="25-Appendix.html"><a href="25-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>D</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="humannmultisampleprocessing" class="section level1 hasAnchor" number="15">
<h1><span class="header-section-number">Chapter 15</span> Multi sample processing<a href="15-Multi_sample_processing.html#humannmultisampleprocessing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/k_w_venn.png" style="width:150px; background: white; border-radius:5px; border: white solid 5px" />
</center>
<p>Looking at the functional profile of one sample in isolation is usually not very informative. First, there is nothing to compare it to and second, there are no biological replicates. We will therefore use all the Korean and Western diet samples.</p>
<p>It would take many hours to analyse all of the data using <code>HUMAnN</code> and is outside the scope of this course. For this reason, samples were analysed prior to the workshop to generate the output files we covered above.</p>
<p>For the purposes of this comparison, we will look at the pathway abundances only. First copy over the data directory containg the gene families tables and have a look in it.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb47-1"><a href="15-Multi_sample_processing.html#cb47-1" tabindex="-1"></a><span class="co">#Ensure you are in the correct directory</span></span>
<span id="cb47-2"><a href="15-Multi_sample_processing.html#cb47-2" tabindex="-1"></a><span class="bu">cd</span> ~/4-FunctionalProfiling</span>
<span id="cb47-3"><a href="15-Multi_sample_processing.html#cb47-3" tabindex="-1"></a><span class="co">#Copy directory with pre made results</span></span>
<span id="cb47-4"><a href="15-Multi_sample_processing.html#cb47-4" tabindex="-1"></a><span class="fu">cp</span> <span class="at">-r</span> /pub14/tea/nsc206/NEOF/Shotgun_metagenomics/DietPathAbundance .</span>
<span id="cb47-5"><a href="15-Multi_sample_processing.html#cb47-5" tabindex="-1"></a><span class="co">#Move into the copied directory</span></span>
<span id="cb47-6"><a href="15-Multi_sample_processing.html#cb47-6" tabindex="-1"></a><span class="bu">cd</span> DietPathAbundance</span>
<span id="cb47-7"><a href="15-Multi_sample_processing.html#cb47-7" tabindex="-1"></a><span class="co">#List files</span></span>
<span id="cb47-8"><a href="15-Multi_sample_processing.html#cb47-8" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>You will see there are 12 files prefixed with K and 12 prefixed with W, for the Korean diet and Western diet samples, respectively. Take a look at the file for K1.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb48-1"><a href="15-Multi_sample_processing.html#cb48-1" tabindex="-1"></a><span class="fu">less</span> K1_pathabundance.tsv</span></code></pre></div>
<p>There are a lot of pathways in the file. Quit out of the <code>less</code> viewer (<code>q</code>) and look at the entries for one specific pathway, COA-PWY-1 (a coenzyme A biosynthesis II pathway).</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb49-1"><a href="15-Multi_sample_processing.html#cb49-1" tabindex="-1"></a><span class="fu">grep</span> COA-PWY-1 K1_pathabundance.tsv <span class="kw">|</span> <span class="fu">less</span></span></code></pre></div>
<p>This shows 30 entries/lines with the top entry/line:</p>
<p><strong><code>COA-PWY-1: superpathway of coenzyme A biosynthesis III (mammals)        6790.1517478104</code></strong></p>
<p>This shows the abundance of the pathway across the entire sample (6790.1517478104).</p>
<p>The other entries show the species stratification information (mentioned above) of the pathway. I.e. the second line:</p>
<p><strong><code>COA-PWY-1: superpathway of coenzyme A biosynthesis III (mammals)|g__Bacteroides.s__Bacteroides_dorei    1292.7711872228</code></strong></p>
<p>shows the abundance of the pathway that is contributed by the species <em>Bacteroides dorei</em> (1292.7711872228).</p>
<p><strong>Note</strong>: The species stratified pathway abundances may not equal the total community pathway abundance. Please see this <a href="https://forum.biobakery.org/t/humann3-pathway-abundance-table-pathway-sum-and-species-sum-different/1471">forum post</a> for details.</p>
<p>With this information we will carry out some comparisons including biomarker detection to determine which pathways are differentially abundant between the Western diet and Korean diet samples.</p>
<p><strong>Note</strong>: The following methods/pipeline can be used for the genefamilies and pathcoverage tables in your own future analyses.</p>
<div id="combining-data" class="section level2 hasAnchor" number="15.1">
<h2><span class="header-section-number">15.1</span> Combining data<a href="15-Multi_sample_processing.html#combining-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/merge.png" style="width:100px; background:white; border-radius:5px" />
</center>
<p>First, we need to combine these 24 tables into one large results table. <code>HUMAnN</code> provides a tool to do this:</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb50-1"><a href="15-Multi_sample_processing.html#cb50-1" tabindex="-1"></a><span class="co">#Change directory to main Functional profiling directory</span></span>
<span id="cb50-2"><a href="15-Multi_sample_processing.html#cb50-2" tabindex="-1"></a><span class="bu">cd</span> ~/4-FunctionalProfiling</span>
<span id="cb50-3"><a href="15-Multi_sample_processing.html#cb50-3" tabindex="-1"></a><span class="co">#Join the tables</span></span>
<span id="cb50-4"><a href="15-Multi_sample_processing.html#cb50-4" tabindex="-1"></a><span class="ex">humann_join_tables</span> <span class="at">--input</span> DietPathAbundance/ <span class="at">--output</span> diet.tsv</span></code></pre></div>
<p>This command will look for all tables in the <code>DietPathAbundance</code> directory and generate a large, 25 column table called <code>diet.tsv</code>. You can inspect the file to ensure that this has worked correctly.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb51-1"><a href="15-Multi_sample_processing.html#cb51-1" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> diet.tsv</span></code></pre></div>
</div>
<div id="split-stratified-table" class="section level2 hasAnchor" number="15.2">
<h2><span class="header-section-number">15.2</span> Split stratified table<a href="15-Multi_sample_processing.html#split-stratified-table" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/split.png" style="width:100px; background:white; border-radius:5px" />
</center>
<p>For this tutorial we do not want the species stratification information. We will therefore split the table to create 2 new files:</p>
<ul>
<li><code>diet_unstratified.tsv</code>: This table only contains the total abundance values for the pathways. It does not contain any species stratification information.</li>
<li><code>diet_stratified.tsv</code>: This table only contains the species stratification abundance values for the pathways. It does not contain the total abundance information.</li>
</ul>
<p>To create the split files and output them to your current directory, run the following command:</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb52-1"><a href="15-Multi_sample_processing.html#cb52-1" tabindex="-1"></a><span class="ex">humann_split_stratified_table</span> <span class="at">--input</span> diet.tsv <span class="at">--output</span> .</span></code></pre></div>
<p>We will use the file <code>diet_unstratified.tsv</code> for our downstream analysis.</p>
<p>Before you move on feel free to inspect the output files with the <code>less</code> command.</p>
<p><strong>Note</strong>: You can use any of the three tables (unsplit table, unstratified table, or stratified table) in your own analysis. This depends on your question and data.</p>
</div>
<div id="humannrenormalise" class="section level2 hasAnchor" number="15.3">
<h2><span class="header-section-number">15.3</span> Renormalising data<a href="15-Multi_sample_processing.html#humannrenormalise" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/normalisation.png" style="width:100px; background:white; border-radius:5px" />
</center>
<p>The next step is to renormalise the data. Currently, all of the abundance values are only normalised within each sample (RPKs). However, they are not normalised between samples, and this is very important to do. For example, if we had sequenced two samples, A and B, and we obtained 5 million reads for sample A and 20 million reads for sample B, without normalisation, it might look there was up to 4x as much functional activity in sample B!</p>
<p>To correct for this, we normalise the abundance values based on the number of reads in each sample. We will normalise to relative abundance (<code>--units relab</code>) where all abundances for each sample add up to 1.</p>
<p>Renormalisation command:</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb53-1"><a href="15-Multi_sample_processing.html#cb53-1" tabindex="-1"></a><span class="ex">humann_renorm_table</span> <span class="dt">\</span></span>
<span id="cb53-2"><a href="15-Multi_sample_processing.html#cb53-2" tabindex="-1"></a>--units relab <span class="dt">\</span></span>
<span id="cb53-3"><a href="15-Multi_sample_processing.html#cb53-3" tabindex="-1"></a>--input diet_unstratified.tsv <span class="dt">\</span></span>
<span id="cb53-4"><a href="15-Multi_sample_processing.html#cb53-4" tabindex="-1"></a>--special n <span class="dt">\</span></span>
<span id="cb53-5"><a href="15-Multi_sample_processing.html#cb53-5" tabindex="-1"></a>--output diet_unstratified.relab.tsv</span></code></pre></div>
<p>This command generates the normalised data in the new table <code>diet_unstratified.relab.tsv</code>. The <code>--special n</code> option tells the script to remove all unmapped and unassigned values (<strong>UNMAPPED</strong> &amp; <strong>UNINTEGRATED</strong>) from the table.</p>
<p><strong>Note:</strong> With the gene families information ensure you normalise by CPM (counts per million) with the option <code>--units cpm</code>. More info can be found on the <a href="https://github.com/biobakery/biobakery/wiki/humann3#31-normalizing-rpks-to-relative-abundance">Normalizing RPKs to relative abundance</a> section of the HUMAnN 3.0 tutorial.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="14-HUMAnN.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="16-Heatmap.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
