<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 16 Megahit Assembly | Shotgun Metagenomics</title>
  <meta name="description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="generator" content="bookdown 0.31 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 16 Megahit Assembly | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 16 Megahit Assembly | Shotgun Metagenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2023-03-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="15-Stitching_reads.html"/>
<link rel="next" href="17-Genome_binning.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Start_conda.html"><a href="04-Start_conda.html"><i class="fa fa-check"></i><b>4</b> Startup &amp; conda</a></li>
<li class="part"><span><b>Data &amp; QC</b></span></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-summary"><i class="fa fa-check"></i><b>7.5</b> Host removal summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy of reads</b></span></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a></li>
<li class="chapter" data-level="9" data-path="09-Kraken2.html"><a href="09-Kraken2.html"><i class="fa fa-check"></i><b>9</b> Kraken2</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-run"><i class="fa fa-check"></i><b>9.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="9.2" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-output"><i class="fa fa-check"></i><b>9.2</b> Kraken2: output</a></li>
<li class="chapter" data-level="9.3" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken-2-mcqs"><i class="fa fa-check"></i><b>9.3</b> Kraken 2: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Krona.html"><a href="10-Krona.html"><i class="fa fa-check"></i><b>10</b> Krona plot</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Krona.html"><a href="10-Krona.html#krona-run"><i class="fa fa-check"></i><b>10.1</b> Krona: run</a></li>
<li class="chapter" data-level="10.2" data-path="10-Krona.html"><a href="10-Krona.html#krona-visualise"><i class="fa fa-check"></i><b>10.2</b> Krona: visualise</a></li>
<li class="chapter" data-level="10.3" data-path="10-Krona.html"><a href="10-Krona.html#krona-mcqs"><i class="fa fa-check"></i><b>10.3</b> Krona: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Bracken.html"><a href="11-Bracken.html"><i class="fa fa-check"></i><b>11</b> Bracken</a>
<ul>
<li class="chapter" data-level="11.0.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-run"><i class="fa fa-check"></i><b>11.0.1</b> Bracken: run</a></li>
<li class="chapter" data-level="11.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-output"><i class="fa fa-check"></i><b>11.1</b> Bracken: output</a></li>
<li class="chapter" data-level="11.2" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-mcqs"><i class="fa fa-check"></i><b>11.2</b> Bracken: MCQs</a></li>
<li class="chapter" data-level="11.3" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-merging-output"><i class="fa fa-check"></i><b>11.3</b> Bracken: merging output</a></li>
<li class="chapter" data-level="11.4" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-extracting-output"><i class="fa fa-check"></i><b>11.4</b> Bracken: extracting output</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Lefse.html"><a href="12-Lefse.html"><i class="fa fa-check"></i><b>12</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2_lefse_metadata"><i class="fa fa-check"></i><b>12.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="12.2" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-conda"><i class="fa fa-check"></i><b>12.2</b> LEfSe: conda</a></li>
<li class="chapter" data-level="12.3" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-format"><i class="fa fa-check"></i><b>12.3</b> LEfSe: format</a></li>
<li class="chapter" data-level="12.4" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-run"><i class="fa fa-check"></i><b>12.4</b> LEfSe: run</a></li>
<li class="chapter" data-level="12.5" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-visualisation"><i class="fa fa-check"></i><b>12.5</b> LEfSe: visualisation</a></li>
<li class="chapter" data-level="12.6" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-mcqs"><i class="fa fa-check"></i><b>12.6</b> LEfSe: MCQs</a></li>
<li class="chapter" data-level="12.7" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>12.7</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="part"><span><b>Functional profiling of reads</b></span></li>
<li class="chapter" data-level="13" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html"><i class="fa fa-check"></i><b>13</b> HUMAnN Functional profiling</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann"><i class="fa fa-check"></i><b>13.1</b> HUMAnN</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-conda-directories-and-files"><i class="fa fa-check"></i><b>13.1.1</b> HUMAnN: conda, directories, and files</a></li>
<li class="chapter" data-level="13.1.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-run"><i class="fa fa-check"></i><b>13.1.2</b> HUMAnN: run</a></li>
<li class="chapter" data-level="13.1.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-output"><i class="fa fa-check"></i><b>13.1.3</b> HUMAnN: output</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humannmultisampleprocessing"><i class="fa fa-check"></i><b>13.2</b> Multi sample processing</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-combining-data"><i class="fa fa-check"></i><b>13.2.1</b> HUMAnN: combining data</a></li>
<li class="chapter" data-level="13.2.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-split-stratified-table"><i class="fa fa-check"></i><b>13.2.2</b> HUMAnN: split stratified table</a></li>
<li class="chapter" data-level="13.2.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humannrenormalise"><i class="fa fa-check"></i><b>13.2.3</b> HUMAnN: renormalising data</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#heatmap"><i class="fa fa-check"></i><b>13.3</b> Heatmap</a></li>
<li class="chapter" data-level="13.4" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann_lefse"><i class="fa fa-check"></i><b>13.4</b> LEfSe</a></li>
<li class="chapter" data-level="13.5" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#optional-task"><i class="fa fa-check"></i><b>13.5</b> Optional task</a></li>
</ul></li>
<li class="part"><span><b>Assembly Approach</b></span></li>
<li class="chapter" data-level="14" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html"><i class="fa fa-check"></i><b>14</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html#metagenome-assembly-conda"><i class="fa fa-check"></i><b>14.1</b> Metagenome assembly: Conda</a></li>
<li class="chapter" data-level="14.2" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>14.2</b> A primer on short read assembly</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html"><i class="fa fa-check"></i><b>15</b> Stitching read pairs</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-run"><i class="fa fa-check"></i><b>15.1</b> FLASH: run</a></li>
<li class="chapter" data-level="15.2" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-output"><i class="fa fa-check"></i><b>15.2</b> FLASH: Output</a></li>
<li class="chapter" data-level="15.3" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-mcqs"><i class="fa fa-check"></i><b>15.3</b> FLASH: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16_Megahit.html"><a href="16_Megahit.html"><i class="fa fa-check"></i><b>16</b> Megahit Assembly</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16_Megahit.html"><a href="16_Megahit.html#megahit"><i class="fa fa-check"></i><b>16.1</b> MEGAHIT</a></li>
<li class="chapter" data-level="16.2" data-path="16_Megahit.html"><a href="16_Megahit.html#quast"><i class="fa fa-check"></i><b>16.2</b> QUAST</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-conda"><i class="fa fa-check"></i><b>16.2.1</b> QUAST: conda</a></li>
<li class="chapter" data-level="16.2.2" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-run"><i class="fa fa-check"></i><b>16.2.2</b> QUAST: run</a></li>
<li class="chapter" data-level="16.2.3" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-visualise"><i class="fa fa-check"></i><b>16.2.3</b> QUAST: visualise</a></li>
<li class="chapter" data-level="16.2.4" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-mcqs"><i class="fa fa-check"></i><b>16.2.4</b> QUAST: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="16_Megahit.html"><a href="16_Megahit.html#metagenome-assembly-summary"><i class="fa fa-check"></i><b>16.3</b> Metagenome assembly summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html"><i class="fa fa-check"></i><b>17</b> Genome binning</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>17.1</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-conda-environment-directory"><i class="fa fa-check"></i><b>17.1.1</b> MetaBAT2: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="17.1.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-depth-calculation"><i class="fa fa-check"></i><b>17.1.2</b> MetaBAT2: depth calculation</a></li>
<li class="chapter" data-level="17.1.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-run"><i class="fa fa-check"></i><b>17.1.3</b> MetaBAT2: run</a></li>
<li class="chapter" data-level="17.1.4" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-output"><i class="fa fa-check"></i><b>17.1.4</b> MetaBAT2: output</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm"><i class="fa fa-check"></i><b>17.2</b> CheckM</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-conda"><i class="fa fa-check"></i><b>17.2.1</b> CheckM: Conda</a></li>
<li class="chapter" data-level="17.2.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-run"><i class="fa fa-check"></i><b>17.2.2</b> CheckM: run</a></li>
<li class="chapter" data-level="17.2.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-output"><i class="fa fa-check"></i><b>17.2.3</b> CheckM: output</a></li>
<li class="chapter" data-level="17.2.4" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-mcqs"><i class="fa fa-check"></i><b>17.2.4</b> CheckM: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#binning-summary"><i class="fa fa-check"></i><b>17.3</b> Binning summary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html"><i class="fa fa-check"></i><b>18</b> Assembly functional annotation</a>
<ul>
<li class="chapter" data-level="18.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#taxonomic-annotation"><i class="fa fa-check"></i><b>18.1</b> Taxonomic annotation</a></li>
<li class="chapter" data-level="18.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#prokka"><i class="fa fa-check"></i><b>18.2</b> Prokka</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#prokka_conda"><i class="fa fa-check"></i><b>18.2.1</b> Prokka: conda and directories</a></li>
<li class="chapter" data-level="18.2.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#prokka-run"><i class="fa fa-check"></i><b>18.2.2</b> Prokka: run</a></li>
<li class="chapter" data-level="18.2.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#prokka-output"><i class="fa fa-check"></i><b>18.2.3</b> Prokka: output</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath"><i class="fa fa-check"></i><b>18.3</b> MinPath</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath-conda-and-directory"><i class="fa fa-check"></i><b>18.3.1</b> MinPath: conda and directory</a></li>
<li class="chapter" data-level="18.3.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath-ec-extraction"><i class="fa fa-check"></i><b>18.3.2</b> MinPath: EC extraction</a></li>
<li class="chapter" data-level="18.3.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpathrun"><i class="fa fa-check"></i><b>18.3.3</b> MinPath: run</a></li>
<li class="chapter" data-level="18.3.4" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath-output"><i class="fa fa-check"></i><b>18.3.4</b> MinPath: output</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="19-Appendix.html"><a href="19-Appendix.html"><i class="fa fa-check"></i><b>A</b> Next steps</a></li>
<li class="chapter" data-level="B" data-path="19-Appendix.html"><a href="19-Appendix.html#manuals"><i class="fa fa-check"></i><b>B</b> Manuals</a></li>
<li class="chapter" data-level="C" data-path="19-Appendix.html"><a href="19-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>C</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="megahit-assembly" class="section level1 hasAnchor" number="16">
<h1><span class="header-section-number">Chapter 16</span> Megahit Assembly<a href="16_Megahit.html#megahit-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/puzzle.png" width="20%" style="display: block; margin: auto;" /></p>
<p>We will use our stitched and unstitched reads to produce an assembly with<code>MEGAHIT</code>.</p>
<div id="megahit" class="section level2 hasAnchor" number="16.1">
<h2><span class="header-section-number">16.1</span> MEGAHIT<a href="16_Megahit.html#megahit" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/play_blue.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Create a new directory to store our assembly in.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb63-1"><a href="16_Megahit.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ..</span>
<span id="cb63-2"><a href="16_Megahit.html#cb63-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 6-Assembly</span>
<span id="cb63-3"><a href="16_Megahit.html#cb63-3" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> 6-Assembly</span></code></pre></div>
<p>Now run the metagenome assembler <code>MEGAHIT</code> using our newly stitched read data.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb64-1"><a href="16_Megahit.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="ex">megahit</span> <span class="dt">\</span></span>
<span id="cb64-2"><a href="16_Megahit.html#cb64-2" aria-hidden="true" tabindex="-1"></a>-r ../5-Stitched/K1.extendedFrags.fastq.gz <span class="dt">\</span></span>
<span id="cb64-3"><a href="16_Megahit.html#cb64-3" aria-hidden="true" tabindex="-1"></a>-1 ../5-Stitched/K1.notCombined_1.fastq.gz <span class="dt">\</span></span>
<span id="cb64-4"><a href="16_Megahit.html#cb64-4" aria-hidden="true" tabindex="-1"></a>-2 ../5-Stitched/K1.notCombined_2.fastq.gz <span class="dt">\</span></span>
<span id="cb64-5"><a href="16_Megahit.html#cb64-5" aria-hidden="true" tabindex="-1"></a>-o K1 <span class="dt">\</span></span>
<span id="cb64-6"><a href="16_Megahit.html#cb64-6" aria-hidden="true" tabindex="-1"></a>-t 12 <span class="dt">\</span></span>
<span id="cb64-7"><a href="16_Megahit.html#cb64-7" aria-hidden="true" tabindex="-1"></a>--k-list 29,49,69,89,109,129,149,169,189</span></code></pre></div>
<div id="parameters-8" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="16_Megahit.html#parameters-8" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/parameter_blue.png" width="10%" style="display: block; margin: auto;" /></p>
<ul>
<li><code>-r</code>: Single-end reads to be used for assembly. We are using our successfully stitched reads.</li>
<li><code>-1</code>: Forward reads of paired end reads to be used for assembly. We are using the reads that did not stitch as they still have useful information.</li>
<li><code>-2</code>: Reverse reads of paired end reads to be used for assembly. We are using the reads that did not stitch as they still have useful information.</li>
<li><code>-o</code>: Output directory.</li>
<li><code>-t</code>: Number of threads to be used for process.</li>
<li><code>--k-list</code>: K-mer list.</li>
</ul>
<p>The <strong>k-mer list</strong> instructs <code>MEGAHIT</code> to first generate an assembly using a k-mer size of 29 bp and when that is complete, integrate the results into an assembly using a k-mer size of 49 bp, and so on up to a final iteration using a k-mer size of 189 bp. This large range of k-mer lengths should give us a good assembly, given the data. However, it may take a while to run. This might be a good time to read on or take a break whilst the command runs.</p>
<p>If you need a command prompt (your current one is busy because MEGAHIT is running), right click on the main screen, choose <code>Applications</code> -&gt; <code>Shell</code> -&gt; <code>bash</code>.</p>
<p>Once the assembly is completed, we can look at the output FASTA file containing the contigs:</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb65-1"><a href="16_Megahit.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> K1/final.contigs.fa</span></code></pre></div>
<p>There is not much to see. When happy, quit the <code>less</code> (<code>q</code>) and carry on to <code>QUAST</code>.</p>
</div>
</div>
<div id="quast" class="section level2 hasAnchor" number="16.2">
<h2><span class="header-section-number">16.2</span> QUAST<a href="16_Megahit.html#quast" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/inspect.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We can generate some metrics based on the assembly with <code>QUAST</code>.</p>
<div id="quast-conda" class="section level3 hasAnchor" number="16.2.1">
<h3><span class="header-section-number">16.2.1</span> QUAST: conda<a href="16_Megahit.html#quast-conda" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Due to a python version conflict we need to use another conda environment.</p>
<p>Open a new terminal (right click on the main screen, choose <code>Applications</code> -&gt; <code>Shell</code> -&gt; <code>bash</code>) and run the below.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb66-1"><a href="16_Megahit.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co">#use script to activate conda env</span></span>
<span id="cb66-2"><a href="16_Megahit.html#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> usegenoassess</span></code></pre></div>
<p>We will use <code>QUAST</code> for genome contiguity assessment but first we will change directory to <code>6-Assembly</code> and create a directory for the <code>QUAST</code> output.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb67-1"><a href="16_Megahit.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Change directory</span></span>
<span id="cb67-2"><a href="16_Megahit.html#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/6-Assembly</span>
<span id="cb67-3"><a href="16_Megahit.html#cb67-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Create QUAST output directory</span></span>
<span id="cb67-4"><a href="16_Megahit.html#cb67-4" aria-hidden="true" tabindex="-1"></a><span class="co">#The option -p will create a directory and any required</span></span>
<span id="cb67-5"><a href="16_Megahit.html#cb67-5" aria-hidden="true" tabindex="-1"></a><span class="co">#parent directories</span></span>
<span id="cb67-6"><a href="16_Megahit.html#cb67-6" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> quast/K1</span></code></pre></div>
</div>
<div id="quast-run" class="section level3 hasAnchor" number="16.2.2">
<h3><span class="header-section-number">16.2.2</span> QUAST: run<a href="16_Megahit.html#quast-run" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/play_green.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Now to run <code>QUAST</code>.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb68-1"><a href="16_Megahit.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="ex">quast</span> <span class="at">-o</span> quast/K1 K1/final.contigs.fa</span></code></pre></div>
</div>
<div id="quast-visualise" class="section level3 hasAnchor" number="16.2.3">
<h3><span class="header-section-number">16.2.3</span> QUAST: visualise<a href="16_Megahit.html#quast-visualise" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/eye_green.png" width="10%" style="display: block; margin: auto;" /></p>
<p><code>QUAST</code> will run relatively quickly. Once complete view the output with <code>firefox</code>.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb69-1"><a href="16_Megahit.html#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="ex">firefox</span> quast/K1/report.html</span></code></pre></div>
<p>The report tells us quite a bit about the assembly quality. Two definitions that you may not be aware are <strong>N50</strong> and <strong>L50</strong>. To calculate these values:</p>
<ul>
<li>Order the contigs from largest to smallest.</li>
<li>Total up the sizes from biggest downwards.</li>
<li>The contig we reach where our total is at least 50% of the size of the whole assembly is the N50 contig.</li>
<li><strong>N50</strong> equals the length of the N50 contig.</li>
<li><strong>L50</strong> is the number of contigs with a length equal to or greater than N50.</li>
</ul>
<p><img src="figures/n50_n90.png" width="40%" style="display: block; margin: auto;" /></p>
</div>
<div id="quast-mcqs" class="section level3 hasAnchor" number="16.2.4">
<h3><span class="header-section-number">16.2.4</span> QUAST: MCQs<a href="16_Megahit.html#quast-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/question_bubble_green.png" width="10%" style="display: block; margin: auto;" /></p>
<ol style="list-style-type: decimal">
<li>What is the total length of the assembly?
<div id="radio_NBFBMGAGPB" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_NBFBMGAGPB" value=""></input> <span><strong>39.88</strong></span></label><label><input type="radio" autocomplete="off" name="radio_NBFBMGAGPB" value=""></input> <span><strong>18,650</strong></span></label><label><input type="radio" autocomplete="off" name="radio_NBFBMGAGPB" value="answer"></input> <span><strong>16,187,295</strong></span></label>
</div></li>
<li>How many contigs does the assembly consist of?
<div id="radio_VYNXECQVUR" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_VYNXECQVUR" value=""></input> <span><strong>39.88</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VYNXECQVUR" value="answer"></input> <span><strong>18,650</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VYNXECQVUR" value=""></input> <span><strong>16,187,295</strong></span></label>
</div></li>
<li>What is the GC% of the assembly?
<div id="radio_PEDCVLINGZ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_PEDCVLINGZ" value="answer"></input> <span><strong>39.88</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PEDCVLINGZ" value=""></input> <span><strong>18,650</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PEDCVLINGZ" value=""></input> <span><strong>16,187,295</strong></span></label>
</div></li>
<li>What is the N50 of the assembly?
<div id="radio_WXTRTHDAFJ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_WXTRTHDAFJ" value="answer"></input> <span><strong>1,363</strong></span></label><label><input type="radio" autocomplete="off" name="radio_WXTRTHDAFJ" value=""></input> <span><strong>2,178</strong></span></label><label><input type="radio" autocomplete="off" name="radio_WXTRTHDAFJ" value=""></input> <span><strong>87,586</strong></span></label>
</div></li>
<li>What is the L50 of the assembly?
<div id="radio_EFVERTHXGQ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_EFVERTHXGQ" value=""></input> <span><strong>1,363</strong></span></label><label><input type="radio" autocomplete="off" name="radio_EFVERTHXGQ" value="answer"></input> <span><strong>2,178</strong></span></label><label><input type="radio" autocomplete="off" name="radio_EFVERTHXGQ" value=""></input> <span><strong>87,586</strong></span></label>
</div></li>
<li>what is the length of the largest contig?
<div id="radio_SLFWQOGZQA" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_SLFWQOGZQA" value=""></input> <span><strong>1,363</strong></span></label><label><input type="radio" autocomplete="off" name="radio_SLFWQOGZQA" value=""></input> <span><strong>2,178</strong></span></label><label><input type="radio" autocomplete="off" name="radio_SLFWQOGZQA" value="answer"></input> <span><strong>87,586</strong></span></label>
</div></li>
</ol>
<p><strong>Questions</strong>
- How do the contig metrics compare to the original reads?</p>
</div>
</div>
<div id="metagenome-assembly-summary" class="section level2 hasAnchor" number="16.3">
<h2><span class="header-section-number">16.3</span> Metagenome assembly summary<a href="16_Megahit.html#metagenome-assembly-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_black.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We now have an assembly. It is not a brilliant one due to us only having used 1 million reads. In real analysis we would prefer fewer but longer contigs. We will explore some tools we can use with our metagenome assembly in the next chapters.</p>
<p>There is also a <code>metaQUAST</code> specifically for metagenome assemblies but it requires reference assemblies be provided.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="15-Stitching_reads.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="17-Genome_binning.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
