<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 17 Genome binning | Shotgun Metagenomics</title>
  <meta name="description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 17 Genome binning | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 17 Genome binning | Shotgun Metagenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2023-06-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="16_Megahit.html"/>
<link rel="next" href="18-Assembly_functional_annotation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Start_conda.html"><a href="04-Start_conda.html"><i class="fa fa-check"></i><b>4</b> Startup &amp; mamba</a></li>
<li class="part"><span><b>Data &amp; QC</b></span></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-summary"><i class="fa fa-check"></i><b>7.5</b> Host removal summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy of reads</b></span></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a></li>
<li class="chapter" data-level="9" data-path="09-Kraken2.html"><a href="09-Kraken2.html"><i class="fa fa-check"></i><b>9</b> Kraken2</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-run"><i class="fa fa-check"></i><b>9.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="9.2" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-output"><i class="fa fa-check"></i><b>9.2</b> Kraken2: output</a></li>
<li class="chapter" data-level="9.3" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken-2-mcqs"><i class="fa fa-check"></i><b>9.3</b> Kraken 2: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Krona.html"><a href="10-Krona.html"><i class="fa fa-check"></i><b>10</b> Krona plot</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Krona.html"><a href="10-Krona.html#krona-run"><i class="fa fa-check"></i><b>10.1</b> Krona: run</a></li>
<li class="chapter" data-level="10.2" data-path="10-Krona.html"><a href="10-Krona.html#krona-visualise"><i class="fa fa-check"></i><b>10.2</b> Krona: visualise</a></li>
<li class="chapter" data-level="10.3" data-path="10-Krona.html"><a href="10-Krona.html#krona-mcqs"><i class="fa fa-check"></i><b>10.3</b> Krona: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Bracken.html"><a href="11-Bracken.html"><i class="fa fa-check"></i><b>11</b> Bracken</a>
<ul>
<li class="chapter" data-level="11.0.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-run"><i class="fa fa-check"></i><b>11.0.1</b> Bracken: run</a></li>
<li class="chapter" data-level="11.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-output"><i class="fa fa-check"></i><b>11.1</b> Bracken: output</a></li>
<li class="chapter" data-level="11.2" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-mcqs"><i class="fa fa-check"></i><b>11.2</b> Bracken: MCQs</a></li>
<li class="chapter" data-level="11.3" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-merging-output"><i class="fa fa-check"></i><b>11.3</b> Bracken: merging output</a></li>
<li class="chapter" data-level="11.4" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-extracting-output"><i class="fa fa-check"></i><b>11.4</b> Bracken: extracting output</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Lefse.html"><a href="12-Lefse.html"><i class="fa fa-check"></i><b>12</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2_lefse_metadata"><i class="fa fa-check"></i><b>12.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="12.2" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-format"><i class="fa fa-check"></i><b>12.2</b> LEfSe: format</a></li>
<li class="chapter" data-level="12.3" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-run"><i class="fa fa-check"></i><b>12.3</b> LEfSe: run</a></li>
<li class="chapter" data-level="12.4" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-visualisation"><i class="fa fa-check"></i><b>12.4</b> LEfSe: visualisation</a></li>
<li class="chapter" data-level="12.5" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-mcqs"><i class="fa fa-check"></i><b>12.5</b> LEfSe: MCQs</a></li>
<li class="chapter" data-level="12.6" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>12.6</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="part"><span><b>Functional profiling of reads</b></span></li>
<li class="chapter" data-level="13" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html"><i class="fa fa-check"></i><b>13</b> HUMAnN Functional profiling</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann"><i class="fa fa-check"></i><b>13.1</b> HUMAnN</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-mamba-directories-and-files"><i class="fa fa-check"></i><b>13.1.1</b> HUMAnN: mamba, directories, and files</a></li>
<li class="chapter" data-level="13.1.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-run"><i class="fa fa-check"></i><b>13.1.2</b> HUMAnN: run</a></li>
<li class="chapter" data-level="13.1.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-output"><i class="fa fa-check"></i><b>13.1.3</b> HUMAnN: output</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humannmultisampleprocessing"><i class="fa fa-check"></i><b>13.2</b> Multi sample processing</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-combining-data"><i class="fa fa-check"></i><b>13.2.1</b> HUMAnN: combining data</a></li>
<li class="chapter" data-level="13.2.2" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann-split-stratified-table"><i class="fa fa-check"></i><b>13.2.2</b> HUMAnN: split stratified table</a></li>
<li class="chapter" data-level="13.2.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humannrenormalise"><i class="fa fa-check"></i><b>13.2.3</b> HUMAnN: renormalising data</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#heatmap"><i class="fa fa-check"></i><b>13.3</b> Heatmap</a></li>
<li class="chapter" data-level="13.4" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#humann_lefse"><i class="fa fa-check"></i><b>13.4</b> LEfSe</a></li>
<li class="chapter" data-level="13.5" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html#optional-task"><i class="fa fa-check"></i><b>13.5</b> Optional task</a></li>
</ul></li>
<li class="part"><span><b>Assembly Approach</b></span></li>
<li class="chapter" data-level="14" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html"><i class="fa fa-check"></i><b>14</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html#metagenome-assembly-conda"><i class="fa fa-check"></i><b>14.1</b> Metagenome assembly: Conda</a></li>
<li class="chapter" data-level="14.2" data-path="14-Metagenome_assembly.html"><a href="14-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>14.2</b> A primer on short read assembly</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html"><i class="fa fa-check"></i><b>15</b> Stitching read pairs</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-run"><i class="fa fa-check"></i><b>15.1</b> FLASH: run</a></li>
<li class="chapter" data-level="15.2" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-output"><i class="fa fa-check"></i><b>15.2</b> FLASH: Output</a></li>
<li class="chapter" data-level="15.3" data-path="15-Stitching_reads.html"><a href="15-Stitching_reads.html#flash-mcqs"><i class="fa fa-check"></i><b>15.3</b> FLASH: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16_Megahit.html"><a href="16_Megahit.html"><i class="fa fa-check"></i><b>16</b> Megahit Assembly</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16_Megahit.html"><a href="16_Megahit.html#megahit"><i class="fa fa-check"></i><b>16.1</b> MEGAHIT</a></li>
<li class="chapter" data-level="16.2" data-path="16_Megahit.html"><a href="16_Megahit.html#quast"><i class="fa fa-check"></i><b>16.2</b> QUAST</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-conda"><i class="fa fa-check"></i><b>16.2.1</b> QUAST: conda</a></li>
<li class="chapter" data-level="16.2.2" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-run"><i class="fa fa-check"></i><b>16.2.2</b> QUAST: run</a></li>
<li class="chapter" data-level="16.2.3" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-visualise"><i class="fa fa-check"></i><b>16.2.3</b> QUAST: visualise</a></li>
<li class="chapter" data-level="16.2.4" data-path="16_Megahit.html"><a href="16_Megahit.html#quast-mcqs"><i class="fa fa-check"></i><b>16.2.4</b> QUAST: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="16_Megahit.html"><a href="16_Megahit.html#metagenome-assembly-summary"><i class="fa fa-check"></i><b>16.3</b> Metagenome assembly summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html"><i class="fa fa-check"></i><b>17</b> Genome binning</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>17.1</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-conda-environment-directory"><i class="fa fa-check"></i><b>17.1.1</b> MetaBAT2: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="17.1.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-depth-calculation"><i class="fa fa-check"></i><b>17.1.2</b> MetaBAT2: depth calculation</a></li>
<li class="chapter" data-level="17.1.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-run"><i class="fa fa-check"></i><b>17.1.3</b> MetaBAT2: run</a></li>
<li class="chapter" data-level="17.1.4" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#metabat2-output"><i class="fa fa-check"></i><b>17.1.4</b> MetaBAT2: output</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm"><i class="fa fa-check"></i><b>17.2</b> CheckM</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-conda"><i class="fa fa-check"></i><b>17.2.1</b> CheckM: Conda</a></li>
<li class="chapter" data-level="17.2.2" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-run"><i class="fa fa-check"></i><b>17.2.2</b> CheckM: run</a></li>
<li class="chapter" data-level="17.2.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-output"><i class="fa fa-check"></i><b>17.2.3</b> CheckM: output</a></li>
<li class="chapter" data-level="17.2.4" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#checkm-mcqs"><i class="fa fa-check"></i><b>17.2.4</b> CheckM: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="17-Genome_binning.html"><a href="17-Genome_binning.html#binning-summary"><i class="fa fa-check"></i><b>17.3</b> Binning summary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html"><i class="fa fa-check"></i><b>18</b> Assembly functional annotation</a>
<ul>
<li class="chapter" data-level="18.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#taxonomic-annotation"><i class="fa fa-check"></i><b>18.1</b> Taxonomic annotation</a></li>
<li class="chapter" data-level="18.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#prokka"><i class="fa fa-check"></i><b>18.2</b> Prokka</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#prokka_conda"><i class="fa fa-check"></i><b>18.2.1</b> Prokka: conda and directories</a></li>
<li class="chapter" data-level="18.2.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#prokka-run"><i class="fa fa-check"></i><b>18.2.2</b> Prokka: run</a></li>
<li class="chapter" data-level="18.2.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#prokka-output"><i class="fa fa-check"></i><b>18.2.3</b> Prokka: output</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath"><i class="fa fa-check"></i><b>18.3</b> MinPath</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath-conda-and-directory"><i class="fa fa-check"></i><b>18.3.1</b> MinPath: conda and directory</a></li>
<li class="chapter" data-level="18.3.2" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath-ec-extraction"><i class="fa fa-check"></i><b>18.3.2</b> MinPath: EC extraction</a></li>
<li class="chapter" data-level="18.3.3" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpathrun"><i class="fa fa-check"></i><b>18.3.3</b> MinPath: run</a></li>
<li class="chapter" data-level="18.3.4" data-path="18-Assembly_functional_annotation.html"><a href="18-Assembly_functional_annotation.html#minpath-output"><i class="fa fa-check"></i><b>18.3.4</b> MinPath: output</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="19-Appendix.html"><a href="19-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="19-Appendix.html"><a href="19-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="19-Appendix.html"><a href="19-Appendix.html#next-steps"><i class="fa fa-check"></i><b>B</b> Next steps</a></li>
<li class="chapter" data-level="C" data-path="19-Appendix.html"><a href="19-Appendix.html#manuals"><i class="fa fa-check"></i><b>C</b> Manuals</a></li>
<li class="chapter" data-level="D" data-path="19-Appendix.html"><a href="19-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>D</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="binning" class="section level1 hasAnchor" number="17">
<h1><span class="header-section-number">Chapter 17</span> Genome binning<a href="17-Genome_binning.html#binning" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/yarn_binning.png" width="40%" style="display: block; margin: auto;" /></p>
<p>A metagenome assembly consists of contigs from many different genomes. At this stage we don't know which contigs are from which species. We could try to taxonomically classify each contig but there are 2 problems with this approach:</p>
<ol style="list-style-type: decimal">
<li>Some contigs may be misclassified which can lead to multiple contigs from the same genome/organism being classified as various taxa.</li>
<li>Databases are incomplete and so some contigs will not be classified at all (microbial dark matter).</li>
</ol>
<p>To alleviate these issues genomic binning can be carried out. This will cluster contigs into bins based on:</p>
<ul>
<li><strong>Coverage</strong>: Contigs with similar coverage are more likely to be from the same genome.</li>
<li><strong>Composition</strong>: Contigs with similar GC content are more likely to belong to the same genome.</li>
</ul>
<p>Genomic binning has been used to discover many new genomes. Additionally, it makes downstream analyses quicker as the downstream steps will be carried out on the sets of bins rather than on one large metagenome assembly.</p>
<p>Binning produces "bins" of contigs of various quality (e.g. draft, complete). These bins are also know as MAGs (Metagenome-assembled genomes). In other words a MAG is a single assembled genome that was assembled with other genomes in a metagenome assembly but later separated from the other assemblies. The term MAG has been adopted by the GSC (Genomics Standards Consortium).</p>
<p>It is recommended to ensure you do not have a poor quality metagenome assembly. Binning requires contigs of good length and good coverage. Extremely low coverage and very short contigs will be excluded from binning.</p>
<div id="metabat2" class="section level2 hasAnchor" number="17.1">
<h2><span class="header-section-number">17.1</span> MetaBAT2<a href="17-Genome_binning.html#metabat2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/bat.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will use <code>MetaBAT2</code> for our genome binning. It is a relatively new binning tool with three major upsides that makes it very popular:</p>
<ol style="list-style-type: decimal">
<li>It has very reliable default parameters meaning virtually no parameter optimisation is required.</li>
<li>It performs very well amongst genome binners.</li>
<li>It is computationally efficient compared to other binners (requires less RAM, cores etc.)</li>
</ol>
<div id="metabat2-conda-environment-directory" class="section level3 hasAnchor" number="17.1.1">
<h3><span class="header-section-number">17.1.1</span> MetaBAT2: Conda environment &amp; directory<a href="17-Genome_binning.html#metabat2-conda-environment-directory" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will use the <code>shotgun_meta</code> conda environment to start for this chapter.</p>
<p>Additionally, make a new directory and move into it.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb69-1"><a href="17-Genome_binning.html#cb69-1" tabindex="-1"></a><span class="co">#Make directory</span></span>
<span id="cb69-2"><a href="17-Genome_binning.html#cb69-2" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> ~/7-Binning/K1</span>
<span id="cb69-3"><a href="17-Genome_binning.html#cb69-3" tabindex="-1"></a><span class="co">#Move into it</span></span>
<span id="cb69-4"><a href="17-Genome_binning.html#cb69-4" tabindex="-1"></a><span class="bu">cd</span> ~/7-Binning/K1</span></code></pre></div>
</div>
<div id="metabat2-depth-calculation" class="section level3 hasAnchor" number="17.1.2">
<h3><span class="header-section-number">17.1.2</span> MetaBAT2: depth calculation<a href="17-Genome_binning.html#metabat2-depth-calculation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/submarine.png" width="10%" style="display: block; margin: auto;" /></p>
<p>To carry out effective genome binning <code>MetaBAT2</code> uses coverage information of the contigs. To calculate depth we need to align the reads to the metagenome assembly.</p>
<p>For the alignment we will use <code>bwa</code>. We need to index our assembly file prior to alignment.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb70-1"><a href="17-Genome_binning.html#cb70-1" tabindex="-1"></a><span class="ex">bwa</span> index ~/6-Assembly/K1/final.contigs.fa</span></code></pre></div>
<p>Next we will align our trimmed paired reads we used to create the stitched reads. We will carry this out with the <code>bwa mem</code> command. <code>bwa mem</code> is a good aligner for short reads. If you are using long reads (PacBio or Nanopore) <code>minimap2</code> will be more appropriate.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb71-1"><a href="17-Genome_binning.html#cb71-1" tabindex="-1"></a><span class="ex">bwa</span> mem ~/6-Assembly/K1/final.contigs.fa <span class="dt">\</span></span>
<span id="cb71-2"><a href="17-Genome_binning.html#cb71-2" tabindex="-1"></a>~/2-Trimmed/K1_R1.fq.gz ~/2-Trimmed/K1_R2.fq.gz <span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb71-3"><a href="17-Genome_binning.html#cb71-3" tabindex="-1"></a>K1.sam</span></code></pre></div>
<p>After alignment we need to get the file ready for the contig depth summarisation step. This requires converting the <code>sam</code> file to a <code>bam</code> (binary form of a <code>sam</code> file) file and then sorting the <code>bam</code> file.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb72-1"><a href="17-Genome_binning.html#cb72-1" tabindex="-1"></a><span class="co"># Convert sam to bam file</span></span>
<span id="cb72-2"><a href="17-Genome_binning.html#cb72-2" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-bu</span> K1.sam <span class="op">&gt;</span> K1.bam</span>
<span id="cb72-3"><a href="17-Genome_binning.html#cb72-3" tabindex="-1"></a><span class="co"># Created sorted bam file</span></span>
<span id="cb72-4"><a href="17-Genome_binning.html#cb72-4" tabindex="-1"></a><span class="ex">samtools</span> sort K1.bam <span class="op">&gt;</span> K1.sort.bam</span></code></pre></div>
<p>Now we can summarise the contig depths from the sorted <code>bam</code> files with <code>MetaBAT2</code>'s <code>jgi_summarize_bam_contig_depths</code> command.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb73-1"><a href="17-Genome_binning.html#cb73-1" tabindex="-1"></a><span class="ex">jgi_summarize_bam_contig_depths</span> <span class="at">--outputDepth</span> K1.depth.txt K1.sort.bam</span></code></pre></div>
<p>You can have a look at the depth file and you will notice there are many contigs with low coverage (&lt;10) and of short length (&lt;1500).</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb74-1"><a href="17-Genome_binning.html#cb74-1" tabindex="-1"></a><span class="fu">less</span> K1.depth.txt</span></code></pre></div>
<p>To get a better look we will open the file in <code>R</code> and look at a summary of the file's table.</p>
<p>Activate <code>R</code>:</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb75-1"><a href="17-Genome_binning.html#cb75-1" tabindex="-1"></a><span class="ex">R</span></span></code></pre></div>
<p>Now in <code>R</code> we will read in the file and get a <code>summary()</code> of it.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="17-Genome_binning.html#cb76-1" tabindex="-1"></a><span class="co">#Read in the table as an object called df (short for data frame)</span></span>
<span id="cb76-2"><a href="17-Genome_binning.html#cb76-2" tabindex="-1"></a><span class="co">#We want the first row to be the column names (header=TRUE)</span></span>
<span id="cb76-3"><a href="17-Genome_binning.html#cb76-3" tabindex="-1"></a><span class="co">#We do not want R to check the column names and &quot;fix&quot; them (check.names=FALSE)</span></span>
<span id="cb76-4"><a href="17-Genome_binning.html#cb76-4" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;K1.depth.txt&quot;</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">check.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb76-5"><a href="17-Genome_binning.html#cb76-5" tabindex="-1"></a><span class="co">#Create a summary of the data</span></span>
<span id="cb76-6"><a href="17-Genome_binning.html#cb76-6" tabindex="-1"></a><span class="fu">summary</span>(df)</span></code></pre></div>
<p>The last command gave us summary information of all the columns. This includes the minimum, maximum, mean, median, and Inter-Quartile Range (IQR) values.</p>
<p>We can see the values of the <code>contigLen</code> and <code>totalAvgDepth</code> are very low. However, this is most likely due to a bunch of short and low coverage contigs which will be ignored by <code>MetaBAT2</code>. Therefore we will remove rows with information on contigs shorter than 1500 and rerun the summary. <code>MetaBAT2</code>'s documentation dictates the minimum contig length should be &gt;=1500 with its default being 2500.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="17-Genome_binning.html#cb77-1" tabindex="-1"></a><span class="co">#Set the new object &quot;df_min1500len&quot; as all rows</span></span>
<span id="cb77-2"><a href="17-Genome_binning.html#cb77-2" tabindex="-1"></a><span class="co">#where the value in the column &quot;contigLen&quot; of &quot;df&quot;</span></span>
<span id="cb77-3"><a href="17-Genome_binning.html#cb77-3" tabindex="-1"></a><span class="co">#Is greater than or equal to 1500</span></span>
<span id="cb77-4"><a href="17-Genome_binning.html#cb77-4" tabindex="-1"></a>df_min1500len <span class="ot">&lt;-</span> df[df<span class="sc">$</span>contigLen <span class="sc">&gt;=</span> <span class="dv">1500</span>,]</span>
<span id="cb77-5"><a href="17-Genome_binning.html#cb77-5" tabindex="-1"></a><span class="co">#Summary of our new data frame</span></span>
<span id="cb77-6"><a href="17-Genome_binning.html#cb77-6" tabindex="-1"></a><span class="fu">summary</span>(df_min1500len)</span></code></pre></div>
<p>That is looking better. The minimum average coverage for <code>MetaBAT2</code> is 1 and our minimum value is 2.844 with a maximum of 92.967. Now you can quit R and continue.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="17-Genome_binning.html#cb78-1" tabindex="-1"></a><span class="co">#quit R</span></span>
<span id="cb78-2"><a href="17-Genome_binning.html#cb78-2" tabindex="-1"></a><span class="fu">q</span>()</span>
<span id="cb78-3"><a href="17-Genome_binning.html#cb78-3" tabindex="-1"></a><span class="co">#On the prompt to save your workspace press &quot;n&quot; and then enter.</span></span></code></pre></div>
<p><strong>Note</strong>: One of the reasons for our short contigs is that we only used a subset of our sequencing dataset for this tutorial due to time concerns.</p>
</div>
<div id="metabat2-run" class="section level3 hasAnchor" number="17.1.3">
<h3><span class="header-section-number">17.1.3</span> MetaBAT2: run<a href="17-Genome_binning.html#metabat2-run" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/play_orange.png" width="10%" style="display: block; margin: auto;" /></p>
<p>With our assembly and its depth information we can run <code>MetaBAT2</code> for binning.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb79-1"><a href="17-Genome_binning.html#cb79-1" tabindex="-1"></a><span class="co">#make a diretcory for the bins</span></span>
<span id="cb79-2"><a href="17-Genome_binning.html#cb79-2" tabindex="-1"></a><span class="fu">mkdir</span> bins</span>
<span id="cb79-3"><a href="17-Genome_binning.html#cb79-3" tabindex="-1"></a><span class="co">#Run MetaBAT2</span></span>
<span id="cb79-4"><a href="17-Genome_binning.html#cb79-4" tabindex="-1"></a><span class="ex">metabat2</span> <span class="dt">\</span></span>
<span id="cb79-5"><a href="17-Genome_binning.html#cb79-5" tabindex="-1"></a>--inFile ~/6-Assembly/K1/final.contigs.fa <span class="dt">\</span></span>
<span id="cb79-6"><a href="17-Genome_binning.html#cb79-6" tabindex="-1"></a>--outFile bins/K1 <span class="dt">\</span></span>
<span id="cb79-7"><a href="17-Genome_binning.html#cb79-7" tabindex="-1"></a>--abdFile K1.depth.txt <span class="dt">\</span></span>
<span id="cb79-8"><a href="17-Genome_binning.html#cb79-8" tabindex="-1"></a>--minContig 1500</span></code></pre></div>
<div id="parameters-9" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="17-Genome_binning.html#parameters-9" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/parameter_orange.png" width="10%" style="display: block; margin: auto;" /></p>
<ul>
<li><code>--inFile</code>: Input metagenome assembly fasta file.</li>
<li><code>--outFile</code>: Prefix of output files.</li>
<li><code>--abdFile</code>: Base depth file.</li>
<li><code>--minContig</code>: Minimum size of contigs to be used for binning.
<ul>
<li>The default is 2500.</li>
<li>We used the minimum value of 1500 as we are using tutorial data. We recommend using the default in your own analysis.</li>
</ul></li>
</ul>
</div>
</div>
<div id="metabat2-output" class="section level3 hasAnchor" number="17.1.4">
<h3><span class="header-section-number">17.1.4</span> MetaBAT2: output<a href="17-Genome_binning.html#metabat2-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/output_file_orange.png" width="10%" style="display: block; margin: auto;" /></p>
<p>List the contents of the output directory and you'll see there are 2 fasta files with the prefix of <code>K1</code>. These are the 2 bins that will hopefully contain 1 MAG (Metagenome-Assembled Genome).</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb80-1"><a href="17-Genome_binning.html#cb80-1" tabindex="-1"></a><span class="fu">ls</span> bins</span></code></pre></div>
</div>
</div>
<div id="checkm" class="section level2 hasAnchor" number="17.2">
<h2><span class="header-section-number">17.2</span> CheckM<a href="17-Genome_binning.html#checkm" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/CheckM.png" width="40%" style="display: block; margin: auto;" /></p>
<p><code>CheckM</code> is a useful tool to assess the quality of assembled bacterial and archaeal genomes. This can be used on assemblies produced from single cell, single isolate, or metagenome data. Additionally, it can be used to identify bins that are likely candidates for merging. This occurs when one genome has been separated into different bins.</p>
<p>An important part of CheckM is the ubiquitous and single-copy genes it utilises. It has sets of these genes for different phylogenetic lineages. With these it can determine:</p>
<ul>
<li>What lineage a bin/MAG belongs to.
<ul>
<li>Does it contain genes only found in <em>Escherichia</em>?</li>
</ul></li>
<li>How complete the bin/MAG is. A set of lineage specific genes should all be found in a genome belonging to the lineage (ubiquitous).
<ul>
<li>What percentage of these lineage specific genes are present in the MAG?</li>
<li>&gt;95% is very good</li>
<li>&gt;80% is good</li>
<li>&gt;70% is ok</li>
<li>&lt;70% is poor to poorer</li>
</ul></li>
<li>How contaminated the bin/MAG is.
<ul>
<li>Only one copy of each gene should be present (single-copy).</li>
<li>Are there any markers for other lineages present?</li>
</ul></li>
</ul>
<div id="checkm-conda" class="section level3 hasAnchor" number="17.2.1">
<h3><span class="header-section-number">17.2.1</span> CheckM: Conda<a href="17-Genome_binning.html#checkm-conda" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Due to program version conflicts we will use the <code>checkm</code> conda environment for this section.</p>
<p>Open a new terminal and activate the <code>checkm</code> environment.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb81-1"><a href="17-Genome_binning.html#cb81-1" tabindex="-1"></a><span class="bu">.</span> usecheckm</span></code></pre></div>
<p>Ensure you are in the correct directory.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb82-1"><a href="17-Genome_binning.html#cb82-1" tabindex="-1"></a><span class="bu">cd</span> ~/7-Binning/K1/</span></code></pre></div>
</div>
<div id="checkm-run" class="section level3 hasAnchor" number="17.2.2">
<h3><span class="header-section-number">17.2.2</span> CheckM: run<a href="17-Genome_binning.html#checkm-run" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/play_black.png" width="10%" style="display: block; margin: auto;" /></p>
<p><code>CheckM</code> has many different commands. We will use one of the common workflows it provides called <code>lineage_wf</code>. This carries out five of its commands in a workflow (i.e. the next step uses output from the previous step).</p>
<ol style="list-style-type: decimal">
<li><code>tree</code> - Places bins in the reference genome tree. This reference tree comes with <code>CheckM</code>.</li>
<li><code>tree_qa</code> - Assess the phylogenetic markers found in each bin.</li>
<li><code>lineage_set</code> - Infers lineage-specific marker sets for each bin.</li>
<li><code>analyze</code> - Identifies marker genes in bins.</li>
<li><code>qa</code> - Assesses the bins for contamination and completeness.</li>
</ol>
<p>Run the <code>CheckM</code> command (this will take a while):</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb83-1"><a href="17-Genome_binning.html#cb83-1" tabindex="-1"></a><span class="ex">checkm</span> lineage_wf <span class="dt">\</span></span>
<span id="cb83-2"><a href="17-Genome_binning.html#cb83-2" tabindex="-1"></a>--tab_table <span class="at">-f</span> MAGS_checkm.tsv <span class="dt">\</span></span>
<span id="cb83-3"><a href="17-Genome_binning.html#cb83-3" tabindex="-1"></a>-x fa <span class="dt">\</span></span>
<span id="cb83-4"><a href="17-Genome_binning.html#cb83-4" tabindex="-1"></a>bins/ checkm_output</span></code></pre></div>
<div id="parameters-10" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="17-Genome_binning.html#parameters-10" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/parameter_black.png" width="10%" style="display: block; margin: auto;" /></p>
<ul>
<li><code>--tab_table</code> : Prints results to a tab separated table.</li>
<li><code>-f</code> : File name to print result to (if not specified results will go to stdout).</li>
<li><code>-x</code> : Suffix/extension of bin files. Other files are ignored in the specified bin directory.
<ul>
<li><code>fa</code> is used as our <code>MetaBAT2</code> produce fasta files that end in <code>.fa</code>.</li>
</ul></li>
<li><code>bins</code> : The second last argument is the <code>bin_dir</code>, the directory containing all the bins to be analysed in fasta format.</li>
<li><code>checkm_output</code> : The last argument is the directory to store the output to. This directory should not exist prior to running.</li>
</ul>
</div>
</div>
<div id="checkm-output" class="section level3 hasAnchor" number="17.2.3">
<h3><span class="header-section-number">17.2.3</span> CheckM: output<a href="17-Genome_binning.html#checkm-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/output_file_black.png" width="10%" style="display: block; margin: auto;" /></p>
<p>As we have only used a subset of data the results are not very good. We'll therefore look at premade results.
These premade results were produced using the entire K1 dataset.
First you will need to copy them.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb84-1"><a href="17-Genome_binning.html#cb84-1" tabindex="-1"></a><span class="bu">cd</span> ~/7-Binning</span>
<span id="cb84-2"><a href="17-Genome_binning.html#cb84-2" tabindex="-1"></a><span class="fu">mkdir</span> K1_fullset</span>
<span id="cb84-3"><a href="17-Genome_binning.html#cb84-3" tabindex="-1"></a><span class="bu">cd</span> K1_fullset</span>
<span id="cb84-4"><a href="17-Genome_binning.html#cb84-4" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub39/tea/nsc006/NEOF/Shotgun_metagenomics/binning/K1_fullset/<span class="pp">*</span> .</span></code></pre></div>
<p>Now we can look at the results table that is in your current directory.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb85-1"><a href="17-Genome_binning.html#cb85-1" tabindex="-1"></a><span class="fu">less</span> MAGS_checkm.tsv</span></code></pre></div>
<div class="webex-solution">
<button>
CheckM statistics definitions
</button>
<p>List of statistics definitions from <a href="https://github.com/Ecogenomics/CheckM/wiki/Reported-Statistics">CheckM wiki</a>)</p>
<ul>
<li><strong>bin id</strong>: unique identifier of genome bin (derived from input fasta file)</li>
<li><strong>marker lineage</strong>: indicates the taxonomic rank of the lineage-specific marker set used to estimated genome completeness, contamination, and strain heterogeneity. More detailed information about the placement of a genome within the reference genome tree can be obtained with the tree_qa command. The UID indicates the branch within the reference tree used to infer the marker set applied to estimate the bins quality.</li>
<li><strong># genomes</strong>: number of reference genomes used to infer the lineage-specific marker set</li>
<li><strong>markers</strong>: number of marker genes within the inferred lineage-specific marker set
marker sets: number of co-located marker sets within the inferred lineage-specific marker set</li>
<li><strong>0-5+</strong>: number of times each marker gene is identified</li>
<li><strong>completeness</strong>: estimated completeness of genome as determined from the presence/absence of marker genes and the expected collocalization of these genes (see Methods in the PeerJ preprint for details)</li>
<li><strong>contamination</strong>: estimated contamination of genome as determined by the presence of multi-copy marker genes and the expected collocalization of these genes (see Methods in the PeerJ preprint for details)</li>
<li><strong>strain heterogeneity</strong>: estimated strain heterogeneity as determined from the number of multi-copy marker pairs which exceed a specified amino acid identity threshold (default = 90%). High strain heterogeneity suggests the majority of reported contamination is from one or more closely related organisms (i.e. potentially the same species), while low strain heterogeneity suggests the majority of contamination is from more phylogenetically diverse sources (see Methods in the CheckM manuscript for more details).</li>
<li><strong>genome size</strong>: number of nucleotides (including unknowns specified by N's) in the genome</li>
<li><strong># ambiguous bases</strong>: number of ambiguous (N's) bases in the genome</li>
<li><strong># scaffolds</strong>: number of scaffolds within the genome</li>
<li><strong># contigs</strong>: number of contigs within the genome as determined by splitting scaffolds at any position consisting of more than 10 consecutive ambiguous bases</li>
<li><strong>N50 (scaffolds)</strong>: N50 statistics as calculated over all scaffolds</li>
<li><strong>N50 (contigs)</strong>: N50 statistics as calculated over all contigs</li>
<li><strong>longest scaffold</strong>: the longest scaffold within the genome</li>
<li><strong>longest contig</strong>: the longest contig within the genome</li>
<li><strong>GC</strong>: number of G/C nucleotides relative to all A,C,G, and T nucleotides in the genome</li>
<li><strong>coding density</strong>: the number of nucleotides within a coding sequence (CDS) relative to all nucleotides in the genome</li>
<li><strong>translation table</strong>: indicates which genetic code was used to translate nucleotides into amino acids</li>
<li><strong># predicted genes</strong>: number of predicted coding sequences (CDS) within the genome as determined using Prodigal</li>
</ul>
</div>
<p>One quick way to calculate the overall quality of a bin is with the following equation:</p>
<p><span class="math display">\[
q = comp - (5 * cont)
\]</span>
Where:</p>
<ul>
<li><strong>q</strong> = Overall quality</li>
<li><strong>comp</strong> = Completeness</li>
<li><strong>cont</strong> = Contamination</li>
</ul>
<p>A score of at least 70-80% would be the aim, with a maximum/perfect value being 100% (100% completeness, 0% contamination). Therefore let us calculate this for the bins with some bash and <code>awk</code> scripting.</p>
<p>First it is good to make a copy of the file in case we make a mistake and want to start over. Additionally, we will make the copy a comma separated file. I find these easier to edit as typing comma characters (<code>,</code>) in commands is more reliable than tab characters (<code>\t</code>).</p>
<p>This is carried out with <code>tr</code> which can translate characters. In the below case we translate/convert the tabs (<code>\t</code>) into commas (<code>,</code>). The converted output is then redirected (<code>&gt;</code>) to the file "MAGS_checkm.csv".</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb86-1"><a href="17-Genome_binning.html#cb86-1" tabindex="-1"></a><span class="fu">cat</span> MAGS_checkm.tsv <span class="kw">|</span> <span class="fu">tr</span> <span class="st">&quot;\t&quot;</span> <span class="st">&quot;,&quot;</span> <span class="op">&gt;</span> MAGS_checkm.csv</span></code></pre></div>
<p>We will create a new file with only the quality information. We'll start by making a file with only a header.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb87-1"><a href="17-Genome_binning.html#cb87-1" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;Quality&quot;</span> <span class="op">&gt;</span> MAGS_quality.csv</span></code></pre></div>
<p>Next is the most complicated command. We will be calculating the Overall quality (see calculation above) for each row except the header row.</p>
<p>We will be using a complicated linux based language called <code>awk</code>. This is very useful as it can carry out calculations on columns or as <code>awk</code> calls them, <strong>fields</strong>.</p>
<p>As this is new and complicated we will build up our command step by step.</p>
<p>First step is to extract the completeness and contamination fields/columns.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb88-1"><a href="17-Genome_binning.html#cb88-1" tabindex="-1"></a><span class="fu">awk</span> <span class="at">-F,</span> <span class="st">&#39;{print $12,$13}&#39;</span> MAGS_checkm.csv</span></code></pre></div>
<ul>
<li><code>-F,</code>: Indicates the input fields are separated by commas (<code>,</code>).</li>
<li><code>''</code>: All the <code>awk</code> options are contained within the quotes.</li>
<li><code>{}</code>: We can supply a function to <code>awk</code> within the braces.</li>
<li><code>print $12,$13</code>: This function instructs <code>awk</code> to print the 12th (completeness) and 13th (contamination) fields. It is common to put commas (<code>,</code>) between fields if printing multiple fields.</li>
<li><code>MAGS_checkm.csv</code>: Our last parameter is the input file. We are not changing the contents of the file, only printing information to screen/stdout.</li>
</ul>
<p>We do not want the header in our calculation so we will add an extra <code>awk</code> option.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb89-1"><a href="17-Genome_binning.html#cb89-1" tabindex="-1"></a><span class="fu">awk</span> <span class="at">-F,</span> <span class="st">&#39;NR&gt;1 {print $12,$13}&#39;</span> MAGS_checkm.csv</span></code></pre></div>
<ul>
<li><code>NR&gt;1</code>: <code>NR</code> stands for number of records. Rows are called records in <code>awk</code>. Therefore <code>NR&gt;1</code> means <code>awk</code> will only carry out the functions on the records numbered greater than 1. I.e. skip row 1, the header row.</li>
</ul>
<p>The <strong>final</strong> step is to carry out the overall quality calculation and append the information to the "MAGS_quality.csv" file.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb90-1"><a href="17-Genome_binning.html#cb90-1" tabindex="-1"></a><span class="fu">awk</span> <span class="at">-F,</span> <span class="st">&#39;NR&gt;1 {print $12 - (5 * $13)}&#39;</span> MAGS_checkm.csv <span class="op">&gt;&gt;</span> MAGS_quality.csv</span></code></pre></div>
<ul>
<li><code>{print $12 - (5 * $13)}</code>: Our new function carries out the overall quality calculation and prints it for each record/row except the first (<code>NR&gt;1</code>).</li>
<li><code>&gt;&gt; MAGS_quality.csv</code>: The printed information is appended (<code>&gt;&gt;</code>) to the file "MAGS_quality.csv". We append because we want to retain the header we added to the file earlier.</li>
</ul>
<p>You can view the file to ensure it worked. The first and second values should be 33.62 and -0.89</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb91-1"><a href="17-Genome_binning.html#cb91-1" tabindex="-1"></a><span class="fu">less</span> MAGS_quality.csv</span></code></pre></div>
<p>The last step is to combine the files "MAGS_checkm.csv" and "MAGS_quality.csv" with the <code>paste</code> command. The <code>-d ","</code> options indicates the merged files will be separated by commas (<code>,</code>), matching the column separation in "MAGS_checkm.csv".</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb92-1"><a href="17-Genome_binning.html#cb92-1" tabindex="-1"></a><span class="fu">paste</span> <span class="at">-d</span> <span class="st">&quot;,&quot;</span> MAGS_checkm.csv MAGS_quality.csv <span class="op">&gt;</span> MAGS_checkm_quality.csv</span></code></pre></div>
</div>
<div id="checkm-mcqs" class="section level3 hasAnchor" number="17.2.4">
<h3><span class="header-section-number">17.2.4</span> CheckM: MCQs<a href="17-Genome_binning.html#checkm-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/question_bubble_blue.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Viewing the file "MAGS_checkm_quality.csv" attempt the below questions.</p>
<p><strong>Tip</strong>: You can use the <code>cut</code> command to look at specific columns. For example:</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb93-1"><a href="17-Genome_binning.html#cb93-1" tabindex="-1"></a><span class="co">#look at the &quot;Bin Id&quot; and &quot;Quality&quot;</span></span>
<span id="cb93-2"><a href="17-Genome_binning.html#cb93-2" tabindex="-1"></a><span class="co">#Convert the printed output&#39;s commas to tabs for readability</span></span>
<span id="cb93-3"><a href="17-Genome_binning.html#cb93-3" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot;,&quot;</span> <span class="at">-f</span> 1,15 MAGS_checkm_quality.csv <span class="kw">|</span> <span class="fu">tr</span> <span class="st">&quot;,&quot;</span> <span class="st">&quot;\t&quot;</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>What lineage was assigned to bin <strong>K1.1</strong>?
<div id="radio_CUZMHOBQIT" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_CUZMHOBQIT" value=""></input> <span><strong>root (UID1)</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CUZMHOBQIT" value="answer"></input> <span><strong>k_Bacteria (UID203)</strong></span></label><label><input type="radio" autocomplete="off" name="radio_CUZMHOBQIT" value=""></input> <span><strong>o_Lachnospiraceae</strong></span></label>
</div></li>
<li>What lineage was assigned to bin <strong>K1.22</strong>?
<div id="radio_IIEIEFZFCP" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_IIEIEFZFCP" value=""></input> <span><strong>root (UID1)</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IIEIEFZFCP" value=""></input> <span><strong>k_Bacteria (UID203)</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IIEIEFZFCP" value="answer"></input> <span><strong>o_Lachnospiraceae</strong></span></label>
</div></li>
<li>What lineage was assigned to bin <strong>K1.8</strong>?
<div id="radio_QAOWCBVSGH" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_QAOWCBVSGH" value="answer"></input> <span><strong>root (UID1)</strong></span></label><label><input type="radio" autocomplete="off" name="radio_QAOWCBVSGH" value=""></input> <span><strong>k_Bacteria (UID203)</strong></span></label><label><input type="radio" autocomplete="off" name="radio_QAOWCBVSGH" value=""></input> <span><strong>o_Lachnospiraceae</strong></span></label>
</div></li>
<li>What is the quality value of K1.1?
<div id="radio_PJZDEIIDQA" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_PJZDEIIDQA" value=""></input> <span><strong>4.17</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PJZDEIIDQA" value="answer"></input> <span><strong>33.62</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PJZDEIIDQA" value=""></input> <span><strong>172</strong></span></label>
</div></li>
<li>What is the completeness value of K1.27?
<div id="radio_HXVFLZSJOM" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_HXVFLZSJOM" value="answer"></input> <span><strong>4.17</strong></span></label><label><input type="radio" autocomplete="off" name="radio_HXVFLZSJOM" value=""></input> <span><strong>33.62</strong></span></label><label><input type="radio" autocomplete="off" name="radio_HXVFLZSJOM" value=""></input> <span><strong>172</strong></span></label>
</div></li>
<li>How many genomes are within the K1.12 bin?
<div id="radio_PGWUZIQLPG" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_PGWUZIQLPG" value=""></input> <span><strong>4.17</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PGWUZIQLPG" value=""></input> <span><strong>33.62</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PGWUZIQLPG" value="answer"></input> <span><strong>172</strong></span></label>
</div></li>
<li>Which bin has the highest quality value (96.38%)?
<div id="radio_IQQRVNIIJX" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_IQQRVNIIJX" value=""></input> <span><strong>K1.20</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IQQRVNIIJX" value=""></input> <span><strong>K1.21</strong></span></label><label><input type="radio" autocomplete="off" name="radio_IQQRVNIIJX" value="answer"></input> <span><strong>K1.22</strong></span></label>
</div></li>
<li>Which bin has the lowest quality value (-300.67%)?
<div id="radio_AOIKFXTHFS" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_AOIKFXTHFS" value="answer"></input> <span><strong>K1.20</strong></span></label><label><input type="radio" autocomplete="off" name="radio_AOIKFXTHFS" value=""></input> <span><strong>K1.21</strong></span></label><label><input type="radio" autocomplete="off" name="radio_AOIKFXTHFS" value=""></input> <span><strong>K1.22</strong></span></label>
</div></li>
<li>Which bin has the highest completeness value (98.27%)?
<div id="radio_YSWYRTSZXH" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_YSWYRTSZXH" value=""></input> <span><strong>K1.20</strong></span></label><label><input type="radio" autocomplete="off" name="radio_YSWYRTSZXH" value="answer"></input> <span><strong>K1.21</strong></span></label><label><input type="radio" autocomplete="off" name="radio_YSWYRTSZXH" value=""></input> <span><strong>K1.22</strong></span></label>
</div></li>
</ol>
</div>
</div>
<div id="binning-summary" class="section level2 hasAnchor" number="17.3">
<h2><span class="header-section-number">17.3</span> Binning summary<a href="17-Genome_binning.html#binning-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_black.png" width="10%" style="display: block; margin: auto;" /></p>
<p>It is always useful to know the quality of your bins so you know which are more reliable than others. With that information you can be more or less certain when concluding your findings.</p>
<p>We have some good quality bins. However, the best bins would only have one genome. Ultimately binning is trying to separate all the genomes from each other. A better metagenome assembly would most likely have led to better binning.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="16_Megahit.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="18-Assembly_functional_annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
