<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 23 Assembly functional annotation | Shotgun Metagenomics</title>
  <meta name="description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 23 Assembly functional annotation | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 23 Assembly functional annotation | Shotgun Metagenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2025-04-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="22-Genome_binning.html"/>
<link rel="next" href="24-Appendix.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Start_conda.html"><a href="04-Start_conda.html"><i class="fa fa-check"></i><b>4</b> Startup &amp; mamba</a></li>
<li class="part"><span><b>Data &amp; QC</b></span></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-summary"><i class="fa fa-check"></i><b>7.5</b> Host removal summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy of reads</b></span></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a></li>
<li class="chapter" data-level="9" data-path="09-Kraken2.html"><a href="09-Kraken2.html"><i class="fa fa-check"></i><b>9</b> Kraken2</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-run"><i class="fa fa-check"></i><b>9.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="9.2" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-output"><i class="fa fa-check"></i><b>9.2</b> Kraken2: output</a></li>
<li class="chapter" data-level="9.3" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken-2-mcqs"><i class="fa fa-check"></i><b>9.3</b> Kraken 2: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Krona.html"><a href="10-Krona.html"><i class="fa fa-check"></i><b>10</b> Krona plot</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Krona.html"><a href="10-Krona.html#krona-run"><i class="fa fa-check"></i><b>10.1</b> Krona: run</a></li>
<li class="chapter" data-level="10.2" data-path="10-Krona.html"><a href="10-Krona.html#krona-visualise"><i class="fa fa-check"></i><b>10.2</b> Krona: visualise</a></li>
<li class="chapter" data-level="10.3" data-path="10-Krona.html"><a href="10-Krona.html#krona-mcqs"><i class="fa fa-check"></i><b>10.3</b> Krona: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Bracken.html"><a href="11-Bracken.html"><i class="fa fa-check"></i><b>11</b> Bracken</a>
<ul>
<li class="chapter" data-level="11.0.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-run"><i class="fa fa-check"></i><b>11.0.1</b> Bracken: run</a></li>
<li class="chapter" data-level="11.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-output"><i class="fa fa-check"></i><b>11.1</b> Bracken: output</a></li>
<li class="chapter" data-level="11.2" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-mcqs"><i class="fa fa-check"></i><b>11.2</b> Bracken: MCQs</a></li>
<li class="chapter" data-level="11.3" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-merging-output"><i class="fa fa-check"></i><b>11.3</b> Bracken: merging output</a></li>
<li class="chapter" data-level="11.4" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-extracting-output"><i class="fa fa-check"></i><b>11.4</b> Bracken: extracting output</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Lefse.html"><a href="12-Lefse.html"><i class="fa fa-check"></i><b>12</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2_lefse_metadata"><i class="fa fa-check"></i><b>12.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="12.2" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-format"><i class="fa fa-check"></i><b>12.2</b> LEfSe: format</a></li>
<li class="chapter" data-level="12.3" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-run"><i class="fa fa-check"></i><b>12.3</b> LEfSe: run</a></li>
<li class="chapter" data-level="12.4" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-visualisation"><i class="fa fa-check"></i><b>12.4</b> LEfSe: visualisation</a></li>
<li class="chapter" data-level="12.5" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-mcqs"><i class="fa fa-check"></i><b>12.5</b> LEfSe: MCQs</a></li>
<li class="chapter" data-level="12.6" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>12.6</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="part"><span><b>Functional profiling of reads</b></span></li>
<li class="chapter" data-level="13" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html"><i class="fa fa-check"></i><b>13</b> Functional profiling</a></li>
<li class="chapter" data-level="14" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html"><i class="fa fa-check"></i><b>14</b> HUMAnN</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#humann-mamba-directories-and-files"><i class="fa fa-check"></i><b>14.1</b> HUMAnN: mamba, directories, and files</a></li>
<li class="chapter" data-level="14.2" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#humann-run"><i class="fa fa-check"></i><b>14.2</b> HUMAnN: run</a></li>
<li class="chapter" data-level="14.3" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#humann-output"><i class="fa fa-check"></i><b>14.3</b> HUMAnN: output</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html"><i class="fa fa-check"></i><b>15</b> Multi sample processing</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#combining-data"><i class="fa fa-check"></i><b>15.1</b> Combining data</a></li>
<li class="chapter" data-level="15.2" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#split-stratified-table"><i class="fa fa-check"></i><b>15.2</b> Split stratified table</a></li>
<li class="chapter" data-level="15.3" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#humannrenormalise"><i class="fa fa-check"></i><b>15.3</b> Renormalising data</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Heatmap.html"><a href="16-Heatmap.html"><i class="fa fa-check"></i><b>16</b> Heatmap</a></li>
<li class="chapter" data-level="17" data-path="17-LEfSe_humann.html"><a href="17-LEfSe_humann.html"><i class="fa fa-check"></i><b>17</b> LEfSe</a></li>
<li class="chapter" data-level="18" data-path="18-HUMAnN_task.html"><a href="18-HUMAnN_task.html"><i class="fa fa-check"></i><b>18</b> Optional task</a></li>
<li class="part"><span><b>Assembly Approach</b></span></li>
<li class="chapter" data-level="19" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html"><i class="fa fa-check"></i><b>19</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="19.1" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html#metagenome-assembly-mamba"><i class="fa fa-check"></i><b>19.1</b> Metagenome assembly: Mamba</a></li>
<li class="chapter" data-level="19.2" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>19.2</b> A primer on short read assembly</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html"><i class="fa fa-check"></i><b>20</b> Stitching read pairs</a>
<ul>
<li class="chapter" data-level="20.1" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html#flash-run"><i class="fa fa-check"></i><b>20.1</b> FLASH: run</a></li>
<li class="chapter" data-level="20.2" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html#flash-output"><i class="fa fa-check"></i><b>20.2</b> FLASH: Output</a></li>
<li class="chapter" data-level="20.3" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html#flash-mcqs"><i class="fa fa-check"></i><b>20.3</b> FLASH: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="21_Megahit.html"><a href="21_Megahit.html"><i class="fa fa-check"></i><b>21</b> Megahit Assembly</a>
<ul>
<li class="chapter" data-level="21.1" data-path="21_Megahit.html"><a href="21_Megahit.html#megahit"><i class="fa fa-check"></i><b>21.1</b> MEGAHIT</a></li>
<li class="chapter" data-level="21.2" data-path="21_Megahit.html"><a href="21_Megahit.html#quast"><i class="fa fa-check"></i><b>21.2</b> QUAST</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="21_Megahit.html"><a href="21_Megahit.html#quast-run"><i class="fa fa-check"></i><b>21.2.1</b> QUAST: run</a></li>
<li class="chapter" data-level="21.2.2" data-path="21_Megahit.html"><a href="21_Megahit.html#quast-visualise"><i class="fa fa-check"></i><b>21.2.2</b> QUAST: visualise</a></li>
<li class="chapter" data-level="21.2.3" data-path="21_Megahit.html"><a href="21_Megahit.html#quast-mcqs"><i class="fa fa-check"></i><b>21.2.3</b> QUAST: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="21_Megahit.html"><a href="21_Megahit.html#metagenome-assembly-summary"><i class="fa fa-check"></i><b>21.3</b> Metagenome assembly summary</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html"><i class="fa fa-check"></i><b>22</b> Genome binning</a>
<ul>
<li class="chapter" data-level="22.1" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>22.1</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="22.1.1" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2-depth-calculation"><i class="fa fa-check"></i><b>22.1.1</b> MetaBAT2: depth calculation</a></li>
<li class="chapter" data-level="22.1.2" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2-run"><i class="fa fa-check"></i><b>22.1.2</b> MetaBAT2: run</a></li>
<li class="chapter" data-level="22.1.3" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2-output"><i class="fa fa-check"></i><b>22.1.3</b> MetaBAT2: output</a></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#checkm"><i class="fa fa-check"></i><b>22.2</b> CheckM</a>
<ul>
<li class="chapter" data-level="22.2.1" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#checkm-mamba"><i class="fa fa-check"></i><b>22.2.1</b> CheckM: Mamba</a></li>
<li class="chapter" data-level="22.2.2" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#checkm-run"><i class="fa fa-check"></i><b>22.2.2</b> CheckM: run</a></li>
<li class="chapter" data-level="22.2.3" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#checkm-output"><i class="fa fa-check"></i><b>22.2.3</b> CheckM: output</a></li>
<li class="chapter" data-level="22.2.4" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#checkm-quality-score"><i class="fa fa-check"></i><b>22.2.4</b> CheckM: Quality score</a></li>
<li class="chapter" data-level="22.2.5" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#checkm-mcqs"><i class="fa fa-check"></i><b>22.2.5</b> CheckM: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="22.3" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#binning-summary"><i class="fa fa-check"></i><b>22.3</b> Binning summary</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="23-Assembly_functional_annotation.html"><a href="23-Assembly_functional_annotation.html"><i class="fa fa-check"></i><b>23</b> Assembly functional annotation</a>
<ul>
<li class="chapter" data-level="23.1" data-path="23-Assembly_functional_annotation.html"><a href="23-Assembly_functional_annotation.html#taxonomic-annotation"><i class="fa fa-check"></i><b>23.1</b> Taxonomic annotation</a></li>
<li class="chapter" data-level="23.2" data-path="23-Assembly_functional_annotation.html"><a href="23-Assembly_functional_annotation.html#bakta"><i class="fa fa-check"></i><b>23.2</b> Bakta</a>
<ul>
<li class="chapter" data-level="23.2.1" data-path="23-Assembly_functional_annotation.html"><a href="23-Assembly_functional_annotation.html#bakta-run"><i class="fa fa-check"></i><b>23.2.1</b> Bakta: run</a></li>
<li class="chapter" data-level="23.2.2" data-path="23-Assembly_functional_annotation.html"><a href="23-Assembly_functional_annotation.html#bakta-premade-results"><i class="fa fa-check"></i><b>23.2.2</b> Bakta: premade results</a></li>
<li class="chapter" data-level="23.2.3" data-path="23-Assembly_functional_annotation.html"><a href="23-Assembly_functional_annotation.html#bakta-output"><i class="fa fa-check"></i><b>23.2.3</b> Bakta: output</a></li>
</ul></li>
<li class="chapter" data-level="23.3" data-path="23-Assembly_functional_annotation.html"><a href="23-Assembly_functional_annotation.html#minpath"><i class="fa fa-check"></i><b>23.3</b> MinPath</a>
<ul>
<li class="chapter" data-level="23.3.1" data-path="23-Assembly_functional_annotation.html"><a href="23-Assembly_functional_annotation.html#minpath-ec-extraction"><i class="fa fa-check"></i><b>23.3.1</b> MinPath: EC extraction</a></li>
<li class="chapter" data-level="23.3.2" data-path="23-Assembly_functional_annotation.html"><a href="23-Assembly_functional_annotation.html#minpathrun"><i class="fa fa-check"></i><b>23.3.2</b> MinPath: run</a></li>
<li class="chapter" data-level="23.3.3" data-path="23-Assembly_functional_annotation.html"><a href="23-Assembly_functional_annotation.html#minpath-output"><i class="fa fa-check"></i><b>23.3.3</b> MinPath: output</a></li>
<li class="chapter" data-level="23.3.4" data-path="23-Assembly_functional_annotation.html"><a href="23-Assembly_functional_annotation.html#keggs"><i class="fa fa-check"></i><b>23.3.4</b> KEGGs</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="24-Appendix.html"><a href="24-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="24-Appendix.html"><a href="24-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="24-Appendix.html"><a href="24-Appendix.html#next-steps"><i class="fa fa-check"></i><b>B</b> Next steps</a></li>
<li class="chapter" data-level="C" data-path="24-Appendix.html"><a href="24-Appendix.html#manuals"><i class="fa fa-check"></i><b>C</b> Manuals</a></li>
<li class="chapter" data-level="D" data-path="24-Appendix.html"><a href="24-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>D</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="funcanno" class="section level1 hasAnchor" number="23">
<h1><span class="header-section-number">Chapter 23</span> Assembly functional annotation<a href="23-Assembly_functional_annotation.html#funcanno" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/pathways.png" style="width:200px; background:white; border-radius:5px" />
</center>
<div id="taxonomic-annotation" class="section level2 hasAnchor" number="23.1">
<h2><span class="header-section-number">23.1</span> Taxonomic annotation<a href="23-Assembly_functional_annotation.html#taxonomic-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/classification.png" style="width:150px; background:white; border-radius:5px" />
</center>
<p>Taxonomic annotation of bins can be carried out with <code>Kraken2</code>. As we have already done this for the reads and taxonomic results from read and assembly approaches have similar performances we will not cover it here. Instead we will move straight onto functional annotation with <a href="https://github.com/oschwengers/bakta"><code>Bakta</code></a>.</p>
<p><strong>Conda Environment</strong>: Go back to your shotgun_meta terminal (or create a new one and use <code>. useshotgun</code>).</p>
</div>
<div id="bakta" class="section level2 hasAnchor" number="23.2">
<h2><span class="header-section-number">23.2</span> Bakta<a href="23-Assembly_functional_annotation.html#bakta" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/bakta_logo.png" style="border-radius: 15px; width: 250px; border: 5px solid white" />
</center>
<p>We will carry out <code>Bakta</code> functional annotation. <code>Bakta</code> can annotate bacterial genomes and plasmids from both isolates and MAGs.</p>
<p>Make a new directory and move into it.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb93-1"><a href="23-Assembly_functional_annotation.html#cb93-1" tabindex="-1"></a><span class="fu">mkdir</span> ~/8-Annotation</span>
<span id="cb93-2"><a href="23-Assembly_functional_annotation.html#cb93-2" tabindex="-1"></a><span class="bu">cd</span> ~/8-Annotation</span></code></pre></div>
<div id="bakta-run" class="section level3 hasAnchor" number="23.2.1">
<h3><span class="header-section-number">23.2.1</span> Bakta: run<a href="23-Assembly_functional_annotation.html#bakta-run" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/play_blue.png" style="width:100px" />
</center>
<p>Now we can annotate one of the bins.</p>
<p>The below will take a long time to run (&gt;1 hour). Instead of running it skip onto the next section to copy pre-made output to continue with. This command is here so you know what to run in your own future analyses.</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb94-1"><a href="23-Assembly_functional_annotation.html#cb94-1" tabindex="-1"></a><span class="ex">bakta</span> <span class="dt">\</span></span>
<span id="cb94-2"><a href="23-Assembly_functional_annotation.html#cb94-2" tabindex="-1"></a>--db /pub14/tea/nsc206/NEOF/databases/bakta/db/ <span class="dt">\</span></span>
<span id="cb94-3"><a href="23-Assembly_functional_annotation.html#cb94-3" tabindex="-1"></a>-o K1.1 <span class="dt">\</span></span>
<span id="cb94-4"><a href="23-Assembly_functional_annotation.html#cb94-4" tabindex="-1"></a>~/7-Binning/K1_fullset/bins/K1.1.fa</span></code></pre></div>
<div id="parameters-11" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="23-Assembly_functional_annotation.html#parameters-11" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<center>
<img src="figures/parameter_blue.png" style="width:100px" />
</center>
<ul>
<li><code>--db</code>: Location of <code>Bakta</code> database. You will need to install this in your own installation. Instructions are in the <a href="24-Appendix.html#mamba_install">appendix</a>.</li>
<li><code>-o</code>: The output directory. This must not exist before running the command.</li>
<li>The last parameter is the fasta file containing the genome/plasmid you would like annotated.</li>
</ul>
</div>
</div>
<div id="bakta-premade-results" class="section level3 hasAnchor" number="23.2.2">
<h3><span class="header-section-number">23.2.2</span> Bakta: premade results<a href="23-Assembly_functional_annotation.html#bakta-premade-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Link the pre-made results for all the K1 bins.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb95-1"><a href="23-Assembly_functional_annotation.html#cb95-1" tabindex="-1"></a><span class="co">#Link all data</span></span>
<span id="cb95-2"><a href="23-Assembly_functional_annotation.html#cb95-2" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub14/tea/nsc206/NEOF/Shotgun_metagenomics/bakta/K1_fullset/<span class="pp">*</span> .</span></code></pre></div>
</div>
<div id="bakta-output" class="section level3 hasAnchor" number="23.2.3">
<h3><span class="header-section-number">23.2.3</span> Bakta: output<a href="23-Assembly_functional_annotation.html#bakta-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/output_file_blue.png" style="width:100px" />
</center>
<p>List the files in the newly created K1.1 directory. Each of the files has the prefix "K1.1" and contains the following information:</p>
<ul>
<li><strong><prefix>.tsv</strong>: annotations as simple human readble TSV</li>
<li><strong><prefix>.gff3</strong>: annotations &amp; sequences in GFF3 format</li>
<li><strong><prefix>.gbff</strong>: annotations &amp; sequences in (multi) GenBank format</li>
<li><strong><prefix>.embl</strong>: annotations &amp; sequences in (multi) EMBL format</li>
<li><strong><prefix>.fna</strong>: replicon/contig DNA sequences as FASTA</li>
<li><strong><prefix>.ffn</strong>: feature nucleotide sequences as FASTA</li>
<li><strong><prefix>.faa</strong>: CDS/sORF amino acid sequences as FASTA</li>
<li><strong><prefix>.hypotheticals.tsv</strong>: further information on hypothetical protein CDS as simple human readble tab separated values</li>
<li><strong><prefix>.hypotheticals.faa</strong>: hypothetical protein CDS amino acid sequences as FASTA</li>
<li><strong><prefix>.json</strong>: all (internal) annotation &amp; sequence information as JSON</li>
<li><strong><prefix>.txt</strong>: summary as TXT</li>
<li><strong><prefix>.png</strong>: circular genome annotation plot as PNG
<ul>
<li>These are only useful for complete/near complete circular genomes.</li>
<li>I would suggest looking at <a href="https://github.com/robotoD/GenoVi/wiki/User-guide#tutorials-">GenoVi</a> for circular genome plots.</li>
</ul></li>
<li><strong><prefix>.svg</strong>: circular genome annotation plot as SVG</li>
<li><strong><prefix>.log</strong>: Log file of command.</li>
</ul>
<p>View the summary file for bin K1.1.</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb96-1"><a href="23-Assembly_functional_annotation.html#cb96-1" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> K1.1/K1.1.txt</span></code></pre></div>
<div class="webex-solution">
<button>
Summary fields
</button>
<p>Sequence information</p>
<ul>
<li>Length: Number of bases.</li>
<li>Count: Number of contigs/scaffolds.</li>
<li>GC: GC%.</li>
<li>N50: <a href="21_Megahit.html#quast-visualise">N50</a>.</li>
<li>N ratio: Ratio of N bases to non-N bases.</li>
<li>coding density: Percentage of bases within coding regions.</li>
</ul>
<p>Annotation information.</p>
<ul>
<li>tRNAs: Transfer RNAs.</li>
<li>tmRNAs: Transfer-messenger RNA.</li>
<li>rRNAs: Ribosomal RNAs.</li>
<li>ncRNAs: Non-coding RNAs.</li>
<li>ncRNA regions: Non-coding RNA regions.</li>
<li>CRISPR arrays: CRISPR arrays.</li>
<li>CDSs: Coding sequences.</li>
<li>pseudogenes: Segments of DNA that structurally resembles a gene but is not capable of coding for a protein</li>
<li>hypotheticals: Hypothetical genes, which are predicted solely by computer algorithms, are experimentally uncharacterized genes.</li>
<li>signal peptides: Short peptides (usually 16-30 amino acids long) normally present at the N-terminus of most newly synthesized proteins that are destined toward the secretory pathway.</li>
<li>sORFs: Short open reading frames (&lt;100 amino acids).</li>
<li>gaps: Gaps in the genome assembly.</li>
<li>oriCs: Chromosome replication origin for bacteria.</li>
<li>oriVs: Plasmid replication origin.</li>
<li>oriTs: An origin of transfer (oriT) is a short sequence ranging from 40-500 base pairs in length. It is necessary for the transfer of DNA from a gram-negative bacterial donor to recipient during bacterial conjugation.</li>
</ul>
</div>
<p>View the gff file for bin K1.1.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb97-1"><a href="23-Assembly_functional_annotation.html#cb97-1" tabindex="-1"></a><span class="fu">less</span> K1.1/K1.1.gff3</span></code></pre></div>
<p>The GFF file is a tab delimited file containing annotation information for the features in the assembly/bin. In this case it is a GFF3 file (most curent version of GFF).</p>
<p>There is quite a lot of information contained in each row so instead of listing all the columns here please have a look at the official documentation:</p>
<p><a href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md" class="uri">https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md</a></p>
<div class="webex-solution">
<button>
Loop used to analyse all bins
</button>
<p>The code below is for your future analysis, do not run it now as it will take to long.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb98-1"><a href="23-Assembly_functional_annotation.html#cb98-1" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> ~/7-Biinning/K1_fullset/bins/<span class="pp">*</span>fa <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">f</span> <span class="kw">;</span> <span class="dt">\</span></span>
<span id="cb98-2"><a href="23-Assembly_functional_annotation.html#cb98-2" tabindex="-1"></a><span class="cf">do</span> <span class="va">s</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$f</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/.fa//&quot;</span><span class="va">)</span> <span class="kw">;</span> <span class="bu">echo</span> <span class="va">$s</span> <span class="kw">;</span> <span class="dt">\</span></span>
<span id="cb98-3"><a href="23-Assembly_functional_annotation.html#cb98-3" tabindex="-1"></a><span class="ex">bakta</span> <span class="at">--db</span> /pub14/tea/nsc206/NEOF/databases/bakta/db/ <span class="dt">\</span></span>
<span id="cb98-4"><a href="23-Assembly_functional_annotation.html#cb98-4" tabindex="-1"></a>-o <span class="va">${s}</span> <span class="dt">\</span></span>
<span id="cb98-5"><a href="23-Assembly_functional_annotation.html#cb98-5" tabindex="-1"></a>~/7-Binning/K1_fullset/bins/K1.1.fa</span></code></pre></div>
</div>
<p>A quick thing we can do with these files is to see if any of the bins contain a specific annotation. For example, if we wanted to know if there were any ATP-binding proteins in any of the bins we could carry out the below command.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb99-1"><a href="23-Assembly_functional_annotation.html#cb99-1" tabindex="-1"></a><span class="fu">grep</span> <span class="st">&quot;ATP-binding protein&quot;</span> <span class="pp">*</span>/<span class="pp">*</span>gff3 <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p>We can now view the lines containing "ATP-binding protein" with the start of the line containing the file name the line belongs to.</p>
<p>In your future analyses you can expect these files further with excel, R, or visualisation software like <code>IGV</code> (<a href="https://software.broadinstitute.org/software/igv/GFF" class="uri">https://software.broadinstitute.org/software/igv/GFF</a>).</p>
<p>What if you want to know about pathways?</p>
</div>
</div>
<div id="minpath" class="section level2 hasAnchor" number="23.3">
<h2><span class="header-section-number">23.3</span> MinPath<a href="23-Assembly_functional_annotation.html#minpath" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/pathways_2.png" style="width:150px; background:white;border-radius:5px" />
</center>
<p><code>MinPath</code> can predict <code>MetaCyc</code> metabolic pathways. These pathway are made up of sets of enzymes that catalyse various reactions.</p>
<p>Ensure you start in the <code>~/8-Annotation</code> directory.</p>
<div id="minpath-ec-extraction" class="section level3 hasAnchor" number="23.3.1">
<h3><span class="header-section-number">23.3.1</span> MinPath: EC extraction<a href="23-Assembly_functional_annotation.html#minpath-ec-extraction" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/exctractor.png" style="width:150px; background:white;border-radius:5px" />
</center>
<p>Before we can estimate the pathways we need to extract the EC numbers predictions from the GFF file. EC (Enzyme Commission) numbers are a numerical classification for enzymes based on the reaction they catalyse.</p>
<p>Unless you know the EC scheme well they are generally not helpful by themselves. An example EC number is EC 3.1.1. The numbers represents the group the enzyme belongs to with the first number being the biggest group. From highest to the lowest grouping 3.1.1 represents:</p>
<ul>
<li><strong>3</strong>: Hydrolase enzymes.</li>
<li><strong>3.1</strong>: Hydrolase enzymes that act on ester bonds.</li>
<li><strong>3.1.1</strong>: Carboxylic Ester Hydrolases.</li>
</ul>
<p>With all that information we will extract the EC annotations from the GFF files. First we'll create a directory for the output.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb100-1"><a href="23-Assembly_functional_annotation.html#cb100-1" tabindex="-1"></a><span class="co">#Create EC directory</span></span>
<span id="cb100-2"><a href="23-Assembly_functional_annotation.html#cb100-2" tabindex="-1"></a><span class="fu">mkdir</span> EC</span></code></pre></div>
<p>We will now use a loop with various parts to create EC annotation files.
The input <code>.ec</code> files required for <code>MinPath</code> are tab delimited with two columns:</p>
<ol style="list-style-type: decimal">
<li>Protein/sequence id. E.g. GDGAPA_12670.</li>
<li>EC annotation. E.g. 6.1.1.4.</li>
</ol>
<p><strong>Note</strong>: The lack of <code>\</code> at the end of most of the lines is intentional.
The below is all one command over multiple lines but loops work slightly different and don't need <code>\</code> in certain parts.
Ensure you do press enter at the end of each line.</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb101-1"><a href="23-Assembly_functional_annotation.html#cb101-1" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="pp">*</span>/<span class="pp">*</span>gff3 <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s|.*/||&quot;</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/.gff3//&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">bin</span></span>
<span id="cb101-2"><a href="23-Assembly_functional_annotation.html#cb101-2" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb101-3"><a href="23-Assembly_functional_annotation.html#cb101-3" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${bin}</span>/<span class="va">${bin}</span>.gff3 <span class="kw">|</span> <span class="fu">grep</span> <span class="st">&quot;EC:&quot;</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 9 <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/ID=//&quot;</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb101-4"><a href="23-Assembly_functional_annotation.html#cb101-4" tabindex="-1"></a><span class="fu">sed</span> <span class="st">&quot;s/;.*EC:/\t/&quot;</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/,.*//&quot;</span> <span class="op">&gt;</span> EC/<span class="va">${bin}</span>.ec</span>
<span id="cb101-5"><a href="23-Assembly_functional_annotation.html#cb101-5" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<div id="code-explanation" class="section level4 hasAnchor" number="23.3.1.1">
<h4><span class="header-section-number">23.3.1.1</span> Code explanation<a href="23-Assembly_functional_annotation.html#code-explanation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>That is quite a bit of code. You don't need to understand it as it should always work for <code>Bakta</code> output.
If you are not currently interested you can skip to the <a href="23-Assembly_functional_annotation.html#minpathrun"><code>MinPath</code>: run section</a></p>
<p>If you are interested in how it works we'll break it down with examples for you to run.</p>
<p>The first part lists all our <code>.gff</code> files on one line each (<code>ls -1 *gff</code>).
Next, we remove the directory name (<code>sed "s|.*||"</code>).
Then the suffix <code>.gff</code> is substituted with nothing <code>sed "s/.gff//"</code>.
This gives us the name of each bin (e.g. <code>K1.1</code>, <code>K1.2</code>).</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb102-1"><a href="23-Assembly_functional_annotation.html#cb102-1" tabindex="-1"></a><span class="co">#List all gff files on one (-1) each</span></span>
<span id="cb102-2"><a href="23-Assembly_functional_annotation.html#cb102-2" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="pp">*</span>/<span class="pp">*</span>gff3</span>
<span id="cb102-3"><a href="23-Assembly_functional_annotation.html#cb102-3" tabindex="-1"></a><span class="co">#Remove the directory name</span></span>
<span id="cb102-4"><a href="23-Assembly_functional_annotation.html#cb102-4" tabindex="-1"></a><span class="co">#You can use any character as the divider in sed</span></span>
<span id="cb102-5"><a href="23-Assembly_functional_annotation.html#cb102-5" tabindex="-1"></a><span class="co">#Useful when you want to move slashes from a file name</span></span>
<span id="cb102-6"><a href="23-Assembly_functional_annotation.html#cb102-6" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="pp">*</span>/<span class="pp">*</span>gff3 <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s|.*/||&quot;</span></span>
<span id="cb102-7"><a href="23-Assembly_functional_annotation.html#cb102-7" tabindex="-1"></a><span class="co">#List all the file prefixes (on one line each)</span></span>
<span id="cb102-8"><a href="23-Assembly_functional_annotation.html#cb102-8" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="pp">*</span>/<span class="pp">*</span>gff3 <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s|.*/||&quot;</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/.gff3//&quot;</span></span></code></pre></div>
<p>With the above code we can loop through each file prefix and use the variable <code>bin</code> (arbitrarily chosen) which contains the file prefix.
This is carried out with <code>while read bin</code>.
All the lines between the <code>do</code> (start of the loop) and <code>done</code> (end of loop) are line run in the loop.</p>
<p>Run the loop with an echo command to show we are using the <code>${bin}</code> variable to specify the input and output files.</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb103-1"><a href="23-Assembly_functional_annotation.html#cb103-1" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="pp">*</span>/<span class="pp">*</span>gff3 <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s|.*/||&quot;</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/.gff3//&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">bin</span></span>
<span id="cb103-2"><a href="23-Assembly_functional_annotation.html#cb103-2" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb103-3"><a href="23-Assembly_functional_annotation.html#cb103-3" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;</span><span class="va">${bin}</span><span class="st">/</span><span class="va">${bin}</span><span class="st">.gff3 ../EC/</span><span class="va">${bin}</span><span class="st">.ec&quot;</span></span>
<span id="cb103-4"><a href="23-Assembly_functional_annotation.html#cb103-4" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Now to look at the command within the loop:</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb104-1"><a href="23-Assembly_functional_annotation.html#cb104-1" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${bin}</span>.gff3 <span class="kw">|</span> <span class="fu">grep</span> <span class="st">&quot;EC:&quot;</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 9 <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb104-2"><a href="23-Assembly_functional_annotation.html#cb104-2" tabindex="-1"></a><span class="fu">sed</span> <span class="st">&quot;s/ID=//&quot;</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/;.*;Dbxref=/,/&quot;</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb104-3"><a href="23-Assembly_functional_annotation.html#cb104-3" tabindex="-1"></a><span class="fu">sed</span> <span class="st">&quot;s/,.*EC:/\t/&quot;</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/,.*//&quot;</span> <span class="op">&gt;</span>../EC/<span class="va">${bin}</span>.ec</span></code></pre></div>
<p>A good way to figure out what a workflow is doing, is by building it up step by step.
Run the first part of workflow and then add the next section, run, repeat.
This shows you how each new section is affecting the output.
Additionally, it is always good to run it on one file and <code>head</code> the output so we have a manageable amount of data to look at.</p>
<p>We'll do that with the <code>K1.1.gff3</code> file.</p>
<p><strong>Note</strong>: Remember we are adding <code>head</code> to the end for ease of viewing.</p>
<p><strong>Tips</strong>:</p>
<ul>
<li>Use the up arrow to go back to previously run commands that you can then edit.</li>
<li>Remember the <code>clear</code> command.</li>
</ul>
<div class="sourceCode" id="cb105"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb105-1"><a href="23-Assembly_functional_annotation.html#cb105-1" tabindex="-1"></a><span class="co">#Grab every line containing &quot;EC:&quot;</span></span>
<span id="cb105-2"><a href="23-Assembly_functional_annotation.html#cb105-2" tabindex="-1"></a><span class="fu">cat</span> K1.1/K1.1.gff3 <span class="kw">|</span> <span class="fu">grep</span> <span class="st">&quot;EC:&quot;</span> <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<div class="sourceCode" id="cb106"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb106-1"><a href="23-Assembly_functional_annotation.html#cb106-1" tabindex="-1"></a><span class="co">#Cut out the 9th column/field (-f) (i.e. only keep the 9th column)</span></span>
<span id="cb106-2"><a href="23-Assembly_functional_annotation.html#cb106-2" tabindex="-1"></a><span class="co">#This is the attributes field in GFF3</span></span>
<span id="cb106-3"><a href="23-Assembly_functional_annotation.html#cb106-3" tabindex="-1"></a><span class="co">#This contains a plethora of information including the EC annotation if present</span></span>
<span id="cb106-4"><a href="23-Assembly_functional_annotation.html#cb106-4" tabindex="-1"></a><span class="co">#cut uses tabs as the default column/field delimiter</span></span>
<span id="cb106-5"><a href="23-Assembly_functional_annotation.html#cb106-5" tabindex="-1"></a><span class="fu">cat</span> K1.1/K1.1.gff3 <span class="kw">|</span> <span class="fu">grep</span> <span class="st">&quot;EC:&quot;</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 9 <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<div class="sourceCode" id="cb107"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb107-1"><a href="23-Assembly_functional_annotation.html#cb107-1" tabindex="-1"></a><span class="co">#The gff3 attributes field starts with the ID</span></span>
<span id="cb107-2"><a href="23-Assembly_functional_annotation.html#cb107-2" tabindex="-1"></a><span class="co">#We want to keep this but remove the &quot;ID=&quot; part</span></span>
<span id="cb107-3"><a href="23-Assembly_functional_annotation.html#cb107-3" tabindex="-1"></a><span class="fu">cat</span> K1.1/K1.1.gff3 <span class="kw">|</span> <span class="fu">grep</span> <span class="st">&quot;EC:&quot;</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 9 <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/ID=//&quot;</span> <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<div class="sourceCode" id="cb108"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb108-1"><a href="23-Assembly_functional_annotation.html#cb108-1" tabindex="-1"></a><span class="co">#We don&#39;t want any of the info between the ID and the EC number</span></span>
<span id="cb108-2"><a href="23-Assembly_functional_annotation.html#cb108-2" tabindex="-1"></a><span class="co">#Therefore we want to remove everything (.*) between</span></span>
<span id="cb108-3"><a href="23-Assembly_functional_annotation.html#cb108-3" tabindex="-1"></a><span class="co"># the first &quot;;&quot; (at the end of the ID info)</span></span>
<span id="cb108-4"><a href="23-Assembly_functional_annotation.html#cb108-4" tabindex="-1"></a><span class="co"># and &quot;EC=&quot;</span></span>
<span id="cb108-5"><a href="23-Assembly_functional_annotation.html#cb108-5" tabindex="-1"></a><span class="co">#We&#39;ll replace this with a \t to seprarate the ID and EC</span></span>
<span id="cb108-6"><a href="23-Assembly_functional_annotation.html#cb108-6" tabindex="-1"></a><span class="co"># columns with a tab (required by MinPath)</span></span>
<span id="cb108-7"><a href="23-Assembly_functional_annotation.html#cb108-7" tabindex="-1"></a><span class="fu">cat</span> K1.1/K1.1.gff3 <span class="kw">|</span> <span class="fu">grep</span> <span class="st">&quot;EC:&quot;</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 9 <span class="kw">|</span>  <span class="fu">sed</span> <span class="st">&quot;s/ID=//&quot;</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb108-8"><a href="23-Assembly_functional_annotation.html#cb108-8" tabindex="-1"></a><span class="fu">sed</span> <span class="st">&quot;s/;.*EC:/\t/&quot;</span> <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<div class="sourceCode" id="cb109"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb109-1"><a href="23-Assembly_functional_annotation.html#cb109-1" tabindex="-1"></a><span class="co">#Finally remove all the info after the EC number</span></span>
<span id="cb109-2"><a href="23-Assembly_functional_annotation.html#cb109-2" tabindex="-1"></a><span class="co">#This info will be after the last ,</span></span>
<span id="cb109-3"><a href="23-Assembly_functional_annotation.html#cb109-3" tabindex="-1"></a><span class="fu">cat</span> K1.1/K1.1.gff3 <span class="kw">|</span> <span class="fu">grep</span> <span class="st">&quot;EC:&quot;</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 9 <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/ID=//&quot;</span> <span class="dt">\</span></span>
<span id="cb109-4"><a href="23-Assembly_functional_annotation.html#cb109-4" tabindex="-1"></a>sed <span class="st">&quot;s/;.*EC:/\t/&quot;</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/,.*//&quot;</span> <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<p>In the looped command, output is written into files: <code>&gt; EC/${bin}.ec</code>.</p>
</div>
</div>
<div id="minpathrun" class="section level3 hasAnchor" number="23.3.2">
<h3><span class="header-section-number">23.3.2</span> MinPath: run<a href="23-Assembly_functional_annotation.html#minpathrun" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/play_green.png" style="width:100px" />
</center>
<p>With our <code>.ec</code> files we can create our MetaCyc predictions.</p>
<p>First we'll change directory into the EC directory. Then create an output directory for the MetaCyc predictions.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb110-1"><a href="23-Assembly_functional_annotation.html#cb110-1" tabindex="-1"></a><span class="bu">cd</span> ./EC</span>
<span id="cb110-2"><a href="23-Assembly_functional_annotation.html#cb110-2" tabindex="-1"></a><span class="fu">mkdir</span> ../MetaCyc</span></code></pre></div>
<p>Now we can loop through the file suffixes to run MinPath.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb111-1"><a href="23-Assembly_functional_annotation.html#cb111-1" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="pp">*</span>ec <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/.ec//&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">bin</span></span>
<span id="cb111-2"><a href="23-Assembly_functional_annotation.html#cb111-2" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb111-3"><a href="23-Assembly_functional_annotation.html#cb111-3" tabindex="-1"></a><span class="ex">python</span> /pub14/tea/nsc206/git_installs/Minpath/MinPath.py <span class="dt">\</span></span>
<span id="cb111-4"><a href="23-Assembly_functional_annotation.html#cb111-4" tabindex="-1"></a>-ec <span class="va">${bin}</span>.ec <span class="dt">\</span></span>
<span id="cb111-5"><a href="23-Assembly_functional_annotation.html#cb111-5" tabindex="-1"></a>-report ../MetaCyc/<span class="va">${bin}</span>.minpath</span>
<span id="cb111-6"><a href="23-Assembly_functional_annotation.html#cb111-6" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>A lot of output will be printed to screen but this can be ignored unless you see warnings.</p>
</div>
<div id="minpath-output" class="section level3 hasAnchor" number="23.3.3">
<h3><span class="header-section-number">23.3.3</span> MinPath: output<a href="23-Assembly_functional_annotation.html#minpath-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/output_file_green.png" style="width:100px" />
</center>
<p>First, change directory into the <code>MetaCyc</code> directory.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb112-1"><a href="23-Assembly_functional_annotation.html#cb112-1" tabindex="-1"></a><span class="co">#Change directory</span></span>
<span id="cb112-2"><a href="23-Assembly_functional_annotation.html#cb112-2" tabindex="-1"></a><span class="bu">cd</span> ../MetaCyc</span>
<span id="cb112-3"><a href="23-Assembly_functional_annotation.html#cb112-3" tabindex="-1"></a><span class="co">#List contents</span></span>
<span id="cb112-4"><a href="23-Assembly_functional_annotation.html#cb112-4" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>From the <code>CheckM</code> results we found that bin K1.22 was very good with a quality score &gt;96%. We will therefore have a look at its output.</p>
<p>Have a look at the <code>.minpath</code> <code>-report</code> file for K1.22.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb113-1"><a href="23-Assembly_functional_annotation.html#cb113-1" tabindex="-1"></a><span class="fu">less</span> K1.22.minpath</span></code></pre></div>
<p>The file contains the following columns</p>
<ol style="list-style-type: decimal">
<li><strong>Pathway ID</strong></li>
<li><strong>Pathway reconstruction</strong>: Only available for genomes annotated in MetaCyc database.</li>
<li><strong>Naive</strong>: Indicates if pathway was reconstructed by the naive mapping approach (1) or not (0).</li>
<li><strong>Minpath</strong>: Indicates if the pathway was kept (1) or removed (0) by <code>MinPath</code>.</li>
<li><strong>Fam0</strong>: The number of families involved in the pathway.</li>
<li><strong>Fam-found</strong>: Number of families in pathway that were annotated/found.</li>
<li><strong>Name</strong>: Description of pathway.</li>
</ol>
<p>Quit (<code>q</code>) <code>less</code> when you are happy.</p>
<p>There are some quick things we can do in bash with these files.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb114-1"><a href="23-Assembly_functional_annotation.html#cb114-1" tabindex="-1"></a><span class="co">#Count number of pathways found in each bin with word count</span></span>
<span id="cb114-2"><a href="23-Assembly_functional_annotation.html#cb114-2" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> <span class="pp">*</span>minpath</span>
<span id="cb114-3"><a href="23-Assembly_functional_annotation.html#cb114-3" tabindex="-1"></a><span class="co">#Grab every line with &quot;PWY-6972&quot; from every file</span></span>
<span id="cb114-4"><a href="23-Assembly_functional_annotation.html#cb114-4" tabindex="-1"></a><span class="fu">grep</span> <span class="st">&quot;PWY-6972&quot;</span> <span class="pp">*</span>minpath <span class="kw">|</span> <span class="fu">less</span></span></code></pre></div>
</div>
<div id="keggs" class="section level3 hasAnchor" number="23.3.4">
<h3><span class="header-section-number">23.3.4</span> KEGGs<a href="23-Assembly_functional_annotation.html#keggs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/KEGG.png" style="width:100px" />
</center>
<p>You can also get KEGG information with <code>MinPath</code>.
The code:</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb115-1"><a href="23-Assembly_functional_annotation.html#cb115-1" tabindex="-1"></a><span class="co">#Change to correct directory</span></span>
<span id="cb115-2"><a href="23-Assembly_functional_annotation.html#cb115-2" tabindex="-1"></a><span class="bu">cd</span> ~/8-Annotation</span>
<span id="cb115-3"><a href="23-Assembly_functional_annotation.html#cb115-3" tabindex="-1"></a><span class="co">#Make directory for KEGGs</span></span>
<span id="cb115-4"><a href="23-Assembly_functional_annotation.html#cb115-4" tabindex="-1"></a><span class="fu">mkdir</span> KEGG</span>
<span id="cb115-5"><a href="23-Assembly_functional_annotation.html#cb115-5" tabindex="-1"></a><span class="co">#KEGG extractions</span></span>
<span id="cb115-6"><a href="23-Assembly_functional_annotation.html#cb115-6" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="pp">*</span>/<span class="pp">*</span>gff3 <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s|.*/||&quot;</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/.gff3//&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">bin</span></span>
<span id="cb115-7"><a href="23-Assembly_functional_annotation.html#cb115-7" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb115-8"><a href="23-Assembly_functional_annotation.html#cb115-8" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${bin}</span>/<span class="va">${bin}</span>.gff3 <span class="kw">|</span> <span class="fu">grep</span> <span class="st">&quot;KEGG:&quot;</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 9 <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/ID=//&quot;</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb115-9"><a href="23-Assembly_functional_annotation.html#cb115-9" tabindex="-1"></a><span class="fu">sed</span> <span class="st">&quot;s/;.*KEGG:/\t/&quot;</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/,.*//&quot;</span> <span class="op">&gt;</span> KEGG/<span class="va">${bin}</span>.kegg</span>
<span id="cb115-10"><a href="23-Assembly_functional_annotation.html#cb115-10" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb115-11"><a href="23-Assembly_functional_annotation.html#cb115-11" tabindex="-1"></a><span class="co">#Make directory for KEGG minpath output</span></span>
<span id="cb115-12"><a href="23-Assembly_functional_annotation.html#cb115-12" tabindex="-1"></a><span class="fu">mkdir</span> KEGG_minpath</span>
<span id="cb115-13"><a href="23-Assembly_functional_annotation.html#cb115-13" tabindex="-1"></a><span class="co">#Change directory to KEGG</span></span>
<span id="cb115-14"><a href="23-Assembly_functional_annotation.html#cb115-14" tabindex="-1"></a><span class="bu">cd</span> KEGG</span>
<span id="cb115-15"><a href="23-Assembly_functional_annotation.html#cb115-15" tabindex="-1"></a><span class="co">#Run MinPath</span></span>
<span id="cb115-16"><a href="23-Assembly_functional_annotation.html#cb115-16" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="pp">*</span>kegg <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/.kegg//&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">bin</span></span>
<span id="cb115-17"><a href="23-Assembly_functional_annotation.html#cb115-17" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb115-18"><a href="23-Assembly_functional_annotation.html#cb115-18" tabindex="-1"></a><span class="ex">python</span> /pub14/tea/nsc206/git_installs/Minpath/MinPath.py <span class="dt">\</span></span>
<span id="cb115-19"><a href="23-Assembly_functional_annotation.html#cb115-19" tabindex="-1"></a>-ko <span class="va">${bin}</span>.kegg <span class="dt">\</span></span>
<span id="cb115-20"><a href="23-Assembly_functional_annotation.html#cb115-20" tabindex="-1"></a>-report ../KEGG_minpath/<span class="va">${bin}</span>.minpath</span>
<span id="cb115-21"><a href="23-Assembly_functional_annotation.html#cb115-21" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>With these files you can then investigate what bins have which pathways. Additionally, with more samples analysed you can determine which samples have which pathways present.</p>

</div>
</div>
</div>



<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="22-Genome_binning.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="24-Appendix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
