<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 23 Depth calculation | Shotgun Metagenomics</title>
  <meta name="description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 23 Depth calculation | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 23 Depth calculation | Shotgun Metagenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2026-02-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="22-Genome_binning.html"/>
<link rel="next" href="24-MetaBAT2.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<link href="libs/htmltools-fill-0.5.9/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/tippyjs-3.2.0/themes/google.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/light-border.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/light.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/translucent.css" rel="stylesheet" />
<script src="libs/tippyjs-3.2.0/tippy.all.min.js"></script>
<script src="libs/tippy-binding-0.1.0/tippy.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Start_conda.html"><a href="04-Start_conda.html"><i class="fa fa-check"></i><b>4</b> Startup &amp; mamba</a></li>
<li class="part"><span><b>Data &amp; QC</b></span></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-summary"><i class="fa fa-check"></i><b>7.5</b> Host removal summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy of reads</b></span></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a></li>
<li class="chapter" data-level="9" data-path="09-Kraken2.html"><a href="09-Kraken2.html"><i class="fa fa-check"></i><b>9</b> Kraken2</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Kraken2.html"><a href="09-Kraken2.html#set-kraken2-database"><i class="fa fa-check"></i><b>9.1</b> Set Kraken2 database</a></li>
<li class="chapter" data-level="9.2" data-path="09-Kraken2.html"><a href="09-Kraken2.html#classify-reads"><i class="fa fa-check"></i><b>9.2</b> Classify reads</a></li>
<li class="chapter" data-level="9.3" data-path="09-Kraken2.html"><a href="09-Kraken2.html#output-and-report-files"><i class="fa fa-check"></i><b>9.3</b> Output and report files</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="09-Kraken2.html"><a href="09-Kraken2.html#output-file"><i class="fa fa-check"></i><b>9.3.1</b> Output file</a></li>
<li class="chapter" data-level="9.3.2" data-path="09-Kraken2.html"><a href="09-Kraken2.html#report-file"><i class="fa fa-check"></i><b>9.3.2</b> Report file</a></li>
<li class="chapter" data-level="9.3.3" data-path="09-Kraken2.html"><a href="09-Kraken2.html#screen-output"><i class="fa fa-check"></i><b>9.3.3</b> Screen output</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="09-Kraken2.html"><a href="09-Kraken2.html#confidence-threshold"><i class="fa fa-check"></i><b>9.4</b> Confidence threshold</a></li>
<li class="chapter" data-level="9.5" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-task"><i class="fa fa-check"></i><b>9.5</b> Kraken2 task</a></li>
<li class="chapter" data-level="9.6" data-path="09-Kraken2.html"><a href="09-Kraken2.html#mcqs"><i class="fa fa-check"></i><b>9.6</b> MCQs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Krona.html"><a href="10-Krona.html"><i class="fa fa-check"></i><b>10</b> Krona</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Krona.html"><a href="10-Krona.html#import-taxonomy"><i class="fa fa-check"></i><b>10.1</b> Import taxonomy</a></li>
<li class="chapter" data-level="10.2" data-path="10-Krona.html"><a href="10-Krona.html#tiered-piecharts"><i class="fa fa-check"></i><b>10.2</b> Tiered piecharts</a></li>
<li class="chapter" data-level="10.3" data-path="10-Krona.html"><a href="10-Krona.html#mcqs-1"><i class="fa fa-check"></i><b>10.3</b> MCQs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Bracken.html"><a href="11-Bracken.html"><i class="fa fa-check"></i><b>11</b> Bracken</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-abundance-reestimation"><i class="fa fa-check"></i><b>11.1</b> Bracken abundance reestimation</a></li>
<li class="chapter" data-level="11.2" data-path="11-Bracken.html"><a href="11-Bracken.html#output-and-report-file"><i class="fa fa-check"></i><b>11.2</b> Output and report file</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="11-Bracken.html"><a href="11-Bracken.html#output"><i class="fa fa-check"></i><b>11.2.1</b> Output</a></li>
<li class="chapter" data-level="11.2.2" data-path="11-Bracken.html"><a href="11-Bracken.html#report-file-1"><i class="fa fa-check"></i><b>11.2.2</b> Report file</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-task"><i class="fa fa-check"></i><b>11.3</b> Bracken task</a></li>
<li class="chapter" data-level="11.4" data-path="11-Bracken.html"><a href="11-Bracken.html#mcqs-2"><i class="fa fa-check"></i><b>11.4</b> MCQs</a></li>
<li class="chapter" data-level="11.5" data-path="11-Bracken.html"><a href="11-Bracken.html#merging-output"><i class="fa fa-check"></i><b>11.5</b> Merging output</a></li>
<li class="chapter" data-level="11.6" data-path="11-Bracken.html"><a href="11-Bracken.html#extracting-output"><i class="fa fa-check"></i><b>11.6</b> Extracting output</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Lefse.html"><a href="12-Lefse.html"><i class="fa fa-check"></i><b>12</b> LEfSe</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2_lefse_metadata"><i class="fa fa-check"></i><b>12.1</b> Add metadata</a></li>
<li class="chapter" data-level="12.2" data-path="12-Lefse.html"><a href="12-Lefse.html#format-file"><i class="fa fa-check"></i><b>12.2</b> Format file</a></li>
<li class="chapter" data-level="12.3" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-run"><i class="fa fa-check"></i><b>12.3</b> LEfSe run</a></li>
<li class="chapter" data-level="12.4" data-path="12-Lefse.html"><a href="12-Lefse.html#out-file"><i class="fa fa-check"></i><b>12.4</b> Out file</a></li>
<li class="chapter" data-level="12.5" data-path="12-Lefse.html"><a href="12-Lefse.html#biomarker-lda-barchart"><i class="fa fa-check"></i><b>12.5</b> Biomarker LDA barchart</a></li>
<li class="chapter" data-level="12.6" data-path="12-Lefse.html"><a href="12-Lefse.html#mcqs-3"><i class="fa fa-check"></i><b>12.6</b> MCQs</a></li>
<li class="chapter" data-level="12.7" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>12.7</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="part"><span><b>Functional profiling of reads</b></span></li>
<li class="chapter" data-level="13" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html"><i class="fa fa-check"></i><b>13</b> Functional profiling</a></li>
<li class="chapter" data-level="14" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html"><i class="fa fa-check"></i><b>14</b> HUMAnN</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#mamba-directories-and-files"><i class="fa fa-check"></i><b>14.1</b> Mamba, directories, and files</a></li>
<li class="chapter" data-level="14.2" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#humann-pipeline"><i class="fa fa-check"></i><b>14.2</b> HUMAnN pipeline</a></li>
<li class="chapter" data-level="14.3" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#output-files"><i class="fa fa-check"></i><b>14.3</b> Output files</a></li>
<li class="chapter" data-level="14.4" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#intermediarytemp-files"><i class="fa fa-check"></i><b>14.4</b> Intermediary/temp files</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#storage-check"><i class="fa fa-check"></i><b>14.4.1</b> Storage check</a></li>
<li class="chapter" data-level="14.4.2" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#delete-intermediary-directory"><i class="fa fa-check"></i><b>14.4.2</b> Delete intermediary directory</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html"><i class="fa fa-check"></i><b>15</b> Multi sample processing</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#pre-made-data"><i class="fa fa-check"></i><b>15.1</b> Pre-made data</a></li>
<li class="chapter" data-level="15.2" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#pathway-abundance-file"><i class="fa fa-check"></i><b>15.2</b> Pathway abundance file</a></li>
<li class="chapter" data-level="15.3" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#combining-data"><i class="fa fa-check"></i><b>15.3</b> Combining data</a></li>
<li class="chapter" data-level="15.4" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#split-stratified-table"><i class="fa fa-check"></i><b>15.4</b> Split stratified table</a></li>
<li class="chapter" data-level="15.5" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#humannrenormalise"><i class="fa fa-check"></i><b>15.5</b> Renormalising data</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Heatmap.html"><a href="16-Heatmap.html"><i class="fa fa-check"></i><b>16</b> Heatmap</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Heatmap.html"><a href="16-Heatmap.html#heatmap_file_edit"><i class="fa fa-check"></i><b>16.1</b> File edit</a></li>
<li class="chapter" data-level="16.2" data-path="16-Heatmap.html"><a href="16-Heatmap.html#hclust2"><i class="fa fa-check"></i><b>16.2</b> hclust2</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-LEfSe_humann.html"><a href="17-LEfSe_humann.html"><i class="fa fa-check"></i><b>17</b> LEfSe</a></li>
<li class="chapter" data-level="18" data-path="18-HUMAnN_task.html"><a href="18-HUMAnN_task.html"><i class="fa fa-check"></i><b>18</b> Optional task</a></li>
<li class="part"><span><b>Assembly</b></span></li>
<li class="chapter" data-level="19" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html"><i class="fa fa-check"></i><b>19</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="19.1" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html#mamba"><i class="fa fa-check"></i><b>19.1</b> Mamba</a></li>
<li class="chapter" data-level="19.2" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>19.2</b> A primer on short read assembly</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="20-FLASH.html"><a href="20-FLASH.html"><i class="fa fa-check"></i><b>20</b> FLASH</a>
<ul>
<li class="chapter" data-level="20.1" data-path="20-FLASH.html"><a href="20-FLASH.html#stitching"><i class="fa fa-check"></i><b>20.1</b> Stitching</a></li>
<li class="chapter" data-level="20.2" data-path="20-FLASH.html"><a href="20-FLASH.html#sticthed-fastqs-and-histogram"><i class="fa fa-check"></i><b>20.2</b> Sticthed fastqs and histogram</a></li>
<li class="chapter" data-level="20.3" data-path="20-FLASH.html"><a href="20-FLASH.html#mcqs-5"><i class="fa fa-check"></i><b>20.3</b> MCQs</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="21-MEGAHIT.html"><a href="21-MEGAHIT.html"><i class="fa fa-check"></i><b>21</b> MEGAHIT</a>
<ul>
<li class="chapter" data-level="21.1" data-path="21-MEGAHIT.html"><a href="21-MEGAHIT.html#megahit-1"><i class="fa fa-check"></i><b>21.1</b> MEGAHIT</a></li>
<li class="chapter" data-level="21.2" data-path="21-MEGAHIT.html"><a href="21-MEGAHIT.html#quast"><i class="fa fa-check"></i><b>21.2</b> QUAST</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="21-MEGAHIT.html"><a href="21-MEGAHIT.html#assembly-assessment"><i class="fa fa-check"></i><b>21.2.1</b> Assembly assessment</a></li>
<li class="chapter" data-level="21.2.2" data-path="21-MEGAHIT.html"><a href="21-MEGAHIT.html#quast_report"><i class="fa fa-check"></i><b>21.2.2</b> Report</a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="21-MEGAHIT.html"><a href="21-MEGAHIT.html#mcqs-6"><i class="fa fa-check"></i><b>21.3</b> MCQs</a></li>
<li class="chapter" data-level="21.4" data-path="21-MEGAHIT.html"><a href="21-MEGAHIT.html#metagenome-assembly-summary"><i class="fa fa-check"></i><b>21.4</b> Metagenome assembly summary</a></li>
</ul></li>
<li class="part"><span><b>Binning</b></span></li>
<li class="chapter" data-level="22" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html"><i class="fa fa-check"></i><b>22</b> Metagenome binning</a></li>
<li class="chapter" data-level="23" data-path="23-Depth_calculation.html"><a href="23-Depth_calculation.html"><i class="fa fa-check"></i><b>23</b> Depth calculation</a>
<ul>
<li class="chapter" data-level="23.1" data-path="23-Depth_calculation.html"><a href="23-Depth_calculation.html#index-assembly"><i class="fa fa-check"></i><b>23.1</b> Index assembly</a></li>
<li class="chapter" data-level="23.2" data-path="23-Depth_calculation.html"><a href="23-Depth_calculation.html#alignment-1"><i class="fa fa-check"></i><b>23.2</b> Alignment</a></li>
<li class="chapter" data-level="23.3" data-path="23-Depth_calculation.html"><a href="23-Depth_calculation.html#sam-to-sorted-bam"><i class="fa fa-check"></i><b>23.3</b> SAM to sorted BAM</a></li>
<li class="chapter" data-level="23.4" data-path="23-Depth_calculation.html"><a href="23-Depth_calculation.html#summarise-depths"><i class="fa fa-check"></i><b>23.4</b> Summarise depths</a></li>
<li class="chapter" data-level="23.5" data-path="23-Depth_calculation.html"><a href="23-Depth_calculation.html#view-depths"><i class="fa fa-check"></i><b>23.5</b> View depths</a></li>
<li class="chapter" data-level="23.6" data-path="23-Depth_calculation.html"><a href="23-Depth_calculation.html#summarise-with-r"><i class="fa fa-check"></i><b>23.6</b> Summarise with R</a>
<ul>
<li class="chapter" data-level="23.6.1" data-path="23-Depth_calculation.html"><a href="23-Depth_calculation.html#column-summaries"><i class="fa fa-check"></i><b>23.6.1</b> Column summaries</a></li>
<li class="chapter" data-level="23.6.2" data-path="23-Depth_calculation.html"><a href="23-Depth_calculation.html#filter-then-summarise"><i class="fa fa-check"></i><b>23.6.2</b> Filter then summarise</a></li>
<li class="chapter" data-level="23.6.3" data-path="23-Depth_calculation.html"><a href="23-Depth_calculation.html#quit-r"><i class="fa fa-check"></i><b>23.6.3</b> Quit R</a></li>
</ul></li>
<li class="chapter" data-level="23.7" data-path="23-Depth_calculation.html"><a href="23-Depth_calculation.html#remove-sam-and-bam-files"><i class="fa fa-check"></i><b>23.7</b> Remove SAM and BAM files</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="24-MetaBAT2.html"><a href="24-MetaBAT2.html"><i class="fa fa-check"></i><b>24</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="24.1" data-path="24-MetaBAT2.html"><a href="24-MetaBAT2.html#metabat2-1"><i class="fa fa-check"></i><b>24.1</b> MetaBAT2</a></li>
<li class="chapter" data-level="24.2" data-path="24-MetaBAT2.html"><a href="24-MetaBAT2.html#bins"><i class="fa fa-check"></i><b>24.2</b> Bins</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="25-CoCoPyE.html"><a href="25-CoCoPyE.html"><i class="fa fa-check"></i><b>25</b> CoCoPyE</a>
<ul>
<li class="chapter" data-level="25.1" data-path="25-CoCoPyE.html"><a href="25-CoCoPyE.html#workflow-concept"><i class="fa fa-check"></i><b>25.1</b> Workflow concept</a>
<ul>
<li class="chapter" data-level="25.1.1" data-path="25-CoCoPyE.html"><a href="25-CoCoPyE.html#input-pre-processing"><i class="fa fa-check"></i><b>25.1.1</b> Input Pre-processing</a></li>
<li class="chapter" data-level="25.1.2" data-path="25-CoCoPyE.html"><a href="25-CoCoPyE.html#prediction-stage-i"><i class="fa fa-check"></i><b>25.1.2</b> Prediction: Stage I</a></li>
<li class="chapter" data-level="25.1.3" data-path="25-CoCoPyE.html"><a href="25-CoCoPyE.html#predition-stage-ii"><i class="fa fa-check"></i><b>25.1.3</b> Predition: Stage II</a></li>
</ul></li>
<li class="chapter" data-level="25.2" data-path="25-CoCoPyE.html"><a href="25-CoCoPyE.html#mamba-1"><i class="fa fa-check"></i><b>25.2</b> Mamba</a></li>
<li class="chapter" data-level="25.3" data-path="25-CoCoPyE.html"><a href="25-CoCoPyE.html#predict-bin-quality"><i class="fa fa-check"></i><b>25.3</b> Predict bin quality</a>
<ul>
<li class="chapter" data-level="" data-path="25-CoCoPyE.html"><a href="25-CoCoPyE.html#parameters-10"><i class="fa fa-check"></i>Parameters</a></li>
</ul></li>
<li class="chapter" data-level="25.4" data-path="25-CoCoPyE.html"><a href="25-CoCoPyE.html#stats"><i class="fa fa-check"></i><b>25.4</b> Stats</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="26-Bin_quality_scores.html"><a href="26-Bin_quality_scores.html"><i class="fa fa-check"></i><b>26</b> Bin quality scores</a>
<ul>
<li class="chapter" data-level="26.1" data-path="26-Bin_quality_scores.html"><a href="26-Bin_quality_scores.html#quality-file"><i class="fa fa-check"></i><b>26.1</b> Quality file</a></li>
<li class="chapter" data-level="26.2" data-path="26-Bin_quality_scores.html"><a href="26-Bin_quality_scores.html#calculate-quality-with-awk"><i class="fa fa-check"></i><b>26.2</b> Calculate quality with awk</a>
<ul>
<li class="chapter" data-level="26.2.1" data-path="26-Bin_quality_scores.html"><a href="26-Bin_quality_scores.html#extract-fieldscolumns"><i class="fa fa-check"></i><b>26.2.1</b> Extract fields/columns</a></li>
<li class="chapter" data-level="26.2.2" data-path="26-Bin_quality_scores.html"><a href="26-Bin_quality_scores.html#ignore-header"><i class="fa fa-check"></i><b>26.2.2</b> Ignore header</a></li>
<li class="chapter" data-level="26.2.3" data-path="26-Bin_quality_scores.html"><a href="26-Bin_quality_scores.html#calculate-quality"><i class="fa fa-check"></i><b>26.2.3</b> Calculate quality</a></li>
<li class="chapter" data-level="26.2.4" data-path="26-Bin_quality_scores.html"><a href="26-Bin_quality_scores.html#fix-values"><i class="fa fa-check"></i><b>26.2.4</b> Fix values</a></li>
<li class="chapter" data-level="26.2.5" data-path="26-Bin_quality_scores.html"><a href="26-Bin_quality_scores.html#append-to-quality-file"><i class="fa fa-check"></i><b>26.2.5</b> Append to quality file</a></li>
</ul></li>
<li class="chapter" data-level="26.3" data-path="26-Bin_quality_scores.html"><a href="26-Bin_quality_scores.html#add-quality-to-the-checkm-results-file"><i class="fa fa-check"></i><b>26.3</b> Add quality to the checkm results file</a></li>
<li class="chapter" data-level="26.4" data-path="26-Bin_quality_scores.html"><a href="26-Bin_quality_scores.html#mcqs-7"><i class="fa fa-check"></i><b>26.4</b> MCQs</a></li>
<li class="chapter" data-level="26.5" data-path="26-Bin_quality_scores.html"><a href="26-Bin_quality_scores.html#bin-quality-summary"><i class="fa fa-check"></i><b>26.5</b> Bin quality summary</a></li>
</ul></li>
<li class="part"><span><b>Functional annotation of bins</b></span></li>
<li class="chapter" data-level="27" data-path="27-Assembly_functional_annotation.html"><a href="27-Assembly_functional_annotation.html"><i class="fa fa-check"></i><b>27</b> Assembly functional annotation</a>
<ul>
<li class="chapter" data-level="" data-path="27-Assembly_functional_annotation.html"><a href="27-Assembly_functional_annotation.html#taxonomic-annotation"><i class="fa fa-check"></i>Taxonomic annotation</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="28-Bakta.html"><a href="28-Bakta.html"><i class="fa fa-check"></i><b>28</b> Bakta</a>
<ul>
<li class="chapter" data-level="28.1" data-path="28-Bakta.html"><a href="28-Bakta.html#annotation"><i class="fa fa-check"></i><b>28.1</b> Annotation</a>
<ul>
<li class="chapter" data-level="28.1.1" data-path="28-Bakta.html"><a href="28-Bakta.html#bakta-premade-results"><i class="fa fa-check"></i><b>28.1.1</b> Bakta: premade results</a></li>
</ul></li>
<li class="chapter" data-level="28.2" data-path="28-Bakta.html"><a href="28-Bakta.html#annotation-files"><i class="fa fa-check"></i><b>28.2</b> Annotation files</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="29-Minpath.html"><a href="29-Minpath.html"><i class="fa fa-check"></i><b>29</b> MinPath</a>
<ul>
<li class="chapter" data-level="29.1" data-path="29-Minpath.html"><a href="29-Minpath.html#ec-extraction"><i class="fa fa-check"></i><b>29.1</b> EC extraction</a>
<ul>
<li class="chapter" data-level="29.1.1" data-path="29-Minpath.html"><a href="29-Minpath.html#code-explanation"><i class="fa fa-check"></i><b>29.1.1</b> Code explanation</a></li>
</ul></li>
<li class="chapter" data-level="29.2" data-path="29-Minpath.html"><a href="29-Minpath.html#minpathrun"><i class="fa fa-check"></i><b>29.2</b> MetaCyc predictions</a></li>
<li class="chapter" data-level="29.3" data-path="29-Minpath.html"><a href="29-Minpath.html#metacyc-report"><i class="fa fa-check"></i><b>29.3</b> MetaCyc report</a></li>
<li class="chapter" data-level="29.4" data-path="29-Minpath.html"><a href="29-Minpath.html#keggs"><i class="fa fa-check"></i><b>29.4</b> KEGGs</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="30-Appendix.html"><a href="30-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="30-Appendix.html"><a href="30-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="30-Appendix.html"><a href="30-Appendix.html#shotgun_meta"><i class="fa fa-check"></i><b>A.1.1</b> shotgun_meta</a></li>
<li class="chapter" data-level="A.1.2" data-path="30-Appendix.html"><a href="30-Appendix.html#cocopye-1"><i class="fa fa-check"></i><b>A.1.2</b> cocopye</a></li>
<li class="chapter" data-level="A.1.3" data-path="30-Appendix.html"><a href="30-Appendix.html#bibakery3.9"><i class="fa fa-check"></i><b>A.1.3</b> bibakery3.9</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="30-Appendix.html"><a href="30-Appendix.html#next-steps"><i class="fa fa-check"></i><b>B</b> Next steps</a></li>
<li class="chapter" data-level="C" data-path="30-Appendix.html"><a href="30-Appendix.html#manuals"><i class="fa fa-check"></i><b>C</b> Manuals</a></li>
<li class="chapter" data-level="D" data-path="30-Appendix.html"><a href="30-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>D</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="depth_calc" class="section level1 hasAnchor" number="23">
<h1><span class="header-section-number">Chapter 23</span> Depth calculation<a href="23-Depth_calculation.html#depth_calc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/submarine.png" style="width:150px; background: white; border-radius:5px; border: white solid 5px" />
</center>
<p>We will use <code>MetaBAT2</code> for genome binning.
It requires <strong>depth</strong> information of the contigs.</p>
<p><strong>Depth</strong> will be calculated for each base of the assembly by aligning the trimmed reads to the metagenome assembly.
The <strong>depth</strong> will be equal to the number of reads/sequences that align to the base, this represents how many times the base was sequenced.
If a base from an assembly aligns to 10 of the reads/sequences used for assembly it will have a depth of 10x.
The average of all the base depths in a contig is its average depth.</p>
<div id="index-assembly" class="section level2 hasAnchor" number="23.1">
<h2><span class="header-section-number">23.1</span> Index assembly<a href="23-Depth_calculation.html#index-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/index.png" style="width:150px; background: white; border-radius:5px; border: white solid 5px" />
</center>
<p>For the alignment we will use <code>bwa</code>. We need to index our assembly file prior to alignment.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb71-1"><a href="23-Depth_calculation.html#cb71-1" tabindex="-1"></a><span class="ex">bwa</span> index ~/6-Assembly/K1/final.contigs.fa</span></code></pre></div>
</div>
<div id="alignment-1" class="section level2 hasAnchor" number="23.2">
<h2><span class="header-section-number">23.2</span> Alignment<a href="23-Depth_calculation.html#alignment-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/dna_alignment.png" style="width:150px; background: white; border-radius:5px; border: white solid 5px" />
</center>
<p>Next we will align our trimmed paired reads we used to create the stitched reads.
We will carry this out with the <code>bwa mem</code> command.</p>
<p><code>bwa mem</code> is a good aligner for short reads.
If you are using long reads (PacBio or Nanopore) <code>minimap2</code> will be more appropriate.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb72-1"><a href="23-Depth_calculation.html#cb72-1" tabindex="-1"></a><span class="ex">bwa</span> mem ~/6-Assembly/K1/final.contigs.fa <span class="dt">\</span></span>
<span id="cb72-2"><a href="23-Depth_calculation.html#cb72-2" tabindex="-1"></a>~/2-Trimmed/K1_R1.fq.gz ~/2-Trimmed/K1_R2.fq.gz <span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb72-3"><a href="23-Depth_calculation.html#cb72-3" tabindex="-1"></a>K1.sam</span></code></pre></div>
</div>
<div id="sam-to-sorted-bam" class="section level2 hasAnchor" number="23.3">
<h2><span class="header-section-number">23.3</span> SAM to sorted BAM<a href="23-Depth_calculation.html#sam-to-sorted-bam" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/sort.png" style="width:150px; background: white; border-radius:5px" />
</center>
<p>After alignment we need to get the file ready for the contig depth summarisation step. This requires converting the <code>SAM</code> file to a <code>BAM</code> (binary form of a <code>SAM</code> file) file and then sorting the <code>BAM</code> file.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb73-1"><a href="23-Depth_calculation.html#cb73-1" tabindex="-1"></a><span class="co"># Convert sam to bam file</span></span>
<span id="cb73-2"><a href="23-Depth_calculation.html#cb73-2" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-bu</span> K1.sam <span class="op">&gt;</span> K1.bam</span>
<span id="cb73-3"><a href="23-Depth_calculation.html#cb73-3" tabindex="-1"></a><span class="co"># Created sorted bam file</span></span>
<span id="cb73-4"><a href="23-Depth_calculation.html#cb73-4" tabindex="-1"></a><span class="ex">samtools</span> sort K1.bam <span class="op">&gt;</span> K1.sort.bam</span></code></pre></div>
</div>
<div id="summarise-depths" class="section level2 hasAnchor" number="23.4">
<h2><span class="header-section-number">23.4</span> Summarise depths<a href="23-Depth_calculation.html#summarise-depths" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/depth.png" style="width:150px; background: white; border-radius:5px" />
</center>
<p>Now we can summarise the contig depths from the sorted <code>bam</code> files with <code>MetaBAT2</code>'s <code>jgi_summarize_bam_contig_depths</code> command.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb74-1"><a href="23-Depth_calculation.html#cb74-1" tabindex="-1"></a><span class="ex">jgi_summarize_bam_contig_depths</span> <span class="at">--outputDepth</span> K1.depth.txt K1.sort.bam</span></code></pre></div>
</div>
<div id="view-depths" class="section level2 hasAnchor" number="23.5">
<h2><span class="header-section-number">23.5</span> View depths<a href="23-Depth_calculation.html#view-depths" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/periscope.png" style="width:150px; background: white; border-radius:5px; border: white solid 5px" />
</center>
<p>Have a look at the depth file.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb75-1"><a href="23-Depth_calculation.html#cb75-1" tabindex="-1"></a><span class="fu">less</span> K1.depth.txt</span></code></pre></div>
<p>Many contigs have low depths of less than 10 (<code>totalAvgDepth</code>) and short lengths of less than &lt;1500 (<code>contigLen</code>).</p>
</div>
<div id="summarise-with-r" class="section level2 hasAnchor" number="23.6">
<h2><span class="header-section-number">23.6</span> Summarise with R<a href="23-Depth_calculation.html#summarise-with-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/R.png" style="width:150px; background: white; border-radius:5px; border: white solid 5px" />
</center>
<p>For tutorial purposes we will have a better at the depth by summarising the contig depths with <code>R</code>.</p>
<p>Activate <code>R</code>:</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb76-1"><a href="23-Depth_calculation.html#cb76-1" tabindex="-1"></a><span class="ex">R</span></span></code></pre></div>
<div id="column-summaries" class="section level3 hasAnchor" number="23.6.1">
<h3><span class="header-section-number">23.6.1</span> Column summaries<a href="23-Depth_calculation.html#column-summaries" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Read in the file and get a <code>summary()</code> of it.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb77-1"><a href="23-Depth_calculation.html#cb77-1" tabindex="-1"></a><span class="co">#Read in the table as an object called df (short for data frame)</span></span>
<span id="cb77-2"><a href="23-Depth_calculation.html#cb77-2" tabindex="-1"></a><span class="co">#We want the first row to be the column names (header=TRUE)</span></span>
<span id="cb77-3"><a href="23-Depth_calculation.html#cb77-3" tabindex="-1"></a><span class="co">#We do not want R to check the column names and &quot;fix&quot; them (check.names=FALSE)</span></span>
<span id="cb77-4"><a href="23-Depth_calculation.html#cb77-4" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;K1.depth.txt&quot;</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">check.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb77-5"><a href="23-Depth_calculation.html#cb77-5" tabindex="-1"></a><span class="co">#Create a summary of the data</span></span>
<span id="cb77-6"><a href="23-Depth_calculation.html#cb77-6" tabindex="-1"></a><span class="fu">summary</span>(df)</span></code></pre></div>
<p>The last command gave us summary information of all the columns.
This includes the minimum, maximum, mean, median, and Inter-Quartile Range (IQR) values.</p>
<p>We can see the values of the <code>contigLen</code> and <code>totalAvgDepth</code> are very low. However, this is most likely due to a bunch of short and low coverage contigs which will be ignored by <code>MetaBAT2</code>, our binning tool.</p>
</div>
<div id="filter-then-summarise" class="section level3 hasAnchor" number="23.6.2">
<h3><span class="header-section-number">23.6.2</span> Filter then summarise<a href="23-Depth_calculation.html#filter-then-summarise" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We'll remove rows with information on contigs shorter than 1500 and rerun the summary.
<code>MetaBAT2</code>'s documentation dictates the minimum contig length should be &gt;=1500 with its default being 2500.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb78-1"><a href="23-Depth_calculation.html#cb78-1" tabindex="-1"></a><span class="co">#Set the new object &quot;df_min1500len&quot; as all rows</span></span>
<span id="cb78-2"><a href="23-Depth_calculation.html#cb78-2" tabindex="-1"></a><span class="co">#where the value in the column &quot;contigLen&quot; of &quot;df&quot;</span></span>
<span id="cb78-3"><a href="23-Depth_calculation.html#cb78-3" tabindex="-1"></a><span class="co">#Is greater than or equal to 1500</span></span>
<span id="cb78-4"><a href="23-Depth_calculation.html#cb78-4" tabindex="-1"></a>df_min1500len <span class="ot">&lt;-</span> df[df<span class="sc">$</span>contigLen <span class="sc">&gt;=</span> <span class="dv">1500</span>,]</span>
<span id="cb78-5"><a href="23-Depth_calculation.html#cb78-5" tabindex="-1"></a><span class="co">#Summary of our new data frame</span></span>
<span id="cb78-6"><a href="23-Depth_calculation.html#cb78-6" tabindex="-1"></a><span class="fu">summary</span>(df_min1500len)</span></code></pre></div>
<p>That is looking better.
The minimum average depth for <code>MetaBAT2</code> is 1 and our minimum value is 2.700 with a maximum of ~93 (your values may differ slightly).</p>
</div>
<div id="quit-r" class="section level3 hasAnchor" number="23.6.3">
<h3><span class="header-section-number">23.6.3</span> Quit R<a href="23-Depth_calculation.html#quit-r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now you can quit R and continue.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb79-1"><a href="23-Depth_calculation.html#cb79-1" tabindex="-1"></a><span class="co">#quit R</span></span>
<span id="cb79-2"><a href="23-Depth_calculation.html#cb79-2" tabindex="-1"></a><span class="fu">q</span>()</span>
<span id="cb79-3"><a href="23-Depth_calculation.html#cb79-3" tabindex="-1"></a><span class="co">#On the prompt to save your workspace press &quot;n&quot; and then enter.</span></span></code></pre></div>
<p><strong>Note</strong>: One of the reasons for our short contigs is that we only used a subset of our sequencing dataset for this tutorial due to time concerns.</p>
</div>
</div>
<div id="remove-sam-and-bam-files" class="section level2 hasAnchor" number="23.7">
<h2><span class="header-section-number">23.7</span> Remove SAM and BAM files<a href="23-Depth_calculation.html#remove-sam-and-bam-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>SAM and BAM files are large so it is good practice to delete them when you don't need them anymore.</p>
<p>Delete you SAM and BAM files.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb80-1"><a href="23-Depth_calculation.html#cb80-1" tabindex="-1"></a><span class="fu">rm</span> K1.sam K1.bam K1.sort.bam</span></code></pre></div>
<p>Great!
With our <code>K1.depth.txt</code> file we will next carry out <code>MetaBAT2</code> genome binning.</p>
<span class="tippy html-widget html-fill-item" height="480" id="htmlwidget-4c4a357625e755aa92d7" width="672"></span>
<script type="application/json" data-for="htmlwidget-4c4a357625e755aa92d7">{"x":{"element":"mag","opts":{"content":"Metagenome-Assembled Genome","arrow":true,"placement":"bottom"}},"evals":[],"jsHooks":[]}</script>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="22-Genome_binning.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="24-MetaBAT2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
