<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A Mamba installs | Shotgun Metagenomics</title>
  <meta name="description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="A Mamba installs | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Mamba installs | Shotgun Metagenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2025-10-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="24-Assembly_functional_annotation.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/tippyjs-3.2.0/themes/google.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/light-border.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/light.css" rel="stylesheet" />
<link href="libs/tippyjs-3.2.0/themes/translucent.css" rel="stylesheet" />
<script src="libs/tippyjs-3.2.0/tippy.all.min.js"></script>
<script src="libs/tippy-binding-0.1.0/tippy.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Start_conda.html"><a href="04-Start_conda.html"><i class="fa fa-check"></i><b>4</b> Startup &amp; mamba</a></li>
<li class="part"><span><b>Data &amp; QC</b></span></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-summary"><i class="fa fa-check"></i><b>7.5</b> Host removal summary</a></li>
</ul></li>
<li class="part"><span><b>Taxonomy of reads</b></span></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a></li>
<li class="chapter" data-level="9" data-path="09-Kraken2.html"><a href="09-Kraken2.html"><i class="fa fa-check"></i><b>9</b> Kraken2</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-run"><i class="fa fa-check"></i><b>9.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="9.2" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken2-output"><i class="fa fa-check"></i><b>9.2</b> Kraken2: output</a></li>
<li class="chapter" data-level="9.3" data-path="09-Kraken2.html"><a href="09-Kraken2.html#kraken-2-mcqs"><i class="fa fa-check"></i><b>9.3</b> Kraken 2: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Krona.html"><a href="10-Krona.html"><i class="fa fa-check"></i><b>10</b> Krona plot</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Krona.html"><a href="10-Krona.html#krona-run"><i class="fa fa-check"></i><b>10.1</b> Krona: run</a></li>
<li class="chapter" data-level="10.2" data-path="10-Krona.html"><a href="10-Krona.html#krona-visualise"><i class="fa fa-check"></i><b>10.2</b> Krona: visualise</a></li>
<li class="chapter" data-level="10.3" data-path="10-Krona.html"><a href="10-Krona.html#krona-mcqs"><i class="fa fa-check"></i><b>10.3</b> Krona: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Bracken.html"><a href="11-Bracken.html"><i class="fa fa-check"></i><b>11</b> Bracken</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-run"><i class="fa fa-check"></i><b>11.1</b> Bracken: run</a></li>
<li class="chapter" data-level="11.2" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-output"><i class="fa fa-check"></i><b>11.2</b> Bracken: output</a></li>
<li class="chapter" data-level="11.3" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-mcqs"><i class="fa fa-check"></i><b>11.3</b> Bracken: MCQs</a></li>
<li class="chapter" data-level="11.4" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-merging-output"><i class="fa fa-check"></i><b>11.4</b> Bracken: merging output</a></li>
<li class="chapter" data-level="11.5" data-path="11-Bracken.html"><a href="11-Bracken.html#bracken-extracting-output"><i class="fa fa-check"></i><b>11.5</b> Bracken: extracting output</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Lefse.html"><a href="12-Lefse.html"><i class="fa fa-check"></i><b>12</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2_lefse_metadata"><i class="fa fa-check"></i><b>12.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="12.2" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-format"><i class="fa fa-check"></i><b>12.2</b> LEfSe: format</a></li>
<li class="chapter" data-level="12.3" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-run"><i class="fa fa-check"></i><b>12.3</b> LEfSe: run</a></li>
<li class="chapter" data-level="12.4" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-visualisation"><i class="fa fa-check"></i><b>12.4</b> LEfSe: visualisation</a></li>
<li class="chapter" data-level="12.5" data-path="12-Lefse.html"><a href="12-Lefse.html#lefse-mcqs"><i class="fa fa-check"></i><b>12.5</b> LEfSe: MCQs</a></li>
<li class="chapter" data-level="12.6" data-path="12-Lefse.html"><a href="12-Lefse.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>12.6</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="part"><span><b>Functional profiling of reads</b></span></li>
<li class="chapter" data-level="13" data-path="13-Functional_profiling.html"><a href="13-Functional_profiling.html"><i class="fa fa-check"></i><b>13</b> Functional profiling</a></li>
<li class="chapter" data-level="14" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html"><i class="fa fa-check"></i><b>14</b> HUMAnN</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#humann-mamba-directories-and-files"><i class="fa fa-check"></i><b>14.1</b> HUMAnN: mamba, directories, and files</a></li>
<li class="chapter" data-level="14.2" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#humann-run"><i class="fa fa-check"></i><b>14.2</b> HUMAnN: run</a></li>
<li class="chapter" data-level="14.3" data-path="14-HUMAnN.html"><a href="14-HUMAnN.html#humann-output"><i class="fa fa-check"></i><b>14.3</b> HUMAnN: output</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html"><i class="fa fa-check"></i><b>15</b> Multi sample processing</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#combining-data"><i class="fa fa-check"></i><b>15.1</b> Combining data</a></li>
<li class="chapter" data-level="15.2" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#split-stratified-table"><i class="fa fa-check"></i><b>15.2</b> Split stratified table</a></li>
<li class="chapter" data-level="15.3" data-path="15-Multi_sample_processing.html"><a href="15-Multi_sample_processing.html#humannrenormalise"><i class="fa fa-check"></i><b>15.3</b> Renormalising data</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Heatmap.html"><a href="16-Heatmap.html"><i class="fa fa-check"></i><b>16</b> Heatmap</a></li>
<li class="chapter" data-level="17" data-path="17-LEfSe_humann.html"><a href="17-LEfSe_humann.html"><i class="fa fa-check"></i><b>17</b> LEfSe</a></li>
<li class="chapter" data-level="18" data-path="18-HUMAnN_task.html"><a href="18-HUMAnN_task.html"><i class="fa fa-check"></i><b>18</b> Optional task</a></li>
<li class="part"><span><b>Assembly Approach</b></span></li>
<li class="chapter" data-level="19" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html"><i class="fa fa-check"></i><b>19</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="19.1" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html#metagenome-assembly-mamba"><i class="fa fa-check"></i><b>19.1</b> Metagenome assembly: Mamba</a></li>
<li class="chapter" data-level="19.2" data-path="19-Metagenome_assembly.html"><a href="19-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>19.2</b> A primer on short read assembly</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html"><i class="fa fa-check"></i><b>20</b> Stitching read pairs</a>
<ul>
<li class="chapter" data-level="20.1" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html#flash-run"><i class="fa fa-check"></i><b>20.1</b> FLASH: run</a></li>
<li class="chapter" data-level="20.2" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html#flash-output"><i class="fa fa-check"></i><b>20.2</b> FLASH: Output</a></li>
<li class="chapter" data-level="20.3" data-path="20-Stitching_reads.html"><a href="20-Stitching_reads.html#flash-mcqs"><i class="fa fa-check"></i><b>20.3</b> FLASH: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="21_Megahit.html"><a href="21_Megahit.html"><i class="fa fa-check"></i><b>21</b> Megahit Assembly</a>
<ul>
<li class="chapter" data-level="21.1" data-path="21_Megahit.html"><a href="21_Megahit.html#megahit"><i class="fa fa-check"></i><b>21.1</b> MEGAHIT</a></li>
<li class="chapter" data-level="21.2" data-path="21_Megahit.html"><a href="21_Megahit.html#quast"><i class="fa fa-check"></i><b>21.2</b> QUAST</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="21_Megahit.html"><a href="21_Megahit.html#quast-run"><i class="fa fa-check"></i><b>21.2.1</b> QUAST: run</a></li>
<li class="chapter" data-level="21.2.2" data-path="21_Megahit.html"><a href="21_Megahit.html#quast-visualise"><i class="fa fa-check"></i><b>21.2.2</b> QUAST: visualise</a></li>
<li class="chapter" data-level="21.2.3" data-path="21_Megahit.html"><a href="21_Megahit.html#quast-mcqs"><i class="fa fa-check"></i><b>21.2.3</b> QUAST: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="21_Megahit.html"><a href="21_Megahit.html#metagenome-assembly-summary"><i class="fa fa-check"></i><b>21.3</b> Metagenome assembly summary</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html"><i class="fa fa-check"></i><b>22</b> Genome binning</a>
<ul>
<li class="chapter" data-level="22.1" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>22.1</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="22.1.1" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2-depth-calculation"><i class="fa fa-check"></i><b>22.1.1</b> MetaBAT2: depth calculation</a></li>
<li class="chapter" data-level="22.1.2" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2-run"><i class="fa fa-check"></i><b>22.1.2</b> MetaBAT2: run</a></li>
<li class="chapter" data-level="22.1.3" data-path="22-Genome_binning.html"><a href="22-Genome_binning.html#metabat2-output"><i class="fa fa-check"></i><b>22.1.3</b> MetaBAT2: output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html"><i class="fa fa-check"></i><b>23</b> Bin quality</a>
<ul>
<li class="chapter" data-level="23.1" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#workflow-concept"><i class="fa fa-check"></i><b>23.1</b> Workflow concept</a>
<ul>
<li class="chapter" data-level="23.1.1" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#input-pre-processing"><i class="fa fa-check"></i><b>23.1.1</b> Input Pre-processing</a></li>
<li class="chapter" data-level="23.1.2" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#stage-i"><i class="fa fa-check"></i><b>23.1.2</b> Stage I</a></li>
<li class="chapter" data-level="23.1.3" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#stage-ii"><i class="fa fa-check"></i><b>23.1.3</b> Stage II</a></li>
</ul></li>
<li class="chapter" data-level="23.2" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#mamba"><i class="fa fa-check"></i><b>23.2</b> Mamba</a></li>
<li class="chapter" data-level="23.3" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#cocopye-run"><i class="fa fa-check"></i><b>23.3</b> CoCoPyE: run</a>
<ul>
<li class="chapter" data-level="" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#parameters-10"><i class="fa fa-check"></i>Parameters</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#cocopye-output"><i class="fa fa-check"></i><b>23.4</b> CoCoPyE: output</a></li>
<li class="chapter" data-level="23.5" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#cocopye-quality-score"><i class="fa fa-check"></i><b>23.5</b> CoCoPyE: Quality score</a>
<ul>
<li class="chapter" data-level="" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#quality-file"><i class="fa fa-check"></i>Quality file</a></li>
<li class="chapter" data-level="" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#calculate-quality-with-awk"><i class="fa fa-check"></i>Calculate quality with awk</a></li>
<li class="chapter" data-level="" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#add-quality-to-the-checkm-results-file"><i class="fa fa-check"></i>Add quality to the checkm results file</a></li>
</ul></li>
<li class="chapter" data-level="23.6" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#checkm-mcqs"><i class="fa fa-check"></i><b>23.6</b> CheckM: MCQs</a></li>
<li class="chapter" data-level="23.7" data-path="23-Bin_quality.html"><a href="23-Bin_quality.html#bin-quality-summary"><i class="fa fa-check"></i><b>23.7</b> Bin quality summary</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html"><i class="fa fa-check"></i><b>24</b> Assembly functional annotation</a>
<ul>
<li class="chapter" data-level="24.1" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#taxonomic-annotation"><i class="fa fa-check"></i><b>24.1</b> Taxonomic annotation</a></li>
<li class="chapter" data-level="24.2" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#bakta"><i class="fa fa-check"></i><b>24.2</b> Bakta</a>
<ul>
<li class="chapter" data-level="24.2.1" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#bakta-run"><i class="fa fa-check"></i><b>24.2.1</b> Bakta: run</a></li>
<li class="chapter" data-level="24.2.2" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#bakta-premade-results"><i class="fa fa-check"></i><b>24.2.2</b> Bakta: premade results</a></li>
<li class="chapter" data-level="24.2.3" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#bakta-output"><i class="fa fa-check"></i><b>24.2.3</b> Bakta: output</a></li>
</ul></li>
<li class="chapter" data-level="24.3" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#minpath"><i class="fa fa-check"></i><b>24.3</b> MinPath</a>
<ul>
<li class="chapter" data-level="24.3.1" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#minpath-ec-extraction"><i class="fa fa-check"></i><b>24.3.1</b> MinPath: EC extraction</a></li>
<li class="chapter" data-level="24.3.2" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#minpathrun"><i class="fa fa-check"></i><b>24.3.2</b> MinPath: run</a></li>
<li class="chapter" data-level="24.3.3" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#minpath-output"><i class="fa fa-check"></i><b>24.3.3</b> MinPath: output</a></li>
<li class="chapter" data-level="24.3.4" data-path="24-Assembly_functional_annotation.html"><a href="24-Assembly_functional_annotation.html#keggs"><i class="fa fa-check"></i><b>24.3.4</b> KEGGs</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="25-Appendix.html"><a href="25-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="25-Appendix.html"><a href="25-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="25-Appendix.html"><a href="25-Appendix.html#next-steps"><i class="fa fa-check"></i><b>B</b> Next steps</a></li>
<li class="chapter" data-level="C" data-path="25-Appendix.html"><a href="25-Appendix.html#manuals"><i class="fa fa-check"></i><b>C</b> Manuals</a></li>
<li class="chapter" data-level="D" data-path="25-Appendix.html"><a href="25-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>D</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mamba-installs" class="section level1 hasAnchor" number="25">
<h1><span class="header-section-number">A</span> Mamba installs<a href="25-Appendix.html#mamba-installs" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/mamba_logo.png" style="width:200px; border-radius:5px" />
</center>
<div id="mamba_install" class="section level2 hasAnchor" number="25.1">
<h2><span class="header-section-number">A.1</span> Mamba installation and environment<a href="25-Appendix.html#mamba_install" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Mamba is a reimplementation of conda. It is a great tool for installing bioinformatic packages including R packages.</p>
<p>Mamba github: <a href="https://github.com/mamba-org/mamba" class="uri">https://github.com/mamba-org/mamba</a></p>
<p>The best way to use Mamba is to install Miniforge. It has both Conda and Mamba commands.</p>
<p>Miniforge installation: <a href="https://github.com/conda-forge/miniforge" class="uri">https://github.com/conda-forge/miniforge</a></p>
<p>Mamba guide: <a href="https://mamba.readthedocs.io/en/latest/user_guide/mamba.html" class="uri">https://mamba.readthedocs.io/en/latest/user_guide/mamba.html</a></p>
<p>To create the mamba environment <code>shotgun_meta</code> run the below commands in your bash. You will need to have installed <code>mamba</code> first.</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb117-1"><a href="25-Appendix.html#cb117-1" tabindex="-1"></a><span class="co">#shotgun_meta</span></span>
<span id="cb117-2"><a href="25-Appendix.html#cb117-2" tabindex="-1"></a><span class="ex">mamba</span> create <span class="at">-n</span> shotgun_meta</span>
<span id="cb117-3"><a href="25-Appendix.html#cb117-3" tabindex="-1"></a><span class="ex">mamba</span> activate shotgun_meta</span>
<span id="cb117-4"><a href="25-Appendix.html#cb117-4" tabindex="-1"></a><span class="co">#Install packages</span></span>
<span id="cb117-5"><a href="25-Appendix.html#cb117-5" tabindex="-1"></a><span class="ex">mamba</span> install <span class="at">-c</span> bioconda fastqc trim-galore multiqc bowtie2 kraken2 <span class="dt">\</span></span>
<span id="cb117-6"><a href="25-Appendix.html#cb117-6" tabindex="-1"></a>krona bracken lefse flash megahit quast metabat2 prokka bbmap bakta circos</span>
<span id="cb117-7"><a href="25-Appendix.html#cb117-7" tabindex="-1"></a><span class="co">#Update krona taxonomy database</span></span>
<span id="cb117-8"><a href="25-Appendix.html#cb117-8" tabindex="-1"></a><span class="ex">ktUpdateTaxonomy.sh</span></span>
<span id="cb117-9"><a href="25-Appendix.html#cb117-9" tabindex="-1"></a><span class="co">#Install GenoVi via pip</span></span>
<span id="cb117-10"><a href="25-Appendix.html#cb117-10" tabindex="-1"></a><span class="ex">pip</span> install genovi</span></code></pre></div>
<p>You will need to install the <code>Bakta</code> database as well. This requires you have a directory to store the database. In the below example you will need a directory with the path <code>~/databases/bakta_db</code>, of course feel free to use a different location you have.</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb118-1"><a href="25-Appendix.html#cb118-1" tabindex="-1"></a><span class="ex">bakta_db</span> download <span class="at">--output</span> ~/databases/bakta <span class="at">--type</span> full</span></code></pre></div>
<p>Install <a href="https://github.com/mgtools/MinPath">MinPath</a> via <code>git</code>.
I suggest creating a directory called "git_installs" in your home directory and running this code there.</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb119-1"><a href="25-Appendix.html#cb119-1" tabindex="-1"></a><span class="co">#git_installs directory</span></span>
<span id="cb119-2"><a href="25-Appendix.html#cb119-2" tabindex="-1"></a><span class="fu">mkdir</span> ~/git_installs</span>
<span id="cb119-3"><a href="25-Appendix.html#cb119-3" tabindex="-1"></a><span class="bu">cd</span> ~/git_installs</span>
<span id="cb119-4"><a href="25-Appendix.html#cb119-4" tabindex="-1"></a><span class="co">#MinPath</span></span>
<span id="cb119-5"><a href="25-Appendix.html#cb119-5" tabindex="-1"></a><span class="fu">git</span> clone https://github.com/mgtools/MinPath</span></code></pre></div>
<p>To run <code>MinPath</code> in your own machines you will need to use the full path of the python file.
If you installed these in your "~/git_installs" examples are below.</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb120-1"><a href="25-Appendix.html#cb120-1" tabindex="-1"></a><span class="co">#MinPath help page</span></span>
<span id="cb120-2"><a href="25-Appendix.html#cb120-2" tabindex="-1"></a><span class="ex">python</span> ~/git_installs/Minpath/MinPath.py <span class="at">-h</span></span></code></pre></div>
<p>To create the mamba environment <code>cocopye</code> run the below commands in your bash. You will need to have installed <code>mamba</code> first.</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb121-1"><a href="25-Appendix.html#cb121-1" tabindex="-1"></a><span class="co">#checkm</span></span>
<span id="cb121-2"><a href="25-Appendix.html#cb121-2" tabindex="-1"></a><span class="ex">mamba</span> create <span class="at">-n</span> cocopye</span>
<span id="cb121-3"><a href="25-Appendix.html#cb121-3" tabindex="-1"></a><span class="ex">mamba</span> activate checkm2</span>
<span id="cb121-4"><a href="25-Appendix.html#cb121-4" tabindex="-1"></a><span class="ex">mamba</span> install bioconda::checkm2=1.1.0</span>
<span id="cb121-5"><a href="25-Appendix.html#cb121-5" tabindex="-1"></a><span class="co">#Download required databases</span></span>
<span id="cb121-6"><a href="25-Appendix.html#cb121-6" tabindex="-1"></a><span class="ex">checkm2</span> database <span class="at">--download</span></span></code></pre></div>
<p>To create the mamba environment <code>biobakery3.9</code> run the below commands in your bash. You will need to have installed <code>mamba</code> first.</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb122-1"><a href="25-Appendix.html#cb122-1" tabindex="-1"></a><span class="co">#biobakery3</span></span>
<span id="cb122-2"><a href="25-Appendix.html#cb122-2" tabindex="-1"></a><span class="ex">mamba</span> create <span class="at">-n</span> biobakery3.9</span>
<span id="cb122-3"><a href="25-Appendix.html#cb122-3" tabindex="-1"></a><span class="ex">mamba</span> activate biobakery3.9</span>
<span id="cb122-4"><a href="25-Appendix.html#cb122-4" tabindex="-1"></a><span class="co">#Add required channels</span></span>
<span id="cb122-5"><a href="25-Appendix.html#cb122-5" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--add</span> channels defaults</span>
<span id="cb122-6"><a href="25-Appendix.html#cb122-6" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--add</span> channels bioconda</span>
<span id="cb122-7"><a href="25-Appendix.html#cb122-7" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--add</span> channels conda-forge</span>
<span id="cb122-8"><a href="25-Appendix.html#cb122-8" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--add</span> channels biobakery</span>
<span id="cb122-9"><a href="25-Appendix.html#cb122-9" tabindex="-1"></a><span class="co">#Install huamnn3.9</span></span>
<span id="cb122-10"><a href="25-Appendix.html#cb122-10" tabindex="-1"></a><span class="ex">mamba</span> install <span class="at">-c</span> biobakery humann=3.9</span>
<span id="cb122-11"><a href="25-Appendix.html#cb122-11" tabindex="-1"></a><span class="co">#Install hclust2 and matplotlib</span></span>
<span id="cb122-12"><a href="25-Appendix.html#cb122-12" tabindex="-1"></a><span class="ex">mamba</span> install bioncda::hclust2=1.0.0 conda-forge::matplotlib=3.7.3</span></code></pre></div>
<p>Next you will need to install the <code>HUMAnN</code> and <code>MetaPhlAn</code> databases.</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb123-1"><a href="25-Appendix.html#cb123-1" tabindex="-1"></a><span class="co">#Ensure you have biobakery3 mamba environment activated</span></span>
<span id="cb123-2"><a href="25-Appendix.html#cb123-2" tabindex="-1"></a><span class="co">#Update HUMAnN databases</span></span>
<span id="cb123-3"><a href="25-Appendix.html#cb123-3" tabindex="-1"></a><span class="ex">humann_databases</span> <span class="at">--download</span> chocophlan full /path/to/databases <span class="at">--update-config</span> yes</span>
<span id="cb123-4"><a href="25-Appendix.html#cb123-4" tabindex="-1"></a><span class="ex">humann_databases</span> <span class="at">--download</span> uniref uniref90_diamond /path/to/databases <span class="at">--update-config</span> yes</span>
<span id="cb123-5"><a href="25-Appendix.html#cb123-5" tabindex="-1"></a><span class="ex">humann_databases</span> <span class="at">--download</span> utility_mapping full /path/to/databases <span class="at">--update-config</span> yes</span>
<span id="cb123-6"><a href="25-Appendix.html#cb123-6" tabindex="-1"></a><span class="co">#Install MetaPhlAn databases</span></span>
<span id="cb123-7"><a href="25-Appendix.html#cb123-7" tabindex="-1"></a><span class="ex">metaphlan</span> <span class="at">--install</span> <span class="at">--index</span> mpa_vOct22_CHOCOPhlAnSGB_202403</span></code></pre></div>
<p>Installing the <code>MetaPhlAn</code> databases may not work with the command above. If not follow the below instructions.</p>
<ul>
<li>Go to the following link: <a href="http://cmprod1.cibio.unitn.it/biobakery4/metaphlan_databases/" class="uri">http://cmprod1.cibio.unitn.it/biobakery4/metaphlan_databases/</a></li>
<li>Download the following files (or the most recent vOct22 files):
<ul>
<li><a href="http://cmprod1.cibio.unitn.it/biobakery4/metaphlan_databases/mpa_vOct22_CHOCOPhlAnSGB_202403.md5"><strong>mpa_vOct22_CHOCOPhlAnSGB_202403.md5</strong></a></li>
<li><a href="http://cmprod1.cibio.unitn.it/biobakery4/metaphlan_databases/mpa_vOct22_CHOCOPhlAnSGB_202403.tar"><strong>mpa_vOct22_CHOCOPhlAnSGB_202403.tar</strong></a></li>
<li><a href="http://cmprod1.cibio.unitn.it/biobakery4/metaphlan_databases/bowtie2_indexes/mpa_vOct22_CHOCOPhlAnSGB_202403_bt2.md5"><strong>bowtie2_indexes/mpa_vOct22_CHOCOPhlAnSGB_202403_bt2.md5</strong></a></li>
<li><a href="http://cmprod1.cibio.unitn.it/biobakery4/metaphlan_databases/bowtie2_indexes/mpa_vOct22_CHOCOPhlAnSGB_202403_bt2.tar"><strong>bowtie2_indexes/mpa_vOct22_CHOCOPhlAnSGB_202403_bt2.tar</strong></a></li>
</ul></li>
<li>Put these in the <code>MetaPhlAn</code> databases directory. This will be in your mamba environment directory.
<ul>
<li>E.g.: ~/mambaforge/envs/biobakery3.9/lib/python3.7/site-packages/metaphlan/metaphlan_databases</li>
<li>Note: this may not be your exact path.</li>
</ul></li>
<li>Run <code>metaphlan --install --index mpa_vOct22_CHOCOPhlAnSGB_202403</code> again.</li>
<li>Untar the CHOCOPhlan tar file
<ul>
<li>Change directory to your <code>MetaPhlAn</code> database directory.</li>
<li><code>tar -xvf mpa_vOct22_CHOCOPhlAnSGB_202403.tar</code></li>
</ul></li>
</ul>
<p>Once the databses are all setup you can test <code>HUMAnN</code>.</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb124-1"><a href="25-Appendix.html#cb124-1" tabindex="-1"></a><span class="ex">humann_test</span></span></code></pre></div>
</div>
</div>
<div id="next-steps" class="section level1 hasAnchor" number="26">
<h1><span class="header-section-number">B</span> Next steps<a href="25-Appendix.html#next-steps" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/step.png" style="width:200px" />
</center>
<p>Below are some good links to start with before carrying out your own projects.</p>
<ul>
<li><code>KrakenTools</code>: A suite of scripts to be used alongside the Kraken, KrakenUniq, Kraken 2, or Bracken programs.
<ul>
<li><a href="https://github.com/jenniferlu717/KrakenTools" class="uri">https://github.com/jenniferlu717/KrakenTools</a></li>
</ul></li>
<li>A review of computational tools for generating metagenome-assembled genomes from metagenomic sequencing data
<ul>
<li><a href="https://www.sciencedirect.com/science/article/pii/S2001037021004931" class="uri">https://www.sciencedirect.com/science/article/pii/S2001037021004931</a></li>
</ul></li>
<li>A review paper on metagenome assembly approaches
<ul>
<li>Genome-resolved metagenomics using environmental and clinical samples: <a href="https://academic.oup.com/bib/advance-article/doi/10.1093/bib/bbab030/6184411" class="uri">https://academic.oup.com/bib/advance-article/doi/10.1093/bib/bbab030/6184411</a></li>
</ul></li>
<li>bioBakery tools for meta'omic profiling
<ul>
<li><a href="https://github.com/biobakery/biobakery" class="uri">https://github.com/biobakery/biobakery</a></li>
</ul></li>
<li>Assessing the performance of different approaches for functional and taxonomic annotation of metagenomes
<ul>
<li><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6289-6" class="uri">https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6289-6</a></li>
</ul></li>
<li>MicrobeAnnotator: Easy-to-use pipeline for the comprehensive metabolic annotation of microbial genomes.
<ul>
<li><a href="https://github.com/cruizperez/MicrobeAnnotator" class="uri">https://github.com/cruizperez/MicrobeAnnotator</a></li>
</ul></li>
<li>MetaEuk: Functional annotation of eukaryotic metagenome
<ul>
<li><a href="https://github.com/soedinglab/metaeuk" class="uri">https://github.com/soedinglab/metaeuk</a>
_ Methods for Metagenomic data visualisation and analysis</li>
<li><a href="https://www.researchgate.net/publication/318252633_Methods_for_The_Metagenomic_Data_Visualization_and_Analysis" class="uri">https://www.researchgate.net/publication/318252633_Methods_for_The_Metagenomic_Data_Visualization_and_Analysis</a></li>
</ul></li>
<li>Visualizing metagenomic and metatranscriptomic data: A comprehensive review
<ul>
<li><a href="https://www.sciencedirect.com/science/article/pii/S2001037024001430" class="uri">https://www.sciencedirect.com/science/article/pii/S2001037024001430</a></li>
</ul></li>
</ul>
</div>
<div id="manuals" class="section level1 hasAnchor" number="27">
<h1><span class="header-section-number">C</span> Manuals<a href="25-Appendix.html#manuals" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/manual.png" style="width:150px" />
</center>
<p>Conda: <a href="https://conda.io/projects/conda/en/latest/user-guide/getting-started.html" class="uri">https://conda.io/projects/conda/en/latest/user-guide/getting-started.html</a></p>
<p>FastQC: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a></p>
<p>MultiQC: <a href="https://multiqc.info/" class="uri">https://multiqc.info/</a></p>
<p>Trim Galore: <a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/</a></p>
<p>Bowtie2: <a href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml" class="uri">http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml</a></p>
<p>samtools: <a href="http://www.htslib.org/" class="uri">http://www.htslib.org/</a></p>
<p>BBTools: <a href="https://jgi.doe.gov/data-and-tools/bbtools/" class="uri">https://jgi.doe.gov/data-and-tools/bbtools/</a></p>
<p>Kraken2: <a href="https://github.com/DerrickWood/kraken2/wiki/Manual" class="uri">https://github.com/DerrickWood/kraken2/wiki/Manual</a></p>
<p>Krona: <a href="https://github.com/marbl/Krona/wiki/KronaTools" class="uri">https://github.com/marbl/Krona/wiki/KronaTools</a></p>
<p>Bracken: <a href="https://ccb.jhu.edu/software/bracken/index.shtml?t=manual" class="uri">https://ccb.jhu.edu/software/bracken/index.shtml?t=manual</a></p>
<p>LEfSe: <a href="https://huttenhower.sph.harvard.edu/lefse/" class="uri">https://huttenhower.sph.harvard.edu/lefse/</a></p>
<p>HUMAnN 3.0: <a href="https://huttenhower.sph.harvard.edu/humann/" class="uri">https://huttenhower.sph.harvard.edu/humann/</a></p>
<p>MetaPhlAn 4.0: <a href="https://github.com/biobakery/MetaPhlAn/wiki/MetaPhlAn-4" class="uri">https://github.com/biobakery/MetaPhlAn/wiki/MetaPhlAn-4</a></p>
<p>Biobakery: <a href="https://github.com/biobakery/biobakery" class="uri">https://github.com/biobakery/biobakery</a></p>
<p>hclust2: <a href="https://github.com/SegataLab/hclust2" class="uri">https://github.com/SegataLab/hclust2</a></p>
<p>MegaHit: <a href="https://github.com/voutcn/megahit" class="uri">https://github.com/voutcn/megahit</a></p>
<p>BWA: <a href="https://github.com/lh3/bwa" class="uri">https://github.com/lh3/bwa</a></p>
<p>minimap2: <a href="https://github.com/lh3/minimap2" class="uri">https://github.com/lh3/minimap2</a></p>
<p>MetaBAT2: <a href="https://bitbucket.org/berkeleylab/metabat/src/master/" class="uri">https://bitbucket.org/berkeleylab/metabat/src/master/</a></p>
<p>CoCoPyE: <a href="https://github.com/gobics/cocopye" class="uri">https://github.com/gobics/cocopye</a></p>
<p>PhyloPhlAn: <a href="https://github.com/biobakery/phylophlan/wiki" class="uri">https://github.com/biobakery/phylophlan/wiki</a></p>
<p>Prokka: <a href="https://github.com/tseemann/prokka" class="uri">https://github.com/tseemann/prokka</a></p>
<p>MinPath: <a href="https://github.com/mgtools/MinPath/blob/master/readme" class="uri">https://github.com/mgtools/MinPath/blob/master/readme</a></p>
<p>MetaCyc: <a href="https://metacyc.org/" class="uri">https://metacyc.org/</a></p>
</div>
<div id="obtaining-read-data" class="section level1 hasAnchor" number="28">
<h1><span class="header-section-number">D</span> Obtaining Read Data<a href="25-Appendix.html#obtaining-read-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The following commands can be used to obtain the sequence data used in this practical, directly from the EBI metagenomics site. It is worth noting that these are the full set of data, not like the miniaturised version you have used in the tutorial.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb125-1"><a href="25-Appendix.html#cb125-1" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K1_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505102/ERR505102_1.fastq.gz</span>
<span id="cb125-2"><a href="25-Appendix.html#cb125-2" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K1_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505102/ERR505102_2.fastq.gz</span>
<span id="cb125-3"><a href="25-Appendix.html#cb125-3" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K2_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505104/ERR505104_1.fastq.gz</span>
<span id="cb125-4"><a href="25-Appendix.html#cb125-4" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K2_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505104/ERR505104_2.fastq.gz</span>
<span id="cb125-5"><a href="25-Appendix.html#cb125-5" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K3_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505095/ERR505095_1.fastq.gz</span>
<span id="cb125-6"><a href="25-Appendix.html#cb125-6" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K3_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505095/ERR505095_2.fastq.gz</span>
<span id="cb125-7"><a href="25-Appendix.html#cb125-7" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K4_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505096/ERR505096_1.fastq.gz</span>
<span id="cb125-8"><a href="25-Appendix.html#cb125-8" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K4_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505096/ERR505096_2.fastq.gz</span>
<span id="cb125-9"><a href="25-Appendix.html#cb125-9" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K5_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505097/ERR505097_1.fastq.gz</span>
<span id="cb125-10"><a href="25-Appendix.html#cb125-10" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K5_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505097/ERR505097_2.fastq.gz</span>
<span id="cb125-11"><a href="25-Appendix.html#cb125-11" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K6_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505098/ERR505098_1.fastq.gz</span>
<span id="cb125-12"><a href="25-Appendix.html#cb125-12" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K6_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505098/ERR505098_2.fastq.gz</span>
<span id="cb125-13"><a href="25-Appendix.html#cb125-13" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K7_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505099/ERR505099_1.fastq.gz</span>
<span id="cb125-14"><a href="25-Appendix.html#cb125-14" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K7_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505099/ERR505099_2.fastq.gz</span>
<span id="cb125-15"><a href="25-Appendix.html#cb125-15" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K8_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505100/ERR505100_1.fastq.gz</span>
<span id="cb125-16"><a href="25-Appendix.html#cb125-16" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K8_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505100/ERR505100_2.fastq.gz</span>
<span id="cb125-17"><a href="25-Appendix.html#cb125-17" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K9_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505101/ERR505101_1.fastq.gz</span>
<span id="cb125-18"><a href="25-Appendix.html#cb125-18" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K9_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505101/ERR505101_2.fastq.gz</span>
<span id="cb125-19"><a href="25-Appendix.html#cb125-19" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K10_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505103/ERR505103_1.fastq.gz</span>
<span id="cb125-20"><a href="25-Appendix.html#cb125-20" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K10_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505103/ERR505103_2.fastq.gz</span>
<span id="cb125-21"><a href="25-Appendix.html#cb125-21" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K11_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505105/ERR505105_1.fastq.gz</span>
<span id="cb125-22"><a href="25-Appendix.html#cb125-22" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K11_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505105/ERR505105_2.fastq.gz</span>
<span id="cb125-23"><a href="25-Appendix.html#cb125-23" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K12_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505106/ERR505106_1.fastq.gz</span>
<span id="cb125-24"><a href="25-Appendix.html#cb125-24" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K12_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505106/ERR505106_2.fastq.gz</span>
<span id="cb125-25"><a href="25-Appendix.html#cb125-25" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W1_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505090/ERR505090_1.fastq.gz</span>
<span id="cb125-26"><a href="25-Appendix.html#cb125-26" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W1_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505090/ERR505090_2.fastq.gz</span>
<span id="cb125-27"><a href="25-Appendix.html#cb125-27" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W2_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505088/ERR505088_1.fastq.gz</span>
<span id="cb125-28"><a href="25-Appendix.html#cb125-28" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W2_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505088/ERR505088_2.fastq.gz</span>
<span id="cb125-29"><a href="25-Appendix.html#cb125-29" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W3_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505084/ERR505084_1.fastq.gz</span>
<span id="cb125-30"><a href="25-Appendix.html#cb125-30" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W3_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505084/ERR505084_2.fastq.gz</span>
<span id="cb125-31"><a href="25-Appendix.html#cb125-31" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W4_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505085/ERR505085_1.fastq.gz</span>
<span id="cb125-32"><a href="25-Appendix.html#cb125-32" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W4_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505085/ERR505085_2.fastq.gz</span>
<span id="cb125-33"><a href="25-Appendix.html#cb125-33" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W5_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505086/ERR505086_1.fastq.gz</span>
<span id="cb125-34"><a href="25-Appendix.html#cb125-34" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W5_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505086/ERR505086_2.fastq.gz</span>
<span id="cb125-35"><a href="25-Appendix.html#cb125-35" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W6_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505087/ERR505087_1.fastq.gz</span>
<span id="cb125-36"><a href="25-Appendix.html#cb125-36" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W6_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505087/ERR505087_2.fastq.gz</span>
<span id="cb125-37"><a href="25-Appendix.html#cb125-37" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W7_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505089/ERR505089_1.fastq.gz</span>
<span id="cb125-38"><a href="25-Appendix.html#cb125-38" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W7_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505089/ERR505089_2.fastq.gz</span>
<span id="cb125-39"><a href="25-Appendix.html#cb125-39" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W8_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505091/ERR505091_1.fastq.gz</span>
<span id="cb125-40"><a href="25-Appendix.html#cb125-40" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W8_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505091/ERR505091_2.fastq.gz</span>
<span id="cb125-41"><a href="25-Appendix.html#cb125-41" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W9_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505092/ERR505092_1.fastq.gz</span>
<span id="cb125-42"><a href="25-Appendix.html#cb125-42" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W9_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505092/ERR505092_2.fastq.gz</span>
<span id="cb125-43"><a href="25-Appendix.html#cb125-43" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W10_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505093/ERR505093_1.fastq.gz</span>
<span id="cb125-44"><a href="25-Appendix.html#cb125-44" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W10_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505093/ERR505093_2.fastq.gz</span>
<span id="cb125-45"><a href="25-Appendix.html#cb125-45" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W11_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505094/ERR505094_1.fastq.gz</span>
<span id="cb125-46"><a href="25-Appendix.html#cb125-46" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W11_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505094/ERR505094_2.fastq.gz</span>
<span id="cb125-47"><a href="25-Appendix.html#cb125-47" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W12_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505083/ERR505083_1.fastq.gz</span>
<span id="cb125-48"><a href="25-Appendix.html#cb125-48" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W12_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505083/ERR505083_2.fastq.gz</span></code></pre></div>

</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="24-Assembly_functional_annotation.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
