<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Background | Phylogenomics</title>
  <meta name="description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Background | Phylogenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Phylogenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Background | Phylogenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Gavin Gouws, Matthew Gemmell, Ewan Harney, Helen Hipperson, Kathryn Maher" />


<meta name="date" content="2024-11-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="01-Intro.html"/>
<link rel="next" href="03-Cluster_Introduction.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Phylogenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-ddradseq-data-of-manta-and-devil-rays"><i class="fa fa-check"></i><b>2.1</b> Example data: ddRADseq data of manta and devil rays</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#manta--and-devil-ray-ddrad-data-set"><i class="fa fa-check"></i><b>4.2</b> Manta- and devil-ray ddRAD data set</a></li>
<li class="chapter" data-level="4.3" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality Control</a></li>
<li class="chapter" data-level="4.5" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#post-qc-check"><i class="fa fa-check"></i><b>4.5</b> Post QC Check</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html"><i class="fa fa-check"></i><b>5</b> Preparing data in Stacks</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#stacks-1"><i class="fa fa-check"></i><b>5.1</b> Stacks</a></li>
<li class="chapter" data-level="5.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#demultiplexing-data-with-process_radtags"><i class="fa fa-check"></i><b>5.2</b> Demultiplexing data with <code>process_radtags</code></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-barcodes"><i class="fa fa-check"></i><b>5.2.1</b> <code>process_radtags</code>: barcodes</a></li>
<li class="chapter" data-level="5.2.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-prep"><i class="fa fa-check"></i><b>5.2.2</b> <code>process_radtags</code>: prep</a></li>
<li class="chapter" data-level="5.2.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-run"><i class="fa fa-check"></i><b>5.2.3</b> <code>process_radtags</code>: run</a></li>
<li class="chapter" data-level="5.2.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-mcqs"><i class="fa fa-check"></i><b>5.2.4</b> <code>process_radtags</code>: MCQs</a></li>
<li class="chapter" data-level="5.2.5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-output"><i class="fa fa-check"></i><b>5.2.5</b> <code>process_radtags</code>: output</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#building-the-catalogue-loci-with-denovo_map.pl"><i class="fa fa-check"></i><b>5.3</b> Building the catalogue loci with <code>denovo_map.pl</code></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-input"><i class="fa fa-check"></i><b>5.3.1</b> <code>denovo_map.pl</code>: input</a></li>
<li class="chapter" data-level="5.3.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-run"><i class="fa fa-check"></i><b>5.3.2</b> <code>denovo_map.pl</code>: run</a></li>
<li class="chapter" data-level="5.3.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-output"><i class="fa fa-check"></i><b>5.3.3</b> <code>denovo_map.pl</code>: output</a></li>
<li class="chapter" data-level="5.3.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-mcqs"><i class="fa fa-check"></i><b>5.3.4</b> <code>denovo_map.pl</code>: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#preparing-a-sequence-alignment-file-in-populations"><i class="fa fa-check"></i><b>5.4</b> Preparing a sequence alignment file in <code>populations</code></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html"><i class="fa fa-check"></i><b>6</b> Model selection, tree reconstruction and support evaluation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#estimating-a-best-fit-model-of-nucleotide-or-sequence-evolution"><i class="fa fa-check"></i><b>6.1</b> Estimating a best-fit model of nucleotide or sequence evolution</a></li>
<li class="chapter" data-level="6.2" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#constructing-a-maximum-likelihood-phylogenetic-tree"><i class="fa fa-check"></i><b>6.2</b> Constructing a maximum-likelihood phylogenetic tree</a></li>
<li class="chapter" data-level="6.3" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#calculating-nodal-support"><i class="fa fa-check"></i><b>6.3</b> Calculating nodal support</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Topology_testing.html"><a href="07-Topology_testing.html"><i class="fa fa-check"></i><b>7</b> Hypothesis and topology testing</a></li>
<li class="chapter" data-level="8" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html"><i class="fa fa-check"></i><b>8</b> Dating phylogenies</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#setting-up-the-date-file"><i class="fa fa-check"></i><b>8.1</b> Setting up the date file</a></li>
<li class="chapter" data-level="8.2" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#running-the-analysis"><i class="fa fa-check"></i><b>8.2</b> Running the analysis</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#dating-our-tree"><i class="fa fa-check"></i><b>8.2.1</b> Dating our tree</a></li>
<li class="chapter" data-level="8.2.2" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#getting-confidence-intervals-around-our-dates"><i class="fa fa-check"></i><b>8.2.2</b> Getting confidence intervals around our dates</a></li>
<li class="chapter" data-level="8.2.3" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#lineage-through-time-plots"><i class="fa fa-check"></i><b>8.2.3</b> Lineage-Through-Time plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html"><i class="fa fa-check"></i><b>9</b> Species delimitation with the Multispecies Coalescent</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#beast2"><i class="fa fa-check"></i><b>9.1</b> beast2</a></li>
<li class="chapter" data-level="9.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#gene-tree---species-tree-conflicts"><i class="fa fa-check"></i><b>9.2</b> Gene tree - species tree conflicts</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis"><i class="fa fa-check"></i><b>9.2.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#starting-the-analysis"><i class="fa fa-check"></i><b>9.2.2</b> Starting the analysis</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#examining-the-results"><i class="fa fa-check"></i><b>9.2.3</b> Examining the results</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#bayes-factor-delimitation"><i class="fa fa-check"></i><b>9.3</b> Bayes Factor Delimitation</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis-1"><i class="fa fa-check"></i><b>9.3.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="9.3.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#running-the-analysis-1"><i class="fa fa-check"></i><b>9.3.2</b> Running the analysis</a></li>
<li class="chapter" data-level="9.3.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#selecting-the-taxonomic-hypothesis"><i class="fa fa-check"></i><b>9.3.3</b> Selecting the taxonomic hypothesis</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="10-Appendix.html"><a href="10-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="10-Appendix.html"><a href="10-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="10-Appendix.html"><a href="10-Appendix.html#loops"><i class="fa fa-check"></i><b>B</b> Wildcards and loops</a>
<ul>
<li class="chapter" data-level="B.1" data-path="10-Appendix.html"><a href="10-Appendix.html#a-simple-wildcard"><i class="fa fa-check"></i><b>B.1</b> A simple wildcard</a></li>
<li class="chapter" data-level="B.2" data-path="10-Appendix.html"><a href="10-Appendix.html#loops-1"><i class="fa fa-check"></i><b>B.2</b> Loops</a>
<ul>
<li class="chapter" data-level="B.2.1" data-path="10-Appendix.html"><a href="10-Appendix.html#an-example-of-a-simple-loop"><i class="fa fa-check"></i><b>B.2.1</b> An example of a simple loop</a></li>
<li class="chapter" data-level="B.2.2" data-path="10-Appendix.html"><a href="10-Appendix.html#stacks-process_radtags-loop-example"><i class="fa fa-check"></i><b>B.2.2</b> Stacks <code>process_radtags</code> loop example</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="C" data-path="10-Appendix.html"><a href="10-Appendix.html#optimising-loci-reconstruction-in-denovo_map.pl-in-stacks"><i class="fa fa-check"></i><b>C</b> Optimising loci reconstruction in <code>denovo_map.pl</code> in Stacks</a></li>
<li class="chapter" data-level="D" data-path="10-Appendix.html"><a href="10-Appendix.html#additional-resources"><i class="fa fa-check"></i><b>D</b> Additional resources</a>
<ul>
<li class="chapter" data-level="D.1" data-path="10-Appendix.html"><a href="10-Appendix.html#snp-variant-calling"><i class="fa fa-check"></i><b>D.1</b> SNP variant calling</a></li>
<li class="chapter" data-level="D.2" data-path="10-Appendix.html"><a href="10-Appendix.html#sequence-alignment-tools"><i class="fa fa-check"></i><b>D.2</b> Sequence alignment tools</a></li>
<li class="chapter" data-level="D.3" data-path="10-Appendix.html"><a href="10-Appendix.html#model-selection-tools"><i class="fa fa-check"></i><b>D.3</b> Model selection tools</a></li>
<li class="chapter" data-level="D.4" data-path="10-Appendix.html"><a href="10-Appendix.html#phylogeny-reconstruction-tools"><i class="fa fa-check"></i><b>D.4</b> Phylogeny reconstruction tools</a></li>
<li class="chapter" data-level="D.5" data-path="10-Appendix.html"><a href="10-Appendix.html#r-packages"><i class="fa fa-check"></i><b>D.5</b> R-packages</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Phylogenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="background" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Background<a href="02-Background.html#background" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/background.png" style="width:200px; border-radius:5px; background:white" />
</center>
<p>We will work through examples of typical phylogenomic and species delimitation analyses.
These can be performed using large datasets of genome-wide genetic variation. We will use Single Nucleotide Polymorphism (SNP) data, obtained through next-generation sequencing.</p>
<p>We will examine raw data, perform routine quality control on the sequences, and compile the SNP data for analysis.
We'll then proceed to:</p>
<ul>
<li>Construct, evaluate, manipulate, and test phylogenetic trees</li>
<li>Date phylogenies</li>
<li>Examine gene tree-species tree conflicts</li>
<li>Determine species boundaries in a multi-species coalescent framework</li>
</ul>
<div id="example-data-ddradseq-data-of-manta-and-devil-rays" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Example data: ddRADseq data of manta and devil rays<a href="02-Background.html#example-data-ddradseq-data-of-manta-and-devil-rays" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For all the exercises in the workbook, we will use various subsets of the data generated by <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/mec.15683">Hosegood <em>et al.</em> (2020)</a>. These authors used a reduced-representation, genome-wide sequencing approach (double-digest Restriction-site Associated DNA sequencing or "ddRAD") to generate SNP data to investigate manta rays and devil rays across their global distribution.</p>
<p>Their study aimed to:</p>
<ul>
<li>Determine the phylogenetic relationships among species groups</li>
<li>Test the validity of the current taxonomy</li>
<li>To examine species delimitation in order to identify meaningful taxonomic or biological units for conservation</li>
</ul>
<center>
<img src="figures/manta_ray.png" style="width:300px; border-radius:5px" />
</center>
<p>The study included 116 individuals, representing all species of <em>Mobula</em> known at the time of sampling, and an additional five representatives of the outgroup, <em>Rhinoptera bonasus</em>.
All data were generated using Illumina sequencing, with the study combining SNP data generated through a pilot MiSeq sequencing run with data generated in a larger HiSeq run.
The authors compiled two data sets, using different filtering thresholds.
This provided different levels of missing data, and 1,762 and 7,926 SNPs were available for analysis in each.</p>
<p>We'll take a more straightforward approach here and use various subsets of the original data set.
This will allow us to run analyses in a timely manner for the workshop.
We will demultiplex samples using sequence reads from one multiplexed sample pool, identify variants and prepare a sequence alignment using data from five samples.
Then we will proceed to construct and examine a phylogenetic tree of 22 individuals.
This will be followed by manipulating the tree to test alternative hypotheses and to date cladogenic events.
Finally, we'll look at two species delimitation analyses using a much smaller subset of the data.</p>
<p>Some of the figures from the Hosegood <em>et al.</em> (2020) study are presented below to give an indication of the taxonomic, geographic and analytical scope of their study, and of the results they reported.
The results we will produce in this workshop will differ because we will be using different subsets of the data, we will filter and treat the data differently, and - in some cases - will use slightly different approaches.</p>
<center>
<img src="figures/map_paper.png" style="width: 700px; border-radius: 5px; border: 3px solid white" />
</center>
<center>
<img src="figures/tree_paper.png" style="width: 700px; border-radius: 5px; border: 3px solid white" />
</center>
<center>
<img src="figures/snapp_paper.png" style="width: 700px; border-radius: 5px; border: 3px solid white" />
</center>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="01-Intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="03-Cluster_Introduction.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
