<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Quality Control | Phylogenomics</title>
  <meta name="description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Quality Control | Phylogenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Phylogenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Quality Control | Phylogenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Gavin Gouws, Matthew Gemmell, Ewan Harney, Helen Hipperson, Kathryn Maher" />


<meta name="date" content="2024-11-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="03-Cluster_Introduction.html"/>
<link rel="next" href="05_Preparing_data_with_Stacks.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Phylogenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-ddradseq-data-of-manta-and-devil-rays"><i class="fa fa-check"></i><b>2.1</b> Example data: ddRADseq data of manta and devil rays</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#manta--and-devil-ray-ddrad-data-set"><i class="fa fa-check"></i><b>4.2</b> Manta- and devil-ray ddRAD data set</a></li>
<li class="chapter" data-level="4.3" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality Control</a></li>
<li class="chapter" data-level="4.5" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#post-qc-check"><i class="fa fa-check"></i><b>4.5</b> Post QC Check</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html"><i class="fa fa-check"></i><b>5</b> Preparing data in Stacks</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#stacks-1"><i class="fa fa-check"></i><b>5.1</b> Stacks</a></li>
<li class="chapter" data-level="5.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#demultiplexing-data-with-process_radtags"><i class="fa fa-check"></i><b>5.2</b> Demultiplexing data with <code>process_radtags</code></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-barcodes"><i class="fa fa-check"></i><b>5.2.1</b> <code>process_radtags</code>: barcodes</a></li>
<li class="chapter" data-level="5.2.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-prep"><i class="fa fa-check"></i><b>5.2.2</b> <code>process_radtags</code>: prep</a></li>
<li class="chapter" data-level="5.2.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-run"><i class="fa fa-check"></i><b>5.2.3</b> <code>process_radtags</code>: run</a></li>
<li class="chapter" data-level="5.2.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-mcqs"><i class="fa fa-check"></i><b>5.2.4</b> <code>process_radtags</code>: MCQs</a></li>
<li class="chapter" data-level="5.2.5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-output"><i class="fa fa-check"></i><b>5.2.5</b> <code>process_radtags</code>: output</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#building-the-catalogue-loci-with-denovo_map.pl"><i class="fa fa-check"></i><b>5.3</b> Building the catalogue loci with <code>denovo_map.pl</code></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-input"><i class="fa fa-check"></i><b>5.3.1</b> <code>denovo_map.pl</code>: input</a></li>
<li class="chapter" data-level="5.3.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-run"><i class="fa fa-check"></i><b>5.3.2</b> <code>denovo_map.pl</code>: run</a></li>
<li class="chapter" data-level="5.3.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-output"><i class="fa fa-check"></i><b>5.3.3</b> <code>denovo_map.pl</code>: output</a></li>
<li class="chapter" data-level="5.3.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-mcqs"><i class="fa fa-check"></i><b>5.3.4</b> <code>denovo_map.pl</code>: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#preparing-a-sequence-alignment-file-in-populations"><i class="fa fa-check"></i><b>5.4</b> Preparing a sequence alignment file in <code>populations</code></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html"><i class="fa fa-check"></i><b>6</b> Model selection, tree reconstruction and support evaluation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#estimating-a-best-fit-model-of-nucleotide-or-sequence-evolution"><i class="fa fa-check"></i><b>6.1</b> Estimating a best-fit model of nucleotide or sequence evolution</a></li>
<li class="chapter" data-level="6.2" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#constructing-a-maximum-likelihood-phylogenetic-tree"><i class="fa fa-check"></i><b>6.2</b> Constructing a maximum-likelihood phylogenetic tree</a></li>
<li class="chapter" data-level="6.3" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#calculating-nodal-support"><i class="fa fa-check"></i><b>6.3</b> Calculating nodal support</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Topology_testing.html"><a href="07-Topology_testing.html"><i class="fa fa-check"></i><b>7</b> Hypothesis and topology testing</a></li>
<li class="chapter" data-level="8" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html"><i class="fa fa-check"></i><b>8</b> Dating phylogenies</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#setting-up-the-date-file"><i class="fa fa-check"></i><b>8.1</b> Setting up the date file</a></li>
<li class="chapter" data-level="8.2" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#running-the-analysis"><i class="fa fa-check"></i><b>8.2</b> Running the analysis</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#dating-our-tree"><i class="fa fa-check"></i><b>8.2.1</b> Dating our tree</a></li>
<li class="chapter" data-level="8.2.2" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#getting-confidence-intervals-around-our-dates"><i class="fa fa-check"></i><b>8.2.2</b> Getting confidence intervals around our dates</a></li>
<li class="chapter" data-level="8.2.3" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#lineage-through-time-plots"><i class="fa fa-check"></i><b>8.2.3</b> Lineage-Through-Time plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html"><i class="fa fa-check"></i><b>9</b> Species delimitation with the Multispecies Coalescent</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#beast2"><i class="fa fa-check"></i><b>9.1</b> beast2</a></li>
<li class="chapter" data-level="9.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#gene-tree---species-tree-conflicts"><i class="fa fa-check"></i><b>9.2</b> Gene tree - species tree conflicts</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis"><i class="fa fa-check"></i><b>9.2.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#starting-the-analysis"><i class="fa fa-check"></i><b>9.2.2</b> Starting the analysis</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#examining-the-results"><i class="fa fa-check"></i><b>9.2.3</b> Examining the results</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#bayes-factor-delimitation"><i class="fa fa-check"></i><b>9.3</b> Bayes Factor Delimitation</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis-1"><i class="fa fa-check"></i><b>9.3.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="9.3.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#running-the-analysis-1"><i class="fa fa-check"></i><b>9.3.2</b> Running the analysis</a></li>
<li class="chapter" data-level="9.3.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#selecting-the-taxonomic-hypothesis"><i class="fa fa-check"></i><b>9.3.3</b> Selecting the taxonomic hypothesis</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="10-Appendix.html"><a href="10-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="10-Appendix.html"><a href="10-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="10-Appendix.html"><a href="10-Appendix.html#loops"><i class="fa fa-check"></i><b>B</b> Wildcards and loops</a>
<ul>
<li class="chapter" data-level="B.1" data-path="10-Appendix.html"><a href="10-Appendix.html#a-simple-wildcard"><i class="fa fa-check"></i><b>B.1</b> A simple wildcard</a></li>
<li class="chapter" data-level="B.2" data-path="10-Appendix.html"><a href="10-Appendix.html#loops-1"><i class="fa fa-check"></i><b>B.2</b> Loops</a>
<ul>
<li class="chapter" data-level="B.2.1" data-path="10-Appendix.html"><a href="10-Appendix.html#an-example-of-a-simple-loop"><i class="fa fa-check"></i><b>B.2.1</b> An example of a simple loop</a></li>
<li class="chapter" data-level="B.2.2" data-path="10-Appendix.html"><a href="10-Appendix.html#stacks-process_radtags-loop-example"><i class="fa fa-check"></i><b>B.2.2</b> Stacks <code>process_radtags</code> loop example</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="C" data-path="10-Appendix.html"><a href="10-Appendix.html#optimising-loci-reconstruction-in-denovo_map.pl-in-stacks"><i class="fa fa-check"></i><b>C</b> Optimising loci reconstruction in <code>denovo_map.pl</code> in Stacks</a></li>
<li class="chapter" data-level="D" data-path="10-Appendix.html"><a href="10-Appendix.html#additional-resources"><i class="fa fa-check"></i><b>D</b> Additional resources</a>
<ul>
<li class="chapter" data-level="D.1" data-path="10-Appendix.html"><a href="10-Appendix.html#snp-variant-calling"><i class="fa fa-check"></i><b>D.1</b> SNP variant calling</a></li>
<li class="chapter" data-level="D.2" data-path="10-Appendix.html"><a href="10-Appendix.html#sequence-alignment-tools"><i class="fa fa-check"></i><b>D.2</b> Sequence alignment tools</a></li>
<li class="chapter" data-level="D.3" data-path="10-Appendix.html"><a href="10-Appendix.html#model-selection-tools"><i class="fa fa-check"></i><b>D.3</b> Model selection tools</a></li>
<li class="chapter" data-level="D.4" data-path="10-Appendix.html"><a href="10-Appendix.html#phylogeny-reconstruction-tools"><i class="fa fa-check"></i><b>D.4</b> Phylogeny reconstruction tools</a></li>
<li class="chapter" data-level="D.5" data-path="10-Appendix.html"><a href="10-Appendix.html#r-packages"><i class="fa fa-check"></i><b>D.5</b> R-packages</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Phylogenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quality" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Quality Control<a href="04-Quality_Control.html#quality" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/magnify_glass.png" style="width:200px" />
</center>
<p>This tutorial will give hands-on experience with the quality control of Illumina sequence data. We will first assess the quality of the RADseq data produced through an Illumina sequencing run before proceeding to some quality control.</p>
<div id="workshop-data" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Workshop data<a href="04-Quality_Control.html#workshop-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before we can carry on with this section, navigate into the sub-directory where the raw data are located.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb5-1"><a href="04-Quality_Control.html#cb5-1" tabindex="-1"></a><span class="co"># Check that you are still in the &#39;phylogenomics&#39; directory</span></span>
<span id="cb5-2"><a href="04-Quality_Control.html#cb5-2" tabindex="-1"></a><span class="bu">pwd</span></span>
<span id="cb5-3"><a href="04-Quality_Control.html#cb5-3" tabindex="-1"></a><span class="co"># Then navigate to the sub-directory</span></span>
<span id="cb5-4"><a href="04-Quality_Control.html#cb5-4" tabindex="-1"></a><span class="bu">cd</span> raw_data</span>
<span id="cb5-5"><a href="04-Quality_Control.html#cb5-5" tabindex="-1"></a><span class="co"># You can navigate to the sub-directory directly, regardless of where you are, with this command</span></span>
<span id="cb5-6"><a href="04-Quality_Control.html#cb5-6" tabindex="-1"></a><span class="bu">cd</span> ~/phylogenomics/raw_data</span></code></pre></div>
</div>
<div id="manta--and-devil-ray-ddrad-data-set" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Manta- and devil-ray ddRAD data set<a href="04-Quality_Control.html#manta--and-devil-ray-ddrad-data-set" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/manta_icon.png" style="width:300px" />
</center>
<p>This dataset contains one set of sequence files, the forward (R1) and reverse (R2) read files of 'Lib1_001_', that were demultiplexed from the paired-end Illumina sequencing run.
These are compressed (gzipped) fastq files, as indicated by the '.gz' extension.
The files contain sequence reads of many RADtags (double-digested DNA sequence fragments) that were isolated and sequenced from a number of individuals of different manta ray and devil ray species, and multiplexed (combined) into a single pool for sequencing.
These are some of the original raw data from Hosegood <em>et al.</em> (2020), but the files have been randomly "down-sampled" (they contain about 10% of the original data) to enable us to complete the exercises in a timely manner.</p>
<p>List the contents of the current directory to confirm that the two sequence files are there. There should also be two additional files ('manta_samples.barcodes' and 'TruSeq3-PE.fa'), which we will use later.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb6-1"><a href="04-Quality_Control.html#cb6-1" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>Have a look at the structure of the reverse (R2) fastq file.</p>
<p><strong>Remember:</strong> You can use 'Tab' to auto-complete and save yourself the typing.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb7-1"><a href="04-Quality_Control.html#cb7-1" tabindex="-1"></a><span class="fu">zcat</span> Lib1_001_R2.fastq.gz <span class="kw">|</span> <span class="fu">head</span> </span></code></pre></div>
<p><code>zcat</code> is a command for viewing the contents of compressed files (equivalent to using <code>cat</code> for other files).
The pipe <code>|</code> symbol then passes this output to <code>head</code> to display only the first few lines of the file.</p>
<p>Fastq files contain a sequence header line, the nucleotide sequence, a quality header line (containing only a '+' symbol) and the quality scores for each nucleotide.</p>
<p>We have opened the R2 file, because there is an important detail that is not present in the R1 file, which we will come back to later. Note and remember the 'GTATGCCG' sequence string at the end of the sequence header line.</p>
</div>
<div id="quality-assessment" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Quality assessment<a href="04-Quality_Control.html#quality-assessment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/fastqc_icon.png" style="max-width: 150px; border-radius: 15px" />
</center>
<p>We’ll run this raw sequence data through <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastQC</a>, a tool for conducting quality assessments on Illumina data, to summarise the data quality.</p>
<p>First, make a directory for the output.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb8-1"><a href="04-Quality_Control.html#cb8-1" tabindex="-1"></a><span class="fu">mkdir</span> fastqc_output</span></code></pre></div>
<p>Now we can run <strong>fastQC</strong>.</p>
<p><strong>Note</strong>:</p>
<ul>
<li>The command below (and others in the workbook) can be run over one line by excluding the <code>\</code> (a bash escape)</li>
<li>Alternatively, you can type <code>\</code> (with a mandatory preceding space) during a command and then press the enter key. The next line on the command line will start with '&gt;'. This will allow you to continue typing the command on the line. This can be used to type one command over multiple lines, which is useful for clarity and for long commands. Please ask if you would like a demonstration of the bash escape.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb9-1"><a href="04-Quality_Control.html#cb9-1" tabindex="-1"></a><span class="ex">fastqc</span> <span class="at">-o</span> fastqc_output <span class="dt">\</span></span>
<span id="cb9-2"><a href="04-Quality_Control.html#cb9-2" tabindex="-1"></a>Lib1_001_R1.fastq.gz Lib1_001_R2.fastq.gz</span></code></pre></div>
<p>Normally, when you run a command it will run the process in the foreground. This means your command line will be locked while the process is running. You will only be able to type a new command in the terminal once the foreground process is finished.</p>
<p>This is normally wanted, but not always - for example, when you run a process like firefox. To run a process in the background, so that you can type and run new commands, you can add the symbol <code>&amp;</code> to the end of a command. We will use this for running firefox.</p>
<p>Please follow <a href="https://linuxconfig.org/understanding-foreground-and-background-linux-processes">this link</a> for more information on foreground and background processing.</p>
<center>
<img src="figures/magnifying_glass_blue.png" style="width:200px" />
</center>
<p>Using firefox, have a look at the output html reports and answer the questions further below.</p>
<p>To look at the R1 fastQC output:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb10-1"><a href="04-Quality_Control.html#cb10-1" tabindex="-1"></a><span class="ex">firefox</span> fastqc_output/Lib1_001_R1_fastqc.html <span class="kw">&amp;</span></span></code></pre></div>
<p>You may see a warning message on your terminal, but you can press enter to continue to type in the terminal.</p>
<p>To look at the R2 fastQC output:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb11-1"><a href="04-Quality_Control.html#cb11-1" tabindex="-1"></a><span class="ex">firefox</span> fastqc_output/Lib1_001_R2_fastqc.html <span class="kw">&amp;</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>How many total reads are there in each of the files?
<div id="radio_JJIERMILPZ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_JJIERMILPZ" value=""></input> <span>53,260</span></label><label><input type="radio" autocomplete="off" name="radio_JJIERMILPZ" value=""></input> <span>165,017</span></label><label><input type="radio" autocomplete="off" name="radio_JJIERMILPZ" value="answer"></input> <span>250,000</span></label>
</div></li>
<li>What is the length of the reads?
<div id="radio_XTPWCLWSLR" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_XTPWCLWSLR" value=""></input> <span>75</span></label><label><input type="radio" autocomplete="off" name="radio_XTPWCLWSLR" value=""></input> <span>100</span></label><label><input type="radio" autocomplete="off" name="radio_XTPWCLWSLR" value="answer"></input> <span>125</span></label><label><input type="radio" autocomplete="off" name="radio_XTPWCLWSLR" value=""></input> <span>151</span></label>
</div></li>
<li>In the read 1 file, what is the last position where each read still has a quality above 28?
<div id="radio_QJLTGWXXFN" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_QJLTGWXXFN" value="answer"></input> <span>84-85</span></label><label><input type="radio" autocomplete="off" name="radio_QJLTGWXXFN" value=""></input> <span>90-91</span></label><label><input type="radio" autocomplete="off" name="radio_QJLTGWXXFN" value=""></input> <span>104-105</span></label><label><input type="radio" autocomplete="off" name="radio_QJLTGWXXFN" value=""></input> <span>It does not</span></label>
</div></li>
<li>In the read 2 file, at what base position does the IQR of all the reads go below 28 (i.e., the position where not all of the box plot, ignoring the whiskers, is in the green)?
<div id="radio_DGVRQUYBFD" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_DGVRQUYBFD" value=""></input> <span>104-105</span></label><label><input type="radio" autocomplete="off" name="radio_DGVRQUYBFD" value=""></input> <span>114-115</span></label><label><input type="radio" autocomplete="off" name="radio_DGVRQUYBFD" value=""></input> <span>124-125</span></label><label><input type="radio" autocomplete="off" name="radio_DGVRQUYBFD" value="answer"></input> <span>It does not</span></label>
</div></li>
<li>Are there any over-represented sequences or sequences with adapter content?
<div id="radio_IDYTXEWCFK" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_IDYTXEWCFK" value="answer"></input> <span>Yes</span></label><label><input type="radio" autocomplete="off" name="radio_IDYTXEWCFK" value=""></input> <span>No</span></label>
</div></li>
</ol>
<p>In this case, the reads seem to be of generally very high quality.
This is likely as these reads are still relatively short, considering the Illumina sequencing kits used.
Longer sequences can be of poorer quality, especially towards the end of the reads, and reads from the sequencing of shorter fragments may contain adapter sequence.</p>
<p>Unfortunately, data from a sequencing run may often not look as nice and will require quality trimming and filtering.</p>
<p>For most datasets:</p>
<ul>
<li>The quality decreases towards the end of the reads</li>
<li>The R2 reads are of poorer quality than the R1 reads</li>
<li>Depending on what has been sequenced and the sequencing kit used, the reads may be a range of sizes compared to being the same size. However, most of the reads will be towards the long end of the range.</li>
</ul>
<p>Even if data does look very good, we would carry out quality control to remove any poor data masked by the high-quality data and to remove any adapter sequences.</p>
<p>The application we will use in the next chapter has an internal quality filter.
However, it is good practice to work through QC steps like these to get a feel for what you may lose from your data set prior to processing it further.
This may put your mind at ease, indicate that you may need to proceed with caution or that you should perhaps revisit your libraries if this was a pilot sequencing run.</p>
</div>
<div id="quality-control" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Quality Control<a href="04-Quality_Control.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/quality_trimming_and_filtering.png" style="width:300px" />
</center>
<p>Quality control generally comes in two forms:</p>
<ol style="list-style-type: decimal">
<li><strong>Trimming</strong>: This involves cutting off bits of sequence. This is typically in the form of trimming off low quality bases from the end of reads, and trimming off adapters or primers at the start of reads.</li>
<li><strong>Filtering</strong>: This occurs when entire reads are removed. A typical occurrence of this is when a read is too short, as we do not want reads below a certain length, or if the read has an excess of uncertain or poor quality base-calls.</li>
</ol>
<center>
<img src="figures/hedge_trimming.png" style="width:200px" />
</center>
<p>To carry this out, we are going to use <a href="https://github.com/usadellab/Trimmomatic"><strong>trimmomatic</strong></a>.</p>
<p>To run trimmomatic with the above sequence data we will use the command below.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb12-1"><a href="04-Quality_Control.html#cb12-1" tabindex="-1"></a><span class="ex">trimmomatic</span> <span class="dt">\</span></span>
<span id="cb12-2"><a href="04-Quality_Control.html#cb12-2" tabindex="-1"></a>  PE <span class="at">-phred33</span> <span class="dt">\</span></span>
<span id="cb12-3"><a href="04-Quality_Control.html#cb12-3" tabindex="-1"></a>  Lib1_001_R1.fastq.gz Lib1_001_R2.fastq.gz <span class="dt">\</span></span>
<span id="cb12-4"><a href="04-Quality_Control.html#cb12-4" tabindex="-1"></a>  Lib1_1_out_paired.fastq Lib1_1_out_unpaired.fastq <span class="dt">\</span></span>
<span id="cb12-5"><a href="04-Quality_Control.html#cb12-5" tabindex="-1"></a>  Lib1_2_out_paired.fastq Lib1_2_out_unpaired.fastq <span class="dt">\</span></span>
<span id="cb12-6"><a href="04-Quality_Control.html#cb12-6" tabindex="-1"></a>  ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 <span class="dt">\</span></span>
<span id="cb12-7"><a href="04-Quality_Control.html#cb12-7" tabindex="-1"></a>  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:30 MINLEN:50</span></code></pre></div>
<p>The parameters in the bash script are:</p>
<ul>
<li><code>PE</code>: Our input data consists of paired-end reads.</li>
<li><code>-phred33</code>: Type of Phred quality encoding. In this case, it is phred33 as our fastq files (like most Illumina data) use the Sanger encoding.</li>
<li>Next, the raw input forward (1) and then reverse (2) read files that we want to perform quality control on are specified.</li>
<li>On the next line, the output files for read 1 are specified, first for paired reads and second for unpaired reads. Unpaired reads come about when one read from a pair is filtered out but the other one is not. We normally ignore this unpaired file after the trimming. However, we must specify this for trimmomatic to run.</li>
<li>Next, the output files for read 2 are specified, first paired then unpaired.</li>
<li><code>ILLUMINACLIP</code>: These settings are used to find and remove Illumina adapters. First, a fasta file of known adapter sequences is given ('TruSeq3-PE.fa'), followed by the number of mismatches allowed between the adapter and read sequence and then thresholds for how accurate we want the alignment between the adapter and read sequence to be. The 'TruSeq3-PE.fa' file is provided in your 'raw_data' directory.</li>
<li><code>LEADING</code>: The minimum quality value required to keep a base at the start of the read.</li>
<li><code>TRAILING</code>: The minimum quality value required to keep a base at the end of the read.</li>
<li><code>SLIDINGWINDOW</code>: This specifies to scan the read quality over a given window (4 bp here), cutting when the average quality is below a certain Phred score (30 here).</li>
<li><code>MINLEN</code>: This specifies the minimum length of a read to keep; any reads shorter than 50 bp are discarded.</li>
</ul>
<p>Once the command is run, it will give you some summary output to the screen.</p>
Based on this output, approximately what percentage of paired reads were kept?
<div id="radio_SZWBXCJDAE" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_SZWBXCJDAE" value=""></input> <span>95</span></label><label><input type="radio" autocomplete="off" name="radio_SZWBXCJDAE" value="answer"></input> <span>66</span></label><label><input type="radio" autocomplete="off" name="radio_SZWBXCJDAE" value=""></input> <span>15</span></label><label><input type="radio" autocomplete="off" name="radio_SZWBXCJDAE" value=""></input> <span>5</span></label>
</div>
</div>
<div id="post-qc-check" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Post QC Check<a href="04-Quality_Control.html#post-qc-check" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/magnify_glass_good.png" style="width:200px" />
</center>
<p>To see the impact of our quality control, we will run fastQC on the trimmomatic output, specifically the two paired read files, and send the output to the 'fastqc_out' directory we created earlier.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb13-1"><a href="04-Quality_Control.html#cb13-1" tabindex="-1"></a><span class="ex">fastqc</span> <span class="at">-o</span> fastqc_output <span class="dt">\</span></span>
<span id="cb13-2"><a href="04-Quality_Control.html#cb13-2" tabindex="-1"></a>Lib1_1_out_paired.fastq Lib1_2_out_paired.fastq</span></code></pre></div>
<p>We can open the fastQC reports for both quality-controlled R1 and R2 files with this command:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb14-1"><a href="04-Quality_Control.html#cb14-1" tabindex="-1"></a><span class="ex">firefox</span> fastqc_output/Lib1_<span class="pp">*</span>_paired_fastqc.html <span class="kw">&amp;</span></span></code></pre></div>
<p>To see if the quality of the reads has improved and to evaluate what we have lost, let’s answer the questions below and compare these to the answers to the questions about the fastQC raw read assessment above.</p>
<ol style="list-style-type: decimal">
<li>How many total reads are there in each of the files?
<div id="radio_ZVIXELETCG" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ZVIXELETCG" value=""></input> <span>53,260</span></label><label><input type="radio" autocomplete="off" name="radio_ZVIXELETCG" value="answer"></input> <span>165,017</span></label><label><input type="radio" autocomplete="off" name="radio_ZVIXELETCG" value=""></input> <span>250,000</span></label><label><input type="radio" autocomplete="off" name="radio_ZVIXELETCG" value=""></input> <span>513,778</span></label>
</div></li>
<li>What is the length of the reads?
<div id="radio_DJRQOIKBXZ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_DJRQOIKBXZ" value=""></input> <span>35-76</span></label><label><input type="radio" autocomplete="off" name="radio_DJRQOIKBXZ" value=""></input> <span>50-76</span></label><label><input type="radio" autocomplete="off" name="radio_DJRQOIKBXZ" value=""></input> <span>75-100</span></label><label><input type="radio" autocomplete="off" name="radio_DJRQOIKBXZ" value="answer"></input> <span>50-125</span></label>
</div></li>
<li>What is the average GC content (%) of the R2 reads?
<div id="radio_VABEVVVQZM" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_VABEVVVQZM" value=""></input> <span>40</span></label><label><input type="radio" autocomplete="off" name="radio_VABEVVVQZM" value="answer"></input> <span>43</span></label><label><input type="radio" autocomplete="off" name="radio_VABEVVVQZM" value=""></input> <span>45</span></label><label><input type="radio" autocomplete="off" name="radio_VABEVVVQZM" value=""></input> <span>50</span></label>
</div></li>
</ol>
<p>Some things to note:</p>
<ul>
<li>There is now a range of read lengths. This reflects the trimming of regions of poor quality at the ends of reads through the sliding window. Depending on the downstream application, you may need to truncate reads to an equal length. We won't do that here, as we can still construct loci with fragments that are overlapping between the R1 and R2 reads. Shorter reads may not overlap, but we may still be able to "stack" them within a RADlocus to assemble alleles and loci (see the next section).</li>
<li>The number of reads has significantly decreased due to quality control. This is expected, but will need to be taken into account. If you do not have enough reads to create sufficient depth or coverage to confidently reconstruct loci and call variants, you will need to be less stringent with the quality filtering you implement in the next section. What is the minimum number of reads required for downstream analysis? Unfortunately, this depends on what you intend to do with your sequence data. For genome assemblies, the general rule of thumb is you want at least 20 times as many bases in the reads as would be in the final assembly (based on the genome size of the organism). For calling SNPs, you need less coverage of the genome, but you do want a sufficiently large enough number of reads of the same RADtag to be able to confidently reconstruct your loci and genotype your samples.</li>
<li>R2 quality is still slightly worse than R1 quality towards the end of the reads. Again, this is normal, and is mostly an artifact of the Illumina chemistry. You could be more stringent, but you may be removing too many reads (including potentially good forward reads).</li>
</ul>
<center>
<img src="figures/loop.png" style="width:200px; background: white; border-radius:5px" />
</center>
<p>In this section, we ran fastQC and trimmomatic on a single set of paired read files. The output of a sequencing run on your library may produce several paired read-files, as would the sequencing of multiple libraries. You may wonder how you would run fastQC and trimmomatic on multiple sets of fastq files. The use of wildcards and loops to accomplish this is explained in the Appendix.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="03-Cluster_Introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="05_Preparing_data_with_Stacks.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
