<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Preparing data in Stacks | Phylogenomics</title>
  <meta name="description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Preparing data in Stacks | Phylogenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Phylogenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Preparing data in Stacks | Phylogenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Gavin Gouws, Matthew Gemmell, Ewan Harney, Helen Hipperson, Kathryn Maher" />


<meta name="date" content="2024-11-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="04-Quality_Control.html"/>
<link rel="next" href="06-Phylogeny.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Phylogenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-ddradseq-data-of-manta-and-devil-rays"><i class="fa fa-check"></i><b>2.1</b> Example data: ddRADseq data of manta and devil rays</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#manta--and-devil-ray-ddrad-data-set"><i class="fa fa-check"></i><b>4.2</b> Manta- and devil-ray ddRAD data set</a></li>
<li class="chapter" data-level="4.3" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality Control</a></li>
<li class="chapter" data-level="4.5" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#post-qc-check"><i class="fa fa-check"></i><b>4.5</b> Post QC Check</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html"><i class="fa fa-check"></i><b>5</b> Preparing data in Stacks</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#stacks-1"><i class="fa fa-check"></i><b>5.1</b> Stacks</a></li>
<li class="chapter" data-level="5.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#demultiplexing-data-with-process_radtags"><i class="fa fa-check"></i><b>5.2</b> Demultiplexing data with <code>process_radtags</code></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-barcodes"><i class="fa fa-check"></i><b>5.2.1</b> <code>process_radtags</code>: barcodes</a></li>
<li class="chapter" data-level="5.2.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-prep"><i class="fa fa-check"></i><b>5.2.2</b> <code>process_radtags</code>: prep</a></li>
<li class="chapter" data-level="5.2.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-run"><i class="fa fa-check"></i><b>5.2.3</b> <code>process_radtags</code>: run</a></li>
<li class="chapter" data-level="5.2.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-mcqs"><i class="fa fa-check"></i><b>5.2.4</b> <code>process_radtags</code>: MCQs</a></li>
<li class="chapter" data-level="5.2.5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-output"><i class="fa fa-check"></i><b>5.2.5</b> <code>process_radtags</code>: output</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#building-the-catalogue-loci-with-denovo_map.pl"><i class="fa fa-check"></i><b>5.3</b> Building the catalogue loci with <code>denovo_map.pl</code></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-input"><i class="fa fa-check"></i><b>5.3.1</b> <code>denovo_map.pl</code>: input</a></li>
<li class="chapter" data-level="5.3.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-run"><i class="fa fa-check"></i><b>5.3.2</b> <code>denovo_map.pl</code>: run</a></li>
<li class="chapter" data-level="5.3.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-output"><i class="fa fa-check"></i><b>5.3.3</b> <code>denovo_map.pl</code>: output</a></li>
<li class="chapter" data-level="5.3.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-mcqs"><i class="fa fa-check"></i><b>5.3.4</b> <code>denovo_map.pl</code>: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#preparing-a-sequence-alignment-file-in-populations"><i class="fa fa-check"></i><b>5.4</b> Preparing a sequence alignment file in <code>populations</code></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html"><i class="fa fa-check"></i><b>6</b> Model selection, tree reconstruction and support evaluation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#estimating-a-best-fit-model-of-nucleotide-or-sequence-evolution"><i class="fa fa-check"></i><b>6.1</b> Estimating a best-fit model of nucleotide or sequence evolution</a></li>
<li class="chapter" data-level="6.2" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#constructing-a-maximum-likelihood-phylogenetic-tree"><i class="fa fa-check"></i><b>6.2</b> Constructing a maximum-likelihood phylogenetic tree</a></li>
<li class="chapter" data-level="6.3" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#calculating-nodal-support"><i class="fa fa-check"></i><b>6.3</b> Calculating nodal support</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Topology_testing.html"><a href="07-Topology_testing.html"><i class="fa fa-check"></i><b>7</b> Hypothesis and topology testing</a></li>
<li class="chapter" data-level="8" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html"><i class="fa fa-check"></i><b>8</b> Dating phylogenies</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#setting-up-the-date-file"><i class="fa fa-check"></i><b>8.1</b> Setting up the date file</a></li>
<li class="chapter" data-level="8.2" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#running-the-analysis"><i class="fa fa-check"></i><b>8.2</b> Running the analysis</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#dating-our-tree"><i class="fa fa-check"></i><b>8.2.1</b> Dating our tree</a></li>
<li class="chapter" data-level="8.2.2" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#getting-confidence-intervals-around-our-dates"><i class="fa fa-check"></i><b>8.2.2</b> Getting confidence intervals around our dates</a></li>
<li class="chapter" data-level="8.2.3" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#lineage-through-time-plots"><i class="fa fa-check"></i><b>8.2.3</b> Lineage-Through-Time plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html"><i class="fa fa-check"></i><b>9</b> Species delimitation with the Multispecies Coalescent</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#beast2"><i class="fa fa-check"></i><b>9.1</b> beast2</a></li>
<li class="chapter" data-level="9.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#gene-tree---species-tree-conflicts"><i class="fa fa-check"></i><b>9.2</b> Gene tree - species tree conflicts</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis"><i class="fa fa-check"></i><b>9.2.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#starting-the-analysis"><i class="fa fa-check"></i><b>9.2.2</b> Starting the analysis</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#examining-the-results"><i class="fa fa-check"></i><b>9.2.3</b> Examining the results</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#bayes-factor-delimitation"><i class="fa fa-check"></i><b>9.3</b> Bayes Factor Delimitation</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis-1"><i class="fa fa-check"></i><b>9.3.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="9.3.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#running-the-analysis-1"><i class="fa fa-check"></i><b>9.3.2</b> Running the analysis</a></li>
<li class="chapter" data-level="9.3.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#selecting-the-taxonomic-hypothesis"><i class="fa fa-check"></i><b>9.3.3</b> Selecting the taxonomic hypothesis</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="10-Appendix.html"><a href="10-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="10-Appendix.html"><a href="10-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="10-Appendix.html"><a href="10-Appendix.html#loops"><i class="fa fa-check"></i><b>B</b> Wildcards and loops</a>
<ul>
<li class="chapter" data-level="B.1" data-path="10-Appendix.html"><a href="10-Appendix.html#a-simple-wildcard"><i class="fa fa-check"></i><b>B.1</b> A simple wildcard</a></li>
<li class="chapter" data-level="B.2" data-path="10-Appendix.html"><a href="10-Appendix.html#loops-1"><i class="fa fa-check"></i><b>B.2</b> Loops</a>
<ul>
<li class="chapter" data-level="B.2.1" data-path="10-Appendix.html"><a href="10-Appendix.html#an-example-of-a-simple-loop"><i class="fa fa-check"></i><b>B.2.1</b> An example of a simple loop</a></li>
<li class="chapter" data-level="B.2.2" data-path="10-Appendix.html"><a href="10-Appendix.html#stacks-process_radtags-loop-example"><i class="fa fa-check"></i><b>B.2.2</b> Stacks <code>process_radtags</code> loop example</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="C" data-path="10-Appendix.html"><a href="10-Appendix.html#optimising-loci-reconstruction-in-denovo_map.pl-in-stacks"><i class="fa fa-check"></i><b>C</b> Optimising loci reconstruction in <code>denovo_map.pl</code> in Stacks</a></li>
<li class="chapter" data-level="D" data-path="10-Appendix.html"><a href="10-Appendix.html#additional-resources"><i class="fa fa-check"></i><b>D</b> Additional resources</a>
<ul>
<li class="chapter" data-level="D.1" data-path="10-Appendix.html"><a href="10-Appendix.html#snp-variant-calling"><i class="fa fa-check"></i><b>D.1</b> SNP variant calling</a></li>
<li class="chapter" data-level="D.2" data-path="10-Appendix.html"><a href="10-Appendix.html#sequence-alignment-tools"><i class="fa fa-check"></i><b>D.2</b> Sequence alignment tools</a></li>
<li class="chapter" data-level="D.3" data-path="10-Appendix.html"><a href="10-Appendix.html#model-selection-tools"><i class="fa fa-check"></i><b>D.3</b> Model selection tools</a></li>
<li class="chapter" data-level="D.4" data-path="10-Appendix.html"><a href="10-Appendix.html#phylogeny-reconstruction-tools"><i class="fa fa-check"></i><b>D.4</b> Phylogeny reconstruction tools</a></li>
<li class="chapter" data-level="D.5" data-path="10-Appendix.html"><a href="10-Appendix.html#r-packages"><i class="fa fa-check"></i><b>D.5</b> R-packages</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Phylogenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="stacks" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Preparing data in Stacks<a href="05_Preparing_data_with_Stacks.html#stacks" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="stacks-1" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Stacks<a href="05_Preparing_data_with_Stacks.html#stacks-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/stacks_logo.png" style="max-width: 400px; border-radius: 15px" />
</center>
<p><a href="https://catchenlab.life.illinois.edu/stacks/">Stacks</a> is a very popular and widely-used application for:</p>
<ul>
<li>Building loci from short-read sequence data</li>
<li>Identifying Single Nucleotide Polymorphisms (SNPs)</li>
<li>Genotyping individuals for population genomic and pedigree/relatedness studies</li>
<li>Building linkage maps</li>
</ul>
<p>It can also prepare SNP data for phylogenetic analyses, as we will do here.</p>
<p>Stacks was principally developed for Illumina RADseq data, but can support other GBS or re-sequencing data sets, and (with a little manipulation) data from other platforms.
It incorporates many elements of the typical tools (like bwa, SAMtools, bcftools and vcftools) used in other SNP/variant calling pipelines in a simple, single-access application.
It also offers the choice of assembling loci using a reference genome, through a <em>de novo</em> assembly procedure (which we will use below) or through a hybrid of the two.</p>
<p>We will use various subsets of the data from the Hosegood <em>et al.</em> (2020) paper to demonstrate demultiplexing the data produced by the Illumina sequencing run, reconstructing loci and identifying variants, and preparing the SNP data for phylogenetic analyses.</p>
</div>
<div id="demultiplexing-data-with-process_radtags" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Demultiplexing data with <code>process_radtags</code><a href="05_Preparing_data_with_Stacks.html#demultiplexing-data-with-process_radtags" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/sort.png" style="width:200px; background: white; border-radius:5px" />
</center>
<p>We will take a quick look in the directory we worked in previously to confirm that the files we need for Stacks are there:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb15-1"><a href="05_Preparing_data_with_Stacks.html#cb15-1" tabindex="-1"></a><span class="bu">cd</span> ~/phylogenomics</span>
<span id="cb15-2"><a href="05_Preparing_data_with_Stacks.html#cb15-2" tabindex="-1"></a><span class="co"># List the files that are present</span></span>
<span id="cb15-3"><a href="05_Preparing_data_with_Stacks.html#cb15-3" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-l</span> raw_data/</span></code></pre></div>
<p>Along with the output you generated in the previous section, you should still have the two sequence data files, the read 1 and read 2 ('Lib1_001') files with '.fastq.gz' extensions, and the barcode file ('manta_samples.barcodes'; see below), an important file for this section. We'll work with the raw read files, rather than the quality-controlled files we generated in the previous chapter, making use of and demonstrating Stacks' internal quality filter.</p>
<p><code>process_radtags</code> is used to demultiplex the individual sequence reads from a sequence data set containing pooled samples and assign these reads to individual samples.</p>
<p>It performs QC at the same time, removing:</p>
<ul>
<li>Low quality sequence reads</li>
<li>Illumina adapters</li>
<li>Reads where the restriction enzyme cut-sites are not detected</li>
<li>Reads with any uncalled/ambiguous bases</li>
</ul>
<p>For this reason, we will be using raw data rather than the read files that we cleaned in the last chapter.</p>
<div id="process_radtags-barcodes" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> <code>process_radtags</code>: barcodes<a href="05_Preparing_data_with_Stacks.html#process_radtags-barcodes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<p><img src="figures/barcode.png" style="width:200px" /></p>
</center>
<p><code>process-radtags</code> requires a barcode file to demultiplex the sequences, identifying the individual samples through a unique barcode or index (or combinations of barcodes/indices), which are either incorporated at the start of the sequence read or annotated in the sequence header. Let's have a look at the barcode file for the sample data we want to demultiplex.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb16-1"><a href="05_Preparing_data_with_Stacks.html#cb16-1" tabindex="-1"></a><span class="fu">less</span> raw_data/manta_samples.barcodes</span></code></pre></div>
<p>This is a tab-delimited file, with two columns listing our barcodes or indexes and the samples identified by these barcode combinations in a third column. Our samples are (mostly) identified by numbers here, but you could use any name or alphanumeric identifier. If you had additional combinatorial barcodes, there will be additional columns for any additional barcodes. You could also just have a single barcode or column. For your own study, you may have many different sequence pools coming off the Illumina platform and you will need to compile barcode files to demultiplex each of the sets of sequences further to get the data for each of your samples.</p>
<p>Press 'q' to exit the <code>less</code> view and return to the command line.</p>
</div>
<div id="process_radtags-prep" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> <code>process_radtags</code>: prep<a href="05_Preparing_data_with_Stacks.html#process_radtags-prep" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<p><img src="figures/files.png" style="width:150px; background:white; border-radius:5px" /></p>
</center>
<p>Before we run <code>process_radtags</code>, create a directory to receive the demultiplexed data:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb17-1"><a href="05_Preparing_data_with_Stacks.html#cb17-1" tabindex="-1"></a><span class="fu">mkdir</span> working_data</span></code></pre></div>
<p>We can now execute <code>process_radtags</code>. The command below contains arguments specific to the way this library was indexed. The Stacks <a href="http://catchenlab.life.illinois.edu/stacks/comp/process_radtags.php">website</a> and <a href="http://catchenlab.life.illinois.edu/stacks/manual/">manual</a> provide detailed examples of code for dealing with various indexing schemas and barcode combinations for single- and paired-end reads. In this case, the library was indexed and each sample is identified using a combination of two different barcodes, which are included in the P1 and the P2 adapters, respectively. The barcoded P1 adapter is appended to the start of the sequence that becomes the R1 read and the barcoded P2 adapter to the start of the R2 read. However, the full sequencing output has already undergone one demultiplexing step, using the P2 barcodes, as they came off the sequencer, and these read files and samples were all indexed with the same P2 barcode. This barcode is now included in the sequence index line (sequence header) in the fastq file, as we saw in the previous chapter.</p>
<p>We can also inspect the sequence data itself to see the inline P1 barcodes in the R1 reads.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb18-1"><a href="05_Preparing_data_with_Stacks.html#cb18-1" tabindex="-1"></a><span class="fu">zcat</span> raw_data/Lib1_001_R1.fastq.gz <span class="kw">|</span> <span class="fu">grep</span> <span class="st">&quot;TGCA&quot;</span> <span class="at">--color</span></span></code></pre></div>
<p>In this command, we are using <code>zcat</code>, piping the output and then using <code>grep</code> (for "grab regular expression"). This is essentially a search function that will pull out each line in the file that contains our search term <code>TGCA</code>. 'TGCA' is the enzyme cutsite overhang for the restriction enzyme <em>SbfI</em>, where the the P1 adapter is ligated to the start of the R1 read. The <code>--color</code> argument will highlight all matches to the search term in colour. Most of these matches will be near the start of the sequences; these are the overhangs. There may be other random nucleotide stretches matching our search term "deeper" in the sequences. The nucleotides to the left of the overhangs are our barcodes and those to the right are the sequence data of the RADtag fragment.</p>
</div>
<div id="process_radtags-run" class="section level3 hasAnchor" number="5.2.3">
<h3><span class="header-section-number">5.2.3</span> <code>process_radtags</code>: run<a href="05_Preparing_data_with_Stacks.html#process_radtags-run" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<p><img src="figures/play_red.png" style="width:100px" /></p>
</center>
<p>The following command will demultiplex the raw read files:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb19-1"><a href="05_Preparing_data_with_Stacks.html#cb19-1" tabindex="-1"></a><span class="ex">process_radtags</span> <span class="dt">\</span></span>
<span id="cb19-2"><a href="05_Preparing_data_with_Stacks.html#cb19-2" tabindex="-1"></a>-1 raw_data/Lib1_001_R1.fastq.gz <span class="dt">\</span></span>
<span id="cb19-3"><a href="05_Preparing_data_with_Stacks.html#cb19-3" tabindex="-1"></a>-2 raw_data/Lib1_001_R2.fastq.gz <span class="dt">\</span></span>
<span id="cb19-4"><a href="05_Preparing_data_with_Stacks.html#cb19-4" tabindex="-1"></a>-P <span class="at">-i</span> gzfastq <span class="dt">\</span></span>
<span id="cb19-5"><a href="05_Preparing_data_with_Stacks.html#cb19-5" tabindex="-1"></a>-b raw_data/manta_samples.barcodes <span class="dt">\</span></span>
<span id="cb19-6"><a href="05_Preparing_data_with_Stacks.html#cb19-6" tabindex="-1"></a>-o working_data/ <span class="dt">\</span></span>
<span id="cb19-7"><a href="05_Preparing_data_with_Stacks.html#cb19-7" tabindex="-1"></a>--inline-index <span class="at">--renz_1</span> sbfI <span class="at">--filter_illumina</span> <span class="dt">\</span></span>
<span id="cb19-8"><a href="05_Preparing_data_with_Stacks.html#cb19-8" tabindex="-1"></a>-c <span class="at">-q</span> <span class="at">-r</span></span></code></pre></div>
<div id="parameters" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="05_Preparing_data_with_Stacks.html#parameters" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<center>
<p><img src="figures/parameter_red.png" style="width:100px" /></p>
</center>
<p>In this instance, the arguments used are:</p>
<ul>
<li><code>-1</code> and <code>-2</code> specify the path to and the names of the raw R1 and R2 sequence data files, respectively.</li>
<li><code>-P</code> indicates that these are read files from paired-end sequencing.</li>
<li><code>-i</code> denotes the input file type. If left out, Stacks will guess the input file type and will use gzipped fastq ('.fastq.gz') as the default if it can't.</li>
<li><code>-b</code>: This specifies the path to and the filename of the barcode file.</li>
<li><code>-o</code> specifies the path to output the demultiplexed data.</li>
<li><code>--inline_index</code> indicates that the barcode used to identify the samples is situated in the sequence itself (inline) at the start of the R1 reads (from the P1 adapter), and in the fastq sequence index (sequence header) in the R2 reads (from the P2 adapter).</li>
<li><code>--renz_1</code> defines one of the restriction enzymes used (<em>SbfI</em>), which will have the cut-site overhang sequence from the P1 adapter on the R1 reads. Ordinarily, one would also specify the second restriction enzyme (<em>SphI</em>) used in the double digest for the cut-site overhang for the P2 adapter with <code>--renz_2</code>. However, we will not do this here, as this enzyme cut-site has already been removed, along with the barcode and Illumina adapters in the first demultiplexing stage.</li>
<li><code>-c</code> cleans the data, removing any reads with uncalled/ambiguous bases.</li>
<li><code>-q</code>: This is a quality filter and discards reads below a given sequence quality (Phred) score.</li>
<li><code>-r</code>: This is an option to rescue sequence reads and avoid them being removed by allowing a specified number of nucleotide mismatches (the default is one) between the specified barcodes and those observed in the reads or between the observed enzyme cut-sites (overhangs) and those expected.</li>
<li><code>--filter_illumina</code> removes reads that were marked as 'failed' by Illumina's internal quality filter.</li>
</ul>
</div>
</div>
<div id="process_radtags-mcqs" class="section level3 hasAnchor" number="5.2.4">
<h3><span class="header-section-number">5.2.4</span> <code>process_radtags</code>: MCQs<a href="05_Preparing_data_with_Stacks.html#process_radtags-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<p><img src="figures/question_bubble_red.png" style="width:100px" /></p>
</center>
<p>After executing the script (it should take about a minute to run), let's consider the terminal output.</p>
<ol style="list-style-type: decimal">
<li>How many reads were retained and assigned to individual samples?
<div id="radio_DRTDOSBUHF" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_DRTDOSBUHF" value=""></input> <span>40,739</span></label><label><input type="radio" autocomplete="off" name="radio_DRTDOSBUHF" value=""></input> <span>61,572</span></label><label><input type="radio" autocomplete="off" name="radio_DRTDOSBUHF" value="answer"></input> <span>392,829</span></label><label><input type="radio" autocomplete="off" name="radio_DRTDOSBUHF" value=""></input> <span>500,000</span></label>
</div></li>
<li>What percentage of reads where not assigned because the RAD cutsite (the correct overhang for enzyme <em>SbfI</em>) could not be found?
<div id="radio_RZWSXCEZQW" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_RZWSXCEZQW" value=""></input> <span>0%</span></label><label><input type="radio" autocomplete="off" name="radio_RZWSXCEZQW" value=""></input> <span>1.1%</span></label><label><input type="radio" autocomplete="off" name="radio_RZWSXCEZQW" value="answer"></input> <span>8.0%</span></label><label><input type="radio" autocomplete="off" name="radio_RZWSXCEZQW" value=""></input> <span>12.3%</span></label>
</div></li>
<li>What percentage of reads failed to pass the quality filters and were discarded?
<div id="radio_IEJWRETMRR" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_IEJWRETMRR" value=""></input> <span>0.0%</span></label><label><input type="radio" autocomplete="off" name="radio_IEJWRETMRR" value="answer"></input> <span>0.4%</span></label><label><input type="radio" autocomplete="off" name="radio_IEJWRETMRR" value=""></input> <span>8.0%</span></label><label><input type="radio" autocomplete="off" name="radio_IEJWRETMRR" value=""></input> <span>13.0%</span></label>
</div></li>
<li>What percentage of the (500 000) reads were not assigned because the barcodes in the sequence could not be matched to any in the barcode file?
<div id="radio_REPCQXHWUO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_REPCQXHWUO" value=""></input> <span>0%</span></label><label><input type="radio" autocomplete="off" name="radio_REPCQXHWUO" value=""></input> <span>1.1%</span></label><label><input type="radio" autocomplete="off" name="radio_REPCQXHWUO" value=""></input> <span>8%</span></label><label><input type="radio" autocomplete="off" name="radio_REPCQXHWUO" value="answer"></input> <span>13.0%</span></label>
</div></li>
</ol>
<div class="webex-solution">
<button>
Why do you think these barcodes were not found?
</button>
<p>Remember that many individuals (up to 96 with this indexing scheme) were multiplexed and included in this pair of read files. We have only specified a few (25) of the potential barcode combinations to identify samples, in order to keep the processing time short for the workshop.</p>
</div>
</div>
<div id="process_radtags-output" class="section level3 hasAnchor" number="5.2.5">
<h3><span class="header-section-number">5.2.5</span> <code>process_radtags</code>: output<a href="05_Preparing_data_with_Stacks.html#process_radtags-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<p><img src="figures/output_file_red.png" style="width:100px" /></p>
</center>
<p>We can peek into the 'working_data' directory to see what files were produced, and also have a look at the 'process_radtags' log, which will give us more detailed information on how the reads were demultiplexed.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb20-1"><a href="05_Preparing_data_with_Stacks.html#cb20-1" tabindex="-1"></a><span class="co"># List the contents of the &#39;working_data&#39; subdirectory</span></span>
<span id="cb20-2"><a href="05_Preparing_data_with_Stacks.html#cb20-2" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-l</span> working_data</span>
<span id="cb20-3"><a href="05_Preparing_data_with_Stacks.html#cb20-3" tabindex="-1"></a><span class="co"># View the contents of the &#39;process_radtags&#39; log</span></span>
<span id="cb20-4"><a href="05_Preparing_data_with_Stacks.html#cb20-4" tabindex="-1"></a><span class="fu">less</span> working_data/process_radtags.raw_data.log</span></code></pre></div>
<p>For each sample (ours are identified by a number), four gzipped fastq sequence files were produced.</p>
<ol style="list-style-type: decimal">
<li><strong>'Sample.1.fq.gz'</strong>: Paired forward reads that were assigned to that sample</li>
<li><strong>'Sample.2.fq.gz'</strong>: Paired reverse reads that were assigned to that sample</li>
<li><strong>'Sample.rem.1.fq.gz'</strong>: Forward reads that were assigned to that sample, but the matching reverse reads could not be assigned or were discarded because of quality issues</li>
<li><strong>'Sample.rem.2.fq.gz'</strong>: Reverse reads that were assigned to that sample, but the matching forward reads could not be assigned or were discarded because of quality issues</li>
</ol>
<p><strong>Note:</strong> We normally would not need the 'remainder' (<code>.rem.</code>) files again.</p>
<p>The 'process_radtags' log expands the summary that we saw in the terminal and indicates at a sample or barcode-combination level how many reads were retained or lost/unassigned because of quality or mismatched barcodes and enzyme cutsites. It also lists and counts potential barcodes that were observed in the data, given the mismatching limits we specified, that were not listed or assigned in the barcode file.</p>
<p>If you need to demultiplex several pairs of read files, as you typically would in a RADseq study, the Appendix gives some guidance on how to do this with loops. With our sequence data demultiplexed and assigned to samples, we can now begin to reconstruct loci and call our SNPs.</p>
</div>
</div>
<div id="building-the-catalogue-loci-with-denovo_map.pl" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Building the catalogue loci with <code>denovo_map.pl</code><a href="05_Preparing_data_with_Stacks.html#building-the-catalogue-loci-with-denovo_map.pl" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/stacking_blocks.png" style="width:200px; background:white; border-radius:5px" />
</center>
<p>We'll use the Stacks pipeline <code>denovo_map.pl</code> to reconstruct loci, identify variants and prepare data for downstream analysis. This pipeline combines several of the core modules of Stacks in a single execution. These modules are:</p>
<ul>
<li><strong>'ustacks':</strong> "Stacks" up sequence reads to identify alleles and compile loci in individual samples</li>
<li><strong>'cstacks':</strong> Combines the loci from each of the individuals into a catalogue of loci for all the included samples</li>
<li><strong>'sstacks':</strong> Refers the samples back to the catalogue to identify shared loci</li>
<li><strong>'tvs2bam':</strong> Transposes the data from being arranged by sample to being arranged by locus</li>
<li><strong>'gstacks':</strong> Creates contigs from paired-end reads, calls SNPs and, importantly for phylogenetic applications, compiles and combines the SNPs into haplotypes</li>
</ul>
<p>Afterwards we'll use the 'populations' modules to generate some basic population-level summary and diversity statistics. This module is probably more relevant for population genomics but can also be used to prepare files for other applications (which we will do below). The module is included by default as a filtering tool and as a controller for which samples are processed down the pipeline.</p>
<div id="denovo_map.pl-input" class="section level3 hasAnchor" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> <code>denovo_map.pl</code>: input<a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-input" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<p><img src="figures/input.png" style="width:150px" /></p>
</center>
<p>For demonstrating <code>denovo_map.pl</code>, we'll use a subset of five samples from the Hosegood <em>et al.</em> (2020) study. These have already been demultiplexed and quality-filtered in 'process_radtags'.</p>
<p>Navigate to the directory where the dataset is located.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb21-1"><a href="05_Preparing_data_with_Stacks.html#cb21-1" tabindex="-1"></a><span class="bu">cd</span> ~/phylogenomics/demultiplexed_data</span></code></pre></div>
<p>List the files that are present:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb22-1"><a href="05_Preparing_data_with_Stacks.html#cb22-1" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-l</span></span></code></pre></div>
<p>You should see 10 fastq files, the R1 ('.1.fq.gz') and R2 ('.2.fq.gz') reads for each of five samples, and a 'popmap' (populations map) file ('5samples.popmap').</p>
<p>Have a look at the 'popmap' file:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb23-1"><a href="05_Preparing_data_with_Stacks.html#cb23-1" tabindex="-1"></a><span class="fu">less</span> 5samples.popmap</span></code></pre></div>
<p>This is also a tab-delimited file, with the names of the sequence files (without the R1, R2, 1 or 2 indicators, and without file extensions) of the samples that we want to analyse in the left column, and a second column on the right for assigning these samples to populations or species. These assignments are important for the calculation of statistics in the 'populations' module, defining units for downstream analysis and implementing data filtering options. Here, the population map contains five individuals, each representing a different species of <em>Mobula</em>.</p>
</div>
<div id="denovo_map.pl-run" class="section level3 hasAnchor" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> <code>denovo_map.pl</code>: run<a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-run" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<p><img src="figures/play_green.png" style="width:100px" /></p>
</center>
<p>We can now create a directory to receive the output...</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb24-1"><a href="05_Preparing_data_with_Stacks.html#cb24-1" tabindex="-1"></a><span class="fu">mkdir</span> ../denovo_map</span></code></pre></div>
<p>...and execute the pipeline:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb25-1"><a href="05_Preparing_data_with_Stacks.html#cb25-1" tabindex="-1"></a><span class="ex">denovo_map.pl</span> <span class="at">--samples</span> . <span class="dt">\</span></span>
<span id="cb25-2"><a href="05_Preparing_data_with_Stacks.html#cb25-2" tabindex="-1"></a>--popmap 5samples.popmap <span class="dt">\</span></span>
<span id="cb25-3"><a href="05_Preparing_data_with_Stacks.html#cb25-3" tabindex="-1"></a>-o ../denovo_map/  <span class="dt">\</span></span>
<span id="cb25-4"><a href="05_Preparing_data_with_Stacks.html#cb25-4" tabindex="-1"></a>-T 8 <span class="at">-d</span> <span class="dt">\</span></span>
<span id="cb25-5"><a href="05_Preparing_data_with_Stacks.html#cb25-5" tabindex="-1"></a>-M 3 <span class="at">-n</span> 3 <span class="dt">\</span></span>
<span id="cb25-6"><a href="05_Preparing_data_with_Stacks.html#cb25-6" tabindex="-1"></a>-p 1 <span class="at">-r</span> 1.0 <span class="dt">\</span></span>
<span id="cb25-7"><a href="05_Preparing_data_with_Stacks.html#cb25-7" tabindex="-1"></a>--paired </span></code></pre></div>
<p>Was this successful? Did it look like all the expected information appeared as output in the terminal? Considering what we are attempting to do, this was rather quick, wasn't it?</p>
<div id="parameters-1" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="05_Preparing_data_with_Stacks.html#parameters-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<center>
<p><img src="figures/parameter_green.png" style="width:100px" /></p>
</center>
<p>Let's take a look at the arguments we used in this script:</p>
<ul>
<li><code>--samples</code> specifies the path to where the sequence reads are located, with <code>.</code> indicating the present directory.</li>
<li><code>--popmap</code> specifies the path to and the name of the 'popmap' file.</li>
<li><code>-o</code>: This is the path to direct your output to.</li>
<li><code>-T</code> specifies the number of threads to use.</li>
<li><code>-M</code> and <code>-n</code> are parameters used to specify the maximum number of mismatches (base differences) allowed between sequence reads when building loci within samples in 'ustacks' (<code>-M</code>), and between individuals when merging loci in the catalogue in 'cstacks' (<code>-n</code>). These are not arbitrary and need to be optimised for each data set. An overview of one optimising procedure is given in the Appendix. We are using the parameters that Hosegood <em>et al.</em> (2020) optimised for the full data set.</li>
<li><code>-p</code> is a data filtering option and specifies the minimum number of populations that a locus must be present in in order to be included and be processed. To maximise the number of loci retained in this initial pass (we will filter this further below), we are specifying a minimum number of one population. <strong>Note:</strong> each individual is a unique taxon in this analysis and we don't have "populations" <em>per se</em>, but in terms of the terminology each individual here is regarded as a different 'population'. You could well have multiple individuals representing a species or 'population'. Depending on your objectives you may choose to treat your samples individually as different terminal representatives of the same species or collectively.</li>
<li><code>-r</code>: Another data filtering option, specifying the minimum percentage, presented as a float or proportion (ranging from 0 to 1.0), of individuals in a population that a locus needs to be present in to be retained. We have set this at 100% here, as we only have one individual in each of our populations and we don't want to exclude any of these.</li>
<li><code>--paired</code> assembles contigs for each locus from the paired-end reads after compiling the RADseq loci.</li>
</ul>
<p><strong>Importantly</strong>,</p>
<ul>
<li><code>-d</code>: 'Dry run' gives you the option of checking whether the command will execute without actually processing any of your called data, allowing you to check for typos, incorrect paths or any other errors. <code>denovo_map.pl</code> is the most time-consuming of the processes we can run in Stacks, and it is good to see that it is working properly before committing time to it. You can check the 'denovo_map.log' for errors and issues, although the output to the terminal should have indicated these.</li>
</ul>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb26-1"><a href="05_Preparing_data_with_Stacks.html#cb26-1" tabindex="-1"></a><span class="fu">less</span> ../denovo_map/denovo_map.log</span></code></pre></div>
<p>Now, remove <code>-d</code> from the code above and execute it again. <strong>Remember</strong> that you can use the up key to scroll through your Linux history to save you from retyping code.</p>
<p>Now would be a good time to go make yourself a cup of tea; it will take Stacks about 12 minutes to take the sets of sample data sequentially through the different modules. As each module executes, a report for each sample will be sent to the terminal.</p>
</div>
</div>
<div id="denovo_map.pl-output" class="section level3 hasAnchor" number="5.3.3">
<h3><span class="header-section-number">5.3.3</span> <code>denovo_map.pl</code>: output<a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<p><img src="figures/output_file_green.png" style="width:100px" /></p>
</center>
<p>Once the command is done, we can have a look at the files that <code>denovo_map.pl</code> produces.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb27-1"><a href="05_Preparing_data_with_Stacks.html#cb27-1" tabindex="-1"></a><span class="fu">ls</span> ../denovo_map</span></code></pre></div>
<p>For each of our samples, four '.tsv.gz' (tab-delimited) output files are produced. These contain:</p>
<ol style="list-style-type: decimal">
<li>The alleles that are constructed from each RADtag locus ('XXX.alleles.tsv.gz')</li>
<li>The assembled loci for each sample ('XXX.tags.tsv.gz')</li>
<li>The loci that were matched to the catalogue ('XXX.matches.tsv.gz')</li>
<li>The final SNP genotype calls ('XXX.snps.tsv.gz')</li>
</ol>
<p>You can consult the Stacks <a href="https://catchenlab.life.illinois.edu/stacks/manual/index.php#files">manual</a> to see exactly what information is contained in each.
There is also a 'XXX.matches.bam' file for each sample. 'BAM' files are compressed, binary versions of <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">Sequence Alignment/Map ('SAM')</a> files. These contain the aligned RADtag loci in that sample that were matched to the catalogue. If we were using a reference genome, these loci will be mapped and the file will also contain reference co-ordinates for where these loci are placed on the reference genome or scaffolds.</p>
<p>There are also individual log files for each of the Stacks modules, which are collated in the main 'denovo_map.log'. There are a few things that we would want to note or consider from this output. These may determine how we proceed with our analyses. A few MCQs will guide you through these.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb28-1"><a href="05_Preparing_data_with_Stacks.html#cb28-1" tabindex="-1"></a><span class="fu">less</span> ../denovo_map/denovo_map.log</span></code></pre></div>
</div>
<div id="denovo_map.pl-mcqs" class="section level3 hasAnchor" number="5.3.4">
<h3><span class="header-section-number">5.3.4</span> <code>denovo_map.pl</code>: MCQs<a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<p><img src="figures/question_bubble_green.png" style="width:100px" /></p>
</center>
<ol style="list-style-type: decimal">
<li>What is the lowest depth of coverage for our samples? (Scroll to the bottom of the 'ustacks' section)
<div id="radio_GOLGTGNRXK" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_GOLGTGNRXK" value=""></input> <span>231.19x</span></label><label><input type="radio" autocomplete="off" name="radio_GOLGTGNRXK" value=""></input> <span>140.44x</span></label><label><input type="radio" autocomplete="off" name="radio_GOLGTGNRXK" value=""></input> <span>99.05x</span></label><label><input type="radio" autocomplete="off" name="radio_GOLGTGNRXK" value="answer"></input> <span>45.78x</span></label><label><input type="radio" autocomplete="off" name="radio_GOLGTGNRXK" value=""></input> <span>37.49x</span></label>
</div></li>
</ol>
<p>As a rule of thumb, we would like at least 30x depth coverage for SNP genotyping in a typical spatial population genomics study and higher coverage for the accuracy needed to build high-resolution pedigrees. For a phylogenomic study, this would also depend on the taxonomic scope and sampling. The coverage in this case is excellent for these samples.</p>
<ol start="2" style="list-style-type: decimal">
<li>How many loci are in the catalogue? (Scroll to the bottom of the 'cstacks' section).
<div id="radio_SEAGILEWOK" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_SEAGILEWOK" value=""></input> <span>5,612</span></label><label><input type="radio" autocomplete="off" name="radio_SEAGILEWOK" value="answer"></input> <span>27,619</span></label><label><input type="radio" autocomplete="off" name="radio_SEAGILEWOK" value=""></input> <span>52,291</span></label><label><input type="radio" autocomplete="off" name="radio_SEAGILEWOK" value=""></input> <span>75,125</span></label>
</div></li>
</ol>
<p>These represent the total number of loci present across all five samples, before one starts filtering to see which are shared among different combinations of samples. With more divergent taxa added to the set of samples, we may see more loci in the catalogue (depending, of course, on our sequencing yield), but we may see a large attrition of data that can actually be informative across samples.</p>
</div>
</div>
<div id="preparing-a-sequence-alignment-file-in-populations" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Preparing a sequence alignment file in <code>populations</code><a href="05_Preparing_data_with_Stacks.html#preparing-a-sequence-alignment-file-in-populations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<p><img src="figures/alignment.png" style="width:200px; background:white; border-radius:5px; border: 5px solid white" /></p>
</center>
<p>We'll use the 'populations' module of Stacks to produce Phylip files for downstream analysis. This data format was developed for the programme <a href="https://phylipweb.github.io/phylip/">PHYLIP</a>, and is a common format for all forms of phylogenetic data, accepted by many applications. The 'populations' module and the filtering and file-creation arguments can be included directly in the <code>denovo_map.pl</code> bash script above, but we'll use the 'populations' module separately to demonstrate the effect of two different filtering strategies. The choice between these two (or of any other along a continuum) will depend on your study, research questions, taxonomic sampling and the data itself.</p>
<p>In the first instance, we will minimise the missing data in our data matrix by only including SNPs that are found in all our taxa. The script is as follows:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb29-1"><a href="05_Preparing_data_with_Stacks.html#cb29-1" tabindex="-1"></a><span class="ex">populations</span> <span class="at">-P</span> ../denovo_map/ <span class="at">-M</span> 5samples.popmap <span class="dt">\ </span></span>
<span id="cb29-2"><a href="05_Preparing_data_with_Stacks.html#cb29-2" tabindex="-1"></a><span class="ex">-t</span> 8 <span class="at">-p</span> 5 <span class="at">-R</span> 1.0 <span class="dt">\</span></span>
<span id="cb29-3"><a href="05_Preparing_data_with_Stacks.html#cb29-3" tabindex="-1"></a>--write-single-snp <span class="dt">\</span></span>
<span id="cb29-4"><a href="05_Preparing_data_with_Stacks.html#cb29-4" tabindex="-1"></a>--phylip</span></code></pre></div>
<p>In this command,</p>
<ul>
<li><code>-P</code> indicates the path to where the files were saved by <code>process_radtags.pl</code> above.</li>
<li><code>-M</code> points to the population map.</li>
<li><code>-t</code>: The number of threads to use for parallel processing</li>
<li><code>-p</code> and <code>-R</code> are the parameters used for data filtering (similar to that above), where <code>-p</code> (an integer) is the minimum number of populations that a SNP locus needs to be present in to be included, while <code>-R</code> (another float) is the minimum percentage of individuals across the full sample that a locus must be present in for it to be maintained. A <code>-p</code> of 5 and a <code>-R</code> of 1.0 (100%) will ensure that the retained loci are present in all five of our taxa.</li>
<li><code>--write-single-snp</code> is called to prevent linked loci from being included in the data set. Multiple SNPs on the same RADtag or locus are much more likely to be linked than to not be. This command will retain only the first SNP from each RADtag. You can also use <code>--write-random-snp</code> to choose a random SNP from each fragment, but this may make exact replication of your analysis difficult.</li>
<li><code>--phylip</code> specifies the output format. This particular argument will include SNPs in the haplotype that are fixed within populations (i.e., species here), but variable between them. Other options allow for including sites that are variable between, as well as within populations (<code>--phylip-var</code>), or all sequences and variable sites (<code>--phylip-var-all</code>).
Take a look at the Phylip file that was produced. An explanation and example of the Phylip format can be found <a href="https://rosalind.info/glossary/phylip-format/">here</a>.</li>
</ul>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb30-1"><a href="05_Preparing_data_with_Stacks.html#cb30-1" tabindex="-1"></a><span class="fu">less</span> ../denovo_map/populations.fixed.phylip</span></code></pre></div>
How many SNP characters are in the data matrix and present in all five taxa? (Your numbers may differ slightly, depending on how the Bayesian genotyper in 'gstacks' calls any marginal genotypes)
<div id="radio_IRJTINLQNR" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_IRJTINLQNR" value=""></input> <span>5</span></label><label><input type="radio" autocomplete="off" name="radio_IRJTINLQNR" value="answer"></input> <span>1,677</span></label><label><input type="radio" autocomplete="off" name="radio_IRJTINLQNR" value=""></input> <span>2,517</span></label><label><input type="radio" autocomplete="off" name="radio_IRJTINLQNR" value=""></input> <span>8,435</span></label>
</div>
<p>In the second scenario, we'll attempt to maximise the number of loci in the data matrix, but this could introduce a lot of missing data:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb31-1"><a href="05_Preparing_data_with_Stacks.html#cb31-1" tabindex="-1"></a><span class="ex">populations</span> <span class="at">-P</span> ../denovo_map/ <span class="at">-M</span> 5samples.popmap <span class="dt">\</span></span>
<span id="cb31-2"><a href="05_Preparing_data_with_Stacks.html#cb31-2" tabindex="-1"></a>-t 8 <span class="at">-p</span> 2 <span class="at">-R</span> 0.4 <span class="dt">\</span></span>
<span id="cb31-3"><a href="05_Preparing_data_with_Stacks.html#cb31-3" tabindex="-1"></a>--write-single-snp <span class="dt">\</span></span>
<span id="cb31-4"><a href="05_Preparing_data_with_Stacks.html#cb31-4" tabindex="-1"></a>--phylip</span></code></pre></div>
<p>Here, we have specified that a locus only needs to be present in two individuals or species (<code>-p 2</code>) and in 40% (<code>-R 0.4</code>) of the overall sample (i.e., those two individuals) to be included. Once executed, open the Phylip file. You can use the same code as above, as the previous output will be overwritten.</p>
How many SNP characters are included in the data matrix now?
<div id="radio_OWHHAZGJDN" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_OWHHAZGJDN" value=""></input> <span>5</span></label><label><input type="radio" autocomplete="off" name="radio_OWHHAZGJDN" value=""></input> <span>1,677</span></label><label><input type="radio" autocomplete="off" name="radio_OWHHAZGJDN" value=""></input> <span>2,517</span></label><label><input type="radio" autocomplete="off" name="radio_OWHHAZGJDN" value="answer"></input> <span>8,435</span></label>
</div>
<p>Now that we have compiled the Phylip data matrix, we will proceed with generating, investigating and further interrogating phylogenetic trees.</p>
<p>If you are interested in exploring SNP identification, variant calling, genotyping, variant filtering and subsetting further, consider registering for the NEOF Population Genomics course. Please visit the <a href="https://neof.org.uk/training/">NEOF Training website</a> periodically to see what courses are on offer.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="04-Quality_Control.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="06-Phylogeny.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
