<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Model selection, tree reconstruction and support evaluation | Phylogenomics</title>
  <meta name="description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Model selection, tree reconstruction and support evaluation | Phylogenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Phylogenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Model selection, tree reconstruction and support evaluation | Phylogenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Gavin Gouws, Matthew Gemmell, Helen Hipperson, Ewan Harney, Kathryn Maher" />


<meta name="date" content="2024-01-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="06-Mapping.html"/>
<link rel="next" href="08-Topology_testing.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>Phylogenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-ddradseq-data-of-manta-and-devil-rays"><i class="fa fa-check"></i><b>2.1</b> Example data: ddRADseq data of manta and devil rays</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#manta--and-devil-ray-ddrad-data-set"><i class="fa fa-check"></i><b>4.2</b> Manta- and devil-ray ddRAD data set</a></li>
<li class="chapter" data-level="4.3" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality Control</a></li>
<li class="chapter" data-level="4.5" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#post-qc-check"><i class="fa fa-check"></i><b>4.5</b> Post QC Check</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html"><i class="fa fa-check"></i><b>5</b> Preparing data in Stacks</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#stacks-1"><i class="fa fa-check"></i><b>5.1</b> Stacks</a></li>
<li class="chapter" data-level="5.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#demultiplexing-data-with-process_radtags"><i class="fa fa-check"></i><b>5.2</b> Demultiplexing data with <code>process_radtags</code></a></li>
<li class="chapter" data-level="5.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#building-the-catalogue-loci-with-denovo_map.pl"><i class="fa fa-check"></i><b>5.3</b> Building the catalogue loci with <code>denovo_map.pl</code></a></li>
<li class="chapter" data-level="5.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#preparing-a-sequence-alignment-file-in-populations"><i class="fa fa-check"></i><b>5.4</b> Preparing a sequence alignment file in <code>populations</code></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Mapping.html"><a href="06-Mapping.html"><i class="fa fa-check"></i><b>6</b> Mapping sequences to the reference genome using BWA</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Mapping.html"><a href="06-Mapping.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>6.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="6.2" data-path="06-Mapping.html"><a href="06-Mapping.html#read-mapping"><i class="fa fa-check"></i><b>6.2</b> Read mapping</a></li>
<li class="chapter" data-level="6.3" data-path="06-Mapping.html"><a href="06-Mapping.html#assessing-mapping-results"><i class="fa fa-check"></i><b>6.3</b> Assessing mapping results</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Phylogeny.html"><a href="07-Phylogeny.html"><i class="fa fa-check"></i><b>7</b> Model selection, tree reconstruction and support evaluation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Phylogeny.html"><a href="07-Phylogeny.html#estimating-a-best-fit-model-of-nucleotide-or-sequence-evolution"><i class="fa fa-check"></i><b>7.1</b> Estimating a best-fit model of nucleotide or sequence evolution</a></li>
<li class="chapter" data-level="7.2" data-path="07-Phylogeny.html"><a href="07-Phylogeny.html#constructing-a-maximum-likelihood-phylogenetic-tree"><i class="fa fa-check"></i><b>7.2</b> Constructing a maximum-likelihood phylogenetic tree</a></li>
<li class="chapter" data-level="7.3" data-path="07-Phylogeny.html"><a href="07-Phylogeny.html#calculating-nodal-support"><i class="fa fa-check"></i><b>7.3</b> Calculating nodal support</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Topology_testing.html"><a href="08-Topology_testing.html"><i class="fa fa-check"></i><b>8</b> Hypothesis and topology testing</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="11-Appendix.html"><a href="11-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="11-Appendix.html"><a href="11-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="11-Appendix.html"><a href="11-Appendix.html#loops-and-wildcards"><i class="fa fa-check"></i><b>B</b> Loops and wildcards</a></li>
<li class="chapter" data-level="C" data-path="11-Appendix.html"><a href="11-Appendix.html#ipyrad"><i class="fa fa-check"></i><b>C</b> iPYRAD</a></li>
<li class="chapter" data-level="D" data-path="11-Appendix.html"><a href="11-Appendix.html#optimising-loci-reconstruction-in-denovo_map.pl-in-stacks"><i class="fa fa-check"></i><b>D</b> Optimising loci reconstruction in <code>denovo_map.pl</code> in Stacks</a></li>
<li class="chapter" data-level="E" data-path="11-Appendix.html"><a href="11-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>E</b> Jupyter-notebook</a></li>
<li class="chapter" data-level="F" data-path="11-Appendix.html"><a href="11-Appendix.html#loops"><i class="fa fa-check"></i><b>F</b> Using loops to process multiple samples</a>
<ul>
<li class="chapter" data-level="F.1" data-path="11-Appendix.html"><a href="11-Appendix.html#simple-loop-example"><i class="fa fa-check"></i><b>F.1</b> Simple loop example</a></li>
<li class="chapter" data-level="F.2" data-path="11-Appendix.html"><a href="11-Appendix.html#trimmomatic-loop-example"><i class="fa fa-check"></i><b>F.2</b> Trimmomatic loop example</a></li>
</ul></li>
<li class="chapter" data-level="G" data-path="11-Appendix.html"><a href="11-Appendix.html#an-overview-of-bayesian-statistics"><i class="fa fa-check"></i><b>G</b> An overview of Bayesian statistics</a></li>
<li class="chapter" data-level="H" data-path="11-Appendix.html"><a href="11-Appendix.html#additional-approaches-for-detecting-selection-and-for-gwas"><i class="fa fa-check"></i><b>H</b> Additional approaches for detecting selection and for GWAS</a>
<ul>
<li class="chapter" data-level="H.1" data-path="11-Appendix.html"><a href="11-Appendix.html#rehh"><i class="fa fa-check"></i><b>H.1</b> rehh</a></li>
<li class="chapter" data-level="H.2" data-path="11-Appendix.html"><a href="11-Appendix.html#gcta"><i class="fa fa-check"></i><b>H.2</b> GCTA</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Phylogenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="IQTree" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Model selection, tree reconstruction and support evaluation<a href="07-Phylogeny.html#IQTree" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>#<code>{r, fig.align = 'center',out.width= '30%', echo=FALSE } #knitr::include_graphics(path = "figures/snp.png", auto_pdf = TRUE) #</code></p>
<p>We will use <a href="http://www.iqtree.org">IQ-TREE</a> to construct and evaluate a phylogenetic tree. IQ-TREE is a popular package that uses a very fast and efficient algorithm to infer phylogenomic trees under a maximum-likelihood framework. It also incorporates modules for evaluating and selecting models prior to tree reconstruction, and for bootstrapping to determine nodal support. These modules, <a href="https://www.nature.com/articles/nmeth.4285">ModelFinder</a> and <a href="https://academic.oup.com/mbe/article/35/2/518/4565479?login=true">UFBoot</a>, are substantially faster than many alternative, "external" packages, such as <a href="https://academic.oup.com/mbe/article/30/5/1188/997508?login=true">jModelTest</a>, or the bootstrapping procedure that is used in <a href="">RAxML</a>, another popular and widely-used, likelihood-based, phylogenetic-construction package. The latest version, <a href="https://academic.oup.com/mbe/article/37/5/1530/5721363?login=true">IQTREE-2</a>, can be installed in Linux, Windows and Mac systems, and offers a range of new tree-search options and sequence evolution models. There is also a IQ-TREE <a href="http://iqtree.cibiv.univie.ac.at/">web-server</a> to which jobs can be submitted.</p>
<p>Navigate to the 'iqtree' directory:</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb37-1"><a href="07-Phylogeny.html#cb37-1" tabindex="-1"></a><span class="bu">cd</span> ~/phylogenomics/iqtree</span></code></pre></div>
<p>We will produce a phylogenetic tree of 10 species of manta- and devil-rays, including two representatives of each of eight <em>Mobula</em> species and two former <em>Manta</em> species, and two <em>Rhinoptera bonasus</em> individuals as outgroup representatives. The data were produced exactly as in the previous exercise, with the second approach in which we maximised the number of SNP loci included.</p>
<p>If we were to use the first approach and eliminate all missing data, we would have been left with only 307 characters to analyse, opposed to 1 678 in the previous exercise! This demonstrates quite nicely how the number of RAD loci and SNPs that are common to all samples in the Stacks catalogue decreases and missing data increase when one adds more distantly-related species to a dataset. In this case, we have added more species, representatives of species that previously belonged to another genus (<em>Manta</em>) and a more distantly-related outgroup.</p>
<p>Have a look at the 'popmap' used to generate this data set to get a feel for the individuals and taxa included.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb38-1"><a href="07-Phylogeny.html#cb38-1" tabindex="-1"></a><span class="fu">less</span> phylogeny.popmap</span></code></pre></div>
<p>Check that you have the Phylip file (manta_phylogeny.phylip) that we will use for the analysis.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb39-1"><a href="07-Phylogeny.html#cb39-1" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-l</span></span></code></pre></div>
<p>You can take a look at this file to see the dimensions (taxa X characters) of the matrix.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb40-1"><a href="07-Phylogeny.html#cb40-1" tabindex="-1"></a><span class="fu">head</span> <span class="at">-n</span> 1 manta_phylogeny.phylip</span></code></pre></div>
<p>We will now work through tree construction. Although many of the steps can be combined in a single command and IQTREE analysis, we will estimate a best-fit model of nucleotide-evolution, construct a phylogenetic tree and evaluate bootstrap support separately in order to pause and consider the procedure.</p>
<div id="estimating-a-best-fit-model-of-nucleotide-or-sequence-evolution" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Estimating a best-fit model of nucleotide or sequence evolution<a href="07-Phylogeny.html#estimating-a-best-fit-model-of-nucleotide-or-sequence-evolution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>IQ-TREE supports and calculates testing the fit of over xxx models of sequence evolution and
For sequence data (traditional rate-matrix (base substitution) base frequency models and newer Lie Markov models.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb41-1"><a href="07-Phylogeny.html#cb41-1" tabindex="-1"></a><span class="ex">iqtree</span> <span class="at">-s</span> manta_phylogeny.phylip <span class="at">-m</span> MFP</span></code></pre></div>
<p>In this code,
- <code>-s</code> denotes our data input file. IQTree can support sequence alignments in PHYLIP, fasta, NEXUS and a number of other formats.
- <code>-m</code> indicates that we wish to use ModelFinder (<code>MFP</code>) to find the most appropriate model.</p>
<p>This analysis should take about two minutes to run and will log useful output to the terminal, which will also be saved as a log file ('manta_phylogeny.log'). For each of the evaluated models, it reports the log-likelihood (-lnL; the likelihood of observing the data given this particular model and the topology of a preliminary tree constructed using parsimony), and the Akaike Information Criterion (AIC), the corrected Akaike Information Criterion (AICc) and Bayesian Inforamtion Criterion (BIC) scores. The BIC is used to choose between competing models, with the model minimising the BIC favoured. You can set this if you want to use one of the other criteria. Once IQTree has identified the optimal model. it will generate a number of of candidate trees to fit and then estimate and optimise the parameters of our chosen model, which is the final output in the terminal.</p>
<p>Scrolling through the output:</p>
<ol style="list-style-type: decimal">
<li>Which is the optimal model for this data set, as selected by the BIC?
<div id="radio_JUQXAWTPZV" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_JUQXAWTPZV" value="answer"></input> <span>K2P+ASC</span></label><label><input type="radio" autocomplete="off" name="radio_JUQXAWTPZV" value=""></input> <span>TVM+F+ASC</span></label>
</div></li>
</ol>
<p>An overview of the various models of sequence or nucleotide evolution can be found <a href="https://en.wikipedia.org/wiki/Models_of_DNA_evolution">here</a>. At the base are a number of simple or straight-forward models (weith vary few parameters), but these are made increasingly more complex as more parameters are allowed to vary or are added (older software only implemented simpler models, but with increasing computing power, we can now make the models a lot more specific).</p>
<p>The model chosen here is a simple two-parameter model. The base frequencies are all equal (at 0.25), and we have differing substitution rates (1.000 and 5.574) for transitions and transversions. In this instance, a correction for ascertainment bias (the '+ASC') has also been included - SNP data (like morphological data) are biased in that we only include variable data and have no constant sites or characters in our data sets. This needs to be considered when calculating and evaluating tree statistics and branch lengths.</p>
</div>
<div id="constructing-a-maximum-likelihood-phylogenetic-tree" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Constructing a maximum-likelihood phylogenetic tree<a href="07-Phylogeny.html#constructing-a-maximum-likelihood-phylogenetic-tree" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will now construct a maximum-likelihood phylogenetic tree, implementing the chosen model and the parameters we have estimated.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb42-1"><a href="07-Phylogeny.html#cb42-1" tabindex="-1"></a><span class="ex">iqtree</span> <span class="at">-s</span> manta_phylogeny.phylip <span class="at">-m</span> K2P+ASC <span class="at">-redo</span></span></code></pre></div>
<p>In this command,
- <code>-s</code> again refers to the input data file.
- <code>-m</code> invokes the model we want to use. We can use <code>MFP</code> if we want to search for a model and construct the tree in one command. We have already performed this step, so here we will just specify the model that was selected above.
- <code>-o</code>: We use this to specify our outgroup, '<em>Rhinoptera_bonasus1</em>' in this case. If you do not specify an outgroup, IQTree will regard the tree as unrooted, but will use the first taxon/entry in your alignment to root it for display purposes.
- <code>-redo</code>: This is a neat safety feature built into IQTree to prevent you accidentally overwriting previous work. If you have run analyses on a particular data file and generated log and output files before and you execute a command without <code>-redo</code>, it will issue an error warning and not execute your command. You are welcome to try the above command without <code>-redo</code> to see the message.</p>
<p>The report and output of the analysis will be written to the output file 'manta_phylogeny.phylip.iqtree'. You can open this to see the summary of the data and the parameters that were used, the tree statistics (likelihood score and length) and the tree itself. You may need to resize your terminal window or it might spread each branch of the tree over multiple lines and make it hard to interpret. If this happens, resize the window and execute the command again.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb43-1"><a href="07-Phylogeny.html#cb43-1" tabindex="-1"></a><span class="fu">less</span> manta_phylogeny.phylip.iqtree</span></code></pre></div>
<p>The tree is also presented in <a href="https://en.wikipedia.org/wiki/Newick_format#:~:text=In%20mathematics%2C%20Newick%20tree%20format,lengths%20using%20parentheses%20and%20commas">Newick</a> format at the bottom of the file. The parentheses (brackets) and commas indicate the relationships and branching patterns in the tree, while the numbers indicate the branch lengths, as proportional mutations along that branch, to the next node or to the taxon (at the tips). Newick annotation is a widely-used format and is accepted by most phylogenetic applications that would need a tree topology as input, and there are several editors (e.g., <a href="">TreeView</a>) available to manipulate tree topologies in this format, and to save the tree graphically for editing and publication. A separate Newick treefile ('manta_phylogeny.phylip.treefile') is written by IQTree and can be used for this purpose.</p>
<p>You will notice a few things about this tree. The two representatives of each species group together, as expected, but those species belonging to the former genus <em>Manta</em> are nested within the genus <em>Mobula</em>. We'll test this in a moment, but first we'll examine support for these relationships.</p>
</div>
<div id="calculating-nodal-support" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Calculating nodal support<a href="07-Phylogeny.html#calculating-nodal-support" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We'll use ultrafast bootstrap approximation, developed by Minh <em>et al.</em> [(2013)] (<a href="https://academic.oup.com/mbe/article/30/5/1188/997508?login=true" class="uri">https://academic.oup.com/mbe/article/30/5/1188/997508?login=true</a>), as implemented in UFBoot, to calculate support for our nodes and relationships in the tree. This approach is orders of magnitude faster than non-parametric (i.e., "traditional") bootstrapping, is less sensitive to model violations and produces less biased estimates of support.</p>
<p>We will execute the command as before, with one addition:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb44-1"><a href="07-Phylogeny.html#cb44-1" tabindex="-1"></a><span class="ex">iqtree</span> <span class="at">-s</span> manta_phylogeny.phylip <span class="at">-m</span> K2P+ASC <span class="at">-bb</span> 1000 <span class="at">-redo</span></span></code></pre></div>
<p>We are specifying (with the <code>-bb</code>) the number of bootstrap replicants we want to perform, with 1000 recommended as a minium. Once complete, we can open the updated 'manta_phylogeny.phylip.iqtree' file to see the resulting tree with the bootstraps appended.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb45-1"><a href="07-Phylogeny.html#cb45-1" tabindex="-1"></a><span class="fu">less</span> manta_phylogeny.phylip.iqtree</span></code></pre></div>
<p>The interpretation of ultrafast bootstrap approximation support is slightly different to that of traditional boostrapping, and the support produced by the two methods can't really be compared. We only consider UFBoot support of &gt; 95% as being meaningful, whereas non-parametric boostrap values of &gt; 75% are generally accepted as supported (because this metric underestimates support). Nevertheless, our tree shows that most of the relationships are well-supported, including the placement of the two <em>Manta</em> species within <em>Mobula</em>, but let's test that further.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="06-Mapping.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="08-Topology_testing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
