<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Dating phylogenies | Phylogenomics</title>
  <meta name="description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Dating phylogenies | Phylogenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Phylogenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Dating phylogenies | Phylogenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Gavin Gouws, Matthew Gemmell, Ewan Harney, Helen Hipperson, Kathryn Maher" />


<meta name="date" content="2025-07-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="07-Topology_testing.html"/>
<link rel="next" href="09-Delimiting_species.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Phylogenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-ddradseq-data-of-manta-and-devil-rays"><i class="fa fa-check"></i><b>2.1</b> Example data: ddRADseq data of manta and devil rays</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#manta--and-devil-ray-ddrad-data-set"><i class="fa fa-check"></i><b>4.2</b> Manta- and devil-ray ddRAD data set</a></li>
<li class="chapter" data-level="4.3" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality Control</a></li>
<li class="chapter" data-level="4.5" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#post-qc-check"><i class="fa fa-check"></i><b>4.5</b> Post QC Check</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html"><i class="fa fa-check"></i><b>5</b> Preparing data in Stacks</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#stacks-1"><i class="fa fa-check"></i><b>5.1</b> Stacks</a></li>
<li class="chapter" data-level="5.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#demultiplexing-data-with-process_radtags"><i class="fa fa-check"></i><b>5.2</b> Demultiplexing data with <code>process_radtags</code></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-barcodes"><i class="fa fa-check"></i><b>5.2.1</b> <code>process_radtags</code>: barcodes</a></li>
<li class="chapter" data-level="5.2.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-prep"><i class="fa fa-check"></i><b>5.2.2</b> <code>process_radtags</code>: prep</a></li>
<li class="chapter" data-level="5.2.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-run"><i class="fa fa-check"></i><b>5.2.3</b> <code>process_radtags</code>: run</a></li>
<li class="chapter" data-level="5.2.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-mcqs"><i class="fa fa-check"></i><b>5.2.4</b> <code>process_radtags</code>: MCQs</a></li>
<li class="chapter" data-level="5.2.5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-output"><i class="fa fa-check"></i><b>5.2.5</b> <code>process_radtags</code>: output</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#building-the-catalogue-loci-with-denovo_map.pl"><i class="fa fa-check"></i><b>5.3</b> Building the catalogue loci with <code>denovo_map.pl</code></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-input"><i class="fa fa-check"></i><b>5.3.1</b> <code>denovo_map.pl</code>: input</a></li>
<li class="chapter" data-level="5.3.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-run"><i class="fa fa-check"></i><b>5.3.2</b> <code>denovo_map.pl</code>: run</a></li>
<li class="chapter" data-level="5.3.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-output"><i class="fa fa-check"></i><b>5.3.3</b> <code>denovo_map.pl</code>: output</a></li>
<li class="chapter" data-level="5.3.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-mcqs"><i class="fa fa-check"></i><b>5.3.4</b> <code>denovo_map.pl</code>: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#preparing-a-sequence-alignment-file-in-populations"><i class="fa fa-check"></i><b>5.4</b> Preparing a sequence alignment file in <code>populations</code></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html"><i class="fa fa-check"></i><b>6</b> Model selection, tree reconstruction and support evaluation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#estimating-a-best-fit-model-of-nucleotide-or-sequence-evolution"><i class="fa fa-check"></i><b>6.1</b> Estimating a best-fit model of nucleotide or sequence evolution</a></li>
<li class="chapter" data-level="6.2" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#constructing-a-maximum-likelihood-phylogenetic-tree"><i class="fa fa-check"></i><b>6.2</b> Constructing a maximum-likelihood phylogenetic tree</a></li>
<li class="chapter" data-level="6.3" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#calculating-nodal-support"><i class="fa fa-check"></i><b>6.3</b> Calculating nodal support</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Topology_testing.html"><a href="07-Topology_testing.html"><i class="fa fa-check"></i><b>7</b> Hypothesis and topology testing</a></li>
<li class="chapter" data-level="8" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html"><i class="fa fa-check"></i><b>8</b> Dating phylogenies</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#setting-up-the-date-file"><i class="fa fa-check"></i><b>8.1</b> Setting up the date file</a></li>
<li class="chapter" data-level="8.2" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#running-the-analysis"><i class="fa fa-check"></i><b>8.2</b> Running the analysis</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#dating-our-tree"><i class="fa fa-check"></i><b>8.2.1</b> Dating our tree</a></li>
<li class="chapter" data-level="8.2.2" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#getting-confidence-intervals-around-our-dates"><i class="fa fa-check"></i><b>8.2.2</b> Getting confidence intervals around our dates</a></li>
<li class="chapter" data-level="8.2.3" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#lineage-through-time-plots"><i class="fa fa-check"></i><b>8.2.3</b> Lineage-Through-Time plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html"><i class="fa fa-check"></i><b>9</b> Species delimitation with the Multispecies Coalescent</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#beast2"><i class="fa fa-check"></i><b>9.1</b> beast2</a></li>
<li class="chapter" data-level="9.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#gene-tree---species-tree-conflicts"><i class="fa fa-check"></i><b>9.2</b> Gene tree - species tree conflicts</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis"><i class="fa fa-check"></i><b>9.2.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#starting-the-analysis"><i class="fa fa-check"></i><b>9.2.2</b> Starting the analysis</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#examining-the-results"><i class="fa fa-check"></i><b>9.2.3</b> Examining the results</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#bayes-factor-delimitation"><i class="fa fa-check"></i><b>9.3</b> Bayes Factor Delimitation</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis-1"><i class="fa fa-check"></i><b>9.3.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="9.3.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#running-the-analysis-1"><i class="fa fa-check"></i><b>9.3.2</b> Running the analysis</a></li>
<li class="chapter" data-level="9.3.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#selecting-the-taxonomic-hypothesis"><i class="fa fa-check"></i><b>9.3.3</b> Selecting the taxonomic hypothesis</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="10-Appendix.html"><a href="10-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="10-Appendix.html"><a href="10-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="10-Appendix.html"><a href="10-Appendix.html#loops"><i class="fa fa-check"></i><b>B</b> Wildcards and loops</a>
<ul>
<li class="chapter" data-level="B.1" data-path="10-Appendix.html"><a href="10-Appendix.html#a-simple-wildcard"><i class="fa fa-check"></i><b>B.1</b> A simple wildcard</a></li>
<li class="chapter" data-level="B.2" data-path="10-Appendix.html"><a href="10-Appendix.html#loops-1"><i class="fa fa-check"></i><b>B.2</b> Loops</a>
<ul>
<li class="chapter" data-level="B.2.1" data-path="10-Appendix.html"><a href="10-Appendix.html#an-example-of-a-simple-loop"><i class="fa fa-check"></i><b>B.2.1</b> An example of a simple loop</a></li>
<li class="chapter" data-level="B.2.2" data-path="10-Appendix.html"><a href="10-Appendix.html#stacks-process_radtags-loop-example"><i class="fa fa-check"></i><b>B.2.2</b> Stacks <code>process_radtags</code> loop example</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="C" data-path="10-Appendix.html"><a href="10-Appendix.html#optimising-loci-reconstruction-in-denovo_map.pl-in-stacks"><i class="fa fa-check"></i><b>C</b> Optimising loci reconstruction in <code>denovo_map.pl</code> in Stacks</a></li>
<li class="chapter" data-level="D" data-path="10-Appendix.html"><a href="10-Appendix.html#additional-resources"><i class="fa fa-check"></i><b>D</b> Additional resources</a>
<ul>
<li class="chapter" data-level="D.1" data-path="10-Appendix.html"><a href="10-Appendix.html#snp-variant-calling"><i class="fa fa-check"></i><b>D.1</b> SNP variant calling</a></li>
<li class="chapter" data-level="D.2" data-path="10-Appendix.html"><a href="10-Appendix.html#sequence-alignment-tools"><i class="fa fa-check"></i><b>D.2</b> Sequence alignment tools</a></li>
<li class="chapter" data-level="D.3" data-path="10-Appendix.html"><a href="10-Appendix.html#model-selection-tools"><i class="fa fa-check"></i><b>D.3</b> Model selection tools</a></li>
<li class="chapter" data-level="D.4" data-path="10-Appendix.html"><a href="10-Appendix.html#phylogeny-reconstruction-tools"><i class="fa fa-check"></i><b>D.4</b> Phylogeny reconstruction tools</a></li>
<li class="chapter" data-level="D.5" data-path="10-Appendix.html"><a href="10-Appendix.html#r-packages"><i class="fa fa-check"></i><b>D.5</b> R-packages</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Phylogenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dating" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Dating phylogenies<a href="08-Dating_phylogenies.html#dating" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/clock.png" style="width:200px" />
</center>
<p>We will use IQ-TREE to date the divergences in our tree. IQ-TREE implements the least-square dating (LSD2) dating procedure and algorithm of <a href="https://academic.oup.com/sysbio/article/65/1/82/2461506?login=true">To <em>et al.</em> (2016)</a>. This approach and IQ-TREE allow us to date the phylogeny using tip dates, e.g., known sampling dates of the terminal taxa as one would have in a study of viral evolution, or by using ancestral dates. This procedure will work as long as you can place a date or age for the most-recent common ancestor (MRCA) on one node of the tree. This date or age can be derived from fossil data or some biogeographic, geological or historical event.</p>
<div id="setting-up-the-date-file" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Setting up the date file<a href="08-Dating_phylogenies.html#setting-up-the-date-file" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For our data and situation, we will use the ancestral dating approach. As a first step, we will need to prepare a 'date file' specifying the date record of the ancestral nodes or at least one node in the tree. This has been prepared already.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb46-1"><a href="08-Dating_phylogenies.html#cb46-1" tabindex="-1"></a><span class="co"># Check whether you are in the &#39;iqtree&#39; directory</span></span>
<span id="cb46-2"><a href="08-Dating_phylogenies.html#cb46-2" tabindex="-1"></a><span class="bu">pwd</span></span>
<span id="cb46-3"><a href="08-Dating_phylogenies.html#cb46-3" tabindex="-1"></a><span class="co"># Open the dating file</span></span>
<span id="cb46-4"><a href="08-Dating_phylogenies.html#cb46-4" tabindex="-1"></a><span class="fu">less</span> manta_phylogeny.dating</span></code></pre></div>
<p>Again, this is a tab-delimited file. In the left column, we have the list of taxa that share a MRCA (i.e., all the decendent taxa of the node we are constraining by date). The age of that node or MRCA in millions of years before present (hence specified as a negative value) is in the right column. In setting this up, it is important that the taxa names match those in our alignment file ('manta_phylogeny.phylip') exactly. These should be separated by commas and there should be no spaces (as spaces can also be used to delimit fields).</p>
<center>
<img src="figures/sharktooth.png" style="width:200px; background:white; border-radius:5px" />
</center>
<p>Part of the reason for the taxonomic uncertainty in this group is that it has undergone very rapid and recent speciation, and Hosegood <em>et al.</em>'s (2020) results also suggested this. <a href="https://www.sciencedirect.com/science/article/pii/S1055790314003637?via%3Dihub">Poortvliet <em>et al.</em> (2015)</a> provided a dated phylogeny, based on mitogenomes, for the manta- and devil-rays, and discussed the fossil evidence used as calibrations. Fossils of extant manta- and devil-rays are recent and only date back to between 11.6 and 3.6 million years ago (Mya). Fossil teeth of <em>Mobula mobular</em> and <em>Mobula japanica</em> date to between 11.6 and 3.6 Mya. The first fossils of <em>Manta</em> appeared 4.8 Mya. Based on molecular and fossil evidence, it is thought that <em>Manta alfredi</em> and <em>Manta birostris</em> diverged within the last one million years.</p>
<p>Based on this information and the species and relationships in our phylogeny, we can specify:</p>
<ol style="list-style-type: decimal">
<li>The age of the MRCA of <em>Manta birostris</em> and <em>Manta alfredi</em> to be as young as 1 million years or to be at least 4.8 million years old</li>
<li>The age of the MRCA of <em>Mobula mobular</em> and <em>Mobula japanica</em> to be at most 11.6 million years</li>
</ol>
<p>All these constraints are included in the date file ('manta_phylogeny.dating'), but we will comment out (using '#') those that we don't want to apply.</p>
<div class="webex-solution">
<button>
Annotation with #
</button>
<p>By putting a '#' in front of a piece of code, text or data, we turn that into annotation. This is not read or executed as a command, but is a rather handy tool for making notes in code (such as comments or reminders to ourselves as to what a command does, what settings to use or how to change parameters). It is also useful for "blocking out" bits of code or data that we don't want to run or use. It is much easier to just remove the '#' when we need to than to try to remember what we deleted before or to have multiple copies of essentially the same script.</p>
</div>
</div>
<div id="running-the-analysis" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Running the analysis<a href="08-Dating_phylogenies.html#running-the-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="dating-our-tree" class="section level3 hasAnchor" number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Dating our tree<a href="08-Dating_phylogenies.html#dating-our-tree" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/calendar.png" style="width:200px; background:white;border-radius:5px; border:white solid 5px" />
</center>
<p>Starting the analysis in IQ-TREE is straightforward:</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb47-1"><a href="08-Dating_phylogenies.html#cb47-1" tabindex="-1"></a><span class="ex">iqtree</span> <span class="at">-s</span> manta_phylogeny.phylip <span class="dt">\</span></span>
<span id="cb47-2"><a href="08-Dating_phylogenies.html#cb47-2" tabindex="-1"></a>--date manta_phylogeny.dating <span class="dt">\</span></span>
<span id="cb47-3"><a href="08-Dating_phylogenies.html#cb47-3" tabindex="-1"></a>--date-tip 0 <span class="dt">\</span></span>
<span id="cb47-4"><a href="08-Dating_phylogenies.html#cb47-4" tabindex="-1"></a>-redo</span></code></pre></div>
<p>In this command:</p>
<ul>
<li><code>-s</code> points to our sequence alignment (as before).</li>
<li><code>--date</code> specifies our date file.</li>
<li><code>--date-tip</code>: This indicates that all our terminal taxa are extant and present at the present date ("time zero"). You could optimistically have ancient DNA from an extinct taxon or a mixture of contemporary tip dates and ancestral dates in a viral study.</li>
<li><strong>Remember</strong> <code>-redo</code>. Our output and log files from the previous chapter are still in the folder.</li>
</ul>
<p>This will run through the full analysis, as we did before. It will use ModelFinder to find the optimal model and parameters of sequence evolution and find a maximum-likelihood tree. The branch lengths will represent the number of substitutions per site. For the dating, it will rescale the branches, using our time constraints, to build a time-calibrated tree.</p>
<p>Once run, a few results are output to the terminal. These include the 'tMRCA', which is the age of the root of the tree, and a substitution rate at the root as the number of subtitutions per site along that branch of the tree.</p>
What is age (approximatety) of the root of the tree?
<div id="radio_NHBYBJEYTB" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_NHBYBJEYTB" value=""></input> <span>310 million years</span></label><label><input type="radio" autocomplete="off" name="radio_NHBYBJEYTB" value="answer"></input> <span>225 million years</span></label><label><input type="radio" autocomplete="off" name="radio_NHBYBJEYTB" value=""></input> <span>185 million years</span></label><label><input type="radio" autocomplete="off" name="radio_NHBYBJEYTB" value=""></input> <span>125 million years</span></label>
</div>
<p>Alongside the typical IQ-TREE output we looked at in the last section, this analysis will also produce three 'timetree' files:</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb48-1"><a href="08-Dating_phylogenies.html#cb48-1" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-l</span></span></code></pre></div>
<p>These are:</p>
<ul>
<li>'manta_phylogeny.phylip.timetree.lsd', which reports the output of the LSD2 process,</li>
<li>'manta_phylogeny.phylip.timetree.nex', the time-tree in a nexus tree format, and</li>
<li>'manta_phylogeny.phylip.timetree.nwk', the time-tree in newick format.</li>
</ul>
<center>
<img src="figures/icytree.png" style="max-width: 300px; border-radius: 15px" />
</center>
<p>We will use a web-browser and <a href="https://academic.oup.com/bioinformatics/article/33/15/2392/3604867">IcyTree</a> to examine our time-calibrated tree.</p>
<ol style="list-style-type: decimal">
<li><p>Right-click in the grey area outside your terminal window and select <code>Firefox</code>. This will open the web browser.</p></li>
<li><p>Navigate to 'icytree.org'.</p></li>
<li><p>Use the <code>File</code> button at the top to navigate to and load your nexus tree. Click on the <code>nsc-xxx</code> button that corresponds to your webVNC account, then open your <code>phylogenomics</code> and <code>iqtree</code> directories to find the nexus tree.</p></li>
<li><p>You are welcome to play around with the various functions and style of the tree, but to see the dates of our nodes and divergences, click the <code>Style</code> button, then <code>Internal node text</code> and <code>date</code>.</p></li>
<li><p>Do not close Firefox quite yet.</p></li>
</ol>
<p>We can now take a quick look to see if changing one of the ancestral node constraints or dates changes the remaining dates in the tree. Return to your main webVNC terminal window.</p>
<p>We'll use <a href="https://www.nano-editor.org/">nano</a>, a Linux text editor, to quickly edit the 'manta_phylogeny.dating' file.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb49-1"><a href="08-Dating_phylogenies.html#cb49-1" tabindex="-1"></a><span class="fu">nano</span> manta_phylogeny.dating</span></code></pre></div>
<p>This will open the file within your terminal. You can edit in this as you would a normal text document (but you'll need to use your cursor rather than your mouse/trackpad to move around).</p>
<ol style="list-style-type: decimal">
<li><p>Delete the '#' from the second line, and place a '#' at the start of the first line. The age of <em>Manta</em> used in the analysis will now be 4.8 million years rather than one million, as the first line will be ignored.</p></li>
<li><p>Press 'ctrl + x'.</p></li>
<li><p>Press 'y' to save the changes.</p></li>
<li><p>Press 'Enter' to use the existing file name. Hopefully, this shows the benefits of using annotations rather than saving lots of files under different names.</p></li>
</ol>
<p>Now you can execute the analysis again!</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb50-1"><a href="08-Dating_phylogenies.html#cb50-1" tabindex="-1"></a><span class="ex">iqtree</span> <span class="at">-s</span> manta_phylogeny.phylip <span class="dt">\</span></span>
<span id="cb50-2"><a href="08-Dating_phylogenies.html#cb50-2" tabindex="-1"></a>--date manta_phylogeny.dating <span class="dt">\</span></span>
<span id="cb50-3"><a href="08-Dating_phylogenies.html#cb50-3" tabindex="-1"></a>--date-tip 0 <span class="dt">\</span></span>
<span id="cb50-4"><a href="08-Dating_phylogenies.html#cb50-4" tabindex="-1"></a>-redo</span></code></pre></div>
<p>Go back to your browser and open a new tab in Firefox. You can open IcyTree and load the latest nexus tree file (it will have the same name) as you did above. Display the dates on the nodes. By flipping between the tabs you can compare the trees to see if there are any obvious or large differences or if the dating was fairly robust to these changes.</p>
</div>
<div id="getting-confidence-intervals-around-our-dates" class="section level3 hasAnchor" number="8.2.2">
<h3><span class="header-section-number">8.2.2</span> Getting confidence intervals around our dates<a href="08-Dating_phylogenies.html#getting-confidence-intervals-around-our-dates" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>IQ-TREE uses a procedure that resamples and combines a mixed Poisson and log-normal distribution of the relaxed clock model. As it is not published yet, it is probably best to use this with caution and to keep an eye on the literature.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb51-1"><a href="08-Dating_phylogenies.html#cb51-1" tabindex="-1"></a><span class="ex">iqtree</span> <span class="at">-s</span> manta_phylogeny.phylip <span class="dt">\</span></span>
<span id="cb51-2"><a href="08-Dating_phylogenies.html#cb51-2" tabindex="-1"></a>--date manta_phylogeny.dating <span class="dt">\</span></span>
<span id="cb51-3"><a href="08-Dating_phylogenies.html#cb51-3" tabindex="-1"></a>--date-ci 100 <span class="dt">\</span></span>
<span id="cb51-4"><a href="08-Dating_phylogenies.html#cb51-4" tabindex="-1"></a>-redo</span></code></pre></div>
<p>The only real difference to our earlier code is the argument <code>--date-ci</code>, which specifies how many resampling replicates we want to use. One can also set the standard deviation of the log-normal distribution (through <code>--clock-sd</code>), but increasing this can make the confidence intervals wider. We will use the default (0.2) here and will not specify this argument.</p>
<p>Once this is run, you can again use IcyTree to open the nexus time-tree file. To display the 95% confidence intervals around the dates we calculated before (with <em>Manta</em> constrained to be 4.8 million years old), click the <code>Style</code> button, select <code>Internal node text</code> and <code>CI_date</code>. This will place the minimum and maximum age on each nodes. You are welcome to repeat the analysis with the alternative dating for <em>Manta</em>.</p>
<p>Do these look reasonable or are they quite wide? Do these suggest our estimated dates are accurate?</p>
<p>They're unlikely to be, given the limitations of the current analysis and our sparse data. In comparing the output, the different dating constraints impacted the older nodes more than the recent ones. The confidence intervals are quite wide.</p>
<p>Generally, we'd expect better dating estimates and narrower confidence intervals</p>
<ul>
<li>on nodes that are temporally- or phylogenetically-close to those that we are constraining,</li>
<li>if we have accurate nodal constraints available,</li>
<li>if we were able to constrain more nodes, and</li>
<li>if our data are relatively free from conflict.</li>
</ul>
<p>You have now dated your phylogeny, and can use these dates to examine various evolutionary, historical or biogeographic scenarios.</p>
</div>
<div id="lineage-through-time-plots" class="section level3 hasAnchor" number="8.2.3">
<h3><span class="header-section-number">8.2.3</span> Lineage-Through-Time plots<a href="08-Dating_phylogenies.html#lineage-through-time-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/LTT.png" style="max-width: 300px; border-radius: 15px" />
</center>
<p>Another nice way to contextualise your phylogeny in terms of evolutionary history, and to examine patterns of speciation and extinction over time is through <strong>Lineage Through Time</strong> (LTT) plots. These show how the lineages in our phylogenetic tree have accumulated or were lost over evolutionary time. There are specialist packages to construct and evaluate these more rigorously (including modules for BEAST, which is introduced in the next chapter), but IcyTree also offers a quick visualisation.</p>
<p>For any of the time-trees that you have open in IcyTree, click the <code>Statistics</code> button and select <code>Lineages-through-time plot</code>. In the plot, you'll see a burst of speciation around 40 Mya and, depending on which you have opened (you are welcome to examine and compare both), and an older period of diversification 150 Mya or 200 Mya. We may be prompted to look at some geological or palaeoclimatic events as possible drivers of this.</p>
<p>In the next chapter, we will return to testing species models, specifically using the Multispecies Coalescent.</p>

</div>
</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="07-Topology_testing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="09-Delimiting_species.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
