<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A Mamba installs | Phylogenomics</title>
  <meta name="description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="A Mamba installs | Phylogenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Phylogenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Mamba installs | Phylogenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Gavin Gouws, Matthew Gemmell, Ewan Harney, Helen Hipperson, Kathryn Maher" />


<meta name="date" content="2025-07-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="09-Delimiting_species.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Phylogenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Intro.html"><a href="01-Intro.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-ddradseq-data-of-manta-and-devil-rays"><i class="fa fa-check"></i><b>2.1</b> Example data: ddRADseq data of manta and devil rays</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#manta--and-devil-ray-ddrad-data-set"><i class="fa fa-check"></i><b>4.2</b> Manta- and devil-ray ddRAD data set</a></li>
<li class="chapter" data-level="4.3" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality Control</a></li>
<li class="chapter" data-level="4.5" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#post-qc-check"><i class="fa fa-check"></i><b>4.5</b> Post QC Check</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html"><i class="fa fa-check"></i><b>5</b> Preparing data in Stacks</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#stacks-1"><i class="fa fa-check"></i><b>5.1</b> Stacks</a></li>
<li class="chapter" data-level="5.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#demultiplexing-data-with-process_radtags"><i class="fa fa-check"></i><b>5.2</b> Demultiplexing data with <code>process_radtags</code></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-barcodes"><i class="fa fa-check"></i><b>5.2.1</b> <code>process_radtags</code>: barcodes</a></li>
<li class="chapter" data-level="5.2.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-prep"><i class="fa fa-check"></i><b>5.2.2</b> <code>process_radtags</code>: prep</a></li>
<li class="chapter" data-level="5.2.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-run"><i class="fa fa-check"></i><b>5.2.3</b> <code>process_radtags</code>: run</a></li>
<li class="chapter" data-level="5.2.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-mcqs"><i class="fa fa-check"></i><b>5.2.4</b> <code>process_radtags</code>: MCQs</a></li>
<li class="chapter" data-level="5.2.5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#process_radtags-output"><i class="fa fa-check"></i><b>5.2.5</b> <code>process_radtags</code>: output</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#building-the-catalogue-loci-with-denovo_map.pl"><i class="fa fa-check"></i><b>5.3</b> Building the catalogue loci with <code>denovo_map.pl</code></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-input"><i class="fa fa-check"></i><b>5.3.1</b> <code>denovo_map.pl</code>: input</a></li>
<li class="chapter" data-level="5.3.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-run"><i class="fa fa-check"></i><b>5.3.2</b> <code>denovo_map.pl</code>: run</a></li>
<li class="chapter" data-level="5.3.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-output"><i class="fa fa-check"></i><b>5.3.3</b> <code>denovo_map.pl</code>: output</a></li>
<li class="chapter" data-level="5.3.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#denovo_map.pl-mcqs"><i class="fa fa-check"></i><b>5.3.4</b> <code>denovo_map.pl</code>: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#preparing-a-sequence-alignment-file-in-populations"><i class="fa fa-check"></i><b>5.4</b> Preparing a sequence alignment file in <code>populations</code></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html"><i class="fa fa-check"></i><b>6</b> Model selection, tree reconstruction and support evaluation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#estimating-a-best-fit-model-of-nucleotide-or-sequence-evolution"><i class="fa fa-check"></i><b>6.1</b> Estimating a best-fit model of nucleotide or sequence evolution</a></li>
<li class="chapter" data-level="6.2" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#constructing-a-maximum-likelihood-phylogenetic-tree"><i class="fa fa-check"></i><b>6.2</b> Constructing a maximum-likelihood phylogenetic tree</a></li>
<li class="chapter" data-level="6.3" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#calculating-nodal-support"><i class="fa fa-check"></i><b>6.3</b> Calculating nodal support</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Topology_testing.html"><a href="07-Topology_testing.html"><i class="fa fa-check"></i><b>7</b> Hypothesis and topology testing</a></li>
<li class="chapter" data-level="8" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html"><i class="fa fa-check"></i><b>8</b> Dating phylogenies</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#setting-up-the-date-file"><i class="fa fa-check"></i><b>8.1</b> Setting up the date file</a></li>
<li class="chapter" data-level="8.2" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#running-the-analysis"><i class="fa fa-check"></i><b>8.2</b> Running the analysis</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#dating-our-tree"><i class="fa fa-check"></i><b>8.2.1</b> Dating our tree</a></li>
<li class="chapter" data-level="8.2.2" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#getting-confidence-intervals-around-our-dates"><i class="fa fa-check"></i><b>8.2.2</b> Getting confidence intervals around our dates</a></li>
<li class="chapter" data-level="8.2.3" data-path="08-Dating_phylogenies.html"><a href="08-Dating_phylogenies.html#lineage-through-time-plots"><i class="fa fa-check"></i><b>8.2.3</b> Lineage-Through-Time plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html"><i class="fa fa-check"></i><b>9</b> Species delimitation with the Multispecies Coalescent</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#beast2"><i class="fa fa-check"></i><b>9.1</b> beast2</a></li>
<li class="chapter" data-level="9.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#gene-tree---species-tree-conflicts"><i class="fa fa-check"></i><b>9.2</b> Gene tree - species tree conflicts</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis"><i class="fa fa-check"></i><b>9.2.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#starting-the-analysis"><i class="fa fa-check"></i><b>9.2.2</b> Starting the analysis</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#examining-the-results"><i class="fa fa-check"></i><b>9.2.3</b> Examining the results</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#bayes-factor-delimitation"><i class="fa fa-check"></i><b>9.3</b> Bayes Factor Delimitation</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis-1"><i class="fa fa-check"></i><b>9.3.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="9.3.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#running-the-analysis-1"><i class="fa fa-check"></i><b>9.3.2</b> Running the analysis</a></li>
<li class="chapter" data-level="9.3.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#selecting-the-taxonomic-hypothesis"><i class="fa fa-check"></i><b>9.3.3</b> Selecting the taxonomic hypothesis</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="10-Appendix.html"><a href="10-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="10-Appendix.html"><a href="10-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="10-Appendix.html"><a href="10-Appendix.html#loops"><i class="fa fa-check"></i><b>B</b> Wildcards and loops</a>
<ul>
<li class="chapter" data-level="B.1" data-path="10-Appendix.html"><a href="10-Appendix.html#a-simple-wildcard"><i class="fa fa-check"></i><b>B.1</b> A simple wildcard</a></li>
<li class="chapter" data-level="B.2" data-path="10-Appendix.html"><a href="10-Appendix.html#loops-1"><i class="fa fa-check"></i><b>B.2</b> Loops</a>
<ul>
<li class="chapter" data-level="B.2.1" data-path="10-Appendix.html"><a href="10-Appendix.html#an-example-of-a-simple-loop"><i class="fa fa-check"></i><b>B.2.1</b> An example of a simple loop</a></li>
<li class="chapter" data-level="B.2.2" data-path="10-Appendix.html"><a href="10-Appendix.html#stacks-process_radtags-loop-example"><i class="fa fa-check"></i><b>B.2.2</b> Stacks <code>process_radtags</code> loop example</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="C" data-path="10-Appendix.html"><a href="10-Appendix.html#optimising-loci-reconstruction-in-denovo_map.pl-in-stacks"><i class="fa fa-check"></i><b>C</b> Optimising loci reconstruction in <code>denovo_map.pl</code> in Stacks</a></li>
<li class="chapter" data-level="D" data-path="10-Appendix.html"><a href="10-Appendix.html#additional-resources"><i class="fa fa-check"></i><b>D</b> Additional resources</a>
<ul>
<li class="chapter" data-level="D.1" data-path="10-Appendix.html"><a href="10-Appendix.html#snp-variant-calling"><i class="fa fa-check"></i><b>D.1</b> SNP variant calling</a></li>
<li class="chapter" data-level="D.2" data-path="10-Appendix.html"><a href="10-Appendix.html#sequence-alignment-tools"><i class="fa fa-check"></i><b>D.2</b> Sequence alignment tools</a></li>
<li class="chapter" data-level="D.3" data-path="10-Appendix.html"><a href="10-Appendix.html#model-selection-tools"><i class="fa fa-check"></i><b>D.3</b> Model selection tools</a></li>
<li class="chapter" data-level="D.4" data-path="10-Appendix.html"><a href="10-Appendix.html#phylogeny-reconstruction-tools"><i class="fa fa-check"></i><b>D.4</b> Phylogeny reconstruction tools</a></li>
<li class="chapter" data-level="D.5" data-path="10-Appendix.html"><a href="10-Appendix.html#r-packages"><i class="fa fa-check"></i><b>D.5</b> R-packages</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Phylogenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mamba-installs" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">A</span> Mamba installs<a href="10-Appendix.html#mamba-installs" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/mamba_logo.png" style="width: 200px; border-radius: 5px" />
</center>
<div id="mamba_install" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">A.1</span> Mamba installation and environment<a href="10-Appendix.html#mamba_install" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Mamba is a re-implementation of conda. It is a great tool for installing bioinformatic packages, including R packages, and creating the environment in which to run them.</p>
<p>Mamba github: <a href="https://github.com/mamba-org/mamba" class="uri">https://github.com/mamba-org/mamba</a></p>
<p>Mamba installation: <a href="https://github.com/conda-forge/miniforge#mambaforge" class="uri">https://github.com/conda-forge/miniforge#mambaforge</a></p>
<p>Mamba guide: <a href="https://mamba.readthedocs.io/en/latest/user_guide/mamba.html" class="uri">https://mamba.readthedocs.io/en/latest/user_guide/mamba.html</a></p>
<p>To create the mamba environment <code>phylogenomics</code>, identical to the one we have used in this workshop, run the following commands in your bash. You will need to have installed <code>mamba</code> first.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb63-1"><a href="10-Appendix.html#cb63-1" tabindex="-1"></a><span class="co">#phylogenomics</span></span>
<span id="cb63-2"><a href="10-Appendix.html#cb63-2" tabindex="-1"></a><span class="ex">mamba</span> create <span class="at">-n</span> phylogenomics</span>
<span id="cb63-3"><a href="10-Appendix.html#cb63-3" tabindex="-1"></a><span class="ex">mamba</span> activate phylogenomics</span>
<span id="cb63-4"><a href="10-Appendix.html#cb63-4" tabindex="-1"></a><span class="co">#Install the packages you need from bioconda channel</span></span>
<span id="cb63-5"><a href="10-Appendix.html#cb63-5" tabindex="-1"></a><span class="ex">mamba</span> install <span class="at">-c</span> bioconda fastqc trimmomatic stacks iqtree beast2</span></code></pre></div>
<p>Once the environment is created and the software is installed, you can activate the environment:</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb64-1"><a href="10-Appendix.html#cb64-1" tabindex="-1"></a><span class="ex">conda</span> activate phylogenomics</span></code></pre></div>
<p>In the active environment, you can begin you your analyses, as well as install the beast2 add-ons used (SNAPP and MODEL-SELECTOR) in the workbook or any others you may need. To do this, open BEAUti:</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb65-1"><a href="10-Appendix.html#cb65-1" tabindex="-1"></a><span class="ex">beauti</span></span></code></pre></div>
<p>This will open a GUI window, as in Chapter 9. Go to <code>File</code> (on the top menu) and select <code>Manage Packages</code>. This will open a window listing all the plug-ins available to install or already installed. You can select those you wish to install or update and click <code>Install/Update</code>.</p>
</div>
</div>
<div id="loops" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">B</span> Wildcards and loops<a href="10-Appendix.html#loops" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/loop.png" style="width:250px; background:white; border-radius:5px" />
</center>
<p>In the main workbook, we ran the quality assessment and quality control on a single set of paired-end reads (the R1 and R2 files) of multiplexed RADseq data. Your sequencing output may consist of many paired read-files. You may wonder if there is a simpler way to run fastQC on multiple sets of fastq files. Likewise, you may wonder how you would go about running <code>process_radtags</code> in Stacks on multiple pairs of files.</p>
<p>You really don't want to type the same command multiple times (even using the Linux history). Besides being tedious and time-consuming, it is a recipe for losing focus, and for errors and typos to creep in.</p>
<p>Fortunately, there are ways around this - like using wildcards or loops.</p>
<div id="a-simple-wildcard" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">B.1</span> A simple wildcard<a href="10-Appendix.html#a-simple-wildcard" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A wildcard is a character that can be used to represent and stand in a 'placeholder' for any number of single or multiple characters. A simple and versatile one is <code>*</code>; other wildcards are explained <a href="https://tldp.org/LDP/GNU-Linux-Tools-Summary/html/x11655.htm">here</a>. This could easily be used in the fastQC example. By executing the command</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb66-1"><a href="10-Appendix.html#cb66-1" tabindex="-1"></a><span class="ex">fastqc</span> <span class="at">-o</span> fastqc_output <span class="pp">*</span>.fastq.gz</span></code></pre></div>
<p>we are asking fastQC to process any file that has the 'fastq.gz' extension, with the <code>*</code> representing all the possible filename prefixes that are present in the directory.</p>
</div>
<div id="loops-1" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">B.2</span> Loops<a href="10-Appendix.html#loops-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Another approach is to run a loop which will cycle through each of your samples and run all the commands that you specify.</p>
<p>Put simply, the syntax for a <code>for loop</code> is as follows:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb67-1"><a href="10-Appendix.html#cb67-1" tabindex="-1"></a><span class="cf">for</span> <span class="op">&lt;</span>variable <span class="ex">name</span><span class="op">&gt;</span> in <span class="op">&lt;</span>a list of items<span class="op">&gt;</span><span class="kw">;</span></span>
<span id="cb67-2"><a href="10-Appendix.html#cb67-2" tabindex="-1"></a><span class="cf">do</span> <span class="op">&lt;</span>run <span class="ex">a</span> command<span class="op">&gt;</span> $<span class="op">&lt;</span>variable name<span class="op">&gt;</span><span class="kw">;</span></span>
<span id="cb67-3"><a href="10-Appendix.html#cb67-3" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>The <code>&lt;variable name&gt;</code> is the name of the variable you will use in the <code>do</code> part of your loop. It contains the item in your list the loop is currently using at that point. In the example of a simple loop below, we call this variable <code>f</code>, but you could use any name as long as you use it consistently in the <code>do</code> section.</p>
<p>The <code>&lt;a list of items&gt;</code> is anything that returns a list of items, e.g. a list of files in our case.</p>
<div id="an-example-of-a-simple-loop" class="section level3 hasAnchor" number="11.2.1">
<h3><span class="header-section-number">B.2.1</span> An example of a simple loop<a href="10-Appendix.html#an-example-of-a-simple-loop" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/data_files.png" style="width:100px" />
</center>
<p>Below is an example of a loop, which can be adjusted for your samples, the software and the code you are running. This simplified loop will clearly demonstrate what the first part of the code is doing.</p>
<p>Make sure that you are in a terminal window and in the directory <code>~/phylogenomics/demultiplexed_data</code>. (This directory contains demultiplexed RADseq data in paired read files for five samples).</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb68-1"><a href="10-Appendix.html#cb68-1" tabindex="-1"></a><span class="co">#This command loops through all files ending in .1.fq.gz</span></span>
<span id="cb68-2"><a href="10-Appendix.html#cb68-2" tabindex="-1"></a><span class="cf">for</span> f <span class="kw">in</span> <span class="pp">*</span>.1.fq.gz<span class="kw">;</span> </span>
<span id="cb68-3"><a href="10-Appendix.html#cb68-3" tabindex="-1"></a><span class="co">#Sets file name as variable &#39;$f&#39;, </span></span>
<span id="cb68-4"><a href="10-Appendix.html#cb68-4" tabindex="-1"></a><span class="co">#It also adds it to a variable called &#39;BASE&#39; removing the &#39;.1.fq.gz&#39; extension</span></span>
<span id="cb68-5"><a href="10-Appendix.html#cb68-5" tabindex="-1"></a><span class="cf">do</span> <span class="va">BASE</span><span class="op">=</span><span class="va">${f</span><span class="op">%</span>.1.fq.gz<span class="va">}</span><span class="kw">;</span> </span>
<span id="cb68-6"><a href="10-Appendix.html#cb68-6" tabindex="-1"></a>    <span class="co">#Print the file name ($f) to the terminal</span></span>
<span id="cb68-7"><a href="10-Appendix.html#cb68-7" tabindex="-1"></a>    <span class="bu">echo</span> <span class="va">$f</span><span class="kw">;</span> </span>
<span id="cb68-8"><a href="10-Appendix.html#cb68-8" tabindex="-1"></a>    <span class="co">#Print the ${BASE} variable to the terminal</span></span>
<span id="cb68-9"><a href="10-Appendix.html#cb68-9" tabindex="-1"></a>    <span class="bu">echo</span> <span class="va">${BASE}</span><span class="kw">;</span> </span>
<span id="cb68-10"><a href="10-Appendix.html#cb68-10" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>By using <code>echo</code> to print the following variables to the console we can see:</p>
<ul>
<li><code>echo $f</code>: prints the full filename.</li>
<li><code>echo ${BASE}</code>: prints just the start of the filename (without the '.1.fq.gz' extension, as specified in our code)</li>
</ul>
<p>The loop then repeats the commands on the next file in the list and will keep going until it has run through all files ending with '1.fq.gz' in the directory.</p>
<p><code>done</code> specifies the end of the commands in the loop.</p>
<p>All commands between <code>do</code> and <code>done</code> in the loop will be executed. In this case we only have five files but this would work with any number of files. This is a useful bit of code as it means we can keep the filename but change the ending/extension of the file in our loops. This is especially useful when a command will produce output with a different file extension, which we would then need to use as input in the next command in the loop.</p>
</div>
<div id="stacks-process_radtags-loop-example" class="section level3 hasAnchor" number="11.2.2">
<h3><span class="header-section-number">B.2.2</span> Stacks <code>process_radtags</code> loop example<a href="10-Appendix.html#stacks-process_radtags-loop-example" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<center>
<img src="figures/quality_trimming_and_filtering.png" style="width:200px" />
</center>
<p>As a more practical example related to the workbook exercise, we can write a loop to run the Stacks module <code>process_radtags</code> to work through and demultiplex several paired read files and assign sequences to samples, using the code above as a base.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb69-1"><a href="10-Appendix.html#cb69-1" tabindex="-1"></a><span class="co">#This will loop through all the files ending in &#39;_001_R1.fastq.gz&#39;</span></span>
<span id="cb69-2"><a href="10-Appendix.html#cb69-2" tabindex="-1"></a><span class="cf">for</span> f <span class="kw">in</span> <span class="pp">*</span>_001_R1.fastq.gz<span class="kw">;</span> </span>
<span id="cb69-3"><a href="10-Appendix.html#cb69-3" tabindex="-1"></a><span class="co">#Assign the file name &#39;$f&#39; to a variable called &#39;BASE&#39; and removes the extension &#39;_001_R1.fastq.gz&#39;. </span></span>
<span id="cb69-4"><a href="10-Appendix.html#cb69-4" tabindex="-1"></a><span class="co">#This means that you can use this filename to refer to files with other extensions</span></span>
<span id="cb69-5"><a href="10-Appendix.html#cb69-5" tabindex="-1"></a><span class="cf">do</span> <span class="va">BASE</span><span class="op">=</span><span class="va">${f</span><span class="op">%</span>_001_R1.fastq.gz<span class="va">}</span><span class="kw">;</span> </span>
<span id="cb69-6"><a href="10-Appendix.html#cb69-6" tabindex="-1"></a>    <span class="ex">process_radtags</span> <span class="at">-1</span> <span class="va">${BASE}</span>_001_R1.fastq.gz <span class="at">-2</span> <span class="va">${BASE}</span>_001_R2.fastq.gz <span class="at">-P</span> <span class="dt">\</span></span>
<span id="cb69-7"><a href="10-Appendix.html#cb69-7" tabindex="-1"></a>    <span class="at">-i</span> gzfastq <span class="dt">\</span></span>
<span id="cb69-8"><a href="10-Appendix.html#cb69-8" tabindex="-1"></a>    <span class="at">-b</span> <span class="va">${BASE}</span>.barcodes <span class="dt">\</span></span>
<span id="cb69-9"><a href="10-Appendix.html#cb69-9" tabindex="-1"></a>    <span class="at">-o</span> processed_data/ <span class="dt">\</span></span>
<span id="cb69-10"><a href="10-Appendix.html#cb69-10" tabindex="-1"></a>    <span class="at">--inline-null</span> <span class="at">--renz_1</span> SbfI <span class="at">--filter_illumina</span> <span class="dt">\</span></span>
<span id="cb69-11"><a href="10-Appendix.html#cb69-11" tabindex="-1"></a>    <span class="at">-c</span> <span class="at">-q</span> <span class="at">-r</span> <span class="dt">\</span></span>
<span id="cb69-12"><a href="10-Appendix.html#cb69-12" tabindex="-1"></a>    <span class="at">--adapter_1</span> AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATC <span class="at">--adapter_mm</span> 2<span class="kw">;</span></span>
<span id="cb69-13"><a href="10-Appendix.html#cb69-13" tabindex="-1"></a><span class="co">#Ends the commands. Here we had only ran one command (`process_radtags`), but we can loop many together.</span></span>
<span id="cb69-14"><a href="10-Appendix.html#cb69-14" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>In the above code, we use the start of the file name that is kept as a variable in '{BASE}' and then specify different extensions to specify the two files of the read pair that we want to run <code>process_radtags</code> on in that iteration of the loop. We also use the start of the filename in '{BASE}' to refer to the corresponding barcode file. For this command to work, you will need to set up a separate barcode file corresponding to each set of reads beforehand, and name it using the filename prefix that is assigned in '{BASE}'.</p>
<p>This code can be edited to run other software in a loop. A nice basic intro to Linux loops can be found <a href="https://www.redhat.com/sysadmin/bash-scripting-loops">here</a>.</p>
</div>
</div>
</div>
<div id="optimising-loci-reconstruction-in-denovo_map.pl-in-stacks" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">C</span> Optimising loci reconstruction in <code>denovo_map.pl</code> in Stacks<a href="10-Appendix.html#optimising-loci-reconstruction-in-denovo_map.pl-in-stacks" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>As indicated in the main workbook, Stacks offers a <em>de novo</em> pipeline for assembling loci and calling variants (SNPs) in order to genotype individuals in the absence of a high-quality reference genome. There are, however, parameters that need to be optimised beforehand to increase accuracy in reconstruction and to maximise the number of loci prior to running <code>denovo_map.pl</code> on the full set of study samples.</p>
<p>The procedure is explained in a book chapter by <a href="https://link.springer.com/protocol/10.1007/978-1-0716-2313-8_7#DOI">Rivera-Colon &amp; Catchen (2022)</a>. The chapter provides a handy introduction to designing a RADseq study, and using the various features, modules and pipelines of Stacks to analyse the resulting data. It also links through to some handy R-script for visualising and evaluating your data as it passes through Stacks.</p>
<p>There are two main parameters that need to be optimised. The first is <code>-M</code>, which is the number of mismatches that <code>ustacks</code> allows when combining stacks (matched sequence reads that represent an allele within an sample) into loci in that sample. If this is set too low, alleles won't be collapsed into loci, and these will be kept apart as separate homozygous loci. If set too high, alleles belonging to different loci will be combined as a single locus. Once loci are assembled in an individual, <code>cstacks</code> combines all the loci from those individuals into a catalogue of loci for the metapopulation. Here, <code>-n</code>, which is the maximum number of mismatches between individual loci and those in the catalogue, needs to be optimised. If <code>-n</code> is too low, loci in the catalogue will be "undermerged" (a locus that is homozygous for one allele will not be merged with an orthologous locus that is homozygous for an alternative allele, and these will be retained as two loci). If too high, different loci will be "overmerged" and considered the same in the catalogue.</p>
<p>Their procedure involves optimising these parameters so that the number of R80 loci (loci present in at least 80% of the metapopulation) is maximised. The <code>denovo_map.pl</code> is run repeatedly, with <code>-M</code> and <code>-n</code> set to 1 and increased in each subsequent iteration (until <code>-M</code> = <code>-n</code> = 10 or 12, depending on the data and biological context). The change (increase or decrease) in the number of R80 loci is then evaluated across the runs. At the optimal value for these parameters, there will be no or negligible change in the number of loci. Above the optimum, the number of R80 loci should decrease again, as in the figure (taken from the book chapter) below.</p>
<center>
<img src="figures/R80loci_graph.png" style="width:500px; border-radius:5px" />
</center>
<p>Selecting the samples on which to optimise these parameters is an important consideration. You obviously don't want to optimise using all your data, unless you have limitless time. These samples you choose should be representative of your library or some biological reality (e.g., representatives of different populations for a population genomic study). You'd want samples that are fairly similar in terms of the amount of input data. Choosing samples with high coverage is sensible, as these will lead to the most accurate allele calling. You may be tempted to include those samples with the highest numbers of reads, disregarding coverage, but in this case you may be optimising for loci that are not going to be present in those samples with lower numbers of reads. There is no "golden rule" and each data set will be different and subject to its own quirks. It is always advisable to do some exploration here and to remember that your results are always a hypothesis, based on the data and a number of assumptions that you make through the course of the analysis.</p>
</div>
<div id="additional-resources" class="section level1 hasAnchor" number="13">
<h1><span class="header-section-number">D</span> Additional resources<a href="10-Appendix.html#additional-resources" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Here are some additional tools and resources for compiling data, determining models of sequence evolution, and constructing phylogenetic trees.</p>
<div id="snp-variant-calling" class="section level2 hasAnchor" number="13.1">
<h2><span class="header-section-number">D.1</span> SNP variant calling<a href="10-Appendix.html#snp-variant-calling" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The following packages are available as alternatives to Stacks to process RADseq or short-read data, identify variants and genotype samples, and produce data matrices for population genomic and phylogenomic analyses.</p>
<ul>
<li><p><a href="https://ipyrad.readthedocs.io/en/master/">iPyrad</a>, developed by <a href="https://academic.oup.com/bioinformatics/article/30/13/1844/2422183">Eaton <em>et al.</em> (2014)</a>, provides much of the same functionality as Stacks, but is proposed to be more interactive. A nice feature of iPyrad is the ability to branch and merge your workflow (much like you can branch and merge code on GitHub) so that you can try different assembly parameters or use different sample sets in reconstructing your loci and calling variants in parallel rather than repetitively starting the process anew. The source code, documentation and installation instructions can be found <a href="https://github.com/dereneaton/ipyrad?tab=readme-ov-file">here</a>.</p></li>
<li><p>AftrRAD: The developers of this tool for assembling and genotyping RADseq data, <a href="https://pubmed.ncbi.nlm.nih.gov/25641221/">Sovic <em>et al.</em> (2015)</a>, maintain that its advantage is that it is computationally more efficient than Stacks or iPyrad. As such, run times are not significantly impacted by the number of samples in the data set; this makes it a useful approach when one has a large sample set or access to limited computing resources (e.g., a lack of parallel processing).</p></li>
<li><p><a href="https://github.com/beiko-lab/RADProc">RADProc</a> was developed by <a href="https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.12954">Ravindran <em>et al.</em> (2018)</a> specifically for <em>de novo</em> locus and catalogue reconstruction, and genotyping. It uses graph-data structures, representing comparisons of sequence similarities, against which sweeps of different parameter sets can be evaluated very efficiently. Performance is reported to be similar to Stacks and iPyrad, but run-times are much quicker.</p></li>
<li><p><a href="https://github.com/evoeco/radtools">radtools: RAD-Seq_tools</a> is a handy tool for converting RADseq data into various formats.</p></li>
</ul>
</div>
<div id="sequence-alignment-tools" class="section level2 hasAnchor" number="13.2">
<h2><span class="header-section-number">D.2</span> Sequence alignment tools<a href="10-Appendix.html#sequence-alignment-tools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We did not work through sequence alignment in the main workbook, as Stacks produced already-aligned haplotypes from the genotyped SNP loci when preparing the data in the 'populations' model. Should we have needed to align partitions of sequence data from othologous gene regions, such as those mined from assembled genomes, low-coverage genome sequencing or Sanger-sequencing of amplicon sets, <a href="https://mafft.cbrc.jp/alignment/software/">MAFFT</a> is one of the most widely-used, powerful and efficient tools for multiple sequence alignment. It implements one of the most accurate iterative alignment algorithms available (L-INS-i), and incorporates a scoring system that tolerates large gaps, like one would find in some genomic regions (e.g. rRNA loop regions). The link above signposts to some useful papers around the methodology and performance of the approach.</p>
</div>
<div id="model-selection-tools" class="section level2 hasAnchor" number="13.3">
<h2><span class="header-section-number">D.3</span> Model selection tools<a href="10-Appendix.html#model-selection-tools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p><a href="https://github.com/ddarriba/jmodeltest2">jModelTest2</a>, developed by <a href="https://www.nature.com/articles/nmeth.2109">Darriba <em>et al.</em> (2012)</a> is the classic Java-based tool for selecting models of sequence/nucleotide evolution on single multiple sequence alignments. It implements five different methods for evaluating models: hierarchical and dynamic likelihood ratio tests, the AIC and BIC (as in Chapter 6), and a Decision Theory method. At present it cannot support a partitioned analysis of multiple alignments, so each alignment will need to be evaluated separately.</p></li>
<li><p><a href="https://github.com/BEAST2-Dev/bModelTest">bModeltest</a> is a plug-in for beast2, developed by <a href="https://bmcecolevol.biomedcentral.com/articles/10.1186/s12862-017-0890-6">Bouckaert &amp; Drummond (2017)</a>. The approach integrates model evaluation, parameter estimation and phylogenetic reconstruction in a Bayesian framework, such that the model does not need to be estimated using likelihood methods prior to tree construction, as is the usual approach.</p></li>
<li><p><a href="https://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a> was originally intended to identify natural partitions in a sequence alignment and estimate the best-fitting models for each of these. It can also be used to determine models for partitions separated by codon position, amino acid, etc. Extending the user-defined partition options, it is then possible to estimate different models and parameters for multiple sequence alignments of different gene regions that have been combined in a concatenated data set.</p></li>
</ul>
</div>
<div id="phylogeny-reconstruction-tools" class="section level2 hasAnchor" number="13.4">
<h2><span class="header-section-number">D.4</span> Phylogeny reconstruction tools<a href="10-Appendix.html#phylogeny-reconstruction-tools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://github.com/stamatak/standard-RAxML">RAxML</a> and its phylogenomic-NGS successor, <a href="https://github.com/amkozlov/raxml-ng">RAxML-ng</a>, were among the first - and are still among the most popular - tools for efficient maximum likelihood-based phylogeny reconstruction and support evaluation. The RAxML (Randomized Axelerated Maximum Likelihood) procedure, developed by <a href="https://academic.oup.com/bioinformatics/article/22/21/2688/251208">Stamakis (2006)</a>, uses fine-scale parallelisation, model approximations to replace specific previous models, and a very efficient tree-search algorithm to reconstruct and evaluate maximum likelihood trees. In improvements to the original software and through the development of RAxML-ng, more models and parameters of sequence and protein evolution can be implemented, and various methods are available to evaluate support - from traditional bootstrapping to a neat new approach which determines how many replicates are needed for support to stabilise. There are also new statistics available to evaluate the final trees, including measures of plausibility, Tree Certainty and Internode Certainty, and a feature to identify "rogue taxa". More information is available <a href="https://academic.oup.com/bioinformatics/article/30/9/1312/238053#">here</a> and <a href="https://academic.oup.com/bioinformatics/article/35/21/4453/5487384">here</a>.</p>
<p><a href="https://nbisweden.github.io/MrBayes/">MrBayes</a>, developed by <a href="https://academic.oup.com/sysbio/article/61/3/539/1674894">Ronquist <em>et al.</em> (2012)</a>, remains the "go to" package for Bayesian Inferences of phylogeny and for determining support through MCMC sampling of the posterior distribution.</p>
<p><a href="http://www.atgc-montpellier.fr/phylobayes/">PhyloBayes</a>, which was compiled for performing MCMC-based Bayesian Inferences of phylogeny on protein alignments, is finding increased use for phylogenomic data sets. The main difference to MrBayes is the use of a probabilistic CAT model that is suited to large multi-gene alignments.</p>
</div>
<div id="r-packages" class="section level2 hasAnchor" number="13.5">
<h2><span class="header-section-number">D.5</span> R-packages<a href="10-Appendix.html#r-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>There are a number of R-based packages for doing phylogenomic analyses. Among the most versatile are:</p>
<ul>
<li><p><a href="https://cran.r-project.org/web/packages/ape/index.html">ape (Analyses of Phylogenetics and Evolution)</a>: This package offers various means for constructing phylogenetic trees, evaluating evolutionary rates, and analysing ancestral characters and patterns of macro-evolution and diversification. There are also tools for manipulating trees and alignments, calculating genetic distances and translating data.</p></li>
<li><p><a href="https://cran.r-project.org/web/packages/phytools/index.html">phytools (Phylogenetic Tools for Comparative Biology and Other Things)</a> offers much the same functionality as 'ape'. Additionally, it offers more tools for comparative biology, such as trait mapping, ancestral state reconstruction, and presentation tools for projecting trees into phenotype space or on geography.</p></li>
</ul>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="09-Delimiting_species.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
