<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A Mamba installs | Phylogenomics</title>
  <meta name="description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="A Mamba installs | Phylogenomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Phylogenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Mamba installs | Phylogenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Phylogenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Gavin Gouws, Matthew Gemmell, Helen Hipperson, Ewan Harney, Kathryn Maher" />


<meta name="date" content="2024-01-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="09-Delimiting_species.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>Phylogenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-ddradseq-data-of-manta-and-devil-rays"><i class="fa fa-check"></i><b>2.1</b> Example data: ddRADseq data of manta and devil rays</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#manta--and-devil-ray-ddrad-data-set"><i class="fa fa-check"></i><b>4.2</b> Manta- and devil-ray ddRAD data set</a></li>
<li class="chapter" data-level="4.3" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality Control</a></li>
<li class="chapter" data-level="4.5" data-path="04-Quality_Control.html"><a href="04-Quality_Control.html#post-qc-check"><i class="fa fa-check"></i><b>4.5</b> Post QC Check</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html"><i class="fa fa-check"></i><b>5</b> Preparing data in Stacks</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#stacks-1"><i class="fa fa-check"></i><b>5.1</b> Stacks</a></li>
<li class="chapter" data-level="5.2" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#demultiplexing-data-with-process_radtags"><i class="fa fa-check"></i><b>5.2</b> Demultiplexing data with <code>process_radtags</code></a></li>
<li class="chapter" data-level="5.3" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#building-the-catalogue-loci-with-denovo_map.pl"><i class="fa fa-check"></i><b>5.3</b> Building the catalogue loci with <code>denovo_map.pl</code></a></li>
<li class="chapter" data-level="5.4" data-path="05_Preparing_data_with_Stacks.html"><a href="05_Preparing_data_with_Stacks.html#preparing-a-sequence-alignment-file-in-populations"><i class="fa fa-check"></i><b>5.4</b> Preparing a sequence alignment file in <code>populations</code></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html"><i class="fa fa-check"></i><b>6</b> Model selection, tree reconstruction and support evaluation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#estimating-a-best-fit-model-of-nucleotide-or-sequence-evolution"><i class="fa fa-check"></i><b>6.1</b> Estimating a best-fit model of nucleotide or sequence evolution</a></li>
<li class="chapter" data-level="6.2" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#constructing-a-maximum-likelihood-phylogenetic-tree"><i class="fa fa-check"></i><b>6.2</b> Constructing a maximum-likelihood phylogenetic tree</a></li>
<li class="chapter" data-level="6.3" data-path="06-Phylogeny.html"><a href="06-Phylogeny.html#calculating-nodal-support"><i class="fa fa-check"></i><b>6.3</b> Calculating nodal support</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Topology_testing.html"><a href="07-Topology_testing.html"><i class="fa fa-check"></i><b>7</b> Hypothesis and topology testing</a></li>
<li class="chapter" data-level="8" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html"><i class="fa fa-check"></i><b>8</b> Species delimitation with the Multispecies Coalescent</a>
<ul>
<li class="chapter" data-level="8.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#beast2"><i class="fa fa-check"></i><b>8.1</b> beast2</a></li>
<li class="chapter" data-level="8.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#gene-tree---species-tree-conflicts"><i class="fa fa-check"></i><b>8.2</b> Gene tree - species tree conflicts</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis"><i class="fa fa-check"></i><b>8.2.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="8.2.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#starting-the-analysis"><i class="fa fa-check"></i><b>8.2.2</b> Starting the analysis</a></li>
<li class="chapter" data-level="8.2.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#examining-the-results"><i class="fa fa-check"></i><b>8.2.3</b> Examining the results</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#bayes-factor-delimitation"><i class="fa fa-check"></i><b>8.3</b> Bayes Factor Delimitation</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#setting-up-the-analysis-1"><i class="fa fa-check"></i><b>8.3.1</b> Setting up the analysis</a></li>
<li class="chapter" data-level="8.3.2" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#running-the-analysis"><i class="fa fa-check"></i><b>8.3.2</b> Running the analysis</a></li>
<li class="chapter" data-level="8.3.3" data-path="09-Delimiting_species.html"><a href="09-Delimiting_species.html#selecting-the-taxonomic-hypothesis"><i class="fa fa-check"></i><b>8.3.3</b> Selecting the taxonomic hypothesis</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="11-Appendix.html"><a href="11-Appendix.html"><i class="fa fa-check"></i><b>A</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="A.1" data-path="11-Appendix.html"><a href="11-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>A.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="11-Appendix.html"><a href="11-Appendix.html#loops-and-wildcards"><i class="fa fa-check"></i><b>B</b> Loops and wildcards</a></li>
<li class="chapter" data-level="C" data-path="11-Appendix.html"><a href="11-Appendix.html#ipyrad"><i class="fa fa-check"></i><b>C</b> iPYRAD</a></li>
<li class="chapter" data-level="D" data-path="11-Appendix.html"><a href="11-Appendix.html#optimising-loci-reconstruction-in-denovo_map.pl-in-stacks"><i class="fa fa-check"></i><b>D</b> Optimising loci reconstruction in <code>denovo_map.pl</code> in Stacks</a></li>
<li class="chapter" data-level="E" data-path="11-Appendix.html"><a href="11-Appendix.html#jupyter_appendix"><i class="fa fa-check"></i><b>E</b> Jupyter-notebook</a></li>
<li class="chapter" data-level="F" data-path="11-Appendix.html"><a href="11-Appendix.html#loops"><i class="fa fa-check"></i><b>F</b> Using loops to process multiple samples</a>
<ul>
<li class="chapter" data-level="F.1" data-path="11-Appendix.html"><a href="11-Appendix.html#simple-loop-example"><i class="fa fa-check"></i><b>F.1</b> Simple loop example</a></li>
<li class="chapter" data-level="F.2" data-path="11-Appendix.html"><a href="11-Appendix.html#trimmomatic-loop-example"><i class="fa fa-check"></i><b>F.2</b> Trimmomatic loop example</a></li>
</ul></li>
<li class="chapter" data-level="G" data-path="11-Appendix.html"><a href="11-Appendix.html#an-overview-of-bayesian-statistics"><i class="fa fa-check"></i><b>G</b> An overview of Bayesian statistics</a></li>
<li class="chapter" data-level="H" data-path="11-Appendix.html"><a href="11-Appendix.html#additional-approaches-for-detecting-selection-and-for-gwas"><i class="fa fa-check"></i><b>H</b> Additional approaches for detecting selection and for GWAS</a>
<ul>
<li class="chapter" data-level="H.1" data-path="11-Appendix.html"><a href="11-Appendix.html#rehh"><i class="fa fa-check"></i><b>H.1</b> rehh</a></li>
<li class="chapter" data-level="H.2" data-path="11-Appendix.html"><a href="11-Appendix.html#gcta"><i class="fa fa-check"></i><b>H.2</b> GCTA</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Phylogenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mamba-installs" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">A</span> Mamba installs<a href="11-Appendix.html#mamba-installs" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/mamba_logo.png" width="20%" style="display: block; margin: auto;" /></p>
<div id="mamba_install" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">A.1</span> Mamba installation and environment<a href="11-Appendix.html#mamba_install" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Mamba is a reimplementation of conda. It is a great tool for installing bioinformatic packages, including R packages, and creating the environment in which to run them.</p>
<p>Mamba github: <a href="https://github.com/mamba-org/mamba" class="uri">https://github.com/mamba-org/mamba</a></p>
<p>Mamba installation: <a href="https://github.com/conda-forge/miniforge#mambaforge" class="uri">https://github.com/conda-forge/miniforge#mambaforge</a></p>
<p>Mamba guide: <a href="https://mamba.readthedocs.io/en/latest/user_guide/mamba.html" class="uri">https://mamba.readthedocs.io/en/latest/user_guide/mamba.html</a></p>
<p>To create the mamba environment <code>phylogenomics</code> run the below commands in your bash. You will need to have installed <code>mamba</code> first.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb57-1"><a href="11-Appendix.html#cb57-1" tabindex="-1"></a><span class="co">#phylogenomics</span></span>
<span id="cb57-2"><a href="11-Appendix.html#cb57-2" tabindex="-1"></a><span class="ex">mamba</span> create <span class="at">-n</span> phylogenomics</span>
<span id="cb57-3"><a href="11-Appendix.html#cb57-3" tabindex="-1"></a><span class="ex">mamba</span> activate phylogenomics</span>
<span id="cb57-4"><a href="11-Appendix.html#cb57-4" tabindex="-1"></a><span class="co">#Install the packages you need from from bioconda channel</span></span>
<span id="cb57-5"><a href="11-Appendix.html#cb57-5" tabindex="-1"></a><span class="co">#NGSadmix is part of angsd</span></span>
<span id="cb57-6"><a href="11-Appendix.html#cb57-6" tabindex="-1"></a><span class="ex">mamba</span> install <span class="at">-c</span> bioconda plink fastqc trimmomatic bwa samtools bcftools angsd r-popgenome gemma</span></code></pre></div>
</div>
</div>
<div id="loops-and-wildcards" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">B</span> Loops and wildcards<a href="11-Appendix.html#loops-and-wildcards" class="anchor-section" aria-label="Anchor link to header"></a></h1>
</div>
<div id="ipyrad" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">C</span> iPYRAD<a href="11-Appendix.html#ipyrad" class="anchor-section" aria-label="Anchor link to header"></a></h1>
</div>
<div id="optimising-loci-reconstruction-in-denovo_map.pl-in-stacks" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">D</span> Optimising loci reconstruction in <code>denovo_map.pl</code> in Stacks<a href="11-Appendix.html#optimising-loci-reconstruction-in-denovo_map.pl-in-stacks" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>As indicated in the main workbook, Stacks offers a <em>de novo</em> pipeline for assembling loci and calling variants (SNPS) in order to genotype individuals in the absence of a high-quality reference genome. There are, however, parameters that need to be optimised beforehand to increase accuracy in reconstruction and to maximimise the number of loci prior to running <code>denovo_map.pl</code> on the full set of study samples.</p>
<p>The procedure is explained in a book chapter by <a href="https://link.springer.com/protocol/10.1007/978-1-0716-2313-8_7#DOI">Rivera-Colon &amp; Catchen (2022)</a>. The chapter provides a handy introduction to designing a RADseq study, and using the various features, modules and pipelines of Stacks to analyse the resulting data. It also links through to some handy R-script for visualising and evaluating the data as it passes through Stacks.</p>
<p>The are two main parameters that need to be optimised. The first is <code>-M</code>, which is the number of mismatches that <code>ustacks</code> allows when combining stacks (matching sequence reads that represent an allele within an sample) into loci in that sample. If this is set too low, alleles won't be collapsed into loci, and these will be kept apart as separate homozygous loci. If set too high, alleles belonging to different loci will be combined as a single locus. Once loci are assembled in an individual, <code>cstacks</code> combines all the loci from those individuals into a catalogue of loci for the metapopulation. Here, <code>-n</code>, which is the maximum number of mismatches between individual loci and those in the catalogue, needs to be optimised. If <code>-n</code> is too low, loci in the catalogue will be "undermerged" (a locus that is homozygous for one allele will not be merged with a orthologous locus that is homozygous for an alternate allele, and these will be retained as two loci). If too high, different loci will be "overmerged" and considered the same in the catalogue.</p>
<p>Their procedure involves optimising these parameters so that the number of R80 loci (loci present in at least 80% of the metapopulation) is maximised. The <code>denovo_map.pl</code> is run repeatedly, with <code>-M</code> and <code>-n</code> set to 1 and increased in each subsequent iteration (until <code>-M</code> = <code>-n</code> = 10 or 12, depending on the data and biological context). The changes in the number of R80 loci are then evaluated across the runs. At the optimal value for these parameters, there will be no or negligible change in the number of loci. Above the optimum, the number of R80 loci should decrease again, as in the figure below.</p>
<p>Selecting the samples on which to optimise these parameters is an important consideration. You obviously don't want to optimise using all your data, unless you have limitless time. These samples you choose should be representative of your library or some biological reality (e.g., representatives of different populations for a population genomic study). You'd want samples that are fairly similar in terms of the amount of input data. Choosing samples with high coverage is sensible, as these will lead to the most accurate allele calling. You may be tempted to include samples with the highest numbers of reads (diresgarding coverage), but in this case you may be optimising for loci that are not going to be present in those samples with lower numbers of reads. There is no "golden rule@ and each data set will be different and subject to its own quirks. It is always advisable to do some exploration here and to remember that your results are always a hypothesis, based on the data and a number of assumptions that you make through the course of your analyses.</p>
</div>
<div id="jupyter_appendix" class="section level1 hasAnchor" number="13">
<h1><span class="header-section-number">E</span> Jupyter-notebook<a href="11-Appendix.html#jupyter_appendix" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/jupyter_logo.png" width="15%" style="display: block; margin: auto;" /></p>
<p>If you are running this on your own computer you can use <code>RStudio</code> for <code>R</code>.
However, you can also use <a href="https://jupyter.org/"><code>Jupyter-notebook</code></a> if you are using an HPC or prefer it.</p>
<p>If using bash you will need to create an environment with <code>Jupyter-notebook</code>.
Ensure you are in the <code>(base)</code> mamba environment.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb58-1"><a href="11-Appendix.html#cb58-1" tabindex="-1"></a><span class="ex">mamba</span> create <span class="at">-n</span> jupyter</span>
<span id="cb58-2"><a href="11-Appendix.html#cb58-2" tabindex="-1"></a><span class="ex">mamba</span> activate jupyter</span>
<span id="cb58-3"><a href="11-Appendix.html#cb58-3" tabindex="-1"></a><span class="ex">mamba</span> install <span class="at">-c</span> anaconda jupyter</span>
<span id="cb58-4"><a href="11-Appendix.html#cb58-4" tabindex="-1"></a><span class="ex">mamba</span> deactivate</span></code></pre></div>
<p>To run <code>Jupyter-notebook</code> with your <code>r_community</code> environment you can run the following.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb59-1"><a href="11-Appendix.html#cb59-1" tabindex="-1"></a><span class="co">#Activate you R_community env</span></span>
<span id="cb59-2"><a href="11-Appendix.html#cb59-2" tabindex="-1"></a><span class="ex">mamba</span> activate r_community</span>
<span id="cb59-3"><a href="11-Appendix.html#cb59-3" tabindex="-1"></a><span class="co">#Run jupyter-notebook (may be a slightly different path)</span></span>
<span id="cb59-4"><a href="11-Appendix.html#cb59-4" tabindex="-1"></a><span class="ex">~/mamba/envs/jupyter/bin/jupyter-notebook</span></span></code></pre></div>
</div>
<div id="loops" class="section level1 hasAnchor" number="14">
<h1><span class="header-section-number">F</span> Using loops to process multiple samples<a href="11-Appendix.html#loops" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/loop.png" width="25%" style="display: block; margin: auto;" /></p>
<p>In our main workflow we learnt how to run the QC and alignment steps using a single sample. You may be wondering how you would run these initial steps when you have many samples.</p>
<p>Ideally you don't want to have to run the same command multiple times - this is a recipe for typos and errors creeping into your code as well as being tedious and time consuming.</p>
<p>One approach to get around this is to run a loop which will cycle through each of your samples running the task you specify.</p>
<p>Put simply the syntax for a <code>for loop</code> is as follows:</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb60-1"><a href="11-Appendix.html#cb60-1" tabindex="-1"></a><span class="cf">for</span> <span class="op">&lt;</span>variable <span class="ex">name</span><span class="op">&gt;</span> in <span class="op">&lt;</span>a list of items<span class="op">&gt;</span><span class="kw">;</span></span>
<span id="cb60-2"><a href="11-Appendix.html#cb60-2" tabindex="-1"></a><span class="cf">do</span> <span class="op">&lt;</span>run <span class="ex">a</span> command<span class="op">&gt;</span> $<span class="op">&lt;</span>variable name<span class="op">&gt;</span><span class="kw">;</span></span>
<span id="cb60-3"><a href="11-Appendix.html#cb60-3" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>The <code>&lt;variable name&gt;</code> is the name of the variable you will use in the <code>do</code> part of your loop.
It contains the item in your list the loop is currently using at that point.
I.e. in the simple loop example (below) we call this variable <code>f</code> but you could use any name as long as you use it consistently in your <code>do</code> section.</p>
<p>The <code>&lt;a list of items&gt;</code> is anything that returns a list of items, e.g. a list of files in our case.</p>
<div id="simple-loop-example" class="section level2 hasAnchor" number="14.1">
<h2><span class="header-section-number">F.1</span> Simple loop example<a href="11-Appendix.html#simple-loop-example" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/data_files.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Below is an example of a loop, which can be adjusted for your samples and the software you are running.
This simplified loop will clearly demonstrate what the first part of the code is doing.</p>
<p>Make sure you are in the directory <code>~/popgenomics/data/fastq/</code> and in a terminal window (not in Jupyter).</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb61-1"><a href="11-Appendix.html#cb61-1" tabindex="-1"></a><span class="cf">for</span> f <span class="kw">in</span> <span class="pp">*</span>_1_small.fastq.gz<span class="kw">;</span> <span class="co">#the following will loop through all files ending _1_small.fastq.gz</span></span>
<span id="cb61-2"><a href="11-Appendix.html#cb61-2" tabindex="-1"></a><span class="cf">do</span> <span class="va">BASE</span><span class="op">=</span><span class="va">${f</span><span class="op">%</span>_1_small.fastq.gz<span class="va">}</span> <span class="co">#assign the file name &#39;$f&#39; to a variable called &#39;BASE&#39; but remove the extension &#39;_1_small.fastq.gz&#39;</span></span>
<span id="cb61-3"><a href="11-Appendix.html#cb61-3" tabindex="-1"></a>    <span class="bu">echo</span> <span class="va">$f</span> <span class="co">#print $f to screen</span></span>
<span id="cb61-4"><a href="11-Appendix.html#cb61-4" tabindex="-1"></a>    <span class="bu">echo</span> <span class="va">${BASE}</span><span class="kw">;</span> <span class="co">#print ${BASE} to screen</span></span>
<span id="cb61-5"><a href="11-Appendix.html#cb61-5" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>By using <code>echo</code> to print the following variables to the console we can see:</p>
<ul>
<li><code>echo $f</code>: prints the full filename.</li>
<li><code>echo ${BASE}</code>: prints just the start of the filename (minus the _1_small.fastq.gz extension as we specified in our code)</li>
</ul>
<p>The loop then repeats the commands on the next file in the list (in this case we only have one file) and will keep going until it has run through all files ending with <code>_1_small.fastq.gz</code> in the given directory.</p>
<p><code>done</code> specifies the end of the commands in the loop.</p>
<p>All commands between <code>do</code> and <code>done</code> in your loop will be executed. In this case we only have two files but this would work with any number of files. This is a useful bit of code as it means we can keep the filename but change the ending/extension of the file in our loops.</p>
</div>
<div id="trimmomatic-loop-example" class="section level2 hasAnchor" number="14.2">
<h2><span class="header-section-number">F.2</span> Trimmomatic loop example<a href="11-Appendix.html#trimmomatic-loop-example" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/quality_trimming_and_filtering.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Now we can write a loop to run trimmomatic starting with the code we used above as a base.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb62-1"><a href="11-Appendix.html#cb62-1" tabindex="-1"></a><span class="cf">for</span> f <span class="kw">in</span> <span class="pp">*</span>_1_small.fastq.gz<span class="kw">;</span> <span class="co">#the following will loop through all files ending _1_small.fastq.gz</span></span>
<span id="cb62-2"><a href="11-Appendix.html#cb62-2" tabindex="-1"></a><span class="cf">do</span> <span class="va">BASE</span><span class="op">=</span><span class="va">${f</span><span class="op">%</span>_1_small.fastq.gz<span class="va">}</span> <span class="co">#assign the file name &#39;$f&#39; to a variable called &#39;BASE&#39; but remove the extension &#39;_1_small.fastq.gz&#39;</span></span>
<span id="cb62-3"><a href="11-Appendix.html#cb62-3" tabindex="-1"></a>    <span class="ex">trimmomatic</span> PE <span class="at">-threads</span> 4 <span class="at">-phred33</span> <span class="dt">\ </span></span>
<span id="cb62-4"><a href="11-Appendix.html#cb62-4" tabindex="-1"></a>    <span class="co">#call trimmomatic and specify parameters</span></span>
<span id="cb62-5"><a href="11-Appendix.html#cb62-5" tabindex="-1"></a>    <span class="va">${BASE}</span><span class="ex">_1_small.fastq.gz</span> <span class="dt">\ </span></span>
<span id="cb62-6"><a href="11-Appendix.html#cb62-6" tabindex="-1"></a>    <span class="co">#specify input R1 extension</span></span>
<span id="cb62-7"><a href="11-Appendix.html#cb62-7" tabindex="-1"></a>    <span class="va">${BASE}</span><span class="ex">_2_small.fastq.gz</span> <span class="dt">\ </span></span>
<span id="cb62-8"><a href="11-Appendix.html#cb62-8" tabindex="-1"></a>    <span class="co">#specify input R2 extension</span></span>
<span id="cb62-9"><a href="11-Appendix.html#cb62-9" tabindex="-1"></a>    <span class="ex">trimmed/</span><span class="va">${BASE}</span><span class="ex">_1_out_paired.fastq</span> <span class="dt">\ </span></span>
<span id="cb62-10"><a href="11-Appendix.html#cb62-10" tabindex="-1"></a>    <span class="co">#specify directory and filename for output of paired R1 read</span></span>
<span id="cb62-11"><a href="11-Appendix.html#cb62-11" tabindex="-1"></a>    <span class="ex">trimmed/</span><span class="va">${BASE}</span><span class="ex">_1_out_unpaired.fastq</span> <span class="dt">\ </span></span>
<span id="cb62-12"><a href="11-Appendix.html#cb62-12" tabindex="-1"></a>    <span class="co">#specify directory and filename for output of unpaired R1 read</span></span>
<span id="cb62-13"><a href="11-Appendix.html#cb62-13" tabindex="-1"></a>    <span class="ex">trimmed/</span><span class="va">${BASE}</span><span class="ex">_2_out_paired.fastq</span> <span class="dt">\ </span></span>
<span id="cb62-14"><a href="11-Appendix.html#cb62-14" tabindex="-1"></a>    <span class="co">#specify directory and filename for output of paired R2 read</span></span>
<span id="cb62-15"><a href="11-Appendix.html#cb62-15" tabindex="-1"></a>    <span class="ex">trimmed/</span><span class="va">${BASE}</span><span class="ex">_2_out_unpaired.fastq</span> <span class="dt">\ </span></span>
<span id="cb62-16"><a href="11-Appendix.html#cb62-16" tabindex="-1"></a>    <span class="co">#specify directory and filename for output of unpaired R2 read</span></span>
<span id="cb62-17"><a href="11-Appendix.html#cb62-17" tabindex="-1"></a>    <span class="ex">ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:12</span> SLIDINGWINDOW:4:30 MINLEN:80<span class="kw">;</span>  <span class="co">#further paramenters for trimmomatic</span></span>
<span id="cb62-18"><a href="11-Appendix.html#cb62-18" tabindex="-1"></a><span class="cf">done</span> <span class="co">#end of commands</span></span></code></pre></div>
<p>You can then edit this code to run other software in a loop. For a nice basic intro to Linux loops please click <a href="https://www.redhat.com/sysadmin/bash-scripting-loops">here</a>.</p>
</div>
</div>
<div id="an-overview-of-bayesian-statistics" class="section level1 hasAnchor" number="15">
<h1><span class="header-section-number">G</span> An overview of Bayesian statistics<a href="11-Appendix.html#an-overview-of-bayesian-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>#<code>{r, fig.align = 'center',out.width= '20%', echo=FALSE } #knitr::include_graphics(path = "figures/Bayes.PNG", auto_pdf = TRUE) #</code>
As indicated in the presentations and used in some of the workshop analyses, many tools incorporate Bayesian approaches or statistics over other approaches. For example, population structure, admixture and ancestry can be examined using NGSadmix, which uses a maximum-likelihood approach, or STRUCTURE, which uses a Bayesian approach, and <code>GEMMA</code> for GWAS can implement a number of different models under a Bayesian framework. But, what does a Bayesian approach actually entail? <a href="https://statswithr.github.io/book/index.html">This resource</a> and <a href="https://vioshyvo.github.io/Bayesian_inference/index.html">this one</a> are useful introductions - they are easy to follow and not too intimidating. (The latter includes some R exercises to demonstrate some principles).</p>
<p>The core philosophy of Bayesian statistics is built around the concept of conditional probability, with analyses aiming to maximise this conditional probability or to minimise the uncertainty in parameter space around this. This conditional probability is the probability or likelihood of observing a particular outcome given (i.e. conditional on) the data, a model and some prior information (a prior). (Paradoxically, it is actually the probability of observing the data, given the model, the priors and the outcome). The ability to include prior information is key, and an advantage of the approach is that it evaluates potential outcomes and all the parameters that underlie these simultaneously. It also lends itself to being iterative; the priors and their influence on the conditional probabilities can be explored, refined or updated through repeated analyses or by incorporating additional data.
#<code>{r, fig.align = 'center',out.width= '40%', echo=FALSE } #knitr::include_graphics(path = "figures/MCMC.PNG", auto_pdf = TRUE) #</code>
In many cases, Bayesian approaches are favoured, not because of the ability to incorporate this prior, but because they are computationally efficient. If one can conceptualise a particular solution or result as lying at a given point in parameter space, a typical maximum likelihood approach may use an exhaustive or heuristic procedure to evaluate all points in that space, calculating a likelihood or probability given the model and data, and eventually compare these. In most cases, this is computationally restrictive and/or time-consuming. Bayesian approaches employ much more efficient sampling algorithms, such as MCMC <a href="https://en.wikipedia.org/wiki/Markov_chain_Monte_Carlo">(Markov chain Monte Carlo)</a>, as conducted when implementing the BSLMM model in <code>GEMMA</code>, to sample posterior probability distributions. Multiple chains will move around and selectively sample various points in parameter space, comparing the conditional probabilities among the chains themselves and to the prior sampled states, and retain those points (and the parameters underlying them) from the posterior probability distribution if there was an improvement in conditional probability. In doing so, eventually the chains and sampling will converge around the “true result”, as in the figure above, taken from <a href="http://faculty.washington.edu/wjs18/MCMC_Bayesian/Lewis_200TREE.pdf">this paper</a>. While the chains might not necessarily hit on the exact result, there will be sufficient sampling around the “true result” that one can synthesise the information from multiple sampled “generations”, and obtain average values around the optimum. As such, it can provide useful information around uncertainty in our data, models and priors, allowing us to evaluate confidence intervals easily, and potentially change parameters, priors or add data. An unfortunate drawback of this approach is the possibility that the sampling gets trapped in local optima in parameter space (the points in the bottom right corner in the figure) rather than sampling near the “true result”. Thus, it is often advised to run multiple iterations of Bayesian analyses, modifying the starting parameters of the search, where the applications allow this.</p>
<p>There are additional aspects to be aware of. There is a longstanding debate as to whether Bayesian statistics have frequentist properties (and may be interpreted as one would interpret more traditional frequentist statistics, especially when evaluating significance) or whether they are entirely probabilistic. Often this conflict results from subtle differences in philosophical perspective; see <a href="https://bids.berkeley.edu/news/bayesians-v-frequentists-tale-old-time">this example</a>. Additionally, when designing an experiment or choosing an analytical approach, Bayesian and frequentist approaches can yield different outcomes, with frequentist approaches often being more sensitive than Bayesian counterparts to how the hypotheses to be tested are framed or approached.</p>
</div>
<div id="additional-approaches-for-detecting-selection-and-for-gwas" class="section level1 hasAnchor" number="16">
<h1><span class="header-section-number">H</span> Additional approaches for detecting selection and for GWAS<a href="11-Appendix.html#additional-approaches-for-detecting-selection-and-for-gwas" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="rehh" class="section level2 hasAnchor" number="16.1">
<h2><span class="header-section-number">H.1</span> rehh<a href="11-Appendix.html#rehh" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>#<code>{r, fig.align = 'center',out.width= '20%', echo=FALSE } #knitr::include_graphics(path = "figures/rehh.PNG", auto_pdf = TRUE) #</code>
<a href="https://cran.r-project.org/package=rehh"><code>rehh</code></a> is a R package which is used to explore SNP data and identify genomic regions that have been under recent positive selection (where advantageous variants are retained and increase in prevalence) in a single population or species, or to examine differential selection between two populations. A handy vignette (from which the above graph was taken) is available <a href="https://cran.r-project.org/web/packages/rehh/vignettes/rehh.html">here</a>. The approach is based on the concept of Extended Haplotype Homozygosity (EHH) (or a long-range haplotype test) proposed by Sabeti et al. <a href="https://www.nature.com/articles/nature01140">2002</a>. Fundamentally, it considers the relationship between allele frequencies and linkage disequilibrium (LD). Under neutral evolution, mutations require time to establish in a population, by which time LD would have broken down around these mutations through recombination. Under recent positive selection, mutations increase in frequency rapidly, before LD has a chance to break down, and these extended haplotypes are then transmitted without recombination.</p>
<p>EHH is the probability that a pair of randomly-chosen chromosomes are identical by descent and homozygous for those SNPs which we have genotyped and mapped at a given distance from a core region of interest (presented as haplotypes of increasing length) on that chromosome. Values range from 0, where all these extended haplotypes are different, to 1, where all are homozygous. EHH is then considered relatively, where the EHH of the tested core haplotype is compared with the EHHs of other core haplotypes in the same region (but excluding other core haplotypes). Through this, the approach detects the breakdown of linkage disequilibrium as one moves further away from the core region. Core haplotypes in regions that have high EHH and are at high frequency in the population indicate mutations that have likely been established through non-neutral evolution.</p>
<p>The approach appears to be more powerful than many traditional metrics, but it can be complex to implement. To reconstruct the extended haplotypes, genotype data are phased, where each of the alleles of a given SNP genotype is assigned to a hypothetical haplotype. This is obviously straightforward when many loci are homozygous, but as one moves away from the SNP of interest (central in the core haplotype) and encounters heterozygous SNP genotypes the number of hypothetical haplotypes increases exponentially. There are a number of tools and approaches for phasing haplotypes, and a useful (albeit dated) review was published by Browning &amp; Browning <a href="http://www.biostat.ulg.ac.be/pages/BIM/Theory/browning_browning_nature_review_genetics2011.pdf">2011</a>. As input, <code>rehh</code> can accept haplotype data in various formats, including vcf-formatted data, and those produced by phasing programmes like <a href="https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html">SHAPEIT2</a> and <a href="http://scheet.org/software.html">fastPHASE</a>.</p>
</div>
<div id="gcta" class="section level2 hasAnchor" number="16.2">
<h2><span class="header-section-number">H.2</span> GCTA<a href="11-Appendix.html#gcta" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>#<code>{r, fig.align = 'center',out.width= '20%', echo=FALSE } #knitr::include_graphics(path = "figures/GCTA.PNG", auto_pdf = TRUE, ) #</code>
GCTA <a href="https://yanglab.westlake.edu.cn/software/gcta/#Overview">Genome-wide Complex Trait Analysis</a> is a package that runs within a Linux/Unix environment, although there are installs and executables for Mac and Windows too. Yang et al. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3014363/">2011</a> originally developed it to determine the proportions of phenotypic variance of a complex trait that can be explained by genome-wide SNP variation. It was since expanded to and focused on many different and powerful GWAS applications, and it now supports a wide range of analyses, including relatedness and inbreeding, heritability and LD estimation, as well as more traditional population genetic statistics.</p>
<p>The program takes binary PED files, generated by PLINK, as input files. As with many applications for GWAS, only biallelic SNPs are supported at the moment, and multiallelic SNPs are automatically ignored.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="09-Delimiting_species.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
