<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Mapping sequences to the reference genome using BWA | Population Genomics</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Mapping sequences to the reference genome using BWA | Population Genomics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Mapping sequences to the reference genome using BWA | Population Genomics" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Helen Hipperson, Katy Maher, Victor Soria-Carrasco, Graeme Fox" />


<meta name="date" content="2023-01-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="04-QC.html"/>
<link rel="next" href="06-SNPs.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Population Genomics</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#example-data-whole-genome-resequencing-of-killer-whales"><i class="fa fa-check"></i><b>2.1</b> Example data: Whole-genome resequencing of killer whales</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03-Cluster_Introduction.html"><a href="03-Cluster_Introduction.html#accessing-the-example-data"><i class="fa fa-check"></i><b>3.3</b> Accessing the example data</a></li>
</ul></li>
<li class="part"><span><b>Generating SNP datasets</b></span></li>
<li class="chapter" data-level="4" data-path="04-QC.html"><a href="04-QC.html"><i class="fa fa-check"></i><b>4</b> QC Tutorials</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QC.html"><a href="04-QC.html#workshop-data"><i class="fa fa-check"></i><b>4.1</b> Workshop data</a></li>
<li class="chapter" data-level="4.2" data-path="04-QC.html"><a href="04-QC.html#killer-whale-genomic-dataset"><i class="fa fa-check"></i><b>4.2</b> Killer whale genomic dataset</a></li>
<li class="chapter" data-level="4.3" data-path="04-QC.html"><a href="04-QC.html#quality-assessment"><i class="fa fa-check"></i><b>4.3</b> Quality assessment</a></li>
<li class="chapter" data-level="4.4" data-path="04-QC.html"><a href="04-QC.html#quality-control"><i class="fa fa-check"></i><b>4.4</b> Quality control</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="04-QC.html"><a href="04-QC.html#post-quality-control-check"><i class="fa fa-check"></i><b>4.4.1</b> Post Quality control check</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Mapping.html"><a href="05-Mapping.html"><i class="fa fa-check"></i><b>5</b> Mapping sequences to the reference genome using BWA</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Mapping.html"><a href="05-Mapping.html#reference-sequence-preparation"><i class="fa fa-check"></i><b>5.1</b> Reference sequence preparation</a></li>
<li class="chapter" data-level="5.2" data-path="05-Mapping.html"><a href="05-Mapping.html#read-mapping"><i class="fa fa-check"></i><b>5.2</b> Read mapping</a></li>
<li class="chapter" data-level="5.3" data-path="05-Mapping.html"><a href="05-Mapping.html#assessing-mapping-results"><i class="fa fa-check"></i><b>5.3</b> Assessing mapping results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-SNPs.html"><a href="06-SNPs.html"><i class="fa fa-check"></i><b>6</b> SNP and genotype calling using SAMtools and BCFtools</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-SNPs.html"><a href="06-SNPs.html#file-preparation"><i class="fa fa-check"></i><b>6.1</b> File preparation</a></li>
<li class="chapter" data-level="6.2" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-calling"><i class="fa fa-check"></i><b>6.2</b> Variant calling</a></li>
<li class="chapter" data-level="6.3" data-path="06-SNPs.html"><a href="06-SNPs.html#variant-filtering"><i class="fa fa-check"></i><b>6.3</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-SNPs.html"><a href="06-SNPs.html#extracting-information"><i class="fa fa-check"></i><b>6.3.1</b> Extracting information</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-SNPs.html"><a href="06-SNPs.html#subsetting-files"><i class="fa fa-check"></i><b>6.3.2</b> Subsetting files</a></li>
<li class="chapter" data-level="6.3.3" data-path="06-SNPs.html"><a href="06-SNPs.html#filtering"><i class="fa fa-check"></i><b>6.3.3</b> Filtering</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="7" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html"><i class="fa fa-check"></i><b>7</b> Population structure with NGSadmix and PCA</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#ngsadmix"><i class="fa fa-check"></i><b>7.1</b> NGSadmix</a></li>
<li class="chapter" data-level="7.2" data-path="07-NGSAdmix.html"><a href="07-NGSAdmix.html#pca"><i class="fa fa-check"></i><b>7.2</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html"><i class="fa fa-check"></i><b>8</b> Calculating population summary statistics using PopGenome</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Summary_stats.html"><a href="08-Summary_stats.html#getting-started"><i class="fa fa-check"></i><b>8.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html"><i class="fa fa-check"></i><b>9</b> Delimitation of contiguous regions of differentiation using Hidden Markov Models (HMM)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#initial-set-up"><i class="fa fa-check"></i><b>9.1</b> Initial set up</a></li>
<li class="chapter" data-level="9.2" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#data-formatting"><i class="fa fa-check"></i><b>9.2</b> Data formatting</a></li>
<li class="chapter" data-level="9.3" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#allele-frequency-estimation"><i class="fa fa-check"></i><b>9.3</b> Allele frequency estimation</a></li>
<li class="chapter" data-level="9.4" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#fst-estimation"><i class="fa fa-check"></i><b>9.4</b> F<sub>ST</sub> estimation</a></li>
<li class="chapter" data-level="9.5" data-path="09-Detecting_selection.html"><a href="09-Detecting_selection.html#delimitation-of-contiguous-regions-of-differentiation-using-a-hmm-model"><i class="fa fa-check"></i><b>9.5</b> Delimitation of contiguous regions of differentiation using a HMM model</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-GWAS.html"><a href="10-GWAS.html"><i class="fa fa-check"></i><b>10</b> Genetic architecture of traits using multi-locus Genome Wide Association (GWA) mapping with GEMMA</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-GWAS.html"><a href="10-GWAS.html#initial-set-up-1"><i class="fa fa-check"></i><b>10.1</b> Initial set up</a></li>
<li class="chapter" data-level="10.2" data-path="10-GWAS.html"><a href="10-GWAS.html#data-formatting-1"><i class="fa fa-check"></i><b>10.2</b> Data formatting</a></li>
<li class="chapter" data-level="10.3" data-path="10-GWAS.html"><a href="10-GWAS.html#running-gemma-to-fit-a-bayesian-sparse-linear-mixed-model-bslmm"><i class="fa fa-check"></i><b>10.3</b> Running GEMMA to fit a Bayesian sparse linear mixed model (BSLMM)</a></li>
<li class="chapter" data-level="10.4" data-path="10-GWAS.html"><a href="10-GWAS.html#analysing-gemma-bslmm-output"><i class="fa fa-check"></i><b>10.4</b> Analysing GEMMA BSLMM output</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Population Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="alignment" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Mapping sequences to the reference genome using BWA<a href="05-Mapping.html#alignment" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>BWA is a program for mapping sequences to a reference genome. Here we will use BWA to map our paired, trimmed sequences to the killer whale reference genome (GCF_000331955.2_Oorc_1.1_genomic.fna), and output a BAM (Binary sequence Alignment/Map format) file of the results.</p>
<p>Further information on the reference genome, such as the sex of the sequenced individual, the technology and assembly programs used, the number of scaffolds/contigs and the sequence coverage can be found on the <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000331955.2">NCBI website page for this assembly</a>.</p>
<div id="reference-sequence-preparation" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Reference sequence preparation<a href="05-Mapping.html#reference-sequence-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/index.png" width="30%" style="display: block; margin: auto;" /></p>
<p>BWA uses indexing of the reference genome to speed up the mapping. The index needs to be prepared before we map our reads. As this takes some time this has been run for you, but for information the command used to do this was:</p>
<blockquote>
<p>bwa index GCF_000331955.2_Oorc_1.1_genomic.fna</p>
</blockquote>
<p>This produces index files with the extensions ‘.sa’, ‘.pac’, ‘.ann’, ‘.amb’ and ‘.bwt’ that will be automatically detected and used in the mapping step below.</p>
</div>
<div id="read-mapping" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Read mapping<a href="05-Mapping.html#read-mapping" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/map%20-%20Copy.png" width="20%" style="display: block; margin: auto;" /></p>
<p>To map our paired-end, trimmed reads type the following:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="05-Mapping.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem <span class="dt">\</span></span>
<span id="cb16-2"><a href="05-Mapping.html#cb16-2" aria-hidden="true" tabindex="-1"></a>/pub39/tea/nsc006/NEOF/popgenomics/genome/GCF_000331955.2_Oorc_1.1_genomic.fna <span class="dt">\</span></span>
<span id="cb16-3"><a href="05-Mapping.html#cb16-3" aria-hidden="true" tabindex="-1"></a>SRR8861574_1_out_paired.fastq SRR8861574_2_out_paired.fastq <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb16-4"><a href="05-Mapping.html#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> sort <span class="at">-o</span> SRR8861574_paired.bam</span></code></pre></div>
<p><code>bwa mem</code> is an alignment algorithm well suited to Illumina-length sequences. The default output is a SAM <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">(Sequence Alignment Map format)</a>. However, here we pipe the output to <a href="http://www.htslib.org/"><code>samtools</code></a>, a program for writing, viewing and manipulating alignment files, to sort and generate a BAM format, a binary, compressed version of SAM format.</p>
<p>The mapping will take a few minutes to run, during which some output messages will be printed to the screen. When it has finished you will see the command prompt <code>$</code> and flashing cursor again.</p>
<p>We can view this file using <code>samtools view</code>, and <code>head</code> to display only the first few lines:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="05-Mapping.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-h</span> SRR8861574_paired.bam <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<p>The header section consists of lines starting with ‘@’. In this case we have a header line for each of the scaffolds in the killer whale reference genome with their name (SN) and length in bp (LN) given.</p>
<p>To look at the information in the alignment section we can leave out the <code>-h</code> option:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="05-Mapping.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view SRR8861574_paired.bam <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<p>Here is an example of one line, representing information from one sequence read.</p>
<p><img src="figures/sam1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>Each column contains information on where and how well the read aligns to the reference. After column 11 there can be several optional tags, often specific to the aligner used and can be useful for downstream analysis.</p>
<p><img src="figures/sam2.png" width="60%" style="display: block; margin: auto auto auto 0;" /></p>
<p>The image above is from a different alignment file to the one you have generated here. On the first line of your alignment file what is the query sequence name? What is the reference contig name?</p>
<p>In order to process the BAM files for further analysis it is usually required that they are indexed. This can also be done with <code>samtools</code>:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="05-Mapping.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> index SRR8861574_paired.bam</span></code></pre></div>
<p>Use <code>ls</code> to view the contents of your directory - you should now have an index file with the extensions <code>bam.bai</code>.</p>
</div>
<div id="assessing-mapping-results" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Assessing mapping results<a href="05-Mapping.html#assessing-mapping-results" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/flag.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Lastly, we can use the Samtools command <code>flagstat</code> to find information on how the reads mapped:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="05-Mapping.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> flagstat SRR8861574_paired.bam</span></code></pre></div>
<p>How many reads have mapped?</p>
<p>This information is a summary of the ‘FLAG’ values - the second field in the sam/bam file. These values can be used if you want to extract or exclude a specific set of reads for downstream analysis. For example, to exclude unmapped reads from our alignment file we can run:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="05-Mapping.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-F</span> 4 <span class="at">-b</span> SRR8861574_paired.bam <span class="op">&gt;</span> SRR8861574_paired_mapped.bam</span></code></pre></div>
<p>The options used are:
-F: only include reads without this flag</p>
<p>-b: output in bam format</p>
<p>Conversely, you can use <code>-f</code> to include the reads with the specified flag.</p>
<p>The Broad Institute has a useful <a href="https://broadinstitute.github.io/picard/explain-flags.html">website to interpret the FLAG values</a>, likewise they are <a href="http://www.htslib.org/doc/samtools-flags.html">listed in the samtools documentation</a>.</p>
<p>Run flagstat on the bam file from which we have excluded unmapped reads.</p>
<p>How many reads are in the file in total, compared to the unfiltered file?</p>
<p>What proportion of the reads in this file are mapped, compared to the unfiltered file?</p>
<p>In the next step we will use alignment files from several samples to detect genetic variation.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="04-QC.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="06-SNPs.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
