# Reading sequence data

<center>![](images/DNA_icon.png){height="300px"}![](images/biopython_logo.png){height="300px"}</center>

Biopython is a suite of libraries and tools designed for analysis of biological datasets in Python, largely centered around the processing of DNA sequence data.

You can find much more information about Biopython on the [website](https://biopython.org/).

During this section on Biopython we are going to be using the SeqIO module. The knowledge and approaches from the previous chapters
will now be used to process some genomic DNA sequence data.

## SeqIO

<center>![](images/DNA_computer.png){width=20%}</center>

The SeqIO tools are the main Sequence Input/Output interface for Biopython. It provides a relatively simple, uniform interface for working with various types of DNA sequence files. The module reads raw sequence files (such as .FASTQ files you will likely receive from your sequencing facility) and converts them to a Python readable format; a [SeqRecord](https://biopython.org/wiki/SeqRecord) class.


## Reading a FASTQ DNA data file into Python
One of the most common types of DNA sequence data files you will be working with is the [FASTQ](https://en.wikipedia.org/wiki/FASTQ_format) file. 
The FASTQ file contains multiple DNA sequences, each one might be a single short read from an Illumina sequencer and the file can contain thousands or millions of sequences depending on which sequencer was used to produce it.

The FASTQ has a very specific format and holds three important pieces of information.

## The FASTQ format

<center>![](images/fastq_icon.png){width=20%}</center>

The file has a repetitive structure of four distinct components:

* A unique sequence identifier (ID)
* The DNA nucleotide sequence
* A small spacer
* The quality score associated with each nucleotide

The ID line always starts with the @ symbol and this is followed by a string of numbers and letters giving details such as the sequencer that produced the data, the run number, and an index adapter used on the sample.

The DNA nucleotide sequence is shown as a string of ATGC characters. 

There is a spacer consisting of the "+" character which separates the sequence from the quality scores.

The quality score associated with each nucleotide above is ASCII encoded with each symbol representing a quality score from 0 to 40 (higher being better). This line is not meant to be easily human-readable.

An example of a single unit of this four line repeat can be seen below:
```
@MISEQ:42:000000000-A8DD3:1:1101:15046:1468
TTCACATCCTGACCATTCAGTTGAGCAAAATAGTTCTTCAGT
+
CCFFFFFGHHHHJIJJJJIJJJIIJJJJIIIJJGFIIIJEDD
```
## Read a FASTQ file into Python

__Scenario:__ You have just received some FASTQ formatted data back from your sequencing facility. The data was generated using a shotgun, whole-genome sequencing approach to sequence the genome of the undulate ray (*Raja undulata*).

<center>![](images/undulate_ray.jpeg){width=20%}</center>

We are going to read the data into Python, ready to do some data processing and analyses.

## Setting up a script to use BioPython

Open a new file in Gedit by clicking the "Create a new document" button (![](images/gedit_create_new_doc.png?display=inline-block){width=4%}) at the top-left.

We need to load a couple of Python libraries: 

`Biopython` (a suite of bioinformatics focused Python tools)

...and...

`re` (allows us to perform searches using wildcards and special characters). 

Add the following to the top of the script.

```{python, eval=FALSE}
from Bio import SeqIO
import re
```

You should hopefully still have the current working directory on your clipboard.
Set a variable called 'wd', (working directory) to the filepath on your clipboard by adding a version of the following:

```{python, eval=FALSE}
wd = "/pub14/tea/nsc201"
```

## Read a fie of sequence data

<center>![](images/bioinformatics_icon2.png){width=20%}</center>

We are going to use `SeqIO` (which stands for sequence in/out) and the `parse` __function__ which splits the FASTQ file into its separate components.

We store the information on which file to parse in the variable called `R_undulata_R1_data`.
`SeqIO.parse` requires two arguments: 

- The filepath to the file which is to be read.
- The format of that file.

```{python, eval=FALSE}
R_undulata_data = SeqIO.parse(wd + "/Python_course_data/R_undulata_WGS.fastq", "fastq")
```

We can then look through the file using a <b>for</b> loop. Each of the four-unit repeats of the FASTQ file is read as a separate 'record' and we can access them in the script. 

To access the ID line of each record, we use '.id' and to access the sequence we use '.seq'. 
Each of these elements can be printed in the terminal using an appropriate print statement.

```{python, eval=FALSE}
for record in R_undulata_data:
  print(record.id)
  print(record.seq)
```

Save the new script as __'script02.py'__ and run it from the terminal window to print the ID and sequences.

## Modifying reads

We can perform operations on each of these elements. For example, Biopython has a built in `translate()` __method__ to convert nucleotide sequence into protein sequence (although bear in mind that we are not considering reading frames and we likely have lots of non-coding data).

Modify the __for__ __loop__ above to the following:

```{python, eval=FALSE}
for record in R_undulata_data:
  print(record.id)
  #print(record.seq)
  print(record.seq.translate())
```

Save the script and run it on the terminal.

## Count reads
<center>![](./images/count.png){width="15%"}</center>

We can implement a counter to tally the total number of reads in the file. Modify the __for__ __loop__ above to the following (you do not need to add the comments however):

```{python, eval=FALSE}
## start a counter to track how many reads
counter = 0

for record in R_undulata_data:
  #print(record.id) 
  #print(record.seq)
  #print(record.seq.translate())
  # increase the counter by 1 each time it sees a new read
  counter = counter + 1

# save the number of raw sequences for use later
number_raw_reads = counter

print("Raw sequence file has " + str(number_raw_reads) + \
" sequence records.")
```

Note that we need to convert the `number_raw_reads` to a __string__ type for the print statement to work using the `str()` __function__. Python will not print variables of different types in the same statement.