<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Normalising your data | Metabarcoding for diet analysis and environmental DNA - supplemental material</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Normalising your data | Metabarcoding for diet analysis and environmental DNA - supplemental material" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Normalising your data | Metabarcoding for diet analysis and environmental DNA - supplemental material" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher, Helen Hipperson and Ewan Harney" />


<meta name="date" content="2025-08-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="01-Metabarcoding_supplemental.html"/>
<link rel="next" href="03-Exporting_data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank">
<img src="figures/neof_rounded_corners_300ppi.png" width=250px></a></li>
<li><a>Environmental metabarcoding supplemental</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Metabarcoding_supplemental.html"><a href="01-Metabarcoding_supplemental.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Metabarcoding_supplemental.html"><a href="01-Metabarcoding_supplemental.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Normalisation.html"><a href="02-Normalisation.html"><i class="fa fa-check"></i><b>2</b> Normalising your data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Normalisation.html"><a href="02-Normalisation.html#relative-abundance"><i class="fa fa-check"></i><b>2.1</b> Relative abundance</a></li>
<li class="chapter" data-level="2.2" data-path="02-Normalisation.html"><a href="02-Normalisation.html#rarefaction"><i class="fa fa-check"></i><b>2.2</b> Rarefaction</a></li>
<li class="chapter" data-level="2.3" data-path="02-Normalisation.html"><a href="02-Normalisation.html#variance-stabilising-transformation-using-deseq2"><i class="fa fa-check"></i><b>2.3</b> Variance stabilising transformation using DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Exporting_data.html"><a href="03-Exporting_data.html"><i class="fa fa-check"></i><b>3</b> Exporting data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Exporting_data.html"><a href="03-Exporting_data.html#fasta-file"><i class="fa fa-check"></i><b>3.1</b> Fasta file</a></li>
<li class="chapter" data-level="3.2" data-path="03-Exporting_data.html"><a href="03-Exporting_data.html#sequence-count-matrix"><i class="fa fa-check"></i><b>3.2</b> Sequence count matrix</a></li>
<li class="chapter" data-level="3.3" data-path="03-Exporting_data.html"><a href="03-Exporting_data.html#table-of-taxon-names"><i class="fa fa-check"></i><b>3.3</b> Table of taxon names</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-BLAST.html"><a href="04-BLAST.html"><i class="fa fa-check"></i><b>4</b> GenBank and BLAST</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-BLAST.html"><a href="04-BLAST.html#blast-againt-the-nt-database"><i class="fa fa-check"></i><b>4.1</b> BLAST againt the nt database</a></li>
<li class="chapter" data-level="4.2" data-path="04-BLAST.html"><a href="04-BLAST.html#blast-against-a-custom-database"><i class="fa fa-check"></i><b>4.2</b> BLAST against a custom database</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-MEGAN.html"><a href="05-MEGAN.html"><i class="fa fa-check"></i><b>5</b> Taxonomizr</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-MEGAN.html"><a href="05-MEGAN.html#filtering-blast-results"><i class="fa fa-check"></i><b>5.1</b> Filtering BLAST results</a></li>
<li class="chapter" data-level="5.2" data-path="05-MEGAN.html"><a href="05-MEGAN.html#importing-blast-results-into-r-and-taxonomizr"><i class="fa fa-check"></i><b>5.2</b> Importing BLAST results into R and taxonomizr</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-blast2megan.html"><a href="06-blast2megan.html"><i class="fa fa-check"></i><b>6</b> More BLAST and taxonomizr</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-blast2megan.html"><a href="06-blast2megan.html#introduction-to-blast2taxonomizr"><i class="fa fa-check"></i><b>6.1</b> Introduction to blast2taxonomizr</a></li>
<li class="chapter" data-level="6.2" data-path="06-blast2megan.html"><a href="06-blast2megan.html#running-blast-in-array-mode"><i class="fa fa-check"></i><b>6.2</b> Running blast in array mode</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="06-blast2megan.html"><a href="06-blast2megan.html#split-the-input-file"><i class="fa fa-check"></i><b>6.2.1</b> Split the input file</a></li>
<li class="chapter" data-level="6.2.2" data-path="06-blast2megan.html"><a href="06-blast2megan.html#make-a-list-of-the-subfiles"><i class="fa fa-check"></i><b>6.2.2</b> Make a list of the subfiles</a></li>
<li class="chapter" data-level="6.2.3" data-path="06-blast2megan.html"><a href="06-blast2megan.html#making-the-array-script"><i class="fa fa-check"></i><b>6.2.3</b> Making the array script</a></li>
<li class="chapter" data-level="6.2.4" data-path="06-blast2megan.html"><a href="06-blast2megan.html#running-the-array-script"><i class="fa fa-check"></i><b>6.2.4</b> Running the array script</a></li>
<li class="chapter" data-level="6.2.5" data-path="06-blast2megan.html"><a href="06-blast2megan.html#arrays-in-blast2taxonomizr"><i class="fa fa-check"></i><b>6.2.5</b> Arrays in blast2taxonomizr</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="06-blast2megan.html"><a href="06-blast2megan.html#taxonomizr"><i class="fa fa-check"></i><b>6.3</b> Taxonomizr</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-blast2megan.html"><a href="06-blast2megan.html#preparing-blast-for-taxonomizr"><i class="fa fa-check"></i><b>6.3.1</b> Preparing Blast for taxonomizr</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-blast2megan.html"><a href="06-blast2megan.html#running-taxonomizr"><i class="fa fa-check"></i><b>6.3.2</b> Running taxonomizr</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="06-blast2megan.html"><a href="06-blast2megan.html#combining-output-with-dada2-output"><i class="fa fa-check"></i><b>6.4</b> Combining output with DADA2 output</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Summary.html"><a href="07-Summary.html"><i class="fa fa-check"></i><b>7</b> Summary</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Metabarcoding for diet analysis and environmental DNA - supplemental material</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="normalising" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Normalising your data<a href="02-Normalisation.html#normalising" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/phyloseq.png" style="width:200px" />
</center>
<p>As mentioned in the main tutorial document there are several different methods for normalising your data for downstream analysis. Here we will look at three different methods of normalising your data for downstream analysis.</p>
<p>To summarise so you have several methods in one place, we will also include the normalisation method that we used in the main tutorial.</p>
<p>Remember these are just a few examples of how to normalise your data and we recommend you read the recent literature and think about the properties of your own dataset when deciding the best way to proceed.</p>
<div id="relative-abundance" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Relative abundance<a href="02-Normalisation.html#relative-abundance" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>transform_sample_counts</code> function transforms the sample counts from a taxa abundance matrix using a user-provided function. The counts of each sample will be transformed individually.</p>
<p>In this case we will transform the raw samples into relative abundances by dividing individual ASV counts by the total ASV counts in a sample.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb2-1"><a href="02-Normalisation.html#cb2-1" tabindex="-1"></a>phylo.prop <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(phylo, <span class="cf">function</span>(otu) otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span></code></pre></div>
</div>
<div id="rarefaction" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Rarefaction<a href="02-Normalisation.html#rarefaction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Rarefaction involves randomly removing reads until you reach a number often equal to or less than the number of reads in the smallest sample.</p>
<p>Here we rarefy to an even depth using the minimum number of reads found in a sample as specified by the <code>sample.size=min(sample_sums(phylo))</code> argument.</p>
<ul>
<li><p>The argument <code>rngseed = 1</code> sets a random number seed used for random number generation, which is then used when subsampling a random number of reads from each sample. By setting the random seed you can make this process reproducible.</p></li>
<li><p>We have set the argument <code>replace</code> as <code>replace = F</code>. From the <code>phylo.rarefied</code> R help page "Two implications to consider are that (1) sampling with replacement is faster and more memory efficient as currently implemented; and (2), sampling with replacement means that there is a chance that the number of reads for a given OTU in a given sample could be larger than the original count value, as opposed to sampling without replacement where the original count value is the maximum possible."</p></li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb3-1"><a href="02-Normalisation.html#cb3-1" tabindex="-1"></a>phylo.rarefied <span class="ot">&lt;-</span> <span class="fu">rarefy_even_depth</span>(phylo, <span class="at">rngseed =</span> <span class="dv">1</span>, </span>
<span id="cb3-2"><a href="02-Normalisation.html#cb3-2" tabindex="-1"></a>                                    <span class="at">sample.size=</span><span class="fu">min</span>(<span class="fu">sample_sums</span>(phylo)), </span>
<span id="cb3-3"><a href="02-Normalisation.html#cb3-3" tabindex="-1"></a>                                    <span class="at">replace =</span> F)</span></code></pre></div>
</div>
<div id="variance-stabilising-transformation-using-deseq2" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Variance stabilising transformation using DESeq2<a href="02-Normalisation.html#variance-stabilising-transformation-using-deseq2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Variance stabilising transformation borrows methods from transcriptomic research using the DESeq2 package.</p>
<p>The workflow below is based on the workflow for variance stabilised data suggested <a href="https://github.com/joey711/phyloseq/issues/283">here</a>.</p>
<p>First we load the DESeq2 library.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb4-1"><a href="02-Normalisation.html#cb4-1" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code></pre></div>
<p>Now transform the phyloseq object into a format appropriate for DESeq2.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb5-1"><a href="02-Normalisation.html#cb5-1" tabindex="-1"></a>ds <span class="ot">&lt;-</span> <span class="fu">phyloseq_to_deseq2</span>(phylo, <span class="sc">~</span>SITE) </span></code></pre></div>
<p>The function <code>estimateSizeFactors</code> is used to estimate the size factors for a DESeqDataSet - for more information on the normalisation methods used by DESeq2 please read the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">DESeq2 paper</a> and <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">manual</a>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb6-1"><a href="02-Normalisation.html#cb6-1" tabindex="-1"></a>ds<span class="ot">&lt;-</span><span class="fu">estimateSizeFactors</span>(ds)</span></code></pre></div>
<p>For our dataset we get the error message "Error in estimateSizeFactorsForMatrix(counts(object), locfunc = locfunc, : every gene contains at least one zero, cannot compute log geometric means"</p>
<p>This indicates we have many zeros in our count data (which is common in amplicon studies).</p>
<p>There are several ways to deal with this problem. One way is to calculate size factors separately using a zero-tolerant variant of geometric mean. For more details of this method see <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html">here</a>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb7-1"><a href="02-Normalisation.html#cb7-1" tabindex="-1"></a><span class="co"># calculate geometric means prior to estimation of size factors</span></span>
<span id="cb7-2"><a href="02-Normalisation.html#cb7-2" tabindex="-1"></a>gm_mean <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">na.rm=</span><span class="cn">TRUE</span>){</span>
<span id="cb7-3"><a href="02-Normalisation.html#cb7-3" tabindex="-1"></a>  <span class="fu">exp</span>(<span class="fu">sum</span>(<span class="fu">log</span>(x[x <span class="sc">&gt;</span> <span class="dv">0</span>]), <span class="at">na.rm=</span>na.rm) <span class="sc">/</span> <span class="fu">length</span>(x))</span>
<span id="cb7-4"><a href="02-Normalisation.html#cb7-4" tabindex="-1"></a>}</span>
<span id="cb7-5"><a href="02-Normalisation.html#cb7-5" tabindex="-1"></a>geoMeans <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">counts</span>(ds), <span class="dv">1</span>, gm_mean)</span></code></pre></div>
<p>We can now proceed and calculate size factors and dispersions of our dataset, before calculating the variance stabilizing transformation and transforming the count data with <code>getVarianceStabilizedData</code>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb8-1"><a href="02-Normalisation.html#cb8-1" tabindex="-1"></a>ds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(ds, <span class="at">geoMeans =</span> geoMeans)</span>
<span id="cb8-2"><a href="02-Normalisation.html#cb8-2" tabindex="-1"></a>ds <span class="ot">&lt;-</span> <span class="fu">estimateDispersions</span>(ds)</span>
<span id="cb8-3"><a href="02-Normalisation.html#cb8-3" tabindex="-1"></a>ds_vst<span class="ot">&lt;-</span><span class="fu">getVarianceStabilizedData</span>(ds)</span></code></pre></div>
<p>We then convert the transformed dataset back into a phyloseq object for further analysis.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb9-1"><a href="02-Normalisation.html#cb9-1" tabindex="-1"></a>ds_count_phylo <span class="ot">&lt;-</span> <span class="fu">otu_table</span>(ds_vst, <span class="at">taxa_are_rows=</span>T)</span>
<span id="cb9-2"><a href="02-Normalisation.html#cb9-2" tabindex="-1"></a>sample_info_tab_phy <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(meta.rmlow)</span>
<span id="cb9-3"><a href="02-Normalisation.html#cb9-3" tabindex="-1"></a>ds_phylo <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(ds_count_phylo, sample_info_tab_phy)</span></code></pre></div>
<p>This is just one way to deal with the problem of zero counts in our dataset. For more discussion on using DESeq2 for metabarcoding data and dealing with zero count data please check out the following links:</p>
<ul>
<li><a href="https://bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-FAQ.html">phyloseq FAQ</a></li>
<li><a href="https://joey711.github.io/phyloseq-extensions/DESeq2.html">DESeq2 with phyloseq</a></li>
</ul>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="01-Metabarcoding_supplemental.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="03-Exporting_data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

</body>

</html>
