<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 More BLAST and MEGAN | Metabarcoding for diet analysis and environmental DNA - supplemental material</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 More BLAST and MEGAN | Metabarcoding for diet analysis and environmental DNA - supplemental material" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 More BLAST and MEGAN | Metabarcoding for diet analysis and environmental DNA - supplemental material" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop - supplemental material" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher and Helen Hipperson" />


<meta name="date" content="2024-07-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="05-MEGAN.html"/>
<link rel="next" href="07-Summary.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>Environmental metabarcoding supplemental</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Metabarcoding_supplemental.html"><a href="01-Metabarcoding_supplemental.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Metabarcoding_supplemental.html"><a href="01-Metabarcoding_supplemental.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Normalisation.html"><a href="02-Normalisation.html"><i class="fa fa-check"></i><b>2</b> Normalising your data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Normalisation.html"><a href="02-Normalisation.html#relative-abundance"><i class="fa fa-check"></i><b>2.1</b> Relative abundance</a></li>
<li class="chapter" data-level="2.2" data-path="02-Normalisation.html"><a href="02-Normalisation.html#rarefaction"><i class="fa fa-check"></i><b>2.2</b> Rarefaction</a></li>
<li class="chapter" data-level="2.3" data-path="02-Normalisation.html"><a href="02-Normalisation.html#variance-stabilising-transformation-using-deseq2"><i class="fa fa-check"></i><b>2.3</b> Variance stabilising transformation using DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Exporting_data.html"><a href="03-Exporting_data.html"><i class="fa fa-check"></i><b>3</b> Exporting data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Exporting_data.html"><a href="03-Exporting_data.html#fasta-file"><i class="fa fa-check"></i><b>3.1</b> Fasta file</a></li>
<li class="chapter" data-level="3.2" data-path="03-Exporting_data.html"><a href="03-Exporting_data.html#sequence-count-matrix"><i class="fa fa-check"></i><b>3.2</b> Sequence count matrix</a></li>
<li class="chapter" data-level="3.3" data-path="03-Exporting_data.html"><a href="03-Exporting_data.html#table-of-taxon-names"><i class="fa fa-check"></i><b>3.3</b> Table of taxon names</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-BLAST.html"><a href="04-BLAST.html"><i class="fa fa-check"></i><b>4</b> GenBank and BLAST</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-BLAST.html"><a href="04-BLAST.html#blast-againt-the-nt-database"><i class="fa fa-check"></i><b>4.1</b> BLAST againt the nt database</a></li>
<li class="chapter" data-level="4.2" data-path="04-BLAST.html"><a href="04-BLAST.html#blast-against-a-custom-database"><i class="fa fa-check"></i><b>4.2</b> BLAST against a custom database</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-MEGAN.html"><a href="05-MEGAN.html"><i class="fa fa-check"></i><b>5</b> MEGAN</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-MEGAN.html"><a href="05-MEGAN.html#filtering-blast-results"><i class="fa fa-check"></i><b>5.1</b> Filtering BLAST results</a></li>
<li class="chapter" data-level="5.2" data-path="05-MEGAN.html"><a href="05-MEGAN.html#importing-blast-results-into-megan"><i class="fa fa-check"></i><b>5.2</b> Importing BLAST results into MEGAN</a></li>
<li class="chapter" data-level="5.3" data-path="05-MEGAN.html"><a href="05-MEGAN.html#exporting-taxonomic-classifications-from-megan"><i class="fa fa-check"></i><b>5.3</b> Exporting taxonomic classifications from MEGAN</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-blast2megan.html"><a href="06-blast2megan.html"><i class="fa fa-check"></i><b>6</b> More BLAST and MEGAN</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-blast2megan.html"><a href="06-blast2megan.html#introduction-to-blast2megan"><i class="fa fa-check"></i><b>6.1</b> Introduction to blast2megan</a></li>
<li class="chapter" data-level="6.2" data-path="06-blast2megan.html"><a href="06-blast2megan.html#running-blast-in-array-mode"><i class="fa fa-check"></i><b>6.2</b> Running blast in array mode</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="06-blast2megan.html"><a href="06-blast2megan.html#split-the-input-file"><i class="fa fa-check"></i><b>6.2.1</b> Split the input file</a></li>
<li class="chapter" data-level="6.2.2" data-path="06-blast2megan.html"><a href="06-blast2megan.html#make-a-list-of-the-subfiles"><i class="fa fa-check"></i><b>6.2.2</b> Make a list of the subfiles</a></li>
<li class="chapter" data-level="6.2.3" data-path="06-blast2megan.html"><a href="06-blast2megan.html#making-the-array-script"><i class="fa fa-check"></i><b>6.2.3</b> Making the array script</a></li>
<li class="chapter" data-level="6.2.4" data-path="06-blast2megan.html"><a href="06-blast2megan.html#running-the-array-script"><i class="fa fa-check"></i><b>6.2.4</b> Running the array script</a></li>
<li class="chapter" data-level="6.2.5" data-path="06-blast2megan.html"><a href="06-blast2megan.html#arrays-in-blast2megan"><i class="fa fa-check"></i><b>6.2.5</b> Arrays in blast2megan</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="06-blast2megan.html"><a href="06-blast2megan.html#megan-for-command-line"><i class="fa fa-check"></i><b>6.3</b> MEGAN for command line</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-blast2megan.html"><a href="06-blast2megan.html#preparing-blast-for-megan"><i class="fa fa-check"></i><b>6.3.1</b> Preparing Blast for MEGAN</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-blast2megan.html"><a href="06-blast2megan.html#running-megan-on-the-command-line"><i class="fa fa-check"></i><b>6.3.2</b> Running MEGAN on the command line</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="06-blast2megan.html"><a href="06-blast2megan.html#combining-output-with-dada2-output"><i class="fa fa-check"></i><b>6.4</b> Combining output with DADA2 output</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Summary.html"><a href="07-Summary.html"><i class="fa fa-check"></i><b>7</b> Summary</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Metabarcoding for diet analysis and environmental DNA - supplemental material</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="blast2megan" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> More BLAST and MEGAN<a href="06-blast2megan.html#blast2megan" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/taxonomy.png" width="200px" style="display: block; margin: auto;" /></p>
<div id="introduction-to-blast2megan" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Introduction to blast2megan<a href="06-blast2megan.html#introduction-to-blast2megan" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We may want to scale up the approaches described in chapters 04 and 05 to allow us to process larger data sets more rapidly. In this chapter we explain two ways you could do this:</p>
<ul>
<li>Using Slurm job arrays to simultaneous blast a split fasta file</li>
<li>Running MEGAN from the command line and formatting the output to provide LCA information</li>
</ul>
<p>We run through some hypothetical examples and draw your attention to some of the key bits of code. This workflow is based on an automated command line workflow called <a href="https://github.com/ewan-harney/hpc_blast2megan/blob/main/README.md">blast2megan</a> which is designed for use on a specific HPC (the University of Sheffield's <a href="https://docs.hpc.shef.ac.uk/en/latest/bessemer/index.html#gsc.tab=0">Bessemer</a> cluster), but the scripts associated with blast2megan can be easily modified for use on your own cluster.</p>
<p><strong>These steps are not intended to be run on the web VNC, but the workflow is provided in case it is useful for your own data.</strong></p>
</div>
<div id="running-blast-in-array-mode" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Running blast in array mode<a href="06-blast2megan.html#running-blast-in-array-mode" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>NCBI nt is an enormous database containing tens of millions of sequences. If you have a long list of ASVs, 'blasting' hundreds or thousands of sequences against nt can take many hours or even days. To speed things up we can use the <a href="https://slurm.schedmd.com/documentation.html">Slurm</a> workload manager (used by many HPCs) to split blast into an array job and run several blast searches simultaneously. Many HPCs provide their own information about how to use slurm effectively (see for example the University of Sheffield's HPC documentation on <a href="https://docs.hpc.shef.ac.uk/en/latest/hpc/scheduler/advanced_job_submission_and_control.html#gsc.tab=0">advanced job submission with slurm</a>).</p>
<p>Below we outline the main steps required to use a slurm job array to process a large number of sequences:</p>
<div id="split-the-input-file" class="section level3 hasAnchor" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Split the input file<a href="06-blast2megan.html#split-the-input-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Say we have an fasta file 'asv.fasta' with 1000 sequences, and we want to split it into 10 smaller files (with 100 sequences each) that can be run simultaneously. We can use <code>awk</code> to split the file into chunks like so:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb18-1"><a href="06-blast2megan.html#cb18-1" tabindex="-1"></a><span class="do">### use awk to split a fasta file</span></span>
<span id="cb18-2"><a href="06-blast2megan.html#cb18-2" tabindex="-1"></a>awk <span class="st">&#39;BEGIN {n=0;} /^&gt;/ {if(n%100==0){file=sprintf(&quot;chunk%d.fa&quot;,n);} print &gt; file; n++; next;} { print &gt;&gt; file; }&#39;</span> <span class="sc">&lt;</span> asv.fasta</span></code></pre></div>
<p>You don't need to fully understand how this is splitting the file, but it's worth noting a couple of things:</p>
<ul>
<li><code>/^&gt;/</code> : the characters between two forward slashes are what we are tell awk to look for (what is known as a <a href="https://en.wikipedia.org/wiki/Regular_expression">regular expression</a>) - in this case a '&gt;' at the start of a line (used by fasta files to denote the sequence name and description),</li>
<li><code>if(n%100==0)</code> : the '100' is how many sequences will be included in each chunk (you can set it to any number),</li>
<li><code>file=sprintf("chunk%d.fa",n)</code> : this is how we define how the output files will be called. in the example it is set to 'chunk%d', where '%d' is the line number of the original file (starting at 0). So our chunk files will be called 'chunk0.fa', 'chunk100.fa' all the way up to 'chunk900.fa',</li>
</ul>
</div>
<div id="make-a-list-of-the-subfiles" class="section level3 hasAnchor" number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> Make a list of the subfiles<a href="06-blast2megan.html#make-a-list-of-the-subfiles" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now we will make a text file containing a list of all the chunks we want to run on the job array. This is easily achieved with the <code>ls</code> command. It's also worth double checking how many chunk files there are in the list, as we will need that information in the following steps:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb19-1"><a href="06-blast2megan.html#cb19-1" tabindex="-1"></a><span class="do">### use ls to make a list of subfiles</span></span>
<span id="cb19-2"><a href="06-blast2megan.html#cb19-2" tabindex="-1"></a>ls chunk<span class="sc">*</span>fa <span class="sc">&gt;</span> list_of_chunks.txt</span>
<span id="cb19-3"><a href="06-blast2megan.html#cb19-3" tabindex="-1"></a><span class="do">### count the number of lines in the file with wc -l (word count - line)</span></span>
<span id="cb19-4"><a href="06-blast2megan.html#cb19-4" tabindex="-1"></a>wc <span class="sc">-</span>l list_of_chunks.txt</span></code></pre></div>
</div>
<div id="making-the-array-script" class="section level3 hasAnchor" number="6.2.3">
<h3><span class="header-section-number">6.2.3</span> Making the array script<a href="06-blast2megan.html#making-the-array-script" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now we have split our fasta file up, we can move on to the array. First, we will have to make a simple bash script that reads in the information from the 'list_of_chunks.txt' file, and uses that as an input for blast.</p>
<p>Below are the main commands you would need to include:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb20-1"><a href="06-blast2megan.html#cb20-1" tabindex="-1"></a><span class="co"># create a variable FASTA corresponding to the information on a numbered line of list_of_chunks.txt</span></span>
<span id="cb20-2"><a href="06-blast2megan.html#cb20-2" tabindex="-1"></a>FASTA<span class="ot">=</span><span class="er">$</span>(sed <span class="st">&quot;${SLURM_ARRAY_TASK_ID}q;d&quot;</span> <span class="sc">&lt;</span> (cat list_of_chunks.txt))</span>
<span id="cb20-3"><a href="06-blast2megan.html#cb20-3" tabindex="-1"></a></span>
<span id="cb20-4"><a href="06-blast2megan.html#cb20-4" tabindex="-1"></a><span class="co"># Run blastn using the newly defined FASTA variable. Make sure the path to the nt database is correct!</span></span>
<span id="cb20-5"><a href="06-blast2megan.html#cb20-5" tabindex="-1"></a>blastn <span class="sc">-</span>query <span class="sc">$</span>{FASTA} \</span>
<span id="cb20-6"><a href="06-blast2megan.html#cb20-6" tabindex="-1"></a>    <span class="sc">-</span>task blastn \</span>
<span id="cb20-7"><a href="06-blast2megan.html#cb20-7" tabindex="-1"></a>    <span class="sc">-</span>db <span class="sc">/</span>path<span class="sc">/</span>to<span class="sc">/</span>ncbi<span class="sc">/</span>nt \</span>
<span id="cb20-8"><a href="06-blast2megan.html#cb20-8" tabindex="-1"></a>    <span class="sc">-</span>out <span class="sc">$</span>{FASTA}_blast.out.tab \</span>
<span id="cb20-9"><a href="06-blast2megan.html#cb20-9" tabindex="-1"></a>    <span class="sc">-</span>outfmt <span class="dv">6</span></span></code></pre></div>
<p>When the job array is submitted this script will be run as many times as specified (it will be 10 times in our example). For each numbered array job (1 to 10):</p>
<ul>
<li><code>FASTA=$(...)</code>: a new variable called 'FASTA' is created based on some information we will provide</li>
<li><code>sed "${SLURM_ARRAY_TASK_ID}q;d"</code>: it is equal to the information on a single line (whose number is determine by which array job is running)</li>
<li><code>&lt; (cat list_of_chunks.txt)</code>: this file is the source of the information</li>
</ul>
<p>Thus for array job 1 <code>FASTA</code> takes the value of line 1 in 'list_of_chunks.txt' (line 1 reads 'chunk0.fa'), while in array job 2 <code>FASTA</code> takes the value of line 2 in 'list_of_chunks.txt' (chunk100.fa). For each job in the array, the unique value of variable <code>${FASTA}</code> is then used when calling blastn.</p>
<p><strong>If you plan to use the command line version of MEGAN, we recommend altering the <code>-outfmt</code> argument slightly (see section 6.3.1 Preparing Blast for MEGAN for more details).</strong></p>
<p>When making a script remember that you will have to include the <a href="https://en.wikipedia.org/wiki/Shebang_(Unix)">shebang</a> and set various other <a href="https://slurm.schedmd.com/pdfs/summary.pdf">sbatch options</a>. If blast is available through a conda environment you will also have to load this as well. Once this is done, you can save the script. We could call it something like 'blast_array.sh'. Depending on how permissions are applied in your HPC you may have to make the script executable using the <code>chmod</code> command (see this <a href="https://cets.seas.upenn.edu/answers/chmod.html">explainer</a> on chmod). A script file can be made executable be running:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb21-1"><a href="06-blast2megan.html#cb21-1" tabindex="-1"></a><span class="co"># make a file executable</span></span>
<span id="cb21-2"><a href="06-blast2megan.html#cb21-2" tabindex="-1"></a>chmod u<span class="sc">+</span>x blast_array.sh</span></code></pre></div>
</div>
<div id="running-the-array-script" class="section level3 hasAnchor" number="6.2.4">
<h3><span class="header-section-number">6.2.4</span> Running the array script<a href="06-blast2megan.html#running-the-array-script" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Finally we can submit the job as an array to slurm. Notice that we have to specify how many jobs our array is composed of. This must be equal to the number of lines in 'list_of_chunks.txt'.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb22-1"><a href="06-blast2megan.html#cb22-1" tabindex="-1"></a><span class="co"># submit script as an array job to slurm</span></span>
<span id="cb22-2"><a href="06-blast2megan.html#cb22-2" tabindex="-1"></a>sbatch <span class="sc">--</span>array<span class="ot">=</span><span class="dv">1-10</span> blast_array.sh</span></code></pre></div>
<p>Once the job has finished, you should have 10 'chunk%d.fa_blast.out.tab' files containing the blast results. The final step is to concatenate these files together with the <code>cat</code> command:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb23-1"><a href="06-blast2megan.html#cb23-1" tabindex="-1"></a><span class="co"># concatenate file together</span></span>
<span id="cb23-2"><a href="06-blast2megan.html#cb23-2" tabindex="-1"></a>cat chunk<span class="sc">*</span>.fa_blast.out.tab <span class="sc">&gt;</span> all_blast.out.tab</span></code></pre></div>
</div>
<div id="arrays-in-blast2megan" class="section level3 hasAnchor" number="6.2.5">
<h3><span class="header-section-number">6.2.5</span> Arrays in blast2megan<a href="06-blast2megan.html#arrays-in-blast2megan" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The blast array in <a href="https://github.com/ewan-harney/hpc_blast2megan">blast2megan</a> makes use of two scripts, a preparatory script that splits the input and makes a list of files (01A_run_prep_for_blast.sh), and then the array itself (01B_run_blastn_array.sh). These scripts do a few additional things compared with the steps listed above, such as making sub-directories to store the 'chunk%d.fa' input files and 'chunk*.fa_blast.out.tab' output files. Please take a look at the <a href="https://github.com/ewan-harney/hpc_blast2megan/blob/main/README.md">readme</a> and have a look inside the scripts themselves for more information.</p>
</div>
</div>
<div id="megan-for-command-line" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> MEGAN for command line<a href="06-blast2megan.html#megan-for-command-line" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In Chapter 5 we provided information about the MEGAN application (GUI), but it is also possible to install a command line version of MEGAN on your cluster and automate the procedure of finding the Lowest Common Ancestor (LCA). Please speak to your HPC administrator for advice on installing new software.</p>
<div id="preparing-blast-for-megan" class="section level3 hasAnchor" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Preparing Blast for MEGAN<a href="06-blast2megan.html#preparing-blast-for-megan" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The command line version of MEGAN requires a blast tab delimited file as input (-outfmt 6). However, it also expects the ncbi subject sequence id (column 2 in the blast tab file) to be a simple accession number (e.g. 'MT539258.1'), yet by default, the output will contain additional gene identifier information (e.g. 'gi|1846592020|gb|MT539258.1|'). Although this information can be filtered out afterwards (e.g. using <code>awk</code>), you can also amend the <code>-outfmt</code> argument so that only the accession is included:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb24-1"><a href="06-blast2megan.html#cb24-1" tabindex="-1"></a><span class="co"># Run blastn with a custom list of column headers</span></span>
<span id="cb24-2"><a href="06-blast2megan.html#cb24-2" tabindex="-1"></a>blastn <span class="sc">-</span>query <span class="sc">$</span>{FASTA} \</span>
<span id="cb24-3"><a href="06-blast2megan.html#cb24-3" tabindex="-1"></a>    <span class="sc">-</span>task blastn \</span>
<span id="cb24-4"><a href="06-blast2megan.html#cb24-4" tabindex="-1"></a>    <span class="sc">-</span>db <span class="sc">/</span>path<span class="sc">/</span>to<span class="sc">/</span>ncbi<span class="sc">/</span>nt \</span>
<span id="cb24-5"><a href="06-blast2megan.html#cb24-5" tabindex="-1"></a>    <span class="sc">-</span>out <span class="sc">$</span>{FASTA}_blast.out.tab \</span>
<span id="cb24-6"><a href="06-blast2megan.html#cb24-6" tabindex="-1"></a>    <span class="sc">-</span>outfmt <span class="st">&quot;6 qseqid saccver pident length mismatch gapopen qstart qend sstart send evalue bitscore&quot;</span></span></code></pre></div>
<p>In this example of blast we specify exactly which column headers to include. Columns 1 and 3-12 are the default values, but we have used <code>saccver</code> (subject accession version) rather than <code>sseqid</code> (subject sequence ID) in column 2 to be compatible with command line MEGAN. This handy <a href="https://www.metagenomics.wiki/tools/blast/blastn-output-format-6">Guide to blastn output format 6</a> has more information about the default and optional columns in blast output format 6. It is possible to specify additional columns such as <code>sscinames</code> and <code>sskingdoms</code> which will provide the scientific names and the kingdom of the ncbi subject sequence (as long as the ncbi taxa database files 'taxdb.btd' and 'taxdb.bti' are present in the current working directory). This taxonomic information can provide a useful sanity check when looking at blast results, but must be removed before using MEGAN (this is the case for both GUI and command line), as the software will only process the blast tab file if it has the 12 columns matching those specified in the code above.</p>
<p>As explained in Chapter 5.1, it is also useful to filter blast results (using <code>awk</code>) for a minimum percentage identity and a minimum length before running MEGAN.</p>
</div>
<div id="running-megan-on-the-command-line" class="section level3 hasAnchor" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Running MEGAN on the command line<a href="06-blast2megan.html#running-megan-on-the-command-line" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Once we have a filtered blast output file prepared, we can create a simple bash script for running the MEGAN blast2lca command. This is the critical piece of code that that must go in the script:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb25-1"><a href="06-blast2megan.html#cb25-1" tabindex="-1"></a><span class="co"># MEGAN blast2lca command</span></span>
<span id="cb25-2"><a href="06-blast2megan.html#cb25-2" tabindex="-1"></a>path<span class="sc">/</span>to<span class="sc">/</span>megan<span class="sc">/</span>tools<span class="sc">/</span>blast2lca <span class="sc">-</span>i filtered_blast.out.tab \</span>
<span id="cb25-3"><a href="06-blast2megan.html#cb25-3" tabindex="-1"></a>    <span class="sc">-</span>m BlastN \</span>
<span id="cb25-4"><a href="06-blast2megan.html#cb25-4" tabindex="-1"></a>    <span class="sc">-</span>o megan_full_out.tsv \</span>
<span id="cb25-5"><a href="06-blast2megan.html#cb25-5" tabindex="-1"></a>    <span class="sc">-</span>mdb megan<span class="sc">-</span>nucl<span class="sc">-</span>Feb2022.db \</span>
<span id="cb25-6"><a href="06-blast2megan.html#cb25-6" tabindex="-1"></a>    <span class="sc">-</span>top <span class="dv">2</span></span></code></pre></div>
<p>Notice that certain arguments have been provided:</p>
<ul>
<li><code>-i</code> the input file (blast tab delimited file)<br />
</li>
<li><code>-m</code> the type of blast that was run (BlastN in our case)<br />
</li>
<li><code>-o</code> what to call the output<br />
</li>
<li><code>-mdb</code> the location of the MEGAN nucleotide database<br />
</li>
<li><code>-top</code> the top percent</li>
</ul>
<p>Other arguments can be set, and depending on your needs may be useful. Running <code>path/to/megan/tools/blast2lca --help</code> will provide more information about the arguments and options that can be set.</p>
<p>Blast2lca only retains blast hits that are less than the 'Top Percent' value away from the highest scoring hit (based on blast bit score). Top Percent has a default of 10, but setting it to a lower value (e.g. 1.5 or 2) may improve taxonomic assignment. For more information about Top Percent and the blast2lca algorithm please see the <a href="https://software-ab.cs.uni-tuebingen.de/download/megan6/manual.pdf">MEGAN manual</a>.</p>
<p>As is the case with the MEGAN GUI, you will need a copy of the file 'megan-nucl-Feb2022.db' in order map the NCBI accession numbers to taxonomic classes. This file is available to download from the <a href="https://uni-tuebingen.de/fakultaeten/mathematisch-naturwissenschaftliche-fakultaet/fachbereiche/informatik/lehrstuehle/algorithms-in-bioinformatics/software/megan6/">MEGAN</a> website.</p>
<p>As explained in Section 6.2.3 Making the array script, you must include certain information in your bash script like the the shebang, and it is useful to provide other sbatch options. You may have to make the script executable with <code>chmod</code>. Once your script is saved and executable, run it like so:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r Rchunk"><code class="sourceCode r"><span id="cb26-1"><a href="06-blast2megan.html#cb26-1" tabindex="-1"></a><span class="co"># Run a script</span></span>
<span id="cb26-2"><a href="06-blast2megan.html#cb26-2" tabindex="-1"></a>sbatch my_script.sh</span></code></pre></div>
The output of blast2lca includes information about the most probable taxa down to species level. This information is known as a taxon path. The default output looks something like this:<br />
<br>
<center>
<img src="figures/megan_full_example.png" style="width:1000px" />
</center>
<p><br>
<br>
At first glance this looks rather a mess. There are many missing values denoted by 'unknown'. This is because of the way blast2lca handles differences between prokaryotes and eukaryotes. Because these domains have a different number of taxonomic ranks (7 in prokaryotes, 8 in eukaryotes), blast2lca first checks the taxon path for prokaryotes (e.g. ASV_1032 in the example above) and writes 'unknown' if the domain is not prokaryotic. Thus for every eukaryotic sequence 6 values of 'unknown' are written between the domain and the kingdom.</p>
<p>Notice also that every taxonomic rank has a trailling number associated with it: this is the percentage of blast hits that match that taxon. If this number is less than 100 then there is some ambiguity about the identity of the taxon. It is recommended to only use taxonomic assignments that have 100% support. The lowest taxonomic level rank with 100% support is the lowest common ancestor.</p>
<p>Occasionally taxon paths will be missing information for intermediate taxonomic ranks. This could be for genuine taxonomic reasons (in the example above ASV_1089 is associated with the <a href="https://en.wikipedia.org/wiki/Lymnaeidae">Lymnaeidae</a> family of snails which belong to a superfamily, a superorder, and an infraclass, but not a regular order), or it may reflect missing data in the databases.</p>
<p>In the blast2megan pipeline, the script '02_run_blast2lca.sh' not only implements the blast2lca command but uses a series of piped <code>awk</code> and <code>sed</code> commands to reformat the output such that:</p>
<ul>
<li>a standardised taxon path is provided for all taxa</li>
<li>only unambiguous taxa with 100% support are retained</li>
<li>missing taxonomic levels are assigned a useful name</li>
</ul>
<p>Two files are output by this script:</p>
<ul>
<li>megan_summary_out.tsv outputs just the LCA rank (s for species, g for genus etc) and the LCA. The output looks like this:<br />
<br>
<center>
<img src="figures/megan_summary_example.png" style="width:250px" />
</center>
<br>
<br></li>
<li>megan_taxonpath_out.tsv contains all the taxonomic information down to the LCA, with NAs for ambiguous levels. The output looks like this:<br />
<br>
<center>
<img src="figures/megan_taxonpath_example.png" style="width:900px" />
</center>
<br>
<br>
The awk and sed commands are not provided here, but if you look inside the script <a href="https://github.com/ewan-harney/hpc_blast2megan/blob/main/b2m_scripts/02_run_blast2lca.sh">02_run_blast2lca.sh</a> you will see that they all make use of <code>awk</code> conditional statements. Although the code appears a little complex at first, the idea behind awk conditional statements is a simple one: <code>if</code> some argument is true, do 'A', <code>else</code> do 'B'. In this script, many awk conditional statements have been joined together using the pipe '|'.</li>
</ul>
</div>
</div>
<div id="combining-output-with-dada2-output" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Combining output with DADA2 output<a href="06-blast2megan.html#combining-output-with-dada2-output" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The final bash script in the blast2megan pipeline is '03_run_make_summary_files.sh'. This script is simply a wrapper for an R script called '03_make_summary_files.R' that combines taxonomic assignment results from blast2megan with sequence data and ASV counts from DADA2 to create several summary files, including:</p>
<ul>
<li>ASV_taxa_seq_counts.tsv: a complete summary of taxonomic results (LCA taxon, taxon rank, and taxon path), sequence, and count results,<br />
</li>
<li>ps_taxamat.tsv : ASV taxonomic results in matrix format for phyloseq,<br />
</li>
<li>ps_countmat.tsv : ASV counts results in matrix format for phyloseq,<br />
</li>
<li>ps_phylogeny.rds : phylogenetic tree for phyloseq prepared according to protocol of <a href="https://f1000research.com/articles/5-1492/v1">Callahan et al. 2016</a>, see subsection <em>Construct the phylogenetic tree</em>.</li>
</ul>
<p>The second, third and fourth files are designed to be inputs for the popular community analysis R package <a href="https://joey711.github.io/phyloseq/">phyloseq</a>.</p>
<p>If you are interested in understanding how it works, take a look at the R code in <a href="https://github.com/ewan-harney/hpc_blast2megan/blob/main/b2m_scripts/03_make_summary_files.R">03_make_summary_files.R</a>. Among other things the script uses the R function <code>merge</code> to join together different data tables based on shared information (either the ASV sequence or the ASV number) and assigns column headers and rownames to the merged data (turning data tables into matrices). Notice that the script requires 'megan_summary_out.tsv' and 'megan_taxonpath_out.tsv (the output from the previous step, which it expects to find in the 'blast_out' directory) as well as the DADA2 output files '06_ASV_seqs.fasta' and '06_ASV_counts.tsv' (which it expects to find in the 'working_data' directory).</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="05-MEGAN.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="07-Summary.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
