<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 De-novo amplicon sequence variants identification | Bacterial 16S metabarcoding</title>
  <meta name="description" content="NEOF book for the Bacterial 16S metabarcoding workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 De-novo amplicon sequence variants identification | Bacterial 16S metabarcoding" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Bacterial 16S metabarcoding workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 De-novo amplicon sequence variants identification | Bacterial 16S metabarcoding" />
  
  <meta name="twitter:description" content="NEOF book for the Bacterial 16S metabarcoding workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Luca Lenzi and Matthew R. Gemmell" />


<meta name="date" content="2024-01-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="08-Trim_PCR_sequences.html"/>
<link rel="next" href="10-Taxonomic_assignment.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/NEOF_bordered_small.png"></a></li>
<li><a>Bacterial 16S metabarcoding</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Bacterial_16S_metabarcoding.html"><a href="01-Bacterial_16S_metabarcoding.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Bacterial_16S_metabarcoding.html"><a href="01-Bacterial_16S_metabarcoding.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#s-rrna"><i class="fa fa-check"></i><b>2.1</b> 16S rRNA</a></li>
<li class="chapter" data-level="2.2" data-path="02-Background.html"><a href="02-Background.html#its-and-other-possible-marker-genes"><i class="fa fa-check"></i><b>2.2</b> ITS and other possible marker genes</a></li>
<li class="chapter" data-level="2.3" data-path="02-Background.html"><a href="02-Background.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.3</b> Amplicon sequencing</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html"><i class="fa fa-check"></i><b>3</b> Introduction to QIIME2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#artifacts"><i class="fa fa-check"></i><b>3.1</b> Artifacts</a></li>
<li class="chapter" data-level="3.2" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#workflow"><i class="fa fa-check"></i><b>3.2</b> Workflow</a></li>
<li class="chapter" data-level="3.3" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#prepare-the-sequence-dataset"><i class="fa fa-check"></i><b>3.3</b> Prepare the sequence dataset</a></li>
<li class="chapter" data-level="3.4" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#amplicon-sequence-variants-identification"><i class="fa fa-check"></i><b>3.4</b> Amplicon Sequence Variants identification</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#otus"><i class="fa fa-check"></i><b>3.4.1</b> OTUs</a></li>
<li class="chapter" data-level="3.4.2" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#asvs"><i class="fa fa-check"></i><b>3.4.2</b> ASVs</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#assign-a-taxonomic-classification-to-each-asv"><i class="fa fa-check"></i><b>3.5</b> Assign a taxonomic classification to each ASV</a></li>
<li class="chapter" data-level="3.6" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#phylogenetic-tree-creation"><i class="fa fa-check"></i><b>3.6</b> Phylogenetic tree creation</a></li>
<li class="chapter" data-level="3.7" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#data-normalisation"><i class="fa fa-check"></i><b>3.7</b> Data normalisation</a></li>
<li class="chapter" data-level="3.8" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#alpha-diversity"><i class="fa fa-check"></i><b>3.8</b> Alpha-diversity</a></li>
<li class="chapter" data-level="3.9" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#beta-diversity"><i class="fa fa-check"></i><b>3.9</b> Beta-diversity</a></li>
<li class="chapter" data-level="3.10" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>3.10</b> Differential Abundance Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Data.html"><a href="05-Data.html"><i class="fa fa-check"></i><b>5</b> Data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Data.html"><a href="05-Data.html#manifestfile"><i class="fa fa-check"></i><b>5.1</b> PairedEndFastqManifestPhred33.csv</a></li>
<li class="chapter" data-level="5.2" data-path="05-Data.html"><a href="05-Data.html#metadata.file.txt"><i class="fa fa-check"></i><b>5.2</b> metadata.file.txt</a></li>
<li class="chapter" data-level="5.3" data-path="05-Data.html"><a href="05-Data.html#primers_seqs.fasta"><i class="fa fa-check"></i><b>5.3</b> primers_seqs.fasta</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-QIIME2_analysis_workflow.html"><a href="06-QIIME2_analysis_workflow.html"><i class="fa fa-check"></i><b>6</b> QIIME2 analysis workflow</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-QIIME2_analysis_workflow.html"><a href="06-QIIME2_analysis_workflow.html#workflow-1"><i class="fa fa-check"></i><b>6.1</b> Workflow</a></li>
<li class="chapter" data-level="6.2" data-path="06-QIIME2_analysis_workflow.html"><a href="06-QIIME2_analysis_workflow.html#activate-environment"><i class="fa fa-check"></i><b>6.2</b> Activate environment</a></li>
</ul></li>
<li class="part"><span><b>Preprocessing of data</b></span></li>
<li class="chapter" data-level="7" data-path="07-Import_sequences_into_QIIME2.html"><a href="07-Import_sequences_into_QIIME2.html"><i class="fa fa-check"></i><b>7</b> Sequence import</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Import_sequences_into_QIIME2.html"><a href="07-Import_sequences_into_QIIME2.html#import-run"><i class="fa fa-check"></i><b>7.1</b> Import: run</a></li>
<li class="chapter" data-level="7.2" data-path="07-Import_sequences_into_QIIME2.html"><a href="07-Import_sequences_into_QIIME2.html#chap7_vis"><i class="fa fa-check"></i><b>7.2</b> Import: visualise</a></li>
<li class="chapter" data-level="7.3" data-path="07-Import_sequences_into_QIIME2.html"><a href="07-Import_sequences_into_QIIME2.html#import-mcqs"><i class="fa fa-check"></i><b>7.3</b> Import: MCQs</a></li>
<li class="chapter" data-level="7.4" data-path="07-Import_sequences_into_QIIME2.html"><a href="07-Import_sequences_into_QIIME2.html#import-summary"><i class="fa fa-check"></i><b>7.4</b> Import: summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Trim_PCR_sequences.html"><a href="08-Trim_PCR_sequences.html"><i class="fa fa-check"></i><b>8</b> Trim the PCR primer sequences</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Trim_PCR_sequences.html"><a href="08-Trim_PCR_sequences.html#cutadapt-run"><i class="fa fa-check"></i><b>8.1</b> Cutadapt: run</a></li>
<li class="chapter" data-level="8.2" data-path="08-Trim_PCR_sequences.html"><a href="08-Trim_PCR_sequences.html#cutadapt-visualise"><i class="fa fa-check"></i><b>8.2</b> Cutadapt: visualise</a></li>
<li class="chapter" data-level="8.3" data-path="08-Trim_PCR_sequences.html"><a href="08-Trim_PCR_sequences.html#cutadapt-mcqs"><i class="fa fa-check"></i><b>8.3</b> Cutadapt: MCQs</a></li>
<li class="chapter" data-level="8.4" data-path="08-Trim_PCR_sequences.html"><a href="08-Trim_PCR_sequences.html#cutadapt-summary"><i class="fa fa-check"></i><b>8.4</b> Cutadapt: summary</a></li>
</ul></li>
<li class="part"><span><b>ASVs, Taxonomy, and phylogeny</b></span></li>
<li class="chapter" data-level="9" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html"><i class="fa fa-check"></i><b>9</b> De-novo amplicon sequence variants identification</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dada2-run"><i class="fa fa-check"></i><b>9.1</b> DADA2: run</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#truncation-length"><i class="fa fa-check"></i><b>9.1.1</b> Truncation length</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#cpdada2"><i class="fa fa-check"></i><b>9.1.2</b> Copy DADA2 output</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dada2-considerations"><i class="fa fa-check"></i><b>9.1.3</b> DADA2 considerations</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#tabledada2qzv"><i class="fa fa-check"></i><b>9.2</b> table-dada2.qzv</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#table-dada2.qzv-mcqs"><i class="fa fa-check"></i><b>9.2.1</b> table-dada2.qzv: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dada2-stats.qzv"><i class="fa fa-check"></i><b>9.3</b> dada2-stats.qzv</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dada2-stats.qzv-mcqs"><i class="fa fa-check"></i><b>9.3.1</b> dada2-stats.qzv: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dada2-summary"><i class="fa fa-check"></i><b>9.4</b> DADA2: summary</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html"><i class="fa fa-check"></i><b>10</b> ASV taxonomic assignment</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#greengenes2-database"><i class="fa fa-check"></i><b>10.1</b> Greengenes2 database</a></li>
<li class="chapter" data-level="10.2" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#extract-16s-region-of-interest"><i class="fa fa-check"></i><b>10.2</b> Extract 16s region of interest</a></li>
<li class="chapter" data-level="10.3" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#classifier-training"><i class="fa fa-check"></i><b>10.3</b> Classifier training</a></li>
<li class="chapter" data-level="10.4" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#taxonomy-assignment"><i class="fa fa-check"></i><b>10.4</b> Taxonomy assignment</a></li>
<li class="chapter" data-level="10.5" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#taxabar"><i class="fa fa-check"></i><b>10.5</b> Taxonomy bar chart production</a></li>
<li class="chapter" data-level="10.6" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#taxonomy-mcqs"><i class="fa fa-check"></i><b>10.6</b> Taxonomy: MCQs</a></li>
<li class="chapter" data-level="10.7" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#taxonomy-summary"><i class="fa fa-check"></i><b>10.7</b> Taxonomy: summary</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html"><i class="fa fa-check"></i><b>11</b> Phylogenetic tree construction</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html#mafft-alignment"><i class="fa fa-check"></i><b>11.1</b> MAFFT alignment</a></li>
<li class="chapter" data-level="11.2" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html#masking"><i class="fa fa-check"></i><b>11.2</b> Masking</a></li>
<li class="chapter" data-level="11.3" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html#un-rooted-tree-creation-with-fasttree"><i class="fa fa-check"></i><b>11.3</b> Un-rooted tree creation with FASTTREE</a></li>
<li class="chapter" data-level="11.4" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html#rooted-tree-creation"><i class="fa fa-check"></i><b>11.4</b> Rooted tree creation</a></li>
<li class="chapter" data-level="11.5" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html#phylogeny-summary"><i class="fa fa-check"></i><b>11.5</b> Phylogeny: summary</a></li>
</ul></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="12" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html"><i class="fa fa-check"></i><b>12</b> Sequencing depth evaluation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#rarefaction-run"><i class="fa fa-check"></i><b>12.1</b> Rarefaction: run</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#cprarefaction"><i class="fa fa-check"></i><b>12.1.1</b> Copy rarefaction output</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#rarefaction-visualise"><i class="fa fa-check"></i><b>12.2</b> Rarefaction: visualise</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#rarefaction-considerations"><i class="fa fa-check"></i><b>12.2.1</b> Rarefaction considerations</a></li>
<li class="chapter" data-level="12.2.2" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#rarefaction-mcqs"><i class="fa fa-check"></i><b>12.2.2</b> Rarefaction: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#rarefaction-summary"><i class="fa fa-check"></i><b>12.3</b> Rarefaction: summary</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Diversity_analysis.html"><a href="13-Diversity_analysis.html"><i class="fa fa-check"></i><b>13</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Diversity_analysis.html"><a href="13-Diversity_analysis.html#diversity-run"><i class="fa fa-check"></i><b>13.1</b> Diversity: run</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="14-Alpha_diversity.html"><a href="14-Alpha_diversity.html"><i class="fa fa-check"></i><b>14</b> Alpha diveristy statistical analysis</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Alpha_diversity.html"><a href="14-Alpha_diversity.html#alpha-diversity-run"><i class="fa fa-check"></i><b>14.1</b> Alpha diversity: run</a></li>
<li class="chapter" data-level="14.2" data-path="14-Alpha_diversity.html"><a href="14-Alpha_diversity.html#alpha-diversity-visualise"><i class="fa fa-check"></i><b>14.2</b> Alpha diversity: visualise</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="14-Alpha_diversity.html"><a href="14-Alpha_diversity.html#alpha-diversity-mcqs"><i class="fa fa-check"></i><b>14.2.1</b> Alpha diversity: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="14-Alpha_diversity.html"><a href="14-Alpha_diversity.html#alpha-diversity-summary"><i class="fa fa-check"></i><b>14.3</b> Alpha diversity: summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html"><i class="fa fa-check"></i><b>15</b> Beta diversity statistical analysis</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html#beta-diveristy-pcoa"><i class="fa fa-check"></i><b>15.1</b> Beta diveristy: PCoA</a></li>
<li class="chapter" data-level="15.2" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html#beta-diversity-run"><i class="fa fa-check"></i><b>15.2</b> Beta diversity: run</a></li>
<li class="chapter" data-level="15.3" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html#beta-diversity-visualise"><i class="fa fa-check"></i><b>15.3</b> Beta diversity: visualise</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html#beta-diversity-mcqs"><i class="fa fa-check"></i><b>15.3.1</b> Beta diversity: MCQs</a></li>
<li class="chapter" data-level="15.3.2" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html#beta-diversity-summary"><i class="fa fa-check"></i><b>15.3.2</b> Beta diversity: summary</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html"><i class="fa fa-check"></i><b>16</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#taxa-collapse"><i class="fa fa-check"></i><b>16.1</b> Taxa collapse</a></li>
<li class="chapter" data-level="16.2" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancom-bc-analysis"><i class="fa fa-check"></i><b>16.2</b> ANCOM-BC analysis</a></li>
<li class="chapter" data-level="16.3" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancombc-tabulate"><i class="fa fa-check"></i><b>16.3</b> ANCOMBC tabulate</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancombc-tabulate-visualise"><i class="fa fa-check"></i><b>16.3.1</b> ANCOMBC tabulate: visualise</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancombc-bar-plots"><i class="fa fa-check"></i><b>16.4</b> ANCOMBC bar plots</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancom-bc-bar-plots-visualise"><i class="fa fa-check"></i><b>16.4.1</b> ANCOM-BC bar plots: visualise</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancom-bc-mcqs"><i class="fa fa-check"></i><b>16.5</b> ANCOM-BC: MCQs</a></li>
<li class="chapter" data-level="16.6" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancom-bc-summary"><i class="fa fa-check"></i><b>16.6</b> ANCOM-BC: summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Final_consideration.html"><a href="17-Final_consideration.html"><i class="fa fa-check"></i><b>17</b> Final consideration</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="18-Appendix.html"><a href="18-Appendix.html"><i class="fa fa-check"></i><b>A</b> Resources</a>
<ul>
<li class="chapter" data-level="A.1" data-path="18-Appendix.html"><a href="18-Appendix.html#suggested-readings"><i class="fa fa-check"></i><b>A.1</b> Suggested readings</a></li>
<li class="chapter" data-level="A.2" data-path="18-Appendix.html"><a href="18-Appendix.html#other-useful-resources"><i class="fa fa-check"></i><b>A.2</b> Other useful resources</a></li>
<li class="chapter" data-level="A.3" data-path="18-Appendix.html"><a href="18-Appendix.html#references"><i class="fa fa-check"></i><b>A.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="18-Appendix.html"><a href="18-Appendix.html#mamba-installs"><i class="fa fa-check"></i><b>B</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="B.1" data-path="18-Appendix.html"><a href="18-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>B.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="18-Appendix.html"><a href="18-Appendix.html#diversity-metrics"><i class="fa fa-check"></i><b>C</b> Diversity metrics</a>
<ul>
<li class="chapter" data-level="C.1" data-path="18-Appendix.html"><a href="18-Appendix.html#alphamet"><i class="fa fa-check"></i><b>C.1</b> Alpha diversity metrics</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="18-Appendix.html"><a href="18-Appendix.html#obvs"><i class="fa fa-check"></i><b>C.1.1</b> Observed features</a></li>
<li class="chapter" data-level="C.1.2" data-path="18-Appendix.html"><a href="18-Appendix.html#evenness"><i class="fa fa-check"></i><b>C.1.2</b> Evenness</a></li>
<li class="chapter" data-level="C.1.3" data-path="18-Appendix.html"><a href="18-Appendix.html#faith"><i class="fa fa-check"></i><b>C.1.3</b> Faith's PD (phylogenetic diversity)</a></li>
<li class="chapter" data-level="C.1.4" data-path="18-Appendix.html"><a href="18-Appendix.html#simpson"><i class="fa fa-check"></i><b>C.1.4</b> Simpson</a></li>
<li class="chapter" data-level="C.1.5" data-path="18-Appendix.html"><a href="18-Appendix.html#simpsone"><i class="fa fa-check"></i><b>C.1.5</b> Simpson evenness measure (simpson_e)</a></li>
<li class="chapter" data-level="C.1.6" data-path="18-Appendix.html"><a href="18-Appendix.html#shannon"><i class="fa fa-check"></i><b>C.1.6</b> Shannon</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="18-Appendix.html"><a href="18-Appendix.html#beta-diversity-metrics"><i class="fa fa-check"></i><b>C.2</b> Beta diversity metrics</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="18-Appendix.html"><a href="18-Appendix.html#unifrac_explanation"><i class="fa fa-check"></i><b>C.2.1</b> Weighted and unweighted UniFrac distances</a></li>
<li class="chapter" data-level="C.2.2" data-path="18-Appendix.html"><a href="18-Appendix.html#bray_explanation"><i class="fa fa-check"></i><b>C.2.2</b> Bray-Curtis</a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="18-Appendix.html"><a href="18-Appendix.html#diversity-resources"><i class="fa fa-check"></i><b>C.3</b> Diversity Resources</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial 16S metabarcoding</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dada2" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> De-novo amplicon sequence variants identification<a href="09-ASV_identification.html#dada2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/clusters.png" width="20%" style="display: block; margin: auto;" /></p>
<p>We will use DADA2 to denoise and identify the amplicon sequence variants (ASVs) among all the samples.</p>
<div id="dada2-run" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> DADA2: run<a href="09-ASV_identification.html#dada2-run" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/play_purple.png" width="15%" style="display: block; margin: auto;" /></p>
<p>To perform the variant sequence identification step with DADA2, the command is (<strong>don't run this command, <a href="09-ASV_identification.html#cpdada2">see below command</a></strong>):</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb10-1"><a href="09-ASV_identification.html#cb10-1" tabindex="-1"></a><span class="ex">qiime</span> dada2 denoise-paired <span class="dt">\</span></span>
<span id="cb10-2"><a href="09-ASV_identification.html#cb10-2" tabindex="-1"></a>--i-demultiplexed-seqs paired-end-demux.trim.qza <span class="dt">\</span></span>
<span id="cb10-3"><a href="09-ASV_identification.html#cb10-3" tabindex="-1"></a>--p-trunc-len-f 220 <span class="dt">\</span></span>
<span id="cb10-4"><a href="09-ASV_identification.html#cb10-4" tabindex="-1"></a>--p-trunc-len-r 220 <span class="dt">\</span></span>
<span id="cb10-5"><a href="09-ASV_identification.html#cb10-5" tabindex="-1"></a>--o-representative-sequences rep-seqs-dada2.qza <span class="dt">\</span></span>
<span id="cb10-6"><a href="09-ASV_identification.html#cb10-6" tabindex="-1"></a>--o-table table-dada2.qza <span class="dt">\</span></span>
<span id="cb10-7"><a href="09-ASV_identification.html#cb10-7" tabindex="-1"></a>--o-denoising-stats dada2-stats.qza <span class="dt">\</span></span>
<span id="cb10-8"><a href="09-ASV_identification.html#cb10-8" tabindex="-1"></a>--verbose</span></code></pre></div>
<div id="parameters-3" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="09-ASV_identification.html#parameters-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/parameter_purple.png" width="10%" style="display: block; margin: auto;" /></p>
<ul>
<li><strong><code>--i-demultiplexed-seqs</code></strong>: Input artifact file containing the trimmed demultiplexed sequences.</li>
<li><strong><code>--p-trunc-len-f</code></strong>: This will truncate (trim) the forward sequences to retain only the first <code>220</code> bases.</li>
<li><strong><code>--p-trunc-len-r</code></strong>: This will truncate (trim) the reverse sequences to retain only the first <code>220</code> bases.</li>
<li><strong><code>--o-representative-sequences</code></strong>: Output artifact containing the identified sequence variants.</li>
<li><strong><code>--o-table</code></strong>: Output artifact containing the abundance table for the identified variants.</li>
<li><strong><code>--o-denoising-stats</code></strong>: Output artifact containing the sequence numbers pre- and post- trimming and denoising.</li>
<li><strong><code>--verbose</code></strong>: Causes command to display verbose output to screen.</li>
</ul>
<p><a href="https://docs.qiime2.org/2023.5/plugins/available/dada2/denoise-paired/">QIIME2 docs on <code>qiime dada2 denoise-paired</code></a></p>
</div>
<div id="truncation-length" class="section level3 hasAnchor" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Truncation length<a href="09-ASV_identification.html#truncation-length" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/measuring_tape.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We chose <code>220</code> for <code>--p-trunc-len-f</code> &amp; <code>--p-trunc-len-r</code> as the read quality was very good up to the 220 bp position for both the forward and reverse reads.
Additionally, our amplicon length is ~291bp. The combined length of our forward and reverse reads is 440bp, therefore we would have an overlap greater than 100bp which is very good (440 - 291 = 149bp).</p>
<p>As a reminder we are using the primer 515FB-806RB for the V4 16S rRNA gene. The name 515FB-806RB represents the location within an <strong>E.coli</strong> 16S gene that the primers will hit (515 for the forward, 806 for the reverse). Therefore the amplicon length for <strong>E.coli</strong> will be 806 - 515 = 291bp. As with all genes this size will vary in length across all the bacterial taxa but only by a few base pairs.</p>
</div>
<div id="cpdada2" class="section level3 hasAnchor" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> Copy DADA2 output<a href="09-ASV_identification.html#cpdada2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/copy.png" width="15%" style="display: block; margin: auto;" /></p>
<p>The command will take a long time, so stop the command (ctrl+c), if you started it, and copy the final output files, which have been previously made, to your current directory:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb11-1"><a href="09-ASV_identification.html#cb11-1" tabindex="-1"></a><span class="co">#Copy the files from the specified directory to your working directory</span></span>
<span id="cb11-2"><a href="09-ASV_identification.html#cb11-2" tabindex="-1"></a><span class="fu">cp</span> /pub14/tea/nsc206/NEOF/16s_workshop/chapter_9/<span class="pp">*</span> ~/Metagenetics</span></code></pre></div>
<p>Through the above command you will now have the following three files:</p>
<ul>
<li><code>rep-seqs-dada2.qza</code>: Artifact containing the identified sequence variants.</li>
<li><code>table-dada2.qza</code>: Artifact containing the ASV abundance table.</li>
<li><code>dada2-stats.qza</code>: Artifact which summarises the statistics after the denoising steps.</li>
</ul>
</div>
<div id="dada2-considerations" class="section level3 hasAnchor" number="9.1.3">
<h3><span class="header-section-number">9.1.3</span> DADA2 considerations<a href="09-ASV_identification.html#dada2-considerations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/considerations.png" width="15%" style="display: block; margin: auto;" /></p>
<p>This may be one of the longest steps of the whole pipeline. The running time is proportional to the complexity of the sequences (how many ASVs are in the dataset) and to the quality of the sequences (how many errors are present). If this step is taking too long in one of your future projects, you may try to use <a href="https://docs.qiime2.org/2023.5/plugins/available/deblur/">‘deblur’</a> to de-noise your data (after joining the reads with vsearch), which is generally faster.</p>
<p>In the case of a marker gene with very variable amplicon sequence lengths, as in the case of ITS amplicon sequencing, a possible alternative denoising strategy is to disable the filtering by truncation length using the options: <code>--p-trunc-len-f 0</code> and <code>--p-trunc-len-r 0</code>. To exclude the low quality tails the <code>--p-trunc-q 20</code> option is now required, this option will trim all the sequences after the first base with quality below ‘20’ (this may be very stringent, so you may want to try with lower values as well). A further possibility may be to use only either the forward or reverse read file, for this the <code>qiime dada2 denoise-single</code> should be used instead of the above dada2 command.</p>
</div>
</div>
<div id="tabledada2qzv" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> table-dada2.qzv<a href="09-ASV_identification.html#tabledada2qzv" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/table.png" width="15%" style="display: block; margin: auto;" />
To visualise information on the ASV table we will run:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb12-1"><a href="09-ASV_identification.html#cb12-1" tabindex="-1"></a><span class="ex">qiime</span> feature-table summarize <span class="dt">\</span></span>
<span id="cb12-2"><a href="09-ASV_identification.html#cb12-2" tabindex="-1"></a>--i-table table-dada2.qza <span class="dt">\</span></span>
<span id="cb12-3"><a href="09-ASV_identification.html#cb12-3" tabindex="-1"></a>--o-visualization table-dada2.qzv <span class="dt">\</span></span>
<span id="cb12-4"><a href="09-ASV_identification.html#cb12-4" tabindex="-1"></a>--m-sample-metadata-file metadata.file.txt</span></code></pre></div>
<div id="parameters-4" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="09-ASV_identification.html#parameters-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/parameter_purple.png" width="10%" style="display: block; margin: auto;" /></p>
<ul>
<li><strong><code>--i-table</code></strong>: Input artifact file containing the abundance table of the ASVs.</li>
<li><strong><code>--o-visualization</code></strong>: Output visualisation artifact.</li>
<li><strong><code>--m-sample-metadata-file</code></strong>: <a href="05-Data.html#metadata.file.txt">Metadata file</a> for our samples.</li>
</ul>
<p><a href="https://docs.qiime2.org/2023.5/plugins/available/feature-table/summarize/">QIIME2 docs on <code>qiime feature-table summarize</code></a></p>
</div>
<div id="output" class="section level4 unnumbered hasAnchor">
<h4>Output<a href="09-ASV_identification.html#output" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/output_file_purple.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Now load the obtained <code>table-dada2.qzv</code> into the ‘view.qiime2.org’ website.</p>
<p>The page opened with <code>table-dada2.qzv</code> contains 3 tabs:</p>
<ol style="list-style-type: decimal">
<li><strong>Overview</strong>: An overview page with various information.
<ul>
<li><strong>Table summary</strong>
<ul>
<li><strong>Number of samples</strong></li>
<li><strong>Number of features</strong>: In this case features are ASVs. If this was a taxa table features would be the number of taxa.</li>
<li><strong>Total frequency</strong>: Total number of sequences (i.e. total abundance across all samples and ASVs).</li>
</ul></li>
<li><strong>Frequency per sample</strong>: Summary of number of sequences.
<ul>
<li><strong>IQR table</strong>: Table showing the Interquartile range of sample frequencies.</li>
<li><strong>Histogram of frequency per sample</strong>: This gives a rough view as the start and end of the histogram is based on the minimum and maximum sample frequencies.</li>
</ul></li>
<li><strong>Frequency per feature</strong>: Summary of number of sequences per feature (ASV/taxa).
<ul>
<li><strong>IQR table</strong>: Table showing the Interquartile range of feature frequencies.</li>
<li><strong>Histogram of frequency per sample</strong>: This shows a rough view of the number of features (y axis) with different frequencies (x axis). Note that the scales are logarithmic. The left side of the plot shows how many rare ASVs there are (ASVs represented by a low amount of sequences, i.e. low abundance ASVs). The right side of the plot shows the amount of high abundance ASVs.</li>
</ul></li>
</ul></li>
<li><strong>Interactive Sample Detail</strong>: This interactive section allows you to explore sampling depth. Experiment using the Plot Controls.
<ul>
<li><strong>Plot</strong>: This shows the number of samples per metadata category.</li>
<li><strong>Feature count table</strong>: Table showing feature count per sample.</li>
<li><strong>Plot Controls</strong>: Controls to change and update the plot and table.
<ul>
<li><strong>Save as..</strong>: Controls to save the current plot.</li>
<li><strong>Metadata Category</strong>: This allows you to change the values on the x axis of the plot.</li>
<li><strong>Sampling Depth</strong>: Choosing a sampling depth will update the following.
<ul>
<li><strong>Plot</strong>: The plot will be updated to grey out the tops of the bars by removing samples with a depth lower than the chosen sampling depth.</li>
<li><strong>Feature count table</strong>: Samples with a lower depth than the sampling depth will be highlighted red.</li>
<li><strong>Plot Controls</strong>: Below the controls is a section of text showing the number of features and samples retained by the chosen sampling depth.</li>
</ul></li>
</ul></li>
</ul></li>
<li><strong>Feature detail</strong>: This is a table with 3 columns.
<ul>
<li>1st column: This is the ASV id. This ID represents the exact amplicon sequence variant. This is useful as if this ASV id is in another sequencing experiment it means they are identical between the two studies. This allows for easy merging of multiple ASV tables.</li>
<li><strong>Frequency</strong>: Number of sequences representing the ASV across all samples. I.e. the total abundance of the ASV.</li>
<li><strong># of Samples Observed in</strong>: Number of samples the ASVs was detected in.</li>
</ul></li>
</ol>
</div>
<div id="table-dada2.qzv-mcqs" class="section level3 hasAnchor" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> table-dada2.qzv: MCQs<a href="09-ASV_identification.html#table-dada2.qzv-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/question_bubble_purple.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Attempt the below MCQs.</p>
<ol style="list-style-type: decimal">
<li>How many ASVs (features) were detected in the dataset?
<div id="radio_BXYNDOAMOI" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BXYNDOAMOI" value="answer"></input> <span><strong>2,139</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BXYNDOAMOI" value=""></input> <span><strong>118,703.5</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BXYNDOAMOI" value=""></input> <span><strong>258,440</strong></span></label>
</div></li>
<li>What is the median number of sequences per sample?
<div id="radio_BUVCXBQGBZ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BUVCXBQGBZ" value=""></input> <span><strong>2,139</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BUVCXBQGBZ" value="answer"></input> <span><strong>118,703.5</strong></span></label><label><input type="radio" autocomplete="off" name="radio_BUVCXBQGBZ" value=""></input> <span><strong>258,440</strong></span></label>
</div></li>
<li>What is the maximum number of sequences per ASV?
<div id="radio_PIWPUCVYOT" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_PIWPUCVYOT" value=""></input> <span><strong>2,139</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PIWPUCVYOT" value=""></input> <span><strong>118,703.5</strong></span></label><label><input type="radio" autocomplete="off" name="radio_PIWPUCVYOT" value="answer"></input> <span><strong>258,440</strong></span></label>
</div></li>
<li>How many samples have &gt;=100,000 sequences?
<div id="radio_EYVPRFLTXX" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_EYVPRFLTXX" value=""></input> <span><strong>4</strong></span></label><label><input type="radio" autocomplete="off" name="radio_EYVPRFLTXX" value=""></input> <span><strong>5</strong></span></label><label><input type="radio" autocomplete="off" name="radio_EYVPRFLTXX" value="answer"></input> <span><strong>13</strong></span></label>
</div></li>
<li>How many Toilet (Metadata Category: Location) samples have &gt;=80,00 sequences?
<div id="radio_HFHWXXWDUO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_HFHWXXWDUO" value=""></input> <span><strong>4</strong></span></label><label><input type="radio" autocomplete="off" name="radio_HFHWXXWDUO" value="answer"></input> <span><strong>5</strong></span></label><label><input type="radio" autocomplete="off" name="radio_HFHWXXWDUO" value=""></input> <span><strong>13</strong></span></label>
</div></li>
<li>How many MainBuilding (Metadata Category: Place) samples have &gt;=120,000 sequences?
<div id="radio_MYBPHFGGIJ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_MYBPHFGGIJ" value="answer"></input> <span><strong>4</strong></span></label><label><input type="radio" autocomplete="off" name="radio_MYBPHFGGIJ" value=""></input> <span><strong>5</strong></span></label><label><input type="radio" autocomplete="off" name="radio_MYBPHFGGIJ" value=""></input> <span><strong>13</strong></span></label>
</div></li>
<li>How many sequences represent the second most abundant ASV?
<div id="radio_FEWFDGZCAC" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_FEWFDGZCAC" value=""></input> <span><strong>15</strong></span></label><label><input type="radio" autocomplete="off" name="radio_FEWFDGZCAC" value="answer"></input> <span><strong>225,752</strong></span></label><label><input type="radio" autocomplete="off" name="radio_FEWFDGZCAC" value=""></input> <span><strong>258,440</strong></span></label>
</div></li>
<li>How many samples is the 10th most abundant ASV observed in?
<div id="radio_EQFDMJIZOF" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_EQFDMJIZOF" value="answer"></input> <span><strong>15</strong></span></label><label><input type="radio" autocomplete="off" name="radio_EQFDMJIZOF" value=""></input> <span><strong>225,752</strong></span></label><label><input type="radio" autocomplete="off" name="radio_EQFDMJIZOF" value=""></input> <span><strong>258,440</strong></span></label>
</div></li>
</ol>
</div>
</div>
<div id="dada2-stats.qzv" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> dada2-stats.qzv<a href="09-ASV_identification.html#dada2-stats.qzv" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/filter.png" width="15%" style="display: block; margin: auto;" />
Next we'll look at the denoising stats. Run the following command.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb13-1"><a href="09-ASV_identification.html#cb13-1" tabindex="-1"></a><span class="ex">qiime</span> metadata tabulate <span class="dt">\</span></span>
<span id="cb13-2"><a href="09-ASV_identification.html#cb13-2" tabindex="-1"></a>--m-input-file dada2-stats.qza <span class="dt">\</span></span>
<span id="cb13-3"><a href="09-ASV_identification.html#cb13-3" tabindex="-1"></a>--o-visualization dada2-stats.qzv</span></code></pre></div>
<div id="parameters-5" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="09-ASV_identification.html#parameters-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/parameter_blue.png" width="10%" style="display: block; margin: auto;" /></p>
<ul>
<li><strong><code>--m-input-file</code></strong>: Input artifact file containing the denoising stats from DADA2.</li>
<li><strong><code>--o-visualization</code></strong>: Output visualisation artifact.</li>
</ul>
<p><a href="https://docs.qiime2.org/2023.5/plugins/available/metadata/tabulate/">QIIME2 docs on <code>qiime metadata tabulate</code></a></p>
</div>
<div id="output-1" class="section level4 unnumbered hasAnchor">
<h4>Output<a href="09-ASV_identification.html#output-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/output_file_blue.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Open the <code>dada2-stats.qzv</code> visualisation with QIIME2 view.</p>
<p>You will now see a denoising stats table with the following columns:</p>
<ul>
<li><strong>sample-id</strong></li>
<li><strong>input</strong>: Number of sequence pairs the sample started with prior to the denoising steps.</li>
<li><strong>filtered</strong>: Number of sequence pairs retained after user specified filtering (e.g. <code>--p-trunc-len-f 220</code> will truncate forward reads but also remove sequence pairs where the forward length is shorter than 220).</li>
<li><strong>percentage of input passed filter</strong>: Percentage of input reads that passed the filtering step.</li>
<li><strong>denoised</strong>: Number of filtered sequence pairs that were successfully denoised.</li>
<li><strong>merged</strong>: Number of filtered and denoised paired sequences that were successfully merged (forward and reverse read merged into one sequence).</li>
<li><strong>percentage of input merged</strong>: Percentage of input reads that passed the filtering, denoising, and merging steps.</li>
<li><strong>non-chimeric</strong>: Number of filtered, denoised, and merged sequences retained after chimera removal. This is the final number of sequences per sample.</li>
<li><strong>percentage of input non-chimeric</strong>: Percentage of input reads retained at the end.</li>
</ul>
<p>This visualisation is very useful to see where our sequences are being filtered. It is possible to be too stringent and not have enough sequences for some samples. This will give you an idea on how you can repeat the denoising stats with changed options to retain more samples.</p>
</div>
<div id="dada2-stats.qzv-mcqs" class="section level3 hasAnchor" number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> dada2-stats.qzv: MCQs<a href="09-ASV_identification.html#dada2-stats.qzv-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/question_bubble_blue.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Attempt the below MCQs?</p>
<ol style="list-style-type: decimal">
<li>In which DADA2 step were most of the sequences lost?
<div id="radio_SHBKNHEPGU" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_SHBKNHEPGU" value="answer"></input> <span><strong>Filtering</strong></span></label><label><input type="radio" autocomplete="off" name="radio_SHBKNHEPGU" value=""></input> <span><strong>Denoising and merging</strong></span></label><label><input type="radio" autocomplete="off" name="radio_SHBKNHEPGU" value=""></input> <span><strong>Chimera removal</strong></span></label>
</div></li>
<li>Which sample retained the smallest percentage of its input at the end (i.e. after filtering, denoising, merging, and chimera removal)?
<div id="radio_MJKUMWAUPU" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_MJKUMWAUPU" value=""></input> <span><strong>1K1MB</strong></span></label><label><input type="radio" autocomplete="off" name="radio_MJKUMWAUPU" value=""></input> <span><strong>1K1E</strong></span></label><label><input type="radio" autocomplete="off" name="radio_MJKUMWAUPU" value="answer"></input> <span><strong>1K2M</strong></span></label>
</div></li>
</ol>
</div>
</div>
<div id="dada2-summary" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> DADA2: summary<a href="09-ASV_identification.html#dada2-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_blue.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Our reads have been filtered, denoised, and merged. Additionally, chimeras have been removed. With this carried out we have an ASV abundance table, representative sequences, and stats on the various steps that have been carried out.</p>
<p>Through visualising the table and stats we can determine that there are a good amount of sequences per sample (&gt;20,000) and there is a good amount of samples per sample group. We are therefore happy with our data and can continue on.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="08-Trim_PCR_sequences.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="10-Taxonomic_assignment.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
