<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 ASV taxonomic assignment | Bacterial 16S metabarcoding</title>
  <meta name="description" content="NEOF book for the Bacterial 16S metabarcoding workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 ASV taxonomic assignment | Bacterial 16S metabarcoding" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Bacterial 16S metabarcoding workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 ASV taxonomic assignment | Bacterial 16S metabarcoding" />
  
  <meta name="twitter:description" content="NEOF book for the Bacterial 16S metabarcoding workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Luca Lenzi and Matthew R. Gemmell" />


<meta name="date" content="2023-11-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="09-ASV_identification.html"/>
<link rel="next" href="11-Phylogentic_tree.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>16S metabarcoding</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Bacterial_16S_metabarcoding.html"><a href="01-Bacterial_16S_metabarcoding.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="01-Bacterial_16S_metabarcoding.html"><a href="01-Bacterial_16S_metabarcoding.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#s-rrna"><i class="fa fa-check"></i><b>2.1</b> 16S rRNA</a></li>
<li class="chapter" data-level="2.2" data-path="02-Background.html"><a href="02-Background.html#its-and-other-possible-marker-genes"><i class="fa fa-check"></i><b>2.2</b> ITS and other possible marker genes</a></li>
<li class="chapter" data-level="2.3" data-path="02-Background.html"><a href="02-Background.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.3</b> Amplicon sequencing</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html"><i class="fa fa-check"></i><b>3</b> Introduction to QIIME2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#artifacts"><i class="fa fa-check"></i><b>3.1</b> Artifacts</a></li>
<li class="chapter" data-level="3.2" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#workflow"><i class="fa fa-check"></i><b>3.2</b> Workflow</a></li>
<li class="chapter" data-level="3.3" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#prepare-the-sequence-dataset"><i class="fa fa-check"></i><b>3.3</b> Prepare the sequence dataset</a></li>
<li class="chapter" data-level="3.4" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#amplicon-sequence-variants-identification"><i class="fa fa-check"></i><b>3.4</b> Amplicon Sequence Variants identification</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#otus"><i class="fa fa-check"></i><b>3.4.1</b> OTUs</a></li>
<li class="chapter" data-level="3.4.2" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#asvs"><i class="fa fa-check"></i><b>3.4.2</b> ASVs</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#assign-a-taxonomic-classification-to-each-asv"><i class="fa fa-check"></i><b>3.5</b> Assign a taxonomic classification to each ASV</a></li>
<li class="chapter" data-level="3.6" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#phylogenetic-tree-creation"><i class="fa fa-check"></i><b>3.6</b> Phylogenetic tree creation</a></li>
<li class="chapter" data-level="3.7" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#data-normalisation"><i class="fa fa-check"></i><b>3.7</b> Data normalisation</a></li>
<li class="chapter" data-level="3.8" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#alpha-diversity"><i class="fa fa-check"></i><b>3.8</b> Alpha-diversity</a></li>
<li class="chapter" data-level="3.9" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#beta-diversity"><i class="fa fa-check"></i><b>3.9</b> Beta-diversity</a></li>
<li class="chapter" data-level="3.10" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>3.10</b> Differential Abundance Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Data.html"><a href="05-Data.html"><i class="fa fa-check"></i><b>5</b> Data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Data.html"><a href="05-Data.html#manifestfile"><i class="fa fa-check"></i><b>5.1</b> PairedEndFastqManifestPhred33.csv</a></li>
<li class="chapter" data-level="5.2" data-path="05-Data.html"><a href="05-Data.html#metadata.file.txt"><i class="fa fa-check"></i><b>5.2</b> metadata.file.txt</a></li>
<li class="chapter" data-level="5.3" data-path="05-Data.html"><a href="05-Data.html#primers_seqs.fasta"><i class="fa fa-check"></i><b>5.3</b> primers_seqs.fasta</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-QIIME2_analysis_workflow.html"><a href="06-QIIME2_analysis_workflow.html"><i class="fa fa-check"></i><b>6</b> QIIME2 analysis workflow</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-QIIME2_analysis_workflow.html"><a href="06-QIIME2_analysis_workflow.html#workflow-1"><i class="fa fa-check"></i><b>6.1</b> Workflow</a></li>
<li class="chapter" data-level="6.2" data-path="06-QIIME2_analysis_workflow.html"><a href="06-QIIME2_analysis_workflow.html#activate-environment"><i class="fa fa-check"></i><b>6.2</b> Activate environment</a></li>
</ul></li>
<li class="part"><span><b>Preprocessing of data</b></span></li>
<li class="chapter" data-level="7" data-path="07-Import_sequences_into_QIIME2.html"><a href="07-Import_sequences_into_QIIME2.html"><i class="fa fa-check"></i><b>7</b> Sequence import</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Import_sequences_into_QIIME2.html"><a href="07-Import_sequences_into_QIIME2.html#import-run"><i class="fa fa-check"></i><b>7.1</b> Import: run</a></li>
<li class="chapter" data-level="7.2" data-path="07-Import_sequences_into_QIIME2.html"><a href="07-Import_sequences_into_QIIME2.html#chap7_vis"><i class="fa fa-check"></i><b>7.2</b> Import: visualise</a></li>
<li class="chapter" data-level="7.3" data-path="07-Import_sequences_into_QIIME2.html"><a href="07-Import_sequences_into_QIIME2.html#import-mcqs"><i class="fa fa-check"></i><b>7.3</b> Import: MCQs</a></li>
<li class="chapter" data-level="7.4" data-path="07-Import_sequences_into_QIIME2.html"><a href="07-Import_sequences_into_QIIME2.html#import-summary"><i class="fa fa-check"></i><b>7.4</b> Import: summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Trim_PCR_sequences.html"><a href="08-Trim_PCR_sequences.html"><i class="fa fa-check"></i><b>8</b> Trim the PCR primer sequences</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Trim_PCR_sequences.html"><a href="08-Trim_PCR_sequences.html#cutadapt-run"><i class="fa fa-check"></i><b>8.1</b> Cutadapt: run</a></li>
<li class="chapter" data-level="8.2" data-path="08-Trim_PCR_sequences.html"><a href="08-Trim_PCR_sequences.html#cutadapt-visualise"><i class="fa fa-check"></i><b>8.2</b> Cutadapt: visualise</a></li>
<li class="chapter" data-level="8.3" data-path="08-Trim_PCR_sequences.html"><a href="08-Trim_PCR_sequences.html#cutadapt-mcqs"><i class="fa fa-check"></i><b>8.3</b> Cutadapt: MCQs</a></li>
<li class="chapter" data-level="8.4" data-path="08-Trim_PCR_sequences.html"><a href="08-Trim_PCR_sequences.html#cutadapt-summary"><i class="fa fa-check"></i><b>8.4</b> Cutadapt: summary</a></li>
</ul></li>
<li class="part"><span><b>ASVs, Taxonomy, and phylogeny</b></span></li>
<li class="chapter" data-level="9" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html"><i class="fa fa-check"></i><b>9</b> De-novo amplicon sequence variants identification</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dada2-run"><i class="fa fa-check"></i><b>9.1</b> DADA2: run</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#truncation-length"><i class="fa fa-check"></i><b>9.1.1</b> Truncation length</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#cpdada2"><i class="fa fa-check"></i><b>9.1.2</b> Copy DADA2 output</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dada2-considerations"><i class="fa fa-check"></i><b>9.1.3</b> DADA2 considerations</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#tabledada2qzv"><i class="fa fa-check"></i><b>9.2</b> table-dada2.qzv</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#table-dada2.qzv-mcqs"><i class="fa fa-check"></i><b>9.2.1</b> table-dada2.qzv: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dada2-stats.qzv"><i class="fa fa-check"></i><b>9.3</b> dada2-stats.qzv</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dada2-stats.qzv-mcqs"><i class="fa fa-check"></i><b>9.3.1</b> dada2-stats.qzv: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html#dada2-summary"><i class="fa fa-check"></i><b>9.4</b> DADA2: summary</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html"><i class="fa fa-check"></i><b>10</b> ASV taxonomic assignment</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#greengenes2-database"><i class="fa fa-check"></i><b>10.1</b> Greengenes2 database</a></li>
<li class="chapter" data-level="10.2" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#extract-16s-region-of-interest"><i class="fa fa-check"></i><b>10.2</b> Extract 16s region of interest</a></li>
<li class="chapter" data-level="10.3" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#classifier-training"><i class="fa fa-check"></i><b>10.3</b> Classifier training</a></li>
<li class="chapter" data-level="10.4" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#taxonomy-assignment"><i class="fa fa-check"></i><b>10.4</b> Taxonomy assignment</a></li>
<li class="chapter" data-level="10.5" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#taxabar"><i class="fa fa-check"></i><b>10.5</b> Taxonomy bar chart production</a></li>
<li class="chapter" data-level="10.6" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#taxonomy-mcqs"><i class="fa fa-check"></i><b>10.6</b> Taxonomy: MCQs</a></li>
<li class="chapter" data-level="10.7" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html#taxonomy-summary"><i class="fa fa-check"></i><b>10.7</b> Taxonomy: summary</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html"><i class="fa fa-check"></i><b>11</b> Phylogenetic tree construction</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html#mafft-alignment"><i class="fa fa-check"></i><b>11.1</b> MAFFT alignment</a></li>
<li class="chapter" data-level="11.2" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html#masking"><i class="fa fa-check"></i><b>11.2</b> Masking</a></li>
<li class="chapter" data-level="11.3" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html#un-rooted-tree-creation-with-fasttree"><i class="fa fa-check"></i><b>11.3</b> Un-rooted tree creation with FASTTREE</a></li>
<li class="chapter" data-level="11.4" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html#rooted-tree-creation"><i class="fa fa-check"></i><b>11.4</b> Rooted tree creation</a></li>
<li class="chapter" data-level="11.5" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html#phylogeny-summary"><i class="fa fa-check"></i><b>11.5</b> Phylogeny: summary</a></li>
</ul></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="12" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html"><i class="fa fa-check"></i><b>12</b> Sequencing depth evaluation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#rarefaction-run"><i class="fa fa-check"></i><b>12.1</b> Rarefaction: run</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#cprarefaction"><i class="fa fa-check"></i><b>12.1.1</b> Copy rarefaction output</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#rarefaction-visualise"><i class="fa fa-check"></i><b>12.2</b> Rarefaction: visualise</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#rarefaction-considerations"><i class="fa fa-check"></i><b>12.2.1</b> Rarefaction considerations</a></li>
<li class="chapter" data-level="12.2.2" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#rarefaction-mcqs"><i class="fa fa-check"></i><b>12.2.2</b> Rarefaction: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html#rarefaction-summary"><i class="fa fa-check"></i><b>12.3</b> Rarefaction: summary</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Diversity_analysis.html"><a href="13-Diversity_analysis.html"><i class="fa fa-check"></i><b>13</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Diversity_analysis.html"><a href="13-Diversity_analysis.html#diversity-run"><i class="fa fa-check"></i><b>13.1</b> Diversity: run</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="14-Alpha_diversity.html"><a href="14-Alpha_diversity.html"><i class="fa fa-check"></i><b>14</b> Alpha diveristy statistical analysis</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Alpha_diversity.html"><a href="14-Alpha_diversity.html#alpha-diversity-run"><i class="fa fa-check"></i><b>14.1</b> Alpha diversity: run</a></li>
<li class="chapter" data-level="14.2" data-path="14-Alpha_diversity.html"><a href="14-Alpha_diversity.html#alpha-diversity-visualise"><i class="fa fa-check"></i><b>14.2</b> Alpha diversity: visualise</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="14-Alpha_diversity.html"><a href="14-Alpha_diversity.html#alpha-diversity-mcqs"><i class="fa fa-check"></i><b>14.2.1</b> Alpha diversity: MCQs</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="14-Alpha_diversity.html"><a href="14-Alpha_diversity.html#alpha-diversity-summary"><i class="fa fa-check"></i><b>14.3</b> Alpha diversity: summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html"><i class="fa fa-check"></i><b>15</b> Beta diversity statistical analysis</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html#beta-diveristy-pcoa"><i class="fa fa-check"></i><b>15.1</b> Beta diveristy: PCoA</a></li>
<li class="chapter" data-level="15.2" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html#beta-diversity-run"><i class="fa fa-check"></i><b>15.2</b> Beta diversity: run</a></li>
<li class="chapter" data-level="15.3" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html#beta-diversity-visualise"><i class="fa fa-check"></i><b>15.3</b> Beta diversity: visualise</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html#beta-diversity-mcqs"><i class="fa fa-check"></i><b>15.3.1</b> Beta diversity: MCQs</a></li>
<li class="chapter" data-level="15.3.2" data-path="15-Beta_diversity_analysis.html"><a href="15-Beta_diversity_analysis.html#beta-diversity-summary"><i class="fa fa-check"></i><b>15.3.2</b> Beta diversity: summary</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html"><i class="fa fa-check"></i><b>16</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#taxa-collapse"><i class="fa fa-check"></i><b>16.1</b> Taxa collapse</a></li>
<li class="chapter" data-level="16.2" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancom-bc-analysis"><i class="fa fa-check"></i><b>16.2</b> ANCOM-BC analysis</a></li>
<li class="chapter" data-level="16.3" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancombc-tabulate"><i class="fa fa-check"></i><b>16.3</b> ANCOMBC tabulate</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancombc-tabulate-visualise"><i class="fa fa-check"></i><b>16.3.1</b> ANCOMBC tabulate: visualise</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancombc-bar-plots"><i class="fa fa-check"></i><b>16.4</b> ANCOMBC bar plots</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancom-bc-bar-plots-visualise"><i class="fa fa-check"></i><b>16.4.1</b> ANCOM-BC bar plots: visualise</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancom-bc-mcqs"><i class="fa fa-check"></i><b>16.5</b> ANCOM-BC: MCQs</a></li>
<li class="chapter" data-level="16.6" data-path="16-Differential_abundance.html"><a href="16-Differential_abundance.html#ancom-bc-summary"><i class="fa fa-check"></i><b>16.6</b> ANCOM-BC: summary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Final_consideration.html"><a href="17-Final_consideration.html"><i class="fa fa-check"></i><b>17</b> Final consideration</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="18-Appendix.html"><a href="18-Appendix.html"><i class="fa fa-check"></i><b>A</b> Resources</a>
<ul>
<li class="chapter" data-level="A.1" data-path="18-Appendix.html"><a href="18-Appendix.html#suggested-readings"><i class="fa fa-check"></i><b>A.1</b> Suggested readings</a></li>
<li class="chapter" data-level="A.2" data-path="18-Appendix.html"><a href="18-Appendix.html#other-useful-resources"><i class="fa fa-check"></i><b>A.2</b> Other useful resources</a></li>
<li class="chapter" data-level="A.3" data-path="18-Appendix.html"><a href="18-Appendix.html#references"><i class="fa fa-check"></i><b>A.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="18-Appendix.html"><a href="18-Appendix.html#mamba-installs"><i class="fa fa-check"></i><b>B</b> Mamba installs</a>
<ul>
<li class="chapter" data-level="B.1" data-path="18-Appendix.html"><a href="18-Appendix.html#mamba_install"><i class="fa fa-check"></i><b>B.1</b> Mamba installation and environment</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="18-Appendix.html"><a href="18-Appendix.html#diversity-metrics"><i class="fa fa-check"></i><b>C</b> Diversity metrics</a>
<ul>
<li class="chapter" data-level="C.1" data-path="18-Appendix.html"><a href="18-Appendix.html#alphamet"><i class="fa fa-check"></i><b>C.1</b> Alpha diversity metrics</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="18-Appendix.html"><a href="18-Appendix.html#obvs"><i class="fa fa-check"></i><b>C.1.1</b> Observed features</a></li>
<li class="chapter" data-level="C.1.2" data-path="18-Appendix.html"><a href="18-Appendix.html#evenness"><i class="fa fa-check"></i><b>C.1.2</b> Evenness</a></li>
<li class="chapter" data-level="C.1.3" data-path="18-Appendix.html"><a href="18-Appendix.html#faith"><i class="fa fa-check"></i><b>C.1.3</b> Faith's PD (phylogenetic diversity)</a></li>
<li class="chapter" data-level="C.1.4" data-path="18-Appendix.html"><a href="18-Appendix.html#simpson"><i class="fa fa-check"></i><b>C.1.4</b> Simpson</a></li>
<li class="chapter" data-level="C.1.5" data-path="18-Appendix.html"><a href="18-Appendix.html#simpsone"><i class="fa fa-check"></i><b>C.1.5</b> Simpson evenness measure (simpson_e)</a></li>
<li class="chapter" data-level="C.1.6" data-path="18-Appendix.html"><a href="18-Appendix.html#shannon"><i class="fa fa-check"></i><b>C.1.6</b> Shannon</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="18-Appendix.html"><a href="18-Appendix.html#beta-diversity-metrics"><i class="fa fa-check"></i><b>C.2</b> Beta diversity metrics</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="18-Appendix.html"><a href="18-Appendix.html#unifrac_explanation"><i class="fa fa-check"></i><b>C.2.1</b> Weighted and unweighted UniFrac distances</a></li>
<li class="chapter" data-level="C.2.2" data-path="18-Appendix.html"><a href="18-Appendix.html#bray_explanation"><i class="fa fa-check"></i><b>C.2.2</b> Bray-Curtis</a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="18-Appendix.html"><a href="18-Appendix.html#diversity-resources"><i class="fa fa-check"></i><b>C.3</b> Diversity Resources</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial 16S metabarcoding</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="taxa" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> ASV taxonomic assignment<a href="10-Taxonomic_assignment.html#taxa" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/classification.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Now that we have defined the sequence variants in the samples, this chapter will show you how to carry out taxonomic classification using <a href="https://www.nature.com/articles/s41587-023-01845-1">Greengenes2</a>. More information on its use can be found in the following <code>QIIME2</code> forum post from one of the Greengenes2 developers: <a href="https://forum.qiime2.org/t/introducing-greengenes2-2022-10/25291" class="uri">https://forum.qiime2.org/t/introducing-greengenes2-2022-10/25291</a></p>
<div id="greengenes2-database" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Greengenes2 database<a href="10-Taxonomic_assignment.html#greengenes2-database" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/green_jeans.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Greengenes2 is a database containing quality-controlled microbial sequence data. It consists of a single massive reference tree that contains taxonomic and sequence information. It was created by combining:</p>
<ul>
<li><strong>Web of Life 2 (WoL2)</strong>: A microbe base phylogenomic tree.
<ul>
<li><strong>15,953</strong> bacterial and archaeal genomes.</li>
<li>Accurate phylogenomic tree reconstructed by summarising <strong>380</strong> global marker genes.</li>
<li>Produced by new workflow <a href="https://github.com/balabanmetin/uDance">uDance</a>.</li>
</ul></li>
<li><strong>16s rRNA sequences</strong> from different sources.
<ul>
<li><strong>18,356</strong> full-length 16S rRNA sequences from the <a href="https://www.nature.com/articles/s41587-020-0501-8">Living Tree Project (LTP)</a> January 2022 release.</li>
<li><strong>1,725,274</strong> near-complete 16S rRNA genes from <a href="https://www.annualreviews.org/doi/10.1146/annurev-ecolsys-012121-095340">Karst et al 2021</a> and the <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-16-S10-S1">Earth Microbiome Project 500 (EMP500)</a>.</li>
<li>All full-length 16S rRNA sequences from <a href="https://gtdb.ecogenomic.org/stats/r207">GTDB r207</a>.</li>
</ul></li>
<li><strong>23,113,447</strong> short V4 16S rRNA ASVs denoised with Deblur v1.1.0 from <a href="https://qiita.ucsd.edu/">Qiita</a></li>
<li>Mitochondria and chloroplast 16S rRNA from <a href="https://www.arb-silva.de/documentation/release-138/">SILVA v138</a> using <a href="https://github.com/yueyujiang/DEPP">deep-learning-enabled phylogenetic placement (DEPP)</a>.</li>
</ul>
<p>This produces a reference tree with the following traits:</p>
<ul>
<li>Genome-based relationships (WoL2) were fixed whilst relationships between full-length 16S rRNA sequences were inferred.</li>
<li>Short fragments were then added independlty from each each with the genome and full-length 16S rRNA relationships fixed.</li>
<li>The final tree covered <strong>21,074,442</strong> sequences from <strong>31</strong> different <a href="https://earthmicrobiome.org/protocols-and-standards/empo/">EMP Ontology 3 (EMPO3)</a> environments.</li>
<li><strong>46.5%</strong> of species-level leaves were covered by a complete genome.</li>
<li>Phylogenies are decorated with taxonomic labels.
<ul>
<li>Taxonomy was based on GTDB r207, combined with the LTP January 2022 release.</li>
<li>Taxonomy is planned to be updated every 6 months with the latest versions of GTDB &amp; LTP.</li>
</ul></li>
</ul>
<p><a href="https://www.nature.com/articles/s41587-023-01845-1/figures/1">Greengenes phylogenetic tree</a>
<img src="figures/greengenes2_phylogeny_tree.png" width="60%" style="display: block; margin: auto;" /></p>
<p>Greengenes2 can be used on shotgun metagenomic data. This can allow you to effectively compare taxonomic data from 16S and Shotgun metagenomic datasets. <a href="https://forum.qiime2.org/t/introducing-greengenes2-2022-10/25291#if-you-have-shotgun-metagenomic-data-6">Tutorial on clasisifying shotgun metageonimc data with Greengenes2</a>.</p>
<p>For your future analysis you can download the databases files from their <a href="http://ftp.microbio.me/greengenes_release/2022.10/">ftp site</a>. However it is also good to check <a href="http://ftp.microbio.me/greengenes_release/">this site page</a> to see if they have a more recent version when you are analysing your data.</p>
<div class="webex-solution">
<button>
Download database files in linux terminal
</button>
<p>You can download the 2 files you need for Greengenes2 taxonomic classification with the following commands.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb14-1"><a href="10-Taxonomic_assignment.html#cb14-1" tabindex="-1"></a><span class="fu">wget</span> http://ftp.microbio.me/greengenes_release/2022.10/2022.10.backbone.full-length.fna.qza</span>
<span id="cb14-2"><a href="10-Taxonomic_assignment.html#cb14-2" tabindex="-1"></a><span class="fu">wget</span> http://ftp.microbio.me/greengenes_release/2022.10/2022.10.backbone.tax.qza</span></code></pre></div>
<p>Remember to first check if 2022.10 is the most recent version before downloading and using it for your own research.</p>
</div>
</div>
<div id="extract-16s-region-of-interest" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Extract 16s region of interest<a href="10-Taxonomic_assignment.html#extract-16s-region-of-interest" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/scissor_cut.png" width="15%" style="display: block; margin: auto;" /></p>
<p>A classifier training step is required to reach optimum assignment performance.</p>
<p>Prior to training we will extract our 16S region of interest. This is carried out by using the amplicon primer sequences. These are the same we used for <a href="08-Trim_PCR_sequences.html#pcrtrim">removing the PCR primer</a>.</p>
<p><strong>Note</strong>: Do not run the below command for this tutorial as it will take a long time. See <code>cp</code> command further down.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb15-1"><a href="10-Taxonomic_assignment.html#cb15-1" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier extract-reads <span class="dt">\</span></span>
<span id="cb15-2"><a href="10-Taxonomic_assignment.html#cb15-2" tabindex="-1"></a>--i-sequences /pub14/tea/nsc206/NEOF/16s_workshop/greengenes2/2022.10.backbone.full-length.fna.qza <span class="dt">\</span></span>
<span id="cb15-3"><a href="10-Taxonomic_assignment.html#cb15-3" tabindex="-1"></a>--p-f-primer NNNNNGTGCCAGCMGCCGCGGTAA <span class="dt">\</span></span>
<span id="cb15-4"><a href="10-Taxonomic_assignment.html#cb15-4" tabindex="-1"></a>--p-r-primer GGACTACHVGGGTWTCTAAT <span class="dt">\</span></span>
<span id="cb15-5"><a href="10-Taxonomic_assignment.html#cb15-5" tabindex="-1"></a>--p-trunc-len 250 <span class="dt">\</span></span>
<span id="cb15-6"><a href="10-Taxonomic_assignment.html#cb15-6" tabindex="-1"></a>--o-reads gg2-v4.ref.seqs.qza <span class="dt">\</span></span>
<span id="cb15-7"><a href="10-Taxonomic_assignment.html#cb15-7" tabindex="-1"></a>--p-n-jobs 12</span></code></pre></div>
<div class="webex-solution">
<button>
Silva 16S rRNA database
</button>
<p>Another popular 16S rRNA database is <a href="https://www.arb-silva.de/">Silva</a>.</p>
<p>You can download the required files for taxonomic classification from <a href="https://docs.qiime2.org/2023.5/data-resources/#silva-16s-18s-rrna">here</a>.</p>
</div>
<div id="parameters-6" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="10-Taxonomic_assignment.html#parameters-6" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/parameter_red.png" width="10%" style="display: block; margin: auto;" /></p>
<ul>
<li><strong><code>--i-sequences</code></strong>: Input artifact containing the feature sequence data.
<ul>
<li>In this case we are using the full length 16S rRNA sequences that are used within the Greenegenes2 phylogenomic tree.</li>
<li>This was downloaded from <a href="http://ftp.microbio.me/greengenes_release/2022.10/2022.10.backbone.full-length.fna.qza" class="uri">http://ftp.microbio.me/greengenes_release/2022.10/2022.10.backbone.full-length.fna.qza</a>.</li>
</ul></li>
<li><strong><code>--p-front-f</code></strong>: Sequence for the forward primer (<code>f</code>). The sequence is ligated to the 5 prime end (<code>front</code>), this is the normal.</li>
<li><strong><code>--p-front-r</code></strong>: Sequence for the reverse primer (<code>r``). The sequence is ligated to the 5 prime end (</code>front`), this is the normal.</li>
<li><strong><code>--p-trunc-len</code></strong>: This cuts the forward and left reads we are creating to the specified truncation length.
<ul>
<li>We are using <code>250</code> as this was the initial length of our Illumina reads.</li>
</ul></li>
<li><strong><code>--o-reads</code></strong>: The output read artifact containing our extracted 16S region as paired reads. This is needed for classifier training.</li>
<li><strong><code>--p-n-jobs</code></strong>: Number of separate processes to use.</li>
</ul>
<p><a href="https://docs.qiime2.org/2023.5/plugins/available/feature-classifier/extract-reads/">QIIME2 docs on <code>qiime feature-classifier extract-reads</code></a></p>
</div>
<div id="copy-premade-results" class="section level4 unnumbered hasAnchor">
<h4>Copy premade results<a href="10-Taxonomic_assignment.html#copy-premade-results" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Given that the above command may require more than 14h to finish, we wonâ€™t be able to run it on this occasion. Copy over the pre-made results with the following command:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb16-1"><a href="10-Taxonomic_assignment.html#cb16-1" tabindex="-1"></a><span class="fu">cp</span> /pub14/tea/nsc206/NEOF/16s_workshop/greengenes2/gg2-v4.ref.seqs.qza .</span></code></pre></div>
</div>
</div>
<div id="classifier-training" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Classifier training<a href="10-Taxonomic_assignment.html#classifier-training" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/train.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Now we train the classifier on this set of extracted reads (approx. 25 mins).</p>
<p>This uses Naive Bayes which is a very complicated algorithm. Thankfully we don't need to know how it works so we won't explain it here.</p>
<p><strong>Note</strong>: You can run the code or copy a pre-made file.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb17-1"><a href="10-Taxonomic_assignment.html#cb17-1" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier fit-classifier-naive-bayes <span class="dt">\</span></span>
<span id="cb17-2"><a href="10-Taxonomic_assignment.html#cb17-2" tabindex="-1"></a>--i-reference-reads gg2-v4.ref.seqs.qza <span class="dt">\</span></span>
<span id="cb17-3"><a href="10-Taxonomic_assignment.html#cb17-3" tabindex="-1"></a>--i-reference-taxonomy /pub14/tea/nsc206/NEOF/16s_workshop/greengenes2/2022.10.backbone.tax.qza <span class="dt">\</span></span>
<span id="cb17-4"><a href="10-Taxonomic_assignment.html#cb17-4" tabindex="-1"></a>--o-classifier gg2-v4.classifier.trained.qza</span></code></pre></div>
<p>Copy pre-made file if you didn't run the code above.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb18-1"><a href="10-Taxonomic_assignment.html#cb18-1" tabindex="-1"></a><span class="fu">cp</span> /pub14/tea/nsc006/NEOF/16s_workshop/greengenes2/gg2-v4.classifier.trained.qza .</span></code></pre></div>
<div id="parameters-7" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="10-Taxonomic_assignment.html#parameters-7" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/parameter_green.png" width="10%" style="display: block; margin: auto;" /></p>
<ul>
<li><strong><code>--i-reference-reads</code></strong>: Input artifact containing the extracted 16S region as paired reads.</li>
<li><strong><code>--i-reference-taxonomy</code></strong>: Input artifact containing the taxonomy information of the features.
<ul>
<li>This was downloaded from: <a href="http://ftp.microbio.me/greengenes_release/2022.10/2022.10.backbone.tax.qza" class="uri">http://ftp.microbio.me/greengenes_release/2022.10/2022.10.backbone.tax.qza</a>.</li>
</ul></li>
<li><strong><code>--o-classifier</code></strong>: Output artifact containing the trained classifier. This will be used for feature (ASV) taxonomy classification.</li>
</ul>
<p><a href="https://docs.qiime2.org/2023.5/plugins/available/feature-classifier/fit-classifier-naive-bayes/">QIIME2 docs on <code>qiime feature-classifier fit-classifier-naive-bayes</code></a></p>
</div>
</div>
<div id="taxonomy-assignment" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Taxonomy assignment<a href="10-Taxonomic_assignment.html#taxonomy-assignment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/classification.png" width="15%" style="display: block; margin: auto;" /></p>
<p>With our trained classifier we can assign taxonomic assignments to the ASV representative sequences.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb19-1"><a href="10-Taxonomic_assignment.html#cb19-1" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier classify-sklearn <span class="dt">\</span></span>
<span id="cb19-2"><a href="10-Taxonomic_assignment.html#cb19-2" tabindex="-1"></a>--i-classifier gg2-v4.classifier.trained.qza <span class="dt">\</span></span>
<span id="cb19-3"><a href="10-Taxonomic_assignment.html#cb19-3" tabindex="-1"></a>--p-confidence 0.7 <span class="dt">\</span></span>
<span id="cb19-4"><a href="10-Taxonomic_assignment.html#cb19-4" tabindex="-1"></a>--i-reads rep-seqs-dada2.qza <span class="dt">\</span></span>
<span id="cb19-5"><a href="10-Taxonomic_assignment.html#cb19-5" tabindex="-1"></a>--o-classification taxonomy.sklearn.qza <span class="dt">\</span></span>
<span id="cb19-6"><a href="10-Taxonomic_assignment.html#cb19-6" tabindex="-1"></a>--p-n-jobs 12</span></code></pre></div>
<div id="parameters-8" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="10-Taxonomic_assignment.html#parameters-8" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/parameter_orange.png" width="10%" style="display: block; margin: auto;" /></p>
<ul>
<li><strong><code>--i-classifier</code></strong>: Input artifact containing the trained classifier.</li>
<li><strong><code>--p-confidence</code></strong>: Confidence threshold for limiting taxonomic depth.
<ul>
<li>We are using 0.7 as it is the default.</li>
<li>Each taxonomic level must reach a confidence level of &lt;= 0.7 to be classified.</li>
<li>This could cause an ASV to be classified to Genus level but not species level as the species level assignment had a confidence of &lt;0.7 whilst the confidence of the genus and higher assignments was &gt;=0.7.</li>
</ul></li>
<li><strong><code>--i-reads</code></strong>: Input artifact containing the representative sequences of the features (ASVs) to be classified.</li>
<li><strong><code>--o-classification</code></strong>: Output artifact containing the taxonomic assignments of the features (ASVs).</li>
<li><strong><code>--p-n-jobs</code></strong>: Number of separate processes to use.</li>
</ul>
<p><a href="https://docs.qiime2.org/2023.5/plugins/available/feature-classifier/classify-sklearn/">QIIME2 docs on <code>qiime feature-classifier classify-sklearn</code></a></p>
</div>
</div>
<div id="taxabar" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> Taxonomy bar chart production<a href="10-Taxonomic_assignment.html#taxabar" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/bar_charts.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Create a taxonomy-abundance bar-chart visualisation artifact:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash Rchunk"><code class="sourceCode bash"><span id="cb20-1"><a href="10-Taxonomic_assignment.html#cb20-1" tabindex="-1"></a><span class="ex">qiime</span> taxa barplot <span class="dt">\</span></span>
<span id="cb20-2"><a href="10-Taxonomic_assignment.html#cb20-2" tabindex="-1"></a>--i-table table-dada2.qza <span class="dt">\</span></span>
<span id="cb20-3"><a href="10-Taxonomic_assignment.html#cb20-3" tabindex="-1"></a>--i-taxonomy taxonomy.sklearn.qza <span class="dt">\</span></span>
<span id="cb20-4"><a href="10-Taxonomic_assignment.html#cb20-4" tabindex="-1"></a>--m-metadata-file metadata.file.txt <span class="dt">\</span></span>
<span id="cb20-5"><a href="10-Taxonomic_assignment.html#cb20-5" tabindex="-1"></a>--o-visualization taxa-bar-plots.dada2.qzv</span></code></pre></div>
<div id="parameters-9" class="section level4 unnumbered hasAnchor">
<h4>Parameters<a href="10-Taxonomic_assignment.html#parameters-9" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="figures/parameter_blue.png" width="10%" style="display: block; margin: auto;" /></p>
<ul>
<li><strong><code>--i-table</code></strong>: Input artifact containing the ASV abundance table.</li>
<li><strong><code>--i-taxonomy</code></strong>: Input artifact containing the taxonomic classifications of the ASVs.</li>
<li><strong><code>--m-metadata-file</code></strong>: <a href="05-Data.html#metadata.file.txt">Metadata file</a> for our samples.</li>
<li><strong><code>--o-visualization</code></strong>: Output visualisation file containing an interactive taxonomy-abundance bar-chart.</li>
</ul>
<p>The visualisation consists of an interactive taxonomy bar chart. You can specify the taxonomic level to look at, as well as a few sorting and colouring options. Experiment with it and then continue to the MCQ section.</p>
<p><a href="https://docs.qiime2.org/2023.5/plugins/available/taxa/barplot/">QIIME2 docs on <code>qiime taxa barplot</code></a></p>
</div>
</div>
<div id="taxonomy-mcqs" class="section level2 hasAnchor" number="10.6">
<h2><span class="header-section-number">10.6</span> Taxonomy: MCQs<a href="10-Taxonomic_assignment.html#taxonomy-mcqs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/question_bubble_blue.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Attempt the below MCQs.</p>
<ol style="list-style-type: decimal">
<li>At taxonomic level 5, what is the taxon with the highest relative abundance across all the samples? (Note: taxa are ordered form most to least abundant in the legend)
<div id="radio_FQACVSUSWQ" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_FQACVSUSWQ" value="answer"></input> <span><strong>Moraxellaceae</strong></span></label><label><input type="radio" autocomplete="off" name="radio_FQACVSUSWQ" value=""></input> <span><strong>Mycobacteriaceae</strong></span></label><label><input type="radio" autocomplete="off" name="radio_FQACVSUSWQ" value=""></input> <span><strong>Pseudomonadaceae</strong></span></label>
</div></li>
<li>At taxonomic level 5, what is the most abundant taxon for sample 1TE?
<div id="radio_VMJQHKQIOA" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_VMJQHKQIOA" value=""></input> <span><strong>Moraxellaceae</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VMJQHKQIOA" value="answer"></input> <span><strong>Mycobacteriaceae</strong></span></label><label><input type="radio" autocomplete="off" name="radio_VMJQHKQIOA" value=""></input> <span><strong>Pseudomonadaceae</strong></span></label>
</div></li>
</ol>
<p><strong>Note</strong>: The "Sort Samples By" drop down will be useful for the following questions. Additionally, switching the "Ascending" (low to high, right sample has the highest relative abundance) drop down to "Descending" may be useful.</p>
<ol start="3" style="list-style-type: decimal">
<li>Which sample has the highest relative abundance of the class Actinomycetia (Taxonomic level 3)?
<div id="radio_OHMQVRDWTO" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_OHMQVRDWTO" value=""></input> <span><strong>1K2E</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OHMQVRDWTO" value="answer"></input> <span><strong>1TM</strong></span></label><label><input type="radio" autocomplete="off" name="radio_OHMQVRDWTO" value=""></input> <span><strong>3TM</strong></span></label>
</div></li>
<li>Which sample has the lowest relative abundance of the genus Corynebacterium (Taxonomic level 6)?
<div id="radio_ZRXCWAYOQN" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_ZRXCWAYOQN" value="answer"></input> <span><strong>1K2E</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ZRXCWAYOQN" value=""></input> <span><strong>1TM</strong></span></label><label><input type="radio" autocomplete="off" name="radio_ZRXCWAYOQN" value=""></input> <span><strong>3TM</strong></span></label>
</div></li>
<li>Which sample has the fourth highest relative abundance of the phylum Bacteroidota (Taxonomic level 2)?
<div id="radio_QKMNAPXFMF" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_QKMNAPXFMF" value=""></input> <span><strong>1K2E</strong></span></label><label><input type="radio" autocomplete="off" name="radio_QKMNAPXFMF" value=""></input> <span><strong>1TM</strong></span></label><label><input type="radio" autocomplete="off" name="radio_QKMNAPXFMF" value="answer"></input> <span><strong>3TM</strong></span></label>
</div></li>
</ol>
</div>
<div id="taxonomy-summary" class="section level2 hasAnchor" number="10.7">
<h2><span class="header-section-number">10.7</span> Taxonomy: summary<a href="10-Taxonomic_assignment.html#taxonomy-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/sum_blue.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We have trimmed and denoised our data, producing ASVs. This has been followed by assigning taxonomic classifications to the ASVs.</p>
<p>In your future analysis if you have sequenced any samples with known composition (e.g. mock community), now is the time to see if they behave as expected. Do they contain what you expect? You may want to change some steps and/or settings of what you have done so far if the composition does not match what you expect. For example, change the de-noising tool or apply more quality filters to further improve the quality of the sequence under examination.</p>
<p>We are close to carrying out some biodiversity analysis. Before this we will produce a phylogenetic tree in the next chapter.</p>
<p><a href="https://docs.qiime2.org/2023.5/tutorials/feature-classifier/">QIIME2 documents on training feature classifiers</a></p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="09-ASV_identification.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="11-Phylogentic_tree.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
