<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>11 Using Conditions to Modify DNA Data | Python for Bioinformatics</title>
  <meta name="description" content="NEOF book for the Python for Bioinformatics workshop" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="11 Using Conditions to Modify DNA Data | Python for Bioinformatics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/images/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Python for Bioinformatics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="11 Using Conditions to Modify DNA Data | Python for Bioinformatics" />
  
  <meta name="twitter:description" content="NEOF book for the Python for Bioinformatics workshop" />
  <meta name="twitter:image" content="/images/NEOF.png" />




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="09_DNA_sequence_data.html"/>
<link rel="next" href="11_reading_and_writing.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="images/neof_small_logo.png"></a></li>
<li><a>Python for Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="01-intro.html"><a href="01-intro.html"><i class="fa fa-check"></i><b>2</b> Python for bioinformatics</a>
<ul>
<li class="chapter" data-level="2.1" data-path="01-intro.html"><a href="01-intro.html#materials-covered-in-the-course"><i class="fa fa-check"></i><b>2.1</b> Materials Covered in the Course</a></li>
<li class="chapter" data-level="2.2" data-path="01-intro.html"><a href="01-intro.html#inputting-commands"><i class="fa fa-check"></i><b>2.2</b> Inputting Commands</a></li>
<li class="chapter" data-level="2.3" data-path="01-intro.html"><a href="01-intro.html#pseudocode"><i class="fa fa-check"></i><b>2.3</b> Pseudocode</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="02_cluster_introduction.html"><a href="02_cluster_introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="02_cluster_introduction.html"><a href="02_cluster_introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon Instructions</a></li>
<li class="chapter" data-level="3.2" data-path="02_cluster_introduction.html"><a href="02_cluster_introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="02_cluster_introduction.html"><a href="02_cluster_introduction.html#load-workshop-data"><i class="fa fa-check"></i><b>3.3</b> Load workshop data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="03_basic_scripting.html"><a href="03_basic_scripting.html"><i class="fa fa-check"></i><b>4</b> Scripting in Python</a>
<ul>
<li class="chapter" data-level="4.1" data-path="03_basic_scripting.html"><a href="03_basic_scripting.html#introduction-to-commands-and-scripts"><i class="fa fa-check"></i><b>4.1</b> Introduction to commands and scripts</a></li>
<li class="chapter" data-level="4.2" data-path="03_basic_scripting.html"><a href="03_basic_scripting.html#issuing-commands-directly-to-the-interpretator"><i class="fa fa-check"></i><b>4.2</b> Issuing commands directly to the interpretator</a></li>
<li class="chapter" data-level="4.3" data-path="03_basic_scripting.html"><a href="03_basic_scripting.html#a-note-on-indentation"><i class="fa fa-check"></i><b>4.3</b> A note on indentation</a></li>
<li class="chapter" data-level="4.4" data-path="03_basic_scripting.html"><a href="03_basic_scripting.html#tabs-vs-spaces"><i class="fa fa-check"></i><b>4.4</b> Tabs vs spaces</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html"><i class="fa fa-check"></i><b>5</b> Python Fundamentals</a>
<ul>
<li class="chapter" data-level="5.1" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#operators"><i class="fa fa-check"></i><b>5.1</b> Operators</a></li>
<li class="chapter" data-level="5.2" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#comments"><i class="fa fa-check"></i><b>5.2</b> Comments</a></li>
<li class="chapter" data-level="5.3" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#variables"><i class="fa fa-check"></i><b>5.3</b> Variables</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#strings"><i class="fa fa-check"></i><b>5.3.1</b> Strings</a></li>
<li class="chapter" data-level="5.3.2" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#integers"><i class="fa fa-check"></i><b>5.3.2</b> Integers</a></li>
<li class="chapter" data-level="5.3.3" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#floats-floating-point-numbers"><i class="fa fa-check"></i><b>5.3.3</b> Floats (floating point numbers)</a></li>
<li class="chapter" data-level="5.3.4" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#booleans"><i class="fa fa-check"></i><b>5.3.4</b> Booleans</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#classes-of-variable"><i class="fa fa-check"></i><b>5.4</b> Classes of variable</a></li>
<li class="chapter" data-level="5.5" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#transforming-variables"><i class="fa fa-check"></i><b>5.5</b> Transforming variables</a></li>
<li class="chapter" data-level="5.6" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#functions-and-methods"><i class="fa fa-check"></i><b>5.6</b> Functions and methods</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#running-a-function"><i class="fa fa-check"></i><b>5.6.1</b> Running a function</a></li>
<li class="chapter" data-level="5.6.2" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#running-a-method"><i class="fa fa-check"></i><b>5.6.2</b> Running a method</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="04_python_fundamentals.html"><a href="04_python_fundamentals.html#recap"><i class="fa fa-check"></i><b>5.7</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="05_data_structures.html"><a href="05_data_structures.html"><i class="fa fa-check"></i><b>6</b> Data Structures</a>
<ul>
<li class="chapter" data-level="6.1" data-path="05_data_structures.html"><a href="05_data_structures.html#lists"><i class="fa fa-check"></i><b>6.1</b> Lists</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="05_data_structures.html"><a href="05_data_structures.html#add-numeric-data-to-a-list"><i class="fa fa-check"></i><b>6.1.1</b> Add numeric data to a list</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="05_data_structures.html"><a href="05_data_structures.html#subsetting-data"><i class="fa fa-check"></i><b>6.2</b> Subsetting data</a></li>
<li class="chapter" data-level="6.3" data-path="05_data_structures.html"><a href="05_data_structures.html#set"><i class="fa fa-check"></i><b>6.3</b> Set</a></li>
<li class="chapter" data-level="6.4" data-path="05_data_structures.html"><a href="05_data_structures.html#transform-a-list-to-a-set"><i class="fa fa-check"></i><b>6.4</b> Transform a list to a set</a></li>
<li class="chapter" data-level="6.5" data-path="05_data_structures.html"><a href="05_data_structures.html#dictionary"><i class="fa fa-check"></i><b>6.5</b> Dictionary</a></li>
<li class="chapter" data-level="6.6" data-path="05_data_structures.html"><a href="05_data_structures.html#recap-1"><i class="fa fa-check"></i><b>6.6</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="06_conditions.html"><a href="06_conditions.html"><i class="fa fa-check"></i><b>7</b> Python Conditions: 'if' and 'else' statements</a>
<ul>
<li class="chapter" data-level="7.1" data-path="06_conditions.html"><a href="06_conditions.html#if-statements"><i class="fa fa-check"></i><b>7.1</b> If statements</a></li>
<li class="chapter" data-level="7.2" data-path="06_conditions.html"><a href="06_conditions.html#if-else-statements"><i class="fa fa-check"></i><b>7.2</b> If else statements</a></li>
<li class="chapter" data-level="7.3" data-path="06_conditions.html"><a href="06_conditions.html#recap-2"><i class="fa fa-check"></i><b>7.3</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="07_scripting_in_python.html"><a href="07_scripting_in_python.html"><i class="fa fa-check"></i><b>8</b> Scripting in Python</a>
<ul>
<li class="chapter" data-level="8.1" data-path="07_scripting_in_python.html"><a href="07_scripting_in_python.html#open-a-plain-text-editor"><i class="fa fa-check"></i><b>8.1</b> Open a plain text editor</a></li>
<li class="chapter" data-level="8.2" data-path="07_scripting_in_python.html"><a href="07_scripting_in_python.html#running-a-script-from-the-terminal"><i class="fa fa-check"></i><b>8.2</b> Running a script from the terminal</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="08_loops.html"><a href="08_loops.html"><i class="fa fa-check"></i><b>9</b> Loops</a>
<ul>
<li class="chapter" data-level="9.1" data-path="08_loops.html"><a href="08_loops.html#data-structure-recap"><i class="fa fa-check"></i><b>9.1</b> Data Structure Recap</a></li>
<li class="chapter" data-level="9.2" data-path="08_loops.html"><a href="08_loops.html#introduction-to-loops"><i class="fa fa-check"></i><b>9.2</b> Introduction To Loops</a></li>
<li class="chapter" data-level="9.3" data-path="08_loops.html"><a href="08_loops.html#for-loops"><i class="fa fa-check"></i><b>9.3</b> For Loops</a></li>
<li class="chapter" data-level="9.4" data-path="08_loops.html"><a href="08_loops.html#for-loop-example"><i class="fa fa-check"></i><b>9.4</b> For Loop Example</a></li>
<li class="chapter" data-level="9.5" data-path="08_loops.html"><a href="08_loops.html#use-a-for-loop-to-complement-a-dna-sequence"><i class="fa fa-check"></i><b>9.5</b> Use a for loop to complement a DNA sequence</a></li>
<li class="chapter" data-level="9.6" data-path="08_loops.html"><a href="08_loops.html#use-a-for-loop-to-reverse-complement-a-dna-sequence"><i class="fa fa-check"></i><b>9.6</b> Use a for loop to reverse-complement a DNA sequence</a></li>
<li class="chapter" data-level="9.7" data-path="08_loops.html"><a href="08_loops.html#while-loops"><i class="fa fa-check"></i><b>9.7</b> While loops</a></li>
<li class="chapter" data-level="9.8" data-path="08_loops.html"><a href="08_loops.html#further-control-statements"><i class="fa fa-check"></i><b>9.8</b> Further control statements</a></li>
<li class="chapter" data-level="9.9" data-path="08_loops.html"><a href="08_loops.html#recap-3"><i class="fa fa-check"></i><b>9.9</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="09_DNA_sequence_data.html"><a href="09_DNA_sequence_data.html"><i class="fa fa-check"></i><b>10</b> Reading sequence data</a>
<ul>
<li class="chapter" data-level="10.1" data-path="09_DNA_sequence_data.html"><a href="09_DNA_sequence_data.html#seqio"><i class="fa fa-check"></i><b>10.1</b> SeqIO</a></li>
<li class="chapter" data-level="10.2" data-path="09_DNA_sequence_data.html"><a href="09_DNA_sequence_data.html#reading-a-fastq-dna-data-file-into-python"><i class="fa fa-check"></i><b>10.2</b> Reading a FASTQ DNA data file into Python</a></li>
<li class="chapter" data-level="10.3" data-path="09_DNA_sequence_data.html"><a href="09_DNA_sequence_data.html#the-fastq-format"><i class="fa fa-check"></i><b>10.3</b> The FASTQ format</a></li>
<li class="chapter" data-level="10.4" data-path="09_DNA_sequence_data.html"><a href="09_DNA_sequence_data.html#read-a-fastq-file-into-python"><i class="fa fa-check"></i><b>10.4</b> Read a FASTQ file into Python</a></li>
<li class="chapter" data-level="10.5" data-path="09_DNA_sequence_data.html"><a href="09_DNA_sequence_data.html#setting-up-a-script-to-use-biopython"><i class="fa fa-check"></i><b>10.5</b> Setting up a script to use BioPython</a></li>
<li class="chapter" data-level="10.6" data-path="09_DNA_sequence_data.html"><a href="09_DNA_sequence_data.html#read-a-fie-of-sequence-data"><i class="fa fa-check"></i><b>10.6</b> Read a fie of sequence data</a></li>
<li class="chapter" data-level="10.7" data-path="09_DNA_sequence_data.html"><a href="09_DNA_sequence_data.html#modifying-reads"><i class="fa fa-check"></i><b>10.7</b> Modifying reads</a></li>
<li class="chapter" data-level="10.8" data-path="09_DNA_sequence_data.html"><a href="09_DNA_sequence_data.html#count-reads"><i class="fa fa-check"></i><b>10.8</b> Count reads</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="10_DNA_conditions.html"><a href="10_DNA_conditions.html"><i class="fa fa-check"></i><b>11</b> Using Conditions to Modify DNA Data</a>
<ul>
<li class="chapter" data-level="11.1" data-path="10_DNA_conditions.html"><a href="10_DNA_conditions.html#remove-short-reads"><i class="fa fa-check"></i><b>11.1</b> Remove short reads</a></li>
<li class="chapter" data-level="11.2" data-path="10_DNA_conditions.html"><a href="10_DNA_conditions.html#remove-pcr-primers"><i class="fa fa-check"></i><b>11.2</b> Remove PCR primers</a></li>
<li class="chapter" data-level="11.3" data-path="10_DNA_conditions.html"><a href="10_DNA_conditions.html#recap-4"><i class="fa fa-check"></i><b>11.3</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="11_reading_and_writing.html"><a href="11_reading_and_writing.html"><i class="fa fa-check"></i><b>12</b> Reading and writing files</a>
<ul>
<li class="chapter" data-level="12.1" data-path="11_reading_and_writing.html"><a href="11_reading_and_writing.html#open"><i class="fa fa-check"></i><b>12.1</b> 'open'</a></li>
<li class="chapter" data-level="12.2" data-path="11_reading_and_writing.html"><a href="11_reading_and_writing.html#with"><i class="fa fa-check"></i><b>12.2</b> 'with'</a></li>
<li class="chapter" data-level="12.3" data-path="11_reading_and_writing.html"><a href="11_reading_and_writing.html#read-a-data-file"><i class="fa fa-check"></i><b>12.3</b> Read a data file</a></li>
<li class="chapter" data-level="12.4" data-path="11_reading_and_writing.html"><a href="11_reading_and_writing.html#write-data-to-a-file"><i class="fa fa-check"></i><b>12.4</b> Write data to a file</a></li>
<li class="chapter" data-level="12.5" data-path="11_reading_and_writing.html"><a href="11_reading_and_writing.html#write-out-the-filtered-undulate-ray-sequence-data"><i class="fa fa-check"></i><b>12.5</b> Write out the filtered Undulate ray sequence data</a></li>
<li class="chapter" data-level="12.6" data-path="11_reading_and_writing.html"><a href="11_reading_and_writing.html#recap-5"><i class="fa fa-check"></i><b>12.6</b> Recap</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Python for Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="using-conditions-to-modify-dna-data" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">11</span> Using Conditions to Modify DNA Data<a href="10_DNA_conditions.html#using-conditions-to-modify-dna-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="images/checks.jpeg" height="200" /><img src="images/DNA_icon2.png" height="200" />
</center>
<p>Suppose we want to filter the raw sequence reads by some parameter.
For example we could remove reads of a certain length or remove sections of a sequence such as PCR primer sequences.
We can use our knowledge of loops and conditions to perform these tasks.</p>
<div id="remove-short-reads" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Remove short reads<a href="10_DNA_conditions.html#remove-short-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="images/comp_filter.png" style="width:20.0%" />
</center>
<p>Using what we know, we can write a script to remove reads below a certain read-length threshold from a FastQ file.</p>
<p>Starting a new section at the bottom of the script, re-specify where <code>SeqIO.parse</code> should read the file.
We need to do this because the <code>SeqIO.parse</code> object gets 'consumed' in the loop we built during the last section.
To loop again, and do something else, we need to re-establish the <code>SeqIO.parse</code> connection.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb54-1"><a href="10_DNA_conditions.html#cb54-1" aria-hidden="true" tabindex="-1"></a>R_undulata_data <span class="op">=</span> SeqIO.parse(wd <span class="op">+</span> <span class="st">&quot;/Python_course_data/R_undulata_WGS.fastq&quot;</span>, <span class="st">&quot;fastq&quot;</span>)</span></code></pre></div>
<p>Suppose we want to find and ultimately remove short reads from the data.
As our loop moves through the file, we can count the number of nucleotides in each read using the Python <strong>function</strong> <code>len()</code>.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb55-1"><a href="10_DNA_conditions.html#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> record <span class="kw">in</span> R_undulata_data:</span>
<span id="cb55-2"><a href="10_DNA_conditions.html#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="bu">print</span>(<span class="bu">len</span>(record.seq))</span></code></pre></div>
<p>Save and run the script to print the lengths of all the sequences.</p>
<p>We can see there is some variation in the length of reads in the file and can use an <b>if</b> statement to select only those reads with a read length of &gt; 100 nucleotides.</p>
<p>Modify the loop above to the following:</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb56-1"><a href="10_DNA_conditions.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># initialise an empty list to hold the records that we want to keep</span></span>
<span id="cb56-2"><a href="10_DNA_conditions.html#cb56-2" aria-hidden="true" tabindex="-1"></a>wanted_sequences <span class="op">=</span> []</span>
<span id="cb56-3"><a href="10_DNA_conditions.html#cb56-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-4"><a href="10_DNA_conditions.html#cb56-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> record <span class="kw">in</span> R_undulata_data:</span>
<span id="cb56-5"><a href="10_DNA_conditions.html#cb56-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#print(len(record.seq))</span></span>
<span id="cb56-6"><a href="10_DNA_conditions.html#cb56-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> <span class="bu">len</span>(record.seq) <span class="op">&gt;</span> <span class="dv">100</span>:</span>
<span id="cb56-7"><a href="10_DNA_conditions.html#cb56-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># add the record to the list of records we want to keep</span></span>
<span id="cb56-8"><a href="10_DNA_conditions.html#cb56-8" aria-hidden="true" tabindex="-1"></a>    wanted_sequences.append(record)</span>
<span id="cb56-9"><a href="10_DNA_conditions.html#cb56-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-10"><a href="10_DNA_conditions.html#cb56-10" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;Found &quot;</span> <span class="op">+</span> <span class="bu">str</span>(<span class="bu">len</span>(wanted_sequences)) <span class="op">+</span> <span class="st">&quot; reads &gt;100bp in length.&quot;</span>)</span></code></pre></div>
<p>Save and run the script. You will now have 2 lines of text as output, one saying the number of raw reads, and one saying the number of reads with a length &gt;100bp.</p>
<p>We used the <code>len()</code> <strong>function</strong> to see how many records are contained in the <code>wanted_sequences</code>.</p>
</div>
<div id="remove-pcr-primers" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Remove PCR primers<a href="10_DNA_conditions.html#remove-pcr-primers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="images/cut.png" style="width:20.0%" /><img src="images/DNA_icon.png" />
</center>
<p>Often you may want to check your sequence file for PCR primers, and remove them when they are found.
Using a <strong>for</strong> <strong>loop</strong> to move through the sequence file, we can check each DNA sequence read for the presence of a pre-defined primer sequence.
If it is present, we'll remove it.</p>
<p>We are going to check for the presence of the 13 nucleotide PCR primer "ACACTGCTGATCG" at the start of the reads using the <code>startswith()</code> <strong>method</strong>.
This <strong>method</strong> checks whether a <strong>string</strong> starts with a particular sub-string and returns <strong>True</strong> or <strong>False</strong>.
It is very useful for checking for primer sequences in our data.</p>
<p>The input for this primer removal step will be the 'wanted_sequences' <strong>variable</strong> from above (those of &gt;100bp in length).</p>
<p>Continue from the bottom of your "script2.py".</p>
<p><strong>Note</strong>: Below is the primer in triplet bases for ease of reading. Ensure you type it as one string with no spaces.</p>
<p><strong>ACA CTG CTG ATC G</strong></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb57-1"><a href="10_DNA_conditions.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co"># loop through the sequence records</span></span>
<span id="cb57-2"><a href="10_DNA_conditions.html#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> record <span class="kw">in</span> wanted_sequences:</span>
<span id="cb57-3"><a href="10_DNA_conditions.html#cb57-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb57-4"><a href="10_DNA_conditions.html#cb57-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># check whether the sequence starts with the primer sequence</span></span>
<span id="cb57-5"><a href="10_DNA_conditions.html#cb57-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> record.seq.startswith(<span class="st">&quot;ACACTGCTGATCG&quot;</span>):</span>
<span id="cb57-6"><a href="10_DNA_conditions.html#cb57-6" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(record)</span></code></pre></div>
<p>Save and run your script.</p>
<p>The code above prints a lot to the terminal. In each record, look at the lines starting "Seq" and note that every sequence does indeed start with the primer we specified above.
This indicates that there are lots of sequences with the primer sequence present.
Now that we have identified their presence, let's trim them off the start of the reads.</p>
<p>Modify the block of code above to the following:</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb58-1"><a href="10_DNA_conditions.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># define a list to hold the trimmed sequences</span></span>
<span id="cb58-2"><a href="10_DNA_conditions.html#cb58-2" aria-hidden="true" tabindex="-1"></a>trimmed_sequences <span class="op">=</span> []</span>
<span id="cb58-3"><a href="10_DNA_conditions.html#cb58-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Start counter to count number of reads with primer</span></span>
<span id="cb58-4"><a href="10_DNA_conditions.html#cb58-4" aria-hidden="true" tabindex="-1"></a>counter <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb58-5"><a href="10_DNA_conditions.html#cb58-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-6"><a href="10_DNA_conditions.html#cb58-6" aria-hidden="true" tabindex="-1"></a><span class="co"># loop through the sequence records</span></span>
<span id="cb58-7"><a href="10_DNA_conditions.html#cb58-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> record <span class="kw">in</span> wanted_sequences:</span>
<span id="cb58-8"><a href="10_DNA_conditions.html#cb58-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-9"><a href="10_DNA_conditions.html#cb58-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># check whether the sequence starts with the primer sequence</span></span>
<span id="cb58-10"><a href="10_DNA_conditions.html#cb58-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> record.seq.startswith(<span class="st">&quot;ACACTGCTGATCG&quot;</span>):</span>
<span id="cb58-11"><a href="10_DNA_conditions.html#cb58-11" aria-hidden="true" tabindex="-1"></a>    <span class="co">#print(record.seq)</span></span>
<span id="cb58-12"><a href="10_DNA_conditions.html#cb58-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb58-13"><a href="10_DNA_conditions.html#cb58-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if it does, subset the read from position 14 onwards </span></span>
<span id="cb58-14"><a href="10_DNA_conditions.html#cb58-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># and store in trimmed_sequences</span></span>
<span id="cb58-15"><a href="10_DNA_conditions.html#cb58-15" aria-hidden="true" tabindex="-1"></a>    trimmed_sequences.append(record[<span class="dv">13</span>:])</span>
<span id="cb58-16"><a href="10_DNA_conditions.html#cb58-16" aria-hidden="true" tabindex="-1"></a>    counter <span class="op">=</span> counter <span class="op">+</span> <span class="dv">1</span></span>
<span id="cb58-17"><a href="10_DNA_conditions.html#cb58-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-18"><a href="10_DNA_conditions.html#cb58-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span>:</span>
<span id="cb58-19"><a href="10_DNA_conditions.html#cb58-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if it doesn&#39;t have the primer, </span></span>
<span id="cb58-20"><a href="10_DNA_conditions.html#cb58-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># retain the whole read and store in trimmed_sequences</span></span>
<span id="cb58-21"><a href="10_DNA_conditions.html#cb58-21" aria-hidden="true" tabindex="-1"></a>    trimmed_sequences.append(record)</span>
<span id="cb58-22"><a href="10_DNA_conditions.html#cb58-22" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb58-23"><a href="10_DNA_conditions.html#cb58-23" aria-hidden="true" tabindex="-1"></a>number_sequences_trimmed <span class="op">=</span> counter</span>
<span id="cb58-24"><a href="10_DNA_conditions.html#cb58-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-25"><a href="10_DNA_conditions.html#cb58-25" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;Found and trimmed &quot;</span> <span class="op">+</span> <span class="bu">str</span>(number_sequences_trimmed) <span class="op">+</span> <span class="op">\</span></span>
<span id="cb58-26"><a href="10_DNA_conditions.html#cb58-26" aria-hidden="true" tabindex="-1"></a><span class="st">&quot; reads with PCR primer present at start.&quot;</span>)</span></code></pre></div>
<p>Save and one the code one more time to see your three lines of information printed to screen.</p>
</div>
<div id="recap-4" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> Recap<a href="10_DNA_conditions.html#recap-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="images/recap.webp" height="300" />
</center>
<p>Using our knowledge of reading FASTQ files, looping through the data, and using conditions to check certain characteristics of the data we have been able to filter out short-reads and also remove primer contamination. We are starting to build a pipeline!</p>
<p>We will learn to write data out and store it in a new file in the next chapter.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="09_DNA_sequence_data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="11_reading_and_writing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
