<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10 Reading sequence data | Python for Bioinformatics</title>
  <meta name="description" content="NEOF book for the Python for Bioinformatics workshop" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="10 Reading sequence data | Python for Bioinformatics" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/images/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Python for Bioinformatics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10 Reading sequence data | Python for Bioinformatics" />
  
  <meta name="twitter:description" content="NEOF book for the Python for Bioinformatics workshop" />
  <meta name="twitter:image" content="/images/NEOF.png" />




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="09_loops.html"/>
<link rel="next" href="11_DNA_conditions.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="images/neof_small_logo.png"></a></li>
<li><a>Python for Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-start.html"><a href="01-start.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-intro.html"><a href="02-intro.html"><i class="fa fa-check"></i><b>2</b> Python for bioinformatics</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-intro.html"><a href="02-intro.html#materials-covered-in-the-course"><i class="fa fa-check"></i><b>2.1</b> Materials Covered in the Course</a></li>
<li class="chapter" data-level="2.2" data-path="02-intro.html"><a href="02-intro.html#inputting-commands"><i class="fa fa-check"></i><b>2.2</b> Inputting Commands</a></li>
<li class="chapter" data-level="2.3" data-path="02-intro.html"><a href="02-intro.html#pseudocode"><i class="fa fa-check"></i><b>2.3</b> Pseudocode</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03_cluster_introduction.html"><a href="03_cluster_introduction.html"><i class="fa fa-check"></i><b>3</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03_cluster_introduction.html"><a href="03_cluster_introduction.html#logon-instructions"><i class="fa fa-check"></i><b>3.1</b> Logon Instructions</a></li>
<li class="chapter" data-level="3.2" data-path="03_cluster_introduction.html"><a href="03_cluster_introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>3.2</b> The Terminal Window</a></li>
<li class="chapter" data-level="3.3" data-path="03_cluster_introduction.html"><a href="03_cluster_introduction.html#load-workshop-data"><i class="fa fa-check"></i><b>3.3</b> Load workshop data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04_basic_scripting.html"><a href="04_basic_scripting.html"><i class="fa fa-check"></i><b>4</b> Scripting in Python</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04_basic_scripting.html"><a href="04_basic_scripting.html#introduction-to-commands-and-scripts"><i class="fa fa-check"></i><b>4.1</b> Introduction to commands and scripts</a></li>
<li class="chapter" data-level="4.2" data-path="04_basic_scripting.html"><a href="04_basic_scripting.html#issuing-commands-directly-to-the-interpreter"><i class="fa fa-check"></i><b>4.2</b> Issuing commands directly to the interpreter</a></li>
<li class="chapter" data-level="4.3" data-path="04_basic_scripting.html"><a href="04_basic_scripting.html#a-note-on-indentation"><i class="fa fa-check"></i><b>4.3</b> A note on indentation</a></li>
<li class="chapter" data-level="4.4" data-path="04_basic_scripting.html"><a href="04_basic_scripting.html#tabs-vs-spaces"><i class="fa fa-check"></i><b>4.4</b> Tabs vs spaces</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html"><i class="fa fa-check"></i><b>5</b> Python Fundamentals</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#operators"><i class="fa fa-check"></i><b>5.1</b> Operators</a></li>
<li class="chapter" data-level="5.2" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#comments"><i class="fa fa-check"></i><b>5.2</b> Comments</a></li>
<li class="chapter" data-level="5.3" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#variables"><i class="fa fa-check"></i><b>5.3</b> Variables</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#strings"><i class="fa fa-check"></i><b>5.3.1</b> Strings</a></li>
<li class="chapter" data-level="5.3.2" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#integers"><i class="fa fa-check"></i><b>5.3.2</b> Integers</a></li>
<li class="chapter" data-level="5.3.3" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#floats-floating-point-numbers"><i class="fa fa-check"></i><b>5.3.3</b> Floats (floating point numbers)</a></li>
<li class="chapter" data-level="5.3.4" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#booleans"><i class="fa fa-check"></i><b>5.3.4</b> Booleans</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#classes-of-variable"><i class="fa fa-check"></i><b>5.4</b> Classes of variable</a></li>
<li class="chapter" data-level="5.5" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#transforming-variables"><i class="fa fa-check"></i><b>5.5</b> Transforming variables</a></li>
<li class="chapter" data-level="5.6" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#functions-and-methods"><i class="fa fa-check"></i><b>5.6</b> Functions and methods</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#running-a-function"><i class="fa fa-check"></i><b>5.6.1</b> Running a function</a></li>
<li class="chapter" data-level="5.6.2" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#running-a-method"><i class="fa fa-check"></i><b>5.6.2</b> Running a method</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="05_python_fundamentals.html"><a href="05_python_fundamentals.html#recap"><i class="fa fa-check"></i><b>5.7</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06_data_structures.html"><a href="06_data_structures.html"><i class="fa fa-check"></i><b>6</b> Data Structures</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06_data_structures.html"><a href="06_data_structures.html#lists"><i class="fa fa-check"></i><b>6.1</b> Lists</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="06_data_structures.html"><a href="06_data_structures.html#add-numeric-data-to-a-list"><i class="fa fa-check"></i><b>6.1.1</b> Add numeric data to a list</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="06_data_structures.html"><a href="06_data_structures.html#subsetting-data"><i class="fa fa-check"></i><b>6.2</b> Subsetting data</a></li>
<li class="chapter" data-level="6.3" data-path="06_data_structures.html"><a href="06_data_structures.html#set"><i class="fa fa-check"></i><b>6.3</b> Set</a></li>
<li class="chapter" data-level="6.4" data-path="06_data_structures.html"><a href="06_data_structures.html#transform-a-list-to-a-set"><i class="fa fa-check"></i><b>6.4</b> Transform a list to a set</a></li>
<li class="chapter" data-level="6.5" data-path="06_data_structures.html"><a href="06_data_structures.html#dictionary"><i class="fa fa-check"></i><b>6.5</b> Dictionary</a></li>
<li class="chapter" data-level="6.6" data-path="06_data_structures.html"><a href="06_data_structures.html#recap-1"><i class="fa fa-check"></i><b>6.6</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07_conditions.html"><a href="07_conditions.html"><i class="fa fa-check"></i><b>7</b> Python Conditions: 'if' and 'else' statements</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07_conditions.html"><a href="07_conditions.html#if-statements"><i class="fa fa-check"></i><b>7.1</b> If statements</a></li>
<li class="chapter" data-level="7.2" data-path="07_conditions.html"><a href="07_conditions.html#if-else-statements"><i class="fa fa-check"></i><b>7.2</b> If else statements</a></li>
<li class="chapter" data-level="7.3" data-path="07_conditions.html"><a href="07_conditions.html#recap-2"><i class="fa fa-check"></i><b>7.3</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08_scripting_in_python.html"><a href="08_scripting_in_python.html"><i class="fa fa-check"></i><b>8</b> Scripting in Python</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08_scripting_in_python.html"><a href="08_scripting_in_python.html#open-a-plain-text-editor"><i class="fa fa-check"></i><b>8.1</b> Open a plain text editor</a></li>
<li class="chapter" data-level="8.2" data-path="08_scripting_in_python.html"><a href="08_scripting_in_python.html#running-a-script-from-the-terminal"><i class="fa fa-check"></i><b>8.2</b> Running a script from the terminal</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09_loops.html"><a href="09_loops.html"><i class="fa fa-check"></i><b>9</b> Loops</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09_loops.html"><a href="09_loops.html#data-structure-recap"><i class="fa fa-check"></i><b>9.1</b> Data Structure Recap</a></li>
<li class="chapter" data-level="9.2" data-path="09_loops.html"><a href="09_loops.html#introduction-to-loops"><i class="fa fa-check"></i><b>9.2</b> Introduction To Loops</a></li>
<li class="chapter" data-level="9.3" data-path="09_loops.html"><a href="09_loops.html#for-loops"><i class="fa fa-check"></i><b>9.3</b> For Loops</a></li>
<li class="chapter" data-level="9.4" data-path="09_loops.html"><a href="09_loops.html#for-loop-example"><i class="fa fa-check"></i><b>9.4</b> For Loop Example</a></li>
<li class="chapter" data-level="9.5" data-path="09_loops.html"><a href="09_loops.html#use-a-for-loop-to-complement-a-dna-sequence"><i class="fa fa-check"></i><b>9.5</b> Use a for loop to complement a DNA sequence</a></li>
<li class="chapter" data-level="9.6" data-path="09_loops.html"><a href="09_loops.html#use-a-for-loop-to-reverse-complement-a-dna-sequence"><i class="fa fa-check"></i><b>9.6</b> Use a for loop to reverse-complement a DNA sequence</a></li>
<li class="chapter" data-level="9.7" data-path="09_loops.html"><a href="09_loops.html#while-loops"><i class="fa fa-check"></i><b>9.7</b> While loops</a></li>
<li class="chapter" data-level="9.8" data-path="09_loops.html"><a href="09_loops.html#further-control-statements"><i class="fa fa-check"></i><b>9.8</b> Further control statements</a></li>
<li class="chapter" data-level="9.9" data-path="09_loops.html"><a href="09_loops.html#recap-3"><i class="fa fa-check"></i><b>9.9</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10_DNA_sequence_data.html"><a href="10_DNA_sequence_data.html"><i class="fa fa-check"></i><b>10</b> Reading sequence data</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10_DNA_sequence_data.html"><a href="10_DNA_sequence_data.html#seqio"><i class="fa fa-check"></i><b>10.1</b> SeqIO</a></li>
<li class="chapter" data-level="10.2" data-path="10_DNA_sequence_data.html"><a href="10_DNA_sequence_data.html#reading-a-fastq-dna-data-file-into-python"><i class="fa fa-check"></i><b>10.2</b> Reading a FASTQ DNA data file into Python</a></li>
<li class="chapter" data-level="10.3" data-path="10_DNA_sequence_data.html"><a href="10_DNA_sequence_data.html#the-fastq-format"><i class="fa fa-check"></i><b>10.3</b> The FASTQ format</a></li>
<li class="chapter" data-level="10.4" data-path="10_DNA_sequence_data.html"><a href="10_DNA_sequence_data.html#read-a-fastq-file-into-python"><i class="fa fa-check"></i><b>10.4</b> Read a FASTQ file into Python</a></li>
<li class="chapter" data-level="10.5" data-path="10_DNA_sequence_data.html"><a href="10_DNA_sequence_data.html#setting-up-a-script-to-use-biopython"><i class="fa fa-check"></i><b>10.5</b> Setting up a script to use BioPython</a></li>
<li class="chapter" data-level="10.6" data-path="10_DNA_sequence_data.html"><a href="10_DNA_sequence_data.html#read-a-fie-of-sequence-data"><i class="fa fa-check"></i><b>10.6</b> Read a fie of sequence data</a></li>
<li class="chapter" data-level="10.7" data-path="10_DNA_sequence_data.html"><a href="10_DNA_sequence_data.html#modifying-reads"><i class="fa fa-check"></i><b>10.7</b> Modifying reads</a></li>
<li class="chapter" data-level="10.8" data-path="10_DNA_sequence_data.html"><a href="10_DNA_sequence_data.html#count-reads"><i class="fa fa-check"></i><b>10.8</b> Count reads</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11_DNA_conditions.html"><a href="11_DNA_conditions.html"><i class="fa fa-check"></i><b>11</b> Using Conditions to Modify DNA Data</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11_DNA_conditions.html"><a href="11_DNA_conditions.html#remove-short-reads"><i class="fa fa-check"></i><b>11.1</b> Remove short reads</a></li>
<li class="chapter" data-level="11.2" data-path="11_DNA_conditions.html"><a href="11_DNA_conditions.html#remove-pcr-primers"><i class="fa fa-check"></i><b>11.2</b> Remove PCR primers</a></li>
<li class="chapter" data-level="11.3" data-path="11_DNA_conditions.html"><a href="11_DNA_conditions.html#recap-4"><i class="fa fa-check"></i><b>11.3</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12_reading_and_writing.html"><a href="12_reading_and_writing.html"><i class="fa fa-check"></i><b>12</b> Reading and writing files</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12_reading_and_writing.html"><a href="12_reading_and_writing.html#open"><i class="fa fa-check"></i><b>12.1</b> 'open'</a></li>
<li class="chapter" data-level="12.2" data-path="12_reading_and_writing.html"><a href="12_reading_and_writing.html#with"><i class="fa fa-check"></i><b>12.2</b> 'with'</a></li>
<li class="chapter" data-level="12.3" data-path="12_reading_and_writing.html"><a href="12_reading_and_writing.html#read-a-data-file"><i class="fa fa-check"></i><b>12.3</b> Read a data file</a></li>
<li class="chapter" data-level="12.4" data-path="12_reading_and_writing.html"><a href="12_reading_and_writing.html#write-data-to-a-file"><i class="fa fa-check"></i><b>12.4</b> Write data to a file</a></li>
<li class="chapter" data-level="12.5" data-path="12_reading_and_writing.html"><a href="12_reading_and_writing.html#write-out-the-filtered-undulate-ray-sequence-data"><i class="fa fa-check"></i><b>12.5</b> Write out the filtered Undulate ray sequence data</a></li>
<li class="chapter" data-level="12.6" data-path="12_reading_and_writing.html"><a href="12_reading_and_writing.html#recap-5"><i class="fa fa-check"></i><b>12.6</b> Recap</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Python for Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="reading-sequence-data" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">10</span> Reading sequence data<a href="10_DNA_sequence_data.html#reading-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="images/DNA_icon.png" height="300" /><img src="images/biopython_logo.png" height="300" />
</center>
<p>Biopython is a suite of libraries and tools designed for analysis of biological datasets in Python, largely centered around the processing of DNA sequence data.</p>
<p>You can find much more information about Biopython on the <a href="https://biopython.org/">website</a>.</p>
<p>During this section on Biopython we are going to be using the SeqIO module. The knowledge and approaches from the previous chapters
will now be used to process some genomic DNA sequence data.</p>
<div id="seqio" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> SeqIO<a href="10_DNA_sequence_data.html#seqio" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="images/DNA_computer.png" style="width:20.0%" />
</center>
<p>The SeqIO tools are the main Sequence Input/Output interface for Biopython. It provides a relatively simple, uniform interface for working with various types of DNA sequence files. The module reads raw sequence files (such as .FASTQ files you will likely receive from your sequencing facility) and converts them to a Python readable format; a <a href="https://biopython.org/wiki/SeqRecord">SeqRecord</a> class.</p>
</div>
<div id="reading-a-fastq-dna-data-file-into-python" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Reading a FASTQ DNA data file into Python<a href="10_DNA_sequence_data.html#reading-a-fastq-dna-data-file-into-python" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>One of the most common types of DNA sequence data files you will be working with is the <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a> file.
The FASTQ file contains multiple DNA sequences, each one might be a single short read from an Illumina sequencer and the file can contain thousands or millions of sequences depending on which sequencer was used to produce it.</p>
<p>The FASTQ has a very specific format and holds three important pieces of information.</p>
</div>
<div id="the-fastq-format" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> The FASTQ format<a href="10_DNA_sequence_data.html#the-fastq-format" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="images/fastq_icon.png" style="width:20.0%" />
</center>
<p>The file has a repetitive structure of four distinct components:</p>
<ul>
<li>A unique sequence identifier (ID)</li>
<li>The DNA nucleotide sequence</li>
<li>A small spacer</li>
<li>The quality score associated with each nucleotide</li>
</ul>
<p>The ID line always starts with the @ symbol and this is followed by a string of numbers and letters giving details such as the sequencer that produced the data, the run number, and an index adapter used on the sample.</p>
<p>The DNA nucleotide sequence is shown as a string of ATGC characters.</p>
<p>There is a spacer consisting of the "+" character which separates the sequence from the quality scores.</p>
<p>The quality score associated with each nucleotide above is ASCII encoded with each symbol representing a quality score from 0 to 40 (higher being better). This line is not meant to be easily human-readable.</p>
<p>An example of a single unit of this four line repeat can be seen below:</p>
<pre><code>@MISEQ:42:000000000-A8DD3:1:1101:15046:1468
TTCACATCCTGACCATTCAGTTGAGCAAAATAGTTCTTCAGT
+
CCFFFFFGHHHHJIJJJJIJJJIIJJJJIIIJJGFIIIJEDD</code></pre>
</div>
<div id="read-a-fastq-file-into-python" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Read a FASTQ file into Python<a href="10_DNA_sequence_data.html#read-a-fastq-file-into-python" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Scenario:</strong> You have just received some FASTQ formatted data back from your sequencing facility. The data was generated using a shotgun, whole-genome sequencing approach to sequence the genome of the undulate ray (<em>Raja undulata</em>).</p>
<center>
<img src="images/undulate_ray.jpeg" style="width:20.0%" />
</center>
<p>We are going to read the data into Python, ready to do some data processing and analyses.</p>
</div>
<div id="setting-up-a-script-to-use-biopython" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> Setting up a script to use BioPython<a href="10_DNA_sequence_data.html#setting-up-a-script-to-use-biopython" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Open a new file in Gedit by clicking the "Create a new document" button (<img src="images/gedit_create_new_doc.png?display=inline-block" style="width:4.0%" />) at the top-left.</p>
<p>We need to load a couple of Python libraries:</p>
<p><code>Biopython</code> (a suite of bioinformatics focused Python tools)</p>
<p>...and...</p>
<p><code>re</code> (allows us to perform searches using wildcards and special characters).</p>
<p>Add the following to the top of the script.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb48-1"><a href="10_DNA_sequence_data.html#cb48-1" tabindex="-1"></a><span class="im">from</span> Bio <span class="im">import</span> SeqIO</span>
<span id="cb48-2"><a href="10_DNA_sequence_data.html#cb48-2" tabindex="-1"></a><span class="im">import</span> re</span></code></pre></div>
<p>You should hopefully still have the current working directory on your clipboard.
Set a variable called 'wd', (working directory) to the filepath on your clipboard by adding a version of the following:</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb49-1"><a href="10_DNA_sequence_data.html#cb49-1" tabindex="-1"></a>wd <span class="op">=</span> <span class="st">&quot;/pub14/tea/nsc201&quot;</span></span></code></pre></div>
</div>
<div id="read-a-fie-of-sequence-data" class="section level2 hasAnchor" number="10.6">
<h2><span class="header-section-number">10.6</span> Read a fie of sequence data<a href="10_DNA_sequence_data.html#read-a-fie-of-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="images/bioinformatics_icon2.png" style="width:20.0%" />
</center>
<p>We are going to use <code>SeqIO</code> (which stands for sequence in/out) and the <code>parse</code> <strong>function</strong> which splits the FASTQ file into its separate components.</p>
<p>We store the information on which file to parse in the variable called <code>R_undulata_R1_data</code>.
<code>SeqIO.parse</code> requires two arguments:</p>
<ul>
<li>The filepath to the file which is to be read.</li>
<li>The format of that file.</li>
</ul>
<div class="sourceCode" id="cb50"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb50-1"><a href="10_DNA_sequence_data.html#cb50-1" tabindex="-1"></a>R_undulata_data <span class="op">=</span> SeqIO.parse(wd <span class="op">+</span> <span class="st">&quot;/Python_course_data/R_undulata_WGS.fastq&quot;</span>, <span class="st">&quot;fastq&quot;</span>)</span></code></pre></div>
<p>We can then look through the file using a <b>for</b> loop. Each of the four-unit repeats of the FASTQ file is read as a separate 'record' and we can access them in the script.</p>
<p>To access the ID line of each record, we use '.id' and to access the sequence we use '.seq'.
Each of these elements can be printed in the terminal using an appropriate print statement.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb51-1"><a href="10_DNA_sequence_data.html#cb51-1" tabindex="-1"></a><span class="cf">for</span> record <span class="kw">in</span> R_undulata_data:</span>
<span id="cb51-2"><a href="10_DNA_sequence_data.html#cb51-2" tabindex="-1"></a>  <span class="bu">print</span>(record.<span class="bu">id</span>)</span>
<span id="cb51-3"><a href="10_DNA_sequence_data.html#cb51-3" tabindex="-1"></a>  <span class="bu">print</span>(record.seq)</span></code></pre></div>
<p>Save the new script as <strong>'script02.py'</strong> and run it from the terminal window to print the ID and sequences.</p>
</div>
<div id="modifying-reads" class="section level2 hasAnchor" number="10.7">
<h2><span class="header-section-number">10.7</span> Modifying reads<a href="10_DNA_sequence_data.html#modifying-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can perform operations on each of these elements. For example, Biopython has a built in <code>translate()</code> <strong>method</strong> to convert nucleotide sequence into protein sequence (although bear in mind that we are not considering reading frames and we likely have lots of non-coding data).</p>
<p>Modify the <strong>for</strong> <strong>loop</strong> above to the following:</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb52-1"><a href="10_DNA_sequence_data.html#cb52-1" tabindex="-1"></a><span class="cf">for</span> record <span class="kw">in</span> R_undulata_data:</span>
<span id="cb52-2"><a href="10_DNA_sequence_data.html#cb52-2" tabindex="-1"></a>  <span class="bu">print</span>(record.<span class="bu">id</span>)</span>
<span id="cb52-3"><a href="10_DNA_sequence_data.html#cb52-3" tabindex="-1"></a>  <span class="co">#print(record.seq)</span></span>
<span id="cb52-4"><a href="10_DNA_sequence_data.html#cb52-4" tabindex="-1"></a>  <span class="bu">print</span>(record.seq.translate())</span></code></pre></div>
<p>Save the script and run it on the terminal.</p>
</div>
<div id="count-reads" class="section level2 hasAnchor" number="10.8">
<h2><span class="header-section-number">10.8</span> Count reads<a href="10_DNA_sequence_data.html#count-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="images/count.png" style="width:15.0%" />
</center>
<p>We can implement a counter to tally the total number of reads in the file. Modify the <strong>for</strong> <strong>loop</strong> above to the following (you do not need to add the comments however):</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb53-1"><a href="10_DNA_sequence_data.html#cb53-1" tabindex="-1"></a><span class="co">## start a counter to track how many reads</span></span>
<span id="cb53-2"><a href="10_DNA_sequence_data.html#cb53-2" tabindex="-1"></a>counter <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb53-3"><a href="10_DNA_sequence_data.html#cb53-3" tabindex="-1"></a></span>
<span id="cb53-4"><a href="10_DNA_sequence_data.html#cb53-4" tabindex="-1"></a><span class="cf">for</span> record <span class="kw">in</span> R_undulata_data:</span>
<span id="cb53-5"><a href="10_DNA_sequence_data.html#cb53-5" tabindex="-1"></a>  <span class="co">#print(record.id) </span></span>
<span id="cb53-6"><a href="10_DNA_sequence_data.html#cb53-6" tabindex="-1"></a>  <span class="co">#print(record.seq)</span></span>
<span id="cb53-7"><a href="10_DNA_sequence_data.html#cb53-7" tabindex="-1"></a>  <span class="co">#print(record.seq.translate())</span></span>
<span id="cb53-8"><a href="10_DNA_sequence_data.html#cb53-8" tabindex="-1"></a>  <span class="co"># increase the counter by 1 each time it sees a new read</span></span>
<span id="cb53-9"><a href="10_DNA_sequence_data.html#cb53-9" tabindex="-1"></a>  counter <span class="op">=</span> counter <span class="op">+</span> <span class="dv">1</span></span>
<span id="cb53-10"><a href="10_DNA_sequence_data.html#cb53-10" tabindex="-1"></a></span>
<span id="cb53-11"><a href="10_DNA_sequence_data.html#cb53-11" tabindex="-1"></a><span class="co"># save the number of raw sequences for use later</span></span>
<span id="cb53-12"><a href="10_DNA_sequence_data.html#cb53-12" tabindex="-1"></a>number_raw_reads <span class="op">=</span> counter</span>
<span id="cb53-13"><a href="10_DNA_sequence_data.html#cb53-13" tabindex="-1"></a></span>
<span id="cb53-14"><a href="10_DNA_sequence_data.html#cb53-14" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;Raw sequence file has &quot;</span> <span class="op">+</span> <span class="bu">str</span>(number_raw_reads) <span class="op">+</span> <span class="op">\</span></span>
<span id="cb53-15"><a href="10_DNA_sequence_data.html#cb53-15" tabindex="-1"></a><span class="st">&quot; sequence records.&quot;</span>)</span></code></pre></div>
<p>Note that we need to convert the <code>number_raw_reads</code> to a <strong>string</strong> type for the print statement to work using the <code>str()</code> <strong>function</strong>. Python will not print variables of different types in the same statement.</p>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="09_loops.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="11_DNA_conditions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
