# (PART\*) Practical python {.unnumbered}

# Reading sequence data {#reading_seq_data}

```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/biopython_logo.png", auto_pdf = TRUE)
```

**Biopython** is a suite of **libraries** and tools designed for analysis of biological datasets in Python, largely centred around the processing of DNA sequence data.

You can find much more information about **Biopython** on the [website](https://biopython.org/).

During this section on **Biopython** we are going to be using the **SeqIO** **module**.

The knowledge and approaches from the previous chapters will now be used to process some genomic DNA sequence data.

Create a new notebook called **"04-Practical_python"**. Use this notebook in this chapter.

## SeqIO

```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/seqio.png", auto_pdf = TRUE)
```

The **SeqIO** tools are the main Sequence Input/Output interface for **Biopython**. It provides a relatively simple, uniform interface for working with various types of DNA sequence files. The **module** reads raw sequence files (such as .FASTQ files you will likely receive from your sequencing facility) and converts them to a Python readable format; a [SeqRecord](https://biopython.org/wiki/SeqRecord) **class**.

## The FASTQ format

```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/fastq.png", auto_pdf = TRUE)
```

One of the most common types of DNA sequence data files you will be working with is the [FASTQ](https://en.wikipedia.org/wiki/FASTQ_format) file. FASTQ files contain multiple DNA sequences. Each one might be a single short read from an Illumina sequencer. The file can contain thousands or millions of sequences depending on which sequencer was used to produce it.

FASTQ files have a very specific format and hold three important pieces of information.

The file has a repetitive structure of four distinct components:

-   A unique sequence identifier (ID)
    -   Always starts with the __"@"__ symbol followed by a string of numbers and letters
    -   This contains details such as the sequencer that produced the data, the run number, and an index adapter used on the sample.
-   The DNA nucleotide sequence
    -   Shown as a string of ATGC characters
-   A small spacer
    -   Consisting of the __"+"__ character which separates the sequence from the quality scores
-   The quality score associated with each nucleotide
    -   The quality score associated with each nucleotide above is ASCII encoded with each symbol representing a quality score from 0 to 40 (higher being better).
    -   This line is not meant to be easily human-readable.

Below is an example of a single FASTQ entry:

```{=html}
<!-- -->
```
    @MISEQ:42:000000000-A8DD3:1:1101:15046:1468
    TTCACATCCTGACCATTCAGTTGAGCAAAATAGTTCTTCAGT
    +
    CCFFFFFGHHHHJIJJJJIJJJIIJJJJIIIJJGFIIIJEDD

## Read a FASTQ file into Python

```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/Leucoraja erinacea.png", auto_pdf = TRUE)
```

**Scenario:** You have just received some FASTQ formatted data back from your sequencing facility. The data was generated using a shotgun, whole-genome sequencing approach to sequence the genome of the undulate ray (*Raja undulata*).

We are going to read the data into Python, ready to do some data processing and analyses.

### Loading the python library

```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/library_1.png", auto_pdf = TRUE)
```

We need to load a __Python library__:

-   **Biopython** (`Bio`): a suite of bioinformatics focused Python tools.
    -   From this __library__ we will import the __module__ `SeqIO`

To load (**import**) this __library__ and **module** add the following towards the top of your notebook.

```{python, eval=FALSE}
#Import libraries and modules
from Bio import SeqIO
```

It is also good to create a **variable** called "wd" that contains your working directory when working with files. Set this __variable__ to your __"Python_course_data"__ filepath by adding a version of the following:

**Note**: The below code is an instructor's directory. Ensure to set your own. A [previous command](#working-directory) may be useful.

```{python, eval=FALSE}
#Working directory
wd = "/pub14/tea/nsc201/Python_course_data"
```

### Read a file of sequence data

```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/mobile_read.png", auto_pdf = TRUE)
```

We are going to use `SeqIO`'s (which stands for sequence in/out) `parse()` **function** which splits the FASTQ file into its separate components.

We'll store the information in the variable called `R_undulata_R1_data`.

`SeqIO.parse` requires two arguments:

-   The filepath of the file to be read.
-   The format of that file.

```{python, eval=FALSE}
#Parse file
R_undulata_data = SeqIO.parse(wd + "/R_undulata_WGS.fastq", "fastq")
```

### Fastq record elements

```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/elements.png", auto_pdf = TRUE)
```

We can then look through the file using a __for loop__. Each of the four-unit repeats of the FASTQ file is read as a separate 'record element' and we can access them in the script.

When looping through the data the different record elements are called by using the __loop variable__ and a suffix for the type of element. The suffixes of interest are:

-   **`.id`**: The sequence identifier.
-   **`.seq`**: The sequence.
- __`.letter_annotations["phred_quality"]`__: A list of the phred quality scores (numbers).

We can print this information for each record with the following **for loop**.

```{python, eval=FALSE}
#Print IDs and sequences
for record in R_undulata_data:
  print(record.id)
  print(record.seq)
```

## Reading and using files

```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/disappear.png", auto_pdf = TRUE)
```

A peculiarity of **Python** is that when a file has been read and then used it is removed from memory.

To demonstrate, create and run the three below **code cells**:

```{python, eval=FALSE}
#Parse file
R_undulata_data = SeqIO.parse(wd + "/R_undulata_WGS.fastq", "fastq")
```

```{python, eval=FALSE}
#Print IDs and sequences
for record in R_undulata_data:
  print(record.id)
```

```{python, eval=FALSE}
#Print IDs and sequences
for record in R_undulata_data:
  print(record.seq)
```

Running them from top to bottom you will find that only the second **code cell** gives you output. The data in the variable **R_undulata_data** was used by the **for loop** in the 2nd code cell. Because it is a **python variable** created by __reading__ a file the information is then removed. Once you have __read__ information and used it once you need to __reread__ the information to use it again.

Try running the code from the 3 **code cells** in 1 **code cell**.

```{python, eval=FALSE}
#Parse file
R_undulata_data = SeqIO.parse(wd + "/R_undulata_WGS.fastq", "fastq")
#Print IDs and sequences
for record in R_undulata_data:
  print(record.id)
#Print IDs and sequences
for record in R_undulata_data:
  print(record.seq)
```

Again, you will only get the **ID** information.

However, if you read the file again after using the **ID** information you can get both.

__Note__: After running the below you will need to scroll through a lot of **IDs** to see the **Sequences**.

```{python, eval=FALSE}
#Parse file
R_undulata_data = SeqIO.parse(wd + "/R_undulata_WGS.fastq", "fastq")
#Print IDs and sequences
for record in R_undulata_data:
  print(record.id)
#Parse file
R_undulata_data = SeqIO.parse(wd + "/R_undulata_WGS.fastq", "fastq")
#Print IDs and sequences
for record in R_undulata_data:
  print(record.seq)
```

This is an important concept so we'll touch on it again later.

## Modifying reads

```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/DNA_translation.png", auto_pdf = TRUE)
```

We can perform operations on each of the record elements.

__Biopython__ has a built in `.translate()` **method** to convert nucleotide sequences into protein sequences. In this case we are not considering reading frames and we'll likely have lots of non-coding data.

See this in action with the below code.

```{python, eval=FALSE}
#Translate
#Parse file
R_undulata_data = SeqIO.parse(wd + "/R_undulata_WGS.fastq", "fastq")
#Print IDs and translated sequences
for record in R_undulata_data:
  print(record.id)
  print(record.seq.translate())
```

## Count reads

```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/count.png", auto_pdf = TRUE)
```

We can implement a counter to tally the total number of reads in the file.

Create the below **for** **loop** to count the number of reads/records:

```{python, eval=FALSE}
#Count reads/records
#Parse file
R_undulata_data = SeqIO.parse(wd + "/R_undulata_WGS.fastq", "fastq")

#Start a counter to track the number of reads/records
counter = 0

#Count reads/records
for record in R_undulata_data:
  #Increase the counter by 1 each time it sees a new read
  counter = counter + 1

#Save the number of reads/records for later use
num_raw_reads = counter

#Print the information
print("Raw sequence file has " + str(num_raw_reads) + " sequence records.")
```

## Reading sequence data MCQ recap

```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/question_bubble_orange.png", auto_pdf = TRUE)
```

Superb! That is the end of this chapter. Ensure to save your **notebook** regularly.

For a quick recap, choose the correct option to complete the below sentences:

```{r, echo = FALSE}
opts_p <- c(answer="__Biopython__", "__Fastq__", "__Read__")
```

1.  \_\_\_\_\_\_\_\_ is a suite of Python **libraries** and tools, useful for analysing biological datasets. `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__Biopython__", "__Fastq__", answer="__Read__")
```

2.  Information on \_\_\_\_\_\_\_\_ files is lost once used. `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__Biopython__", answer="__Fastq__", "__Read__")
```

3.  \_\_\_\_\_\_\_\_ files have 4 elements per record/entry. `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__`.seq`__", answer="__`SeqIO.parse()`__", "__`.translate()`__")
```

4.  The **function** \_\_\_\_\_\_\_\_ can read/parse fastq files. `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__`.seq`__", "__`SeqIO.parse()`__", answer="__`.translate()`__")
```

5.  The `SeqIO` **method** \_\_\_\_\_\_\_\_ can be used to translate DNA sequences to amino acid sequences. `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c(answer="__`.seq`__", "__`SeqIO.parse()`__", "__`.translate()`__")
```

6.  The sequence element in a parsed fastq file is referred to by the suffix \_\_\_\_\_\_\_\_. `r longmcq(opts_p)`
