<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Background | Metabarcoding for diet analysis and environmental DNA</title>
  <meta name="description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Background | Metabarcoding for diet analysis and environmental DNA" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Background | Metabarcoding for diet analysis and environmental DNA" />
  
  <meta name="twitter:description" content="NEOF book for the Metabarcoding for diet analysis and environmental DNA workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Katy Maher, Helen Hipperson &amp; Graeme Fox" />


<meta name="date" content="2025-12-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="01-Metabarcoding.html"/>
<link rel="next" href="03-R_DADA2_intro.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Environmental metabarcoding</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting started</b></span></li>
<li class="chapter" data-level="1" data-path="01-Metabarcoding.html"><a href="01-Metabarcoding.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Example_dataset.html"><a href="02-Example_dataset.html"><i class="fa fa-check"></i><b>2</b> Background</a></li>
<li class="chapter" data-level="3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html"><i class="fa fa-check"></i><b>3</b> Introduction to DADA2 and R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#dada2"><i class="fa fa-check"></i><b>3.1</b> DADA2</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#otus-vs-asvs"><i class="fa fa-check"></i><b>3.1.1</b> OTUs vs ASVs</a></li>
<li class="chapter" data-level="3.1.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#workflow"><i class="fa fa-check"></i><b>3.1.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#r"><i class="fa fa-check"></i><b>3.2</b> R</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#the-console-window"><i class="fa fa-check"></i><b>3.2.1</b> The console window</a></li>
<li class="chapter" data-level="3.2.2" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#your-working-directory"><i class="fa fa-check"></i><b>3.2.2</b> Your working directory</a></li>
<li class="chapter" data-level="3.2.3" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#packages"><i class="fa fa-check"></i><b>3.2.3</b> Packages</a></li>
<li class="chapter" data-level="3.2.4" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#functions"><i class="fa fa-check"></i><b>3.2.4</b> Functions</a></li>
<li class="chapter" data-level="3.2.5" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#assigning-values-to-variables"><i class="fa fa-check"></i><b>3.2.5</b> Assigning values to variables</a></li>
<li class="chapter" data-level="3.2.6" data-path="03-R_DADA2_intro.html"><a href="03-R_DADA2_intro.html#a-few-useful-commands-and-tips"><i class="fa fa-check"></i><b>3.2.6</b> A few useful commands and tips</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>DADA2</b></span></li>
<li class="chapter" data-level="5" data-path="05-Loading_data.html"><a href="05-Loading_data.html"><i class="fa fa-check"></i><b>5</b> Loading data into R and DADA2</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-r"><i class="fa fa-check"></i><b>5.1</b> Loading R</a></li>
<li class="chapter" data-level="5.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-dada2"><i class="fa fa-check"></i><b>5.2</b> Loading DADA2</a></li>
<li class="chapter" data-level="5.3" data-path="05-Loading_data.html"><a href="05-Loading_data.html#loading-our-data"><i class="fa fa-check"></i><b>5.3</b> Loading our data</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="05-Loading_data.html"><a href="05-Loading_data.html#saving-the-paths-for-our-input-and-output-directories"><i class="fa fa-check"></i><b>5.3.1</b> Saving the paths for our input and output directories</a></li>
<li class="chapter" data-level="5.3.2" data-path="05-Loading_data.html"><a href="05-Loading_data.html#inputting-the-forward-and-reverse-reads"><i class="fa fa-check"></i><b>5.3.2</b> Inputting the forward and reverse reads</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-quality_checks.html"><a href="06-quality_checks.html"><i class="fa fa-check"></i><b>6</b> Checking the quality of your data</a></li>
<li class="chapter" data-level="7" data-path="07-clean.html"><a href="07-clean.html"><i class="fa fa-check"></i><b>7</b> Cleaning your data</a></li>
<li class="chapter" data-level="8" data-path="08-ASV_identification.html"><a href="08-ASV_identification.html"><i class="fa fa-check"></i><b>8</b> Identification of ASVs</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-ASV_identification.html"><a href="08-ASV_identification.html#generate-an-error-model"><i class="fa fa-check"></i><b>8.1</b> Generate an error model</a></li>
<li class="chapter" data-level="8.2" data-path="08-ASV_identification.html"><a href="08-ASV_identification.html#dereplication"><i class="fa fa-check"></i><b>8.2</b> Dereplication</a></li>
<li class="chapter" data-level="8.3" data-path="08-ASV_identification.html"><a href="08-ASV_identification.html#inferrence-of-asvs"><i class="fa fa-check"></i><b>8.3</b> Inferrence of ASVs</a></li>
<li class="chapter" data-level="8.4" data-path="08-ASV_identification.html"><a href="08-ASV_identification.html#merging-paired-end-reads"><i class="fa fa-check"></i><b>8.4</b> Merging paired end reads</a></li>
<li class="chapter" data-level="8.5" data-path="08-ASV_identification.html"><a href="08-ASV_identification.html#making-our-asv-matrix"><i class="fa fa-check"></i><b>8.5</b> Making our ASV matrix</a></li>
<li class="chapter" data-level="8.6" data-path="08-ASV_identification.html"><a href="08-ASV_identification.html#chimera-detection-and-removal"><i class="fa fa-check"></i><b>8.6</b> Chimera detection and removal</a></li>
<li class="chapter" data-level="8.7" data-path="08-ASV_identification.html"><a href="08-ASV_identification.html#sequence-tracking-sanity-check"><i class="fa fa-check"></i><b>8.7</b> Sequence tracking sanity check</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Assign_taxonomy.html"><a href="09-Assign_taxonomy.html"><i class="fa fa-check"></i><b>9</b> Assigning taxonomy</a></li>
<li class="part"><span><b>Analysis</b></span></li>
<li class="chapter" data-level="10" data-path="10-Further_analysis.html"><a href="10-Further_analysis.html"><i class="fa fa-check"></i><b>10</b> Further analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Further_analysis.html"><a href="10-Further_analysis.html#r-packages-for-further-analysis"><i class="fa fa-check"></i><b>10.1</b> R packages for further analysis</a></li>
<li class="chapter" data-level="10.2" data-path="10-Further_analysis.html"><a href="10-Further_analysis.html#rarefaction-curves"><i class="fa fa-check"></i><b>10.2</b> Rarefaction curves</a></li>
<li class="chapter" data-level="10.3" data-path="10-Further_analysis.html"><a href="10-Further_analysis.html#alpha-diversity"><i class="fa fa-check"></i><b>10.3</b> Alpha diversity</a></li>
<li class="chapter" data-level="10.4" data-path="10-Further_analysis.html"><a href="10-Further_analysis.html#beta-diversity"><i class="fa fa-check"></i><b>10.4</b> Beta diversity</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Exporting_data.html"><a href="11-Exporting_data.html"><i class="fa fa-check"></i><b>11</b> Exporting data</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Exporting_data.html"><a href="11-Exporting_data.html#fasta-file"><i class="fa fa-check"></i><b>11.1</b> Fasta file</a></li>
<li class="chapter" data-level="11.2" data-path="11-Exporting_data.html"><a href="11-Exporting_data.html#sequence-count-matrix"><i class="fa fa-check"></i><b>11.2</b> Sequence count matrix</a></li>
<li class="chapter" data-level="11.3" data-path="11-Exporting_data.html"><a href="11-Exporting_data.html#table-of-taxon-names"><i class="fa fa-check"></i><b>11.3</b> Table of taxon names</a></li>
</ul></li>
<li class="part"><span><b>Second marker</b></span></li>
<li class="chapter" data-level="12" data-path="12-Summary.html"><a href="12-Summary.html"><i class="fa fa-check"></i><b>12</b> Summary</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Summary.html"><a href="12-Summary.html#neof-training-courses"><i class="fa fa-check"></i><b>12.1</b> NEOF training courses</a></li>
<li class="chapter" data-level="12.2" data-path="12-Summary.html"><a href="12-Summary.html#feedback-and-access-to-example-data"><i class="fa fa-check"></i><b>12.2</b> Feedback and access to example data</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Metabarcoding for diet analysis and environmental DNA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="background" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Background<a href="02-Example_dataset.html#background" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/background.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Advances in sequencing technology mean that it is now feasible to characterise eDNA and diet content through DNA-based approaches by simultaneously sequencing short standardised DNA sequences (DNA barcodes) from a variety of environmental samples, a process known as DNA metabarcoding.</p>
<p>The dataset we will use today to work through our metabarcoding analysis is from a study by <a href="https://onlinelibrary.wiley.com/doi/full/10.1002/edn3.43">Doble <em>et al.</em> 2019</a> on tropical fish communities from Lake Tanganyika.</p>
<p>The study’s authors collected water eDNA samples from sites along the shore of Lake Tanganyika in Tanzania. Samples were then filtered, extracted and made into individual libraries using a two-step PCR method to amplify barcoding genes and attach indexes and Illumina adapters.</p>
<p><img src="figures/Lake_Tang.png" width="60%" style="display: block; margin: auto;" /></p>
<p>Four different primer sets were used to assess the community structure of fishes in Lake Tanganyika. We will examine one of these primer sets today:</p>
<ul>
<li>MiFish‐U (<a href="https://royalsocietypublishing.org/doi/10.1098/rsos.150088">Miya <em>et al.</em> 2015</a>)</li>
</ul>
<p>These primers target a hypervariable region of the mitochondrial 12S rRNA gene. The 12S gene occupies 1/16th of the mitochondrial genome. It is the mitochondrial homologue of the prokaryotic 16S and eukaryotic nuclear 18S ribosomal RNAs.</p>
<p><img src="figures/cichlid.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Sequencing was carried out with an Illumina MiSeq 300 cycle sequencing run (2x150bp paired end sequencing).</p>
<p>The subset of samples you will analyse today were sampled in triplicate from sites along the shore of Lake Tanganyika in 2017. Also included is a filter control, an extraction control and 5 aquarium samples. The aquarium samples contained five cichlid species endemic to the lake.</p>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="01-Metabarcoding.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="03-R_DADA2_intro.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": null,
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
