<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Exporting Artifacts | Bacterial 16S metabarcoding supplemental</title>
  <meta name="description" content="NEOF book for the Introduction to command line bioinformatics workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Exporting Artifacts | Bacterial 16S metabarcoding supplemental" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Introduction to command line bioinformatics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Exporting Artifacts | Bacterial 16S metabarcoding supplemental" />
  
  <meta name="twitter:description" content="NEOF book for the Introduction to command line bioinformatics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-10-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="01-16S_supplemental.html"/>
<link rel="next" href="03-Handy_QIIME2_commands.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>16S supplemental</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-16S_supplemental.html"><a href="01-16S_supplemental.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html"><i class="fa fa-check"></i><b>2</b> Exporting Artifacts</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#biom-files"><i class="fa fa-check"></i><b>2.1</b> BIOM files</a></li>
<li class="chapter" data-level="2.2" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#setup"><i class="fa fa-check"></i><b>2.2</b> Setup</a></li>
<li class="chapter" data-level="2.3" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#abundance-table-export"><i class="fa fa-check"></i><b>2.3</b> Abundance table export</a></li>
<li class="chapter" data-level="2.4" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#taxonomy-info-export"><i class="fa fa-check"></i><b>2.4</b> Taxonomy info export</a></li>
<li class="chapter" data-level="2.5" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#representative-sequences-export"><i class="fa fa-check"></i><b>2.5</b> Representative sequences export</a></li>
<li class="chapter" data-level="2.6" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#phylogenetic-tree-export"><i class="fa fa-check"></i><b>2.6</b> Phylogenetic tree export</a></li>
<li class="chapter" data-level="2.7" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#asv-table-with-taxonomy-export"><i class="fa fa-check"></i><b>2.7</b> ASV table with taxonomy export</a></li>
<li class="chapter" data-level="2.8" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#rarified-table-export"><i class="fa fa-check"></i><b>2.8</b> Rarified table export</a></li>
<li class="chapter" data-level="2.9" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#relative-abundance-table-export"><i class="fa fa-check"></i><b>2.9</b> Relative abundance table export</a></li>
<li class="chapter" data-level="2.10" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#taxonomy-abundance-table-export"><i class="fa fa-check"></i><b>2.10</b> Taxonomy abundance table export</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html"><i class="fa fa-check"></i><b>3</b> Handy QIIME2 commands</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#create-a-manifest-file"><i class="fa fa-check"></i><b>3.1</b> Create a manifest file</a></li>
<li class="chapter" data-level="3.2" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#filter-asvs-from-a-table"><i class="fa fa-check"></i><b>3.2</b> Filter ASVs from a table</a></li>
<li class="chapter" data-level="3.3" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#split-tables"><i class="fa fa-check"></i><b>3.3</b> Split tables</a></li>
<li class="chapter" data-level="3.4" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#merge-tables"><i class="fa fa-check"></i><b>3.4</b> Merge tables</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-16S_R_packages.html"><a href="04-16S_R_packages.html"><i class="fa fa-check"></i><b>4</b> 16S R packages</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-16S_R_packages.html"><a href="04-16S_R_packages.html#qiime2r"><i class="fa fa-check"></i><b>4.1</b> qiime2R</a></li>
<li class="chapter" data-level="4.2" data-path="04-16S_R_packages.html"><a href="04-16S_R_packages.html#phyloseq"><i class="fa fa-check"></i><b>4.2</b> phyloseq</a></li>
<li class="chapter" data-level="4.3" data-path="04-16S_R_packages.html"><a href="04-16S_R_packages.html#microbiome"><i class="fa fa-check"></i><b>4.3</b> microbiome</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial 16S metabarcoding supplemental</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="export" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Exporting Artifacts<a href="02-Exporting_artifacts.html#export" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/artefact_to_tsv.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Artifact files are very useful when carrying out analysis in QIIME2. However they are impossible to directly use (I think) with anything outside of QIIME2. Therefore it can be very useful to know how to export data out of Artifact files.</p>
<p>This section will show how to export various different artifact files into text files. However, for some data types they must be converted into a BIOM file first.</p>
<div id="biom-files" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> BIOM files<a href="02-Exporting_artifacts.html#biom-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/biom.png" width="20%" style="display: block; margin: auto;" />
BIOM stands for BIological Observation Matrix. BIOM files generally contain abundance tables. These files are not human-readable but can be used with certain programs. Here we will only manipulate them in such ways to get the text files we need.</p>
<p>For more information on the BIOM format please see: <a href="https://biom-format.org/" class="uri">https://biom-format.org/</a></p>
<p>We will be using the command <code>biom convert</code> to convert BIOM files to TSV (Tab Separated Value) files. For more info on this please see: <a href="https://biom-format.org/documentation/biom_conversion.html" class="uri">https://biom-format.org/documentation/biom_conversion.html</a></p>
</div>
<div id="setup" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Setup<a href="02-Exporting_artifacts.html#setup" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/setup.png" width="15%" style="display: block; margin: auto;" />
First copy a new directory that will contain all the QIIME2 output you need and then change directory to it.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="02-Exporting_artifacts.html#cb2-1" tabindex="-1"></a><span class="fu">cp</span> <span class="at">-r</span> /pub14/tea/nsc206/NEOF/16s_workshop/16s_export ~/Metagenetics</span>
<span id="cb2-2"><a href="02-Exporting_artifacts.html#cb2-2" tabindex="-1"></a><span class="bu">cd</span> ~/Metagenetics/16s_export</span></code></pre></div>
<p>Before we start exporting it is best to make a few directories.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="02-Exporting_artifacts.html#cb3-1" tabindex="-1"></a><span class="fu">mkdir</span> export</span>
<span id="cb3-2"><a href="02-Exporting_artifacts.html#cb3-2" tabindex="-1"></a><span class="fu">mkdir</span> qiime2_exports</span>
<span id="cb3-3"><a href="02-Exporting_artifacts.html#cb3-3" tabindex="-1"></a><span class="fu">mkdir</span> final_exports</span></code></pre></div>
<ul>
<li><code>export</code>: This directory is where we will point the various <code>qiime tools export</code> commands. You can only specify an output directory for this command and the command will give the exported file very generic names. This makes it very easy to overwrite files when using this command.</li>
<li><code>qiime2_exports</code>: For some of the exported files we will move and rename them from <code>export</code> to this directory. This is an temporary directory for files we will not want at the end (mostly BIOM files).</li>
<li><code>final_exports</code>: This directory will contain the final exported files we would generaly want to keep.</li>
</ul>
</div>
<div id="abundance-table-export" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Abundance table export<a href="02-Exporting_artifacts.html#abundance-table-export" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/noun_Table_250445_000000.png" width="10%" style="display: block; margin: auto;" /></p>
<p>First we will export our ASV abundance table. We will have to convert the BIOM file produced by QIIME2 to a TSV file.</p>
<p>Export the table to a directory. This will create a BIOM file called <code>feature-table.biom</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="02-Exporting_artifacts.html#cb4-1" tabindex="-1"></a><span class="ex">qiime</span> tools export <span class="at">--input-path</span> table-dada2.qza <span class="at">--output-path</span> export</span></code></pre></div>
<p>Move the created file whilst renaming it.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="02-Exporting_artifacts.html#cb5-1" tabindex="-1"></a><span class="fu">mv</span> export/feature-table.biom qiime2_exports/ASV_table.biom</span></code></pre></div>
<p>Finall we will use <code>biom convert</code> to create our TSV file.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="02-Exporting_artifacts.html#cb6-1" tabindex="-1"></a><span class="ex">biom</span> convert <span class="dt">\</span></span>
<span id="cb6-2"><a href="02-Exporting_artifacts.html#cb6-2" tabindex="-1"></a>-i qiime2_exports/ASV_table.biom <span class="dt">\</span></span>
<span id="cb6-3"><a href="02-Exporting_artifacts.html#cb6-3" tabindex="-1"></a>-o final_exports/ASV_table.tsv <span class="dt">\</span></span>
<span id="cb6-4"><a href="02-Exporting_artifacts.html#cb6-4" tabindex="-1"></a>--to-tsv</span></code></pre></div>
<p>The option <code>--to-tsv</code> (or alternative) must be provided or the command will not run and only complain.</p>
<p>Use a text viewer/editor of your choice to inspect the final file. Make sure to do this for all the files in this tutorial.</p>
</div>
<div id="taxonomy-info-export" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Taxonomy info export<a href="02-Exporting_artifacts.html#taxonomy-info-export" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/noun_ranking_165760.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Exporting the taxonomy info.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="02-Exporting_artifacts.html#cb7-1" tabindex="-1"></a><span class="ex">qiime</span> tools export <span class="at">--input-path</span> taxonomy.sklearn.qza <span class="dt">\</span></span>
<span id="cb7-2"><a href="02-Exporting_artifacts.html#cb7-2" tabindex="-1"></a>--output-path export</span>
<span id="cb7-3"><a href="02-Exporting_artifacts.html#cb7-3" tabindex="-1"></a><span class="co">#Move the produced tsv file to the final_exports directory</span></span>
<span id="cb7-4"><a href="02-Exporting_artifacts.html#cb7-4" tabindex="-1"></a><span class="fu">mv</span> export/taxonomy.tsv final_exports</span></code></pre></div>
</div>
<div id="representative-sequences-export" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Representative sequences export<a href="02-Exporting_artifacts.html#representative-sequences-export" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/noun_DNA_947999.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Export the representative sequences to a fasta file.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="02-Exporting_artifacts.html#cb8-1" tabindex="-1"></a><span class="ex">qiime</span> tools export <span class="at">--input-path</span> rep-seqs-dada2.qza <span class="dt">\</span></span>
<span id="cb8-2"><a href="02-Exporting_artifacts.html#cb8-2" tabindex="-1"></a>--output-path export</span>
<span id="cb8-3"><a href="02-Exporting_artifacts.html#cb8-3" tabindex="-1"></a><span class="co">#Move and rename the produced fasta file</span></span>
<span id="cb8-4"><a href="02-Exporting_artifacts.html#cb8-4" tabindex="-1"></a><span class="fu">mv</span> export/dna-sequences.fasta final_exports/ASV_rep_seqs.fasta</span></code></pre></div>
</div>
<div id="phylogenetic-tree-export" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Phylogenetic tree export<a href="02-Exporting_artifacts.html#phylogenetic-tree-export" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/noun_Tree_1857655.png" width="10%" style="display: block; margin: auto;" /></p>
<p>The below exports the specified phylogenetic tree into a newick format file. For more info on the Newick format please see: <a href="https://en.wikipedia.org/wiki/Newick_format" class="uri">https://en.wikipedia.org/wiki/Newick_format</a></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="02-Exporting_artifacts.html#cb9-1" tabindex="-1"></a><span class="ex">qiime</span> tools export <span class="at">--input-path</span> rooted-tree.qza <span class="dt">\</span></span>
<span id="cb9-2"><a href="02-Exporting_artifacts.html#cb9-2" tabindex="-1"></a>--output-path export</span>
<span id="cb9-3"><a href="02-Exporting_artifacts.html#cb9-3" tabindex="-1"></a><span class="co">#Move and rename the exported newick file</span></span>
<span id="cb9-4"><a href="02-Exporting_artifacts.html#cb9-4" tabindex="-1"></a><span class="fu">mv</span> export/tree.nwk final_exports/fasttree_rooted_tree.nwk</span></code></pre></div>
<p>If you would like to view the tree you can open <code>firefox</code> and go to __“<a href="https://itol.embl.de%22__" class="uri">https://itol.embl.de"__</a>. Then you can upload the <code>.nwk</code> to quickly view it.</p>
</div>
<div id="asv-table-with-taxonomy-export" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> ASV table with taxonomy export<a href="02-Exporting_artifacts.html#asv-table-with-taxonomy-export" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/table_w_tax.png" width="25%" style="display: block; margin: auto;" /></p>
<p>This step will produce a TSV file containing the ASV abundance table with the last column containing the taxonomy assignment of the ASV.</p>
<p>To carry out this step you will need the BIOM format produced in the abundance table export and the exported taxonomy file.</p>
<p>We need to change the header of the taxonomy file for it to be compatible with the biom command.</p>
<p>First create a copy which we will edit</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="02-Exporting_artifacts.html#cb10-1" tabindex="-1"></a><span class="fu">cp</span> final_exports/taxonomy.tsv final_exports/taxonomy.header.tsv</span></code></pre></div>
<p>Then using a text editor of your choice (nano, vim, emacs, etc.) change the first and second column headers to <code>#OTUID</code> and <code>taxonomy</code>.</p>
<p>Now we can create a new BIOM file which contains the taxonomy information</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="02-Exporting_artifacts.html#cb11-1" tabindex="-1"></a><span class="ex">biom</span> add-metadata <span class="dt">\</span></span>
<span id="cb11-2"><a href="02-Exporting_artifacts.html#cb11-2" tabindex="-1"></a>--input-fp qiime2_exports/ASV_table.biom <span class="dt">\</span></span>
<span id="cb11-3"><a href="02-Exporting_artifacts.html#cb11-3" tabindex="-1"></a>--observation-metadata-fp final_exports/taxonomy.header.tsv <span class="dt">\</span></span>
<span id="cb11-4"><a href="02-Exporting_artifacts.html#cb11-4" tabindex="-1"></a>--output-fp qiime2_exports/ASV_table.tax.biom</span></code></pre></div>
<p>Export the newly created BIOM file. We will need to include option <code>--header-key taxonomy</code> to include the taxonomy info in the newly TSV file.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="02-Exporting_artifacts.html#cb12-1" tabindex="-1"></a><span class="ex">biom</span> convert <span class="dt">\</span></span>
<span id="cb12-2"><a href="02-Exporting_artifacts.html#cb12-2" tabindex="-1"></a>-i qiime2_exports/ASV_table.tax.biom <span class="dt">\</span></span>
<span id="cb12-3"><a href="02-Exporting_artifacts.html#cb12-3" tabindex="-1"></a>-o final_exports/ASV_table.tax.tsv <span class="dt">\</span></span>
<span id="cb12-4"><a href="02-Exporting_artifacts.html#cb12-4" tabindex="-1"></a>--header-key taxonomy <span class="dt">\</span></span>
<span id="cb12-5"><a href="02-Exporting_artifacts.html#cb12-5" tabindex="-1"></a>--to-tsv</span></code></pre></div>
</div>
<div id="rarified-table-export" class="section level2 hasAnchor" number="2.8">
<h2><span class="header-section-number">2.8</span> Rarified table export<a href="02-Exporting_artifacts.html#rarified-table-export" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/noun_curve_3542841.png" width="10%" style="display: block; margin: auto;" /></p>
<p>You can export a rarefied table. The first step of this is to rarefy our ASV table artifact.</p>
<p>The option <code>--p-sampling-depth</code> is our rarefaction threshold. We will choose 20,000 for this example but make sure to pick an appropriate one for your own projects.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="02-Exporting_artifacts.html#cb13-1" tabindex="-1"></a><span class="ex">qiime</span> feature-table rarefy <span class="dt">\</span></span>
<span id="cb13-2"><a href="02-Exporting_artifacts.html#cb13-2" tabindex="-1"></a>--i-table table-dada2.qza <span class="dt">\</span></span>
<span id="cb13-3"><a href="02-Exporting_artifacts.html#cb13-3" tabindex="-1"></a>--p-sampling-depth 20000 <span class="dt">\</span></span>
<span id="cb13-4"><a href="02-Exporting_artifacts.html#cb13-4" tabindex="-1"></a>--o-rarefied-table table-dada2.rarefied_20000.qza</span></code></pre></div>
<p>After that it is the normal export and convert commands. To be fancy we will include the taxonomy information in the final file.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="02-Exporting_artifacts.html#cb14-1" tabindex="-1"></a><span class="co">#Qiime tools export</span></span>
<span id="cb14-2"><a href="02-Exporting_artifacts.html#cb14-2" tabindex="-1"></a><span class="ex">qiime</span> tools export <span class="dt">\</span></span>
<span id="cb14-3"><a href="02-Exporting_artifacts.html#cb14-3" tabindex="-1"></a>--input-path table-dada2.rarefied_20000.qza <span class="dt">\</span></span>
<span id="cb14-4"><a href="02-Exporting_artifacts.html#cb14-4" tabindex="-1"></a>--output-path export</span>
<span id="cb14-5"><a href="02-Exporting_artifacts.html#cb14-5" tabindex="-1"></a><span class="co">#Move and rename exported biom</span></span>
<span id="cb14-6"><a href="02-Exporting_artifacts.html#cb14-6" tabindex="-1"></a><span class="fu">mv</span> export/feature-table.biom qiime2_exports/ASV_table.rarefied_20000.biom</span>
<span id="cb14-7"><a href="02-Exporting_artifacts.html#cb14-7" tabindex="-1"></a><span class="co">#Add taxonomy info to biom file</span></span>
<span id="cb14-8"><a href="02-Exporting_artifacts.html#cb14-8" tabindex="-1"></a><span class="co">#Ensure to use the taxonomy with the altered headers</span></span>
<span id="cb14-9"><a href="02-Exporting_artifacts.html#cb14-9" tabindex="-1"></a><span class="ex">biom</span> add-metadata <span class="dt">\</span></span>
<span id="cb14-10"><a href="02-Exporting_artifacts.html#cb14-10" tabindex="-1"></a>--input-fp qiime2_exports/ASV_table.rarefied_20000.biom <span class="dt">\</span></span>
<span id="cb14-11"><a href="02-Exporting_artifacts.html#cb14-11" tabindex="-1"></a>--observation-metadata-fp final_exports/taxonomy.header.tsv <span class="dt">\</span></span>
<span id="cb14-12"><a href="02-Exporting_artifacts.html#cb14-12" tabindex="-1"></a>--output-fp qiime2_exports/ASV_table.rarefied_20000.tax.biom</span>
<span id="cb14-13"><a href="02-Exporting_artifacts.html#cb14-13" tabindex="-1"></a><span class="co">#Convert biom to tsv with taxonomy</span></span>
<span id="cb14-14"><a href="02-Exporting_artifacts.html#cb14-14" tabindex="-1"></a><span class="ex">biom</span> convert <span class="dt">\</span></span>
<span id="cb14-15"><a href="02-Exporting_artifacts.html#cb14-15" tabindex="-1"></a>-i qiime2_exports/ASV_table.rarefied_20000.tax.biom <span class="dt">\</span></span>
<span id="cb14-16"><a href="02-Exporting_artifacts.html#cb14-16" tabindex="-1"></a>-o final_exports/ASV_table.rarefied_20000.tax.tsv <span class="dt">\</span></span>
<span id="cb14-17"><a href="02-Exporting_artifacts.html#cb14-17" tabindex="-1"></a>--header-key taxonomy <span class="dt">\</span></span>
<span id="cb14-18"><a href="02-Exporting_artifacts.html#cb14-18" tabindex="-1"></a>--to-tsv</span></code></pre></div>
</div>
<div id="relative-abundance-table-export" class="section level2 hasAnchor" number="2.9">
<h2><span class="header-section-number">2.9</span> Relative abundance table export<a href="02-Exporting_artifacts.html#relative-abundance-table-export" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/noun_Percent_3560962.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Instead of an abundance table you may want a relative abundance table. Thankfully this can easily be done in QIIME 2 with the following command.</p>
<p><strong>Note</strong>: You will most likely want to get the relative abundance from a rarefied table.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="02-Exporting_artifacts.html#cb15-1" tabindex="-1"></a><span class="ex">qiime</span> feature-table relative-frequency <span class="dt">\</span></span>
<span id="cb15-2"><a href="02-Exporting_artifacts.html#cb15-2" tabindex="-1"></a>--i-table table-dada2.rarefied_20000.qza <span class="dt">\</span></span>
<span id="cb15-3"><a href="02-Exporting_artifacts.html#cb15-3" tabindex="-1"></a>--o-relative-frequency-table table-dada2.rarefied_20000.relabund.qza</span></code></pre></div>
<p>Then you can export it the same way as the rarefied abundance table.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="02-Exporting_artifacts.html#cb16-1" tabindex="-1"></a><span class="co">#Qiime tools export</span></span>
<span id="cb16-2"><a href="02-Exporting_artifacts.html#cb16-2" tabindex="-1"></a><span class="ex">qiime</span> tools export <span class="dt">\</span></span>
<span id="cb16-3"><a href="02-Exporting_artifacts.html#cb16-3" tabindex="-1"></a>--input-path table-dada2.rarefied_20000.relabund.qza <span class="dt">\</span></span>
<span id="cb16-4"><a href="02-Exporting_artifacts.html#cb16-4" tabindex="-1"></a>--output-path export</span>
<span id="cb16-5"><a href="02-Exporting_artifacts.html#cb16-5" tabindex="-1"></a><span class="co">#Move and rename exported biom</span></span>
<span id="cb16-6"><a href="02-Exporting_artifacts.html#cb16-6" tabindex="-1"></a><span class="fu">mv</span> export/feature-table.biom <span class="dt">\</span></span>
<span id="cb16-7"><a href="02-Exporting_artifacts.html#cb16-7" tabindex="-1"></a>qiime2_exports/ASV_table.rarefied_20000.relabund.biom</span>
<span id="cb16-8"><a href="02-Exporting_artifacts.html#cb16-8" tabindex="-1"></a><span class="co">#Add taxonomy info to biom file</span></span>
<span id="cb16-9"><a href="02-Exporting_artifacts.html#cb16-9" tabindex="-1"></a><span class="ex">biom</span> add-metadata <span class="dt">\</span></span>
<span id="cb16-10"><a href="02-Exporting_artifacts.html#cb16-10" tabindex="-1"></a>--input-fp qiime2_exports/ASV_table.rarefied_20000.relabund.biom <span class="dt">\</span></span>
<span id="cb16-11"><a href="02-Exporting_artifacts.html#cb16-11" tabindex="-1"></a>--observation-metadata-fp final_exports/taxonomy.header.tsv <span class="dt">\</span></span>
<span id="cb16-12"><a href="02-Exporting_artifacts.html#cb16-12" tabindex="-1"></a>--output-fp qiime2_exports/ASV_table.rarefied_20000.relabund.tax.biom</span>
<span id="cb16-13"><a href="02-Exporting_artifacts.html#cb16-13" tabindex="-1"></a><span class="co">#Convert biom to tsv with taxonomy</span></span>
<span id="cb16-14"><a href="02-Exporting_artifacts.html#cb16-14" tabindex="-1"></a><span class="ex">biom</span> convert <span class="dt">\</span></span>
<span id="cb16-15"><a href="02-Exporting_artifacts.html#cb16-15" tabindex="-1"></a>-i qiime2_exports/ASV_table.rarefied_20000.relabund.tax.biom <span class="dt">\</span></span>
<span id="cb16-16"><a href="02-Exporting_artifacts.html#cb16-16" tabindex="-1"></a>-o final_exports/ASV_table.rarefied_20000.relabund.tax.tsv <span class="dt">\</span></span>
<span id="cb16-17"><a href="02-Exporting_artifacts.html#cb16-17" tabindex="-1"></a>--header-key taxonomy <span class="dt">\</span></span>
<span id="cb16-18"><a href="02-Exporting_artifacts.html#cb16-18" tabindex="-1"></a>--to-tsv</span></code></pre></div>
</div>
<div id="taxonomy-abundance-table-export" class="section level2 hasAnchor" number="2.10">
<h2><span class="header-section-number">2.10</span> Taxonomy abundance table export<a href="02-Exporting_artifacts.html#taxonomy-abundance-table-export" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/noun_Family_137805.png" width="10%" style="display: block; margin: auto;" /></p>
<p>The last recipe I will show you is how to create and then export a taxonomy abundance table.</p>
<p>First we collapse our ASV table to a specific taxonomy level. This is carried out by specifying a number to the option <code>--p-level</code> in the command below. With Silva and Greengenes2 the numbers normally correspond to:</p>
<ol style="list-style-type: decimal">
<li>Kingdom</li>
<li>Phylum</li>
<li>Class</li>
<li>Order</li>
<li>Family</li>
<li>Genus</li>
<li>Species</li>
</ol>
<p>Be careful though as with some taxonomy databases the levels are not consistent between different organisms. This occurs if sub-level information (such as sub-order, sub-family etc.) is included in some but not all taxonomic classifications. This could mean that the 5th level taxonomy of one organisms could be its sub-order whilst another’s could be its family. Check your <code>export/taxonomy.tsv</code> to see if this could be an issue or not.</p>
<p>We will now run our taxonomy collapse command on our rarefied table to produce a Family abundance table.</p>
<p><strong>Note</strong>: The below command can not be run on a relative abundance table. You will have to run the relative frequency command on its output if you want relative abundance family table.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="02-Exporting_artifacts.html#cb17-1" tabindex="-1"></a><span class="ex">qiime</span> taxa collapse <span class="dt">\</span></span>
<span id="cb17-2"><a href="02-Exporting_artifacts.html#cb17-2" tabindex="-1"></a>--i-table table-dada2.rarefied_20000.qza <span class="dt">\</span></span>
<span id="cb17-3"><a href="02-Exporting_artifacts.html#cb17-3" tabindex="-1"></a>--i-taxonomy taxonomy.sklearn.qza <span class="dt">\</span></span>
<span id="cb17-4"><a href="02-Exporting_artifacts.html#cb17-4" tabindex="-1"></a>--p-level 5 <span class="dt">\</span></span>
<span id="cb17-5"><a href="02-Exporting_artifacts.html#cb17-5" tabindex="-1"></a>--o-collapsed-table table-dada2-family.rarefied_20000.qza</span></code></pre></div>
<p>Then you can export it in a similar way as the other tables. However, we should not include the step that adds the taxonomy for this.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="02-Exporting_artifacts.html#cb18-1" tabindex="-1"></a><span class="co">#Qiime tools export</span></span>
<span id="cb18-2"><a href="02-Exporting_artifacts.html#cb18-2" tabindex="-1"></a><span class="ex">qiime</span> tools export <span class="dt">\</span></span>
<span id="cb18-3"><a href="02-Exporting_artifacts.html#cb18-3" tabindex="-1"></a>--input-path table-dada2-family.rarefied_20000.qza <span class="dt">\</span></span>
<span id="cb18-4"><a href="02-Exporting_artifacts.html#cb18-4" tabindex="-1"></a>--output-path export</span>
<span id="cb18-5"><a href="02-Exporting_artifacts.html#cb18-5" tabindex="-1"></a><span class="co">#Move and rename exported biom</span></span>
<span id="cb18-6"><a href="02-Exporting_artifacts.html#cb18-6" tabindex="-1"></a><span class="fu">mv</span> export/feature-table.biom qiime2_exports/family_table.rarefied_20000.biom</span>
<span id="cb18-7"><a href="02-Exporting_artifacts.html#cb18-7" tabindex="-1"></a><span class="co">#Convert biom to tsv with taxonomy</span></span>
<span id="cb18-8"><a href="02-Exporting_artifacts.html#cb18-8" tabindex="-1"></a><span class="ex">biom</span> convert <span class="dt">\</span></span>
<span id="cb18-9"><a href="02-Exporting_artifacts.html#cb18-9" tabindex="-1"></a>-i qiime2_exports/family_table.rarefied_20000.biom <span class="dt">\</span></span>
<span id="cb18-10"><a href="02-Exporting_artifacts.html#cb18-10" tabindex="-1"></a>-o final_exports/family_table.rarefied_20000.tsv <span class="dt">\</span></span>
<span id="cb18-11"><a href="02-Exporting_artifacts.html#cb18-11" tabindex="-1"></a>--to-tsv</span></code></pre></div>
<p>I hope this is a nice quick reference for your future use.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="01-16S_supplemental.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="03-Handy_QIIME2_commands.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
