<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 16S R packages | Bacterial 16S metabarcoding supplemental</title>
  <meta name="description" content="NEOF book for the Introduction to command line bioinformatics workshop" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 16S R packages | Bacterial 16S metabarcoding supplemental" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Introduction to command line bioinformatics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 16S R packages | Bacterial 16S metabarcoding supplemental" />
  
  <meta name="twitter:description" content="NEOF book for the Introduction to command line bioinformatics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2023-10-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="03-Handy_QIIME2_commands.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>16S supplemental</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-16S_supplemental.html"><a href="01-16S_supplemental.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html"><i class="fa fa-check"></i><b>2</b> Exporting Artifacts</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#biom-files"><i class="fa fa-check"></i><b>2.1</b> BIOM files</a></li>
<li class="chapter" data-level="2.2" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#setup"><i class="fa fa-check"></i><b>2.2</b> Setup</a></li>
<li class="chapter" data-level="2.3" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#abundance-table-export"><i class="fa fa-check"></i><b>2.3</b> Abundance table export</a></li>
<li class="chapter" data-level="2.4" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#taxonomy-info-export"><i class="fa fa-check"></i><b>2.4</b> Taxonomy info export</a></li>
<li class="chapter" data-level="2.5" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#representative-sequences-export"><i class="fa fa-check"></i><b>2.5</b> Representative sequences export</a></li>
<li class="chapter" data-level="2.6" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#phylogenetic-tree-export"><i class="fa fa-check"></i><b>2.6</b> Phylogenetic tree export</a></li>
<li class="chapter" data-level="2.7" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#asv-table-with-taxonomy-export"><i class="fa fa-check"></i><b>2.7</b> ASV table with taxonomy export</a></li>
<li class="chapter" data-level="2.8" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#rarified-table-export"><i class="fa fa-check"></i><b>2.8</b> Rarified table export</a></li>
<li class="chapter" data-level="2.9" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#relative-abundance-table-export"><i class="fa fa-check"></i><b>2.9</b> Relative abundance table export</a></li>
<li class="chapter" data-level="2.10" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#taxonomy-abundance-table-export"><i class="fa fa-check"></i><b>2.10</b> Taxonomy abundance table export</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html"><i class="fa fa-check"></i><b>3</b> Handy QIIME2 commands</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#create-a-manifest-file"><i class="fa fa-check"></i><b>3.1</b> Create a manifest file</a></li>
<li class="chapter" data-level="3.2" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#filter-asvs-from-a-table"><i class="fa fa-check"></i><b>3.2</b> Filter ASVs from a table</a></li>
<li class="chapter" data-level="3.3" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#split-tables"><i class="fa fa-check"></i><b>3.3</b> Split tables</a></li>
<li class="chapter" data-level="3.4" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#merge-tables"><i class="fa fa-check"></i><b>3.4</b> Merge tables</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-16S_R_packages.html"><a href="04-16S_R_packages.html"><i class="fa fa-check"></i><b>4</b> 16S R packages</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-16S_R_packages.html"><a href="04-16S_R_packages.html#qiime2r"><i class="fa fa-check"></i><b>4.1</b> qiime2R</a></li>
<li class="chapter" data-level="4.2" data-path="04-16S_R_packages.html"><a href="04-16S_R_packages.html#phyloseq"><i class="fa fa-check"></i><b>4.2</b> phyloseq</a></li>
<li class="chapter" data-level="4.3" data-path="04-16S_R_packages.html"><a href="04-16S_R_packages.html#microbiome"><i class="fa fa-check"></i><b>4.3</b> microbiome</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial 16S metabarcoding supplemental</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rpackages" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> 16S R packages<a href="04-16S_R_packages.html#rpackages" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/R.png" width="20%" style="display: block; margin: auto;" /></p>
<p>There are various R packages that can be used to analyse 16S data. We have a workshop on how to use the below tools, check our <a href="https://neof.org.uk/training/">training page</a> to see if it open for registration. Generally these will take the output from QIIME2 so you can carry out further downstream analysis on the QIIME2 created ASV table, taxonomy, and phylogenetic tree.</p>
<p>I would recommend only those with a good foundation in R to use these packages. If you do not have this I would recommend trying to gain it.</p>
<div id="qiime2r" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> qiime2R<a href="04-16S_R_packages.html#qiime2r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/qiime2r.png" width="40%" style="display: block; margin: auto;" /></p>
<p><a href="https://github.com/jbisanz/qiime2R" class="uri">https://github.com/jbisanz/qiime2R</a></p>
<p>qiime2R is an R package to import QIIME2 artifacts into R compatible files. It can convert artifacts into data structures that are useable in R. Additionally it can also create some plots.</p>
<p>One of its most useful commands is <code>qza_to_phyloseq()</code> which converts a .qza artifact into the phyloseq data structure.</p>
</div>
<div id="phyloseq" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> phyloseq<a href="04-16S_R_packages.html#phyloseq" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/phyloseq_logo.png" width="15%" style="display: block; margin: auto;" /></p>
<p><a href="https://joey711.github.io/phyloseq/" class="uri">https://joey711.github.io/phyloseq/</a></p>
<p>phyloseq is a very powerful R package to carry out microbiome profile analysis. It is a great way to produce custom plots for alpha and beta diversity, heatmaps, network plotting, etc. It can be used to analyse any type of community based data such as shotgun metagenomic data.</p>
<p>There are two large caveats:</p>
<ol style="list-style-type: decimal">
<li>It can be quite difficult and laborious to convert data to the required phyloseq R object (a data structure created for phyloseq). qiime2R does make this much easier for 16S data though.</li>
<li>The documentation can be quite hard to follow and assumes you already have a lot of specific knowledge. It uses ggplot2 coding conventions so it is essential you learn the R package “ggplot2” prior to trying phyloseq.</li>
</ol>
</div>
<div id="microbiome" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> microbiome<a href="04-16S_R_packages.html#microbiome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://microbiome.github.io/tutorials/" class="uri">https://microbiome.github.io/tutorials/</a></p>
<p>Once you start using phyloseq you will probably notice there are many commands missing you may require. Thankfully the microbiome R package is a very good supplemental package to phyloseq.</p>
<p>This package supports the phyloseq data format and acts to expand the available toolkit to analyse this data format. I would highly recommend using it if you start using phyloseq.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="03-Handy_QIIME2_commands.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
