<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Other QIIME2 resources | Bacterial 16S metabarcoding supplemental</title>
  <meta name="description" content="NEOF book for the Introduction to command line bioinformatics workshop" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Other QIIME2 resources | Bacterial 16S metabarcoding supplemental" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Introduction to command line bioinformatics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Other QIIME2 resources | Bacterial 16S metabarcoding supplemental" />
  
  <meta name="twitter:description" content="NEOF book for the Introduction to command line bioinformatics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2024-12-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="03-Handy_QIIME2_commands.html"/>
<link rel="next" href="05-16S_R_packages.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_rounded_corners.png" width="250"></a></li>
<li><a>16S supplemental</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-16S_supplemental.html"><a href="01-16S_supplemental.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html"><i class="fa fa-check"></i><b>2</b> Exporting Artifacts</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#biom-files"><i class="fa fa-check"></i><b>2.1</b> BIOM files</a></li>
<li class="chapter" data-level="2.2" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#setup"><i class="fa fa-check"></i><b>2.2</b> Setup</a></li>
<li class="chapter" data-level="2.3" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#abundance-table-export"><i class="fa fa-check"></i><b>2.3</b> Abundance table export</a></li>
<li class="chapter" data-level="2.4" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#taxonomy-info-export"><i class="fa fa-check"></i><b>2.4</b> Taxonomy info export</a></li>
<li class="chapter" data-level="2.5" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#representative-sequences-export"><i class="fa fa-check"></i><b>2.5</b> Representative sequences export</a></li>
<li class="chapter" data-level="2.6" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#phylogenetic-tree-export"><i class="fa fa-check"></i><b>2.6</b> Phylogenetic tree export</a></li>
<li class="chapter" data-level="2.7" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#asv-table-with-taxonomy-export"><i class="fa fa-check"></i><b>2.7</b> ASV table with taxonomy export</a></li>
<li class="chapter" data-level="2.8" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#rarified-table-export"><i class="fa fa-check"></i><b>2.8</b> Rarified table export</a></li>
<li class="chapter" data-level="2.9" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#relative-abundance-table-export"><i class="fa fa-check"></i><b>2.9</b> Relative abundance table export</a></li>
<li class="chapter" data-level="2.10" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#taxonomy-abundance-table-export"><i class="fa fa-check"></i><b>2.10</b> Taxonomy abundance table export</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html"><i class="fa fa-check"></i><b>3</b> Handy QIIME2 commands</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#create-a-manifest-file"><i class="fa fa-check"></i><b>3.1</b> Create a manifest file</a></li>
<li class="chapter" data-level="3.2" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#filter-asvs-from-a-table"><i class="fa fa-check"></i><b>3.2</b> Filter ASVs from a table</a></li>
<li class="chapter" data-level="3.3" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#split-tables"><i class="fa fa-check"></i><b>3.3</b> Split tables</a></li>
<li class="chapter" data-level="3.4" data-path="03-Handy_QIIME2_commands.html"><a href="03-Handy_QIIME2_commands.html#merge-tables"><i class="fa fa-check"></i><b>3.4</b> Merge tables</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-QIIME2_resources.html"><a href="04-QIIME2_resources.html"><i class="fa fa-check"></i><b>4</b> Other QIIME2 resources</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-QIIME2_resources.html"><a href="04-QIIME2_resources.html#qiime2-plugins"><i class="fa fa-check"></i><b>4.1</b> QIIME2 Plugins</a></li>
<li class="chapter" data-level="4.2" data-path="04-QIIME2_resources.html"><a href="04-QIIME2_resources.html#qiime2-tutorials"><i class="fa fa-check"></i><b>4.2</b> QIIME2 tutorials</a></li>
<li class="chapter" data-level="4.3" data-path="04-QIIME2_resources.html"><a href="04-QIIME2_resources.html#fungal-its"><i class="fa fa-check"></i><b>4.3</b> Fungal ITS</a></li>
<li class="chapter" data-level="4.4" data-path="04-QIIME2_resources.html"><a href="04-QIIME2_resources.html#longitudinal-analysis"><i class="fa fa-check"></i><b>4.4</b> Longitudinal analysis</a></li>
<li class="chapter" data-level="4.5" data-path="04-QIIME2_resources.html"><a href="04-QIIME2_resources.html#iterative-rarefaction"><i class="fa fa-check"></i><b>4.5</b> Iterative rarefaction</a></li>
<li class="chapter" data-level="4.6" data-path="04-QIIME2_resources.html"><a href="04-QIIME2_resources.html#managaing-reference-databases"><i class="fa fa-check"></i><b>4.6</b> Managaing reference databases</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-16S_R_packages.html"><a href="05-16S_R_packages.html"><i class="fa fa-check"></i><b>5</b> 16S R packages</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-16S_R_packages.html"><a href="05-16S_R_packages.html#qiime2r"><i class="fa fa-check"></i><b>5.1</b> qiime2R</a></li>
<li class="chapter" data-level="5.2" data-path="05-16S_R_packages.html"><a href="05-16S_R_packages.html#phyloseq"><i class="fa fa-check"></i><b>5.2</b> phyloseq</a></li>
<li class="chapter" data-level="5.3" data-path="05-16S_R_packages.html"><a href="05-16S_R_packages.html#microbiome"><i class="fa fa-check"></i><b>5.3</b> microbiome</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial 16S metabarcoding supplemental</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="qiime2_resources" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Other QIIME2 resources<a href="04-QIIME2_resources.html#qiime2_resources" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<center>
<img src="figures/resources.png" style="width:200px; background:white; border-radius:5px" />
</center>
<div id="qiime2-plugins" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> QIIME2 Plugins<a href="04-QIIME2_resources.html#qiime2-plugins" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/qiime2_plugin.png" style="width:200px; background:white; border-radius:5px" />
</center>
<p>QIIME2 commands are grouped into “Plugins”.
All the base QIIME2 PLugins are listed in the following page.</p>
<p><a href="https://docs.qiime2.org/2024.10/plugins/" class="uri">https://docs.qiime2.org/2024.10/plugins/</a></p>
<p>Each Plugin can have a variety of “Pipelines”, “Methods”, and “Visualizers</p>
<p>An example is the <code>diversity</code> plugin. Its page is below:</p>
<p><a href="https://docs.qiime2.org/2024.10/plugins/available/diversity/" class="uri">https://docs.qiime2.org/2024.10/plugins/available/diversity/</a></p>
</div>
<div id="qiime2-tutorials" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> QIIME2 tutorials<a href="04-QIIME2_resources.html#qiime2-tutorials" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/computer_tutorial.png" style="width:200px; background:white; border-radius:5px" />
</center>
<p>There are many QIIME2 tutorials online you can attempt.
It is always good to practice with other data sets.</p>
<ul>
<li>QIIME2 Tutorials main page: <a href="https://docs.qiime2.org/2024.10/tutorials/" class="uri">https://docs.qiime2.org/2024.10/tutorials/</a></li>
<li>“Moving Pictures”: <a href="https://docs.qiime2.org/2024.10/tutorials/moving-pictures/" class="uri">https://docs.qiime2.org/2024.10/tutorials/moving-pictures/</a></li>
<li>Fecal microbiota transplant (FMT) study: <a href="https://docs.qiime2.org/2024.10/tutorials/fmt/" class="uri">https://docs.qiime2.org/2024.10/tutorials/fmt/</a></li>
<li>“Atacama soil microbiome” tutorial: <a href="https://docs.qiime2.org/2024.10/tutorials/atacama-soils/" class="uri">https://docs.qiime2.org/2024.10/tutorials/atacama-soils/</a></li>
<li>Parkinson’s Mouse Tutorial: <a href="https://docs.qiime2.org/2024.10/tutorials/pd-mice/" class="uri">https://docs.qiime2.org/2024.10/tutorials/pd-mice/</a></li>
</ul>
</div>
<div id="fungal-its" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Fungal ITS<a href="04-QIIME2_resources.html#fungal-its" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/fungal_its.png" style="width:200px" />
</center>
<p>Carrying out analysis of Fungal ITS data has a lot of similar steps as 16S analysis.
However there are some key changes.
Below is a link for a tutorial on ITS analysis:</p>
<p><a href="https://forum.qiime2.org/t/q2-itsxpress-a-tutorial-on-a-qiime-2-plugin-to-trim-its-sequences/5780" class="uri">https://forum.qiime2.org/t/q2-itsxpress-a-tutorial-on-a-qiime-2-plugin-to-trim-its-sequences/5780</a></p>
<p>The UNITE database is used by many for fungal ITS classification.</p>
<p>UNITE UNITE v10.0 v04.04.2024 for qiime2-2024.5 (ensure this is the latest version before use): <a href="https://github.com/colinbrislawn/unite-train/releases/tag/v10.0-v04.04.2024-qiime2-2024.5" class="uri">https://github.com/colinbrislawn/unite-train/releases/tag/v10.0-v04.04.2024-qiime2-2024.5</a></p>
<p>UNITE website: <a href="https://unite.ut.ee/repository.php" class="uri">https://unite.ut.ee/repository.php</a></p>
</div>
<div id="longitudinal-analysis" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Longitudinal analysis<a href="04-QIIME2_resources.html#longitudinal-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/calender.png" style="width:150px; background:white; border-radius:5px; border:white solid 5px" />
</center>
<p>Analysis of a longitudinal 16S dataset can be carried out with the <code>q2-longitudinal</code> plugin.
It is complicated and very stats heavy.
Thankfully there is a tutorial.</p>
<p><a href="https://docs.qiime2.org/2024.10/tutorials/longitudinal/" class="uri">https://docs.qiime2.org/2024.10/tutorials/longitudinal/</a></p>
<center>
<img src="figures/q2-longitudinal.png" style="width:800px; background:white; border-radius:5px" />
</center>
</div>
<div id="iterative-rarefaction" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Iterative rarefaction<a href="04-QIIME2_resources.html#iterative-rarefaction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/iterative_rarefaction.png" style="width:200px" />
</center>
<p>The Plugin <code>q2-boots</code> works similar to the <code>diversity</code> Plugin, allowing for alpha and beta diversity analysis.
However, instead of running only a single step of rarefaction it can run multiple iterations and produce averaged diversity values.</p>
<p>It is currently in beta mode and you can see more info in the below link:</p>
<p><a href="https://forum.qiime2.org/t/q2-boots-bootstrapped-and-rarefaction-based-diversity-analysis/31134" class="uri">https://forum.qiime2.org/t/q2-boots-bootstrapped-and-rarefaction-based-diversity-analysis/31134</a></p>
</div>
<div id="managaing-reference-databases" class="section level2 hasAnchor" number="4.6">
<h2><span class="header-section-number">4.6</span> Managaing reference databases<a href="04-QIIME2_resources.html#managaing-reference-databases" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<center>
<img src="figures/rescript.png" style="width:200px; background:white; border-radius:5px; border:white solid 5px" />
</center>
<p>You can work with non 16S metabarcodes in QIIME2.
However, some of the non 16S databases were not created with QIIME2 analysis in mind.
You will therefore need to manage and curate the databases, this can be carried out with <code>RESCRIPt</code>.</p>
<p>The GitHub page: <a href="https://github.com/bokulich-lab/RESCRIPt" class="uri">https://github.com/bokulich-lab/RESCRIPt</a></p>
<p>QIIME2 rescript plugin page: <a href="https://docs.qiime2.org/2024.10/plugins/available/rescript/" class="uri">https://docs.qiime2.org/2024.10/plugins/available/rescript/</a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="03-Handy_QIIME2_commands.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="05-16S_R_packages.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
